#!/bin/sh

##20160211
##File Change History Log/Explanations/TOC'ish
~~~~

##20150701
#Created a 'vs2' of this MainScript1.sh file because reorganizing large chunks of it so that 'file downloads', 'top SNP hits' and 'multivariate analysis' are done in separate sections all together for organizational purposes
#Think this might make things a bit easier to traverse through for the time being or at least organize mentally in my head
#Also somewhat purposely separating the Global Lipids and GIANT work from the rest of the datasets since it seems/feels like those two datasets (or groups of datasets) can be the anchors of the project whereas most other datasets will be accessories/supportive; also this is how my mind is naturally separating things out at the moment anyways

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/MainScript1.sh /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/MainScript1.vs2.sh

##20150812      
#Reversed the above decision, going back to a 'per cohort/dataset' organization standpoint -- just going to process each dataset at similar stages one at a time even though in this file they will appear as separate block entries

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/MainScript1.vs2.sh /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/MainScript1.vs3.sh

##20150830      
#I reversed this initial decision and continued using it as a 'three-way split' type of setup, but just with the vs3 title. Making some more organizational changes so continuing on with a vs4 of things
cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/MainScript1.vs2.sh /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/MainScript1.vs3.sh

##20160211
#.....thinking of trying to do 'per study' organization again...

Proposed organization -- 
DATASET-START
DATASET-download
DATASET-tophits
DATASET-process
DATASET-analysis

Datasets -- 
GlobalLipids2010
GlobalLipids2013
GIANT2010
GIANT2014/5
ICBP2011
HaemgenRBC2012
Choongwon2015
Choongwon2016
Stein2016



20140611
~~~~~~~

cd /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013

git clone https://github.com/stephens999/multivariate

mkdir /data/external_public/GlobalLipids2013

cd /data/external_public/GlobalLipids2013

wget -r -l1 --no-parent -A "jointGwas*" http://www.sph.umich.edu/csg/abecasis/public/lipids2013/

wget -r -l1 --no-parent -A "Mc*" http://www.sph.umich.edu/csg/abecasis/public/lipids2013/

#NOTE -- these files were actually already downloaded by Xiang Zhu in /data/external_public/GlobalLipid
#So removed the /data/external_public/GlobalLipids2013 directory

mkdir /data/external_public/GlobalLipids2010

cd /data/external_public/GlobalLipids2010

wget -r -l1 --no-parent -A "*2010*" http://www.sph.umich.edu/csg/abecasis/public/lipids2010/

wget -r -l1 --no-parent -A "*with_Effect*" http://www.sph.umich.edu/csg/abecasis/public/lipids2010/

#Unzipped both *2010* and *with_Effect* files

mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013 

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/multivariate/globallipids/process.txt /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/process.MTedits.vs1.R

#Confirming column 2 in file /data/external_public/GlobalLipid/Mc_LDL.txt.gz has only unique entries
zcat /data/external_public/GlobalLipid/Mc_LDL.txt.gz | awk '{ print $2 }' | sort | uniq -c | sort -k 1,1 | tail -n 20

cd /mnt/lustre/home/mturchin20/Software

wget http://cran.r-project.org/src/contrib/data.table_1.9.2.tar.gz

#install.packages("/mnt/lustre/home/mturchin20/Software/data.table_1.9.2.tar.gz", repos=NULL, type="source")

scp /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/process.MTedits.vs1.R mturchin20@wolfy.uchicago.edu:/Users/mturchin20/LabStuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/. 

cp -p /Users/mturchin20/LabStuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/process.MTedits.vs1.R /Users/mturchin20/LabStuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/process.MTedits.For2013.vs1.R

#Copy and pasted commands from /Users/mturchin20/LabStuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/process.MTedits.For2013.vs1.R into R terminal, but could have run script as well

scp mturchin20@wolfy.uchicago.edu:/Users/mturchin20/LabStuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.*.txt /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/.

##NOTE_A (see below)
#Supplementary file ng.2797-S1.pdf was downloaded from http://www.nature.com/ng/journal/v45/n11/extref/ng.2797-S1.pdf and the .pdf was saved as a .txt file using Adobe Acrobat
#In excel, all tables except for supplementary tables 2 and 3 were removed manually, producing file ng.2797-S1.edited.vs2.txt
#Supplementary table 2 contains 62 novel loci and supplementary table 3 contains 157 previously identified loci

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.txt | perl -lane 'foreach my $val1 (@F) { print $val1 ; } ' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.txt

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.txt | grep ^rs > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.txt

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.txt | awk '{ print $3 }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.rsIDs.txt

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.txt | grep -v -f /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.rsIDs.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.missingSNPs.txt

#/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.txt was run through HapMart to get the chromosome and position locations for each rsID. 2 SNPs were missing and their information was manually included. These 2 SNPs were rs1047891 and rs9411489. Information was retrieved from UCSC Genome Browser hg18 and included in the /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.txt file
#hg18 because HapMart only had hg18 but switching to hg19 with the final annotated file
#rs1047891 chr2 211248752
#rs9411489 chr 
#NOTE -- Previously reported ABO SNP rs9411489 has merged into rs635634 (source: supplementary of http://hmg.oxfordjournals.org/content/21/6/1444.full or https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=2&cad=rja&uact=8&ved=0CCsQFjAB&url=http%3A%2F%2Fhmg.oxfordjournals.org%2Fcontent%2Fsuppl%2F2011%2F12%2F07%2Fddr581.DC1%2Fddr581supp_table1.doc&ei=HxKiU8_OIYOOyAS-tICoCg&usg=AFQjCNHO1Honaz_0R7CcVLsZan3LNJulKQ&sig2=TIdN-hlZUyRQZeMxiDGWHg&bvm=bv.69137298,d.aWw)
#rs635634 chr9 135144821

mkdir /mnt/lustre/home/mturchin20/Scripts/Python

#Below scp done in Miranda server from NU
scp /home/michaelt/Scripts/Python/2fileTemplate.vs1.py mturchin20@wolfy.uchicago.edu:/Users/mturchin20/clstrHme/. 

mv /mnt/lustre/home/mturchin20/2fileTemplate.vs1.py /mnt/lustre/home/mturchin20/Scripts/Python/2fileTemplate.vs1.py

cp -p /mnt/lustre/home/mturchin20/Scripts/Python/2fileTemplate.vs1.py /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/DataOrganization.PythonVLookUp.CreateFullAnnotFile.vs1.py

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/DataOrganization.PythonVLookUp.CreateFullAnnotFile.vs1.py /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/DataOrganization.PythonVLookUp.CreateProcessFile.vs1.py

#python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/DataOrganization.PythonVLookUp.CreateProcessFile.vs1.py --file1 /data/external_public/GlobalLipid/jointGwasMc_LDL.txt.gz,/data/external_public/GlobalLipid/jointGwasMc_HDL.txt.gz,/data/external_public/GlobalLipid/jointGwasMc_TG.txt.gz,/data/external_public/GlobalLipid/jointGwasMc_TC.txt.gz > 

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/DataOrganization.PythonVLookUp.CreateFullAnnotFile.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.txt --file2 /data/external_public/GlobalLipid/jointGwasMc_LDL.txt.gz,/data/external_public/GlobalLipid/jointGwasMc_HDL.txt.gz,/data/external_public/GlobalLipid/jointGwasMc_TG.txt.gz,/data/external_public/GlobalLipid/jointGwasMc_TC.txt.gz | perl -lane 'my @vals1 = split(/:/, $F[1]); my $chr = "NA"; if ($vals1[0] =~ m/chr(\d+)/) { $chr = $1; } splice(@F, 1, 1, ($chr, $vals1[1])); push(@F, "NA"); print join("\t", @F);' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/jointGwasMc_AllPheno.Annot.txt.gz 

#For troubleshooting purposes with the /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/DataOrganization.PythonVLookUp.CreateFullAnnotFile.vs1.py script
zcat /data/external_public/GlobalLipid/jointGwasMc_LDL.txt.gz | head -n 20 > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Data/jointGwasMc_LDL.txt.top20
zcat /data/external_public/GlobalLipid/jointGwasMc_HDL.txt.gz | head -n 20 > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Data/jointGwasMc_HDL.txt.top20
zcat /data/external_public/GlobalLipid/jointGwasMc_TG.txt.gz | head -n 20 > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Data/jointGwasMc_TG.txt.top20
zcat /data/external_public/GlobalLipid/jointGwasMc_TC.txt.gz | head -n 20 > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Data/jointGwasMc_TC.txt.top20
zcat /data/external_public/GlobalLipid/jointGwasMc_LDL.txt.gz | head -n 5000 > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Data/jointGwasMc_LDL.txt.top5000
zcat /data/external_public/GlobalLipid/jointGwasMc_HDL.txt.gz | head -n 5000 > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Data/jointGwasMc_HDL.txt.top5000
zcat /data/external_public/GlobalLipid/jointGwasMc_TG.txt.gz | head -n 5000 > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Data/jointGwasMc_TG.txt.top5000
zcat /data/external_public/GlobalLipid/jointGwasMc_TC.txt.gz | head -n 5000 > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Data/jointGwasMc_TC.txt.top5000

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/DataOrganization.PythonVLookUp.CreateFullAnnotFile.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Data/jointGwasMc_LDL.txt.top20,/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Data/jointGwasMc_HDL.txt.top20,/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Data/jointGwasMc_TG.txt.top20,/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Data/jointGwasMc_TC.txt.top20 > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/jointGwasMc_AllPheno.Annot.txt.top20
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/DataOrganization.PythonVLookUp.CreateFullAnnotFile.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Data/jointGwasMc_LDL.txt.top5000,/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Data/jointGwasMc_HDL.txt.top5000,/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Data/jointGwasMc_TG.txt.top5000,/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Data/jointGwasMc_TC.txt.top5000 > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/jointGwasMc_AllPheno.Annot.txt.top5000

#Added the following header to /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/jointGwasMc_AllPheno.Annot.txt.gz
#posHg18 chr pos snp a1 a2 maf beta_LDL se_LDL n_LDL beta_HDL se_HDL n_HDL beta_TG se_TG n_TG beta_TC se_TC n_TC annot gene 

cp -p /mnt/lustre/home/mturchin20/Scripts/Python/2fileTemplate.vs1.py /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/DataOrganization.PythonVLookUp.PutTogetherProcessAndAnnotFiles.vs1.py

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/DataOrganization.PythonVLookUp.PutTogetherProcessAndAnnotFiles.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/dtlesssignif.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/jointGwasMc_AllPheno.Annot.txt.gz | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/dtlesssignif.annot.txt.gz

#Added the following header to /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/dtlesssignif.annot.txt.gz
#posHg18 chr pos snp a1 a2 maf beta_LDL se_LDL n_LDL beta_HDL se_HDL n_HDL beta_TG se_TG n_TG beta_TC se_TC n_TC annot gene Z.tg Z.tc Z.ldl Z.hdl mvstat mvp unip

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/multivariate/globallipids/GLC.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GLC.MTedits.For2013.vs1.R

cp -p /mnt/lustre/home/mturchin20/Scripts/Python/2fileTemplate.vs1.py /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/DataOrganization.PythonVLookUp.UpdateAnnotFileAnnotations.vs1.py

##NOTE_B
#sebanti.novel_lipid_loci.txt and sebanti.known_lipid_loci.txt were provided by Sebanti Sengupta and Xiaoquan Wen from UMichigan after being asked by Matthew

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/sebanti.known_lipid_loci.txt /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/sebanti.novel_lipid_loci.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/sebanti.all_lipid_loci.txt
#NOTE -- replaced rs9411489 with rs635634 in /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/sebanti.all_lipid_loci.txt file since /data/external_public/GlobalLipid/jointGwasMc_* files do not contain rs9411489 but contain rs635634
#Running this code paste <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.txt | awk '{ if (NR > 1) { print $3 } }' | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/sebanti.all_lipid_loci.txt | sort) | vi - makes it look like the previous list of rsIDs -- the ones scraped from the 2013 paper's supplement -- and the list sent to us by Sebanti, are in fact the same
#As a result, going to continue using the list of rsIDs and information from the HapMart set of information since we use positional information to tag 'nearby' SNPs. Sebanti did not send this information (we did not ask for it), so I would just be redoing the HapMart information and getting the same results


#NOTE -- I was using qnorm(p-value) to get Z-scores, but this was producing one-sided Z-scores when the original test statistic was two-sided (the original H0 is Beta == 0 and H1 is Beta != 0). So need to do qnorm(p-value/2, lower.tail=FALSE) to get proper Z-scores. So redoing things to get those results
cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/process.MTedits.For2013.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/process.MTedits.For2013.vs2.R

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GLC.MTedits.For2013.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GLC.MTedits.For2013.vs2.R

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/DataOrganization.PythonVLookUp.PutTogetherProcessAndAnnotFiles.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/dtlesssignif.vs2.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/jointGwasMc_AllPheno.Annot.txt.gz | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/dtlesssignif.annot.vs2.txt.gz

#Manually searched UCSC Genome Browser, GTEx portal and dbSNP for information regarding the top 74 hits that come out of our /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GLC.MTedits.For2013.vs2.R analysis (e.g. SNPs with lbfavs > 4.6 or so, the minimum of the prior hits)

zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/dtlesssignif.annot.vs2.txt.gz | perl -lane 'print $F[1], "\t", $F[2], "\t", $F[2], "\t", uc($F[4]), "\t", uc($F[5]);' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/dtlesssignif.annot.vs2.txt.AnnovarFormat

#Manually deleted header line from /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/dtlesssignif.annot.vs2.txt.AnnovarFormat

mv /mnt/lustre/home/mturchin20/Software/annovar.latest.tar.gz 20130509_annovar.latest.tar.gz

wget http://www.openbioinformatics.org/annovar/download/Ht8qRwQSTi/annovar.latest.tar.gz

mkdir /mnt/lustre/home/mturchin20/Software/20130509annovar

mv /mnt/lustre/home/mturchin20/Software/annovar/* /mnt/lustre/home/mturchin20/Software/20130509annovar/.

tar -xvzf /mnt/lustre/home/mturchin20/Software/annovar.latest.tar.gz

perl /mnt/lustre/home/mturchin20/Software/annovar/annotate_variation.pl --downdb -webfrom annovar refGene /mnt/lustre/home/mturchin20/Software/annovar/humandb/ -build hg19

perl /mnt/lustre/home/mturchin20/Software/annovar/annotate_variation.pl --downdb genomicSuperDups /mnt/lustre/home/mturchin20/Software/annovar/humandb/ -build hg19

perl /mnt/lustre/home/mturchin20/Software/annovar/annotate_variation.pl --downdb -webfrom annovar snp129 /mnt/lustre/home/mturchin20/Software/annovar/humandb/ -build hg19

perl /mnt/lustre/home/mturchin20/Software/annovar/annotate_variation.pl --downdb -webfrom annovar avsift /mnt/lustre/home/mturchin20/Software/annovar/humandb/ -build hg19

perl /mnt/lustre/home/mturchin20/Software/annovar/annotate_variation.pl --downdb -webfrom annovar ljb_all /mnt/lustre/home/mturchin20/Software/annovar/humandb/ -build hg19

perl /mnt/lustre/home/mturchin20/Software/annovar/annotate_variation.pl --downdb -webfrom annovar ljb23_all /mnt/lustre/home/mturchin20/Software/annovar/humandb/ -build hg19

perl /mnt/lustre/home/mturchin20/Software/annovar/annotate_variation.pl --downdb -webfrom annovar esp6500_all /mnt/lustre/home/mturchin20/Software/annovar/humandb/ -build hg19

perl /mnt/lustre/home/mturchin20/Software/annovar/annotate_variation.pl --downdb -webfrom annovar 1000g2012apr /mnt/lustre/home/mturchin20/Software/annovar/humandb/ -build hg19

perl /mnt/lustre/home/mturchin20/Software/annovar/annotate_variation.pl --downdb phastConsElements46way /mnt/lustre/home/mturchin20/Software/annovar/humandb/ -build hg19

cd /mnt/lustre/home/mturchin20/Software/annovar/humandb/. ; nbthis *

cd /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1

perl /mnt/lustre/home/mturchin20/Software/annovar/summarize_annovar.pl /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/dtlesssignif.annot.vs2.txt.AnnovarFormat /mnt/lustre/home/mturchin20/Software/annovar/humandb/ --ver1000g 1000g2012apr --verdbsnp 129 --veresp 6500 --buildver hg19 --remove

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/newhits.vs2.txt | awk '{ print $1 }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/newhits.vs2.txt.rsIDs

cat dtlesssignif.annot.vs2.txt.AnnovarFormat.genome_summary.csv | grep -f /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/newhits.vs2.txt.rsIDs > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/newhits.vs2.genome_summary.csv
#Doing above line of code only got 39 of the 74 hits

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/newhits.vs2.txt | awk '{ print $3 }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/newhits.vs2.txt.pos

cat dtlesssignif.annot.vs2.txt.AnnovarFormat.genome_summary.csv | grep -f /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/newhits.vs2.txt.pos > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/newhits.vs2.genome_summary.csv

[  mturchin20@bigmem02  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/dtlesssignif.annot.vs2.txt.AnnovarFormat.genome_summary.csv | head -n 1 | perl -F, -lane 'print $F[19];'
LJB_MutationTaster_Pred

#cat newhits.vs2.genome_summary.csv | perl -F, -lane 'print join(",", @F[0..8]), ",", $F[13], ",", $F[15], ",", $F[17], ",", $F[19];' | paste <(cat newhits.vs2.txt | tail -n +2 | perl -lane 'print join(",", @F);') - | perl -lane 'print join(",", @F);' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/newhits.vs2.wAnnovar.txt
cat newhits.vs2.genome_summary.csv | perl -F, -lane 'if ($#F == 26) { $F[1] = $F[1] . ";".  $F[2]; splice(@F, 2, 1);} if ($F[2] =~ m/synon/) { my @vals1 = split(/\s+/, $F[2]); $F[2] = join("_", @vals1); } print join(",", @F[0..8]), ",", $F[13], ",", $F[15], ",", $F[17], ",", $F[19];' | paste <(cat newhits.vs2.txt | tail -n +2 | perl -lane 'print join(",", @F);') - | perl -lane 'print join(",", @F);' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/newhits.vs2.wAnnovar.txt
#Manually included the below header into /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/newhits.vs2.wAnnovar.txt
#snp,chr,pos,maf,Z.tg,Z.tc,Z.ldl,Z.hdl,lbfav,lbfall,lbfuni,Func,Gene,ExonicFunc,AAChange,Conserved,SegDup,ESP6500_ALL,1000g2012apr_ALL,dbSNP129,LJB_SIFT_Pred,LJB_PolyPhen2_Pred,LJB_LRT_Pred,LJB_MutationTaster_Pred
#Manually included Novel,Previous,Enriched categories by checking whether each gene ID was present in 2013 paper supplement; also check whether annotations made sense using UCSC Genome Browser via rsIDs

#NOTE_C
#ERLIN1 previously reported with rs1408579 by Matthew -- rs2862954 is top hit now. Differences shown below
[  mturchin20@spudling17  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1]$zcat /data/external_public/GlobalLipid/jointGwasMc_HDL.txt.gz | grep -w rs1408579
chr10:101902184 chr10:101912194 rs1408579       t       c       0.0198  0.0048  94205.00        1.787e-05       0.4617
[  mturchin20@spudling17  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1]$zcat /data/external_public/GlobalLipid/jointGwasMc_LDL.txt.gz | grep -w rs1408579
chr10:101902184 chr10:101912194 rs1408579       t       c       0.0028  0.0052  89784.00        0.4581  0.4617
[  mturchin20@spudling17  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1]$zcat /data/external_public/GlobalLipid/jointGwasMc_TC.txt.gz | grep -w rs1408579
chr10:101902184 chr10:101912194 rs1408579       t       c       0.0157  0.0051  94484.00        0.0009035       0.4617
[  mturchin20@spudling17  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1]$zcat /data/external_public/GlobalLipid/jointGwasMc_TG.txt.gz | grep -w rs1408579
chr10:101902184 chr10:101912194 rs1408579       t       c       0.0093  0.0047  90902.00        0.04111 0.4617

[  mturchin20@spudling17  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1]$zcat /data/external_public/GlobalLipid/jointGwasMc_HDL.txt.gz | grep -w rs2862954
chr10:101902054 chr10:101912064 rs2862954       c       t       0.0166  0.0034  186893.00       1.287e-06       0.4631
[  mturchin20@spudling17  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1]$zcat /data/external_public/GlobalLipid/jointGwasMc_LDL.txt.gz | grep -w rs2862954
chr10:101902054 chr10:101912064 rs2862954       c       t       0.0013  0.0037  172821.00       0.5875  0.4631
[  mturchin20@spudling17  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1]$zcat /data/external_public/GlobalLipid/jointGwasMc_TC.txt.gz | grep -w rs2862954
chr10:101902054 chr10:101912064 rs2862954       c       t       0.0124  0.0035  187083.00       0.0002526       0.4631
[  mturchin20@spudling17  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1]$zcat /data/external_public/GlobalLipid/jointGwasMc_TG.txt.gz | grep -w rs2862954
chr10:101902054 chr10:101912064 rs2862954       c       t       0.0083  0.0033  177587.10       0.01393 0.4631

#rs12739698 5' upstream of NROB2 did not show up, and there's nothing nearby that is a top hit
[  mturchin20@spudling17  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1]$zcat /data/external_public/GlobalLipid/jointGwasMc_HDL.txt.gz | grep -w rs12739698
chr1:27102620   chr1:27230033   rs12739698      g       a       0.0422  0.0088  94311.00        8.852e-07       0.92744
[  mturchin20@spudling17  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1]$zcat /data/external_public/GlobalLipid/jointGwasMc_LDL.txt.gz | grep -w rs12739698
chr1:27102620   chr1:27230033   rs12739698      a       g       0.0442  0.0095  89888.00        2.67e-05        0.07256
[  mturchin20@spudling17  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1]$zcat /data/external_public/GlobalLipid/jointGwasMc_TC.txt.gz | grep -w rs12739698
chr1:27102620   chr1:27230033   rs12739698      a       g       0.0340  0.0093  94595.00        0.001193        0.07256
[  mturchin20@spudling17  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1]$zcat /data/external_public/GlobalLipid/jointGwasMc_TG.txt.gz | grep -w rs12739698
chr1:27102620   chr1:27230033   rs12739698      a       g       0.0271  0.0083  91013.00        0.0008306       0.07256
#NOTE -- gene is spelt NR0B2, it is in fact tagged by a 'novel' hit in the 2013 paper and the above SNP is within .5Mb

#rs10490632 is within .5Mb of a hit, rs10490626 which is in INSIG2 which is downstream of DDX2

#rs762861 is within .5Mb of a hit, rs6831256 which is in LRPAP1 which is downstream of HGFAC/RGS12

#rs2862954 is not within .5Mb of a hit, but it is in ERLIN1 which is upstream of BLOC1S2 and CHUK, both of which come up as 'enriched genes' in the pathway analyses. BF is on the lower end (4.8) in the Matthew paper table 2, though other loci with lower BFs did get picked up, e.g. VEGFA with 4.5 and STAB1 with 4.7

#Both rs11229252 and rs11227638 are in the OR4* (olfactory receptor) region of chromosome 11 -- they are also not showing up via those rsID #s in the new dataset, but presumably they are being considered 'tagged' by being nearby any type of OR4*, which has already shown up in the 2013 paper in the form of OR4C46

##NOTE_E


##NOTE_D1 -- MAGENTA work
wget http://www.broadinstitute.org/mpg/magenta/MAGENTA_software_package_vs2_hg18_hg19_July2011_Feb16_12.tar.gz

tar -xvzf /mnt/lustre/home/mturchin20/Software/MAGENTA_software_package_vs2_July2011/MAGENTA_software_package_vs2_hg18_hg19_July2011_Feb16_12.tar.gz

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.txt | awk '{ print $3 }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.rsIDs

cp -p /mnt/lustre/home/mturchin20/Scripts/Python/2fileTemplate.vs1.py /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/DataOrganization.PythonVLookUp.Basic.vs1.py

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/DataOrganization.PythonVLookUp.Basic.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.txt --file2 /data/external_public/GlobalLipid/jointGwasMc_HDL.txt.gz > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Data/jointGwasMc_HDL.PrevHits.vs1.txt 
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/DataOrganization.PythonVLookUp.Basic.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.txt --file2 /data/external_public/GlobalLipid/jointGwasMc_LDL.txt.gz > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Data/jointGwasMc_LDL.PrevHits.vs1.txt 
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/DataOrganization.PythonVLookUp.Basic.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.txt --file2 /data/external_public/GlobalLipid/jointGwasMc_TG.txt.gz > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Data/jointGwasMc_TC.PrevHits.vs1.txt 
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/DataOrganization.PythonVLookUp.Basic.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.txt --file2 /data/external_public/GlobalLipid/jointGwasMc_TC.txt.gz > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Data/jointGwasMc_TG.PrevHits.vs1.txt 

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Data/jointGwasMc_HDL.PrevHits.vs1.txt | perl -lane 'my @vals1 = split(/:/, $F[1]); my $chr = "NA"; if ($vals1[0] =~ m/chr(\d+)/) { $chr = $1 }; print $chr, "\t", $vals1[1], "\t", $F[8];' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Data/jointGwasMc_HDL_PrevHits_vs1_MAGENTA_txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Data/jointGwasMc_LDL.PrevHits.vs1.txt | perl -lane 'my @vals1 = split(/:/, $F[1]); my $chr = "NA"; if ($vals1[0] =~ m/chr(\d+)/) { $chr = $1 }; print $chr, "\t", $vals1[1], "\t", $F[8];' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Data/jointGwasMc_LDL_PrevHits_vs1_MAGENTA_txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Data/jointGwasMc_TC.PrevHits.vs1.txt | perl -lane 'my @vals1 = split(/:/, $F[1]); my $chr = "NA"; if ($vals1[0] =~ m/chr(\d+)/) { $chr = $1 }; print $chr, "\t", $vals1[1], "\t", $F[8];' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Data/jointGwasMc_TC_PrevHits_vs1_MAGENTA_txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Data/jointGwasMc_TG.PrevHits.vs1.txt | perl -lane 'my @vals1 = split(/:/, $F[1]); my $chr = "NA"; if ($vals1[0] =~ m/chr(\d+)/) { $chr = $1 }; print $chr, "\t", $vals1[1], "\t", $F[8];' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Data/jointGwasMc_TG_PrevHits_vs1_MAGENTA_txt

zcat /data/external_public/GlobalLipid/jointGwasMc_HDL.txt.gz | perl -lane 'my @vals1 = split(/:/, $F[1]); my $chr = "NA"; if ($vals1[0] =~ m/chr(\d+)/) { $chr = $1 }; print $chr, "\t", $vals1[1], "\t", $F[8];' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Data/jointGwasMc_HDL.MAGENTA.txt 

ssh mturchin20@hegel.uchicago.edu

mv /mnt/lustre/home/mturchin20/Software/MAGENTA_software_package_vs2_July2011/ /mnt/lustre/home/mturchin20/Software/MAGENTA_software_package_vs2_July2011_Feb16_12/

wget http://www.broadinstitute.org/mpg/magenta/MAGENTA_software_package_vs2_hg18_hg19_July2011.tar.gz

#In /mnt/lustre/home/mturchin20/Software/MAGENTA_software_package_vs2_July2011_Feb16_12/
ln -s /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Data/jointGwasMc_HDL.PrevHits.vs1.MAGENTA.txt jointGwasMc_HDL_PrevHits_vs1_MAGENTA_txt

#On mturchin20@hegel.uchicago.edu

mkdir Lipids

wget http://www.broadinstitute.org/mpg/magenta/MAGENTA_software_package_vs2_hg18_hg19_July2011_Feb16_12.tar.gz 

scp /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Data/jointGwasMc_HDL.PrevHits.vs1.MAGENTA.txt mturchin20@hegel.uchicago.edu:/home/mturchin20/Lipids/MAGENTA_software_package_vs2_July2011/.

mv jointGwasMc_HDL.PrevHits.vs1.MAGENTA.txt jointGwasMc_HDL_PrevHits_vs1_MAGENTA_txt

scp /mnt/lustre/home/mturchin20/Software/MAGENTA_software_package_vs2_July2011_Feb16_12/Run_MAGENTA_vs2_July_2011.m mturchin20@hegel.uchicago.edu:/home/mturchin20/Lipids/MAGENTA_software_package_vs2_July2011/.

scp /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Data/jointGwasMc_HDL.MAGENTA.txt mturchin20@hegel.uchicago.edu:/home/mturchin20/Lipids/MAGENTA_software_package_vs2_July2011/.

mv /home/mturchin20/Lipids/MAGENTA_software_package_vs2_July2011/jointGwasMc_HDL.MAGENTA.txt /home/mturchin20/Lipids/MAGENTA_software_package_vs2_July2011/jointGwasMc_HDL_MAGENTA_txt

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/newhits.vs2.wAnnovar.txt | perl -F, -ane 'print $F[0], "\t", $F[1], "\t", $F[2], "\t", $F[12], "\n";' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/newhits.vs2.wAnnovar.ChrBP_And_Genes.txt

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/newhits.vs2.wAnnovar.ChrBP_And_Genes.txt /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/newhits.vs2.wAnnovar.ChrBP_And_Genes.wEdits.txt

#Manually edited /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/newhits.vs2.wAnnovar.ChrBP_And_Genes.wEdits.txt after looking at each SNP/gene in UCSC and doing a Google/literature search 

scp /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/newhits.vs2.txt.rsIDs mturchin20@wolfy.uchicago.edu:/Users/mturchin20/LabStuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/.

cat /home/mturchin20/Lipids/Output_MAGENTA_GlobalLipids2013_HDL_AllSNPs_10000perm_Jul03_14/MAGENTA_pval_GeneSetEnrichAnalysis_GlobalLipids2013_HDL_AllSNPs_110kb_upstr_40kb_downstr_10000perm_Jul03_14.results | perl -lane 'if ($#F > 17) { my $diff = $#F - 17; my $end = 1 + $diff; $F[1] = join("_", @F[1..$end]); splice(@F, 2, $diff); } print join("\t", @F);' | perl -lane 'print $#F;' | sort | uniq -c


cp -p /home/mturchin20/Lipids/MAGENTA_software_package_vs2_July2011/Run_MAGENTA_vs2_July_2011_prevHits_HDL.m /home/mturchin20/Lipids/MAGENTA_software_package_vs2_July2011/Run_MAGENTA_vs2_July_2011_prevHits_LDL.m
cp -p /home/mturchin20/Lipids/MAGENTA_software_package_vs2_July2011/Run_MAGENTA_vs2_July_2011_prevHits_HDL.m /home/mturchin20/Lipids/MAGENTA_software_package_vs2_July2011/Run_MAGENTA_vs2_July_2011_prevHits_TC.m
cp -p /home/mturchin20/Lipids/MAGENTA_software_package_vs2_July2011/Run_MAGENTA_vs2_July_2011_prevHits_HDL.m /home/mturchin20/Lipids/MAGENTA_software_package_vs2_July2011/Run_MAGENTA_vs2_July_2011_prevHits_TG.m
cp -p /home/mturchin20/Lipids/MAGENTA_software_package_vs2_July2011/Run_MAGENTA_vs2_July_2011_prevHits_HDL.m /home/mturchin20/Lipids/MAGENTA_software_package_vs2_July2011/Run_MAGENTA_vs2_July_2011_currHits_HDL.m

#Editted above files to include proper inputs files, e.g. jointGwasMc_*.PrevHits.vs1.txt or jointGwasMc_*.CurrHits.vs1.txt

scp /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Data/jointGwasMc_*_PrevHits_vs1_MAGENTA_txt mturchin20@hegel.uchicago.edu:/home/mturchin20/Lipids/MAGENTA_software_package_vs2_July2011/.

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/DataOrganization.PythonVLookUp.Basic.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/newhits.vs2.txt.rsIDs --file2 /data/external_public/GlobalLipid/jointGwasMc_HDL.txt.gz > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Data/jointGwasMc_HDL.newHits.vs1.txt 
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/DataOrganization.PythonVLookUp.Basic.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/newhits.vs2.txt.rsIDs --file2 /data/external_public/GlobalLipid/jointGwasMc_LDL.txt.gz > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Data/jointGwasMc_LDL.newHits.vs1.txt 
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/DataOrganization.PythonVLookUp.Basic.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/newhits.vs2.txt.rsIDs --file2 /data/external_public/GlobalLipid/jointGwasMc_TC.txt.gz > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Data/jointGwasMc_TC.newHits.vs1.txt 
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/DataOrganization.PythonVLookUp.Basic.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/newhits.vs2.txt.rsIDs --file2 /data/external_public/GlobalLipid/jointGwasMc_TG.txt.gz > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Data/jointGwasMc_TG.newHits.vs1.txt 

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Data/jointGwasMc_HDL.PrevHits.vs1.txt /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Data/jointGwasMc_HDL.newHits.vs1.txt | perl -lane 'my @vals1 = split(/:/, $F[1]); my $chr = "NA"; if ($vals1[0] =~ m/chr(\d+)/) { $chr = $1 }; print $chr, "\t", $vals1[1], "\t", $F[8];' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Data/jointGwasMc_HDL_CurrHits_vs1_MAGENTA_txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Data/jointGwasMc_LDL.PrevHits.vs1.txt /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Data/jointGwasMc_LDL.newHits.vs1.txt | perl -lane 'my @vals1 = split(/:/, $F[1]); my $chr = "NA"; if ($vals1[0] =~ m/chr(\d+)/) { $chr = $1 }; print $chr, "\t", $vals1[1], "\t", $F[8];' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Data/jointGwasMc_LDL_CurrHits_vs1_MAGENTA_txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Data/jointGwasMc_TC.PrevHits.vs1.txt /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Data/jointGwasMc_TC.newHits.vs1.txt | perl -lane 'my @vals1 = split(/:/, $F[1]); my $chr = "NA"; if ($vals1[0] =~ m/chr(\d+)/) { $chr = $1 }; print $chr, "\t", $vals1[1], "\t", $F[8];' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Data/jointGwasMc_TC_CurrHits_vs1_MAGENTA_txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Data/jointGwasMc_TG.PrevHits.vs1.txt /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Data/jointGwasMc_TG.newHits.vs1.txt | perl -lane 'my @vals1 = split(/:/, $F[1]); my $chr = "NA"; if ($vals1[0] =~ m/chr(\d+)/) { $chr = $1 }; print $chr, "\t", $vals1[1], "\t", $F[8];' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Data/jointGwasMc_TG_CurrHits_vs1_MAGENTA_txt

scp /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Data/jointGwasMc_*_CurrHits_vs1_MAGENTA_txt mturchin20@hegel.uchicago.edu:/home/mturchin20/Lipids/MAGENTA_software_package_vs2_July2011/.

cp -p /home/mturchin20/Lipids/MAGENTA_software_package_vs2_July2011/Run_MAGENTA_vs2_July_2011_currHits_HDL.m /home/mturchin20/Lipids/MAGENTA_software_package_vs2_July2011/Run_MAGENTA_vs2_July_2011_currHits_LDL.m
cp -p /home/mturchin20/Lipids/MAGENTA_software_package_vs2_July2011/Run_MAGENTA_vs2_July_2011_currHits_HDL.m /home/mturchin20/Lipids/MAGENTA_software_package_vs2_July2011/Run_MAGENTA_vs2_July_2011_currHits_TC.m
cp -p /home/mturchin20/Lipids/MAGENTA_software_package_vs2_July2011/Run_MAGENTA_vs2_July_2011_currHits_HDL.m /home/mturchin20/Lipids/MAGENTA_software_package_vs2_July2011/Run_MAGENTA_vs2_July_2011_currHits_TG.m

##NOTED_2

#Back in directory /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1

zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/jointGwasMc_AllPheno.Annot.vs2.txt.gz | perl -lane 'print $F[1], "\t", $F[2], "\t", $F[2], "\t", uc($F[4]), "\t", uc($F[5]);' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/jointGwasMc_AllPheno.Annot.vs2.txt.AnnovarFormat

perl /mnt/lustre/home/mturchin20/Software/annovar/summarize_annovar.pl /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/jointGwasMc_AllPheno.Annot.vs2.txt.AnnovarFormat /mnt/lustre/home/mturchin20/Software/annovar/humandb/ --ver1000g 1000g2012apr --verdbsnp 129 --veresp 6500 --buildver hg19 --remove

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/DataOrganization.PythonVLookUp.Basic.vs1.py /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/DataOrganization.PythonVLookUp.PutHG19IntoHapMartResults.vs1.py

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/DataOrganization.PythonVLookUp.PutHG19IntoHapMartResults.vs1.py --file1 /data/external_public/GlobalLipid/jointGwasMc_HDL.txt.gz --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.hg19.txt

#Manually included the below header in /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.hg19.txt
#chromosome      position        marker id       alleles population id   reference allele

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/DataOrganization.PythonVLookUp.Basic.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.hg19.txt --file2 jointGwasMc_AllPheno.Annot.vs2.txt.AnnovarFormat.genome_summary.csv > jointGwasMc_AllPheno.Annot.vs2.txt.AnnovarFormat.genome_summary.prevHits.csv

cp -p /mnt/lustre/home/mturchin20/Software/MAGENTA_software_package_vs2_July2011/Extract_genes_around_SNPs.m /mnt/lustre/home/mturchin20/Software/MAGENTA_software_package_vs2_July2011/Extract_genes_around_SNPs.dtlesssignif.m

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/newhits.vs2.wAnnovar.txt | perl -F, -lane 'print $F[0], "\t", $F[1], "\t", $F[2];' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/newhits.vs2.wAnnovar.Extract_genes_around_SNPs.Format.txt

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.hg19.txt | perl -lane 'my $chr = "NA"; if ($F[0] =~ m/chr(\d+)/) { $chr = $1; } print $F[2], "\t", $chr, "\t", $F[1];' | grep -v marker > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.hg19.Extract_genes_around_SNPs.Format.txt

zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/dtlesssignif.annot.txt.gz | perl -lane 'print $F[3], "\t", $F[1], "\t", $F[2]; ' | grep -v chr > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/dtlesssignif.annot.Extract_genes_around_SNPs.Format.txt

#NOTE -- 50 SNPs do not have rsIDs, so they might fail 'Extract_genes_around_SNPs.m'. These SNPs are visible via the code below
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/dtlesssignif.annot.txt.gz | perl -lane 'if ($F[3] !~ /rs/) { print join("\t", @F); }' | wc

scp /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/newhits.vs2.wAnnovar.Extract_genes_around_SNPs.Format.txt /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.hg19.Extract_genes_around_SNPs.Format.txt /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/dtlesssignif.annot.Extract_genes_around_SNPs.Format.txt mturchin20@hegel.uchicago.edu:/home/mturchin20/Lipids/MAGENTA_software_package_vs2_July2011/.

#Ran /mnt/lustre/home/mturchin20/Software/MAGENTA_software_package_vs2_July2011/Extract_genes_around_SNPs.m in matlab on mturchin20@hegel.uchicago.edu
# Extract_genes_around_SNPs('ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.hg19.Extract_genes_around_SNPs.Format.txt',100000,'GlobalLipids2013_prevHits')
#Output file names are: Genes_near_SNP_100kb_boundary_GlobalLipids2013_prevHits_Jul10_14 and Distance_genes_near_SNP_100kb_boundary_GlobalLipids2013_prevHits_Jul10_14
#There are 27 SNPs with no genes within 100000 base pairs around the SNP.
#There are a total of 458 genes within 100000 base pairs around all SNPs.
#
#Extract_genes_around_SNPs('newhits.vs2.wAnnovar.Extract_genes_around_SNPs.Format.txt',100000,'GlobalLipids2013_currHits')
#Output file names are: Genes_near_SNP_100kb_boundary_GlobalLipids2013_currHits_Jul10_14 and Distance_genes_near_SNP_100kb_boundary_GlobalLipids2013_currHits_Jul10_14
#There are 12 SNPs with no genes within 100000 base pairs around the SNP.
#There are a total of 215 genes within 100000 base pairs around all SNPs.
#
#Output file names are: Genes_near_SNP_100kb_boundary_GlobalLipids2013_dtlesssignif_Jul10_14 and Distance_genes_near_SNP_100kb_boundary_GlobalLipids2013_dtlesssignif_Jul10_14
#There are 2024 SNPs with no genes within 100000 base pairs around the SNP.
#There are a total of 36341 genes within 100000 base pairs around all SNPs.
#
#Output is in mturchin20@hegel.uchicago.edu:/home/mturchin20/Lipids/MAGENTA_software_package_vs2_July2011

#Issue with the following line in supplement of 2013 paper
#Locus Lead SNP Chr hg19 Position (Mb) Traits GWS Nearest Gene Nearest Gene (kb)   No. of Genes within 100kb
#ASAP3 rs1077514 1 23.77 TC ASAP3 0 6
#'no of genes within 100kb' is 6 but get 3 when run the following command Extract_genes_around_SNPs('ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.hg19.Extract_genes_around_SNPs.Format.txt',100000,'GlobalLipids2013_prevHits') and don't see 6 when look in UCSC genome browser using Hg19 database (see like 3 or so within a window of 100kb with SNP at center)
#However, interestingly, don't see ASAP3 anywhere in MAGENTA results tables, and ASAP3 doesn't exist in the GeneID data they provide to automatically use. But a bunch of hand-picked Gene IDs presented in the MAGENTA supplementary tables do exist in the GeneID list MAGENTA provides -- maybe they used the lists given and then hand-curated a number of the SNPs later?

scp mturchin20@hegel.uchicago.edu:/home/mturchin20/Lipids/MAGENTA_software_package_vs2_July2011/Distance_genes_near_SNP* mturchin20@wolfy.uchicago.edu:/Users/mturchin20/clstrHme/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/. 

cat /home/mturchin20/Lipids/MAGENTA_software_package_vs2_July2011/Distance_genes_near_SNP_100kb_boundary_GlobalLipids2013_prevHits_Jul10_14 | awk '{ print $5 }' | sort | uniq | perl -lane 'if ($F[0] !~ m/^$/) { print join("\t", @F); }' | grep -v \# > /home/mturchin20/Lipids/MAGENTA_software_package_vs2_July2011/Distance_genes_near_SNP_100kb_boundary_GL2013_prevHits_GeneIDs

cat /home/mturchin20/Lipids/MAGENTA_software_package_vs2_July2011/Distance_genes_near_SNP_100kb_boundary_GlobalLipids2013_currHits_Jul10_14 | awk '{ print $5 }' | sort | uniq | perl -lane 'if ($F[0] !~ m/^$/) { print join("\t", @F); }' | grep -v \# > /home/mturchin20/Lipids/MAGENTA_software_package_vs2_July2011/Distance_genes_near_SNP_100kb_boundary_GL2013_currHits_GeneIDs

cat /home/mturchin20/Lipids/MAGENTA_software_package_vs2_July2011/Distance_genes_near_SNP_100kb_boundary_GlobalLipids2013_dtlesssignif_Jul10_14 | awk '{ print $5 }' | sort | uniq | perl -lane 'if ($F[0] !~ m/^$/) { print join("\t", @F); }' | grep -v \# > /home/mturchin20/Lipids/MAGENTA_software_package_vs2_July2011/Distance_genes_near_SNP_100kb_boundary_GL2013_dtlesssignif_GeneIDs

cat /home/mturchin20/Lipids/MAGENTA_software_package_vs2_July2011/Distance_genes_near_SNP_100kb_boundary_GlobalLipids2013_prevHits_Jul10_14 | awk '{ print $4 }' | sort | uniq | perl -lane 'if ($F[0] !~ m/^$/) { print join("\t", @F); }' > /home/mturchin20/Lipids/MAGENTA_software_package_vs2_July2011/Distance_genes_near_SNP_100kb_boundary_GL2013_prevHits_GeneSymbols

cat /home/mturchin20/Lipids/MAGENTA_software_package_vs2_July2011/Distance_genes_near_SNP_100kb_boundary_GlobalLipids2013_currHits_Jul10_14 | awk '{ print $4 }' | sort | uniq | perl -lane 'if ($F[0] !~ m/^$/) { print join("\t", @F); }' > /home/mturchin20/Lipids/MAGENTA_software_package_vs2_July2011/Distance_genes_near_SNP_100kb_boundary_GL2013_currHits_GeneSymbols

cat /home/mturchin20/Lipids/MAGENTA_software_package_vs2_July2011/Distance_genes_near_SNP_100kb_boundary_GlobalLipids2013_dtlesssignif_Jul10_14 | awk '{ print $4 }' | sort | uniq | perl -lane 'if ($F[0] !~ m/^$/) { print join("\t", @F); }' > /home/mturchin20/Lipids/MAGENTA_software_package_vs2_July2011/Distance_genes_near_SNP_100kb_boundary_GL2013_dtlesssignif_GeneSymbols









##20150515
##Going back into this project and need to re-understand what I had done
##I will be explaining things below and continuing my log of things from this point on. I will not copy/paste relevant lines of code from above but I will give context and annotations to what was important/what will be used/moved going forward
##The goals, as of writing this header, are the following:
##	Identify where 2013 paper recapitulated results of Matthew's earlier paper
##	Identify new top hits compared to 2013 paper
##	Do a sort of SUMSTAT based on BF? to get a sense of 'enrichment' for relevant biological pathways
##	Do same setup in GIANT data with older and newer paper to see if anything gained?

##	Note -- Matthew was particularly interested in variants there were close to the threshold but then went over it in our analysis versus in the published paper's analysis ??

##NOTE_A: I manually took all the supplementary tables from the 2013 paper and tried to identify ever rsID# present in order to roughly determine what variants we had 'discovered' were also present in the 2013 paper in any regard. This approach is clearly making a number of assumptions but as a first approximation should probably help guide/inform things

##NOTE_B: I received a list of SNPs, those which were already known and those which were new in 2013 paper (known_lipid_loci.txt and novel_lipid_loci.txt), from UMich people and identified that my 'scraping' of the .pdf pulled the same list. So should have the right list of SNPs now from 2013 paper.

##NOTE_C: I think this is me looking for Matthew's previous SNPs in the newer data to try and figure out, if the rsID was not specifically present in the 2013 results, if Matthew's result was being tagged by something else

##NOTE_D1: The following section is mostly MAGENTA work on 'Hagel', the computer John helped me get one, and the portion at/after NOTE_D2 is on the PPS cluster. Note, while I do have 'results' from my earlier, initial runs of MAGENTA, I was not using full genome-wide data, just top hits, and MAGENTA assumes you are providing it with full genome-wide data. If I were to do this I would need to run Matthew's code on every SNP and then convert the joint BF back to a p-value somehow that's on the same scale as the univariate p-values originally were

##NOTE_E: It would appear that these are the current state of things -- 
#	2010	95_hits	(59_novel)	Matthew_18
#	2013	157_hits (62_novel)	Matthew_75
#	CHECK_0: Look at whether the 18 loci from before are specifically in the 'novel' loci list I received from Sebanti, or can use Annovar to tag them, tag Matthew's 18 and see if the same loci are being tagged at least since rsIDs might be different between those two lists
#
#	/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GLC.MTedits.For2013.vs2.R contains the code and results that were used to come up with the 'newhits.vs2.txt' file shown below

#	CHECK_0: Include a line or some code about where the 'previous hits' were included, how the 'RSS0.vs2.txt' file was created which I presume is where the 'previous hits' come into play
#	Line from GLC.MTedits.For2013.vs2.R showing the 'lbfav' cutoff from the least significant 'true' hit
#	newhits = sub[sub$annot==0 & sub$lbfav>4.608818 & sub$nmin>50000,c(4,2:3,7,22:25,30)]
#	
#	Matthew was interested in SNPs whose BFs 'should' have been picked up, so look at SNPs with BFs > 8....guess this is close to 'genome-wide significance' or something as far as a cutoff to use was?
#	Also looked at how many of the new hits have univariate BFs greater than joint BFs and you get the following

~~~
lbf.all.newhits[lbf.uni.newhits < lbf.all.newhits]
[1]  5.113600  9.238360  6.887086  4.339257  4.501904  4.378643  5.930985
[8]  4.371434  5.184725  4.125246  5.819409 17.316921  4.514223  4.770906
[15]  5.259777  5.139357  6.664481  5.397071  4.275684  4.560132  4.154638
[22]  4.857058  5.761283  5.226976  6.581458  4.150895  5.575327  4.252227
[29]  4.292621  4.981670  6.722394 25.754510  5.658189 11.867786  4.854878
[36]  7.653760  6.827153  5.810097  5.007101  5.123809  5.980351  4.851327
[43] 31.807344  4.781310  5.977829  4.754488  6.047135  5.258339  9.998777
[50]  6.890773  5.988893 11.425993  4.190938  6.475299  4.686178  4.280777
[57]  6.004999  5.289372  4.777772 11.061500 36.511011  8.021977  5.313751
[64]  4.773029  5.407697  5.001026  5.908662

lbf.all.newhits[lbf.uni.newhits > lbf.all.newhits]
[1]  4.767202  8.055220 20.992954  5.048607  5.396590  6.225892  4.796701
~~~

~~~
[  mturchin20@spudhead  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/newhits.vs2.txt | wc
     75     825   11000
~~~

#Things moving forward:
#	CHECK_0: Figure out state of those 18 SNPs, which ones are completely reproduced and which ones are not?...get details as if for paper?
#	CHECK_0: Look at the bf > 8 hits and compare univariate vs multivariate BFs -- if no univariate BF is close to multivariate BF makes sense (e.g. multivariate really increased power), but if a single univariate is close to multivariate then this leads to wonder why that SNP wasn't included in the first place. Look at genome browser to see if near any other genes or hits, even though it should have been pruned based on distance already? May also want to create a list of these variants to send to UMich people to see if they have any understanding/reasons to believe why those SNPs didn't make the cut
#	CHECK_0: Manually go through each 75 top hit to get information/details as if for publication, such as eQTL, GenomeBrowser stuff, other  functional information, ANNOVAR
#	CHECK_0: How many of the new hits are functional (e.g. exonic/intronic vs. intergenic?)
#	CHECK_0: Do SUMSTAT on original 157, new 75, then both together? Lookign at both 'do new hits add information to original 157' as well as 'do new 57 represent any novel biology?' This latter question may be better addresssed to while going through each gene manually as well?
#	CHECK_0: Do GIANT results 2010 and 2014 for height, BMI, wasitsizeratio....maybe see from Joel whether there are any other phenotypes that can be used as well?
#	CHECK_0: Reread paper

#	CHECK_0: More down the road -- look for other datasets that may even have just a single release timepoint that have multiple (presumably correlated) phenotypes? Ask Xiang about this potentially to?
#	CHECK_0: Read further into literature, compare Tim's paper ('sharing genetic effects across tissues, assuming independence') to Matthew's paper ('in fact assuming correlation and utilizing that')

#20150521
cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GLC.MTedits.For2013.vs2.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GLC.MTedits.For2013.vs3.R


##20150518
##GIANT work

#Note -- renamed the directory /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/GlobalLipids2013/... to /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/... so conducted the below command on this entire file
#:.,455 s/StephensLab\/GlobalLipids2013/StephensLab\/Multivariate\/GlobalLipids2013/g

mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate

mv /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/GlobalLipids2013/ /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/.

mv /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/MainScript1.sh /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/.

#CHECK_0: Figure out github stuff along with these changes

mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010
mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5
mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013

mkdir /mnt/gluster/data/external_public_supp/GIANT2010
mkdir /mnt/gluster/data/external_public_supp/GIANT2014_5
mkdir /mnt/gluster/data/external_public_supp/GIANT2013

#CHECK_0: Do sex-specific analysis on 2013 or 2014_5 data since it's there/present? 2014_5 doesn't have height stratified....could probably get from Joel if asked? 2013 has all traits stratified by sex

cd /mnt/gluster/data/external_public_supp/GIANT2013

#Sex-specific files from 2013
wget http://www.broadinstitute.org/collaboration/giant/images/3/30/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_BMI_MEN_N.txt 
wget http://www.broadinstitute.org/collaboration/giant/images/8/8d/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_BMI_WOMEN_N.txt
wget http://www.broadinstitute.org/collaboration/giant/images/5/5c/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HEIGHT_MEN_N.txt
wget http://www.broadinstitute.org/collaboration/giant/images/f/fe/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HEIGHT_WOMEN_N.txt
wget http://www.broadinstitute.org/collaboration/giant/images/2/2f/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HIPadjBMI_MEN_N.txt
wget http://www.broadinstitute.org/collaboration/giant/images/e/e5/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HIPadjBMI_WOMEN_N.txt
wget http://www.broadinstitute.org/collaboration/giant/images/0/08/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WCadjBMI_MEN_N.txt
wget http://www.broadinstitute.org/collaboration/giant/images/2/24/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WCadjBMI_WOMEN_N.txt
wget http://www.broadinstitute.org/collaboration/giant/images/f/f7/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WHRadjBMI_MEN_N.txt
wget http://www.broadinstitute.org/collaboration/giant/images/4/40/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WHRadjBMI_WOMEN_N.txt
wget http://www.broadinstitute.org/collaboration/giant/images/a/a9/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HIP_MEN_N.txt
wget http://www.broadinstitute.org/collaboration/giant/images/9/99/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HIP_WOMEN_N.txt
wget http://www.broadinstitute.org/collaboration/giant/images/3/33/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WC_MEN_N.txt
wget http://www.broadinstitute.org/collaboration/giant/images/e/e7/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WC_WOMEN_N.txt
wget http://www.broadinstitute.org/collaboration/giant/images/f/fe/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WHR_MEN_N.txt
wget http://www.broadinstitute.org/collaboration/giant/images/f/f0/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WHR_WOMEN_N.txt
wget http://www.broadinstitute.org/collaboration/giant/images/f/f8/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WEIGHT_MEN_N.txt
wget http://www.broadinstitute.org/collaboration/giant/images/c/cc/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WEIGHT_WOMEN_N.txt

cd /mnt/gluster/data/external_public_supp/GIANT2010

wget http://www.broadinstitute.org/collaboration/giant/images/b/b7/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.txt.gz
wget http://www.broadinstitute.org/collaboration/giant/images/4/49/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.txt.gz
wget http://www.broadinstitute.org/collaboration/giant/images/8/87/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.txt.gz


#From -- http://www.broadinstitute.org/collaboration/giant/index.php/GIANT_consortium_data_files
~~~
BMI (download GZIP)

MD5 (GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.txt -- 79 MB; 2,471,517 lines) = 38c836542807a3830101bcf48bb34472
If you use these Body Mass Index data, please cite: Speliotes, E.K., Willer, C.J., Berndt, S.I., Monda, K.L., Thorleifsson, G., Jackson, A.U., Allen, H.L., Lindgren, C.M., Luan, J., Magi, R., et al. (2010). Association analyses of 249,796 individuals reveal 18 new loci associated with body mass index. Nat Genet 42, 937-948.
Height (download GZIP)

MD5 (GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.txt -- 82 MB; 2,469,636 lines) = b51b4c4ff1f03bd33c4b2dfd6b10cb82
If you use these height data, please cite: Lango Allen, H., Estrada, K., Lettre, G., Berndt, S.I., Weedon, M.N., Rivadeneira, F., Willer, C.J., Jackson, A.U., Vedantam, S., Raychaudhuri, S., et al. (2010). Hundreds of variants clustered in genomic loci and biological pathways affect human height. Nature 467, 832-838.
WHRadjBMI (download GZIP)

MD5 (GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.txt -- 75 MB; 2,483,326 lines) = 8f7e2ca61c33a120db9e7bfe51e3c053
If you use these waist-hip ratio adjusted for BMI data, please cite: Heid, I.M., Jackson, A.U., Randall, J.C., Winkler, T.W., Qi, L., Steinthorsdottir, V., Thorleifsson, G., Zillikens, M.C., Speliotes, E.K., Magi, R., et al. (2010). Meta-analysis identifies 13 new loci associated with waist-hip ratio and reveals sexual dimorphism in the genetic basis of fat distribution. Nat Genet 42, 949-960.
~~~

~~~
[  mturchin20@spudhead  /mnt/gluster/data/external_public_supp/GIANT2010]$zcat GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.txt.gz | wc
2471517 14829102 82685870
[  mturchin20@spudhead  /mnt/gluster/data/external_public_supp/GIANT2010]$zcat GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.txt.gz | wc
2469636 14817816 85061092
[  mturchin20@spudhead  /mnt/gluster/data/external_public_supp/GIANT2010]$zcat GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.txt.gz | wc
2483326 14899956 78051825
[  mturchin20@spudhead  /mnt/gluster/data/external_public_supp/GIANT2010]$zcat GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.txt.gz | head -n 10
MarkerName Allele1 Allele2 Freq.Allele1.HapMapCEU p N
rs10 a c 0.0333 0.708 80566
rs1000000 g a 0.6333 0.506 123865
rs10000010 c t 0.425 0.736 123827
rs10000012 c g 0.8083 0.042 123809
rs10000013 c a 0.1667 0.0689 123863
rs10000017 t c 0.2333 0.457 123262
rs1000002 c t 0.475 0.0322 123783
rs10000023 t g 0.5917 0.939 123756
rs10000029 t c 0.975 0.24 103623
[  mturchin20@spudhead  /mnt/gluster/data/external_public_supp/GIANT2010]$zcat GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.txt.gz | head -n 10
MarkerName Allele1 Allele2 Freq.Allele1.HapMapCEU p N
rs10 a c 0.0333 0.8826 78380
rs1000000 a g 0.3667 0.1858 133822
rs10000010 t c 0.575 0.8947 132858
rs10000012 c g 0.8083 0.1312 133785
rs10000013 a c 0.8333 0.628 133843
rs10000017 t c 0.2333 0.3073 133174
rs1000002 t c 0.525 0.221 133711
rs10000023 t g 0.5917 0.354 131895
rs10000029 c t 0.025 0.9831 116885
[  mturchin20@spudhead  /mnt/gluster/data/external_public_supp/GIANT2010]$zcat GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.txt.gz | head -n 10
MarkerName Allele1 Allele2 Freq.Allele1.HapMapCEU p N
rs10 c a 0.9667 0.42 57031
rs1000000 g a 0.6333 0.55 77168
rs10000010 t c 0.575 0.0029 77152
rs10000012 g c 0.1917 0.99 77117
rs10000013 a c 0.8333 0.89 77167
rs10000017 c t 0.7667 0.64 77166
rs1000002 c t 0.475 0.73 77095
rs10000023 g t 0.4083 0.041 77140
rs10000029 t c 0.975 0.84 61337
[  mturchin20@spudhead  /mnt/gluster/data/external_public_supp/GIANT2010]$md5sum GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.txt.gz
57baef8af0bceb3c09c6084954770afb  GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.txt.gz
[  mturchin20@spudhead  /mnt/gluster/data/external_public_supp/GIANT2010]$md5sum GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.txt.gz
484dace64c588d524438eae0d02ea2c4  GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.txt.gz
[  mturchin20@spudhead  /mnt/gluster/data/external_public_supp/GIANT2010]$md5sum GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.txt.gz
282989e5678f2e4785c2707349ccff38  GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.txt.gz
~~~


cd /mnt/gluster/data/external_public_supp/GIANT2014_5

wget http://www.broadinstitute.org/collaboration/giant/images/0/01/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.txt.gz
wget http://www.broadinstitute.org/collaboration/giant/images/f/f0/All_ancestries_SNP_gwas_mc_merge_nogc.tbl.uniq.gz
wget http://www.broadinstitute.org/collaboration/giant/images/1/15/SNP_gwas_mc_merge_nogc.tbl.uniq.gz
wget http://www.broadinstitute.org/collaboration/giant/images/e/eb/GIANT_2015_WHRadjBMI_COMBINED_EUR.txt.gz
wget http://www.broadinstitute.org/collaboration/giant/images/f/f6/GIANT_2015_WHRadjBMI_COMBINED_AllAncestries.txt.gz
wget http://www.broadinstitute.org/collaboration/giant/images/5/52/GIANT_2015_HIPadjBMI_COMBINED_EUR.txt.gz
wget http://www.broadinstitute.org/collaboration/giant/images/0/0e/GIANT_2015_HIPadjBMI_COMBINED_AllAncestries.txt.gz
wget http://www.broadinstitute.org/collaboration/giant/images/7/73/GIANT_2015_WCadjBMI_COMBINED_EUR.txt.gz
wget http://www.broadinstitute.org/collaboration/giant/images/3/3f/GIANT_2015_WCadjBMI_COMBINED_AllAncestries.txt.gz
wget http://www.broadinstitute.org/collaboration/giant/images/5/54/GIANT_2015_WHR_COMBINED_EUR.txt.gz
wget http://www.broadinstitute.org/collaboration/giant/images/d/d7/GIANT_2015_WHR_COMBINED_AllAncestries.txt.gz
wget http://www.broadinstitute.org/collaboration/giant/images/e/e4/GIANT_2015_HIP_COMBINED_EUR.txt.gz
wget http://www.broadinstitute.org/collaboration/giant/images/6/6f/GIANT_2015_HIP_COMBINED_AllAncestries.txt.gz
wget http://www.broadinstitute.org/collaboration/giant/images/5/57/GIANT_2015_WC_COMBINED_EUR.txt.gz
wget http://www.broadinstitute.org/collaboration/giant/images/e/ea/GIANT_2015_WC_COMBINED_AllAncestries.txt.gz


~~~

~~~

#2010 
#Copying/pasting/downoading/finding (however) list of top hits from each study

mkdir /mnt/lustre/home/mturchin20/Data/GIANT/2010/ 

#Height -- 180 SNPs
#Copy/pasted supplementary table 1 from http://www.nature.com.proxy.uchicago.edu/nature/journal/v467/n7317/full/nature09410.html (http://www.nature.com.proxy.uchicago.edu/nature/journal/v467/n7317/extref/nature09410-s1.pdf) into /mnt/lustre/home/mturchin20/Data/GIANT/2010/LangoAllen2010.SupplTable1.txt

#BMI -- 32 SNPs 
#Copy/pasted table 1 from http://www.nature.com.proxy.uchicago.edu/ng/journal/v42/n11/full/ng.686.html into /mnt/lustre/home/mturchin20/Data/GIANT/2010/Speliotes2010.Table1.txt

#WHRadjBMI -- 14 SNPs
#Copy/pasted table 1 from http://www.nature.com.proxy.uchicago.edu/ng/journal/v42/n11/full/ng.685.html into /mnt/lustre/home/mturchin20/Data/GIANT/2010/Heid2010.Table1.txt
#Manually removed the below three lines from table1
#Further SNPs evaluated in follow up but not achieving genome-wide significance in the combined analysis
#rs2076529       6       32,471,933      BTNL2   C       0.430   2.22 Ã 10â8     0.041   34,532  0.012   0.011   92,778  3.71 Ã 10â7     0.020
#rs7081678       10      32,030,629      ZEB1    A       0.085   5.76 Ã 10â7     0.045   76,270  0.094   0.013   100,527 5.57 Ã 10â6     0.027

#2014_5
#Copying/pasting/downoading/finding (however) list of top hits from each study

mkdir /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/ 

#Height -- 697 SNPs
#Downloaded Supplementary Table 1 from http://www.nature.com/ng/journal/v46/n11/full/ng.3097.html#supplementary-information and saved it as a .txt manually
scp -p mturchin20@wolfy.uchicago.edu:/Users/mturchin20/LabStuff/Data/GIANT/2014_5/ng.3097-S2.txt /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/. 
cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/ng.3097-S2.txt | sed 's/\r/\n/g' | perl -lane 'print join("\t", @F);' > /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/ng.3097-S2.noCarriageReturn.txt
cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/ng.3097-S2.noCarriageReturn.txt | sed 's/"//g' | sed 's/,//g' > /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/ng.3097-S2.noCarriageReturn.noQuotes.noCommas.txt

#BMI -- 97 SNPs
#Copy/pasted tables 1 & 2 from http://www.nature.com/nature/journal/v518/n7538/full/nature14177.html into /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/Locke2015.SupplTables1_2.txt
#As well as 'Extended Data Table 2' where I downloaded the jpg nature14177-st2.jpg and used http://www.free-ocr.com/ to extract the text from it
#Note -- for 'Extended Data Table 2' double-checked a few of the entries and corrected positions/rsIDs; feel this is still a better starting point than copying everything manually
cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/Locke2015.SupplTables1_2.txt > /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/Locke2015.SupplTables1_2.wManualOCREdits.txt

~~~
[  mturchin20@spudhead  ~/Data/GIANT/2014_5]$cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/Locke2015.SupplTables1_2.txt | grep rs | wc
     97     979    7660
[  mturchin20@spudhead  ~/Data/GIANT/2014_5]$cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/Locke2015.SupplTables1_2.txt | grep rs | awk '{ print $1 }' | sort | uniq -c | wc
     97     194    1774
~~~

#WHRadjBMI -- 49 SNPs (48 SNPs just with EUR; 1 SNP with 'ALL ancestries')
#Downlaoded Supplemetary Table 4 from http://www.nature.com/nature/journal/v518/n7538/full/nature14132.html#supplementary-information and saved it as a .txt manually
scp -p mturchin20@wolfy.uchicago.edu:/Users/mturchin20/LabStuff/Data/GIANT/2014_5/nature14132-s2.ST4.txt /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/.
cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/nature14132-s2.ST4.txt | sed 's/\r/\n/g' > /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/nature14132-s2.ST4.noCarriageReturn.txt
cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/nature14132-s2.ST4.noCarriageReturn.txt | grep WHRadjBMI | grep -vE '2nd|ALL' > /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/nature14132-s2.ST4.noCarriageReturn.WHRadjBMI.Eur.txt
cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/nature14132-s2.ST4.noCarriageReturn.txt | grep WHRadjBMI | grep -vE '2nd' > /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/nature14132-s2.ST4.noCarriageReturn.WHRadjBMI.ALL.txt
#CHECK_1: FIgure out the correct combinations for including the other waist traits -- do the '2nd' hits count if they are just conditional analyses? Solution -- just using all SNPs regardless of GWAS only, GWAS+metabochip, all vs Euro ancestries, all sex vs. male/female only, or joint/conditional. Not sure if can parse out all these factors for every trait (e.g. height results from GATC COJO directly and not sure if I can get 'non-conditional' hits so specifically like I can for WHRadjBMI or BMI)

~~~
[  mturchin20@spudhead  ~/Data/GIANT/2014_5]$cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/nature14132-s2.ST4.noCarriageReturn.WHRadjBMI.Eur.txt | grep rs | awk '{ print $1 }' | sort | uniq -c | wc
     48      96     871
~~~

#2013
#Downloaded TableS2 from from http://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1003500 and saved it as a .txt manually
mkdir /mnt/lustre/home/mturchin20/Data/GIANT/2013
scp -p mturchin20@wolfy.uchicago.edu:/Users/mturchin20/LabStuff/Data/GIANT/2013/journal.pgen.1003500.s007.txt /mnt/lustre/home/mturchin20/Data/GIANT/2013/.

cat /mnt/lustre/home/mturchin20/Data/GIANT/2013/journal.pgen.1003500.s007.txt | sed 's/\r/\n/g' > /mnt/lustre/home/mturchin20/Data/GIANT/2013/journal.pgen.1003500.s007.noCarriageReturn.txt

#46 unique SNPs across either or both sexes with pval < 5e-8 (from Figure 3)
~~~
[  mturchin20@spudhead  ~/Software]$cat /mnt/lustre/home/mturchin20/Data/GIANT/2013/journal.pgen.1003500.s007.noCarriageReturn.txt | awk '{ if (($25 < .00000005) || ($26 < .00000005)) { print $1 } }' | sort | uniq | wc
     46      46     464
[  mturchin20@spudhead  ~/Software]$cat /mnt/lustre/home/mturchin20/Data/GIANT/2013/journal.pgen.1003500.s007.noCarriageReturn.txt | awk '{ if ($25 < .00000005 ) { print $1 } }' | wc
     15      15     151
[  mturchin20@spudhead  ~/Software]$cat /mnt/lustre/home/mturchin20/Data/GIANT/2013/journal.pgen.1003500.s007.noCarriageReturn.txt | awk '{ if ($26 < .00000005 ) { print $1 } }' | wc
     38      38     385
~~~

#Male -- 15 SNPs
cat /mnt/lustre/home/mturchin20/Data/GIANT/2013/journal.pgen.1003500.s007.noCarriageReturn.txt | awk '{ if ($25 < .00000005 ) { print $1 } }' > /mnt/lustre/home/mturchin20/Data/GIANT/2013/journal.pgen.1003500.s007.noCarriageReturn.Male.rsIDs.txt
cat /mnt/lustre/home/mturchin20/Data/GIANT/2013/journal.pgen.1003500.s007.noCarriageReturn.txt | awk '{ if ($25 < .00000005 ) { print $1, "\t", $5, "\t", $6 } }' > /mnt/lustre/home/mturchin20/Data/GIANT/2013/journal.pgen.1003500.s007.noCarriageReturn.Male.MarkerChrBP.txt 

#Female -- 38 SNPs
cat /mnt/lustre/home/mturchin20/Data/GIANT/2013/journal.pgen.1003500.s007.noCarriageReturn.txt | awk '{ if ($26 < .00000005 ) { print $1 } }' > /mnt/lustre/home/mturchin20/Data/GIANT/2013/journal.pgen.1003500.s007.noCarriageReturn.Female.rsIDs.txt
cat /mnt/lustre/home/mturchin20/Data/GIANT/2013/journal.pgen.1003500.s007.noCarriageReturn.txt | awk '{ if ($26 < .00000005 ) { print $1, "\t", $5, "\t", $6 } }' > /mnt/lustre/home/mturchin20/Data/GIANT/2013/journal.pgen.1003500.s007.noCarriageReturn.Female.MarkerChrBP.txt




#2010
#Conducting multivariate analyses

mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1
mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/process.MTedits.For2013.vs2.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/process.MTedits.ForGIANT2010.vs1.R

#Converted process.MTedits.For2013.vs2.R to GIANT phenotypes, moving commands to height, BMI, WHRadjBMI instead of tg, tc, ldl and hdl
#Copy/pasted commands from /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/process.MTedits.ForGIANT2010.vs1.R into an interactive sessions in $PWD /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1

#mv /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/.RData /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/RData.20150519
#Redid /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/process.MTedits.ForGIANT2010.vs1.R to switch dt2 columns so that order of phenotypes was Z.height Z.BMI Z.WHRadjBMI, not Z.WHRadjBMI Z.height Z.BMI
mv /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/.RData /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/RData.20150521

gzip /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/RData.20150519
gzip /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/RData.20150521

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GLC.MTedits.For2013.vs2.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GLC.MTedits.ForGIANT2010.vs1.R

#For GIANT2010.dtlesssignif.annot.vs1.txt.gz will need (except in GIANT vals)
#posHg18 chr pos snp a1 a2 maf beta_LDL se_LDL n_LDL beta_HDL se_HDL n_HDL beta_TG se_TG n_TG beta_TC se_TC n_TC annot gene Z.tg Z.tc Z.ldl Z.hdl mvstat mvp unip (from /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/GlobalLipids2013/Vs1/dtlesssignif.annot.vs2.txt.gz)
#chr pos snp a1 a2 maf beta_height se_height n_height beta_BMI se_BMI n_BMI beta_WHRadjBMI se_WHRadjBMI n_WHRadjBMI annot gene Z.height Z.BMI Z.WHRadjBMI mvstat mvp unip

~~~
[  mturchin20@spudhead  ~/Data/GIANT/2014_5]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/dtlesssignif.vs2.txt | head -n 10
rs,Z.tg,Z.tc,Z.ldl,Z.hdl,mvstat,mvp,unip
chr10:101902054,2.45906343998997,3.65960938616346,0.54246240431255,4.84175096013916,43.8856399660704,8.16899883851383,5.89042145309562
chr10:101902184,2.04241886517916,3.31897041855539,0.741979097659704,4.28996658928395,33.5923747854106,6.04416453147192,4.74787544749436
chr10:101950956,1.89987138538418,4.80442095262784,2.98132929795538,3.49559840808192,36.4681968880102,6.63489650218934,5.80910828307783
chr10:101957202,2.3172521539636,4.44716747735029,2.65105825613512,3.51059798302414,34.6265665431121,6.25629717529947,5.0604308313441
chr10:101966491,2.38390551204578,4.24376554263135,2.1138275382868,3.50881965277072,33.4617468909834,6.01739588476946,4.65797231191253
chr10:101989736,1.89426382584585,4.15303118888146,2.03680088096434,3.87804325071873,33.8172794310246,6.0902663254914,4.48399376961395
chr10:101990691,1.97401532860328,4.24234125347809,2.21939299224744,3.75592849665782,33.7012328521064,6.06647651253057,4.65521487736734
chr10:101996419,1.96555531599126,4.668233848697,2.92978188856575,3.38349312248405,34.6728122799936,6.26579130288765,5.51741223047323
chr10:101998159,2.16811418290611,4.79021825506647,3.0723226578465,3.4849657712978,37.0775367184447,6.7603876155226,5.77832500292923
[  mturchin20@bigmem02  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT2010.dtlesssignif.vs1.txt | head -n 10
rs,Z.height,Z.BMI,Z.WHRadjBMI,mvstat,mvp,unip
rs10006067,7.08776007395249,1.65020996204492,1.8119106729526,55.8346783977573,10.663188034222,11.8655041441653
rs1000972,7.98723399811821,3.43709008602989,3.37805680205288,86.1131635514716,17.0552250763639,14.8601209135988
rs10011200,6.22294771858863,0.186567181836519,1.59819313992282,41.1157838111925,7.5945727241859,9.31166918188773
rs10013023,6.04641969222953,1.28727056310794,0.371856089385075,38.2024656839996,6.99233732659037,8.82944494147879
rs10016839,5.71547189846515,1.22387337228946,0.994457883209753,34.9475904413173,6.32221671701731,7.96098267800259
rs10017744,6.15675623449777,1.34075503369022,0.495850347347453,39.7751132103379,7.31716845232464,9.12930354201075
rs10020593,5.7374722838198,1.22652812003661,0.994457883209753,35.2050664771305,6.3751109638301,8.01727661233145
rs10023833,5.974709888153,0.0401168101849681,2.51214432793046,41.7298228825345,7.72176818166663,8.63732907027433
rs10028610,5.55041024777321,1.50237611995585,0.568051498338983,33.208638751781,5.96554504181686,7.54515513999149
[  mturchin20@spudhead  ~/Data/GIANT/2014_5]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/jointGwasMc_AllPheno.Annot.txt.gz | head -n 10
posHg18 chr pos snp a1 a2 maf beta_LDL se_LDL n_LDL beta_HDL se_HDL n_HDL beta_TG se_TG n_TG beta_TC se_TC n_TC annot gene
chr6:54599910   6       54491951        rs1280728       t       a       0.0     0.0108  0.0081  89888.00        0.0107  0.0074  94311.00        0.0139  0.0073  91013.00        0.0125  0.0080  94595.00        0       NA
chr9:3746564    9       3756564 rs10814541      c       t       0.2982  0.0011  0.0064  89888.00        0.0036  0.0058  94311.00        0.0119  0.0057  91013.00        0.0025  0.0062  94595.00        0       NA
chr5:133405061  5       133377162       rs4958172       g       a       0.1187  0       0.0089  89817.00        0.0072  0.0082  94239.00        0.0029  0.0080  90938.00        0.0029  0.0087  94520.00        0       NA
chr12:69055675  12      70769408        rs2053102       g       c       0.0     0.0069  0.0075  89888.00        0.0075  0.0069  94311.00        0.0076  0.0068  91013.00        0.0009  0.0073  94595.00        0       NA
chr5:155873914  5       155941336       rs256837        c       a       0.2005  0.0016  0.0066  88199.00        0.0002  0.0061  92620.00        0.0013  0.0060  89323.00        0.0031  0.0065  92905.00        2       NA
chr3:60337349   3       60362309        rs1569354       t       a       0.0     0.0021  0.0054  89888.00        0.0060  0.0050  94311.00        0.0035  0.0048  91013.00        0.0025  0.0053  94595.00        0       NA
chr4:65034366   4       65351771        rs6851961       a       g       0.4631  0.0003  0.0053  89877.00        0.0041  0.0049  94300.00        0.0025  0.0048  91002.00        0.0022  0.0052  94584.00        0       NA
chr7:62372905   7       62735470        rs11560217      c       a       0.03034 0.0126  0.0161  83513.00        0.0236  0.0147  88983.00        0.0041  0.0148  81387.00        0.0008  0.0157  88090.00        0       NA
chr17:6058839   17      6118115 rs17803227      a       g       0.1016  0.0046  0.0094  88422.00        0.0055  0.0086  92809.00        0.0081  0.0084  89474.00        0.0013  0.0092  93056.00        0       NA
[  mturchin20@spudhead  ~]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/multivariate/globallipids/dtlesssignif.annot.txt | wc
8066  209716 1794390
[  mturchin20@spudhead  ~]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/multivariate/globallipids/dtlesssignif.annot.txt | perl -lane 'if ($F[$#F-1] > $F[$#F]) { print join("\t", @F) ; } ' | wc
3724   96824  830117
[  mturchin20@spudhead  ~]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/dtlesssignif.annot.vs2.txt.gz | wc
12463  348964 3301334
[  mturchin20@spudhead  ~]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/dtlesssignif.annot.vs2.txt.gz | perl -lane 'if ($F[$#F-1] > $F[$#F]) { print join("\t", @F) ; } ' | wc
8859  248052 2346362
[  mturchin20@bigmem02  ~]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT2010.dtlesssignif.vs1.annot.vs1.txt.gz | wc
   8842  203366 1641132
   [  mturchin20@bigmem02  ~]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT2010.dtlesssignif.vs1.annot.vs1.txt.gz | perl -lane 'if ($F[$#F-1] > $F[$#F]) { print join("\t", @F) ; } ' | wc
   1097   25231  202912
~~~

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/DataOrganization.PythonVLookUp.CreateFullAnnotFile.vs1.py /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/DataOrganization.GIANT2010.PythonVLookUp.CreateFullAnnotFile.vs1.py

cat /mnt/lustre/home/mturchin20/Data/GIANT/2010/LangoAllen2010.SupplTable1.txt | grep rs | awk '{ print $1, "\t", $2, "\t", $3 }' > /mnt/lustre/home/mturchin20/Data/GIANT/2010/LangoAllen2010.SupplTable1.rsIDs.MarkerChrBP.txt

cat /mnt/lustre/home/mturchin20/Data/GIANT/2010/Speliotes2010.Table1.txt | grep rs | perl -ane 'print $F[0], "\t"; for (my $i = 0; $i <= $#F; $i++) { if ($F[$i] =~ m/\d,\d/) { print $F[$i-1], "\t", $F[$i], "\t"; } } print "\n";' | awk '{ print $1, "\t", $2, "\t", $3 }' | sed 's/,//g' > /mnt/lustre/home/mturchin20/Data/GIANT/2010/Speliotes2010.Table1.rsIDs.MarkerChrBP.txt

cat /mnt/lustre/home/mturchin20/Data/GIANT/2010/Heid2010.Table1.txt | grep rs | awk '{ print $1, "\t", $2, "\t", $3 }' | sed 's/,//g' > /mnt/lustre/home/mturchin20/Data/GIANT/2010/Heid2010.Table1.rsIDs.MarkerChrBP.txt

cat /mnt/lustre/home/mturchin20/Data/GIANT/2010/LangoAllen2010.SupplTable1.rsIDs.MarkerChrBP.txt /mnt/lustre/home/mturchin20/Data/GIANT/2010/Speliotes2010.Table1.rsIDs.MarkerChrBP.txt /mnt/lustre/home/mturchin20/Data/GIANT/2010/Heid2010.Table1.rsIDs.MarkerChrBP.txt > /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt

~~~
[  mturchin20@spudhead  ~]$cat /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt | wc
226     678    5794
[  mturchin20@spudhead  ~]$cat /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt | sort | uniq -c | wc
226     904    7602
[  mturchin20@spudhead  ~]$cat /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt | awk '{ print $1 }' | sort | uniq -c | wc
226     452    4074
~~~

#Need to get basepair position for the GIANT GWAS summary files -- only have rsIDs for all of them. Guessing/hoping dbSNP main vcf has this information
mkdir /mnt/lustre/home/mturchin20/Data/dbSNP 
cd /mnt/lustre/home/mturchin20/Data/dbSNP 
wget ftp://ftp.ncbi.nih.gov/snp/organisms/human_9606_b142_GRCh37p13/VCF/All_20150415.vcf.gz 
wget ftp://ftp.ncbi.nih.gov/snp/organisms/human_9606_b142_GRCh37p13/VCF/All_20150415.vcf.gz.tbi
wget ftp://ftp.ncbi.nih.gov/snp/organisms/human_9606_b142_GRCh38/VCF/All_20150416.vcf.gz
mv /mnt/lustre/home/mturchin20/Data/dbSNP/All_20150416.vcf.gz /mnt/lustre/home/mturchin20/Data/dbSNP/dbSNP_human_9606_b142_GRCh38_All_20150416.vcf.gz
wget ftp://ftp.ncbi.nih.gov/snp/organisms/human_9606_b142_GRCh38/VCF/All_20150416.vcf.gz.tbi
mv /mnt/lustre/home/mturchin20/Data/dbSNP/All_20150416.vcf.gz.tbi /mnt/lustre/home/mturchin20/Data/dbSNP/dbSNP_human_9606_b142_GRCh38_All_20150416.vcf.gz.tbi

~~~
[  mturchin20@spudhead  ~/Data/dbSNP]$zcat All_20150415.vcf.gz | head -n 80 | tail -n 10
1       10250   rs199706086     A       C       .       .       RS=199706086;RSPOS=10250;dbSNPBuildID=137;SSR=0;SAO=0;VP=0x050000020005000002000100;WGT=1;VC=SNV;R5;ASP
1       10254   rs140194106     TA      T       .       .       RS=140194106;RSPOS=10255;dbSNPBuildID=134;SSR=0;SAO=0;VP=0x050000020005000002000200;WGT=1;VC=DIV;R5;ASP
1       10257   rs111200574     A       C       .       .       RS=111200574;RSPOS=10257;dbSNPBuildID=132;SSR=0;SAO=0;VP=0x050100020005000102000100;WGT=1;VC=SNV;SLO;R5;ASP;GNO
1       10259   rs200940095     C       A       .       .       RS=200940095;RSPOS=10259;dbSNPBuildID=137;SSR=0;SAO=0;VP=0x050000020005000002000100;WGT=1;VC=SNV;R5;ASP
1       10291   rs145427775     C       T       .       .       RS=145427775;RSPOS=10291;dbSNPBuildID=134;SSR=0;SAO=0;VP=0x050000020005000002000100;WGT=1;VC=SNV;R5;ASP
1       10327   rs112750067     T       C       .       .       RS=112750067;RSPOS=10327;dbSNPBuildID=132;SSR=0;SAO=0;VP=0x050000020005000002000100;WGT=1;VC=SNV;R5;ASP
1       10328   rs201106462     AACCCCTAACCCTAACCCTAACCCT       A       .       .       RS=201106462;RSPOS=10329;dbSNPBuildID=137;SSR=0;SAO=0;VP=0x050000020005000002000200;WGT=1;VC=DIV;R5;ASP
1       10329   rs150969722     AC      A       .       .       RS=150969722;RSPOS=10330;dbSNPBuildID=134;SSR=0;SAO=0;VP=0x050000020005000002000200;WGT=1;VC=DIV;R5;ASP
1       10352   rs145072688     T       TA      .       .       RS=145072688;RSPOS=10353;dbSNPBuildID=134;SSR=0;SAO=0;VP=0x050000020015000002000200;WGT=1;VC=DIV;R5;OTH;ASP;CAF=0.5625,0.4375;COMMON=1
1       10352   rs555500075     T       TA      .       .       RS=555500075;RSPOS=10352;dbSNPBuildID=142;SSR=0;SAO=0;VP=0x050000020015140024000200;WGT=1;VC=DIV;R5;OTH;ASP;VLD;KGPhase3;CAF=0.5625,0.4375;COMMON=1
[  mturchin20@spudhead  ~/Data/dbSNP]$zcat All_20150415.vcf.gz | awk '{ print $1, "\t", $2, "\t", $3 }' | wc
111198367 333595084 3048046196
~~~

cp -p /mnt/lustre/home/mturchin20/Scripts/Python/2fileTemplate.vs1.py /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs1.py
cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs1.py /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs2.py


python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/dbSNP/All_20150415.vcf.gz --file2 /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.txt.gz | gzip > /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.txt.gz
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/dbSNP/All_20150415.vcf.gz --file2 /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.txt.gz | gzip > /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.txt.gz
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/dbSNP/All_20150415.vcf.gz --file2 /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.txt.gz | gzip > /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.txt.gz

#Ordering the two files this way due to large 
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs2.py --file1 /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.txt.gz --file2 /mnt/lustre/home/mturchin20/Data/dbSNP/All_20150415.vcf.gz | gzip > /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.vs2.txt.gz
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs2.py --file1 /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.txt.gz --file2 /mnt/lustre/home/mturchin20/Data/dbSNP/All_20150415.vcf.gz | gzip > /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.vs2.txt.gz
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs2.py --file1 /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.txt.gz --file2 /mnt/lustre/home/mturchin20/Data/dbSNP/All_20150415.vcf.gz | gzip > /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.vs2.txt.gz

~~~
[  mturchin20@bigmem02  ~/Data/dbSNP]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.txt.gz | wc
2469636 17287452 113790927
[  mturchin20@bigmem02  ~/Data/dbSNP]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.vs2.txt.gz | wc
2465677 17259739 113641249
[  mturchin20@bigmem02  ~/Data/dbSNP]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.vs3.txt.gz | wc
2469636 17287452 113790924
[  mturchin20@bigmem02  ~/Data/dbSNP]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.txt.gz | wc
2483326 17383282 106629662
[  mturchin20@bigmem02  ~/Data/dbSNP]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.vs3.txt.gz | wc
2483326 17383282 106629659
[  mturchin20@bigmem02  ~/Data/dbSNP]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.txt.gz | wc
2471517 17300619 111437629
[  mturchin20@bigmem02  ~/Data/dbSNP]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.vs3.txt.gz | wc
2471517 17300619 111437626
[  mturchin20@bigmem02  ~/Data/dbSNP]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.txt.gz | wc             2471517 14829102 82685870
[  mturchin20@bigmem02  ~/Data/dbSNP]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.txt.gz | wc            2483326 14899956 78051825
[  mturchin20@bigmem02  ~/Data/dbSNP]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.txt.gz | wc         2469636 14817816 85061092
[  mturchin20@bigmem02  ~/Data/dbSNP]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.vs3.txt.gz | grep NA | wc
3972   27804  145898
[  mturchin20@bigmem02  ~/Data/dbSNP]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.txt.gz | grep NA | wc
3971   27797  145841
[  mturchin20@bigmem02  ~/Data/dbSNP]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.vs3.txt.gz | grep Marker
MarkerName      Allele1 Allele2 Freq.Allele1.HapMapCEU  p       N       NA
[  mturchin20@bigmem02  ~/Data/dbSNP]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.txt.gz | grep Marker
MarkerName      Allele1 Allele2 Freq.Allele1.HapMapCEU  p       N       ChrBP
[  mturchin20@bigmem02  ~/Data/dbSNP]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.txt.gz | grep NA | wc
3958   27706  149618
[  mturchin20@bigmem02  ~/Data/dbSNP]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.vs3.txt.gz | grep NA | wc
3959   27713  149675
[  mturchin20@bigmem02  ~/Data/dbSNP]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.txt.gz | grep NA | wc
39972  279804 1229461
[  mturchin20@bigmem02  ~/Data/dbSNP]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.vs3.txt.gz | grep NA | wc
39973  279811 1229518
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.vs3.txt.gz | grep NA | grep -v rs
MarkerName      Allele1 Allele2 Freq.Allele1.HapMapCEU  p       N       NA
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.txt.gz | grep NA | grep -v rs
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.txt.gz | grep NA | grep -v rs
~~~

scp -p mturchin20@wolfy.uchicago.edu:/Users/mturchin20/LabStuff/Data/UCSC/GenomeBrowser/dbSNP130.txt.gz /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/. 

~~~
[  mturchin20@spudhead  ~/Data/dbSNP]$zcat /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/dbSNP130.txt.gz | wc
18833532 188335320 1426536856
~~~

#Redoing the ChrBP annotation steps with dbSNP130 now
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/dbSNP130.txt.gz --file2 /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.txt.gz | gzip > /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/dbSNP130.txt.gz --file2 /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.txt.gz | gzip > /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/dbSNP130.txt.gz --file2 /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.txt.gz | gzip > /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz &

zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep , | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | grep -v rs | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep , | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | grep -v rs | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep , | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | grep -v rs | wc

~~~
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | wc
2469636 17287452 114167633
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep , | wc
10809   75663  865971
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | wc
1981   13867   75103
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | wc
2471517 17300619 111814122
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep , | wc
10805   75635  848551
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | wc
1980   13860   73049
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | wc
2483326 17383282 106990904
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep , | wc
10446   73122  797793
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | wc
37936  265552 1159088
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | grep -v rs | wc
0       0       0
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | grep -v rs | wc
0       0       0
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | grep -v rs | wc
0       0       0
~~~

#Going to be conservative and allow a rsID to exist in multiple locations if there are multiple locations for now from dbSNP130 (hence the ',' marks). Will include code that accounts for this in the 'annotation' script as well as to identify whether there is no 'rsID' for the marker and just chr#:##### which a few of the files contain (though I think most are 100% rs##### -- which is a bit interesting/odd since thought the same platform was being used across all phenotypes....but I guess there might be differences due to imputation quality/results or something for a given dataset/phenotype too?....different cohorts in total are being used across each dataest too)

#20150520 -- Realized the 2010 GIANT data doesn't have betas or SEs......so can't use them for this yet. Might be able to get that data from Joel if ask but for now can only do the 2014_5 data and the sex-stratified 2013 data
#20150520 -- 30 minutes later I realized/remembered the method only uses Zscores, maf and n, not betas or SEs. For some reason we just keep that information, when present, in the prep files but it's not actually used by the method. So....I can go ahead with this like I originally planned. yay. \o/

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/DataOrganization.GIANT2010.PythonVLookUp.CreateFullAnnotFile.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt --file2 /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz,/mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz,/mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT_AllPheno_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz 
#python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/DataOrganization.GIANT2010.PythonVLookUp.CreateFullAnnotFile.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt --file2 /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT_AllPheno_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.Trbl1.gz
#python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/DataOrganization.GIANT2010.PythonVLookUp.CreateFullAnnotFile.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt --file2 /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz,/mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT_AllPheno_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.Trbl2.gz
#python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/DataOrganization.GIANT2010.PythonVLookUp.CreateFullAnnotFile.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt --file2 /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz,/mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT_AllPheno_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.Trbl3.gz

#Added the following header to /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT_AllPheno_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz
#MarkerName a1 a2 maf annot beta_height se_height n_height beta_BMI se_BMI n_BMI beta_WHRadjBMI se_WHRadjBMI n_WHRadjBMI

#Troubleshooting /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/DataOrganization.GIANT2010.PythonVLookUp.CreateFullAnnotFile.vs1.py stuff
~~~
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep -E 'rs2154319|rs10863936|rs11205277|rs4665736' | gzip > /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.4SNPs.txt.gz
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/DataOrganization.GIANT2010.PythonVLookUp.CreateFullAnnotFile.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt --file2 /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.4SNPs.txt.gz,/mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.4SNPs.txt.gz
/mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.4SNPs.txt.gz,/mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.4SNPs.txt.gz
Happening1
rs11205277      g       a       NA      2       NA      NA      127114  NA      NA      127114
Happening1
rs10863936      g       a       NA      2       NA      NA      133820  NA      NA      133820
Happening1
rs2154319       c       t       NA      2       NA      NA      133480  NA      NA      133480
Happening1
rs4665736       t       c       0.4667  2       NA      NA      133780  NA      NA      133780
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT_AllPheno_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.Trbl1.gz | wc
2469635 34574890 132103187
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT_AllPheno_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.Trbl2.gz | wc
2469635 34574890 141891923
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT_AllPheno_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.Trbl1.gz | head -n 10
rs999721        a       g       0.4667  0       NA      NA      133732  NA      NA      NA      NA      NA      NA
rs9821657       g       a       0.175   0       NA      NA      133582  NA      NA      NA      NA      NA      NA
rs1919329       t       g       0.4833  0       NA      NA      133834  NA      NA      NA      NA      NA      NA
rs6444035       c       t       0.0667  0       NA      NA      133803  NA      NA      NA      NA      NA      NA
rs3957240       c       t       0.05    0       NA      NA      122588  NA      NA      NA      NA      NA      NA
rs1919324       c       t       0.1417  0       NA      NA      133796  NA      NA      NA      NA      NA      NA
rs2739330       c       t       0.3833  0       NA      NA      133781  NA      NA      NA      NA      NA      NA
rs973978        a       g       NA      0       NA      NA      133825  NA      NA      NA      NA      NA      NA
rs17597444      t       g       0.25    0       NA      NA      133772  NA      NA      NA      NA      NA      NA
rs17597445      t       c       NA      0       NA      NA      133810  NA      NA      NA      NA      NA      NA
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT_AllPheno_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.Trbl2.gz | head -n 10
rs999721        a       g       0.4667  0       NA      NA      133732  NA      NA      133732  NA      NA      NA
rs9821657       g       a       0.175   0       NA      NA      133582  NA      NA      133582  NA      NA      NA
rs1919329       t       g       0.4833  0       NA      NA      133834  NA      NA      133834  NA      NA      NA
rs6444035       c       t       0.0667  0       NA      NA      133803  NA      NA      133803  NA      NA      NA
rs3957240       c       t       0.05    0       NA      NA      122588  NA      NA      122588  NA      NA      NA
rs1919324       c       t       0.1417  0       NA      NA      133796  NA      NA      133796  NA      NA      NA
rs2739330       c       t       0.3833  0       NA      NA      133781  NA      NA      133781  NA      NA      NA
rs973978        a       g       NA      0       NA      NA      133825  NA      NA      133825  NA      NA      NA
rs17597444      t       g       0.25    0       NA      NA      133772  NA      NA      133772  NA      NA      NA
rs17597445      t       c       NA      0       NA      NA      133810  NA      NA      133810  NA      NA      NA
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT_AllPheno_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | wc
2516306 35228284 151538382
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT_AllPheno_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | head -n 10
rs999721        a       g       0.4667  0       NA      NA      133732  NA      NA      123835  NA      NA      77154
rs9821657       g       a       0.175   0       NA      NA      133582  NA      NA      123732  NA      NA      77043
rs1919329       t       g       0.4833  0       NA      NA      133834  NA      NA      123862  NA      NA      77166
rs6444035       c       t       0.0667  0       NA      NA      133803  NA      NA      123814  NA      NA      77155
rs3957240       c       t       0.05    0       NA      NA      122588  NA      NA      114400  NA      NA      68863
rs1919324       c       t       0.1417  0       NA      NA      133796  NA      NA      123864  NA      NA      77168
rs2739330       c       t       0.3833  0       NA      NA      133781  NA      NA      123892  NA      NA      77209
rs973978        a       g       NA      0       NA      NA      133825  NA      NA      123846  NA      NA      77155
rs17597444      t       g       0.25    0       NA      NA      133772  NA      NA      123864  NA      NA      77168
rs17597445      t       c       NA      0       NA      NA      133810  NA      NA      123867  NA      NA      77169
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT_AllPheno_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | awk '{ if ($4 == "NA") { print $0 } } ' | wc
 225095 3151330 12559530
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/DataOrga
nization.GIANT2010.PythonVLookUp.CreateFullAnnotFile.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt --file2 /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.4SNPs.txt.gz,/mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.4SNPs.txt.gz
/mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.4SNPs.txt.gz,/mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.4SNPs.txt.gz
rs11205277      g       a       NA      1       NA      NA      127114  NA      NA      127114  NA      NA      NA
rs10863936      g       a       NA      1       NA      NA      133820  NA      NA      133820  NA      NA      NA
rs2154319       c       t       NA      1       NA      NA      133480  NA      NA      133480  NA      NA      NA
rs4665736       t       c       0.4667  1       NA      NA      133780  NA      NA      133780  NA      NA      NA
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT_AllPheno_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | awk '{ print $5 }' | sort | uniq -c
2292802 0
226 1
223278 2
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$cat /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt | wc
226     678    5794
[  mturchin20@bigmem02  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1]$python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/DataOrganization.GIANT2010.PythonVLookUp.CreateFullAnnotFile.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt --file2 /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.4SNPs.txt.gz,/mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.4SNPs.txt.gz
/mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.4SNPs.txt.gz,/mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.4SNPs.txt.gz
rs11205277      g       a       1_148159495     NA      1       NA      NA      NA      127114  NA      NA      NA      127114  NA      NA      NA      NA
rs10863936      g       a       1_210304420     NA      1       NA      NA      NA      133820  NA      NA      NA      133820  NA      NA      NA      NA
rs2154319       c       t       1_41518356      NA      1       NA      NA      NA      133480  NA      NA      NA      133480  NA      NA      NA      NA
rs4665736       t       c       2_25041102      0.4667  1       0.4667  NA      NA      133780  0.4667  NA      NA      133780  NA      NA      NA      NA
[  mturchin20@bigmem02  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1]$zcat GIANT_AllPheno_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | awk '{ print $6 }' | sort | uniq -c
2292802 0
    226 1
[  mturchin20@bigmem02  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1]$zcat GIANT_AllPheno_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | awk '{ print $5, "\t", $7, "\t", $11, "\t", $15 }' | sort | uniq -c                                                                                             
.
.
.
223278 2
    14 0.2      NA      NA      0.2
    30 0.2      NA      NA      NA
 33331 0.3      0.3     0.3     0.3
   442 0.3      0.3     0.3     NA
     1 0.3      0.3     NA      0.3
     5 0.3      0.3     NA      NA
 33582 0.3083   0.3083          0.3083          0.3083
   450 0.3083   0.3083          0.3083          NA
.
.
.
[  mturchin20@bigmem02  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1]$zcat GIANT_AllPheno_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | awk '{ if ($5 == "NA") { print $5, "\t", $7, "\t", $11, "\t", $15 } }' | sort | uniq -c
     2 NA       NA      NA      0.0
225093 NA       NA      NA      NA
[  mturchin20@bigmem02  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1]$zcat GIANT_AllPheno_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | awk '{ if (($5 == "0.2") && ($7 == "NA")) { print $0 } } ' | head -n 10
rs3006263       c       g       10_26198595     0.2     0       NA      NA      NA      NA      NA      NA      0.2     NA      NA      41946   NA      NA
rs9646391       c       t       17_69523401     0.2     0       NA      NA      NA      NA      NA      NA      0.2     NA      NA      43049   NA      NA
rs4789011       t       c       17_69345766     0.2     0       NA      NA      NA      NA      NA      NA      0.2     NA      NA      43206   NA      NA
rs11124487      a       g       2_35975673      0.2     0       NA      NA      NA      NA      NA      NA      0.2     NA      NA      41710   NA      NA
rs267154        t       c       3_76963429      0.2     0       NA      NA      NA      NA      NA      NA      0.2     NA      NA      41885   NA      NA
rs4085550       g       c       13_22416770     0.2     0       NA      NA      NA      NA      NA      NA      0.2     NA      NA      46353   NA      NA
rs1593477       c       t       18_2291963      0.2     0       NA      NA      NA      0.2     NA      NA      62647   NA      0.2     NA      NA      48836
rs6051233       g       c       20_2622734      0.2     0       NA      NA      NA      0.2     NA      NA      69572   NA      0.2     NA      NA      43449
rs11768737      c       a       7_41341790      0.2     0       NA      NA      NA      NA      NA      NA      0.2     NA      NA      40685   NA      NA
rs17406991      c       t       7_78515066      0.2     0       NA      NA      NA      0.2     NA      NA      62027   NA      0.2     NA      NA      46607
[  mturchin20@bigmem02  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1]$zcat GIANT_AllPheno_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | perl -lane 'print $#F;' | sort | uniq -c
2516306 17
[  mturchin20@bigmem02  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep rs4789011
rs4789011       t       c       0.2     0.21    43206   17_69345766
[  mturchin20@bigmem02  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep -E 'rs1593477|rs4789011|rs3006263|rs17406991' | gzip > /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.4otherSNPs.txt.gz
[  mturchin20@bigmem02  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.4otherSNPs.txt.gz
rs1593477       c       t       0.2     0.24    48836   18_2291963
rs1593477       c       t       0.2     0.357   62647   18_2291963
rs17406991      c       t       0.8     0.446   62027   7_78515066
rs17406991      t       c       0.2     0.39    46607   7_78515066
rs3006263       c       g       0.2     0.32    41946   10_26198595
rs4789011       t       c       0.2     0.21    43206   17_69345766
[  mturchin20@bigmem02  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1]$python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/DataOrganization.GIANT2010.PythonVLookUp.CreateFullAnnotFile.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt --file2 /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.4otherSNPs.txt.gz,/mnt/gluster/data/external_public_supp/GIANT2010/GIANT_2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.4otherSNPs.txt.gz
/mnt/gluster/data/external_public_supp/GIANT2010/GIANT_2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.4otherSNPs.txt.gz,/mnt/gluster/data/external_public_supp/GIANT2010/GIANT_2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.4otherSNPs.txt.gz
Error2a -- line2[0] present more than once in a particular file (0: rs1593477,c,t,0.2,0.357,62647,18_2291963)
Error2a -- line2[0] present more than once in a particular file (0: rs17406991,t,c,0.2,0.39,46607,7_78515066)
Error2a -- line2[0] present more than once in a particular file (1: rs1593477,c,t,0.2,0.357,62647,18_2291963)
Error2a -- line2[0] present more than once in a particular file (1: rs17406991,t,c,0.2,0.39,46607,7_78515066)
rs1593477       c       t       18_2291963      0.2     0       0.2     NA      NA      62647   0.2     NA      NA      62647   NA      NA      NA      NA
rs4789011       t       c       17_69345766     0.2     0       0.2     NA      NA      43206   0.2     NA      NA      43206   NA      NA      NA      NA
rs3006263       c       g       10_26198595     0.2     0       0.2     NA      NA      41946   0.2     NA      NA      41946   NA      NA      NA      NA
rs17406991      c       t       7_78515066      0.2     0       0.2     NA      NA      46607   0.2     NA      NA      46607   NA      NA      NA      NA
[  mturchin20@bigmem02  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1]$zcat GIANT_AllPheno_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | wc
2516306 45293508 227795829
[  mturchin20@bigmem02  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1]$zcat GIANT_AllPheno_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | perl -lane 'print $#F;' | sort | uniq -c
2516306 17
[  mturchin20@bigmem02  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1]$zcat GIANT_AllPheno_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | awk '{ if ($5 == "NA") { print $5, "\t", $7, "\t", $11, "\t", $15 } }' | sort | uniq -c
 225093 NA       NA      NA      NA
[  mturchin20@bigmem02  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1]$zcat GIANT_AllPheno_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | awk '{ print $8, "\t", $9, "\t", $12, "\t", $13, "\t", $16, "\t", $17 }' | sort | uniq -c
2516306 NA   NA      NA      NA      NA      NA
[  mturchin20@bigmem02  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1]$zcat GIANT_AllPheno_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | perl -lane 'if (($F[9] =~ m/\D+/) && ($F[9] ne "NA")) { print join("\t", @F); } ' | sort | uniq -c
1 rs796188        a       c       13_88518668     0.0167  0       0.0167  NA      NA      1e+05   0.0167  NA      NA      92198   0.0167  NA      NA      56914
1 rs9842319       t       g       3_144835684     0.0333  0       0.0333  NA      NA      1e+05   0.0333  NA      NA      95468   0.0333  NA      NA      54663
[  mturchin20@bigmem02  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1]$zcat GIANT_AllPheno_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | perl -lane 'if (($F[13] =~ m/\D+/) && ($F[13] ne "NA")) { print join("\t", @F); } ' | sort | uniq -c
1 rs11244235      g       c       9_132864356     0.025   2       0.025   NA      NA      108437  0.025   NA      NA      1e+05   0.025   NA      NA      62450
1 rs12917986      c       g       16_72309046     0.2083  0       0.2083  NA      NA      109441  0.2083  NA      NA      1e+05   0.2083  NA      NA      61723
1 rs17599562      g       t       10_122917026    0.05    0       0.05    NA      NA      95129   0.05    NA      NA      1e+05   0.05    NA      NA      66847
[  mturchin20@bigmem02  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1]$zcat GIANT_AllPheno_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | perl -lane 'if (($F[17] =~ m/\D+/) && ($F[17] ne "NA")) { print join("\t", @F); } ' | sort | uniq -c
~~~

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/DataOrganization.PythonVLookUp.PutTogetherProcessAndAnnotFiles.vs1.py /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/DataOrganization.GIANT2010.PythonVLookUp.PutTogetherProcessAndAnnotFiles.vs1.py

#python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/DataOrganization.GIANT2010.PythonVLookUp.PutTogetherProcessAndAnnotFiles.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT2010.dtlesssignif.vs1.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT_AllPheno_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT2010.dtlesssignif.vs1.annot.vs1.txt.gz 
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/DataOrganization.GIANT2010.PythonVLookUp.PutTogetherProcessAndAnnotFiles.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT2010.dtlesssignif.vs1.txt --file2 <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT_AllPheno_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | perl -lane 'my $chr; my $pos; if ($F[3] eq "NA") { $chr = "NA"; $pos = "NA"; } else { $chr = ((split(/_/, $F[3]))[0]); $pos = ((split(/_/, $F[3]))[1]); } splice(@F, 1, 0, ($chr, $pos)); splice(@F, 8, 1); splice(@F, 11, 1); splice(@F, 14, 1); print join("\t", @F);') | \
gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT2010.dtlesssignif.vs1.annot.vs1.txt.gz
#Added the following header to /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT2010.dtlesssignif.vs1.annot.vs1.txt.gz
#snp chr pos a1 a2 chrbp maf annot beta_height se_height n_height beta_BMI se_BMI n_BMI beta_WHRadjBMI se_WHRadjBMI n_WHRadjBMI Z.height Z.BMI Z.WHRadjBMI mvstat mvp unip

#Converted GLC.MTedits.For2013.vs2.R to GIANT phenotypes, moving commands to height, BMI, WHRadjBMI instead of tg, tc, ldl and hdl
#Copy/pasted commands from /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GLC.MTedits.ForGIANT2010.vs1.R into an interactive sessions in $PWD /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1

cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt | grep -f <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT2010.newtophits.vs1.txt | awk '{ print $1 }') 

~~~
[  mturchin20@spudhead  ~]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT2010.newtophits.vs1.txt | wc                                20     160     861
[  mturchin20@spudhead  ~]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT2010.newtophits.vs1.txt | awk '{ print $1 }' | grep -v -f <(cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt | grep -f <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT2010.newtophits.vs1.txt | awk '{ print $1 }') | awk '{ print $1 }') | wc
14      14     138
[  mturchin20@spudhead  ~]$cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt | grep -f <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT2010.newtophits.vs1.txt | awk '{ print $1 }') | wc
6      18     155
[  mturchin20@spudhead  ~]$cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt | grep -f <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT2010.newtophits.vs1.txt | awk '{ print $1 }')
rs34651          5       72179761
rs12204421       6       33736841
rs648831         6       81012927
rs17783015       12      88755517
rs11835818       12      120979192
rs1809889        12      123367179
[  mturchin20@spudhead  ~]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT2010.newtophits.vs1.txt | awk '{ print $1 }' | grep -v -f <(cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt | grep -f <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT2010.newtophits.vs1.txt | awk '{ print $1 }') | awk '{ print $1 }')
snp
rs17016663
rs10805383
rs7601531
rs2025151
rs6824258
rs4735692
rs7614120
rs7081678
rs12534698
rs2390312
rs389883
rs10040888
rs10822129
~~~

mv /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/.RData /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/RData.GIANT2010.GLC.20150522

gzip /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/RData.GIANT2010.GLC.20150522





#2014_5
#Conducting multivariate analyses

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/process.MTedits.ForGIANT2010.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/process.MTedits.ForGIANT2014_5.vs1.R

#A bit anachronistic here -- editing the 'dbSNP130' versions of the GWAS file to correct the 'headers' so they can be properly used by R and other processes

cp -p /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.txt.gz
cp -p /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHR_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHR_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.txt.gz
cp -p /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_HIPadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_HIPadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.txt.gz
cp -p /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_HIP_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_HIP_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.txt.gz
cp -p /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WCadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WCadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.txt.gz
cp -p /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WC_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WC_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.txt.gz

#Copy/pasted commands from /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/process.MTedits.ForGIANT2014_5.vs1.R into an interactive sessions in $PWD /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/

mv /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/.RData /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/RData.GIANT2014_5.process.20150522
gzip /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/RData.GIANT2014_5.process.20150522

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GLC.MTedits.ForGIANT2010.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GLC.MTedits.ForGIANT2014_5.Orig3.vs1.R

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/DataOrganization.GIANT2010.PythonVLookUp.CreateFullAnnotFile.vs1.py /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/DataOrganization.GIANT2014_5.PythonVLookUp.CreateFullAnnotFile.vs1.py

cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/ng.3097-S2.noCarriageReturn.noQuotes.noCommas.txt | grep rs | awk '{ print $1, "\t", $2, "\t", $3 }' > /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/ng.3097-S2.noCarriageReturn.noQuotes.noCommas.rsIDs.MarkerChrBP.txt

cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/Locke2015.SupplTables1_2.wManualOCREdits.txt | grep rs | awk '{ print $1, "\t", $2 }' | sed 's/,//g' | perl -lane 'print $F[0], "\t", ((split(/:/, $F[1]))[0]), "\t", ((split(/:/, $F[1]))[1]);' > /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/Locke2015.SupplTables1_2.wManualOCREdits.rsIDs.MarkerChrBP.txt

~~~
[  mturchin20@spudhead  ~]$cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/Locke2015.SupplTables1_2.wManualOCREdits.txt| grep rs | awk '{ print $1, "\t", $2 }' | sed 's/,//g' | perl -lane 'print $F[0], "\t", ((split(/:/, $F[1]))[0]), "\t", ((split(/:/, $F[1]))[1]);' | wc                                                97     291    2130
~~~

#cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/nature14132-s2.ST4.noCarriageReturn.WHRadjBMI.Eur.txt | grep rs | awk '{ print $1, "\t", $3, "\t", $4 }' | sed 's/"//g' | sed 's/,//g' | grep -v This > /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/nature14132-s2.ST4.noCarriageReturn.WHRadjBMI.Eur.rsIDs.MarkerChrBP.txt
cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/nature14132-s2.ST4.noCarriageReturn.txt | grep WHRadjBMI | grep rs | awk '{ print $1, "\t", $3, "\t", $4 }' | sed 's/"//g' | sed 's/,//g' | grep -v This > /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/nature14132-s2.ST4.noCarriageReturn.WHRadjBMI.rsIDs.MarkerChrBP.txt
cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/nature14132-s2.ST4.noCarriageReturn.txt | grep rs | awk '{ print $1, "\t", $3, "\t", $4 }' | sed 's/"//g' | sed 's/,//g' | grep -v This > /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/nature14132-s2.ST4.noCarriageReturn.rsIDs.MarkerChrBP.txt

~~~
[  mturchin20@spudhead  ~/Data/GIANT/2014_5]$cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/nature14132-s2.ST4.noCarriageReturn.WHRadjBMI.rsIDs.MarkerChrBP.txt | wc
    68     204    1760
[  mturchin20@spudhead  ~/Data/GIANT/2014_5]$cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/nature14132-s2.ST4.noCarriageReturn.rsIDs.MarkerChrBP.txt | wc
    87     261    2245
~~~

cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/ng.3097-S2.noCarriageReturn.noQuotes.noCommas.rsIDs.MarkerChrBP.txt /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/Locke2015.SupplTables1_2.wManualOCREdits.rsIDs.MarkerChrBP.txt /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/nature14132-s2.ST4.noCarriageReturn.WHRadjBMI.rsIDs.MarkerChrBP.txt > /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt

cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/ng.3097-S2.noCarriageReturn.noQuotes.noCommas.rsIDs.MarkerChrBP.txt /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/Locke2015.SupplTables1_2.wManualOCREdits.rsIDs.MarkerChrBP.txt /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/nature14132-s2.ST4.noCarriageReturn.rsIDs.MarkerChrBP.txt > /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.AllPheno.MarkerChrBP.txt

~~~
[  mturchin20@spudhead  ~/Data/GIANT/2014_5]$cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt | wc
862    2586   21825
[  mturchin20@spudhead  ~/Data/GIANT/2014_5]$cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.AllPheno.MarkerChrBP.txt | wc
881    2643   22310
[  mturchin20@spudhead  ~]$cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt | awk '{ print $1 }' | sort | uniq | wc
862     862    8731
[  mturchin20@spudhead  ~]$cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.AllPheno.MarkerChrBP.txt | awk '{ print $1 }' | sort | uniq | wc
881     881    8921
~~~

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/dbSNP/All_20150415.vcf.gz --file2 /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.txt.gz | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/dbSNP/All_20150415.vcf.gz --file2 /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.gz | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/dbSNP/All_20150415.vcf.gz --file2 /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.txt.gz | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/dbSNP/All_20150415.vcf.gz --file2 /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_HIPadjBMI_COMBINED_EUR.txt.gz | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_HIPadjBMI_COMBINED_EUR.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/dbSNP/All_20150415.vcf.gz --file2 /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WCadjBMI_COMBINED_EUR.txt.gz | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WCadjBMI_COMBINED_EUR.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/dbSNP/All_20150415.vcf.gz --file2 /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHR_COMBINED_EUR.txt.gz | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHR_COMBINED_EUR.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/dbSNP/All_20150415.vcf.gz --file2 /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_HIP_COMBINED_EUR.txt.gz | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_HIP_COMBINED_EUR.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/dbSNP/All_20150415.vcf.gz --file2 /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WC_COMBINED_EUR.txt.gz | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WC_COMBINED_EUR.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.txt.gz &

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/dbSNP/dbSNP_human_9606_b142_GRCh38_All_20150416.vcf.gz --file2 /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_HIPadjBMI_COMBINED_EUR.txt.gz | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_HIPadjBMI_COMBINED_EUR.wdbSNP_human_9606_b142_GRCh38.ChrBP.txt.gz

~~~
[  mturchin20@spudhead  ~/Data/dbSNP]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/All_ancestries_SNP_gwas_mc_merge_nogc.tbl.uniq.gz | wc
2555511 20444088 125133265
[  mturchin20@spudhead  ~/Data/dbSNP]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.gz | wc
2554638 20437104 125031676
[  mturchin20@bigmem02  ~/Data/dbSNP]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.txt.gz | wc
2550859 22957731 148851105
[  mturchin20@bigmem02  ~/Data/dbSNP]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.txt.gz | wc      2550859 20406872 119190473
[  mturchin20@bigmem02  ~/Data/dbSNP]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.gz | wc
2554638 22991742 154763475
[  mturchin20@bigmem02  ~/Data/dbSNP]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.gz | wc                                   2554638 20437104 125031676
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.txt.gz | wc
wc: standard input:8: Invalid or incomplete multibyte or wide character
2542440 22881925 147378256
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.txt.gz | wc
wc: standard input:8: Invalid or incomplete multibyte or wide character
2542439 20339485 118046720
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.txt.gz | grep NA | wc
8176   73584  414776
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.gz | grep NA | wc
56907  512163 3248718
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.txt.gz | awk '{ print $9 }' | sort | tail -n 508853 | grep NA | wc
  32523   32523   97569
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_HIPadjBMI_COMBINED_EUR.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.txt.gz | wc
wc: standard input:8: Invalid or incomplete multibyte or wide character
2540934 27950223 177978334
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_HIPadjBMI_COMBINED_EUR.txt.gz | wc
wc: standard input:8: Invalid or incomplete multibyte or wide character
2540933 25409289 148664320
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_HIPadjBMI_COMBINED_EUR.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.txt.gz | awk '{ print $11 }' | sort | tail -n 508853 | grep NA | wc
  32510   32510   97530
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WCadjBMI_COMBINED_EUR.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.txt.gz | wc
wc: standard input:8: Invalid or incomplete multibyte or wide character
2546082 28006851 177414406
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WCadjBMI_COMBINED_EUR.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.txt.gz | awk '{ print $11 }' | sort | tail -n 508853 | grep NA | wc
  33608   33608  100824
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.txt.gz | grep NA | grep -v rs
SNP_A-2097957   A       C       0.681   0.012   0.0033  0.00040 231091  NA
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.gz | grep NA | grep -v rs
SNP     A1      A2      Freq1.Hapmap    b       se      p       N       NA
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.txt.gz | grep NA | grep -v rs
Binary file (standard input) matches
~~~


#Realized, because /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WCadjBMI_COMBINED_EUR.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.txt.gz has ChrBP information in it, that I am mapping to the wrong reference database (dbSNP142 I guess?). Comparing the 'GWAS hit' files, which do have ChrBP info, with this WCadjBMI file to see if they are mapping to the same region, for both 2010 and 2014_5 'GWAS hits'. It looks like everything was mapped to the same build, 2010 to 2014_5, which is NCBI36/hg18 & dbSNP130. So able to download the dbSNP130 list of variants from UCSC genome browser, the commands for which are at the bottom of this chunk of labcode/results
~~~
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$cat /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt | tail -n 20
rs2890652        2       142676401
rs1555543        1       96717385
rs4771122        13      26918180
rs4836133        5       124360002
rs4929949        11      8561169
rs206936         6       34410847
rs9491696        6       127494332
rs6905288        6       43866851
rs984222         1       119305366
rs1055144        7       25837634
rs10195252       2       165221337
rs4846567        1       217817340
rs1011731        1       170613171
rs718314         12      26344550
rs1294421        6       6688148
rs1443512        12      52628951
rs6795735        3       64680405
rs4823006        22      27781671
rs6784615        3       52481466
rs6861681        5       173295064
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WCadjBMI_COMBINED_EUR.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.txt.gz | perl -lane 'if ($F[0] =~ m/rs4823006/) { print join("\t", @F); }'
rs4823006       22      27781671        A       G       0.5333  0.018   0.0034  2.2e-07 231058  22_29451671
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WCadjBMI_COMBINED_EUR.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.txt.gz | perl -lane 'if ($F[0] =~ m/rs1555543/) { print join("\t", @F); }'
rs1555543       1       96717385        A       C       0.425   -0.0026 0.0043  0.55    153380  1_96944797
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$cat /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt | head -n 10
rs425277         1       2059032
rs2284746        1       17179262
rs1738475        1       23409478
rs4601530        1       24916698
rs7532866        1       26614131
rs2154319        1       41518357
rs17391694       1       78396214
rs6699417        1       88896031
rs10874746       1       93096559
rs9428104        1       118657110
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WCadjBMI_COMBINED_EUR.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.txt.gz | perl -lane 'if ($F[0] =~ m/rs7532866/) { print join("\t", @F); }'
rs7532866       1       26614131        A       G       NA      0       0.0036  0.99    229730  1_26741544
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WCadjBMI_COMBINED_EUR.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.txt.gz | perl -lane 'if ($F[0] =~ m/rs10874746/) { print join("\t", @F); }'
rs10874746      1       93096559        T       C       0.3417  -0.0084 0.0035  0.016   231211  1_93323971
~~~

#Redoing the ChrBP annotation steps with dbSNP130 now

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/dbSNP130.txt.gz --file2 /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.txt.gz | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/dbSNP130.txt.gz --file2 /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.gz | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/dbSNP130.txt.gz --file2 /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.txt.gz | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/dbSNP130.txt.gz --file2 /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_HIPadjBMI_COMBINED_EUR.txt.gz | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_HIPadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/dbSNP130.txt.gz --file2 /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WCadjBMI_COMBINED_EUR.txt.gz | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WCadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/dbSNP130.txt.gz --file2 /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHR_COMBINED_EUR.txt.gz | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHR_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/dbSNP130.txt.gz --file2 /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_HIP_COMBINED_EUR.txt.gz | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_HIP_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/dbSNP130.txt.gz --file2 /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WC_COMBINED_EUR.txt.gz | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WC_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz &

zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep , | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | grep -v rs | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.gz | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.gz | grep , | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.gz | grep NA | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.gz | grep NA | grep -v rs | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | tail -n +20 | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | tail -n +20 | grep , | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | tail -n +20 | grep NA | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | tail -n +20 | grep NA | grep -v rs | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_HIPadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | tail -n +20 | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_HIPadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | tail -n +20 | grep , | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_HIPadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | tail -n +20 | grep NA | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_HIPadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | tail -n +20 | grep NA | grep -v rs | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WCadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | tail -n +20 | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WCadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | tail -n +20 | grep , | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WCadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | tail -n +20 | grep NA | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WCadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | tail -n +20 | grep NA | grep -v rs | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHR_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | tail -n +20 | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHR_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | tail -n +20 | grep , | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHR_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | tail -n +20 | grep NA | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHR_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | tail -n +20 | grep NA | grep -v rs | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_HIP_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | tail -n +20 | wc 
zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_HIP_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | tail -n +20 | grep , | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_HIP_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | tail -n +20 | grep NA | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_HIP_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | tail -n +20 | grep NA | grep -v rs | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WC_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | tail -n +20 | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WC_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | tail -n +20 | grep , | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WC_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | tail -n +20 | grep NA | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WC_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | tail -n +20 | grep NA | grep -v rs | wc

~~~
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | wc
2550859 22957731 149252489
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep , | wc
11574  104166 1060266
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | wc
5913   53217  306525
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | grep -v rs | wc
1       9      55
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.gz | wc
2554638 22991742 155169515
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.gz | grep , | wc
11705  105345 1094041
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.gz | grep NA | wc
54778  493002 3162501
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.gz | grep NA | grep -v rs | wc
1       9      35
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.gz | grep NA | grep -v rs
SNP     A1      A2      Freq1.Hapmap    b       se      p       N       NA
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | tail -n +20 | wc
2542421 22881781 147770052
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | tail -n +20 | grep , | wc 
11320  101880 1032095
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | tail -n +20 | grep NA | wc
54931  494371 2829544
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | tail -n +20 | grep NA | grep -v rs | wc
30223  271999 1488151
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_HIPadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | tail -n +20 | wc 
2540915 27950055 178369746
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_HIPadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | tail -n +20 | grep , | wc
11308  124388 1161471
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_HIPadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | tail -n +20 | grep NA | wc 
54712  601822 3463146
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_HIPadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | tail -n +20 | grep NA | grep -v rs | wc
30211  332311 1847730
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WCadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | tail -n +20 | wc
2546063 28006683 177806542
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WCadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | tail -n +20 | grep , | wc
11329  124619 1157287
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WCadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | tail -n +20 | grep NA | wc 
56657  623217 3581917
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WCadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | tail -n +20 | grep NA | grep -v rs | wc
31299  344279 1909638
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHR_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | tail -n +20 | wc
2560771 28168471 179038050
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHR_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | tail -n +20 | grep , | wc
11561  127171 1180926
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHR_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | tail -n +20 | grep NA | wc
75202  827212 4807405
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHR_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | tail -n +20 | grep NA | grep -v rs | wc
30242  332652 1852424
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_HIP_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | tail -n +20 | wc
2559728 23037544 149736440
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_HIP_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | tail -n +20 | grep , | wc
11452  103068 1048156
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_HIP_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | tail -n +20 | grep NA | wc
73932  665380 3853906
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_HIP_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | tail -n +20 | grep NA | grep -v rs | wc
30130  271162 1487498
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WC_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | tail -n +20 | wc
2565397 23088565 149759722
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WC_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | tail -n +20 | grep , | wc
11461  103149 1045870
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WC_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | tail -n +20 | grep NA | wc
76732  690580 3999491
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WC_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | tail -n +20 | grep NA | grep -v rs | wc
31579  284203 1559698
~~~

zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHR_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.txt.gz | perl -lane 'splice(@F, 1, 2); print join("\t", @F);' | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHR_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.ColRightFormat.txt.gz
zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_HIPadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.txt.gz | perl -lane 'splice(@F, 1, 2); print join("\t", @F);' | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_HIPadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.ColRightFormat.txt.gz
zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WCadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.txt.gz | perl -lane 'splice(@F, 1, 2); print join("\t", @F);' | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WCadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.ColRightFormat.txt.gz

zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep -E 'rs6694089|rs4652773|rs2298265|rs1368380' | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.4SNPs.txt.gz

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/DataOrganization.GIANT2014_5.PythonVLookUp.CreateFullAnnotFile.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt --file2 /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.4SNPs.txt.gz,/mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.4SNPs.txt.gz

~~~
[  mturchin20@bigmem02  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1]$python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/DataOrganization.GIANT2014_5.PythonVLookUp.CreateFullAnnotFile.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt --file2 /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.4SNPs.txt.gz,/mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.4SNPs.txt.gz
/mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.4SNPs.txt.gz,/mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.4SNPs.txt.gz
rs2298265       T       C       1_149525666     0.117   1       0.117   -0.030  0.0046  8.3e-11 0.117   -0.030  0.0046  8.3e-11
rs6694089       A       G       1_170350503     0.217   1       0.217   0.039   0.0032  4.2e-33 0.217   0.039   0.0032  4.2e-33
rs4652773       A       G       1_181321449     0.474   1       0.474   0.024   0.0029  1.1e-15 0.474   0.024   0.0029  1.1e-15
rs1368380       T       C       5_171218236     0.425   1       0.425   0.032   0.0031  6.1e-25 0.425   0.032   0.0031  6.1e-25
~~~

zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | head -n 1000 | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.top1000.gz
zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.gz | head -n 1000 | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.top1000.gz

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/DataOrganization.GIANT2014_5.PythonVLookUp.CreateFullAnnotFile.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt --file2 /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.top1000.gz,/mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.top1000.gz

~~~
[  mturchin20@bigmem02  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1]$python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/DataOrganization.GIANT2014_5.PythonVLookUp.CreateFullAnnotFile.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt --file2 /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.top1000.gz,/mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.top1000.gz | head -n 25
/mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.top1000.gz,/mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.top1000.gz
rs10002873      T       C       4_60322082      0.075   0       NA      NA      NA      NA      0.075   -0.0068 0.0069  0.3244
rs10883118      A       G       10_100283886    0.289   0       0.289   -0.0028 0.0032  0.39    NA      NA      NA      NA
rs10002758      G       T       4_41478453      0.2417  0       NA      NA      NA      NA      0.2417  -0.0079 0.0046  0.08591
rs10002976      A       G       4_180507195     0.1333  0       NA      NA      NA      NA      0.1333  5e-04   0.0051  0.9219
rs12268065      A       G       10_10035354     0.02    0       0.02    -0.00030        0.0059  0.96    NA      NA      NA      NA
rs9988679       A       G       10_100604778    0.397   0       0.397   0.0063  0.0031  0.044   NA      NA      NA      NA
rs7924209       T       C       10_100145916    0.288   0       0.288   0.0060  0.0032  0.059   NA      NA      NA      NA
rs10002199      T       G       4_20417561      0.35    0       NA      NA      NA      NA      0.35    0.008   0.0037  0.03061
rs11599208      T       C       10_100182361    0.195   0       0.195   -0.00050        0.0040  0.91    NA      NA      NA      NA
rs12569898      A       G       10_10048290     0.5     0       0.5     0.0023  0.0030  0.44    NA      NA      NA      NA
rs10002195      G       A       4_125014932     0.2417  0       NA      NA      NA      NA      0.2417  0.0024  0.0036  0.5054
rs10000017      C       T       4_84997148      0.2333  0       NA      NA      NA      NA      0.2333  -0.0034 0.0046  0.4598
rs10000545      G       C       4_110577326     0.1583  0       NA      NA      NA      NA      0.1583  -0.0023 0.0042  0.5871
rs10001548      C       T       4_166098830     0.3417  0       NA      NA      NA      NA      0.3417  -4e-04  0.0041  0.9223
rs17110576      A       G       10_100399966    0.092   0       0.092   0.0031  0.0048  0.52    NA      NA      NA      NA
rs9664571       T       C       10_10063767     0.033   0       0.033   0.010   0.0097  0.28    NA      NA      NA      NA
rs10002751      G       C       4_10652353      0.275   0       NA      NA      NA      NA      0.275   -8e-04  0.0043  0.8524
rs10002370      G       A       4_46893928      0.1167  0       NA      NA      NA      NA      0.1167  0.006   0.0053  0.2576
rs7097807       A       G       10_100081571    0.207   0       0.207   0.0050  0.0037  0.17    NA      NA      NA      NA
rs10001545      C       A       4_88395053      0.2583  0       NA      NA      NA      NA      0.2583  -0.0083 0.0044  0.05925
rs12766763      A       T       10_10068697     0.025   0       0.025   -0.0057 0.0081  0.48    NA      NA      NA      NA
rs11815392      A       C       10_10061543     0.033   0       0.033   0.0094  0.0098  0.34    NA      NA      NA      NA
rs7907555       T       C       10_100145799    0.3     0       0.3     -0.0061 0.0032  0.053   NA      NA      NA      NA
rs11815889      A       C       10_100313990    0.25    0       0.25    -0.0018 0.0032  0.57    NA      NA      NA      NA
rs7902229       A       G       10_10004243     0.458   0       0.458   -0.0028 0.0037  0.44    NA      NA      NA      NA
~~~

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/DataOrganization.GIANT2014_5.PythonVLookUp.CreateFullAnnotFile.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt --file2 /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz,/mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.gz,/mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.txt.gz | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5_Orig3_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz

zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.gz /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.txt.gz | awk '{ print $1 }' | sort | uniq -c | awk '{ if ($1 >= 2) { print $0 } } ' | vi -

#Memory issue with so many GWAS hit SNPs (50g interactive session not enough -- currentl trying with a 70g session); so going to try and create a 'vs2' file of /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/DataOrganization.GIANT2014_5.PythonVLookUp.CreateFullAnnotFile.vs1.py that approaches this differently
#Actually it looks like using ql 70g is enough to get past this issue, so just kept using /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/DataOrganization.GIANT2014_5.PythonVLookUp.CreateFullAnnotFile.vs1.py
#cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/DataOrganization.GIANT2014_5.PythonVLookUp.CreateFullAnnotFile.vs1.py /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/DataOrganization.GIANT2014_5.PythonVLookUp.CreateFullAnnotFile.vs2.py
#mv /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/DataOrganization.GIANT2014_5.PythonVLookUp.CreateFullAnnotFile.vs2.py /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/DataOrganization.GIANT2014_5.PythonVLookUp.CreateFullAnnotFile.vs2.py.20150527.DidntEndUpUsing

zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5_Orig3_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | wc
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5_Orig3_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | perl -lane 'print $#F;' | sort | uniq -c
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5_Orig3_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | awk '{ print $6 }' | sort | uniq -c 
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5_Orig3_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | awk '{ print $4 }' | sort | uniq -c | perl -lane 'if ($F[1] !~ m/\d+/) { print join("\t", @F); }'
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5_Orig3_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | awk '{ print $5 }' | sort | uniq -c | perl -lane 'if ($F[1] !~ m/\d+/) { print join("\t", @F); }'

~~~
[  mturchin20@bigmem02  ~]$Error2a -- line2[0] present more than once in a particular file (1: rs13598,C,A,0.1583,-0.003,0.0096,0.7547,69744.9,15_46489749)
Error2a -- line2[0] present more than once in a particular file (1: rs2160421,T,A,0.5583,-0.001,0.0037,0.787,233968,15_34746878)
Error2a -- line2[0] present more than once in a particular file (1: rs372268,G,A,0.975,-0.0113,0.0214,0.5975,150660,4_139504797)
Error2a -- line2[0] present more than once in a particular file (1: rs3752976,G,A,0.05172,0.0071,0.0089,0.425,224029,1_199381979)
Error2a -- line2[0] present more than once in a particular file (1: rs3780697,T,C,0.725,0.0035,0.0048,0.4659,233977,9_131618930)
Error2a -- line2[0] present more than once in a particular file (1: rs3819262,T,G,0.7167,0.0051,0.004,0.2023,233953,21_37382759)
Error2a -- line2[0] present more than once in a particular file (1: rs573071,A,G,0.6833,-0.0042,0.004,0.2937,232167,9_246993)
Error2a -- line2[0] present more than once in a particular file (1: rs5820007,C,T,0.1271,-0.0029,0.0061,0.6345,216740,17_28342660)
Error2a -- line2[0] present more than once in a particular file (1: rs5820007,T,C,0.875,0.0029,0.0061,0.6345,216740,17_28342660)
Error2a -- line2[0] present more than once in a particular file (1: rs627141,T,C,0.2833,0,0.0051,1,229425,18_8332476)
Error2a -- line2[0] present more than once in a particular file (1: rs7591686,A,G,0.7333,0.001,0.0043,0.8161,233849,2_45609171)
Error2a -- line2[0] present more than once in a particular file (1: rs969206,G,A,0.3917,-0.005,0.0037,0.1766,234003,9_74502957)
[  mturchin20@spudhead  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $1 }' | sort | uniq -c | awk '{ if ($1 >= 2) { print $0 } } '
[  mturchin20@spudhead  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.gz | awk '{ print $1 }' | sort | uniq -c | awk '{ if ($1 >= 2) { print $0 } } '
2 rs11576885
2 rs11771665
2 rs13598
2 rs2160421
2 rs372268
2 rs3752976
2 rs3780697
2 rs3819262
2 rs573071
3 rs5820007
2 rs627141
2 rs7591686
2 rs969206
[  mturchin20@spudhead  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.txt.gz | awk '{ print $1 }' | sort | uniq -c | awk '{ if ($1 >= 2) { print $0 } } '
[  mturchin20@spudhead  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.gz | grep -w -E 'rs11576885|rs11771665|rs13598|rs2160421|rs372268|rs3752976|rs3780697|rs3819262|rs573071|rs5820007|rs627141|rs7591686|rs969206'
rs11576885      C       G       0.4083  0.0014  0.0042  0.7389  219514  1_204737275
rs11576885      G       C       0.4083  -0.0014 0.0042  0.7389  219514  1_204737275
rs11771665      G       A       0.15    -0.0043 0.0056  0.4426  222149  7_86704151
rs11771665      G       A       0.1525  -0.0043 0.0056  0.4426  222149  7_86704151
rs13598 A       C       0.8417  0.003   0.0096  0.7547  69744.9 15_46489749
rs13598 C       A       0.1583  -0.003  0.0096  0.7547  69744.9 15_46489749
rs2160421       A       T       0.5583  0.001   0.0037  0.787   233968  15_34746878
rs2160421       T       A       0.5583  -0.001  0.0037  0.787   233968  15_34746878
rs372268        A       G       0.975   0.0113  0.0214  0.5975  150660  4_139504797
rs372268        G       A       0.975   -0.0113 0.0214  0.5975  150660  4_139504797
rs3752976       A       G       0.05    -0.0071 0.0089  0.425   224029  1_199381979
rs3752976       G       A       0.05172 0.0071  0.0089  0.425   224029  1_199381979
rs3780697       C       T       0.1833  -0.0035 0.0048  0.4659  233977  9_131618930
rs3780697       T       C       0.725   0.0035  0.0048  0.4659  233977  9_131618930
rs3819262       T       G       0.6917  0.0051  0.004   0.2023  233953  21_37382759
rs3819262       T       G       0.7167  0.0051  0.004   0.2023  233953  21_37382759
rs573071        A       G       0.675   -0.0042 0.004   0.2937  232167  9_246993
rs573071        A       G       0.6833  -0.0042 0.004   0.2937  232167  9_246993
rs5820007       C       T       0.125   -0.0029 0.0061  0.6345  216740  17_28342660
rs5820007       C       T       0.1271  -0.0029 0.0061  0.6345  216740  17_28342660
rs5820007       T       C       0.875   0.0029  0.0061  0.6345  216740  17_28342660
rs627141        T       C       0.1833  0       0.0051  1       229425  18_8332476
rs627141        T       C       0.2833  0       0.0051  1       229425  18_8332476
rs7591686       A       G       0.7167  0.001   0.0043  0.8161  233849  2_45609171
rs7591686       A       G       0.7333  0.001   0.0043  0.8161  233849  2_45609171
rs969206        A       G       0.3917  0.005   0.0037  0.1766  234003  9_74502957
rs969206        G       A       0.3917  -0.005  0.0037  0.1766  234003  9_74502957
[  mturchin20@bigmem02  ~]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5_Orig3_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | wc
2589848 46617264 297715972
[  mturchin20@bigmem02  ~]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5_Orig3_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | perl -lane 'print $#F;' | sort | uniq -c
2589848 17
[  mturchin20@bigmem02  ~]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5_Orig3_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | awk '{ print $6 }' | sort | uniq -c
2017735 0
    861 1
 571252 2
[  mturchin20@bigmem02  ~]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5_Orig3_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | awk '{ print $4 }' | sort | uniq -c | perl -lane 'if ($F[1] !~ m/\d+/) { print join("\t", @F); }'
3448    NA
3       None
[  mturchin20@bigmem02  ~]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5_Orig3_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | awk '{ if ($4 == "None") { print $0 } } '
rs34102154      A       G       None    NA      0       NA      NA      NA      NA      NA      NA      NA      NA      NA      -0.0035 0.0084  51224
rs1053838       A       C       None    0.009   0       0.009   -0.013  0.039   101301  0.008621        0.0295  0.0502  97215.8 0.008621        0.044   0.059   57703
rs1632856       T       C       None    0.283   0       0.283   -0.0073 0.0032  251823  0.2833  0.0037  0.0041  232620  0.2833  0.0031  0.0047  141363
[  mturchin20@bigmem02  ~]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5_Orig3_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | awk '{ print $5 }' | sort | uniq -c | perl -lane 'if ($F[1] !~ m/\d+/) { print join("\t", @F); }'
38285   NA
~~~

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/DataOrganization.GIANT2010.PythonVLookUp.PutTogetherProcessAndAnnotFiles.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5.Orig3.dtlesssignif.vs1.txt --file2 <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5_Orig3_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | perl -lane 'my $chr; my $pos; if (($F[3] eq "NA") || ($F[3] eq "None")) { $chr = "NA"; $pos = "NA"; } else { $chr = ((split(/_/, $F[3]))[0]); $pos = ((split(/_/, $F[3]))[1]); } splice(@F, 1, 0, ($chr, $pos)); splice(@F, 8, 1); splice(@F, 11, 1); splice(@F, 14, 1); print join("\t", @F);') | \
gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5.Orig3.dtlesssignif.vs1.annot.vs1.txt.gz
#Added the following header to /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5.Orig3.dtlesssignif.vs1.annot.vs1.txt.gz
#snp chr pos a1 a2 chrbp maf annot beta_height se_height n_height beta_BMI se_BMI n_BMI beta_WHRadjBMI se_WHRadjBMI n_WHRadjBMI Z.height Z.BMI Z.WHRadjBMI mvstat mvp unip

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5.Orig3.dtlesssignif.vs1.txt | wc
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5.Orig3.dtlesssignif.vs1.annot.vs1.txt.gz | wc
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5.Orig3.dtlesssignif.vs1.annot.vs1.txt.gz | perl -lane 'if ($F[$#F-1] > $F[$#F]) { print join("\t", @F) ; } ' | wc

~~~
[  mturchin20@spudhead  ~]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/multivariate/globallipids/dtlesssignif.annot.txt | wc
8066  209716 1794390
[  mturchin20@spudhead  ~]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/multivariate/globallipids/dtlesssignif.annot.txt | perl -lane 'if ($F[$#F-1] > $F[$#F]) { print join("\t", @F) ; } ' | wc
3724   96824  830117
[  mturchin20@spudhead  ~]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/dtlesssignif.annot.vs2.txt.gz | wc
12463  348964 3301334
[  mturchin20@spudhead  ~]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/dtlesssignif.annot.vs2.txt.gz | perl -lane 'if ($F[$#F-1] > $F[$#F]) { print join("\t", @F) ; } ' | wc
8859  248052 2346362
[  mturchin20@bigmem02  ~]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT2010.dtlesssignif.vs1.annot.vs1.txt.gz | wc
8842  203366 1641132
[  mturchin20@bigmem02  ~]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT2010.dtlesssignif.vs1.annot.vs1.txt.gz | perl -lane 'if ($F[$#F-1] > $F[$#F]) { print join("\t", @F) ; } ' | wc
1097   25231  202912
[  mturchin20@bigmem02  ~]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5.Orig3.dtlesssignif.vs1.txt | wc
42023   42023 4708300
[  mturchin20@bigmem02  ~]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5.Orig3.dtlesssignif.vs1.annot.vs1.txt.gz | wc
42023  966529 8803594
[  mturchin20@bigmem02  ~]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5.Orig3.dtlesssignif.vs1.annot.vs1.txt.gz | perl -lane 'if ($F[$#F-1] > $F[$#F]) { print join("\t", @F) ; } ' | wc
5677  130571 1185565
~~~       

#Copy/pasted commands from /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GLC.MTedits.ForGIANT2014_5.Orig3.vs1.R into an inteactve session in #PWD /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/ 

mv /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/.RData /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/RData.GIANT2014_5.Orig3.GLC.20150528

gzip /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/RData.GIANT2014_5.Orig3.GLC.20150528



python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/DataOrganization.GIANT2014_5.PythonVLookUp.CreateFullAnnotFile.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.AllPheno.MarkerChrBP.txt --file2 /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz,/mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.gz,/mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.txt.gz,/mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHR_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.ColRightFormat.txt.gz,/mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_HIPadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.ColRightFormat.txt.gz,/mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_HIP_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.txt.gz,/mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WCadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.ColRightFormat.txt.gz,/mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WC_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.txt.gz | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5_AllPheno_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz 

zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5_AllPheno_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | wc
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5_AllPheno_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | perl -lane 'print $#F;' | sort | uniq -c
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5_AllPheno_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | awk '{ print $4 }' | sort | uniq -c | perl -lane 'if ($F[1] !~ m/\d+/) { print join("\t", @F); }'
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5_AllPheno_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | awk '{ if ($4 == "None") { print $0 } } '
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5_AllPheno_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | awk '{ print $5 }' | sort | uniq -c | perl -lane 'if ($F[1] !~ m/\d+/) { print join("\t", @F); }'
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5_AllPheno_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | awk '{ print $6 }' | sort | uniq -c 

~~~
[  mturchin20@bigmem02  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5_AllPheno_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | wc
2591354 98471452 649782849
[  mturchin20@bigmem02  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5_AllPheno_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | perl -lane 'print $#F;' | sort | uniq -c
2591354 37
[  mturchin20@bigmem02  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5_AllPheno_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | awk '{ print $4 }' | sort | uniq -c | perl -lane 'if ($F[1] !~ m/\d+/) { print join("\t", @F); }'
3454    NA
4       None
[  mturchin20@bigmem02  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5_AllPheno_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | awk '{ if ($4 == "None") { print $0 } } '
rs34102154      A       G       None    NA      0       NA      NA      NA      NA      NA      NA      NA      NA      NA      -0.0035 0.0084  51224   NA      -0.0036 0.0079  51473   NA      -0.003  0.0088  51197   NA      NA      NA      NA      NA      -0.0004 0.0081  61707   NA      NA      NA      NA
rs1053838       A       C       None    0.009   0       0.009   -0.013  0.039   101301  0.008621        0.0295  0.0502  97215.8 0.008621        0.044   0.059   57703   0.008621        0.035   0.063   50566   0.008621        -0.065  0.062   58456   0.008621        -0.036  0.065   52190   0.0086  0.01    0.058   63735       0.008621        0.042   0.064   54959
rs1632856       T       C       None    0.283   0       0.283   -0.0073 0.0032  251823  0.2833  0.0037  0.0041  232620  0.2833  0.0031  0.0047  141363  0.2833  0.0054  0.0046  143200  0.2833  0.0036  0.0049  142955  0.2833  0.0051  0.0048  144058  0.2833  0.0098  0.0046  152083  0.2833  0.0078  0.0047  152552
rs6603251       T       C       None    NA      0       NA      NA      NA      NA      NA      NA      NA      NA      NA      NA      NA      NA      NA      NA      NA      NA      NA      NA      NA      NA      NA      NA      NA      NA      NA      0.0083  0.0068  56955   NA      NA      NA      NA
[  mturchin20@bigmem02  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5_AllPheno_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | awk '{ print $5 }' | sort | uniq -c | perl -lane 'if ($F[1] !~ m/\d+/) { print join("\t", @F); }'
39789   NA
[  mturchin20@bigmem02  ~]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5_AllPheno_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | awk '{ print $6 }' | sort | uniq -c
2003823 0
    880 1
 586651 2
~~~

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/DataOrganization.GIANT2010.PythonVLookUp.PutTogetherProcessAndAnnotFiles.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5.AllPheno.dtlesssignif.vs1.txt --file2 <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5_AllPheno_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | perl -lane 'my $chr; my $pos; if (($F[3] eq "NA") || ($F[3] eq "None")) { $chr = "NA"; $pos = "NA"; } else { $chr = ((split(/_/, $F[3]))[0]); $pos = ((split(/_/, $F[3]))[1]); } splice(@F, 1, 0, ($chr, $pos)); splice(@F, 8, 1); splice(@F, 11, 1); splice(@F, 14, 1); splice(@F, 17, 1); splice(@F, 20, 1); splice(@F, 23, 1); splice(@F, 27, 1); splice(@F, 30, 1); print join("\t", @F);') | \
gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5.AllPheno.dtlesssignif.vs1.annot.vs1.txt.gz
#Added the following header to /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5.AllPheno.dtlesssignif.vs1.annot.vs1.txt.gz
#snp chr pos a1 a2 chrbp maf annot beta_height se_height n_height beta_BMI se_BMI n_BMI beta_WHRadjBMI se_WHRadjBMI n_WHRadjBMI  beta_WHR se_WHR n_WHR beta_HIPadjBMI se_HIPadjBMI n_HIPadjBMI beta_HIP se_HIP n_HIP beta_WCadjBMI se_WCadjBMI n_WCadjBMI beta_WC se_WC n_WC Z.height Z.BMI Z.WHRadjBMI Z.WHR Z.HIPadjBMI Z.HIP Z.WCadjBMI Z.WC mvstat mvp unip

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5.AllPheno.dtlesssignif.vs1.txt | wc
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5.AllPheno.dtlesssignif.vs1.annot.vs1.txt.gz | wc
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5.AllPheno.dtlesssignif.vs1.annot.vs1.txt.gz | perl -lane 'if ($F[$#F-1] > $F[$#F]) { print join("\t", @F) ; } ' | wc
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5.AllPheno.dtlesssignif.vs1.annot.vs1.txt.gz | perl -F, -lane 'print $#F;' | sort | uniq -c

~~~
[  mturchin20@spudhead  ~/Data/GIANT/2014_5]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5_AllPheno_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | perl -lane 'my $chr; my $pos; if (($F[3] eq "NA") || ($F[3] eq "None")) { $chr = "NA"; $pos = "NA"; } else { $chr = ((split(/_/, $F[3]))[0]); $pos = ((split(/_/, $F[3]))[1]); } splice(@F, 1, 0, ($chr, $pos)); splice(@F, 8, 1); splice(@F, 11, 1); splice(@F, 14, 1); splice(@F, 17, 1); splice(@F, 20, 1); splice(@F, 23, 1); splice(@F, 27, 1); splice(@F, 30, 1); print join("\t", @F);' | head -n 10
rs999721        4       85416326        A       G       4_85416326      0.466   0       0.0097  0.0030  241371  -0.0039 0.0038  222236  0.0034  0.0044  130986  0.0006  0.0043  132832  0.0046  0.0046  132042  -0.0001 0.0046  133691  0.4667  0.0044  140848  0.4667  0.0044  142183
rs942176        9       8420741 A       G       9_8420741       0.433   0       0.0040  0.0030  253133  0.0029  0.0038  233938  6e-04   0.0043  142727  0.0031  0.0042  144568  0.0033  0.0045  143781  0.0064  0.0045  145424  0.4333  0.0043  151195  0.4333  0.0044  153921
rs9821657       3       186113796       A       G       3_186113796     0.175   0       0.0061  0.0035  252892  4e-04   0.0046  233804  0       0.005   142556  -0.0034 0.0051  144392  0.0031  0.0053  143637  0.0017  0.0053  145195  0.175   0.005   153275  0.175   0.0052  153744
rs1919329       4       165827169       T       G       4_165827169     0.483   0       0.0078  0.0030  246965  -2e-04  0.0038  230282  0.0022  0.0043  140365  0.002   0.0042  142384  0.005   0.0045  141418  0.0024  0.0045  143242  0.4833  0.0043  151084  0.4833  0.0044  151736
rs3957240       6       37262200        T       C       6_37262200      0.05    0       -0.024  0.011   215562  -0.0106 0.014   202410  -0.011  0.016   115779  -0.027  0.016   117279  -0.0046 0.017   116818  -0.021  0.017   118121  0.05    0.016   126473  0.05    0.016   126597
rs1919324       4       165805562       T       C       4_165805562     0.142   0       -0.0016 0.0039  253148  0.0028  0.0051  233978  0.0085  0.0057  142610  0.0089  0.0056  144588  -0.0083 0.0061  143664  -0.0025 0.006   145448  0.1417  0.0057  153330  0.1417  0.0058  153940
rs8030214       15      49019617        T       C       15_49019617     0.336   2       -0.015  0.0030  253230  2e-04   0.0038  234018  9e-04   0.0043  142752  0.0001  0.0043  144595  -0.0098 0.0045  143805  -0.0078 0.0045  145454  0.3333  0.0043  153471  0.3333  0.0044  153947
rs17597444      13      42193631        T       G       13_42193631     0.237   0       0.0067  0.0033  253153  0.0041  0.0042  234007  0.0097  0.0047  142518  0.0075  0.0046  144592  -0.012  0.0049  143571  -0.0039 0.0049  145451  0.25    0.0047  153237  0.25    0.0047  153944
rs2664913       10      8169772 A       G       10_8169772      0.425   0       -0.0016 0.0031  240207  -0.004  0.004   221984  0.0044  0.0045  130431  0.0007  0.0044  132745  -0.0015 0.0047  131498  -0.0025 0.0047  133604  0.425   0.0045  141150  0.425   0.0045  142096
rs16876997      5       6246251 A       G       5_6246251       0.017   0       0.0065  0.011   238862  -0.0022 0.0138  220635  -0.016  0.015   130612  -0.024  0.015   132437  -0.008  0.016   131665  -0.01   0.016   133289  0.0167  0.015   141330  0.0167  0.016   141788
[  mturchin20@bigmem02  ~]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5.Orig3.dtlesssignif.vs1.txt | wc
42023   42023 4708300
[  mturchin20@bigmem02  ~]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5.Orig3.dtlesssignif.vs1.annot.vs1.txt.gz | wc
42023  966529 8803594
[  mturchin20@bigmem02  ~]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5.Orig3.dtlesssignif.vs1.annot.vs1.txt.gz | perl -lane 'if ($F[$#F-1] > $F[$#F]) { print join("\t", @F) ; } ' | wc
5677  130571 1185565
[  mturchin20@bigmem02  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5.AllPheno.dtlesssignif.vs1.txt | wc
56323   56323 11147251
[  mturchin20@bigmem02  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5.AllPheno.dtlesssignif.vs1.annot.vs1.txt.gz | wc
56323 2421889 22448471
[  mturchin20@bigmem02  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5.AllPheno.dtlesssignif.vs1.annot.vs1.txt.gz | perl -lane 'if ($F[$#F-1] > $F[$#F]) { print join("\t", @F) ; } ' | wc
41783 1796669 16626655
[  mturchin20@bigmem02  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5.AllPheno.dtlesssignif.vs1.txt | perl -F, -lane 'print $#F;' | sort | uniq -c
56323 11
~~~

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GLC.MTedits.ForGIANT2014_5.Orig3.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GLC.MTedits.ForGIANT2014_5.AllPheno.vs1.R

cd /mnt/lustre/home/mturchin20/Software
wget http://cran.r-project.org/src/contrib/bigmemory_4.4.6.tar.gz 
wget http://cran.r-project.org/src/contrib/bigmemory.sri_0.1.3.tar.gz
wget http://cran.r-project.org/src/contrib/BH_1.58.0-1.tar.gz
wget http://cran.r-project.org/src/contrib/pryr_0.1.1.tar.gz
wget http://cran.r-project.org/src/base/R-3/R-3.1.2.tar.gz
wget http://cran.r-project.org/src/contrib/stringi_0.4-1.tar.gz
wget http://cran.r-project.org/src/contrib/magrittr_1.5.tar.gz
wget http://cran.r-project.org/src/contrib/stringr_1.0.0.tar.gz
wget http://cran.r-project.org/src/contrib/Rcpp_0.11.6.tar.gz
wget http://cran.r-project.org/src/contrib/plyr_1.8.2.tar.gz
wget http://cran.r-project.org/src/contrib/reshape2_1.4.1.tar.gz
wget http://cran.r-project.org/src/contrib/chron_2.3-45.tar.gz
wget http://cran.r-project.org/src/contrib/data.table_1.9.4.tar.gz
wget http://cran.r-project.org/src/contrib/assertthat_0.1.tar.gz
wget http://cran.r-project.org/src/contrib/R6_2.0.1.tar.gz  
wget http://cran.r-project.org/src/contrib/lazyeval_0.1.10.tar.gz
wget http://cran.r-project.org/src/contrib/DBI_0.3.1.tar.gz
wget http://cran.r-project.org/src/contrib/dplyr_0.4.1.tar.gz
tar -xvzf /mnt/lustre/home/mturchin20/Software/R-3.1.2.tar.gz
./configure --prefix=/mnt/lustre/home/mturchin20/lib --with-x

##Having memory issues running as large a matrix as I needed to in /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GLC.MTedits.ForGIANT2014_5.AllPheno.vs1.R 
##Particularly going from:
~~~
#lbf=compute.allBFs.fromZscores(Z,VYY,gl$nmin,gl$maf,sigmaa)
#lbf.bigmat=do.call(rbind,lbf$lbf) #lbf$lbf is a list of matrices, with one element (a matrix) for each sigma; this stacks these matrices together into a big matrix with nsnp columns, and nsigma*nmodels rows
~~~
#So going to try and use the 'more significant' list of SNPs for this phenotype to cut down on computation

##20150601 -- Realized I can break up the matrix into groups of 10000 columns to overcome the memory issue. So have 7 lbf.bigmat#s and then can rbind all 7 together with 1:10000, 10001:20000, etc.. etc.. 6 times (up until 56301)
##Also I do the GWAS prior hit steps before the lbf.bigmat creation process too now
##As far as I can tell the compute.allBFs.fromZscores(...) steps and all related processes just need every model present per snp but doesn't need any informatino across SNPs to work properl; so breaking things up on a per SNP basis should be fine as long as I'm keeping all the model information for a given SNP (and maintaing the SNP order so that the results are consistent)

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GLC.MTedits.ForGIANT2014_5.AllPheno.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GLC.MTedits.ForGIANT2014_5.AllPheno.vs2.R

#python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/DataOrganization.GIANT2010.PythonVLookUp.PutTogetherProcessAndAnnotFiles.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5.AllPheno.dtsignif.vs1.txt --file2 <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5_AllPheno_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | perl -lane 'my $chr; my $pos; if (($F[3] eq "NA") || ($F[3] eq "None")) { $chr = "NA"; $pos = "NA"; } else { $chr = ((split(/_/, $F[3]))[0]); $pos = ((split(/_/, $F[3]))[1]); } splice(@F, 1, 0, ($chr, $pos)); splice(@F, 8, 1); splice(@F, 11, 1); splice(@F, 14, 1); splice(@F, 17, 1); splice(@F, 20, 1); splice(@F, 23, 1); splice(@F, 27, 1); splice(@F, 30, 1); print join("\t", @F);') | \
#gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5.AllPheno.dtsignif.vs1.annot.vs1.txt.gz
##Added the following header to /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5.AllPheno.dtsignif.vs1.annot.vs1.txt.gz
##snp chr pos a1 a2 chrbp maf annot beta_height se_height n_height beta_BMI se_BMI n_BMI beta_WHRadjBMI se_WHRadjBMI n_WHRadjBMI  beta_WHR se_WHR n_WHR beta_HIPadjBMI se_HIPadjBMI n_HIPadjBMI beta_HIP se_HIP n_HIP beta_WCadjBMI se_WCadjBMI n_WCadjBMI beta_WC se_WC n_WC Z.height Z.BMI Z.WHRadjBMI Z.WHR Z.HIPadjBMI Z.HIP Z.WCadjBMI Z.WC mvstat mvp unip

#~~~
#[  mturchin20@spudhead  ~/Software]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5.AllPheno.dtsignif.vs1.txt | wc
#  38978   38978 7708677
#~~~
#
#cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GLC.MTedits.ForGIANT2014_5.AllPheno.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GLC.MTedits.ForGIANT2014_5.AllPheno.dtsignif.vs1.R 

#Copy/pasted commands from /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GLC.MTedits.ForGIANT2014_5.AllPheno.vs2.R into an inteactve session in #PWD /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/

##20150601 -- .RData file became >10gb so decided to just remove it
#mv /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/.RData /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/RData.GIANT2014_5.AllPheno.GLC.MTedits.vs2.20150601
#gzip /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/RData.GIANT2014_5.AllPheno.GLC.MTedits.vs2.20150601

#CHECK_0: Print out allele information for all versions of 'newtophits' in previous runs (GlobalLipids, GIANT2010, 2014_5_Orig3)
#CHECK_0: Think about ways to explore the diffuse qualities of posterior 'bestclass' classifications, e.g. how many of the top hits don't have a single model with evidence above .75 or .5

cp -p GIANT2014_5/Vs1/GLC.MTedits.ForGIANT2014_5.Orig3.vs1.SignCrrct.vs1.R GIANT2014_5/Vs1/GLC.MTedits.ForGIANT2014_5.Orig3.vs1.SignCrrct.vs1.ForCSHLPoster.R






#2013
#Conducting multivariate analyses

#CHECK_0: Assuming for the moment that these analyses/studies were mapped to the same build that 2010 & 2014_5 were (should really be the case)
#But may want to double, double check this at a later point when there's a bit more time

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/dbSNP130.txt.gz --file2 /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HEIGHT_MEN_N.txt | gzip > /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HEIGHT_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz 
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/dbSNP130.txt.gz --file2 /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HEIGHT_WOMEN_N.txt | gzip > /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HEIGHT_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz 
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/dbSNP130.txt.gz --file2 /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_BMI_MEN_N.txt | gzip > /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_BMI_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz 
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/dbSNP130.txt.gz --file2 /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_BMI_WOMEN_N.txt | gzip > /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_BMI_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz 
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/dbSNP130.txt.gz --file2 /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WHRadjBMI_MEN_N.txt | gzip > /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WHRadjBMI_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz 
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/dbSNP130.txt.gz --file2 /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WHRadjBMI_WOMEN_N.txt | gzip > /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WHRadjBMI_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz 
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/dbSNP130.txt.gz --file2 /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WHR_MEN_N.txt | gzip > /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WHR_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz 
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/dbSNP130.txt.gz --file2 /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WHR_WOMEN_N.txt | gzip > /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WHR_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz 
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/dbSNP130.txt.gz --file2 /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HIPadjBMI_MEN_N.txt | gzip > /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HIPadjBMI_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz 
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/dbSNP130.txt.gz --file2 /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HIPadjBMI_WOMEN_N.txt | gzip > /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HIPadjBMI_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz 
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/dbSNP130.txt.gz --file2 /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HIP_MEN_N.txt | gzip > /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HIP_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz 
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/dbSNP130.txt.gz --file2 /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HIP_WOMEN_N.txt | gzip > /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HIP_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz 
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/dbSNP130.txt.gz --file2 /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WCadjBMI_MEN_N.txt | gzip > /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WCadjBMI_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz 
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/dbSNP130.txt.gz --file2 /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WCadjBMI_WOMEN_N.txt | gzip > /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WCadjBMI_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz 
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/dbSNP130.txt.gz --file2 /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WC_MEN_N.txt | gzip > /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WC_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz 
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/dbSNP130.txt.gz --file2 /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WC_WOMEN_N.txt | gzip > /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WC_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz 
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/dbSNP130.txt.gz --file2 /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WEIGHT_MEN_N.txt | gzip > /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WEIGHT_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz 
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/dbSNP130.txt.gz --file2 /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WEIGHT_WOMEN_N.txt | gzip > /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WEIGHT_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz 

zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_*_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | perl -lane 'print $#F;' | sort | uniq -c

zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HEIGHT_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | wc 
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HEIGHT_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep , | wc 
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HEIGHT_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HEIGHT_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | grep -v rs | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HEIGHT_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | wc 
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HEIGHT_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep , | wc 
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HEIGHT_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HEIGHT_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | grep -v rs | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_BMI_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | wc 
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_BMI_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep , | wc 
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_BMI_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_BMI_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | grep -v rs | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_BMI_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | wc 
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_BMI_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep , | wc 
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_BMI_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WHRadjBMI_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | wc 
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WHRadjBMI_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep , | wc 
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WHRadjBMI_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WHRadjBMI_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | wc 
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WHRadjBMI_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep , | wc 
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WHRadjBMI_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WHR_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | wc 
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WHR_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep , | wc 
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WHR_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WHR_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | wc 
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WHR_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep , | wc 
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WHR_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HIPadjBMI_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | wc 
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HIPadjBMI_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep , | wc 
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HIPadjBMI_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HIPadjBMI_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | wc 
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HIPadjBMI_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep , | wc 
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HIPadjBMI_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HIP_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | wc 
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HIP_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep , | wc 
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HIP_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HIP_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | wc 
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HIP_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep , | wc 
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HIP_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WCadjBMI_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | wc 
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WCadjBMI_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep , | wc 
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WCadjBMI_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WCadjBMI_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | wc 
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WCadjBMI_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep , | wc 
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WCadjBMI_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WC_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | wc 
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WC_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep , | wc 
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WC_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WC_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | wc 
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WC_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep , | wc 
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WC_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WEIGHT_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | wc 
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WEIGHT_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep , | wc 
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WEIGHT_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | wc
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WEIGHT_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | wc 
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WEIGHT_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep , | wc 
zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WEIGHT_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | wc

#~~~
[  mturchin20@spudling52  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_*_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | perl -lane 'print $#F;' | sort | uniq -c
46583187 8
[  mturchin20@spudling74  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HEIGHT_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep , | head -n 10
rs10884221      t       c       0.8083  -0.0098 0.0072  0.17    59229.5 10_107486155,6_cox_hap1_48402
rs11193825      c       g       0.15    -0.0042 0.0089  0.64    60555.2 10_109725754,6_cox_hap1_4021800
rs11527583      a       c       0.4417  -0.0022 0.0061  0.72    60506.9 10_20698187,5_h2_hap1_1641990
rs2503868       c       g       .       0.0033  0.014   0.82    29451.1 10_43117647,6_cox_hap1_4404084,6_qbl_hap2_4209369
rs11818958      a       c       0.4583  -0.0022 0.0061  0.72    60556.6 10_43808197,6_cox_hap1_48162
rs10824920      t       c       0.5583  0.0040  0.0063  0.52    60505.1 10_54585619,6_cox_hap1_4449605,6_qbl_hap2_4255532
rs1046747       a       g       0.375   -0.0045 0.0063  0.47    60500.6 10_71580635,6_cox_hap1_498071,6_qbl_hap2_300095
rs15801 t       c       0.4     -0.0053 0.0061  0.39    60538.7 10_71580668,6_cox_hap1_498102,6_qbl_hap2_300126
rs2394643       t       c       0.4     -0.0054 0.0061  0.38    60536.9 10_71581019,6_cox_hap1_498457,6_qbl_hap2_300481
rs17343324      a       g       .       -0.0010 0.015   0.95    12317.3 10_88799902,10_random_105971
[  mturchin20@spudling49  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HEIGHT_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep , | wc
13827  124443 1245567
[  mturchin20@spudling49  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HEIGHT_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | wc
1       9      44
[  mturchin20@spudling49  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HEIGHT_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | grep -v rs | wc
0       0       0
[  mturchin20@spudling49  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HEIGHT_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | wc
2748520 24736680 160765048
[  mturchin20@spudling49  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HEIGHT_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep , | wc
13760  123840 1238783
[  mturchin20@spudling49  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HEIGHT_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | wc
1       9      44
[  mturchin20@spudling49  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HEIGHT_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | grep -v rs | wc
0       0       0
[  mturchin20@spudling52  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_BMI_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | wc
2736850 24631650 159851677
[  mturchin20@spudling52  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_BMI_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep , | wc
13678  123102 1230687
[  mturchin20@spudling52  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_BMI_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | wc
1       9      44
[  mturchin20@spudling52  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_BMI_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | grep -v rs | wc
0       0       0
[  mturchin20@spudling52  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_BMI_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | wc
2736850 24631650 159924050
[  mturchin20@spudling52  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_BMI_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep , | wc
13678  123102 1229700
[  mturchin20@spudling52  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_BMI_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | wc
1       9      44
[  mturchin20@spudling52  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_BMI_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | grep -v rs | wc
0       0       0
[  mturchin20@spudling52  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_BMI_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA
rs4510589       a       g       .       0.0079  0.0096  0.41    27670   NA
[  mturchin20@spudling52  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_BMI_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep -v rs | wc
1       9      59
[  mturchin20@spudling52  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_BMI_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep -v rs
MarkerName      A1      A2      Freq.Hapmap.Ceu BETA    SE.2gc  P.2gc   N       ChrBP
[  mturchin20@spudling52  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WHRadjBMI_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | wc
2742719 24684471 159937922
[  mturchin20@spudling52  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WHRadjBMI_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep , | wc
13723  123507 1232512
[  mturchin20@spudling52  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WHRadjBMI_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | wc
1       9      45
[  mturchin20@spudling52  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WHRadjBMI_WOMEN_N.wUCSC
GenomeBrowser_dbSNP130.vs1.txt.gz | wc
2737330 24635970 159690178
[  mturchin20@spudling52  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WHRadjBMI_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep , | wc
13621  122589 1224175
[  mturchin20@spudling52  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WHRadjBMI_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | wc
1       9      43
[  mturchin20@spudling52  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WHR_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | wc
2743314 24689826 159474582
[  mturchin20@spudling52  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WHR_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep , | wc
13724  123516 1230202
[  mturchin20@spudling52  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WHR_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | wc
1       9      44
[  mturchin20@spudling52  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WHR_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | wc
2738277 24644493 159488583
[  mturchin20@spudling52  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WHR_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep , | wc
13627  122643 1223158
[  mturchin20@spudling52  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WHR_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | wc
1       9      44
[  mturchin20@spudling52  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HIPadjBMI_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | wc
2725770 24531930 158583551
[  mturchin20@spudling52  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HIPadjBMI_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep , | wc
13532  121788 1215259
[  mturchin20@spudling52  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HIPadjBMI_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | wc
1       9      43
[  mturchin20@spudling52  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HIPadjBMI_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | wc
2725770 24531930 158193487
[  mturchin20@spudling52  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HIPadjBMI_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep , | wc
13532  121788 1209939
[  mturchin20@spudling52  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HIPadjBMI_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | wc
1       9      43
[  mturchin20@spudling52  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HIP_MEN_N.wUCSCGenomeBr
owser_dbSNP130.vs1.txt.gz | wc
2725730 24531570 158233458
[  mturchin20@spudling52  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HIP_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep , | wc
13530  121770 1213552
[  mturchin20@spudling52  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HIP_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | wc
1       9      42
[  mturchin20@spudling52  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HIP_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | wc
2725730 24531570 158587868
[  mturchin20@spudling52  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HIP_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep , | wc
13530  121770 1214335
[  mturchin20@spudling52  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HIP_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | wc
1       9      43
[  mturchin20@spudling52  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WCadjBMI_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | wc
2744340 24699060 161028253
[  mturchin20@spudling52  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WCadjBMI_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep , | wc
13724  123516 1238481
[  mturchin20@spudling52  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WCadjBMI_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | wc
1       9      45
[  mturchin20@spudling52  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WCadjBMI_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | wc
2738312 24644808 160585060
[  mturchin20@spudling52  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WCadjBMI_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep , | wc
13626  122634 1227995
[  mturchin20@spudling52  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WCadjBMI_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | wc
1       9      44
[  mturchin20@spudling52  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WC_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | wc
2745518 24709662 159596765
[  mturchin20@spudling52  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WC_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep , | wc
13725  123525 1229833
[  mturchin20@spudling52  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WC_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | wc
1       9      43
[  mturchin20@spudling52  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WC_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | wc
2740359 24663231 159625213
[  mturchin20@spudling52  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WC_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep , | wc
13631  122679 1223141
[  mturchin20@spudling52  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WC_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | wc
1       9      44
[  mturchin20@spudling52  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WEIGHT_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | wc
2760764 24846876 161121016
[  mturchin20@spudling52  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WEIGHT_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep , | wc
13829  124461 1244620
[  mturchin20@spudling52  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WEIGHT_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | wc
1       9      43
[  mturchin20@spudling52  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WEIGHT_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | wc
2746981 24722829 160322057
[  mturchin20@spudling52  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WEIGHT_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep , | wc
13755  123795 1235871
[  mturchin20@spudling52  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WEIGHT_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep NA | wc
1       9      43
#~~~


mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1
cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/DataOrganization.GIANT2014_5.PythonVLookUp.CreateFullAnnotFile.vs1.py /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/DataOrganization.GIANT2013.PythonVLookUp.CreateFullAnnotFile.vs1.py

zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HEIGHT_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep -E 'rs17524355|rs12219605|rs4747841|rs737656' | gzip > /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HEIGHT_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.4SNPs.txt.gz

#~~~
[  mturchin20@spudling49  ~]$zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HEIGHT_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.4SNPs.txt.gz
rs4747841       a       g       0.55    0.0025  0.0061  0.68    60558.2 10_10000134
rs737656        a       g       0.3667  -0.0073 0.0064  0.25    60529.2 10_100002728
rs17524355      t       c       .       -0.046  0.046   0.32    1700    10_100003427
rs12219605      t       g       0.45    -0.0025 0.0061  0.68    60558.2 10_10000458
rs7376568       t       g       0.2833  0.0069  0.0078  0.38    59638.3 4_189370200
[  mturchin20@spudling49  ~]$python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/DataOrganization.GIANT2013.PythonVLookUp.CreateFullAnnotFile.vs1.py --file1 <(cat /mnt/lustre/home/mturchin20/Data/GIANT/2013/journal.pgen.1003500.s007.noCarriageReturn.Male.MarkerChrBP.txt <(zcat /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HEIGHT_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.4SNPs.txt.gz | head -n 1 | perl -lane 'print $F[0], "\t", ((split(/_/, $F[8]))[0]), "\t", ((split(/_/, $F[8]))[1]);')) --file2 /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HEIGHT_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.4SNPs.txt.gz,/mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HEIGHT_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.4SNPs.txt.gz
/mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HEIGHT_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.4SNPs.txt.gz,/mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HEIGHT_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.4SNPs.txt.gz
rs4747841       a       g       10_10000134     0.45    1       0.45    0.0025  0.0061  60558.2 0.45    0.0025  0.0061  60558.2
rs12219605      t       g       10_10000458     0.45    2       0.45    -0.0025 0.0061  60558.2 0.45    -0.0025 0.0061  60558.2
rs737656        a       g       10_100002728    0.3667  0       0.3667  -0.0073 0.0064  60529.2 0.3667  -0.0073 0.0064  60529.2
rs7376568       t       g       4_189370200     0.2833  0       0.2833  0.0069  0.0078  59638.3 0.2833  0.0069  0.0078  59638.3
rs17524355      t       c       10_100003427    NA      0       NA      -0.046  0.046   1700    NA      -0.046  0.046   1700
#~~~

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/DataOrganization.GIANT2013.PythonVLookUp.CreateFullAnnotFile.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/GIANT/2013/journal.pgen.1003500.s007.noCarriageReturn.Male.MarkerChrBP.txt --file2 /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HEIGHT_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz,/mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_BMI_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz,/mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WHRadjBMI_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz,/mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WHR_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz,/mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HIPadjBMI_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz,/mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HIP_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz,/mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WCadjBMI_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz,/mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WC_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz,/mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WEIGHT_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_AllPheno_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz

zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_AllPheno_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | wc
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_AllPheno_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | perl -lane 'print $#F;' | sort | uniq -c
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_AllPheno_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | awk '{ print $4 }' | sort | uniq -c | perl -lane 'if ($F[1] !~ m/\d+/) { print join("\t", @F); }'
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_AllPheno_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | awk '{ if ($4 == "None") { print $0 } } '
Error2a -- line2[0] present more than once in a particular file (count: 0 line2[0]: rs12996248,t,g,0.2833,0.0050,0.0074,0.50,58638.6,2_121661536)
Error2a -- line2[0] present more than once in a particular file (count: 0 line2[0]: rs7591686,a,g,0.7333,-0.0020,0.0072,0.78,60529.7,2_45609171)
Error2a -- line2[0] present more than once in a particular file (count: 0 line2[0]: rs2936687,a,g,0.8167,-0.015,0.0082,0.068,60556.5,8_75279924)
Error2a -- line2[0] present more than once in a particular file (count: 0 line2[0]: rs3780697,t,c,0.725,0.0042,0.0079,0.60,60527.2,9_131618930)
Error2a -- line2[0] present more than once in a particular file (count: 0 line2[0]: rs573071,a,g,0.675,-0.0054,0.0065,0.40,60495.8,9_246993)
Error2a -- line2[0] present more than once in a particular file (count: 1 line2[0]: rs424881,a,g,0.225,0.0093,0.0078,0.23,58618.1,10_132097974)
Error2a -- line2[0] present more than once in a particular file (count: 1 line2[0]: rs2160421,a,t,0.4417,0.0039,0.0064,0.54,58617.6,15_34746878)
Error2a -- line2[0] present more than once in a particular file (count: 1 line2[0]: rs13598,a,c,0.8417,-0.017,0.0091,0.059,58619.2,15_46489749)
Error2a -- line2[0] present more than once in a particular file (count: 1 line2[0]: rs5820007,t,c,.,-0.011,0.0099,0.27,57091,17_28342660)
Error2a -- line2[0] present more than once in a particular file (count: 1 line2[0]: rs627146,a,g,0.2833,0.00080,0.0067,0.90,57248.1,18_8332477)
Error2a -- line2[0] present more than once in a particular file (count: 1 line2[0]: rs3752976,a,g,.,-0.011,0.014,0.44,58619.7,1_199381979)
Error2a -- line2[0] present more than once in a particular file (count: 1 line2[0]: rs11576885,c,g,.,-0.00050,0.0070,0.94,58611.3,1_204737275)
Error2a -- line2[0] present more than once in a particular file (count: 1 line2[0]: rs3819262,t,g,0.6917,-0.00020,0.0067,0.98,58618.3,21_37382759)
Error2a -- line2[0] present more than once in a particular file (count: 1 line2[0]: rs12996248,t,g,0.2833,-0.0055,0.0078,0.48,56724.5,2_121661536)
Error2a -- line2[0] present more than once in a particular file (count: 1 line2[0]: rs7591686,a,g,0.7167,0.012,0.0073,0.094,58613.6,2_45609171)
Error2a -- line2[0] present more than once in a particular file (count: 1 line2[0]: rs2936687,a,g,0.8167,0.024,0.0084,0.0041,58618.4,8_75279924)
Error2a -- line2[0] present more than once in a particular file (count: 1 line2[0]: rs3780697,t,c,0.725,-0.0013,0.0082,0.87,58607.2,9_131618930)
Error2a -- line2[0] present more than once in a particular file (count: 1 line2[0]: rs573071,a,g,0.6833,-0.0019,0.0068,0.78,58614.7,9_246993)
Error2a -- line2[0] present more than once in a particular file (count: 2 line2[0]: rs424881,a,g,0.225,0.00E+00,0.0083,1.0,34601.4,10_132097974)
Error2a -- line2[0] present more than once in a particular file (count: 2 line2[0]: rs2160421,a,t,0.4417,0.0017,0.0067,0.80,34600.9,15_34746878)
Error2a -- line2[0] present more than once in a particular file (count: 2 line2[0]: rs13598,a,c,0.8417,0.0071,0.0098,0.47,34601.8,15_46489749)
Error2a -- line2[0] present more than once in a particular file (count: 2 line2[0]: rs5820007,t,c,0.875,-0.0021,0.010,0.84,33318.2,17_28342660)
Error2a -- line2[0] present more than once in a particular file (count: 2 line2[0]: rs627146,a,g,0.2833,0.0010,0.0070,0.89,34595.9,18_8332477)
Error2a -- line2[0] present more than once in a particular file (count: 2 line2[0]: rs3752976,a,g,.,0.0055,0.014,0.70,34602.2,1_199381979)
Error2a -- line2[0] present more than once in a particular file (count: 2 line2[0]: rs11576885,c,g,.,-0.0059,0.0075,0.43,34599.3,1_204737275)
Error2a -- line2[0] present more than once in a particular file (count: 2 line2[0]: rs3819262,t,g,0.7167,0.012,0.0071,0.080,34600.8,21_37382759)
Error2a -- line2[0] present more than once in a particular file (count: 2 line2[0]: rs12996248,t,g,0.2833,-0.0037,0.0083,0.66,32709.5,2_121661536)
Error2a -- line2[0] present more than once in a particular file (count: 2 line2[0]: rs7591686,a,g,0.7333,0.0029,0.0077,0.71,34596.2,2_45609171)
Error2a -- line2[0] present more than once in a particular file (count: 2 line2[0]: rs2936687,a,g,0.8167,0.0078,0.0089,0.38,34601.1,8_75279924)
Error2a -- line2[0] present more than once in a particular file (count: 2 line2[0]: rs3780698,t,g,0.8167,0.0065,0.0078,0.40,34601.2,9_131618931)
Error2a -- line2[0] present more than once in a particular file (count: 2 line2[0]: rs573071,a,g,0.675,-0.00090,0.0071,0.90,34599.2,9_246993)
Error2a -- line2[0] present more than once in a particular file (count: 3 line2[0]: rs424881,a,g,0.775,-0.0014,0.0094,0.88,34581.2,10_132097974)
Error2a -- line2[0] present more than once in a particular file (count: 3 line2[0]: rs2160421,a,t,0.4417,0.0049,0.0076,0.52,34580.7,15_34746878)
Error2a -- line2[0] present more than once in a particular file (count: 3 line2[0]: rs13598,a,c,.,-0.0018,0.011,0.87,34581.6,15_46489749)
Error2a -- line2[0] present more than once in a particular file (count: 3 line2[0]: rs5820007,t,c,.,0.0036,0.012,0.76,33297,17_28342660)
Error2a -- line2[0] present more than once in a particular file (count: 3 line2[0]: rs627146,a,g,0.2833,0.0055,0.0079,0.49,34575.8,18_8332477)
Error2a -- line2[0] present more than once in a particular file (count: 3 line2[0]: rs3752976,a,g,.,-0.0097,0.016,0.55,34582,1_199381979)
Error2a -- line2[0] present more than once in a particular file (count: 3 line2[0]: rs11576885,c,g,.,-0.0068,0.0085,0.43,34579.1,1_204737275)
Error2a -- line2[0] present more than once in a particular file (count: 3 line2[0]: rs3819262,t,g,0.7167,0.013,0.0080,0.097,34580.6,21_37382759)
Error2a -- line2[0] present more than once in a particular file (count: 3 line2[0]: rs12996248,t,g,0.2833,-0.0043,0.0095,0.65,32689.3,2_121661536)
Error2a -- line2[0] present more than once in a particular file (count: 3 line2[0]: rs7591686,a,g,0.7167,0.0057,0.0087,0.51,34576,2_45609171)
Error2a -- line2[0] present more than once in a particular file (count: 3 line2[0]: rs2936687,a,g,0.8167,0.025,0.010,0.013,34580.9,8_75279924)
Error2a -- line2[0] present more than once in a particular file (count: 3 line2[0]: rs3780698,t,g,0.725,0.0057,0.0088,0.52,34581,9_131618931)
Error2a -- line2[0] present more than once in a particular file (count: 3 line2[0]: rs573071,a,g,0.675,-0.0024,0.0081,0.77,34579,9_246993)
Error2a -- line2[0] present more than once in a particular file (count: 4 line2[0]: rs424881,a,g,0.775,0.0037,0.0099,0.71,32930.1,10_132097974)
Error2a -- line2[0] present more than once in a particular file (count: 4 line2[0]: rs2160421,a,t,.,0.00070,0.0079,0.93,32930.2,15_34746878)
Error2a -- line2[0] present more than once in a particular file (count: 4 line2[0]: rs13598,a,c,.,0.0032,0.011,0.77,32930.4,15_46489749)
Error2a -- line2[0] present more than once in a particular file (count: 4 line2[0]: rs5820007,t,c,.,0.0028,0.013,0.83,31647.1,17_28342660)
Error2a -- line2[0] present more than once in a particular file (count: 4 line2[0]: rs627146,a,g,0.2833,-0.0025,0.0082,0.76,32925.3,18_8332477)
Error2a -- line2[0] present more than once in a particular file (count: 4 line2[0]: rs3752976,a,g,.,0.0019,0.018,0.92,32931.5,1_199381979)
Error2a -- line2[0] present more than once in a particular file (count: 4 line2[0]: rs11576885,c,g,.,-0.011,0.0087,0.20,32927.8,1_204737275)
Error2a -- line2[0] present more than once in a particular file (count: 4 line2[0]: rs3819262,t,g,0.6917,-0.0012,0.0083,0.89,32929.4,21_37382759)
Error2a -- line2[0] present more than once in a particular file (count: 4 line2[0]: rs12996248,t,g,0.2833,0.0058,0.0097,0.55,32924,2_121661536)
Error2a -- line2[0] present more than once in a particular file (count: 4 line2[0]: rs7591686,a,g,0.7167,-0.0068,0.0093,0.46,32925.5,2_45609171)
Error2a -- line2[0] present more than once in a particular file (count: 4 line2[0]: rs2936687,a,g,0.8167,-0.016,0.011,0.14,32930.1,8_75279924)
Error2a -- line2[0] present more than once in a particular file (count: 4 line2[0]: rs3780698,t,g,0.725,-0.012,0.0092,0.18,32930.5,9_131618931)
Error2a -- line2[0] present more than once in a particular file (count: 4 line2[0]: rs573071,a,g,0.6833,0.010,0.0084,0.21,32928.2,9_246993)
Error2a -- line2[0] present more than once in a particular file (count: 5 line2[0]: rs424881,a,g,0.775,0.0046,0.011,0.67,30603,10_132097974)
Error2a -- line2[0] present more than once in a particular file (count: 5 line2[0]: rs2160421,a,t,.,0.0064,0.0083,0.44,32849,15_34746878)
Error2a -- line2[0] present more than once in a particular file (count: 5 line2[0]: rs13598,a,c,0.8417,-0.0059,0.012,0.62,32849.2,15_46489749)
Error2a -- line2[0] present more than once in a particular file (count: 5 line2[0]: rs5820007,t,c,.,0.018,0.013,0.18,31564.9,17_28342660)
Error2a -- line2[0] present more than once in a particular file (count: 5 line2[0]: rs627146,a,g,.,0.0014,0.0087,0.87,32844.3,18_8332477)
Error2a -- line2[0] present more than once in a particular file (count: 5 line2[0]: rs3752976,a,g,.,-0.029,0.019,0.13,32850.3,1_199381979)
Error2a -- line2[0] present more than once in a particular file (count: 5 line2[0]: rs11576885,c,g,.,0.00070,0.0093,0.94,32846.7,1_204737275)
Error2a -- line2[0] present more than once in a particular file (count: 5 line2[0]: rs3819262,t,g,0.6917,-0.0046,0.0087,0.60,32848.2,21_37382759)
Error2a -- line2[0] present more than once in a particular file (count: 5 line2[0]: rs12996248,t,g,0.2833,0.0041,0.010,0.69,32842.8,2_121661536)
Error2a -- line2[0] present more than once in a particular file (count: 5 line2[0]: rs7591686,a,g,0.7167,-0.0020,0.0096,0.84,32844.3,2_45609171)
Error2a -- line2[0] present more than once in a particular file (count: 5 line2[0]: rs2936687,a,g,0.8167,0.016,0.011,0.15,32848.9,8_75279924)
Error2a -- line2[0] present more than once in a particular file (count: 5 line2[0]: rs3780698,t,g,0.8167,0.00060,0.0096,0.95,32849.3,9_131618931)
Error2a -- line2[0] present more than once in a particular file (count: 5 line2[0]: rs573071,a,g,0.6833,0.010,0.0089,0.24,32847,9_246993)
Error2a -- line2[0] present more than once in a particular file (count: 6 line2[0]: rs424881,a,g,0.225,0.0033,0.0057,0.57,38369.6,10_132097974)
Error2a -- line2[0] present more than once in a particular file (count: 6 line2[0]: rs2160421,a,t,.,0.0024,0.0047,0.61,38369.1,15_34746878)
Error2a -- line2[0] present more than once in a particular file (count: 6 line2[0]: rs13598,a,c,.,0.0024,0.0069,0.73,38370,15_46489749)
Error2a -- line2[0] present more than once in a particular file (count: 6 line2[0]: rs5820007,t,c,.,0.0054,0.0075,0.47,36845.4,17_28342660)
Error2a -- line2[0] present more than once in a particular file (count: 6 line2[0]: rs627146,a,g,0.2833,-0.0012,0.0050,0.81,37013.2,18_8332477)
Error2a -- line2[0] present more than once in a particular file (count: 6 line2[0]: rs3752976,a,g,.,-0.0058,0.0099,0.56,38370.4,1_199381979)
Error2a -- line2[0] present more than once in a particular file (count: 6 line2[0]: rs11576885,c,g,.,-0.0044,0.0053,0.41,38367.5,1_204737275)
Error2a -- line2[0] present more than once in a particular file (count: 6 line2[0]: rs3819262,t,g,0.7167,0.00080,0.0049,0.87,38369,21_37382759)
Error2a -- line2[0] present more than once in a particular file (count: 6 line2[0]: rs12996248,t,g,0.2833,0.0036,0.0057,0.53,36477.7,2_121661536)
Error2a -- line2[0] present more than once in a particular file (count: 6 line2[0]: rs7591686,a,g,0.7167,-0.00040,0.0054,0.94,38364.4,2_45609171)
Error2a -- line2[0] present more than once in a particular file (count: 6 line2[0]: rs2936687,a,g,0.8167,-0.0024,0.0063,0.70,38369.3,8_75279924)
Error2a -- line2[0] present more than once in a particular file (count: 6 line2[0]: rs3780697,t,c,0.725,-0.0058,0.0061,0.34,38364.2,9_131618930)
Error2a -- line2[0] present more than once in a particular file (count: 6 line2[0]: rs573071,a,g,0.6833,-0.0014,0.0050,0.78,38367.4,9_246993)
Error2a -- line2[0] present more than once in a particular file (count: 7 line2[0]: rs424881,a,g,0.775,0.0052,0.0092,0.57,38305,10_132097974)
Error2a -- line2[0] present more than once in a particular file (count: 7 line2[0]: rs2160421,a,t,.,0.0098,0.0074,0.19,38304.5,15_34746878)
Error2a -- line2[0] present more than once in a particular file (count: 7 line2[0]: rs13598,a,c,0.8417,-0.0076,0.011,0.48,38305.4,15_46489749)
Error2a -- line2[0] present more than once in a particular file (count: 7 line2[0]: rs5820007,t,c,.,0.0068,0.012,0.56,36779.8,17_28342660)
Error2a -- line2[0] present more than once in a particular file (count: 7 line2[0]: rs627146,a,g,0.2833,0.0025,0.0078,0.75,36948.8,18_8332477)
Error2a -- line2[0] present more than once in a particular file (count: 7 line2[0]: rs3752976,a,g,.,-0.030,0.016,0.069,38305.8,1_199381979)
Error2a -- line2[0] present more than once in a particular file (count: 7 line2[0]: rs11576885,c,g,.,-0.0015,0.0084,0.86,38302.9,1_204737275)
Error2a -- line2[0] present more than once in a particular file (count: 7 line2[0]: rs3819262,t,g,0.6917,0.0029,0.0078,0.71,38304.4,21_37382759)
Error2a -- line2[0] present more than once in a particular file (count: 7 line2[0]: rs12996248,t,g,0.2833,-0.00020,0.0092,0.98,36413.1,2_121661536)
Error2a -- line2[0] present more than once in a particular file (count: 7 line2[0]: rs7591686,a,g,0.7167,0.0075,0.0086,0.38,38299.8,2_45609171)
Error2a -- line2[0] present more than once in a particular file (count: 7 line2[0]: rs2936687,a,g,0.8167,0.026,0.0099,0.0093,38304.7,8_75279924)
Error2a -- line2[0] present more than once in a particular file (count: 7 line2[0]: rs3780697,t,c,0.725,-0.0015,0.0096,0.88,38299.9,9_131618930)
Error2a -- line2[0] present more than once in a particular file (count: 7 line2[0]: rs573071,a,g,0.6833,-0.0013,0.0079,0.87,38302.8,9_246993)
Error2a -- line2[0] present more than once in a particular file (count: 8 line2[0]: rs424881,a,g,0.775,0.0080,0.0078,0.31,58322.7,10_132097974)
Error2a -- line2[0] present more than once in a particular file (count: 8 line2[0]: rs2160421,a,t,0.4417,0.00020,0.0064,0.98,58322.2,15_34746878)
Error2a -- line2[0] present more than once in a particular file (count: 8 line2[0]: rs13598,a,c,0.8417,-0.015,0.0091,0.099,58323.8,15_46489749)
Error2a -- line2[0] present more than once in a particular file (count: 8 line2[0]: rs5820007,t,c,0.875,-0.0052,0.010,0.60,56795.6,17_28342660)
Error2a -- line2[0] present more than once in a particular file (count: 8 line2[0]: rs627146,a,g,.,-0.0033,0.0068,0.62,56953.3,18_8332477)
Error2a -- line2[0] present more than once in a particular file (count: 8 line2[0]: rs3752976,a,g,.,-0.0072,0.015,0.62,58324.3,1_199381979)
Error2a -- line2[0] present more than once in a particular file (count: 8 line2[0]: rs11576885,c,g,.,0.00080,0.0072,0.91,58315.9,1_204737275)
Error2a -- line2[0] present more than once in a particular file (count: 8 line2[0]: rs3819262,t,g,0.7167,-0.0058,0.0068,0.39,58322.9,21_37382759)
Error2a -- line2[0] present more than once in a particular file (count: 8 line2[0]: rs12996248,t,g,0.2833,0.00070,0.0078,0.93,56429.1,2_121661536)
Error2a -- line2[0] present more than once in a particular file (count: 8 line2[0]: rs7591686,a,g,0.7167,0.0081,0.0074,0.27,58318.2,2_45609171)
Error2a -- line2[0] present more than once in a particular file (count: 8 line2[0]: rs2936687,a,g,0.8167,0.016,0.0085,0.060,58323,8_75279924)
Error2a -- line2[0] present more than once in a particular file (count: 8 line2[0]: rs3780697,t,c,0.725,0.0045,0.0083,0.59,58312.2,9_131618930)
Error2a -- line2[0] present more than once in a particular file (count: 8 line2[0]: rs573071,a,g,0.6833,-0.0034,0.0069,0.62,58319.3,9_246993)
#~~~

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/DataOrganization.GIANT2013.PythonVLookUp.CreateFullAnnotFile.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/GIANT/2013/journal.pgen.1003500.s007.noCarriageReturn.Female.MarkerChrBP.txt --file2 /mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HEIGHT_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz,/mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_BMI_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz,/mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WHRadjBMI_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz,/mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WHR_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz,/mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HIPadjBMI_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz,/mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HIP_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz,/mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WCadjBMI_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz,/mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WC_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz,/mnt/gluster/data/external_public_supp/GIANT2013/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_WEIGHT_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_AllPheno_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz

zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_AllPheno_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | wc
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_AllPheno_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | perl -lane 'print $#F;' | sort | uniq -c
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_AllPheno_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | awk '{ print $4 }' | sort | uniq -c | perl -lane 'if ($F[1] !~ m/\d+/) { print join("\t", @F); }'
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_AllPheno_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | awk '{ if ($4 == "None") { print $0 } } '
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_AllPheno_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | awk '{ print $5 }' | sort | uniq -c | perl -lane 'if ($F[1] !~ m/\d+/) { print join("\t", @F); }'
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_AllPheno_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | awk '{ print $6 }' | sort | uniq -c

~~~
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2013]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_AllPheno_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | wc
2752595 115608990 769011235
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2013]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_AllPheno_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | perl -lane 'print $#F;' | sort | uniq -c
2752595 41
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2013]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_AllPheno_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | awk '{ print $4 }' | sort | uniq -c | perl -lane 'if ($F[1] !~ m/\d+/) { print join("\t", @F); }'
1       NA
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2013]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_AllPheno_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | awk '{ if ($4 == "None") { print $0 } } '
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2013]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_AllPheno_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | awk '{ print $5 }' | sort | uniq -c | perl -lane 'if ($F[1] !~ m/\d+/) { print join("\t", @F); }'
492041  NA
[  mturchin20@bigmem02  /mnt/gluster/data/external_public_supp/GIANT2013]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_AllPheno_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | awk '{ print $6 }' | sort | uniq -c
2717426 0
     38 1
  35131 2
~~~

#~~~
Error2a -- line2[0] present more than once in a particular file (count: 0 line2[0]: 0: rs424881,a,g,0.775,0.0023,0.0069,0.74,73128.9,10_132097974)
Error2a -- line2[0] present more than once in a particular file (count: 0 line2[0]: 0: rs2160421,a,t,0.4417,-0.0091,0.0056,0.11,73120.5,15_34746878)
Error2a -- line2[0] present more than once in a particular file (count: 0 line2[0]: 0: rs13598,a,c,.,-0.0095,0.0082,0.25,73130,15_46489749)
Error2a -- line2[0] present more than once in a particular file (count: 0 line2[0]: 0: rs5820007,t,c,.,0.0031,0.0091,0.73,70813.9,17_28342660)
Error2a -- line2[0] present more than once in a particular file (count: 0 line2[0]: 0: rs627146,a,g,.,0.0079,0.0060,0.19,71183.1,18_8332477)
Error2a -- line2[0] present more than once in a particular file (count: 0 line2[0]: 0: rs3752976,a,g,.,-0.0098,0.013,0.45,73133.5,1_199381979)
Error2a -- line2[0] present more than once in a particular file (count: 0 line2[0]: 0: rs11576885,c,g,.,0.0016,0.0063,0.80,72203.5,1_204737275)
Error2a -- line2[0] present more than once in a particular file (count: 0 line2[0]: 0: rs3819262,t,g,0.7167,-0.0074,0.0061,0.22,73136.1,21_37382759)
Error2a -- line2[0] present more than once in a particular file (count: 0 line2[0]: 0: rs12996248,t,g,0.2833,-0.0025,0.0069,0.72,70681,2_121661536)
Error2a -- line2[0] present more than once in a particular file (count: 0 line2[0]: 0: rs7591686,a,g,0.7333,-0.0065,0.0065,0.32,73112.3,2_45609171)
Error2a -- line2[0] present more than once in a particular file (count: 0 line2[0]: 0: rs2936687,a,g,0.8167,-0.0017,0.0075,0.82,73134.7,8_75279924)
Error2a -- line2[0] present more than once in a particular file (count: 0 line2[0]: 0: rs3780697,t,c,0.8167,-0.00050,0.0074,0.95,73092.2,9_131618930)
Error2a -- line2[0] present more than once in a particular file (count: 0 line2[0]: 0: rs573071,a,g,0.675,-0.00060,0.0061,0.92,73077.3,9_246993)
Error2a -- line2[0] present more than once in a particular file (count: 1 line2[0]: 1: rs424881,a,g,0.225,-0.0025,0.0075,0.74,67956.5,10_132097974)
Error2a -- line2[0] present more than once in a particular file (count: 1 line2[0]: 1: rs2160421,a,t,.,0.0043,0.0060,0.48,67956.1,15_34746878)
Error2a -- line2[0] present more than once in a particular file (count: 1 line2[0]: 1: rs13598,a,c,.,-0.0021,0.0088,0.81,67957.6,15_46489749)
Error2a -- line2[0] present more than once in a particular file (count: 1 line2[0]: 1: rs5820007,t,c,.,0.0098,0.0098,0.32,65653.5,17_28342660)
Error2a -- line2[0] present more than once in a particular file (count: 1 line2[0]: 1: rs627146,a,g,.,-0.0080,0.0065,0.22,66069.8,18_8332477)
Error2a -- line2[0] present more than once in a particular file (count: 1 line2[0]: 1: rs3752976,a,g,.,0.0046,0.014,0.74,67958.1,1_199381979)
Error2a -- line2[0] present more than once in a particular file (count: 1 line2[0]: 1: rs11576885,c,g,.,-0.00040,0.0068,0.95,67952.1,1_204737275)
Error2a -- line2[0] present more than once in a particular file (count: 1 line2[0]: 1: rs3819262,t,g,0.6917,0.013,0.0065,0.047,67956.7,21_37382759)
Error2a -- line2[0] present more than once in a particular file (count: 1 line2[0]: 1: rs12996248,t,g,0.2833,0.0058,0.0075,0.44,65534.6,2_121661536)
Error2a -- line2[0] present more than once in a particular file (count: 1 line2[0]: 1: rs7591686,a,g,0.7333,-0.0028,0.0070,0.69,67952.9,2_45609171)
Error2a -- line2[0] present more than once in a particular file (count: 1 line2[0]: 1: rs2936687,a,g,0.8167,0.010,0.0081,0.21,67956.3,8_75279924)
Error2a -- line2[0] present more than once in a particular file (count: 1 line2[0]: 1: rs3780697,t,c,0.725,0.00060,0.0080,0.94,67938,9_131618930)
Error2a -- line2[0] present more than once in a particular file (count: 1 line2[0]: 1: rs573071,a,g,0.6833,-0.0057,0.0066,0.39,67953.9,9_246993)
Error2a -- line2[0] present more than once in a particular file (count: 2 line2[0]: rs424881,a,g,0.225,0.0048,0.0081,0.55,42735.4,10_132097974)
Error2a -- line2[0] present more than once in a particular file (count: 2 line2[0]: rs2160421,a,t,0.4417,-0.0059,0.0066,0.37,42735,15_34746878)
Error2a -- line2[0] present more than once in a particular file (count: 2 line2[0]: rs13598,a,c,0.8417,-0.00090,0.0098,0.93,42735.8,15_46489749)
Error2a -- line2[0] present more than once in a particular file (count: 2 line2[0]: rs5820007,t,c,0.875,0.0089,0.010,0.39,40783.3,17_28342660)
Error2a -- line2[0] present more than once in a particular file (count: 2 line2[0]: rs627146,a,g,0.2833,0.0022,0.0069,0.75,42728.4,18_8332477)
Error2a -- line2[0] present more than once in a particular file (count: 2 line2[0]: rs3752976,a,g,.,0.0087,0.015,0.55,42736.3,1_199381979)
Error2a -- line2[0] present more than once in a particular file (count: 2 line2[0]: rs11576885,c,g,.,0.013,0.0074,0.072,42733.3,1_204737275)
Error2a -- line2[0] present more than once in a particular file (count: 2 line2[0]: rs3819262,t,g,0.6917,-0.0033,0.0070,0.64,42734.9,21_37382759)
Error2a -- line2[0] present more than once in a particular file (count: 2 line2[0]: rs12996248,t,g,0.2833,0.0032,0.0080,0.69,40315,2_121661536)
Error2a -- line2[0] present more than once in a particular file (count: 2 line2[0]: rs7591686,a,g,0.7333,-0.00040,0.0076,0.96,42731.2,2_45609171)
Error2a -- line2[0] present more than once in a particular file (count: 2 line2[0]: rs2936687,a,g,0.8167,0.0096,0.0088,0.28,42734.8,8_75279924)
Error2a -- line2[0] present more than once in a particular file (count: 2 line2[0]: rs3780698,t,g,0.725,-0.0032,0.0077,0.68,42735.3,9_131618931)
Error2a -- line2[0] present more than once in a particular file (count: 2 line2[0]: rs573071,a,g,0.6833,0.0050,0.0071,0.48,42733.2,9_246993)
Error2a -- line2[0] present more than once in a particular file (count: 3 line2[0]: rs424881,a,g,0.225,-0.00070,0.0087,0.94,42732.1,10_132097974)
Error2a -- line2[0] present more than once in a particular file (count: 3 line2[0]: rs2160421,a,t,0.4417,-0.0067,0.0070,0.34,42731.7,15_34746878)
Error2a -- line2[0] present more than once in a particular file (count: 3 line2[0]: rs13598,a,c,0.8417,-0.00070,0.010,0.95,42732.5,15_46489749)
Error2a -- line2[0] present more than once in a particular file (count: 3 line2[0]: rs5820007,t,c,0.875,0.0073,0.011,0.51,40780,17_28342660)
Error2a -- line2[0] present more than once in a particular file (count: 3 line2[0]: rs627146,a,g,0.2833,0.00060,0.0074,0.94,42725.2,18_8332477)
Error2a -- line2[0] present more than once in a particular file (count: 3 line2[0]: rs3752976,a,g,.,0.011,0.016,0.46,42733,1_199381979)
Error2a -- line2[0] present more than once in a particular file (count: 3 line2[0]: rs11576885,c,g,.,0.0098,0.0079,0.21,42730,1_204737275)
Error2a -- line2[0] present more than once in a particular file (count: 3 line2[0]: rs3819262,t,g,0.7167,0.0024,0.0075,0.75,42731.6,21_37382759)
Error2a -- line2[0] present more than once in a particular file (count: 3 line2[0]: rs12996248,t,g,0.2833,0.011,0.0086,0.21,40311.7,2_121661536)
Error2a -- line2[0] present more than once in a particular file (count: 3 line2[0]: rs7591686,a,g,0.7167,-0.0025,0.0081,0.76,42727.9,2_45609171)
Error2a -- line2[0] present more than once in a particular file (count: 3 line2[0]: rs2936687,a,g,0.8167,0.011,0.0094,0.26,42731.5,8_75279924)
Error2a -- line2[0] present more than once in a particular file (count: 3 line2[0]: rs3780698,t,g,0.725,0.0011,0.0082,0.89,42732,9_131618931)
Error2a -- line2[0] present more than once in a particular file (count: 3 line2[0]: rs573071,a,g,0.6833,0.0021,0.0075,0.78,42729.9,9_246993)
Error2a -- line2[0] present more than once in a particular file (count: 4 line2[0]: rs424881,a,g,0.225,0.0032,0.0092,0.73,40724.6,10_132097974)
Error2a -- line2[0] present more than once in a particular file (count: 4 line2[0]: rs2160421,a,t,0.4417,0.00030,0.0074,0.97,40724.7,15_34746878)
Error2a -- line2[0] present more than once in a particular file (count: 4 line2[0]: rs13598,a,c,.,-0.010,0.011,0.34,40725,15_46489749)
Error2a -- line2[0] present more than once in a particular file (count: 4 line2[0]: rs5820007,t,c,0.875,0.0040,0.012,0.74,38772.7,17_28342660)
Error2a -- line2[0] present more than once in a particular file (count: 4 line2[0]: rs627146,a,g,.,-0.00040,0.0077,0.96,40718.2,18_8332477)
Error2a -- line2[0] present more than once in a particular file (count: 4 line2[0]: rs3752976,a,g,.,-0.0032,0.017,0.85,40387,1_199381979)
Error2a -- line2[0] present more than once in a particular file (count: 4 line2[0]: rs11576885,c,g,.,-0.0020,0.0082,0.81,40722.3,1_204737275)
Error2a -- line2[0] present more than once in a particular file (count: 4 line2[0]: rs3819262,t,g,0.7167,0.0036,0.0077,0.64,40724,21_37382759)
Error2a -- line2[0] present more than once in a particular file (count: 4 line2[0]: rs12996248,t,g,0.2833,0.0023,0.0089,0.80,40720,2_121661536)
Error2a -- line2[0] present more than once in a particular file (count: 4 line2[0]: rs7591686,a,g,0.7333,0.0074,0.0087,0.39,40720.9,2_45609171)
Error2a -- line2[0] present more than once in a particular file (count: 4 line2[0]: rs2936687,a,g,0.8167,-0.0077,0.0099,0.44,40724.4,8_75279924)
Error2a -- line2[0] present more than once in a particular file (count: 4 line2[0]: rs3780698,t,g,0.725,0.0093,0.0086,0.28,40725,9_131618931)
Error2a -- line2[0] present more than once in a particular file (count: 4 line2[0]: rs573071,a,g,0.6833,0.0010,0.0079,0.90,40722.6,9_246993)
Error2a -- line2[0] present more than once in a particular file (count: 5 line2[0]: rs424881,a,g,0.775,-0.014,0.0097,0.16,40355.4,10_132097974)
Error2a -- line2[0] present more than once in a particular file (count: 5 line2[0]: rs2160421,a,t,0.4417,-0.00080,0.0078,0.92,40355.4,15_34746878)
Error2a -- line2[0] present more than once in a particular file (count: 5 line2[0]: rs13598,a,c,.,-0.0081,0.012,0.48,40355.8,15_46489749)
Error2a -- line2[0] present more than once in a particular file (count: 5 line2[0]: rs5820007,t,c,0.875,-0.0023,0.012,0.85,38403.4,17_28342660)
Error2a -- line2[0] present more than once in a particular file (count: 5 line2[0]: rs627146,a,g,0.2833,-0.0060,0.0084,0.47,40349.3,18_8332477)
Error2a -- line2[0] present more than once in a particular file (count: 5 line2[0]: rs3752976,a,g,.,0.016,0.018,0.37,40356.8,1_199381979)
Error2a -- line2[0] present more than once in a particular file (count: 5 line2[0]: rs11576885,c,g,.,-0.0091,0.0089,0.31,40353.1,1_204737275)
Error2a -- line2[0] present more than once in a particular file (count: 5 line2[0]: rs3819262,t,g,0.6917,0.014,0.0083,0.098,40354.8,21_37382759)
Error2a -- line2[0] present more than once in a particular file (count: 5 line2[0]: rs12996248,t,g,0.2833,0.019,0.0095,0.044,40350.8,2_121661536)
Error2a -- line2[0] present more than once in a particular file (count: 5 line2[0]: rs7591686,a,g,0.7167,-0.0052,0.0092,0.57,40351.7,2_45609171)
Error2a -- line2[0] present more than once in a particular file (count: 5 line2[0]: rs2936687,a,g,0.8167,-0.0047,0.011,0.66,40355.2,8_75279924)
Error2a -- line2[0] present more than once in a particular file (count: 5 line2[0]: rs3780698,t,g,0.725,0.017,0.0091,0.060,40355.8,9_131618931)
Error2a -- line2[0] present more than once in a particular file (count: 5 line2[0]: rs573071,a,g,0.6833,-0.0025,0.0085,0.77,40353.4,9_246993)
Error2a -- line2[0] present more than once in a particular file (count: 6 line2[0]: rs424881,a,g,0.225,0.0014,0.0056,0.80,47471.7,10_132097974)
Error2a -- line2[0] present more than once in a particular file (count: 6 line2[0]: rs2160421,a,t,.,-0.0030,0.0046,0.51,47471.3,15_34746878)
Error2a -- line2[0] present more than once in a particular file (count: 6 line2[0]: rs13598,a,c,.,-0.0050,0.0068,0.46,47472.1,15_46489749)
Error2a -- line2[0] present more than once in a particular file (count: 6 line2[0]: rs5820007,t,c,.,0.0020,0.0072,0.78,45180.6,17_28342660)
Error2a -- line2[0] present more than once in a particular file (count: 6 line2[0]: rs627146,a,g,.,0.0023,0.0049,0.64,45600.6,18_8332477)
Error2a -- line2[0] present more than once in a particular file (count: 6 line2[0]: rs3752976,a,g,.,0.0046,0.010,0.65,47472.6,1_199381979)
Error2a -- line2[0] present more than once in a particular file (count: 6 line2[0]: rs11576885,c,g,.,0.0095,0.0051,0.063,47469.6,1_204737275)
Error2a -- line2[0] present more than once in a particular file (count: 6 line2[0]: rs3819262,t,g,0.7167,-0.00010,0.0048,0.98,47471.2,21_37382759)
Error2a -- line2[0] present more than once in a particular file (count: 6 line2[0]: rs12996248,t,g,0.2833,0.0036,0.0055,0.51,45050.3,2_121661536)
Error2a -- line2[0] present more than once in a particular file (count: 6 line2[0]: rs7591686,a,g,0.7167,0.0029,0.0052,0.58,47467.5,2_45609171)
Error2a -- line2[0] present more than once in a particular file (count: 6 line2[0]: rs2936687,a,g,0.8167,0.0076,0.0060,0.21,47471.1,8_75279924)
Error2a -- line2[0] present more than once in a particular file (count: 6 line2[0]: rs3780697,t,c,0.8167,-0.0015,0.0059,0.80,47465.6,9_131618930)
Error2a -- line2[0] present more than once in a particular file (count: 6 line2[0]: rs573071,a,g,0.675,0.0019,0.0049,0.70,47469.5,9_246993)
Error2a -- line2[0] present more than once in a particular file (count: 7 line2[0]: rs424881,a,g,0.775,-0.0080,0.0086,0.35,47320.6,10_132097974)
Error2a -- line2[0] present more than once in a particular file (count: 7 line2[0]: rs2160421,a,t,.,-0.0047,0.0070,0.50,47320.2,15_34746878)
Error2a -- line2[0] present more than once in a particular file (count: 7 line2[0]: rs13598,a,c,.,-0.0051,0.010,0.62,47321,15_46489749)
Error2a -- line2[0] present more than once in a particular file (count: 7 line2[0]: rs5820007,t,c,.,0.0049,0.011,0.66,45029.5,17_28342660)
Error2a -- line2[0] present more than once in a particular file (count: 7 line2[0]: rs627146,a,g,.,-0.0033,0.0074,0.66,45447.9,18_8332477)
Error2a -- line2[0] present more than once in a particular file (count: 7 line2[0]: rs3752976,a,g,.,0.011,0.016,0.50,47321.5,1_199381979)
Error2a -- line2[0] present more than once in a particular file (count: 7 line2[0]: rs11576885,c,g,.,0.0041,0.0079,0.60,47318.5,1_204737275)
Error2a -- line2[0] present more than once in a particular file (count: 7 line2[0]: rs3819262,t,g,0.7167,0.012,0.0073,0.088,47320.1,21_37382759)
Error2a -- line2[0] present more than once in a particular file (count: 7 line2[0]: rs12996248,t,g,0.2833,0.013,0.0085,0.14,44899.1,2_121661536)
Error2a -- line2[0] present more than once in a particular file (count: 7 line2[0]: rs7591686,a,g,0.7333,0.00060,0.0081,0.94,47316.4,2_45609171)
Error2a -- line2[0] present more than once in a particular file (count: 7 line2[0]: rs2936687,a,g,0.8167,0.0077,0.0093,0.41,47320,8_75279924)
Error2a -- line2[0] present more than once in a particular file (count: 7 line2[0]: rs3780697,t,c,0.725,-0.0018,0.0091,0.84,47314.9,9_131618930)
Error2a -- line2[0] present more than once in a particular file (count: 7 line2[0]: rs573071,a,g,0.6833,-0.0051,0.0074,0.49,47318.4,9_246993)
Error2a -- line2[0] present more than once in a particular file (count: 8 line2[0]: rs424881,a,g,0.225,-0.0029,0.0075,0.70,67592.2,10_132097974)
Error2a -- line2[0] present more than once in a particular file (count: 8 line2[0]: rs2160421,a,t,.,-0.0021,0.0061,0.73,67591.8,15_34746878)
Error2a -- line2[0] present more than once in a particular file (count: 8 line2[0]: rs13598,a,c,.,-0.0082,0.0089,0.35,67593.3,15_46489749)
Error2a -- line2[0] present more than once in a particular file (count: 8 line2[0]: rs5820007,t,c,0.875,0.0089,0.010,0.37,65288.2,17_28342660)
Error2a -- line2[0] present more than once in a particular file (count: 8 line2[0]: rs627146,a,g,0.2833,-0.0068,0.0065,0.30,65706.2,18_8332477)
Error2a -- line2[0] present more than once in a particular file (count: 8 line2[0]: rs3752976,a,g,.,0.0016,0.014,0.91,67593.8,1_199381979)
Error2a -- line2[0] present more than once in a particular file (count: 8 line2[0]: rs11576885,c,g,.,0.0010,0.0069,0.88,67587.8,1_204737275)
Error2a -- line2[0] present more than once in a particular file (count: 8 line2[0]: rs3819262,t,g,0.7167,0.0096,0.0065,0.14,67592.4,21_37382759)
Error2a -- line2[0] present more than once in a particular file (count: 8 line2[0]: rs12996248,t,g,0.2833,0.0049,0.0074,0.51,65170.3,2_121661536)
Error2a -- line2[0] present more than once in a particular file (count: 8 line2[0]: rs7591686,a,g,0.7333,-0.0035,0.0071,0.62,67588.6,2_45609171)
Error2a -- line2[0] present more than once in a particular file (count: 8 line2[0]: rs2936687,a,g,0.8167,0.011,0.0082,0.16,67592,8_75279924)
Error2a -- line2[0] present more than once in a particular file (count: 8 line2[0]: rs3780697,t,c,0.725,0.0015,0.0081,0.85,67574.2,9_131618930)
Error2a -- line2[0] present more than once in a particular file (count: 8 line2[0]: rs573071,a,g,0.675,-0.0073,0.0066,0.27,67589.6,9_246993)
#~~~

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/process.MTedits.ForGIANT2014_5.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/process.MTedits.ForGIANT2013.vs1.R

#Copy/pasted commands from /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/process.MTedits.ForGIANT2013.vs1.R into an interactive sessions in $PWD /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/

mv /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/.RData /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/RData.GIANT2013.process.20150602
gzip /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/RData.GIANT2013.process.20150602

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/DataOrganization.GIANT2010.PythonVLookUp.PutTogetherProcessAndAnnotFiles.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GIANT2013.Male.dtlesssignif.vs1.txt --file2 <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_AllPheno_MEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | perl -lane 'my $chr; my $pos; if (($F[3] eq "NA") || ($F[3] eq "None")) { $chr = "NA"; $pos = "NA"; } else { $chr = ((split(/_/, $F[3]))[0]); $pos = ((split(/_/, $F[3]))[1]); } splice(@F, 1, 0, ($chr, $pos)); splice(@F, 8, 1); splice(@F, 11, 1); splice(@F, 14, 1); splice(@F, 17, 1); splice(@F, 20, 1); splice(@F, 23, 1); splice(@F, 27, 1); splice(@F, 30, 1); splice(@F, 33, 1); print join("\t", @F);') | \ 
gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GIANT2013.Male.dtlesssignif.vs1.annot.vs1.txt.gz

#Added the following header to /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GIANT2013.Male.dtlesssignif.vs1.annot.vs1.txt.gz
#snp chr pos a1 a2 chrbp maf annot beta_height se_height n_height beta_BMI se_BMI n_BMI beta_WHRadjBMI se_WHRadjBMI n_WHRadjBMI  beta_WHR se_WHR n_WHR beta_HIPadjBMI se_HIPadjBMI n_HIPadjBMI beta_HIP se_HIP n_HIP beta_WCadjBMI se_WCadjBMI n_WCadjBMI beta_WC se_WC n_WC beta_weight se_weight n_weight Z.height Z.BMI Z.WHRadjBMI Z.WHR Z.HIPadjBMI Z.HIP Z.WCadjBMI Z.WC Z.weight mvstat mvp unip

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GIANT2013.Male.dtlesssignif.vs1.txt | wc
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GIANT2013.Male.dtlesssignif.vs1.annot.vs1.txt.gz | wc
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GIANT2013.Male.dtlesssignif.vs1.annot.vs1.txt.gz | perl -lane 'if ($F[$#F-1] > $F[$#F]) { print join("\t", @F) ; } ' | wc
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GIANT2013.Male.dtlesssignif.vs1.annot.vs1.txt.gz | perl -F, -lane 'print $#F;' | sort | uniq -c
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GIANT2013.Male.dtlesssignif.vs1.annot.vs1.txt.gz | head -n 5

~~~
[  mturchin20@bigmem03  ~]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GIANT2013.Male.dtlesssignif.vs1.txt | wc
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GIANT2013.Male.dtlesssignif.vs1.annot.vs1.txt.gz | wc
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GIANT2013.Male.dtlesssignif.vs1.annot.vs1.txt.gz | perl -lane 'if ($F[$#F-1] > $F[$#F]) { print join("\t", @F) ; } ' | wc
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GIANT2013.Male.dtlesssignif.vs1.annot.vs1.txt.gz | perl -F, -lane 'print $#F;' | sort | uniq -c
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GIANT2013.Male.dtlesssignif.vs1.annot.vs1.txt.gz | head -n 5
7624    7624 1638075
[  mturchin20@bigmem03  ~]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GIANT2013.Male.dtlesssignif.vs1.annot.vs1.txt.gz | wc
7624  358328 3362517
[  mturchin20@bigmem03  ~]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GIANT2013.Male.dtlesssignif.vs1.annot.vs1.txt.gz | perl -lane 'if ($F[$#F-1] > $F[$#F]) { print join("\t", @F) ; } ' | wc
7453  350291 3287270
[  mturchin20@bigmem03  ~]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GIANT2013.Male.dtlesssignif.vs1.annot.vs1.txt.gz | perl -F, -lane 'print $#F;' | sort | uniq -c
7624 0
[  mturchin20@bigmem03  ~]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GIANT2013.Male.dtlesssignif.vs1.annot.vs1.txt.gz | head -n 5
snp chr pos a1 a2 chrbp maf annot beta_height se_height n_height beta_BMI se_BMI n_BMI beta_WHRadjBMI se_WHRadjBMI n_WHRadjBMI  beta_WHR se_WHR n_WHR beta_HIPadjBMI se_HIPadjBMI n_HIPadjBMI beta_HIP se_HIP n_HIP beta_WCadjBMI se_WCadjBMI n_WCadjBMI beta_WC se_WC n_WC beta_weight se_weight n_weight Z.height Z.BMI Z.WHRadjBMI Z.WHR Z.HIPadjBMI Z.HIP Z.WCadjBMI Z.WC Z.weight mvstat mvp unip
rs13078872      3       142732650       c       g       3_142732650     0.3417  0       -0.039  0.0066  60556.4 -0.0022 0.0068  58618.3 0.0062  0.0072  34600.8 -0.00020
0.0081  34580.6 -0.031  0.0085  32929.5 -0.029  0.0089  32848.2 0.3417  0.0050  38369   0.3417  0.0080  38304.4 0.3417  0.0069  58322.9 5.92727656693212       0.318639363964375        0.859617364241911       0.0250689082587111      3.64250283822469        3.26361637006925        2.12007168974215        2.0641868904004 3.09023230616781        64.3274543043609        12.4478658862156        8.51144928349955
rs545708        18      55991198        t       c       18_55991198     0.4667  0       0.016   0.0061  60540.8 0.039   0.0064  58617.7 -0.0038 0.0068  34600.4 0.022  0.0076   34580.2 0.0044  0.0079  32929.3 0.038   0.0083  32848.1 0.4667  0.0047  38368.6 0.4667  0.0075  38304   0.4667  0.0064  58322.3 2.5758293035489 6.11944811150025
0.568051498338983       2.82690683268606        0.553384719555673       4.58684245215462        0.125661346855074       5.01259803621023        7.00499197014344
78.0663903750382        15.3494810269617        11.6073030467403
rs10484733      6       142752680       c       g       6_142752680     0.1917  0       0.035   0.0077  60398.8 -0.012  0.0080  58377.7 -0.010  0.0083  34601.2 -0.021 0.0095   34581   0.026   0.010   32930.4 0.0012  0.010   32849.2 0.1917  0.0060  38128.4 0.1917  0.0093  38063.8 0.1917  0.0081  58082.3 4.56311578837316        1.51410188761928        1.22652812003661        2.21151780918668        2.63961605297416        0.113038540644565       2.38670773449225        0.789191652658222       1.17498679206609        37.1300181865462        6.77120095160744        5.29756946355448
rs17514738      1       217055376       t       c       1_217055376     NA      0       0.024   0.0063  60543.8 0.0044  0.0065  58615.7 -0.0023 0.0069  34601.1 0.0029 0.0077   34580.9 0.026   0.0080  32929   0.030   0.0085  32847.8 NA      0.0048  38369.3 NA      0.0075  38304.7 NA      0.0065  58320.3 3.89722801810903        0.674489750196082       0.331853346436817       0.371856089385075       3.19465105376329        3.48075640434621        2.10835839916911        2.45726339020544        2.36561812686429        40.3777497032728        7.44180904800661        4.01188715973165
~~~

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/DataOrganization.GIANT2010.PythonVLookUp.PutTogetherProcessAndAnnotFiles.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GIANT2013.Female.dtlesssignif.vs1.txt --file2 <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_AllPheno_WOMEN_N.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.txt.gz | perl -lane 'my $chr; my $pos; if (($F[3] eq "NA") || ($F[3] eq "None")) { $chr = "NA"; $pos = "NA"; } else { $chr = ((split(/_/, $F[3]))[0]); $pos = ((split(/_/, $F[3]))[1]); } splice(@F, 1, 0, ($chr, $pos)); splice(@F, 8, 1); splice(@F, 11, 1); splice(@F, 14, 1); splice(@F, 17, 1); splice(@F, 20, 1); splice(@F, 23, 1); splice(@F, 27, 1); splice(@F, 30, 1); splice(@F, 33, 1); print join("\t", @F);') | \ 
gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GIANT2013.Female.dtlesssignif.vs1.annot.vs1.txt.gz

#Added the following header to /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GIANT2013.Female.dtlesssignif.vs1.annot.vs1.txt.gz
#snp chr pos a1 a2 chrbp maf annot beta_height se_height n_height beta_BMI se_BMI n_BMI beta_WHRadjBMI se_WHRadjBMI n_WHRadjBMI  beta_WHR se_WHR n_WHR beta_HIPadjBMI se_HIPadjBMI n_HIPadjBMI beta_HIP se_HIP n_HIP beta_WCadjBMI se_WCadjBMI n_WCadjBMI beta_WC se_WC n_WC beta_weight se_weight n_weight Z.height Z.BMI Z.WHRadjBMI Z.WHR Z.HIPadjBMI Z.HIP Z.WCadjBMI Z.WC Z.weight mvstat mvp unip

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GIANT2013.Female.dtlesssignif.vs1.txt | wc
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GIANT2013.Female.dtlesssignif.vs1.annot.vs1.txt.gz | wc
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GIANT2013.Female.dtlesssignif.vs1.annot.vs1.txt.gz | perl -lane 'if ($F[$#F-1] > $F[$#F]) { print join("\t", @F) ; } ' | wc
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GIANT2013.Female.dtlesssignif.vs1.annot.vs1.txt.gz | perl -F, -lane 'print $#F;' | sort | uniq -c
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GIANT2013.Female.dtlesssignif.vs1.annot.vs1.txt.gz | head -n 5

~~~
[  mturchin20@bigmem03  ~]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GIANT2013.Female.dtlesssignif.vs1.txt | wc
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GIANT2013.Female.dtlesssignif.vs1.annot.vs1.txt.gz | wc
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GIANT2013.Female.dtlesssignif.vs1.annot.vs1.txt.gz | perl -lane 'if ($F[$#F-1] > $F[$#F]) { print join("\t", @F) ; } ' | wc
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GIANT2013.Female.dtlesssignif.vs1.annot.vs1.txt.gz | perl -F, -lane 'print $#F;' | sort | uniq -c
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GIANT2013.Female.dtlesssignif.vs1.annot.vs1.txt.gz | head -n 5
9437    9437 2032741
[  mturchin20@bigmem03  ~]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GIANT2013.Female.dtlesssignif.vs1.annot.vs1.txt.gz | wc
9437  443539 4172992
[  mturchin20@bigmem03  ~]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GIANT2013.Female.dtlesssignif.vs1.annot.vs1.txt.gz | perl -lane 'if ($F[$#F-1] > $F[$#F]) { print join("\t", @F) ; } ' | wc
8725  410075 3855206
[  mturchin20@bigmem03  ~]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GIANT2013.Female.dtlesssignif.vs1.annot.vs1.txt.gz | perl -F, -lane 'print $#F;' | sort | uniq -c
9437 0
[  mturchin20@bigmem03  ~]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GIANT2013.Female.dtlesssignif.vs1.annot.vs1.txt.gz | head -n 5
snp chr pos a1 a2 chrbp maf annot beta_height se_height n_height beta_BMI se_BMI n_BMI beta_WHRadjBMI se_WHRadjBMI n_WHRadjBMI  beta_WHR se_WHR n_WHR beta_HIPadjBMI se_HIPadjBMI n_HIPadjBMI beta_HIP se_HIP n_HIP beta_WCadjBMI se_WCadjBMI n_WCadjBMI beta_WC se_WC n_WC beta_weight se_weight n_weight Z.height Z.BMI Z.WHRadjBMI Z.WHR Z.HIPadjBMI Z.HIP Z.WCadjBMI Z.WC Z.weight mvstat mvp unip
rs13078872      3       142732650       c       g       3_142732650     0.3417  0       -0.037  0.0061  73130.9 -0.0033 0.0065  67956.5 -0.0068 0.0071  42734.7 -0.00750.0076   42731.4 -0.0095 0.0079  40723.9 -0.011  0.0085  40354.7 0.3417  0.0049  47471   0.3417  0.0075  47319.9 0.3417  0.0066  67592.2 6.14139478519449        0.510073456968595       0.954165253146194       0.994457883209753       1.20035885803086        1.25356543847045        1.88079360815125        1.51410188761928        2.89430405305143        52.1090569513165        9.88309797640845        9.08724669632868
rs4360313       8       4802188 t       g       8_4802188       0.25    0       0.025   0.0060  73125   -0.020  0.0064  67956.6 0.0033  0.0070  42735.1 -0.0022 0.0075 42731.8  0.016   0.0078  40724.7 -0.0029 0.0084  40355.5 0.25    0.0048  47471.4 0.25    0.0074  47320.3 0.25    0.0065  67592.3 4.19406451274716        3.12138914935986
0.467698799114508       0.292374896226804       1.97736842818195        0.345125531470472       1.31057911216813        0.994457883209753       0.877896295051229       37.0427275273345        6.75321592565396        4.56224943717961
rs17378391      20      47206007        t       c       20_47206007     0.3083  2       -0.032  0.0067  73116.3 0.017   0.0073  67956.9 -0.0034 0.0079  42735.1 0.000600.0084   42731.8 -0.014  0.0088  40724.9 0.0088  0.0093  40355.7 0.3083  0.0054  47471.4 0.3083  0.0083  47320.3 0.3083  0.0073  67592.6 4.75647768915719        2.30798447494596        0.426148007841278       0.0752698620998299      1.59819313992282        0.954165253146194       2.58280745200824        0.10043372051147        0.125661346855074       43.391819913533 8.06646637431228        5.70553377383841
rs545708        18      55991198        t       c       18_55991198     0.4667  0       0.012   0.0057  73118.7 0.030   0.0062  67956.3 0.010   0.0067  42734.5 0.026  0.0071   42731.2 0.0037  0.0074  40723.9 0.031   0.0080  40354.7 0.4667  0.0046  47470.8 0.4667  0.0070  47319.7 0.4667  0.0062  67592   2.15707270447901        4.79215420637313        1.55477359459685        3.58274690211504        0.495850347347453       3.89747712566571        1.05812161768478        4.66474498631497        4.98581864328374        54.6991692870824        10.4252269927113        6.20971483596676
~~~

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GLC.MTedits.ForGIANT2014_5.AllPheno.vs2.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GLC.MTedits.ForGIANT2013.Male.vs1.R
#Copy/pasted commands from /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GLC.MTedits.ForGIANT2013.Male.vs1.R into an inteactve session in #PWD /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/

#CHECK_0: Redo this part in GIANT2014_5.AllPheno -- think I may have mesed it up; the nmin & l==1 portion
#lbf.newhits <- rbind(lbf.bigmat1[,gl$nmin>30000][,l==1],lbf.bigmat2[,gl$nmin>30000][,l==1],lbf.bigmat3[,gl$nmin>30000][,l==1],lbf.bigmat4[,gl$nmin>30000][,l==1],lbf.bigmat5[,gl$nmin>30000][,l==1],lbf.bigmat6[,gl$nmin>30000][,l==1],lbf.bigmat7[,gl$nmin>30000][,l==1])

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GLC.MTedits.ForGIANT2013.Male.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GLC.MTedits.ForGIANT2013.Female.vs1.R
#Copy/pasted commands from /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GLC.MTedits.ForGIANT2013.Female.vs1.R into an inteactve session in #PWD /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/

~~~
[  mturchin20@spudhead  ~/Data/dbSNP]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GLC.MTedits.ForGIANT2013.Male.vs1.newtophits.vs1.txt /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/GLC.MTedits.ForGIANT2013.Female.vs1.newtophits.vs1.txt | awk '{ print $1 }' | sort | uniq -c | awk '{ if ($1 >= 2) { print $0 } } '
2 rs1421085
2 rs143384
2 rs1812175
2 rs2145270
2 rs2280470
2 rs724016
2 rs7498665
2 rs7759938
2 rs806794
2 snp
~~~

#CHECK_0: Make a consistent criteria, or as much as possible, with what SNPs are included as 'GWAS/Hit SNPs', e.g. do you include replication findings or not, just primary analyses/first-round results? etc...?



~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

##20150828
##Interim work on GlobalLipids and/or GIANT datasets prior to creating the suite of scripts that will then just be applied to all datasets below
##E.g. sort of a scratch space for things as I work through them so I don't have this type of 'freehand work' in the middle of what (hopefully) should be just straightforward application of the main scripts (both processing/formatting and analysis itself) below


~~~
[  mturchin20@spudling70  /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/combined_controls]$cat /data/external_public/GlobalLipids2010/HDL_ONE_Europeans.tbl | head -n 10
MarkerName      Allele1 Allele2 Weight  GC.Zscore       GC.Pvalue       Overall Direction
rs3764261       a       c       94225.00        41.691  7.10e-380       +       ++++++++++++++++++?++++++
rs173539        t       c       94225.00        41.661  2.50e-379       +       ++++++++++++++++++?++++++
rs247616        t       c       94225.00        41.576  8.54e-378       +       ++++++++++++++++++?++++++
rs7205804       a       g       98409.00        38.806  2.00e-329       +       +++++++++++++++++++++++++
rs1532624       a       c       98409.00        38.678  2.93e-327       +       +++++++++++++++++++++++++
rs11508026      t       c       98409.00        38.448  2.08e-323       +       +++++++++++++++++++++++++
rs1800775       a       c       96794.00        37.417  2.06e-306       +       ++++++++++++?++++++++++++
rs711752        a       g       90602.00        36.859  2.07e-297       +       +++++++++++++++++++++++++
rs1864163       a       g       94225.00        -35.831 3.69e-281       -       ------------------?------
> Data1 <- read.table("/data/external_public/GlobalLipids2010/HDL_ONE_Europeans.tbl", header=TRUE)
> head(Data1)
  MarkerName Allele1 Allele2 Weight GC.Zscore     GC.Pvalue Overall
  1  rs3764261       a       c  94225    41.691  0.000000e+00       +
  2   rs173539       t       c  94225    41.661  0.000000e+00       +
  3   rs247616       t       c  94225    41.576  0.000000e+00       +
  4  rs7205804       a       g  98409    38.806  0.000000e+00       +
  5  rs1532624       a       c  98409    38.678  0.000000e+00       +
  6 rs11508026       t       c  98409    38.448 1.976263e-323       +
                    Direction
		    1 ++++++++++++++++++?++++++
		    2 ++++++++++++++++++?++++++
		    3 ++++++++++++++++++?++++++
		    4 +++++++++++++++++++++++++
		    5 +++++++++++++++++++++++++
		    6 +++++++++++++++++++++++++
> GetZScore <- function(x) { val1 <- qnorm(log(x/2), lower.tail=FALSE, log.p=TRUE); return(val1) }
> tg <- cbind(Data1, apply(as.matrix(tg$GC.Pvalue), 2, GetZScore))
> tg[1:50,9]
 [1]      Inf      Inf      Inf      Inf      Inf 38.44939 37.41710 36.85937
  [9] 35.83059 35.24572 35.23606 35.21521 33.24590 31.30166 29.40863 27.79082
  [17] 27.14425 26.94528 26.36972 23.37030 21.18964 20.98135 20.84285 20.81887
  [25] 20.60629 20.59302 20.55601 20.55171 20.47032 20.45584 20.41873 20.34816
  [33] 20.25378 20.23548 20.20857 20.20791 20.18025 20.16569 20.14965 20.12878
  [41] 20.12355 20.11633 20.10081 20.09854 20.09075 20.06125 20.05767 20.05301
  [49] 20.04372 20.01759
  > Data1[1:50,5]
   [1]  41.691  41.661  41.576  38.806  38.678  38.448  37.417  36.859 -35.831
   [10]  35.246 -35.236 -35.215  33.246 -31.302 -29.409 -27.791 -27.144  26.945
   [19]  26.370  23.370 -21.190 -20.981 -20.843  20.819  20.606 -20.593  20.556
   [28]  20.552  20.470  20.456 -20.419  20.348  20.254 -20.236  20.209  20.208
   [37]  20.180 -20.166 -20.150 -20.129 -20.124 -20.116 -20.101 -20.099 -20.091
   [46]  20.061  20.058  20.053 -20.044  20.018
~~~   

##Permutation stuff? ##Permutation stuff? ##Permutation stuff? ##Permutation stuff? ##Permutation stuff? ##Permutation stuff? ##Permutation stuff? ##Permutation stuff? ##Permutation stuff? 

cp -p 




















~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

##20150701
##Reorganizing things currently into 'downloading datasets', 'top SNP hits' and 'main analyses' sections, at least for the time being/until something else comes up
#PGC, IBD, TAG, BP(?dbGAP), MAGIC, GEFOS, SSGAC, EGG, GERA, BEAGESS, GWASofVTE, HaemgenRBC, GPC, 23andMe(?), WTCCC

##20150701 && 20150804
##Reorganizing as described below and with the creation of the 'vs2' of this MainScript1 file to separate everything into 3 sections, 'file downloads/file processing', 'top SNP hits' and 'multivariate analysis', divided by study within each section (versus sections being based on each study and then divded three times for every study) 
##20150812
##I think I'm going back to the original organization but I'll be copying this file over to a 'vs3' of everything to keep the organizational order of these decisions
##Goal is to create a standardized beginning/setup though for each dataset as was the same for the initiative behind moving towards 'vs2'
##20150828
##I think I decided (before this current note) to actually stick with the 'vs2' setup....but since I've been editting things in the 'vs3' file I'm just going to continue using the 'vs3' file...even though 'vs3' and 'vs2', organizationally speaking are still the same now (sorry? notsorry?)

#Format
#Chr BP snpID MAF N pval

#General QC scripts
#for i in `ls -lrt | awk '{ print $9 }' | grep results`; do echo $i; zcat $i | head -n 5; done
#for i in `ls -lrt | awk '{ print $9 }' | grep results`; do paste <(zcat $i | wc) <(zcat $i | perl -lane 'print $#F;' | sort | uniq -c) | xargs echo $i; done
#for i in `ls -lrt | awk '{ print $9 }' | grep txt.gz`; do echo $i; paste -d @ <(zcat $i | sed 's/E-//g'| perl -lane 'if ($F[0] !~ m/^rs\d+/) { print join("\t", @F); }' | awk '{ print $1 }' | sort  | uniq -c) <(zcat $i | sed 's/E-//g'| perl -lane 'if ($F[1] !~ m/(1|2|3|4|5|6|7|8|9|10|11|12|13|14|15|16|17|18|19|20|21|22|X|Y|M)/) { print join("\t", @F); }' | awk '{ print $2 }' | sort  | uniq -c) <(zcat $i | sed 's/E-//g'| perl -lane 'if ($F[2] =~ m/[^\d\.]/) { print join("\t", @F); }' | awk '{ print $3 }' | sort  | uniq -c) <(zcat $i | sed 's/E-//g'| perl -lane 'if ($F[7] =~ m/[^\d\.]/) { print join("\t", @F); }' | awk '{ print $8 }' | sort  | uniq -c) <(zcat $i | sed 's/E-//g'| perl -lane 'if ($F[10] =~ m/[^\d\.]/) { print join("\t", @F); }' | awk '{ print $11 }' | sort  | uniq -c) | sed 's/@/@ /g'| column -s $'@' -t ; done
#for i in `ls -lrt | awk '{ print $9 }' | grep results`; do echo $i; zcat $i | perl -lane 'print join("\t", @F);' | R -q -e "library(data.table); Data1 <- fread('file:///dev/stdin', header=TRUE); quantile(Data1\$MAF, na.rm=TRUE); quantile(Data1\$Sample_size, na.rm=TRUE); quantile(Data1\$P_value, na.rm=TRUE); png(\""$i".QCCheck.vs1.png\", height=2000, width=2000, res=200); par(mfrow=c(2,2)); hist(Data1\$MAF, breaks=100); hist(Data1\$Sample_size, breaks=100); hist(Data1\$P_value, breaks=100); expP <- -log10(c(1:length(Data1\$P_value["\!"is.na(Data1\$P_value)])) / (length(Data1\$P_value["\!"is.na(Data1\$P_value)])+1)); plot(expP, -log10(sort(Data1\$P_value["\!"is.na(Data1\$P_value)])), xlab=\"Expected p-Values\", ylab=\"Observed p-Values\"); abline(0, 1, col=\"RED\"); dev.off();"; done





#20150831 CHECK_0: mvstat df values in process...R scripts?



###########################
#
#
# Part B/C -- Downloading/Processing main datasets
#
#
###########################





##GlobalLipids2010 dbsnp ###/HapMap Phase II CEU , GWAS = 95, N = provided per SNP
##20150828

#HapMap Phase II CEU mentioned multiple times in http://www.nature.com.proxy.uchicago.edu/nature/journal/v466/n7307/extref/nature09270-s1.pdf, e.g.:
~~~
...
 To facilitate meta-analysis, each group performed
 genotype imputation using BIMBAM, IMPUTE, or MACH, with reference to the Phase II CEU
 HapMap74. Study-specific details are presented in Supplementary Table 3. 
...
~~~
#Going to assume this goes with release 22 (as is the majority of late 2000s and early 2010s studies are?)

mkdir /mnt/lustre/home/mturchin20/Data/HapMap/Releases
mkdir /mnt/lustre/home/mturchin20/Data/HapMap/Releases/Release23
mkdir /mnt/lustre/home/mturchin20/Data/HapMap/Releases/Release23/Frequencies

cd /mnt/lustre/home/mturchin20/Data/HapMap/Releases/Release23/Frequencies

wget http://hapmap.ncbi.nlm.nih.gov/downloads/frequencies/2008-01/fwd_strand/non-redundant/

mv /mnt/lustre/home/mturchin20/Data/HapMap/Releases/Release23/Frequencies/index.html /mnt/lustre/home/mturchin20/Data/HapMap/Releases/Release23/Frequencies/index.http:__hapmap.ncbi.nlm.nih.gov_downloads_frequencies_2008-01_fwd_strand_non-redundant.html

for i in `cat /mnt/lustre/home/mturchin20/Data/HapMap/Releases/Release23/Frequencies/index.http:__hapmap.ncbi.nlm.nih.gov_downloads_frequencies_2008-01_fwd_strand_non-redundant.html | perl -lane 'my $fullLine = join("", @F); if ($fullLine =~ m/.*(allele.*.txt.gz).*/) { print $1 ; } '`; do wget http://hapmap.ncbi.nlm.nih.gov/downloads/frequencies/2008-01/fwd_strand/non-redundant/$i; done

nbthis *

####Taken from earlier in script
##mkdir /data/external_public/GlobalLipids2010

##cd /data/external_public/GlobalLipids2010

##wget -r -l1 --no-parent -A "*2010*" http://www.sph.umich.edu/csg/abecasis/public/lipids2010/

##wget -r -l1 --no-parent -A "*with_Effect*" http://www.sph.umich.edu/csg/abecasis/public/lipids2010/

###Unzipped both *2010* and *with_Effect* files

gzip *tbl

~~~
[  mturchin20@spudling70  /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/combined_controls]$ls -lrt /data/external_public/GlobalLipids2010/.
total 1846740
-rw-r--r-- 1 mturchin20 mturchin20 174142389 Aug  6  2010 HDL_ONE_Europeans.tbl.gz
-rw-r--r-- 1 mturchin20 mturchin20 174151558 Aug  6  2010 LDL_ONE_Europeans.tbl.gz
-rw-r--r-- 1 mturchin20 mturchin20 174154742 Aug  6  2010 TG_ONE_Europeans.tbl.gz
-rw-r--r-- 1 mturchin20 mturchin20 175468147 Aug  6  2010 TC_ONE_Europeans.tbl.gz
-rw-rw-r-- 1 mturchin20 mturchin20  39057098 Aug  6  2010 HDL2010.zip
-rw-rw-r-- 1 mturchin20 mturchin20  39076634 Aug  6  2010 LDL2010.zip
-rw-rw-r-- 1 mturchin20 mturchin20  39094453 Aug  6  2010 TC2010.zip
-rw-rw-r-- 1 mturchin20 mturchin20  39033516 Aug  6  2010 TG2010.zip
-rw-rw-r-- 1 mturchin20 mturchin20    132125 Aug  6  2010 Manhattan2010.png
-rw-r--r-- 1 mturchin20 mturchin20 203754736 May  7  2013 HDL_with_Effect.tbl.gz
-rw-r--r-- 1 mturchin20 mturchin20 203994603 May  7  2013 LDL_with_Effect.tbl.gz
-rw-r--r-- 1 mturchin20 mturchin20 205290701 May  7  2013 TC_with_Effect.tbl.gz
-rw-r--r-- 1 mturchin20 mturchin20 203832172 May  7  2013 TG_with_Effect.tbl.gz
-rw-rw-r-- 1 mturchin20 mturchin20  56249598 May  7  2013 HDL_with_Effect.zip
-rw-rw-r-- 1 mturchin20 mturchin20  54574432 May  7  2013 LDL_with_Effect.zip
-rw-rw-r-- 1 mturchin20 mturchin20  54581074 May  7  2013 TC_with_Effect.zip
-rw-rw-r-- 1 mturchin20 mturchin20  54374508 May  7  2013 TG_with_Effect.zip
[  mturchin20@spudling26  /data/external_public/GlobalLipids2010]$for i in `ls -lrt | awk '{ print $9 }' | grep with_Effect.tbl`; do echo $i; zcat $i | head -n 5; done
HDL_with_Effect.tbl.gz
MarkerName      Allele1 Allele2 Weight  GC.Zscore       GC.Pvalue       Overall Direction       Effect  StdErr
rs10    a       c       85853   -0.096  0.9232  -       ---+-+++-+?+--??--++??+++       -0.0019 0.0147
rs1000000       a       g       99900   -0.937  0.349   -       ---+-+---++-+---+++-++++-       -0.0056 0.0058
rs10000010      t       c       99900   0.185   0.8532  +       -+++-+-+--+++-++++--+----       0.0013  0.0049
rs10000012      c       g       99843   -0.752  0.4523  -       +-+-+-+--+----+--+++++-++       -0.0065 0.0071
LDL_with_Effect.tbl.gz
MarkerName      Allele1 Allele2 Weight  GC.Zscore       GC.Pvalue       Overall Direction       Effect  StdErr
rs10    a       c       81680   2.051   0.04027 +       ++-+++++-+?++-??-++-??++-       0.0294  0.0152
rs1000000       a       g       95454   0.662   0.5077  +       -----++-+--+-+++-+++-+-++       0.0044  0.0063
rs10000010      t       c       95454   1.583   0.1135  +       ++--++---+-+-++--+++---++       0.0073  0.0052
rs10000012      c       g       95397   0.155   0.877   +       +-+-----++-++---+++---+-+       -2e-04  0.0076
TC_with_Effect.tbl.gz
MarkerName      Allele1 Allele2 Weight  GC.Zscore       GC.Pvalue       Overall Direction       Effect  StdErr
rs10    a       c       86136   2.229   0.02582 +       +--+++++-+?+++??--+-??-+-       0.0304  0.0151
rs1000000       a       g       100184  0.018   0.9855  +       ---+-+----++-+++++++-+-++       7e-04   0.0061
rs10000010      t       c       100184  1.748   0.0805  +       ++-+++-+-+-++++---++---+-       0.0084  0.0051
rs10000012      c       g       100125  0.319   0.7498  +       +-+----+-+--+---+++--++-+       9e-04   0.0074
TG_with_Effect.tbl.gz
MarkerName      Allele1 Allele2 Weight  GC.Zscore       GC.Pvalue       Overall Direction       Effect  StdErr
rs10    a       c       82546   0.75    0.4533  +       ++------+-?+++??++--??-+-       0.0133  0.0141
rs1000000       a       g       96598   1.244   0.2134  +       ++--+++++-++-+++----+---+       0.0075  0.0057
rs10000010      t       c       96598   0.675   0.4999  +       -+-+-----+--+++-+-+++--+-       0.0022  0.0048
rs10000012      c       g       96539   0.579   0.5624  +       -+-+-+----+-+++-++-+-+++-       0.0032  0.007
[  mturchin20@spudling26  /data/external_public/GlobalLipids2010]$for i in `ls -lrt | awk '{ print $9 }' | grep with_Effect.tbl`; do paste <(zcat $i | wc) <(zcat $i | perl -lane 'print $#F;' | sort | uniq -c) | xargs echo $i; done
HDL_with_Effect.tbl.gz 2692430 26924300 203754736 2692430 9
LDL_with_Effect.tbl.gz 2692565 26925650 203994603 2692565 9
TC_with_Effect.tbl.gz 2692414 26924140 205290701 2692414 9
TG_with_Effect.tbl.gz 2692561 26925610 203832172 2692561 9
~~~

##for i in `ls -lrt | awk '{ print $9 }' | grep txt.gz`; do echo $i; paste -d @ <(zcat $i | sed 's/E-//g'| perl -lane 'if ($F[0] !~ m/^rs\d+/) { print join("\t", @F); }' | awk '{ print $1 }' | sort  | uniq -c) <(zcat $i | sed 's/E-//g'| perl -lane 'if ($F[1] !~ m/(1|2|3|4|5|6|7|8|9|10|11|12|13|14|15|16|17|18|19|20|21|22|X|Y|M)/) { print join("\t", @F); }' | awk '{ print $2 }' | sort  | uniq -c) <(zcat $i | sed 's/E-//g'| perl -lane 'if ($F[2] =~ m/[^\d\.]/) { print join("\t", @F); }' | awk '{ print $3 }' | sort  | uniq -c) <(zcat $i | sed 's/E-//g'| perl -lane 'if ($F[7] =~ m/[^\d\.]/) { print join("\t", @F); }' | awk '{ print $8 }' | sort  | uniq -c) <(zcat $i | sed 's/E-//g'| perl -lane 'if ($F[10] =~ m/[^\d\.]/) { print join("\t", @F); }' | awk '{ print $11 }' | sort  | uniq -c) | sed 's/@/@ /g'| column -s $'@' -t ; done
#for i in `ls -lrt | awk '{ print $9 }' | grep results`; do echo $i; zcat $i | perl -lane 'print join("\t", @F);' | R -q -e "library(data.table); Data1 <- fread('file:///dev/stdin', header=TRUE); quantile(Data1\$MAF, na.rm=TRUE); quantile(Data1\$Sample_size, na.rm=TRUE); quantile(Data1\$P_value, na.rm=TRUE); png(\""$i".QCCheck.vs1.png\", height=2000, width=2000, res=200); par(mfrow=c(2,2)); hist(Data1\$MAF, breaks=100); hist(Data1\$Sample_size, breaks=100); hist(Data1\$P_value, breaks=100); expP <- -log10(c(1:length(Data1\$P_value["\!"is.na(Data1\$P_value)])) / (length(Data1\$P_value["\!"is.na(Data1\$P_value)])+1)); plot(expP, -log10(sort(Data1\$P_value["\!"is.na(Data1\$P_value)])), xlab=\"Expected p-Values\", ylab=\"Observed p-Values\"); abline(0, 1, col=\"RED\"); dev.off();"; done

#Going to get MAF and ChrBP before doing graph stuff

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /data/external_public/GlobalLipids2010/HDL_with_Effect.tbl.gz | gzip > /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap22.tbl.gz &

~~~
[  mturchin20@spudling26  /data/external_public/GlobalLipids2010]$zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap22.tbl.gz | head -n 10             MarkerName      Allele1 Allele2 Weight  GC.Zscore       GC.Pvalue       Overall Direction       Effect  StdErr  ChrBPAFInfo
rs10    a       c       85853   -0.096  0.9232  -       ---+-+++-+?+--??--++??+++       -0.0019 0.0147  chr7_92221824_A_0.033_C_0.967
rs1000000       a       g       99900   -0.937  0.349   -       ---+-+---++-+---+++-++++-       -0.0056 0.0058  chr12_125456933_C_0.627_T_0.373
rs10000010      t       c       99900   0.185   0.8532  +       -+++-+-+--+++-++++--+----       0.0013  0.0049  chr4_21227772_T_0.575_C_0.425
rs10000012      c       g       99843   -0.752  0.4523  -       +-+-+-+--+----+--+++++-++       -0.0065 0.0071  chr4_1347325_C_0.808_G_0.192
rs10000013      a       c       99900   0.642   0.5209  +       +++-++--+--++--+---+-++--       3e-04   0.0058  chr4_36901464_C_0.167_A_0.833
rs10000017      t       c       99900   2.043   0.04106 +       +-++++++---++++---++--+++       0.0105  0.0062  chr4_84997149_C_0.777_T_0.223
rs1000002       t       c       99797   2.249   0.02453 +       -+-+---+-+-+++--++-+-+-++       0.0101  0.0049  chr3_185118462_G_0.475_A_0.525
rs10000023      t       g       99900   -0.008  0.9935  -       -+-+--+--+++++--++------+       4e-04   0.005   chr4_95952929_G_0.408_T_0.592
rs10000027      c       g       7807    -0.686  0.4925  -       ?????????????-???????????       -0.0141 0.0205  NA
[  mturchin20@spudling26  /data/external_public/GlobalLipids2010]$zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap22.tbl.gz | wc
2692430 29616730 280821208
[  mturchin20@spudling26  /data/external_public/GlobalLipids2010]$zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap22.tbl.gz | awk '{ print $11 }' | grep NA | wc
 157247  157247  471741
~~~

#Going to try HapMap23 since there are a number of missing SNPs from release 22? Might try 21 too? http://hapmap.ncbi.nlm.nih.gov/old_news.html.en has details

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /data/external_public/GlobalLipids2010/HDL_with_Effect.tbl.gz | gzip > /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap21.tbl.gz 
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /data/external_public/GlobalLipids2010/HDL_with_Effect.tbl.gz | gzip > /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.tbl.gz 

~~~
[  mturchin20@spudling26  /data/external_public/GlobalLipids2010]$zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap21.tbl.gz |  awk '{ print $11 }' | grep NA | wc
 143425  143425  430275
 [  mturchin20@spudling26  /data/external_public/GlobalLipids2010]$zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.tbl.gz |  awk '{ print $11 }' | grep NA | wc
   64542   64542  193626
~~~

#Using HapMap23 then

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /data/external_public/GlobalLipids2010/HDL_with_Effect.tbl.gz | perl -lane 'if ($F[0] =~ m/^rs\d+/) { if ($F[$#F] eq "NA") { push (@F, ("NA", "NA", "NA")); } else { my $chr = -9; my $bp = -9; my $maf = -9; my @vals1 = split(/_/, $F[$#F]); if ($vals1[0] =~ m/chr(.*)/) { $chr = $1; } $bp = $vals1[1]; if ($vals1[3] > .5) { $vals1[3] = 1 - $vals1[3]; } $maf = $vals1[3]; push(@F, ($chr, $bp, $maf)); } } else { push(@F, ("Chr", "BP", "MAF")); } print join("\t", @F);' | gzip > /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.tbl.gz 
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /data/external_public/GlobalLipids2010/LDL_with_Effect.tbl.gz | perl -lane 'if ($F[0] =~ m/^rs\d+/) { if ($F[$#F] eq "NA") { push (@F, ("NA", "NA", "NA")); } else { my $chr = -9; my $bp = -9; my $maf = -9; my @vals1 = split(/_/, $F[$#F]); if ($vals1[0] =~ m/chr(.*)/) { $chr = $1; } $bp = $vals1[1]; if ($vals1[3] > .5) { $vals1[3] = 1 - $vals1[3]; } $maf = $vals1[3]; push(@F, ($chr, $bp, $maf)); } } else { push(@F, ("Chr", "BP", "MAF")); } print join("\t", @F);' | gzip > /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.tbl.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /data/external_public/GlobalLipids2010/TG_with_Effect.tbl.gz | perl -lane 'if ($F[0] =~ m/^rs\d+/) { if ($F[$#F] eq "NA") { push (@F, ("NA", "NA", "NA")); } else { my $chr = -9; my $bp = -9; my $maf = -9; my @vals1 = split(/_/, $F[$#F]); if ($vals1[0] =~ m/chr(.*)/) { $chr = $1; } $bp = $vals1[1]; if ($vals1[3] > .5) { $vals1[3] = 1 - $vals1[3]; } $maf = $vals1[3]; push(@F, ($chr, $bp, $maf)); } } else { push(@F, ("Chr", "BP", "MAF")); } print join("\t", @F);' | gzip > /data/external_public/GlobalLipids2010/TG_with_Effect.wHapMap23.expanded.tbl.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /data/external_public/GlobalLipids2010/TC_with_Effect.tbl.gz | perl -lane 'if ($F[0] =~ m/^rs\d+/) { if ($F[$#F] eq "NA") { push (@F, ("NA", "NA", "NA")); } else { my $chr = -9; my $bp = -9; my $maf = -9; my @vals1 = split(/_/, $F[$#F]); if ($vals1[0] =~ m/chr(.*)/) { $chr = $1; } $bp = $vals1[1]; if ($vals1[3] > .5) { $vals1[3] = 1 - $vals1[3]; } $maf = $vals1[3]; push(@F, ($chr, $bp, $maf)); } } else { push(@F, ("Chr", "BP", "MAF")); } print join("\t", @F);' | gzip > /data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.tbl.gz &

~~~
[  mturchin20@spudling26  /data/external_public/GlobalLipids2010]$for i in `ls -lrt | awk '{ print $9 }' | grep with_Effect.wHapMap23.expanded`; do echo $i; paste -d @ <(zcat $i | sed 's/e-//g'| perl -lane 'if ($F[0] !~ m/^rs\d+/) { print join("\t", @F); }' | awk '{ print $1 }' | sort  | uniq -c) <(zcat $i | sed 's/e-//g'| perl -lane 'if ($F[5] =~ m/[^\d\.]/) { print join("\t", @F); }' | awk '{ print $6 }' | sort  | uniq -c) <(zcat $i | sed 's/e-//g'| perl -lane 'if ($F[3] =~ m/[^\d]/) { print join("\t", @F); }' | awk '{ print $4 }' | sort  | uniq -c) <(zcat $i | perl -lane ' if ($F[10] !~ m/chr(1|2|3|4|5|6|7|8|9|10|11|12|13|14|15|16|17|18|19|20|21|22|X|Y|M)_\d+_\D_.*_\D_.*\d+/) { print $F[10]; }' | sort | uniq -c ) | sed 's/@/ @ /g'| column -s $'@' -t ; done
HDL_with_Effect.wHapMap23.expanded.tbl.gz
1 MarkerName          1 GC.Pvalue          1 Weight          1 chr12_132361639_0_0_0.375_45
								1 chr12_132362221_0_0_0.629_73
								1 chr4_191301184_0_0_0.308_37
								1 ChrBPAFInfo
								64542 NA
LDL_with_Effect.wHapMap23.expanded.tbl.gz
1 MarkerName          1 GC.Pvalue          1 Weight          1 chr12_132361639_0_0_0.375_45
								1 chr12_132362221_0_0_0.629_73
								1 chr4_191301184_0_0_0.308_37
								1 ChrBPAFInfo
								64546 NA
TC_with_Effect.wHapMap23.expanded.tbl.gz
1 MarkerName          1 GC.Pvalue        229 1e+05           1 chr12_132361639_0_0_0.375_45
					1 Weight          1 chr12_132362221_0_0_0.629_73
								1 chr4_191301184_0_0_0.308_37
								1 ChrBPAFInfo
								64544 NA
TG_with_Effect.wHapMap23.expanded.tbl.gz
1 MarkerName          1 GC.Pvalue          1 Weight          1 chr12_132361639_0_0_0.375_45
								1 chr12_132362221_0_0_0.629_73
								1 chr4_191301184_0_0_0.308_37
								1 ChrBPAFInfo
								64541 NA
[  mturchin20@spudling26  /data/external_public/GlobalLipids2010]$for i in `ls -lrt | awk '{ print $9 }' | grep with_Effect.wHapMap23.expanded`; do echo $i; zcat $i | perl -lane 'print join("\t", @F);' | R -q -e "library(data.table); Data1 <- fread('file:///dev/stdin', header=TRUE); quantile(Data1\$MAF, na.rm=TRUE); quantile(Data1\$Weight, na.rm=TRUE); quantile(Data1\$GC.Pvalue, na.rm=TRUE); png(\""$i".QCCheck.vs1.png\", height=2000, width=2000, res=200); par(mfrow=c(2,2)); hist(Data1\$MAF, breaks=100); hist(Data1\$Weight, breaks=100); hist(Data1\$GC.Pvalue, breaks=100); expP <- -log10(c(1:length(Data1\$GC.Pvalue["\!"is.na(Data1\$GC.Pvalue)])) / (length(Data1\$GC.Pvalue["\!"is.na(Data1\$GC.Pvalue)])+1)); plot(expP, -log10(sort(Data1\$GC.Pvalue["\!"is.na(Data1\$GC.Pvalue)])), xlab=\"Expected p-Values\", ylab=\"Observed p-Values\"); abline(0, 1, col=\"RED\"); dev.off();"; done
HDL_with_Effect.wHapMap23.expanded.tbl.gz
> library(data.table); Data1 <- fread('file:///dev/stdin', header=TRUE); quantile(Data1$MAF, na.rm=TRUE); quantile(Data1$Weight, na.rm=TRUE); quantile(Data1$GC.Pvalue, na.rm=TRUE); png("HDL_with_Effect.wHapMap23.expanded.tbl.gz.QCCheck.vs1.png", height=2000, width=2000, res=200); par(mfrow=c(2,2)); hist(Data1$MAF, breaks=100); hist(Data1$Weight, breaks=100); hist(Data1$GC.Pvalue, breaks=100); expP <- -log10(c(1:length(Data1$GC.Pvalue[!is.na(Data1$GC.Pvalue)])) / (length(Data1$GC.Pvalue[!is.na(Data1$GC.Pvalue)])+1)); plot(expP, -log10(sort(Data1$GC.Pvalue[!is.na(Data1$GC.Pvalue)])), xlab="Expected p-Values", ylab="Observed p-Values"); abline(0, 1, col="RED"); dev.off();
Read 2692429 rows and 14 (of 14) columns from 0.309 GB file in 00:01:33
0%   25%   50%   75%  100%
0.000 0.086 0.208 0.347 0.500
0%   25%   50%   75%  100%
279 95716 99150 99900 99900
0%    25%    50%    75%   100%
0.0000 0.2487 0.5000 0.7493 1.0000
null device
1
>
>
LDL_with_Effect.wHapMap23.expanded.tbl.gz
> library(data.table); Data1 <- fread('file:///dev/stdin', header=TRUE); quantile(Data1$MAF, na.rm=TRUE); quantile(Data1$Weight, na.rm=TRUE); quantile(Data1$GC.Pvalue, na.rm=TRUE); png("LDL_with_Effect.wHapMap23.expanded.tbl.gz.QCCheck.vs1.png", height=2000, width=2000, res=200); par(mfrow=c(2,2)); hist(Data1$MAF, breaks=100); hist(Data1$Weight, breaks=100); hist(Data1$GC.Pvalue, breaks=100); expP <- -log10(c(1:length(Data1$GC.Pvalue[!is.na(Data1$GC.Pvalue)])) / (length(Data1$GC.Pvalue[!is.na(Data1$GC.Pvalue)])+1)); plot(expP, -log10(sort(Data1$GC.Pvalue[!is.na(Data1$GC.Pvalue)])), xlab="Expected p-Values", ylab="Observed p-Values"); abline(0, 1, col="RED"); dev.off();
Read 2692564 rows and 14 (of 14) columns from 0.309 GB file in 00:01:32
0%   25%   50%   75%  100%
0.000 0.086 0.208 0.347 0.500
0%   25%   50%   75%  100%
277 91293 94704 95454 95454
0%        25%        50%        75%       100%
9.700e-171  2.477e-01  5.000e-01  7.503e-01  1.000e+00
null device
1
>
>
TC_with_Effect.wHapMap23.expanded.tbl.gz
> library(data.table); Data1 <- fread('file:///dev/stdin', header=TRUE); quantile(Data1$MAF, na.rm=TRUE); quantile(Data1$Weight, na.rm=TRUE); quantile(Data1$GC.Pvalue, na.rm=TRUE); png("TC_with_Effect.wHapMap23.expanded.tbl.gz.QCCheck.vs1.png", height=2000, width=2000, res=200); par(mfrow=c(2,2)); hist(Data1$MAF, breaks=100); hist(Data1$Weight, breaks=100); hist(Data1$GC.Pvalue, breaks=100); expP <- -log10(c(1:length(Data1$GC.Pvalue[!is.na(Data1$GC.Pvalue)])) / (length(Data1$GC.Pvalue[!is.na(Data1$GC.Pvalue)])+1)); plot(expP, -log10(sort(Data1$GC.Pvalue[!is.na(Data1$GC.Pvalue)])), xlab="Expected p-Values", ylab="Observed p-Values"); abline(0, 1, col="RED"); dev.off();
Read 2692413 rows and 14 (of 14) columns from 0.310 GB file in 00:01:33
0%   25%   50%   75%  100%
0.000 0.086 0.208 0.347 0.500
0%    25%    50%    75%   100%
309  96000  99434 100184 100184
0%        25%        50%        75%       100%
5.770e-131  2.466e-01  5.000e-01  7.508e-01  1.000e+00
null device
1
>
>
TG_with_Effect.wHapMap23.expanded.tbl.gz
> library(data.table); Data1 <- fread('file:///dev/stdin', header=TRUE); quantile(Data1$MAF, na.rm=TRUE); quantile(Data1$Weight, na.rm=TRUE); quantile(Data1$GC.Pvalue, na.rm=TRUE); png("TG_with_Effect.wHapMap23.expanded.tbl.gz.QCCheck.vs1.png", height=2000, width=2000, res=200); par(mfrow=c(2,2)); hist(Data1$MAF, breaks=100); hist(Data1$Weight, breaks=100); hist(Data1$GC.Pvalue, breaks=100); expP <- -log10(c(1:length(Data1$GC.Pvalue[!is.na(Data1$GC.Pvalue)])) / (length(Data1$GC.Pvalue[!is.na(Data1$GC.Pvalue)])+1)); plot(expP, -log10(sort(Data1$GC.Pvalue[!is.na(Data1$GC.Pvalue)])), xlab="Expected p-Values", ylab="Observed p-Values"); abline(0, 1, col="RED"); dev.off();
Read 2692560 rows and 14 (of 14) columns from 0.309 GB file in 00:01:34
0%   25%   50%   75%  100%
0.000 0.086 0.208 0.347 0.500
0%   25%   50%   75%  100%
182 92418 95848 96598 96598
0%        25%        50%        75%       100%
6.710e-240  2.472e-01  5.000e-01  7.509e-01  1.000e+00
null device
1
>
>
[  mturchin20@spudling26  /data/external_public/GlobalLipids2010]$for i in `ls -lrt | awk '{ print $9 }' | grep with_Effect.wHapMap23.expanded`; do echo $i; paste -d @ <(zcat $i | awk '{ print $11 }' | grep NA | wc) <(zcat $i | awk '{ print $11 }' | grep , | wc) <(zcat $i | awk '{ print $11 }' | perl -lane 'my @vals1 = split(/,/, $F[0]); print scalar(@vals1);' | sort | uniq -c) <(zcat $i | awk '{ print $14 }' | perl -lane 'if ($F[0] !~ m/0\.0?[1-9]+/) { print $F[0]; }' | sort | uniq -c) | sed 's/@/ @ /g'| column -s $'@' -t ; done
HDL_with_Effect.wHapMap23.expanded.tbl.gz
  64542   64542  193626          0       0       0    2692430 1      29809 0
									9582 0.008
									56270 0.00800000000000001
									2343 0.009
									14270 0.00900000000000001
									1 MAF
									64542 NA
LDL_with_Effect.wHapMap23.expanded.tbl.gz
64546   64546  193638          0       0       0    2692565 1      29788 0
									9594 0.008
									56372 0.00800000000000001
									2348 0.009
									14327 0.00900000000000001
									1 MAF
									64546 NA
TC_with_Effect.wHapMap23.expanded.tbl.gz
64544   64544  193632          0       0       0    2692414 1      29816 0
									9580 0.008
									56302 0.00800000000000001
									2342 0.009
									14274 0.00900000000000001
									1 MAF
									64544 NA
TG_with_Effect.wHapMap23.expanded.tbl.gz
64541   64541  193623          0       0       0    2692561 1      29827 0
									9602 0.008
									56393 0.00800000000000001
									2345 0.009
									14316 0.00900000000000001
									1 MAF
									64541 NA
~~~

mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010
mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs1

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/MAGIC2010/Vs1/process.MTedits.ForMAGIC2010.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs1/process.MTedits.ForGlobalLipids2010.vs1.R

#20150902 Note -- Some SNPs have 'NA' or '0' for MAF

zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.tbl.gz | awk '{ print $12, "\t", $13, "\t", $1, "\t", $14, "\t", $6, "\t", $4 }' | gzip > /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.formatted.tbl.gz &
zcat /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.tbl.gz | awk '{ print $12, "\t", $13, "\t", $1, "\t", $14, "\t", $6, "\t", $4 }' | gzip > /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.formatted.tbl.gz & 
zcat /data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.tbl.gz | awk '{ print $12, "\t", $13, "\t", $1, "\t", $14, "\t", $6, "\t", $4 }' | gzip > /data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.formatted.tbl.gz &
zcat /data/external_public/GlobalLipids2010/TG_with_Effect.wHapMap23.expanded.tbl.gz | awk '{ print $12, "\t", $13, "\t", $1, "\t", $14, "\t", $6, "\t", $4 }' | gzip > /data/external_public/GlobalLipids2010/TG_with_Effect.wHapMap23.expanded.formatted.tbl.gz & 

#join -1 3 -2 3 -e -9 -o 0 1.1 1.2 1.4 1.5 1.6 2.4 2.5 2.6 <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.formatted.tbl.gz | sort -g -k 3,3) <(zcat /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.formatted.tbl.gz | sort -g -k 3,3) | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
#join -1 1 -2 3 -e -9 -o 0 1.2 2.4 2.5 2.6 - <(zcat /data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.formatted.tbl.gz | sort -g -k 3,3) | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
#join -1 1 -2 3 -e -9 -o 0 1.2 2.4 2.5 2.6 - <(zcat /data/external_public/GlobalLipids2010/TG_with_Effect.wHapMap23.expanded.formatted.tbl.gz | sort -g -k 3,3) | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
#perl -lane 'print $F[0], "\t", join("\t", split(/_/, $F[1]));' | gzip > /data/external_public/GlobalLipids2010/AllPhenos_with_Effect.wHapMap23.expanded.formatted.tbl.gz 

#join -a 1 -a 2 -1 3 -2 3 -e NA2 -o 0 1.1 1.2 1.4 1.5 1.6 2.4 2.5 2.6 <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.formatted.tbl.gz | sort -g -k 3,3) <(zcat /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.formatted.tbl.gz | sort -g -k 3,3) | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
#join -a 1 -a 2 -1 1 -2 3 -e NA2 -o 0 1.2 2.4 2.5 2.6 - <(zcat /data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.formatted.tbl.gz | sort -g -k 3,3) | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
#join -a 1 -a 2 -1 1 -2 3 -e NA2 -o 0 1.2 2.4 2.5 2.6 - <(zcat /data/external_public/GlobalLipids2010/TG_with_Effect.wHapMap23.expanded.formatted.tbl.gz | sort -g -k 3,3) | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
#perl -lane 'print $F[0], "\t", join("\t", split(/_/, $F[1]));' | gzip > /data/external_public/GlobalLipids2010/AllPhenos_with_Effect.wHapMap23.expanded.formatted.tbl.gz 

~~~
[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/GPC2012]$zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.formatted.tbl.gz | head -n 10
Chr      BP      MarkerName      MAF     GC.Zscore       Weight
7        92221824        rs10    0.033   -0.096          85853
12       125456933       rs1000000       0.373   -0.937          99900
4        21227772        rs10000010      0.425   0.185   99900
4        1347325         rs10000012      0.192   -0.752          99843
4        36901464        rs10000013      0.167   0.642   99900
4        84997149        rs10000017      0.223   2.043   99900
[  mturchin20@spudling70  ~]$join -a1 -a2  -o 0 1.2 2.2 -eNA nana1.txt nana2.txt 
1 2 NA
3 4 NA
5 6 6
7 8 8
9 NA 10
11 NA 12
[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/GPC2012]$zcat /data/external_public/GlobalLipids2010/AllPhenos_with_Effect.wHapMap23.expanded.formatted.tbl.gz | head -n 10
rs4747841       10      10000135        0.449   -0.822  99150   0.449   0.036   94704   0.449   0.202   99434   0.449   0.782   95848                                                                                           rs4749917       10      10000265        0.436   0.854   99150   0.436   0.04    94704   0.436   -0.151  99434   0.436   -0.801  95848                                                                                           rs737656        10      100002729       0.367   -2.122  99900   0.367   2.144   95454   0.367   1.337   100184  0.367   2.053   96598                                                                                           rs737657        10      100002880       0.358   -2.113  99900   0.358   1.826   95454   0.358   1.086   100184  0.358   1.958   96598
rs7086391       10      100003553       0.12    0.191   99900   0.12    -1.109  95454   0.12    -1.209  100184  0.12    -1.509  96598
rs878177        10      100003805       0.3     1.557   99900   0.3     -1.361  95454   0.3     -0.417  100184  0.3     -0.703  96598
rs878178        10      100003967       0.356   2.082   99900   0.356   -1.912  95454   0.356   -1.169  100184  0.356   -1.957  96598
rs12219605      10      10000459        0.427   0.839   99150   0.427   -0.006  94704   0.427   -0.165  99434   0.427   -0.779  95848
rs3763688       10      100005553       0.144   1.627   99900   0.144   0.819   95454   0.144   0.858   100184  0.144   -0.599  96598
rs3763689       10      100005593       0.217   -0.178  99900   0.217   -0.994  95454   0.217   -1.191  100184  0.217   -1.18   96598
[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/GPC2012]$zcat /data/external_public/GlobalLipids2010/AllPhenos_with_Effect.wHapMap23.expanded.formatted.tbl.gz | grep NA2
[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/GPC2012]$  
~~~

join -a 1 -a 2 -1 3 -2 3 -e NA2 -o 0 1.1 1.2 1.4 1.5 1.6 2.1 2.2 2.4 2.5 2.6 <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.formatted.tbl.gz | sort -g -k 3,3) <(zcat /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.formatted.tbl.gz | sort -g -k 3,3) | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
join -a 1 -a 2 -1 1 -2 3 -e NA2 -o 0 1.2 2.1 2.2 2.4 2.5 2.6 - <(zcat /data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.formatted.tbl.gz | sort -g -k 3,3) | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); print join("\t", @F);' | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); if ($#F == 6) { splice(@F, 1, 0, (("NA2") x 9)); } my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
join -a 1 -a 2 -1 1 -2 3 -e NA2 -o 0 1.2 2.1 2.2 2.4 2.5 2.6 - <(zcat /data/external_public/GlobalLipids2010/TG_with_Effect.wHapMap23.expanded.formatted.tbl.gz | sort -g -k 3,3) | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); print join("\t", @F);' | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); if ($#F == 6) { splice(@F, 1, 0, (("NA2") x 14)); } my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
perl -lane 'print $F[0], "\t", join("\t", split(/_/, $F[1]));' | gzip > /data/external_public/GlobalLipids2010/AllPhenos_with_Effect.wHapMap23.expanded.formatted.tbl.pre1.gz

zcat /data/external_public/GlobalLipids2010/AllPhenos_with_Effect.wHapMap23.expanded.formatted.tbl.pre1.gz | grep -v NA2 | gzip > /data/external_public/GlobalLipids2010/AllPhenos_with_Effect.wHapMap23.expanded.formatted.tbl.pre2.gz

zcat /data/external_public/GlobalLipids2010/AllPhenos_with_Effect.wHapMap23.expanded.formatted.tbl.gz | awk '{ if (($4 != $9) && ($4 != $14) && ($4 != $19)) { print $0 } } ' | wc
zcat /data/external_public/GlobalLipids2010/AllPhenos_with_Effect.wHapMap23.expanded.formatted.tbl.gz | awk '{ if (($4 != $9) || ($4 != $14) || ($4 != $19)) { print $0 } } ' | wc
zcat /data/external_public/GlobalLipids2010/AllPhenos_with_Effect.wHapMap23.expanded.formatted.tbl.gz | awk '{ if (($3 != $8) && ($3 != $13) && ($3 != $18)) { print $0 } } ' | wc
zcat /data/external_public/GlobalLipids2010/AllPhenos_with_Effect.wHapMap23.expanded.formatted.tbl.gz | awk '{ if (($3 != $8) || ($3 != $13) || ($3 != $18)) { print $0 } } ' | wc
zcat /data/external_public/GlobalLipids2010/AllPhenos_with_Effect.wHapMap23.expanded.formatted.tbl.gz | awk '{ if (($2 != $7) && ($2 != $12) && ($2 != $17)) { print $0 } } ' | wc
zcat /data/external_public/GlobalLipids2010/AllPhenos_with_Effect.wHapMap23.expanded.formatted.tbl.gz | awk '{ if (($2 != $7) || ($2 != $12) || ($2 != $17)) { print $0 } } ' | wc

#Lines of code while troubleshooting getting the above/below code working properly
~~~
[  mturchin20@spudling26  ~]$join -a 1 -a 2 -1 3 -2 3 -e NA2 -o 0 1.1 1.2 1.4 1.5 1.6 2.1 2.2 2.4 2.5 2.6 <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.formatted.tbl.gz | sort -g -k 3,3) <(zcat /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.formatted.tbl.gz | sort -g -k 3,3) | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | join -a 1 -a 2 -1 1 -2 3 -e NA2 -o 0 1.2 2.1 2.2 2.4 2.5 2.6 - <(zcat /data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.formatted.tbl.gz | sort -g -k 3,3) | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); if ($#F == 6) { splice(@F, 1, 0, (("NA2") x 9)); } print $#F;' | sort | uniq -c
2716910 15
[  mturchin20@spudling26  ~]$join -a 1 -a 2 -1 3 -2 3 -e NA2 -o 0 1.1 1.2 1.4 1.5 1.6 2.1 2.2 2.4 2.5 2.6 <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.formatted.tbl.gz | sort -g -k 3,3) <(zcat /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.formatted.tbl.gz | sort -g -k 3,3) | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | join -a 1 -a 2 -1 1 -2 3 -e NA2 -o 0 1.2 2.1 2.2 2.4 2.5 2.6 - <(zcat /data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.formatted.tbl.gz | sort -g -k 3,3) | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); print join("\t", @F);' | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); if ($#F == 6) { splice(@F, 1, 0, (("NA2") x 9)); } my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | perl -lane 'print $#F;' | sort | uniq -c 
2716910 1
[  mturchin20@spudling26  ~]$join -a 1 -a 2 -1 3 -2 3 -e NA2 -o 0 1.1 1.2 1.4 1.5 1.6 2.1 2.2 2.4 2.5 2.6 <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.formatted.tbl.gz | sort -g -k 3,3) <(zcat /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.formatted.tbl.gz | sort -g -k 3,3) | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | join -a 1 -a 2 -1 1 -2 3 -e NA2 -o 0 1.2 2.1 2.2 2.4 2.5 2.6 - <(zcat /data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.formatted.tbl.gz | sort -g -k 3,3) | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); print join("\t", @F);' | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); if ($#F == 6) { splice(@F, 1, 0, (("NA2") x 9)); } my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | join -a 1 -a 2 -1 1 -2 3 -e NA2 -o 0 1.2 2.1 2.2 2.4 2.5 2.6 - <(zcat /data/external_public/GlobalLipids2010/TG_with_Effect.wHapMap23.expanded.formatted.tbl.gz | sort -g -k 3,3) | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); print join("\t", @F);' | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); if ($#F == 6) { splice(@F, 1, 0, (("NA2") x 14)); } my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | perl -lane 'print $F[0], "\t", join("\t", split(/_/, $F[1]));' | perl -lane 'print $#F;' | sort | uniq -c
2718780 20
[  mturchin20@spudling26  ~]$zcat /data/external_public/GlobalLipids2010/AllPhenos_with_Effect.wHapMap23.expanded.formatted.tbl.pre.gz | wc
2718780 57094380 357916176
[  mturchin20@spudling26  ~]$zcat /data/external_public/GlobalLipids2010/AllPhenos_with_Effect.wHapMap23.expanded.formatted.tbl.gz | wc
2666628 55999188 352132352
[  mturchin20@spudling26  ~]$zcat /data/external_public/GlobalLipids2010/AllPhenos_with_Effect.wHapMap23.expanded.formatted.tbl.gz | awk '{ if (($4 != $9) && ($4 != $14) && ($4 != $19)) { print $0 } } ' | wc
    0       0       0
[  mturchin20@spudling26  ~]$zcat /data/external_public/GlobalLipids2010/AllPhenos_with_Effect.wHapMap23.expanded.formatted.tbl.gz | awk '{ if (($4 != $9) || ($4 != $14) || ($4 != $19)) { print $0 } } ' | wc
    0       0       0
[  mturchin20@spudling26  ~]$zcat /data/external_public/GlobalLipids2010/AllPhenos_with_Effect.wHapMap23.expanded.formatted.tbl.gz | awk '{ if (($3 != $8) && ($3 != $13) && ($3 != $18)) { print $0 } } ' | wc
      0       0       0
[  mturchin20@spudling26  ~]$zcat /data/external_public/GlobalLipids2010/AllPhenos_with_Effect.wHapMap23.expanded.formatted.tbl.gz | awk '{ if (($3 != $8) || ($3 != $13) || ($3 != $18)) { print $0 } } ' | wc
0       0       0
[  mturchin20@spudling26  ~]$zcat /data/external_public/GlobalLipids2010/AllPhenos_with_Effect.wHapMap23.expanded.formatted.tbl.gz | awk '{ if (($2 != $7) && ($2 != $12) && ($2 != $17)) { print $0 } } ' | wc
0       0       0
[  mturchin20@spudling26  ~]$zcat /data/external_public/GlobalLipids2010/AllPhenos_with_Effect.wHapMap23.expanded.formatted.tbl.gz | awk '{ if (($2 != $7) || ($2 != $12) || ($2 != $17)) { print $0 } } ' | wc
0       0       0
~~~

#20150919 CHECK_0: Only include SNPs which have full entries in all files?

zcat /data/external_public/GlobalLipids2010/AllPhenos_with_Effect.wHapMap23.expanded.formatted.tbl.pre2.gz | perl -lane 'splice(@F, 16, 3); splice(@F, 11, 3); splice(@F, 6, 3); print join("\t", @F);' | gzip > /data/external_public/GlobalLipids2010/AllPhenos_with_Effect.wHapMap23.expanded.formatted.tbl.gz

#201509__ -- Realized need to do 'annot' column so the below setup, with just joining the ...formatted.tbl.gz and ...dtlesssignif.vs1... files with their current states would not work
####join -a 2 -1 1 -2 1 -e NA3 -o 0 1.2 1.3 1.4 1.5 1.6 1.7 1.8 1.9 1.10 1.11 1.12 2.2 2.3 2.4 2.5 2.6 2.7 2.8 <(zcat /data/external_public/GlobalLipids2010/AllPhenos_with_Effect.wHapMap23.expanded.formatted.tbl.gz | sort -g -k 1,1 ) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs1/GlobalLipids2010.dtlesssignif.vs1.txt | perl -F, -lane 'print join("\t", @F);' | sort -g -k 1,1) | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs1/GlobalLipids2010.dtlesssignif.vs1.annot.txt.pre1.gz

####zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs1/GlobalLipids2010.dtlesssignif.vs1.annot.txt.pre1.gz | grep -v NA3 | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs1/GlobalLipids2010.dtlesssignif.vs1.annot.txt.gz

#~~~
#[  mturchin20@spudling26  ~]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs1/GlobalLipids2010.dtlesssignif.vs1.annot.txt.pre1.gz | wc
#8077  153463 1299269
#[  mturchin20@spudling26  ~]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs1/GlobalLipids2010.dtlesssignif.vs1.annot.txt.gz | wc
#7452  141588 1218351
#[  mturchin20@spudling26  ~]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs1/GlobalLipids2010.dtlesssignif.vs1.annot.txt.pre1.gz | grep NA3 | head -n 10
#rs1002687 NA3 NA3 NA3 NA3 NA3 NA3 NA3 NA3 NA3 NA3 NA3 3.03 8.32 13.088 13.471 301.068726650517 63.1957322003633 40.6350618925778
#rs1004712 NA3 NA3 NA3 NA3 NA3 NA3 NA3 NA3 NA3 NA3 NA3 -0.017 -0.266 -1.777 -5.073 30.2199529910874 5.35508450691846 6.4071656349204
#rs1005980 NA3 NA3 NA3 NA3 NA3 NA3 NA3 NA3 NA3 NA3 NA3 -0.289 4.089 5.321 3.477 32.4549125136963 5.81127355182767 6.98632702436835
#rs1006176 NA3 NA3 NA3 NA3 NA3 NA3 NA3 NA3 NA3 NA3 NA3 3.342 2.422 4.713 2.202 34.5523196114194 6.24105592525177 5.61243898676993
#rs1007205 NA3 NA3 NA3 NA3 NA3 NA3 NA3 NA3 NA3 NA3 NA3 -2.943 -8.392 -13.038 -13.411 297.597252539787 62.4469144731202 40.2829029824426
#rs1008420 NA3 NA3 NA3 NA3 NA3 NA3 NA3 NA3 NA3 NA3 NA3 -6.056 0.227 -2.363 0.389 42.6263057629326 7.90762374730948 8.85527637916135
#rs1009136 NA3 NA3 NA3 NA3 NA3 NA3 NA3 NA3 NA3 NA3 NA3 -0.036 -3.163 -5.115 -5.222 42.4818451591699 7.87766266851917 6.7520243551741
#rs1011685 NA3 NA3 NA3 NA3 NA3 NA3 NA3 NA3 NA3 NA3 NA3 20.552 -0.111 0.139 -22.515 699.42707840491 149.333707790455 111.528798405764
#rs10119 NA3 NA3 NA3 NA3 NA3 NA3 NA3 NA3 NA3 NA3 NA3 -2.495 0.012 1.675 5.531 36.3260514632445 6.60563776715105 7.49701285715082
#rs1012167 NA3 NA3 NA3 NA3 NA3 NA3 NA3 NA3 NA3 NA3 NA3 0.489 5.073 5.796 2.778 34.8951427338811 6.31144476812792 8.16803261162228
> vals1$X2
[1] 3 4
> string1 <- "X2"
> paste("vals1$", string1, sep="")
[1] "vals1$X2"
> eval(parse(text=paste("vals1$", string1, sep="")))
[1] 3 4
~~~

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/DataOrganization.GIANT2010.PythonVLookUp.CreateFullAnnotFile.vs1.py /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.PythonVLookUp.AnnotateGWASHits.vs1.py

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.PythonVLookUp.AnnotateGWASHits.vs1.py --file1 --file2 

#Go through and get top gwas hits of all studies that I'll be doing/using
#Then get this general-use script and create it for everyone...or go through everyone up until this point first and then go GWAS top hits?
#Go straight to magic, icbp, haemgen, gpc first?
#maybe just icbp and haemgen?






#20151007
#CSHL Poster work

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/multivariate/globallipids/GLC.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs1/GLC.MTedits.ForGlobalLipids2010.vs0.ForCSHLPoster.R 



#20161009
#Applying bmass while still in development via devtools (eg this tag that was later used/adopted: Vs2 -- Running things through version of bmass() code)

cd /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/bmass

git fetch --all

git reset --hard origin/master
#20170218 NOTE -- using the above command will remove any differences/changes in current directory from what you're pulling. It's less a merge than just a rewrite of the current directory's state with what you pulled. If looking for a merge, where say if the current directory has files that should be unaffected by the pull (eg changes in files that are not involved with the pull), then should figure out/use different command otherwise you're  going to lose those untracked/changed files in the current working directory so far

#While in R, can load a package using an absolute path via devtools like so:
~~~
library("devtools")
devtools::load_all("/Users/mturchin20/Documents/Work/LabMisc/StephensLab/bmass")
~~~

#ExpectedColumnNames=c("Chr", "BP", "MAF", "Direction", "pValue", "N")

#zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.tbl.gz | perl -lane 'my $direction; if ($F[4] > 0) { $direction = "+"; } elsif ($F[4] < 0) { $direction = "-"; } else { $direction = "NA"; } print $F[$#F-2], "\t", $F[$#F-1], "\t", $F[$#F], "\t", $F[1], "\t", $direction, "\t", $F[5], "\t", $F[3];' | grep -v Allele1 | cat <(echo -e "Chr\tBP\tMAF\tA1\tDirection\tpValue\tN") - | perl -lane 'if (gzip > /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.bmass.formatted.tbl.gz 
#zcat /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.tbl.gz | perl -lane 'my $direction; if ($F[4] > 0) { $direction = "+"; } elsif ($F[4] < 0) { $direction = "-"; } else { $direction = "NA"; } print $F[$#F-2], "\t", $F[$#F-1], "\t", $F[$#F], "\t", $F[1], "\t", $direction, "\t", $F[5], "\t", $F[3];' | grep -v Allele1 | cat <(echo -e "Chr\tBP\tMAF\tA1\tDirection\tpValue\tN") - | gzip > /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.bmass.formatted.tbl.gz 
#zcat /data/external_public/GlobalLipids2010/TG_with_Effect.wHapMap23.expanded.tbl.gz | perl -lane 'my $direction; if ($F[4] > 0) { $direction = "+"; } elsif ($F[4] < 0) { $direction = "-"; } else { $direction = "NA"; } print $F[$#F-2], "\t", $F[$#F-1], "\t", $F[$#F], "\t", $F[1], "\t", $direction, "\t", $F[5], "\t", $F[3];' | grep -v Allele1 | cat <(echo -e "Chr\tBP\tMAF\tA1\tDirection\tpValue\tN") - | gzip > /data/external_public/GlobalLipids2010/TG_with_Effect.wHapMap23.expanded.bmass.formatted.tbl.gz 
#zcat /data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.tbl.gz | perl -lane 'my $direction; if ($F[4] > 0) { $direction = "+"; } elsif ($F[4] < 0) { $direction = "-"; } else { $direction = "NA"; } print $F[$#F-2], "\t", $F[$#F-1], "\t", $F[$#F], "\t", $F[1], "\t", $direction, "\t", $F[5], "\t", $F[3];' | grep -v Allele1 | cat <(echo -e "Chr\tBP\tMAF\tA1\tDirection\tpValue\tN") - | gzip > /data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.bmass.formatted.tbl.gz 
zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.tbl.gz | perl -lane 'if ($F[6] eq "-") { ($F[1], $F[2]) = ($F[2], $F[1]); $F[6] = "+"; $F[$#F] = 1 - $F[$#F]; } print $F[$#F-2], "\t", $F[$#F-1], "\t", $F[$#F], "\t", $F[1], "\t", $F[2], "\t", $F[6], "\t", $F[5], "\t", $F[3]; ' | grep -v Allele1 | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | gzip > /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.tbl.gz
zcat /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.tbl.gz | perl -lane 'if ($F[6] eq "-") { ($F[1], $F[2]) = ($F[2], $F[1]); $F[6] = "+"; $F[$#F] = 1 - $F[$#F]; } print $F[$#F-2], "\t", $F[$#F-1], "\t", $F[$#F], "\t", $F[1], "\t", $F[2], "\t", $F[6], "\t", $F[5], "\t", $F[3]; ' | grep -v Allele1 | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | gzip > /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.tbl.gz
zcat /data/external_public/GlobalLipids2010/TG_with_Effect.wHapMap23.expanded.tbl.gz | perl -lane 'if ($F[6] eq "-") { ($F[1], $F[2]) = ($F[2], $F[1]); $F[6] = "+"; $F[$#F] = 1 - $F[$#F]; } print $F[$#F-2], "\t", $F[$#F-1], "\t", $F[$#F], "\t", $F[1], "\t", $F[2], "\t", $F[6], "\t", $F[5], "\t", $F[3]; ' | grep -v Allele1 | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | gzip > /data/external_public/GlobalLipids2010/TG_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.tbl.gz
zcat /data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.tbl.gz | perl -lane 'if ($F[6] eq "-") { ($F[1], $F[2]) = ($F[2], $F[1]); $F[6] = "+"; $F[$#F] = 1 - $F[$#F]; } print $F[$#F-2], "\t", $F[$#F-1], "\t", $F[$#F], "\t", $F[1], "\t", $F[2], "\t", $F[6], "\t", $F[5], "\t", $F[3]; ' | grep -v Allele1 | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | gzip > /data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.tbl.gz

#join <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.tbl.gz | awk '{ print $1 "_" $2 "\t" $4 }' | grep -v NA | sort -g -k 1,1) <(zcat /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.tbl.gz | awk '{ print $1 "_" $2 "\t" $0 }' | grep -v NA | sort -g -k 1,1) | perl -lane 'my $switch = "Same"; if ($F[1] ne $F[5]) { ($F[5], $F[6]) = ($F[6], $F[5]); $F[4] = 1 - $F[4]; if ($F[7] eq "-") { $F[7] = "+"; } elsif ($F[7] eq "+") { $F[7] = "-"; } else { $PH = 1; } $switch = "Switch"; } print join("\t", @F[2..$#F]), "\t", $switch;' | gzip > /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL.wSwitchTag.tbl.gz
join <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.tbl.gz | awk '{ print $1 "_" $2 "\t" $4 }' | grep -v NA | sort -g -k 1,1) <(zcat /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.tbl.gz | awk '{ print $1 "_" $2 "\t" $0 }' | grep -v NA | sort -g -k 1,1) | perl -lane 'if ($F[1] eq $F[6]) { ($F[5], $F[6]) = ($F[6], $F[5]); $F[4] = 1 - $F[4]; if ($F[7] eq "-") { $F[7] = "+"; } elsif ($F[7] eq "+") { $F[7] = "-"; } else { $PH = 1; } } elsif ($F[1] eq $F[5]) { $PH = 1; } else { push(@F, "NoAlleleMatch"); } print join("\t", @F[2..$#F]);' | gzip > /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL.tbl.gz
join <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.tbl.gz | awk '{ print $1 "_" $2 "\t" $4 }' | grep -v NA | sort -g -k 1,1) <(zcat /data/external_public/GlobalLipids2010/TG_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.tbl.gz | awk '{ print $1 "_" $2 "\t" $0 }' | grep -v NA | sort -g -k 1,1) | perl -lane 'if ($F[1] eq $F[6]) { ($F[5], $F[6]) = ($F[6], $F[5]); $F[4] = 1 - $F[4]; if ($F[7] eq "-") { $F[7] = "+"; } elsif ($F[7] eq "+") { $F[7] = "-"; } else { $PH = 1; } } elsif ($F[1] eq $F[5]) { $PH = 1; } else { push(@F, "NoAlleleMatch"); } print join("\t", @F[2..$#F]);' | gzip > /data/external_public/GlobalLipids2010/TG_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL.tbl.gz
join <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.tbl.gz | awk '{ print $1 "_" $2 "\t" $4 }' | grep -v NA | sort -g -k 1,1) <(zcat /data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.tbl.gz | awk '{ print $1 "_" $2 "\t" $0 }' | grep -v NA | sort -g -k 1,1) | perl -lane 'if ($F[1] eq $F[6]) { ($F[5], $F[6]) = ($F[6], $F[5]); $F[4] = 1 - $F[4]; if ($F[7] eq "-") { $F[7] = "+"; } elsif ($F[7] eq "+") { $F[7] = "-"; } else { $PH = 1; } } elsif ($F[1] eq $F[5]) { $PH = 1; } else { push(@F, "NoAlleleMatch"); } print join("\t", @F[2..$#F]);' | gzip > /data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL.tbl.gz

zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.tbl.gz | perl -lane 'if ($F[5] ne "0") { print join("\t", @F); }' | gzip > /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.No0Dir.tbl.gz 
zcat /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL.tbl.gz | grep -v NoAlleleMatch | perl -lane 'if ($F[5] ne "0") { print join("\t", @F); }' | gzip > /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.No0Dir.tbl.gz
zcat /data/external_public/GlobalLipids2010/TG_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL.tbl.gz | grep -v NoAlleleMatch | perl -lane 'if ($F[5] ne "0") { print join("\t", @F); }' | gzip > /data/external_public/GlobalLipids2010/TG_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.No0Dir.tbl.gz
zcat /data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL.tbl.gz | grep -v NoAlleleMatch | perl -lane 'if ($F[5] ne "0") { print join("\t", @F); }' | gzip > /data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.No0Dir.tbl.gz

~~~
[  mturchin20@spudling96  ~]$zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.tbl.gz | grep 100003553
10      100003553       0.12    t       c       +       0.8488  99900
[  mturchin20@spudling96  ~]$zcat /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.tbl.gz | grep 100003553
10      100003553       0.88    c       t       +       0.2675  95454
[  mturchin20@spudling96  ~]$join <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.tbl.gz | awk '{ print $1 "_" $2 "\t" $4 }' | sort) <(zcat /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.tbl.gz | awk '{ print $1 "_" $2 "\t" $0 }' | sort) | perl -lane 'if ($F[1] ne $F[5]) { ($F[5], $F[6]) = ($F[6], $F[5]); $F[4] = 1 - $F[4]; if ($F[7] eq "-") { $F[7] = "+"; } elsif ($F[7] eq "+") { $F[7] = "-"; } else { $PH = 1; } } print join("\t", @F[2..$#F]);' | grep 100003553
10      100003553       0.12    t       c       -       0.2675  95454
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3]$zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL.wSwitchTag.tbl.gz | perl -lane 'print $F[$#F];' | sort | uniq -c
1245182 Same
1379176 Switch
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3]$zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL.wSwitchTag.tbl.gz | grep Switch | head -n 15000 | tail -n 1
1       153074559       0.432   a       g       -       0.6879  95443   Switch
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3]$zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL.wSwitchTag.tbl.gz | grep Switch | head -n 150000 | tail -n 1
2       181018761       0.892   c       t       -       0.3692  95446   Switch
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3]$zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL.wSwitchTag.tbl.gz | grep Switch | head -n 1500000 | tail -n 1
22      49564000        0.442   a       g       -       0.6262  69309   Switch
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3]$zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL.wSwitchTag.tbl.gz | grep Switch | head -n 760000 | tail -n 1
8       97585747        0.958   t       a       -       0.7256  95454   Switch
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3]$zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL.wSwitchTag.tbl.gz | grep Switch | head -n 1000000 | tail -n 1
12      28635908        0.925   c       t       -       0.08854 95446   Switch
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3]$zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.tbl.gz | grep 153074559
1       153074559       0.432   a       g       +       0.005438        99888
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3]$zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.tbl.gz | grep 181018761
2       181018761       0.892   c       t       +       0.6379  99892
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3]$zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.tbl.gz | grep 49564000
22      49564000        0.442   a       g       +       0.4254  73432
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3]$zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.tbl.gz | grep 97585747
8       97585747        0.958   t       a       +       0.646   99900
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3]$zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.tbl.gz | grep 28635908
12      28635908        0.925   c       t       +       0.07426 99892
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3]$zcat /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.tbl.gz | grep 153074559
1       153074559       0.568   g       a       +       0.6879  95443
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3]$zcat /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.tbl.gz | grep 181018761
2       181018761       0.108   t       c       +       0.3692  95446
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3]$zcat /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.tbl.gz | grep 49564000 
22      49564000        0.558   g       a       +       0.6262  69309
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3]$zcat /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.tbl.gz | grep 97585747
8       97585747        0.042   a       t       +       0.7256  95454
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3]$zcat /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.tbl.gz | grep 28635908
12      28635908        0.075   t       c       +       0.08854 95446
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2]$join <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.tbl.gz | awk '{ print $1 "_" $2 "\t" $4 }' | grep -v NA | sort -g -k 1,1) <(zcat /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.tbl.gz | awk '{ print $1 "_" $2 "\t" $0 }' | grep -v NA | sort -g -k 1,1) | perl -lane 'if ($F[1] eq $F[6]) { ($F[5], $F[6]) = ($F[6], $F[5]); $F[4] = 1 - $F[4]; if ($F[7] eq "-") { $F[7] = "+"; } elsif ($F[7] eq "+") { $F[7] = "-"; } else { $PH = 1; } } elsif ($F[1] eq $F[5]) { $PH = 1; } else { push(@F, "NoAlleleMatch"); } print join("\t", @F[2..$#F]);' | grep NoAlleleMatch
1  33381798     0.008   a       g       +       0.9507  13015   NoAlleleMatch
2  180144902    0.167   t       c       +       0.9237  95446   NoAlleleMatch
2  180144902    0       a       g       +       0.6696  7776    NoAlleleMatch
2  209886487    0       a       t       +       0.6401  7776    NoAlleleMatch
2  209886487    0.517   c       a       +       0.1491  92495   NoAlleleMatch
4  70240740     1       g       a       +       0.2427  7776    NoAlleleMatch
4  70240740     0.683   c       t       +       0.3575  90533   NoAlleleMatch
5  168120106    0.00800000000000001     t       c       +       0.7115  12722   NoAlleleMatch
7  92339640     0.042   c       g       +       0.02041 89181   NoAlleleMatch
16 12215381     1       g       t       +       0.6599  2630    NoAlleleMatch
22 40969589     0       a       g       +       0.6822  7773    NoAlleleMatch
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2]$join <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.tbl.gz | awk '{ print $1 "_" $2 "\t" $4 }' | grep -v NA | sort -g -k 1,1) <(zcat /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.tbl.gz | awk '{ print $1 "_" $2 "\t" $0 }' | grep -v NA | sort -g -k 1,1) | perl -lane 'if ($F[1] eq $F[6]) { ($F[5], $F[6]) = ($F[6], $F[5]); $F[4] = 1 - $F[4]; if ($F[7] eq "-") { $F[7] = "+"; } elsif ($F[7] eq "+") { $F[7] = "-"; } else { $PH = 1; } } elsif ($F[1] eq $F[5]) { $PH = 1; } else { push(@F, "NoAlleleMatch"); } print join("\t", @F[2..$#F]);' | perl -lane 'print $#F;' | sort | uniq -c
2627146 7
     11 8
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2]$for i in `ls -lrt /data/external_public/GlobalLipids2010/*_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL.tbl.gz | awk '{ print $9 }'`; do echo $i; zcat $i | perl -lane 'print $#F;' | sort | uniq -c; done
/data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL.tbl.gz
2627146 7
     11 8
/data/external_public/GlobalLipids2010/TG_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL.tbl.gz
2627396 7
     12 8
/data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL.tbl.gz
2627528 7
      9 8
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2]$for i in `ls -lrt /data/external_public/GlobalLipids2010/*_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL* | awk '{ print $9 }'`; do echo $i; zcat $i | wc; done
/data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL.wSwitchTag.tbl.gz
2624358 23619222 112557168
/data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL.No0Dir.tbl.gz
2624227 20993816 96672892
/data/external_public/GlobalLipids2010/TG_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL.No0Dir.tbl.gz
2624466 20995728 96703993
/data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL.No0Dir.tbl.gz
2624603 20996824 97999391
/data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL.tbl.gz
2627157 21017267 96776029
/data/external_public/GlobalLipids2010/TG_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL.tbl.gz
2627408 21019276 96807521
/data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL.tbl.gz
2627537 21020305 98104076
/data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.No0Dir.tbl.gz
2627015 21016120 96771348
/data/external_public/GlobalLipids2010/TG_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.No0Dir.tbl.gz
2627253 21018024 96802408
/data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.No0Dir.tbl.gz
2627393 21019144 98099275
~~~

#HDL <- read.table("/data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.No0Dir.tbl.gz", header=T);
#LDL <- read.table("/data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL.No0Dir.tbl.gz", header=T);
#TG <- read.table("/data/external_public/GlobalLipids2010/TG_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL.No0Dir.tbl.gz", header=T);
#TC <- read.table("/data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL.No0Dir.tbl.gz", header=T);

R -q -e "library("devtools"); devtools::load_all("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/bmass"); 
HDL <- read.table("/data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.No0Dir.tbl.gz", header=T);
LDL <- read.table("/data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.No0Dir.tbl.gz", header=T);
TG <- read.table("/data/external_public/GlobalLipids2010/TG_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.No0Dir.tbl.gz", header=T);
TC <- read.table("/data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.No0Dir.tbl.gz", header=T);
GWASsnps <- read.table("/mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.txt", header=F);
names(GWASsnps) <- c("Chr", "BP")
#bmassOutput1 <- bmass(c("HDL", "LDL", "TG", "TC"), GWASsnps=GWASsnps, NminThreshold = 1, bmassSeedValue=150);
#print(bmassOutput1);" > shana1.txt
DataSources = c("HDL", "LDL", "TG", "TC"); NminThreshold = 50000; bmassSeedValue=150;
ExpectedColumnNames=c("Chr", "BP", "MAF", "Direction", "pValue", "N"); GWASsnps_AnnotateWindow = 5e5; SNPMarginalUnivariateThreshold = 1e-6; SNPMarginalMultivariateThreshold = 1e-6; SigmaAlphas = c(0.005,0.0075,0.01,0.015,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.15); ProvidedPriors=NULL; UseFlatPriors=FALSE; PruneMarginalSNPs=TRUE; PruneMarginalSNPs_bpWindow=5e5;

#bmassOutput1 <- bmass(c("bmass_TestData1", "bmass_TestData2"), GWASsnps=bmass_TestSigSNPs, NminThreshold = 2000, bmassSeedValue=NULL)

#From 20161227
bmassOutput1 <- bmass(c("HDL", "LDL", "TG", "TC"), GWASsnps=GWASsnps, NminThreshold = 50000, bmassSeedValue=150);
bmassOutput2 <- bmass(c("HDL", "LDL", "TG", "TC"), GWASsnps=GWASsnps, NminThreshold = 50000, bmassSeedValue=150, PruneMarginalSNPs=FALSE);
bmassOutput3 <- bmass(c("HDL", "LDL", "TG", "TC"), GWASsnps=GWASsnps, NminThreshold = 50000, bmassSeedValue=150, PrintMergedData=TRUE, PruneMarginalSNPs=FALSE);

write.table(bmassOutput1$NewSNPs$SNPs, file="/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.NewSNPs.SNPs.vs1.txt", quote=FALSE, row.names=FALSE)
##Haven't done these below yet -- should I? Eventually right, yeah?
##write.table(bmassOutput1$PreviousSNPs$SNPs, file="/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.PreviousSNPs.SNPs.vs1.txt", quote=FALSE, row.names=FALSE)
##write.table(bmassOutput1$MarginalSNPs$SNPs, file="/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MarginalSNPs.Pruned.SNPs.vs1.txt", quote=FALSE, row.names=FALSE)
##write.table(bmassOutput3$MarginalSNPs$SNPs, file="/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MarginalSNPs.NotPruned.SNPs.vs1.txt", quote=FALSE, row.names=FALSE)
write.table(bmassOutput3$MergedDataSources, file="/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.txt", quote=FALSE, row.names=FALSE)

system("gzip /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.txt")


#CommandLine:
~~~
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3]$zcat /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL.tbl.gz | perl -lane 'print $F[5];' | sort | uniq -c
1379112 -
1245114 +
    131 0
      1 Direction
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3]$zcat /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL.No0Dir.tbl.gz | perl -lane 'print $F[5];' | sort | uniq -c
1379112 -
1245114 +
      1 Direction

zcat /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL.tbl.gz | perl -lane 'if ($F[5] ne "0") { print join("\t", @F); }' | gzip > /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL.No0Dir.tbl.gz
~~~
#R console:
~~~
> library("devtools"); devtools::load_all("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/bmass");
Loading bmass
> HDL <- read.table("/data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.tbl.gz", header=T);
> LDL <- read.table("/data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL.tbl.gz", header=T);
> TG <- read.table("/data/external_public/GlobalLipids2010/TG_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL.tbl.gz", header=T);
> TC <- read.table("/data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL.tbl.gz", header=T);
> GWASsnps <- read.table("/mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.txt", header=F);
names(GWASsnps) <- c("Chr", "BP")
> bmassOutput1 <- bmass(c("HDL", "LDL", "TG", "TC"), GWASsnps=GWASsnps, NminThreshold = 1, bmassSeedValue=150);
Error in CheckIndividualDataSources(DataSources, GWASsnps, ExpectedColumnNames,  :
  2016-10-07 13:28:27 -- the following data sources have entries other than + and - in the Direction column. Please fix and rerun bmass: HDL LDL TG TC
> bmassOutput1 <- bmass(c("HDL", "LDL", "TG", "TC"), GWASsnps=GWASsnps, NminThreshold = 1, bmassSeedValue=150);
Error in if (GWASsnps1[snpIndex, ]$Chr == as.numeric(as.character(MergedDataSource1["Chr"]))) { :
.
.
.
>         bmassOutput <- list()
>         bmassOutput$MergedDataSources <- NULL
>         if (!is.null(MergedDataSources)) {
+                 bmassOutput$MergedDataSources <- MergedDataSources
+         }
Error: object 'MergedDataSources' not found
>         bmassOutput$ZScoresCorMatrix <- NULL
>         bmassOutput$MarginalSNPs <- list()
>         bmassOutput$Models <- NULL
>         bmassOutput$ModelPriors <- NULL
>         bmassOutput$PreviousSNPs <- list()
>         bmassOutput$NewSNPs <- list()
>         bmassOutput$GWASlogBFMinThreshold <- NULL
>         #LogFile <- c()
>         bmassOutput$LogFile <- c()
>
>         bmassOutput$LogFile <- rbind(bmassOutput$LogFile, paste(format(Sys.time()), " -- beginning bmass.", sep=""))
>
>                 bmassOutput$LogFile <- CheckIndividualDataSources(DataSources, GWASsnps, ExpectedColumnNames, SigmaAlphas, bmassOutput$MergedDataSources, ProvidedPriors, UseFlatPriors, PruneMarginalSNPs, PruneMarginalSNPs_bpWindow, SNPMarginalUnivariateThreshold, SNPMarginalMultivariateThreshold, NminThreshold, bmassSeedValue, bmassOutput$LogFile)
>
>
>                 bmassOutput[c("MergedDataSources", "LogFile")] <- MergeDataSources(DataSources, bmassOutput$LogFile)[c("MergedDataSources", "LogFile")]
>                 system.time(bmassOutput[c("MergedDataSources", "LogFile")] <- MergeDataSources(DataSources, bmassOutput$LogFile)[c("MergedDataSources", "LogFile")])
   user  system elapsed
   269.592   0.246 269.864
>
>         bmassOutput[c("MergedDataSources", "LogFile")] <- AnnotateMergedDataWithGWASSNPs(bmassOutput$MergedDataSources, GWASsnps, GWASsnps_AnnotateWindow, bmassOutput$LogFile)[c("MergedDataSources", "LogFile")]
>
>         bmassOutput[c("MarginalSNPs", "ZScoresCorMatrix", "LogFile")] <- ProcessMergedAndAnnotatedDataSources(DataSources, bmassOutput$MergedDataSources, SNPMarginalUnivariateThreshold, SNPMarginalMultivariateThreshold, bmassOutput$LogFile)[c("MarginalSNPs", "ZScoresCorMatrix", "LogFile")]
Warning message:
In ProcessMergedAndAnnotatedDataSources(DataSources, bmassOutput$MergedDataSources,  :
  2016-11-08 15:38:24 -- One of your datafiles has p-values less than the threshold which R can properly convert them to log-scale, meaning their log-values return as 'Infinite'. bmass automatically replaces these 'Infinite' values with the max, non-infinite Z-scores available, but it is recommended you self-check this as well.
>
>         bmassOutput[c("MarginalSNPs", "Models", "ModelPriors", "LogFile")] <- GetLogBFsFromData(DataSources, bmassOutput$MarginalSNPs, bmassOutput$ZScoresCorMatrix, SigmaAlphas, bmassOutput$LogFile)[c("MarginalSNPs", "Models", "ModelPriors", "LogFile")]
> 
> 
>         bmassOutput[c("MarginalSNPs", "PreviousSNPs", "ModelPriors", "GWASlogBFMinThreshold", "LogFile")] <- DetermineAndApplyPriors(DataSources, bmassOutput$MarginalSNPs, GWASsnps, SigmaAlphas, bmassOutput$Models, bmassOutput$ModelPriors, ProvidedPriors, UseFlatPriors, bmassSeedValue, bmassOutput$LogFile)[c("MarginalSNPs", "PreviousSNPs", "ModelPriors", "GWASlogBFMinThreshold", "LogFile")]
> 
>         bmassOutput[c("MarginalSNPs", "PreviousSNPs", "NewSNPs", "LogFile")] <- FinalizeAndFormatResults(DataSources, bmassOutput$MarginalSNPs, bmassOutput$PreviousSNPs, GWASsnps, bmassOutput$GWASlogBFMinThreshold, SigmaAlphas, bmassOutput$Models, bmassOutput$ModelPriors, NminThreshold, PruneMarginalSNPs, PruneMarginalSNPs_bpWindow, bmassOutput$LogFile)[c("MarginalSNPs", "PreviousSNPs", "NewSNPs", "LogFile")]
Error in if (x == 1) { : missing value where TRUE/FALSE needed
> head(bmassOutput$MergedDataSources)
         ChrBP Chr        BP   MAF A1 HDL_A2 HDL_Direction HDL_pValue HDL_N
1  10_10000135  10  10000135 0.551  g      a             +    0.41100 99150
2  10_10000265  10  10000265 0.436  t      c             +    0.39330 99150
3 10_100002729  10 100002729 0.633  g      a             +    0.03387 99900
4 10_100002880  10 100002880 0.642  g      a             +    0.03461 99900
5 10_100003553  10 100003553 0.120  t      c             +    0.84880 99900
6 10_100003805  10 100003805 0.300  t      c             +    0.11940 99900
  HDL_ZScore LDL_Direction LDL_pValue LDL_N  LDL_ZScore TG_Direction TG_pValue
1  0.8221351             -    0.97120 94704 -0.03610329            -   0.43440
2  0.8536481             +    0.96830 94704  0.03974052            -   0.42320
3  2.1216160             -    0.03202 95454 -2.14416086            -   0.04005
4  2.1128921             -    0.06781 95454 -1.82626736            -   0.05027
5  0.1906498             -    0.26750 95454 -1.10883817            -   0.13120
6  1.5572969             -    0.17340 95454 -1.36135984            -   0.48220
   TG_N  TG_ZScore TC_Direction TC_pValue   TC_N  TC_ZScore GWASannot
1 95848 -0.7816845            -    0.8399  99434 -0.2020214         0
2 95848 -0.8008819            -    0.8803  99434 -0.1505889         0
3 96598 -2.0532328            -    0.1812 100184 -1.3370694         0
4 96598 -1.9576593            -    0.2776 100184 -1.0857265         0
5 96598 -1.5093867            -    0.2265 100184 -1.2094240         0
6 96598 -0.7027685            -    0.6764 100184 -0.4173807         0
> table(is.na(bmassOutput$MergedDataSources$HDL_ZScore))

  FALSE 
2628101 
> table(is.na(bmassOutput$MergedDataSources$LDL_ZScore))

  FALSE 
2628101 
> table(is.na(bmassOutput$MergedDataSources$TG_ZScore))

  FALSE 
2628101 
> table(is.na(bmassOutput$MergedDataSources$TC_ZScore))

  FALSE 
2628101 
> PruneMarginalSNPs
[1] TRUE
> table(apply(bmassOutput$MarginalSNPs$logBFs, c(1,2), is.na))

  FALSE    TRUE 
9836876   39130 
> table(bmassOutput$MarginalSNPs$SNPs$MAF > .5)

FALSE  TRUE 
 4539  4170 
> head(bmassOutput$MarginalSNPs$SNPs[bmassOutput$MarginalSNPs$SNPs$MAF > .5,])
             ChrBP Chr        BP   MAF A1 HDL_A2 HDL_Direction HDL_pValue HDL_N
2237  10_101902054  10 101902054 0.533  c      t             +  8.152e-05 99702
15986 10_113892408  10 113892408 0.717  c      a             +  3.578e-07 99900
16002 10_113906301  10 113906301 0.538  g      a             +  8.961e-04 96908
16003 10_113907075  10 113907075 0.700  t      a             +  2.555e-07 96908
16006 10_113910455  10 113910455 0.576  g      a             +  2.172e-03 96908
16008 10_113911344  10 113911344 0.700  g      a             +  2.157e-07 96908
      HDL_ZScore LDL_Direction LDL_pValue LDL_N LDL_ZScore TG_Direction
2237    3.939887             +  5.060e-01 95261  0.6650789            +
15986   5.090140             +  2.221e-08 95454  5.5938413            -
16002   3.321266             +  7.160e-06 92503  4.4888759            -
16003   5.153624             +  3.819e-08 92503  5.4990236            -
16006   3.065646             +  2.647e-06 92503  4.6964670            -
16008   5.185272             +  5.466e-09 92503  5.8323260            -
.
.
.
#Edited bmass code and reran
>                 bmassOutput[c("MergedDataSources", "LogFile")] <- MergeDataSources(DataSources, bmassOutput$LogFile)[c("MergedDataSources", "LogFile")]
> table(bmassOutput$MergedDataSources$MAF > .5)

  FALSE 
2628101 
> head(HDL[HDL$MAF > .5,])
   Chr        BP   MAF A1 A2 Direction  pValue     N
1    7  92221824 0.967  c  a         + 0.92320 85853
2   12 125456933 0.627  g  a         + 0.34900 99900
4    4   1347325 0.808  g  c         + 0.45230 99843
8    4  95952929 0.592  g  t         + 0.99350 99900
9    4 156533822 1.000  g  c         + 0.49250  7807
11   3  99825597 0.875  g  a         + 0.06873 99743
> bmassOutput$MergedDataSources[bmassOutput$MergedDataSources$BP == 92221824,]
             ChrBP Chr       BP   MAF A1 HDL_A2 HDL_Direction HDL_pValue HDL_N
2342864 7_92221824   7 92221824 0.033  c      a             +     0.9232 85853
         HDL_ZScore LDL_Direction LDL_pValue LDL_N LDL_ZScore TG_Direction
2342864 -0.09640364             -    0.04027 81680   2.050969            -
        TG_pValue  TG_N TG_ZScore TC_Direction TC_pValue  TC_N TC_ZScore
2342864    0.4533 82546 0.7499248            -   0.02582 86136  2.228908
> bmassOutput$MergedDataSources[bmassOutput$MergedDataSources$BP == 125456933,]
              ChrBP Chr        BP   MAF A1 HDL_A2 HDL_Direction HDL_pValue
302329 12_125456933  12 125456933 0.373  g      a             +      0.349
       HDL_N HDL_ZScore LDL_Direction LDL_pValue LDL_N LDL_ZScore TG_Direction
302329 99900 -0.9365307             -     0.5077 95454  0.6624233            -
       TG_pValue  TG_N TG_ZScore TC_Direction TC_pValue   TC_N  TC_ZScore
302329    0.2134 96598  1.244271            -    0.9855 100184 0.01817406
> bmassOutput$MergedDataSources[bmassOutput$MergedDataSources$BP == 99825597,]
             ChrBP Chr       BP   MAF A1 HDL_A2 HDL_Direction HDL_pValue HDL_N
1669999 3_99825597   3 99825597 0.125  g      a             +    0.06873 99743
        HDL_ZScore LDL_Direction LDL_pValue LDL_N LDL_ZScore TG_Direction
1669999   -1.82019             +      0.316 95300  -1.002712            +
        TG_pValue  TG_N  TG_ZScore TC_Direction TC_pValue   TC_N TC_ZScore
1669999    0.5112 96437 -0.6569703            +    0.1468 100023 -1.450928
.
.
.
>         bmassOutput[c("MarginalSNPs", "PreviousSNPs", "NewSNPs", "LogFile")] <- FinalizeAndFormatResults(DataSources, bmassOutput$MarginalSNPs, bmassOutput$PreviousSNPs, GWASsnps, bmassOutput$GWASlogBFMinThreshold, SigmaAlphas, bmassOutput$Models, bmassOutput$ModelPriors, NminThreshold, PruneMarginalSNPs, PruneMarginalSNPs_bpWindow, bmassOutput$LogFile)[c("MarginalSNPs", "PreviousSNPs", "NewSNPs", "LogFile")]
Error in if (x == 1) { : missing value where TRUE/FALSE needed
>
> table(apply(bmassOutput$MarginalSNPs$logBFs, c(1,2), is.na))

  FALSE    TRUE
9477384   43680
> apply(bmassOutput$MarginalSNPs$logBFs[1:10,1:10], 2, is.na)
       [,1]  [,2]  [,3]  [,4]  [,5]  [,6]  [,7]  [,8]  [,9] [,10]
 [1,] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [2,] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [3,] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [4,] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [5,] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [6,] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [7,] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [8,] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [9,] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
[10,] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
> apply(apply(bmassOutput$MarginalSNPs$logBFs[1:10,1:10], c(1,2), is.na), 2, function(x) { return(TRUE %in% x); })
 [1] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
> length(apply(apply(bmassOutput$MarginalSNPs$logBFs, c(1,2), is.na), 2, function(x) { return(TRUE %in% x); }))
[1] 8396
> table(apply(apply(bmassOutput$MarginalSNPs$logBFs, c(1,2), is.na), 2, function(x) { return(TRUE %in% x); }))

FALSE  TRUE 
 8348    48 
> head(bmassOutput$MarginalSNPs$SNPs[apply(apply(bmassOutput$MarginalSNPs$logBFs, c(1,2), is.na), 2, function(x) { return(TRUE %in% x); }),])
              ChrBP Chr        BP MAF A1 HDL_A2 HDL_Direction HDL_pValue HDL_N
26172  10_122804658  10 122804658   0  g      a             +  4.195e-10  1456
97134   10_61078772  10  61078772   0  a      g             +  1.047e-09  1456
584250  14_86135518  14  86135518   0  g      c             +  7.364e-02  1456
603657  15_24151016  15  24151016   0  t      c             +  4.205e-02   711
619158  15_42033223  15  42033223   0  a      t             +  2.857e-06 98409
732176  16_80263472  16  80263472   0  c      t             +  2.269e-07  1456
       HDL_ZScore LDL_Direction LDL_pValue LDL_N LDL_ZScore TG_Direction
26172   -6.246594             +     0.8262  1375 -0.2195778            +
97134    6.102075             -     0.5842  1375 -0.5472602            +
584250  -1.788844             +     0.7190  1375 -0.3597956            -
603657   2.033025             +     0.1975  1101  1.2887069            -
619158   4.680841             +     0.7666 93999  0.2968252            -
732176  -5.175830             -     0.2270  1375  1.2081229            -
       TG_pValue  TG_N   TG_ZScore TC_Direction TC_pValue  TC_N  TC_ZScore
26172  9.551e-01  1458 -0.05630354            -   0.73470  1459  0.3388802
97134  3.055e-01  1458  1.02471009            -   0.72690  1459 -0.3492522
584250 3.671e-01  1458  0.90191881            -   0.04739  1459  1.9827980
603657 3.787e-01   716 -0.88029411            -   0.43580   710 -0.7793051
619158 1.631e-11 95070 -6.73573796            -   0.64100 98656 -0.4663011
732176 4.083e-02  1458  2.04525221            +   0.49080  1459 -0.6890370
       GWASannot   mvstat mvstat_log10pVal   unistat unistat_log10pVal  Nmin
26172          0 88.80556        17.626801 39.019935          9.377268  1375
97134          0 82.13563        16.211576 37.235323          8.980053  1375
584250         0 44.98902         8.398274  3.931488          1.324313  1375
603657         0 37.19156         6.783883  4.133191          1.376234   710
619158         1 54.07996        10.295536 45.370166         10.787546 93999
732176         2 28.98333         5.103553 26.789216          6.644166  1375
       logBFWeightedAvg
26172               NaN
97134               NaN
584250              NaN
603657              NaN
619158              NaN
732176              NaN
> head(HDL[HDL$BP == 42033223 & !is.na(HDL$BP),])
        Chr       BP   MAF A1 A2 Direction    pValue     N
207918   12 42033223 0.865  c  t         + 1.064e-01 99900
1502337  15 42033223 0.000  a  t         + 2.857e-06 98409
> head(LDL[LDL$BP == 42033223,])
        Chr       BP   MAF A1 A2 Direction pValue     N
1909876  12 42033223 0.865  c  t         - 0.1670 95454
2183114  15 42033223 0.000  a  t         + 0.7666 93999
> head(TG[TG$BP == 42033223,])
        Chr       BP   MAF A1 A2 Direction    pValue     N
1910055  12 42033223 0.865  c  t         - 7.502e-01 96598
2183316  15 42033223 0.000  a  t         - 1.631e-11 95070
> head(TC[TC$BP == 42033223,])
        Chr       BP   MAF A1 A2 Direction pValue      N
1910149  12 42033223 0.865  c  t         - 0.4864 100184
2183421  15 42033223 0.000  a  t         - 0.6410  98656
#From /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.No0Dir.tbl.gz 
#.
#18      63418292        0       t       c       +       0.1144  1456
#15      42033223        0       a       t       +       2.857e-06       98409
#9       7650299 0       a       g       +       0.8839  8581
#.
.
.
.
> table(bmassOutput$MergedDataSources$GWASannot)

      0       1       2 
2513953      99   84205 
.
.
.
> library("devtools"); devtools::load_all("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/bmass");
txt
DataSources = c("HDL", "LDL", "TG", "TC"); NminThreshold = 50000; bmassSeedValue=150;
ExpectedColumnNames=c("Chr", "BP", "MAF", "Direction", "pValue", "N"); GWASsnps_AnnotateWindow = 5e5; SNPMarginalUnivariateThreshold = 1e-6; SNPMarginalMultivariateThreshold = 1e-6; SigmaAlphas = c(0.005,0.0075,0.01,0.015,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.15); ProvidedPriors=NULL; UseFlatPriors=FALSE; PruneMarginalSNPs=TRUE; PruneMarginalSNPs_bpWindow=5e5;Loading bmass
> HDL <- read.table("/data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.No0Dir.tbl.gz", header=T);
> LDL <- read.table("/data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL.No0Dir.tbl.gz", header=T);
> TG <- read.table("/data/external_public/GlobalLipids2010/TG_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL.No0Dir.tbl.gz", header=T);
> TC <- read.table("/data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL.No0Dir.tbl.gz", header=T);
> GWASsnps <- read.table("/mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.txt", header=F);
> names(GWASsnps) <- c("Chr", "BP")
> #bmassOutput1 <- bmass(c("HDL", "LDL", "TG", "TC"), GWASsnps=GWASsnps, NminThreshold = 1, bmassSeedValue=150);
> #print(bmassOutput1);" > shana1.txt
> DataSources = c("HDL", "LDL", "TG", "TC"); NminThreshold = 50000; bmassSeedValue=150;
> ExpectedColumnNames=c("Chr", "BP", "MAF", "Direction", "pValue", "N"); GWASsnps_AnnotateWindow = 5e5; SNPMarginalUnivariateThreshold = 1e-6; SNPMarginalMultivariateThreshold = 1e-6; SigmaAlphas = c(0.005,0.0075,0.01,0.015,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.15); ProvidedPriors=NULL; UseFlatPriors=FALSE; PruneMarginalSNPs=TRUE; PruneMarginalSNPs_bpWindow=5e5;
>
>         bmassOutput <- list()
>         bmassOutput$MergedDataSources <- NULL
>         if (!is.null(MergedDataSources)) {
+                 bmassOutput$MergedDataSources <- MergedDataSources
+         }
Error: object 'MergedDataSources' not found
>         bmassOutput$ZScoresCorMatrix <- NULL
>         bmassOutput$MarginalSNPs <- list()
>         bmassOutput$Models <- NULL
>         bmassOutput$ModelPriors <- NULL
>         bmassOutput$PreviousSNPs <- list()
>         bmassOutput$NewSNPs <- list()
>         bmassOutput$GWASlogBFMinThreshold <- NULL
>         #LogFile <- c()
>         bmassOutput$LogFile <- c()
>
>         bmassOutput$LogFile <- rbind(bmassOutput$LogFile, paste(format(Sys.time()), " -- beginning bmass.", sep=""))
>
>                 bmassOutput$LogFile <- CheckIndividualDataSources(DataSources, GWASsnps, ExpectedColumnNames, SigmaAlphas, bmassOutput$MergedDataSources, ProvidedPriors, UseFlatPriors, PruneMarginalSNPs, PruneMarginalSNPs_bpWindow, SNPMarginalUnivariateThreshold, SNPMarginalMultivariateThreshold, NminThreshold, bmassSeedValue, bmassOutput$LogFile)
>
>                 bmassOutput[c("MergedDataSources", "LogFile")] <- MergeDataSources(DataSources, bmassOutput$LogFile)[c("MergedDataSources", "LogFile")]
> 
>         bmassOutput[c("MergedDataSources", "LogFile")] <- AnnotateMergedDataWithGWASSNPs(bmassOutput$MergedDataSources, GWASsnps, GWASsnps_AnnotateWindow, bmassOutput$LogFile)[c("MergedDataSources", "LogFile")]
> 
>         bmassOutput[c("MarginalSNPs", "ZScoresCorMatrix", "LogFile")] <- ProcessMergedAndAnnotatedDataSources(DataSources, bmassOutput$MergedDataSources, SNPMarginalUnivariateThreshold, SNPMarginalMultivariateThreshold, bmassOutput$LogFile)[c("MarginalSNPs", "ZScoresCorMatrix", "LogFile")]
Warning message:
In ProcessMergedAndAnnotatedDataSources(DataSources, bmassOutput$MergedDataSources,  :
  2016-11-09 15:41:26 -- One of your datafiles has p-values less than the threshold which R can properly convert them to log-scale, meaning their log-values return as 'Infinite'. bmass automatically replaces these 'Infinite' values with the max, non-infinite Z-scores available, but it is recommended you self-check this as well.
> 
>         bmassOutput[c("MarginalSNPs", "Models", "ModelPriors", "LogFile")] <- GetLogBFsFromData(DataSources, bmassOutput$MarginalSNPs, bmassOutput$ZScoresCorMatrix, SigmaAlphas, bmassOutput$LogFile)[c("MarginalSNPs", "Models", "ModelPriors", "LogFile")]
> 
>         bmassOutput[c("MarginalSNPs", "PreviousSNPs", "ModelPriors", "GWASlogBFMinThreshold", "LogFile")] <- DetermineAndApplyPriors(DataSources, bmassOutput$MarginalSNPs, GWASsnps, SigmaAlphas, bmassOutput$Models, bmassOutput$ModelPriors, ProvidedPriors, UseFlatPriors, bmassSeedValue, bmassOutput$LogFile)[c("MarginalSNPs", "PreviousSNPs", "ModelPriors", "GWASlogBFMinThreshold", "LogFile")]
> 
>         bmassOutput[c("MarginalSNPs", "PreviousSNPs", "NewSNPs", "LogFile")] <- FinalizeAndFormatResults(DataSources, bmassOutput$MarginalSNPs, bmassOutput$PreviousSNPs, GWASsnps, bmassOutput$GWASlogBFMinThreshold, SigmaAlphas, bmassOutput$Models, bmassOutput$ModelPriors, NminThreshold, PruneMarginalSNPs, PruneMarginalSNPs_bpWindow, bmassOutput$LogFile)[c("MarginalSNPs", "PreviousSNPs", "NewSNPs", "LogFile")]
> 
> dim(bmassOutput$NewSNPs$SNPs)
[1] 27 29
> bmassOutput$GWASlogBFMinThreshold
[1] 4.341153
> bmassOutput1 <- bmass(c("HDL", "LDL", "TG", "TC"), GWASsnps=GWASsnps, NminThreshold = 50000, bmassSeedValue=150);

Warning message:
In ProcessMergedAndAnnotatedDataSources(DataSources, bmassOutput$MergedDataSources,  :
  2016-11-09 15:51:54 -- One of your datafiles has p-values less than the threshold which R can properly convert them to log-scale, meaning their log-values return as 'Infinite'. bmass automatically replaces these 'Infinite' values with the max, non-infinite Z-scores available, but it is recommended you self-check this as well.
>
> summary(bmassOutput1)
                      Length Class  Mode
MarginalSNPs             4   -none- list
PreviousSNPs             4   -none- list
NewSNPs                  3   -none- list
LogFile                 18   -none- character
ZScoresCorMatrix        16   -none- numeric
Models                 324   -none- numeric
ModelPriors           1134   -none- numeric
GWASlogBFMinThreshold    1   -none- numeric
> dim(bmassOutput1$NewSNPs$SNPs)
[1] 27 29
> bmassOutput1$GWASlogBFMinThreshold
[1] 4.341153
.
.
.
#After dropping non-matching SNPs
> dim(bmassOutput1$NewSNPs$SNPs)
[1] 26 29
> bmassOutput1$GWASlogBFMinThreshold
[1] 4.34246
.
.
.
#20161227 NOTE -- continuing work from seen above
> devtools::load_all("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/bmass");
> HDL <- read.table("/data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.No0Dir.tbl.gz", header=T);
> LDL <- read.table("/data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.No0Dir.tbl.gz", header=T);
> TG <- read.table("/data/external_public/GlobalLipids2010/TG_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.No0Dir.tbl.gz", header=T);
> TC <- read.table("/data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.No0Dir.tbl.gz", header=T);
> GWASsnps <- read.table("/mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.txt", header=F);
> names(GWASsnps) <- c("Chr", "BP")
> #bmassOutput1 <- bmass(c("HDL", "LDL", "TG", "TC"), GWASsnps=GWASsnps, NminThreshold = 1, bmassSeedValue=150);
> #print(bmassOutput1);" > shana1.txt
> DataSources = c("HDL", "LDL", "TG", "TC"); NminThreshold = 50000; bmassSeedValue=150;
> ExpectedColumnNames=c("Chr", "BP", "MAF", "Direction", "pValue", "N"); GWASsnps_AnnotateWindow = 5e5; SNPMarginalUnivariateThreshold = 1e-6; SNPMarginalMultivariateThreshold = 1e-6; SigmaAlphas = c(0.005,0.0075,0.01,0.015,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.15); ProvidedPriors=NULL; UseFlatPriors=FALSE; PruneMarginalSNPs=TRUE; PruneMarginalSNPs_bpWindow=5e5;
> 
> bmassOutput1 <- bmass(c("HDL", "LDL", "TG", "TC"), GWASsnps=GWASsnps, NminThreshold = 50000, bmassSeedValue=150);
> bmassOutput2 <- bmass(c("HDL", "LDL", "TG", "TC"), GWASsnps=GWASsnps, NminThreshold = 50000, bmassSeedValue=150, PruneMarginalSNPs=FALSE);
#20170105
> bmassOutput3 <- bmass(c("HDL", "LDL", "TG", "TC"), GWASsnps=GWASsnps, NminThreshold = 50000, bmassSeedValue=150, PrintMergedData=TRUE, PruneMarginalSNPs=FALSE);
>
> write.table(bmassOutput1$NewSNPs$SNPs, file="/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.NewSNPs.SNPs.vs1.txt", quote=FALSE, row.names=FALSE)
> write.table(bmassOutput3$MergedDataSources, file="/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.txt", quote=FALSE, row.names=FALSE)
>
> system("gzip /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.txt")
~~~

mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/multivariate/globallipids/dtlesssignif.annot.txt /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/20161109_edits_dtlesssignif.annot.txt

#Put below header into /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/20161109_edits_dtlesssignif.annot.txt
#SNP Chr BP MAF HDL_N HDL_Beta HDL_SE LDL_N LDL_Beta LDL_SE TC_N TC_Beta TC_SE TG_N TG_Beta TG_SE GWASannot Gene RS TG_ZScore TC_ZScore LDL_ZScore HDL_ZScore mvstat mvstat_log10pVal unistat_log10pVal

R code to use below
~~~

	library("devtools"); devtools::load_all("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/bmass");

        bmassOutput <- list()
        bmassOutput$MergedDataSources <- NULL
        if (!is.null(MergedDataSources)) {
                bmassOutput$MergedDataSources <- MergedDataSources
        }
        bmassOutput$ZScoresCorMatrix <- NULL
        bmassOutput$MarginalSNPs <- list()
        bmassOutput$Models <- NULL
        bmassOutput$ModelPriors <- NULL
        bmassOutput$PreviousSNPs <- list()
        bmassOutput$NewSNPs <- list()
        bmassOutput$GWASlogBFMinThreshold <- NULL
        #LogFile <- c()
        bmassOutput$LogFile <- c()

        bmassOutput$LogFile <- rbind(bmassOutput$LogFile, paste(format(Sys.time()), " -- beginning bmass.", sep=""))

	bmassOutput$MergedDataSources <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/20161109_edits_dtlesssignif.annot.txt", header=TRUE)
	bmassOutput$ZScoresCorMatrix <- as.matrix(read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/multivariate/globallipids/RSS0.txt", header=TRUE, sep=",")) 
	GWASsnps <- read.table("/mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.txt", header=F);
	names(GWASsnps) <- c("Chr", "BP")
#	DataSources = c("HDL", "LDL", "TG", "TC"); 
	DataSources = c("TC", "TG", "HDL", "LDL"); NminThreshold = 50000; bmassSeedValue=150;
	ExpectedColumnNames=c("Chr", "BP", "MAF", "Direction", "pValue", "N"); GWASsnps_AnnotateWindow = 5e5; SNPMarginalUnivariateThreshold = 1e-6; SNPMarginalMultivariateThreshold = 1e-6; SigmaAlphas = c(0.005,0.0075,0.01,0.015,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.15); ProvidedPriors=NULL; UseFlatPriors=FALSE; PruneMarginalSNPs=TRUE; PruneMarginalSNPs_bpWindow=5e5;
	bmassOutput$MergedDataSources$ChrBP <- paste(bmassOutput$MergedDataSources$Chr, "_", bmassOutput$MergedDataSources$BP, sep="")

	bmassOutput$MarginalSNPs$SNPs <- bmassOutput$MergedDataSources[bmassOutput$MergedDataSources$mvstat_log10pVal > -log10(SNPMarginalUnivariateThreshold) | bmassOutput$MergedDataSources$unistat_log10pVal > -log10(SNPMarginalMultivariateThreshold),]

        bmassOutput[c("MarginalSNPs", "Models", "ModelPriors", "LogFile")] <- GetLogBFsFromData(DataSources, bmassOutput$MarginalSNPs, bmassOutput$ZScoresCorMatrix, SigmaAlphas, bmassOutput$LogFile)[c("MarginalSNPs", "Models", "ModelPriors", "LogFile")]

#	Nmins <- apply(bmassOutput$MarginalSNPs$SNPs[,c("TC_N", "TG_N", "LDL_N", "HDL_N")], 1, min)
#	ZScores <- bmassOutput$MarginalSNPs$SNPs[,c("TC_ZScore", "TG_ZScore", "HDL_ZScore", "LDL_ZScore")]
#	MarginalSNPs_logBFs <- compute.allBFs.fromZscores(ZScores, bmassOutput$ZScoresCorMatrix, Nmins, bmassOutput$MarginalSNPs$SNPs$MAF, SigmaAlphas)

        bmassOutput[c("MarginalSNPs", "PreviousSNPs", "ModelPriors", "GWASlogBFMinThreshold", "LogFile")] <- DetermineAndApplyPriors(DataSources, bmassOutput$MarginalSNPs, GWASsnps, SigmaAlphas, bmassOutput$Models, bmassOutput$ModelPriors, ProvidedPriors, UseFlatPriors, bmassSeedValue, bmassOutput$LogFile)[c("MarginalSNPs", "PreviousSNPs", "ModelPriors", "GWASlogBFMinThreshold", "LogFile")]
	
	write.table(bmassOutput$MarginalSNPs$SNPs, file="/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.MarginalSNPs.NotPruned.SNPs.vs1.txt", quote=FALSE, row.names=FALSE) #20170222

        bmassOutput[c("MarginalSNPs", "PreviousSNPs", "NewSNPs", "LogFile")] <- FinalizeAndFormatResults(DataSources, bmassOutput$MarginalSNPs, bmassOutput$PreviousSNPs, GWASsnps, bmassOutput$GWASlogBFMinThreshold, SigmaAlphas, bmassOutput$Models, bmassOutput$ModelPriors, NminThreshold, PruneMarginalSNPs, PruneMarginalSNPs_bpWindow, bmassOutput$LogFile)[c("MarginalSNPs", "PreviousSNPs", "NewSNPs", "LogFile")]

	write.table(bmassOutput$NewSNPs$SNPs, file="/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.txt", quote=FALSE, row.names=FALSE)
	#20170222
	write.table(bmassOutput$PreviousSNPs$SNPs, file="/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.PreviousSNPs.SNPs.vs1.txt", quote=FALSE, row.names=FALSE)
	write.table(bmassOutput$MarginalSNPs$SNPs, file="/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.MarginalSNPs.Pruned.SNPs.vs1.txt", quote=FALSE, row.names=FALSE)
	

~~~

#Output from above
~~~
>         library("devtools"); devtools::load_all("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/bmass");
oresCorMatrix <- as.matrix(read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/multivariate/globallipids/RSS0.txt", header=TRUE, sep=","))
        GWASsnps <- read.table("/mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.txt", header=F);
        names(GWASsnps) <- c("Chr", "BP")
#       DataSources = c("HDL", "LDL", "TG", "TC");
        DataSources = c("TC", "TG", "HDL", "LDL"); NminThreshold = 50000; bmassSeedValue=150;
        ExpectedColumnNames=c("Chr", "BP", "MAF", "Direction", "pValue", "N"); GWASsnps_AnnotateWindow = 5e5; SNPMarginalUnivariateThreshold = 1e-6; SNPMarginalMultivariateThreshold = 1e-6; SigmaAlphas = c(0.005,0.0075,0.01,0.015,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.15); ProvidedPriors=NULL; UseFlatPriors=FALSE; PruneMarginalSNPs=TRUE; PruneMarginalSNPs_bpWindow=5e5;
        bmassOutput$MergedDataSources$ChrBP <- paste(bmassOutput$MergedDataSources$Chr, "_", bmassOutput$MergedDataSources$BP, sep="")
Loading bmass
>         
>         bmassOutput <- list()
>         bmassOutput$MergedDataSources <- NULL
>         if (!is.null(MergedDataSources)) {
+                 bmassOutput$MergedDataSources <- MergedDataSources
+         }       
Error: object 'MergedDataSources' not found
>         bmassOutput$ZScoresCorMatrix <- NULL
>         bmassOutput$MarginalSNPs <- list()
>         bmassOutput$Models <- NULL
>         bmassOutput$ModelPriors <- NULL
>         bmassOutput$PreviousSNPs <- list()
>         bmassOutput$NewSNPs <- list()
>         bmassOutput$GWASlogBFMinThreshold <- NULL
>         #LogFile <- c()
>         bmassOutput$LogFile <- c()
>         
>         bmassOutput$LogFile <- rbind(bmassOutput$LogFile, paste(format(Sys.time()), " -- beginning bmass.", sep=""))
>         
>         bmassOutput$MergedDataSources <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/20161109_edits_dtlesssignif.annot.txt", header=TRUE)
>         bmassOutput$ZScoresCorMatrix <- as.matrix(read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/multivariate/globallipids/RSS0.txt", header=TRUE, sep=","))
>         GWASsnps <- read.table("/mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.txt", header=F);
>         names(GWASsnps) <- c("Chr", "BP")
> #       DataSources = c("HDL", "LDL", "TG", "TC");
>         DataSources = c("TC", "TG", "HDL", "LDL"); NminThreshold = 50000; bmassSeedValue=150;
>         ExpectedColumnNames=c("Chr", "BP", "MAF", "Direction", "pValue", "N"); GWASsnps_AnnotateWindow = 5e5; SNPMarginalUnivariateThreshold = 1e-6; SNPMarginalMultivariateThreshold = 1e-6; SigmaAlphas = c(0.005,0.0075,0.01,0.015,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.15); ProvidedPriors=NULL; UseFlatPriors=FALSE; PruneMarginalSNPs=TRUE; PruneMarginalSNPs_bpWindow=5e5;
>         bmassOutput$MergedDataSources$ChrBP <- paste(bmassOutput$MergedDataSources$Chr, "_", bmassOutput$MergedDataSources$BP, sep="")
>
>         bmassOutput$MarginalSNPs$SNPs <- bmassOutput$MergedDataSources[bmassOutput$MergedDataSources$mvstat_log10pVal > -log10(SNPMarginalUnivariateThreshold) | bmassOutput$MergedDataSources$unistat_log10pVal > -log10(SNPMarginalMultivariateThreshold),]
>
>         bmassOutput[c("MarginalSNPs", "Models", "ModelPriors", "LogFile")] <- GetLogBFsFromData(DataSources, bmassOutput$MarginalSNPs, bmassOutput$ZScoresCorMatrix, SigmaAlphas, bmassOutput$LogFile)[c("MarginalSNPs", "Models", "ModelPriors", "LogFile")]
>
> 
>         bmassOutput[c("MarginalSNPs", "PreviousSNPs", "ModelPriors", "GWASlogBFMinThreshold", "LogFile")] <- DetermineAndApplyPriors(DataSources, bmassOutput$MarginalSNPs, GWASsnps, SigmaAlphas, bmassOutput$Models, bmassOutput$ModelPriors, ProvidedPriors, UseFlatPriors, bmassSeedValue, bmassOutput$LogFile)[c("MarginalSNPs", "PreviousSNPs", "ModelPriors", "GWASlogBFMinThreshold", "LogFile")]
> 
>         bmassOutput[c("MarginalSNPs", "PreviousSNPs", "NewSNPs", "LogFile")] <- FinalizeAndFormatResults(DataSources, bmassOutput$MarginalSNPs, bmassOutput$PreviousSNPs, GWASsnps, bmassOutput$GWASlogBFMinThreshold, SigmaAlphas, bmassOutput$Models, bmassOutput$ModelPriors, NminThreshold, PruneMarginalSNPs, PruneMarginalSNPs_bpWindow, bmassOutput$LogFile)[c("MarginalSNPs", "PreviousSNPs", "NewSNPs", "LogFile")]
> 
> dim(bmassOutput$NewSNPs$SNPs)
[1] 19 29
> bmassOutput$GWASlogBFMinThreshold
[1] 4.352589
#20170223
> bmassOutput$ZScoresCorMatrix
          Z.tc       Z.tg       Z.hdl       Z.ldl
[1,] 1.0000000  0.3237916  0.16898079  0.83938874
[2,] 0.3237916  1.0000000 -0.36682333  0.18159499
[3,] 0.1689808 -0.3668233  1.00000000 -0.07931303
[4,] 0.8393887  0.1815950 -0.07931303  1.00000000
> dim(bmassOutput$PreviousSNPs$SNPs)
[1] 102  29
> dim(GWASsnps)
[1] 101   2
#From /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs1/process.MTedits.ForGlobalLipids2010.vs1.R 
##Z.LDL       Z.HDL      Z.TC       Z.TG
##Z.LDL  1.00000000 -0.07931303 0.8393887  0.1815950
##Z.HDL -0.07931303  1.00000000 0.1689808 -0.3668233
##Z.TC   0.83938874  0.16898079 1.0000000  0.3237916
##Z.TG   0.18159499 -0.36682333 0.3237916  1.0000000
#From /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs1/GLC.MTedits.ForGlobalLipids2010.vs0.ForCSHLPoster.R  
#min(lbf.av.glhits)
##[1] 4.349679
#cbind(as.character(newhits$snp),bestclass,apply(pp.newhits.classmatrix,2,max))
##                   bestclass
##  [1,] "rs12739698" "1"       "0.811711793061224"
#.
#.
#.
## [19,] "rs10422101" "1"       "0.898137410416297"
~~~


#20170106
#Found some duplicate entries in the original 2010 files due to use of HapMap references to get Chr & BP information per rsID#
#These appear to only be a handful of double entries and a number of 'NAs' that end up getting super-duplicated due to use of join statements. These 'NAs' are all eventually removed -- and so therefore likely haven't affected analyses up until this point -- but have probably messed with interim reports of total line counts from files

join <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.tbl.gz | perl -lane 'print $F[$#F-2], "_", $F[$#F-1];' | grep -v Allele1 | sort | uniq -u) <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.tbl.gz | perl -lane 'if ($F[6] eq "-") { ($F[1], $F[2]) = ($F[2], $F[1]); $F[6] = "+"; $F[$#F] = 1 - $F[$#F]; } print $F[$#F-2], "_", $F[$#F-1], "\t", $F[$#F-2], "\t", $F[$#F-1], "\t", $F[$#F], "\t", $F[1], "\t", $F[2], "\t", $F[6], "\t", $F[5], "\t", $F[3]; ' | grep -v Allele1 | sort -k 1,1) | perl -lane 'splice(@F, 0, 1); print join("\t", @F);' | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | gzip > /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.tbl.gz
join <(zcat /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.tbl.gz | perl -lane 'print $F[$#F-2], "_", $F[$#F-1];' | grep -v Allele1 | sort | uniq -u) <(zcat /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.tbl.gz | perl -lane 'if ($F[6] eq "-") { ($F[1], $F[2]) = ($F[2], $F[1]); $F[6] = "+"; $F[$#F] = 1 - $F[$#F]; } print $F[$#F-2], "_", $F[$#F-1], "\t", $F[$#F-2], "\t", $F[$#F-1], "\t", $F[$#F], "\t", $F[1], "\t", $F[2], "\t", $F[6], "\t", $F[5], "\t", $F[3]; ' | grep -v Allele1 | sort -k 1,1) | perl -lane 'splice(@F, 0, 1); print join("\t", @F);' | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | gzip > /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.tbl.gz
join <(zcat /data/external_public/GlobalLipids2010/TG_with_Effect.wHapMap23.expanded.tbl.gz | perl -lane 'print $F[$#F-2], "_", $F[$#F-1];' | grep -v Allele1 | sort | uniq -u) <(zcat /data/external_public/GlobalLipids2010/TG_with_Effect.wHapMap23.expanded.tbl.gz | perl -lane 'if ($F[6] eq "-") { ($F[1], $F[2]) = ($F[2], $F[1]); $F[6] = "+"; $F[$#F] = 1 - $F[$#F]; } print $F[$#F-2], "_", $F[$#F-1], "\t", $F[$#F-2], "\t", $F[$#F-1], "\t", $F[$#F], "\t", $F[1], "\t", $F[2], "\t", $F[6], "\t", $F[5], "\t", $F[3]; ' | grep -v Allele1 | sort -k 1,1) | perl -lane 'splice(@F, 0, 1); print join("\t", @F);' | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | gzip > /data/external_public/GlobalLipids2010/TG_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.tbl.gz
join <(zcat /data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.tbl.gz | perl -lane 'print $F[$#F-2], "_", $F[$#F-1];' | grep -v Allele1 | sort | uniq -u) <(zcat /data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.tbl.gz | perl -lane 'if ($F[6] eq "-") { ($F[1], $F[2]) = ($F[2], $F[1]); $F[6] = "+"; $F[$#F] = 1 - $F[$#F]; } print $F[$#F-2], "_", $F[$#F-1], "\t", $F[$#F-2], "\t", $F[$#F-1], "\t", $F[$#F], "\t", $F[1], "\t", $F[2], "\t", $F[6], "\t", $F[5], "\t", $F[3]; ' | grep -v Allele1 | sort -k 1,1) | perl -lane 'splice(@F, 0, 1); print join("\t", @F);' | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | gzip > /data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.tbl.gz

join <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.tbl.gz | awk '{ print $1 "_" $2 "\t" $4 }' | grep -v NA | sort -g -k 1,1) <(zcat /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.tbl.gz | awk '{ print $1 "_" $2 "\t" $0 }' | grep -v NA | sort -g -k 1,1) | perl -lane 'if ($F[1] eq $F[6]) { ($F[5], $F[6]) = ($F[6], $F[5]); $F[4] = 1 - $F[4]; if ($F[7] eq "-") { $F[7] = "+"; } elsif ($F[7] eq "+") { $F[7] = "-"; } else { $PH = 1; } } elsif ($F[1] eq $F[5]) { $PH = 1; } else { push(@F, "NoAlleleMatch"); } print join("\t", @F[2..$#F]);' | gzip > /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.MatchedToHDL.tbl.gz
join <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.tbl.gz | awk '{ print $1 "_" $2 "\t" $4 }' | grep -v NA | sort -g -k 1,1) <(zcat /data/external_public/GlobalLipids2010/TG_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.tbl.gz | awk '{ print $1 "_" $2 "\t" $0 }' | grep -v NA | sort -g -k 1,1) | perl -lane 'if ($F[1] eq $F[6]) { ($F[5], $F[6]) = ($F[6], $F[5]); $F[4] = 1 - $F[4]; if ($F[7] eq "-") { $F[7] = "+"; } elsif ($F[7] eq "+") { $F[7] = "-"; } else { $PH = 1; } } elsif ($F[1] eq $F[5]) { $PH = 1; } else { push(@F, "NoAlleleMatch"); } print join("\t", @F[2..$#F]);' | gzip > /data/external_public/GlobalLipids2010/TG_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.MatchedToHDL.tbl.gz
join <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.tbl.gz | awk '{ print $1 "_" $2 "\t" $4 }' | grep -v NA | sort -g -k 1,1) <(zcat /data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.tbl.gz | awk '{ print $1 "_" $2 "\t" $0 }' | grep -v NA | sort -g -k 1,1) | perl -lane 'if ($F[1] eq $F[6]) { ($F[5], $F[6]) = ($F[6], $F[5]); $F[4] = 1 - $F[4]; if ($F[7] eq "-") { $F[7] = "+"; } elsif ($F[7] eq "+") { $F[7] = "-"; } else { $PH = 1; } } elsif ($F[1] eq $F[5]) { $PH = 1; } else { push(@F, "NoAlleleMatch"); } print join("\t", @F[2..$#F]);' | gzip > /data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.MatchedToHDL.tbl.gz

zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.tbl.gz | perl -lane 'if ($F[5] ne "0") { print join("\t", @F); }' | gzip > /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.No0Dir.tbl.gz
zcat /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.MatchedToHDL.tbl.gz | grep -v NoAlleleMatch | perl -lane 'if ($F[5] ne "0") { print join("\t", @F); }' | gzip > /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.MatchedToHDL.DropNoMatch.No0Dir.tbl.gz
zcat /data/external_public/GlobalLipids2010/TG_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.MatchedToHDL.tbl.gz | grep -v NoAlleleMatch | perl -lane 'if ($F[5] ne "0") { print join("\t", @F); }' | gzip > /data/external_public/GlobalLipids2010/TG_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.MatchedToHDL.DropNoMatch.No0Dir.tbl.gz
zcat /data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.MatchedToHDL.tbl.gz | grep -v NoAlleleMatch | perl -lane 'if ($F[5] ne "0") { print join("\t", @F); }' | gzip > /data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.MatchedToHDL.DropNoMatch.No0Dir.tbl.gz

#20170125
#Made change to force MAF from start now

zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.No0Dir.tbl.gz | perl -lane 'if ($F[2] > .5) { $F[2] = 1 - $F[2]; } print join("\t", @F);' | gzip > /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.No0Dir.forceMAF.tbl.gz
zcat /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.MatchedToHDL.DropNoMatch.No0Dir.tbl.gz | perl -lane 'if ($F[2] > .5) { $F[2] = 1 - $F[2]; } print join("\t", @F);' | gzip > /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.MatchedToHDL.DropNoMatch.No0Dir.forceMAF.tbl.gz
zcat /data/external_public/GlobalLipids2010/TG_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.MatchedToHDL.DropNoMatch.No0Dir.tbl.gz | perl -lane 'if ($F[2] > .5) { $F[2] = 1 - $F[2]; } print join("\t", @F);' | gzip > /data/external_public/GlobalLipids2010/TG_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.MatchedToHDL.DropNoMatch.No0Dir.forceMAF.tbl.gz
zcat /data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.MatchedToHDL.DropNoMatch.No0Dir.tbl.gz | perl -lane 'if ($F[2] > .5) { $F[2] = 1 - $F[2]; } print join("\t", @F);' | gzip > /data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.MatchedToHDL.DropNoMatch.No0Dir.forceMAF.tbl.gz

#20170126
#Redoing full setup with orienting towards MAF allele from get go now. 
#Thinking about explicitly dropping 'fixed' alleles...but maybe in this one instance prefer not to, since if one of say 4 files has >0 MAF, then that should be fine for all files. Should really average MAF across all files being used for final 'merged' MAF. Deciding with going with explicitly dropping though. If a SNP doesn't have MAF > 0 in all four populations than it 'wasn't measured using the exact same subset of individuals across phenotypes' (somewhat disregarding measurement error and other things, etc...).
#NOTE -- also including a check that ref allele from dataset matches either allele in HapMap23 (dropping if ref doesn't match either alleles and not correcting/checking for strandedness here)

#NoDup, noNA, noComma (mainly affects dbSNP), Check Alleles (drop if no match, includes strand check), Flip to Minor, MatchToPheno1, Drop 0Dir, force flip to minor AF, drop if fixed

join <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.tbl.gz | perl -lane 'print $F[$#F-2], "_", $F[$#F-1];' | grep -v Allele1 | sort | uniq -u) <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.tbl.gz | grep -v NA | grep -v , | sed 's/_/ /g' | sed 's/MAF/RAF/g' | perl -lane 'if (uc($F[1]) eq $F[12]) { $F[$#F] = $F[13]; } elsif (uc($F[1]) eq $F[14]) { $F[$#F] = $F[15]; } else { my $flipa1 = uc($F[1]); $flipa1 =~ tr/ATGC/TACG/; my $flipa2 = uc($F[2]); $flipa2 =~ tr/ATGC/TACG/; if ((($flipa1 eq $F[12]) && ($flipa2 eq $F[14])) || (($flipa1 eq $F[14]) && ($flipa2 eq $F[12]))) { if ($flipa1 eq $F[12]) { $F[$#F] = $F[13]; } elsif ($flipa1 eq $F[14]) { $F[$#F] = $F[15]; } else { print STDERR "Error1a";} } else { $F[$#F] = "NoHapMapMatch"; } } print join("\t", @F[0..9]), "\t", join("_", @F[10..15]), "\t", join("\t", @F[16..$#F]);' | grep -v NoHapMapMatch | perl -lane 'if ($F[$#F] > .5) { ($F[1], $F[2]) = ($F[2], $F[1]); if ($F[6] eq "+") { $F[6] = "-"; } elsif ($F[6] eq "-") { $F[6] = "+"; } else { $PH = 1; } $F[$#F] = 1 - $F[$#F]; } print $F[$#F-2], "_", $F[$#F-1], "\t", $F[$#F-2], "\t", $F[$#F-1], "\t", $F[$#F], "\t", uc($F[1]), "\t", uc($F[2]), "\t", $F[6], "\t", $F[5], "\t", $F[3]; ' | grep -v Allele1 | sort -k 1,1) | perl -lane 'splice(@F, 0, 1); print join("\t", @F);' | gzip > /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.MinorAllele.bmass.formatted.noDups.noNA.noComma.DropNoHapMap.tbl.gz
join <(zcat /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.tbl.gz | perl -lane 'print $F[$#F-2], "_", $F[$#F-1];' | grep -v Allele1 | sort | uniq -u) <(zcat /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.tbl.gz | grep -v NA | grep -v , | sed 's/_/ /g' | sed 's/MAF/RAF/g' | perl -lane 'if (uc($F[1]) eq $F[12]) { $F[$#F] = $F[13]; } elsif (uc($F[1]) eq $F[14]) { $F[$#F] = $F[15]; } else { my $flipa1 = uc($F[1]); $flipa1 =~ tr/ATGC/TACG/; my $flipa2 = uc($F[2]); $flipa2 =~ tr/ATGC/TACG/; if ((($flipa1 eq $F[12]) && ($flipa2 eq $F[14])) || (($flipa1 eq $F[14]) && ($flipa2 eq $F[12]))) { if ($flipa1 eq $F[12]) { $F[$#F] = $F[13]; } elsif ($flipa1 eq $F[14]) { $F[$#F] = $F[15]; } else { print STDERR "Error1a";} } else { $F[$#F] = "NoHapMapMatch"; } } print join("\t", @F[0..9]), "\t", join("_", @F[10..15]), "\t", join("\t", @F[16..$#F]);' | grep -v NoHapMapMatch | perl -lane 'if ($F[$#F] > .5) { ($F[1], $F[2]) = ($F[2], $F[1]); if ($F[6] eq "+") { $F[6] = "-"; } elsif ($F[6] eq "-") { $F[6] = "+"; } else { $PH = 1; } $F[$#F] = 1 - $F[$#F]; } print $F[$#F-2], "_", $F[$#F-1], "\t", $F[$#F-2], "\t", $F[$#F-1], "\t", $F[$#F], "\t", uc($F[1]), "\t", uc($F[2]), "\t", $F[6], "\t", $F[5], "\t", $F[3]; ' | grep -v Allele1 | sort -k 1,1) | perl -lane 'splice(@F, 0, 1); print join("\t", @F);' | gzip > /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.MinorAllele.bmass.formatted.noDups.noNA.noComma.DropNoHapMap.tbl.gz
join <(zcat /data/external_public/GlobalLipids2010/TG_with_Effect.wHapMap23.expanded.tbl.gz | perl -lane 'print $F[$#F-2], "_", $F[$#F-1];' | grep -v Allele1 | sort | uniq -u) <(zcat /data/external_public/GlobalLipids2010/TG_with_Effect.wHapMap23.expanded.tbl.gz | grep -v NA | grep -v , | sed 's/_/ /g' | sed 's/MAF/RAF/g' | perl -lane 'if (uc($F[1]) eq $F[12]) { $F[$#F] = $F[13]; } elsif (uc($F[1]) eq $F[14]) { $F[$#F] = $F[15]; } else { my $flipa1 = uc($F[1]); $flipa1 =~ tr/ATGC/TACG/; my $flipa2 = uc($F[2]); $flipa2 =~ tr/ATGC/TACG/; if ((($flipa1 eq $F[12]) && ($flipa2 eq $F[14])) || (($flipa1 eq $F[14]) && ($flipa2 eq $F[12]))) { if ($flipa1 eq $F[12]) { $F[$#F] = $F[13]; } elsif ($flipa1 eq $F[14]) { $F[$#F] = $F[15]; } else { print STDERR "Error1a";} } else { $F[$#F] = "NoHapMapMatch"; } } print join("\t", @F[0..9]), "\t", join("_", @F[10..15]), "\t", join("\t", @F[16..$#F]);' | grep -v NoHapMapMatch | perl -lane 'if ($F[$#F] > .5) { ($F[1], $F[2]) = ($F[2], $F[1]); if ($F[6] eq "+") { $F[6] = "-"; } elsif ($F[6] eq "-") { $F[6] = "+"; } else { $PH = 1; } $F[$#F] = 1 - $F[$#F]; } print $F[$#F-2], "_", $F[$#F-1], "\t", $F[$#F-2], "\t", $F[$#F-1], "\t", $F[$#F], "\t", uc($F[1]), "\t", uc($F[2]), "\t", $F[6], "\t", $F[5], "\t", $F[3]; ' | grep -v Allele1 | sort -k 1,1) | perl -lane 'splice(@F, 0, 1); print join("\t", @F);' | gzip > /data/external_public/GlobalLipids2010/TG_with_Effect.wHapMap23.expanded.MinorAllele.bmass.formatted.noDups.noNA.noComma.DropNoHapMap.tbl.gz
join <(zcat /data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.tbl.gz | perl -lane 'print $F[$#F-2], "_", $F[$#F-1];' | grep -v Allele1 | sort | uniq -u) <(zcat /data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.tbl.gz | grep -v NA | grep -v , | sed 's/_/ /g' | sed 's/MAF/RAF/g' | perl -lane 'if (uc($F[1]) eq $F[12]) { $F[$#F] = $F[13]; } elsif (uc($F[1]) eq $F[14]) { $F[$#F] = $F[15]; } else { my $flipa1 = uc($F[1]); $flipa1 =~ tr/ATGC/TACG/; my $flipa2 = uc($F[2]); $flipa2 =~ tr/ATGC/TACG/; if ((($flipa1 eq $F[12]) && ($flipa2 eq $F[14])) || (($flipa1 eq $F[14]) && ($flipa2 eq $F[12]))) { if ($flipa1 eq $F[12]) { $F[$#F] = $F[13]; } elsif ($flipa1 eq $F[14]) { $F[$#F] = $F[15]; } else { print STDERR "Error1a";} } else { $F[$#F] = "NoHapMapMatch"; } } print join("\t", @F[0..9]), "\t", join("_", @F[10..15]), "\t", join("\t", @F[16..$#F]);' | grep -v NoHapMapMatch | perl -lane 'if ($F[$#F] > .5) { ($F[1], $F[2]) = ($F[2], $F[1]); if ($F[6] eq "+") { $F[6] = "-"; } elsif ($F[6] eq "-") { $F[6] = "+"; } else { $PH = 1; } $F[$#F] = 1 - $F[$#F]; } print $F[$#F-2], "_", $F[$#F-1], "\t", $F[$#F-2], "\t", $F[$#F-1], "\t", $F[$#F], "\t", uc($F[1]), "\t", uc($F[2]), "\t", $F[6], "\t", $F[5], "\t", $F[3]; ' | grep -v Allele1 | sort -k 1,1) | perl -lane 'splice(@F, 0, 1); print join("\t", @F);' | gzip > /data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.MinorAllele.bmass.formatted.noDups.noNA.noComma.DropNoHapMap.tbl.gz

/data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.MinorAllele.bmass.formatted.noDups.noNA.noComma.DropNoHapMap.tbl.gz

join <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.MinorAllele.bmass.formatted.noDups.noNA.noComma.DropNoHapMap.tbl.gz | awk '{ print $1 "_" $2 "\t" $4 }' | sort -g -k 1,1) <(zcat /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.MinorAllele.bmass.formatted.noDups.noNA.noComma.DropNoHapMap.tbl.gz | awk '{ print $1 "_" $2 "\t" $0 }' | sort -g -k 1,1) | perl -lane 'if ($F[1] eq $F[6]) { ($F[5], $F[6]) = ($F[6], $F[5]); $F[4] = 1 - $F[4]; if ($F[7] eq "-") { $F[7] = "+"; } elsif ($F[7] eq "+") { $F[7] = "-"; } else { $PH = 1; } } elsif ($F[1] eq $F[5]) { $PH = 1; } else { push(@F, "NoAlleleMatch"); } print join("\t", @F[2..$#F]);' | gzip > /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.MinorAllele.bmass.formatted.noDups.noNA.noComma.DropNoHapMap.MatchedToHDL.tbl.gz
join <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.MinorAllele.bmass.formatted.noDups.noNA.noComma.DropNoHapMap.tbl.gz | awk '{ print $1 "_" $2 "\t" $4 }' | sort -g -k 1,1) <(zcat /data/external_public/GlobalLipids2010/TG_with_Effect.wHapMap23.expanded.MinorAllele.bmass.formatted.noDups.noNA.noComma.DropNoHapMap.tbl.gz | awk '{ print $1 "_" $2 "\t" $0 }' | sort -g -k 1,1) | perl -lane 'if ($F[1] eq $F[6]) { ($F[5], $F[6]) = ($F[6], $F[5]); $F[4] = 1 - $F[4]; if ($F[7] eq "-") { $F[7] = "+"; } elsif ($F[7] eq "+") { $F[7] = "-"; } else { $PH = 1; } } elsif ($F[1] eq $F[5]) { $PH = 1; } else { push(@F, "NoAlleleMatch"); } print join("\t", @F[2..$#F]);' | gzip > /data/external_public/GlobalLipids2010/TG_with_Effect.wHapMap23.expanded.MinorAllele.bmass.formatted.noDups.noNA.noComma.DropNoHapMap.MatchedToHDL.tbl.gz
join <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.MinorAllele.bmass.formatted.noDups.noNA.noComma.DropNoHapMap.tbl.gz | awk '{ print $1 "_" $2 "\t" $4 }' | sort -g -k 1,1) <(zcat /data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.MinorAllele.bmass.formatted.noDups.noNA.noComma.DropNoHapMap.tbl.gz | awk '{ print $1 "_" $2 "\t" $0 }' | sort -g -k 1,1) | perl -lane 'if ($F[1] eq $F[6]) { ($F[5], $F[6]) = ($F[6], $F[5]); $F[4] = 1 - $F[4]; if ($F[7] eq "-") { $F[7] = "+"; } elsif ($F[7] eq "+") { $F[7] = "-"; } else { $PH = 1; } } elsif ($F[1] eq $F[5]) { $PH = 1; } else { push(@F, "NoAlleleMatch"); } print join("\t", @F[2..$#F]);' | gzip > /data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.MinorAllele.bmass.formatted.noDups.noNA.noComma.DropNoHapMap.MatchedToHDL.tbl.gz

zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.MinorAllele.bmass.formatted.noDups.noNA.noComma.DropNoHapMap.tbl.gz | perl -lane 'if ($F[5] ne "0") { if ($F[2] > .5) { $F[2] = 1 - $F[2]; } if ($F[2] > 0) { print join("\t", @F); } }' | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | gzip > /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.MinorAllele.bmass.formatted.noDups.noNA.noComma.DropNoHapMap.no0Dir.noFixed.forceMAF.tbl.gz
zcat /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.MinorAllele.bmass.formatted.noDups.noNA.noComma.DropNoHapMap.MatchedToHDL.tbl.gz | grep -v NoAlleleMatch | perl -lane 'if ($F[5] ne "0") { if ($F[2] > .5) { $F[2] = 1 - $F[2]; } if ($F[2] > 0) { print join("\t", @F); } }' | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | gzip > /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.MinorAllele.bmass.formatted.noDups.noNA.noComma.DropNoHapMap.MatchedToHDL.DropNoMatch.no0Dir.noFixed.forceMAF.tbl.gz
zcat /data/external_public/GlobalLipids2010/TG_with_Effect.wHapMap23.expanded.MinorAllele.bmass.formatted.noDups.noNA.noComma.DropNoHapMap.MatchedToHDL.tbl.gz | grep -v NoAlleleMatch | perl -lane 'if ($F[5] ne "0") { if ($F[2] > .5) { $F[2] = 1 - $F[2]; } if ($F[2] > 0) { print join("\t", @F); } }' | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | gzip > /data/external_public/GlobalLipids2010/TG_with_Effect.wHapMap23.expanded.MinorAllele.bmass.formatted.noDups.noNA.noComma.DropNoHapMap.MatchedToHDL.DropNoMatch.no0Dir.noFixed.forceMAF.tbl.gz
zcat /data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.MinorAllele.bmass.formatted.noDups.noNA.noComma.DropNoHapMap.MatchedToHDL.tbl.gz | grep -v NoAlleleMatch | perl -lane 'if ($F[5] ne "0") { if ($F[2] > .5) { $F[2] = 1 - $F[2]; } if ($F[2] > 0) { print join("\t", @F); } }' | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | gzip > /data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.MinorAllele.bmass.formatted.noDups.noNA.noComma.DropNoHapMap.MatchedToHDL.DropNoMatch.no0Dir.noFixed.forceMAF.tbl.gz

#20170112
#Note the change to /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.txt from /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.txt
#20170125
#Note change to using forceMAF files
#20170126
#Note change to MinorAllele files
#20170202
#Note change to `noDups.noNA.noComma.DropNoHapMap.MatchedToHDL.DropNoMatch.no0Dir.noFixed.forceMAF`

R -q -e "library("devtools"); devtools::load_all("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/bmass");
#HDL <- read.table("/data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.No0Dir.tbl.gz", header=T);
#LDL <- read.table("/data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.MatchedToHDL.DropNoMatch.No0Dir.tbl.gz", header=T);
#TG <- read.table("/data/external_public/GlobalLipids2010/TG_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.MatchedToHDL.DropNoMatch.No0Dir.tbl.gz", header=T);
#TC <- read.table("/data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.MatchedToHDL.DropNoMatch.No0Dir.tbl.gz", header=T);
#HDL <- read.table("/data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.No0Dir.forceMAF.tbl.gz", header=T);
#LDL <- read.table("/data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.MatchedToHDL.DropNoMatch.No0Dir.forceMAF.tbl.gz", header=T);
#TG <- read.table("/data/external_public/GlobalLipids2010/TG_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.MatchedToHDL.DropNoMatch.No0Dir.forceMAF.tbl.gz", header=T);
#TC <- read.table("/data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.MatchedToHDL.DropNoMatch.No0Dir.forceMAF.tbl.gz", header=T);
#HDL <- read.table("/data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.MinorAllele.bmass.formatted.noNAs.noDups.no0Dir.noFixed.forceMAF.tbl.gz", header=T);
#LDL <- read.table("/data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.MinorAllele.bmass.formatted.noNAs.noDups.MatchedToHDL.DropNoMatch.no0Dir.noFixed.forceMAF.tbl.gz", header=T);
#TG <- read.table("/data/external_public/GlobalLipids2010/TG_with_Effect.wHapMap23.expanded.MinorAllele.bmass.formatted.noNAs.noDups.MatchedToHDL.DropNoMatch.no0Dir.noFixed.forceMAF.tbl.gz", header=T);
#TC <- read.table("/data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.MinorAllele.bmass.formatted.noNAs.noDups.MatchedToHDL.DropNoMatch.no0Dir.noFixed.forceMAF.tbl.gz", header=T);
HDL <- read.table("/data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.MinorAllele.bmass.formatted.noDups.noNA.noComma.DropNoHapMap.no0Dir.noFixed.forceMAF.tbl.gz", header=T);
LDL <- read.table("/data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.MinorAllele.bmass.formatted.noDups.noNA.noComma.DropNoHapMap.MatchedToHDL.DropNoMatch.no0Dir.noFixed.forceMAF.tbl.gz", header=T);
TG <- read.table("/data/external_public/GlobalLipids2010/TG_with_Effect.wHapMap23.expanded.MinorAllele.bmass.formatted.noDups.noNA.noComma.DropNoHapMap.MatchedToHDL.DropNoMatch.no0Dir.noFixed.forceMAF.tbl.gz", header=T);
TC <- read.table("/data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.MinorAllele.bmass.formatted.noDups.noNA.noComma.DropNoHapMap.MatchedToHDL.DropNoMatch.no0Dir.noFixed.forceMAF.tbl.gz", header=T);
##GWASsnps <- read.table("/mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.txt", header=F);
GWASsnps <- read.table("/mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.txt", header=F);
names(GWASsnps) <- c("Chr", "BP")
#bmassOutput1 <- bmass(c("HDL", "LDL", "TG", "TC"), GWASsnps=GWASsnps, NminThreshold = 1, bmassSeedValue=150);
#print(bmassOutput1);" > shana1.txt
DataSources = c("HDL", "LDL", "TG", "TC"); NminThreshold = 50000; bmassSeedValue=150;
ExpectedColumnNames=c("Chr", "BP", "MAF", "Direction", "pValue", "N"); GWASsnps_AnnotateWindow = 5e5; SNPMarginalUnivariateThreshold = 1e-6; SNPMarginalMultivariateThreshold = 1e-6; SigmaAlphas = c(0.005,0.0075,0.01,0.015,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.15); ProvidedPriors=NULL; UseFlatPriors=FALSE; PruneMarginalSNPs=TRUE; PruneMarginalSNPs_bpWindow=5e5;

#bmassOutput1 <- bmass(c("bmass_TestData1", "bmass_TestData2"), GWASsnps=bmass_TestSigSNPs, NminThreshold = 2000, bmassSeedValue=NULL)

#From 20161227
bmassOutput1 <- bmass(c("HDL", "LDL", "TG", "TC"), GWASsnps=GWASsnps, NminThreshold = 50000, bmassSeedValue=150);
bmassOutput2 <- bmass(c("HDL", "LDL", "TG", "TC"), GWASsnps=GWASsnps, NminThreshold = 50000, bmassSeedValue=150, PruneMarginalSNPs=FALSE);
bmassOutput3 <- bmass(c("HDL", "LDL", "TG", "TC"), GWASsnps=GWASsnps, NminThreshold = 50000, bmassSeedValue=150, PrintMergedData=TRUE, PruneMarginalSNPs=FALSE);
 
write.table(bmassOutput1$NewSNPs$SNPs, file="/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.NewSNPs.SNPs.vs1.txt", quote=FALSE, row.names=FALSE)
write.table(bmassOutput3$MergedDataSources, file="/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.txt", quote=FALSE, row.names=FALSE) 
#2017022
write.table(bmassOutput1$PreviousSNPs$SNPs, file="/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.PreviousSNPs.SNPs.vs1.txt", quote=FALSE, row.names=FALSE)
write.table(bmassOutput1$MarginalSNPs$SNPs, file="/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MarginalSNPs.Pruned.SNPs.vs1.txt", quote=FALSE, row.names=FALSE)
write.table(bmassOutput3$MarginalSNPs$SNPs, file="/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MarginalSNPs.NotPruned.SNPs.vs1.txt", quote=FALSE, row.names=FALSE)

system("gzip /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.txt")

~~~
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.No0Dir.tbl.gz | awk '{ print $1 "_" $2 }' | sort | uniq -u | wc
2627690 2627690 30597403
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.No0Dir.tbl.gz | awk '{ print $1 "_" $2 "\t" $0 }' | sort -k 1,1) <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.No0Dir.tbl.gz | awk '{ print $1 "_" $2 }' | sort | uniq -u) | wc
2627690 23649210 127286752
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$for i in `ls -lrt /data/external_public/GlobalLipids2010/*_with_Effect.wHapMap23.expanded.tbl.gz | awk '{ print $9 }'`; do echo $i; zcat $i | wc; zcat $i | perl -lane 'print $F[$#F-2], "_", $F[$#F-1];' | sort | uniq | wc; done
/data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.tbl.gz
2692430 37694020 331765755
2627853 2627853 30599302
/data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.tbl.gz
2692565 37695910 332014156
2627984 2627984 30600736
/data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.tbl.gz
2692414 37693796 333301157
2627835 2627835 30599076
/data/external_public/GlobalLipids2010/TG_with_Effect.wHapMap23.expanded.tbl.gz
2692561 37695854 331851585
2627985 2627985 30600838
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.tbl.gz | grep 91950526
rs17884995      a       g       7776    -0.226  0.8215  -       ?????????????-???????????       -0.0042 0.0185  chr1_91950526_A_0.508_G_0.492   1       91950526        0.492
rs1805113       a       g       87588   -0.712  0.4764  -       -+---+---+--+++-------+-+       -0.0026 0.0054  chr1_91950526_A_0.492_G_0.508   1       91950526        0.492
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.tbl.gz | grep 104850499
rs1475276       t       c       94483   -0.827  0.4081  -       --+--+---+-+-----++-+----       -0.0047 0.0065  chr13_104850499_C_0.783_T_0.217 13      104850499       0.217
rs34921609      t       c       2334    0.887   0.3749  +       ????????????????????++???       0.032   0.0364  chr13_104850499_C_0.224_T_0.776 13      104850499       0.224
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.tbl.gz | grep 40969589 
rs5751250       t       g       93996   -1.181  0.2375  -       ---+----+--+---++-----++-       -0.0056 0.0052  chr22_40969589_T_0.533_G_0.467  22      40969589        0.467
rs5758681       a       g       7773    0.41    0.6822  +       ?????????????+???????????       0.0115  0.0279  chr22_40969589_T_0_G_0.208      22      40969589        0
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs3.py --file1 <(zcat /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/snp130.txt.gz | perl -lane 'if ($F[1] =~ m/(chr[0-9MXY]+)_.*/) { $F[1] = $1; } print join("\t", @F);') --file2 /data/external_public/GlobalLipids2010/HDL_with_Effect.tbl.gz | head -n 10
MarkerName      Allele1 Allele2 Weight  GC.Zscore       GC.Pvalue       Overall Direction       Effect  StdErr  ChrBP
rs10 a  c       85853   -0.096  0.9232  -       ---+-+++-+?+--??--++??+++       -0.0019 0.0147  7_92221824
rs1000000       a       g       99900   -0.937  0.349   -       ---+-+---++-+---+++-++++-       -0.0056 0.0058  12_125456933
rs10000010      t       c       99900   0.185   0.8532  +       -+++-+-+--+++-++++--+----       0.0013  0.0049  4_21227772
rs10000012      c       g       99843   -0.752  0.4523  -       +-+-+-+--+----+--+++++-++       -0.0065 0.0071  4_1347325
rs10000013      a       c       99900   0.642   0.5209  +       +++-++--+--++--+---+-++--       3e-04   0.0058  4_36901464
rs10000017      t       c       99900   2.043   0.04106 +       +-++++++---++++---++--+++       0.0105  0.0062  4_84997149
rs1000002       t       c       99797   2.249   0.02453 +       -+-+---+-+-+++--++-+-+-++       0.0101  0.0049  3_185118462
rs10000023      t       g       99900   -0.008  0.9935  -       -+-+--+--+++++--++------+       4e-04   0.005   4_95952929
rs10000027      c       g       7807    -0.686  0.4925  -       ?????????????-???????????       -0.0141 0.0205  4_156395667
[  mturchin20@spudhead  ~/Software/bedtools2-2.25.0]$for i in `ls -lrt /data/external_public/GlobalLipids2010/*_with_Effect.wHapMap23.expanded.tbl.gz | awk '{ print $9 }'`; do echo $i; zcat $i | wc; zcat $i | perl -lane 'print $F[$#F-2], "_", $F[$#F-1];' | sort | uniq | wc; zcat $i | perl -lane 'print $F[$#F-2], "_", $F[$#F-1];' | sort | uniq -u | wc; done
/data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.tbl.gz
2692430 37694020 331765755
2627853 2627853 30599302
2627816 2627816 30598877
/data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.tbl.gz
2692565 37695910 332014156
2627984 2627984 30600736
2627947 2627947 30600311
/data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.tbl.gz
2692414 37693796 333301157
2627835 2627835 30599076
2627798 2627798 30598651
/data/external_public/GlobalLipids2010/TG_with_Effect.wHapMap23.expanded.tbl.gz
2692561 37695854 331851585
2627985 2627985 30600838
2627948 2627948 30600413
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.tbl.gz | perl -lane 'print $F[$#F-2], "_", $F[$#F-1];' | sort | uniq -c | awk '{ if ($1 > 1) { print $0 } }'                           2 10_107974310
      2 10_44190021
      2 1_151269919
      2 1_181792961
      2 1_181801167
      2 1_181822914
      2 1_243218867
      2 1_244679662
      2 13_104850499
      2 13_27851104
      2 13_52264136
      2 13_92475302
      2 1_40088475
      2 15_46489749
      2 17_28342660
      2 1_91950526
      2 2_166781097
      2 2_180144902
      2 2_209886487
      2 22_40969589
      2 3_46444018
      2 4_118924724
      2 4_70240740
      2 6_147339146
      2 6_40760233
      2 6_76639833
      2 7_112790973
      2 7_150832232
      2 7_75772576
      2 7_92339640
      2 7_94835296
      2 7_96313573
      2 8_56053572
      2 8_8538134
      2 9_102507732
      2 9_8919077
  64542 NA_NA
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.tbl.gz | perl -lane 'print $F[$#F-2], "_", $F[$#F-1];' | sort | uniq -u | head -n 10                                             10_10000135
10_10000265
10_100002729
10_100002880
10_100003553
10_100003805
10_100003967
10_10000459
10_100005553
10_100005593
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.tbl.gz | perl -lane 'print $F[$#F-2], "_", $F[$#F-1];' | grep -v Allele1 | sort | uniq -u) <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.tbl.gz | perl -lane 'if ($F[6] eq "-") { ($F[1], $F[2]) = ($F[2], $F[1]); $F[6] = "+"; $F[$#F] = 1 - $F[$#F]; } print $F[$#F-2], "_", $F[$#F-1], "\t", $F[$#F-2], "\t", $F[$#F-1], "\t", $F[$#F], "\t", $F[1], "\t", $F[2], "\t", $F[6], "\t", $F[5], "\t", $F[3]; ' | grep -v Allele1 | sort -k 1,1) | wc
2627815 23650335 127292283
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.tbl.gz | perl -lane 'print $F[$#F-2], "_", $F[$#F-1];' | grep -v Allele1 | sort | uniq -u | wc
2627816 2627816 30598877
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.tbl.gz | perl -lane 'print $F[$#F-2], "_", $F[$#F-1];' | grep -v Allele1 | sort | uniq -u) <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.tbl.gz | perl -lane 'if ($F[6] eq "-") { ($F[1], $F[2]) = ($F[2], $F[1]); $F[6] = "+"; $F[$#F] = 1 - $F[$#F]; } print $F[$#F-2], "_", $F[$#F-1], "\t", $F[$#F-2], "\t", $F[$#F-1], "\t", $F[$#F], "\t", $F[1], "\t", $F[2], "\t", $F[6], "\t", $F[5], "\t", $F[3]; ' | grep -v Allele1 | sort -k 1,1) | head -n 10
10_10000135 10 10000135 0.551 g a + 0.411 99150
10_10000265 10 10000265 0.436 t c + 0.3933 99150
10_100002729 10 100002729 0.633 g a + 0.03387 99900
10_100002880 10 100002880 0.642 g a + 0.03461 99900
10_100003553 10 100003553 0.12 t c + 0.8488 99900
10_100003805 10 100003805 0.3 t c + 0.1194 99900
10_100003967 10 100003967 0.356 a t + 0.03739 99900
10_10000459 10 10000459 0.427 t g + 0.4015 99150
10_100005553 10 100005553 0.144 c g + 0.1038 99900
10_100005593 10 100005593 0.783 g t + 0.8587 99900
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.tbl.gz | perl -lane 'print $F[$#F-2], "_", $F[$#F-1];' | grep -v Allele1 | sort | uniq -u) <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.tbl.gz | perl -lane 'if ($F[6] eq "-") { ($F[1], $F[2]) = ($F[2], $F[1]); $F[6] = "+"; $F[$#F] = 1 - $F[$#F]; } print $F[$#F-2], "_", $F[$#F-1], "\t", $F[$#F-2], "\t", $F[$#F-1], "\t", $F[$#F], "\t", $F[1], "\t", $F[2], "\t", $F[6], "\t", $F[5], "\t", $F[3]; ' | grep -v Allele1 | sort -k 1,1) | perl -lane 'splice(@F, 0, 1); print join("\t", @F);' | head -n 10
10 10000135     0.551   g       a       +       0.411   99150
10 10000265     0.436   t       c       +       0.3933  99150
10 100002729    0.633   g       a       +       0.03387 99900
10 100002880    0.642   g       a       +       0.03461 99900
10 100003553    0.12    t       c       +       0.8488  99900
10 100003805    0.3     t       c       +       0.1194  99900
10 100003967    0.356   a       t       +       0.03739 99900
10 10000459     0.427   t       g       +       0.4015  99150
10 100005553    0.144   c       g       +       0.1038  99900
10 100005593    0.783   g       t       +       0.8587  99900
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.tbl.gz | perl -lane 'print $F[$#F-2], "_", $F[$#F-1];' | grep -v Allele1 | sort | uniq -u) <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.tbl.gz | perl -lane 'if ($F[6] eq "-") { ($F[1], $F[2]) = ($F[2], $F[1]); $F[6] = "+"; $F[$#F] = 1 - $F[$#F]; } print $F[$#F-2], "_", $F[$#F-1], "\t", $F[$#F-2], "\t", $F[$#F-1], "\t", $F[$#F], "\t", $F[1], "\t", $F[2], "\t", $F[6], "\t", $F[5], "\t", $F[3]; ' | grep -v Allele1 | sort -k 1,1) | perl -lane 'splice(@F, 0, 1); print join("\t", @F);' | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | gzip > /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.tbl.gz
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.tbl.gz | head -n 10
Chr BP  MAF     A1      A2      Direction       pValue  N
10 10000135     0.551   g       a       +       0.411   99150
10 10000265     0.436   t       c       +       0.3933  99150
10 100002729    0.633   g       a       +       0.03387 99900
10 100002880    0.642   g       a       +       0.03461 99900
10 100003553    0.12    t       c       +       0.8488  99900
10 100003805    0.3     t       c       +       0.1194  99900
10 100003967    0.356   a       t       +       0.03739 99900
10 10000459     0.427   t       g       +       0.4015  99150
10 100005553    0.144   c       g       +       0.1038  99900
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$for i in `ls -lrt /data/external_public/GlobalLipids2010/*_with_Effect.*No0Dir.tbl.gz | awk '{ print $9 }'`; do echo $i; zcat $i | wc; done
/data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.No0Dir.tbl.gz
2692298 21538384 98440960
/data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL.No0Dir.tbl.gz
2624227 20993816 96672892
/data/external_public/GlobalLipids2010/TG_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL.No0Dir.tbl.gz
2624466 20995728 96703993
/data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL.No0Dir.tbl.gz
2624603 20996824 97999391
/data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.No0Dir.tbl.gz
2627015 21016120 96771348
/data/external_public/GlobalLipids2010/TG_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.No0Dir.tbl.gz
2627253 21018024 96802408
/data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.No0Dir.tbl.gz
2627393 21019144 98099275
/data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.No0Dir.tbl.gz
2627690 21021520 96689349
/data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.MatchedToHDL.DropNoMatch.No0Dir.tbl.gz
2624080 20992640 96667639
/data/external_public/GlobalLipids2010/TG_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.MatchedToHDL.DropNoMatch.No0Dir.tbl.gz
2624318 20994544 96698691
/data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.MatchedToHDL.DropNoMatch.No0Dir.tbl.gz
2624458 20995664 97994197
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.MinorAllele.bmass.formatted.NoDups.tbl.gz | grep NA2 | wc
      0       0       0
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.tbl.gz | head -n 10
Chr BP  MAF A1  A2 Direction  pValue    N
10 10000135 0.551  g    a  +  0.411     99150
10 10000265 0.436  t    c  +  0.3933    99150
10 100002729    0.633   g  a  + 0.03387 99900
10 100002880    0.642   g  a  + 0.03461 99900
10 100003553    0.12    t  c  + 0.8488  99900
10 100003805    0.3 t   c  +  0.1194    99900
10 100003967    0.356   a  t  + 0.03739 99900
10 10000459 0.427  t    g  +  0.4015    99150
10 100005553    0.144   c  g  + 0.1038  99900
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.MinorAllele.bmass.formatted.NoDups.tbl.gz | head -n 10
Chr BP  MAF A1  A2 Direction  pValue    N
10 10000135 0.449  a    g  -  0.411     99150
10 10000265 0.436  t    c  +  0.3933    99150
10 100002729    0.367   a  g  - 0.03387 99900
10 100002880    0.358   a  g  - 0.03461 99900
10 100003553    0.12    t  c  + 0.8488  99900
10 100003805    0.3 t   c  +  0.1194    99900
10 100003967    0.356   a  t  + 0.03739 99900
10 10000459 0.427  t    g  +  0.4015    99150
10 100005553    0.144   c  g  + 0.1038  99900
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.tbl.gz | head -n 10
Chr BP  MAF A1  A2 Direction  pValue    N
10 10000135 0.449  a    g  +  0.8399    99434
10 10000265 0.564  c    t  +  0.8803    99434
10 100002729    0.367   a  g  + 0.1812  100184
10 100002880    0.358   a  g  + 0.2776  100184
10 100003553    0.88    c  t  + 0.2265  100184
10 100003805    0.7 c   t  +  0.6764    100184
10 100003967    0.644   t  a  + 0.2426  100184
10 10000459 0.573  g    t  +  0.8693    99434
10 100005553    0.144   c  g  + 0.3909  100184
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.MinorAllele.bmass.formatted.NoDups.tbl.gz | head -n 10
Chr BP  MAF A1  A2 Direction  pValue    N
10 10000135 0.449  a    g  +  0.8399    99434
10 10000265 0.436  t    c  -  0.8803    99434
10 100002729    0.367   a  g  + 0.1812  100184
10 100002880    0.358   a  g  + 0.2776  100184
10 100003553    0.12    t  c  - 0.2265  100184
10 100003805    0.3 t   c  -  0.6764    100184
10 100003967    0.356   a  t  - 0.2426  100184
10 10000459 0.427  t    g  -  0.8693    99434
10 100005553    0.144   c  g  + 0.3909  100184
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.tbl.gz | sed 's/_/ /g' | sed 's/MAF/RAF/g' | perl -lane 'if (uc($F[1]) eq $F[12]) { $F[$#F] = $F[13]; } elsif (uc($F[1]) eq $F[14]) { $F[$#F] = $F[15]; } else { $F[$#F] = "NoAlleleMatch"; } print join("\t", @F);' | grep NoAlleleMatch | wc
  72085 1046900 7319704
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.tbl.gz | sed 's/_/ /g' | sed 's/MAF/RAF/g' | perl -lane 'if (uc($F[1]) eq $F[12]) { $F[$#F] = $F[13]; } elsif (uc($F[1]) eq $F[14]) { $F[$#F] = $F[15]; } else { $F[$#F] = "NoAlleleMatch"; } print join("\t", @F);' | grep NoAlleleMatch | wc
  72084 1046861 7318714
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /data/external_public/GlobalLipids2010/TG_with_Effect.wHapMap23.expanded.tbl.gz | sed 's/_/ /g' | sed 's/MAF/RAF/g' | perl -lane 'if (uc($F[1]) eq $F[12]) { $F[$#F] = $F[13]; } elsif (uc($F[1]) eq $F[14]) { $F[$#F] = $F[15]; } else { $F[$#F] = "NoAlleleMatch"; } print join("\t", @F);' | grep NoAlleleMatch | wc
  72080 1046810 7318828
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.tbl.gz | sed 's/_/ /g' | sed 's/MAF/RAF/g' | perl -lane 'if (uc($F[1]) eq $F[12]) { $F[$#F] = $F[13]; } elsif (uc($F[1]) eq $F[14]) { $F[$#F] = $F[15]; } else { $F[$#F] = "NoAlleleMatch"; } print join("\t", @F);' | grep NoAlleleMatch | wc
  72086 1046909 7318291
#20170202
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.tbl.gz | sed 's/_/ /g' | sed 's/MAF/RAF/g' | perl -lane 'if (uc($F[1]) eq $F[12]) { $F[$#F] = $F[13]; } elsif (uc($F[1]) eq $F[14]) { $F[$#F] = $F[15]; } else { my $flipa1 = uc($F[1]); $flipa1 =~ tr/ATGC/TACG/; my $flipa2 = uc($F[2]); $flipa2 =~ tr/ATGC/TACG/; if ((($flipa1 eq $F[12]) && ($flipa2 eq $F[14])) || (($flipa1 eq $F[14]) && ($flipa2 eq $F[12]))) { if ($flipa1 eq $F[12]) { $F[$#F] = $F[13]; } elsif ($flipa1 eq $F[14]) { $F[$#F] = $F[15]; } else { print STDERR "Error1a";} print join("\t", @F); }}' | head -n 10
rs10000947  a   g  3841 -0.201  0.8407  -       ??????????-?????????-+???       -0.0045 0.0256  chr4    137731533     T 0.417   C       0.583   4       137731533      0.417
rs10002903  a   g  9135 1.706 0.08797   +       ??????????-??+???????????       0.0289  0.017   chr4    74614160 T    0.575     C       0.425   4       74614160       0.575
rs10005574  t   c  11649   0.717        0.4737  +       ??????????+??-??????++???       0.0118  0.0155  chr4   104910105        G       0.392   A       0.608   4      104910105        0.608
rs10006225  a   c  1313 1.709 0.08751   +       ??????????+??????????????       0.318   0.1853  chr4    183945990     G 1       T       0       4       183945990      0
rs10010161  t   c  11638   0.562        0.5739  +       ??????????+??-??????++???       0.0157  0.0186  chr4   94638079 A       0.783   G       0.217   4       94638079        0.783
rs10015332  a   g  9071 -0.357  0.7212  -       ??????????-??-???????????       -0.0271 0.0716  chr4    69889512 T    0.975     C       0.025   4       69889512       0.975
rs10017130  t   c  11638   0.986        0.3241  +       ??????????-??+??????++???       0.0143  0.0146  chr4   117036008        G       0.458   A       0.542   4      117036008        0.542
rs10021898  a   g  13547   -0.918       0.3588  -       ??????????+??--?????++???       -0.0187 0.0189  chr4   154425811        C       0.167   T       0.833   4      154425811        0.833
rs10022250  t   c  11654   -0.833       0.4047  -       ??????????-??-??????-+???       -0.0143 0.0156  chr4   42526579 G       0.517   A       0.483   4       42526579        0.483
rs10023240  t   c  9150 2.016 0.04381   +       ??????????+??+???????????       0.0408  0.0202  chr4    101362449     G 0.225   A       0.775   4       101362449      0.775
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.tbl.gz | sed 's/_/ /g' | sed 's/MAF/RAF/g' | perl -lane 'if (uc($F[1]) eq $F[12]) { $F[$#F] = $F[13]; } elsif (uc($F[1]) eq $F[14]) { $F[$#F] = $F[15]; } else { my $flipa1 = uc($F[1]); $flipa1 =~ tr/ATGC/TACG/; my $flipa2 = uc($F[2]); $flipa2 =~ tr/ATGC/TACG/; if ((($flipa1 eq $F[12]) && ($flipa2 eq $F[14])) || (($flipa1 eq $F[14]) && ($flipa2 eq $F[12]))) { if ($flipa1 eq $F[12]) { $F[$#F] = $F[13]; } elsif ($flipa1 eq $F[14]) { $F[$#F] = $F[15]; } else { print STDERR "Error1a";} print join("\t", @F); }}' | wc
   2129   40451  254282
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.tbl.gz | sed 's/_/ /g' | sed 's/MAF/RAF/g' | perl -lane 'if (uc($F[1]) eq $F[12]) { $F[$#F] = $F[13]; } elsif (uc($F[1]) eq $F[14]) { $F[$#F] = $F[15]; } else { my $flipa1 = uc($F[1]); $flipa1 =~ tr/ATGC/TACG/; my $flipa2 = uc($F[2]); $flipa2 =~ tr/ATGC/TACG/; if ((($flipa1 eq $F[12]) && ($flipa2 eq $F[14])) || (($flipa1 eq $F[14]) && ($flipa2 eq $F[12]))) { if ($flipa1 eq $F[12]) { $F[$#F] = $F[13]; } elsif ($flipa1 eq $F[14]) { $F[$#F] = $F[15]; } else { print STDERR "Error1a";} print join("\t", @F); }}' | wc
   2123   40337  253515
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /data/external_public/GlobalLipids2010/TG_with_Effect.wHapMap23.expanded.tbl.gz | sed 's/_/ /g' | sed 's/MAF/RAF/g' | perl -lane 'if (uc($F[1]) eq $F[12]) { $F[$#F] = $F[13]; } elsif (uc($F[1]) eq $F[14]) { $F[$#F] = $F[15]; } else { my $flipa1 = uc($F[1]); $flipa1 =~ tr/ATGC/TACG/; my $flipa2 = uc($F[2]); $flipa2 =~ tr/ATGC/TACG/; if ((($flipa1 eq $F[12]) && ($flipa2 eq $F[14])) || (($flipa1 eq $F[14]) && ($flipa2 eq $F[12]))) { if ($flipa1 eq $F[12]) { $F[$#F] = $F[13]; } elsif ($flipa1 eq $F[14]) { $F[$#F] = $F[15]; } else { print STDERR "Error1a";} print join("\t", @F); }}' | wc
   2125   40375  253840
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.tbl.gz | sed 's/_/ /g' | sed 's/MAF/RAF/g' | perl -lane 'if (uc($F[1]) eq $F[12]) { $F[$#F] = $F[13]; } elsif (uc($F[1]) eq $F[14]) { $F[$#F] = $F[15]; } else { my $flipa1 = uc($F[1]); $flipa1 =~ tr/ATGC/TACG/; my $flipa2 = uc($F[2]); $flipa2 =~ tr/ATGC/TACG/; if ((($flipa1 eq $F[12]) && ($flipa2 eq $F[14])) || (($flipa1 eq $F[14]) && ($flipa2 eq $F[12]))) { if ($flipa1 eq $F[12]) { $F[$#F] = $F[13]; } elsif ($flipa1 eq $F[14]) { $F[$#F] = $F[15]; } else { print STDERR "Error1a";} print join("\t", @F); }}' | wc
   2128   40432  254160
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.tbl.gz | grep -v NA | grep -v , | sed 's/_/ /g' | sed 's/MAF/RAF/g' | perl -lane 'if (uc($F[1]) eq $F[12]) { $F[$#F] = $F[13]; } elsif (uc($F[1]) eq $F[14]) { $F[$#F] = $F[15]; } else { my $flipa1 = uc($F[1]); $flipa1 =~ tr/ATGC/TACG/; my $flipa2 = uc($F[2]); $flipa2 =~ tr/ATGC/TACG/; if ((($flipa1 eq $F[12]) && ($flipa2 eq $F[14])) || (($flipa1 eq $F[14]) && ($flipa2 eq $F[12]))) { if ($flipa1 eq $F[12]) { $F[$#F] = $F[13]; } elsif ($flipa1 eq $F[14]) { $F[$#F] = $F[15]; } else { print STDERR "Error1a";} } else { $F[$#F] = "NoHapMapMatch"; } } print join("\t", @F[0..9]), "\t", join("_", @F[10..15]), "\t", join("\t", @F[16..$#F]);' | grep -v NoHapMapMatch | head -n 10
rs10 a  c   85853  -0.096  0.9232       -       ---+-+++-+?+--??--++??+++       -0.0019 0.0147  chr7_92221824_A_0.033_C_0.967   7       92221824        0.033
rs1000000   a   g  99900   -0.937       0.349   -       ---+-+---++-+---+++-++++-       -0.0056 0.0058  chr12_125456933_G_0.627_A_0.373 12      125456933       0.373
rs10000010  t   c  99900   0.185        0.8532  +       -+++-+-+--+++-++++--+----       0.0013  0.0049  chr4_21227772_T_0.575_C_0.425   4       21227772        0.575
rs10000012  c   g  99843   -0.752       0.4523  -       +-+-+-+--+----+--+++++-++       -0.0065 0.0071  chr4_1347325_C_0.808_G_0.192    4       1347325 0.808
rs10000013  a   c  99900   0.642        0.5209  +       +++-++--+--++--+---+-++--       3e-04   0.0058  chr4_36901464_C_0.167_A_0.833   4       36901464        0.833
rs10000017  t   c  99900   2.043        0.04106 +       +-++++++---++++---++--+++       0.0105  0.0062  chr4_84997149_C_0.777_T_0.223   4       84997149        0.223
rs1000002   t   c  99797   2.249        0.02453 +       -+-+---+-+-+++--++-+-+-++       0.0101  0.0049  chr3_185118462_C_0.475_T_0.525  3       185118462       0.525
rs10000023  t   g  99900   -0.008       0.9935  -       -+-+--+--+++++--++------+       4e-04   0.005   chr4_95952929_G_0.408_T_0.592   4       95952929        0.592
rs10000029  t   c  11323   0.451        0.6519  +       ??+????+?????????+?-+-???       0.0255  0.0712  chr4_138905074_T_0.975_C_0.025  4       138905074       0.975
rs1000003   a   g  99743   -1.82        0.06873 -       +------+--++-----+-+-++-+       -0.0121 0.0066  chr3_99825597_A_0.875_G_0.125   3       99825597        0.875
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.tbl.gz | grep -v NA | grep -v , | sed 's/_/ /g' | sed 's/MAF/RAF/g' | perl -lane 'if (uc($F[1]) eq $F[12]) { $F[$#F] = $F[13]; } elsif (uc($F[1]) eq $F[14]) { $F[$#F] = $F[15]; } else { my $flipa1 = uc($F[1]); $flipa1 =~ tr/ATGC/TACG/; my $flipa2 = uc($F[2]); $flipa2 =~ tr/ATGC/TACG/; if ((($flipa1 eq $F[12]) && ($flipa2 eq $F[14])) || (($flipa1 eq $F[14]) && ($flipa2 eq $F[12]))) { if ($flipa1 eq $F[12]) { $F[$#F] = $F[13]; } elsif ($flipa1 eq $F[14]) { $F[$#F] = $F[15]; } else { print STDERR "Error1a";} } else { $F[$#F] = "NoHapMapMatch"; } } print join("\t", @F[0..9]), "\t", join("_", @F[10..15]), "\t", join("\t", @F[16..$#F]);' | grep -v NoHapMapMatch | perl -lane 'if ($F[$#F] > .5) { ($F[1], $F[2]) = ($F[2], $F[1]); if ($F[6] eq "+") { $F[6] = "-"; } elsif ($F[6] eq "-") { $F[6] = "+"; } else { $PH = 1; } $F[$#F] = 1 - $F[$#F]; } print join ("\t", @F);' | head -n 10
rs10 a  c   85853  -0.096  0.9232       -       ---+-+++-+?+--??--++??+++       -0.0019 0.0147  chr7_92221824_A_0.033_C_0.967   7       92221824        0.033
rs1000000   a   g  99900   -0.937       0.349   -       ---+-+---++-+---+++-++++-       -0.0056 0.0058  chr12_125456933_G_0.627_A_0.373 12      125456933       0.373
rs10000010  c   t  99900   0.185        0.8532  -       -+++-+-+--+++-++++--+----       0.0013  0.0049  chr4_21227772_T_0.575_C_0.425   4       21227772        0.425
rs10000012  g   c  99843   -0.752       0.4523  +       +-+-+-+--+----+--+++++-++       -0.0065 0.0071  chr4_1347325_C_0.808_G_0.192    4       1347325 0.192
rs10000013  c   a  99900   0.642        0.5209  -       +++-++--+--++--+---+-++--       3e-04   0.0058  chr4_36901464_C_0.167_A_0.833   4       36901464        0.167
rs10000017  t   c  99900   2.043        0.04106 +       +-++++++---++++---++--+++       0.0105  0.0062  chr4_84997149_C_0.777_T_0.223   4       84997149        0.223
rs1000002   c   t  99797   2.249        0.02453 -       -+-+---+-+-+++--++-+-+-++       0.0101  0.0049  chr3_185118462_C_0.475_T_0.525  3       185118462       0.475
rs10000023  g   t  99900   -0.008       0.9935  +       -+-+--+--+++++--++------+       4e-04   0.005   chr4_95952929_G_0.408_T_0.592   4       95952929        0.408
rs10000029  c   t  11323   0.451        0.6519  -       ??+????+?????????+?-+-???       0.0255  0.0712  chr4_138905074_T_0.975_C_0.025  4       138905074       0.025
rs1000003   g   a  99743   -1.82        0.06873 +       +------+--++-----+-+-++-+       -0.0121 0.0066  chr3_99825597_A_0.875_G_0.125   3       99825597        0.125
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.tbl.gz | grep -v NA | grep -v , | sed 's/_/ /g' | sed 's/MAF/RAF/g' | perl -lane 'if (uc($F[1]) eq $F[12]) { $F[$#F] = $F[13]; } elsif (uc($F[1]) eq $F[14]) { $F[$#F] = $F[15]; } else { my $flipa1 = uc($F[1]); $flipa1 =~ tr/ATGC/TACG/; my $flipa2 = uc($F[2]); $flipa2 =~ tr/ATGC/TACG/; if ((($flipa1 eq $F[12]) && ($flipa2 eq $F[14])) || (($flipa1 eq $F[14]) && ($flipa2 eq $F[12]))) { if ($flipa1 eq $F[12]) { $F[$#F] = $F[13]; } elsif ($flipa1 eq $F[14]) { $F[$#F] = $F[15]; } else { print STDERR "Error1a";} } else { $F[$#F] = "NoHapMapMatch"; } } print join("\t", @F[0..9]), "\t", join("_", @F[10..15]), "\t", join("\t", @F[16..$#F]);' | grep -v NoHapMapMatch | perl -lane 'if ($F[$#F] > .5) { ($F[1], $F[2]) = ($F[2], $F[1]); if ($F[6] eq "+") { $F[6] = "-"; } elsif ($F[6] eq "-") { $F[6] = "+"; } else { $PH = 1; } $F[$#F] = 1 - $F[$#F]; } print $F[$#F-2], "_", $F[$#F-1], "\t", $F[$#F-2], "\t", $F[$#F-1], "\t", $F[$#F], "\t", uc($F[1]), "\t", uc($F[2]), "\t", $F[6], "\t", $F[5], "\t", $F[3]; ' | grep -v Allele1 | head -n 10
7_92221824  7   92221824   0.033        A       C       -  0.9232       85853
12_125456933    12 125456933  0.373     A       G       -  0.349        99900
4_21227772  4   21227772   0.425        C       T       -  0.8532       99900
4_1347325   4   1347325 0.192 G C       +       0.4523  99843
4_36901464  4   36901464   0.167        C       A       -  0.5209       99900
4_84997149  4   84997149   0.223        T       C       +  0.04106      99900
3_185118462 3   185118462  0.475        C       T       -  0.02453      99797
4_95952929  4   95952929   0.408        G       T       +  0.9935       99900
4_138905074 4   138905074  0.025        C       T       -  0.6519       11323
3_99825597  3   99825597   0.125        G       A       +  0.06873      99743
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.tbl.gz | grep -v NA | grep -v , | sed 's/_/ /g' | sed 's/MAF/RAF/g' | perl -lane 'if (uc($F[1]) eq $F[12]) { $F[$#F] = $F[13]; } elsif (uc($F[1]) eq $F[14]) { $F[$#F] = $F[15]; } else { my $flipa1 = uc($F[1]); $flipa1 =~ tr/ATGC/TACG/; my $flipa2 = uc($F[2]); $flipa2 =~ tr/ATGC/TACG/; if ((($flipa1 eq $F[12]) && ($flipa2 eq $F[14])) || (($flipa1 eq $F[14]) && ($flipa2 eq $F[12]))) { if ($flipa1 eq $F[12]) { $F[$#F] = $F[13]; } elsif ($flipa1 eq $F[14]) { $F[$#F] = $F[15]; } else { print STDERR "Error1a";} } else { $F[$#F] = "NoHapMapMatch"; } } print join("\t", @F[0..9]), "\t", join("_", @F[10..15]), "\t", join("\t", @F[16..$#F]);' | grep -v NoHapMapMatch | perl -lane 'if ($F[$#F] > .5) { ($F[1], $F[2]) = ($F[2], $F[1]); if ($F[6] eq "+") { $F[6] = "-"; } elsif ($F[6] eq "-") { $F[6] = "+"; } else { $PH = 1; } $F[$#F] = 1 - $F[$#F]; } print $F[$#F-2], "_", $F[$#F-1], "\t", $F[$#F-2], "\t", $F[$#F-1], "\t", $F[$#F], "\t", uc($F[1]), "\t", uc($F[2]), "\t", $F[6], "\t", $F[5], "\t", $F[3]; ' | grep -v Allele1 | sort -k 1,1 | head -n 10
10_10000135 10  10000135   0.449        G       A       +  0.411        99150
10_10000265 10  10000265   0.436        T       C       +  0.3933       99150
10_100002729    10 100002729  0.367     A       G       -  0.03387      99900
10_100002880    10 100002880  0.358     A       G       -  0.03461      99900
10_100003553    10 100003553  0.12      T       C       +  0.8488       99900
10_100003805    10 100003805  0.3       T       C       +  0.1194       99900
10_100003967    10 100003967  0.356     T       A       -  0.03739      99900
10_10000459 10  10000459   0.427        T       G       +  0.4015       99150
10_100005553    10 100005553  0.144     C       G       +  0.1038       99900
10_100005593    10 100005593  0.217     T       G       -  0.8587       99900
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$join <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.tbl.gz | perl -lane 'print $F[$#F-2], "_", $F[$#F-1];' | grep -v Allele1 | sort | uniq -u) <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.tbl.gz | grep -v NA | grep -v , | sed 's/_/ /g' | sed 's/MAF/RAF/g' | perl -lane 'if (uc($F[1]) eq $F[12]) { $F[$#F] = $F[13]; } elsif (uc($F[1]) eq $F[14]) { $F[$#F] = $F[15]; } else { my $flipa1 = uc($F[1]); $flipa1 =~ tr/ATGC/TACG/; my $flipa2 = uc($F[2]); $flipa2 =~ tr/ATGC/TACG/; if ((($flipa1 eq $F[12]) && ($flipa2 eq $F[14])) || (($flipa1 eq $F[14]) && ($flipa2 eq $F[12]))) { if ($flipa1 eq $F[12]) { $F[$#F] = $F[13]; } elsif ($flipa1 eq $F[14]) { $F[$#F] = $F[15]; } else { print STDERR "Error1a";} } else { $F[$#F] = "NoHapMapMatch"; } } print join("\t", @F[0..9]), "\t", join("_", @F[10..15]), "\t", join("\t", @F[16..$#F]);' | grep -v NoHapMapMatch | perl -lane 'if ($F[$#F] > .5) { ($F[1], $F[2]) = ($F[2], $F[1]); if ($F[6] eq "+") { $F[6] = "-"; } elsif ($F[6] eq "-") { $F[6] = "+"; } else { $PH = 1; } $F[$#F] = 1 - $F[$#F]; } print $F[$#F-2], "_", $F[$#F-1], "\t", $F[$#F-2], "\t", $F[$#F-1], "\t", $F[$#F], "\t", uc($F[1]), "\t", uc($F[2]), "\t", $F[6], "\t", $F[5], "\t", $F[3]; ' | grep -v Allele1 | sort -k 1,1) | perl -lane 'splice(@F, 0, 1); print join("\t", @F);' | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | head -n 10
Chr BP  MAF A1  A2 Direction  pValue    N
10 10000135 0.449  G    A  +  0.411     99150
10 10000265 0.436  T    C  +  0.3933    99150
10 100002729    0.367   A  G  - 0.03387 99900
10 100002880    0.358   A  G  - 0.03461 99900
10 100003553    0.12    T  C  + 0.8488  99900
10 100003805    0.3 T   C  +  0.1194    99900
10 100003967    0.356   T  A  - 0.03739 99900
10 10000459 0.427  T    G  +  0.4015    99150
10 100005553    0.144   C  G  + 0.1038  99900
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$join <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.tbl.gz | perl -lane 'print $F[$#F-2], "_", $F[$#F-1];' | grep -v Allele1 | sort | uniq -u) <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.tbl.gz | grep -v NA | grep -v , | sed 's/_/ /g' | sed 's/MAF/RAF/g' | perl -lane 'if (uc($F[1]) eq $F[12]) { $F[$#F] = $F[13]; } elsif (uc($F[1]) eq $F[14]) { $F[$#F] = $F[15]; } else { my $flipa1 = uc($F[1]); $flipa1 =~ tr/ATGC/TACG/; my $flipa2 = uc($F[2]); $flipa2 =~ tr/ATGC/TACG/; if ((($flipa1 eq $F[12]) && ($flipa2 eq $F[14])) || (($flipa1 eq $F[14]) && ($flipa2 eq $F[12]))) { if ($flipa1 eq $F[12]) { $F[$#F] = $F[13]; } elsif ($flipa1 eq $F[14]) { $F[$#F] = $F[15]; } else { print STDERR "Error1a";} } else { $F[$#F] = "NoHapMapMatch"; } } print join("\t", @F[0..9]), "\t", join("_", @F[10..15]), "\t", join("\t", @F[16..$#F]);' | grep -v NoHapMapMatch | perl -lane 'if ($F[$#F] > .5) { ($F[1], $F[2]) = ($F[2], $F[1]); if ($F[6] eq "+") { $F[6] = "-"; } elsif ($F[6] eq "-") { $F[6] = "+"; } else { $PH = 1; } $F[$#F] = 1 - $F[$#F]; } print $F[$#F-2], "_", $F[$#F-1], "\t", $F[$#F-2], "\t", $F[$#F-1], "\t", $F[$#F], "\t", uc($F[1]), "\t", uc($F[2]), "\t", $F[6], "\t", $F[5], "\t", $F[3]; ' | grep -v Allele1 | sort -k 1,1) | perl -lane 'splice(@F, 0, 1); print join("\t", @F);' | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | wc
2622407 20979256 96617893
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$join <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.tbl.gz | perl -lane 'print $F[$#F-2], "_", $F[$#F-1];' | grep -v Allele1 | sort | uniq -u) <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.tbl.gz | perl -lane 'if ($F[6] eq "-") { ($F[1], $F[2]) = ($F[2], $F[1]); $F[6] = "+"; $F[$#F] = 1 - $F[$#F]; } print $F[$#F-2], "_", $F[$#F-1], "\t", $F[$#F-2], "\t", $F[$#F-1], "\t", $F[$#F], "\t", $F[1], "\t", $F[2], "\t", $F[6], "\t", $F[5], "\t", $F[3]; ' | grep -v Allele1 | sort -k 1,1) | perl -lane 'splice(@F, 0, 1); print join("\t", @F);' | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | wc
2627816 21022528 96693449
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$R -q -e "2627816 - 2622407"
> 2627816 - 2622407
[1] 5409
> 
> 
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.tbl.gz | grep -v NA | grep -v , | sed 's/_/ /g' | sed 's/MAF/RAF/g' | perl -lane 'if (uc($F[1]) eq $F[12]) { $F[$#F] = $F[13]; } elsif (uc($F[1]) eq $F[14]) { $F[$#F] = $F[15]; } else { my $flipa1 = uc($F[1]); $flipa1 =~ tr/ATGC/TACG/; my $flipa2 = uc($F[2]); $flipa2 =~ tr/ATGC/TACG/; if ((($flipa1 eq $F[12]) && ($flipa2 eq $F[14])) || (($flipa1 eq $F[14]) && ($flipa2 eq $F[12]))) { if ($flipa1 eq $F[12]) { $F[$#F] = $F[13]; } elsif ($flipa1 eq $F[14]) { $F[$#F] = $F[15]; } else { print STDERR "Error1a";} } else { $F[$#F] = "NoHapMapMatch"; } } print join("\t", @F[0..9]), "\t", join("_", @F[10..15]), "\t", join("\t", @F[16..$#F]);' | grep -v NoHapMapMatch | perl -lane 'if ($F[$#F] > .5) { ($F[1], $F[2]) = ($F[2], $F[1]); if ($F[6] eq "+") { $F[6] = "-"; } elsif ($F[6] eq "-") { $F[6] = "+"; } else { $PH = 1; } $F[$#F] = 1 - $F[$#F]; } print $F[$#F-2], "_", $F[$#F-1], "\t", $F[$#F-2], "\t", $F[$#F-1], "\t", $F[$#F], "\t", uc($F[1]), "\t", uc($F[2]), "\t", $F[6], "\t", $F[5], "\t", $F[3]; ' | grep -v Allele1 | sort -k 1,1 | wc
2622474 23602266 127157324
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.tbl.gz | perl -lane 'print $F[$#F-2], "_", $F[$#F-1];' | grep -v Allele1 | sort | uniq -u | wc
2627816 2627816 30598877
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.tbl.gz | grep -v NA | grep -v , | sed 's/_/ /g' | sed 's/MAF/RAF/g' | perl -lane 'if (uc($F[1]) eq $F[12]) { $F[$#F] = $F[13]; } elsif (uc($F[1]) eq $F[14]) { $F[$#F] = $F[15]; } else { my $flipa1 = uc($F[1]); $flipa1 =~ tr/ATGC/TACG/; my $flipa2 = uc($F[2]); $flipa2 =~ tr/ATGC/TACG/; if ((($flipa1 eq $F[12]) && ($flipa2 eq $F[14])) || (($flipa1 eq $F[14]) && ($flipa2 eq $F[12]))) { if ($flipa1 eq $F[12]) { $F[$#F] = $F[13]; } elsif ($flipa1 eq $F[14]) { $F[$#F] = $F[15]; } else { print STDERR "Error1a";} } else { $F[$#F] = "NoHapMapMatch"; } } print join("\t", @F[0..9]), "\t", join("_", @F[10..15]), "\t", join("\t", @F[16..$#F]);' | wc
2627888 36790429 324087192
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.tbl.gz | grep -v NA | grep -v , | wc
2627888 36790432 326116206
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.tbl.gz | wc
2692430 37694020 331765755
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.tbl.gz | grep -v NA | grep -v , | sed 's/_/ /g' | sed 's/MAF/RAF/g' | perl -lane 'if (uc($F[1]) eq $F[12]) { $F[$#F] = $F[13]; } elsif (uc($F[1]) eq $F[14]) { $F[$#F] = $F[15]; } else { my $flipa1 = uc($F[1]); $flipa1 =~ tr/ATGC/TACG/; my $flipa2 = uc($F[2]); $flipa2 =~ tr/ATGC/TACG/; if ((($flipa1 eq $F[12]) && ($flipa2 eq $F[14])) || (($flipa1 eq $F[14]) && ($flipa2 eq $F[12]))) { if ($flipa1 eq $F[12]) { $F[$#F] = $F[13]; } elsif ($flipa1 eq $F[14]) { $F[$#F] = $F[15]; } else { print STDERR "Error1a";} } else { $F[$#F] = "NoHapMapMatch"; } } print join("\t", @F[0..9]), "\t", join("_", @F[10..15]), "\t", join("\t", @F[16..$#F]);' | grep NoHapMapMatch | wc
   5414   75793  685955
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.tbl.gz | perl -lane 'print $F[$#F-2], "_", $F[$#F-1];' | grep -v Allele1 | sort | wc
2692430 2692430 30986967
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.tbl.gz | grep -v NA | grep -v , | perl -lane 'print $F[$#F-2], "_", $F[$#F-1];' | grep -v Allele1 | sort | uniq -u | wc
2627816 2627816 30598877
[  mturchin20@spudling89  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2]$zcat /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.tbl.gz | wc
2692565 37695910 332014156
[  mturchin20@spudling89  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2]$zcat /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.tbl.gz | grep -v NA | wc
2628019 36792266 326365196
[  mturchin20@spudling89  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2]$zcat /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.tbl.gz | grep -v NA | grep -v , | sed 's/_/ /g' | sed 's/MAF/RAF/g' | perl -lane 'if (uc($F[1]) eq $F[12]) { $F[$#F] = $F[13]; } elsif (uc($F[1]) eq $F[14]) { $F[$#F] = $F[15]; } else { my $flipa1 = uc($F[1]); $flipa1 =~ tr/ATGC/TACG/; my $flipa2 = uc($F[2]); $flipa2 =~ tr/ATGC/TACG/; if ((($flipa1 eq $F[12]) && ($flipa2 eq $F[14])) || (($flipa1 eq $F[14]) && ($flipa2 eq $F[12]))) { if ($flipa1 eq $F[12]) { $F[$#F] = $F[13]; } elsif ($flipa1 eq $F[14]) { $F[$#F] = $F[15]; } else { print STDERR "Error1a";} } else { $F[$#F] = "NoHapMapMatch"; } } print join("\t", @F[0..9]), "\t", join("_", @F[10..15]), "\t", join("\t", @F[16..$#F]);' | grep -v NoHapMapMatch | wc
2622604 36716456 323647706
[  mturchin20@spudling89  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2]$zcat /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.tbl.gz | perl -lane 'print $F[$#F-2], "_", $F[$#F-1];' | grep -v Allele1 | sort | uniq -u | wc
[  mturchin20@spudling89  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2]$zcat /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.tbl.gz | grep -v NA | grep -v , | sed 's/_/ /g' | sed 's/MAF/RAF/g' | perl -lane 'if (uc($F[1]) eq $F[12]) { $F[$#F] = $F[13]; } elsif (uc($F[1]) eq $F[14]) { $F[$#F] = $F[15]; } else { my $flipa1 = uc($F[1]); $flipa1 =~ tr/ATGC/TACG/; my $flipa2 = uc($F[2]); $flipa2 =~ tr/ATGC/TACG/; if ((($flipa1 eq $F[12]) && ($flipa2 eq $F[14])) || (($flipa1 eq $F[14]) && ($flipa2 eq $F[12]))) { if ($flipa1 eq $F[12]) { $F[$#F] = $F[13]; } elsif ($flipa1 eq $F[14]) { $F[$#F] = $F[15]; } else { print STDERR "Error1a";} } else { $F[$#F] = "NoHapMapMatch"; } } print join("\t", @F[0..9]), "\t", join("_", @F[10..15]), "\t", join("\t", @F[16..$#F]);' | grep NoHapMapMatch | wc
   5415   75807  686341
[  mturchin20@spudling89  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2]$zcat /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.tbl.gz | perl -lane 'print $F[$#F-2], "_", $F[$#F-1];' | grep -v Allele1 | sort | uniq -u | wc
2627947 2627947 30600311
~~~

```
> library("devtools"); devtools::load_all("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/bmass");
Loading bmass
> HDL <- read.table("/data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.No0Dir.tbl.gz", header=T);
> LDL <- read.table("/data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.MatchedToHDL.DropNoMatch.No0Dir.tbl.gz", header=T);
> TG <- read.table("/data/external_public/GlobalLipids2010/TG_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.MatchedToHDL.DropNoMatch.No0Dir.tbl.gz", header=T);
> TC <- read.table("/data/external_public/GlobalLipids2010/TC_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.MatchedToHDL.DropNoMatch.No0Dir.tbl.gz", header=T);
> GWASsnps <- read.table("/mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.txt", header=F);
#20170112
> GWASsnps <- read.table("/mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.txt", header=F);
#
> names(GWASsnps) <- c("Chr", "BP")
> #bmassOutput1 <- bmass(c("HDL", "LDL", "TG", "TC"), GWASsnps=GWASsnps, NminThreshold = 1, bmassSeedValue=150);
> #print(bmassOutput1);" > shana1.txt
> DataSources = c("HDL", "LDL", "TG", "TC"); NminThreshold = 50000; bmassSeedValue=150;
> ExpectedColumnNames=c("Chr", "BP", "MAF", "Direction", "pValue", "N"); GWASsnps_AnnotateWindow = 5e5; SNPMarginalUnivariateThreshold = 1e-6; SNPMarginalMultivariateThreshold = 1e-6; SigmaAlphas = c(0.005,0.0075,0.01,0.015,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.15); ProvidedPriors=NULL; UseFlatPriors=FALSE; PruneMarginalSNPs=TRUE; PruneMarginalSNPs_bpWindow=5e5;
>
> #bmassOutput1 <- bmass(c("bmass_TestData1", "bmass_TestData2"), GWASsnps=bmass_TestSigSNPs, NminThreshold = 2000, bmassSeedValue=NULL)
>
> #From 20161227
> bmassOutput1 <- bmass(c("HDL", "LDL", "TG", "TC"), GWASsnps=GWASsnps, NminThreshold = 50000, bmassSeedValue=150);
Warning message:
In ProcessMergedAndAnnotatedDataSources(DataSources, bmassOutput$MergedDataSources,  :
  2017-01-10 16:08:20 -- One of your datafiles has p-values less than the threshold which R can properly convert them to log-scale, meaning their log-values return as 'Infinite'. bmass automatically replaces these 'Infinite' values with the max, non-infinite Z-scores available, but it is recommended you self-check this as well.
> bmassOutput2 <- bmass(c("HDL", "LDL", "TG", "TC"), GWASsnps=GWASsnps, NminThreshold = 50000, bmassSeedValue=150, PruneMarginalSNPs=FALSE);
Warning message:
In ProcessMergedAndAnnotatedDataSources(DataSources, bmassOutput$MergedDataSources,  :
  2017-01-10 16:13:44 -- One of your datafiles has p-values less than the threshold which R can properly convert them to log-scale, meaning their log-values return as 'Infinite'. bmass automatically replaces these 'Infinite' values with the max, non-infinite Z-scores available, but it is recommended you self-check this as well.
> bmassOutput3 <- bmass(c("HDL", "LDL", "TG", "TC"), GWASsnps=GWASsnps, NminThreshold = 50000, bmassSeedValue=150, PrintMergedData=TRUE, PruneMarginalSNPs=FALSE);
Warning message:
In ProcessMergedAndAnnotatedDataSources(DataSources, bmassOutput$MergedDataSources,  :
  2017-01-10 16:19:46 -- One of your datafiles has p-values less than the threshold which R can properly convert them to log-scale, meaning their log-values return as 'Infinite'. bmass automatically replaces these 'Infinite' values with the max, non-infinite Z-scores available, but it is recommended you self-check this as well.
>
> dim(bmassOutput1$NewSNPs$SNPs)
[1] 26 29
> bmassOutput1$GWASlogBFMinThreshold
[1] 4.339657
#From prior to `NoDups`
#> dim(bmassOutput1$NewSNPs$SNPs)
#[1] 26 29
#> bmassOutput1$GWASlogBFMinThreshold
#[1] 4.34246
#> dim(bmassOutput3$MergedDataSources)
#[1] 2600747      23
> dim(bmassOutput3$MergedDataSources)
[1] 2593966      23
#20170112 -- after adding in the NA'ed GWAS snp via /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.txt
> dim(bmassOutput1$NewSNPs$SNPs)
[1] 26 29
> bmassOutput1$GWASlogBFMinThreshold
[1] 4.339657
> dim(bmassOutput3$MergedDataSources)
[1] 2593966      23
> dim(GWASsnps)
[1] 102   2
#20170125 -- after including forceMAF now
> dim(bmassOutput1$NewSNPs$SNPs)
[1] 30 29
> bmassOutput1$GWASlogBFMinThreshold
[1] 4.538565
> bmassOutput1$ZScoresCorMatrix     
            HDL_ZScore LDL_ZScore  TG_ZScore  TC_ZScore
HDL_ZScore  1.00000000 -0.0478679 -0.2159197 0.09801589
LDL_ZScore -0.04786790  1.0000000  0.1734726 0.85805232
TG_ZScore  -0.21591968  0.1734726  1.0000000 0.38865754
TC_ZScore   0.09801589  0.8580523  0.3886575 1.00000000
> dim(bmassOutput3$MergedDataSources)
[1] 2593966      23
#20170202 -- orienting to MinorAllele and doing proper amount of drops/removals/etc
> dim(bmassOutput1$NewSNPs$SNPs)
[1] 26 29
> dim(bmassOutput1$PreviousSNPs$SNPs)
[1] 100  29
> dim(bmassOutput3$MergedDataSources)
[1] 2596925      23
> bmassOutput1$GWASlogBFMinThreshold
[1] 4.340976
> bmassOutput1$ZScoresCorMatrix
            HDL_ZScore  LDL_ZScore  TG_ZScore TC_ZScore
HDL_ZScore  1.00000000 -0.08061013 -0.3650923 0.1684484
LDL_ZScore -0.08061013  1.00000000  0.1846387 0.8391409
TG_ZScore  -0.36509228  0.18463868  1.0000000 0.3255277
TC_ZScore   0.16844842  0.83914088  0.3255277 1.0000000
> dim(GWASsnps)
[1] 102   2
```

#20170221
#Extras added to main script after done with bmassOutput1$...etc

```
#GetModelPriorMatrix(bmassOutput1$Models, bmassOutput1$ModelPriors, bmassOutput1$LogFile)$ModelPriorMatrix
#
#GetTopModelsPerSNPViaPosteriors(DataSources, bmassOutput1$PreviousSNPs, GetModelPriorMatrix(bmassOutput1$Models, bmassOutput1$ModelPriors, bmassOutput1$LogFile)$ModelPriorMatrix, bmassOutput1$LogFile)
#GetTopModelsPerSNPViaPosteriors(DataSources, bmassOutput1$NewSNPs, GetModelPriorMatrix(bmassOutput1$Models, bmassOutput1$ModelPriors, bmassOutput1$LogFile)$ModelPriorMatrix, bmassOutput1$LogFile)
#
##marginal.postprobs(bmassOutput1$NewSNPs$Posteriors[,5:ncol(bmassOutput1$NewSNPs$Posteriors)], bmassOutput1$Models, 1)
#
#nana2 <- GetMarginalPosteriors(DataSources, bmassOutput1$NewSNPs, bmassOutput1$Models, SigmaAlphas = c(0.005,0.0075,0.01,0.015,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.15), bmassOutput1$LogFile)
#nana2$ListSNPs$Marginals$pU[,1:10]
#
##Make 'Analysis folder for marginal stuff?'
#jpeg("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.NewSNPs.MarginalHeatplots.DirectAssoc.vs1.jpg", width=4000, height=2000, res=300)
##	NewSNPs.Marginals.DirectAssoc.Clustered <- PlotMarginalPosteriors(DataSources, nana2$ListSNPs, "D", Main="BlahBlah")
#	NewSNPs.Marginals.DirectAssoc.Clustered <- PlotMarginalPosteriors(DataSources, bmassOutput1$NewSNPs, "D", Main="BlahBlah")
#	#PlotMarginalPosteriors <- function (DataSources, ListSNPs, Marginal) {
#dev.off()
#
#jpeg("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.NewSNPs.MarginalHeatplots.DirectAssoc.wDirection.vs1.jpg", width=4000, height=2000, res=300)
#	NewSNPs.Marginals.DirectAssoc.Clustered <- PlotMarginalPosteriors.withDirection(DataSources, bmassOutput1$NewSNPs, "D", Main="BlahBlah")
#dev.off()

bmassOutput4$ModelPriorMatrix <- GetModelPriorMatrix(bmassOutput1$Models, bmassOutput1$ModelPriors, bmassOutput1$LogFile)$ModelPriorMatrix

bmassOutput4[c("PreviousSNPs", "LogFile")] <- GetTopModelsPerSNPViaPosteriors(DataSources, bmassOutput1$PreviousSNPs, bmassOutput4$ModelPriorMatrix, bmassOutput1$LogFile)
bmassOutput4[c("NewSNPs", "LogFile")] <- GetTopModelsPerSNPViaPosteriors(DataSources, bmassOutput1$NewSNPs, bmassOutput4$ModelPriorMatrix, bmassOutput1$LogFile)

bmassOutput4[c("NewSNPs", "LogFile")] <- GetMarginalPosteriors(DataSources, bmassOutput4$NewSNPs, bmassOutput1$Models, SigmaAlphas, bmassOutput1$LogFile)

write.table(bmassOutput4$ModelPriorMatrix, file="/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.ModelPriorMatrix.vs1.txt", quote=FALSE, row.names=FALSE)
write.table(bmassOutput4$PreviousSNPs$TopModels, file="/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.PreviousSNPs.TopModels.vs1.txt", quote=FALSE, row.names=FALSE)
write.table(bmassOutput4$NewSNPs$TopModels, file="/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.NewSNPs.TopModels.vs1.txt", quote=FALSE, row.names=FALSE)

jpeg("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.NewSNPs.MarginalHeatplots.DirectAssoc.vs1.jpg", width=4000, height=2000, res=300)
	bmassOutput1$NewSNPs$HeatPlotOutput <- PlotMarginalPosteriors(DataSources, bmassOutput4$NewSNPs, "D", Main="GlobalLipids2010 NewSNPs DirAssoc")
dev.off()
jpeg("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.NewSNPs.MarginalHeatplots.DirectAssoc.wDirection.vs1.jpg", width=4000, height=2000, res=300)
	NewSNPs.Marginals.DirectAssoc.Clustered <- PlotMarginalPosteriors.withDirection(DataSources, bmassOutput4$NewSNPs, "D", Main="GlobalLipids2010 NewSNPs DirAssoc")
dev.off()




print out now before moving/going on?

print/or copy-paste ModelPriorMatrix? 
print top models 
print marginals? do plots? w/out and w/ direction?









```

~~~

~~~




#20161102
#Scrap/Misc related to getting DAVID/RDAVIDWebServices to work

#https://bioconductor.org/packages/release/bioc/html/RDAVIDWebService.html & from `browseVignettes("RDAVIDWebService")` (http://127.0.0.1:15191/library/RDAVIDWebService/doc/RDavidWS-vignette.pdf)
~~~
> david<-DAVIDWebService(email="mturchin20@uchicago.edu", url="https://david.ncifcrf.gov/webservice/services/DAVIDWebService.DAVIDWebServiceHttpSoap12Endpoint/")
>  data(demoList1)
> result<-addList(david, demoList1, idType="AFFYMETRIX_3PRIME_IVT_ID", listName="demoList1", listType="Gene")
> termCluster<-getClusterReport(david, type="Term")
> summary(termCluster)
   Cluster Enrichment Members
1        1  5.3760826       9
2        2  3.4798417       4
3        3  2.1212619       9
4        4  1.5909857       5
5        5  1.3927257       6
6        6  1.3880614      11
7        7  1.1177393       4
8        8  1.0543396       5
9        9  0.8970393      13
10      10  0.6050857       6
11      11  0.3995537      13
12      12  0.3453586      25
13      13  0.1870551      12
> members(termCluster)[[2]]
DAVID Result object
Result type:  FunctionalAnnotationChart 
          Category                                      Term Count       X.
1 GOTERM_BP_DIRECT     GO:0050832~defense response to fungus     5 3.448276
2 GOTERM_BP_DIRECT GO:0019731~antibacterial humoral response     5 3.448276
3      UP_KEYWORDS                                Antibiotic     6 4.137931
4      UP_KEYWORDS                             Antimicrobial     6 4.137931
        PValue
1 0.0000557092
2 0.0004239358
3 0.0005382324
4 0.0009471880
                                                                           Genes
1           679_AT, 31793_AT, 33284_AT, 31506_S_AT, 37105_AT, 34623_AT, 34546_AT
2                   38604_AT, 31793_AT, 31506_S_AT, 34623_AT, 33979_AT, 34546_AT
3 33963_AT, 679_AT, 31793_AT, 31506_S_AT, 37105_AT, 34623_AT, 33979_AT, 34546_AT
4 33963_AT, 679_AT, 31793_AT, 31506_S_AT, 37105_AT, 34623_AT, 33979_AT, 34546_AT
  List.Total Pop.Hits Pop.Total Fold.Enrichment Bonferroni  Benjamini
1        133       27     16787       23.373712 0.05828839 0.05828839
2        133       45     16787       14.024227 0.36688275 0.20431334
3        140       98     20568        8.994752 0.13482501 0.02047652
4        140      111     20568        7.941313 0.22501711 0.03136197
         FDR
1 0.08882436
2 0.67407811
3 0.70496204
4 1.23752805
~~~






















##GlobalLipids2013 dbsnp ### (ChrBP & 1000G MAF provided), GWAS = , N = provided per SNP
##20150828

####Taken from earlier in script

##mkdir /data/external_public/GlobalLipids2013

##cd /data/external_public/GlobalLipids2013

##wget -r -l1 --no-parent -A "jointGwas*" http://www.sph.umich.edu/csg/abecasis/public/lipids2013/

##wget -r -l1 --no-parent -A "Mc*" http://www.sph.umich.edu/csg/abecasis/public/lipids2013/

###NOTE -- these files were actually already downloaded by Xiang Zhu in /data/external_public/GlobalLipid

###So removed the /data/external_public/GlobalLipids2013 directory     

cd /data/external_public/GlobalLipid

mkdir /mnt/lustre/home/mturchin20/Data/GlobalLipids2013

~~~
[  mturchin20@spudling26  /data/external_public/GlobalLipid]$ls -lrt
total 245376
-rw-r--r-- 1 xiangzhu xiangzhu  3828870 Jun  4  2014 Mc_LDL.txt.gz
-rw-r--r-- 1 xiangzhu xiangzhu  3851583 Jun  4  2014 Mc_HDL.txt.gz
-rw-rw-r-- 1 xiangzhu xiangzhu  3834784 Jun  4  2014 Mc_TG.txt.gz
-rw-rw-r-- 1 xiangzhu xiangzhu  3856060 Jun  4  2014 Mc_TC.txt.gz
-rw-r--r-- 1 xiangzhu xiangzhu 58851380 Jun  4  2014 jointGwasMc_LDL.txt.gz
-rw-r--r-- 1 xiangzhu xiangzhu 59025777 Jun  4  2014 jointGwasMc_HDL.txt.gz
-rw-r--r-- 1 xiangzhu xiangzhu 58866740 Jun  4  2014 jointGwasMc_TG.txt.gz
-rw-r--r-- 1 xiangzhu xiangzhu 59110550 Jun  4  2014 jointGwasMc_TC.txt.gz
-rw-rw-r-- 1 xiangzhu xiangzhu      862 Jun  4  2014 README.txt
[  mturchin20@spudling26  /data/external_public/GlobalLipid]$for i in `ls -lrt | awk '{ print $9 }' | grep jointGwas`; do echo $i; zcat $i | head -n 5; done
jointGwasMc_LDL.txt.gz
SNP_hg18        SNP_hg19        rsid    A1      A2      beta    se      N       P-value Freq.A1.1000G.EUR
chr10:10000135  chr10:9960129   rs4747841       a       g       0.0037  0.0052  89138.00        0.7158  0.4908
chr10:10000265  chr10:9960259   rs4749917       c       t       0.0033  0.0052  89138.00        0.7748  0.4908
chr10:100002729 chr10:100012739 rs737656        a       g       0.0099  0.0054  89888.00        0.04    0.3206
chr10:100002880 chr10:100012890 rs737657        a       g       0.0084  0.0054  89888.00        0.08428 0.3206
jointGwasMc_HDL.txt.gz
SNP_hg18        SNP_hg19        rsid    A1      A2      beta    se      N       P-value Freq.A1.1000G.EUR
chr10:10000135  chr10:9960129   rs4747841       g       a       0.0026  0.0048  93561.00        0.7538  0.5092
chr10:10000265  chr10:9960259   rs4749917       t       c       0.0028  0.0048  93561.00        0.7295  0.5092
chr10:100002729 chr10:100012739 rs737656        g       a       0.0098  0.0049  94311.00        0.09234 0.6794
chr10:100002880 chr10:100012890 rs737657        g       a       0.0102  0.0049  94311.00        0.0934  0.6794
jointGwasMc_TG.txt.gz
SNP_hg18        SNP_hg19        rsid    A1      A2      beta    se      N       P-value Freq.A1.1000G.EUR
chr10:10000135  chr10:9960129   rs4747841       a       g       0.0020  0.0047  90263.00        0.7567  0.4908
chr10:10000265  chr10:9960259   rs4749917       c       t       0.002   0.0047  90263.00        0.7612  0.4908
chr10:100002729 chr10:100012739 rs737656        a       g       0.0094  0.0048  91013.00        0.06856 0.3206
chr10:100002880 chr10:100012890 rs737657        a       g       0.0094  0.0048  91013.00        0.07994 0.3206
jointGwasMc_TC.txt.gz
SNP_hg18        SNP_hg19        rsid    A1      A2      beta    se      N       P-value Freq.A1.1000G.EUR
chr10:10000135  chr10:9960129   rs4747841       a       g       0.0026  0.0051  93845.00        0.653   0.4908
chr10:10000265  chr10:9960259   rs4749917       c       t       0.0024  0.0051  93845.00        0.6915  0.4908
chr10:100002729 chr10:100012739 rs737656        a       g       0.0071  0.0052  94595.00        0.1653  0.3206
chr10:100002880 chr10:100012890 rs737657        a       g       0.0062  0.0052  94595.00        0.255   0.3206
[  mturchin20@spudling26  /data/external_public/GlobalLipid]$for i in `ls -lrt | awk '{ print $9 }' | grep jointGwas`; do paste <(zcat $i | wc) <(zcat $i | perl -lane 'print $#F;' | sort | uniq -c) | xargs echo $i; done
jointGwasMc_LDL.txt.gz 2437752 24377520 194868366 2437752 9
jointGwasMc_HDL.txt.gz 2447442 24474420 195649445 2447442 9
jointGwasMc_TG.txt.gz 2439433 24394330 194998474 2439433 9
jointGwasMc_TC.txt.gz 2446982 24469820 195629190 2446982 9
[  mturchin20@spudling26  /data/external_public/GlobalLipid]$for i in `ls -lrt | awk '{ print $9 }' | grep jointGwas`; do echo $i; paste -d @ <(zcat $i | perl -lane ' if ($F[0] !~ m/chr(1|2|3|4|5|6|7|8|9|10|11|12|13|14|15|16|17|18|19|20|21|22|X|Y|M):\d+/) { $F[0]; } ' | sort | uniq -c ) <(zcat $i | perl -lane ' if ($F[1] !~ m/chr(1|2|3|4|5|6|7|8|9|10|11|12|13|14|15|16|17|18|19|20|21|22|X|Y|M):\d+/) { $F[1]; } ' | sort | uniq -c ) <(zcat $i | sed 's/e-//g'| perl -lane 'if ($F[2] !~ m/^rs\d+/) { print join("\t", @F); }' | awk '{ print $3 }' | sort | uniq -c) <(zcat $i | sed 's/e-//g'| perl -lane 'if ($F[7] =~ m/[^\d\.]/) { print join("\t", @F); }' | awk '{ print $8 }' | sort | uniq -c ) <(zcat $i | sed 's/e-//g'| perl -lane 'if ($F[8] =~ m/[^\d\.]/) { print join("\t", @F); }' | awk '{ print $9 }' | sort | uniq -c ) <(zcat $i | sed 's/e-//g'| perl -lane 'if ($F[9] =~ m/[^\d\.]/) { print join("\t", @F); }' | awk '{ print $10 }' | sort | uniq -c ) <(zcat $i | sed 's/e-//g'| perl -lane 'if ($F[9] !~ m/\.0?[1-9]+/) { print $F[9]; } ' | sort | uniq -c ) | sed 's/@/ @ /g'| column -s $'@' -t ; done
jointGwasMc_LDL.txt.gz
	795 .             1 N          1 P-value          1 Freq.A1.1000G.EUR        103 0
	1 rsid                                     185802 NA                       329 0.001319
											391 0.002639
											574 0.003958
											717 0.005277
											1129 0.006596
											1327 0.007916
											1740 0.009235
											187475 1
											185802 NA
jointGwasMc_HDL.txt.gz
	803 .             1 N          1 P-value          1 Freq.A1.1000G.EUR        136 0
	1 rsid                                     186081 NA                       401 0.001319
											446 0.002639
											618 0.003958
											918 0.005277
											1138 0.006596
											1421 0.007916
											1896 0.009235
											189040 1
											186081 NA
jointGwasMc_TG.txt.gz
	796 .             1 N          1 P-value          1 Freq.A1.1000G.EUR        100 0
	1 rsid                                     185919 NA                       368 0.001319
											433 0.002639
											631 0.003958
											744 0.005277
											1147 0.006596
											1324 0.007916
											1810 0.009235
											187791 1
											185919 NA
jointGwasMc_TC.txt.gz
	805 .             1 N          1 P-value          1 Freq.A1.1000G.EUR        123 0
	1 rsid                                     186648 NA                       446 0.001319
											470 0.002639
											663 0.003958
											822 0.005277
											1288 0.006596
											1461 0.007916
											1831 0.009235
											188424 1
											186648 NA
[  mturchin20@spudling26  /data/external_public/GlobalLipid]$for i in `ls -lrt | awk '{ print $9 }' | grep jointGwas`; do echo $i; zcat $i | sed 's/P-value/P_value/g' |
perl -lane 'print join("\t", @F);' | R -q -e "library(data.table); Data1 <- fread('file:///dev/stdin', header=TRUE); quantile(Data1\$Freq.A1.1000G.EUR, na.rm=TRUE); qu
antile(Data1\$N, na.rm=TRUE); quantile(Data1\$P_value, na.rm=TRUE); png(\"/mnt/lustre/home/mturchin20/Data/GlobalLipids2013/"$i".QCCheck.vs1.png\", height=2000, width=2
000, res=200); par(mfrow=c(2,2)); hist(Data1\$Freq.A1.1000G.EUR, breaks=100); hist(Data1\$N, breaks=100); hist(Data1\$P_value, breaks=100); expP <- -log10(c(1:length(Da
ta1\$P_value["\!"is.na(Data1\$P_value)])) / (length(Data1\$P_value["\!"is.na(Data1\$P_value)])+1)); plot(expP, -log10(sort(Data1\$P_value["\!"is.na(Data1\$P_value)])),
xlab=\"Expected p-Values\", ylab=\"Observed p-Values\"); abline(0, 1, col=\"RED\"); dev.off();"; done
jointGwasMc_LDL.txt.gz
> library(data.table); Data1 <- fread('file:///dev/stdin', header=TRUE); quantile(Data1$Freq.A1.1000G.EUR, na.rm=TRUE); quantile(Data1$N, na.rm=TRUE); quantile(Data1$P_value, na.rm=TRUE); png("/mnt/lustre/home/mturchin20/Data/GlobalLipids2013/jointGwasMc_LDL.txt.gz.QCCheck.vs1.png", height=2000, width=2000, res=200); par(mfrow=c(2,2)); hist(Data1$Freq.A1.1000G.EUR, breaks=100); hist(Data1$N, breaks=100); hist(Data1$P_value, breaks=100); expP <- -log10(c(1:length(Data1$P_value[!is.na(Data1$P_value)])) / (length(Data1$P_value[!is.na(Data1$P_value)])+1)); plot(expP, -log10(sort(Data1$P_value[!is.na(Data1$P_value)])), xlab="Expected p-Values", ylab="Observed p-Values"); abline(0, 1, col="RED"); dev.off();
Read 2437751 rows and 10 (of 10) columns from 0.181 GB file in 00:01:35
Warning message:
In fread("file:///dev/stdin", header = TRUE) :
C function strtod() returned ERANGE for one or more fields. The first was string input '3.85e-326'. It was read using (double)strtold() as numeric value 0.0000000000000000E+00 (displayed here using %.16E); loss of accuracy likely occurred. This message is designed to tell you exactly what has been done by fread's C code, so you can search yourself online for many references about double precision accuracy and these specific C functions. You may wish to use colClasses to read the column as character instead and then coerce that column using the Rmpfr package for greater accuracy.
0%    25%    50%    75%   100%
0.0000 0.2388 0.5541 0.8470 1.0000
0%    25%    50%    75%   100%
50000  87745  89872  89888 173082
0%    25%    50%    75%   100%
0.0000 0.2434 0.4968 0.7488 1.0000
null device
1
>
>
jointGwasMc_HDL.txt.gz
> library(data.table); Data1 <- fread('file:///dev/stdin', header=TRUE); quantile(Data1$Freq.A1.1000G.EUR, na.rm=TRUE); quantile(Data1$N, na.rm=TRUE); quantile(Data1$P_value, na.rm=TRUE); png("/mnt/lustre/home/mturchin20/Data/GlobalLipids2013/jointGwasMc_HDL.txt.gz.QCCheck.vs1.png", height=2000, width=2000, res=200); par(mfrow=c(2,2)); hist(Data1$Freq.A1.1000G.EUR, breaks=100); hist(Data1$N, breaks=100); hist(Data1$P_value, breaks=100); expP <- -log10(c(1:length(Data1$P_value[!is.na(Data1$P_value)])) / (length(Data1$P_value[!is.na(Data1$P_value)])+1)); plot(expP, -log10(sort(Data1$P_value[!is.na(Data1$P_value)])), xlab="Expected p-Values", ylab="Observed p-Values"); abline(0, 1, col="RED"); dev.off();
Read 2447441 rows and 10 (of 10) columns from 0.182 GB file in 00:01:35
Warning message:
In fread("file:///dev/stdin", header = TRUE) :
C function strtod() returned ERANGE for one or more fields. The first was string input '1.79e-339'. It was read using (double)strtold() as numeric value 0.0000000000000000E+00 (displayed here using %.16E); loss of accuracy likely occurred. This message is designed to tell you exactly what has been done by fread's C code, so you can search yourself online for many references about double precision accuracy and these specific C functions. You may wish to use colClasses to read the column as character instead and then coerce that column using the Rmpfr package for greater accuracy.
0%    25%    50%    75%   100%
0.0000 0.2388 0.5515 0.8496 1.0000
0%    25%    50%    75%   100%
50000  92613  94294  94311 187167
0%    25%    50%    75%   100%
0.0000 0.2435 0.4968 0.7483 1.0000
null device
1
>
>
jointGwasMc_TG.txt.gz
> library(data.table); Data1 <- fread('file:///dev/stdin', header=TRUE); quantile(Data1$Freq.A1.1000G.EUR, na.rm=TRUE); quantile(Data1$N, na.rm=TRUE); quantile(Data1$P_value, na.rm=TRUE); png("/mnt/lustre/home/mturchin20/Data/GlobalLipids2013/jointGwasMc_TG.txt.gz.QCCheck.vs1.png", height=2000, width=2000, res=200); par(mfrow=c(2,2)); hist(Data1$Freq.A1.1000G.EUR, breaks=100); hist(Data1$N, breaks=100); hist(Data1$P_value, breaks=100); expP <- -log10(c(1:length(Data1$P_value[!is.na(Data1$P_value)])) / (length(Data1$P_value[!is.na(Data1$P_value)])+1)); plot(expP, -log10(sort(Data1$P_value[!is.na(Data1$P_value)])), xlab="Expected p-Values", ylab="Observed p-Values"); abline(0, 1, col="RED"); dev.off();
Read 2439432 rows and 10 (of 10) columns from 0.182 GB file in 00:01:29
0%    25%    50%    75%   100%
0.0000 0.2401 0.5567 0.8470 1.0000
0%      25%      50%      75%     100%
50002.0  88741.0  90996.0  91013.0 177860.9
0%        25%        50%        75%       100%
1.100e-249  2.463e-01  4.983e-01  7.491e-01  1.000e+00
null device
1
>
>
jointGwasMc_TC.txt.gz
> library(data.table); Data1 <- fread('file:///dev/stdin', header=TRUE); quantile(Data1$Freq.A1.1000G.EUR, na.rm=TRUE); quantile(Data1$N, na.rm=TRUE); quantile(Data1$P_value, na.rm=TRUE); png("/mnt/lustre/home/mturchin20/Data/GlobalLipids2013/jointGwasMc_TC.txt.gz.QCCheck.vs1.png", height=2000, width=2000, res=200); par(mfrow=c(2,2)); hist(Data1$Freq.A1.1000G.EUR, breaks=100); hist(Data1$N, breaks=100); hist(Data1$P_value, breaks=100); expP <- -log10(c(1:length(Data1$P_value[!is.na(Data1$P_value)])) / (length(Data1$P_value[!is.na(Data1$P_value)])+1)); plot(expP, -log10(sort(Data1$P_value[!is.na(Data1$P_value)])), xlab="Expected p-Values", ylab="Observed p-Values"); abline(0, 1, col="RED"); dev.off();
Read 2446981 rows and 10 (of 10) columns from 0.182 GB file in 00:01:31
0%    25%    50%    75%   100%
0.0000 0.2375 0.5528 0.8483 1.0000
0%       25%       50%       75%      100%
50001.00  92609.49  94577.00  94595.00 187365.00
0%        25%        50%        75%       100%
1.560e-283  2.425e-01  4.970e-01  7.484e-01  1.000e+00
null device
1
>
>
~~~

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs1/process.MTedits.ForGlobalLipids2010.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/process.MTedits.ForGlobalLipids2013.vs1.R

#20150832 Note -- there are multiple duplicate rsIDs in some of these files so use column 2 (chrbp information). This was discovered by me already previously, see code below
~~~
...

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/multivariate/globallipids/process.txt /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/process.MTedits.vs1.R

#Confirming column 2 in file /data/external_public/GlobalLipid/Mc_LDL.txt.gz has only unique entries
zcat /data/external_public/GlobalLipid/Mc_LDL.txt.gz | awk '{ print $2 }' | sort | uniq -c | sort -k 1,1 | tail -n 20
...
~~~

#Looks like results from before are replicated (aside from order....my b)
~~~
[  mturchin20@spudling29  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1]$cat GlobalLipids2013.dtlesssignif.vs1.txt | wc
  12463   12463 1665961
[  mturchin20@spudling29  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1]$cat dtlesssignif.vs2.txt | wc
  12463   12463 1665957
[  mturchin20@spudling29  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1]$cat GlobalLipids2013.RSS0.vs1.txt
Z.LDL,Z.HDL,Z.TC,Z.TG
1,0.00372130546934003,0.615708639149167,0.0162974156495297
0.00372130546934003,1,0.0152772659325678,0.0952387554368848
0.615708639149167,0.0152772659325678,1,0.061496929582914
0.0162974156495297,0.0952387554368848,0.061496929582914,1
[  mturchin20@spudling29  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1]$cat RSS0.
RSS0.txt      RSS0.vs2.txt
[  mturchin20@spudling29  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1]$cat RSS0.vs2.txt
Z.tc,Z.tg,Z.hdl,Z.ldl
1,0.061496929582914,0.0152772659325678,0.615708639149167
0.061496929582914,1,0.0952387554368848,0.0162974156495297
0.0152772659325678,0.0952387554368848,1,0.00372130546934003
0.615708639149167,0.0162974156495297,0.00372130546934003,1
~~~    

zcat /data/external_public/GlobalLipid/jointGwasMc_LDL.txt.gz | awk '{ print $2, "\t", $3, "\t", $10, "\t", $9, "\t", $8 }' | perl -lane 'my @vals1 = split(/:/, $F[0]); splice(@F, 0, 1); splice(@F, 0, 0, ($vals1[0], $vals1[1])); print join("\t", @F);' | gzip >  /mnt/lustre/home/mturchin20/Data/GlobalLipids2013/jointGwasMc_LDL.formatted.txt.gz &
zcat /data/external_public/GlobalLipid/jointGwasMc_HDL.txt.gz | awk '{ print $2, "\t", $3, "\t", $10, "\t", $9, "\t", $8 }' | perl -lane 'my @vals1 = split(/:/, $F[0]); splice(@F, 0, 1); splice(@F, 0, 0, ($vals1[0], $vals1[1])); print join("\t", @F);' | gzip >  /mnt/lustre/home/mturchin20/Data/GlobalLipids2013/jointGwasMc_HDL.formatted.txt.gz & 
zcat /data/external_public/GlobalLipid/jointGwasMc_TC.txt.gz | awk '{ print $2, "\t", $3, "\t", $10, "\t", $9, "\t", $8 }' | perl -lane 'my @vals1 = split(/:/, $F[0]); splice(@F, 0, 1); splice(@F, 0, 0, ($vals1[0], $vals1[1])); print join("\t", @F);' | gzip >  /mnt/lustre/home/mturchin20/Data/GlobalLipids2013/jointGwasMc_TC.formatted.txt.gz &
zcat /data/external_public/GlobalLipid/jointGwasMc_TG.txt.gz | awk '{ print $2, "\t", $3, "\t", $10, "\t", $9, "\t", $8 }' | perl -lane 'my @vals1 = split(/:/, $F[0]); splice(@F, 0, 1); splice(@F, 0, 0, ($vals1[0], $vals1[1])); print join("\t", @F);' | gzip >  /mnt/lustre/home/mturchin20/Data/GlobalLipids2013/jointGwasMc_TG.formatted.txt.gz & 




##20151008 Note -- Checking to see if new hits from 2010 run are within 1Mb of the 2013 hits

cat ../../GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.txt | tail -n +2 | perl -lane 'print $F[2], "\t", ((split(/hr/, $F[0])))[1], "\t", $F[1];' > ../../GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.MarkerChrBP.txt

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/multivariate/globallipids/newtophits.txt | perl -lane 'print $F[0], "\t", $F[2], "\t", $F[4];' | tail -n +2 | python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 - --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.MarkerChrBP.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.MarkerChrBP.wGWASannot.txt

#Note -- the marker missing is rs1408579 -- ERLIN1
#The two olfactory region SNPs rs11229252 and rs11227638 may have been represented by another farther away SNP?
~~~
[  mturchin20@spudling26  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs1]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.MarkerChrBP.wGWASannot.txt | awk '{ print $4 }' | sort | uniq -c   
142 0
8 1
7 2
[  mturchin20@spudling26  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs1]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.MarkerChrBP.wGWASannot.txt | awk '{ print $4, "\t", $5 }' | sort | uniq -c
142 0   
1 1   rs10904908
1 1   rs11246602
1 1   rs13326165
1 1   rs267733
1 1   rs4722551
1 1   rs499974
1 1   rs970548
1 1   rs998584
1 2   rs10422101
1 2   rs10490632
1 2   rs12739698
1 2   rs17134533
1 2   rs4942505
1 2   rs6951245
1 2   rs762861
~~~


#20151007
#CSHL Poster work

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GLC.MTedits.For2013.vs3.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GLC.MTedits.For2013.vs3.ForCSHLPoster.R

#20160428 NOTE -- Editing the above 'ForCSHLPoster' a little to get figures/results for NHS presentation as well




#20160220
#Correcting sign issue -- quick hack to get results

#NOTE -- flipping sign on p-value to make coding downstream easier for this, ie I'm not implying something by creating negative p-values here. This is just for a quick run through.

join <(zcat /data/external_public/GlobalLipid/jointGwasMc_HDL.txt.gz | awk '{ print $1, "\t", $4 }') <(zcat /data/external_public/GlobalLipid/jointGwasMc_LDL.txt.gz) | perl -lane 'if ($F[1] ne $F[4]) { $F[9] = -1 * $F[9]; } splice(@F, 1, 1); print join("\t", @F);' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/jointGwasMc_LDL.MatchedToHDL.txt.gz
join <(zcat /data/external_public/GlobalLipid/jointGwasMc_HDL.txt.gz | awk '{ print $1, "\t", $4 }') <(zcat /data/external_public/GlobalLipid/jointGwasMc_TG.txt.gz) | perl -lane 'if ($F[1] ne $F[4]) { $F[9] = -1 * $F[9]; } splice(@F, 1, 1); print join("\t", @F);' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/jointGwasMc_TG.MatchedToHDL.txt.gz 
join <(zcat /data/external_public/GlobalLipid/jointGwasMc_HDL.txt.gz | awk '{ print $1, "\t", $4 }') <(zcat /data/external_public/GlobalLipid/jointGwasMc_TC.txt.gz) | perl -lane 'if ($F[1] ne $F[4]) { $F[9] = -1 * $F[9]; } splice(@F, 1, 1); print join("\t", @F);' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/jointGwasMc_TC.MatchedToHDL.txt.gz 

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/DataOrganization.PythonVLookUp.CreateFullAnnotFile.vs1.py /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/DataOrganization.PythonVLookUp.CreateFullAnnotFile.vs1.SignCrrct.vs1.py

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/DataOrganization.PythonVLookUp.CreateFullAnnotFile.vs1.SignCrrct.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/jointGwasMc_LDL.MatchedToHDL.txt.gz,/data/external_public/GlobalLipid/jointGwasMc_HDL.txt.gz,/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/jointGwasMc_TG.MatchedToHDL.txt.gz,/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/jointGwasMc_TC.MatchedToHDL.txt.gz | perl -lane 'my @vals1 = split(/:/, $F[1]); my $chr = "NA"; if ($vals1[0] =~ m/chr(\d+)/) { $chr = $1; } splice(@F, 1, 1, ($chr, $vals1[1])); push(@F, "NA"); print join("\t", @F);' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/jointGwasMc_AllPheno.Annot.vs1.SignCrrct.RAF.txt.gz 

#Added the following header to /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/jointGwasMc_AllPheno.Annot.vs1.SignCrrct.RAF.txt.gz
#posHg18 chr pos snp a1 a2 raf beta_LDL se_LDL n_LDL beta_HDL se_HDL n_HDL beta_TG se_TG n_TG beta_TC se_TC n_TC annot gene

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/process.MTedits.ForGlobalLipids2013.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/process.MTedits.ForGlobalLipids2013.vs1.SignCrrct.vs1.R

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/DataOrganization.PythonVLookUp.PutTogetherProcessAndAnnotFiles.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GlobalLipids2013.dtlesssignif.vs1.SignCrrct.vs1.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/jointGwasMc_AllPheno.Annot.vs1.SignCrrct.RAF.txt.gz | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GlobalLipids2013.dtlesssignif.vs1.SignCrrct.vs1.annot.RAF.txt.gz 

#Added the following header /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GlobalLipids2013.dtlesssignif.vs1.SignCrrct.vs1.annot.RAF.txt.gz 
#posHg18 chr pos snp a1 a2 raf beta_LDL se_LDL n_LDL beta_HDL se_HDL n_HDL beta_TG se_TG n_TG beta_TC se_TC n_TC annot gene Z.tg Z.tc Z.ldl Z.hdl mvstat mvp unip 

zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GlobalLipids2013.dtlesssignif.vs1.SignCrrct.vs1.annot.RAF.txt.gz | perl -lane 'if ($F[6] > .5) { $F[6] = 1 - $F[6]; $F[21] = $F[21] * -1; $F[22] = $F[22] * -1; $F[23] = $F[23] * -1; $F[24] = $F[24] * -1; } print join("\t", @F);' | sed 's/raf/maf/g' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GlobalLipids2013.dtlesssignif.vs1.SignCrrct.vs1.annot.MAF.txt.gz

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GLC.MTedits.For2013.vs3.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GLC.MTedits.For2013.vs3.SignCrrct.vs1.R 

#20160630

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GLC.MTedits.For2013.vs3.SignCrrct.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GLC.MTedits.For2013.vs3.SignCrrct.vs1.R


#20161109
#Vs2 -- Running things through version of bmass() code

mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2 

#2017023 NOTE -- keeping the code below unchange since I used it and it was copy/pasted elsewhere, but should /not/ use `== "-"`, it is erroneous; use `eq "-"` instead as is normal procedure in perl. Using the numeric equivalence operator == messes up with a string such as `"-"`; `eq` handles this properly. I've corrected this elsewhere and I'm 'commenting' the below lines so as to create a visual cue that something is amiss here if I quickly grab/copy/paste the code chunk here again
#zcat /data/external_public/GlobalLipid/jointGwasMc_HDL.txt.gz | sed 's/:/ /g' | sed 's/chr//g' | awk '{ print $1 "\t" $2 "\t" $12 "\t" $6 "\t" $7 "\t" $8 "\t" $11 "\t" $10 }' | grep -v hg | grep -v rs | perl -lane 'if ($F[5] > 0) { $F[5] = "+"; } elsif ($F[5] < 0) { $F[5] = "-"; } else { $F[5] = 0;} if ($F[5] == "-") { ($F[3], $F[4]) = ($F[4], $F[3]); $F[5] = "+"; $F[2] = 1 - $F[2]; } print join("\t", @F); ' | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_HDL.IncAllele.bmass.formatted.txt.gz
#zcat /data/external_public/GlobalLipid/jointGwasMc_LDL.txt.gz | sed 's/:/ /g' | sed 's/chr//g' | awk '{ print $1 "\t" $2 "\t" $12 "\t" $6 "\t" $7 "\t" $8 "\t" $11 "\t" $10 }' | grep -v hg | grep -v rs | perl -lane 'if ($F[5] > 0) { $F[5] = "+"; } elsif ($F[5] < 0) { $F[5] = "-"; } else { $F[5] = 0;} if ($F[5] == "-") { ($F[3], $F[4]) = ($F[4], $F[3]); $F[5] = "+"; $F[2] = 1 - $F[2]; } print join("\t", @F); ' | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_LDL.IncAllele.bmass.formatted.txt.gz
#zcat /data/external_public/GlobalLipid/jointGwasMc_TG.txt.gz | sed 's/:/ /g' | sed 's/chr//g' | awk '{ print $1 "\t" $2 "\t" $12 "\t" $6 "\t" $7 "\t" $8 "\t" $11 "\t" $10 }' | grep -v hg | grep -v rs | perl -lane 'if ($F[5] > 0) { $F[5] = "+"; } elsif ($F[5] < 0) { $F[5] = "-"; } else { $F[5] = 0;} if ($F[5] == "-") { ($F[3], $F[4]) = ($F[4], $F[3]); $F[5] = "+"; $F[2] = 1 - $F[2]; } print join("\t", @F); ' | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TG.IncAllele.bmass.formatted.txt.gz
#zcat /data/external_public/GlobalLipid/jointGwasMc_TC.txt.gz | sed 's/:/ /g' | sed 's/chr//g' | awk '{ print $1 "\t" $2 "\t" $12 "\t" $6 "\t" $7 "\t" $8 "\t" $11 "\t" $10 }' | grep -v hg | grep -v rs | perl -lane 'if ($F[5] > 0) { $F[5] = "+"; } elsif ($F[5] < 0) { $F[5] = "-"; } else { $F[5] = 0;} if ($F[5] == "-") { ($F[3], $F[4]) = ($F[4], $F[3]); $F[5] = "+"; $F[2] = 1 - $F[2]; } print join("\t", @F); ' | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TC.IncAllele.bmass.formatted.txt.gz

join <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_HDL.IncAllele.bmass.formatted.txt.gz | awk '{ print $1 "_" $2 "\t" $4 }' | grep -v NA | sort -g -k 1,1) <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_LDL.IncAllele.bmass.formatted.txt.gz | awk '{ print $1 "_" $2 "\t" $0 }' | grep -v NA | sort -g -k 1,1) | perl -lane 'if ($F[1] eq $F[6]) { ($F[5], $F[6]) = ($F[6], $F[5]); $F[4] = 1 - $F[4]; if ($F[7] eq "-") { $F[7] = "+"; } elsif ($F[7] eq "+") { $F[7] = "-"; } else { $PH = 1; } } elsif ($F[1] eq $F[5]) { $PH = 1; } else { push(@F, "NoAlleleMatch"); } print join("\t", @F[2..$#F]);' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_LDL.IncAllele.bmass.formatted.MatchedToHDL.txt.gz 
join <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_HDL.IncAllele.bmass.formatted.txt.gz | awk '{ print $1 "_" $2 "\t" $4 }' | grep -v NA | sort -g -k 1,1) <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TG.IncAllele.bmass.formatted.txt.gz | awk '{ print $1 "_" $2 "\t" $0 }' | grep -v NA | sort -g -k 1,1) | perl -lane 'if ($F[1] eq $F[6]) { ($F[5], $F[6]) = ($F[6], $F[5]); $F[4] = 1 - $F[4]; if ($F[7] eq "-") { $F[7] = "+"; } elsif ($F[7] eq "+") { $F[7] = "-"; } else { $PH = 1; } } elsif ($F[1] eq $F[5]) { $PH = 1; } else { push(@F, "NoAlleleMatch"); } print join("\t", @F[2..$#F]);' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TG.IncAllele.bmass.formatted.MatchedToHDL.txt.gz 
join <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_HDL.IncAllele.bmass.formatted.txt.gz | awk '{ print $1 "_" $2 "\t" $4 }' | grep -v NA | sort -g -k 1,1) <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TC.IncAllele.bmass.formatted.txt.gz | awk '{ print $1 "_" $2 "\t" $0 }' | grep -v NA | sort -g -k 1,1) | perl -lane 'if ($F[1] eq $F[6]) { ($F[5], $F[6]) = ($F[6], $F[5]); $F[4] = 1 - $F[4]; if ($F[7] eq "-") { $F[7] = "+"; } elsif ($F[7] eq "+") { $F[7] = "-"; } else { $PH = 1; } } elsif ($F[1] eq $F[5]) { $PH = 1; } else { push(@F, "NoAlleleMatch"); } print join("\t", @F[2..$#F]);' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TC.IncAllele.bmass.formatted.MatchedToHDL.txt.gz 

zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_LDL.IncAllele.bmass.formatted.MatchedToHDL.txt.gz | grep -v NoAlleleMatch | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_LDL.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.txt.gz
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TG.IncAllele.bmass.formatted.MatchedToHDL.txt.gz | grep -v NoAlleleMatch | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TG.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.txt.gz
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TC.IncAllele.bmass.formatted.MatchedToHDL.txt.gz | grep -v NoAlleleMatch | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TC.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.txt.gz

~~~
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2]$zcat /data/external_public/GlobalLipid/jointGwasMc_HDL.txt.gz | sed 's/:/ /g' | sed 's/chr//g' | awk '{ print $1 "\t" $2 "\t" $12 "\t" $6 "\t" $7 "\t" $8 "\t" $11 "\t" $10 }' | grep -v hg | perl -lane 'if ($F[5] > 0) { $F[5] = "+"; } elsif ($F[5] < 0) { $F[5] = "-"; } else { $F[5] = 0;} if ($F[5] == "-") { ($F[3], $F[4]) = ($F[4], $F[3]); $F[5] = "+"; $F[2] = 1 - $F[2]; } print join("\t", @F); ' | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | awk '{ print $6 }' | sort | uniq -c
2447441 +
      1 Direction
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_LDL.IncAllele.bmass.formatted.txt.gz | awk '{ print $6 }' | sort | uniq -c
2437751 +
      1 Direction
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TG.IncAllele.bmass.formatted.txt.gz | awk '{ print $6 }' | sort | uniq -c
2439432 +
      1 Direction
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TC.IncAllele.bmass.formatted.txt.gz | awk '{ print $6 }' | sort | uniq -c
2446981 +
      1 Direction
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2]$for i in `ls -lrt /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_*.IncAllele.bmass.formatted.txt.gz | awk '{ print $9 }'`; do echo $i; zcat $i | wc; zcat $i | grep -v NA | wc; done
/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_HDL.IncAllele.bmass.formatted.txt.gz
2447442 19579531 98456444
2447442 19579531 98456444
/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_LDL.IncAllele.bmass.formatted.txt.gz
2437752 19502013 98039596
2437752 19502013 98039596
/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TC.IncAllele.bmass.formatted.txt.gz
2446982 19575851 98439224
2446982 19575851 98439224
/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TG.IncAllele.bmass.formatted.txt.gz
2439433 19515461 98113749
2439433 19515461 98113749
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2]$zcat /data/external_public/GlobalLipid/jointGwasMc_HDL.txt.gz | awk '{ print $1 }' | grep rs | wc        
      5       5      48
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2]$zcat /data/external_public/GlobalLipid/jointGwasMc_LDL.txt.gz | awk '{ print $1 }' | grep rs | wc
      3       3      28
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2]$zcat /data/external_public/GlobalLipid/jointGwasMc_TC.txt.gz | awk '{ print $1 }' | grep rs | wc
      5       5      48
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2]$zcat /data/external_public/GlobalLipid/jointGwasMc_TG.txt.gz | awk '{ print $1 }' | grep rs | wc
      3       3      28
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2]$for i in `ls -lrt /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_*.IncAllele.bmass.formatted.txt.gz | awk '{ print $9 }'`; do echo $i; zcat $i | wc; zcat $i | grep -v rs | wc; done
/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_HDL.IncAllele.bmass.formatted.txt.gz
2447442 19579531 98456444
2447437 19579496 98456260
/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_LDL.IncAllele.bmass.formatted.txt.gz
2437752 19502013 98039596
2437749 19501992 98039485
/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TC.IncAllele.bmass.formatted.txt.gz
2446982 19575851 98439224
2446977 19575816 98439039
/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TG.IncAllele.bmass.formatted.txt.gz
2439433 19515461 98113749
2439430 19515440 98113638
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2]$for i in `ls -lrt  /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_*.IncAllele.bmass.formatted.MatchedToHDL.txt.gz | awk '{ print $9 }'`; do echo $i; zcat $i | awk '{ print $6 }' | sort | uniq -c; done
/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_LDL.IncAllele.bmass.formatted.MatchedToHDL.txt.gz
1147511 +
1290001 -
      1 Direction
/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TG.IncAllele.bmass.formatted.MatchedToHDL.txt.gz
 886926 +
1552083 -
      1 Direction
/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TC.IncAllele.bmass.formatted.MatchedToHDL.txt.gz
1325486 +
1120942 -
      1 Direction
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2]$for i in `ls -lrt  /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_*.IncAllele.bmass.formatted.MatchedToHDL.txt.gz | awk '{ print $9 }'`; do echo $i; zcat $i | perl -lane 'print $#F;' | sort | uniq -c; done;
/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_LDL.IncAllele.bmass.formatted.MatchedToHDL.txt.gz
2437513 7
/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TG.IncAllele.bmass.formatted.MatchedToHDL.txt.gz
2439010 7
/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TC.IncAllele.bmass.formatted.MatchedToHDL.txt.gz
2446429 7
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2]$for i in `ls -lrt /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_*.IncAllele.bmass.formatted.MatchedToHDL* | awk '{ print $9 }'`; do echo $i; zcat $i | wc; done
/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_LDL.IncAllele.bmass.formatted.MatchedToHDL.txt.gz
2437513 19500104 98036573
/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TG.IncAllele.bmass.formatted.MatchedToHDL.txt.gz
2439010 19512080 98100496
/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TC.IncAllele.bmass.formatted.MatchedToHDL.txt.gz
2446429 19571432 98426865
/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_LDL.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.txt.gz
2437513 19500104 98036573
/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TG.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.txt.gz
2439010 19512080 98100496
/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TC.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.txt.gz
2446429 19571432 98426865
~~~

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.MarkerChrBP.txt | awk '{ print $2 "\t" $3 }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.ChrBP.txt

R -q -e "library("devtools"); devtools::load_all("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/bmass"); 
HDL <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_HDL.IncAllele.bmass.formatted.txt.gz", header=T);
LDL <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_LDL.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.txt.gz", header=T);
TG <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TG.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.txt.gz", header=T);
TC <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TC.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.txt.gz", header=T);
GWASsnps <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.ChrBP.txt", header=F);
names(GWASsnps) <- c("Chr", "BP")
#bmassOutput1 <- bmass(c("HDL", "LDL", "TG", "TC"), GWASsnps=GWASsnps, NminThreshold = 1, bmassSeedValue=150);
#print(bmassOutput1);" > shana1.txt
DataSources = c("HDL", "LDL", "TG", "TC"); NminThreshold = 50000; bmassSeedValue=150;
ExpectedColumnNames=c("Chr", "BP", "MAF", "Direction", "pValue", "N"); GWASsnps_AnnotateWindow = 5e5; SNPMarginalUnivariateThreshold = 1e-6; SNPMarginalMultivariateThreshold = 1e-6; SigmaAlphas = c(0.005,0.0075,0.01,0.015,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.15); ProvidedPriors=NULL; UseFlatPriors=FALSE; PruneMarginalSNPs=TRUE; PruneMarginalSNPs_bpWindow=5e5;

#Output of R Code
~~~
> library("devtools"); devtools::load_all("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/bmass");                                                                                                                   HDL <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_HDL.IncAllele.bmass.formatted.txt.gz", header=T);                                                           LDL <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_LDL.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.txt.gz", header=T);
gz", header=T);                                    GWASsnps <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.ChrBP.txt", header=F);                                             names(GWASsnps) <- c("Chr", "BP")
#bmassOutput1 <- bmass(c("HDL", "LDL", "TG", "TC"), GWASsnps=GWASsnps, NminThreshold = 1, bmassSeedValue=150);
#print(bmassOutput1);" > shana1.txt
DataSources = c("HDL", "LDL", "TG", "TC"); NminThreshold = 50000; bmassSeedValue=150;
ExpectedColumnNames=c("Chr", "BP", "MAF", "Direction", "pValue", "N"); GWASsnps_AnnotateWindow = 5e5; SNPMarginalUnivariateThreshold = 1e-6; SNPMarginalMultivariateThreshold = 1e-6; SigmaAlphas = c(0.005,0.0075,0.01,0.015,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.15); ProvidedPriors=NULL; UseFlatPriors=FALSE; PruneMarginalSNPs=TRUE; PruneMarginalSNPs_bpWindow=5e5;
Loading bmass
> TG <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TG.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.txt.gz", header=T);                                    TC <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TC.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.txt.gz", header=T);                                    GWASsnps <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.ChrBP.txt", header=F);                                             names(GWASsnps) <- c("Chr", "BP")
> #bmassOutput1 <- bmass(c("HDL", "LDL", "TG", "TC"), GWASsnps=GWASsnps, NminThreshold = 1, bmassSeedValue=150);
> #print(bmassOutput1);" > shana1.txt
> DataSources = c("HDL", "LDL", "TG", "TC"); NminThreshold = 50000; bmassSeedValue=150;
> ExpectedColumnNames=c("Chr", "BP", "MAF", "Direction", "pValue", "N"); GWASsnps_AnnotateWindow = 5e5; SNPMarginalUnivariateThreshold = 1e-6; SNPMarginalMultivariateThreshold = 1e-6; SigmaAlphas = c(0.005,0.0075,0.01,0.015,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.15); ProvidedPriors=NULL; UseFlatPriors=FALSE; PruneMarginalSNPs=TRUE; PruneMarginalSNPs_bpWindow=5e5;
>
>
>         bmassOutput <- list()
>         bmassOutput$MergedDataSources <- NULL
>         if (!is.null(MergedDataSources)) {
+                 bmassOutput$MergedDataSources <- MergedDataSources
+         }
Error: object 'MergedDataSources' not found
>         bmassOutput$ZScoresCorMatrix <- NULL
>         bmassOutput$MarginalSNPs <- list()
>         bmassOutput$Models <- NULL
>         bmassOutput$ModelPriors <- NULL
>         bmassOutput$PreviousSNPs <- list()
>         bmassOutput$NewSNPs <- list()
>         bmassOutput$GWASlogBFMinThreshold <- NULL
>         #LogFile <- c()
>         bmassOutput$LogFile <- c()
>
>         bmassOutput$LogFile <- rbind(bmassOutput$LogFile, paste(format(Sys.time()), " -- beginning bmass.", sep=""))
>
>                 bmassOutput$LogFile <- CheckIndividualDataSources(DataSources, GWASsnps, ExpectedColumnNames, SigmaAlphas, bmassOutput$MergedDataSources, ProvidedPriors, UseFlatPriors, PruneMarginalSNPs, PruneMarginalSNPs_bpWindow, SNPMarginalUnivariateThreshold, SNPMarginalMultivariateThreshold, NminThreshold, bmassSeedValue, bmassOutput$LogFile)
>
>                 bmassOutput[c("MergedDataSources", "LogFile")] <- MergeDataSources(DataSources, bmassOutput$LogFile)[c("MergedDataSources", "LogFile")]
>
>         bmassOutput[c("MergedDataSources", "LogFile")] <- AnnotateMergedDataWithGWASSNPs(bmassOutput$MergedDataSources, GWASsnps, GWASsnps_AnnotateWindow, bmassOutput$LogFile)[c("MergedDataSources", "LogFile")]
>
>         bmassOutput[c("MarginalSNPs", "ZScoresCorMatrix", "LogFile")] <- ProcessMergedAndAnnotatedDataSources(DataSources, bmassOutput$MergedDataSources, SNPMarginalUnivariateThreshold, SNPMarginalMultivariateThreshold, bmassOutput$LogFile)[c("MarginalSNPs", "ZScoresCorMatrix", "LogFile")]
Warning message:
In ProcessMergedAndAnnotatedDataSources(DataSources, bmassOutput$MergedDataSources,  :
  2016-11-10 00:07:29 -- One of your datafiles has p-values less than the threshold which R can properly convert them to log-scale, meaning their log-values return as 'Infinite'. bmass automatically replaces these 'Infinite' values with the max, non-infinite Z-scores available, but it is recommended you self-check this as well.
>
>
>         bmassOutput[c("MarginalSNPs", "Models", "ModelPriors", "LogFile")] <- GetLogBFsFromData(DataSources, bmassOutput$MarginalSNPs, bmassOutput$ZScoresCorMatrix, SigmaAlphas, bmassOutput$LogFile)[c("MarginalSNPs", "Models", "ModelPriors", "LogFile")]
>
>         bmassOutput[c("MarginalSNPs", "PreviousSNPs", "ModelPriors", "GWASlogBFMinThreshold", "LogFile")] <- DetermineAndApplyPriors(DataSources, bmassOutput$MarginalSNPs, GWASsnps, SigmaAlphas, bmassOutput$Models, bmassOutput$ModelPriors, ProvidedPriors, UseFlatPriors, bmassSeedValue, bmassOutput$LogFile)[c("MarginalSNPs", "PreviousSNPs", "ModelPriors", "GWASlogBFMinThreshold", "LogFile")]
>                  
>         bmassOutput[c("MarginalSNPs", "PreviousSNPs", "NewSNPs", "LogFile")] <- FinalizeAndFormatResults(DataSources, bmassOutput$MarginalSNPs, bmassOutput$PreviousSNPs, GWASsnps, bmassOutput$GWASlogBFMinThreshold, SigmaAlphas, bmassOutput$Models, bmassOutput$ModelPriors, NminThreshold, PruneMarginalSNPs, PruneMarginalSNPs_bpWindow, bmassOutput$LogFile)[c("MarginalSNPs", "PreviousSNPs", "NewSNPs", "LogFile")]
> dim(bmassOutput$NewSNPs$SNPs)
[1] 65 29
> dim(bmassOutput$PreviousSNPs$SNPs)
[1] 148  29
> bmassOutput$GWASlogBFMinThreshold
[1] 4.28084
> bmassOutput$ZScoresCorMatrix
            HDL_ZScore  LDL_ZScore  TG_ZScore TC_ZScore
HDL_ZScore  1.00000000 -0.08715854 -0.3635593 0.1531368
LDL_ZScore -0.08715854  1.00000000  0.1585048 0.8197523
TG_ZScore  -0.36355928  0.15850485  1.0000000 0.2877814
TC_ZScore   0.15313683  0.81975235  0.2877814 1.0000000
> dim(bmassOutput1$NewSNPs$SNPs)
[1] 65 29
> dim(bmassOutput1$PreviousSNPs$SNPs)
[1] 148  29
> bmassOutput1$GWASlogBFMinThreshold
[1] 4.28084
> bmassOutput1$ZScoresCorMatrix     
            HDL_ZScore  LDL_ZScore  TG_ZScore TC_ZScore
HDL_ZScore  1.00000000 -0.08715854 -0.3635593 0.1531368
LDL_ZScore -0.08715854  1.00000000  0.1585048 0.8197523
TG_ZScore  -0.36355928  0.15850485  1.0000000 0.2877814
TC_ZScore   0.15313683  0.81975235  0.2877814 1.0000000
~~~


#20161115 -- checking to see if I start with same merged file from most recent `Vs1` run if I also get same `NewSNPs` count, eg liek ~84 or around there or something

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GlobalLipids2013.dtlesssignif.vs1.SignCrrct.vs1.annot.MAF.txt.gz /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/20161115_edits_GlobalLipids2013.dtlesssignif.vs1.SignCrrct.vs1.annot.MAF.txt.gz

#Put below header into /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/20161115_edits_GlobalLipids2013.dtlesssignif.vs1.SignCrrct.vs1.annot.MAF.txt.gz
#SNP Chr BP SNP A1 A2 MAF LDL_Beta LDL_SE LDL_N HDL_Beta HDL_SE HDL_N TG_Beta TG_SE TG_N TC_Beta TC_SE TC_N GWASannot Gene TG_ZScore TC_ZScore LDL_ZScore HDL_ZScore mvstat mvstat_log10pVal unistat_log10pVal


R code to use below
~~~

	library("devtools"); devtools::load_all("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/bmass");

        bmassOutput <- list()
        bmassOutput$MergedDataSources <- NULL
        if (!is.null(MergedDataSources)) {
                bmassOutput$MergedDataSources <- MergedDataSources
        }
        bmassOutput$ZScoresCorMatrix <- NULL
        bmassOutput$MarginalSNPs <- list()
        bmassOutput$Models <- NULL
        bmassOutput$ModelPriors <- NULL
        bmassOutput$PreviousSNPs <- list()
        bmassOutput$NewSNPs <- list()
        bmassOutput$GWASlogBFMinThreshold <- NULL
        #LogFile <- c()
        bmassOutput$LogFile <- c()

        bmassOutput$LogFile <- rbind(bmassOutput$LogFile, paste(format(Sys.time()), " -- beginning bmass.", sep=""))

	bmassOutput$MergedDataSources <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/20161115_edits_GlobalLipids2013.dtlesssignif.vs1.SignCrrct.vs1.annot.MAF.txt.gz", header=TRUE)
	bmassOutput$MergedDataSources <- bmassOutput$MergedDataSources[bmassOutput$MergedDataSources$MAF > 0 & !is.na(bmassOutput$MergedDataSources$MAF),]	

	bmassOutput$ZScoresCorMatrix <- as.matrix(read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GlobalLipids2013.RSS0.vs1.SignCrrct.vs1.txt", header=TRUE, sep=","))
	GWASsnps <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.ChrBP.txt", header=F);
	names(GWASsnps) <- c("Chr", "BP")
	#Z.LDL,Z.HDL,Z.TC,Z.TG
	DataSources = c("LDL", "HDL", "TC", "TG"); NminThreshold = 50000; bmassSeedValue=150;
	ExpectedColumnNames=c("Chr", "BP", "MAF", "Direction", "pValue", "N"); GWASsnps_AnnotateWindow = 5e5; SNPMarginalUnivariateThreshold = 1e-6; SNPMarginalMultivariateThreshold = 1e-6; SigmaAlphas = c(0.005,0.0075,0.01,0.015,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.15); ProvidedPriors=NULL; UseFlatPriors=FALSE; PruneMarginalSNPs=TRUE; PruneMarginalSNPs_bpWindow=5e5;
	bmassOutput$MergedDataSources$ChrBP <- paste(bmassOutput$MergedDataSources$Chr, "_", bmassOutput$MergedDataSources$BP, sep="")

	bmassOutput$MarginalSNPs$SNPs <- bmassOutput$MergedDataSources[bmassOutput$MergedDataSources$mvstat_log10pVal > -log10(SNPMarginalUnivariateThreshold) | bmassOutput$MergedDataSources$unistat_log10pVal > -log10(SNPMarginalMultivariateThreshold),]

        bmassOutput[c("MarginalSNPs", "Models", "ModelPriors", "LogFile")] <- GetLogBFsFromData(DataSources, bmassOutput$MarginalSNPs, bmassOutput$ZScoresCorMatrix, SigmaAlphas, bmassOutput$LogFile)[c("MarginalSNPs", "Models", "ModelPriors", "LogFile")]

#	Nmins <- apply(bmassOutput$MarginalSNPs$SNPs[,c("TC_N", "TG_N", "LDL_N", "HDL_N")], 1, min)
#	ZScores <- bmassOutput$MarginalSNPs$SNPs[,c("TC_ZScore", "TG_ZScore", "HDL_ZScore", "LDL_ZScore")]
#	MarginalSNPs_logBFs <- compute.allBFs.fromZscores(ZScores, bmassOutput$ZScoresCorMatrix, Nmins, bmassOutput$MarginalSNPs$SNPs$MAF, SigmaAlphas)

        bmassOutput[c("MarginalSNPs", "PreviousSNPs", "ModelPriors", "GWASlogBFMinThreshold", "LogFile")] <- DetermineAndApplyPriors(DataSources, bmassOutput$MarginalSNPs, GWASsnps, SigmaAlphas, bmassOutput$Models, bmassOutput$ModelPriors, ProvidedPriors, UseFlatPriors, bmassSeedValue, bmassOutput$LogFile)[c("MarginalSNPs", "PreviousSNPs", "ModelPriors", "GWASlogBFMinThreshold", "LogFile")]

        bmassOutput[c("MarginalSNPs", "PreviousSNPs", "NewSNPs", "LogFile")] <- FinalizeAndFormatResults(DataSources, bmassOutput$MarginalSNPs, bmassOutput$PreviousSNPs, GWASsnps, bmassOutput$GWASlogBFMinThreshold, SigmaAlphas, bmassOutput$Models, bmassOutput$ModelPriors, NminThreshold, PruneMarginalSNPs, PruneMarginalSNPs_bpWindow, bmassOutput$LogFile)[c("MarginalSNPs", "PreviousSNPs", "NewSNPs", "LogFile")]

HDL2 <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_HDL.IncAllele.bmass.formatted.txt.gz", header=T);
LDL2 <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_LDL.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.txt.gz", header=T);
TG2 <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TG.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.txt.gz", header=T);
TC2 <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TC.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.txt.gz", header=T);
GWASsnps2 <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.ChrBP.txt", header=F);
names(GWASsnps2) <- c("Chr", "BP")
bmassOutput2 <- bmass(c("HDL2", "LDL2", "TG2", "TC2"), GWASsnps=GWASsnps2, NminThreshold = 50000, PrintMergedData=TRUE, bmassSeedValue=150);
bmassOutput3 <- bmass(c("HDL2", "LDL2", "TG2", "TC2"), GWASsnps=GWASsnps2, NminThreshold = 50000, PrintMergedData=TRUE, PruneMarginalSNPs=FALSE, bmassSeedValue=150);

~~~

#Output from above
~~~
> table(bmassOutput$MergedDataSources$GWASannot)

   0    1    2 
1783  148 9642 
.
.
.
> FinalizeAndFormatResults(DataSources, bmassOutput$MarginalSNPs, bmassOutput$PreviousSNPs, GWASsnps, bmassOutput$GWASlogBFMinThreshold, SigmaAlphas, bmassOutput$Models, bmassOutput$ModelPriors, NminThreshold, PruneMarginalSNPs, PruneMarginalSNPs_bpWindow, bmassOutput$LogFile)[c("MarginalSNPs", "PreviousSNPs", "NewSNPs", "LogFile")]
Error in if (x == 1) { : missing value where TRUE/FALSE needed
> table(apply(bmassOutput$MarginalSNPs$logBFs, c(1,2), is.na))


   FALSE     TRUE
12883542   240240
>
> table(bmassOutput$MarginalSNPs$SNPs$MAF > .5)

FALSE 
11445 
> table(bmassOutput$MarginalSNPs$SNPs$MAF == 0)

FALSE  TRUE 
11309   136 
> table(bmassOutput$MarginalSNPs$SNPs$MAF == .5)

FALSE  TRUE 
11431    14 
> table(bmassOutput$MarginalSNPs$SNPs$MAF == 1)

FALSE 
11445 
.
.
.
> dim(bmassOutput$MergedDataSources)
[1] 11573    28
> dim(bmassOutput$MergedDataSources[bmassOutput$MergedDataSource$MAF == 0,])
[1] 264  28
> dim(bmassOutput$MergedDataSources[bmassOutput$MergedDataSource$MAF > 0,])
[1] 11437    28
> bmassOutput$MergedDataSources <- bmassOutput$MergedDataSources[bmassOutput$MergedDataSources$MAF > 0,]
> dim(bmassOutput$MergedDataSources)                                                                                                                                                                                           [1] 11437    28
> dim(bmassOutput$MergedDataSources[bmassOutput$MergedDataSource$MAF == 0,])
[1] 128  28
> dim(bmassOutput$MergedDataSources[bmassOutput$MergedDataSource$MAF > 0,])
[1] 11437    28
> head(bmassOutput$MergedDataSources[bmassOutput$MergedDataSource$MAF == 0,])
      SNP Chr BP SNP.1   A1   A2 MAF LDL_Beta LDL_SE LDL_N HDL_Beta HDL_SE
NA   <NA>  NA NA  <NA> <NA> <NA>  NA       NA     NA    NA       NA     NA
NA.1 <NA>  NA NA  <NA> <NA> <NA>  NA       NA     NA    NA       NA     NA
NA.2 <NA>  NA NA  <NA> <NA> <NA>  NA       NA     NA    NA       NA     NA
NA.3 <NA>  NA NA  <NA> <NA> <NA>  NA       NA     NA    NA       NA     NA
NA.4 <NA>  NA NA  <NA> <NA> <NA>  NA       NA     NA    NA       NA     NA
NA.5 <NA>  NA NA  <NA> <NA> <NA>  NA       NA     NA    NA       NA     NA
.
.
.
> table(apply(bmassOutput$MergedDataSources, c(1,2), is.na))

 FALSE   TRUE 
312343  11701 
> table(is.na(bmassOutput$MergedDataSources$MAF))

FALSE  TRUE 
11445   128 
.
.
.
> dim(bmassOutput$MergedDataSources)
[1] 11573    28
> table(is.na(bmassOutput$MergedDataSources$MAF))

FALSE  TRUE 
11445   128 
> dim(bmassOutput$MergedDataSources[bmassOutput$MergedDataSources$MAF == 0,])
[1] 264  28
>         bmassOutput$MergedDataSources <- bmassOutput$MergedDataSources[bmassOutput$MergedDataSources$MAF > 0 & !is.na(bmassOutput$MergedDataSources$MAF),]
> dim(bmassOutput$MergedDataSources)
[1] 11309    28
> table(is.na(bmassOutput$MergedDataSources$MAF))

FALSE 
11309 
> dim(bmassOutput$MergedDataSources[bmassOutput$MergedDataSources$MAF == 0,])
[1]  0 28
.
.
.
> dim(bmassOutput$NewSNPs$SNPs)
[1] 92 31
> bmassOutput$GWASlogBFMinThreshold
[1] 3.989499
> dim(bmassOutput$MergedDataSources)
[1] 11309    29
> quantile(bmassOutput$MergedDataSources$mvstat_log10pVal)
         0%         25%         50%         75%        100% 
   4.153174    6.991042   10.456705   21.017019 5554.224810 
> quantile(bmassOutput$MergedDataSources$unistat_log10pVal)
        0%        25%        50%        75%       100% 
  1.076808   6.936291   9.519993  17.633577 317.777283 
> LDL2 <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_LDL.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.txt.gz", header=T);
> TG2 <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TG.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.txt.gz", header=T);
> TC2 <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TC.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.txt.gz", header=T);
> GWASsnps2 <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.ChrBP.txt", header=F);
> names(GWASsnps2) <- c("Chr", "BP")
>
> bmassOutput2 <- bmass(c("HDL2", "LDL2", "TG2", "TC2"), GWASsnps=GWASsnps2, NminThreshold = 50000, PrintMergedData=TRUE, bmassSeedValue=150);
Warning message:
In ProcessMergedAndAnnotatedDataSources(DataSources, bmassOutput$MergedDataSources,  :
  2016-11-15 17:57:26 -- One of your datafiles has p-values less than the threshold which R can properly convert them to log-scale, meaning their log-values return as 'Infinite'. bmass automatically replaces these 'Infinite' values with the max, non-infinite Z-scores available, but it is recommended you self-check this as well.
> dim(bmassOutput2$MergedDataSources)
[1] 2063845      23
> dim(bmassOutput2$NewSNPs$SNPs)
[1] 65 29
> dim(bmassOutput2$MarginalSNPs$SNPs)
[1] 252  29
> table(bmassOutput2$MergedDataSources$GWASannot)

      0       1       2
1945082     149  118614
> table(bmassOutput2$MarginalSNPs$SNPs$GWASannot)

 0  1  2
96 87 69
> table(MergeDataSources$
MergeDataSources$
> table(bmassOutput$MergedDataSources$GWASannot)

   0    1    2
1744  148 9417
> table(bmassOutput$MarginalSNPs$SNPs$GWASannot)

  0   1   2
106  79  75
> bmassOutput3 <- bmass(c("HDL2", "LDL2", "TG2", "TC2"), GWASsnps=GWASsnps2, NminThreshold = 50000, PrintMergedData=TRUE, PruneMarginalSNPs=FALSE, bmassSeedValue=150);


~~~

#20161227 NOTE -- This got kind of paused midway, so picking it up and restarting a bit here

zcat /data/external_public/GlobalLipid/jointGwasMc_HDL.txt.gz | sed 's/:/ /g' | sed 's/chr//g' | awk '{ print $1 "\t" $2 "\t" $12 "\t" $6 "\t" $7 "\t" $8 "\t" $11 "\t" $10 }' | grep -v hg | grep -v rs | perl -lane 'if ($F[5] > 0) { $F[5] = "+"; } elsif ($F[5] < 0) { $F[5] = "-"; } else { $F[5] = 0;} print join("\t", @F); ' | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_HDL.bmass.formatted.txt.gz
zcat /data/external_public/GlobalLipid/jointGwasMc_LDL.txt.gz | sed 's/:/ /g' | sed 's/chr//g' | awk '{ print $1 "\t" $2 "\t" $12 "\t" $6 "\t" $7 "\t" $8 "\t" $11 "\t" $10 }' | grep -v hg | grep -v rs | perl -lane 'if ($F[5] > 0) { $F[5] = "+"; } elsif ($F[5] < 0) { $F[5] = "-"; } else { $F[5] = 0;} print join("\t", @F); ' | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_LDL.bmass.formatted.txt.gz
zcat /data/external_public/GlobalLipid/jointGwasMc_TG.txt.gz | sed 's/:/ /g' | sed 's/chr//g' | awk '{ print $1 "\t" $2 "\t" $12 "\t" $6 "\t" $7 "\t" $8 "\t" $11 "\t" $10 }' | grep -v hg | grep -v rs | perl -lane 'if ($F[5] > 0) { $F[5] = "+"; } elsif ($F[5] < 0) { $F[5] = "-"; } else { $F[5] = 0;} print join("\t", @F); ' | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TG.bmass.formatted.txt.gz
zcat /data/external_public/GlobalLipid/jointGwasMc_TC.txt.gz | sed 's/:/ /g' | sed 's/chr//g' | awk '{ print $1 "\t" $2 "\t" $12 "\t" $6 "\t" $7 "\t" $8 "\t" $11 "\t" $10 }' | grep -v hg | grep -v rs | perl -lane 'if ($F[5] > 0) { $F[5] = "+"; } elsif ($F[5] < 0) { $F[5] = "-"; } else { $F[5] = 0;} print join("\t", @F); ' | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TC.bmass.formatted.txt.gz

#/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_HDL.bmass.formatted.txt.gz
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/jointGwasMc_LDL.MatchedToHDL.txt.gz | sed 's/:/ /g' | sed 's/chr//g' | awk '{ print $1 "\t" $2 "\t" $12 "\t" $6 "\t" $7 "\t" $8 "\t" $11 "\t" $10 }' | grep -v hg | grep -v rs | perl -lane 'if ($F[6] > 0) { $F[5] = "+"; } elsif ($F[6] < 0) { $F[5] = "-"; } else { $F[5] = 0;} $F[6] = abs($F[6]); print join("\t", @F); ' | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_LDL.bmass.formatted.MatchedToHDL.txt.gz
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/jointGwasMc_TG.MatchedToHDL.txt.gz | sed 's/:/ /g' | sed 's/chr//g' | awk '{ print $1 "\t" $2 "\t" $12 "\t" $6 "\t" $7 "\t" $8 "\t" $11 "\t" $10 }' | grep -v hg | grep -v rs | perl -lane 'if ($F[6] > 0) { $F[5] = "+"; } elsif ($F[6] < 0) { $F[5] = "-"; } else { $F[5] = 0;} $F[6] = abs($F[6]); print join("\t", @F); ' | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TG.bmass.formatted.MatchedToHDL.txt.gz
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/jointGwasMc_TC.MatchedToHDL.txt.gz | sed 's/:/ /g' | sed 's/chr//g' | awk '{ print $1 "\t" $2 "\t" $12 "\t" $6 "\t" $7 "\t" $8 "\t" $11 "\t" $10 }' | grep -v hg | grep -v rs | perl -lane 'if ($F[6] > 0) { $F[5] = "+"; } elsif ($F[6] < 0) { $F[5] = "-"; } else { $F[5] = 0;} $F[6] = abs($F[6]); print join("\t", @F); ' | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TC.bmass.formatted.MatchedToHDL.txt.gz

#NOTE -- So I realized that /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GlobalLipids2013.dtlesssignif.vs1.SignCrrct.vs1.annot.RAF.txt.gz  was given the header seen below which incorrectly orders the phenotype ZScores. This order seems to be what was used in the very early, first runs of the analyses (eg see /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/process.MTedits.For2013.vs1.R). So somehow got it mixed up by copying/returning to that order from earlier with the newer file, though the newer file had a different order of HDL LDL TC TG. 
This appears to account for the previously 'most recent run' having a 'larger' number of new hits. See below:
#20170119 NOTE -- additionally, using the 'correct' ZScore matrix helps lower the total SNPs to the amount we're seeing now when running bmass() on the properly formatted globallipids 2013 data, among a few other tweaks
#~~~
#python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/DataOrganization.PythonVLookUp.PutTogetherProcessAndAnnotFiles.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GlobalLipids2013.dtlesssignif.vs1.SignCrrct.vs1.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/jointGwasMc_AllPheno.Annot.vs1.SignCrrct.RAF.txt.gz | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GlobalLipids2013.dtlesssignif.vs1.SignCrrct.vs1.annot.RAF.txt.gz
#
##Added the following header /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GlobalLipids2013.dtlesssignif.vs1.SignCrrct.vs1.annot.RAF.txt.gz
##posHg18 chr pos snp a1 a2 raf beta_LDL se_LDL n_LDL beta_HDL se_HDL n_HDL beta_TG se_TG n_TG beta_TC se_TC n_TC annot gene Z.tg Z.tc Z.ldl Z.hdl mvstat mvp unip
#~~~

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/20161115_edits_GlobalLipids2013.dtlesssignif.vs1.SignCrrct.vs1.annot.MAF.txt.gz /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/20161227_edits_GlobalLipids2013.dtlesssignif.vs1.SignCrrct.vs1.annot.MAF.txt.gz

#Put below header into /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/20161227_edits_GlobalLipids2013.dtlesssignif.vs1.SignCrrct.vs1.annot.MAF.txt.gz
#SNP Chr BP SNP A1 A2 MAF LDL_Beta LDL_SE LDL_N HDL_Beta HDL_SE HDL_N TG_Beta TG_SE TG_N TC_Beta TC_SE TC_N GWASannot Gene HDL_ZScore LDL_ZScore TC_ZScore TG_ZScore mvstat mvstat_log10pVal unistat_log10pVal

zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/20161227_edits_GlobalLipids2013.dtlesssignif.vs1.SignCrrct.vs1.annot.MAF.txt.gz | sed 's/:/ /g' | sed 's/chr//g' | awk '{ print $5 "\t" $1 "\t" $2 "\t" $3 "\t" $4 "\t" $21 }' | python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.MarkerChrBP.txt --file2 - | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/20161227_edits_GlobalLipids2013.dtlesssignif.vs1.SignCrrct.vs1.annot.MAF.Hg18vsHg19GWASProxTest.gz

join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/20161228.bmassOutput3.4.NotPruned.MaringalSNPs.txt | awk '{ print $1 "\t" $10 }' | sort -k 1,1) <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/20161227_edits_GlobalLipids2013.dtlesssignif.vs1.SignCrrct.vs1.annot.MAF.txt.gz | sed 's/chr//g' | sed 's/:/_/g ' | sort -k 1,1) | perl -lane 'if ($F[1] != $F[22]) { $F[22] = -1 * $F[22]; $F[23] = -1 * $F[23]; $F[24] = -1 * $F[24]; $F[25] = -1 * $F[25]; } splice(@F, 1, 1); print join("\t", @F);' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/20161227_edits_GlobalLipids2013.dtlesssignif.vs1.SignCrrct.vs1.annot.MAF.bmassOutput3.4_ZScoreSignMatch.txt.gz

#Put below header into /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/20161227_edits_GlobalLipids2013.dtlesssignif.vs1.SignCrrct.vs1.annot.MAF.bmassOutput3.4_ZScoreSignMatch.txt.gz
#SNP Chr BP SNP A1 A2 MAF LDL_Beta LDL_SE LDL_N HDL_Beta HDL_SE HDL_N TG_Beta TG_SE TG_N TC_Beta TC_SE TC_N GWASannot Gene HDL_ZScore LDL_ZScore TC_ZScore TG_ZScore mvstat mvstat_log10pVal unistat_log10pVal

#20170125
#Made change to force MAF from start now
#20170202 NOTE -- this lead to results of 87 new SNPs (as well as doing MAF check/flip direclty in R code as a check, see below)
#20170203 NOTE -- this is using the `...HDL.IncAllele.bmass.formatted...` type files that had the `== "-"` error, hence why back to '87' SNPs using these results but back to '64' when using the updated `MinorAllele` files created below.

zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_HDL.IncAllele.bmass.formatted.txt.gz | perl -lane 'if ($F[2] > .5) { $F[2] = 1 - $F[2]; } print join("\t", @F);' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_HDL.IncAllele.bmass.formatted.forceMAF.txt.gz 
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_LDL.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.txt.gz | perl -lane 'if ($F[2] > .5) { $F[2] = 1 - $F[2]; } print join("\t", @F);' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_LDL.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.forceMAF.txt.gz
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TG.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.txt.gz | perl -lane 'if ($F[2] > .5) { $F[2] = 1 - $F[2]; } print join("\t", @F);' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TG.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.forceMAF.txt.gz
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TC.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.txt.gz | perl -lane 'if ($F[2] > .5) { $F[2] = 1 - $F[2]; } print join("\t", @F);' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TC.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.forceMAF.txt.gz

#20170126
#Redoing full setup with orienting towards MAF allele from get go now. Also including checks between reference allele and database (HapMap/dbSNP/etc) allele (including strandedness where possible) when needed. 
#Note this immediate code below is what's being used and is viewed as correct as of 20170306, which produces 65 SNPs

#NoDup, noNA, noComma (mainly affects dbSNP), Check Alleles (drop if no match, includes strand check), Flip to Minor, MatchToPheno1, Drop 0Dir, force flip to minor AF, drop if fixed

join <(zcat /data/external_public/GlobalLipid/jointGwasMc_HDL.txt.gz | sed 's/:/_/g' | sed 's/chr//g' | awk '{ print $1 }' | grep -v hg18 | sort | uniq -u) <(zcat /data/external_public/GlobalLipid/jointGwasMc_HDL.txt.gz | sed 's/:/ /g' | sed 's/chr//g' | awk '{ print $1 "\t" $2 "\t" $12 "\t" $6 "\t" $7 "\t" $8 "\t" $11 "\t" $10 }' | grep -v hg | grep -v rs | grep -v NA | perl -lane '$F[3] = uc($F[3]); $F[4] = uc($F[4]); if ($F[5] > 0) { $F[5] = "+"; } elsif ($F[5] < 0) { $F[5] = "-"; } else { $F[5] = 0;} if ($F[2] > .5) { ($F[3], $F[4]) = ($F[4], $F[3]); if ($F[5] eq "+") { $F[5] = "-"; } elsif ($F[5] eq "-") { $F[5] = "+"; } else { $PH = 1; } $F[2] = 1 - $F[2]; } print $F[0], "_", $F[1], "\t", join("\t", @F);' | sort -k 1,1) | perl -lane 'splice(@F, 0, 1); print join("\t", @F);' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_HDL.MinorAllele.bmass.formatted.noDups.noNA.txt.gz 
join <(zcat /data/external_public/GlobalLipid/jointGwasMc_LDL.txt.gz | sed 's/:/_/g' | sed 's/chr//g' | awk '{ print $1 }' | grep -v hg18 | sort | uniq -u) <(zcat /data/external_public/GlobalLipid/jointGwasMc_LDL.txt.gz | sed 's/:/ /g' | sed 's/chr//g' | awk '{ print $1 "\t" $2 "\t" $12 "\t" $6 "\t" $7 "\t" $8 "\t" $11 "\t" $10 }' | grep -v hg | grep -v rs | grep -v NA | perl -lane '$F[3] = uc($F[3]); $F[4] = uc($F[4]); if ($F[5] > 0) { $F[5] = "+"; } elsif ($F[5] < 0) { $F[5] = "-"; } else { $F[5] = 0;} if ($F[2] > .5) { ($F[3], $F[4]) = ($F[4], $F[3]); if ($F[5] eq "+") { $F[5] = "-"; } elsif ($F[5] eq "-") { $F[5] = "+"; } else { $PH = 1; } $F[2] = 1 - $F[2]; } print $F[0], "_", $F[1], "\t", join("\t", @F);' | sort -k 1,1) | perl -lane 'splice(@F, 0, 1); print join("\t", @F);' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_LDL.MinorAllele.bmass.formatted.noDups.noNA.txt.gz 
join <(zcat /data/external_public/GlobalLipid/jointGwasMc_TG.txt.gz | sed 's/:/_/g' | sed 's/chr//g' | awk '{ print $1 }' | grep -v hg18 | sort | uniq -u) <(zcat /data/external_public/GlobalLipid/jointGwasMc_TG.txt.gz | sed 's/:/ /g' | sed 's/chr//g' | awk '{ print $1 "\t" $2 "\t" $12 "\t" $6 "\t" $7 "\t" $8 "\t" $11 "\t" $10 }' | grep -v hg | grep -v rs | grep -v NA | perl -lane '$F[3] = uc($F[3]); $F[4] = uc($F[4]); if ($F[5] > 0) { $F[5] = "+"; } elsif ($F[5] < 0) { $F[5] = "-"; } else { $F[5] = 0;} if ($F[2] > .5) { ($F[3], $F[4]) = ($F[4], $F[3]); if ($F[5] eq "+") { $F[5] = "-"; } elsif ($F[5] eq "-") { $F[5] = "+"; } else { $PH = 1; } $F[2] = 1 - $F[2]; } print $F[0], "_", $F[1], "\t", join("\t", @F);' | sort -k 1,1) | perl -lane 'splice(@F, 0, 1); print join("\t", @F);' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TG.MinorAllele.bmass.formatted.noDups.noNA.txt.gz 
join <(zcat /data/external_public/GlobalLipid/jointGwasMc_TC.txt.gz | sed 's/:/_/g' | sed 's/chr//g' | awk '{ print $1 }' | grep -v hg18 | sort | uniq -u) <(zcat /data/external_public/GlobalLipid/jointGwasMc_TC.txt.gz | sed 's/:/ /g' | sed 's/chr//g' | awk '{ print $1 "\t" $2 "\t" $12 "\t" $6 "\t" $7 "\t" $8 "\t" $11 "\t" $10 }' | grep -v hg | grep -v rs | grep -v NA | perl -lane '$F[3] = uc($F[3]); $F[4] = uc($F[4]); if ($F[5] > 0) { $F[5] = "+"; } elsif ($F[5] < 0) { $F[5] = "-"; } else { $F[5] = 0;} if ($F[2] > .5) { ($F[3], $F[4]) = ($F[4], $F[3]); if ($F[5] eq "+") { $F[5] = "-"; } elsif ($F[5] eq "-") { $F[5] = "+"; } else { $PH = 1; } $F[2] = 1 - $F[2]; } print $F[0], "_", $F[1], "\t", join("\t", @F);' | sort -k 1,1) | perl -lane 'splice(@F, 0, 1); print join("\t", @F);' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TC.MinorAllele.bmass.formatted.noDups.noNA.txt.gz 

join <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_HDL.MinorAllele.bmass.formatted.noDups.noNA.txt.gz | awk '{ print $1 "_" $2 "\t" $4 }' | sort -g -k 1,1) <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_LDL.MinorAllele.bmass.formatted.noDups.noNA.txt.gz | awk '{ print $1 "_" $2 "\t" $0 }' | sort -g -k 1,1) | perl -lane 'if ($F[1] eq $F[6]) { ($F[5], $F[6]) = ($F[6], $F[5]); $F[4] = 1 - $F[4]; if ($F[7] eq "-") { $F[7] = "+"; } elsif ($F[7] eq "+") { $F[7] = "-"; } else { $PH = 1; } } elsif ($F[1] eq $F[5]) { $PH = 1; } else { push(@F, "NoAlleleMatch"); } print join("\t", @F[2..$#F]);' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_LDL.MinorAllele.bmass.formatted.noDups.noNA.MatchedToHDL.txt.gz
join <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_HDL.MinorAllele.bmass.formatted.noDups.noNA.txt.gz | awk '{ print $1 "_" $2 "\t" $4 }' | sort -g -k 1,1) <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TG.MinorAllele.bmass.formatted.noDups.noNA.txt.gz | awk '{ print $1 "_" $2 "\t" $0 }' | sort -g -k 1,1) | perl -lane 'if ($F[1] eq $F[6]) { ($F[5], $F[6]) = ($F[6], $F[5]); $F[4] = 1 - $F[4]; if ($F[7] eq "-") { $F[7] = "+"; } elsif ($F[7] eq "+") { $F[7] = "-"; } else { $PH = 1; } } elsif ($F[1] eq $F[5]) { $PH = 1; } else { push(@F, "NoAlleleMatch"); } print join("\t", @F[2..$#F]);' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TG.MinorAllele.bmass.formatted.noDups.noNA.MatchedToHDL.txt.gz
join <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_HDL.MinorAllele.bmass.formatted.noDups.noNA.txt.gz | awk '{ print $1 "_" $2 "\t" $4 }' | sort -g -k 1,1) <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TC.MinorAllele.bmass.formatted.noDups.noNA.txt.gz | awk '{ print $1 "_" $2 "\t" $0 }' | sort -g -k 1,1) | perl -lane 'if ($F[1] eq $F[6]) { ($F[5], $F[6]) = ($F[6], $F[5]); $F[4] = 1 - $F[4]; if ($F[7] eq "-") { $F[7] = "+"; } elsif ($F[7] eq "+") { $F[7] = "-"; } else { $PH = 1; } } elsif ($F[1] eq $F[5]) { $PH = 1; } else { push(@F, "NoAlleleMatch"); } print join("\t", @F[2..$#F]);' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TC.MinorAllele.bmass.formatted.noDups.noNA.MatchedToHDL.txt.gz

zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_HDL.MinorAllele.bmass.formatted.noDups.noNA.txt.gz | perl -lane 'if ($F[5] ne "0") { if ($F[2] > .5) { $F[2] = 1 - $F[2]; } if ($F[2] > 0) { print join("\t", @F); } }' | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_HDL.MinorAllele.bmass.formatted.noDups.noNA.no0Dir.noFixed.forceMAF.txt.gz 
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_LDL.MinorAllele.bmass.formatted.noDups.noNA.MatchedToHDL.txt.gz | grep -v NoAlleleMatch | perl -lane 'if ($F[5] ne "0") { if ($F[2] > .5) { $F[2] = 1 - $F[2]; } if ($F[2] > 0) { print join("\t", @F); } }' | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_LDL.MinorAllele.bmass.formatted.noDups.noNA.MatchedToHDL.DropNoMatch.no0Dir.noFixed.forceMAF.txt.gz
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TG.MinorAllele.bmass.formatted.noDups.noNA.MatchedToHDL.txt.gz | grep -v NoAlleleMatch | perl -lane 'if ($F[5] ne "0") { if ($F[2] > .5) { $F[2] = 1 - $F[2]; } if ($F[2] > 0) { print join("\t", @F); } }' | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TG.MinorAllele.bmass.formatted.noDups.noNA.MatchedToHDL.DropNoMatch.no0Dir.noFixed.forceMAF.txt.gz
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TC.MinorAllele.bmass.formatted.noDups.noNA.MatchedToHDL.txt.gz | grep -v NoAlleleMatch | perl -lane 'if ($F[5] ne "0") { if ($F[2] > .5) { $F[2] = 1 - $F[2]; } if ($F[2] > 0) { print join("\t", @F); } }' | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TC.MinorAllele.bmass.formatted.noDups.noNA.MatchedToHDL.DropNoMatch.no0Dir.noFixed.forceMAF.txt.gz

#Below produced list of SNPs with 65 hits so not totally appropriate
##zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_HDL.MinorAllele.bmass.formatted.noDups.noNA.txt.gz | perl -lane 'if ($F[5] eq "0") { $F[5] = "+"; } if ($F[2] > .5) { $F[2] = 1 - $F[2]; } if ($F[2] > 0) { print join("\t", @F); }' | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/CHECK1.jointGwasMc_HDL.MinorAllele.bmass.formatted.noDups.noNA.Flip0Dir.noFixed.forceMAF.txt.gz 
##zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_LDL.MinorAllele.bmass.formatted.noDups.noNA.MatchedToHDL.txt.gz | grep -v NoAlleleMatch | perl -lane 'if ($F[5] eq "0") { $F[5] = "+"; } if ($F[2] > .5) { $F[2] = 1 - $F[2]; } if ($F[2] > 0) { print join("\t", @F); }' | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/CHECK1.jointGwasMc_LDL.MinorAllele.bmass.formatted.noDups.noNA.MatchedToHDL.DropNoMatch.Flip0Dir.noFixed.forceMAF.txt.gz
##zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TG.MinorAllele.bmass.formatted.noDups.noNA.MatchedToHDL.txt.gz | grep -v NoAlleleMatch | perl -lane 'if ($F[5] eq "0") { $F[5] = "+"; } if ($F[2] > .5) { $F[2] = 1 - $F[2]; } if ($F[2] > 0) { print join("\t", @F); }' | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/CHECK1.jointGwasMc_TG.MinorAllele.bmass.formatted.noDups.noNA.MatchedToHDL.DropNoMatch.Flip0Dir.noFixed.forceMAF.txt.gz
##zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TC.MinorAllele.bmass.formatted.noDups.noNA.MatchedToHDL.txt.gz | grep -v NoAlleleMatch | perl -lane 'if ($F[5] eq "0") { $F[5] = "+"; } if ($F[2] > .5) { $F[2] = 1 - $F[2]; } if ($F[2] > 0) { print join("\t", @F); }' | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/CHECK1.jointGwasMc_TC.MinorAllele.bmass.formatted.noDups.noNA.MatchedToHDL.DropNoMatch.Flip0Dir.noFixed.forceMAF.txt.gz

#Below is testing whether if starting with IncAllele but using the same exact new code as MinorAllele otherwise, get the 87 SNP results. This includes the error of using  `=="-"` which is what was found to be causing the erroneous direction flip of 0 to +. Prefacing these lines with `~` to somehow indicate 'something is odd about these lines here' -- eg this set of code is purposefully testing an incorrect approach, but was in fact used 'in production'. This is in contrast to the meaning I try to reserve for prefacing lines with `#`, which usually means the lines were not actually kept or used 'in production'.
~join <(zcat /data/external_public/GlobalLipid/jointGwasMc_HDL.txt.gz | sed 's/:/_/g' | sed 's/chr//g' | awk '{ print $1 }' | grep -v hg18 | sort | uniq -u) <(zcat /data/external_public/GlobalLipid/jointGwasMc_HDL.txt.gz | sed 's/:/ /g' | sed 's/chr//g' | awk '{ print $1 "\t" $2 "\t" $12 "\t" $6 "\t" $7 "\t" $8 "\t" $11 "\t" $10 }' | grep -v hg | grep -v rs | grep -v NA | perl -lane '$F[3] = uc($F[3]); $F[4] = uc($F[4]); if ($F[5] > 0) { $F[5] = "+"; } elsif ($F[5] < 0) { $F[5] = "-"; } else { $F[5] = 0;} if ($F[5] == "-") { ($F[3], $F[4]) = ($F[4], $F[3]); $F[5] = "+"; $F[2] = 1 - $F[2]; } print $F[0], "_", $F[1], "\t", join("\t", @F);' | sort -k 1,1) | perl -lane 'splice(@F, 0, 1); print join("\t", @F);' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/CHECK2.jointGwasMc_HDL.IncAllele.bmass.formatted.noDups.noNA.txt.gz 
~join <(zcat /data/external_public/GlobalLipid/jointGwasMc_LDL.txt.gz | sed 's/:/_/g' | sed 's/chr//g' | awk '{ print $1 }' | grep -v hg18 | sort | uniq -u) <(zcat /data/external_public/GlobalLipid/jointGwasMc_LDL.txt.gz | sed 's/:/ /g' | sed 's/chr//g' | awk '{ print $1 "\t" $2 "\t" $12 "\t" $6 "\t" $7 "\t" $8 "\t" $11 "\t" $10 }' | grep -v hg | grep -v rs | grep -v NA | perl -lane '$F[3] = uc($F[3]); $F[4] = uc($F[4]); if ($F[5] > 0) { $F[5] = "+"; } elsif ($F[5] < 0) { $F[5] = "-"; } else { $F[5] = 0;} if ($F[5] == "-") { ($F[3], $F[4]) = ($F[4], $F[3]); $F[5] = "+"; $F[2] = 1 - $F[2]; } print $F[0], "_", $F[1], "\t", join("\t", @F);' | sort -k 1,1) | perl -lane 'splice(@F, 0, 1); print join("\t", @F);' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/CHECK2.jointGwasMc_LDL.IncAllele.bmass.formatted.noDups.noNA.txt.gz 
~join <(zcat /data/external_public/GlobalLipid/jointGwasMc_TG.txt.gz | sed 's/:/_/g' | sed 's/chr//g' | awk '{ print $1 }' | grep -v hg18 | sort | uniq -u) <(zcat /data/external_public/GlobalLipid/jointGwasMc_TG.txt.gz | sed 's/:/ /g' | sed 's/chr//g' | awk '{ print $1 "\t" $2 "\t" $12 "\t" $6 "\t" $7 "\t" $8 "\t" $11 "\t" $10 }' | grep -v hg | grep -v rs | grep -v NA | perl -lane '$F[3] = uc($F[3]); $F[4] = uc($F[4]); if ($F[5] > 0) { $F[5] = "+"; } elsif ($F[5] < 0) { $F[5] = "-"; } else { $F[5] = 0;} if ($F[5] == "-") { ($F[3], $F[4]) = ($F[4], $F[3]); $F[5] = "+"; $F[2] = 1 - $F[2]; } print $F[0], "_", $F[1], "\t", join("\t", @F);' | sort -k 1,1) | perl -lane 'splice(@F, 0, 1); print join("\t", @F);' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/CHECK2.jointGwasMc_TG.IncAllele.bmass.formatted.noDups.noNA.txt.gz 
~join <(zcat /data/external_public/GlobalLipid/jointGwasMc_TC.txt.gz | sed 's/:/_/g' | sed 's/chr//g' | awk '{ print $1 }' | grep -v hg18 | sort | uniq -u) <(zcat /data/external_public/GlobalLipid/jointGwasMc_TC.txt.gz | sed 's/:/ /g' | sed 's/chr//g' | awk '{ print $1 "\t" $2 "\t" $12 "\t" $6 "\t" $7 "\t" $8 "\t" $11 "\t" $10 }' | grep -v hg | grep -v rs | grep -v NA | perl -lane '$F[3] = uc($F[3]); $F[4] = uc($F[4]); if ($F[5] > 0) { $F[5] = "+"; } elsif ($F[5] < 0) { $F[5] = "-"; } else { $F[5] = 0;} if ($F[5] == "-") { ($F[3], $F[4]) = ($F[4], $F[3]); $F[5] = "-"; $F[2] = 1 - $F[2]; } print $F[0], "_", $F[1], "\t", join("\t", @F);' | sort -k 1,1) | perl -lane 'splice(@F, 0, 1); print join("\t", @F);' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/CHECK2.jointGwasMc_TC.IncAllele.bmass.formatted.noDups.noNA.txt.gz 
~
~join <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/CHECK2.jointGwasMc_HDL.IncAllele.bmass.formatted.noDups.noNA.txt.gz | awk '{ print $1 "_" $2 "\t" $4 }' | sort -g -k 1,1) <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/CHECK2.jointGwasMc_LDL.IncAllele.bmass.formatted.noDups.noNA.txt.gz | awk '{ print $1 "_" $2 "\t" $0 }' | sort -g -k 1,1) | perl -lane 'if ($F[1] eq $F[6]) { ($F[5], $F[6]) = ($F[6], $F[5]); $F[4] = 1 - $F[4]; if ($F[7] eq "-") { $F[7] = "+"; } elsif ($F[7] eq "+") { $F[7] = "-"; } else { $PH = 1; } } elsif ($F[1] eq $F[5]) { $PH = 1; } else { push(@F, "NoAlleleMatch"); } print join("\t", @F[2..$#F]);' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/CHECK2.jointGwasMc_LDL.IncAllele.bmass.formatted.noDups.noNA.MatchedToHDL.txt.gz
~join <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/CHECK2.jointGwasMc_HDL.IncAllele.bmass.formatted.noDups.noNA.txt.gz | awk '{ print $1 "_" $2 "\t" $4 }' | sort -g -k 1,1) <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/CHECK2.jointGwasMc_TG.IncAllele.bmass.formatted.noDups.noNA.txt.gz | awk '{ print $1 "_" $2 "\t" $0 }' | sort -g -k 1,1) | perl -lane 'if ($F[1] eq $F[6]) { ($F[5], $F[6]) = ($F[6], $F[5]); $F[4] = 1 - $F[4]; if ($F[7] eq "-") { $F[7] = "+"; } elsif ($F[7] eq "+") { $F[7] = "-"; } else { $PH = 1; } } elsif ($F[1] eq $F[5]) { $PH = 1; } else { push(@F, "NoAlleleMatch"); } print join("\t", @F[2..$#F]);' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/CHECK2.jointGwasMc_TG.IncAllele.bmass.formatted.noDups.noNA.MatchedToHDL.txt.gz
~join <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/CHECK2.jointGwasMc_HDL.IncAllele.bmass.formatted.noDups.noNA.txt.gz | awk '{ print $1 "_" $2 "\t" $4 }' | sort -g -k 1,1) <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/CHECK2.jointGwasMc_TC.IncAllele.bmass.formatted.noDups.noNA.txt.gz | awk '{ print $1 "_" $2 "\t" $0 }' | sort -g -k 1,1) | perl -lane 'if ($F[1] eq $F[6]) { ($F[5], $F[6]) = ($F[6], $F[5]); $F[4] = 1 - $F[4]; if ($F[7] eq "-") { $F[7] = "+"; } elsif ($F[7] eq "+") { $F[7] = "-"; } else { $PH = 1; } } elsif ($F[1] eq $F[5]) { $PH = 1; } else { push(@F, "NoAlleleMatch"); } print join("\t", @F[2..$#F]);' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/CHECK2.jointGwasMc_TC.IncAllele.bmass.formatted.noDups.noNA.MatchedToHDL.txt.gz
~
~zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/CHECK2.jointGwasMc_HDL.IncAllele.bmass.formatted.noDups.noNA.txt.gz | perl -lane 'if ($F[5] ne "0") { if ($F[2] > .5) { $F[2] = 1 - $F[2]; } if ($F[2] > 0) { print join("\t", @F); } }' | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/CHECK2.jointGwasMc_HDL.IncAllele.bmass.formatted.noDups.noNA.no0Dir.noFixed.forceMAF.txt.gz 
~zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/CHECK2.jointGwasMc_LDL.IncAllele.bmass.formatted.noDups.noNA.MatchedToHDL.txt.gz | grep -v NoAlleleMatch | perl -lane 'if ($F[5] ne "0") { if ($F[2] > .5) { $F[2] = 1 - $F[2]; } if ($F[2] > 0) { print join("\t", @F); } }' | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/CHECK2.jointGwasMc_LDL.IncAllele.bmass.formatted.noDups.noNA.MatchedToHDL.DropNoMatch.no0Dir.noFixed.forceMAF.txt.gz
~zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/CHECK2.jointGwasMc_TG.IncAllele.bmass.formatted.noDups.noNA.MatchedToHDL.txt.gz | grep -v NoAlleleMatch | perl -lane 'if ($F[5] ne "0") { if ($F[2] > .5) { $F[2] = 1 - $F[2]; } if ($F[2] > 0) { print join("\t", @F); } }' | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/CHECK2.jointGwasMc_TG.IncAllele.bmass.formatted.noDups.noNA.MatchedToHDL.DropNoMatch.no0Dir.noFixed.forceMAF.txt.gz
~zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/CHECK2.jointGwasMc_TC.IncAllele.bmass.formatted.noDups.noNA.MatchedToHDL.txt.gz | grep -v NoAlleleMatch | perl -lane 'if ($F[5] ne "0") { if ($F[2] > .5) { $F[2] = 1 - $F[2]; } if ($F[2] > 0) { print join("\t", @F); } }' | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/CHECK2.jointGwasMc_TC.IncAllele.bmass.formatted.noDups.noNA.MatchedToHDL.DropNoMatch.no0Dir.noFixed.forceMAF.txt.gz


#20170125
#Note change to using forceMAF files
#20170203
#Note change to MinorAllele files & `noDups.noNA.MatchedToHDL.DropNoMatch.forceMAF`

```
library("devtools"); devtools::load_all("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/bmass"); 
#HDL <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_HDL.IncAllele.bmass.formatted.txt.gz", header=T);
#LDL <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_LDL.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.txt.gz", header=T);
#TG <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TG.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.txt.gz", header=T);
#TC <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TC.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.txt.gz", header=T);
#HDL <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_HDL.IncAllele.bmass.formatted.forceMAF.txt.gz", header=T);
#LDL <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_LDL.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.forceMAF.txt.gz", header=T);
#TG <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TG.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.forceMAF.txt.gz", header=T);
#TC <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TC.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.forceMAF.txt.gz", header=T);
HDL <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_HDL.MinorAllele.bmass.formatted.noDups.noNA.no0Dir.noFixed.forceMAF.txt.gz", header=T);
LDL <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_LDL.MinorAllele.bmass.formatted.noDups.noNA.MatchedToHDL.DropNoMatch.no0Dir.noFixed.forceMAF.txt.gz", header=T);
TG <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TG.MinorAllele.bmass.formatted.noDups.noNA.MatchedToHDL.DropNoMatch.no0Dir.noFixed.forceMAF.txt.gz", header=T);
TC <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TC.MinorAllele.bmass.formatted.noDups.noNA.MatchedToHDL.DropNoMatch.no0Dir.noFixed.forceMAF.txt.gz", header=T);
#HDL <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/CHECK1.jointGwasMc_HDL.MinorAllele.bmass.formatted.noDups.noNA.Flip0Dir.noFixed.forceMAF.txt.gz", header=T);
#LDL <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/CHECK1.jointGwasMc_LDL.MinorAllele.bmass.formatted.noDups.noNA.MatchedToHDL.DropNoMatch.Flip0Dir.noFixed.forceMAF.txt.gz", header=T);
#TG <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/CHECK1.jointGwasMc_TG.MinorAllele.bmass.formatted.noDups.noNA.MatchedToHDL.DropNoMatch.Flip0Dir.noFixed.forceMAF.txt.gz", header=T);
#TC <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/CHECK1.jointGwasMc_TC.MinorAllele.bmass.formatted.noDups.noNA.MatchedToHDL.DropNoMatch.Flip0Dir.noFixed.forceMAF.txt.gz", header=T);
#HDL <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/CHECK2.jointGwasMc_HDL.IncAllele.bmass.formatted.noDups.noNA.no0Dir.noFixed.forceMAF.txt.gz", header=T);
#LDL <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/CHECK2.jointGwasMc_LDL.IncAllele.bmass.formatted.noDups.noNA.MatchedToHDL.DropNoMatch.no0Dir.noFixed.forceMAF.txt.gz", header=T);
#TG <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/CHECK2.jointGwasMc_TG.IncAllele.bmass.formatted.noDups.noNA.MatchedToHDL.DropNoMatch.no0Dir.noFixed.forceMAF.txt.gz", header=T);
#TC <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/CHECK2.jointGwasMc_TC.IncAllele.bmass.formatted.noDups.noNA.MatchedToHDL.DropNoMatch.no0Dir.noFixed.forceMAF.txt.gz", header=T);
GWASsnps <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.ChrBP.txt", header=F);
names(GWASsnps) <- c("Chr", "BP")
#bmassOutput1 <- bmass(c("HDL", "LDL", "TG", "TC"), GWASsnps=GWASsnps, NminThreshold = 1, bmassSeedValue=150);
#print(bmassOutput1);" > shana1.txt
DataSources = c("HDL", "LDL", "TG", "TC"); NminThreshold = 50000; bmassSeedValue=150;
ExpectedColumnNames=c("Chr", "BP", "MAF", "Direction", "pValue", "N"); GWASsnps_AnnotateWindow = 5e5; SNPMarginalUnivariateThreshold = 1e-6; SNPMarginalMultivariateThreshold = 1e-6; SigmaAlphas = c(0.005,0.0075,0.01,0.015,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.15); ProvidedPriors=NULL; UseFlatPriors=FALSE; PruneMarginalSNPs=TRUE; PruneMarginalSNPs_bpWindow=5e5;

#Commented below out for 20170203
#HDL <- HDL[HDL$Direction != 0,]
#LDL <- LDL[LDL$Direction != 0,]
#TG <- TG[TG$Direction != 0,]
#TC <- TC[TC$Direction != 0,]
#HDL <- HDL[!is.na(HDL$MAF),]
#LDL <- LDL[!is.na(LDL$MAF),]
#TG <- TG[!is.na(TG$MAF),]
#TC <- TC[!is.na(TC$MAF),]
#Inserted this part 20170125
#HDL <- HDL[HDL$MAF != 0,]
#LDL <- LDL[LDL$MAF != 0,]
#TG <- TG[TG$MAF != 0,]
#TC <- TC[TC$MAF != 0,]

#For all
#ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.txt

HDL2 <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_HDL.bmass.formatted.txt.gz", header=T);
LDL2 <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_LDL.bmass.formatted.txt.gz", header=T);
TG2 <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TG.bmass.formatted.txt.gz", header=T);
TC2 <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TC.bmass.formatted.txt.gz", header=T);
GWASsnps2 <- GWASsnps

HDL2 <- HDL2[HDL2$Direction != 0,]
LDL2 <- LDL2[LDL2$Direction != 0,]
TG2 <- TG2[TG2$Direction != 0,]
TC2 <- TC2[TC2$Direction != 0,]
HDL2 <- HDL2[!is.na(HDL2$MAF),]
LDL2 <- LDL2[!is.na(LDL2$MAF),]
TG2 <- TG2[!is.na(TG2$MAF),]
TC2 <- TC2[!is.na(TC2$MAF),]

HDL3 <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_HDL.bmass.formatted.txt.gz", header=T);
LDL3 <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_LDL.bmass.formatted.MatchedToHDL.txt.gz", header=T);
TG3 <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TG.bmass.formatted.MatchedToHDL.txt.gz", header=T);
TC3 <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TC.bmass.formatted.MatchedToHDL.txt.gz", header=T);
GWASsnps3 <- GWASsnps

HDL3 <- HDL3[HDL3$Direction != 0,]
LDL3 <- LDL3[LDL3$Direction != 0,]
TG3 <- TG3[TG3$Direction != 0,]
TC3 <- TC3[TC3$Direction != 0,]
HDL3 <- HDL3[!is.na(HDL3$MAF),]
LDL3 <- LDL3[!is.na(LDL3$MAF),]
TG3 <- TG3[!is.na(TG3$MAF),]
TC3 <- TC3[!is.na(TC3$MAF),]

bmassOutput1 <- bmass(c("HDL", "LDL", "TG", "TC"), GWASsnps=GWASsnps, NminThreshold = 50000, bmassSeedValue=150);
bmassOutput3 <- bmass(c("HDL", "LDL", "TG", "TC"), GWASsnps=GWASsnps, NminThreshold = 50000, PrintMergedData=TRUE, PruneMarginalSNPs=FALSE, bmassSeedValue=150);
#bmassOutput1.wInc.force <- bmass(c("HDL", "LDL", "TG", "TC"), GWASsnps=GWASsnps, NminThreshold = 50000, bmassSeedValue=150);
#bmassOutput3.wInc.force <- bmass(c("HDL", "LDL", "TG", "TC"), GWASsnps=GWASsnps, NminThreshold = 50000, PrintMergedData=TRUE, PruneMarginalSNPs=FALSE, bmassSeedValue=150);
#bmassOutput1.wMinor.Flip0dir <- bmass(c("HDL", "LDL", "TG", "TC"), GWASsnps=GWASsnps, NminThreshold = 50000, bmassSeedValue=150);
#bmassOutput1.wInc.CHECK2.NewSetup <- bmass(c("HDL", "LDL", "TG", "TC"), GWASsnps=GWASsnps, NminThreshold = 50000, bmassSeedValue=150);
#bmassOutput3.wInc.CHECK2.NewSetup <- bmass(c("HDL", "LDL", "TG", "TC"), GWASsnps=GWASsnps, NminThreshold = 50000, PrintMergedData=TRUE, PruneMarginalSNPs=FALSE, bmassSeedValue=150);
bmassOutput2 <- bmass(c("HDL2", "LDL2", "TG2", "TC2"), GWASsnps=GWASsnps2, NminThreshold = 50000, PrintMergedData=TRUE, bmassSeedValue=150);
bmassOutput3 <- bmass(c("HDL3", "LDL3", "TG3", "TC3"), GWASsnps=GWASsnps3, NminThreshold = 50000, PrintMergedData=TRUE, bmassSeedValue=150);
bmassOutput3.2 <- bmass(c("HDL3", "LDL3", "TG3", "TC3"), GWASsnps=GWASsnps3, NminThreshold = 50000, PrintMergedData=TRUE, bmassSeedValue=34987435);

#From 20170119
 
write.table(bmassOutput1$NewSNPs$SNPs, file="/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.txt", quote=FALSE, row.names=FALSE)
write.table(bmassOutput3$MergedDataSources, file="/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/bmass.GlobalLipids2013.Vs2.MergedDataSources.vs1.txt", quote=FALSE, row.names=FALSE) 
#write.table(bmassOutput3.wInc.force$MergedDataSources, file="/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/OLD1.WRONG1.Inc.force.bmass.GlobalLipids2013.Vs2.MergedDataSources.vs1.txt", quote=FALSE, row.names=FALSE)

system("gzip /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/bmass.GlobalLipids2013.Vs2.MergedDataSources.vs1.txt")
#system("gzip /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/OLD1.WRONG1.Inc.force.bmass.GlobalLipids2013.Vs2.MergedDataSources.vs1.txt")

        bmassOutput4 <- list()
        bmassOutput4$MergedDataSources <- NULL
        if (!is.null(MergedDataSources)) {
                bmassOutput4$MergedDataSources <- MergedDataSources
        }       
        bmassOutput4$ZScoresCorMatrix <- NULL
        bmassOutput4$MarginalSNPs <- list()
        bmassOutput4$Models <- NULL
        bmassOutput4$ModelPriors <- NULL
        bmassOutput4$PreviousSNPs <- list()
        bmassOutput4$NewSNPs <- list()
        bmassOutput4$GWASlogBFMinThreshold <- NULL
        #LogFile <- c()
        bmassOutput4$LogFile <- c()

        bmassOutput4$LogFile <- rbind(bmassOutput4$LogFile, paste(format(Sys.time()), " -- beginning bmass.", sep=""))

	bmassOutput5 <- bmassOutput4
	bmassOutput5.2 <- bmassOutput5
	#bmassOutput5.3 <- bmassOutput5
	bmassOutput5.4 <- bmassOutput5

        bmassOutput4$MergedDataSources <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/20161115_edits_GlobalLipids2013.dtlesssignif.vs1.SignCrrct.vs1.annot.MAF.txt.gz", header=TRUE)
        bmassOutput4$MergedDataSources <- bmassOutput4$MergedDataSources[bmassOutput4$MergedDataSources$MAF > 0 & !is.na(bmassOutput4$MergedDataSources$MAF),]
        bmassOutput4$ZScoresCorMatrix <- as.matrix(read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GlobalLipids2013.RSS0.vs1.SignCrrct.vs1.txt", header=TRUE, sep=","))
        GWASsnps <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.ChrBP.txt", header=F);
        names(GWASsnps) <- c("Chr", "BP")   
	#Z.LDL,Z.HDL,Z.TC,Z.TG
        DataSources = c("LDL", "HDL", "TC", "TG"); NminThreshold = 50000; bmassSeedValue=150;
        ExpectedColumnNames=c("Chr", "BP", "MAF", "Direction", "pValue", "N"); GWASsnps_AnnotateWindow = 5e5; SNPMarginalUnivariateThreshold = 1e-6; SNPMarginalMultivariateThreshold = 1e-6; SigmaAlphas = c(0.005,0.0075,0.01,0.015,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.15); ProvidedPriors=NULL; UseFlatPriors=FALSE; PruneMarginalSNPs=TRUE; PruneMarginalSNPs_bpWindow=5e5;
        bmassOutput4$MergedDataSources$ChrBP <- paste(bmassOutput4$MergedDataSources$Chr, "_", bmassOutput4$MergedDataSources$BP, sep="")

        bmassOutput4$MarginalSNPs$SNPs <- bmassOutput4$MergedDataSources[bmassOutput4$MergedDataSources$mvstat_log10pVal > -log10(SNPMarginalUnivariateThreshold) | bmassOutput4$MergedDataSources$unistat_log10pVal > -log10(SNPMarginalMultivariateThreshold),]

        bmassOutput4[c("MarginalSNPs", "Models", "ModelPriors", "LogFile")] <- GetLogBFsFromData(DataSources, bmassOutput4$MarginalSNPs, bmassOutput4$ZScoresCorMatrix, SigmaAlphas, bmassOutput4$LogFile)[c("MarginalSNPs", "Models", "ModelPriors", "LogFile")]

#       Nmins <- apply(bmassOutput4$MarginalSNPs$SNPs[,c("TC_N", "TG_N", "LDL_N", "HDL_N")], 1, min)
#       ZScores <- bmassOutput4$MarginalSNPs$SNPs[,c("TC_ZScore", "TG_ZScore", "HDL_ZScore", "LDL_ZScore")]
#       MarginalSNPs_logBFs <- compute.allBFs.fromZscores(ZScores, bmassOutput4$ZScoresCorMatrix, Nmins, bmassOutput4$MarginalSNPs$SNPs$MAF, SigmaAlphas)

        bmassOutput4[c("MarginalSNPs", "PreviousSNPs", "ModelPriors", "GWASlogBFMinThreshold", "LogFile")] <- DetermineAndApplyPriors(DataSources, bmassOutput4$MarginalSNPs, GWASsnps, SigmaAlphas, bmassOutput4$Models, bmassOutput4$ModelPriors, ProvidedPriors, UseFlatPriors, bmassSeedValue, bmassOutput4$LogFile)[c("MarginalSNPs", "PreviousSNPs", "ModelPriors", "GWASlogBFMinThreshold", "LogFile")]

        bmassOutput4[c("MarginalSNPs", "PreviousSNPs", "NewSNPs", "LogFile")] <- FinalizeAndFormatResults(DataSources, bmassOutput4$MarginalSNPs, bmassOutput4$PreviousSNPs, GWASsnps, bmassOutput4$GWASlogBFMinThreshold, SigmaAlphas, bmassOutput4$Models, bmassOutput4$ModelPriors, NminThreshold, PruneMarginalSNPs, PruneMarginalSNPs_bpWindow, bmassOutput4$LogFile)[c("MarginalSNPs", "PreviousSNPs", "NewSNPs", "LogFile")]



	#Testing using /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/20161227_edits_GlobalLipids2013.dtlesssignif.vs1.SignCrrct.vs1.annot.MAF.txt.gz which has the correct header for the ZScores
        bmassOutput5$MergedDataSources <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/20161227_edits_GlobalLipids2013.dtlesssignif.vs1.SignCrrct.vs1.annot.MAF.txt.gz", header=TRUE)
        bmassOutput5$MergedDataSources <- bmassOutput5$MergedDataSources[bmassOutput5$MergedDataSources$MAF > 0 & !is.na(bmassOutput5$MergedDataSources$MAF),]
        bmassOutput5$ZScoresCorMatrix <- as.matrix(read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GlobalLipids2013.RSS0.vs1.SignCrrct.vs1.txt", header=TRUE, sep=","))
        GWASsnps <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.ChrBP.txt", header=F);
        names(GWASsnps) <- c("Chr", "BP")   
	#Z.LDL,Z.HDL,Z.TC,Z.TG
        DataSources = c("LDL", "HDL", "TC", "TG"); NminThreshold = 50000; bmassSeedValue=150;
        ExpectedColumnNames=c("Chr", "BP", "MAF", "Direction", "pValue", "N"); GWASsnps_AnnotateWindow = 5e5; SNPMarginalUnivariateThreshold = 1e-6; SNPMarginalMultivariateThreshold = 1e-6; SigmaAlphas = c(0.005,0.0075,0.01,0.015,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.15); ProvidedPriors=NULL; UseFlatPriors=FALSE; PruneMarginalSNPs=TRUE; PruneMarginalSNPs_bpWindow=5e5;
        bmassOutput5$MergedDataSources$ChrBP <- paste(bmassOutput5$MergedDataSources$Chr, "_", bmassOutput5$MergedDataSources$BP, sep="")

        bmassOutput5$MarginalSNPs$SNPs <- bmassOutput5$MergedDataSources[bmassOutput5$MergedDataSources$mvstat_log10pVal > -log10(SNPMarginalUnivariateThreshold) | bmassOutput5$MergedDataSources$unistat_log10pVal > -log10(SNPMarginalMultivariateThreshold),]

        bmassOutput5[c("MarginalSNPs", "Models", "ModelPriors", "LogFile")] <- GetLogBFsFromData(DataSources, bmassOutput5$MarginalSNPs, bmassOutput5$ZScoresCorMatrix, SigmaAlphas, bmassOutput5$LogFile)[c("MarginalSNPs", "Models", "ModelPriors", "LogFile")]

        bmassOutput5[c("MarginalSNPs", "PreviousSNPs", "ModelPriors", "GWASlogBFMinThreshold", "LogFile")] <- DetermineAndApplyPriors(DataSources, bmassOutput5$MarginalSNPs, GWASsnps, SigmaAlphas, bmassOutput5$Models, bmassOutput5$ModelPriors, ProvidedPriors, UseFlatPriors, bmassSeedValue, bmassOutput5$LogFile)[c("MarginalSNPs", "PreviousSNPs", "ModelPriors", "GWASlogBFMinThreshold", "LogFile")]

        bmassOutput5[c("MarginalSNPs", "PreviousSNPs", "NewSNPs", "LogFile")] <- FinalizeAndFormatResults(DataSources, bmassOutput5$MarginalSNPs, bmassOutput5$PreviousSNPs, GWASsnps, bmassOutput5$GWASlogBFMinThreshold, SigmaAlphas, bmassOutput5$Models, bmassOutput5$ModelPriors, NminThreshold, PruneMarginalSNPs, PruneMarginalSNPs_bpWindow, bmassOutput5$LogFile)[c("MarginalSNPs", "PreviousSNPs", "NewSNPs", "LogFile")]

        
	#Testing using bmassOutput1$ZScoresCorMatrix on the bmassOutput5 background
	bmassOutput5.2$MergedDataSources <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/20161227_edits_GlobalLipids2013.dtlesssignif.vs1.SignCrrct.vs1.annot.MAF.txt.gz", header=TRUE)
        bmassOutput5.2$MergedDataSources <- bmassOutput5.2$MergedDataSources[bmassOutput5.2$MergedDataSources$MAF > 0 & !is.na(bmassOutput5.2$MergedDataSources$MAF),]
        bmassOutput5.2$ZScoresCorMatrix <- bmassOutput1$ZScoresCorMatrix
	GWASsnps <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.ChrBP.txt", header=F);
        names(GWASsnps) <- c("Chr", "BP")   
	#Z.LDL,Z.HDL,Z.TC,Z.TG
        DataSources = c("HDL", "LDL", "TC", "TG"); NminThreshold = 50000; bmassSeedValue=150;
        ExpectedColumnNames=c("Chr", "BP", "MAF", "Direction", "pValue", "N"); GWASsnps_AnnotateWindow = 5e5; SNPMarginalUnivariateThreshold = 1e-6; SNPMarginalMultivariateThreshold = 1e-6; SigmaAlphas = c(0.005,0.0075,0.01,0.015,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.15); ProvidedPriors=NULL; UseFlatPriors=FALSE; PruneMarginalSNPs=TRUE; PruneMarginalSNPs_bpWindow=5e5;
        bmassOutput5.2$MergedDataSources$ChrBP <- paste(bmassOutput5.2$MergedDataSources$Chr, "_", bmassOutput5.2$MergedDataSources$BP, sep="")

        bmassOutput5.2$MarginalSNPs$SNPs <- bmassOutput5.2$MergedDataSources[bmassOutput5.2$MergedDataSources$mvstat_log10pVal > -log10(SNPMarginalUnivariateThreshold) | bmassOutput5.2$MergedDataSources$unistat_log10pVal > -log10(SNPMarginalMultivariateThreshold),]

        bmassOutput5.2[c("MarginalSNPs", "Models", "ModelPriors", "LogFile")] <- GetLogBFsFromData(DataSources, bmassOutput5.2$MarginalSNPs, bmassOutput5.2$ZScoresCorMatrix, SigmaAlphas, bmassOutput5.2$LogFile)[c("MarginalSNPs", "Models", "ModelPriors", "LogFile")]

	bmassOutput5.3 <- bmassOutput5.2

        bmassOutput5.2[c("MarginalSNPs", "PreviousSNPs", "ModelPriors", "GWASlogBFMinThreshold", "LogFile")] <- DetermineAndApplyPriors(DataSources, bmassOutput5.2$MarginalSNPs, GWASsnps, SigmaAlphas, bmassOutput5.2$Models, bmassOutput5.2$ModelPriors, ProvidedPriors, UseFlatPriors, bmassSeedValue, bmassOutput5.2$LogFile)[c("MarginalSNPs", "PreviousSNPs", "ModelPriors", "GWASlogBFMinThreshold", "LogFile")]

        bmassOutput5.2[c("MarginalSNPs", "PreviousSNPs", "NewSNPs", "LogFile")] <- FinalizeAndFormatResults(DataSources, bmassOutput5.2$MarginalSNPs, bmassOutput5.2$PreviousSNPs, GWASsnps, bmassOutput5.2$GWASlogBFMinThreshold, SigmaAlphas, bmassOutput5.2$Models, bmassOutput5.2$ModelPriors, NminThreshold, PruneMarginalSNPs, PruneMarginalSNPs_bpWindow, bmassOutput5.2$LogFile)[c("MarginalSNPs", "PreviousSNPs", "NewSNPs", "LogFile")]


	#Testing putting in bmassOutput3$ModelPriors to the bmassOutput5 background
	bmassOutput5.3[c("MarginalSNPs", "PreviousSNPs", "ModelPriors", "GWASlogBFMinThreshold", "LogFile")] <- DetermineAndApplyPriors(DataSources, bmassOutput5.2$MarginalSNPs, GWASsnps, SigmaAlphas, bmassOutput5.2$Models, bmassOutput5.2$ModelPriors, ProvidedPriors=bmassOutput3$ModelPriors, UseFlatPriors, bmassSeedValue, bmassOutput5.2$LogFile)[c("MarginalSNPs", "PreviousSNPs", "ModelPriors", "GWASlogBFMinThreshold", "LogFile")]
        bmassOutput5.3[c("MarginalSNPs", "PreviousSNPs", "NewSNPs", "LogFile")] <- FinalizeAndFormatResults(DataSources, bmassOutput5.3$MarginalSNPs, bmassOutput5.3$PreviousSNPs, GWASsnps, bmassOutput5.3$GWASlogBFMinThreshold, SigmaAlphas, bmassOutput5.3$Models, bmassOutput5.3$ModelPriors, NminThreshold, PruneMarginalSNPs, PruneMarginalSNPs_bpWindow, bmassOutput5.3$LogFile)[c("MarginalSNPs", "PreviousSNPs", "NewSNPs", "LogFile")]


	
	#Testing using the corrected Zscores (some were flipped to matched bmassOutput3.4) on the bmassOutput5 background and with the bmassOutput3 ZScore Correlation Matrix
	bmassOutput5.4$MergedDataSources <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/20161227_edits_GlobalLipids2013.dtlesssignif.vs1.SignCrrct.vs1.annot.MAF.bmassOutput3.4_ZScoreSignMatch.txt.gz", header=TRUE)
        bmassOutput5.4$MergedDataSources <- bmassOutput5.4$MergedDataSources[bmassOutput5.4$MergedDataSources$MAF > 0 & !is.na(bmassOutput5.4$MergedDataSources$MAF),]
        bmassOutput5.4$ZScoresCorMatrix <- bmassOutput3$ZScoresCorMatrix
	GWASsnps <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.ChrBP.txt", header=F);
        names(GWASsnps) <- c("Chr", "BP")   
	#Z.LDL,Z.HDL,Z.TC,Z.TG
        DataSources = c("HDL", "LDL", "TC", "TG"); NminThreshold = 50000; bmassSeedValue=150;
        ExpectedColumnNames=c("Chr", "BP", "MAF", "Direction", "pValue", "N"); GWASsnps_AnnotateWindow = 5e5; SNPMarginalUnivariateThreshold = 1e-6; SNPMarginalMultivariateThreshold = 1e-6; SigmaAlphas = c(0.005,0.0075,0.01,0.015,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.15); ProvidedPriors=NULL; UseFlatPriors=FALSE; PruneMarginalSNPs=TRUE; PruneMarginalSNPs_bpWindow=5e5;
        bmassOutput5.4$MergedDataSources$ChrBP <- paste(bmassOutput5.4$MergedDataSources$Chr, "_", bmassOutput5.4$MergedDataSources$BP, sep="")

        bmassOutput5.4$MarginalSNPs$SNPs <- bmassOutput5.4$MergedDataSources[bmassOutput5.4$MergedDataSources$mvstat_log10pVal > -log10(SNPMarginalUnivariateThreshold) | bmassOutput5.4$MergedDataSources$unistat_log10pVal > -log10(SNPMarginalMultivariateThreshold),]

        bmassOutput5.4[c("MarginalSNPs", "Models", "ModelPriors", "LogFile")] <- GetLogBFsFromData(DataSources, bmassOutput5.4$MarginalSNPs, bmassOutput5.4$ZScoresCorMatrix, SigmaAlphas, bmassOutput5.4$LogFile)[c("MarginalSNPs", "Models", "ModelPriors", "LogFile")]

        bmassOutput5.4[c("MarginalSNPs", "PreviousSNPs", "ModelPriors", "GWASlogBFMinThreshold", "LogFile")] <- DetermineAndApplyPriors(DataSources, bmassOutput5.4$MarginalSNPs, GWASsnps, SigmaAlphas, bmassOutput5.4$Models, bmassOutput5.4$ModelPriors, ProvidedPriors, UseFlatPriors, bmassSeedValue, bmassOutput5.4$LogFile)[c("MarginalSNPs", "PreviousSNPs", "ModelPriors", "GWASlogBFMinThreshold", "LogFile")]

        bmassOutput5.4[c("MarginalSNPs", "PreviousSNPs", "NewSNPs", "LogFile")] <- FinalizeAndFormatResults(DataSources, bmassOutput5.4$MarginalSNPs, bmassOutput5.4$PreviousSNPs, GWASsnps, bmassOutput5.4$GWASlogBFMinThreshold, SigmaAlphas, bmassOutput5.4$Models, bmassOutput5.4$ModelPriors, NminThreshold, PruneMarginalSNPs, PruneMarginalSNPs_bpWindow, bmassOutput5.4$LogFile)[c("MarginalSNPs", "PreviousSNPs", "NewSNPs", "LogFile")]


	#bmassOutput3.3 <- bmass(c("HDL3", "LDL3", "TG3", "TC3"), GWASsnps=GWASsnps3, NminThreshold = 50000, PrintMergedData=TRUE, bmassSeedValue=150, ProvidedPriors=bmassOutput5$ModelPriors);
	bmassOutput3.4 <- bmass(c("HDL3", "LDL3", "TG3", "TC3"), GWASsnps=GWASsnps3, NminThreshold = 50000, PrintMergedData=TRUE, bmassSeedValue=150, PruneMarginalSNPs=FALSE)
```

~~~
> library("devtools"); devtools::load_all("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/bmass");
Loading bmass
> HDL <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_HDL.IncAllele.bmass.formatted.txt.gz", header=T);
> LDL <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_LDL.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.txt.gz", header=T);
> TG <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TG.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.txt.gz", header=T);
> TC <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TC.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.txt.gz", header=T);
> GWASsnps <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.ChrBP.txt", header=F);
> names(GWASsnps) <- c("Chr", "BP")
> #bmassOutput1 <- bmass(c("HDL", "LDL", "TG", "TC"), GWASsnps=GWASsnps, NminThreshold = 1, bmassSeedValue=150);
> #print(bmassOutput1);" > shana1.txt
> DataSources = c("HDL", "LDL", "TG", "TC"); NminThreshold = 50000; bmassSeedValue=150;
> ExpectedColumnNames=c("Chr", "BP", "MAF", "Direction", "pValue", "N"); GWASsnps_AnnotateWindow = 5e5; SNPMarginalUnivariateThreshold = 1e-6; SNPMarginalMultivariateThreshold = 1e-6; SigmaAlphas = c(0.005,0.0075,0.01,0.015,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.15); ProvidedPriors=NULL; UseFlatPriors=FALSE; PruneMarginalSNPs=TRUE; PruneMarginalSNPs_bpWindow=5e5;
>
> bmassOutput1 <- bmass(c("HDL", "LDL", "TG", "TC"), GWASsnps=GWASsnps, NminThreshold = 50000, PrintMergedData=TRUE, bmassSeedValue=150);
Warning message:
In ProcessMergedAndAnnotatedDataSources(DataSources, bmassOutput$MergedDataSources,  :
  2016-12-27 16:21:27 -- One of your datafiles has p-values less than the threshold which R can properly convert them to log-scale, meaning their log-values return as 'Infinite'. bmass automatically replaces these 'Infinite' values with the max, non-infinite Z-scores available, but it is recommended you self-check this as well.
>
> summary(bmassOutput1)
                      Length Class      Mode     
MarginalSNPs             4   -none-     list     
PreviousSNPs             4   -none-     list     
NewSNPs                  3   -none-     list     
LogFile                 18   -none-     character
MergedDataSources       23   data.frame list     
ZScoresCorMatrix        16   -none-     numeric  
Models                 324   -none-     numeric  
ModelPriors           1134   -none-     numeric  
GWASlogBFMinThreshold    1   -none-     numeric  
> head(bmassOutput1$MergedDataSources)
         ChrBP Chr        BP    MAF A1 HDL_A2 HDL_Direction HDL_pValue HDL_N
1  10_10000135  10  10000135 0.4908  a      g             +    0.75380 93561
2  10_10000265  10  10000265 0.4908  c      t             +    0.72950 93561
3 10_100002729  10 100002729 0.3206  a      g             +    0.09234 94311
4 10_100002880  10 100002880 0.3206  a      g             +    0.09340 94311
5 10_100003553  10 100003553 0.2190  t      c             +    0.87230 94311
6 10_100003805  10 100003805 0.3483  c      t             +    0.12400 94311
  HDL_ZScore LDL_Direction LDL_pValue LDL_N LDL_ZScore TG_Direction TG_pValue
1  0.3136327             -    0.71580 89138 -0.3640777            -   0.75670
2  0.3457907             -    0.77480 89138 -0.2861020            -   0.76120
3  1.6831813             -    0.04000 89888 -2.0537489            -   0.06856
4  1.6777291             -    0.08428 89888 -1.7263748            -   0.07994
5  0.1607377             +    0.26890 89888  1.1055993            +   0.11250
6 -1.5381989             -    0.13760 89888  1.4847880            -   0.73140
   TG_N  TG_ZScore TC_Direction TC_pValue  TC_N  TC_ZScore GWASannot
1 90263 -0.3098172            -    0.6530 93845 -0.4495985         0
2 90263 -0.3039054            -    0.6915 93845 -0.3968203         0
3 91013 -1.8213083            -    0.1653 94595 -1.3874651         0
4 91013 -1.7510343            -    0.2550 94595 -1.1382886         0
5 91013  1.5870558            +    0.1513 94595  1.4349547         0
6 91013  0.3432638            -    0.7058 94595  0.3775028         0
> dim(bmassOutput1$PreviousSNPs$SNPs)
[1] 148  29
> dim(bmassOutput1$MarginalSNPs$SNPs)
[1] 252  29
> head(bmassOutput1$MarginalSNPs$SNPs)
             ChrBP Chr        BP    MAF A1 HDL_A2 HDL_Direction HDL_pValue
2044  10_101902054  10 101902054 0.4631  t      c             +  1.287e-06
15280 10_113911344  10 113911344 0.3193  a      g             +  2.022e-17
17437 10_115779365  10 115779365 0.3430  c      t             +  1.712e-07
45109  10_17300296  10  17300296 0.4538  a      g             +  2.648e-04
74515  10_45333283  10  45333283 0.2770  a      c             +  1.706e-10
78810   10_5244847  10   5244847 0.1319  g      a             +  1.141e-01
       HDL_N HDL_ZScore LDL_Direction LDL_pValue  LDL_N LDL_ZScore TG_Direction
2044  186893  -4.841751             +  5.875e-01 172821 -0.5424624            +
15280 184088  -8.492522             +  3.935e-13 170057 -7.2577782            -
17437 187073   5.228166             +  3.965e-01 173006  0.8478886            -
45109 186923  -3.647504             +  3.135e-07 172857 -5.1151450            +
74515 186596  -6.385714             +  6.647e-04 172518 -3.4037400            +
78810 186767   1.580030             +  1.061e-01 172749  1.6159737            +
      TG_pValue     TG_N  TG_ZScore TC_Direction TC_pValue     TC_N TC_ZScore
2044  1.393e-02 177587.1 -2.4590634            + 2.526e-04 187083.0 -3.659609
15280 1.210e-10 174733.9  6.4380728            + 1.472e-15 184217.9 -7.979271
17437 3.631e-03 177771.0 -2.9085578            + 6.209e-02 187258.0  1.865652
45109 7.254e-03 177627.0 -2.6849530            + 2.601e-11 187112.1 -6.667555
74515 4.594e-01 177293.0 -0.7398352            + 8.431e-09 186782.0 -5.759607
78810 1.718e-12 177504.0  7.0556466            + 4.919e-06 186960.0  4.568213
      GWASannot    mvstat mvstat_log10pVal  unistat unistat_log10pVal   Nmin
2044          0  48.52281         9.134137 23.44255          5.890421 172821
15280         2 165.63939        34.044791 72.12293         16.694219 170057
17437         0  31.08461         5.531340 27.33372          6.766496 173006
45109         1  57.81554        11.078700 44.45629         10.584860 172857
74515         1  66.88366        12.986515 40.77734          9.768021 172518
78810         1  72.97818        14.273105 49.78215         11.764977 172749
      logBFWeightedAvg
2044          7.054519
15280        31.249129
17437         4.181650
45109         9.221851
74515        11.555365
78810        12.473907
> dim(bmassOutput1$NewSNPs$SNPs)
[1] 65 29
> bmassOutput1$ZScoresCorMatrix
            HDL_ZScore  LDL_ZScore  TG_ZScore TC_ZScore
HDL_ZScore  1.00000000 -0.08715854 -0.3635593 0.1531368
LDL_ZScore -0.08715854  1.00000000  0.1585048 0.8197523
TG_ZScore  -0.36355928  0.15850485  1.0000000 0.2877814
TC_ZScore   0.15313683  0.81975235  0.2877814 1.0000000
> bmassOutput1$GWASlogBFMinThreshold
[1] 4.28084
> HDL2 <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_HDL.bmass.formatted.txt.gz", header=T);
> LDL2 <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_LDL.bmass.formatted.txt.gz", header=T);
> TG2 <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TG.bmass.formatted.txt.gz", header=T);
> TC2 <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TC.bmass.formatted.txt.gz", header=T);
> GWASsnp2 <- GWASsnps
> 
> table(HDL2$Direction)

      +       0 
2433288   14148 
> table(LDL2$Direction)

      +       0 
2424368   13380 
> table(TG2$Direction)

      +       0 
2424745   14684 
> table(TC2$Direction)

      +       0 
2433334   13642 
> HDL2 <- HDL2[HDL2$Direction != 0,]
> LDL2 <- LDL2[LDL2$Direction != 0,]
> TG2 <- TG2[TG2$Direction != 0,]
> TC2 <- TC2[TC2$Direction != 0,]
> 
> table(HDL2$Direction)

      +       0 
2433288       0 
> table(LDL2$Direction)

      +       0 
2424368       0 
> table(TG2$Direction)

      +       0 
2424745       0 
> table(TC2$Direction)

      +       0 
2433334       0 
> HDL3 <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_HDL.bmass.formatted.txt.gz", header=T);
> LDL3 <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_LDL.bmass.formatted.MatchedToHDL.txt.gz", header=T);
> TG3 <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TG.bmass.formatted.MatchedToHDL.txt.gz", header=T);
> TC3 <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TC.bmass.formatted.MatchedToHDL.txt.gz", header=T);
> GWASsnp3 <- GWASsnps
>
> table(HDL3$Direction)

      +       0 
2433288   14148 
> table(LDL3$Direction)

      -       +       0 
1146655 1020189       3 
> table(TG3$Direction)

      -       + 
1380007  788165 
> table(TC3$Direction)

      -       + 
1120423 1324948 
> 
> HDL3 <- HDL3[HDL3$Direction != 0,]
> LDL3 <- LDL3[LDL3$Direction != 0,]
> TG3 <- TG3[TG3$Direction != 0,]
> TC3 <- TC3[TC3$Direction != 0,]
> 
> table(HDL3$Direction)

      +       0 
2433288       0 
> table(LDL3$Direction)

      -       +       0 
1146655 1020189       0 
> table(TG3$Direction)

      -       + 
1380007  788165 
> table(TC3$Direction)

      -       + 
1120423 1324948 
> bmassOutput2 <- bmass(c("HDL2", "LDL2", "TG2", "TC2"), GWASsnps=GWASsnps2, NminThreshold = 50000, PrintMergedData=TRUE, bmassSeedValue=150);
Error in `[<-.data.frame`(`*tmp*`, MAF_CheckList, , value = list(ChrBP = c("10_10000135",  : 
  missing values are not allowed in subscripted assignments of data frames
> bmassOutput3 <- bmass(c("HDL3", "LDL3", "TG3", "TC3"), GWASsnps=GWASsnps3, NminThreshold = 50000, PrintMergedData=TRUE, bmassSeedValue=150);
Error in `[<-.data.frame`(`*tmp*`, MAF_CheckList, , value = list(ChrBP = c("10_10000135",  : 
  missing values are not allowed in subscripted assignments of data frames
>         
> head(HDL2)
  Chr        BP    MAF A1 A2 Direction  pValue     N
1  10  10000135 0.5092  g  a         + 0.75380 93561
2  10  10000265 0.5092  t  c         + 0.72950 93561
3  10 100002729 0.6794  g  a         + 0.09234 94311
4  10 100002880 0.6794  g  a         + 0.09340 94311
5  10 100003553 0.7810  c  t         + 0.87230 94311
6  10 100003805 0.3483  t  c         + 0.12400 94311
> table(is.na(HDL2[,1]))

  FALSE 
2433288 
> table(is.na(HDL2[,2]))

  FALSE 
2433288 
> table(is.na(HDL2[,3]))

  FALSE    TRUE 
2247211  186077 
> table(is.na(HDL2[,4]))

  FALSE 
2433288 
> table(is.na(HDL2[,5]))

  FALSE 
2433288 
> table(is.na(HDL2[,6]))

  FALSE 
2433288 
> table(is.na(HDL2[,7]))

  FALSE 
2433288 
> table(is.na(HDL2[,8]))

  FALSE 
2433288 
> table(is.na(LDL2[,3]))

  FALSE    TRUE 
2238566  185802 
> table(is.na(TG2[,3]))

  FALSE    TRUE 
2238827  185918 
> table(is.na(TC2[,3]))

  FALSE    TRUE 
2246688  186646 
> table(is.na(HDL3[,3]))

  FALSE    TRUE 
2247211  186077 
> table(is.na(LDL3[,3]))

  FALSE    TRUE 
2115718   51126 
> table(is.na(TG3[,3]))

  FALSE    TRUE 
2116907   51265 
> table(is.na(TC3[,3]))

  FALSE    TRUE 
2259342  186029 
> HDL2 <- HDL2[!is.na(HDL2$MAF),]
> table(is.na(HDL2[,3]))

  FALSE 
2247211 
> HDL2 <- HDL2[!is.na(HDL2$MAF),]
> LDL2 <- LDL2[!is.na(LDL2$MAF),]
> TG2 <- TG2[!is.na(TG2$MAF),]
> TC2 <- TC2[!is.na(TC2$MAF),]
> HDL3 <- HDL3[!is.na(HDL3$MAF),]
> LDL3 <- LDL3[!is.na(LDL3$MAF),]
> TG3 <- TG3[!is.na(TG3$MAF),]
> TC3 <- TC3[!is.na(TC3$MAF),]
> table(is.na(LDL[,3]))      

  FALSE 
2437512 
> table(is.na(LDL2[,3]))

  FALSE 
2238566 
> table(is.na(TG2[,3]))

  FALSE 
2238827 
> table(is.na(TC2[,3]))

  FALSE 
2246688 
> table(is.na(LDL3[,3]))

  FALSE 
2115718 
> table(is.na(HDL3[,3]))

  FALSE 
2247211 
> table(is.na(TG3[,3]))

  FALSE 
2116907 
> table(is.na(TC3[,3]))

  FALSE 
2259342 
> bmassOutput2 <- bmass(c("HDL2", "LDL2", "TG2", "TC2"), GWASsnps=GWASsnps2, NminThreshold = 50000, PrintMergedData=TRUE, bmassSeedValue=150);
Warning message:
In ProcessMergedAndAnnotatedDataSources(DataSources, bmassOutput$MergedDataSources,  :
  2016-12-27 23:05:26 -- One of your datafiles has p-values less than the threshold which R can properly convert them to log-scale, meaning their log-values return as 'Infinite'. bmass automatically replaces these 'Infinite' values with the max, non-infinite Z-scores available, but it is recommended you self-check this as well.
> bmassOutput3 <- bmass(c("HDL3", "LDL3", "TG3", "TC3"), GWASsnps=GWASsnps3, NminThreshold = 50000, PrintMergedData=TRUE, bmassSeedValue=150);
Warning message:
In ProcessMergedAndAnnotatedDataSources(DataSources, bmassOutput$MergedDataSources,  :
  2016-12-27 23:10:50 -- One of your datafiles has p-values less than the threshold which R can properly convert them to log-scale, meaning their log-values return as 'Infinite'. bmass automatically replaces these 'Infinite' values with the max, non-infinite Z-scores available, but it is recommended you self-check this as well.
> summary(bmassOutput2)
                      Length Class      Mode     
MarginalSNPs             4   -none-     list     
PreviousSNPs             4   -none-     list     
NewSNPs                  3   -none-     list     
LogFile                 18   -none-     character
MergedDataSources       23   data.frame list     
ZScoresCorMatrix        16   -none-     numeric  
Models                 324   -none-     numeric  
ModelPriors           1134   -none-     numeric  
GWASlogBFMinThreshold    1   -none-     numeric  
> summary(bmassOutput3)
                      Length Class      Mode     
MarginalSNPs             4   -none-     list     
PreviousSNPs             4   -none-     list     
NewSNPs                  3   -none-     list     
LogFile                 18   -none-     character
MergedDataSources       23   data.frame list     
ZScoresCorMatrix        16   -none-     numeric  
Models                 324   -none-     numeric  
ModelPriors           1134   -none-     numeric  
GWASlogBFMinThreshold    1   -none-     numeric  
> dim(bmassOutput2$NewSNPs$SNPs)
[1] 199  29
> dim(bmassOutput3$NewSNPs$SNPs)
[1] 65 29
>         bmassOutput4 <- list()
>         bmassOutput4$MergedDataSources <- NULL
>         if (!is.null(MergedDataSources)) {
+                 bmassOutput4$MergedDataSources <- MergedDataSources
+         }
Error: object 'MergedDataSources' not found
>         bmassOutput4$ZScoresCorMatrix <- NULL
>         bmassOutput4$MarginalSNPs <- list()
>         bmassOutput4$Models <- NULL
>         bmassOutput4$ModelPriors <- NULL
>         bmassOutput4$PreviousSNPs <- list()
>         bmassOutput4$NewSNPs <- list()
>         bmassOutput4$GWASlogBFMinThreshold <- NULL
>         #LogFile <- c()
>         bmassOutput4$LogFile <- c()
> 
>         bmassOutput4$LogFile <- rbind(bmassOutput4$LogFile, paste(format(Sys.time()), " -- beginning bmass.", sep=""))
> 
>         bmassOutput4$MergedDataSources <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/20161115_edits_GlobalLipids2013.dtlesssignif.vs1.SignCrrct.vs1.annot.MAF.txt.gz", h
eader=TRUE)
>         bmassOutput4$ZScoresCorMatrix <- as.matrix(read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GlobalLipids2013.RSS0.vs1.SignCrrct.vs1.txt", header=TRUE, sep=","))
>         GWASsnps <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.ChrBP.txt", header=F);
>         names(GWASsnps) <- c("Chr", "BP")
>         #Z.LDL,Z.HDL,Z.TC,Z.TG
>         DataSources = c("LDL", "HDL", "TC", "TG"); NminThreshold = 50000; bmassSeedValue=150;
>         ExpectedColumnNames=c("Chr", "BP", "MAF", "Direction", "pValue", "N"); GWASsnps_AnnotateWindow = 5e5; SNPMarginalUnivariateThreshold = 1e-6; SNPMarginalMultivariateThreshold = 1e-6; SigmaAlphas = c(0.005,0.0075,0.01,0.015,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.15); ProvidedPriors=NULL; UseFlatPriors=FALSE; PruneMarginalSNPs=TRUE; PruneMarginalSNPs_bpWindow=5e5;
>         bmassOutput4$MergedDataSources$ChrBP <- paste(bmassOutput4$MergedDataSources$Chr, "_", bmassOutput4$MergedDataSources$BP, sep="")
>
> dim(bmassOutput4$MarginalSNPs$SNPs)
[1] 11309    29
#From /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/process.MTedits.ForGlobalLipids2013.vs1.SignCrrct.vs1.R 
#> dim(dtlesssignif)
#[1] 11573     8
>
>         bmassOutput4$MarginalSNPs$SNPs <- bmassOutput4$MergedDataSources[bmassOutput4$MergedDataSources$mvstat_log10pVal > -log10(SNPMarginalUnivariateThreshold) | bmassOutput4$MergedDataSources$unistat_log10pVal > -log10(SNPMarginalMultivariateThreshold),]
> 
>         bmassOutput4[c("MarginalSNPs", "Models", "ModelPriors", "LogFile")] <- GetLogBFsFromData(DataSources, bmassOutput4$MarginalSNPs, bmassOutput4$ZScoresCorMatrix, SigmaAlphas, bmassOutput4$LogFile)[c("MarginalSNPs", "Models", "ModelPriors", "LogFile")]
> 
> #       Nmins <- apply(bmassOutput4$MarginalSNPs$SNPs[,c("TC_N", "TG_N", "LDL_N", "HDL_N")], 1, min)
> #       ZScores <- bmassOutput4$MarginalSNPs$SNPs[,c("TC_ZScore", "TG_ZScore", "HDL_ZScore", "LDL_ZScore")]
> #       MarginalSNPs_logBFs <- compute.allBFs.fromZscores(ZScores, bmassOutput4$ZScoresCorMatrix, Nmins, bmassOutput4$MarginalSNPs$SNPs$MAF, SigmaAlphas)
> 
>         bmassOutput4[c("MarginalSNPs", "PreviousSNPs", "ModelPriors", "GWASlogBFMinThreshold", "LogFile")] <- DetermineAndApplyPriors(DataSources, bmassOutput4$MarginalSNPs, GWASsnps, SigmaAlphas, bmassOutput4$Models, bmassOutput4$ModelPriors, ProvidedPriors, UseFlatPriors, bmassSeedValue, bmassOutput4$LogFile)[c("MarginalSNPs", "PreviousSNPs", "ModelPriors", "GWASlogBFMinThreshold", "LogFile")]
> 
>         bmassOutput4[c("MarginalSNPs", "PreviousSNPs", "NewSNPs", "LogFile")] <- FinalizeAndFormatResults(DataSources, bmassOutput4$MarginalSNPs, bmassOutput4$PreviousSNPs, GWASsnps, bmassOutput4$GWASlogBFMinThreshold, SigmaAlphas, bmassOutput4$Models, bmassOutput4$ModelPriors, NminThreshold, PruneMarginalSNPs, PruneMarginalSNPs_bpWindow, bmassOutput4$LogFile)[c("MarginalSNPs", "PreviousSNPs", "NewSNPs", "LogFile")]
> 
> dim(bmassOutput4$NewSNPs$SNPs)
[1] 92 31
> bmassOutput2$ZScoresCorMatrix
            HDL2_ZScore LDL2_ZScore TG2_ZScore TC2_ZScore
HDL2_ZScore   1.0000000   0.6764348  0.7045817  0.6792574
LDL2_ZScore   0.6764348   1.0000000  0.6798776  0.8750846
TG2_ZScore    0.7045817   0.6798776  1.0000000  0.6938238
TC2_ZScore    0.6792574   0.8750846  0.6938238  1.0000000
> bmassOutput2$GWASlogBFMinThreshold
[1] 4.284365
> bmassOutput3$ZScoresCorMatrix
            HDL3_ZScore LDL3_ZScore TG3_ZScore TC3_ZScore
HDL3_ZScore  1.00000000 -0.08744893 -0.3647370  0.1536905
LDL3_ZScore -0.08744893  1.00000000  0.1585641  0.8196124
TG3_ZScore  -0.36473696  0.15856412  1.0000000  0.2871750
TC3_ZScore   0.15369046  0.81961243  0.2871750  1.0000000
> bmassOutput3$GWASlogBFMinThreshold
[1] 4.281287
> bmassOutput4$ZScoresCorMatrix
           Z.LDL       Z.HDL       Z.TC       Z.TG
[1,]  1.00000000 -0.05101127 0.83617727  0.1446093
[2,] -0.05101127  1.00000000 0.09840344 -0.2298445
[3,]  0.83617727  0.09840344 1.00000000  0.3389824
[4,]  0.14460929 -0.22984452 0.33898241  1.0000000
> bmassOutput4$GWASlogBFMinThreshold
[1] 3.989499
> quantile(bmassOutput1$PreviousSNPs$SNPs$logBFWeightedAvg)
        0%        25%        50%        75%       100% 
  4.280840   6.897337  10.107299  18.401784 481.572551 
> quantile(bmassOutput4$PreviousSNPs$SNPs$logBFWeightedAvg)
         0%         25%         50%         75%        100% 
   3.989499    9.452513   15.628748   29.678137 1235.633742 
> bmassOutput1$PreviousSNPs$SNPs[bmassOutput1$PreviousSNPs$SNPs$logBFWeightedAvg > 250,]
              ChrBP Chr        BP     MAF A1 HDL_A2 HDL_Direction HDL_pValue
141903  1_109619829   1 109619829 0.21240  t      g             +  7.608e-13
729318  15_56470658  15  56470658 0.36680  g      a             + 1.240e-188
805943  16_55550825  16  55550825 0.29420  c      a             +  0.000e+00
1001775 19_11063306  19  11063306 0.09763  g      t             +  6.321e-05
1304810  2_27584444   2  27584444 0.41290  t      c             +  1.737e-03
2239873  8_19888502   8  19888502 0.12140  a      g             + 1.380e-149
           HDL_N HDL_ZScore LDL_Direction LDL_pValue    LDL_N LDL_ZScore
141903  155813.0  -7.168038             - 5.400e-241 142644.0 33.1514351
729318  185482.0 -29.293909             +  6.473e-01 171461.0 -0.4575163
805943  177533.0 -38.141507             -  2.220e-34 164865.0 12.2273740
1001775 184617.2  -4.000495             - 3.850e-262 170607.9 34.5878329
1304810 187062.0   3.131865             -  1.509e-07 172995.0 -5.2514579
2239873 187049.3 -26.049025             -  5.045e-01 172984.9  0.6674261
        TG_Direction  TG_pValue     TG_N  TG_ZScore TC_Direction  TC_pValue
141903             -  6.652e-01 146759.0   0.432745            - 1.700e-170
729318             +  2.318e-18 176156.0  -8.740637            +  7.334e-47
805943             -  1.584e-25 169275.0  10.442607            +  3.459e-31
1001775            -  1.043e-01 175280.0   1.624355            - 5.430e-202
1304810            - 2.290e-239 177765.0 -33.038308            -  3.079e-42
2239873            - 1.820e-199 177749.9  30.132478            +  7.352e-01
            TC_N   TC_ZScore GWASannot   mvstat mvstat_log10pVal   unistat
141903  155873.0  27.8336313         1 1218.835         261.8811 1099.0176
729318  185639.0 -14.3758724         1 1311.783         282.0325  858.1331
805943  177497.0 -11.6149724         1 2238.365         483.0054 1454.7746
1001775 184763.8  30.3246241         1 1271.658         273.3330 1196.3182
1304810 187251.0 -13.6192053         1 1189.525         255.5270 1091.5298
2239873 187233.3  -0.3382166         1 1195.548         256.8326  907.9662
        unistat_log10pVal     Nmin logBFWeightedAvg
141903           240.2676 142644.0         259.4150
729318           187.9066 171461.0         279.3921
805943           317.5800 164865.0         481.5726
1001775          261.4145 170607.9         270.2174
1304810          238.6402 172995.0         252.4891
2239873          198.7399 172984.9         255.0694
> bmassOutput4$PreviousSNPs$SNPs[bmassOutput4$PreviousSNPs$SNPs$logBFWeightedAvg > 500,]
                SNP Chr       BP     SNP.1 A1 A2    MAF LDL_Beta LDL_SE  LDL_N
3142 chr16:55550825  16 56993324 rs3764261  c  a 0.2942   0.0528 0.0042 164865
7853 chr15:56470658  15 58683366 rs1532085  a  g 0.3668   0.0026 0.0037 171461
     HDL_Beta HDL_SE  HDL_N TG_Beta  TG_SE   TG_N TC_Beta  TC_SE   TC_N
3142   0.2412 0.0039 177533  0.0395 0.0038 169275  0.0503 0.0040 177497
7853   0.1068 0.0035 185482  0.0310 0.0034 176156  0.0545 0.0036 185639
     GWASannot Gene TG_ZScore  TC_ZScore LDL_ZScore HDL_ZScore   mvstat
3142         1   NA -38.15340 12.2273740  -11.61497  10.442607 2852.227
7853         1   NA  29.29391  0.4575163   14.37587   8.740637 1266.253
     mvstat_log10pVal unistat_log10pVal       ChrBP   Nmin logBFWeightedAvg
3142         616.1989          317.7773 16_56993324 164865        1235.6337
7853         272.1611          187.9066 15_58683366 171461         703.3767
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GlobalLipids2013.dtlesssignif.vs1.SignCrrct.vs1.txt | head -n 10
rs,Z.HDL,Z.LDL,Z.TC,Z.TG,mvstat,mvp,unip
chr10:101902054,4.84175096013916,0.54246240431255,3.65960938616346,2.45906343998997,48.768882385891,9.18546064700021,5.89042145309562
chr10:101902184,4.28996658928395,0.741979097659704,3.31897041855539,2.04241886517916,36.3776853917822,6.61626523237262,4.74787544749436
chr10:101914408,4.16769148760827,0.694450074537328,3.21159569456144,2.34730043800811,35.6324406198061,6.46295377959646,4.51187250375254
chr10:101950956,3.49559840808192,2.98132929795538,4.80442095262784,1.89987138538418,34.3755765844435,6.2047817211044,5.80910828307783
chr10:101989736,3.87804325071873,2.03680088096434,4.15303118888146,1.89426382584585,33.651359603728,6.05625379108408,4.48399376961395
chr10:113892408,4.80727691171168,5.40513388872724,5.94718082470692,-2.54333468717218,68.150118444231,13.2536117777103,8.56415563401556
chr10:113900711,8.38226223514305,7.38382982844603,8.0385200273586,-5.79780128305849,175.072295946929,36.069345409152,16.2846653162077
chr10:113903212,4.76030564571111,5.5312813228826,6.04447208869014,-2.04932853694305,65.6788629080588,12.7325608116026,8.82419836715172
chr10:113906292,4.92644305955787,5.60832002246494,6.01116043492249,-2.27626756787699,68.3490493954776,13.2955792602323,8.73471037413917
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GlobalLipids2013.dtlesssignif.vs1.SignCrrct.vs1.txt | grep chr16:55550825
chr16:55550825,38.153404647743,-12.2273739740071,11.6149724163073,-10.4426069018792,2852.22747879747,616.198868999995,317.777283224373
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GlobalLipids2013.dtlesssignif.vs1.SignCrrct.vs1.annot.MAF.txt.gz | head -n 10
posHg18 chr     pos     snp     a1      a2      maf     beta_LDL        se_LDL  n_LDL   beta_HDL        se_HDL  n_HDL   beta_TG se_TG   n_TG    beta_TC se_TC   n_TC    annot   gene    Z.tg    Z.tc    Z.ldl   Z.hdl   mvstatmvp unip
chr2:27647654   2       27794150        rs4260197       g       c       0.4723  0.0014  0.0036  172858.00       0.0034  0.0034  186895.00       0.0226  0.0033  177617.00       0.0063  0.0035  187085.00       2       NA    1.11441777795726 -0.37736819416554       -1.64791555971734       -6.72515966642187       46.041078916398 8.617110500481  10.75597041097
chr11:122011218 11      122506008       rs6589938       t       g       0.3958  0.0239  0.0053  89888.00        0.0217  0.0048  94311.00        0.0030  0.0047  91013.00        0.0315  0.0051  94595.00        2       NA    4.86999689907688 4.40070446560701        6.13816525139913        0.517653567562213       56.5330861681492        10.8096322565807        9.07841755960658
chr11:55768424  11      56011848        rs1905053       c       t       0.128   0.0007  0.0080  84452.00        0.0351  0.0073  88834.00        0.0055  0.0077  85492.00        0.0146  0.0078  89074.00        0       NA    -4.54675250180027        0.582098998069215       -2.31450568087186       -0.908905519584347      37.453897736508 6.83795063746814        5.26376290109527
chr11:116101674 11      116596464       rs60583202      a       g       0.06069 0.0684  0.0104  77810.60        0.0903  0.0098  86114.26        0.2003  0.0093  81176.36        0.0973  0.0099  85850.56        2       NA    8.32912796628562 -6.305912969167 -9.39916075941876       -19.8357217790965       422.06640802648 89.3241521253346        86.8344589232776
chr8:126543488  8       126474306       rs2980885       g       a       0.2177  0.0313  0.0045  172989.02       0.0354  0.0042  187053.05       0.0578  0.0041  177756.98       0.0337  0.0044  187245.96       2       NA    7.66903140951649 -6.53744176280611       -7.46611048441843       -13.2806701245226       219.973486284765        45.7213645218472        39.5233132570543
chr8:9707185    8       9669775 rs10104807      g       a       0.3456  0.0042  0.0038  172681.10       0.0098  0.0035  186722.00       0.0212  0.0035  177440.10       0.001   0.0037  186893.00       2       NA      2.26553285841545       1.13303394170561        0.336757120162384       -5.56697470766763       36.9188966062519        6.72770601364781        7.58636500280144
chr8:19895890   8       19851610        rs17091908      c       t       0.02639 0.0109  0.0172  83107.38        0.1670  0.0163  92749.71        0.1343  0.0158  86746.33        0.0173  0.0163  92645.65        2       NA    9.26985451347923 -0.50522615781421       1.23831306836121        -7.57389399535419       133.137018473767        27.0805918929069        19.729554091982
chr1:109597131  1       109795608       rs437444        c       t       0.03562 0.0818  0.0099  159110.00       0.0330  0.0093  172747.00       0.0186  0.0091  164020.00       0.0621  0.0096  172047.00       2       NA    -3.45696292677496        7.76746888414013        6.23620972254678        1.79861249151656        69.8711064154011        13.6167936453774        14.0965301714928
chr2:136168146  2       136451676       rs12465544      t       c       0.2309  0.0333  0.0071  86806.00        0.0183  0.0066  91192.00        0.0051  0.0066  87892.00        0.0352  0.0070  91474.00        0       NA    2.55137112261234 4.72177086137641        5.09797838876236        -0.866346816134571      35.7441624448977        6.48592645802003        6.46432619657425
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GlobalLipids2013.dtlesssignif.vs1.SignCrrct.vs1.annot.RAF.txt.gz | head -n 10
posHg18 chr pos snp a1 a2 raf beta_LDL se_LDL n_LDL beta_HDL se_HDL n_HDL beta_TG se_TG n_TG beta_TC se_TC n_TC annot gene Z.tg Z.tc Z.ldl Z.hdl mvstat mvp unip
chr2:27647654   2       27794150        rs4260197       g       c       0.4723  0.0014  0.0036  172858.00       0.0034  0.0034  186895.00       0.0226  0.0033  177617.00       0.0063  0.0035  187085.00       2       NA    1.11441777795726 -0.37736819416554       -1.64791555971734       -6.72515966642187       46.041078916398 8.617110500481  10.75597041097
chr11:122011218 11      122506008       rs6589938       t       g       0.3958  0.0239  0.0053  89888.00        0.0217  0.0048  94311.00        0.0030  0.0047  91013.00        0.0315  0.0051  94595.00        2       NA    4.86999689907688 4.40070446560701        6.13816525139913        0.517653567562213       56.5330861681492        10.8096322565807        9.07841755960658
chr11:55768424  11      56011848        rs1905053       c       t       0.872   0.0007  0.0080  84452.00        0.0351  0.0073  88834.00        0.0055  0.0077  85492.00        0.0146  0.0078  89074.00        0       NA    4.54675250180027 -0.582098998069215      2.31450568087186        0.908905519584347       37.453897736508 6.83795063746814        5.26376290109527
chr11:116101674 11      116596464       rs60583202      a       g       0.06069 0.0684  0.0104  77810.60        0.0903  0.0098  86114.26        0.2003  0.0093  81176.36        0.0973  0.0099  85850.56        2       NA    8.32912796628562 -6.305912969167 -9.39916075941876       -19.8357217790965       422.06640802648 89.3241521253346        86.8344589232776
chr8:126543488  8       126474306       rs2980885       g       a       0.2177  0.0313  0.0045  172989.02       0.0354  0.0042  187053.05       0.0578  0.0041  177756.98       0.0337  0.0044  187245.96       2       NA    7.66903140951649 -6.53744176280611       -7.46611048441843       -13.2806701245226       219.973486284765        45.7213645218472        39.5233132570543
chr8:9707185    8       9669775 rs10104807      g       a       0.3456  0.0042  0.0038  172681.10       0.0098  0.0035  186722.00       0.0212  0.0035  177440.10       0.001   0.0037  186893.00       2       NA      2.26553285841545       1.13303394170561        0.336757120162384       -5.56697470766763       36.9188966062519        6.72770601364781        7.58636500280144
chr8:19895890   8       19851610        rs17091908      c       t       0.02639 0.0109  0.0172  83107.38        0.1670  0.0163  92749.71        0.1343  0.0158  86746.33        0.0173  0.0163  92645.65        2       NA    9.26985451347923 -0.50522615781421       1.23831306836121        -7.57389399535419       133.137018473767        27.0805918929069        19.729554091982
chr1:109597131  1       109795608       rs437444        c       t       0.96438 0.0818  0.0099  159110.00       0.0330  0.0093  172747.00       0.0186  0.0091  164020.00       0.0621  0.0096  172047.00       2       NA    3.45696292677496 -7.76746888414013       -6.23620972254678       -1.79861249151656       69.8711064154011        13.6167936453774        14.0965301714928
chr2:136168146  2       136451676       rs12465544      t       c       0.2309  0.0333  0.0071  86806.00        0.0183  0.0066  91192.00        0.0051  0.0066  87892.00        0.0352  0.0070  91474.00        0       NA    2.55137112261234 4.72177086137641        5.09797838876236        -0.866346816134571      35.7441624448977        6.48592645802003        6.46432619657425
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/jointGwasMc_AllPheno.Annot.vs1.SignCrrct.RAF.txt.gz | head -n 10
posHg18 chr pos snp a1 a2 raf beta_LDL se_LDL n_LDL beta_HDL se_HDL n_HDL beta_TG se_TG n_TG beta_TC se_TC n_TC annot gene
chr6:54599910   6       54491951        rs1280728       a       t       NA      NA      NA      NA      0.0107  0.0074  94311.00        NA      NA      NA      0.0125  0.0080  94595.00        0       NA
chr9:3746564    9       3756564 rs10814541      c       t       0.7018  0.0011  0.0064  89888.00        0.0036  0.0058  94311.00        0.0119  0.0057  91013.00        0.0025  0.0062  94595.00        0       NA
chr5:133405061  5       133377162       rs4958172       g       a       0.8813  0       0.0089  89817.00        0.0072  0.0082  94239.00        0.0029  0.0080  90938.00        0.0029  0.0087  94520.00        0       NA
chr12:69055675  12      70769408        rs2053102       c       g       NA      NA      NA      NA      0.0075  0.0069  94311.00        NA      NA      NA      0.0009  0.0073  94595.00        0       NA
chr5:155873914  5       155941336       rs256837        c       a       0.7995  0.0016  0.0066  88199.00        0.0002  0.0061  92620.00        0.0013  0.0060  89323.00        0.0031  0.0065  92905.00        2       NA
chr3:60337349   3       60362309        rs1569354       a       t       NA      NA      NA      NA      0.0060  0.0050  94311.00        NA      NA      NA      0.0025  0.0053  94595.00        0       NA
chr4:65034366   4       65351771        rs6851961       a       g       0.5369  0.0003  0.0053  89877.00        0.0041  0.0049  94300.00        0.0025  0.0048  91002.00        0.0022  0.0052  94584.00        0       NA
chr7:62372905   7       62735470        rs11560217      c       a       0.96966 0.0126  0.0161  83513.00        0.0236  0.0147  88983.00        0.0041  0.0148  81387.00        0.0008  0.0157  88090.00        0       NA
chr17:6058839   17      6118115 rs17803227      a       g       0.1016  0.0046  0.0094  88422.00        0.0055  0.0086  92809.00        0.0081  0.0084  89474.00        0.0013  0.0092  93056.00        0       NA
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GlobalLipids2013.dtlesssignif.vs1.SignCrrct.vs1.txt | head -n 10
rs,Z.HDL,Z.LDL,Z.TC,Z.TG,mvstat,mvp,unip
chr10:101902054,4.84175096013916,0.54246240431255,3.65960938616346,2.45906343998997,48.768882385891,9.18546064700021,5.89042145309562
chr10:101902184,4.28996658928395,0.741979097659704,3.31897041855539,2.04241886517916,36.3776853917822,6.61626523237262,4.74787544749436
chr10:101914408,4.16769148760827,0.694450074537328,3.21159569456144,2.34730043800811,35.6324406198061,6.46295377959646,4.51187250375254
chr10:101950956,3.49559840808192,2.98132929795538,4.80442095262784,1.89987138538418,34.3755765844435,6.2047817211044,5.80910828307783
chr10:101989736,3.87804325071873,2.03680088096434,4.15303118888146,1.89426382584585,33.651359603728,6.05625379108408,4.48399376961395
chr10:113892408,4.80727691171168,5.40513388872724,5.94718082470692,-2.54333468717218,68.150118444231,13.2536117777103,8.56415563401556
chr10:113900711,8.38226223514305,7.38382982844603,8.0385200273586,-5.79780128305849,175.072295946929,36.069345409152,16.2846653162077
chr10:113903212,4.76030564571111,5.5312813228826,6.04447208869014,-2.04932853694305,65.6788629080588,12.7325608116026,8.82419836715172
chr10:113906292,4.92644305955787,5.60832002246494,6.01116043492249,-2.27626756787699,68.3490493954776,13.2955792602323,8.73471037413917
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/dtlesssignif.txt | head -n 10
rs,Z.tg,Z.tc,Z.ldl,Z.hdl,mvstat,mvp,unip
chr10:101902054,-2.19925212246208,-3.47798404219811,0.221118712997571,-4.70217875377794,43.9274041593846,8.17767271147827,5.58939145743163
chr10:101998159,-1.87859364172538,-4.64921546125538,-2.85913347775579,-3.29494769718191,33.6897163071804,6.06411584439104,5.47729500726525
chr10:107242452,Inf,1.29303197614424,1.9497996777359,0.528143126468276,NA,NA,Inf
chr10:1081195,0.932651371845857,Inf,0.451540558796654,-0.0298333012789937,NA,NA,Inf
chr10:113892408,-2.29105902640745,-5.83263142306766,-5.27952387879639,-4.66675085358473,62.5066384973055,12.0645696755321,8.26312563835158
chr10:113900711,-5.68040012829455,-7.95312818670872,-7.29102748537499,-8.30031009432857,160.378697998669,32.9162927187097,15.9836353205437
chr10:113903212,-1.74571780292366,-5.93170769988567,-5.40842895118219,-4.61845918556053,62.2754393895005,12.0159247839646,8.52316837148774
chr10:113903542,-1.71459230796557,-5.89284606552626,-5.40108236790352,-4.53159500164139,61.0931696686151,11.7672608841947,8.42067379624475
chr10:113906292,-1.99852289798056,-5.89779108321291,-5.48709388302838,-4.78916108555607,64.574092320775,12.499810726148,8.43368037847519
>
>         bmassOutput4 <- list()
>         bmassOutput4$MergedDataSources <- NULL
>         if (!is.null(MergedDataSources)) {
+                 bmassOutput4$MergedDataSources <- MergedDataSources
+         }
Error: object 'MergedDataSources' not found
>         bmassOutput4$ZScoresCorMatrix <- NULL
>         bmassOutput4$MarginalSNPs <- list()
>         bmassOutput4$Models <- NULL
>         bmassOutput4$ModelPriors <- NULL
>         bmassOutput4$PreviousSNPs <- list()
>         bmassOutput4$NewSNPs <- list()
>         bmassOutput4$GWASlogBFMinThreshold <- NULL
>         #LogFile <- c()
>         bmassOutput4$LogFile <- c()
>
>         bmassOutput4$LogFile <- rbind(bmassOutput4$LogFile, paste(format(Sys.time()), " -- beginning bmass.", sep=""))
>
>         bmassOutput5 <- bmassOutput4
>
>         bmassOutput5$MergedDataSources <- bmassOutput5$MergedDataSources[bmassOutput5$MergedDataSources$MAF > 0 & !is.na(bmassOutput5$MergedDataSources$MAF),]
>         bmassOutput5$ZScoresCorMatrix <- as.matrix(read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GlobalLipids2013.RSS0.vs1.SignCrrct.vs1.txt", header=TRUE, sep=","))
>         GWASsnps <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.ChrBP.txt", header=F);
>         names(GWASsnps) <- c("Chr", "BP")
>         #Z.LDL,Z.HDL,Z.TC,Z.TG
>         DataSources = c("LDL", "HDL", "TC", "TG"); NminThreshold = 50000; bmassSeedValue=150;
>         ExpectedColumnNames=c("Chr", "BP", "MAF", "Direction", "pValue", "N"); GWASsnps_AnnotateWindow = 5e5; SNPMarginalUnivariateThreshold = 1e-6; SNPMarginalMultivariateThreshold = 1e-6; SigmaAlphas = c(0.005,0.0075,0.01,0.015,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.15); ProvidedPriors=NULL; UseFlatPriors=FALSE; PruneMarginalSNPs=TRUE; PruneMarginalSNPs_bpWindow=5e5;
>         bmassOutput5$MergedDataSources$ChrBP <- paste(bmassOutput5$MergedDataSources$Chr, "_", bmassOutput5$MergedDataSources$BP, sep="")
>         
>         bmassOutput5$MarginalSNPs$SNPs <- bmassOutput5$MergedDataSources[bmassOutput5$MergedDataSources$mvstat_log10pVal > -log10(SNPMarginalUnivariateThreshold) | bmassOutput5$MergedDataSources$unistat_log10pVal > -log10(SNPMarginalMultivariateThreshold),]
>         
>         bmassOutput5[c("MarginalSNPs", "Models", "ModelPriors", "LogFile")] <- GetLogBFsFromData(DataSources, bmassOutput5$MarginalSNPs, bmassOutput5$ZScoresCorMatrix, SigmaAlphas, bmassOutput5$LogFile)[c("MarginalSNPs", "Models", "ModelPriors", "LogFile")]
>         
>         bmassOutput5[c("MarginalSNPs", "PreviousSNPs", "ModelPriors", "GWASlogBFMinThreshold", "LogFile")] <- DetermineAndApplyPriors(DataSources, bmassOutput5$MarginalSNPs, GWASsnps, SigmaAlphas, bmassOutput5$Models, bmassOutput5$ModelPriors, ProvidedPriors, UseFlatPriors, bmassSeedValue, bmassOutput5$LogFile)[c("MarginalSNPs", "PreviousSNPs", "ModelPriors", "GWASlogBFMinThreshold", "LogFile")]
>         
>         bmassOutput5[c("MarginalSNPs", "PreviousSNPs", "NewSNPs", "LogFile")] <- FinalizeAndFormatResults(DataSources, bmassOutput5$MarginalSNPs, bmassOutput5$PreviousSNPs, GWASsnps, bmassOutput5$GWASlogBFMinThreshold, SigmaAlphas, bmassOutput5$Models, bmassOutput5$ModelPriors, NminThreshold, PruneMarginalSNPs, PruneMarginalSNPs_bpWindow, bmassOutput5$LogFile)[c("MarginalSNPs", "PreviousSNPs", "NewSNPs", "LogFile")]
> 
> 
> dim(bmassOutput5$NewSNPs$SNPs)
[1] 87 31
> 
> quantile(bmassOutput5$NewSNPs$SNPs$logBFWeightedAvg)
       0%       25%       50%       75%      100% 
 4.080139  4.586978  5.210187  6.742220 37.734890 
> bmassOutput5$GWASlogBFMinThreshold
[1] 4.078411
> bmassOutput1$GWASlogBFMinThreshold
[1] 4.28084
> quantile(bmassOutput1$NewSNPs$SNPs$logBFWeightedAvg)
       0%       25%       50%       75%      100% 
 4.289660  4.703371  5.427240  7.501618 33.345835 
> dim(bmassOutput1$MarginalSNPs$SNPs)
[1] 252  29
> dim(bmassOutput1$MarginalSNPs$SNPs[bmassOutput1$MarginalSNPs$SNPs$logBFWeightedAvg > 4,])
[1] 232  29
> dim(bmassOutput1$MarginalSNPs$SNPs[bmassOutput1$MarginalSNPs$SNPs$logBFWeightedAvg > 4 & bmassOutput1$MarginalSNPs$SNPs$GWASannot == 0,])
[1] 76 29
> dim(bmassOutput5$MarginalSNPs$SNPs[bmassOutput5$MarginalSNPs$SNPs$logBFWeightedAvg > 4.2 & bmassOutput5$MarginalSNPs$SNPs$GWASannot == 0,])
[1] 82 31
> bmassOutput5$NewSNPs$SNPs[bmassOutput5$NewSNPs$SNPs$logBFWeightedAvg > 20,]
                 SNP Chr       BP      SNP.1 A1 A2     MAF LDL_Beta LDL_SE
5780  chr19:50709959  19 46018119  rs7255743  g  a 0.02639   0.1647 0.0156
6449   chr6:31373469   6 31265490  rs2247056  c  t 0.21770   0.0248 0.0043
10243 chr11:47251202  11 47294626 rs10501321  t  c 0.31400   0.0087 0.0039
         LDL_N HDL_Beta HDL_SE    HDL_N TG_Beta  TG_SE     TG_N TC_Beta  TC_SE
5780  123958.8   0.0027 0.0146 137200.5  0.0323 0.0138 127634.9  0.1066 0.0150
6449  169288.0   0.0121 0.0040 183368.0  0.0378 0.0039 174062.0  0.0391 0.0041
10243 172916.0   0.0483 0.0036 186984.0  0.0216 0.0035 177680.0  0.0060 0.0038
          TC_N GWASannot Gene  HDL_ZScore LDL_ZScore TC_ZScore TG_ZScore
5780  135895.3         0   NA  -0.6782708  10.453779  7.135980 -2.624603
6449  183557.0         0   NA  -2.8949657  -5.671080 -9.252015 -9.436320
10243 187172.0         0   NA -12.9184728   2.029671 -1.512133  5.671927
        mvstat mvstat_log10pVal unistat_log10pVal       ChrBP     Nmin
5780  129.0127         26.19846          24.85140 19_46018119 123958.8
6449  147.2293         30.09762          20.41330  6_31265490 169288.0
10243 195.7057         40.50196          37.45087 11_47294626 172916.0
      logBFWeightedAvg
5780          22.98328
6449          27.95503
10243         37.73489
> bmassOutput1$NewSNPs$SNPs[bmassOutput1$NewSNPs$SNPs$logBFWeightedAvg > 20,]
              ChrBP Chr       BP     MAF A1 HDL_A2 HDL_Direction HDL_pValue
233267  11_47251202  11 47251202 0.31400  t      c             +  3.541e-38
234096  11_47857845  11 47857845 0.14780  c      t             +  2.705e-27
1026499 19_50709959  19 50709959 0.02639  g      a             +  4.976e-01
1964137  6_31373469   6 31373469 0.21770  t      c             +  3.792e-03
           HDL_N  HDL_ZScore LDL_Direction LDL_pValue    LDL_N LDL_ZScore
233267  186984.0 -12.9184728             -  4.239e-02 172916.0   2.029671
234096  178710.0 -10.8221024             +  2.521e-01 164709.0  -1.145264
1026499 137200.5  -0.6782708             -  1.408e-25 123958.8  10.453779
1964137 183368.0   2.8949657             +  1.419e-08 169288.0   5.671080
        TG_Direction TG_pValue     TG_N TG_ZScore TC_Direction TC_pValue
233267             - 1.412e-08 177680.0  5.671927            + 1.305e-01
234096             - 2.279e-02 169360.0  2.276937            + 1.080e-05
1026499            + 8.675e-03 127634.9 -2.624603            - 9.610e-13
1964137            + 3.861e-21 174062.0  9.436320            + 2.203e-20
            TC_N TC_ZScore GWASannot   mvstat mvstat_log10pVal   unistat
233267  187172.0 -1.512133         0 169.9167         34.96265 166.88694
234096  178854.3 -4.400503         0 125.5053         25.44864 117.11790
1026499 135895.3  7.135980         0 130.5222         26.52128 109.28150
1964137 183557.0  9.252015         0 160.9693         33.04297  89.04414
        unistat_log10pVal     Nmin logBFWeightedAvg
233267           37.45087 172916.0         33.34583
234096           26.56783 164709.0         23.91387
1026499          24.85140 123958.8         23.33516
1964137          20.41330 169288.0         30.99773
> bmassOutput3.2 <- bmass(c("HDL3", "LDL3", "TG3", "TC3"), GWASsnps=GWASsnps3, NminThreshold = 50000, PrintMergedData=TRUE, bmassSeedValue=34987435);
Warning message:
In ProcessMergedAndAnnotatedDataSources(DataSources, bmassOutput$MergedDataSources,  :
  2016-12-28 00:39:31 -- One of your datafiles has p-values less than the threshold which R can properly convert them to log-scale, meaning their log-values return as 'Infinite'. bmass automatically replaces these 'Infinite' values with the max, non-infinite Z-scores available, but it is recommended you self-check this as well.
> 
> dim(bmassOutput3.2$NewSNPs$SNPs)
[1] 65 29
> head(sort(bmassOutput3$ModelPriors, decreasing=TRUE))
[1] 0.28186232 0.13024238 0.12325361 0.06253280 0.04978415 0.03813073
> head(sort(bmassOutput5$ModelPriors, decreasing=TRUE))
[1] 0.23906274 0.09421731 0.08242790 0.07670760 0.07123819 0.04993153
>
>         bmassOutput4 <- list()
>         bmassOutput4$MergedDataSources <- NULL
>         if (!is.null(MergedDataSources)) {
+                 bmassOutput4$MergedDataSources <- MergedDataSources
+         }
Error: object 'MergedDataSources' not found
>         bmassOutput4$ZScoresCorMatrix <- NULL
>         bmassOutput4$MarginalSNPs <- list()
>         bmassOutput4$Models <- NULL
>         bmassOutput4$ModelPriors <- NULL
>         bmassOutput4$PreviousSNPs <- list()
>         bmassOutput4$NewSNPs <- list()
>         bmassOutput4$GWASlogBFMinThreshold <- NULL
>         #LogFile <- c()
>         bmassOutput4$LogFile <- c()
>
>         bmassOutput4$LogFile <- rbind(bmassOutput4$LogFile, paste(format(Sys.time()), " -- beginning bmass.", sep=""))
>
>         bmassOutput5 <- bmassOutput4
>         bmassOutput5.2 <- bmassOutput5
.
.
.
>         bmassOutput5.2$MergedDataSources <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/20161227_edits_GlobalLipids2013.dtlesssignif.vs1.SignCrrct.vs1.annot.MAF.txt.gz", header=TRUE)
>         bmassOutput5.2$MergedDataSources <- bmassOutput5.2$MergedDataSources[bmassOutput5.2$MergedDataSources$MAF > 0 & !is.na(bmassOutput5.2$MergedDataSources$MAF),]
>         bmassOutput5.2$ZScoresCorMatrix <- bmassOutput1$ZScoresCorMatrix
>         GWASsnps <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.ChrBP.txt", header=F);
>         names(GWASsnps) <- c("Chr", "BP")
>         #Z.LDL,Z.HDL,Z.TC,Z.TG
>         DataSources = c("HDL", "LDL", "TC", "TG"); NminThreshold = 50000; bmassSeedValue=150;
>         ExpectedColumnNames=c("Chr", "BP", "MAF", "Direction", "pValue", "N"); GWASsnps_AnnotateWindow = 5e5; SNPMarginalUnivariateThreshold = 1e-6; SNPMarginalMultivariateThreshold = 1e-6; SigmaAlphas = c(0.005,0.0075,0.01,0.015,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.15); ProvidedPriors=NULL; UseFlatPriors=FALSE; PruneMarginalSNPs=TRUE; PruneMarginalSNPs_bpWindow=5e5;
>         bmassOutput5.2$MergedDataSources$ChrBP <- paste(bmassOutput5.2$MergedDataSources$Chr, "_", bmassOutput5.2$MergedDataSources$BP, sep="")
>
>         bmassOutput5.2$MarginalSNPs$SNPs <- bmassOutput5.2$MergedDataSources[bmassOutput5.2$MergedDataSources$mvstat_log10pVal > -log10(SNPMarginalUnivariateThreshold) | bmassOutput5.2$MergedDataSources$unistat_log10pVal > -log10(SNPMarginalMultivariateThreshold),]
>
>         bmassOutput5.2[c("MarginalSNPs", "Models", "ModelPriors", "LogFile")] <- GetLogBFsFromData(DataSources, bmassOutput5.2$MarginalSNPs, bmassOutput5.2$ZScoresCorMatrix, SigmaAlphas, bmassOutput5.2$LogFile)[c("MarginalSNPs", "Models", "ModelPriors", "LogFile")]
>
>         bmassOutput5.2[c("MarginalSNPs", "PreviousSNPs", "ModelPriors", "GWASlogBFMinThreshold", "LogFile")] <- DetermineAndApplyPriors(DataSources, bmassOutput5.2$MarginalSNPs, GWASsnps, SigmaAlphas, bmassOutput5.2$Models, bmassOutput5.2$ModelPriors, ProvidedPriors, UseFlatPriors, bmassSeedValue, bmassOutput5.2$LogFile)[c("MarginalSNPs", "PreviousSNPs", "ModelPriors", "GWASlogBFMinThreshold", "LogFile")]
>
>         bmassOutput5.2[c("MarginalSNPs", "PreviousSNPs", "NewSNPs", "LogFile")] <- FinalizeAndFormatResults(DataSources, bmassOutput5.2$MarginalSNPs, bmassOutput5.2$PreviousSNPs, GWASsnps, bmassOutput5.2$GWASlogBFMinThreshold, SigmaAlphas, bmassOutput5.2$Models, bmassOutput5.2$ModelPriors, NminThreshold, PruneMarginalSNPs, PruneMarginalSNPs_bpWindow, bmassOutput5.2$LogFile)[c("MarginalSNPs", "PreviousSNPs", "NewSNPs", "LogFile")]
>
> dim(bmassOutput5.2$NewSNPs$SNPs)
[1] 84 31
> write.table(bmassOutput3$PreviousSNPs$SNPs, file="nanana3.txt", quote=FALSE, row.names=FALSE)
> write.table(bmassOutput5.2$PreviousSNPs$SNPs, file="nanana5.2.txt", quote=FALSE, row.names=FALSE)
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/nanana3.txt | perl -lane 'print $F[0], "\t", $F[9], "\t", $F[13], "\t", $F[17], "\t", $F[21], "\t", join("\t", @F[22..26]), "\t", $F[28]; ' | sort -k 1,1 | head -n 10
10_113923876    8.47494035962306        7.40464678868388        -6.02377044685236       8.07929924807389        1       162.407222586343        33.3513893982575        71.8246140991678        16.6285626825959        30.4230894747641
10_17300296     3.64750418481346        5.11514502107085        2.68495302878402        6.66755500152202        1       57.8527371818585        11.0865082398778        44.4562896983213        10.5848596478041        9.22681814814558
10_45333283     6.38571362584799        3.40373996191713        0.739835195937606       5.75960700206589        1       66.9017971256983        12.9903401067078        40.7773385113407        9.76802097316849        11.5562865653172
10_5244847      -1.58002998406243       -1.6159736979577        -7.05564655784335       -4.568213009728 1       73.0575763928857        14.2898866030454        49.7821483492066        11.7649768405048        12.4930006165342
10_94829632     4.7387882766992 -4.11421453385679       -6.73126008177982       -3.76105236579225       1       60.4465330338601        11.6313195372535        45.3098622885624        10.7741740085381        9.84491786890623
11_10345358     -5.45060058651368       2.35807461854458        2.74931480194631        0.00626661170175034     1       37.2682722275453        6.79969069182732        29.7090467537033        7.29929628285498        4.80578085033094
11_117991277    3.05361771166883        4.33194182314675        0.569377474575075       5.7068294621188 1       37.4902470996644        6.84544386472414        32.5679025097072        7.93892467637021        5.62681215613522
11_122027585    7.72483309131196        4.70513340698104        -0.986271298702238      7.62747295996659        1       102.58526432914 20.5576666086493        59.6730462886284        13.9507819773298        19.3029965699057
11_125749162    -2.62167043158378       9.37981274181744        3.21268593453459        7.81521337484289        1       92.9016234537775        18.4970875667416        87.9808870715607        20.1798642481296        16.820865295778
11_18589560     -1.1947109058546        -7.14189934068621       -2.29244461575788       -6.79879200526399       1       58.1681080551622        11.1527078156039        51.0067261924942        12.03597620716  9.37021161205544
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/nanana5.2.txt | perl -lane 'print $F[28], "\t", $F[19], "\t", join("\t", @F[21..27]), "\t", $F[30];' | sort -k 1,1 | head -n 10
10_113933886    1       8.47494035962306        7.40464678868388        8.07929924807389        -6.02377044685236       180.500977544004        37.2350540633722        16.6285626825959        346.86843701254
10_17260290     1       3.64750418481346        5.11514502107085        6.66755500152202        2.68495302878402        55.4858725775983        10.5900730097836        10.5848596478041        38.5412265937624
10_46013277     1       6.38571362584799        3.40373996191713        5.75960700206589        0.739835195937606       68.7674948655753        13.3838679508119        9.76802097316849        55.5550837078421
10_5254847      1       -1.58002998406243       -1.6159736979577        -4.568213009728 -7.05564655784335       66.2079133047116        12.8440609356507        11.7649768405048        19.4444693208997
10_94839642     1       -4.7387882766992        4.11421453385679        3.76105236579225        6.73126008177982        66.5673303570534        12.9198248795096        10.7741740085381        19.5301684395352
11_10388782     1       5.45060058651368        -2.35807461854458       -0.00626661170175034    -2.74931480194631       46.721487072851 8.75875147136635        7.29929628285498        11.6969505021368
11_118486067    1       3.05361771166883        4.33194182314675        5.7068294621188 0.569377474575075       39.7092386839387        7.30354935340201        7.93892467637021        43.2152554732928
11_122522375    1       7.72483309131196        4.70513340698104        7.62747295996659        -0.986271298702238      115.855453436528        23.3873723754993        13.9507819773298        129.4346518154
11_126243952    1       2.62167043158378        -9.37981274181744       -7.81521337484289       -3.21268593453459       94.6719391878239        18.8734799554428        20.1798642481296        55.6299080973941
11_18632984     1       -1.1947109058546        -7.14189934068621       -6.79879200526399       -2.29244461575788       56.3664410241416        10.7746839339184        12.03597620716  51.2044261282904
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/nanana3.txt | perl -lane 'print $F[0], "\t", $F[9], "\t", $F[13], "\t", $F[17], "\t", $F[21], "\t", join("\t", @F[22..26]), "\t", $F[28]; ' | head -n 1 
ChrBP   HDL3_ZScore     LDL3_ZScore     TG3_ZScore      TC3_ZScore      GWASannot       mvstat  mvstat_log10pVal        unistat unistat_log10pVal       logBFWeightedAvg
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/nanana5.2.txt | perl -lane 'print $F[28], "\t", $F[19], "\t", join("\t", @F[21..27]), "\t", $F[30];' | head -n 1 
ChrBP   GWASannot       HDL_ZScore      LDL_ZScore      TC_ZScore       TG_ZScore       mvstat  mvstat_log10pVal        unistat_log10pVal       logBFWeightedAvg
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/20161227_edits_GlobalLipids2013.dtlesssignif.vs1.SignCrrct.vs1.annot.MAF.txt.gz | sed 's/:/ /g' | awk '{ print $2 "\t" $4 }' | R -q -e "Data1 <- read.table(file('stdin'), header=T); quantile(Data1[,1] - Data1[,2]);"
> Data1 <- read.table(file('stdin'), header=T); quantile(Data1[,1] - Data1[,2]);
      0%      25%      50%      75%     100% 
-2975874  -566586  -146496    35757  4691840 
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/20161227_edits_GlobalLipids2013.dtlesssignif.vs1.SignCrrct.vs1.annot.MAF.Hg18vsHg19GWASProxTest.gz | awk '{ print $6 }' | sort | uniq -c
   1783 0
    148 1
   9642 2
      1 Gene
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/20161227_edits_GlobalLipids2013.dtlesssignif.vs1.SignCrrct.vs1.annot.MAF.Hg18vsHg19GWASProxTest.gz | awk '{ print $7 }' | sort | uniq -c
   1784 0
    146 1
   9644 2
> write.table(bmassOutput3.4$MarginalSNPs$SNPs, "20161228.bmassOutput3.4.NotPruned.MaringalSNPs.txt", quote=FALSE, row.names=FALSE)
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/20161228.bmassOutput3.4.NotPruned.MaringalSNPs.txt | awk '{ print $1 "\t" $10 }' | sort -k 1,1) <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/20161227_edits_GlobalLipids2013.dtlesssignif.vs1.SignCrrct.vs1.annot.MAF.txt.gz | sed 's/chr//g' | sed 's/:/_/g ' | sort -k 1,1) | wc
  10888  315752 3071885
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/20161228.bmassOutput3.4.NotPruned.MaringalSNPs.txt | awk '{ print $1 "\t" $10 "\t" $14 "\t" $18 "\t" $22 }' | sort -k 1,1) <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/20161227_edits_GlobalLipids2013.dtlesssignif.vs1.SignCrrct.vs1.annot.MAF.txt.gz | sed 's/chr//g' | sed 's/:/_/g ' | awk '{ print $1 "\t" $22 "\t" $23 "\t" $24 "\t" $25 }' | sort -k 1,1) | awk '{ if ($2 != $6) { print $0 } }' | wc  
   5401   48609  824008
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/20161228.bmassOutput3.4.NotPruned.MaringalSNPs.txt | awk '{ print $1 "\t" $10 }' | sort -k 1,1) <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/20161227_edits_GlobalLipids2013.dtlesssignif.vs1.SignCrrct.vs1.annot.MAF.txt.gz | sed 's/chr//g' | sed 's/:/_/g ' | sort -k 1,1) | perl -lane 'if ($F[1] != $F[22]) { print "yaya2"; }' | sort | uniq -c
   5401 yaya2
.
.
.

>         bmassOutput5.4$MergedDataSources <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/20161227_edits_GlobalLipids2013.dtlesssignif.vs1.SignCrrct.vs1.annot.MAF.bmassOutput3.4_ZScoreSignMatch.txt.gz", header=TRUE)
old = 1e-6; SNPMarginalMultivariateThreshold = 1e-6; SigmaAlphas = c(0.005,0.0075,0.01,0.015,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.15); ProvidedPriors=NULL; UseFlatPriors=FALSE; PruneMarginalSNPs=TRUE; PruneMarginalSNPs_bpWindow=5e5;
        bmassOutput5.4$MergedDataSources$ChrBP <- paste(bmassOutput5.4$MergedDataSources$Chr, "_", bmassOutput5.4$MergedDataSources$BP, sep="")

        bmassOutput5.4$MarginalSNPs$SNPs <- bmassOutput5.4$MergedDataSources[bmassOutput5.4$MergedDataSources$mvstat_log10pVal > -log10(SNPMarginalUnivariateThreshold) | bmassOutput5.4$MergedDataSources$unistat_log10pVal > -log10(SNPMarginalMultivariateThreshold),]

        bmassOutput5.4[c("MarginalSNPs", "Models", "ModelPriors", "LogFile")] <- GetLogBFsFromData(DataSources, bmassOutput5.4$MarginalSNPs, bmassOutput5.4$ZScoresCorMatrix, SigmaAlphas, bmassOutput5.4$LogFile)[c("MarginalSNPs", "Models", "ModelPriors", "LogFile")]

        bmassOutput5.4[c("MarginalSNPs", "PreviousSNPs", "ModelPriors", "GWASlogBFMinThreshold", "LogFile")] <- DetermineAndApplyPriors(DataSources, bmassOutput5.4$MarginalSNPs, GWASsnps, SigmaAlphas, bmassOutput5.4$Models, bmassOutput5.4$ModelPriors, ProvidedPriors, UseFlatPriors, bmassSeedValue, bmassOutput5.4$LogFile)[c("MarginalSNPs", "PreviousSNPs", "ModelPriors", "GWASlogBFMinThreshold", "LogFile")]

        bmassOutput5.4[c("MarginalSNPs", "PreviousSNPs", "NewSNPs", "LogFile")] <- FinalizeAndFormatResults(DataSources, bmassOutput5.4$MarginalSNPs, bmassOutput5.4$PreviousSNPs, GWASsnps, bmassOutput5.4$GWASlogBFMinThreshold, SigmaAlphas, bmassOutput5.4$Models, bmassOutput5.4$ModelPriors, NminThreshold, PruneMarginalSNPs, PruneMarginalSNPs_bpWindow, bmassOutput5.4$LogFile)[c("MarginalSNPs", "PreviousSNPs", "NewSNPs", "LogFile")]


>         bmassOutput5.4$MergedDataSources <- bmassOutput5.4$MergedDataSources[bmassOutput5.4$MergedDataSources$MAF > 0 & !is.na(bmassOutput5.4$MergedDataSources$MAF),]
>         bmassOutput5.4$ZScoresCorMatrix <- bmassOutput3$ZScoresCorMatrix
>         GWASsnps <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.ChrBP.txt", header=F);
>         names(GWASsnps) <- c("Chr", "BP")
>         #Z.LDL,Z.HDL,Z.TC,Z.TG
>         DataSources = c("HDL", "LDL", "TC", "TG"); NminThreshold = 50000; bmassSeedValue=150;
>         ExpectedColumnNames=c("Chr", "BP", "MAF", "Direction", "pValue", "N"); GWASsnps_AnnotateWindow = 5e5; SNPMarginalUnivariateThreshold = 1e-6; SNPMarginalMultivariateThreshold = 1e-6; SigmaAlphas = c(0.005,0.0075,0.01,0.015,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.15); ProvidedPriors=NULL; UseFlatPriors=FALSE; PruneMarginalSNPs=TRUE; PruneMarginalSNPs_bpWindow=5e5;
>         bmassOutput5.4$MergedDataSources$ChrBP <- paste(bmassOutput5.4$MergedDataSources$Chr, "_", bmassOutput5.4$MergedDataSources$BP, sep="")
>
>         bmassOutput5.4$MarginalSNPs$SNPs <- bmassOutput5.4$MergedDataSources[bmassOutput5.4$MergedDataSources$mvstat_log10pVal > -log10(SNPMarginalUnivariateThreshold) | bmassOutput5.4$MergedDataSources$unistat_log10pVal > -log10(SNPMarginalMultivariateThreshold),]
>
>         bmassOutput5.4[c("MarginalSNPs", "Models", "ModelPriors", "LogFile")] <- GetLogBFsFromData(DataSources, bmassOutput5.4$MarginalSNPs, bmassOutput5.4$ZScoresCorMatrix, SigmaAlphas, bmassOutput5.4$LogFile)[c("MarginalSNPs", "Models", "ModelPriors", "LogFile")]
>
>         bmassOutput5.4[c("MarginalSNPs", "PreviousSNPs", "ModelPriors", "GWASlogBFMinThreshold", "LogFile")] <- DetermineAndApplyPriors(DataSources, bmassOutput5.4$MarginalSNPs, GWASsnps, SigmaAlphas, bmassOutput5.4$Models, bmassOutput5.4$ModelPriors, ProvidedPriors, UseFlatPriors, bmassSeedValue, bmassOutput5.4$LogFile)[c("MarginalSNPs", "PreviousSNPs", "ModelPriors", "GWASlogBFMinThreshold", "LogFile")]
> 
>         bmassOutput5.4[c("MarginalSNPs", "PreviousSNPs", "NewSNPs", "LogFile")] <- FinalizeAndFormatResults(DataSources, bmassOutput5.4$MarginalSNPs, bmassOutput5.4$PreviousSNPs, GWASsnps, bmassOutput5.4$GWASlogBFMinThreshold, SigmaAlphas, bmassOutput5.4$Models, bmassOutput5.4$ModelPriors, NminThreshold, PruneMarginalSNPs, PruneMarginalSNPs_bpWindow, bmassOutput5.4$LogFile)[c("MarginalSNPs", "PreviousSNPs", "NewSNPs", "LogFile")]
> dim(bmassOutput5.4$NewSNPs$SNPs)
[1] 72 31
.
.
.
#20170119
> dim(bmassOutput1$MergedDataSources)
[1] 2063845      23
> dim(bmassOutput1$NewSNPs$SNPs)
[1] 65 29
> dim(bmassOutput3$MergedDataSources)
[1] 2051822      23
> dim(bmassOutput3$NewSNPs$SNPs)
[1] 65 29
> bmassOutput1$GWASlogBFMinThreshold
[1] 4.28084
> bmassOutput3$GWASlogBFMinThreshold
[1] 4.281287
> bmassOutput3.2$GWASlogBFMinThreshold
[1] 4.281287
>
> write.table(bmassOutput1$NewSNPs$SNPs, file="/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.txt", quote=FALSE, row.names=FALSE)
> write.table(bmassOutput1$MergedDataSources, file="/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/bmass.GlobalLipids2013.Vs2.MergedDataSources.vs1.txt", quote=FALSE, row.names=FALSE)
>
> system("gzip /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/bmass.GlobalLipids2013.Vs2.MergedDataSources.vs1.txt")
>
#20170223
> bmassOutput1$ZScoresCorMatrix
            HDL_ZScore  LDL_ZScore  TG_ZScore TC_ZScore
HDL_ZScore  1.00000000 -0.08715854 -0.3635593 0.1531368
LDL_ZScore -0.08715854  1.00000000  0.1585048 0.8197523
TG_ZScore  -0.36355928  0.15850485  1.0000000 0.2877814
TC_ZScore   0.15313683  0.81975235  0.2877814 1.0000000
> dim(bmassOutput1$PreviousSNPs$SNPs)
[1] 148  29
> dim(GWASsnps)
[1] 157   2
#From /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/process.MTedits.ForGlobalLipids2013.vs1.SignCrrct.vs1.R
#> RSS0
#            Z.LDL       Z.HDL       Z.TC       Z.TG
#Z.LDL  1.00000000 -0.05101127 0.83617727  0.1446093
#Z.HDL -0.05101127  1.00000000 0.09840344 -0.2298445
#Z.TC   0.83617727  0.09840344 1.00000000  0.3389824
#Z.TG   0.14460929 -0.22984452 0.33898241  1.0000000
#From /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GLC.MTedits.For2013.vs3.SignCrrct.vs1.ForCSHLPoster.R
#min(lbf.av.glhits)
##[1] 4.608818
##[1] 7.312795
#[1] 3.989499
#84 in new results
#74 in new results
#20170125 NOTE -- rechecking after creating and using forceMAF files
> bmassOutput1$GWASlogBFMinThreshold
[1] 4.078202
> dim(bmassOutput1$NewSNPs$SNPs)
[1] 87 29
> dim(bmassOutput1$MergedDataSources)
[1] 2063842      23
> dim(GWASsnps)
[1] 157   2
> dim(bmassOutput1$PreviousSNPs$SNPs)
[1] 148  29
> bmassOutput1$ZScoresCorMatrix
            HDL_ZScore  LDL_ZScore  TG_ZScore  TC_ZScore
HDL_ZScore  1.00000000 -0.05099993 -0.2301535 0.09830688
LDL_ZScore -0.05099993  1.00000000  0.1445420 0.83613234
TG_ZScore  -0.23015345  0.14454195  1.0000000 0.33892587
TC_ZScore   0.09830688  0.83613234  0.3389259 1.00000000
> library("devtools"); devtools::load_all("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/bmass");
Loading bmass
> HDL <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_HDL.IncAllele.bmass.formatted.txt.gz", header=T);
> LDL <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_LDL.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.txt.gz", header=T);
> TG <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TG.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.txt.gz", header=T);
> TC <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TC.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.txt.gz", header=T);
> GWASsnps <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.ChrBP.txt", header=F);
> names(GWASsnps) <- c("Chr", "BP")
> #bmassOutput1 <- bmass(c("HDL", "LDL", "TG", "TC"), GWASsnps=GWASsnps, NminThreshold = 1, bmassSeedValue=150);
> #print(bmassOutput1);" > shana1.txt
> DataSources = c("HDL", "LDL", "TG", "TC"); NminThreshold = 50000; bmassSeedValue=150;
> ExpectedColumnNames=c("Chr", "BP", "MAF", "Direction", "pValue", "N"); GWASsnps_AnnotateWindow = 5e5; SNPMarginalUnivariateThreshold = 1e-6; SNPMarginalMultivariateThreshold = 1e-6; SigmaAlphas = c(0.005,0.0075,0.01,0.015,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.15); ProvidedPriors=NULL; UseFlatPriors=FALSE; PruneMarginalSNPs=TRUE; PruneMarginalSNPs_bpWindow=5e5;
>
>
> bmassOutput1 <- bmass(c("HDL", "LDL", "TG", "TC"), GWASsnps=GWASsnps, NminThreshold = 50000, PrintMergedData=TRUE, bmassSeedValue=150);
Warning message:
In ProcessMergedAndAnnotatedDataSources(DataSources, bmassOutput$MergedDataSources,  :
  2017-01-25 23:46:30 -- One of your datafiles has p-values less than the threshold which R can properly convert them to log-scale, meaning their log-values return as 'Infinite'. bmass automatically replaces these 'Infinite' values with the max, non-infinite Z-scores available, but it is recommended you self-check this as well.
> dim(bmassOutput1$NewSNPs$SNPs)
[1] 65 29
> HDL[HDL$MAF > .5,]$MAF <- 1 - HDL[HDL$MAF > .5,]$MAF
> LDL[LDL$MAF > .5,]$MAF <- 1 - LDL[LDL$MAF > .5,]$MAF
> TG[TG$MAF > .5,]$MAF <- 1 - TG[TG$MAF > .5,]$MAF
> TC[TC$MAF > .5,]$MAF <- 1 - TC[TC$MAF > .5,]$MAF
> bmassOutput1 <- bmass(c("HDL", "LDL", "TG", "TC"), GWASsnps=GWASsnps, NminThreshold = 50000, PrintMergedData=TRUE, bmassSeedValue=150);
Warning message:
In ProcessMergedAndAnnotatedDataSources(DataSources, bmassOutput$MergedDataSources,  :
  2017-01-25 23:55:21 -- One of your datafiles has p-values less than the threshold which R can properly convert them to log-scale, meaning their log-values return as 'Infinite'. bmass automatically replaces these 'Infinite' values with the max, non-infinite Z-scores available, but it is recommended you self-check this as well.
> dim(bmassOutput1$NewSNPs$SNPs)
[1] 87 29
> bmassOutput1$ZScoresCorMatrix
            HDL_ZScore  LDL_ZScore  TG_ZScore  TC_ZScore
HDL_ZScore  1.00000000 -0.05099922 -0.2301516 0.09830801
LDL_ZScore -0.05099922  1.00000000  0.1445408 0.83613207
TG_ZScore  -0.23015158  0.14454084  1.0000000 0.33892580
TC_ZScore   0.09830801  0.83613207  0.3389258 1.00000000
#20170202
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /data/external_public/GlobalLipid/jointGwasMc_HDL.txt.gz | awk '{ print $1 }' | sort | wc
2447442 2447442 35840951
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /data/external_public/GlobalLipid/jointGwasMc_HDL.txt.gz | awk '{ print $1 }' | sort | uniq -u | wc
2447442 2447442 35840951
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /data/external_public/GlobalLipid/jointGwasMc_HDL.txt.gz | grep NA | wc
 186081 1860810 14234173
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /data/external_public/GlobalLipid/jointGwasMc_HDL.txt.gz | grep -v NA | wc
2261361 22613610 181415272
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /data/external_public/GlobalLipid/jointGwasMc_HDL.txt.gz | grep NA | head -n 10
chr1:178149073  chr1:179882450  rs524166        a       t  0.0104       0.0116  94311.00        0.5041  NA
chr16:8922306   chr16:9014805 rs7199803 a       t       0.0074  0.0049  92820.00        0.0835  NA
chr18:10420507  chr18:10430507  rs206414        c       g  0.0031       0.0054  92820.00        0.29    NA
chr21:36264150  chr21:37342280  rs8128314       a       t  0.0031       0.0060  92751.00        0.8686  NA
chr3:24917609   chr3:24942605 rs2068192 a       t       0.0028  0.0062  94171.00        0.5245  NA
chr8:61295862   chr8:61133308 rs10103858        a       t  0.0016       0.0048  94311.00        0.922   NA
chr1:228661432  chr1:230594809  rs11585829      c       g  0.0027       0.0049  94298.00        0.6007  NA
chr10:119197901 chr10:119207911 rs7095181       c       g  0.0273       0.0175  94311.00        0.6289  NA
chr4:176891997  chr4:176655003  rs13136033      a       t  0.0045       0.0054  94311.00        0.5108  NA
chr3:112530769  chr3:111048079  rs2971363       a       t  0.0105       0.0186  88360.00        0.2342  NA
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /data/external_public/GlobalLipid/jointGwasMc_HDL.txt.gz | grep , | head -n 10
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /data/external_public/GlobalLipid/jointGwasMc_HDL.txt.gz | sed 's/:/ /g' | sed 's/chr//g' | awk '{ print $1 "\t" $2 "\t" $12 "\t" $6 "\t" $7 "\t" $8 "\t" $11 "\t" $10 }' | grep -v hg | grep rs | head -n 10
rs2390477   7      g    0.0022  0.0103  NA      0.987
rs2806791   13     c    0.0050  0.0059  NA      0.2625
rs2516473   6      g    0.0203  0.0154  NA      0.09822
rs2067426   2      c    0.0038  0.0246  NA      0.783
rs13598 15      a  0.0058  0.0089       1       0.5364
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /data/external_public/GlobalLipid/jointGwasMc_HDL.txt.gz | sed 's/:/ /g' | sed 's/chr//g' | awk '{ print $1 "\t" $2 "\t" $12 "\t" $6 "\t" $7 "\t" $8 "\t" $11 "\t" $10 }' | grep -v hg | grep rs | wc        
      5      35     193
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$join <(zcat /data/external_public/GlobalLipid/jointGwasMc_HDL.txt.gz | sed 's/:/_/g' | sed 's/chr//g' | awk '{ print $1 }' | grep -v hg18 | sort | uniq -u) <(zcat /data/external_public/GlobalLipid/jointGwasMc_HDL.txt.gz | sed 's/:/ /g' | sed 's/chr//g' | awk '{ print $1 "\t" $2 "\t" $12 "\t" $6 "\t" $7 "\t" $8 "\t" $11 "\t" $10 }' | grep -v hg | grep -v rs | grep -v NA | perl -lane 'if ($F[5] > 0) { $F[5] = "+"; } elsif ($F[5] < 0) { $F[5] = "-"; } else { $F[5] = 0;} if ($F[2] > .5) { ($F[3], $F[4]) = ($F[4], $F[3]); if ($F[5] eq "+") { $F[5] = "-"; } elsif ($F[5] eq "-") { $F[5] = "+"; } else { $PH = 1; } $F[2] = 1 - $F[2]; } print $F[0], "_", $F[1], "\t", join("\t", @F);' | sort -k 1,1) | perl -lane 'splice(@F, 0, 1); print join("\t", @F);' | wc        
2261359 18090872 91826331
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$join <(zcat /data/external_public/GlobalLipid/jointGwasMc_HDL.txt.gz | sed 's/:/_/g' | sed 's/chr//g' | awk '{ print $1 }' | grep -v hg18 | sort | uniq -u) <(zcat /data/external_public/GlobalLipid/jointGwasMc_HDL.txt.gz | sed 's/:/ /g' | sed 's/chr//g' | awk '{ print $1 "\t" $2 "\t" $12 "\t" $6 "\t" $7 "\t" $8 "\t" $11 "\t" $10 }' | grep -v hg | grep -v rs | grep -v NA | perl -lane 'if ($F[5] > 0) { $F[5] = "+"; } elsif ($F[5] < 0) { $F[5] = "-"; } else { $F[5] = 0;} if ($F[2] > .5) { ($F[3], $F[4]) = ($F[4], $F[3]); if ($F[5] eq "+") { $F[5] = "-"; } elsif ($F[5] eq "-") { $F[5] = "+"; } else { $PH = 1; } $F[2] = 1 - $F[2]; } print $F[0], "_", $F[1], "\t", join("\t", @F);' | sort -k 1,1) | perl -lane 'splice(@F, 0, 1); print join("\t", @F);' | R -q -e "Data1 <- read.table(file('stdin'), header=F); quantile(Data1[,3]);"
> Data1 <- read.table(file('stdin'), header=F); quantile(Data1[,3]);
    0%    25%    50%    75%   100%
0.0000 0.0752 0.1939 0.3377 0.5000
>
>
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /data/external_public/GlobalLipid/jointGwasMc_HDL.txt.gz | sed 's/:/ /g' | sed 's/chr//g' | awk '{ print $1 "\t" $2 "\t" $12 "\t" $6 "\t" $7 "\t" $8 "\t" $11 "\t" $10 }' | grep -v hg | grep -v rs | grep -v NA | perl -lane 'if ($F[5] > 0) { $F[5] = "+"; } elsif ($F[5] < 0) { $F[5] = "-"; } else { $F[5] = 0;} print join("\t", @F);' | awk '{ print $6 }' | sort | uniq -c
2247211 +
  14148 0
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /data/external_public/GlobalLipid/jointGwasMc_HDL.txt.gz | sed 's/:/ /g' | sed 's/chr//g' | awk '{ print $1 "\t" $2 "\t" $12 "\t" $6 "\t" $7 "\t" $8 "\t" $11 "\t" $10 }' | grep -v hg | grep -v rs | grep -v NA | perl -lane 'if ($F[5] > 0) { $F[5] = "+"; } elsif ($F[5] < 0) { $F[5] = "-"; } else { $F[5] = 0;} if ($F[2] > .5) { ($F[3], $F[4]) = ($F[4], $F[3]); if ($F[5] eq "+") { $F[5] = "-"; } elsif ($F[5] eq "-") { $F[5] = "+"; } else { $PH = 1; } $F[2] = 1 - $F[2]; } print join("\t", @F);' | awk '{ print $6 }' | sort | uniq -c
1038583 +
1208628 -
  14148 0
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /data/external_public/GlobalLipid/jointGwasMc_HDL.txt.gz | sed 's/:/ /g' | sed 's/chr//g' | awk '{ print $1 "\t" $2 "\t" $12 "\t" $6 "\t" $7 "\t" $8 "\t" $11 "\t" $10 }' | grep -v hg | grep -v rs | grep -v NA | perl -lane 'if ($F[5] > 0) { $F[5] = "+"; } elsif ($F[5] < 0) { $F[5] = "-"; } else { $F[5] = 0;} print join("\t", @F);' | R -q -e "Data1 <- read.table(file('stdin'), header=F); table(cut(Data1[,3], c(0,.5,1)));"
> Data1 <- read.table(file('stdin'), header=F); table(cut(Data1[,3], c(0,.5,1)));

(0,0.5] (0.5,1]
1044027 1217196
>
>
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /data/external_public/GlobalLipid/jointGwasMc_HDL.txt.gz | sed 's/:/ /g' | sed 's/chr//g' | awk '{ print $1 "\t" $2 "\t" $12 "\t" $6 "\t" $7 "\t" $8 "\t" $11 "\t" $10 }' | grep -v hg | grep -v rs | grep -v NA | perl -lane 'if ($F[5] > 0) { $F[5] = "+"; } elsif ($F[5] < 0) { $F[5] = "-"; } else { $F[5] = 0;} print join("\t", @F);' | awk '{ if ($3 == .5) { print $0 } } ' | wc
   2275   18200   85834
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /data/external_public/GlobalLipid/jointGwasMc_HDL.txt.gz | sed 's/:/ /g' | sed 's/chr//g' | awk '{ print $1 "\t" $2 "\t" $12 "\t" $6 "\t" $7 "\t" $8 "\t" $11 "\t" $10 }' | grep -v hg | grep -v rs | grep -v NA | perl -lane 'if ($F[5] > 0) { $F[5] = "+"; } elsif ($F[5] < 0) { $F[5] = "-"; } else { $F[5] = 0;} print join("\t", @F);' | awk '{ if (($3 > .5) && ($6 == 0)) { print $0 } } ' | wc
   8568   68544  337187
> dim(bmassOutput1$NewSNPs$SNPs)
[1] 64 29
> bmassOutput1$GWASlogBFMinThreshold
[1] 4.285889
> bmassOutput1$ZScoresCorMatrix     
            HDL_ZScore  LDL_ZScore  TG_ZScore TC_ZScore
HDL_ZScore  1.00000000 -0.08725994 -0.3655873 0.1523855
LDL_ZScore -0.08725994  1.00000000  0.1607749 0.8220478
TG_ZScore  -0.36558729  0.16077491  1.0000000 0.2892950
TC_ZScore   0.15238551  0.82204776  0.2892950 1.0000000
> dim(bmassOutput3$MergedDataSources)
[1] 2017038      23
> dim(bmassOutput1$PreviousSNPs$SNPs)
[1] 147  29
> dim(bmassOutput1$MarginalSNPs$SNPs[bmassOutput1$MarginalSNPs$SNPs$logBFWeightedAvg>4.02 & bmassOutput1$MarginalSNPs$SNPs$GWASannot == 0,])
[1] 74 29
> bmassOutput3$MarginalSNPs$SNPs[bmassOutput3$MarginalSNPs$SNPs$ChrBP == "16_55542640",]
             ChrBP Chr       BP   MAF A1 HDL_A2 HDL_Direction HDL_pValue  HDL_N
670509 16_55542640  16 55542640 0.405  A      G             -          0 177532
       HDL_ZScore LDL_Direction LDL_pValue  LDL_N LDL_ZScore TG_Direction
670509   -38.1534             +  2.494e-12 163625   7.003638            +
       TG_pValue   TG_N TG_ZScore TC_Direction TC_pValue   TC_N TC_ZScore
670509 1.052e-11 168198  6.799202            - 1.026e-15 177683 -8.023708
       GWASannot   mvstat mvstat_log10pVal  unistat unistat_log10pVal   Nmin
670509         2 1586.397         341.5818 1455.682          317.7773 163625
       logBFWeightedAvg
670509         338.8718
> bmassOutput3$MergedDataSources[bmassOutput3$MergedDataSources$ChrBP == "16_55542640",]
             ChrBP Chr       BP   MAF A1 HDL_A2 HDL_Direction HDL_pValue  HDL_N
670509 16_55542640  16 55542640 0.405  A      G             -          0 177532
       HDL_ZScore LDL_Direction LDL_pValue  LDL_N LDL_ZScore TG_Direction
670509       -Inf             +  2.494e-12 163625   7.003638            +
       TG_pValue   TG_N TG_ZScore TC_Direction TC_pValue   TC_N TC_ZScore
670509 1.052e-11 168198  6.799202            - 1.026e-15 177683 -8.023708
       GWASannot
670509         2
> dim(bmassOutput3$MergedDataSources)
[1] 2017038      23
> dim(bmassOutput3.wInc.force$MergedDataSources)
[1] 2063842      23
> quantile(bmassOutput3$MergedDataSources$MAF)
      0%      25%      50%      75%     100% 
0.001319 0.102900 0.216400 0.350900 0.500000 
> table(bmassOutput3$MergedDataSources$HDL_Direction)

      -       + 
1000335 1016703 
> table(bmassOutput3$MergedDataSources$LDL_Direction)

      -       + 
1008677 1008361 
> table(bmassOutput3$MergedDataSources$TG_Direction)

      -       + 
1012772 1004266 
> table(bmassOutput3$MergedDataSources$TC_Direction)

      -       + 
1005288 1011750 
> quantile(bmassOutput3.wInc.force$MergedDataSources$MAF)
      0%      25%      50%      75%     100% 
0.001319 0.104200 0.217700 0.352200 0.500000 
> table(bmassOutput3.wInc.force$MergedDataSources$HDL_Direction)

      + 
2063842 
> table(bmassOutput3.wInc.force$MergedDataSources$LDL_Direction)

      -       + 
1091907  971935 
> table(bmassOutput3.wInc.force$MergedDataSources$TG_Direction)

      -       + 
1313565  750277 
> table(bmassOutput3.wInc.force$MergedDataSources$TC_Direction)

      -       + 
 944883 1118959 
> quantile(bmassOutput3$MergedDataSources$HDL_ZScore)
         0%         25%         50%         75%        100% 
       -Inf -0.67622154  0.01046536  0.69476913         Inf 
> quantile(bmassOutput3$MergedDataSources$LDL_ZScore)
           0%           25%           50%           75%          100% 
         -Inf -0.6864966121 -0.0002506628  0.6890369744 31.2583360822 
> quantile(bmassOutput3$MergedDataSources$TG_ZScore)
           0%           25%           50%           75%          100% 
-30.132477619  -0.691382818  -0.005389277   0.675591563  33.749155699 
> quantile(bmassOutput3$MergedDataSources$TC_ZScore)
           0%           25%           50%           75%          100% 
-35.982698483  -0.682219763   0.004135948   0.693333953  26.762094800 
> quantile(bmassOutput3.wInc.force$MergedDataSources$HDL_ZScore)
       0%       25%       50%       75%      100% 
0.0000000 0.3215416 0.6809549 1.1692573       Inf 
> quantile(bmassOutput3.wInc.force$MergedDataSources$LDL_ZScore)
         0%         25%         50%         75%        100% 
       -Inf -0.75191862 -0.07288202  0.60481790 18.67579573 
> quantile(bmassOutput3.wInc.force$MergedDataSources$TG_ZScore)
         0%         25%         50%         75%        100% 
-33.7491557  -0.9725038  -0.3296029   0.3088968  19.2028547 
> quantile(bmassOutput3.wInc.force$MergedDataSources$TC_ZScore)
         0%         25%         50%         75%        100% 
-35.9826985  -0.5735092   0.1038354   0.7803243  19.1311404 
> bmassOutput1$ZScoresCorMatrix
            HDL_ZScore  LDL_ZScore  TG_ZScore TC_ZScore
HDL_ZScore  1.00000000 -0.08725994 -0.3655873 0.1523855
LDL_ZScore -0.08725994  1.00000000  0.1607749 0.8220478
TG_ZScore  -0.36558729  0.16077491  1.0000000 0.2892950
TC_ZScore   0.15238551  0.82204776  0.2892950 1.0000000
> bmassOutput1.wInc.force$ZScoresCorMatrix
            HDL_ZScore  LDL_ZScore  TG_ZScore  TC_ZScore
HDL_ZScore  1.00000000 -0.05099993 -0.2301535 0.09830688
LDL_ZScore -0.05099993  1.00000000  0.1445420 0.83613234
TG_ZScore  -0.23015345  0.14454195  1.0000000 0.33892587
TC_ZScore   0.09830688  0.83613234  0.3389259 1.00000000
> dim(bmassOutput1.wInc.force$PreviousSNPs$SNPs)
[1] 148  29
> dim(bmassOutput1$PreviousSNPs$SNPs)
[1] 147  29
[  mturchin20@spudling89  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2]$zcat jointGwasMc_HDL.IncAllele.bmass.formatted.txt.gz | awk '{ print $6 }' | sort | uniq -c
2447436 +
      1 Direction
[  mturchin20@spudling89  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2]$zcat jointGwasMc_HDL.bmass.formatted.txt.gz | awk '{ print $6 }' | sort | uniq -c
2433288 +
  14148 0
      1 Direction
[  mturchin20@spudling89  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2]$zcat jointGwasMc_LDL.bmass.formatted.txt.gz | awk '{ print $6 }' | sort | uniq -c
2424368 +
  13380 0
      1 Direction
[  mturchin20@spudling89  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2]$zcat jointGwasMc_LDL.IncAllele.bmass.formatted.txt.gz | awk '{ print $6 }' | sort | uniq -c
2437748 +
      1 Direction
[  mturchin20@spudling89  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2]$join -v 2 <(zcat bmass.GlobalLipids2013.Vs2.MergedDataSources.vs1.txt.gz | sort -g -k 1,1) <(zcat OLD1.WRONG1.Inc.force.bmass.GlobalLipids2013.Vs2.MergedDataSources.vs1.txt.gz | sort -g -k 1,1) | wc
  46804 1076492 7949471
[  mturchin20@spudling89  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2]$zcat jointGwasMc_HDL.bmass.formatted.txt.gz | awk '{ if ($6 == 0) { print $0 } }' | head -n 10
10      100063147       0.6768  g       a       0       0.922   94311.00
10      100179128       0.5488  g       a       0       0.5806  86431.00
10      10017987        0.5211  c       t       0       0.8692  94311.00
10      10033271        0.5211  g       t       0       0.8775  94311.00
10      100631865       0.7084  c       t       0       0.9285  94303.00
10      100789840       0.3707  g       a       0       0.8183  94311.00
10      100800257       0.6346  c       t       0       0.8237  94298.00
10      100801035       0.2032  c       t       0       0.7842  94311.00
10      101047389       0.157   g       a       0       0.9087  92657.00
10      101057518       0.1675  c       a       0       0.9347  94311.00
[  mturchin20@spudling89  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2]$join <(zcat jointGwasMc_HDL.bmass.formatted.txt.gz | awk '{ if ($6 == 0) { print $1 "_" $2 "\t" $0 } }' | sort -g -k 1,1) <(zcat jointGwasMc_LDL.bmass.formatted.txt.gz | awk '{ if ($6 == 0) { print $1 "_" $2 "\t" $0 } } ' | sort -g -k 1,1) | wc
     37     629    3313
[  mturchin20@spudling89  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2]$join -v 1 <(zcat jointGwasMc_HDL.bmass.formatted.txt.gz | awk '{ if ($6 == 0) { print $1 "_" $2 "\t" $0 } }' | sort -g -k 1,1) <(zcat jointGwasMc_LDL.bmass.formatted.txt.gz | awk '{ if ($6 == 0) { print $1 "_" $2 "\t" $0 } } ' | sort -g -k 1,1) | head -n 10
1_100160067 1 100160067 0.6385 g t 0 0.819 94311.00
1_100877176 1 100877176 0.6359 g a 0 0.7666 91319.00
1_101249300 1 101249300 1 g c 0 0.9823 91192.00
1_101379778 1 101379778 0.09499 g a 0 0.7076 91319.00
1_101391964 1 101391964 0.09631 g a 0 0.6548 91277.00
1_101525689 1 101525689 0.3615 c t 0 0.8743 94230.00
1_101645453 1 101645453 0.1715 c t 0 0.9522 94311.00
1_101810892 1 101810892 1 t a 0 0.7609 89252.00
1_102064973 1 102064973 0.6583 g t 0 0.9032 94311.00
1_102176835 1 102176835 0.4024 c t 0 0.7365 94296.00
[  mturchin20@spudling89  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2]$zcat jointGwasMc_LDL.bmass.formatted.txt.gz | grep 100160067
1       100160067       0.3615  t       g       +       0.7383  89888.00
[  mturchin20@spudling89  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2]$zcat jointGwasMc_LDL.bmass.formatted.txt.gz | grep 101379778
1       101379778       0.90501 a       g       +       0.2352  86937.00
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/CHECK1.jointGwasMc_HDL.MinorAllele.bmass.formatted.noDups.noNA.Flip0Dir.noFixed.forceMAF.txt.gz | awk '{ print $6 }' | sort | uniq -c
1050454 +
1021730 -
      1 Direction
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_HDL.MinorAllele.bmass.formatted.noDups.noNA.no0Dir.noFixed.forceMAF.txt.gz | awk '{ print $6 }' | sort | uniq -c
1038447 +
1021730 -
      1 Direction
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_HDL.MinorAllele.bmass.formatted.noDups.noNA.txt.gz | awk '{ print $6 }' | sort | uniq -c                        
1038583 +
1208628 -
  14148 0
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$R -q -e "1038583 + 14148"
> 1038583 + 14148
[1] 1052731
> 
> 
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$R -q -e "1038447 + 14148"
> 1038447 + 14148
[1] 1052595
> 
> 
> HDL <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/CHECK2.jointGwasMc_HDL.IncAllele.bmass.formatted.noDups.noNA.no0Dir.noFixed.forceMAF.txt.gz", header=T);
> LDL <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/CHECK2.jointGwasMc_LDL.IncAllele.bmass.formatted.noDups.noNA.MatchedToHDL.DropNoMatch.no0Dir.noFixed.forceMAF.txt.gz", header=T);
> TG <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/CHECK2.jointGwasMc_TG.IncAllele.bmass.formatted.noDups.noNA.MatchedToHDL.DropNoMatch.no0Dir.noFixed.forceMAF.txt.gz", header=T);
> TC <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/CHECK2.jointGwasMc_TC.IncAllele.bmass.formatted.noDups.noNA.MatchedToHDL.DropNoMatch.no0Dir.noFixed.forceMAF.txt.gz", header=T);
> GWASsnps <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.ChrBP.txt", header=F);
> names(GWASsnps) <- c("Chr", "BP")
> #bmassOutput1 <- bmass(c("HDL", "LDL", "TG", "TC"), GWASsnps=GWASsnps, NminThreshold = 1, bmassSeedValue=150);
> #print(bmassOutput1);" > shana1.txt
> DataSources = c("HDL", "LDL", "TG", "TC"); NminThreshold = 50000; bmassSeedValue=150;                                                                                                                                                                                                                                      ExpectedColumnNames=c("Chr", "BP", "MAF", "Direction", "pValue", "N"); GWASsnps_AnnotateWindow = 5e5; SNPMarginalUnivariateThreshold = 1e-6; SNPMarginalMultivariateThreshold = 1e-6; SigmaAlphas = c(0.005,0.0075,0.01,0.015,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.15); ProvidedPriors=NULL; UseFlatPriors=FALSE; PruneMarginalSNPs=TRUE; PruneMarginalSNPs_bpWindow=5e5;
>
> bmassOutput1.wInc.CHECK2.NewSetup <- bmass(c("HDL", "LDL", "TG", "TC"), GWASsnps=GWASsnps, NminThreshold = 50000, bmassSeedValue=150);
Warning message:
In ProcessMergedAndAnnotatedDataSources(DataSources, bmassOutput$MergedDataSources,  :
  2017-02-03 18:46:41 -- One of your datafiles has p-values less than the threshold which R can properly convert them to log-scale, meaning their log-values return as 'Infinite'. bmass automatically replaces these 'Infinite' values with the max, non-infinite Z-scores available, but it is recommended you self-check this as well.
> dim(bmassOutput1.wInc.CHECK2.NewSetup$NewSNPs$SNPs)
[1] 87 29
> bmassOutput1.wInc.CHECK2.NewSetup$GWASlogBFMinThreshold
[1] 4.078202
> dim(bmassOutput1.wInc.CHECK2.NewSetup$PreviousSNPs$SNPs)
[1] 148  29
> bmassOutput1.wInc.CHECK2.NewSetup$ZScoresCorMatrix
            HDL_ZScore  LDL_ZScore  TG_ZScore   TC_ZScore
HDL_ZScore  1.00000000 -0.05099993 -0.2301535 -0.09830688
LDL_ZScore -0.05099993  1.00000000  0.1445420 -0.83613234
TG_ZScore  -0.23015345  0.14454195  1.0000000 -0.33892587
TC_ZScore  -0.09830688 -0.83613234 -0.3389259  1.00000000
> bmassOutput3.wInc.CHECK2.NewSetup <- bmass(c("HDL", "LDL", "TG", "TC"), GWASsnps=GWASsnps, NminThreshold = 50000, PrintMergedData=TRUE, PruneMarginalSNPs=FALSE, bmassSeedValue=150);
Warning message:
In ProcessMergedAndAnnotatedDataSources(DataSources, bmassOutput$MergedDataSources,  :
  2017-02-03 18:55:38 -- One of your datafiles has p-values less than the threshold which R can properly convert them to log-scale, meaning their log-values return as 'Infinite'. bmass automatically replaces these 'Infinite' values with the max, non-infinite Z-scores available, but it is recommended you self-check this as well.
> dim(bmassOutput3.wInc.CHECK2.NewSetup$MergedDataSources)
[1] 2063842      23
~~~















##GIANT2010 dbsnp 130, GWAS = , N = provided per SNP
##20150829

cd /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/
wget http://hgdownload.cse.ucsc.edu/goldenPath/hg18/database/snp130.txt.gz

####Taken from earlier in script

##mkdir /mnt/gluster/data/external_public_supp/GIANT2010
##mkdir /mnt/gluster/data/external_public_supp/GIANT2014_5
##mkdir /mnt/gluster/data/external_public_supp/GIANT2013

##cd /mnt/gluster/data/external_public_supp/GIANT2010

##wget http://www.broadinstitute.org/collaboration/giant/images/b/b7/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.txt.gz
##wget http://www.broadinstitute.org/collaboration/giant/images/4/49/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.txt.gz
##wget http://www.broadinstitute.org/collaboration/giant/images/8/87/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.txt.gz

~~~~
[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/GIANT2010]$ls -lrt
total 432452
-rw-rw-r-- 1 mturchin20 mturchin20 22469737 Jun 30  2011 GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.txt.gz
-rw-rw-r-- 1 mturchin20 mturchin20 24655014 Jun 30  2011 GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.txt.gz
-rw-rw-r-- 1 mturchin20 mturchin20 19937269 Jun 30  2011 GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.txt.gz
-rw-rw-r-- 1 mturchin20 mturchin20 36383913 May 19 19:55 GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.vs2.txt.gz
-rw-rw-r-- 1 mturchin20 mturchin20 43392872 May 19 19:57 GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.vs3.txt.gz
-rw-rw-r-- 1 mturchin20 mturchin20 38174702 May 19 20:03 GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.txt.gz
-rw-rw-r-- 1 mturchin20 mturchin20 33378341 May 19 20:12 GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.txt.gz
-rw-rw-r-- 1 mturchin20 mturchin20 36031144 May 19 20:20 GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.txt.gz
-rw-rw-r-- 1 mturchin20 mturchin20 38991094 May 19 20:36 GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.vs3.txt.gz
-rw-rw-r-- 1 mturchin20 mturchin20 41427585 May 19 20:36 GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wdbSNP_human_9606_b142_GRCh37p13.ChrBP.vs3.txt.gz
-rw-rw-r-- 1 mturchin20 mturchin20      139 May 20 21:16 GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.4SNPs.txt.gz
-rw-rw-r-- 1 mturchin20 mturchin20 38308587 May 20 21:57 GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz
-rw-rw-r-- 1 mturchin20 mturchin20 36168037 May 20 22:10 GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz
-rw-rw-r-- 1 mturchin20 mturchin20 33507782 May 20 22:10 GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz
-rw-rw-r-- 1 mturchin20 mturchin20      138 May 21 15:55 GIANT_2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.4otherSNPs.txt.gz
[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/GIANT2010]$for i in `ls -lrt | awk '{ print $9 }' | grep publicrelease_HapMapCeuFreq.txt.gz`; do echo $i; zcat $i | head -n 5; done
GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.txt.gz
MarkerName Allele1 Allele2 Freq.Allele1.HapMapCEU p N
rs10 a c 0.0333 0.708 80566
rs1000000 g a 0.6333 0.506 123865
rs10000010 c t 0.425 0.736 123827
rs10000012 c g 0.8083 0.042 123809
GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.txt.gz
MarkerName Allele1 Allele2 Freq.Allele1.HapMapCEU p N
rs10 a c 0.0333 0.8826 78380
rs1000000 a g 0.3667 0.1858 133822
rs10000010 t c 0.575 0.8947 132858
rs10000012 c g 0.8083 0.1312 133785
GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.txt.gz
MarkerName Allele1 Allele2 Freq.Allele1.HapMapCEU p N
rs10 c a 0.9667 0.42 57031
rs1000000 g a 0.6333 0.55 77168
rs10000010 t c 0.575 0.0029 77152
rs10000012 g c 0.1917 0.99 77117
[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/GIANT2010]$for i in `ls -lrt | awk '{ print $9 }' | grep publicrelease_HapMapCeuFreq.txt.gz`;  do paste <(zcat $i | wc) <(zcat $i | perl -lane 'print $#F;' | sort | uniq -c) | xargs echo $i; done
GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.txt.gz 2471517 14829102 82685870 2471517 5
GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.txt.gz 2469636 14817816 85061092 2469636 5
GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.txt.gz 2483326 14899956 78051825 2483326 5
[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/GIANT2010]$for i in `ls -lrt | awk '{ print $9 }' | grep publicrelease_HapMapCeuFreq.txt.gz`; do echo $i; paste -d @ <(zcat $i | sed 's/e-//g'| perl -lane 'if ($F[0] !~ m/^rs\d+/) { print join("\t", @F); }' | awk '{ print $1 }' | sort | uniq -c) <(zcat $i | sed 's/e-//g'| perl -lane 'if ($F[3] =~ m/[^\d\.]/) { print join("\t", @F); }' | awk '{ print $4 }' | sort | uniq -c ) <(zcat $i | sed 's/e-//g'| perl -lane 'if ($F[4] =~ m/[^\d\.]/) { print join("\t", @F); }' | awk '{ print $5 }' | sort | uniq -c ) <(zcat $i | sed 's/e-//g'| perl -lane 'if ($F[5] =~ m/[^\d\.]/) { print join("\t", @F); }' | awk '{ print $6 }' | sort | uniq -c ) | sed 's/@/ @ /g'| column -s $'@' -t ; done
GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.txt.gz
1 MarkerName          1 Freq.Allele1.HapMapCEU          1 p          3 1e+05
									1 N
GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.txt.gz
1 MarkerName          1 Freq.Allele1.HapMapCEU          1 p          2 1e+05
									1 N
GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.txt.gz
1 MarkerName          1 Freq.Allele1.HapMapCEU          1 p          1 N
[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/GIANT2010]$for i in `ls -lrt | awk '{ print $9 }' | grep publicrelease_HapMapCeuFreq.txt.gz`; do echo $i; zcat $i | perl -lane 'if ($F[3] eq ".") { $F[3] = "NA"; }; print join("\t", @F);' | R -q -e "library(data.table); Data1 <- fread('file:///dev/stdin', header=TRUE); quantile(Data1\$Freq.Allele1.HapMapCEU, na.rm=TRUE); quantile(Data1\$N, na.rm=TRUE); quantile(Data1\$p, na.rm=TRUE); png(\""$i".QCCheck.vs1.png\", height=2000, width=2000, res=200); par(mfrow=c(2,2)); hist(Data1\$Freq.Allele1.HapMapCEU, breaks=100); hist(Data1\$N, breaks=100); hist(Data1\$p, breaks=100); expP <- -log10(c(1:length(Data1\$p["\!"is.na(Data1\$p)])) / (length(Data1\$p["\!"is.na(Data1\$p)])+1)); plot(expP, -log10(sort(Data1\$p["\!"is.na(Data1\$p)])),xlab=\"Expected p-Values\", ylab=\"Observed p-Values\"); abline(0, 1, col=\"RED\"); dev.off();"; done
GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.txt.gz
> library(data.table); Data1 <- fread('file:///dev/stdin', header=TRUE); quantile(Data1$Freq.Allele1.HapMapCEU, na.rm=TRUE); quantile(Data1$N, na.rm=TRUE); quantile(Data1$p, na.rm=TRUE); png("GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.txt.gz.QCCheck.vs1.png", height=2000, width=2000, res=200); par(mfrow=c(2,2)); hist(Data1$Freq.Allele1.HapMapCEU, breaks=100); hist(Data1$N, breaks=100); hist(Data1$p, breaks=100); expP <- -log10(c(1:length(Data1$p[!is.na(Data1$p)])) / (length(Data1$p[!is.na(Data1$p)])+1)); plot(expP, -log10(sort(Data1$p[!is.na(Data1$p)])),xlab="Expected p-Values", ylab="Observed p-Values"); abline(0, 1, col="RED"); dev.off();
Read 2471516 rows and 6 (of 6) columns from 0.077 GB file in 00:00:18
0%    25%    50%    75%   100%
0.0083 0.2083 0.4917 0.7833 0.9917
0%    25%    50%    75%   100%
61957 123254 123842 123862 123912
0%      25%      50%      75%     100%
2.05e-62 2.39e-01 4.92e-01 7.47e-01 1.00e+00
null device
1
>
>
GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.txt.gz
> library(data.table); Data1 <- fread('file:///dev/stdin', header=TRUE); quantile(Data1$Freq.Allele1.HapMapCEU, na.rm=TRUE); quantile(Data1$N, na.rm=TRUE); quantile(Data1$p, na.rm=TRUE); png("GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.txt.gz.QCCheck.vs1.png", height=2000, width=2000, res=200); par(mfrow=c(2,2)); hist(Data1$Freq.Allele1.HapMapCEU, breaks=100); hist(Data1$N, breaks=100); hist(Data1$p, breaks=100); expP <- -log10(c(1:length(Data1$p[!is.na(Data1$p)])) / (length(Data1$p[!is.na(Data1$p)])+1)); plot(expP, -log10(sort(Data1$p[!is.na(Data1$p)])),xlab="Expected p-Values", ylab="Observed p-Values"); abline(0, 1, col="RED"); dev.off();
Read 2469635 rows and 6 (of 6) columns from 0.079 GB file in 00:00:17
0%    25%    50%    75%   100%
0.0083 0.2333 0.5250 0.8083 0.9917
0%    25%    50%    75%   100%
66930 131979 133732 133818 133859
0%       25%       50%       75%      100%
4.470e-52 2.286e-01 4.898e-01 7.471e-01 1.000e+00
null device
1
>
>
GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.txt.gz
> library(data.table); Data1 <- fread('file:///dev/stdin', header=TRUE); quantile(Data1$Freq.Allele1.HapMapCEU, na.rm=TRUE); quantile(Data1$N, na.rm=TRUE); quantile(Data1$p, na.rm=TRUE); png("GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.txt.gz.QCCheck.vs1.png", height=2000, width=2000, res=200); par(mfrow=c(2,2)); hist(Data1$Freq.Allele1.HapMapCEU, breaks=100); hist(Data1$N, breaks=100); hist(Data1$p, breaks=100); expP <- -log10(c(1:length(Data1$p[!is.na(Data1$p)])) / (length(Data1$p[!is.na(Data1$p)])+1)); plot(expP, -log10(sort(Data1$p[!is.na(Data1$p)])),xlab="Expected p-Values", ylab="Observed p-Values"); abline(0, 1, col="RED"); dev.off();
Read 2483325 rows and 6 (of 6) columns from 0.073 GB file in 00:00:16
0%    25%    50%    75%   100%
0.0083 0.2083 0.5000 0.7833 1.0000
0%   25%   50%   75%  100%
38611 77010 77160 77167 77221
0%      25%      50%      75%     100%
7.66e-15 2.40e-01 5.00e-01 7.50e-01 1.00e+00
null device
1
>
>
~~~

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs3.py --file1 <(zcat /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/snp130.txt.gz | perl -lane 'if ($F[1] =~ m/(chr[0-9MXY]+)_.*/) { $F[1] = $1; } print join("\t", @F);') --file2 /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.txt.gz | gzip > /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGB_snp130.vs3.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs3.py --file1 <(zcat /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/snp130.txt.gz | perl -lane 'if ($F[1] =~ m/(chr[0-9MXY]+)_.*/) { $F[1] = $1; } print join("\t", @F);') --file2 /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.txt.gz | gzip > /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGB_snp130.vs3.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs3.py --file1 <(zcat /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/snp130.txt.gz | perl -lane 'if ($F[1] =~ m/(chr[0-9MXY]+)_.*/) { $F[1] = $1; } print join("\t", @F);') --file2 /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.txt.gz | gzip > /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGB_snp130.vs3.txt.gz &

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/process.MTedits.ForGlobalLipids2013.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/process.MTedits.ForGIANT2010.vs2.R

#RSS0 same, # of SNPs slightly different with change in chisq df at the end (correctly changing from 4 to 3)
~~~
[  mturchin20@spudling70  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1]$cat GIANT2010.vs2.RSS0.vs1.txt
Z.BMI,Z.Height,Z.WHRadjBMI
1,0.00738322905943783,0.000344548919060848
0.00738322905943783,1,0.00939104110463606
0.000344548919060848,0.00939104110463606,1
[  mturchin20@spudling70  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1]$cat GIANT2010.RSS0.vs1.txt
Z.BMI,Z.height,Z.WHRadjBMI
1,0.00738322905943783,0.000344548919060848
0.00738322905943783,1,0.00939104110463606
0.000344548919060848,0.00939104110463606,1
[  mturchin20@spudling70  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1]$cat GIANT2010.vs2.dtlesssignif.vs1.txt | wc
   9119    9119 1025000
[  mturchin20@spudling70  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1]$cat GIANT2010.dtlesssignif.vs1.txt | wc
   8842    8842  993994
~~~

~~~
[  mturchin20@spudling70  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1]$for i in `ls -lrt | awk '{ print $9 }' | grep publicrelease_HapMapCeuFreq.wUCSCGB_snp130.vs3.txt.gz`; do echo $i; paste -d @ <(zcat $i | awk '{ print $7 }' | grep NA | wc) <(zcat $i | awk '{ print $7 }' | grep , | wc) <(zcat $i | awk '{ print $7 }' | perl -lane 'my @vals1 = split(/,/, $F[0]); print scalar(@vals1);' | sort | uniq -c) <(zcat $i | awk '{ print $4 }' | perl -lane 'if ($F[0] !~ m/0\.0?[1-9]+/) { print $F[0]; }' | sort | uniq -c) | sed 's/@/ @ /g'| column -s $'@' -t ; done
GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGB_snp130.vs3.txt.gz
37936   37936  113808      10446   10446  303082    2472880 1     222024 .
							2029 2      29718 0.00829999999999997
							8354 3        307 0.0083
							49 4         12 1
							9 5          1 Freq.Allele1.HapMapCEU
							3 6
							1 8
							1 9
GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGB_snp130.vs3.txt.gz
1980    1980    5940      10805   10805  314863    2460712 1     187321 .
							1944 2      30160 0.00829999999999997
							8798 3        289 0.0083
							49 4          1 Freq.Allele1.HapMapCEU
							9 5    
							3 6    
							1 8    
							1 9    
GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGB_snp130.vs3.txt.gz
1981    1981    5943      10809   10809  315035    2458827 1     187070 .
							1938 2      24859 0.00829999999999997
							8808 3        256 0.0083
							49 4          1 Freq.Allele1.HapMapCEU
							9 5    
							3 6    
							1 8    
							1 9    
~~~

for i in `ls -lrt | awk '{ print $9 }' | grep publicrelease_HapMapCeuFreq.wUCSCGB_snp130.vs3.txt.gz`; do echo $i; paste -d @ <(zcat $i | awk '{ print $7 }' | grep NA | wc) <(zcat $i | awk '{ print $7 }' | grep , | wc) <(zcat $i | awk '{ print $7 }' | perl -lane 'my @vals1 = split(/,/, $F[0]); print scalar(@vals1);' | sort | uniq -c) <(zcat $i | awk '{ print $4 }' | perl -lane 'if ($F[0] !~ m/0\.0?[1-9]+/) { print $F[0]; }' | sort | uniq -c) | sed 's/@/ @ /g'| column -s $'@' -t ; done
<(zcat $i | sed 's/e-//g'| perl -lane 'if ($F[9] !~ m/\.0?[1-9]+/) { print $F[9]; } ' | sort | uniq -c )

[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/GIANT2010]$zcat GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGB_snp130.vs3.txt.gz | head -n 10
MarkerName      Allele1 Allele2 Freq.Allele1.HapMapCEU  p       N       ChrBP
rs10    a       c       0.0333  0.708   80566   7_92221824
rs1000000       g       a       0.6333  0.506   123865  12_125456933
rs10000010      c       t       0.425   0.736   123827  4_21227772
rs10000012      c       g       0.8083  0.042   123809  4_1347325
rs10000013      c       a       0.1667  0.0689  123863  4_36901464
rs10000017      t       c       0.2333  0.457   123262  4_84997149
rs1000002       c       t       0.475   0.0322  123783  3_185118462

#20150903 Note -- considering moving entirely from dbSNP database files to get Chr/Position information to just using the hapmap files. I think initially resistant to this because 1) I just originally remembered/thought to use dbSNP than the hapmap files and 2) I think more often than not I was able to find a specific dbSNP release ##, such as 126 vs. 130, whereas with hapmap information it was less consistent to find a specific release ## than say just 'HapMap Phase II' (which can have multiple releases, e.g. 21, 22, 23 etc...). However, think might move to hapmap files entirely now since ultimately SNPs with multiple genomic position mapping may have to just be dropped which, while not 'a lot' of SNPs still feels unnecessary if there is a datasource (e.g. hapmap) that has this info
#Info -- https://www.biostars.org/p/12223/ and https://www.biostars.org/p/2323/

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.txt.gz | gzip > /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wHapMap21.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.txt.gz | gzip > /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wHapMap22.vs3.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.txt.gz | gzip > /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wHapMap23.txt.gz &

~~~
[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/GIANT2010]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wHapMap21.txt.gz | perl -lane 'my $maf1 = sprintf("%.3f", $F[3]); my $maf2 = sprintf("%.3f", ((split(/_/, $F[6]))[3])); if ($maf1 > .5) { $maf1 = sprintf("%.3f", 1 - $maf1); } if ($maf2 > .5) { $maf2 = sprintf("%.3f", 1 - $maf2); } if ($maf1 eq $maf2) { print "match"; } else { print "nope"; }' | sort | uniq -c
1443816 match
1027701 nope
[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/GIANT2010]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wHapMap22.txt.gz | perl -lane 'my $maf1 = sprintf("%.3f", $F[3]); my $maf2 = sprintf("%.3f", ((split(/_/, $F[6]))[3])); if ($maf1 > .5) { $maf1 = sprintf("%.3f", 1 - $maf1); } if ($maf2 > .5) { $maf2 = sprintf("%.3f", 1 - $maf2); } if ($maf1 eq $maf2) { print "match"; } else { print "nope"; }' | sort | uniq -c
1476010 match
 995507 nope
[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/GIANT2010]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wHapMap23.txt.gz | perl -lane 'my $maf1 = sprintf("%.3f", $F[3]); my $maf2 = sprintf("%.3f", ((split(/_/, $F[6]))[3])); if ($maf1 > .5) { $maf1 = sprintf("%.3f", 1 - $maf1); } if ($maf2 > .5) { $maf2 = sprintf("%.3f", 1 - $maf2); } if ($maf1 eq $maf2) { print "match"; } else { print "nope"; }' | sort | uniq -c 
1468909 match
1002608 nope
~~~

#Looks like none of these releases are the correct one yet? 

mkdir /mnt/lustre/home/mturchin20/Data/HapMap/Releases
mkdir /mnt/lustre/home/mturchin20/Data/HapMap/Releases/Release24
mkdir /mnt/lustre/home/mturchin20/Data/HapMap/Releases/Release24/Frequencies

cd /mnt/lustre/home/mturchin20/Data/HapMap/Releases/Release24/Frequencies

wget http://hapmap.ncbi.nlm.nih.gov/downloads/frequencies/2008-10_phaseII/fwd_strand/non-redundant/

mv /mnt/lustre/home/mturchin20/Data/HapMap/Releases/Release24/Frequencies/index.html /mnt/lustre/home/mturchin20/Data/HapMap/Releases/Release24/Frequencies/index.http:__hapmap.ncbi.nlm.nih.gov_downloads_frequencies_2008-10_phaseII_fwd_strand_non-redundant.html

for i in `cat /mnt/lustre/home/mturchin20/Data/HapMap/Releases/Release24/Frequencies/index.http:__hapmap.ncbi.nlm.nih.gov_downloads_frequencies_2008-10_phaseII_fwd_strand_non-redundant.html | perl -lane 'my $fullLine = join("", @F); if ($fullLine =~ m/.*(allele.*.txt.gz).*/) { print $1 ; } '`; do wget http://hapmap.ncbi.nlm.nih.gov/downloads/frequencies/2008-10_phaseII/fwd_strand/non-redundant/$i; done

nbthis *


mkdir /mnt/lustre/home/mturchin20/Data/HapMap/Releases
mkdir /mnt/lustre/home/mturchin20/Data/HapMap/Releases/Release27
mkdir /mnt/lustre/home/mturchin20/Data/HapMap/Releases/Release27/Frequencies

cd /mnt/lustre/home/mturchin20/Data/HapMap/Releases/Release27/Frequencies

wget http://hapmap.ncbi.nlm.nih.gov/downloads/frequencies/2009-02_phaseII+III/forward/non-redundant/

mv /mnt/lustre/home/mturchin20/Data/HapMap/Releases/Release27/Frequencies/index.html /mnt/lustre/home/mturchin20/Data/HapMap/Releases/Release27/Frequencies/index.http:__hapmap.ncbi.nlm.nih.gov_downloads_frequencies_2009-02_phaseIIplusIII_forward_non-redundant.html

for i in `cat /mnt/lustre/home/mturchin20/Data/HapMap/Releases/Release27/Frequencies/index.http:__hapmap.ncbi.nlm.nih.gov_downloads_frequencies_2009-02_phaseIIplusIII_forward_non-redundant.html | perl -lane 'my $fullLine = join("", @F); if ($fullLine =~ m/.*(allele.*.txt.gz).*/) { print $1 ; } ' | grep -E 'CEU|CHB|YRI'`; do wget http://hapmap.ncbi.nlm.nih.gov/downloads/frequencies/2009-02_phaseII+III/forward/non-redundant/$i; done

nbthis *

mkdir /mnt/lustre/home/mturchin20/Data/HapMap/Releases
mkdir /mnt/lustre/home/mturchin20/Data/HapMap/Releases/Release28
mkdir /mnt/lustre/home/mturchin20/Data/HapMap/Releases/Release28/Frequencies

cd /mnt/lustre/home/mturchin20/Data/HapMap/Releases/Release28/Frequencies

wget http://hapmap.ncbi.nlm.nih.gov/downloads/frequencies/2010-08_phaseII+III/

mv /mnt/lustre/home/mturchin20/Data/HapMap/Releases/Release28/Frequencies/index.html /mnt/lustre/home/mturchin20/Data/HapMap/Releases/Release28/Frequencies/index.http:__hapmap.ncbi.nlm.nih.gov_downloads_frequencies_2010-08_phaseIIplusIII.html

for i in `cat /mnt/lustre/home/mturchin20/Data/HapMap/Releases/Release28/Frequencies/index.http:__hapmap.ncbi.nlm.nih.gov_downloads_frequencies_2010-08_phaseIIplusIII.html | perl -lane 'my $fullLine = join("", @F); if ($fullLine =~ m/.*(allele.*.txt.gz).*/) { print $1 ; } ' | grep -E 'CEU|CHB|YRI'`; do wget http://hapmap.ncbi.nlm.nih.gov/downloads/frequencies/2010-08_phaseII+III/$i; done

nbthis *

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.txt.gz | gzip > /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wHapMap24.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.txt.gz | gzip > /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wHapMap27.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.txt.gz | gzip > /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wHapMap28.txt.gz &

~~~
[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/GIANT2010]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wHapMap24.txt.gz | perl -lane 'my $maf1 = sprintf("%.3f", $F[3]); my $maf2 = sprintf("%.3f", ((split(/_/, $F[6]))[3])); if ($maf1 > .5) { $maf1 = sprintf("%.3f", 1 - $maf1); } if ($maf2 > .5) { $maf2 = sprintf("%.3f", 1 - $maf2); } if ($maf1 eq $maf2) { print "match"; } else { print "nope"; }' | sort | uniq -c
1468128 match
1003389 nope
[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/GIANT2010]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wHapMap27.txt.gz | perl -lane 'my $maf1 = sprintf("%.3f", $F[3]); my $maf2 = sprintf("%.3f", ((split(/_/, $F[6]))[3])); if ($maf1 > .5) { $maf1 = sprintf("%.3f", 1 - $maf1); } if ($maf2 > .5) { $maf2 = sprintf("%.3f", 1 - $maf2); } if ($maf1 eq $maf2) { print "match"; } else { print "nope"; }' | sort | uniq -c
 703734 match
 1767783 nope
 [  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/GIANT2010]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wHapMap28.txt.gz | perl -lane 'my $maf1 = sprintf("%.3f", $F[3]); my $maf2 = sprintf("%.3f", ((split(/_/, $F[6]))[3])); if ($maf1 > .5) { $maf1 = sprintf("%.3f", 1 - $maf1); } if ($maf2 > .5) { $maf2 = sprintf("%.3f", 1 - $maf2); } if ($maf1 eq $maf2) { print "match"; } else { print "nope"; }' | sort | uniq -c
  116485 match
  2355032 nope
~~~

#20150903 Note -- No release matches 'perfectly' and the 'most recent' releases match the worst

~~~
[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/GIANT2010]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wHapMap21.txt.gz | grep NA | wc
25994  181958  908406
[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/GIANT2010]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wHapMap22.txt.gz | grep NA | wc
28     196     868
[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/GIANT2010]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wHapMap23.txt.gz | grep NA | wc
6      42     186
[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/GIANT2010]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wHapMap24.txt.gz | grep NA | wc
1905   13335   68671
[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/GIANT2010]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wHapMap27.txt.gz | grep NA | wc
1722   12054   62470
[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/GIANT2010]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wHapMap28.txt.gz | grep NA | wc
2142   14994   77859
~~~

#2015093 Note -- going to go with HapMap22 because 2nd best 'no NA' rate and best 'MAF matching' rate. Also it seems like HapMap22 has been used elsewhere and it seems like, at least during this era, some of these design aspects were common across studies (see: http://www.nature.com.proxy.uchicago.edu/nature/journal/v466/n7307/extref/nature09270-s1.pdf and "...MACH was used to obtain imputed genotypes for ~2.5 million SNPs in the HapMap release 22 for each of the samples...")

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.txt.gz | gzip > /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wHapMap22.vs3.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.txt.gz | gzip > /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wHapMap22.vs3.txt.gz &

~~~
[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/GIANT2010]$for i in `ls -lrt | awk '{ print $9 }' | grep publicrelease_HapMapCeuFreq.wHapMap22`; do echo $i; paste -d @ <(zcat $i | awk '{ print $7 }' | grep NA | wc) <(zcat $i | awk '{ print $7 }' | grep , | wc) <(zcat $i | awk '{ print $7 }' | perl -lane 'my @vals1 = split(/,/, $F[0]); print scalar(@vals1);' | sort | uniq -c) <(zcat $i | awk '{ print $4 }' | perl -lane 'if ($F[0] !~ m/0\.0?[1-9]+/) { print $F[0]; }' | sort | uniq -c) | sed 's/@/ @ /g'| column -s $'@' -t ; done
GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wHapMap22.vs3.txt.gz
28      28      84          0       0       0    2471517 1     187321 .
								30160 0.00829999999999997
								289 0.0083
								1 Freq.Allele1.HapMapCEU
GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wHapMap22.vs3.txt.gz
29      29      87          0       0       0    2469636 1     187070 .
								24859 0.00829999999999997
								256 0.0083
								1 Freq.Allele1.HapMapCEU
GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wHapMap22.vs3.txt.gz
37239   37239  111717          0       0       0    2483326 1     222024 .
								29718 0.00829999999999997
								307 0.0083
								12 1
								1 Freq.Allele1.HapMapCEU
~~~

zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wHapMap22.vs3.txt.gz | perl -lane 'my @vals1 = split(/_/, $F[6]); push(@F, @vals1); if ($F[3] eq ".") { $F[3] = "NA"; } print join("\t", @F);' | awk '{ print $8, "\t", $9, "\t", $1, "\t", $4, "\t", $5, "\t", $6 }' | gzip > /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wHapMap22.vs3.formatted.txt.gz
zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wHapMap22.vs3.txt.gz | perl -lane 'my @vals1 = split(/_/, $F[6]); push(@F, @vals1); if ($F[3] eq ".") { $F[3] = "NA"; } print join("\t", @F);' | awk '{ print $8, "\t", $9, "\t", $1, "\t", $4, "\t", $5, "\t", $6 }' | gzip > /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wHapMap22.vs3.formatted.txt.gz
zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wHapMap22.vs3.txt.gz | perl -lane 'my @vals1 = split(/_/, $F[6]); push(@F, @vals1); if ($F[3] eq ".") { $F[3] = "NA"; } print join("\t", @F);' | awk '{ print $8, "\t", $9, "\t", $1, "\t", $4, "\t", $5, "\t", $6 }' | gzip > /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wHapMap22.vs3.formatted.txt.gz








##20151006 Note -- Checking to see if new hits from 2010 run are within 1Mb of the 2014_5 hits

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT2010.newtophits.vs1.txt

#python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT2010.newtophits.vs1.txt 
#cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt | perl -lane 'for (my $i = 1; $i <= 500000; $i++) { print $F[0], "\t", $F[1], "\t", $F[2] + $i; print $F[0], "\t", $F[1], "\t", $F[2] - $i; }' > /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.w1MbBorder.txt

tail -n +2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT2010.newtophits.vs1.txt | python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 - --file2 /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt > /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.w2010Annot.txt

~~~
[  mturchin20@spudling26  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1]$cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.w2010Annot.txt | awk '{ print $4 }' | sort | uniq -c
838 0
6 1
18 2
[  mturchin20@spudling26  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT2010.newtophits.vs1.txt | wc
20     160     861
[  mturchin20@spudling26  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1]$cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.w2010Annot.txt | awk '{ print $4, "\t", $5 }' | sort | uniq -c
838 0
6 1
1 2   rs10040888
1 2   rs11835818
1 2   rs12204421
2 2   rs12534698
3 2   rs1809889
1 2   rs2025151
1 2   rs2390312
2 2   rs389883
1 2   rs4735692
2 2   rs648831
1 2   rs6824258
1 2   rs7601531
1 2   rs7614120
[  mturchin20@spudling26  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2010/V

~~~



#20151008
#CSHL Poster work

#/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GLC.MTedits.For2013.vs3.ForCSHLPoster.R
cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GLC.MTedits.ForGIANT2010.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GLC.MTedits.ForGIANT2010.vs1.ForCSHLPoster.R 

scp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GLC.MTedits.ForGIANT2010.vs1.ForCSHLPoster.ModelsVslbfavNewOld.vs1.jpeg mturchin20@wolfy.uchicago.edu:/Users/mturchin20/LabStuff/StephensLab/Multivariate/GIANT2010/Vs1/. 



#20160221
#Correcting sign issue -- quick hack to get results

#NOTE -- flipping sign on p-value to make coding downstream easier for this, ie I'm not implying something by creating negative p-values here. This is just for a quick run through.

join <(zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $1, "\t", $2 }' | sort) <(zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz) | perl -lane 'if ($F[1] ne $F[2]) { $F[5] = -1 * $F[5]; } splice(@F, 1, 1); print join("\t", @F);' | gzip > /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.MatchedToHeight.txt.gz
join <(zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $1, "\t", $2 }' | sort) <(zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz) | perl -lane 'if ($F[1] ne $F[2]) { $F[5] = -1 * $F[5]; } splice(@F, 1, 1); print join("\t", @F);' | gzip > /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.MatchedToHeight.txt.gz

join <(zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.txt.gz | awk '{ print $1, "\t", $2 }' | sort) <(zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.txt.gz | sort) | perl -lane 'if ($F[1] ne $F[2]) { $F[5] = -1 * $F[5]; } splice(@F, 1, 1); print join("\t", @F);' | gzip > /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.MatchedToHeight.txt.gz 
join <(zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.txt.gz | awk '{ print $1, "\t", $2 }' | sort) <(zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.txt.gz | sort) | perl -lane 'if ($F[1] ne $F[2]) { $F[5] = -1 * $F[5]; } splice(@F, 1, 1); print join("\t", @F);' | gzip > /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.MatchedToHeight.txt.gz

#cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/process.MTedits.ForGIANT2010.vs2.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/process.MTedits.ForGIANT2010.vs2.SignCrrct.vs1.R
cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/process.MTedits.ForGIANT2010.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/process.MTedits.ForGIANT2010.vs1.SignCrrct.vs1.R

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/DataOrganization.GIANT2010.PythonVLookUp.CreateFullAnnotFile.vs1.py /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/DataOrganization.GIANT2010.PythonVLookUp.CreateFullAnnotFile.vs1.SignCrrct.vs1.py

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/DataOrganization.GIANT2010.PythonVLookUp.CreateFullAnnotFile.vs1.SignCrrct.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt --file2 /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz,/mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.MatchedToHeight.txt.gz,/mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.MatchedToHeight.txt.gz | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT_AllPheno_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.SignCrrct.vs1.RAF.txt.gz

#Added the following header to /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT_AllPheno_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.SignCrrct.vs1.RAF.txt.gz
###MarkerName a1 a2 raf annot beta_height se_height n_height beta_BMI se_BMI n_BMI beta_WHRadjBMI se_WHRadjBMI n_WHRadjBMI

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/DataOrganization.GIANT2010.PythonVLookUp.PutTogetherProcessAndAnnotFiles.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT2010.dtlesssignif.vs1.SignCrrct.vs1.txt --file2 <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT_AllPheno_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.SignCrrct.vs1.RAF.txt.gz | perl -lane 'my $chr; my $pos; if ($F[3] eq "NA") { $chr = "NA"; $pos = "NA"; } else { $chr = ((split(/_/, $F[3]))[0]); $pos = ((split(/_/, $F[3]))[1]); } splice(@F, 1, 0, ($chr, $pos)); splice(@F, 8, 1); splice(@F, 11, 1); splice(@F, 14, 1); print join("\t", @F);') | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT2010.dtlesssignif.vs1.SignCrrect.vs1.annot.vs1.RAF.txt.gz

#Added the following header to /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT2010.dtlesssignif.vs1.SignCrrect.vs1.annot.vs1.RAF.txt.gz
#snp chr pos a1 a2 chrbp raf annot beta_height se_height n_height beta_BMI se_BMI n_BMI beta_WHRadjBMI se_WHRadjBMI n_WHRadjBMI Z.height Z.BMI Z.WHRadjBMI mvstat mvp unip

zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT2010.dtlesssignif.vs1.SignCrrect.vs1.annot.vs1.RAF.txt.gz | perl -lane 'if ($F[6] > .5) { $F[6] = 1 - $F[6]; $F[17] = $F[17] * -1; $F[18] = $F[18] * -1; $F[19] = $F[19] * -1; } print join("\t", @F);' | sed 's/raf/maf/g' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT2010.dtlesssignif.vs1.SignCrrect.vs1.annot.vs1.MAF.txt.gz 

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GLC.MTedits.ForGIANT2010.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GLC.MTedits.ForGIANT2010.vs1.SignCrrct.vs1.R



#20161109
#Vs2 -- Running things through version of bmass() code

mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2
cd /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2

join <(zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $7 }' | grep -v , | grep -v NA | grep -v ChrBP | sort | uniq -u) <(zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $7 "\t" $4 "\t" $2 "\t" $3 "\t+\t" $5 "\t" $6 }' | grep -v Freq | sed 's/_/ /g' | awk '{ print $1 "_" $2 "\t" $0 }' | sort -k 1,1) | perl -lane 'splice(@F, 0, 1); print join("\t", @F);' | perl -lane 'if (($F[2] ne ".") && ($F[2] > 0) && ($F[2] < 1)) { print join("\t", @F); }' | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | gzip > /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.txt.gz 
join <(zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $7 }' | grep -v , | grep -v NA | grep -v ChrBP | sort | uniq -u) <(zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $7 "\t" $4 "\t" $2 "\t" $3 "\t+\t" $5 "\t" $6 }' | grep -v Freq | sed 's/_/ /g' | awk '{ print $1 "_" $2 "\t" $0 }' | sort -k 1,1) | perl -lane 'splice(@F, 0, 1); print join("\t", @F);' | perl -lane 'if (($F[2] ne ".") && ($F[2] > 0) && ($F[2] < 1)) { print join("\t", @F); }' | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | gzip > /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.txt.gz 
join <(zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $7 }' | grep -v , | grep -v NA | grep -v ChrBP | sort | uniq -u) <(zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $7 "\t" $4 "\t" $2 "\t" $3 "\t+\t" $5 "\t" $6 }' | grep -v Freq | sed 's/_/ /g' | awk '{ print $1 "_" $2 "\t" $0 }' | sort -k 1,1) | perl -lane 'splice(@F, 0, 1); print join("\t", @F);' | perl -lane 'if (($F[2] ne ".") && ($F[2] > 0) && ($F[2] < 1)) { print join("\t", @F); }' | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | gzip > /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.txt.gz 

join <(zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.txt.gz | awk '{ print $1 "_" $2 "\t" $4 }' | sort -g -k 1,1) <(zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.txt.gz | awk '{ print $1 "_" $2 "\t" $0 }' | sort -g -k 1,1) | perl -lane 'if ($F[1] eq $F[6]) { ($F[5], $F[6]) = ($F[6], $F[5]); $F[4] = 1 - $F[4]; if ($F[7] eq "-") { $F[7] = "+"; } elsif ($F[7] eq "+") { $F[7] = "-"; } else { $PH = 1; } } elsif ($F[1] eq $F[5]) { $PH = 1; } else { push(@F, "NoAlleleMatch"); } print join("\t", @F[2..$#F]);' | gzip > /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.MatchedToHeight.txt.gz 
join <(zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.txt.gz | awk '{ print $1 "_" $2 "\t" $4 }' | sort -g -k 1,1) <(zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.txt.gz | awk '{ print $1 "_" $2 "\t" $0 }' | sort -g -k 1,1) | perl -lane 'if ($F[1] eq $F[6]) { ($F[5], $F[6]) = ($F[6], $F[5]); $F[4] = 1 - $F[4]; if ($F[7] eq "-") { $F[7] = "+"; } elsif ($F[7] eq "+") { $F[7] = "-"; } else { $PH = 1; } } elsif ($F[1] eq $F[5]) { $PH = 1; } else { push(@F, "NoAlleleMatch"); } print join("\t", @F[2..$#F]);' | gzip > /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.MatchedToHeight.txt.gz 

zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.txt.gz | perl -lane 'if ($F[5] ne "0") { print join("\t", @F); }' | gzip > /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.No0Dir.txt.gz 
zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.MatchedToHeight.txt.gz | grep -v NoAlleleMatch | perl -lane 'if ($F[5] ne "0") { print join("\t", @F); }' | gzip > /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.MatchedToHeight.DropNoMatch.No0Dir.txt.gz
zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.MatchedToHeight.txt.gz | grep -v NoAlleleMatch | perl -lane 'if ($F[5] ne "0") { print join("\t", @F); }' | gzip > /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.MatchedToHeight.DropNoMatch.No0Dir.txt.gz

#20170125
#Made change to force MAF from start now

zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.No0Dir.txt.gz | perl -lane 'if ($F[2] > .5) { $F[2] = 1 - $F[2]; } print join("\t", @F);' | gzip > /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.No0Dir.forceMAF.txt.gz
zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.MatchedToHeight.DropNoMatch.No0Dir.txt.gz | perl -lane 'if ($F[2] > .5) { $F[2] = 1 - $F[2]; } print join("\t", @F);' | gzip > /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.MatchedToHeight.DropNoMatch.No0Dir.forceMAF.txt.gz
zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.MatchedToHeight.DropNoMatch.No0Dir.txt.gz | perl -lane 'if ($F[2] > .5) { $F[2] = 1 - $F[2]; } print join("\t", @F);' | gzip > /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.MatchedToHeight.DropNoMatch.No0Dir.forceMAF.txt.gz

#20170206
#Redoing full setup with orienting towards MAF allele from get go now. Also including checks between reference allele and database (HapMap/dbSNP/etc) allele (including strandedness where possible) when needed. 

#NoDup, noNA, noComma (mainly affects dbSNP), Check Alleles (drop if no match, includes strand check), Flip to Minor, MatchToPheno1, Drop 0Dir, force flip to minor AF, drop if fixed

/mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz
/mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz
/mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz

join <(zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $7 }' | grep -v ChrBP | sort | uniq -u) <(zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $7 "\t" $4 "\t" $2 "\t" $3 "\t+\t" $5 "\t" $6 }' | grep -v NA | grep -v , | grep -v Freq | sed 's/_/ /g' | awk '{ print $1 "_" $2 "\t" $0 }' | perl -lane '$F[4] = uc($F[4]); $F[5] = uc($F[5]); if ($F[3] > .5) {  ($F[4], $F[5]) = ($F[5], $F[4]); if ($F[6] eq "+") { $F[6] = "-"; } elsif ($F[6] eq "-") { $F[6] = "+"; } else { $PH = 1; } $F[3] = 1 - $F[3]; } print join("\t", @F); ' | sort -k 1,1) | perl -lane 'splice(@F, 0, 1); print join("\t", @F);' | perl -lane 'if ($F[2] ne ".") { print join("\t", @F); }' | gzip > /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.MinorAllele.bmass.formatted.noDups.noNA.noComma.noEmptyMAF.txt.gz 
join <(zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $7 }' | grep -v ChrBP | sort | uniq -u) <(zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $7 "\t" $4 "\t" $2 "\t" $3 "\t+\t" $5 "\t" $6 }' | grep -v NA | grep -v , | grep -v Freq | sed 's/_/ /g' | awk '{ print $1 "_" $2 "\t" $0 }' | perl -lane '$F[4] = uc($F[4]); $F[5] = uc($F[5]); if ($F[3] > .5) {  ($F[4], $F[5]) = ($F[5], $F[4]); if ($F[6] eq "+") { $F[6] = "-"; } elsif ($F[6] eq "-") { $F[6] = "+"; } else { $PH = 1; } $F[3] = 1 - $F[3]; } print join("\t", @F); ' | sort -k 1,1) | perl -lane 'splice(@F, 0, 1); print join("\t", @F);' | perl -lane 'if ($F[2] ne ".") { print join("\t", @F); }' | gzip > /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.MinorAllele.bmass.formatted.noDups.noNA.noComma.noEmptyMAF.txt.gz 
join <(zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $7 }' | grep -v ChrBP | sort | uniq -u) <(zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $7 "\t" $4 "\t" $2 "\t" $3 "\t+\t" $5 "\t" $6 }' | grep -v NA | grep -v , | grep -v Freq | sed 's/_/ /g' | awk '{ print $1 "_" $2 "\t" $0 }' | perl -lane '$F[4] = uc($F[4]); $F[5] = uc($F[5]); if ($F[3] > .5) {  ($F[4], $F[5]) = ($F[5], $F[4]); if ($F[6] eq "+") { $F[6] = "-"; } elsif ($F[6] eq "-") { $F[6] = "+"; } else { $PH = 1; } $F[3] = 1 - $F[3]; } print join("\t", @F); ' | sort -k 1,1) | perl -lane 'splice(@F, 0, 1); print join("\t", @F);' | perl -lane 'if ($F[2] ne ".") { print join("\t", @F); }' | gzip > /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.MinorAllele.bmass.formatted.noDups.noNA.noComma.noEmptyMAF.txt.gz 

join <(zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.MinorAllele.bmass.formatted.noDups.noNA.noComma.noEmptyMAF.txt.gz | awk '{ print $1 "_" $2 "\t" $4 }' | sort -g -k 1,1) <(zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.MinorAllele.bmass.formatted.noDups.noNA.noComma.noEmptyMAF.txt.gz | awk '{ print $1 "_" $2 "\t" $0 }' | sort -g -k 1,1) | perl -lane 'if ($F[1] eq $F[6]) { ($F[5], $F[6]) = ($F[6], $F[5]); $F[4] = 1 - $F[4]; if ($F[7] eq "-") { $F[7] = "+"; } elsif ($F[7] eq "+") { $F[7] = "-"; } else { $PH = 1; } } elsif ($F[1] eq $F[5]) { $PH = 1; } else { push(@F, "NoAlleleMatch"); } print join("\t", @F[2..$#F]);' | gzip > /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.MinorAllele.bmass.formatted.noDups.noNA.noComma.noEmptyMAF.MatchedToHeight.txt.gz 
join <(zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.MinorAllele.bmass.formatted.noDups.noNA.noComma.noEmptyMAF.txt.gz | awk '{ print $1 "_" $2 "\t" $4 }' | sort -g -k 1,1) <(zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.MinorAllele.bmass.formatted.noDups.noNA.noComma.noEmptyMAF.txt.gz | awk '{ print $1 "_" $2 "\t" $0 }' | sort -g -k 1,1) | perl -lane 'if ($F[1] eq $F[6]) { ($F[5], $F[6]) = ($F[6], $F[5]); $F[4] = 1 - $F[4]; if ($F[7] eq "-") { $F[7] = "+"; } elsif ($F[7] eq "+") { $F[7] = "-"; } else { $PH = 1; } } elsif ($F[1] eq $F[5]) { $PH = 1; } else { push(@F, "NoAlleleMatch"); } print join("\t", @F[2..$#F]);' | gzip > /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.MinorAllele.bmass.formatted.noDups.noNA.noComma.noEmptyMAF.MatchedToHeight.txt.gz 

zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.MinorAllele.bmass.formatted.noDups.noNA.noComma.noEmptyMAF.txt.gz | perl -lane 'if ($F[5] ne "0") { if ($F[2] > .5) { $F[2] = 1 - $F[2]; } if ($F[2] > 0) { print join("\t", @F); } }' | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | gzip > /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.MinorAllele.bmass.formatted.noDups.noNA.noComma.noEmptyMAF.no0Dir.noFixed.forceMAF.txt.gz
zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.MinorAllele.bmass.formatted.noDups.noNA.noComma.noEmptyMAF.MatchedToHeight.txt.gz | grep -v NoAlleleMatch | perl -lane 'if ($F[5] ne "0") { if ($F[2] > .5) { $F[2] = 1 - $F[2]; } if ($F[2] > 0) { print join("\t", @F); } }' | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | gzip > /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.MinorAllele.bmass.formatted.noDups.noNA.noComma.noEmptyMAF.MatchedToHeight.DropNoMatch.no0Dir.noFixed.forceMAF.txt.gz
zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.MinorAllele.bmass.formatted.noDups.noNA.noComma.noEmptyMAF.MatchedToHeight.txt.gz | grep -v NoAlleleMatch | perl -lane 'if ($F[5] ne "0") { if ($F[2] > .5) { $F[2] = 1 - $F[2]; } if ($F[2] > 0) { print join("\t", @F); } }' | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | gzip > /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.MinorAllele.bmass.formatted.noDups.noNA.noComma.noEmptyMAF.MatchedToHeight.DropNoMatch.no0Dir.noFixed.forceMAF.txt.gz

~~~
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | head -n 10
MarkerName      Allele1 Allele2 Freq.Allele1.HapMapCEU  p       N       ChrBP
rs10 a  c       0.0333  0.8826  78380   7_92221823
rs1000000       a       g       0.3667  0.1858  133822  12_125456932
rs10000010      t       c       0.575   0.8947  132858  4_21227771
rs10000012      c       g       0.8083  0.1312  133785  4_1347324
rs10000013      a       c       0.8333  0.628   133843  4_36901463
rs10000017      t       c       0.2333  0.3073  133174  4_84997148
rs1000002       t       c       0.525   0.221   133711  3_185118461
rs10000023      t       g       0.5917  0.354   131895  4_95952928
rs10000029      c       t       0.025   0.9831  116885  4_138905073
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_HDL.IncAllele.bmass.formatted.txt.gz | head -n 10
Chr BP  MAF A1  A2 Direction  pValue    N
10 10000135 0.4908 a    g  +  0.7538    93561.00
10 10000265 0.4908 c    t  +  0.7295    93561.00
10 100002729    0.3206  a  g  + 0.09234 94311.00
10 100002880    0.3206  a  g  + 0.0934  94311.00
10 100003553    0.219   t  c  + 0.8723  94311.00
10 100003805    0.6517  c  t  + 0.124   94311.00
10 10000459 0.4908 g    t  +  0.7409    93561.00
10 100005593    0.2203  t  g  + 0.6502  94311.00
10 100006329    0.5409  c  t  + 0.05496 94311.00
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $7 "\t" $4 "\t" $2 "\t" $3 "\t+\t" $5 $7 }' | grep -v Freq | sed 's/_/ /g' | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | head -n 10
Chr BP  MAF A1  A2 Direction  pValue    N
7 92221823  0.0333 a    c  +  0.88267 92221823
12 125456932    0.3667  a  g  + 0.185812 125456932
4 21227771  0.575  t    c  +  0.89474 21227771
4 1347324   0.8083 c    g  +  0.13124 1347324
4 36901463  0.8333 a    c  +  0.6284 36901463
4 84997148  0.2333 t    c  +  0.30734 84997148
3 185118461 0.525  t    c  +  0.2213 185118461
4 95952928  0.5917 t    g  +  0.3544 95952928
4 138905073 0.025  c    t  +  0.98314 138905073
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $7 "\t" $4 "\t" $2 "\t" $3 "\t+\t" $5 $7 }' | grep -v Freq | sed 's/_/ /g' | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | head -n 10
Chr BP  MAF A1  A2 Direction  pValue    N
7 92221823  0.0333 a    c  +  0.7087 92221823
12 125456932    0.6333  g  a  + 0.50612 125456932
4 21227771  0.425  c    t  +  0.7364 21227771
4 1347324   0.8083 c    g  +  0.0424 1347324
4 36901463  0.1667 c    a  +  0.06894 36901463
4 84997148  0.2333 t    c  +  0.4574 84997148
3 185118461 0.475  c    t  +  0.03223 185118461
4 95952928  0.5917 t    g  +  0.9394 95952928
4 138905073 0.975  t    c  +  0.244 138905073
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $7 "\t" $4 "\t" $2 "\t" $3 "\t+\t" $5 $7 }' | grep -v Freq | sed 's/_/ /g' | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | head -n 10
Chr BP  MAF A1  A2 Direction  pValue    N
7 92221823  0.9667 c    a  +  0.427 92221823
12 125456932    0.6333  g  a  + 0.5512 125456932
4 21227771  0.575  t    c  +  0.00294 21227771
4 1347324   0.1917 g    c  +  0.994 1347324
4 36901463  0.8333 a    c  +  0.894 36901463
4 84997148  0.7667 c    t  +  0.644 84997148
3 185118461 0.475  c    t  +  0.733 185118461
4 95952928  0.4083 g    t  +  0.0414 95952928
4 138905073 0.975  t    c  +  0.844 138905073
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | wc
2469636 17287452 114167633
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.IncAllele.bmass.formatted.txt.gz | wc
2469636 19869006 103329784
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | wc
2471517 17300619 111814122
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.IncAllele.bmass.formatted.txt.gz | wc
2471517 19883954 100991229
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | wc
2483326 17383282 106990904
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.IncAllele.bmass.formatted.txt.gz | wc
2483326 19901350 98146932
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.IncAllele.bmass.formatted.txt.gz | perl -lane 'print $#F;' | sort | uniq -c
    163 11
   1348 13
     12 15
      5 17
   8771 19
      4 21
      1 23
     29 25
      1 27
      5 31
      1 37
      1 49
   1981 5
2456846 7
    468 9
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.IncAllele.bmass.formatted.txt.gz | perl -lane 'if ($#F == 15) { print join("\t", @F); }' | head -n 10
9  42108182,9   46428850,9 67656949,21  random  955799  0.95    t       c       +       0.41339 42108182,9 46428850,9 67656949,21       random  955799
5  22197867,5   70727046,5 h2 hap1      1352128 0.0833  c  t    +       0.21435 22197867,5      70727046,5 h2  hap1   1352128
5  22235756,5   70764552,5 h2 hap1      1389627 0.55    c  a    +       0.31435 22235756,5      70764552,5 h2  hap1   1389627
15 80414779,15  80722106,15   83598845,15       random  502320  0.9833  c       t       +       0.546615   80414779,15  80722106,15     83598845,15     random  502320
13 18239547,13  18258865,18   14287000,18       14303348,21     14186182        0.55    t       g   +   0.172213 18239547,13    18258865,18     14287000,18     14303348,21     14186182
5  22245138,5   70774005,5 h2 hap1      1399068 0.8583  t  c    +       0.77985 22245138,5      70774005,5 h2  hap1   1399068
2  148840166,Y  57656780,6 qbl  hap2    3442364 0.4417  g  a    +       0.41462 148840166,Y     57656780,6 qbl hap2   3442364
5  22184070,5   70713218,5 h2 hap1      1338298 0.7083  c  t    +       0.84365 22184070,5      70713218,5 h2  hap1   1338298
5  22252602,5   70781465,5 h2 hap1      1406528 0.8667  g  a    +       0.7885  22252602,5      70781465,5 h2  hap1   1406528
5  22240880,5   70769727,5 h2 hap1      1394790 0.00829999999999997     a       g       +       0.47625 22240880,5    70769727,5        h2      hap1    1394790
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep 42108182
rs10908023  t   c  0.95 0.4133  95180   9_42108182,9_46428850,9_67656949,21_random_955799
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.IncAllele.bmass.formatted.txt.gz | grep -v , | perl -lane 'print $#F;' | sort | uniq -c
   1981 5
2456846 7
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.IncAllele.bmass.formatted.txt.gz | grep -v , | perl -lane 'if ($#F == 5) { print join("\t", @F); }' | head -n 10
NA 0.7  g   t   +  0.9533NA
NA 0.2917   g   c  +    0.5317NA
NA 0.2  g   t   +  0.08391NA
NA 0.5333   g   t  +    0.4531NA
NA 0.7167   g   a  +    0.2233NA
NA 0.6833   t   c  +    0.6829NA
NA 0.6833   a   c  +    0.7059NA
NA 0.3917   c   t  +    0.01341NA
NA 0.95 a   g   +  0.4679NA
NA 0.2  g   a   +  0.07273NA
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.IncAllele.bmass.formatted.txt.gz | grep -v , | grep -v NA | perl -lane 'print $#F;' | sort | uniq -c
2456846 7
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.IncAllele.bmass.formatted.txt.gz | perl -lane 'print $#F;' | sort | uniq -c
    162 11
   1350 13
     12 15
      5 17
   8761 19
      4 21
      1 23
     29 25
      1 27
      5 31
      1 37
      1 49
   1980 5
2458732 7
    473 9
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.IncAllele.bmass.formatted.txt.gz | grep -v , | perl -lane 'print $#F;' | sort | uniq -c
   1980 5
2458732 7
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.IncAllele.bmass.formatted.txt.gz | grep -v , | grep -v NA | perl -lane 'print $#F;' | sort | uniq -c
2458732 7
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.IncAllele.bmass.formatted.txt.gz | perl -lane 'print $#F;' | sort | uniq -c
    220 11
   1320 13
     12 15
      6 17
   8316 19
      4 21
      1 23
     29 25
      1 27
      5 31
      1 37
      1 49
  37936 5
2434944 7
    530 9
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.IncAllele.bmass.formatted.txt.gz | grep -v , | perl -lane 'print $#F;' | sort | uniq -c
  37936 5
2434944 7
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.IncAllele.bmass.formatted.txt.gz | grep -v , | grep -v NA | perl -lane 'print $#F;' | sort | uniq -c
2434944 7
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | wc
2469636 17287452 114167633
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep -v , | wc
2458827 17211789 113301662
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep -v , | grep -v NA | wc
2456846 17197922 113226559
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | wc
2471517 17300619 111814122
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep -v , | wc
2460712 17224984 110965571
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep -v , | grep -v NA | wc
2458732 17211124 110892522
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | wc
2483326 17383282 106990904
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep -v , | wc
2472880 17310160 106193111
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep -v , | grep -v NA | wc
2434944 17044608 105034023
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.IncAllele.bmass.formatted.txt.gz | perl -lane 'print $#F;' | sort | uniq -c
2456846 7
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.IncAllele.bmass.formatted.txt.gz | perl -lane 'print $#F;' | sort | uniq -c
2458732 7
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.IncAllele.bmass.formatted.txt.gz | perl -lane 'print $#F;' | sort | uniq -c
2434944 7
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.IncAllele.bmass.formatted.txt.gz | wc
2456846 19654768 102101011
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.IncAllele.bmass.formatted.No0Dir.txt.gz | wc
2456846 19654768 102101011
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.IncAllele.bmass.formatted.MatchedToHeight.txt.gz | wc
2454431 19635498 99533894
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.IncAllele.bmass.formatted.MatchedToHeight.DropNoMatch.No0Dir.txt.gz | wc
2454381 19635048 99531280
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.IncAllele.bmass.formatted.MatchedToHeight.txt.gz | wc
2424926 19399426 95878067
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.IncAllele.bmass.formatted.MatchedToHeight.DropNoMatch.No0Dir.txt.gz | wc
2424908 19399264 95877102
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.IncAllele.bmass.formatted.MatchedToHeight.DropNoMatch.No0Dir.txt.gz | awk '{ print $6 }' | sort | uniq -c
1174759 +
1279621 -
      1 Direction
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.IncAllele.bmass.formatted.MatchedToHeight.DropNoMatch.No0Dir.txt.gz | awk '{ print $6 }' | sort | uniq -c
1211320 +
1213587 -
      1 Direction
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.IncAllele.bmass.formatted.MatchedToHeight.DropNoMatch.No0Dir.txt.gz | awk '{ print $1 "_" $2 }' | wc
2454381 2454381 28585546
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.IncAllele.bmass.formatted.MatchedToHeight.DropNoMatch.No0Dir.txt.gz | awk '{ print $1 "_" $2 }' | sort | uniq | wc
2454355 2454355 28585244
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.IncAllele.bmass.formatted.MatchedToHeight.DropNoMatch.No0Dir.txt.gz | awk '{ print $3 }' | grep -rE '^.$|^0$|^1$' | sort | uniq -c
  90070 .
  96611 1
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.txt.gz | awk '{ print $1 "_" $2 }' | sort | wc 
2269949 2269949 26458664
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.txt.gz | awk '{ print $1 "_" $2 }' | sort | uniq | wc
2269949 2269949 26458664
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.txt.gz | awk '{ print $3 }' | grep -E '^\\.$|^0$|^1$' | sort | uniq -c
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.txt.gz | awk '{ print $1 "_" $2 }' | sort | wc   
2271588 2271588 26477854
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.txt.gz | awk '{ print $1 "_" $2 }' | sort | uniq | wc
2271588 2271588 26477854
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.txt.gz | awk '{ print $3 }' | grep -E '^\\.$|^0$|^1$' | sort | uniq -c 
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.txt.gz | awk '{ print $1 "_" $2 }' | sort | wc
2269949 2269949 26458664
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.txt.gz | awk '{ print $1 "_" $2 }' | sort | uniq | wc
2269949 2269949 26458664
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.txt.gz | awk '{ print $3 }' | grep -E '^\\.$|^0$|^1$' | sort | uniq -c
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$
.
.
.
#20170206
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $7 "\t" $4 "\t" $2 "\t" $3 "\t+\t" $5 "\t" $6 }' | grep -v NA | grep -v , | grep -v Freq | sed 's/_/ /g' | awk '{ print $1 "_" $2 "\t" $0 }' | perl -lane '$F[4] = uc($F[4]); $F[5] = uc($F[5]); if ($F[3] > .5) {  ($F[4], $F[5]) = ($F[5], $F[4]); if ($F[6] eq "+") { $F[6] = "-"; } elsif ($F[6] eq "-") { $F[6] = "+"; } else { $PH = 1; } $F[3] = 1 - $F[3]; } print join("\t", @F); ' | head -n 10
7_92221823  7   92221823   0.0333       A       C       +  0.8826       78380
12_125456932    12 125456932  0.3667    A       G       +  0.1858       133822
4_21227771  4   21227771   0.425        C       T       -  0.8947       132858
4_1347324   4   1347324 0.1917  G       C       -       0.1312  133785
4_36901463  4   36901463   0.1667       C       A       -  0.628        133843
4_84997148  4   84997148   0.2333       T       C       +  0.3073       133174
3_185118461 3   185118461  0.475        C       T       -  0.221        133711
4_95952928  4   95952928   0.4083       G       T       -  0.354        131895
4_138905073 4   138905073  0.025        C       T       +  0.9831       116885
3_99825596  3   99825596   0.125        G       A       -  0.06799      133682
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $7 "\t" $4 "\t" $2 "\t" $3 "\t+\t" $5 "\t" $6 }' | grep -v NA | grep -v , | grep -v Freq | sed 's/_/ /g' | awk '{ print $1 "_" $2 "\t" $0 }' | perl -lane '$F[4] = uc($F[4]); $F[5] = uc($F[5]); if ($F[3] > .5) {  ($F[4], $F[5]) = ($F[5], $F[4]); if ($F[6] eq "+") { $F[6] = "-"; } elsif ($F[6] eq "-") { $F[6] = "+"; } else { $PH = 1; } $F[3] = 1 - $F[3]; } print join("\t", @F); ' | awk '{ print $5 }' | sort | uniq -c
 629526 A
 597594 C
 598541 G
 631184 T
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $7 "\t" $4 "\t" $2 "\t" $3 "\t+\t" $5 "\t" $6 }' | grep -v NA | grep -v , | grep -v Freq | sed 's/_/ /g' | awk '{ print $1 "_" $2 "\t" $0 }' | perl -lane '$F[4] = uc($F[4]); $F[5] = uc($F[5]); if ($F[3] > .5) {  ($F[4], $F[5]) = ($F[5], $F[4]); if ($F[6] eq "+") { $F[6] = "-"; } elsif ($F[6] eq "-") { $F[6] = "+"; } else { $PH = 1; } $F[3] = 1 - $F[3]; } print join("\t", @F); ' | awk '{ print $6 }' | sort | uniq -c
 571391 A
 657804 C
 656385 G
 571265 T
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $7 "\t" $4 "\t" $2 "\t" $3 "\t+\t" $5 "\t" $6 }' | grep -v NA | grep -v , | grep -v Freq | sed 's/_/ /g' | awk '{ print $1 "_" $2 "\t" $0 }' | perl -lane '$F[4] = uc($F[4]); $F[5] = uc($F[5]); if ($F[3] > .5) {  ($F[4], $F[5]) = ($F[5], $F[4]); if ($F[6] eq "+") { $F[6] = "-"; } elsif ($F[6] eq "-") { $F[6] = "+"; } else { $PH = 1; } $F[3] = 1 - $F[3]; } print join("\t", @F); ' | awk '{ print $7 }' | sort | uniq -c
1279486 +
1177359 -
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $7 "\t" $4 "\t" $2 "\t" $3 "\t+\t" $5 "\t" $6 }' | grep -v NA | grep -v , | grep -v Freq | sed 's/_/ /g' | awk '{ print $1 "_" $2 "\t" $0 }' | head -n 10
7_92221823  7 92221823  0.0333  a       c       +       0.8826  78380
12_125456932    12 125456932  0.3667    a       g       +  0.1858       133822
4_21227771  4 21227771  0.575 t c       +       0.8947  132858
4_1347324   4 1347324   0.8083  c       g       +       0.1312  133785
4_36901463  4 36901463  0.8333  a       c       +       0.628   133843
4_84997148  4 84997148  0.2333  t       c       +       0.3073  133174
3_185118461 3 185118461 0.525 t c       +       0.221   133711
4_95952928  4 95952928  0.5917  t       g       +       0.354   131895
4_138905073 4 138905073 0.025 c t       +       0.9831  116885
3_99825596  3 99825596  0.875 a g       +       0.06799 133682
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $7 "\t" $4 "\t" $2 "\t" $3 "\t+\t" $5 "\t" $6 }' | grep -v NA | grep -v , | grep -v Freq | sed 's/_/ /g' | awk '{ print $1 "_" $2 "\t" $0 }' | perl -lane '$F[4] = uc($F[4]); $F[5] = uc($F[5]); if ($F[3] > .5) {  ($F[4], $F[5]) = ($F[5], $F[4]); if ($F[6] eq "+") { $F[6] = "-"; } elsif ($F[6] eq "-") { $F[6] = "+"; } else { $PH = 1; } $F[3] = 1 - $F[3]; } print join("\t", @F); ' | tail -n 10
4_84291760  4   84291760   0.0583       C       T       -  0.7137       131551
4_69885704  4   69885704   0.025        C       A       -  0.8107       132804
4_5978255   4   5978255 0.1083  T       C       +       0.9696  131133
4_121201085 4   121201085  0.0417       C       T       +  0.4169       129978
4_135953340 4   135953340  0.475        T       C       -  0.6987       132008
4_4987061   4   4987061 0.0333  T       C       +       0.4898  130313
4_123121533 4   123121533  0.05 A       G       -       0.8546  91239
4_98781693  4   98781693   0.4583       T       A       +  0.00346      133850
4_69817055  4   69817055   0.1917       C       A       -  0.5583       133833
4_164089927 4   164089927  0.45 G       A       +       0.511   133818
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $7 "\t" $4 "\t" $2 "\t" $3 "\t+\t" $5 "\t" $6 }' | grep -v NA | grep -v , | grep -v Freq | sed 's/_/ /g' | awk '{ print $1 "_" $2 "\t" $0 }' | tail -n 10
4_84291760  4 84291760  0.9417  t       c       +       0.7137  131551
4_69885704  4 69885704  0.975 a c       +       0.8107  132804
4_5978255   4 5978255   0.1083  t       c       +       0.9696  131133
4_121201085 4 121201085 0.0417  c       t       +       0.4169  129978
4_135953340 4 135953340 0.525 c t       +       0.6987  132008
4_4987061   4 4987061   0.0333  t       c       +       0.4898  130313
4_123121533 4 123121533 0.95  g a       +       0.8546  91239
4_98781693  4 98781693  0.4583  t       a       +       0.00346 133850
4_69817055  4 69817055  0.8083  a       c       +       0.5583  133833
4_164089927 4 164089927 0.45  g a       +       0.511   133818
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $7 "\t" $4 "\t" $2 "\t" $3 "\t+\t" $5 "\t" $6 }' | grep -v NA | grep -v , | grep -v Freq | sed 's/_/ /g' | awk '{ print $1 "_" $2 "\t" $0 }' | perl -lane '$F[4] = uc($F[4]); $F[5] = uc($F[5]); if ($F[3] > .5) {  ($F[4], $F[5]) = ($F[5], $F[4]); if ($F[6] eq "+") { $F[6] = "-"; } elsif ($F[6] eq "-") { $F[6] = "+"; } else { $PH = 1; } $F[3] = 1 - $F[3]; } print join("\t", @F); ' | awk '{ print $4 }' | grep ^\.$ | sort | uniq -c
 186869 .
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $7 }' | grep -v ChrBP | sort | wc     
2469635 2469635 29106535
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $7 }' | grep -v ChrBP | sort | uniq -u | wc
2467622 2467622 29100218
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $7 }' | grep , | wc                        
  10809   10809  486392
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $7 }' | grep NA | wc
   1981    1981    5943
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$join <(zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $7 }' | grep -v ChrBP | sort | uniq -u) <(zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $7 "\t" $4 "\t" $2 "\t" $3 "\t+\t" $5 "\t" $6 }' | grep -v NA | grep -v , | grep -v Freq | sed 's/_/ /g' | awk '{ print $1 "_" $2 "\t" $0 }' | perl -lane '$F[4] = uc($F[4]); $F[5] = uc($F[5]); if ($F[3] > .5) {  ($F[4], $F[5]) = ($F[5], $F[4]); if ($F[6] eq "+") { $F[6] = "-"; } elsif ($F[6] eq "-") { $F[6] = "+"; } else { $PH = 1; } $F[3] = 1 - $F[3]; } print join("\t", @F); ' | sort -k 1,1) | perl -lane 'splice(@F, 0, 1); print join("\t", @F);' | perl -lane 'if ($F[2] ne ".") { print join("\t", @F); }' | head -n 10
10 10000134 0.45   G    A  +  0.9278    132999
10 10000264 0.45   T    C  +  0.9278    132999
10 100002728    0.3667  A  G  - 0.1165  133791
10 100002879    0.3583  A  G  - 0.1688  133790
10 100003552    0.2 T   C  -  0.006853  132043
10 100003804    0.3 T   C  +  0.004653  132044
10 100003966    0.3583  T  A  - 0.1897  133803
10 10000458 0.45   T    G  +  0.9278    132999
10 100005552    0.15    C  G  + 0.1403  133784
10 100005592    0.2167  T  G  - 0.01622 133812
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$join <(zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $7 }' | grep -v ChrBP | sort | uniq -u) <(zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $7 "\t" $4 "\t" $2 "\t" $3 "\t+\t" $5 "\t" $6 }' | grep -v NA | grep -v , | grep -v Freq | sed 's/_/ /g' | awk '{ print $1 "_" $2 "\t" $0 }' | perl -lane '$F[4] = uc($F[4]); $F[5] = uc($F[5]); if ($F[3] > .5) {  ($F[4], $F[5]) = ($F[5], $F[4]); if ($F[6] eq "+") { $F[6] = "-"; } elsif ($F[6] eq "-") { $F[6] = "+"; } else { $PH = 1; } $F[3] = 1 - $F[3]; } print join("\t", @F); ' | sort -k 1,1) | perl -lane 'splice(@F, 0, 1); print join("\t", @F);' | perl -lane 'if ($F[2] ne ".") { print join("\t", @F); }' | R -q -e "Data1 <- read.table(file('stdin'), header=F); quantile(Data1[,3]); quantile(Data1[,7]); quantile(Data1[,8]);"
> Data1 <- read.table(file('stdin'), header=F); quantile(Data1[,3]); quantile(Data1[,7]); quantile(Data1[,8]);
    0%    25%    50%    75%   100% 
0.0083 0.1000 0.2167 0.3500 0.5000 
       0%       25%       50%       75%      100% 
4.470e-52 2.301e-01 4.914e-01 7.471e-01 1.000e+00 
    0%    25%    50%    75%   100% 
 66930 131986 133732 133819 133859 
> 
> 
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$join <(zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $7 }' | grep -v ChrBP | sort | uniq -u) <(zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $7 "\t" $4 "\t" $2 "\t" $3 "\t+\t" $5 "\t" $6 }' | grep -v NA | grep -v , | grep -v Freq | sed 's/_/ /g' | awk '{ print $1 "_" $2 "\t" $0 }' | perl -lane '$F[4] = uc($F[4]); $F[5] = uc($F[5]); if ($F[3] > .5) {  ($F[4], $F[5]) = ($F[5], $F[4]); if ($F[6] eq "+") { $F[6] = "-"; } elsif ($F[6] eq "-") { $F[6] = "+"; } else { $PH = 1; } $F[3] = 1 - $F[3]; } print join("\t", @F); ' | sort -k 1,1) | perl -lane 'splice(@F, 0, 1); print join("\t", @F);' | perl -lane 'if ($F[2] ne ".") { print join("\t", @F); }' | head -n 10
10 10000134 0.45   G    A  +  0.446     123864
10 10000264 0.45   T    C  +  0.444     123864
10 100002728    0.3667  A  G  + 0.62    123864
10 100002879    0.3583  A  G  + 0.563   123854
10 100003552    0.2 T   C  +  0.134     123863
10 100003804    0.3 T   C  -  0.113     123864
10 100003966    0.3583  T  A  + 0.547   123864
10 10000458 0.45   T    G  +  0.45      123864
10 100005552    0.15    C  G  - 0.479   123866
10 100005592    0.2167  T  G  + 0.114   123865
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$join <(zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $7 }' | grep -v ChrBP | sort | uniq -u) <(zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $7 "\t" $4 "\t" $2 "\t" $3 "\t+\t" $5 "\t" $6 }' | grep -v NA | grep -v , | grep -v Freq | sed 's/_/ /g' | awk '{ print $1 "_" $2 "\t" $0 }' | perl -lane '$F[4] = uc($F[4]); $F[5] = uc($F[5]); if ($F[3] > .5) {  ($F[4], $F[5]) = ($F[5], $F[4]); if ($F[6] eq "+") { $F[6] = "-"; } elsif ($F[6] eq "-") { $F[6] = "+"; } else { $PH = 1; } $F[3] = 1 - $F[3]; } print join("\t", @F); ' | sort -k 1,1) | perl -lane 'splice(@F, 0, 1); print join("\t", @F);' | perl -lane 'if ($F[2] ne ".") { print join("\t", @F); }'  | R -q -e "Data1 <- read.table(file('stdin'), header=F); quantile(Data1[,3]); quantile(Data1[,7]); quantile(Data1[,8]);"
> Data1 <- read.table(file('stdin'), header=F); quantile(Data1[,3]); quantile(Data1[,7]); quantile(Data1[,8]);
    0%    25%    50%    75%   100% 
0.0083 0.1000 0.2167 0.3500 0.5000 
      0%      25%      50%      75%     100% 
2.05e-62 2.39e-01 4.91e-01 7.47e-01 1.00e+00 
    0%    25%    50%    75%   100% 
 61957 123259 123842 123862 123912 
> 
> 
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$join <(zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $7 }' | grep -v ChrBP | sort | uniq -u) <(zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $7 "\t" $4 "\t" $2 "\t" $3 "\t+\t" $5 "\t" $6 }' | grep -v NA | grep -v , | grep -v Freq | sed 's/_/ /g' | awk '{ print $1 "_" $2 "\t" $0 }' | perl -lane '$F[4] = uc($F[4]); $F[5] = uc($F[5]); if ($F[3] > .5) {  ($F[4], $F[5]) = ($F[5], $F[4]); if ($F[6] eq "+") { $F[6] = "-"; } elsif ($F[6] eq "-") { $F[6] = "+"; } else { $PH = 1; } $F[3] = 1 - $F[3]; } print join("\t", @F); ' | sort -k 1,1) | perl -lane 'splice(@F, 0, 1); print join("\t", @F);' | perl -lane 'if ($F[2] ne ".") { print join("\t", @F); }' | head -n 10
10 10000134 0.45   G    A  +  0.91      77166
10 10000264 0.45   T    C  +  0.92      77166
10 100002728    0.3667  A  G  - 0.67    77167
10 100002879    0.3583  A  G  - 0.56    77167
10 100003552    0.2 T   C  -  0.39      77167
10 100003804    0.3 T   C  +  0.41      77167
10 100003966    0.3583  T  A  - 0.56    77166
10 10000458 0.45   T    G  +  0.92      77166
10 100005552    0.15    C  G  - 0.89    77168
10 100005592    0.2167  T  G  - 0.51    77167
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$join <(zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $7 }' | grep -v ChrBP | sort | uniq -u) <(zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $7 "\t" $4 "\t" $2 "\t" $3 "\t+\t" $5 "\t" $6 }' | grep -v NA | grep -v , | grep -v Freq | sed 's/_/ /g' | awk '{ print $1 "_" $2 "\t" $0 }' | perl -lane '$F[4] = uc($F[4]); $F[5] = uc($F[5]); if ($F[3] > .5) {  ($F[4], $F[5]) = ($F[5], $F[4]); if ($F[6] eq "+") { $F[6] = "-"; } elsif ($F[6] eq "-") { $F[6] = "+"; } else { $PH = 1; } $F[3] = 1 - $F[3]; } print join("\t", @F); ' | sort -k 1,1) | perl -lane 'splice(@F, 0, 1); print join("\t", @F);' | perl -lane 'if ($F[2] ne ".") { print join("\t", @F); }' | R -q -e "Data1 <- read.table(file('stdin'), header=F); quantile(Data1[,3]); quantile(Data1[,7]); quantile(Data1[,8]);"
> Data1 <- read.table(file('stdin'), header=F); quantile(Data1[,3]); quantile(Data1[,7]); quantile(Data1[,8]);
    0%    25%    50%    75%   100% 
0.0000 0.0917 0.2083 0.3500 0.5000 
      0%      25%      50%      75%     100% 
7.66e-15 2.50e-01 5.00e-01 7.50e-01 1.00e+00 
   0%   25%   50%   75%  100% 
38611 77051 77160 77167 77221 
> 
> 
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$join <(zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $7 }' | grep -v ChrBP | sort | uniq -u) <(zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $7 "\t" $4 "\t" $2 "\t" $3 "\t+\t" $5 "\t" $6 }' | grep -v NA | grep -v , | grep -v Freq | sed 's/_/ /g' | awk '{ print $1 "_" $2 "\t" $0 }' | perl -lane '$F[4] = uc($F[4]); $F[5] = uc($F[5]); if ($F[3] > .5) {  ($F[4], $F[5]) = ($F[5], $F[4]); if ($F[6] eq "+") { $F[6] = "-"; } elsif ($F[6] eq "-") { $F[6] = "+"; } else { $PH = 1; } $F[3] = 1 - $F[3]; } print join("\t", @F); ' | sort -k 1,1) | perl -lane 'splice(@F, 0, 1); print join("\t", @F);' | perl -lane 'if ($F[2] ne ".") { print join("\t", @F); }' | awk '{ if ($8 <= 50000) { print $0 } } ' | wc
  27329  218632 1086910

~~~

#20170125
#Note change to using forceMAF files
#20170206
#Note change to MinorAllele files & `MinorAllele.bmass.formatted.noDups.noNA.noComma.noEmptyMAF.MatchedToHeight.DropNoMatch.no0Dir.noFixed.forceMAF`

```
library("devtools"); devtools::load_all("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/bmass");
#Height <- read.table("/mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.No0Dir.txt.gz", header=T)
#BMI <- read.table("/mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.MatchedToHeight.DropNoMatch.No0Dir.txt.gz", header=T);
#WHRadjBMI <- read.table("/mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.MatchedToHeight.DropNoMatch.No0Dir.txt.gz", header=T);
#Height <- read.table("/mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.No0Dir.forceMAF.txt.gz", header=T)
#BMI <- read.table("/mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.MatchedToHeight.DropNoMatch.No0Dir.forceMAF.txt.gz", header=T);
#WHRadjBMI <- read.table("/mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.MatchedToHeight.DropNoMatch.No0Dir.forceMAF.txt.gz", header=T);
Height <- read.table("/mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.MinorAllele.bmass.formatted.noDups.noNA.noComma.noEmptyMAF.no0Dir.noFixed.forceMAF.txt.gz", header=T)
BMI <- read.table("/mnt/gluster/data/external_public_supp/GIANT2010/GIANT_BMI_Speliotes2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.MinorAllele.bmass.formatted.noDups.noNA.noComma.noEmptyMAF.MatchedToHeight.DropNoMatch.no0Dir.noFixed.forceMAF.txt.gz", header=T);
WHRadjBMI <- read.table("/mnt/gluster/data/external_public_supp/GIANT2010/GIANT_WHRadjBMI_Heid2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.MinorAllele.bmass.formatted.noDups.noNA.noComma.noEmptyMAF.MatchedToHeight.DropNoMatch.no0Dir.noFixed.forceMAF.txt.gz", header=T);
##GWASsnps <- read.table("/mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt", header=F);
GWASsnps <- read.table("/mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.txt", header=F);
GWASsnps <- GWASsnps[,2:3]
names(GWASsnps) <- c("Chr", "BP")
#bmassOutput1 <- bmass(c("HDL", "LDL", "TG", "TC"), GWASsnps=GWASsnps, NminThreshold = 1, bmassSeedValue=150);
#print(bmassOutput1);" > shana1.txt
DataSources = c("Height", "BMI", "WHRadjBMI"); NminThreshold = 50000; bmassSeedValue=150;
ExpectedColumnNames=c("Chr", "BP", "MAF", "Direction", "pValue", "N"); GWASsnps_AnnotateWindow = 5e5; SNPMarginalUnivariateThreshold = 1e-6; SNPMarginalMultivariateThreshold = 1e-6; SigmaAlphas = c(0.005,0.0075,0.01,0.015,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.15); ProvidedPriors=NULL; UseFlatPriors=FALSE; PruneMarginalSNPs=TRUE; PruneMarginalSNPs_bpWindow=5e5;

#For all
#ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.txt

bmassOutput1 <- bmass(c("Height", "BMI", "WHRadjBMI"), GWASsnps=GWASsnps, NminThreshold = 50000, PrintMergedData=FALSE, bmassSeedValue=150);
bmassOutput2 <- bmass(c("Height", "BMI", "WHRadjBMI"), GWASsnps=GWASsnps, NminThreshold = 50000, PrintMergedData=FALSE, PruneMarginalSNPs=FALSE, bmassSeedValue=150);
bmassOutput3 <- bmass(c("Height", "BMI", "WHRadjBMI"), GWASsnps=GWASsnps, NminThreshold = 50000, PrintMergedData=TRUE, PruneMarginalSNPs=FALSE, bmassSeedValue=150);

write.table(bmassOutput1$NewSNPs$SNPs, file="/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.txt", quote=FALSE, row.names=FALSE)
write.table(bmassOutput3$MergedDataSources, file="/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/bmass.GIANT2010.Vs2.MergedDataSources.vs1.txt", quote=FALSE, row.names=FALSE)

system("gzip /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/bmass.GIANT2010.Vs2.MergedDataSources.vs1.txt")


#        bmassOutput4 <- list()
#        bmassOutput4$MergedDataSources <- NULL
#        if (!is.null(MergedDataSources)) {
#                bmassOutput4$MergedDataSources <- MergedDataSources
#        }
#        bmassOutput4$ZScoresCorMatrix <- NULL
#        bmassOutput4$MarginalSNPs <- list()
#        bmassOutput4$Models <- NULL
#        bmassOutput4$ModelPriors <- NULL
#        bmassOutput4$PreviousSNPs <- list()
#        bmassOutput4$NewSNPs <- list()
#        bmassOutput4$GWASlogBFMinThreshold <- NULL
#        #LogFile <- c()
#        bmassOutput4$LogFile <- c()
#	
#        bmassOutput4$LogFile <- rbind(bmassOutput4$LogFile, paste(format(Sys.time()), " -- beginning bmass.", sep=""))
#
#	bmassOutput4$LogFile <- CheckIndividualDataSources(DataSources, GWASsnps, ExpectedColumnNames, SigmaAlphas, bmassOutput4$MergedDataSources, ProvidedPriors, UseFlatPriors, PruneMarginalSNPs, PruneMarginalSNPs_bpWindow, SNPMarginalUnivariateThreshold, SNPMarginalMultivariateThreshold, NminThreshold, bmassSeedValue, bmassOutput4$LogFile)
#
#        bmassOutput4[c("MergedDataSources", "LogFile")] <- MergeDataSources(DataSources, bmassOutput4$LogFile)[c("MergedDataSources", "LogFile")]
#
#	bmassOutput41 <- bmassOutput4
# 
#        bmassOutput41[c("MergedDataSources", "LogFile")] <- AnnotateMergedDataWithGWASSNPs(bmassOutput41$MergedDataSources, GWASsnps, GWASsnps_AnnotateWindow, bmassOutput41$LogFile)[c("MergedDataSources", "LogFile")]
# 
#        bmassOutput41[c("MarginalSNPs", "ZScoresCorMatrix", "LogFile")] <- ProcessMergedAndAnnotatedDataSources(DataSources, bmassOutput41$MergedDataSources, SNPMarginalUnivariateThreshold, SNPMarginalMultivariateThreshold, bmassOutput41$LogFile)[c("MarginalSNPs", "ZScoresCorMatrix", "LogFile")]
#
##	bmassOutput4$MergedDataSources$ChrBP <- paste(bmassOutput4$MergedDataSources$Chr, "_", bmassOutput4$MergedDataSources$BP, sep="")
##	bmassOutput4$MarginalSNPs$SNPs <- bmassOutput4$MergedDataSources[bmassOutput4$MergedDataSources$mvstat_log10pVal > -log10(SNPMarginalUnivariateThreshold) | bmassOutput4$MergedDataSources$unistat_log10pVal > -log10(SNPMarginalMultivariateThreshold),]
#
#        bmassOutput41[c("MarginalSNPs", "Models", "ModelPriors", "LogFile")] <- GetLogBFsFromData(DataSources, bmassOutput41$MarginalSNPs, bmassOutput41$ZScoresCorMatrix, SigmaAlphas, bmassOutput41$LogFile)[c("MarginalSNPs", "Models", "ModelPriors", "LogFile")]
#
#        bmassOutput41[c("MarginalSNPs", "PreviousSNPs", "ModelPriors", "GWASlogBFMinThreshold", "LogFile")] <- DetermineAndApplyPriors(DataSources, bmassOutput41$MarginalSNPs, GWASsnps, SigmaAlphas, bmassOutput41$Models, bmassOutput41$ModelPriors, ProvidedPriors, UseFlatPriors, bmassSeedValue, bmassOutput41$LogFile)[c("MarginalSNPs", "PreviousSNPs", "ModelPriors", "GWASlogBFMinThreshold", "LogFile")]
#
#        bmassOutput41[c("MarginalSNPs", "PreviousSNPs", "NewSNPs", "LogFile")] <- FinalizeAndFormatResults(DataSources, bmassOutput41$MarginalSNPs, bmassOutput41$PreviousSNPs, GWASsnps, bmassOutput41$GWASlogBFMinThreshold, SigmaAlphas, bmassOutput41$Models, bmassOutput41$ModelPriors, NminThreshold, PruneMarginalSNPs, PruneMarginalSNPs_bpWindow, bmassOutput41$LogFile)[c("MarginalSNPs", "PreviousSNPs", "NewSNPs", "LogFile")]

```

~~~
> dim(bmassOutput1$NewSNPs$SNPs)
[1] 19 25
> dim(bmassOutput1$PrevSNPs$SNPs)
NULL
> dim(bmassOutput1$PreviousSNPs$SNPs)
[1] 174  25
> dim(GWASsnps)
[1] 226   2
> table(bmassOutput3$MergedDataSources$GWASannot)

      0       1       2 
2082599     198  157894 
> dim(bmassOutput3$PreviousSNPs$SNPs)
[1] 174  25
> bmassOutput1$GWASlogBFMinThreshold
[1] 3.330696
> bmassOutput1$ZScoresCorMatrix
                 Height_ZScore  BMI_ZScore WHRadjBMI_ZScore
Height_ZScore       1.00000000 -0.04723303      -0.00452808
BMI_ZScore         -0.04723303  1.00000000       0.01527599
WHRadjBMI_ZScore   -0.00452808  0.01527599       1.00000000
> dim(bmassOutput3$MergedDataSources)           
[1] 2240691      19
#20170223
#From /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GLC.MTedits.ForGIANT2010.vs1.SignCrrct.vs1.R
#> min(lbf.av.glhits)
#[1] 3.324269       
#19 in results
#From /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/process.MTedits.ForGIANT2010.vs2.SignCrrct.vs1.R
#> RSS0
#                  Z.BMI     Z.Height  Z.WHRadjBMI 
#Z.BMI        1.00000000 -0.028708246  0.014570846
#Z.Height    -0.02870825  1.000000000 -0.003307104
#Z.WHRadjBMI  0.01457085 -0.003307104  1.000000000
#20170125 -- from forceMAF now
> dim(bmassOutput1$NewSNPs$SNPs)
[1] 21 25
> bmassOutput1$GWASlogBFMinThreshold
[1] 3.325272
> dim(bmassOutput3$MergedDataSources)   
[1] 2240691      19
> bmassOutput1$ZScoresCorMatrix
                 Height_ZScore  BMI_ZScore WHRadjBMI_ZScore
Height_ZScore      1.000000000 -0.02928493     -0.003028615
BMI_ZScore        -0.029284934  1.00000000      0.015141630
WHRadjBMI_ZScore  -0.003028615  0.01514163      1.000000000
> dim(bmassOutput1$PreviousSNPs$SNPs)
[1] 174  25
> dim(GWASsnps)
[1] 226   2
#20170206
> dim(bmassOutput1$NewSNPs$SNPs)
[1] 19 25
> dim(bmassOutput3$MergedDataSources)
[1] 2240691      19
> bmassOutput1$GWASlogBFMinThreshold
[1] 3.330696
> bmassOutput1$ZScoresCorMatrix
                 Height_ZScore  BMI_ZScore WHRadjBMI_ZScore
Height_ZScore       1.00000000 -0.04723303      -0.00452808
BMI_ZScore         -0.04723303  1.00000000       0.01527599
WHRadjBMI_ZScore   -0.00452808  0.01527599       1.00000000
> dim(GWASsnps)
[1] 226   2
> dim(bmassOutput1$PreviousSNPs$SNPs)
[1] 174  25
~~~













mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2
cd /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2

#Chr BP  MAF     A1      A2      Direction       pValue  N
#ExpectedColumnNames=c("Chr", "BP", "MAF", "Direction", "pValue", "N")

1) Expand/Correct/Make format of files
2) Match to a phenotype
3) Extra steps such as drop no allele matches or 0dir step (latter maybe specific to GlobalLipids 2010)

library("devtools"); devtools::load_all("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/bmass");
HDL <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_HDL.IncAllele.bmass.formatted.txt.gz", header=T);
LDL <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_LDL.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.txt.gz", header=T);
TG <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TG.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.txt.gz", header=T);
TC <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TC.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.txt.gz", header=T);
GWASsnps <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.ChrBP.txt", header=F);
names(GWASsnps) <- c("Chr", "BP")
#bmassOutput1 <- bmass(c("HDL", "LDL", "TG", "TC"), GWASsnps=GWASsnps, NminThreshold = 1, bmassSeedValue=150);
#print(bmassOutput1);" > shana1.txt
DataSources = c("HDL", "LDL", "TG", "TC"); NminThreshold = 50000; bmassSeedValue=150;
ExpectedColumnNames=c("Chr", "BP", "MAF", "Direction", "pValue", "N"); GWASsnps_AnnotateWindow = 5e5; SNPMarginalUnivariateThreshold = 1e-6; SNPMarginalMultivariateThreshold = 1e-6; SigmaAlphas = c(0.005,0.0075,0.01,0.015,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.15); ProvidedPriors=NULL; UseFlatPriors=FALSE; PruneMarginalSNPs=TRUE; PruneMarginalSNPs_bpWindow=5e5;

#For all
#ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.txt

bmassOutput1 <- bmass(c("HDL", "LDL", "TG", "TC"), GWASsnps=GWASsnps, NminThreshold = 50000, PrintMergedData=FALSE, bmassSeedValue=150);
bmassOutput2 <- bmass(c("HDL", "LDL", "TG", "TC"), GWASsnps=GWASsnps, NminThreshold = 50000, PrintMergedData=FALSE, PruneMarginalSNPs=FALSE, bmassSeedValue=150);
bmassOutput3 <- bmass(c("HDL", "LDL", "TG", "TC"), GWASsnps=GWASsnps, NminThreshold = 50000, PrintMergedData=TRUE, PruneMarginalSNPs=FALSE, bmassSeedValue=150);

write.table(bmassOutput1$NewSNPs$SNPs, file="/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.txt", quote=FALSE, row.names=FALSE)
write.table(bmassOutput1$MergedDataSources, file="/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/bmass.GlobalLipids2013.Vs2.MergedDataSources.vs1.txt", quote=FALSE, row.names=FALSE)

system("gzip /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/bmass.GlobalLipids2013.Vs2.MergedDataSources.vs1.txt")
























##GIANT2014/5 dbsnp 130, GWAS = , N = provided per SNP
##20150829

####Taken from earlier in script

##mkdir /mnt/gluster/data/external_public_supp/GIANT2010
##mkdir /mnt/gluster/data/external_public_supp/GIANT2014_5
##mkdir /mnt/gluster/data/external_public_supp/GIANT2013

##cd /mnt/gluster/data/external_public_supp/GIANT2014_5

##wget http://www.broadinstitute.org/collaboration/giant/images/0/01/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.txt.gz
##wget http://www.broadinstitute.org/collaboration/giant/images/f/f0/All_ancestries_SNP_gwas_mc_merge_nogc.tbl.uniq.gz
##wget http://www.broadinstitute.org/collaboration/giant/images/1/15/SNP_gwas_mc_merge_nogc.tbl.uniq.gz
##wget http://www.broadinstitute.org/collaboration/giant/images/e/eb/GIANT_2015_WHRadjBMI_COMBINED_EUR.txt.gz
##wget http://www.broadinstitute.org/collaboration/giant/images/f/f6/GIANT_2015_WHRadjBMI_COMBINED_AllAncestries.txt.gz
##wget http://www.broadinstitute.org/collaboration/giant/images/5/52/GIANT_2015_HIPadjBMI_COMBINED_EUR.txt.gz
##wget http://www.broadinstitute.org/collaboration/giant/images/0/0e/GIANT_2015_HIPadjBMI_COMBINED_AllAncestries.txt.gz
##wget http://www.broadinstitute.org/collaboration/giant/images/7/73/GIANT_2015_WCadjBMI_COMBINED_EUR.txt.gz
##wget http://www.broadinstitute.org/collaboration/giant/images/3/3f/GIANT_2015_WCadjBMI_COMBINED_AllAncestries.txt.gz
##wget http://www.broadinstitute.org/collaboration/giant/images/5/54/GIANT_2015_WHR_COMBINED_EUR.txt.gz
##wget http://www.broadinstitute.org/collaboration/giant/images/d/d7/GIANT_2015_WHR_COMBINED_AllAncestries.txt.gz
##wget http://www.broadinstitute.org/collaboration/giant/images/e/e4/GIANT_2015_HIP_COMBINED_EUR.txt.gz
##wget http://www.broadinstitute.org/collaboration/giant/images/6/6f/GIANT_2015_HIP_COMBINED_AllAncestries.txt.gz
##wget http://www.broadinstitute.org/collaboration/giant/images/5/57/GIANT_2015_WC_COMBINED_EUR.txt.gz
##wget http://www.broadinstitute.org/collaboration/giant/images/e/ea/GIANT_2015_WC_COMBINED_AllAncestries.txt.gz

#Focusing on 'European Ancestry' files since in http://www.nature.com.proxy.uchicago.edu/ng/journal/v46/n11/full/ng.3097.html we have this statement -- "...We first performed a GWAS meta-analysis of adult height using the summary statistics from 79 studies consisting of 253,288 individuals of European ancestry (Online Methods)...."

ln -s /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.txt.gz /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2014_Height_publicrelease_HapMapCeuFreq.ForAnalysis.txt.gz
ln -s /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.gz /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_BMI.ForAnalysis.txt.gz
ln -s /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.txt.gz /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.ForAnalysis.txt.gz

~~~
[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/GIANT2014_5]$ls -lrt | grep ForAnalysis
lrwxrwxrwx 1 mturchin20 mturchin20      114 Aug 29 13:33 GIANT_2014_Height_publicrelease_HapMapCeuFreq.ForAnalysis.txt.gz -> /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.txt.gz
lrwxrwxrwx 1 mturchin20 mturchin20       85 Aug 29 13:33 GIANT_2015_BMI.ForAnalysis.txt.gz -> /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.gz
lrwxrwxrwx 1 mturchin20 mturchin20       91 Aug 29 13:33 GIANT_2015_WHRadjBMI_COMBINED_EUR.ForAnalysis.txt.gz -> /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.txt.gz
[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/GIANT2014_5]$for i in `ls -lrt | awk '{ print $9 }' | grep ForAnalysis`; do echo $i; zcat $i | head -n 5; done
GIANT_2014_Height_publicrelease_HapMapCeuFreq.ForAnalysis.txt.gz
MarkerName      Allele1 Allele2 Freq.Allele1.HapMapCEU  b       SE      p       N
rs4747841       A       G       0.551   -0.0011 0.0029  0.70    253213
rs4749917       T       C       0.436   0.0011  0.0029  0.70    253213
rs737656        A       G       0.367   -0.0062 0.0030  0.042   253116
rs737657        A       G       0.358   -0.0062 0.0030  0.041   252156
GIANT_2015_BMI.ForAnalysis.txt.gz
SNP     A1      A2      Freq1.Hapmap    b       se      p       N
rs1000000       G       A       0.6333  1e-04   0.0044  0.9819  231410
rs10000010      T       C       0.575   -0.0029 0.003   0.3374  322079
rs10000012      G       C       0.1917  -0.0095 0.0054  0.07853 233933
rs10000013      A       C       0.8333  -0.0095 0.0044  0.03084 233886
GIANT_2015_WHRadjBMI_COMBINED_EUR.ForAnalysis.txt.gz
PaxHeader/GIANT_2015_WHRadjBMI_COMBINED_EUR.txt000644 777777 000120 00000000234 12466640262 021316 xustar00med-dsnadmin000000 000000 17 uid=334083809
20 ctime=1423655090
20 atime=1423655079
38 LIBARCHIVE.creationtime=1423655076
23 SCHILY.dev=16777228
~~~

#Editing GIANT_2015_WHRadjBMI_COMBINED_EUR.ForAnalysis.txt.gz so as to fix the header issue

cp -p /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.ForAnalysis.txt.gz /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.ForAnalysis.HeaderFix.txt.gz

#Manually edited /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.ForAnalysis.HeaderFix.txt.gz using vi; removed the following lines up until the beginning of the visible header
#Also removed the very last line which has an issue
~~~
PaxHeader/GIANT_2015_WHRadjBMI_COMBINED_EUR.txt000644 777777 000120 00000000234 12466640262 021316 xustar00med-dsnadmin000000 000000 17 uid=334083809
20 ctime=1423655090
20 atime=1423655079
38 LIBARCHIVE.creationtime=1423655076
23 SCHILY.dev=16777228
20 SCHILY.ino=80523
18 SCHILY.nlink=1
GIANT_2015_WHRadjBMI_COMBINED_EUR.txt000644 ï¿½ï¿½ï¿½000120 00702223325 12466640262 020237 0ustar00med-dsnadmin000000 000000 MarkerName       Allele1 Allele2 FreqAllele1HapMapCEU    b       se      p       N
.
.
.

~~~

rm /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.txt.gz

~~~
[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/GIANT2014_5]$for i in `ls -lrt | awk '{ print $9 }' | grep ForAnalysis`; do echo $i; zcat $i | head -n 5; done
GIANT_2014_Height_publicrelease_HapMapCeuFreq.ForAnalysis.txt.gz
MarkerName      Allele1 Allele2 Freq.Allele1.HapMapCEU  b       SE      p       N
rs4747841       A       G       0.551   -0.0011 0.0029  0.70    253213
rs4749917       T       C       0.436   0.0011  0.0029  0.70    253213
rs737656        A       G       0.367   -0.0062 0.0030  0.042   253116
rs737657        A       G       0.358   -0.0062 0.0030  0.041   252156
GIANT_2015_BMI.ForAnalysis.txt.gz
SNP     A1      A2      Freq1.Hapmap    b       se      p       N
rs1000000       G       A       0.6333  1e-04   0.0044  0.9819  231410
rs10000010      T       C       0.575   -0.0029 0.003   0.3374  322079
rs10000012      G       C       0.1917  -0.0095 0.0054  0.07853 233933
rs10000013      A       C       0.8333  -0.0095 0.0044  0.03084 233886
GIANT_2015_WHRadjBMI_COMBINED_EUR.ForAnalysis.HeaderFix.txt.gz
MarkerName      Allele1 Allele2 FreqAllele1HapMapCEU    b       se      p       N
rs10011200      C       G       0.5333  0.017   0.0043  0.00011 142475
rs8051831       T       C       0.0917  0.034   0.0089  0.00011 138860
rs17542520      A       G       0.0917  0.028   0.0071  0.00011 142581
rs6954671       G       C       0.4167  0.016   0.0042  0.00011 142637
[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/GIANT2014_5]$for i in `ls -lrt | awk '{ print $9 }' | grep ForAnalysis`; do paste <(zcat $i | wc) <(zcat $i | perl -lane 'print $#F;' | sort | uniq -c) | xargs echo $i; done
GIANT_2014_Height_publicrelease_HapMapCeuFreq.ForAnalysis.txt.gz 2550859 20406872 119190473 2550859 7
GIANT_2015_BMI.ForAnalysis.txt.gz 2554638 20437104 125031676 2554638 7
GIANT_2015_WHRadjBMI_COMBINED_EUR.ForAnalysis.HeaderFix.txt.gz 2542433 20339456 118045185 1 0 2542432 7
[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/GIANT2014_5]$for i in `ls -lrt | awk '{ print $9 }' | grep ForAnalysis`; do echo $i; paste -d @ <(zcat $i | sed 's/e-//g'| perl -lane 'if ($F[0] !~ m/^rs\d+/) { if ($F[0] !~ m/.*\d+(_|:)\d+/) { print join("\t", @F); } }' | awk '{ print $1 }' | sort | uniq -c) <(zcat $i | sed 's/e-//g'| perl -lane 'if ($F[3] =~ m/[^\d\.]/) { print join("\t", @F); }' | awk '{ print $4 }' | sort | uniq -c ) <(zcat $i | sed 's/e-//g'| perl -lane 'if ($F[6] =~ m/[^\d\.]/) { print join("\t", @F); }' | awk '{ print $7 }' | sort | uniq -c ) <(zcat $i | sed 's/e-//g'| perl -lane 'if ($F[7] =~ m/[^\d\.]/) { print join("\t", @F); }' | awk '{ print $8 }' | sort | uniq -c ) | sed 's/@/ @ /g'| column -s $'@' -t ; done
GIANT_2014_Height_publicrelease_HapMapCeuFreq.ForAnalysis.txt.gz
1 MarkerName             1 Freq.Allele1.HapMapCEU          1 p          1 N
1 SNP_A-2097957       3577 NA
GIANT_2015_BMI.ForAnalysis.txt.gz
1 SNP          1 Freq1.Hapmap          1 p          5 2e+05
54674 NA	                                 1 3e+05
							1 N
GIANT_2015_WHRadjBMI_COMBINED_EUR.ForAnalysis.HeaderFix.txt.gz
column: line too long
1 MarkerName      54824 NA              1 N
[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/GIANT2014_5]$for i in `ls -lrt | awk '{ print $9 }' | grep ForAnalysis`; do echo $i; zcat $i | sed 's/F
req1.Hapmap/Freq.Allele1.HapMapCEU/g' | sed 's/FreqAllele1HapMapCEU/Freq.Allele1.HapMapCEU/g' | sed 's/SNP/MarkerName/g' | R -q -e "library(data.table); Data1 <- fread(
'file:///dev/stdin', header=TRUE); quantile(Data1\$Freq.Allele1.HapMapCEU, na.rm=TRUE); quantile(Data1\$N, na.rm=TRUE); quantile(Data1\$p, na.rm=TRUE); png(\""$i".QCChe
ck.vs1.png\", height=2000, width=2000, res=200); par(mfrow=c(2,2)); hist(Data1\$Freq.Allele1.HapMapCEU, breaks=100); hist(Data1\$N, breaks=100); hist(Data1\$p, breaks=1
00); expP <- -log10(c(1:length(Data1\$p["\!"is.na(Data1\$p)])) / (length(Data1\$p["\!"is.na(Data1\$p)])+1)); plot(expP, -log10(sort(Data1\$p["\!"is.na(Data1\$p)])),xlab=\"Expected p-Values\", ylab=\"Observed p-Values\"); abline(0, 1, col=\"RED\"); dev.off();"; done
GIANT_2014_Height_publicrelease_HapMapCeuFreq.ForAnalysis.txt.gz
> library(data.table); Data1 <- fread('file:///dev/stdin', header=TRUE); quantile(Data1$Freq.Allele1.HapMapCEU, na.rm=TRUE); quantile(Data1$N, na.rm=TRUE); quantile(Data1$p, na.rm=TRUE); png("GIANT_2014_Height_publicrelease_HapMapCeuFreq.ForAnalysis.txt.gz.QCCheck.vs1.png", height=2000, width=2000, res=200); par(mfrow=c(2,2)); hist(Data1$Freq.Allele1.HapMapCEU, breaks=100); hist(Data1$N, breaks=100); hist(Data1$p, breaks=100); expP <- -log10(c(1:length(Data1$p[!is.na(Data1$p)])) / (length(Data1$p[!is.na(Data1$p)])+1)); plot(expP, -log10(sort(Data1$p[!is.na(Data1$p)])),xlab="Expected p-Values", ylab="Observed p-Values"); abline(0, 1, col="RED"); dev.off();
Read 2550858 rows and 8 (of 8) columns from 0.111 GB file in 00:00:30
0%   25%   50%   75%  100%
0.000 0.192 0.467 0.774 1.000
0%    25%    50%    75%   100%
50003 241207 251631 253085 253280
0%      25%      50%      75%     100%
3.2e-158  9.8e-02  3.5e-01  6.6e-01  1.0e+00
null device
1
>
>
GIANT_2015_BMI.ForAnalysis.txt.gz
> library(data.table); Data1 <- fread('file:///dev/stdin', header=TRUE); quantile(Data1$Freq.Allele1.HapMapCEU, na.rm=TRUE); quantile(Data1$N, na.rm=TRUE); quantile(Data1$p, na.rm=TRUE); png("GIANT_2015_BMI.ForAnalysis.txt.gz.QCCheck.vs1.png", height=2000, width=2000, res=200); par(mfrow=c(2,2)); hist(Data1$Freq.Allele1.HapMapCEU, breaks=100); hist(Data1$N, breaks=100); hist(Data1$p, breaks=100); expP <- -log10(c(1:length(Data1$p[!is.na(Data1$p)])) / (length(Data1$p[!is.na(Data1$p)])+1)); plot(expP, -log10(sort(Data1$p[!is.na(Data1$p)])),xlab="Expected p-Values", ylab="Observed p-Values"); abline(0, 1, col="RED"); dev.off();
Read 2554637 rows and 8 (of 8) columns from 0.116 GB file in 00:00:27
0%    25%    50%    75%   100%
0.0083 0.1833 0.4464 0.7583 1.0000
0%      25%      50%      75%     100%
50004.7 222251.0 233524.0 233952.0 322154.0
0%        25%        50%        75%       100%
7.510e-153  2.318e-01  4.870e-01  7.447e-01  1.000e+00
null device
1
>
>
> library(data.table); Data1 <- fread('file:///dev/stdin', header=TRUE); quantile(Data1$Freq.Allele1.HapMapCEU, na.rm=TRUE); quantile(Data1$N, na.rm=TRUE); quantile(Data1$p, na.rm=TRUE); png("GIANT_2015_WHRadjBMI_COMBINED_EUR.ForAnalysis.HeaderFix.txt.gz.QCCheck.vs1.png", height=2000, width=2000, res=200); par(mfrow=c(2,2)); hist(Data1$Freq.Allele1.HapMapCEU, breaks=100); hist(Data1$N, breaks=100); hist(Data1$p, breaks=100); expP <- -log10(c(1:length(Data1$p[!is.na(Data1$p)])) / (length(Data1$p[!is.na(Data1$p)])+1)); plot(expP, -log10(sort(Data1$p[!is.na(Data1$p)])),xlab="Expected p-Values", ylab="Observed p-Values"); abline(0, 1, col="RED"); dev.off();
Read 2542431 rows and 8 (of 8) columns from 0.110 GB file in 00:00:34
0%    25%    50%    75%   100%
0.0083 0.1833 0.4500 0.7583 0.9917
0%    25%    50%    75%   100%
50001 131441 142322 142675 210086
0%     25%     50%     75%    100%
3.6e-35 2.4e-01 5.0e-01 7.5e-01 1.0e+00
null device
1
>
>
~~~

#Note -- GIANT_2015_WHRadjBMI_COMBINED_EUR.ForAnalysis.HeaderFix.txt.gz has 'MarkerName's that are chr##:###### and hg18_##_######

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs3.py --file1 <(zcat /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/snp130.txt.gz | perl -lane 'if ($F[1] =~ m/(chr[0-9MXY]+)_.*/) { $F[1] = $1; } print join("\t", @F);') --file2 /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2014_Height_publicrelease_HapMapCeuFreq.ForAnalysis.txt.gz | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2014_Height_publicrelease_HapMapCeuFreq.ForAnalysis.wUCSCGB_snp130.vs3.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs3.py --file1 <(zcat /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/snp130.txt.gz | perl -lane 'if ($F[1] =~ m/(chr[0-9MXY]+)_.*/) { $F[1] = $1; } print join("\t", @F);') --file2 /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_BMI.ForAnalysis.txt.gz | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_BMI.ForAnalysis.wUCSCGB_snp130.vs3.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs3.py --file1 <(zcat /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/snp130.txt.gz | perl -lane 'if ($F[1] =~ m/(chr[0-9MXY]+)_.*/) { $F[1] = $1; } print join("\t", @F);') --file2 /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.ForAnalysis.HeaderFix.txt.gz | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.ForAnalysis.wUCSCGB_snp130.vs3.HeaderFix.txt.gz &

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/process.MTedits.ForGIANT2010.vs2.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/process.MTedits.ForGIANT2014_5.vs2.R 

#RSS0 same, # of SNPs slightly different with change in chisq df at the end (correctly changing from 4 to 3)
~~~
[  mturchin20@spudling70  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1]$cat GIANT2014_5.vs2.RSS0.vs1.txt
Z.BMI,Z.Height,Z.WHRadjBMI
1,0.00402473151403864,0.00476194401287714
0.00402473151403864,1,0.0106619753044063
0.00476194401287714,0.0106619753044063,1
[  mturchin20@spudling70  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1]$cat GIANT2014_5.Orig3.RSS0.vs1.txt
Z.BMI,Z.height,Z.WHRadjBMI
1,0.00402473151403864,0.00476194401287714
0.00402473151403864,1,0.0106619753044063
0.00476194401287714,0.0106619753044063,1
[  mturchin20@spudling70  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1]$cat GIANT2014_5.vs2.dtlesssignif.vs1.txt | wc
  43087   43087 4827200
[  mturchin20@spudling70  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1]$cat GIANT2014_5.Orig3.dtlesssignif.vs1.txt | wc
  42023   42023 4708300
~~~

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2014_Height_publicrelease_HapMapCeuFreq.ForAnalysis.txt.gz | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2014_Height_publicrelease_HapMapCeuFreq.ForAnalysis.wHapMap21.vs3.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2014_Height_publicrelease_HapMapCeuFreq.ForAnalysis.txt.gz | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2014_Height_publicrelease_HapMapCeuFreq.ForAnalysis.wHapMap23.vs3.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2014_Height_publicrelease_HapMapCeuFreq.ForAnalysis.txt.gz | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2014_Height_publicrelease_HapMapCeuFreq.ForAnalysis.wHapMap24.vs3.txt.gz &

~~~
[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2014_Height_publicrelease_HapMapCeuFreq.ForAnalysis.wHapMap21.vs3.txt.gz | awk '{ print $9 }' | grep NA | wc
65518   65518  196554
[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2014_Height_publicrelease_HapMapCeuFreq.ForAnalysis.wHapMap22.vs3.txt.gz | awk '{ print $9 }' | grep NA | wc
40014   40014  120042
[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2014_Height_publicrelease_HapMapCeuFreq.ForAnalysis.wHapMap23.vs3.txt.gz | awk '{ print $9 }' | grep NA | wc
2645    2645    7935
[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2014_Height_publicrelease_HapMapCeuFreq.ForAnalysis.wHapMap24.vs3.txt.gz | awk '{ print $9 }' | grep NA | wc
4187    4187   12561
[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2014_Height_publicrelease_HapMapCeuFreq.ForAnalysis.wHapMap21.vs3.txt.gz | perl -lane 'my $maf1 = sprintf("%.3f", $F[3]); my $maf2 = sprintf("%.3f", ((split(/_/, $F[8]))[3])); if ($maf1 > .5) { $maf1 = sprintf("%.3f", 1 - $maf1); } if ($maf2 > .5) { $maf2 = sprintf("%.3f", 1 - $maf2); } if ($maf1 eq $maf2) { print "match"; } else { print "nope"; }' | sort | uniq -c
2408241 match
142618 nope
[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2014_Height_publicrelease_HapMapCeuFreq.ForAnalysis.wHapMap22.vs3.txt.gz | perl -lane 'my $maf1 = sprintf("%.3f", $F[3]); my $maf2 = sprintf("%.3f", ((split(/_/, $F[8]))[3])); if ($maf1 > .5) { $maf1 = sprintf("%.3f", 1 - $maf1); } if ($maf2 > .5) { $maf2 = sprintf("%.3f", 1 - $maf2); } if ($maf1 eq $maf2) { print "match"; } else { print "nope"; }' | sort | uniq -c
2488302 match
62557 nope
[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2014_Height_publicrelease_HapMapCeuFreq.ForAnalysis.wHapMap23.vs3.txt.gz | perl -lane 'my $maf1 = sprintf("%.3f", $F[3]); my $maf2 = sprintf("%.3f", ((split(/_/, $F[8]))[3])); if ($maf1 > .5) { $maf1 = sprintf("%.3f", 1 - $maf1); } if ($maf2 > .5) { $maf2 = sprintf("%.3f", 1 - $maf2); } if ($maf1 eq $maf2) { print "match"; } else { print "nope"; }' | sort | uniq -c
2538038 match
12821 nope
[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/GIANT2014_5]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2014_Height_publicrelease_HapMapCeuFreq.ForAnalysis.wHapMap24.vs3.txt.gz | perl -lane 'my $maf1 = sprintf("%.3f", $F[3]); my $maf2 = sprintf("%.3f", ((split(/_/, $F[8]))[3])); if ($maf1 > .5) { $maf1 = sprintf("%.3f", 1 - $maf1); } if ($maf2 > .5) { $maf2 = sprintf("%.3f", 1 - $maf2); } if ($maf1 eq $maf2) { print "match"; } else { print "nope"; }' | sort | uniq -c
2546669 match
4190 nope
~~~

#20150903 Note -- Going with HapMap23 this round since drop in 'NA' is pretty apparent and stays in same temporal ballpark as HapMap22 (which I'm using for the first round of GIANT datasets)


python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2014_Height_publicrelease_HapMapCeuFreq.ForAnalysis.txt.gz | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2014_Height_publicrelease_HapMapCeuFreq.ForAnalysis.wHapMap23.vs3.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_BMI.ForAnalysis.txt.gz | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_BMI.ForAnalysis.wHapMap23.vs3.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.ForAnalysis.HeaderFix.txt.gz | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.ForAnalysis.HeaderFix.wHapMap23.vs3.txt.gz &

~~~
[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/GIANT2014_5]$for i in `ls -lrt | awk '{ print $9 }' | grep wHapMap23.vs3.txt.gz`; do echo $i; paste -d @ <(zcat $i | awk '{ print $9 }' | grep NA | wc) <(zcat $i | awk '{ print $9 }' | grep , | wc) <(zcat $i | awk '{ print $9 }' | perl -lane 'my @vals1 = split(/,/, $F[0]); print scalar(@vals1);' | sort | uniq -c) <(zcat $i | awk '{ print $4 }' | perl -lane 'if ($F[0] !~ m/0\.0?[1-9]+/) { print $F[0]; }' | sort | uniq -c) | sed 's/@/ @ /g'| column -s $'@' -t ; done
GIANT_2014_Height_publicrelease_HapMapCeuFreq.ForAnalysis.wHapMap23.vs3.txt.gz
2645    2645    7935          0       0       0    2550859 1       2396 0
									1 0.004
									38635 0.008
									8558 0.009
									1590 1
									1 Freq.Allele1.HapMapCEU
3577 NA
GIANT_2015_WHRadjBMI_COMBINED_EUR.ForAnalysis.HeaderFix.wHapMap23.vs3.txt.gz
3716    3716   11148          0       0       0    2542432 1      29540 0.00829999999999997
									307 0.0083
									9918 0.008333
									2595 0.008475
									1485 0.008621
									699 0.008772
									436 0.008929
									317 0.009091
									258 0.009259
									1 FreqAllele1HapMapCEU
									54824 NA
GIANT_2015_BMI.ForAnalysis.wHapMap23.vs3.txt.gz
6129    6129   18387          0       0       0    2554638 1      29717 0.00829999999999997
									307 0.0083
									15299 0.008333
									3831 0.008475
									2119 0.008621
									995 0.008772
									645 0.008929
									470 0.009091
									360 0.009259
									12 1
									1 Freq1.Hapmap
									54674 NA
~~~

zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2014_Height_publicrelease_HapMapCeuFreq.ForAnalysis.wHapMap23.vs3.txt.gz | perl -lane 'my @vals1 = split(/_/, $F[8]); push(@F, @vals1); print join("\t", @F);' | awk '{ print $10, "\t", $11, "\t", $1, "\t", $4, "\t", $7, "\t", $8 }' | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2014_Height_publicrelease_HapMapCeuFreq.ForAnalysis.wHapMap23.vs3.formatted.txt.gz  
zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_BMI.ForAnalysis.wHapMap23.vs3.txt.gz | perl -lane 'my @vals1 = split(/_/, $F[8]); push(@F, @vals1); print join("\t", @F);' | awk '{ print $10, "\t", $11, "\t", $1, "\t", $4, "\t", $7, "\t", $8 }' | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_BMI.ForAnalysis.wHapMap23.vs3.formatted.txt.gz
zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.ForAnalysis.HeaderFix.wHapMap23.vs3.txt.gz | perl -lane 'my @vals1 = split(/_/, $F[8]); push(@F, @vals1); print join("\t", @F);' | awk '{ print $10, "\t", $11, "\t", $1, "\t", $4, "\t", $7, "\t", $8 }' | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.ForAnalysis.HeaderFix.wHapMap23.vs3.formatted.txt.gz




#20151008
#CSHL Poster work

#/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GLC.MTedits.For2013.vs3.ForCSHLPoster.R
cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GLC.MTedits.ForGIANT2014_5.Orig3.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GLC.MTedits.ForGIANT2014_5.Orig3.vs1.ForCSHLPoster.R 

scp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GLC.MTedits.ForGIANT2014_5.Orig3.vs1.ForCSHLPoster.ModelsVslbfavNewOld.vs1.jpeg mturchin20@wolfy.uchicago.edu:/Users/mturchin20/LabStuff/StephensLab/Multivariate/GIANT2014_5/Vs1/. 




#20160221
#Correcting sign issue -- quick hack to get results

#NOTE -- flipping sign on p-value to make coding downstream easier for this, ie I'm not implying something by creating negative p-values here. This is just for a quick run through.

~~~
[  mturchin20@spudhead  /mnt/gluster/data/external_public_supp/HaemgenRBC2012]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | perl -lane 'if ($F[4] < 0) { ($F[1], $F[2]) = ($F[2], $F[1]); $F[4] = $F[4] * -1; } print join("\t", @F);' | head -n 10
MarkerName      Allele1 Allele2 Freq.Allele1.HapMapCEU  b       SE      p       N       ChrBP
rs4747841       G       A    0.551      0.0011  0.0029  0.70    253213  10_10000134
rs4749917       T       C    0.436      0.0011  0.0029  0.70    253213  10_10000264
rs737656        G       A    0.367      0.0062  0.0030  0.042   253116  10_100002728
rs737657        G       A    0.358      0.0062  0.0030  0.041   252156  10_100002879
rs7086391       C       T    0.12       0.0087  0.0038  0.024   248425  10_100003552
rs878177        T       C    0.3        0.014   0.0031  8.2e-06 251271  10_100003804
rs878178        A       T    0.644      0.0067  0.0031  0.029   253086  10_100003966
rs12219605      T       G    0.427      0.0011  0.0029  0.70    253213  10_10000458
rs3763688       G       C    0.144      0.0022  0.0045  0.62    253056  10_100005552
[  mturchin20@spudhead  /mnt/gluster/data/external_public_supp/HaemgenRBC2012]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | head -n 10
MarkerName      Allele1 Allele2 Freq.Allele1.HapMapCEU  b       SE      p       N       ChrBP
rs4747841       A       G    0.551      -0.0011 0.0029  0.70    253213  10_10000134
rs4749917       T       C    0.436      0.0011  0.0029  0.70    253213  10_10000264
rs737656        A       G    0.367      -0.0062 0.0030  0.042   253116  10_100002728
rs737657        A       G    0.358      -0.0062 0.0030  0.041   252156  10_100002879
rs7086391       T       C    0.12       -0.0087 0.0038  0.024   248425  10_100003552
rs878177        T       C    0.3        0.014   0.0031  8.2e-06 251271  10_100003804
rs878178        A       T    0.644      0.0067  0.0031  0.029   253086  10_100003966
rs12219605      T       G    0.427      0.0011  0.0029  0.70    253213  10_10000458
rs3763688       C       G    0.144      -0.0022 0.0045  0.62    253056  10_100005552
[  mturchin20@spudhead  /mnt/gluster/data/external_public_supp/HaemgenRBC2012]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | perl -lane 'if ($F[4] < 0) { ($F[1], $F[2]) = ($F[2], $F[1]); $F[4] = $F[4] * -1; } print join("\t", @F);' | awk '{ if ($5 < 0) { print $0 } } '
[  mturchin20@spudhead  /mnt/gluster/data/external_public_supp/HaemgenRBC2012]$
~~~

zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | perl -lane 'if ($F[4] < 0) { ($F[1], $F[2]) = ($F[2], $F[1]); $F[3] = 1 - $F[3]; $F[4] = $F[4] * -1; } print join("\t", @F);' | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.IncAllele.txt.gz

join <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.IncAllele.txt.gz | awk '{ print $1, "\t", $2 }' | sort) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.gz | perl -lane 'if ($F[4] < 0) { ($F[1], $F[2]) = ($F[2], $F[1]); $F[3] = 1 - $F[3]; $F[4] = $F[4] * -1; } print join("\t", @F);' | sort) | perl -lane 'if ($F[1] ne $F[2]) { $F[7] = -1 * $F[7]; } splice(@F, 1, 1); print join("\t", @F);' | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.MatchedToHeight.IncAllele.gz

#Included the following header into /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.MatchedToHeight.IncAllele.gz
#SNP     A1      A2      Freq1.Hapmap    b       se      p       N       ChrBP

join <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.IncAllele.txt.gz | awk '{ print $1, "\t", $2 }' | sort) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.txt.gz | perl -lane 'if ($F[4] < 0) { ($F[1], $F[2]) = ($F[2], $F[1]); $F[3] = 1 - $F[3]; $F[4] = $F[4] * -1; } print join("\t", @F);' | sort) | perl -lane 'if ($F[1] ne $F[2]) { $F[7] = -1 * $F[7]; } splice(@F, 1, 1); print join("\t", @F);' | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.txt.gz

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/DataOrganization.GIANT2014_5.PythonVLookUp.CreateFullAnnotFile.vs1.py /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/DataOrganization.GIANT2014_5.PythonVLookUp.CreateFullAnnotFile.vs1.SignCrrct.vs1.py

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/DataOrganization.GIANT2014_5.PythonVLookUp.CreateFullAnnotFile.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt --file2 /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.IncAllele.txt.gz,/mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.MatchedToHeight.IncAllele.gz,/mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.txt.gz | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5_Orig3_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.SignCrrct.vs1.RAF.txt.gz

~~~
[  mturchin20@spudling12  ~]$python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/DataOrganization.GIANT2014_5.PythonVLookUp.CreateFullAnnotFile.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt --file2 /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.IncAllele.txt.gz,/mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.MatchedToHeight.IncAllele.gz,/mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.txt.gz | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5_Orig3_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.SignCrrct.vs1.RAF.txt.gz
Error2a -- line2[0] present more than once in a particular file (count: 1 line2[0]: 1: rs11576885,C,G,0.5917,0.0014,0.0042,0.7389,219514,1_204737275)
Error2a -- line2[0] present more than once in a particular file (count: 1 line2[0]: 1: rs11771665,A,G,0.85,0.0043,0.0056,0.4426,222149,7_86704151)
Error2a -- line2[0] present more than once in a particular file (count: 1 line2[0]: 1: rs13598,A,C,0.8417,0.003,0.0096,0.7547,69744.9,15_46489749)
Error2a -- line2[0] present more than once in a particular file (count: 1 line2[0]: 1: rs2160421,A,T,0.5583,0.001,0.0037,-0.787,233968,15_34746878)
Error2a -- line2[0] present more than once in a particular file (count: 1 line2[0]: 1: rs372268,A,G,0.975,0.0113,0.0214,-0.5975,150660,4_139504797)
Error2a -- line2[0] present more than once in a particular file (count: 1 line2[0]: 1: rs3752976,G,A,0.95,0.0071,0.0089,-0.425,224029,1_199381979)
Error2a -- line2[0] present more than once in a particular file (count: 1 line2[0]: 1: rs3780697,T,C,0.8167,0.0035,0.0048,0.4659,233977,9_131618930)
Error2a -- line2[0] present more than once in a particular file (count: 1 line2[0]: 1: rs3819262,T,G,0.7167,0.0051,0.004,-0.2023,233953,21_37382759)
Error2a -- line2[0] present more than once in a particular file (count: 1 line2[0]: 1: rs573071,G,A,0.325,0.0042,0.004,0.2937,232167,9_246993)
Error2a -- line2[0] present more than once in a particular file (count: 1 line2[0]: 1: rs627141,T,C,0.2833,0,0.0051,1,229425,18_8332476)
Error2a -- line2[0] present more than once in a particular file (count: 1 line2[0]: 1: rs7591686,A,G,0.7333,0.001,0.0043,-0.8161,233849,2_45609171)
Error2a -- line2[0] present more than once in a particular file (count: 1 line2[0]: 1: rs969206,A,G,0.6083,0.005,0.0037,0.1766,234003,9_74502957)
~~~

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/process.MTedits.ForGIANT2014_5.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/process.MTedits.ForGIANT2014_5.vs1.SignCrrct.vs1.R

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/DataOrganization.GIANT2010.PythonVLookUp.PutTogetherProcessAndAnnotFiles.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5.Orig3.dtlesssignif.vs1.SignCrrct.vs1.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5_Orig3_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.SignCrrct.vs1.RAF.txt.gz | perl -lane 'my $chr; my $pos; if (($F[3] eq "NA") || ($F[3] eq "None")) { $chr = "NA"; $pos = "NA"; } else { $chr = ((split(/_/, $F[3]))[0]); $pos = ((split(/_/, $F[3]))[1]); } splice(@F, 1, 0, ($chr, $pos)); splice(@F, 8, 1); splice(@F, 11, 1); splice(@F, 14, 1); print join("\t", @F);' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5.Orig3.dtlesssignif.vs1.SignCrrct.vs1.annot.vs1.RAF.txt.gz 

#Added the following header to /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5.Orig3.dtlesssignif.vs1.SignCrrct.vs1.annot.vs1.RAF.txt.gz
#snp chr pos a1 a2 chrbp raf annot beta_height se_height n_height beta_BMI se_BMI n_BMI beta_WHRadjBMI se_WHRadjBMI n_WHRadjBMI Z.height Z.BMI Z.WHRadjBMI mvstat mvp unip

zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5.Orig3.dtlesssignif.vs1.SignCrrct.vs1.annot.vs1.RAF.txt.gz | perl -lane 'if ($F[6] > .5) { $F[6] = 1 - $F[6]; $F[17] = $F[17] * -1; $F[18] = $F[18] * -1; $F[19] = $F[19] * -1; } print join("\t", @F);' | sed 's/raf/maf/g' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5.Orig3.dtlesssignif.vs1.SignCrrct.vs1.annot.vs1.MAF.txt.gz

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GLC.MTedits.ForGIANT2014_5.Orig3.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GLC.MTedits.ForGIANT2014_5.Orig3.vs1.SignCrrct.vs1.R










#20161222
#Vs2 -- Running things through version of bmass() code

mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2
cd /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2

#Dropping SNPs with multiple GenomeBrowser SNP location hits via grep -v ,
~~~
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.gz | grep 126943397
rs17212015      A       T       0.8583  0.0094  0.0057  0.09912 232876  X_126943397,13_81046747
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep 107486155                                                                                                                                                                                                      rs10884221      T       C       0.808   -0.0024 0.0035  0.48    249684  10_107486155,6_cox_hap1_48402                                                                                                                          
~~~

#20161227 NOTE -- I was doing `...if ($F[5] == "-")...` which was causing everything to get flipped automatically, when I should have been doing `...if ($F[5] eq "-")...`. This is where the BMI/WHRadjBMI-corr-of-.6 results were coming from. So edited things below to fix  
##zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | sed 's/_/ /g' | awk '{ print $9 "\t" $10 "\t" $4 "\t" $2 "\t" $3 "\t" $5 "\t" $7 "\t" $8 }' | perl -lane 'if ($F[5] > 0) { $F[5] = "+"; } elsif ($F[5] < 0) { $F[5] = "-"; } else { $F[5] = 0;} if ($F[5] == "-") { ($F[3], $F[4]) = ($F[4], $F[3]); $F[5] = "+"; $F[2] = 1 - $F[2]; } print join("\t", @F); ' | grep -v Freq | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | grep -v , | grep -v NA | grep -v cox | grep -v ChrBP | perl -lane 'if ($F[0] eq "X") { $F[0] = 23; } if ($F[0] eq "Y") { $F[0] = 24; } print join("\t", @F); ' | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.IncAllele.bmass.formatted.txt.gz  
##zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.gz | sed 's/_/ /g' | awk '{ print $9 "\t" $10 "\t" $4 "\t" $2 "\t" $3 "\t" $5 "\t" $7 "\t" $8 }' | perl -lane 'if ($F[5] > 0) { $F[5] = "+"; } elsif ($F[5] < 0) { $F[5] = "-"; } else { $F[5] = 0;} if ($F[5] == "-") { ($F[3], $F[4]) = ($F[4], $F[3]); $F[5] = "+"; $F[2] = 1 - $F[2]; } print join("\t", @F); ' | grep -v Freq | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | grep -v , | grep -v NA | grep -v cox | perl -lane 'if ($F[0] eq "X") { $F[0] = 23; } if ($F[0] eq "Y") { $F[0] = 24; } print join("\t", @F); ' | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.IncAllele.bmass.formatted.txt.gz
##zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.txt.gz | sed 's/_/ /g' | awk '{ print $9 "\t" $10 "\t" $4 "\t" $2 "\t" $3 "\t" $5 "\t" $7 "\t" $8 }' | perl -lane 'if ($F[5] > 0) { $F[5] = "+"; } elsif ($F[5] < 0) { $F[5] = "-"; } else { $F[5] = 0;} if ($F[5] == "-") { ($F[3], $F[4]) = ($F[4], $F[3]); $F[5] = "+"; $F[2] = 1 - $F[2]; } print join("\t", @F); ' | grep -v Freq | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | grep -v , | grep -v NA | grep -v cox | perl -lane 'if ($F[0] eq "X") { $F[0] = 23; } if ($F[0] eq "Y") { $F[0] = 24; } print join("\t", @F); ' | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.IncAllele.bmass.formatted.txt.gz
zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | sed 's/_/ /g' | awk '{ print $9 "\t" $10 "\t" $4 "\t" $2 "\t" $3 "\t" $5 "\t" $7 "\t" $8 }' | perl -lane 'if ($F[5] > 0) { $F[5] = "+"; } elsif ($F[5] < 0) { $F[5] = "-"; } else { $F[5] = 0;} if ($F[5] eq "-") { ($F[3], $F[4]) = ($F[4], $F[3]); $F[5] = "+"; $F[2] = 1 - $F[2]; } print join("\t", @F); ' | grep -v Freq | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | grep -v , | grep -v NA | grep -v cox | grep -v ChrBP | perl -lane 'if ($F[0] eq "X") { $F[0] = 23; } if ($F[0] eq "Y") { $F[0] = 24; } print join("\t", @F); ' | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.IncAllele.bmass.formatted.txt.gz  
zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.gz | sed 's/_/ /g' | awk '{ print $9 "\t" $10 "\t" $4 "\t" $2 "\t" $3 "\t" $5 "\t" $7 "\t" $8 }' | perl -lane 'if ($F[5] > 0) { $F[5] = "+"; } elsif ($F[5] < 0) { $F[5] = "-"; } else { $F[5] = 0;} if ($F[5] eq "-") { ($F[3], $F[4]) = ($F[4], $F[3]); $F[5] = "+"; $F[2] = 1 - $F[2]; } print join("\t", @F); ' | grep -v Freq | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | grep -v , | grep -v NA | grep -v cox | perl -lane 'if ($F[0] eq "X") { $F[0] = 23; } if ($F[0] eq "Y") { $F[0] = 24; } print join("\t", @F); ' | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.IncAllele.bmass.formatted.txt.gz
zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.txt.gz | sed 's/_/ /g' | awk '{ print $9 "\t" $10 "\t" $4 "\t" $2 "\t" $3 "\t" $5 "\t" $7 "\t" $8 }' | perl -lane 'if ($F[5] > 0) { $F[5] = "+"; } elsif ($F[5] < 0) { $F[5] = "-"; } else { $F[5] = 0;} if ($F[5] eq "-") { ($F[3], $F[4]) = ($F[4], $F[3]); $F[5] = "+"; $F[2] = 1 - $F[2]; } print join("\t", @F); ' | grep -v Freq | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | grep -v , | grep -v NA | grep -v cox | perl -lane 'if ($F[0] eq "X") { $F[0] = 23; } if ($F[0] eq "Y") { $F[0] = 24; } print join("\t", @F); ' | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.IncAllele.bmass.formatted.txt.gz

#Chr BP  MAF     A1      A2      Direction       pValue  N
#ExpectedColumnNames=c("Chr", "BP", "MAF", "Direction", "pValue", "N")

join <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.IncAllele.bmass.formatted.txt.gz | awk '{ print $1 "_" $2 "\t" $4 }' | grep -v NA | sort -g -k 1,1) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.IncAllele.bmass.formatted.txt.gz | awk '{ print $1 "_" $2 "\t" $0 }' | grep -v NA | sort -g -k 1,1) | perl -lane 'if ($F[1] eq $F[6]) { ($F[5], $F[6]) = ($F[6], $F[5]); $F[4] = 1 - $F[4]; if ($F[7] eq "-") { $F[7] = "+"; } elsif ($F[7] eq "+") { $F[7] = "-"; } else { $PH = 1; } } elsif ($F[1] eq $F[5]) { $PH = 1; } else { push(@F, "NoAlleleMatch"); } print join("\t", @F[2..$#F]);' | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.IncAllele.bmass.formatted.MatchedToHeight.txt.gz
join <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.IncAllele.bmass.formatted.txt.gz | awk '{ print $1 "_" $2 "\t" $4 }' | grep -v NA | sort -g -k 1,1) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.IncAllele.bmass.formatted.txt.gz | awk '{ print $1 "_" $2 "\t" $0 }' | grep -v NA | sort -g -k 1,1) | perl -lane 'if ($F[1] eq $F[6]) { ($F[5], $F[6]) = ($F[6], $F[5]); $F[4] = 1 - $F[4]; if ($F[7] eq "-") { $F[7] = "+"; } elsif ($F[7] eq "+") { $F[7] = "-"; } else { $PH = 1; } } elsif ($F[1] eq $F[5]) { $PH = 1; } else { push(@F, "NoAlleleMatch"); } print join("\t", @F[2..$#F]);' | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.IncAllele.bmass.formatted.MatchedToHeight.txt.gz

zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.IncAllele.bmass.formatted.MatchedToHeight.txt.gz | grep -v NoAlleleMatch | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.IncAllele.bmass.formatted.MatchedToHeight.DropNoMatch.txt.gz 
zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.IncAllele.bmass.formatted.MatchedToHeight.txt.gz | grep -v NoAlleleMatch | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.IncAllele.bmass.formatted.MatchedToHeight.DropNoMatch.txt.gz


~~~
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | head -n 10
MarkerName      Allele1 Allele2 Freq.Allele1.HapMapCEU  p       N       ChrBP
rs10 a  c       0.0333  0.8826  78380   7_92221823

#bmass code thing
#ChrBP final SNPs thing

~~~
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.IncAllele.bmass.formatted.txt.gz | head -n 10
Chr BP  MAF     A1      A2      Direction       pValue  N
10 10000134     0.449   G       A       +       0.70    253213
10 10000264     0.564   C       T       +       0.70    253213
10 100002728    0.633   G       A       +       0.042   253116
10 100002879    0.642   G       A       +       0.041   252156
10 100003552    0.88    C       T       +       0.024   248425
10 100003804    0.7     C       T       +       8.2e-06 251271
10 100003966    0.356   T       A       +       0.029   253086
10 10000458     0.573   G       T       +       0.70    253213
10 100005552    0.856   G       C       +       0.62    253056
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.IncAllele.bmass.formatted.txt.gz | wc
2535924 20287392 91982463
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.IncAllele.bmass.formatted.txt.gz | perl -lane 'print $#F;' | sort | uniq -c
2535924 7
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.IncAllele.bmass.formatted.MatchedToHeight.DropNoMatch.txt.gz | head -n 10
Chr BP  MAF     A1      A2      Direction       pValue  N
1  100002698    0.7     C       T       +       0.1273  233918
1  100002784    0.93333 C       T       +       0.1885  234025
1  100003454    1       C       T       +       0.2486  217165
1  100006954    0.93333 C       A       +       0.2797  222182
1  100007330    0.7167  G       A       +       0.09721 233809
1  100007518    0.0625  T       A       -       0.249   233266
1  100007855    1       C       T       +       0.2301  222236
1  100008013    0.2167  C       T       -       0.3633  233837
1  100008024    1       C       T       +       0.3284  234002
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.IncAllele.bmass.formatted.MatchedToHeight.DropNoMatch.txt.gz | wc
2534376 20275008 97022409
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.IncAllele.bmass.formatted.MatchedToHeight.DropNoMatch.txt.gz | perl -lane 'print $#F;' | sort | uniq -c
2534376 7
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.IncAllele.bmass.formatted.MatchedToHeight.DropNoMatch.txt.gz | head -n 10
Chr BP  MAF     A1      A2      Direction       pValue  N
1  100002698    0.7     C       T       +       0.93    142700
1  100002784    0.93333 C       T       +       0.56    142749
1  100003454    1       C       T       +       0.63    127489
1  100006954    0.93333 C       A       +       0.44    130999
1  100007330    0.7167  G       A       +       0.63    142695
1  100007518    0.0625  T       A       -       0.53    142316
1  100007855    1       C       T       +       0.38    131011
1  100008013    0.2167  C       T       -       0.97    142741
1  100008024    1       C       T       +       0.97    142729
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.IncAllele.bmass.formatted.MatchedToHeight.DropNoMatch.txt.gz | wc
2494521 19956168 90436894
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.IncAllele.bmass.formatted.MatchedToHeight.DropNoMatch.txt.gz | perl -lane 'print $#F;' | sort | uniq -c
2494521 7
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.IncAllele.bmass.formatted.txt.gz | grep cox | head -n 10              
6  cox  0.717   C       T       +       0.024   251823
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.IncAllele.bmass.formatted.txt.gz | grep ChrBP           
231091  ChrBP   1       A       A-2097957       +       0.0033  0.00040
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.IncAllele.bmass.formatted.txt.gz | awk '{ print $2 }' | sort -g | head -n 5
BP
ChrBP
cox
1522
1542
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.IncAllele.bmass.formatted.MatchedToHeight.DropNoMatch.txt.gz | awk '{ print $2 }' | sort -g | head -n 5
BP
cox
1522
1542
2182
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.IncAllele.bmass.formatted.MatchedToHeight.DropNoMatch.txt.gz | awk '{ print $2 }' | sort -g | head -n 5
BP
cox
1522
1542
2182
.
.
.
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.AllPheno.MarkerChrBP.txt | head -n 10
rs425277         1       2059032
rs9434723        1       9214869
rs10779751       1       11206923
rs2284746        1       17179262
rs12137162       1       19635983
rs212524         1       21455898
rs1014987        1       22371411
rs2806561        1       23377382
rs4601530        1       24916698
rs926438         1       25626225
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep rs2806561
rs2806561       A       G       0.567   0.027   0.0029  1.9e-20 253049  1_23377381
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep rs425277
rs425277        T       C       0.28    0.028   0.0033  1.2e-17 252521  1_2059031
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep rs12137162
rs12137162      A       C       0.271   0.019   0.0032  4.1e-09 253175  1_19635982
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.AllPheno.MarkerChrBP.txt | head -n 150 | tail -n 10
rs1658351        3       57988613
rs6794009        3       61488535
rs17806888       3       67499012
rs2175513        3       68705056
rs12330322       3       72538045
rs7633464        3       100198513
rs9825951        3       100752611
rs1797625        3       114309105
rs1533269        3       115697301
rs1546391        3       116180147
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep rs17806888
rs17806888      T       C       0.908   0.034   0.0048  2.9e-12 245828  3_67499011
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep rs1533269 
rs1533269       A       C       0.317   -0.020  0.0033  1.1e-09 250818  3_115697300
.
.
.
#>         bmassOutput1[c("MergedDataSources", "LogFile")] <- AnnotateMergedDataWithGWASSNPs(bmassOutput1$MergedDataSources, GWASsnps, GWASsnps_AnnotateWindow, bmassOutput1$LogFile)[c("MergedDataSources", "LogFile")]
#> table(bmassOutput1$MergedDataSources$GWASannot)
#
#      0       1       2
#1942212     873  546528
#> write.table(bmassOutput1$MergedDataSources[bmassOutput1$MergedDataSources$GWASannot == 1,3]+1, file="blahblah.BP.txt")
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join -v 2 <(cat ../../../../blahblah.BP.txt | awk '{ print $2 }' | sort) <(cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.AllPheno.MarkerChrBP.txt | awk '{ print $3 "\t" $2 "\t" $1; }' | sort -k 1,1)
127496586 6 rs72959041
28859706 6 rs1233627
29912144 6 rs9404952
31214438 6 rs1265097
31380240 6 rs6457374
31696363 6 rs2857693
32489714 6 rs7759742
33216265 6 rs3129254
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep rs1265097
rs12650977      A       C       0.025   0.0043  0.014   0.75    228549  4_122462644
rs12650975      T       G       0.991   -0.00050        0.016   0.98    187678  4_148841903
rs1265097       A       C       0.075   -0.059  0.0050  1.7e-32 236690  6_31214437,6_cox_hap1_2557687,6_qbl_hap2_2356251
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep rs7759742
rs7759742       A       T       0.525   -0.015  0.0030  5.7e-07 249717  6_32489713,6_cox_hap1_3788644,6_qbl_hap2_3596549
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep rs72959041
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.gz | grep rs72959041
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat ../../../../blahblah.BP.txt | wc
    855    1709   12910
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join -v 2 <(cat ../../../../blahblah.BP.txt | awk '{ print $2 }' | sort) <(cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt | awk '{ print $3 "\t" $2 "\t" $1; }' | sort -k 1,1) | wc
      8      24     170
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join -v 2 <(cat ../../../../blahblah.BP.txt | awk '{ print $2 }' | sort) <(cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt | awk '{ print $3 "\t" $2 "\t" $1; }' | sort -k 1,1)
127496586 6 rs72959041
28859706 6 rs1233627
29912144 6 rs9404952
31214438 6 rs1265097
31380240 6 rs6457374
31696363 6 rs2857693
32489714 6 rs7759742
33216265 6 rs3129254
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHR_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.ColRightFormat.txt.gz | grep rs72959041
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.AllPheno.MarkerChrBP.txt | wc
    881    2643   22310
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt | sort -k 1,1) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | perl -lane 'print $F[0], "\t", $F[$#F];' | sort -k 1,1) | wc
    861    3444   29026
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt | wc
    862    2586   21825
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt | sort -k 1,1) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | perl -lane 'print $F[0], "\t", $F[$#F];' | sort -k 1,1) | head -n 10
rs1000940 17 5223976 17_5223975
rs1004202 5 36101220 5_36101219
rs10048625 2 1754655 2_1754654
rs10059761 5 67191839 5_67191838
rs1007358 7 46167880 7_46167879
rs10083886 17 67434950 17_67434949
rs10119624 9 117345259 9_117345258
rs10131337 14 36214267 14_36214266
rs10132280 14 24998019 14_24998018
rs10140101 14 74108442 14_74108441
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt | sort -k 1,1) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | perl -lane 'print $F[0], "\t", $F[$#F];' | sort -k 1,1) | grep ,    
rs1233627 6 28859706 6_28859705,6_cox_hap1_211260,6_qbl_hap2_9083
rs1265097 6 31214438 6_31214437,6_cox_hap1_2557687,6_qbl_hap2_2356251
rs2857693 6 31696363 6_31696362,6_cox_hap1_3034507,6_qbl_hap2_2835971
rs3129254 6 33216265 6_33216264,6_cox_hap1_4488425,6_qbl_hap2_4294418
rs6457374 6 31380240 6_31380239,6_cox_hap1_2722003,6_qbl_hap2_2519738
rs7759742 6 32489714 6_32489713,6_cox_hap1_3788644,6_qbl_hap2_3596549
rs9404952 6 29912144 6_29912143,6_cox_hap1_1256589,6_qbl_hap2_1058722
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt | sort -k 1,1) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | perl -lane 'print $F[0], "\t", $F[$#F];' | sort -k 1,1) | grep , | perl -lane '$F[$#F] = ((split(/,/, $F[$#F]))[0]); print join("\t", @F);'
rs1233627       6       28859706        6_28859705
rs1265097       6       31214438        6_31214437
rs2857693       6       31696363        6_31696362
rs3129254       6       33216265        6_33216264
rs6457374       6       31380240        6_31380239
rs7759742       6       32489714        6_32489713
rs9404952       6       29912144        6_29912143
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt | sort -k 1,1) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | perl -lane 'print $F[0], "\t", $F[$#F];' | sort -k 1,1) | grep , | perl -lane '$F[$#F] = ((split(/,/, $F[$#F]))[0]); print join("\t", @F);' | sed 's/_/ /g' | awk '{ print $1 "\t" $4 "\t" $5 }' 
rs1233627       6       28859705
rs1265097       6       31214437
rs2857693       6       31696362
rs3129254       6       33216264
rs6457374       6       31380239
rs7759742       6       32489713
rs9404952       6       29912143
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt | sort -k 1,1) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | perl -lane 'print $F[0], "\t", $F[$#F];' | sort -k 1,1) | perl -lane '$F[$#F] = ((split(/,/, $F[$#F]))[0]); print join("\t", @F);' | sed 's/_/ /g' | awk '{ print $1 "\t" $4 "\t" $5 }' | head -n 10
rs1000940       17      5223975
rs1004202       5       36101219
rs10048625      2       1754654
rs10059761      5       67191838
rs1007358       7       46167879
rs10083886      17      67434949
rs10119624      9       117345258
rs10131337      14      36214266
rs10132280      14      24998018
rs10140101      14      74108441
~~~

cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/ng.3097-S2.noCarriageReturn.noQuotes.noCommas.rsIDs.MarkerChrBP.txt /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/Locke2015.SupplTables1_2.wManualOCREdits.rsIDs.MarkerChrBP.txt /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/nature14132-s2.ST4.noCarriageReturn.rsIDs.MarkerChrBP.txt > /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMI.MarkerChrBP.txt

#20161226 NOTE -- It looks like the ChrBP coordinates given in the manuscripts are off-by-1, at least compared to what the dbSNP130 values present for the same rsIDs. I'm going to go with the dbSNP130 values and subtract 1 from all BP values of the 'GWASsnps' to make the matches okay. Matches were done using both rsID# and ChrBP in the python script used in the first runs of things, hence why they 'matched' previous but then seemed to 'not match' now
#This is where the `GWASsnps[,2] <- GWASsnps[,2] - 1` type calls are now coming from

#20161226 NOTE -- A few SNPs get dropped out of 'MergedDataSources' because they have multiple entries for ChrBP in dbSNP130, so going to a direct grab of ChrBP based on marker from the Wood et al. 2014 dataset just to be complete and easy here
#Also it looks like rs72959041 wasn't even found in the original runs since there were only '880' SNPs with '1' annotations, and there's no header in /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.AllPheno.MarkerChrBP.txt so the total count was 881.
#I realized as I was doing this that this wouldn't change anything? The SNP hit list is fine, which is what I'm attempting to change here -- regardless, the 'MergedDataSources' will still be missing these SNPs unless I feel like changing my default QC of 'dropping SNPs with multiple dbSNP130 entries' in the individual GWAS dataset file prep steps

#20161226 NOTE -- Below code/work is to compare the new results to the old results, trying to figure out where the differences are

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5.Orig3.dtlesssignif.vs1.SignCrrct.vs1.annot.vs1.MAF.txt.gz /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/20161217_edits_GIANT2014_5.Orig3.dtlesssignif.vs1.SignCrrct.vs1.annot.vs1.MAF.txt.gz 

#Put below header into /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/20161217_edits_GIANT2014_5.Orig3.dtlesssignif.vs1.SignCrrct.vs1.annot.vs1.MAF.txt.gz
#SNP Chr BP A1 A2 chrbp MAF GWASannot Height_Beta Height_SE Height_N BMI_Beta BMI_SE BMI_N WHRadjBMI_Beta WHRadjBMI_SE WHRadjBMI_N Height_ZScore BMI_ZScore WHRadjBMI_ZScore mvstat mvstat_log10pVal unistat_log10pVal

zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | sed 's/_/ /g' | perl -lane 'my $dir = "+"; if ($F[6] < 0) { $dir = "-"; } print $F[$#F-1], "\t", $F[$#F], "\t", $F[3], "\t", $F[1], "\t", $F[2], "\t", $dir, "\t", abs($F[6]), "\t", $F[7];' | grep -v ChrBP | grep -v NA | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/20161227edits.GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz 
zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.MatchedToHeight.IncAllele.gz | sed 's/_/ /g' | perl -lane 'my $dir = "+"; if ($F[6] < 0) { $dir = "-"; } print $F[$#F-1], "\t", $F[$#F], "\t", $F[3], "\t", $F[1], "\t", $F[2], "\t", $dir, "\t", abs($F[6]), "\t", $F[7];' | grep -v ChrBP | grep -v NA | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/20161227edits.SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.MatchedToHeight.IncAllele.gz 
zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.txt.gz | sed 's/_/ /g' | perl -lane 'my $dir = "+"; if ($F[6] < 0) { $dir = "-"; } print $F[$#F-1], "\t", $F[$#F], "\t", $F[3], "\t", $F[1], "\t", $F[2], "\t", $dir, "\t", abs($F[6]), "\t", $F[7];' | grep -v ChrBP | grep -v NA | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/20161227edits.GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.txt.gz
      
mv ../../../../blahblah1.txt ../../../../GIANT2014_5/Vs2/20161227.Troubleshooting.bmassOutput1.MergedDataSources.txt
mv ../../../../blahblah2.txt ../../../../GIANT2014_5/Vs2/20161227.Troubleshooting.bmassOutput1.1.MergedDataSources.txt
gzip ../../../../GIANT2014_5/Vs2/20161227.Troubleshooting.bmassOutput1.MergedDataSources.txt ../../../../GIANT2014_5/Vs2/20161227.Troubleshooting.bmassOutput1.1.MergedDataSources.txt

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/GIANT2014_5.Vs2.HeightBMIWHRadjBMI.NewSNPs.SNPs.vs1.txt | awk '{ print $1 "\t" $2 "\t" $3 }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/GIANT2014_5.Vs2.HeightBMIWHRadjBMI.NewSNPs.SNPs.vs1.ChrBPChrBP.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/GIANT2014_5.Vs2.HeightBMI.NewSNPs.SNPs.vs1.txt | awk '{ print $1 "\t" $2 "\t" $3 }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/GIANT2014_5.Vs2.HeightBMI.NewSNPs.SNPs.vs1.ChrBPChrBP.txt

#In R
```
library("devtools"); devtools::load_all("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/bmass"); 

Height <- read.table("/mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.IncAllele.bmass.formatted.txt.gz", header=T);
BMI <- read.table("/mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.IncAllele.bmass.formatted.MatchedToHeight.DropNoMatch.txt.gz", header=T);
WHRadjBMI <- read.table("/mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.IncAllele.bmass.formatted.MatchedToHeight.DropNoMatch.txt.gz", header=T);
Height2 <- read.table("/mnt/gluster/data/external_public_supp/GIANT2014_5/20161227edits.GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz", header=T)
BMI2 <- read.table("/mnt/gluster/data/external_public_supp/GIANT2014_5/20161227edits.SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.MatchedToHeight.IncAllele.gz", header=T)
WHRadjBMI2 <- read.table("/mnt/gluster/data/external_public_supp/GIANT2014_5/20161227edits.GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.txt.gz", header=T)

Height <- Height[Height$Direction != 0,]
BMI <- BMI[BMI$Direction != 0,]
WHRadjBMI <- WHRadjBMI[WHRadjBMI$Direction != 0,]

#GWASsnps <- read.table("/mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.AllPheno.MarkerChrBP.txt", header=F);
GWASsnps <- read.table("/mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt", header=F);
GWASsnps <- GWASsnps[,2:3]
GWASsnps[,2] <- GWASsnps[,2] - 1
GWASsnps2 <- read.table("/mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMI.MarkerChrBP.txt", header=F);
GWASsnps2 <- GWASsnps2[,2:3]
GWASsnps2[,2] <- GWASsnps2[,2] - 1
names(GWASsnps) <- c("Chr", "BP");
names(GWASsnps2) <- c("Chr", "BP");

bmassOutput1 <- bmass(c("Height", "BMI", "WHRadjBMI"), GWASsnps=GWASsnps, NminThreshold = 50000, PrintMergedData=TRUE, bmassSeedValue=150);
bmassOutput1.1 <- bmass(c("Height2", "BMI2", "WHRadjBMI2"), GWASsnps=GWASsnps, NminThreshold = 50000, PrintMergedData=TRUE, bmassSeedValue=150);
bmassOutput2 <- bmass(c("Height", "BMI"), GWASsnps=GWASsnps2, NminThreshold = 50000, PrintMergedData=TRUE, bmassSeedValue=150);

#write.table(bmassOutput1$NewSNPs$SNPs, file="/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/GIANT2014_5.Vs2.HeightBMIWHRadjBMI.NewSNPs.SNPs.vs1.txt", quote=FALSE, row.names=FALSE)
#write.table(bmassOutput2$NewSNPs$SNPs, file="/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/GIANT2014_5.Vs2.HeightBMI.NewSNPs.SNPs.vs1.txt", quote=FALSE, row.names=FALSE)
write.table(bmassOutput1$NewSNPs$SNPs, file="/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.txt", quote=FALSE, row.names=FALSE)
write.table(bmassOutput1$MergedDataSources, file="/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/bmass.GIANT2014_5.Vs2.MergedDataSources.vs1.txt", quote=FALSE, row.names=FALSE)

system("gzip /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/bmass.GIANT2014_5.Vs2.MergedDataSources.vs1.txt")




system("gzip 
        
	bmassOutput1.2 <- list()
        bmassOutput1.2$MergedDataSources <- NULL
        if (!is.null(MergedDataSources)) {
                bmassOutput1.2$MergedDataSources <- MergedDataSources
        }
        bmassOutput1.2$ZScoresCorMatrix <- NULL
        bmassOutput1.2$MarginalSNPs <- list()
        bmassOutput1.2$Models <- NULL
        bmassOutput1.2$ModelPriors <- NULL
        bmassOutput1.2$PreviousSNPs <- list()
        bmassOutput1.2$NewSNPs <- list()
        bmassOutput1.2$GWASlogBFMinThreshold <- NULL
        #LogFile <- c()
        bmassOutput1.2$LogFile <- c()

        bmassOutput1.2$LogFile <- rbind(bmassOutput1.2$LogFile, paste(format(Sys.time()), " -- beginning bmass.", sep=""))


system("gzip 
        
	bmassOutput1.2 <- list()
        bmassOutput1.2$MergedDataSources <- NULL
        if (!is.null(MergedDataSources)) {
                bmassOutput1.2$MergedDataSources <- MergedDataSources
        }
        bmassOutput1.2$ZScoresCorMatrix <- NULL
        bmassOutput1.2$MarginalSNPs <- list()
        bmassOutput1.2$Models <- NULL
        bmassOutput1.2$ModelPriors <- NULL
        bmassOutput1.2$PreviousSNPs <- list()
        bmassOutput1.2$NewSNPs <- list()
        bmassOutput1.2$GWASlogBFMinThreshold <- NULL
        #LogFile <- c()
        bmassOutput1.2$LogFile <- c()

        bmassOutput1.2$LogFile <- rbind(bmassOutput1.2$LogFile, paste(format(Sys.time()), " -- beginning bmass.", sep=""))

	bmassOutput1.3 <- bmassOutput1.2
	bmassOutput1.4 <- bmassOutput1.2

#	DataSources = c("Height", "BMI", "WHRadjBMI"); NminThreshold = 50000; bmassSeedValue=150;
#	ExpectedColumnNames=c("Chr", "BP", "MAF", "Direction", "pValue", "N"); GWASsnps_AnnotateWindow = 5e5; SNPMarginalUnivariateThreshold = 1e-6; SNPMarginalMultivariateThreshold = 1e-6; SigmaAlphas = c(0.005,0.0075,0.01,0.015,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.15); ProvidedPriors=NULL; UseFlatPriors=FALSE; PruneMarginalSNPs=TRUE; PruneMarginalSNPs_bpWindow=5e5;
#         
#	bmassOutput$LogFile <- CheckIndividualDataSources(DataSources, GWASsnps, ExpectedColumnNames, SigmaAlphas, bmassOutput$MergedDataSources, ProvidedPriors, UseFlatPriors, PruneMarginalSNPs, PruneMarginalSNPs_bpWindow, SNPMarginalUnivariateThreshold, SNPMarginalMultivariateThreshold, NminThreshold, bmassSeedValue, bmassOutput$LogFile)
#
#        bmassOutput[c("MergedDataSources", "LogFile")] <- MergeDataSources(DataSources, bmassOutput$LogFile)[c("MergedDataSources", "LogFile")]
#
#	bmassOutput1 <- bmassOutput
# 
#        bmassOutput1[c("MergedDataSources", "LogFile")] <- AnnotateMergedDataWithGWASSNPs(bmassOutput1$MergedDataSources, GWASsnps, GWASsnps_AnnotateWindow, bmassOutput1$LogFile)[c("MergedDataSources", "LogFile")]
# 
#        bmassOutput1[c("MarginalSNPs", "ZScoresCorMatrix", "LogFile")] <- ProcessMergedAndAnnotatedDataSources(DataSources, bmassOutput1$MergedDataSources, SNPMarginalUnivariateThreshold, SNPMarginalMultivariateThreshold, bmassOutput1$LogFile)[c("MarginalSNPs", "ZScoresCorMatrix", "LogFile")]
#
##	bmassOutput$MergedDataSources$ChrBP <- paste(bmassOutput$MergedDataSources$Chr, "_", bmassOutput$MergedDataSources$BP, sep="")
##	bmassOutput$MarginalSNPs$SNPs <- bmassOutput$MergedDataSources[bmassOutput$MergedDataSources$mvstat_log10pVal > -log10(SNPMarginalUnivariateThreshold) | bmassOutput$MergedDataSources$unistat_log10pVal > -log10(SNPMarginalMultivariateThreshold),]
#
#        bmassOutput1[c("MarginalSNPs", "Models", "ModelPriors", "LogFile")] <- GetLogBFsFromData(DataSources, bmassOutput1$MarginalSNPs, bmassOutput1$ZScoresCorMatrix, SigmaAlphas, bmassOutput1$LogFile)[c("MarginalSNPs", "Models", "ModelPriors", "LogFile")]
#
##	Nmins <- apply(bmassOutput$MarginalSNPs$SNPs[,c("TC_N", "TG_N", "LDL_N", "HDL_N")], 1, min)
##	ZScores <- bmassOutput$MarginalSNPs$SNPs[,c("TC_ZScore", "TG_ZScore", "HDL_ZScore", "LDL_ZScore")]
##	MarginalSNPs_logBFs <- compute.allBFs.fromZscores(ZScores, bmassOutput$ZScoresCorMatrix, Nmins, bmassOutput$MarginalSNPs$SNPs$MAF, SigmaAlphas)
#
#        bmassOutput1[c("MarginalSNPs", "PreviousSNPs", "ModelPriors", "GWASlogBFMinThreshold", "LogFile")] <- DetermineAndApplyPriors(DataSources, bmassOutput1$MarginalSNPs, GWASsnps, SigmaAlphas, bmassOutput1$Models, bmassOutput1$ModelPriors, ProvidedPriors, UseFlatPriors, bmassSeedValue, bmassOutput1$LogFile)[c("MarginalSNPs", "PreviousSNPs", "ModelPriors", "GWASlogBFMinThreshold", "LogFile")]
#
#        bmassOutput1[c("MarginalSNPs", "PreviousSNPs", "NewSNPs", "LogFile")] <- FinalizeAndFormatResults(DataSources, bmassOutput1$MarginalSNPs, bmassOutput1$PreviousSNPs, GWASsnps, bmassOutput1$GWASlogBFMinThreshold, SigmaAlphas, bmassOutput1$Models, bmassOutput1$ModelPriors, NminThreshold, PruneMarginalSNPs, PruneMarginalSNPs_bpWindow, bmassOutput1$LogFile)[c("MarginalSNPs", "PreviousSNPs", "NewSNPs", "LogFile")]
.
.
.

	bmassOutput1.4$MergedDataSources <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/20161217_edits_GIANT2014_5.Orig3.dtlesssignif.vs1.SignCrrct.vs1.annot.vs1.MAF.txt.gz", header=TRUE) 
	bmassOutput1.4$ZScoresCorMatrix <- as.matrix(read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5.Orig3.RSS0.vs1.SignCrrct.vs1.txt",header=T,sep=",")) 
	#Same as GWASsnps but whatevers
	GWASsnps1.4 <- read.table("/mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt", header=F);
	GWASsnps1.4 <- GWASsnps1.4[,2:3]
	GWASsnps1.4[,2] <- GWASsnps1.4[,2] - 1
	names(GWASsnps1.4) <- c("Chr", "BP");
	
	DataSources1.4 = c("Height", "BMI", "WHRadjBMI"); NminThreshold = 50000; bmassSeedValue=150;
        ExpectedColumnNames=c("Chr", "BP", "MAF", "Direction", "pValue", "N"); GWASsnps_AnnotateWindow = 5e5; SNPMarginalUnivariateThreshold = 1e-6; SNPMarginalMultivariateThreshold = 1e-6; SigmaAlphas = c(0.005,0.0075,0.01,0.015,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.15); ProvidedPriors=NULL; UseFlatPriors=FALSE; PruneMarginalSNPs=TRUE; PruneMarginalSNPs_bpWindow=5e5;
        bmassOutput1.4$MergedDataSources$ChrBP <- paste(bmassOutput1.4$MergedDataSources$Chr, "_", bmassOutput1.4$MergedDataSources$BP, sep="")

        bmassOutput1.4$MarginalSNPs$SNPs <- bmassOutput1.4$MergedDataSources[bmassOutput1.4$MergedDataSources$mvstat_log10pVal > -log10(SNPMarginalUnivariateThreshold) | bmassOutput1.4$MergedDataSources$unistat_log10pVal > -log10(SNPMarginalMultivariateThreshold),]

        bmassOutput1.4[c("MarginalSNPs", "Models", "ModelPriors", "LogFile")] <- GetLogBFsFromData(DataSources1.4, bmassOutput1.4$MarginalSNPs, bmassOutput1.4$ZScoresCorMatrix, SigmaAlphas, bmassOutput1.4$LogFile)[c("MarginalSNPs", "Models", "ModelPriors", "LogFile")]

        bmassOutput1.4[c("MarginalSNPs", "PreviousSNPs", "ModelPriors", "GWASlogBFMinThreshold", "LogFile")] <- DetermineAndApplyPriors(DataSources1.4, bmassOutput1.4$MarginalSNPs, GWASsnps1.4, SigmaAlphas, bmassOutput1.4$Models, bmassOutput1.4$ModelPriors, ProvidedPriors, UseFlatPriors, bmassSeedValue, bmassOutput1.4$LogFile)[c("MarginalSNPs", "PreviousSNPs", "ModelPriors", "GWASlogBFMinThreshold", "LogFile")]

        bmassOutput1.4[c("MarginalSNPs", "PreviousSNPs", "NewSNPs", "LogFile")] <- FinalizeAndFormatResults(DataSources1.4, bmassOutput1.4$MarginalSNPs, bmassOutput1.4$PreviousSNPs, GWASsnps1.4, bmassOutput1.4$GWASlogBFMinThreshold, SigmaAlphas, bmassOutput1.4$Models, bmassOutput1.4$ModelPriors, NminThreshold, PruneMarginalSNPs, PruneMarginalSNPs_bpWindow, bmassOutput1.4$LogFile)[c("MarginalSNPs", "PreviousSNPs", "NewSNPs", "LogFile")]
```

~~~
.
.
.
> summary(bmassOutput1)
                  Length Class      Mode     
MarginalSNPs       0     -none-     list     
PreviousSNPs       0     -none-     list     
NewSNPs            0     -none-     list     
LogFile           10     -none-     character
MergedDataSources 19     data.frame list     
> table(bmassOutput1$MergedDataSources$GWASannot)

      0       2 
1942213  547400 
.
.
.
> head(GWASsnps)
  Chr       BP
1   1  2059032
2   1  9214869
3   1 11206923
4   1 17179262
5   1 19635983
6   1 21455898
> head(GWASsnps[,2] - 1)
[1]  2059031  9214868 11206922 17179261 19635982 21455897
> GWASsnps[,2] <- GWASsnps[,2] - 1
> head(GWASsnps)
  Chr       BP
1   1  2059031
2   1  9214868
3   1 11206922
4   1 17179261
5   1 19635982
6   1 21455897
>         bmassOutput1[c("MergedDataSources", "LogFile")] <- AnnotateMergedDataWithGWASSNPs(bmassOutput1$MergedDataSources, GWASsnps, GWASsnps_AnnotateWindow, bmassOutput1$LogFile)[c("MergedDataSources", "LogFile")]
> table(bmassOutput1$MergedDataSources$GWASannot)

      0       1       2 
1942212     873  546528 
> write.table(bmassOutput1$MergedDataSources[bmassOutput1$MergedDataSources$GWASannot == 1,3]+1, file="blahblah.BP.txt")
> GWASsnps <- read.table("/mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt", header=F);
> GWASsnps <- GWASsnps[,2:3]
> GWASsnps[,2] <- GWASsnps[,2] - 1
> names(GWASsnps) <- c("Chr", "BP");
> head(GWASsnps)
  Chr       BP
1   1  2059031
2   1  9214868
3   1 11206922
4   1 17179261
5   1 19635982
6   1 21455897
>         bmassOutput1[c("MergedDataSources", "LogFile")] <- AnnotateMergedDataWithGWASSNPs(bmassOutput1$MergedDataSources, GWASsnps, GWASsnps_AnnotateWindow, bmassOutput1$LogFile)[c("MergedDataSources", "LogFile")]
> table(bmassOutput1$MergedDataSources$GWASannot)

      0       1       2 
1956477     854  532282 
> write.table(bmassOutput1$MergedDataSources[bmassOutput1$MergedDataSources$GWASannot == 1,3]+1, file="blahblah.BP.txt")

.
.
.
> dim(bmassOutput1$NewSNPs$SNPs)
[1] 302  25
> bmassOutput1$ZScoresCorMatrix
                 Height_ZScore BMI_ZScore WHRadjBMI_ZScore
Height_ZScore       1.00000000 0.01157916       0.01108188
BMI_ZScore          0.01157916 1.00000000       0.67291581
WHRadjBMI_ZScore    0.01108188 0.67291581       1.00000000
> bmassOutput1$GWASlogBFMinThreshold
[1] 3.564426
.
.
.
> Height2 <- read.table("/mnt/gluster/data/external_public_supp/GIANT2014_5/20161227edits.GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz", header=T)
> BMI2 <- read.table("/mnt/gluster/data/external_public_supp/GIANT2014_5/20161227edits.SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.MatchedToHeight.IncAllele.gz", header=T)
> WHRadjBMI2 <- read.table("/mnt/gluster/data/external_public_supp/GIANT2014_5/20161227edits.GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.txt.gz", header=T)
>
> bmassOutput1.1 <- bmass(c("Height2", "BMI2", "WHRadjBMI2"), GWASsnps=GWASsnps, NminThreshold = 50000, PrintMergedData=TRUE, bmassSeedValue=150);
> summary(bmassOutput1.1)
                      Length Class      Mode     
MarginalSNPs            4    -none-     list     
PreviousSNPs            4    -none-     list     
NewSNPs                 3    -none-     list     
LogFile                17    -none-     character
MergedDataSources      19    data.frame list     
ZScoresCorMatrix        9    -none-     numeric  
Models                 81    -none-     numeric  
ModelPriors           378    -none-     numeric  
GWASlogBFMinThreshold   1    -none-     numeric  
> dim(bmassOutput1.1$NewSNPs$SNPs)
[1] 240  25
> bmassOutput1.1$ZScoresCorMatrix
                  Height2_ZScore BMI2_ZScore WHRadjBMI2_ZScore
Height2_ZScore       1.000000000 -0.05096709      -0.009691941
BMI2_ZScore         -0.050967089  1.00000000       0.015522429
WHRadjBMI2_ZScore   -0.009691941  0.01552243       1.000000000
> bmassOutput1.1$GWASlogBFMinThreshold
[1] 2.907321
.
.
.
>         bmassOutput1.4$MergedDataSources <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/20161217_edits_GIANT2014_5.Orig3.dtlesssignif.vs1.SignCrrct.vs1.annot.vs1.MAF.txt.gz", header=TRUE)
>         bmassOutput1.4$ZScoresCorMatrix <- as.matrix(read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5.Orig3.RSS0.vs1.SignCrrct.vs1.txt",header=T,sep=","))
>         #Same as GWASsnps but whatevers
>         GWASsnps1.4 <- read.table("/mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt", header=F);
>         GWASsnps1.4 <- GWASsnps1.4[,2:3]
>         GWASsnps1.4[,2] <- GWASsnps1.4[,2] - 1
>         names(GWASsnps1.4) <- c("Chr", "BP");
>         DataSources1.4 = c("Height", "BMI", "WHRadjBMI"); NminThreshold = 50000; bmassSeedValue=150;
>         ExpectedColumnNames=c("Chr", "BP", "MAF", "Direction", "pValue", "N"); GWASsnps_AnnotateWindow = 5e5; SNPMarginalUnivariateThreshold = 1e-6; SNPMarginalMultivariateThreshold = 1e-6; SigmaAlphas = c(0.005,0.0075,0.01,0.015,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.15); ProvidedPriors=NULL; UseFlatPriors=FALSE; PruneMarginalSNPs=TRUE; PruneMarginalSNPs_bpWindow=5e5;
>         bmassOutput1.4$MergedDataSources$ChrBP <- paste(bmassOutput1.4$MergedDataSources$Chr, "_", bmassOutput1.4$MergedDataSources$BP, sep="")
>
>         bmassOutput1.4$MarginalSNPs$SNPs <- bmassOutput1.4$MergedDataSources[bmassOutput1.4$MergedDataSources$mvstat_log10pVal > -log10(SNPMarginalUnivariateThreshold) | bmassOutput1.4$MergedDataSources$unistat_log10pVal > -log10(SNPMarginalMultivariateThreshold),]
>
> dim(bmassOutput1.4$MarginalSNPs$SNPs)
[1] 43187    24
> dim(bmassOutput1.4$MergedDataSources)
[1] 43187    24
>         bmassOutput1.4[c("MarginalSNPs", "Models", "ModelPriors", "LogFile")] <- GetLogBFsFromData(DataSources1.4, bmassOutput1.4$MarginalSNPs, bmassOutput1.4$ZScoresCorMatrix, SigmaAlphas, bmassOutput1.4$LogFile)[c("MarginalSNPs", "Models", "ModelPriors", "LogFile")]
GWASlogBFMinThreshold", "LogFile")]
>         bmassOutput1.4[c("MarginalSNPs", "PreviousSNPs", "ModelPriors", "GWASlogBFMinThreshold", "LogFile")] <- DetermineAndApplyPriors(DataSources1.4, bmassOutput1.4$MarginalSNPs, GWASsnps1.4, SigmaAlphas, bmassOutput1.4$Models, bmassOutput1.4$ModelPriors, ProvidedPriors, UseFlatPriors, bmassSeedValue, bmassOutput1.4$LogFile)[c("MarginalSNPs", "PreviousSNPs", "ModelPriors", "GWASlogBFMinThreshold", "LogFile")]
>         bmassOutput1.4[c("MarginalSNPs", "PreviousSNPs", "NewSNPs", "LogFile")] <- FinalizeAndFormatResults(DataSources1.4, bmassOutput1.4$MarginalSNPs, bmassOutput1.4$PreviousSNPs, GWASsnps1.4, bmassOutput1.4$GWASlogBFMinThreshold, SigmaAlphas, bmassOutput1.4$Models, bmassOutput1.4$ModelPriors, NminThreshold, PruneMarginalSNPs, PruneMarginalSNPs_bpWindow, bmassOutput1.4$LogFile)[c("MarginalSNPs", "PreviousSNPs", "NewSNPs", "LogFile")]
Error in if (x == 1) { : missing value where TRUE/FALSE needed
>
> dim(bmassOutput1.4$NewSNPs$SNPs)
NULL
> bmassOutput1.4$GWASlogBFMinThreshold
[1] 2.953744
> MarginalSNPs_PrunedList <- indephits(bmassOutput1.4$MarginalSNPs$SNPs$logBFWeightedAvg, bmassOutput1.4$MarginalSNPs$SNPs$Chr, bmassOutput1.4$MarginalSNPs$SNPs$BP, T=PruneMarginalSNPs_bpWindow)
> MarginalSNPs_Pruned <- bmassOutput1.4$MarginalSNPs$SNPs[MarginalSNPs_PrunedList == 1,]
> NewSNPs$SNPs <- MarginalSNPs_Pruned[MarginalSNPs_Pruned$GWASanno == 0 & MarginalSNPs_Pruned$logBFWeightedAvg >= 2.953744,]
> dim(NewSNPs$SNPs)
[1] 3911   26
> table(is.na((bmassOutput1.4$MarginalSNPs$SNPs$Chr)))

FALSE
43187
> table(is.na((bmassOutput1.4$MarginalSNPs$SNPs$BP)))

FALSE
43187
> table(is.na((bmassOutput1.4$MarginalSNPs$SNPs$logBFWeightedAvg)))

FALSE  TRUE
43163    24
> MarginalSNPs <- bmassOutput1.4$MarginalSNPs$SNPs[!is.na(bmassOutput1.4$MarginalSNPs$SNPs$logBFWeightedAvg),]
> MarginalSNPs_PrunedList <- indephits(MarginalSNPs$logBFWeightedAvg, MarginalSNPs$Chr, MarginalSNPs$BP, T=PruneMarginalSNPs_bpWindow)
> MarginalSNPs_Pruned <- MarginalSNPs[MarginalSNPs_PrunedList == 1,]
> NewSNPs$SNPs <- MarginalSNPs_Pruned[MarginalSNPs_Pruned$GWASanno == 0 & MarginalSNPs_Pruned$logBFWeightedAvg >= 2.953744,]
> dim(NewSNPs$SNPs)
[1] 217  26
[  mturchin20@spudhead  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs1]$cat ../../GIANT2014_5/Vs1/GIANT2014_5.Orig3.newtophits.vs1.SignCrrct.vs1.MarkerChrBP.txt | wc
    218     654    5624
> summary(bmassOutput1)
                      Length Class      Mode     
MarginalSNPs            4    -none-     list     
PreviousSNPs            4    -none-     list     
NewSNPs                 3    -none-     list     
LogFile                17    -none-     character
MergedDataSources      19    data.frame list     
ZScoresCorMatrix        9    -none-     numeric  
Models                 81    -none-     numeric  
ModelPriors           378    -none-     numeric  
GWASlogBFMinThreshold   1    -none-     numeric  
> summary(bmassOutput1.1)
                      Length Class      Mode     
MarginalSNPs            4    -none-     list     
PreviousSNPs            4    -none-     list     
NewSNPs                 3    -none-     list     
LogFile                17    -none-     character
MergedDataSources      19    data.frame list     
ZScoresCorMatrix        9    -none-     numeric  
Models                 81    -none-     numeric  
ModelPriors           378    -none-     numeric  
GWASlogBFMinThreshold   1    -none-     numeric  
> dim(bmassOutput1$MergedDataSources)
[1] 2489613      19
> dim(bmassOutput1.1$MergedDataSources)
[1] 2479684      19
> write.table(bmassOutput1$MergedDataSources, "blahblah1.txt")
> write.table(bmassOutput1.1$MergedDataSources, "blahblah2.txt")
.
.
.
> Height <- read.table("/mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.IncAllele.bmass.formatted.txt.gz", header=T);
> BMI <- read.table("/mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.IncAllele.bmass.formatted.MatchedToHeight.DropNoMatch.txt.gz", header=T);
> WHRadjBMI <- read.table("/mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.IncAllele.bmass.formatted.MatchedToHeight.DropNoMatch.txt.gz", header=T);
>
> #GWASsnps <- read.table("/mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.AllPheno.MarkerChrBP.txt", header=F);
> GWASsnps <- read.table("/mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt", header=F);
> GWASsnps <- GWASsnps[,2:3]
> GWASsnps[,2] <- GWASsnps[,2] - 1
> GWASsnps2 <- read.table("/mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMI.MarkerChrBP.txt", header=F);
> GWASsnps2 <- GWASsnps2[,2:3]
> GWASsnps2[,2] <- GWASsnps2[,2] - 1
> names(GWASsnps) <- c("Chr", "BP");
> names(GWASsnps2) <- c("Chr", "BP");
>
> bmassOutput1 <- bmass(c("Height", "BMI", "WHRadjBMI"), GWASsnps=GWASsnps, NminThreshold = 50000, PrintMergedData=TRUE, bmassSeedValue=150);
Error in CheckIndividualDataSources(DataSources, GWASsnps, ExpectedColumnNames,  :
  2016-12-27 14:42:06 -- the following data sources have entries other than + and - in the Direction column. Please fix and rerun bmass: Height BMI WHRadjBMI
>
> table(Height$Direction)

      +       0
2517440   17224


> head(Height[Height$Direction==0,])
     Chr        BP   MAF A1 A2 Direction pValue      N
377   10 100215646 0.350  T  C         0   0.99 253177
426   10 100265663 0.025  T  C         0   1.00 225416
490   10 100303552 0.333  T  C         0   0.99 244803
835   10   1005246 0.017  T  G         0   1.00 229858
852   10 100546415 0.192  A  G         0   0.99 253218
1046  10 100748568 0.093  A  C         0   1.00 252700
> table(BMI$Direction)

      -       +       0 
1314277 1176949   17949 
> table(WHRadjBMI$Direction)

      -       +       0 
1246373 1220497   16655 
> Height <- Height[Height$Direction != 0,] 
> BMI <- BMI[BMI$Direction != 0,]
> WHRadjBMI <- WHRadjBMI[WHRadjBMI$Direction != 0,]
> 
> table(Height$Direction)

      +       0 
2517440       0 
> dim(bmassOutput1$NewSNPs$SNPs)
[1] 218  25
> bmassOutput1$ZScoresCorMatrix
                 Height_ZScore  BMI_ZScore WHRadjBMI_ZScore
Height_ZScore      1.000000000 -0.05115997     -0.009512951
BMI_ZScore        -0.051159975  1.00000000      0.015944658
WHRadjBMI_ZScore  -0.009512951  0.01594466      1.000000000
> bmassOutput1$GWASlogBFMinThreshold
[1] 2.899102
>
> bmassOutput2 <- bmass(c("Height", "BMI"), GWASsnps=GWASsnps2, NminThreshold = 50000, PrintMergedData=TRUE, bmassSeedValue=150);
>
> dim(bmassOutput2$NewSNPs$SNPs)
[1] 200  21
> bmassOutput2$GWASlogBFMinThreshold
[1] 3.577093
> bmassOutput2$ZScoresCorMatrix
              Height_ZScore  BMI_ZScore
Height_ZScore    1.00000000 -0.04949805
BMI_ZScore      -0.04949805  1.00000000
~~~

#20170120
#Above was a 'quicker' run-through to get results for Sarah MASH comparisons. Just going over things again with a few small tweaks and to be consistent with what I've done with other datasets now too. 
#Going to include the 'no0Dir' and 'off-by-1' corrections prior to the R code, unlike what was done before

join <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $9 }' | grep -v , | grep -v NA | grep -v cox | grep -v ChrBP | sort | uniq -u) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | sed 's/_/ /g' | awk '{ print $9 "\t" $10 "\t" $4 "\t" $2 "\t" $3 "\t" $5 "\t" $7 "\t" $8 }' | perl -lane 'if ($F[5] > 0) { $F[5] = "+"; } elsif ($F[5] < 0) { $F[5] = "-"; } else { $F[5] = 0;} if ($F[5] eq "-") { ($F[3], $F[4]) = ($F[4], $F[3]); $F[5] = "+"; $F[2] = 1 - $F[2]; } print join("\t", @F); ' | grep -v Freq | awk '{ print $1 "_" $2 "\t" $0 }' | sort -k 1,1) | perl -lane 'splice(@F, 0, 1); print join("\t", @F);' | perl -lane 'if (($F[2] ne "NA") && ($F[2] > 0) && ($F[2] < 1)) { print join("\t", @F); }' | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | perl -lane 'if ($F[0] eq "X") { $F[0] = 23; } if ($F[0] eq "Y") { $F[0] = 24; } print join("\t", @F); ' | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.txt.gz  
join <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.gz | awk '{ print $9 }' | grep -v , | grep -v NA | grep -v cox | grep -v ChrBP | sort | uniq -u) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.gz | sed 's/_/ /g' | awk '{ print $9 "\t" $10 "\t" $4 "\t" $2 "\t" $3 "\t" $5 "\t" $7 "\t" $8 }' | perl -lane 'if ($F[5] > 0) { $F[5] = "+"; } elsif ($F[5] < 0) { $F[5] = "-"; } else { $F[5] = 0;} if ($F[5] eq "-") { ($F[3], $F[4]) = ($F[4], $F[3]); $F[5] = "+"; $F[2] = 1 - $F[2]; } print join("\t", @F); ' | grep -v Freq | awk '{ print $1 "_" $2 "\t" $0 }' | sort -k 1,1) | perl -lane 'splice(@F, 0, 1); print join("\t", @F);' | perl -lane 'if (($F[2] ne "NA") && ($F[2] > 0) && ($F[2] < 1)) { print join("\t", @F); }' | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | perl -lane 'if ($F[0] eq "X") { $F[0] = 23; } if ($F[0] eq "Y") { $F[0] = 24; } print join("\t", @F); ' | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.txt.gz  
join <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $9 }' | tail -n +5 | grep -v , | grep -v NA | grep -v cox | grep -v ChrBP | sort | uniq -u) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | tail -n +9 | sed 's/_/ /g' | awk '{ print $9 "\t" $10 "\t" $4 "\t" $2 "\t" $3 "\t" $5 "\t" $7 "\t" $8 }' | perl -lane 'if ($F[5] > 0) { $F[5] = "+"; } elsif ($F[5] < 0) { $F[5] = "-"; } else { $F[5] = 0;} if ($F[5] eq "-") { ($F[3], $F[4]) = ($F[4], $F[3]); $F[5] = "+"; $F[2] = 1 - $F[2]; } print join("\t", @F); ' | grep -v Freq | awk '{ print $1 "_" $2 "\t" $0 }' | sort -k 1,1) | perl -lane 'splice(@F, 0, 1); print join("\t", @F);' | perl -lane 'if (($F[2] ne "NA") && ($F[2] > 0) && ($F[2] < 1)) { print join("\t", @F); }' | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | perl -lane 'if ($F[0] eq "X") { $F[0] = 23; } if ($F[0] eq "Y") { $F[0] = 24; } print join("\t", @F); ' | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.txt.gz  

join <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.txt.gz | awk '{ print $1 "_" $2 "\t" $4 }' | grep -v NA | sort -g -k 1,1) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.txt.gz | awk '{ print $1 "_" $2 "\t" $0 }' | grep -v NA | sort -g -k 1,1) | perl -lane 'if ($F[1] eq $F[6]) { ($F[5], $F[6]) = ($F[6], $F[5]); $F[4] = 1 - $F[4]; if ($F[7] eq "-") { $F[7] = "+"; } elsif ($F[7] eq "+") { $F[7] = "-"; } else { $PH = 1; } } elsif ($F[1] eq $F[5]) { $PH = 1; } else { push(@F, "NoAlleleMatch"); } print join("\t", @F[2..$#F]);' | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.MatchedToHeight.txt.gz
join <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.txt.gz | awk '{ print $1 "_" $2 "\t" $4 }' | grep -v NA | sort -g -k 1,1) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.txt.gz | awk '{ print $1 "_" $2 "\t" $0 }' | grep -v NA | sort -g -k 1,1) | perl -lane 'if ($F[1] eq $F[6]) { ($F[5], $F[6]) = ($F[6], $F[5]); $F[4] = 1 - $F[4]; if ($F[7] eq "-") { $F[7] = "+"; } elsif ($F[7] eq "+") { $F[7] = "-"; } else { $PH = 1; } } elsif ($F[1] eq $F[5]) { $PH = 1; } else { push(@F, "NoAlleleMatch"); } print join("\t", @F[2..$#F]);' | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.MatchedToHeight.txt.gz

zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.txt.gz | perl -lane 'if ($F[5] ne "0") { print join("\t", @F); }' | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.No0Dir.txt.gz
zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.MatchedToHeight.txt.gz | grep -v NoAlleleMatch | perl -lane 'if ($F[5] ne "0") { print join("\t", @F); }' | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.MatchedToHeight.DropNoMatch.No0Dir.txt.gz 
zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.MatchedToHeight.txt.gz | grep -v NoAlleleMatch | perl -lane 'if ($F[5] ne "0") { print join("\t", @F); }' | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.MatchedToHeight.DropNoMatch.No0Dir.txt.gz

#20170125
#Made change to force MAF from start now

zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.No0Dir.txt.gz | perl -lane 'if ($F[2] > .5) { $F[2] = 1 - $F[2]; } print join("\t", @F);' | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.No0Dir.forceMAF.txt.gz
zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.MatchedToHeight.DropNoMatch.No0Dir.txt.gz | perl -lane 'if ($F[2] > .5) { $F[2] = 1 - $F[2]; } print join("\t", @F);' | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.MatchedToHeight.DropNoMatch.No0Dir.forceMAF.txt.gz
zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.MatchedToHeight.DropNoMatch.No0Dir.txt.gz | perl -lane 'if ($F[2] > .5) { $F[2] = 1 - $F[2]; } print join("\t", @F);' | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.MatchedToHeight.DropNoMatch.No0Dir.forceMAF.txt.gz

#20170126
#Redoing full setup with orienting towards MAF allele from get go now. 
#NOTE -- also including a check that ref allele from dataset matches either allele in HapMap23 (dropping if ref doesn't match either alleles and not correcting/checking for strandedness here)

#NoDup, noNA, noComma (mainly affects dbSNP), Check Alleles (drop if no match, includes strand check), Flip to Minor, MatchToPheno1, Drop 0Dir, force flip to minor AF, drop if fixed

GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130
SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130
GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130

join <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $9 }' | grep -v ChrBP | sort | uniq -u) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep -v , | grep -v NA | grep -v cox | sed 's/_/ /g' | awk '{ print $9 "\t" $10 "\t" $4 "\t" $2 "\t" $3 "\t" $5 "\t" $7 "\t" $8 }' | perl -lane 'if ($F[5] > 0) { $F[5] = "+"; } elsif ($F[5] < 0) { $F[5] = "-"; } else { $F[5] = 0;} if ($F[2] > .5) { ($F[3], $F[4]) = ($F[4], $F[3]); if ($F[5] eq "+") { $F[5] = "-"; } elsif ($F[5] eq "-") { $F[5] = "+"; } else { $PH = 1; } $F[2] = 1 - $F[2]; } print join("\t", @F); ' | grep -v ChrBP | grep -v Freq | awk '{ print $1 "_" $2 "\t" $0 }' | sort -k 1,1) | perl -lane 'splice(@F, 0, 1); print join("\t", @F);' | perl -lane 'if ($F[0] eq "X") { $F[0] = 23; } if ($F[0] eq "Y") { $F[0] = 24; } print join("\t", @F); ' | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.MinorAllele.bmass.formatted.noDups.noNA.noComma.noCox.txt.gz  
join <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.gz | awk '{ print $9 }' | grep -v ChrBP | sort | uniq -u) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.gz | grep -v , | grep -v NA | grep -v cox | sed 's/_/ /g' | awk '{ print $9 "\t" $10 "\t" $4 "\t" $2 "\t" $3 "\t" $5 "\t" $7 "\t" $8 }' | perl -lane 'if ($F[5] > 0) { $F[5] = "+"; } elsif ($F[5] < 0) { $F[5] = "-"; } else { $F[5] = 0;} if ($F[2] > .5) { ($F[3], $F[4]) = ($F[4], $F[3]); if ($F[5] eq "+") { $F[5] = "-"; } elsif ($F[5] eq "-") { $F[5] = "+"; } else { $PH = 1; } $F[2] = 1 - $F[2]; } print join("\t", @F); ' | grep -v ChrBP | grep -v Freq | awk '{ print $1 "_" $2 "\t" $0 }' | sort -k 1,1) | perl -lane 'splice(@F, 0, 1); print join("\t", @F);' | perl -lane 'if ($F[0] eq "X") { $F[0] = 23; } if ($F[0] eq "Y") { $F[0] = 24; } print join("\t", @F); ' | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.MinorAllele.bmass.formatted.noDups.noNA.noComma.noCox.txt.gz  
join <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $9 }' | tail -n +5 | grep -v ChrBP | sort | uniq -u) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | tail -n +9 | grep -v , | grep -v NA | grep -v cox | sed 's/_/ /g' | awk '{ print $9 "\t" $10 "\t" $4 "\t" $2 "\t" $3 "\t" $5 "\t" $7 "\t" $8 }' | perl -lane 'if ($F[5] > 0) { $F[5] = "+"; } elsif ($F[5] < 0) { $F[5] = "-"; } else { $F[5] = 0;} if ($F[2] > .5) { ($F[3], $F[4]) = ($F[4], $F[3]); if ($F[5] eq "+") { $F[5] = "-"; } elsif ($F[5] eq "-") { $F[5] = "+"; } else { $PH = 1; } $F[2] = 1 - $F[2]; } print join("\t", @F); ' | grep -v ChrBP | grep -v Freq | awk '{ print $1 "_" $2 "\t" $0 }' | sort -k 1,1) | perl -lane 'splice(@F, 0, 1); print join("\t", @F);' | perl -lane 'if ($F[0] eq "X") { $F[0] = 23; } if ($F[0] eq "Y") { $F[0] = 24; } print join("\t", @F); ' | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.MinorAllele.bmass.formatted.noDups.noNA.noComma.noCox.txt.gz  

join <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.MinorAllele.bmass.formatted.noDups.noNA.noComma.noCox.txt.gz | awk '{ print $1 "_" $2 "\t" $4 }' | grep -v NA | sort -g -k 1,1) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.MinorAllele.bmass.formatted.noDups.noNA.noComma.noCox.txt.gz | awk '{ print $1 "_" $2 "\t" $0 }' | grep -v NA | sort -g -k 1,1) | perl -lane 'if ($F[1] eq $F[6]) { ($F[5], $F[6]) = ($F[6], $F[5]); $F[4] = 1 - $F[4]; if ($F[7] eq "-") { $F[7] = "+"; } elsif ($F[7] eq "+") { $F[7] = "-"; } else { $PH = 1; } } elsif ($F[1] eq $F[5]) { $PH = 1; } else { push(@F, "NoAlleleMatch"); } print join("\t", @F[2..$#F]);' | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.MinorAllele.bmass.formatted.noDups.noNA.noComma.noCox.MatchedToHeight.txt.gz
join <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.MinorAllele.bmass.formatted.noDups.noNA.noComma.noCox.txt.gz | awk '{ print $1 "_" $2 "\t" $4 }' | grep -v NA | sort -g -k 1,1) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.MinorAllele.bmass.formatted.noDups.noNA.noComma.noCox.txt.gz | awk '{ print $1 "_" $2 "\t" $0 }' | grep -v NA | sort -g -k 1,1) | perl -lane 'if ($F[1] eq $F[6]) { ($F[5], $F[6]) = ($F[6], $F[5]); $F[4] = 1 - $F[4]; if ($F[7] eq "-") { $F[7] = "+"; } elsif ($F[7] eq "+") { $F[7] = "-"; } else { $PH = 1; } } elsif ($F[1] eq $F[5]) { $PH = 1; } else { push(@F, "NoAlleleMatch"); } print join("\t", @F[2..$#F]);' | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.MinorAllele.bmass.formatted.noDups.noNA.noComma.noCox.MatchedToHeight.txt.gz

zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.MinorAllele.bmass.formatted.noDups.noNA.noComma.noCox.txt.gz | perl -lane 'if ($F[5] ne "0") { if ($F[2] > .5) { $F[2] = 1 - $F[2]; } if ($F[2] > 0) { print join("\t", @F); } }' | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.MinorAllele.bmass.formatted.noDups.noNA.noComma.noCox.no0Dir.noFixed.forceMAF.txt.gz
zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.MinorAllele.bmass.formatted.noDups.noNA.noComma.noCox.MatchedToHeight.txt.gz | grep -v NoAlleleMatch | perl -lane 'if ($F[5] ne "0") { if ($F[2] > .5) { $F[2] = 1 - $F[2]; } if ($F[2] > 0) { print join("\t", @F); } }' | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.MinorAllele.bmass.formatted.noDups.noNA.noComma.noCox.MatchedToHeight.DropNoMatch.no0Dir.noFixed.forceMAF.txt.gz 
zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.MinorAllele.bmass.formatted.noDups.noNA.noComma.noCox.MatchedToHeight.txt.gz | grep -v NoAlleleMatch | perl -lane 'if ($F[5] ne "0") { if ($F[2] > .5) { $F[2] = 1 - $F[2]; } if ($F[2] > 0) { print join("\t", @F); } }' | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.MinorAllele.bmass.formatted.noDups.noNA.noComma.noCox.MatchedToHeight.DropNoMatch.no0Dir.noFixed.forceMAF.txt.gz 

#20170125
#Note change to using forceMAF files
#20170206
#Note change to MinorAllele files & `MinorAllele.bmass.formatted.noDups.noNA.noComma.noCox.MatchedToHeight.DropNoMatch.no0Dir.noFixed.forceMAF`

```
library("devtools"); devtools::load_all("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/bmass"); 

#Height <- read.table("/mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.No0Dir.txt.gz", header=T);
#BMI <- read.table("/mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.MatchedToHeight.DropNoMatch.No0Dir.txt.gz", header=T);
#WHRadjBMI <- read.table("/mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.MatchedToHeight.DropNoMatch.No0Dir.txt.gz", header=T);
#Height <- read.table("/mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.No0Dir.forceMAF.txt.gz", header=T);
#BMI <- read.table("/mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.MatchedToHeight.DropNoMatch.No0Dir.forceMAF.txt.gz", header=T);
#WHRadjBMI <- read.table("/mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.MatchedToHeight.DropNoMatch.No0Dir.forceMAF.txt.gz", header=T);
Height <- read.table("/mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.MinorAllele.bmass.formatted.noDups.noNA.noComma.noCox.no0Dir.noFixed.forceMAF.txt.gz", header=T);
BMI <- read.table("/mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.MinorAllele.bmass.formatted.noDups.noNA.noComma.noCox.MatchedToHeight.DropNoMatch.no0Dir.noFixed.forceMAF.txt.gz", header=T);
WHRadjBMI <- read.table("/mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.MinorAllele.bmass.formatted.noDups.noNA.noComma.noCox.MatchedToHeight.DropNoMatch.no0Dir.noFixed.forceMAF.txt.gz", header=T);

#Height <- Height[Height$Direction != 0,]
#BMI <- BMI[BMI$Direction != 0,]
#WHRadjBMI <- WHRadjBMI[WHRadjBMI$Direction != 0,]

GWASsnps <- read.table("/mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.txt", header=F);
GWASsnps <- GWASsnps[,2:3]
##GWASsnps[,2] <- GWASsnps[,2] - 1
names(GWASsnps) <- c("Chr", "BP");

bmassOutput1 <- bmass(c("Height", "BMI", "WHRadjBMI"), GWASsnps=GWASsnps, NminThreshold = 50000, PrintMergedData=FALSE, bmassSeedValue=150);
bmassOutput2 <- bmass(c("Height", "BMI", "WHRadjBMI"), GWASsnps=GWASsnps, NminThreshold = 50000, PrintMergedData=FALSE, PruneMarginalSNPs=FALSE, bmassSeedValue=150);
bmassOutput3 <- bmass(c("Height", "BMI", "WHRadjBMI"), GWASsnps=GWASsnps, NminThreshold = 50000, PrintMergedData=TRUE, PruneMarginalSNPs=FALSE, bmassSeedValue=150);

#write.table(bmassOutput1$NewSNPs$SNPs, file="/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/GIANT2014_5.Vs2.HeightBMIWHRadjBMI.NewSNPs.SNPs.vs1.txt", quote=FALSE, row.names=FALSE)
#write.table(bmassOutput2$NewSNPs$SNPs, file="/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/GIANT2014_5.Vs2.HeightBMI.NewSNPs.SNPs.vs1.txt", quote=FALSE, row.names=FALSE)
write.table(bmassOutput1$NewSNPs$SNPs, file="/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.txt", quote=FALSE, row.names=FALSE)
#write.table(bmassOutput1$MergedDataSources, file="/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/bmass.GIANT2014_5.Vs2.MergedDataSources.vs1.txt", quote=FALSE, row.names=FALSE)
write.table(bmassOutput3$MergedDataSources, file="/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/bmass.GIANT2014_5.Vs2.MergedDataSources.vs1.txt", quote=FALSE, row.names=FALSE)

system("gzip /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/bmass.GIANT2014_5.Vs2.MergedDataSources.vs1.txt")
```

~~~
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $9 }' | sort | grep -v NA | grep -v , | grep -v cox | uniq | wc
2535906 2535906 29534883
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $9 }' | sort | grep -v NA | grep -v , | grep -v cox | wc       
2535923 2535923 29535076
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $9 }' | sort | grep -v NA | grep -v , | grep -v cox | uniq -d  
10_110193236
11_64148485
14_103679890
14_63629844
16_1546445
1_224022541
1_225248177
2_18013703
4_190979685
5_109391943
6_28439106
7_82377142
8_127880389
8_59431959
9_29789139
9_71795273
ChrBP
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $9 }' | sort | grep -v NA | grep -v , | grep -v cox | wc
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.gz | awk '{ print $9 }' | sort | grep -v NA | grep -v , | grep -v cox | wc
2542707 2542707 29614568
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.gz | awk '{ print $9 }' | sort | grep -v NA | grep -v , | grep -v cox | uniq | wc
2542677 2542677 29614220
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.txt.gz | awk '{ print $9 }' | sort | grep -v NA | grep -v , | grep -v cox | wc
2500731 2500731 29126166
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.txt.gz | awk '{ print $9 }' | sort | grep -v NA | grep -v , | grep -v cox | uniq | wc
2500715 2500715 29125979
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | wc
2550859 22957731 149252492
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.txt.gz | wc
2529547 20236376 91114082
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.txt.gz | head -n 10
Chr BP  MAF A1  A2 Direction  pValue    N
10 10000134 0.449  G    A  +  0.70      253213
10 10000264 0.436  T    C  +  0.70      253213
10 100002728    0.633   G  A  + 0.042   253116
10 100002879    0.642   G  A  + 0.041   252156
10 100003552    0.88    C  T  + 0.024   248425
10 100003804    0.3 T   C  +  8.2e-06   251271
10 100003966    0.644   A  T  + 0.029   253086
10 10000458 0.427  T    G  +  0.70      253213
10 100005552    0.856   G  C  + 0.62    253056
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.txt.gz | grep NA | wc
      0       0       0
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | wc
2550859 22957731 149252492
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.txt.gz | wc
2529547 20236376 91114082
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.txt.gz | head -n 10
Chr BP  MAF A1  A2 Direction  pValue    N
10 10000134 0.449  G    A  +  0.70      253213
10 10000264 0.436  T    C  +  0.70      253213
10 100002728    0.633   G  A  + 0.042   253116
10 100002879    0.642   G  A  + 0.041   252156
10 100003552    0.88    C  T  + 0.024   248425
10 100003804    0.3 T   C  +  8.2e-06   251271
10 100003966    0.644   A  T  + 0.029   253086
10 10000458 0.427  T    G  +  0.70      253213
10 100005552    0.856   G  C  + 0.62    253056
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.txt.gz | grep -E '\,|NA|cox' | wc
      0       0       0
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.txt.gz | perl -lane 'if ($F[2] !~ m/\d/) { print join("\t", @F); }'
Chr BP  MAF A1  A2 Direction  pValue    N
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.txt.gz | perl -lane 'if (($F[2] <= 0) || ($F[2] >= 1)) { print join("\t", @F); }'
Chr BP  MAF A1  A2 Direction  pValue    N
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.gz | wc
2554638 22991742 155169518
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.txt.gz | wc
2488820 19910560 95401829
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.txt.gz | head -n 10
Chr BP  MAF A1  A2 Direction  pValue    N
10 10000134 0.45   G    A  +  0.3173    233983
10 10000264 0.45   T    C  +  0.3044    233984
10 100002728    0.3667  A  G  + 0.1712  233950
10 100002879    0.3583  A  G  + 0.1269  233968
10 100003552    0.2 T   C  +  0.3173    232547
10 100003804    0.7 C   T  +  0.004372  233853
10 100003966    0.3583  T  A  + 0.1542  233909
10 10000458 0.45   T    G  +  0.3306    233983
10 100005552    0.15    C  G  + 0.6416  233900
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.txt.gz | grep -E '\,|NA|cox' | wc
      0       0       0
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.txt.gz | perl -lane 'if ($F[2] !~ m/\d/) { print join("\t", @F); }'
Chr BP  MAF A1  A2 Direction  pValue    N
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.txt.gz | perl -lane 'if (($F[2] <= 0) || ($F[2] >= 1)) { print join("\t", @F); }'
Chr BP  MAF A1  A2 Direction  pValue    N
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $9 }' | tail -n +6 | head -n 20


Allele1
4_145749152
16_4343664
15_50622399
7_77248249
2_43698832
6_45685111
7_111703558
3_52571437
17_17774368
6_148642523
2_119154355
11_111353740
14_90580590
11_64330164
20_33208564
7_46293246
10_27959726
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $9 }' | tail -n +6 | grep -v , | head -n 10


Allele1
4_145749152
16_4343664
15_50622399
7_77248249
2_43698832
6_45685111
7_111703558
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $9 }' | grep -v , | head -n 10
Binary file (standard input) matches
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | sed 's/_/ /g' | awk '{ print $9 "\t" $10 "\t" $4 "\t" $2 "\t" $3 "\t" $5 "\t" $7 "\t" $8 }' | perl -lane 'if ($F[5] > 0) { $F[5] = "+"; } elsif ($F[5] < 0) { $F[5] = "-"; } else { $F[5] = 0;} if ($F[5] eq "-") { ($F[3], $F[4]) = ($F[4], $F[3]); $F[5] = "+"; $F[2] = 1 - $F[2]; } print join("\t", @F); ' | grep -v Freq | awk '{ print $1 "_" $2 "\t" $0 }' | sort -k 1,1 | wc
      1       6      49
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | tail -n +5 | sed 's/_/ /g' | awk '{ print $9 "\t" $10 "\t" $4 "\t" $2 "\t" $3 "\t" $5 "\t" $7 "\t" $8 }' | perl -lane 'if ($F[5] > 0) { $F[5] = "+"; } elsif ($F[5] < 0) { $F[5] = "-"; } else { $F[5] = 0;} if ($F[5] eq "-") { ($F[3], $F[4]) = ($F[4], $F[3]); $F[5] = "+"; $F[2] = 1 - $F[2]; } print join("\t", @F); ' | grep -v Freq | awk '{ print $1 "_" $2 "\t" $0 }' | sort -k 1,1 | wc
      1       6      49
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | tail -n +9 | sed 's/_/ /g' | awk '{ print $9 "\t" $10 "\t" $4 "\t" $2 "\t" $3 "\t" $5 "\t" $7 "\t" $8 }' | perl -lane 'if ($F[5] > 0) { $F[5] = "+"; } elsif ($F[5] < 0) { $F[5] = "-"; } else { $F[5] = 0;} if ($F[5] eq "-") { ($F[3], $F[4]) = ($F[4], $F[3]); $F[5] = "+"; $F[2] = 1 - $F[2]; } print join("\t", @F); ' | grep -v Freq | awk '{ print $1 "_" $2 "\t" $0 }' | sort -k 1,1 | wc
2542432 22851608 121247379
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | wc
2542440 22881925 147772335
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.txt.gz | wc
2476553 19812424 89841980
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.txt.gz | head -n 10
Chr BP  MAF A1  A2 Direction  pValue    N
10 10000134 0.45   G    A  +  0.74      142749
10 10000264 0.45   T    C  +  0.74      142749
10 100002728    0.6333  G  A  + 0.91    142676
10 100002879    0.6417  G  A  + 0.82    142589
10 100003552    0.8 C   T  +  0.2       142245
10 100003804    0.3 T   C  +  0.11      142373
10 100003966    0.6417  A  T  + 0.74    142575
10 10000458 0.45   T    G  +  0.76      142749
10 100005552    0.85    G  C  + 0.41    142595
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.txt.gz | grep -E '\,|NA|cox' | wc
      0       0       0
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.txt.gz | perl -lane 'if ($F[2] !~ m/\d/) { print join("\t", @F); }'
Chr BP  MAF A1  A2 Direction  pValue    N
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.txt.gz | perl -lane 'if (($F[2] <= 0) || ($F[2] >= 1)) { print join("\t", @F); }'
Chr BP  MAF A1  A2 Direction  pValue    N
> dim(bmassOutput1$NewSNPs$SNPs)
[1] 217  25
> dim(bmassOutput1$MergedDataSources)
[1] 2418714      19
> bmassOutput1$GWASlogBFMinThreshold
[1] 2.855675
> bmassOutput1$ZScoresCorMatrix
                 Height_ZScore  BMI_ZScore WHRadjBMI_ZScore
Height_ZScore      1.000000000 -0.05108351     -0.009431116
BMI_ZScore        -0.051083508  1.00000000      0.014632198
WHRadjBMI_ZScore  -0.009431116  0.01463220      1.000000000
> dim(bmassOutput1$PreviousSNPs$SNPs)
[1] 793  25
> dim(GWASsnps)
[1] 843   2
> dim(bmassOutput3$MergedDataSources)
[1] 2418714      19
> dim(bmassOutput3$NewSNPs$SNPs)
[1] 3015   25
> bmassOutput3$GWASlogBFMinThreshold
[1] 2.855675
#20170123
#From /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GLC.MTedits.ForGIANT2014_5.Orig3.vs1.SignCrrct.vs1.ForCSHLPoster.R
#> min(lbf.av.glhits)
#[1] 2.953744
#194 in results
#From /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/process.MTedits.ForGIANT2014_5.vs1.SignCrrct.vs1.R
#> RSS0
#                 Z.BMI    Z.height Z.WHRadjBMI
#Z.BMI        1.0000000 -0.02537170  0.01383690
#Z.height    -0.0253717  1.00000000 -0.00338066
#Z.WHRadjBMI  0.0138369 -0.00338066  1.00000000
#20170125 -- forceMAF update
> dim(bmassOutput1$NewSNPs$SNPs)
[1] 215  25
> bmassOutput1$GWASlogBFMinThreshold
[1] 2.851709
> bmassOutput1$ZScoresCorMatrix
                 Height_ZScore  BMI_ZScore WHRadjBMI_ZScore
Height_ZScore      1.000000000 -0.02542650     -0.003160168
BMI_ZScore        -0.025426501  1.00000000      0.014215151
WHRadjBMI_ZScore  -0.003160168  0.01421515      1.000000000
> dim(bmassOutput3$MergedDataSources)
[1] 2418714      19
> dim(bmassOutput1$PreviousSNPs$SNPs)
[1] 793  25
> dim(GWASsnps)
[1] 843   2
.
.
.
#20170207
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$join <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $9 }' | grep -v ChrBP | sort | uniq -u) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep -v , | grep -v NA | grep -v cox | sed 's/_/ /g' | awk '{ print $9 "\t" $10 "\t" $4 "\t" $2 "\t" $3 "\t" $5 "\t" $7 "\t" $8 }' | perl -lane 'if ($F[5] > 0) { $F[5] = "+"; } elsif ($F[5] < 0) { $F[5] = "-"; } else { $F[5] = 0;} if ($F[2] > .5) { ($F[3], $F[4]) = ($F[4], $F[3]); if ($F[5] eq "+") { $F[5] = "-"; } elsif ($F[5] eq "-") { $F[5] = "+"; } else { $PH = 1; } $F[2] = 1 - $F[2]; } print join("\t", @F); ' | grep -v ChrBP | grep -v Freq | awk '{ print $1 "_" $2 "\t" $0 }' | sort -k 1,1) | perl -lane 'splice(@F, 0, 1); print join("\t", @F);' | perl -lane 'if ($F[0] eq "X") { $F[0] = 23; } if ($F[0] eq "Y") { $F[0] = 24; } print join("\t", @F); ' | head -n 10
10 10000134 0.449  G    A  +  0.70      253213
10 10000264 0.436  T    C  +  0.70      253213
10 100002728    0.367   A  G  - 0.042   253116
10 100002879    0.358   A  G  - 0.041   252156
10 100003552    0.12    T  C  - 0.024   248425
10 100003804    0.3 T   C  +  8.2e-06   251271
10 100003966    0.356   T  A  - 0.029   253086
10 10000458 0.427  T    G  +  0.70      253213
10 100005552    0.144   C  G  - 0.62    253056
10 100005592    0.217   T  G  - 0.024   253179
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | wc
2550859 22957731 149252492
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $9 }' | grep -v ChrBP | sort | uniq -u | wc
2547463 2547463 30051547
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$join <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $9 }' | grep -v ChrBP | sort | uniq -u) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep -v , | grep -v NA | grep -v cox | sed 's/_/ /g' | awk '{ print $9 "\t" $10 "\t" $4 "\t" $2 "\t" $3 "\t" $5 "\t" $7 "\t" $8 }' | perl -lane 'if ($F[5] > 0) { $F[5] = "+"; } elsif ($F[5] < 0) { $F[5] = "-"; } else { $F[5] = 0;} if ($F[2] > .5) { ($F[3], $F[4]) = ($F[4], $F[3]); if ($F[5] eq "+") { $F[5] = "-"; } elsif ($F[5] eq "-") { $F[5] = "+"; } else { $PH = 1; } $F[2] = 1 - $F[2]; } print join("\t", @F); ' | grep -v ChrBP | grep -v Freq | awk '{ print $1 "_" $2 "\t" $0 }' | sort -k 1,1) | perl -lane 'splice(@F, 0, 1); print join("\t", @F);' | perl -lane 'if ($F[0] eq "X") { $F[0] = 23; } if ($F[0] eq "Y") { $F[0] = 24; } print join("\t", @F); ' | awk '{ print $4 }' | sort | uniq -c
 657513 A
 608986 C
 608297 G
 658680 T
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$join <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $9 }' | grep -v ChrBP | sort | uniq -u) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep -v , | grep -v NA | grep -v cox | sed 's/_/ /g' | awk '{ print $9 "\t" $10 "\t" $4 "\t" $2 "\t" $3 "\t" $5 "\t" $7 "\t" $8 }' | perl -lane 'if ($F[5] > 0) { $F[5] = "+"; } elsif ($F[5] < 0) { $F[5] = "-"; } else { $F[5] = 0;} if ($F[2] > .5) { ($F[3], $F[4]) = ($F[4], $F[3]); if ($F[5] eq "+") { $F[5] = "-"; } elsif ($F[5] eq "-") { $F[5] = "+"; } else { $PH = 1; } $F[2] = 1 - $F[2]; } print join("\t", @F); ' | grep -v ChrBP | grep -v Freq | awk '{ print $1 "_" $2 "\t" $0 }' | sort -k 1,1) | perl -lane 'splice(@F, 0, 1); print join("\t", @F);' | perl -lane 'if ($F[0] eq "X") { $F[0] = 23; } if ($F[0] eq "Y") { $F[0] = 24; } print join("\t", @F); ' | awk '{ print $5 }' | sort | uniq -c
 580497 A
 685729 C
 685882 G
 581368 T
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep -v , | wc
2539285 22853565 148192226
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep -v NA | wc
2544947 22904523 148946022
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep -v cox | wc
2540914 22868226 148313900
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep cox | head -n 10
rs10884221  T   C  0.808   -0.0024      0.0035  0.48    249684  10_107486155,6_cox_hap1_48402
rs11193825  C   G  0.15 -0.00070        0.0043  0.87    253232  10_109725754,6_cox_hap1_4021800
rs11818958  A   C  0.436   0.0044       0.0030  0.14    252729  10_43808197,6_cox_hap1_48162
rs10824920  T   C  0.558   0.0014       0.0030  0.65    253004  10_54585619,6_cox_hap1_4449605,6_qbl_hap2_4255532
rs1046747   A   G  0.375   0.0026       0.0030  0.38    252865  10_71580635,6_cox_hap1_498071,6_qbl_hap2_300095
rs15801 T   C   0.4 0.0043 0.0030       0.15    252930  10_71580668,6_cox_hap1_498102,6_qbl_hap2_300126
rs2394643   T   C  0.377   0.0041       0.0030  0.17    252971  10_71581019,6_cox_hap1_498457,6_qbl_hap2_300481
rs170485    A   G  0.517   0.0028       0.0032  0.37    250334  11_110164198,6_cox_hap1_3701029,6_qbl_hap2_3508865
rs12786590  A   G  0.542   -0.0031      0.0030  0.29    253213  11_120667035,6_cox_hap1_515920,6_qbl_hap2_317886
rs7941169   T   C  0.845   -0.014       0.0038  0.00033 253215  11_50174233,6_cox_hap1_219189,6_qbl_hap2_17020
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$join <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $9 }' | grep -v ChrBP | sort | uniq -u) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep -v , | grep -v NA | grep -v cox | sed 's/_/ /g' | awk '{ print $9 "\t" $10 "\t" $4 "\t" $2 "\t" $3 "\t" $5 "\t" $7 "\t" $8 }' | perl -lane 'if ($F[5] > 0) { $F[5] = "+"; } elsif ($F[5] < 0) { $F[5] = "-"; } else { $F[5] = 0;} if ($F[2] > .5) { ($F[3], $F[4]) = ($F[4], $F[3]); if ($F[5] eq "+") { $F[5] = "-"; } elsif ($F[5] eq "-") { $F[5] = "+"; } else { $PH = 1; } $F[2] = 1 - $F[2]; } print join("\t", @F); ' | grep -v ChrBP | grep -v Freq | awk '{ print $1 "_" $2 "\t" $0 }' | sort -k 1,1) | perl -lane 'splice(@F, 0, 1); print join("\t", @F);' | perl -lane 'if ($F[0] eq "X") { $F[0] = 23; } if ($F[0] eq "Y") { $F[0] = 24; } print join("\t", @F); ' | R -q -e "Data1 <- read.table(file('stdin'), header=F); quantile(Data1[,3]); quantile(Data1[,7]); quantile(Data1[,8]);"
> Data1 <- read.table(file('stdin'), header=F); quantile(Data1[,3]); quantile(Data1[,7]); quantile(Data1[,8]);
   0%   25%   50%   75%  100%
0.000 0.092 0.208 0.347 0.500
      0%      25%      50%      75%     100%
3.2e-158  9.9e-02  3.5e-01  6.6e-01  1.0e+00
    0%    25%    50%    75%   100%
 50003 241240 251654 253087 253280
>
>
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$join <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.gz | awk '{ print $9 }' | grep -v ChrBP | sort | uniq -u) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.gz | grep -v , | grep -v NA | grep -v cox | sed 's/_/ /g' | awk '{ print $9 "\t" $10 "\t" $4 "\t" $2 "\t" $3 "\t" $5 "\t" $7 "\t" $8 }' | perl -lane 'if ($F[5] > 0) { $F[5] = "+"; } elsif ($F[5] < 0) { $F[5] = "-"; } else { $F[5] = 0;} if ($F[2] > .5) { ($F[3], $F[4]) = ($F[4], $F[3]); if ($F[5] eq "+") { $F[5] = "-"; } elsif ($F[5] eq "-") { $F[5] = "+"; } else { $PH = 1; } $F[2] = 1 - $F[2]; } print join("\t", @F); ' | grep -v ChrBP | grep -v Freq | awk '{ print $1 "_" $2 "\t" $0 }' | sort -k 1,1) | perl -lane 'splice(@F, 0, 1); print join("\t", @F);' | perl -lane 'if ($F[0] eq "X") { $F[0] = 23; } if ($F[0] eq "Y") { $F[0] = 24; } print join("\t", @F); ' | R -q -e "Data1 <- read.table(file('stdin'), header=F); quantile(Data1[,3]); quantile(Data1[,7]); quantile(Data1[,8]);"
> Data1 <- read.table(file('stdin'), header=F); quantile(Data1[,3]); quantile(Data1[,7]); quantile(Data1[,8]);
    0%    25%    50%    75%   100%
0.0000 0.0917 0.2083 0.3500 0.5000
        0%        25%        50%        75%       100%
7.510e-153  2.301e-01  4.863e-01  7.444e-01  1.000e+00
      0%      25%      50%      75%     100%
 50004.7 223574.0 233565.0 233955.0 322154.0
>
>
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$join <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $9 }' | tail -n +5 | grep -v ChrBP | sort | uniq -u) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | tail -n +9 | grep -v , | grep -v NA | grep -v cox | sed 's/_/ /g' | awk '{ print $9 "\t" $10 "\t" $4 "\t" $2 "\t" $3 "\t" $5 "\t" $7 "\t" $8 }' | perl -lane 'if ($F[5] > 0) { $F[5] = "+"; } elsif ($F[5] < 0) { $F[5] = "-"; } else { $F[5] = 0;} if ($F[2] > .5) { ($F[3], $F[4]) = ($F[4], $F[3]); if ($F[5] eq "+") { $F[5] = "-"; } elsif ($F[5] eq "-") { $F[5] = "+"; } else { $PH = 1; } $F[2] = 1 - $F[2]; } print join("\t", @F); ' | grep -v ChrBP | grep -v Freq | awk '{ print $1 "_" $2 "\t" $0 }' | sort -k 1,1) | perl -lane 'splice(@F, 0, 1); print join("\t", @F);' | perl -lane 'if ($F[0] eq "X") { $F[0] = 23; } if ($F[0] eq "Y") { $F[0] = 24; } print join("\t", @F); ' | R -q -e "Data1 <- read.table(file('stdin'), header=F); quantile(Data1[,3]); quantile(Data1[,7]); quantile(Data1[,8]);"
> Data1 <- read.table(file('stdin'), header=F); quantile(Data1[,3]); quantile(Data1[,7]); quantile(Data1[,8]);
    0%    25%    50%    75%   100% 
0.0083 0.0917 0.2083 0.3500 0.5000 
     0%     25%     50%     75%    100% 
3.6e-35 2.4e-01 5.0e-01 7.5e-01 1.0e+00 
    0%    25%    50%    75%   100% 
 50003 134513 142356 142678 210086 
> 
> 
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep -v , | grep -v NA | grep -v cox | sed 's/_/ /g' | awk '{ print $9 "\t" $10 "\t" $4 "\t" $2 "\t" $3 "\t" $5 "\t" $7 "\t" $8 }' | head -n 10
ChrBP       Freq.Allele1.HapMapCEU      Allele1 Allele2 b  p    N
10 10000134 0.551  A    G  -0.0011      0.70    253213
10 10000264 0.436  T    C  0.0011       0.70    253213
10 100002728    0.367   A  G  -0.0062   0.042   253116
10 100002879    0.358   A  G  -0.0062   0.041   252156
10 100003552    0.12    T  C  -0.0087   0.024   248425
10 100003804    0.3 T   C  0.014        8.2e-06 251271
10 100003966    0.644   A  T  0.0067    0.029   253086
10 10000458 0.427  T    G  0.0011       0.70    253213
10 100005552    0.144   C  G  -0.0022   0.62    253056
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep -v , | grep -v NA | grep -v cox | sed 's/_/ /g' | awk '{ print $9 "\t" $10 "\t" $4 "\t" $2 "\t" $3 "\t" $5 "\t" $7 "\t" $8 }' | perl -lane 'if ($F[5] > 0) { $F[5] = "+"; } elsif ($F[5] < 0) { $F[5] = "-"; } else { $F[5] = 0;} if ($F[2] > .5) { ($F[3], $F[4]) = ($F[4], $F[3]); if ($F[5] eq "+") { $F[5] = "-"; } elsif ($F[5] eq "-") { $F[5] = "+"; } else { $PH = 1; } $F[2] = 1 - $F[2]; } print join("\t", @F); ' | head -n 10
ChrBP   Freq.Allele1.HapMapCEU  Allele1 Allele2 b       0  N
10 10000134 0.449  G    A  +  0.70      253213
10 10000264 0.436  T    C  +  0.70      253213
10 100002728    0.367   A  G  - 0.042   253116
10 100002879    0.358   A  G  - 0.041   252156
10 100003552    0.12    T  C  - 0.024   248425
10 100003804    0.3 T   C  +  8.2e-06   251271
10 100003966    0.356   T  A  - 0.029   253086
10 10000458 0.427  T    G  +  0.70      253213
10 100005552    0.144   C  G  - 0.62    253056
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep -v , | grep -v NA | grep -v cox | sed 's/_/ /g' | awk '{ print $9 "\t" $10 "\t" $4 "\t" $2 "\t" $3 "\t" $5 "\t" $7 "\t" $8 }' | tail -n 10
1  187536162    0.446   A  G  -0.0053   0.38    62113
4  179992349    0.992   T  C  0.027     0.0059  140223
2  62726619 0.241  A    G  -0.0032      0.62    73844
4  143260026    0.119   A  T  -0.015    0.091   73853
5  87454155 0.827  A    C  -0.0041      0.58    56282
2  209836736    0.967   A  G  -0.012    0.50    73062
1  65362420 0.769  T    C  0.013        0.057   73851
2  233399107    0.678   T  C  0.011     0.058   73845
1  228200511    0.2 A   G  0.024        0.0028  62111
2  189895357    0.033   A  G  -0.00090  0.95    69057
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep -v , | grep -v NA | grep -v cox | sed 's/_/ /g' | awk '{ print $9 "\t" $10 "\t" $4 "\t" $2 "\t" $3 "\t" $5 "\t" $7 "\t" $8 }' | perl -lane 'if ($F[5] > 0) { $F[5] = "+"; } elsif ($F[5] < 0) { $F[5] = "-"; } else { $F[5] = 0;} if ($F[2] > .5) { ($F[3], $F[4]) = ($F[4], $F[3]); if ($F[5] eq "+") { $F[5] = "-"; } elsif ($F[5] eq "-") { $F[5] = "+"; } else { $PH = 1; } $F[2] = 1 - $F[2]; } print join("\t", @F); ' | tail -n 10
1  187536162    0.446   A  G  - 0.38    62113
4  179992349    0.00800000000000001     C       T       -  0.0059       140223
2  62726619 0.241  A    G  -  0.62      73844
4  143260026    0.119   A  T  - 0.091   73853
5  87454155 0.173  C    A  +  0.58      56282
2  209836736    0.033   G  A  + 0.50    73062
1  65362420 0.231  C    T  -  0.057     73851
2  233399107    0.322   C  T  - 0.058   73845
1  228200511    0.2 A   G  +  0.0028    62111
2  189895357    0.033   A  G  - 0.95    69057
> dim(bmassOutput1$NewSNPs$SNPs)
[1] 217  25
> dim(bmassOutput1$PreviousSNPs$SNPs)
[1] 793  25
> dim(GWASsnps)
[1] 843   2
> dim(bmassOutput3$MergedDataSources)
[1] 2418714      19
> bmassOutput1$GWASlogBFMinThreshold
[1] 2.855602
> bmassOutput1$ZScoresCorMatrix
                 Height_ZScore  BMI_ZScore WHRadjBMI_ZScore
Height_ZScore      1.000000000 -0.05101508     -0.009438275
BMI_ZScore        -0.051015076  1.00000000      0.014631505
WHRadjBMI_ZScore  -0.009438275  0.01463151      1.000000000
~~~


































##PGC 2013 dbsnp ###, GWAS = , N = 
##20150601

mkdir /mnt/gluster/data/external_public_supp/PGC2013 
cd /mnt/gluster/data/external_public_supp/PGC2013
#Accessed 20150601
wget https://www.med.unc.edu/pgc/files/resultfiles/pgc.cross.full.2013-03.zip
wget https://www.med.unc.edu/pgc/files/resultfiles/pgc.cross.add.zip
wget https://www.med.unc.edu/pgc/files/resultfiles/pgc.cross.aut.zip
wget https://www.med.unc.edu/pgc/files/resultfiles/pgc.cross.bip.zip
wget https://www.med.unc.edu/pgc/files/resultfiles/pgc.cross.mdd.zip
wget https://www.med.unc.edu/pgc/files/resultfiles/pgc.cross.scz.zip
wget https://www.med.unc.edu/pgc/files/resultfiles/pgc-cross-scoring-5files.zip

unzip /mnt/gluster/data/external_public_supp/PGC2013/pgc.cross.full.2013-03.zip
unzip /mnt/gluster/data/external_public_supp/PGC2013/pgc.cross.add.zip
unzip /mnt/gluster/data/external_public_supp/PGC2013/pgc.cross.aut.zip
unzip /mnt/gluster/data/external_public_supp/PGC2013/pgc.cross.bip.zip
unzip /mnt/gluster/data/external_public_supp/PGC2013/pgc.cross.mdd.zip
unzip /mnt/gluster/data/external_public_supp/PGC2013/pgc.cross.scz.zip
unzip /mnt/gluster/data/external_public_supp/PGC2013/pgc-cross-scoring-5files.zip

#~~~
[  mturchin20@spudhead  /mnt/gluster/data/external_public_supp/PGC2013]$cat pgc.cross.ADD4.2013-05.txt | head -n 5
snpid hg18chr bp a1 a2 or se pval info ngt CEUaf
rs3131972       1       742584  A       G       1.137   0.1189  0.2799  0.731   0       0.16055
rs3131969       1       744045  A       G       1.13    0.1136  0.2813  0.966   0       0.133028
rs3131967       1       744197  T       C       1.14    0.1208  0.2794  0.899   0       .
rs1048488       1       750775  T       C       0.8794  0.1189  0.2798  0.731   0       0.836449
[  mturchin20@spudhead  /mnt/gluster/data/external_public_supp/PGC2013]$cat pgc.cross.AUT8.2013-05.txt | head -n 5
snpid hg18chr bp a1 a2 or se pval info ngt CEUaf
rs3131972       1       742584  A       G       1.03    0.0478  0.5351  0.972   0       0.16055
rs3131969       1       744045  A       G       1.067   0.0538  0.2268  0.926   0       0.133028
rs3131967       1       744197  T       C       1.082   0.0579  0.1754  0.854   0       .
rs1048488       1       750775  T       C       0.9829  0.048   0.7195  0.974   0       0.836449
[  mturchin20@spudhead  /mnt/gluster/data/external_public_supp/PGC2013]$cat pgc.cross.BIP11.2013-05.txt | head -n 5
snpid hg18chr bp a1 a2 or se pval info ngt CEUaf
rs3131972       1       742584  A       G       1.092   0.0817  0.2819  0.694   0       0.16055
rs3131969       1       744045  A       G       1.087   0.0781  0.2855  0.939   0       0.133028
rs3131967       1       744197  T       C       1.093   0.0835  0.2859  0.869   0       .
rs1048488       1       750775  T       C       0.9158  0.0817  0.2817  0.694   0       0.836449
[  mturchin20@spudhead  /mnt/gluster/data/external_public_supp/PGC2013]$cat pgc.cross.MDD9.2013-05.txt | head -n 5
snpid hg18chr bp a1 a2 or se pval info ngt CEUaf
rs12562034      1       758311  A       G       0.9028  0.0894  0.2527  1.02    0       0.0925926
rs4970383       1       828418  A       C       1.018   0.0769  0.8186  0.436   0       0.201835
rs4475691       1       836671  T       C       1.016   0.0546  0.7751  0.979   0       0.146789
rs1806509       1       843817  A       C       0.99    0.0737  0.8911  0.372   0       0.600917
[  mturchin20@spudhead  /mnt/gluster/data/external_public_supp/PGC2013]$cat pgc.cross.SCZ17.2013-05.txt | head -n 5
snpid hg18chr bp a1 a2 or se pval info ngt CEUaf
rs3131972       1       742584  A       G       1       0.0966  0.9991  0.702   0       0.16055
rs3131969       1       744045  A       G       1       0.0925  0.9974  0.938   0       0.133028
rs3131967       1       744197  T       C       1.001   0.0991  0.9928  0.866   0       .
rs1048488       1       750775  T       C       0.9999  0.0966  0.9991  0.702   0       0.836449
#~~~

##IBD 2012
##20150601

mkdir /mnt/gluster/data/external_public_supp/IBDConsortium2012
cd /mnt/gluster/data/external_public_supp/IBDConsortium2012
#Accessed 20150601
wget ftp://ftp.sanger.ac.uk/pub4/ibdgenetics/gwas_ichip_meta_release.txt.gz
#Accessed 20150602
wget ftp://ftp.sanger.ac.uk/pub4/ibdgenetics/cd-meta.txt.gz
wget ftp://ftp.sanger.ac.uk/pub4/ibdgenetics/ucmeta-sumstats.txt.gz

#~~~
[  mturchin20@spudhead  /mnt/gluster/data/external_public_supp/IBD2012]$zcat gwas_ichip_meta_release.txt.gz | head -n 10
CHR BP GWAS_SNP GWAS_A1A2 ICHIP_SNP ICHIP_A1A2 LD-info Strand/Effectallele CD_GWAS_P CD_ICHIP_P CD_META_P UC_GWAS_P UC_ICHIP_P UC_META_P IBD_GWAS_P IBD_ICHIP_P IBD_META_P MAF CD_OR CD_OR_CI UC_OR UC_OR_CI IBD_OR IBD_OR_CI
1 1110294 rs1320571 AG same AG same same_strand,_same_effect-allele - - - - - - 0.004646 0.8236 0.105 0.0442 - - - - 1.05 0.989652-1.11403
1 1125105 rs9729550 AC same CA same same_strand,_different_effect-allele 0.001286 0.03004 0.000316 - - - - - - 0.738 0.959692898272553 0.924956-0.995735 - - - -
1 1130298 rs1815606 TG same AC same strand_flip,_same_effect-allele 0.0001352 0.04842 0.0002186 - - - 0.00123 0.009144 6.327e-05 0.31 1.036 1.00062-1.07263 - - 1.051 1.02556-1.07707
1 1142494 rs11721 AC rs6697886 AG (0.485,AA/CG) same_strand,_same_effect-allele 0.003183 0.0003092 5.165e-06 - - - 0.004786 1.81e-06 2.954e-08 0.0781 1.089 1.03957-1.14078 - - 1.102 1.06484-1.14045
1 1153667 rs7515488 TC same AG same strand_flip,_same_effect-allele 0.000283 0.0002496 3.62e-07 0.006893 1.753e-05 4.92e-07 6.046e-05 3.443e-06 7.67e-10 0.147 1.088 1.04018-1.13801 1.11 1.05823-1.16431 1.1 1.06708-1.13393
1 1155173 rs11260562 AG same AG same same_strand,_same_effect-allele 0.0003004 0.08256 0.0004346 0.005253 0.04672 0.001047 1.053e-05 0.02667 6.605e-06 0.0546 1.063 0.992095-1.13897 1.077 1.00084-1.15896 1.114 1.06302-1.16743
1 1163474 rs6697886 AG same AG same same_strand,_same_effect-allele 0.00109 0.0003092 1.57e-06 - - - 0.0003566 1.81e-06 2.441e-09 0.131 1.089 1.03957-1.14078 - - 1.101 1.06659-1.13652
1 1164145 rs4970364 TC rs11804831 GA (0.627,CC/TT) strand_flip,_different_effect-allel 6.571e-05 0.0005104 2.77e-07 0.004123 5.014e-07 1.243e-08 5.14e-05 9.839e-07 2.326e-10 0.76 0.928505106778087 0.890662-0.967956 0.892857142857143 0.854237-0.933223 0.9138 0.888718-0.93959
1 1166460 rs6675798 TC rs6697886 AG (0.657,CA/TG) same_strand,_same_effect-allele - - - - - - 0.001967 1.81e-06 1.267e-08 0.9 - - - - 0.9069 0.876836-0.937995
#~~~

#Got links for the updated IDBGC information from Stephan Ripke & co -- see e-mail change copy/pasted below near end of this record file
#Accessed 20150618

mkdir /mnt/gluster/data/external_public_supp/IBDConsortium2015
cd /mnt/gluster/data/external_public_supp/IBDConsortium2015

wget http://www.broadinstitute.org/~sripke/share_links/ALHySN0Z6i4O6PZeQgpqXN9Ewtqxit_IBD15_1KG.sh2/daner_IBD15_1KG.sh2_mds10.CD7b.gz
wget http://www.broadinstitute.org/~sripke/share_links/ALHySN0Z6i4O6PZeQgpqXN9Ewtqxit_IBD15_1KG.sh2/daner_IBD15_1KG.sh2_mds10.UC8b.gz
wget http://www.broadinstitute.org/~sripke/share_links/ALHySN0Z6i4O6PZeQgpqXN9Ewtqxit_IBD15_1KG.sh2/daner_IBD15_1KG.sh2_mds15.IBD15bb.gz
wget http://www.broadinstitute.org/~sripke/share_links/ALHySN0Z6i4O6PZeQgpqXN9Ewtqxit_IBD15_1KG.sh2/download_all.txt

~~~
[  mturchin20@spudhead  /mnt/gluster/data/external_public_supp/IBDConsortium2015]$zcat daner_IBD15_1KG.sh2_mds10.CD7b.gz | head -n 10
CHR     SNP     BP      A1   A2 FRQ_A_5956      FRQ_U_14927     INFO    OR      SE      P       ngt     Direction       HetISqt HetChiSq        HetDf   HetPVa
10      rs185339560     2392426 T       C       0.00954 0.00942 0.633   0.91411 0.1485  0.5453  0       +----++ 22.5    10.316  6       0.112
10      rs11250701      1689546 A       G       0.679   0.669   0.962   1.02398 0.0248  0.3394  0       -++---- -14.0   7.018   6       0.3191
10      chr10_2622752_D 2622752 I2      D       0.97    0.968   0.929   1.06184 0.0691  0.3857  0       -+-+--- 37.9    12.884  6       0.04492
10      rs7085086       151476  A       G       0.319   0.317   0.935   0.99452 0.0253  0.8283  0       ---+-++ 0.0     4.809   6       0.5685
10      rs113494187     1593759 T       G       0.986   0.985   0.829   1.00391 0.1064  0.9708  0       --++++- -30.4   6.136   6       0.4081
10      rs117915320     1708106 A       C       0.0113  0.012   0.616   1.01227 0.1373  0.9292  0       -+--+++ 0.0     5.392   6       0.4946
10      rs182753344     790310  T       C       0.0993  0.0996  0.606   1.04561 0.0489  0.3612  0       -++++++ 0.0     2.893   6       0.8221
10      rs188913771     1273049 A       G       0.0126  0.0131  0.672   0.88577 0.1253  0.3331  0       -----++ -7.5    7.443   6       0.2818
10      rs7911665       2067236 T       G       0.681   0.686   0.915   0.98442 0.0254  0.5368  0       ---+-+- 0.0     1.667   6       0.9476
[  mturchin20@spudhead  /mnt/gluster/data/external_public_supp/IBDConsortium2015]$zcat daner_IBD15_1KG.sh2_mds10.UC8b.gz | head -n 10
CHR     SNP     BP      A1   A2 FRQ_A_6968      FRQ_U_20464     INFO    OR      SE      P       ngt     Direction       HetISqt HetChiSq        HetDf   HetPVa
10      rs185339560     2392426 T       C       0.00802 0.00774 0.626   1.03076 0.1554  0.8452  0       --+++--?        0.0     3.126   6       0.7928
10      rs11250701      1689546 A       G       0.669   0.668   0.961   1.00833 0.0229  0.7183  0       -+-+--+-        -2.6    8.776   7       0.2692
10      chr10_2622752_D 2622752 I2      D       0.968   0.969   0.93    0.91229 0.063   0.1452  0       -+-+++-+        12.1    10.239  7       0.1754
10      rs7085086       151476  A       G       0.311   0.313   0.971   1.00070 0.0231  0.9747  0       -+-++-++        -19.6   7.523   7       0.3765
10      rs113494187     1593759 T       G       0.987   0.985   0.866   1.07993 0.0979  0.4321  0       ++--++++        0.0     1.801   7       0.97
10      rs117915320     1708106 A       C       0.0125  0.0121  0.642   1.05686 0.1208  0.6472  0       --++-+++        -15.5   7.791   7       0.3514
10      rs182753344     790310  T       C       0.0948  0.0998  0.616   0.94233 0.0458  0.1951  0       ----+++-        0.0     5.427   7       0.608
10      rs188913771     1273049 A       G       0.01    0.0113  0.629   0.96194 0.1349  0.7735  0       ++---+-+        0.0     1.865   7       0.9669
10      rs7911665       2067236 T       G       0.691   0.689   0.92    1.03159 0.0238  0.1909  0       +-+++-+-        23.6    11.785  7       0.1078
[  mturchin20@spudhead  /mnt/gluster/data/external_public_supp/IBDConsortium2015]$zcat daner_IBD15_1KG.sh2_mds15.IBD15bb.gz | head -n 10
CHR     SNP     BP      A1   A2 FRQ_A_12882     FRQ_U_21770     INFO    OR      SE      P       ngt     Direction       HetISqt HetChiSq        HetDf   HetPVa
10      rs185339560     2392426 T       C       0.00879 0.00848 0.634   0.92090 0.1184  0.4865  0       -----+-+-++--+? 25.3    20.073  13      0.09342
10      rs11250701      1689546 A       G       0.673   0.668   0.957   1.01440 0.0182  0.4314  0       --+++--+-+--++- 0.6     16.102  14      0.3072
10      chr10_2622752_D 2622752 I2      D       0.969   0.968   0.923   0.97385 0.0504  0.5986  0       --+-+-+--+-+--+ 22.7    20.704  14      0.1095
10      rs7085086       151476  A       G       0.315   0.316   0.961   0.99302 0.0184  0.7057  0       ----+-+-+-+-+++ -5.3    15.196  14      0.3649
10      rs113494187     1593759 T       G       0.987   0.985   0.849   1.05085 0.0775  0.5225  0       -+++--+-+++++-+ 0.0     8.947   14      0.8344
10      rs117915320     1708106 A       C       0.0118  0.0119  0.623   1.01990 0.0992  0.8423  0       --+--+-++-+++++ 0.0     10.974  14      0.6881
10      rs182753344     790310  T       C       0.0969  0.0988  0.612   0.98639 0.0361  0.705   0       --++--+-++-++-- 0.0     9.530   14      0.7957
10      rs188913771     1273049 A       G       0.0112  0.0117  0.638   0.90620 0.1001  0.325   0       ----+--+--++-++ 0.0     7.930   14      0.8929
10      rs7911665       2067236 T       G       0.686   0.689   0.917   1.01319 0.0188  0.4844  0       -++---+++++-+-- 0.0     12.066  14      0.601
[  mturchin20@spudhead  /mnt/gluster/data/external_public_supp/IBDConsortium2015]$zcat daner_IBD15_1KG.sh2_mds10.CD7b.gz | wc
12276507 208700614 1186149792
[  mturchin20@spudhead  /mnt/gluster/data/external_public_supp/IBDConsortium2015]$zcat daner_IBD15_1KG.sh2_mds10.UC8b.gz | wc
12255264 208339153 1197560812
[  mturchin20@spudhead  /mnt/gluster/data/external_public_supp/IBDConsortium2015]$zcat daner_IBD15_1KG.sh2_mds15.IBD15bb.gz | wc
12716151 216174232 1348769658
~~~












##TAG 2010 dbsnp ###, GWAS = , N = 74053
##20150601

mkdir /mnt/gluster/data/external_public_supp/TAG2010
cd /mnt/gluster/data/external_public_supp/TAG2010
#Accessed 20150601
wget https://www.med.unc.edu/pgc/files/resultfiles/readme.tag.txt
wget https://www.med.unc.edu/pgc/files/resultfiles/tag.cpd.tbl.gz
wget https://www.med.unc.edu/pgc/files/resultfiles/tag.evrsmk.tbl.gz
wget https://www.med.unc.edu/pgc/files/resultfiles/tag.former.tbl.gz
wget https://www.med.unc.edu/pgc/files/resultfiles/tag.logonset.tbl.gz

#~~~
[  mturchin20@spudhead  /mnt/gluster/data/external_public_supp/TAG2010]$zcat tag.logonset.tbl.gz | head -n 5
CHR     SNP     BP      A1   A2 FRQ_A   FRQ_U   INFO    OR      SE      P
1       rs12565286      711153  C       G       0.0434  0.0434  1       -0.0054 0.0123  0.6592
1       rs11804171      713682  A       T       0.0434  0.0434  1       -0.0049 0.0123  0.6912
1       rs3094315       742429  A       G       0.8211  0.8211  1       -0.0046 0.0042  0.2692
1       rs3131968       744055  A       G       0.1462  0.1462  1       -0.0141 0.0101  0.1626
[  mturchin20@spudhead  /mnt/gluster/data/external_public_supp/TAG2010]$zcat tag.former.tbl.gz | head -n 5
CHR     SNP     BP      A1   A2 FRQ_A   FRQ_U   INFO    OR      SE      P
1       rs12565286      711153  C       G       0.0457  0.0457  1       0.0533  0.0934  0.568
1       rs11804171      713682  A       T       0.0457  0.0457  1       0.0553  0.0937  0.5547
1       rs3094315       742429  A       G       0.8238  0.8238  1       -0.0479 0.0316  0.1301
1       rs3131968       744055  A       G       0.1346  0.1346  1       0.0912  0.0529  0.08477
[  mturchin20@spudhead  /mnt/gluster/data/external_public_supp/TAG2010]$zcat tag.evrsmk.tbl.gz | head -n 5
CHR     SNP     BP      A1   A2 FRQ_A   FRQ_U   INFO    OR      SE      P
1       rs12565286      711153  C       G       0.241   0.241   1       -0.0885 0.0736  0.2293
1       rs11804171      713682  A       T       0.2394  0.2394  1       -0.089  0.0739  0.229
1       rs3094315       742429  A       G       0.7737  0.7737  1       -0.0031 0.0237  0.8949
1       rs3131968       744055  A       G       0.1349  0.1349  1       -0.021  0.0435  0.6293
[  mturchin20@spudhead  /mnt/gluster/data/external_public_supp/TAG2010]$zcat tag.cpd.tbl.gz | head -n 5
CHR     SNP     BP      A1   A2 FRQ_A   FRQ_U   INFO    OR      SE      P
1       rs12565286      711153  C       G       0.0434  0.0434  1       -0.4029 0.5634  0.4745
1       rs11804171      713682  A       T       0.0434  0.0434  1       -0.4092 0.5649  0.4689
1       rs3094315       742429  A       G       0.824   0.824   1       -0.035  0.174   0.8407
1       rs3131968       744055  A       G       0.1348  0.1348  1       0.2606  0.3068  0.3956
#~~~

##20150830 --- Original location of these lines was below
#mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/TAG2010 
#mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/TAG2010/Vs1 
#
#cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/PGC2013/Vs1/process.MTedits.ForPGC2013.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/TAG2010/Vs1/process.MTedits.ForTAG2010.vs1.R 
#
#mv /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/TAG2010/Vs1/.RData /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/TAG2010/Vs1/RData.TAG2010.process.20150602
#gzip /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/TAG2010/Vs1/RData.TAG2010.process.20150602

#20150830 CHECK_0: Do QC stuff on TAG data
#20150830 CHECK_0: Maybe can just do 'N = max N * (1-stderr^2)' ?

#cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/TAG2010/Vs1/process.MTedits.ForTAG2010.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/TAG2010/Vs1/process.MTedits.ForTAG2010.vs2.R

zcat /mnt/gluster/data/external_public_supp/TAG2010/tag.logonset.tbl.gz | awk '{ print $1, "\t", $3, "\t", $2, "\t", $6, "\t", $11, "\t74053" }' | gzip >  /mnt/gluster/data/external_public_supp/TAG2010/tag.logonset.formatted.maxN.tbl.gz
zcat /mnt/gluster/data/external_public_supp/TAG2010/tag.former.tbl.gz | awk '{ print $1, "\t", $3, "\t", $2, "\t", $6, "\t", $11, "\t74053" }' | gzip >  /mnt/gluster/data/external_public_supp/TAG2010/tag.former.formatted.maxN.tbl.gz
zcat /mnt/gluster/data/external_public_supp/TAG2010/tag.evrsmk.tbl.gz | awk '{ print $1, "\t", $3, "\t", $2, "\t", $6, "\t", $11, "\t74053" }' | gzip >  /mnt/gluster/data/external_public_supp/TAG2010/tag.evrsmk.formatted.maxN.tbl.gz
zcat /mnt/gluster/data/external_public_supp/TAG2010/tag.cpd.tbl.gz | awk '{ print $1, "\t", $3, "\t", $2, "\t", $6, "\t", $11, "\t74053" }' | gzip >  /mnt/gluster/data/external_public_supp/TAG2010/tag.cpd.formatted.maxN.tbl.gz

~~~
[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/TAG2010]$zcat /mnt/gluster/data/external_public_supp/TAG2010/tag.cpd.tbl.gz | perl -lane 'my $SEovrBet; if ($F[8] != 0) { $SEovrBet = $F[9] / $F[8]; } else { $SEovrBet = $F[9]; } print $F[8], "\t", $F[9], "\t", abs($SEovrBet);' | sort -k 3,3 -rg | head -n 10
0.0001  4.4321  44321
-0.0002 2.4013  12006.5
0.0001  0.9381  9381
-0.0001 0.9091  9091
-0.0001 0.8273  8273
0.0001  0.7681  7681
-0.0007 5.1579  7368.42857142857
0.0002  1.4694  7347
0.0001  0.5955  5955
-0.0002 1.1889  5944.5
join <(zcat /mnt/gluster/data/external_public_supp/TAG2010/tag.logonset.IncAllele.tbl.gz | awk '{ print $2, "\t", $4 }' | sort) <(zcat /mnt/gluster/data/external_public_supp/TAG2010/tag.evrsmk.tbl.gz | perl -lane 'if ($F[8] < 0) { ($F[3], $F[4]) = ($F[4], $F[3]); $F[5] = 1 - $F[5]; $F[8] = $F[8] * -1; } ($F[0], $F[1]) = ($F[1], $F[0]); print join("\t", @F);' | sort) | perl -lane 'if ($F[1] ne $F[4]) { $F[11] = $F[11] * -1; } splice(@F, 1, 1); ($F[0], $F[1]) = ($F[1], $F[0]); print join("\t", @F);' | grep -v -w NA | gzip > /mnt/gluster/data/external_public_supp/TAG2010/tag.evrsmk.MatchedTologonset.IncAllele.tbl.gz
join <(zcat /mnt/gluster/data/external_public_supp/TAG2010/tag.logonset.IncAllele.tbl.gz | awk '{ print $2, "\t", $4 }' | sort) <(zcat /mnt/gluster/data/external_public_supp/TAG2010/tag.cpd.tbl.gz | perl -lane 'if ($F[8] < 0) { ($F[3], $F[4]) = ($F[4], $F[3]); $F[5] = 1 - $F[5]; $F[8] = $F[8] * -1; } ($F[0], $F[1]) = ($F[1], $F[0]); print join("\t", @F);' | sort) | perl -lane 'if ($F[1] ne $F[4]) { $F[11] = $F[11] * -1; } splice(@F, 1, 1); ($F[0], $F[1]) = ($F[1], $F[0]); print join("\t", @F);' | grep -v -w NA | gzip > /mnt/gluster/data/external_public_supp/TAG2010/tag.cpd.MatchedTologonset.IncAllele.tbl.gz

LANG=${LANG2}
LC_ALL=${LC_ALL2}
export LC_ALL LANG

~~~
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/TAG2010/Vs1]$zcat /mnt/gluster/data/external_public_supp/TAG2010/tag.logonset.IncAllele.tbl.gz | sort -k 2,2 | head -n 10
CHR     SNP     BP      A1      A2      FRQ_A   FRQ_U   INFO    OR      SE      P
12      rs1000000       125456933       A       G       0.2149  0.2149  1       0.0015  0.0028  0.5847
4       rs10000010      21227772        C       T       0.503   0.497   1       0.0019  0.0022  0.3898
4       rs10000012      1347325 G       C       0.1252  0.8748  1       0.0018  0.0035  0.6073
4       rs10000013      36901464        A       C       0.7617  0.7617  1       0.0011  0.0027  0.6714
4       rs10000017      84997149        T       C       0.2181  0.2181  1       0.0024  0.0028  0.4005
3       rs1000002       185118462       T       C       0.4848  0.4848  1       0.0003  0.0023  0.8782
4       rs10000023      95952929        T       G       0.6007  0.6007  1       0.0037  0.0023  0.1137
4       rs10000029      138905074       C       T       0.0178  0.9822  1       0.0073  0.0152  0.6312
3       rs1000003       99825597        G       A       0.1601  0.8399  1       0.0001  0.0031  0.9695
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/TAG2010/Vs1]$zcat /mnt/gluster/data/external_public_supp/TAG2010/tag.former.MatchedTologonset.IncAllele.tbl.gz | head -n 10
CHR     SNP     BP      A1      A2      FRQ_A   FRQ_U   INFO    OR      SE      P
12      rs1000000       125456933       A       G       0.2174  0.2174  1       0.027   0.0194  0.1647
4       rs10000010      21227772        C       T       0.498   0.502   1       0.0052  0.0158  0.7428
4       rs10000012      1347325 G       C       0.1304  0.8696  1       0.0181  0.0242  0.454
4       rs10000013      36901464        C       A       0.2327  0.7673  1       0.0049  0.0191  -0.7969
4       rs10000017      84997149        T       C       0.2208  0.2208  1       0.0101  0.0211  0.631
3       rs1000002       185118462       C       T       0.5149  0.4851  1       0.0262  0.016   -0.1004
4       rs10000023      95952929        G       T       0.4025  0.5975  1       0.0054  0.0163  -0.7399
4       rs10000029      138905074       C       T       0.0184  0.9816  1       0.0176  0.1325  0.8943
3       rs1000003       99825597        G       A       0.1574  0.8426  1       0.0006  0.0218  0.9781
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/TAG2010/Vs1]$zcat /mnt/gluster/data/external_public_supp/TAG2010/tag.evrsmk.MatchedTologonset.IncAllele.tbl.gz | head -n 10
CHR     SNP     BP      A1      A2      FRQ_A   FRQ_U   INFO    OR      SE      P
12      rs1000000       125456933       G       A       0.7634  0.2366  1       0.0063  0.0141  -0.6532
4       rs10000010      21227772        C       T       0.4965  0.5035  1       0.0105  0.0116  0.3663
4       rs10000012      1347325 G       C       0.1637  0.8363  1       0.0131  0.0177  0.4572
4       rs10000013      36901464        C       A       0.2492  0.7508  1       0.0009  0.0139  -0.9479
4       rs10000017      84997149        T       C       0.2451  0.2451  1       0.0086  0.0152  0.571
3       rs1000002       185118462       T       C       0.4801  0.4801  1       0.0176  0.0116  0.1291
4       rs10000023      95952929        G       T       0.4126  0.5874  1       0.0041  0.0118  -0.7252
4       rs10000029      138905074       C       T       0.1855  0.8145  1       0.0099  0.1235  0.9362
3       rs1000003       99825597        A       G       0.8152  0.8152  1       0.0077  0.0159  -0.6266
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/TAG2010/Vs1]$zcat /mnt/gluster/data/external_public_supp/TAG2010/tag.cpd.MatchedTologonset.IncAllele.tbl.gz | head -n 10
CHR     SNP     BP      A1      A2      FRQ_A   FRQ_U   INFO    OR      SE      P
12      rs1000000       125456933       G       A       0.7836  0.2164  1       0.1189  0.0995  -0.2322
4       rs10000010      21227772        C       T       0.4972  0.5028  1       0.0847  0.0809  0.2955
4       rs10000012      1347325 G       C       0.1308  0.8692  1       0.0459  0.1225  0.7082
4       rs10000013      36901464        C       A       0.231   0.769   1       0.1827  0.0986  -0.06375
4       rs10000017      84997149        T       C       0.2209  0.2209  1       0.0564  0.1064  0.5958
3       rs1000002       185118462       C       T       0.5145  0.4855  1       0.0489  0.0819  -0.5508
4       rs10000023      95952929        G       T       0.4026  0.5974  1       0.0554  0.0841  -0.51
4       rs10000029      138905074       T       C       0.9827  0.9827  1       0.6773  0.7148  -0.3434
3       rs1000003       99825597        A       G       0.8435  0.8435  1       0.1192  0.1125  -0.2893
~~~

zcat /mnt/gluster/data/external_public_supp/TAG2010/tag.logonset.IncAllele.tbl.gz | awk '{ print $1, "\t", $3, "\t", $2, "\t", $6, "\t", $11, "\t74053" }' | gzip > /mnt/gluster/data/external_public_supp/TAG2010/tag.logonset.IncAllele.formatted.maxN.tbl.gz
zcat /mnt/gluster/data/external_public_supp/TAG2010/tag.former.MatchedTologonset.IncAllele.tbl.gz | awk '{ print $1, "\t", $3, "\t", $2, "\t", $6, "\t", $11, "\t74053" }' | gzip > /mnt/gluster/data/external_public_supp/TAG2010/tag.former.MatchedTologonset.IncAllele.formatted.maxN.tbl.gz
zcat /mnt/gluster/data/external_public_supp/TAG2010/tag.evrsmk.MatchedTologonset.IncAllele.tbl.gz | awk '{ print $1, "\t", $3, "\t", $2, "\t", $6, "\t", $11, "\t74053" }' | gzip > /mnt/gluster/data/external_public_supp/TAG2010/tag.evrsmk.MatchedTologonset.IncAllele.formatted.maxN.tbl.gz
zcat /mnt/gluster/data/external_public_supp/TAG2010/tag.cpd.MatchedTologonset.IncAllele.tbl.gz | awk '{ print $1, "\t", $3, "\t", $2, "\t", $6, "\t", $11, "\t74053" }' | gzip > /mnt/gluster/data/external_public_supp/TAG2010/tag.cpd.MatchedTologonset.IncAllele.formatted.maxN.tbl.gz


LANG2=${LANG}
LC_ALL2=${LC_ALL}
LANG=C
LC_ALL=C
export LC_ALL LANG

join -a 1 -a 2 -1 3 -2 3 -e NA2 -o 0 1.1 1.2 1.4 1.5 1.6 2.1 2.2 2.4 2.5 2.6 <(zcat /mnt/gluster/data/external_public_supp/TAG2010/tag.logonset.IncAllele.formatted.maxN.tbl.gz | sort -g -k 3,3) <(zcat /mnt/gluster/data/external_public_supp/TAG2010/tag.former.MatchedTologonset.IncAllele.formatted.maxN.tbl.gz | sort -g -k 3,3 ) | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
join -a 1 -a 2 -1 1 -2 3 -e NA2 -o 0 1.2 2.1 2.2 2.4 2.5 2.6 - <(zcat /mnt/gluster/data/external_public_supp/TAG2010/tag.evrsmk.MatchedTologonset.IncAllele.formatted.maxN.tbl.gz  | sort -g -k 3,3) | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); print join("\t", @F);' | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); if ($#F == 6) { splice(@F, 1, 0, (("NA2") x 9)); } my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
join -a 1 -a 2 -1 1 -2 3 -e NA2 -o 0 1.2 2.1 2.2 2.4 2.5 2.6 - <(zcat /mnt/gluster/data/external_public_supp/TAG2010/tag.cpd.MatchedTologonset.IncAllele.formatted.maxN.tbl.gz | sort -g -k 3,3) | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); print join("\t", @F);' | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); if ($#F == 6) { splice(@F, 1, 0, (("NA2") x 14)); } my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
sed 's/FRQ_A/RAF/g' | perl -lane 'print $F[0], "\t", join("\t", split(/_/, $F[1]));' | gzip > /mnt/gluster/data/external_public_supp/TAG2010/TAG2010.AllPhenos.MatchedTologonset.IncAllele.formatted.RAF.txt.pre1.gz

LANG=${LANG2}
LC_ALL=${LC_ALL2}
export LC_ALL LANG

~~~
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/TAG2010/Vs1]$zcat /mnt/gluster/data/external_public_supp/TAG2010/TAG2010.AllPhenos.MatchedTologonset.IncAllele.formatted.RAF.txt.pre1.gz | head -n 10
rs1172982  1    100002699       0.6717  0.7307  74053   1       100002699       0.6656  0.1343  74053   1 100002699     0.6563  0.6712  74053   1 100002699     0.6681  0.6283  74053
rs1172981  1    100002785       0.0736  0.5609  74053   1       100002785       0.9214  -0.2204 74053   1 100002785     0.8893  -0.9351 74053   1 100002785     0.9227  -0.849  74053
rs11166327 1    100003455       0.0691000000000001      0.6123  74053   1       100003455       0.9284  -0.2026 74053   1 100003455     0.8962  -0.6423 74053   1       100003455 0.9299        -0.9616 74053
rs4908018  1    100006955       0.0738  0.5364  74053   1       100006955       0.9211  -0.2506 74053   1 100006955     0.8891  -0.9414 74053   1 100006955     0.9224  -0.8526 74053
rs2392072  1    100007331       0.6739  0.7426  74053   1       100007331       0.6681  0.1538  74053   1 100007331     0.6595  0.7959  74053   1 100007331     0.671   0.4886  74053
rs12129588 1    100007519       0.0865  0.8712  74053   1       100007519       0.909   -0.3236 74053   1 100007519     0.124   0.74    74053   1 100007519     0.909   -0.6867 74053
rs10875256 1    100007856       0.074   0.5412  74053   1       100007856       0.9208  -0.2383 74053   1 100007856     0.8888  -0.9605 74053   1 100007856     0.9221  -0.8255 74053
rs7541580  1    100008014       0.7545  0.7303  74053   1       100008014       0.7538  0.3041  74053   1 100008014     0.7371  0.7455  74053   1 100008014     0.755   0.5979  74053
rs883532   1    100008025       0.7543  0.7289  74053   1       100008025       0.7537  0.3126  74053   1 100008025     0.7371  0.7511  74053   1 100008025     0.7549  0.613   74053
rs883533   1    100008439       0.074   0.5386  74053   1       100008439       0.9207  -0.2291 74053   1 100008439     0.8887  -0.9712 74053   1 100008439     0.922   -0.8316 74053
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/TAG2010/Vs1]$zcat /mnt/gluster/data/external_public_supp/TAG2010/TAG2010.AllPhenos.MatchedTologonset.IncAllele.formatted.RAF.txt.pre1.gz | wc
2530634 53143318 342155691
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/TAG2010/Vs1]$zcat /mnt/gluster/data/external_public_supp/TAG2010/TAG2010.AllPhenos.MatchedTologonset.IncAllele.formatted.RAF.txt.pre1.gz | perl -lane 'print $#F;' | sort | uniq -c
2530633 20
      1 24
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/TAG2010/Vs1]$zcat /mnt/gluster/data/external_public_supp/TAG2010/TAG2010.AllPhenos.MatchedTologonset.IncAllele.formatted.RAF.txt.pre1.gz | perl -lane 'if ($#F == 24) { print join("\t", @F); }' 
SNP     CHR     BP      FRQ     A       P       74053   CHR     BP      FRQ     A       P       74053   CHR     BP      FRQ     A       P       74053   CHR     BP      FRQ     A P     74053
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/TAG2010/Vs1]$zcat /mnt/gluster/data/external_public_supp/TAG2010/TAG2010.AllPhenos.MatchedTologonset.IncAllele.formatted.RAF.txt.pre1.gz | perl -lane 'print $#F;' | sort | uniq -c
2530634 20
~~~

zcat /mnt/gluster/data/external_public_supp/TAG2010/TAG2010.AllPhenos.MatchedTologonset.IncAllele.formatted.RAF.txt.pre1.gz | grep -v NA2 | gzip > /mnt/gluster/data/external_public_supp/TAG2010/TAG2010.AllPhenos.MatchedTologonset.IncAllele.formatted.RAF.txt.pre2.gz

zcat /mnt/gluster/data/external_public_supp/TAG2010/TAG2010.AllPhenos.MatchedTologonset.IncAllele.formatted.RAF.txt.pre2.gz | perl -lane 'splice(@F, 21, 3); splice(@F, 16, 3); splice(@F, 11, 3); splice(@F, 6, 3); print join("\t", @F);' | gzip > /mnt/gluster/data/external_public_supp/TAG2010/TAG2010.AllPhenos.MatchedTologonset.IncAllele.formatted.RAF.txt.gz

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/TAG/2010/TAG2010.Table2.MarkerChrBP.txt --file2 /mnt/gluster/data/external_public_supp/TAG2010/TAG2010.AllPhenos.MatchedTologonset.IncAllele.formatted.RAF.txt.gz | gzip > /mnt/gluster/data/external_public_supp/TAG2010/TAG2010.AllPhenos.MatchedTologonset.IncAllele.formatted.RAF.wGWASannot.txt.gz 

~~~
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/TAG2010/Vs1]$zcat /mnt/gluster/data/external_public_supp/TAG2010/TAG2010.AllPhenos.MatchedTologonset.IncAllele.formatted.RAF.wGWASannot.txt.gz | perl -lane 'print $F[$#F];' | sort | uniq -c
2372914 0
     14 1
   3647 2
~~~

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/TAG2010/Vs1/process.MTedits.ForTAG2010.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/TAG2010/Vs1/process.MTedits.ForTAG2010.vs1.SignCrrct.vs1.R

LANG2=${LANG}   
LC_ALL2=${LC_ALL}
LANG=C
LC_ALL=C
export LC_ALL LANG

#join -a 2 -1 1 -2 1 -e NA4 -o 0 1.2 1.3 1.4 1.5 1.6 1.7 1.8 1.9 1.10 1.11 1.12 1.13 2.2 2.3 2.4 2.5 2.6 2.7 2.8 <(zcat /mnt/gluster/data/external_public_supp/TAG2010/TAG2010.AllPhenos.MatchedTologonset.IncAllele.formatted.RAF.wGWASannot.txt.gz | sort -g -k 1,1 ) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/TAG2010/Vs1/TAG2010.dtlesssignif.vs1.SignCrrct.vs1.txt | perl -F, -lane 'print join("\t", @F);' | sort -g -k 1,1) | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/TAG2010/Vs1/TAG2010.dtlesssignif.vs1.SignCrrct.vs1.annot.vs1.RAF.txt.pre1.gz
join -a 2 -1 1 -2 1 -e NA4 -o 0 1.2 1.3 1.4 1.5 1.6 1.7 1.8 1.9 1.10 1.11 1.12 1.13 2.2 2.3 2.4 2.5 2.6 2.7 2.8 <(zcat /mnt/gluster/data/external_public_supp/TAG2010/TAG2010.AllPhenos.MatchedTologonset.IncAllele.formatted.RAF.wGWASannot.txt.gz | sort -g -k 1,1 ) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/TAG2010/Vs1/TAG2010.dtlesslesssignif.vs1.SignCrrct.vs1.txt | perl -F, -lane 'print join("\t", @F);' | sort -g -k 1,1) | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/TAG2010/Vs1/TAG2010.dtlesslesssignif.vs1.SignCrrct.vs1.annot.vs1.RAF.txt.pre1.gz

LANG=${LANG2}
LC_ALL=${LC_ALL2}
export LC_ALL LANG

zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/TAG2010/Vs1/TAG2010.dtlesslesssignif.vs1.SignCrrct.vs1.annot.vs1.RAF.txt.pre1.gz | grep -v NA4 | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/TAG2010/Vs1/TAG2010.dtlesslesssignif.vs1.SignCrrct.vs1.annot.vs1.RAF.txt.gz

#Added the following header to /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/TAG2010/Vs1/TAG2010.dtlesslesssignif.vs1.SignCrrct.vs1.annot.vs1.RAF.txt.gz
#snp chr pos raf p_logonset n_logonset p_former n_former p_evrsmk n_evrsmk p_cpd n_cpd annot Z.logonset Z.former Z.cpd Z.evrsmk mvstat mvp unip

zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/TAG2010/Vs1/TAG2010.dtlesslesssignif.vs1.SignCrrct.vs1.annot.vs1.RAF.txt.gz | perl -lane 'if ($F[3] > .5) { $F[3] = 1 - $F[3]; $F[13] = $F[13] * -1; $F[14] = $F[14] * -1; $F[15] = $F[15] * -1; $F[16] = $F[16] * -1; } print join("\t", @F);' | sed 's/raf/maf/g' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/TAG2010/Vs1/TAG2010.dtlesslesssignif.vs1.SignCrrct.vs1.annot.vs1.MAF.txt.gz

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/MAGIC2010/Vs1/GLC.MTedits.ForMAGIC2010.vs1.SignCrrct.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/TAG2010/Vs1/GLC.MTedits.ForTAG2010.vs1.SignCrrct.vs1.R















##ICBP2011 dbSNP 126, GWAS = (29 SBP/DBP w/ combined follow-up, 7 in MAP/PP w/ combined follow-up), N = provided per SNP
##20150811

#For dbSNP 126
#source: http://www.nature.com.proxy.uchicago.edu/ng/journal/v43/n10/full/ng.922.html
#"...Imputation of the allele dosage of ungenotyped SNPs in HapMap CEU v21a or v22...All study-specific effect estimates and coded alleles were oriented to the forward strand of the HapMap release 22, with the alphabetically higher allele as the coded allele..." 

#From wolfy.uchicago.edu (e.g. home desktop) because need to download from dbGaP webpage using AsperaConnect or something
#Below source: https://dbgap.ncbi.nlm.nih.gov/aa/wga.cgi?page=dr_files&drid=45646&filter=drid (only accessible to me I guess but it's the generic 'instructions' page for an 'authorized access' download from dbGaP I think
~~~
Browser download (using AsperaConnect plugin) into:

 new directory Ã¢ÂÂ you will be asked for download location.

 if process fails - it can be continued later from this page (which can be accessed from "Downloads" tab).
    default location Ã¢ÂÂ change it in Aspera preferences

    default download directory is usually set to "Desktop", which is hardly good place for large files

Command line (ascp) download

Use dbgap-aspera-download.pl perl script that internally calls ascp, but hides some specific usage details; 
OR
Run ascp manually using command line below. 
"%ASPERA_CONNECT_DIR%\bin\ascp" -QTr -l 300M -k 1 -i "%ASPERA_CONNECT_DIR%\etc\asperaweb_id_dsa.openssh" -W A869B89C8E1A84D3EA8F7D3D242B62CA62482ECA6FA4B45B48A99342E22060B8D6EDEB65E21E3478D92444D39011EF190D dbtest@gap-upload.ncbi.nlm.nih.gov:data/instant/mturchin20/45646 . 
This command includes a download ticket which will be valid for a limited time. Once the ticket is expired you will need to visit this page again to get the command with an updated ticket.
%ASPERA_CONNECT_DIR% Ã¢ÂÂ path to Aspera Connect directory. You may have to adjust some ascp-specific command line keys for your environment and network connectivity. Please refer to the AsperaConnect user manual for more information.
~~~

#mkdir /Users/mturchin20/clstrHme/Data/dbGaPTempDownloads/
mkdir /Users/mturchin20/clstrHme2/Data/dbGaPTempDownloads/

#Downloaded from Firefox browser to the above directory

#From source: http://www.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=std
cd /mnt/lustre/home/mturchin20/Software/
wget ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/current/sratoolkit.current-centos_linux64.tar.gz
tar -xvzf sratoolkit.current-centos_linux64.tar.gz

#Trying to decrypt from mac using http://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/2.5.2/decryption.2.5.2-mac64.tar.gz
#So back on wolfy.uchicago.edu
cd /Users/mturchin20/LabStuff/Software 
wget http://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/2.5.2/sratoolkit.2.5.2-mac64.tar.gz
tar -xvzf /Users/mturchin20/LabStuff/Software/sratoolkit.2.5.2-mac64.tar.gz
wget http://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/2.5.2/decryption.2.5.2-mac64.tar.gz
tar -xvzf /Users/mturchin20/LabStuff/Software/decryption.2.5.2-mac64.tar.gz

#Downloading files from dbGaP is a bit of a pain, use these directions -- http://www.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=toolkit_doc&f=dbgap_use
#Need to download the project password (a '.ngc' file), upload it via vdb-config into the SRA toolkit, then use the vdb-decrypt utility /in the directory you specificy during vdb-config/ on the encrypted file of interest

~~~
[  mturchin20@wolfy  ~/clstrHme2/Data/dbGaPTempDownloads/dbGaP-9195]$export PATH=/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/texbin:/usr/X11/bin:/Users/mturchin20/LabStuff/Software/decryption.2.5.2-mac64/bin/
[  mturchin20@wolfy  ~/clstrHme2/Data/dbGaPTempDownloads/dbGaP-9195]$vdb-decrypt 45646/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/phs000585.ICBP.analysis-PI.c1.DS-CVD-IRB.update.tar.gz.ncbi_enc
~~~

#Ran vdb-decrypt on other StudyFiles/ too

mkdir /mnt/gluster/data/external_private_supp/ICBP2011 

mv /mnt/lustre/home/mturchin20/Data/dbGaPTempDownloads/dbGaP-9195/45646/* /mnt/gluster/data/external_private_supp/ICBP2011/.

cd /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles

tar -xvzf /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/phs000585.ICBP.analysis-PI.c1.DS-CVD-IRB.update.tar.gz 

#So technically it looks like the .../dump/pha003588.v1.p1.tgz files contain Chr/Bp position information but I'm unsure if there's been post-processing between those files and the 'ICBP2011_SBP_results_v2.txt' type files. I think I will go with the 'result' files and add in the ChrBP information myself so it's more in line with how I am approaching the other files too...
~~~
[  mturchin20@spudling26  /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data]$zcat ../dump/pha003588.v1.p1.tgz | awk '{ print $3, "\t", $4 }' | tail -n +5 | grep rs | perl -lane 'if ($F[0] =~ m/rs(\d+)/) { $F[0] = $1; } print join("\t", @F);' | sort -rg -k 1,1 | head -n 5
56149945        0.40451364016227
28969975        0.877195032949737
28968498        0.0595783379209321
28968497        0.492645641683983
28968494        0.0692415847883944
[  mturchin20@spudling26  /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data]$cat ICBP2011_SBP_results_v2.txt | awk '{ print $1, "\t", $11 }' | grep rs | perl -lane 'if ($F[0] =~ m/rs(\d+)/) { $F[0] = $1; } print join("\t", @F);' | sort -rg -k 1,1 | head -n 5
28969975        0.877195032949737
28968498        0.0595783379209321
28968497        0.492645641683983
28968494        0.0692415847883944
28968490        0.611171816089357
~~~

gzip /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/*results*

~~~
[  mturchin20@spudling26  /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data]$for i in `ls -lrt | awk '{ print $9 }' | grep results`; do zcat $i | head -n 5; done
SNP_ID  Phenotype       Minor_allele    MAF     Sample_size     Number_contrib_studies  Estimate_effect Standard_error  Coded_target_allele     Alternative_allele     P_value
rs10    MAP     A       0.033   11771.436234    46      0.149907976115096       0.192216538424306       C       A       0.43545495142108
rs1000000       MAP     A       0.373   28834.981022    52      0.132731566188609       0.07107920224497        G       A       0.0618491581791517
rs10000010      MAP     C       0.425   28605.3784795   52      0.00772478995450941     0.0587842002547057      T       C       0.895451540899128
rs10000012      MAP     G       0.192   29024.693561    52      0.0095346253630703      0.0891027368922516      G       C       0.914783348127538
SNP_ID  Phenotype       Minor_allele    MAF     Sample_size     Number_contrib_studies  Estimate_effect Standard_error  Coded_target_allele     Alternative_allele     P_value
rs10    PP      A       0.033   30864.245347    46      -0.0765807416199527     0.191745923428818       C       A       0.689608407493433
rs1000000       PP      A       0.373   73344.2702230173        52      0.116212543195141       0.0711710088206335      G       A       0.102497706366436
rs10000010      PP      C       0.425   73058.533850557 52      0.059499089797371       0.058975444263077       T       C       0.313032646787397
rs10000012      PP      G       0.192   72686.5791272688        52      -0.204123677118336      0.0879102178534104      G       C       0.0202352931299413
SNP_ID  Phenotype       Minor_allele    MAF     Sample_size     Number_contrib_studies  Estimate_effect Standard_error  Coded_target_allele     Alternative_allele     P_value
rs6650104       DBP     G       0.00585623702001679     604.665344      2       -1.55397498982592       1.5508839921798 G       A       0.316346946938566
rs10458597      DBP     T       0.00106272794931912     3310    2       2.26365236006724        4.25070190350682        T       C       0.594354722067332
rs12565286      DBP     C       0.05702869341078        23940.60202     18      0.270170019660784       0.235156231033854       G       C       0.250598926068886
rs11804171      DBP     A       0.057004405108274       23786.253334    18      0.266784434520624       0.235301852343122       T       A       0.256879887582599
SNP_ID  Phenotype       Minor_allele    MAF     Sample_size     Number_contrib_studies  Estimate_effect Standard_error  Coded_target_allele     Alternative_allele     P_value
rs6650104       SBP     G       0.00585623702001679     577.256101      2       -0.282039274730107      2.89491273284234        G       A       0.922388233210622
rs10458597      SBP     T       0.00128949065119278     1560    1       -3.19642        9.61968834610419        T       C       0.739678628776108
rs12565286      SBP     C       0.057307343656075       23657.53802     17      0.350754962559607       0.38334382989527        G       C       0.360197986930745
rs11804171      SBP     A       0.056712027544547       23724.943674    18      0.380649554738311       0.381267839842624       T       A       0.318095930520557
[  mturchin20@spudling26  /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data]$for i in `ls -lrt | awk '{ print $9 }' | grep results`; do paste <(zcat $i | wc) <(zcat $i | perl -lane 'print $#F;' | sort | uniq -c) | xargs echo $i; done
ICBP2011_MAP_results.txt.gz 2651710 29168810 258341514 2651710 10
ICBP2011_PP_results.txt.gz 2652061 29172671 262169603 2652061 10
ICBP2011_DBP_results_v2.txt.gz 2687670 29564370 300818547 2687670 10
ICBP2011_SBP_results_v2.txt.gz 2687670 29564370 297456943 2687670 10
[  mturchin20@spudling26  /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data]$for i in `ls -lrt | awk '{ print $9 }' | grep results`; do echo $i; paste -d @ <(zcat $i | sed 's/e-//g'| perl -lane 'if ($F[3] =~ m/[^\d\.]/) { print join("\t", @F); }' | awk '{ print $4 }' | sort  | uniq -c) <(zcat $i | sed 's/e-//g'| perl -lane 'if ($F[4] =~ m/[^\d\.]/) { print join("\t", @F); }' | awk '{ print $5 }' | sort  | uniq -c) <(zcat $i | sed 's/e-//g'| perl -lane 'if ($F[10] =~ m/[^\d\.]/) { print join("\t", @F); }' | awk '{ print $11 }' | sort  | uniq -c) | sed 's/@/@ /g'| column -s $'@' -t ; done
ICBP2011_MAP_results.txt.gz
1 MAF     29652 NA                  1 P_value
56 NA          1 Sample_size
ICBP2011_PP_results.txt.gz
1 MAF     29678 NA                  1 P_value
56 NA          1 Sample_size
ICBP2011_DBP_results_v2.txt.gz
1 MAF         1 Sample_size      1066 NA
1067 NA                                1 P_value
ICBP2011_SBP_results_v2.txt.gz
1 MAF         1 Sample_size     22887 NA
22888 NA                                1 P_value
[  mturchin20@spudling26  /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data]$for i in `ls -lrt | awk '{ print $9 }' | grep results`; do echo $i; zcat $i | perl -lane 'print join("\t", @F);' | R -q -e "library(data.table); Data1 <- fread('file:///dev/stdin', header=TRUE); quantile(Data1\$MAF, na.rm=TRUE); quantile(Data1\$Sample_size, na.rm=TRUE); quantile(Data1\$P_value, na.rm=TRUE); png(\""$i".QCCheck.vs1.png\", height=2000, width=2000, res=200); par(mfrow=c(2,2)); hist(Data1\$MAF, breaks=100); hist(Data1\$Sample_size, breaks=100); hist(Data1\$P_value, breaks=100); expP <- -log10(c(1:length(Data1\$P_value["\!"is.na(Data1\$P_value)])) / (length(Data1\$P_value["\!"is.na(Data1\$P_value)])+1)); plot(expP, -log10(sort(Data1\$P_value["\!"is.na(Data1\$P_value)])), xlab=\"Expected p-Values\", ylab=\"Observed p-Values\"); abline(0, 1, col=\"RED\"); dev.off();"; done
ICBP2011_MAP_results.txt.gz
> library(data.table); Data1 <- fread('file:///dev/stdin', header=TRUE); quantile(Data1$MAF, na.rm=TRUE); quantile(Data1$Sample_size, na.rm=TRUE); quantile(Data1$P_value, na.rm=TRUE); png("ICBP2011_MAP_results.txt.gz.QCCheck.vs1.png", height=2000, width=2000, res=200); par(mfrow=c(2,2)); hist(Data1$MAF, breaks=100); hist(Data1$Sample_size, breaks=100); hist(Data1$P_value, breaks=100); expP <- -log10(c(1:length(Data1$P_value[!is.na(Data1$P_value)])) / (length(Data1$P_value[!is.na(Data1$P_value)])+1)); plot(expP, -log10(sort(Data1$P_value[!is.na(Data1$P_value)])), xlab="Expected p-Values", ylab="Observed p-Values"); abline(0, 1, col="RED"); dev.off();
Read 2651709 rows and 11 (of 11) columns from 0.241 GB file in 00:00:22
0%   25%   50%   75%  100%
0.000 0.075 0.192 0.339 0.500
0%      25%      50%      75%     100%
0.00 25002.59 27785.67 28820.52 29182.00
0%          25%          50%          75%         100%
2.450516e-70 2.499265e-01 5.000000e-01 7.502246e-01 9.999994e-01

null device
1
>
>
ICBP2011_PP_results.txt.gz
> library(data.table); Data1 <- fread('file:///dev/stdin', header=TRUE); quantile(Data1$MAF, na.rm=TRUE); quantile(Data1$Sample_size, na.rm=TRUE); quantile(Data1$P_value, na.rm=TRUE); png("ICBP2011_PP_results.txt.gz.QCCheck.vs1.png", height=2000, width=2000, res=200); par(mfrow=c(2,2)); hist(Data1$MAF, breaks=100); hist(Data1$Sample_size, breaks=100); hist(Data1$P_value, breaks=100); expP <- -log10(c(1:length(Data1$P_value[!is.na(Data1$P_value)])) / (length(Data1$P_value[!is.na(Data1$P_value)])+1)); plot(expP, -log10(sort(Data1$P_value[!is.na(Data1$P_value)])), xlab="Expected p-Values", ylab="Observed p-Values"); abline(0, 1, col="RED"); dev.off();
Read 2652060 rows and 11 (of 11) columns from 0.244 GB file in 00:00:22
0%   25%   50%   75%  100%
0.000 0.075 0.192 0.339 0.500
0%         25%         50%         75%        100%
72.93118 63266.29581 70402.24453 72738.04595 74078.79094
0%          25%          50%          75%         100%
9.814230e-99 2.488532e-01 4.999993e-01 7.501796e-01 9.999987e-01
null device
1
>
>
ICBP2011_DBP_results_v2.txt.gz
> library(data.table); Data1 <- fread('file:///dev/stdin', header=TRUE); quantile(Data1$MAF, na.rm=TRUE); quantile(Data1$Sample_size, na.rm=TRUE); quantile(Data1$P_value, na.rm=TRUE); png("ICBP2011_DBP_results_v2.txt.gz.QCCheck.vs1.png", height=2000, width=2000, res=200); par(mfrow=c(2,2)); hist(Data1$MAF, breaks=100); hist(Data1$Sample_size, breaks=100); hist(Data1$P_value, breaks=100); expP <- -log10(c(1:length(Data1$P_value[!is.na(Data1$P_value)])) / (length(Data1$P_value[!is.na(Data1$P_value)])+1)); plot(expP, -log10(sort(Data1$P_value[!is.na(Data1$P_value)])), xlab="Expected p-Values", ylab="Observed p-Values"); abline(0, 1, col="RED"); dev.off();
Read 2687669 rows and 11 (of 11) columns from 0.280 GB file in 00:00:27
0%        25%        50%        75%       100%
0.00000000 0.07505255 0.19326217 0.33783551 0.50000000
0%      25%      50%      75%     100%
0.00 57610.78 66009.65 68535.06 69899.00
0%          25%          50%          75%         100%
1.643488e-14 2.491495e-01 5.000000e-01 7.500955e-01 1.000000e+00
null device
1
>
>
ICBP2011_SBP_results_v2.txt.gz
> library(data.table); Data1 <- fread('file:///dev/stdin', header=TRUE); quantile(Data1$MAF, na.rm=TRUE); quantile(Data1$Sample_size, na.rm=TRUE); quantile(Data1$P_value, na.rm=TRUE); png("ICBP2011_SBP_results_v2.txt.gz.QCCheck.vs1.png", height=2000, width=2000, res=200); par(mfrow=c(2,2)); hist(Data1$MAF, breaks=100); hist(Data1$Sample_size, breaks=100); hist(Data1$P_value, breaks=100); expP <- -log10(c(1:length(Data1$P_value[!is.na(Data1$P_value)])) / (length(Data1$P_value[!is.na(Data1$P_value)])+1)); plot(expP, -log10(sort(Data1$P_value[!is.na(Data1$P_value)])), xlab="Expected p-Values", ylab="Observed p-Values"); abline(0, 1, col="RED"); dev.off();
Read 2687669 rows and 11 (of 11) columns from 0.277 GB file in 00:00:27
0%        25%        50%        75%       100%
0.00000000 0.07762993 0.19541978 0.33911372 0.50000000
0%      25%      50%      75%     100%
0.00 57618.27 66018.67 68544.74 69909.00
0%          25%          50%          75%         100%
9.728597e-13 2.490807e-01 5.000000e-01 7.502985e-01 1.000000e+00
null device
1
>
>
~~~

#20150813 CHECK_0: Is it weird that p-values for PP and MAP are much more significant than either SBP or DBP? Maybe this was the (or 'there was a') reason behind only SBP/DBP being publicly being released and PP/MAP being released behind dbGaP?
#20150830 -- Should probably try running things with SBP/DBP and then seperately with all four

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs3.py --file1 <(zcat /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/snp126.txt.gz | perl -lane 'if ($F[1] =~ m/(chr[0-9MXY]+)_.*/) { $F[1] = $1; } print join("\t", @F);')  --file2 /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_SBP_results_v2.txt.gz | gzip > /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_SBP_results_v2.wUCSCGB_snp126.vs3.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs3.py --file1 <(zcat /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/snp126.txt.gz | perl -lane 'if ($F[1] =~ m/(chr[0-9MXY]+)_.*/) { $F[1] = $1; } print join("\t", @F);')  --file2 /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_DBP_results_v2.txt.gz | gzip > /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_DBP_results_v2.wUCSCGB_snp126.vs3.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs3.py --file1 <(zcat /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/snp126.txt.gz | perl -lane 'if ($F[1] =~ m/(chr[0-9MXY]+)_.*/) { $F[1] = $1; } print join("\t", @F);')  --file2 /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_MAP_results.txt.gz | gzip > /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_MAP_results.wUCSCGB_snp126.vs3.txt.gz & 
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs3.py --file1 <(zcat /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/snp126.txt.gz | perl -lane 'if ($F[1] =~ m/(chr[0-9MXY]+)_.*/) { $F[1] = $1; } print join("\t", @F);')  --file2 /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_PP_results.txt.gz | gzip > /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_PP_results.wUCSCGB_snp126.vs3.txt.gz & 

~~~
[  mturchin20@spudling26  /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data]$for i in `ls -lrt | awk '{ print $9 }' | grep wUCSCGB_snp126.vs3`; do echo $i; paste -d @ <(zcat $i | wc) <(zcat $i | perl -lane 'print $F[$#F];' | grep NA | wc) <(zcat $i | perl -lane 'print $F[$#F];' | grep , | wc) <(zcat $i | perl -lane 'print $F[$#F];' | grep , | perl -F, -lane 'print scalar(@F);' | sort | uniq -c) | sed 's/@/@ /g'| column -s $'@' -t ; done
ICBP2011_SBP_results_v2.wUCSCGB_snp126.vs3.txt.gz
2687670 32252040 328408458     41304   41304  123912       442     442   14776       308 2
											38 3
											19 4
											23 5
											17 6
											16 7
											14 8
											7 9
ICBP2011_DBP_results_v2.wUCSCGB_snp126.vs3.txt.gz
2687670 32252040 331770062     41304   41304  123912       442     442   14776       308 2
											38 3
											19 4
											23 5
											17 6
											16 7
											14 8
											7 9
ICBP2011_PP_results.wUCSCGB_snp126.vs3.txt.gz
2652061 31824732 293054152         0       0       0        35      35     737        35 2
ICBP2011_MAP_results.wUCSCGB_snp126.vs3.txt.gz
2651710 31820520 289221970         0       0       0        35      35     737        35 2
~~~

mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011
mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1

cd /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/TAG2010/Vs1/process.MTedits.ForTAG2010.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/process.MTedits.ForICBP2011.vs1.R

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/process.MTedits.ForICBP2011.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/process.MTedits.ForICBP2011.JustSBPDBP.vs1.R

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_SBP_results_v2.txt.gz | gzip > /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_SBP_results_v2.wHapMap21.vs3.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_SBP_results_v2.txt.gz | gzip > /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_SBP_results_v2.wHapMap22.vs3.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_SBP_results_v2.txt.gz | gzip > /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_SBP_results_v2.wHapMap23.vs3.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_SBP_results_v2.txt.gz | gzip > /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_SBP_results_v2.wHapMap24.vs3.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_SBP_results_v2.txt.gz | gzip > /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_SBP_results_v2.wHapMap27.vs3.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_DBP_results_v2.txt.gz | gzip > /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_DBP_results_v2.wHapMap21.vs3.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_MAP_results.txt.gz | gzip > /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_MAP_results.wHapMap21.vs3.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_PP_results.txt.gz | gzip > /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_PP_results.wHapMap21.vs3.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_DBP_results_v2.txt.gz | gzip > /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_DBP_results_v2.wHapMap22.vs3.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_MAP_results.txt.gz | gzip > /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_MAP_results.wHapMap22.vs3.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_PP_results.txt.gz | gzip > /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_PP_results.wHapMap22.vs3.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_DBP_results_v2.txt.gz | gzip > /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_DBP_results_v2.wHapMap23.vs3.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_MAP_results.txt.gz | gzip > /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_MAP_results.wHapMap23.vs3.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_PP_results.txt.gz | gzip > /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_PP_results.wHapMap23.vs3.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_DBP_results_v2.txt.gz | gzip > /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_DBP_results_v2.wHapMap24.vs3.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_MAP_results.txt.gz | gzip > /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_MAP_results.wHapMap24.vs3.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_PP_results.txt.gz | gzip > /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_PP_results.wHapMap24.vs3.txt.gz &


~~~
[  mturchin20@spudling26  /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data]$zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_SBP_results_v2.wHapMap21.vs3.txt.gz | awk '{ print $12 }' | grep NA | wc
0       0       0
[  mturchin20@spudling26  /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data]$zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_SBP_results_v2.wHapMap22.vs3.txt.gz | awk '{ print $12 }' | grep NA | wc
41913   41913  125739
[  mturchin20@spudling26  /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data]$zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_SBP_results_v2.wHapMap23.vs3.txt.gz | awk '{ print $12 }' | grep NA | wc
41831   41831  125493
[  mturchin20@spudling26  /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data]$zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_SBP_results_v2.wHapMap24.vs3.txt.gz | awk '{ print $12 }' | grep NA | wc
43530   43530  130590
[  mturchin20@spudling26  /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data]$zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_SBP_results_v2.wHapMap25.vs3.txt.gz | awk '{ print $12 }' | grep NA | wc
zcat: /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_SBP_results_v2.wHapMap25.vs3.txt.gz: No such file or directory
0       0       0
[  mturchin20@spudling26  /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data]$zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_SBP_results_v2.wHapMap27.vs3.txt.gz | awk '{ print $12 }' | grep NA | wc
43206   43206  129618
[  mturchin20@spudling26  /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data]$zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_DBP_results_v2.wHapMap21.vs3.txt.gz | awk '{ print $12 }' | grep NA | wc
0       0       0
[  mturchin20@spudling26  /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data]$zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_DBP_results_v2.wHapMap22.vs3.txt.gz | awk '{ print $12 }' | grep NA | wc
41913   41913  125739
[  mturchin20@spudling26  /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data]$zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_DBP_results_v2.wHapMap23.vs3.txt.gz | awk '{ print $12 }' | grep NA | wc
41831   41831  125493
[  mturchin20@spudling26  /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data]$zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_DBP_results_v2.wHapMap24.vs3.txt.gz | awk '{ print $12 }' | grep NA | wc
43530   43530  130590
[  mturchin20@spudling26  /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data]$zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_MAP_results.wHapMap21.vs3.txt.gz | awk '{ print $12 }' | grep NA | wc
29652   29652   88956
[  mturchin20@spudling26  /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data]$zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_MAP_results.wHapMap22.vs3.txt.gz | awk '{ print $12 }' | grep NA | wc
2028    2028    6084
[  mturchin20@spudling26  /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data]$zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_MAP_results.wHapMap23.vs3.txt.gz | awk '{ print $12 }' | grep NA | wc
46      46     138
[  mturchin20@spudling26  /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data]$zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_MAP_results.wHapMap24.vs3.txt.gz | awk '{ print $12 }' | grep NA | wc
56      56     168
[  mturchin20@spudling26  /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data]$zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_PP_results.wHapMap21.vs3.txt.gz | awk '{ print $12 }' | grep NA | wc
29678   29678   89034
[  mturchin20@spudling26  /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data]$zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_PP_results.wHapMap22.vs3.txt.gz | awk '{ print $12 }' | grep NA | wc
2028    2028    6084
[  mturchin20@spudling26  /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data]$zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_PP_results.wHapMap23.vs3.txt.gz | awk '{ print $12 }' | grep NA | wc
46      46     138
[  mturchin20@spudling26  /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data]$zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_PP_results.wHapMap24.vs3.txt.gz | awk '{ print $12 }' | grep NA | wc
56      56     168
~~~

#Going forward with both HapMap21 and HapMap22 to see if results change/matter based on which reference used

~~~
[  mturchin20@spudling26  /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data]$for i in `ls -lrt | awk '{ print $9 }' | grep wHapMap21.vs3.txt.gz`; do echo $i; paste -d @ <(zcat $i | awk '{ print $12 }' | grep NA | wc) <(zcat $i | awk '{ print $12 }' | grep , | wc) <(zcat $i | awk '{ print $12 }' | perl -lane 'my @vals1 = split(/,/, $F[0]); print scalar(@vals1);' | sort | uniq -c) <(zcat $i | awk '{ print $4 }' | perl -lane 'if ($F[0] !~ m/0\.0?[1-9]+/) { if ($F[0] =~ m/\d+/) { if ($F[0] > .01) { print $F[0]; } } else { print $F[0]; } }' | sort | uniq -c) | sed 's/@/ @ /g'| column -s $'@' -t; done
ICBP2011_SBP_results_v2.wHapMap21.vs3.txt.gz
0       0       0          0       0       0    2687670 1          1 MAF
								22888 NA
ICBP2011_PP_results.wHapMap21.vs3.txt.gz
29678   29678   89034          0       0       0    2652061 1          1 MAF
									56 NA
ICBP2011_MAP_results.wHapMap21.vs3.txt.gz
29652   29652   88956          0       0       0    2651710 1          1 MAF
									56 NA
ICBP2011_DBP_results_v2.wHapMap21.vs3.txt.gz
0       0       0          0       0       0    2687670 1          1 MAF
								1067 NA
[  mturchin20@spudling26  /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data]$for i in `ls -lrt | awk '{ print $9 }' | grep wHapMap22.vs3.txt.gz`; do echo $i; paste -d @ <(zcat $i | awk '{ print $12 }' | grep NA | wc) <(zcat $i | awk '{ print $12 }' | grep , | wc) <(zcat $i | awk '{ print $12 }' | perl -lane 'my @vals1 = split(/,/, $F[0]); print scalar(@vals1);' | sort | uniq -c) <(zcat $i | awk '{ print $4 }' | perl -lane 'if ($F[0] !~ m/0\.0?[1-9]+/) { if ($F[0] =~ m/\d+/) { if ($F[0] > .01) { print $F[0]; } } else { print $F[0]; } }' | sort | uniq -c) | sed 's/@/ @ /g'| column -s $'@' -t; done
ICBP2011_SBP_results_v2.wHapMap22.vs3.txt.gz
41913   41913  125739          0       0       0    2687670 1          1 MAF
									22888 NA
ICBP2011_PP_results.wHapMap22.vs3.txt.gz
2028    2028    6084          0       0       0    2652061 1          1 MAF
									56 NA
ICBP2011_DBP_results_v2.wHapMap22.vs3.txt.gz
41913   41913  125739          0       0       0    2687670 1          1 MAF
									1067 NA
ICBP2011_MAP_results.wHapMap22.vs3.txt.gz
2028    2028    6084          0       0       0    2651710 1          1 MAF
									56 NA
[  mturchin20@spudling26  /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data]$for i in `ls -lrt | awk '{ print $9 }' | grep wHapMap21.vs3.txt.gz | grep -v _v2`; do echo $i; paste -d @ <(zcat $i | awk '{ print $12 }' | grep NA | wc) <(zcat $i | awk '{ print $12 }' | grep , | wc) <(zcat $i | awk '{ print $12 }' | perl -lane 'my @vals1 = split(/,/, $F[0]); print scalar(@vals1);' | sort | uniq -c) <(zcat $i | awk '{ print $4 }' | perl -lane 'if ($F[0] !~ m/0\.0?[1-9]+/) { print $F[0]; }' | sort | uniq -c) | sed 's/@/ @ /g'| column -s $'@' -t ; done
ICBP2011_PP_results.wHapMap21.vs3.txt.gz
29678   29678   89034          0       0       0    2652061 1     114010 0
									83666 0.008
									21001 0.009
									1 MAF
									56 NA
ICBP2011_MAP_results.wHapMap21.vs3.txt.gz
29652   29652   88956          0       0       0    2651710 1     113994 0
									83572 0.008
									20913 0.009
									1 MAF
									56 NA
[  mturchin20@spudling26  /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data]$for i in `ls -lrt | awk '{ print $9 }' | grep wHapMap22.vs3.txt.gz | grep -v _v2`; do echo $i; paste -d @ <(zcat $i | awk '{ print $12 }' | grep NA | wc) <(zcat $i | awk '{ print $12 }' | grep , | wc) <(zcat $i | awk '{ print $12 }' | perl -lane 'my @vals1 = split(/,/, $F[0]); print scalar(@vals1);' | sort | uniq -c) <(zcat $i | awk '{ print $4 }' | perl -lane 'if ($F[0] !~ m/0\.0?[1-9]+/) { print $F[0]; }' | sort | uniq -c) | sed 's/@/ @ /g'| column -s $'@' -t ; done
ICBP2011_PP_results.wHapMap22.vs3.txt.gz
2028    2028    6084          0       0       0    2652061 1     114010 0
									83666 0.008
									21001 0.009
									1 MAF
									56 NA
ICBP2011_MAP_results.wHapMap22.vs3.txt.gz
2028    2028    6084          0       0       0    2651710 1     113994 0
									83572 0.008
									20913 0.009
									1 MAF
									56 NA
~~~


zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_SBP_results_v2.wHapMap21.vs3.txt.gz | perl -lane 'my @vals1 = split(/_/, $F[11]); push(@F, @vals1); print join("\t", @F);' | awk '{ print $13, "\t", $14, "\t", $1, "\t", $4, "\t", $11, "\t", $5 }' | gzip > /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_SBP_results_v2.wHapMap21.vs3.formatted.txt.gz
zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_DBP_results_v2.wHapMap21.vs3.txt.gz | perl -lane 'my @vals1 = split(/_/, $F[11]); push(@F, @vals1); print join("\t", @F);' | awk '{ print $13, "\t", $14, "\t", $1, "\t", $4, "\t", $11, "\t", $5 }' | gzip > /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_DBP_results_v2.wHapMap21.vs3.formatted.txt.gz
zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_MAP_results.wHapMap21.vs3.txt.gz | perl -lane 'my @vals1 = split(/_/, $F[11]); push(@F, @vals1); print join("\t", @F);' | awk '{ print $13, "\t", $14, "\t", $1, "\t", $4, "\t", $11, "\t", $5 }' | gzip > /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_MAP_results.wHapMap21.vs3.formatted.txt.gz
zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_PP_results.wHapMap21.vs3.txt.gz | perl -lane 'my @vals1 = split(/_/, $F[11]); push(@F, @vals1); print join("\t", @F);' | awk '{ print $13, "\t", $14, "\t", $1, "\t", $4, "\t", $11, "\t", $5 }' | gzip > /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_PP_results.wHapMap21.vs3.formatted.txt.gz
zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_SBP_results_v2.wHapMap22.vs3.txt.gz | perl -lane 'my @vals1 = split(/_/, $F[11]); push(@F, @vals1); print join("\t", @F);' | awk '{ print $13, "\t", $14, "\t", $1, "\t", $4, "\t", $11, "\t", $5 }' | gzip > /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_SBP_results_v2.wHapMap22.vs3.formatted.txt.gz
zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_DBP_results_v2.wHapMap22.vs3.txt.gz | perl -lane 'my @vals1 = split(/_/, $F[11]); push(@F, @vals1); print join("\t", @F);' | awk '{ print $13, "\t", $14, "\t", $1, "\t", $4, "\t", $11, "\t", $5 }' | gzip > /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_DBP_results_v2.wHapMap22.vs3.formatted.txt.gz
zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_MAP_results.wHapMap22.vs3.txt.gz | perl -lane 'my @vals1 = split(/_/, $F[11]); push(@F, @vals1); print join("\t", @F);' | awk '{ print $13, "\t", $14, "\t", $1, "\t", $4, "\t", $11, "\t", $5 }' | gzip > /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_MAP_results.wHapMap22.vs3.formatted.txt.gz
zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_PP_results.wHapMap22.vs3.txt.gz | perl -lane 'my @vals1 = split(/_/, $F[11]); push(@F, @vals1); print join("\t", @F);' | awk '{ print $13, "\t", $14, "\t", $1, "\t", $4, "\t", $11, "\t", $5 }' | gzip > /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_PP_results.wHapMap22.vs3.formatted.txt.gz

#Going with 21 for now?

#go all the way to completion now just all the way through

join -a 1 -a 2 -1 3 -2 3 -e NA2 -o 0 1.1 1.2 1.4 1.5 1.6 2.1 2.2 2.4 2.5 2.6 <(zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_SBP_results_v2.wHapMap21.vs3.formatted.txt.gz | sort -g -k 3,3) <(zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_DBP_results_v2.wHapMap21.vs3.formatted.txt.gz | sort -g -k 3,3 ) | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
join -a 1 -a 2 -1 1 -2 3 -e NA2 -o 0 1.2 2.1 2.2 2.4 2.5 2.6 - <(zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_MAP_results.wHapMap21.vs3.formatted.txt.gz  | sort -g -k 3,3) | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); print join("\t", @F);' | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); if ($#F == 6) { splice(@F, 1, 0, (("NA2") x 9)); } my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
join -a 1 -a 2 -1 1 -2 3 -e NA2 -o 0 1.2 2.1 2.2 2.4 2.5 2.6 - <(zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_PP_results.wHapMap21.vs3.formatted.txt.gz | sort -g -k 3,3) | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); print join("\t", @F);' | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); if ($#F == 6) { splice(@F, 1, 0, (("NA2") x 14)); } my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
perl -lane 'print $F[0], "\t", join("\t", split(/_/, $F[1]));' | sed 's/chr//g' | gzip > /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_AllPhenos_results.wHapMap21.vs3.formatted.txt.pre1.gz

zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_AllPhenos_results.wHapMap21.vs3.formatted.txt.pre1.gz | grep -v NA2 | gzip > /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_AllPhenos_results.wHapMap21.vs3.formatted.txt.pre2.gz

~~~
#[  mturchin20@spudling26  ~]$zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_AllPhenos_results.wHapMap21.vs3.formatted.txt.pre1.gz | wc
#14780175 310383687 2001169629
#[  mturchin20@spudling26  ~]$zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_AllPhenos_results.wHapMap21.vs3.formatted.txt.pre2.gz | wc
#  11112  233352 2632635
[  mturchin20@spudling26  ~]$zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_AllPhenos_results.wHapMap21.vs3.formatted.txt.pre1.gz | wc
12106717 254241069 1759959144
[  mturchin20@spudling26  ~]$zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_AllPhenos_results.wHapMap21.vs3.formatted.txt.pre2.gz | wc
2133187 44796927 502688223
~~~

##Note 20150921 -- Something odd going on? See above (majority of SNPs don't have overlapping/mathing rsIDs between all the files??)
##Note 20150921 -- Had to do with join being conducted on 'reverse sorted' files....don't do this, join appears to be expecting files to be 'normally sorted', not reversed
#
#zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_SBP_results_v2.wHapMap21.vs3.formatted.txt.gz /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_DBP_results_v2.wHapMap21.vs3.formatted.txt.gz /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_MAP_results.wHapMap21.vs3.formatted.txt.gz /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_PP_results.wHapMap21.vs3.formatted.txt.gz | awk '{ print $3 }' | sort | uniq -c | sort -g -k 1,1 | vi -
#(or > kbsdfjdfgoudfgkbjdfg)
#
#~~~
#[  mturchin20@spudling26  ~]$cat kbsdfjdfgoudfgkbjdfg | awk '{ if ($1 == 1) { print $0 } } ' | wc
#312     624    5718
#[  mturchin20@spudling26  ~]$cat kbsdfjdfgoudfgkbjdfg | awk '{ if ($1 == 2) { print $0 } } ' | wc
#66704  133408 1221730
#[  mturchin20@spudling26  ~]$cat kbsdfjdfgoudfgkbjdfg | awk '{ if ($1 == 3) { print $0 } } ' | wc
#221     442    4046
#[  mturchin20@spudling26  ~]$cat kbsdfjdfgoudfgkbjdfg | awk '{ if ($1 == 4) { print $0 } } ' | wc
#2621387 5242774 47767906
#~~~

zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_AllPhenos_results.wHapMap21.vs3.formatted.txt.pre2.gz | awk '{ if (($4 != $9) && ($4 != $14) && ($4 != $19)) { print $0 } } ' | wc
zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_AllPhenos_results.wHapMap21.vs3.formatted.txt.pre2.gz | awk '{ if (($4 != $9) || ($4 != $14) || ($4 != $19)) { print $0 } } ' | wc
zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_AllPhenos_results.wHapMap21.vs3.formatted.txt.pre2.gz | awk '{ if (($3 != $8) && ($3 != $13) && ($3 != $18)) { print $0 } } ' | wc
zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_AllPhenos_results.wHapMap21.vs3.formatted.txt.pre2.gz | awk '{ if (($3 != $8) || ($3 != $13) || ($3 != $18)) { print $0 } } ' | wc
zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_AllPhenos_results.wHapMap21.vs3.formatted.txt.pre2.gz | awk '{ if (($2 != $7) && ($2 != $12) && ($2 != $17)) { print $0 } } ' | wc
zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_AllPhenos_results.wHapMap21.vs3.formatted.txt.pre2.gz | awk '{ if (($2 != $7) || ($2 != $12) || ($2 != $17)) { print $0 } } ' | wc

~~~
[  mturchin20@spudling26  ~]$zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_AllPhenos_results.wHapMap21.vs3.formatted.txt.pre2.gz | awk '{ if (($4 != $9) && ($4 != $14) && ($4 != $19)) { print $0 } } ' | wc
2045356 42952476 485284589
[  mturchin20@spudling26  ~]$zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_AllPhenos_results.wHapMap21.vs3.formatted.txt.pre2.gz | awk '{ if (($4 != $9) || ($4 != $14) || ($4 != $19)) { print $0 } } ' | wc
2122643 44575503 500905623
[  mturchin20@spudling26  ~]$zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_AllPhenos_results.wHapMap21.vs3.formatted.txt.pre2.gz | awk '{ if (($3 != $8) && ($3 != $13) && ($3 != $18)) { print $0 } } ' | wc
0       0       0
[  mturchin20@spudling26  ~]$zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_AllPhenos_results.wHapMap21.vs3.formatted.txt.pre2.gz | awk '{ if (($3 != $8) || ($3 != $13) || ($3 != $18)) { print $0 } } ' | wc
0       0       0
[  mturchin20@spudling26  ~]$zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_AllPhenos_results.wHapMap21.vs3.formatted.txt.pre2.gz | awk '{ if (($2 != $7) && ($2 != $12) && ($2 != $17)) { print $0 } } ' | wc
0       0       0
[  mturchin20@spudling26  ~]$zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_AllPhenos_results.wHapMap21.vs3.formatted.txt.pre2.gz | awk '{ if (($2 != $7) || ($2 != $12) || ($2 != $17)) { print $0 } } ' | wc
0       0       0
~~~

#201507__ NOTE -- Do all gwas hit stuff now?
#Finish things off on first GlobalLipids then jump back to ICBP2011?

zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_AllPhenos_results.wHapMap21.vs3.formatted.txt.pre2.gz | perl -lane 'splice(@F, 21, 3); splice(@F, 16, 3); splice(@F, 11, 3); splice(@F, 6, 3); print join("\t", @F);' | gzip > /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_AllPhenos_results.wHapMap21.vs3.formatted.txt.gz

~~~
[  mturchin20@spudhead  /mnt/gluster/data/external_public_supp/HaemgenRBC2012]$ls -lrt /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/*/Vs1/*CreateFullAnnotFile*
-rw-r--r-- 1 mturchin20 mturchin20 3288 Jun 19  2014 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/DataOrganization.PythonVLookUp.CreateFullAnnotFile.vs1.py
-rw-r--r-- 1 mturchin20 mturchin20 5460 May 26 14:24 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/DataOrganization.GIANT2010.PythonVLookUp.CreateFullAnnotFile.vs1.py
-rw-r--r-- 1 mturchin20 mturchin20 5923 May 26 18:45 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/DataOrganization.GIANT2014_5.PythonVLookUp.CreateFullAnnotFile.vs2.py.20150527.DidntEndUpUsing
-rw-r--r-- 1 mturchin20 mturchin20 5677 May 27 00:56 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/DataOrganization.GIANT2014_5.PythonVLookUp.CreateFullAnnotFile.vs1.py
-rw-r--r-- 1 mturchin20 mturchin20 5764 Jun  2 00:48 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2013/Vs1/DataOrganization.GIANT2013.PythonVLookUp.CreateFullAnnotFile.vs1.py
~~~

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/DataOrganization.GIANT2014_5.PythonVLookUp.CreateFullAnnotFile.vs1.py /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py

##20151005 Note -- Some GWAS SNPs do not make it into final all phenotype file, so took ChrBP information from each of the GWAS hit tables/supplemental tables that were used to get the original set of rsIDs.
####zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_AllPhenos_results.wHapMap21.vs3.formatted.txt.gz | grep -w -f /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.AllPheno.GWASHits.rsIDs.txt | awk '{ print $1, "\t", $2, "\t", $3 }' > /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011_AllPhenos_results.wHapMap21.vs3.formatted.GWASHits.MarkerChrBP.txt

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.AllPheno.GWASHits.MarkerChrBP.txt --file2 /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_AllPhenos_results.wHapMap21.vs3.formatted.txt.gz | gzip > /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_AllPhenos_results.wHapMap21.vs3.formatted.wGWASannot.txt.gz 

##20151005 Note -- There are multiple copies of rs9507310 and rs2155163 in the MAP and PP files, which produces the oddity below. I believe these two SNPs are not included in the dtlesssignif file so just going to pass over this issue for now (versus manually removing them or making code to target these two SNPs in particular)
~~~
[  mturchin20@spudling26  ~]$python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.AllPheno.GWASHits.MarkerChrBP.txt --file2 /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_AllPhenos_results.wHapMap21.vs3.formatted.txt.gz | gzip > /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_AllPhenos_results.wHapMap21.vs3.formatted.wGWASannot.txt.gz
/mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_AllPhenos_results.wHapMap21.vs3.formatted.txt.gz
Error2a -- rsID present more than once (line2: rs2155163,11,126029458,0.286676594454589,0.183522204450074,61788.420529,0.360809080767912,61804.241442,0.881524020104083,26700.9677505,0.437296926934427,66312.9360645)
Error2a -- rsID present more than once (line2: rs2155163,11,126029458,0.286676594454589,0.183522204450074,61788.420529,0.360809080767912,61804.241442,0.881524020104083,26700.9677505,0.554989319526114,65908.5681245)
Error2a -- rsID present more than once (line2: rs2155163,11,126029458,0.286676594454589,0.183522204450074,61788.420529,0.360809080767912,61804.241442,0.881524020104083,26700.9677505,0.554989319526114,66312.9360645)
Error2a -- rsID present more than once (line2: rs2155163,11,126029458,0.286676594454589,0.183522204450074,61788.420529,0.360809080767912,61804.241442,0.996984251080975,26700.9677505,0.437296926934427,65908.5681245)
Error2a -- rsID present more than once (line2: rs2155163,11,126029458,0.286676594454589,0.183522204450074,61788.420529,0.360809080767912,61804.241442,0.996984251080975,26700.9677505,0.437296926934427,66312.9360645)
Error2a -- rsID present more than once (line2: rs2155163,11,126029458,0.286676594454589,0.183522204450074,61788.420529,0.360809080767912,61804.241442,0.996984251080975,26700.9677505,0.554989319526114,65908.5681245)
Error2a -- rsID present more than once (line2: rs2155163,11,126029458,0.286676594454589,0.183522204450074,61788.420529,0.360809080767912,61804.241442,0.996984251080975,26700.9677505,0.554989319526114,66312.9360645)
Error2a -- rsID present more than once (line2: rs9507310,13,23813546,0.339812753749424,0.146924314614668,58667.5711156003,0.211004958883424,58645.1777896003,0.22209504104413,25849.361839,0.319938411013575,63846.1046376902)
Error2a -- rsID present more than once (line2: rs9507310,13,23813546,0.339812753749424,0.146924314614668,58667.5711156003,0.211004958883424,58645.1777896003,0.22209504104413,25849.361839,0.365385851509647,63439.2649526902)
Error2a -- rsID present more than once (line2: rs9507310,13,23813546,0.339812753749424,0.146924314614668,58667.5711156003,0.211004958883424,58645.1777896003,0.22209504104413,25849.361839,0.365385851509647,63846.1046376902)
Error2a -- rsID present more than once (line2: rs9507310,13,23813546,0.339812753749424,0.146924314614668,58667.5711156003,0.211004958883424,58645.1777896003,0.405127885366005,25849.361839,0.319938411013575,63439.2649526902)
Error2a -- rsID present more than once (line2: rs9507310,13,23813546,0.339812753749424,0.146924314614668,58667.5711156003,0.211004958883424,58645.1777896003,0.405127885366005,25849.361839,0.319938411013575,63846.1046376902)
Error2a -- rsID present more than once (line2: rs9507310,13,23813546,0.339812753749424,0.146924314614668,58667.5711156003,0.211004958883424,58645.1777896003,0.405127885366005,25849.361839,0.365385851509647,63439.2649526902)
Error2a -- rsID present more than once (line2: rs9507310,13,23813546,0.339812753749424,0.146924314614668,58667.5711156003,0.211004958883424,58645.1777896003,0.405127885366005,25849.361839,0.365385851509647,63846.1046376902)
[  mturchin20@spudling70  ~]$zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_SBP_results_v2.wHapMap21.vs3.formatted.txt.gz | grep rs9507310
chr13    23813546        rs9507310       0.339812753749424       0.146924314614668       58667.5711156003
[  mturchin20@spudling70  ~]$zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_DBP_results_v2.wHapMap21.vs3.formatted.txt.gz | grep rs9507310
chr13    23813546        rs9507310       0.339806139252346       0.211004958883424       58645.1777896003
[  mturchin20@spudling70  ~]$zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_MAP_results.wHapMap21.vs3.formatted.txt.gz | grep rs9507310
chr13    23813546        rs9507310       0.358   0.22209504104413        25849.361839
chr13    23813546        rs9507310       0.358   0.405127885366005       25849.361839
[  mturchin20@spudling70  ~]$zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_MAP_results.wHapMap21.vs3.formatted.txt.gz | grep rs2155163
chr11    126029458       rs2155163       0.333   0.881524020104083       26700.9677505
chr11    126029458       rs2155163       0.333   0.996984251080975       26700.9677505
[  mturchin20@spudling70  ~]$zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_PP_results.wHapMap21.vs3.formatted.txt.gz | grep rs2155163
chr11    126029458       rs2155163       0.333   0.554989319526114       65908.5681245
chr11    126029458       rs2155163       0.333   0.437296926934427       65908.5681245
chr11    126029458       rs2155163       0.333   0.554989319526114       66312.9360645
chr11    126029458       rs2155163       0.333   0.437296926934427       66312.9360645
[  mturchin20@spudling70  ~]$zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_MAP_results.wHapMap21.vs3.formatted.txt.gz | grep rs9507310
chr13    23813546        rs9507310       0.358   0.22209504104413        25849.361839
chr13    23813546        rs9507310       0.358   0.405127885366005       25849.361839
~~~

~~~
[  mturchin20@spudling26  ~]$zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_AllPhenos_results.wHapMap21.vs3.formatted.wGWASannot.txt.gz | grep NA3
[  mturchin20@spudling26  ~]$
[  mturchin20@spudling26  ~]$zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_AllPhenos_results.wHapMap21.vs3.formatted.wGWASannot.txt.gz | perl -lane 'print $F[$#F];' | sort | uniq -c
2104031 0
     40 1
  29116 2
~~~

LANG2=${LANG}
LC_ALL2=${LC_ALL}
LANG=C
LC_ALL=C
export LC_ALL LANG

join -a 2 -1 1 -2 1 -e NA4 -o 0 1.2 1.3 1.4 1.5 1.6 1.7 1.8 1.9 1.10 1.11 1.12 1.13 2.2 2.3 2.4 2.5 2.6 2.7 2.8 <(zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_AllPhenos_results.wHapMap21.vs3.formatted.wGWASannot.txt.gz | sort -g -k 1,1 ) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/ICBP2011.dtlesssignif.vs1.txt | perl -F, -lane 'print join("\t", @F);' | sort -g -k 1,1) | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/ICBP2011.dtlesssignif.vs1.annot.txt.pre1.gz

##cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.JoinOnFrontrsIDs.vs1.py 

##python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.JoinOnFrontrsIDs.vs1.py --file1 /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_AllPhenos_results.wHapMap21.vs3.formatted.wGWASannot.txt.gz --file2 <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/ICBP2011.dtlesssignif.vs1.txt | perl -F, -lane 'print join("\t", @F);') | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/ICBP2011.dtlesssignif.vs1.annot.txt.pre1.gz

##rs1016669


~~~
[  mturchin20@spudling26  ~]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/ICBP2011.dtlesssignif.vs1.annot.txt.pre1.gz | grep NA4 | wc
    130    2600   22880
~~~    

LANG=${LANG2}
LC_ALL=${LC_ALL2}
export LC_ALL LANG

zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/ICBP2011.dtlesssignif.vs1.annot.txt.pre1.gz | grep -v NA4 | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/ICBP2011.dtlesssignif.vs1.annot.txt.gz

#Added the following header to /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/ICBP2011.dtlesssignif.vs1.annot.txt.gz
#snp chr pos maf p_SBP n_SBP p_DBP n_DBP p_MAP n_MAP p_PP n_PP annot Z.SBP Z.DBP Z.PP Z.MAP mvstat mvp unip

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GLC.MTedits.For2013.vs2.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/GLC.MTedits.ForICBP2011.vs1.R

#Converted GLC.MTedits.For2013.vs2.R to ICBP2011 phenotypes, moving commands to SBP, DBP, PP and MAP  instead of tg, tc, ldl and hdl

#Copy/pasted commands from /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/GLC.MTedits.ForICBP2011.vs1.R into an interactive sessions in $PWD /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1




#20151008
#CSHL Poster work

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/GLC.MTedits.ForICBP2011.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/GLC.MTedits.ForICBP2011.vs1.ForCSHLPoster.R

scp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/GLC.MTedits.ForICBP2011.vs1.ForCSHLPoster.ModelsVslbfavNewOld.vs1.jpeg mturchin20@wolfy.uchicago.edu:/Users/mturchin20/LabStuff/StephensLab/Multivariate/ICBP2011/Vs1/. 





#20160221
#Correcting sign issue -- quick hack to get results

#NOTE -- flipping sign on p-value to make coding downstream easier for this, ie I'm not implying something by creating negative p-values here. This is just for a quick run through.

~~~
[  mturchin20@spudling26  ~/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1]$zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_SBP_results_v2.wUCSCGB_snp126.vs3.txt.gz | head -n 10
SNP_ID  Phenotype       Minor_allele    MAF     Sample_size     Number_contrib_studies  Estimate_effect Standard_error  Coded_target_allele     Alternative_allele      P_value ChrBP
rs6650104  SBP  G       0.00585623702001679     577.256101      2       -0.282039274730107      2.89491273284234        G       A       0.922388233210622       1_554340
rs10458597 SBP  T       0.00128949065119278     1560    1       -3.19642        9.61968834610419        T       C       0.739678628776108       1_554484
rs12565286 SBP  C       0.057307343656075       23657.53802     17      0.350754962559607       0.38334382989527        G       C       0.360197986930745       1_711153
rs11804171 SBP  A       0.056712027544547       23724.943674    18      0.380649554738311       0.381267839842624       T       A       0.318095930520557       1_713682
rs12082473 SBP  A       0.00290322580645197     1560    1       7.03478 6.42825576829291        G       A       0.273800255957098       1_730720
rs12138618 SBP  A       0.05119847359311        10099.368892    13      -0.859471671129419      0.58007141086085        G       A       0.138429378994208       1_740098
rs3094315  SBP  G       0.172988451107781       47235.43444     30      -0.197307418321165      0.161681234743116       G       A       0.222332871025571       1_742429
rs17160906 SBP  A       0.130407314439507       26226.575896    20      0.333229488819729       0.246162748842702       G       A       0.175833436207162       NA
rs12562034 SBP  A       0.083660577761897       16480.526426    13      -0.730863120709895      0.37155669627385        G       A       0.049179767043305       1_758311
[  mturchin20@spudling26  ~/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1]$zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_SBP_results_v2.wUCSCGB_snp126.vs3.txt.gz | perl -lane 'if ($F[6] < 0) { ($F[8], $F[9]) = ($F[9], $F[8]); $F[3] = 1 - $F[3]; $F[6] = $F[6] * -1; } print join("\t", @F);' | gzip > /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_SBP_results_v2.wUCSCGB_snp126.vs3.IncAllele.txt.gz
[  mturchin20@spudling26  ~/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1]$zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_SBP_results_v2.wUCSCGB_snp126.vs3.IncAllele.txt.gz | head -n 10
SNP_ID  Phenotype       Minor_allele    MAF     Sample_size     Number_contrib_studies  Estimate_effect Standard_error  Coded_target_allele     Alternative_allele      P_value ChrBP
rs6650104  SBP  G       0.994143762979983       577.256101      2       0.282039274730107       2.89491273284234        A       G       0.922388233210622       1_554340
rs10458597 SBP  T       0.998710509348807       1560    1       3.19642 9.61968834610419        C       T       0.739678628776108       1_554484
rs12565286 SBP  C       0.057307343656075       23657.53802     17      0.350754962559607       0.38334382989527        G       C       0.360197986930745       1_711153
rs11804171 SBP  A       0.056712027544547       23724.943674    18      0.380649554738311       0.381267839842624       T       A       0.318095930520557       1_713682
rs12082473 SBP  A       0.00290322580645197     1560    1       7.03478 6.42825576829291        G       A       0.273800255957098       1_730720
rs12138618 SBP  A       0.94880152640689        10099.368892    13      0.859471671129419       0.58007141086085        A       G       0.138429378994208       1_740098
rs3094315  SBP  G       0.827011548892219       47235.43444     30      0.197307418321165       0.161681234743116       A       G       0.222332871025571       1_742429
rs17160906 SBP  A       0.130407314439507       26226.575896    20      0.333229488819729       0.246162748842702       G       A       0.175833436207162       NA
rs12562034 SBP  A       0.916339422238103       16480.526426    13      0.730863120709895       0.37155669627385        A       G       0.049179767043305       1_758311
[  mturchin20@spudling26  ~/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1]$zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_SBP_results_v2.wUCSCGB_snp126.vs3.IncAllele.txt.gz | awk '{ if ($7 < 0) { print $0 } } '
[  mturchin20@spudling26  ~/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1]$
~~~

zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_SBP_results_v2.wUCSCGB_snp126.vs3.txt.gz 

zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_SBP_results_v2.wUCSCGB_snp126.vs3.txt.gz | perl -lane 'if ($F[2] ne $F[8]) { $F[3] = 1 - $F[3]; } print join("\t", @F);' | perl -lane 'if ($F[6] < 0) { ($F[8], $F[9]) = ($F[9], $F[8]); $F[3] = 1 - $F[3]; $F[6] = $F[6] * -1; } print join("\t", @F);' | gzip > /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_SBP_results_v2.wUCSCGB_snp126.vs3.IncAllele.txt.gz

#20160226 -- try doing below to help with prev issue -- have not rerun icbp yet with this
#LANG2=${LANG}
#LC_ALL2=${LC_ALL}
#LANG=C
#LC_ALL=C
#export LC_ALL LANG

join <(zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_SBP_results_v2.wUCSCGB_snp126.vs3.IncAllele.txt.gz | awk '{ print $1, "\t", $9 }' | sort) <(zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_DBP_results_v2.wUCSCGB_snp126.vs3.txt.gz | perl -lane 'if ($F[2] ne $F[8]) { $F[3] = 1 - $F[3]; } print join("\t", @F);' | perl -lane 'if ($F[6] < 0) { ($F[8], $F[9]) = ($F[9], $F[8]); $F[3] = 1 - $F[3]; $F[6] = $F[6] * -1; } print join("\t", @F);' | sort) | perl -lane 'if ($F[1] ne $F[9]) { $F[11] = -1 * $F[11]; } splice(@F, 1, 1); print join("\t", @F);' | gzip > /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_DBP_results_v2.wUCSCGB_snp126.vs3.MatchedToSBP.IncAllele.txt.gz
join <(zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_SBP_results_v2.wUCSCGB_snp126.vs3.IncAllele.txt.gz | awk '{ print $1, "\t", $9 }' | sort) <(zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_PP_results.wUCSCGB_snp126.vs3.txt.gz | perl -lane 'if ($F[2] ne $F[8]) { $F[3] = 1 - $F[3]; } print join("\t", @F);' | perl -lane 'if ($F[6] < 0) { ($F[8], $F[9]) = ($F[9], $F[8]); $F[3] = 1 - $F[3]; $F[6] = $F[6] * -1; } print join("\t", @F);' | sort) | perl -lane 'if ($F[1] ne $F[9]) { $F[11] = -1 * $F[11]; } splice(@F, 1, 1); print join("\t", @F);' | gzip > /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_PP_results.wUCSCGB_snp126.vs3.MatchedToSBP.IncAllele.txt.gz
join <(zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_SBP_results_v2.wUCSCGB_snp126.vs3.IncAllele.txt.gz | awk '{ print $1, "\t", $9 }' | sort) <(zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_MAP_results.wUCSCGB_snp126.vs3.txt.gz | perl -lane 'if ($F[2] ne $F[8]) { $F[3] = 1 - $F[3]; } print join("\t", @F);' | perl -lane 'if ($F[6] < 0) { ($F[8], $F[9]) = ($F[9], $F[8]); $F[3] = 1 - $F[3]; $F[6] = $F[6] * -1; } print join("\t", @F);' | sort) | perl -lane 'if ($F[1] ne $F[9]) { $F[11] = -1 * $F[11]; } splice(@F, 1, 1); print join("\t", @F);' | gzip > /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_MAP_results.wUCSCGB_snp126.vs3.MatchedToSBP.IncAllele.txt.gz

zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_SBP_results_v2.wHapMap21.vs3.txt.gz | perl -lane 'if ($F[2] ne $F[8]) { $F[3] = 1 - $F[3]; } print join("\t", @F);' | perl -lane 'if ($F[6] < 0) { ($F[8], $F[9]) = ($F[9], $F[8]); $F[3] = 1 - $F[3]; $F[6] = $F[6] * -1; } print join("\t", @F);' | gzip > /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_SBP_results_v2.wHapMap21.vs3.IncAllele.txt.gz

join <(zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_SBP_results_v2.wHapMap21.vs3.IncAllele.txt.gz | awk '{ print $1, "\t", $9 }' | sort) <(zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_DBP_results_v2.wHapMap21.vs3.txt.gz | perl -lane 'if ($F[2] ne $F[8]) { $F[3] = 1 - $F[3]; } print join("\t", @F);' | perl -lane 'if ($F[6] < 0) { ($F[8], $F[9]) = ($F[9], $F[8]); $F[3] = 1 - $F[3]; $F[6] = $F[6] * -1; } print join("\t", @F);' | sort) | perl -lane 'if ($F[1] ne $F[9]) { $F[11] = -1 * $F[11]; } splice(@F, 1, 1); print join("\t", @F);' | gzip > /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_DBP_results_v2.wHapMap21.vs3.MatchedToSBP.IncAllele.txt.gz
join <(zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_SBP_results_v2.wHapMap21.vs3.IncAllele.txt.gz | awk '{ print $1, "\t", $9 }' | sort) <(zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_PP_results.wHapMap21.vs3.txt.gz | perl -lane 'if ($F[2] ne $F[8]) { $F[3] = 1 - $F[3]; } print join("\t", @F);' | perl -lane 'if ($F[6] < 0) { ($F[8], $F[9]) = ($F[9], $F[8]); $F[3] = 1 - $F[3]; $F[6] = $F[6] * -1; } print join("\t", @F);' | sort) | perl -lane 'if ($F[1] ne $F[9]) { $F[11] = -1 * $F[11]; } splice(@F, 1, 1); print join("\t", @F);' | gzip > /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_PP_results.wHapMap21.vs3.MatchedToSBP.IncAllele.txt.gz
join <(zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_SBP_results_v2.wHapMap21.vs3.IncAllele.txt.gz | awk '{ print $1, "\t", $9 }' | sort) <(zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_MAP_results.wHapMap21.vs3.txt.gz | perl -lane 'if ($F[2] ne $F[8]) { $F[3] = 1 - $F[3]; } print join("\t", @F);' | perl -lane 'if ($F[6] < 0) { ($F[8], $F[9]) = ($F[9], $F[8]); $F[3] = 1 - $F[3]; $F[6] = $F[6] * -1; } print join("\t", @F);' | sort) | perl -lane 'if ($F[1] ne $F[9]) { $F[11] = -1 * $F[11]; } splice(@F, 1, 1); print join("\t", @F);' | gzip > /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_MAP_results.wHapMap21.vs3.MatchedToSBP.IncAllele.txt.gz

zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_SBP_results_v2.wHapMap21.vs3.IncAllele.txt.gz | perl -lane 'my @vals1 = split(/_/, $F[11]); push(@F, @vals1); print join("\t", @F);' | awk '{ print $13, "\t", $14, "\t", $1, "\t", $4, "\t", $11, "\t", $5 }' | gzip > /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_SBP_results_v2.wHapMap21.vs3.IncAllele.formatted.txt.gz
zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_DBP_results_v2.wHapMap21.vs3.MatchedToSBP.IncAllele.txt.gz | perl -lane 'my @vals1 = split(/_/, $F[11]); push(@F, @vals1); print join("\t", @F);' | awk '{ print $13, "\t", $14, "\t", $1, "\t", $4, "\t", $11, "\t", $5 }' | gzip > /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_DBP_results_v2.wHapMap21.vs3.MatchedToSBP.IncAllele.formatted.txt.gz
zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_PP_results.wHapMap21.vs3.MatchedToSBP.IncAllele.txt.gz | perl -lane 'my @vals1 = split(/_/, $F[11]); push(@F, @vals1); print join("\t", @F);' | awk '{ print $13, "\t", $14, "\t", $1, "\t", $4, "\t", $11, "\t", $5 }' | gzip > /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_PP_results.wHapMap21.vs3.MatchedToSBP.IncAllele.formatted.txt.gz
zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_MAP_results.wHapMap21.vs3.MatchedToSBP.IncAllele.txt.gz | perl -lane 'my @vals1 = split(/_/, $F[11]); push(@F, @vals1); print join("\t", @F);' | awk '{ print $13, "\t", $14, "\t", $1, "\t", $4, "\t", $11, "\t", $5 }' | gzip > /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_MAP_results.wHapMap21.vs3.MatchedToSBP.IncAllele.formatted.txt.gz

LANG2=${LANG}
LC_ALL2=${LC_ALL}
LANG=C
LC_ALL=C
export LC_ALL LANG

join -a 1 -a 2 -1 3 -2 3 -e NA2 -o 0 1.1 1.2 1.4 1.5 1.6 2.1 2.2 2.4 2.5 2.6 <(zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_SBP_results_v2.wHapMap21.vs3.IncAllele.formatted.txt.gz | sort -g -k 3,3) <(zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_DBP_results_v2.wHapMap21.vs3.MatchedToSBP.IncAllele.formatted.txt.gz | sort -g -k 3,3 ) | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
join -a 1 -a 2 -1 1 -2 3 -e NA2 -o 0 1.2 2.1 2.2 2.4 2.5 2.6 - <(zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_MAP_results.wHapMap21.vs3.MatchedToSBP.IncAllele.formatted.txt.gz  | sort -g -k 3,3) | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); print join("\t", @F);' | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); if ($#F == 6) { splice(@F, 1, 0, (("NA2") x 9)); } my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
join -a 1 -a 2 -1 1 -2 3 -e NA2 -o 0 1.2 2.1 2.2 2.4 2.5 2.6 - <(zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_PP_results.wHapMap21.vs3.MatchedToSBP.IncAllele.formatted.txt.gz | sort -g -k 3,3) | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); print join("\t", @F);' | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); if ($#F == 6) { splice(@F, 1, 0, (("NA2") x 14)); } my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
perl -lane 'print $F[0], "\t", join("\t", split(/_/, $F[1]));' | sed 's/chr//g' | gzip > /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_AllPhenos_results.wHapMap21.vs3.MatchedToSBP.IncAllele.formatted.RAF.txt.pre1.gz

LANG=${LANG2}
LC_ALL=${LC_ALL2}
export LC_ALL LANG

zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_AllPhenos_results.wHapMap21.vs3.MatchedToSBP.IncAllele.formatted.RAF.txt.pre1.gz | grep -v NA2 | gzip > /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_AllPhenos_results.wHapMap21.vs3.MatchedToSBP.IncAllele.formatted.RAF.txt.pre2.gz

zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_AllPhenos_results.wHapMap21.vs3.MatchedToSBP.IncAllele.formatted.RAF.txt.pre2.gz | perl -lane 'splice(@F, 21, 3); splice(@F, 16, 3); splice(@F, 11, 3); splice(@F, 6, 3); print join("\t", @F);' | gzip > /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_AllPhenos_results.wHapMap21.vs3.MatchedToSBP.IncAllele.formatted.RAF.txt.gz 

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.AllPheno.GWASHits.MarkerChrBP.txt --file2 /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_AllPhenos_results.wHapMap21.vs3.MatchedToSBP.IncAllele.formatted.RAF.txt.gz | gzip > /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_AllPhenos_results.wHapMap21.vs3.MatchedToSBP.IncAllele.formatted.RAF.wGWASannot.txt.gz

~~~
[  mturchin20@spudling12  ~]$python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.AllPheno.GWASHits.MarkerChrBP.txt --file2 /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_AllPhenos_results.wHapMap21.vs3.MatchedToSBP.IncAllele.formatted.RAF.txt.gz | gzip > /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_AllPhenos_results.wHapMap21.vs3.MatchedToSBP.IncAllele.formatted.RAF.wGWASannot.txt.gz
/mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_AllPhenos_results.wHapMap21.vs3.MatchedToSBP.IncAllele.formatted.txt.gz
Error2a -- rsID present more than once (line2: rs2155163,11,126029458,0.286676594454589,0.183522204450074,61788.420529,0.360809080767912,61804.241442,-0.881524020104083,26700.9677505,0.437296926934427,66312.9360645)
Error2a -- rsID present more than once (line2: rs2155163,11,126029458,0.286676594454589,0.183522204450074,61788.420529,0.360809080767912,61804.241442,-0.881524020104083,26700.9677505,0.554989319526114,65908.5681245)
Error2a -- rsID present more than once (line2: rs2155163,11,126029458,0.286676594454589,0.183522204450074,61788.420529,0.360809080767912,61804.241442,-0.881524020104083,26700.9677505,0.554989319526114,66312.9360645)
Error2a -- rsID present more than once (line2: rs2155163,11,126029458,0.286676594454589,0.183522204450074,61788.420529,0.360809080767912,61804.241442,-0.996984251080975,26700.9677505,0.437296926934427,65908.5681245)
Error2a -- rsID present more than once (line2: rs2155163,11,126029458,0.286676594454589,0.183522204450074,61788.420529,0.360809080767912,61804.241442,-0.996984251080975,26700.9677505,0.437296926934427,66312.9360645)
Error2a -- rsID present more than once (line2: rs2155163,11,126029458,0.286676594454589,0.183522204450074,61788.420529,0.360809080767912,61804.241442,-0.996984251080975,26700.9677505,0.554989319526114,65908.5681245)
Error2a -- rsID present more than once (line2: rs2155163,11,126029458,0.286676594454589,0.183522204450074,61788.420529,0.360809080767912,61804.241442,-0.996984251080975,26700.9677505,0.554989319526114,66312.9360645)
Error2a -- rsID present more than once (line2: rs9507310,13,23813546,0.339812753749424,0.146924314614668,58667.5711156003,0.211004958883424,58645.1777896003,0.22209504104413,25849.361839,0.319938411013575,63846.1046376902)
Error2a -- rsID present more than once (line2: rs9507310,13,23813546,0.339812753749424,0.146924314614668,58667.5711156003,0.211004958883424,58645.1777896003,0.22209504104413,25849.361839,0.365385851509647,63439.2649526902)
Error2a -- rsID present more than once (line2: rs9507310,13,23813546,0.339812753749424,0.146924314614668,58667.5711156003,0.211004958883424,58645.1777896003,0.22209504104413,25849.361839,0.365385851509647,63846.1046376902)
Error2a -- rsID present more than once (line2: rs9507310,13,23813546,0.339812753749424,0.146924314614668,58667.5711156003,0.211004958883424,58645.1777896003,0.405127885366005,25849.361839,0.319938411013575,63439.2649526902)
Error2a -- rsID present more than once (line2: rs9507310,13,23813546,0.339812753749424,0.146924314614668,58667.5711156003,0.211004958883424,58645.1777896003,0.405127885366005,25849.361839,0.319938411013575,63846.1046376902)
Error2a -- rsID present more than once (line2: rs9507310,13,23813546,0.339812753749424,0.146924314614668,58667.5711156003,0.211004958883424,58645.1777896003,0.405127885366005,25849.361839,0.365385851509647,63439.2649526902)
Error2a -- rsID present more than once (line2: rs9507310,13,23813546,0.339812753749424,0.146924314614668,58667.5711156003,0.211004958883424,58645.1777896003,0.405127885366005,25849.361839,0.365385851509647,63846.1046376902)
[  mturchin20@spudling12  ~]$
~~~

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/process.MTedits.ForICBP2011.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/process.MTedits.ForICBP2011.vs1.SignCrrct.vs1.R

#20160222 CHECK_0 -- Prob: Correct header situation for the DBP, PP and MAP files needed for process file above

LANG2=${LANG}
LC_ALL2=${LC_ALL}
LANG=C
LC_ALL=C
export LC_ALL LANG

join -a 2 -1 1 -2 1 -e NA4 -o 0 1.2 1.3 1.4 1.5 1.6 1.7 1.8 1.9 1.10 1.11 1.12 1.13 2.2 2.3 2.4 2.5 2.6 2.7 2.8 <(zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_AllPhenos_results.wHapMap21.vs3.MatchedToSBP.IncAllele.formatted.RAF.wGWASannot.txt.gz | sort -g -k 1,1 ) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/ICBP2011.dtlesssignif.vs1.SignCrrct.vs1.txt | perl -F, -lane 'print join("\t", @F);' | sort -g -k 1,1) | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/ICBP2011.dtlesssignif.vs1.annot.vs1.SignCrrct.vs1.RAF.txt.pre1.gz

LC_ALL=${LC_ALL2}
export LC_ALL LANG

zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/ICBP2011.dtlesssignif.vs1.annot.vs1.SignCrrct.vs1.RAF.txt.pre1.gz | grep -v NA4 | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/ICBP2011.dtlesssignif.vs1.annot.vs1.SignCrrct.vs1.RAF.txt.gz

#Added the following header to /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/ICBP2011.dtlesssignif.vs1.annot.vs1.SignCrrct.vs1.RAF.txt.gz
#snp chr pos raf p_SBP n_SBP p_DBP n_DBP p_MAP n_MAP p_PP n_PP annot Z.SBP Z.DBP Z.PP Z.MAP mvstat mvp unip

zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/ICBP2011.dtlesssignif.vs1.annot.vs1.SignCrrct.vs1.RAF.txt.gz | perl -lane 'if ($F[3] > .5) { $F[3] = 1 - $F[3]; $F[13] = $F[13] * -1; $F[14] = $F[14] * -1; $F[15] = $F[15] * -1; $F[16] = $F[16] * -1; } print join("\t", @F);' | sed 's/raf/maf/g' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/ICBP2011.dtlesssignif.vs1.annot.vs1.SignCrrct.vs1.MAF.txt.gz

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/GLC.MTedits.ForICBP2011.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/GLC.MTedits.ForICBP2011.vs1.SignCrrct.vs1.R






#20161227
#Vs2 -- Running things through version of bmass() code

mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs2
cd /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs2
















##iGENBP2015
##20160421

mkdir /mnt/gluster/data/external_public_supp/iGENBP2015

cd /mnt/gluster/data/external_public_supp/iGENBP2015

#Downloaded from EGA using Matthew's access, https://www.ebi.ac.uk/ega/datasets/EGAD00010000819 and https://www.ebi.ac.uk/ega/about/your_EGA_account/download_streaming_client

#On MacBook Air
#cd 
#java -jar
#login mstephens@uchicago.edu
#password: ?????
#request dataset EGAD00010000819 abc request_EGAD00010000819
#download request_EGAD00010000819
#decrypt _ega-box-493_DBP_summary_stats.txt.cip abc
#decrypt _ega-box-493_SBP_summary_stats.txt.cip abc
#decrypt _ega-box-493_HT_summary_stats.txt.cip abc
#decrypt _ega-box-493_MAP_summary_stats.txt.cip abc
#decrypt _ega-box-493_PP_summary_stats.txt.cip abc

gzip /mnt/gluster/data/external_public_supp/iGENBP2015/*

paste <(zcat /mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_DBP_summary_stats.txt.gz | awk '{ print $4 }' | sort -g) <(zcat /mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_SBP_summary_stats.txt.gz | awk '{ print $4 }' | sort -g) <(zcat /mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_PP_summary_stats.txt.gz | awk '{ print $4 }' | sort -g) <(zcat /mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_MAP_summary_stats.txt.gz | awk '{ print $4 }' | sort -g) | vi - 

for i in `ls -lrt /mnt/gluster/data/external_public_supp/iGENBP2015/. | grep ega-box | awk '{ print $9 }'`; do
	echo /mnt/gluster/data/external_public_supp/iGENBP2015/$i;
	zcat /mnt/gluster/data/external_public_supp/iGENBP2015/$i | head -n 10	
done

for i in `ls -lrt /mnt/gluster/data/external_public_supp/iGENBP2015/. | grep ega-box | awk '{ print $9 }'`; do
	echo /mnt/gluster/data/external_public_supp/iGENBP2015/$i;
	zcat /mnt/gluster/data/external_public_supp/iGENBP2015/$i | awk '{ print $4 }' | grep -v Beta | R -q -e "Data1 <- read.table(file('stdin'), header=F); quantile(Data1[,1]); var(Data1[,1]);";
done

~~~
[  mturchin20@spudling12  ~/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/PathwayWork]$for i in `ls -lrt /mnt/gluster/data/external_public_supp/iGENBP2015/. | grep ega-box | awk '{ print $9 }'`; do
>         echo /mnt/gluster/data/external_public_supp/iGENBP2015/$i;
>         zcat /mnt/gluster/data/external_public_supp/iGENBP2015/$i | head -n 10 
> done
/mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_DBP_summary_stats.txt.gz
SNP     CHR     POS     Beta    N
rs11099098      4       81388936        0.5092  83890.1
rs13149993      4       81377569        0.464   92729.4
rs11105378      12      88614872        -0.4897 93131.4
rs12579302      12      88574634        0.478   93302.1
rs11105354      12      88550654        0.4782  93193.5
rs7136259       12      88605319        -0.4433 99947.3
rs12230074      12      88614998        0.4806  90515.1
rs17249754      12      88584717        -0.5308 64491.7
rs11105368      12      88598572        -0.5284 64482.6
/mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_SBP_summary_stats.txt.gz
SNP     CHR     POS     Beta    N
rs11105354      12      88550654        1.0347  93206.5
rs11105378      12      88614872        -1.0509 93144.4
rs12579302      12      88574634        1.0286  93315.1
rs12230074      12      88614998        1.0472  90528.2
rs7136259       12      88605319        -0.898  99960.3
rs2681485       12      88549753        0.8682  97835.7
rs11105368      12      88598572        -1.0764 64495.6
rs11105364      12      88593407        1.0741  64505.6
rs17249754      12      88584717        -1.0728 64504.7
/mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_HT_summary_stats.txt.gz
SNP     CHR     POS     Beta    N
rs1558902       16      52361075        0.1215  44430.5
rs11099098      4       81388936        0.1256  34088.5
rs7193144       16      52368187        -0.1163 44427.7
rs8050136       16      52373776        0.1154  44431
rs1458038       4       81383747        0.1228  34085
rs9939609       16      52378028        0.1161  44428.5
rs13149993      4       81377569        0.1129  37523.3
rs9935401       16      52374339        0.1145  44435.5
rs17817449      16      52370868        -0.1156 42441.4
/mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_MAP_summary_stats.txt.gz
SNP     CHR     POS     Beta    N
rs11105354      12      88550654        0.6666  92197.5
rs11105378      12      88614872        -0.6781 92135.4
rs12579302      12      88574634        0.6638  92306.1
rs12230074      12      88614998        0.6713  89519.1
rs7136259       12      88605319        -0.5878 98951.3
rs11105368      12      88598572        -0.7066 64482.6
rs11105364      12      88593407        0.7055  64492.6
rs17249754      12      88584717        -0.7069 64491.7
rs2681485       12      88549753        0.5647  96826.8
/mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_PP_summary_stats.txt.gz
SNP     CHR     POS     Beta    N
rs11105354      12      88550654        0.5208  93193.5
rs11105378      12      88614872        -0.527  93131.4
rs12579302      12      88574634        0.5153  93302.1
rs12230074      12      88614998        0.5298  90515.1
rs740406        19      2183221 -0.7295 91115
rs1558118       19      2183083 0.6605  91115.1
rs17477177      7       106199094       -0.5308 99344.1
rs12705390      7       106198013       0.5263  99271
rs17398575      7       106196688       0.531   99040.5
[  mturchin20@spudling12  ~/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/PathwayWork]$for i in `ls -lrt /mnt/gluster/data/external_public_supp/iGENBP2015/. | grep ega-box | awk '{ print $9 }'`; do
>         echo /mnt/gluster/data/external_public_supp/iGENBP2015/$i;
>         zcat /mnt/gluster/data/external_public_supp/iGENBP2015/$i | awk '{ print $4 }' | grep -v Beta | R -q -e "Data1 <- read.table(file('stdin'), header=F); quantile(Data1[,1]); var(Data1[,1]);";
> done
/mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_DBP_summary_stats.txt.gz
> Data1 <- read.table(file('stdin'), header=F); quantile(Data1[,1]); var(Data1[,1]);
     0%     25%     50%     75%    100% 
-0.9704 -0.0507  0.0001  0.0508  0.9292 
[1] 0.007225688
> 
> 
/mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_SBP_summary_stats.txt.gz
> Data1 <- read.table(file('stdin'), header=F); quantile(Data1[,1]); var(Data1[,1]);
     0%     25%     50%     75%    100% 
-1.6847 -0.0876  0.0002  0.0879  1.8349 
[1] 0.02180375
> 
> 
/mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_HT_summary_stats.txt.gz
> Data1 <- read.table(file('stdin'), header=F); quantile(Data1[,1]); var(Data1[,1]);
     0%     25%     50%     75%    100% 
-0.4023 -0.0161  0.0000  0.0161  0.3398 
[1] 0.000767995
> 
> 
/mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_MAP_summary_stats.txt.gz
> Data1 <- read.table(file('stdin'), header=F); quantile(Data1[,1]); var(Data1[,1]);
     0%     25%     50%     75%    100% 
-1.0147 -0.0592  0.0001  0.0594  1.1646 
[1] 0.009853744
> 
> 
/mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_PP_summary_stats.txt.gz
> Data1 <- read.table(file('stdin'), header=F); quantile(Data1[,1]); var(Data1[,1]);
     0%     25%     50%     75%    100% 
-1.0536 -0.0584  0.0001  0.0587  1.0419 
[1] 0.009765907
> 
> 
~~~

cat /mnt/lustre/home/mturchin20/Data/iGENBP2015/iGENBP2015.SupplTable4.txt | perl -lane 'print join("\t", @F[0..11]);' | grep SBP | perl -lane 'my $varY = $F[10]**2 * 2 * $F[7] * $F[8] * (1 - $F[8]); print $varY;' | R -q -e "Data1 <- read.table(file('stdin'), header=F); print(c(length(Data1[,]), mean(Data1[,1]), sd(Data1[,1])));"
cat /mnt/lustre/home/mturchin20/Data/iGENBP2015/iGENBP2015.SupplTable4.txt | perl -lane 'print join("\t", @F[0..11]);' | grep DBP | perl -lane 'my $varY = $F[10]**2 * 2 * $F[7] * $F[8] * (1 - $F[8]); print $varY;' | R -q -e "Data1 <- read.table(file('stdin'), header=F); print(c(length(Data1[,]), mean(Data1[,1]), sd(Data1[,1])));"
cat /mnt/lustre/home/mturchin20/Data/iGENBP2015/iGENBP2015.SupplTable4.txt | perl -lane 'print join("\t", @F[0..11]);' | grep PP | perl -lane 'my $varY = $F[10]**2 * 2 * $F[7] * $F[8] * (1 - $F[8]); print $varY;' | R -q -e "Data1 <- read.table(file('stdin'), header=F); print(c(length(Data1[,]), mean(Data1[,1]), sd(Data1[,1])));"
cat /mnt/lustre/home/mturchin20/Data/iGENBP2015/iGENBP2015.SupplTable4.txt | perl -lane 'print join("\t", @F[0..11]);' | grep MAP | perl -lane 'my $varY = $F[10]**2 * 2 * $F[7] * $F[8] * (1 - $F[8]); print $varY;' | R -q -e "Data1 <- read.table(file('stdin'), header=F); print(c(length(Data1[,]), mean(Data1[,1]), sd(Data1[,1])));"

~~~
[  mturchin20@bigmem01  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Data/iGENBP2015/iGENBP2015.SupplTable4.txt | perl -lane 'print join("\t", @F[0..11]);' | grep SBP | perl -lane 'my $varY = $F[10]**2 * 2 * $F[7] * $F[8] * (1 - $F[8]); print $varY;' | R -q -e "Data1 <- read.table(file('stdin'), header=F); print(c(length(Data1[,]), mean(Data1[,1]), sd(Data1[,1])));"
in20/Data/iGENBP2015/iGENBP2015.SupplTable4.txt | perl -lane 'print join("\t", @F[0..11]);' | grep MAP | perl -lane 'my $varY = $F[10]**2 * 2 * $F[7] * $F[8] * (1 - $F[8]); print $varY;' | R -q -e "Data1 <- read.table(file('stdin'), header=F); print(c(length(Data1[,]), mean(Data1[,1]), sd(Data1[,1])));"
> Data1 <- read.table(file('stdin'), header=F); print(c(length(Data1[,]), mean(Data1[,1]), sd(Data1[,1])));
[1]  11.00000 434.46853  43.88172
> 
> 
[  mturchin20@bigmem01  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Data/iGENBP2015/iGENBP2015.SupplTable4.txt | perl -lane 'print join("\t", @F[0..11]);' | grep DBP | perl -lane 'my $varY = $F[10]**2 * 2 * $F[7] * $F[8] * (1 - $F[8]); print $varY;' | R -q -e "Data1 <- read.table(file('stdin'), header=F); print(c(length(Data1[,]), mean(Data1[,1]), sd(Data1[,1])));"
> Data1 <- read.table(file('stdin'), header=F); print(c(length(Data1[,]), mean(Data1[,1]), sd(Data1[,1])));
[1]   5.00000 162.42747  16.25914
> 
> 
[  mturchin20@bigmem01  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Data/iGENBP2015/iGENBP2015.SupplTable4.txt | perl -lane 'print join("\t", @F[0..11]);' | grep PP | perl -lane 'my $varY = $F[10]**2 * 2 * $F[7] * $F[8] * (1 - $F[8]); print $varY;' | R -q -e "Data1 <- read.table(file('stdin'), header=F); print(c(length(Data1[,]), mean(Data1[,1]), sd(Data1[,1])));"
> Data1 <- read.table(file('stdin'), header=F); print(c(length(Data1[,]), mean(Data1[,1]), sd(Data1[,1])));
[1]  10.00000 219.57919  41.20232
> 
> 
[  mturchin20@bigmem01  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Data/iGENBP2015/iGENBP2015.SupplTable4.txt | perl -lane 'print join("\t", @F[0..11]);' | grep MAP | perl -lane 'my $varY = $F[10]**2 * 2 * $F[7] * $F[8] * (1 - $F[8]); print $varY;' | R -q -e "Data1 <- read.table(file('stdin'), header=F); print(c(length(Data1[,]), mean(Data1[,1]), sd(Data1[,1])));"
> Data1 <- read.table(file('stdin'), header=F); print(c(length(Data1[,]), mean(Data1[,1]), sd(Data1[,1])));
[1]  14.00000 209.79683  50.63987
> 
> 
~~~

#var(Y) values for each phenotype
#SBP = 434.46853
#DBP = 162.42747
#PP = 219.57919
#MAP = 209.79683

##python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs3.py --file1 <(zcat /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/snp129.txt.gz | perl -lane 'if ($F[1] =~ m/(chr[0-9MXY]+)_.*/) { $F[1] = $1; } print join("\t", @F);')  --file2 /mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_SBP_summary_stats.txt.gz | gzip > /mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_SBP_summary_stats.wUCSCGB_snp129.vs3.txt.gz & 
##python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs3.py --file1 <(zcat /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/snp130.txt.gz | perl -lane 'if ($F[1] =~ m/(chr[0-9MXY]+)_.*/) { $F[1] = $1; } print join("\t", @F);')  --file2 /mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_SBP_summary_stats.txt.gz | gzip > /mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_SBP_summary_stats.wUCSCGB_snp130.vs3.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_SBP_summary_stats.txt.gz | gzip > /mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_SBP_summary_stats.wHapMap21.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_SBP_summary_stats.txt.gz | gzip > /mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_SBP_summary_stats.wHapMap22.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_SBP_summary_stats.txt.gz | gzip > /mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_SBP_summary_stats.wHapMap23.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_SBP_summary_stats.txt.gz | gzip > /mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_SBP_summary_stats.wHapMap24.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_SBP_summary_stats.txt.gz | gzip > /mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_SBP_summary_stats.wHapMap27.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_SBP_summary_stats.txt.gz | gzip > /mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_SBP_summary_stats.wHapMap28.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_DBP_summary_stats.txt.gz | gzip > /mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_DBP_summary_stats.wHapMap27.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_PP_summary_stats.txt.gz | gzip > /mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_PP_summary_stats.wHapMap27.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_MAP_summary_stats.txt.gz | gzip > /mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_MAP_summary_stats.wHapMap27.txt.gz &

#Going with HapMap27 -- see below
~~~
[  mturchin20@bigmem01  ~]$zcat /mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_SBP_summary_stats.wHapMap21.txt.gz | awk '{ print $6 }' | grep NA | wc
  25610   25610   76830
[  mturchin20@bigmem01  ~]$zcat /mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_SBP_summary_stats.wHapMap22.txt.gz | awk '{ print $6 }' | grep NA | wc
   2826    2826    8478
[  mturchin20@bigmem01  ~]$zcat /mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_SBP_summary_stats.wHapMap23.txt.gz | awk '{ print $6 }' | grep NA | wc
    375     375    1125
[  mturchin20@bigmem01  ~]$zcat /mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_SBP_summary_stats.wHapMap24.txt.gz | awk '{ print $6 }' | grep NA | wc
    364     364    1092
[  mturchin20@bigmem01  ~]$zcat /mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_SBP_summary_stats.wHapMap27.txt.gz | awk '{ print $6 }' | grep NA | wc
     40      40     120
[  mturchin20@bigmem01  ~]$zcat /mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_SBP_summary_stats.wHapMap28.txt.gz | awk '{ print $6 }' | grep NA | wc
    121     121     363
[  mturchin20@bigmem01  ~]$zcat /mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_SBP_summary_stats.wHapMap21.txt.gz | perl -lane 'my @vals1 = split(/_/, $F[5]); if ($F[2] eq $vals1[1]) { print "match"; } else { print "nope"; }' | sort | uniq -c
1308179 match
 819704 nope
[  mturchin20@bigmem01  ~]$zcat /mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_SBP_summary_stats.wHapMap22.txt.gz | perl -lane 'my @vals1 = split(/_/, $F[5]); if ($F[2] eq $vals1[1]) { print "match"; } else { print "nope"; }' | sort | uniq -c
2125048 match
   2835 nope
[  mturchin20@bigmem01  ~]$zcat /mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_SBP_summary_stats.wHapMap23.txt.gz | perl -lane 'my @vals1 = split(/_/, $F[5]); if ($F[2] eq $vals1[1]) { print "match"; } else { print "nope"; }' | sort | uniq -c
2126749 match
   1134 nope
[  mturchin20@bigmem01  ~]$zcat /mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_SBP_summary_stats.wHapMap24.txt.gz | perl -lane 'my @vals1 = split(/_/, $F[5]); if ($F[2] eq $vals1[1]) { print "match"; } else { print "nope"; }' | sort | uniq -c
2127513 match
    370 nope
[  mturchin20@bigmem01  ~]$zcat /mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_SBP_summary_stats.wHapMap27.txt.gz | perl -lane 'my @vals1 = split(/_/, $F[5]); if ($F[2] eq $vals1[1]) { print "match"; } else { print "nope"; }' | sort | uniq -c
2127840 match
     43 nope
[  mturchin20@bigmem01  ~]$zcat /mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_SBP_summary_stats.wHapMap28.txt.gz | perl -lane 'my @vals1 = split(/_/, $F[5]); if ($F[2] eq $vals1[1]) { print "match"; } else { print "nope"; }' | sort | uniq -c
2127756 match
    127 nope
~~~

zcat /mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_SBP_summary_stats.wHapMap27.txt.gz | perl -lane 'my @vals1 = split(/_/, $F[5]); if (($F[5] ne "NA") && ($vals1[3] > 0) && ($vals1[3] < 1)) { push(@F, sqrt(434.46853 / (2 * $F[4] * $vals1[3] * (1 - $vals1[3]))));} else {push (@F, "NA"); } print join("\t", @F);' | gzip > /mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_SBP_summary_stats.wHapMap27.wSE.txt.gz
zcat /mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_DBP_summary_stats.wHapMap27.txt.gz | perl -lane 'my @vals1 = split(/_/, $F[5]); if (($F[5] ne "NA") && ($vals1[3] > 0) && ($vals1[3] < 1)) { push(@F, sqrt(162.42747 / (2 * $F[4] * $vals1[3] * (1 - $vals1[3]))));} else {push (@F, "NA"); } print join("\t", @F);' | gzip > /mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_DBP_summary_stats.wHapMap27.wSE.txt.gz
zcat /mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_PP_summary_stats.wHapMap27.txt.gz | perl -lane 'my @vals1 = split(/_/, $F[5]); if (($F[5] ne "NA") && ($vals1[3] > 0) && ($vals1[3] < 1)) { push(@F, sqrt(219.57919 / (2 * $F[4] * $vals1[3] * (1 - $vals1[3]))));} else {push (@F, "NA"); } print join("\t", @F);' | gzip > /mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_PP_summary_stats.wHapMap27.wSE.txt.gz
zcat /mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_MAP_summary_stats.wHapMap27.txt.gz | perl -lane 'my @vals1 = split(/_/, $F[5]); if (($F[5] ne "NA") && ($vals1[3] > 0) && ($vals1[3] < 1)) { push(@F, sqrt(209.79683 / (2 * $F[4] * $vals1[3] * (1 - $vals1[3]))));} else {push (@F, "NA"); } print join("\t", @F);' | gzip > /mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_MAP_summary_stats.wHapMap27.wSE.txt.gz



[  mturchin20@bigmem01  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Data/iGENBP2015/iGENBP2015.SupplTable4.txt | perl -lane 'print join("\t", @F[0..11]);' | grep MAP | perl -lane 'my $varY = $F[10]**2 * 2 * $F[7] * $F[8] * (1 - $F[8]); print $varY;' | R -q -e "Data1 <- read.table(file('stdin'), header=F); print(c(length(Data1[,]), mean(Data1[,1]), sd(Data1[,1])));"
> Data1 <- read.table(file('stdin'), header=F); print(c(length(Data1[,]), mean(Data1[,1]), sd(Data1[,1])));
[1]  14.00000 209.79683  50.63987
>
>
~

#var(Y) values for each phenotype
#SBP = 434.46853
#DBP = 162.42747
#PP = 219.57919
#MAP = 209.79683























##MAGIC2010 -- dbSNP 129, no GWAS (18 < 5e-8 w/ follow-up), N = "...up to 46,186 nondiabetic participants" (source: http://www.nature.com/ng/journal/v42/n2/full/ng.520.html)
##20150806

mkdir /mnt/gluster/data/external_public_supp/MAGIC2010
mkdir /mnt/gluster/data/external_public_supp/MAGIC2012

cd /mnt/gluster/data/external_public_supp/MAGIC2010

wget ftp://ftp.sanger.ac.uk/pub/magic/MAGIC_ln_*.txt
wget ftp://ftp.sanger.ac.uk/pub/magic/MAGIC_FastingGlucose.txt

~~~
[  mturchin20@spudling50  /mnt/gluster/data/external_public_supp/MAGIC2010]$head -n 5 *
==> MAGIC_FastingGlucose.txt <==
snp     effect_allele   other_allele    maf     effect  stderr  pvalue
rs10    a       c       .033    -1.2e-02        1.2e-02 .3211
rs1000000       a       g       .274    3.0e-03 4.4e-03 .5052
rs10000010      t       c       .446    3.0e-04 3.7e-03 .9311
rs10000012      c       g       .146    4.3e-03 5.4e-03 .4268

==> MAGIC_ln_FastingInsulin.txt <==
snp     effect_allele   other_allele    maf     effect  stderr  pvalue
rs10    a       c       .033    1.4e-03 1.2e-02 .9058
rs1000000       a       g       .274    2.8e-03 4.6e-03 .543
rs10000010      t       c       .446    3.4e-03 3.9e-03 .3826
rs10000012      c       g       .146    -5.0e-04        5.7e-03 .9293

==> MAGIC_ln_HOMA-B.txt <==
snp     effect_allele   other_allele    maf     effect  stderr  pvalue
rs10    a       c       .033    9.4e-03 1.0e-02 .3533
rs1000000       a       g       .274    6.0e-04 3.9e-03 .8824
rs10000010      t       c       .446    3.5e-03 3.3e-03 .2925
rs10000012      c       g       .146    -1.0e-04        5.0e-03 .9834

==> MAGIC_ln_HOMA-IR.txt <==
snp     effect_allele   other_allele    maf     effect  stderr  pvalue
rs10    a       c       .033    2.5e-03 1.3e-02 .8436
rs1000000       a       g       .274    4.1e-03 4.8e-03 .3917
rs10000010      t       c       .446    2.4e-03 4.0e-03 .5571
rs10000012      c       g       .146    -5.0e-04        6.0e-03 .9272

[  mturchin20@spudling50  /mnt/gluster/data/external_public_supp/MAGIC2010]$ls -lrt | awk '{ print $9 }' | perl -lane 'system("cat $F[0] | wc | xargs echo $F[0]");'
0 0 0
MAGIC_ln_HOMA-IR.txt 2458074 17206518 102203586
MAGIC_ln_HOMA-B.txt 2456946 17198622 102153416
MAGIC_ln_FastingInsulin.txt 2461106 17227742 102330320
MAGIC_FastingGlucose.txt 2470477 17293339 102729940
[  mturchin20@spudling50  /mnt/gluster/data/external_public_supp/MAGIC2010]$for i in `ls -lrt | awk '{ print $9}'`; do cat $i | wc | xargs echo $i ; done
MAGIC_ln_HOMA-IR.txt 2458074 17206518 102203586
MAGIC_ln_HOMA-B.txt 2456946 17198622 102153416
MAGIC_ln_FastingInsulin.txt 2461106 17227742 102330320
MAGIC_FastingGlucose.txt 2470477 17293339 102729940
~~~

gzip * 

#20150806 CHECK_0: Create processed, ready files with dbSNP Chr BP snpID maf N pval
#20150806 20150806 CHECK_1: Downlaod dbSNP129 version of file from UCSC Genome Browser

#On how to get dbSNP database versions as I did for dbSNP130 (but didn't include any info on that here in this log)
#Source:
~~~
having said that, the best places where you may bulk download all dbSNP SNPs from are:

dbSNP itself, through its FTP site (there's no single file containing all the SNPs, so you'd need some reading and guidance here)
UCSC Browser, on the "Tables" section, selecting group "Variation and repeats"
Ensembl's Biomart, selecting "Ensembl Variation 61" as database if you want dbSNP132 variants, or through the latest version on the archive mapped to hg18 (Ensembl 54), searching through its BioMart section for dbSNP129 (since dbSNP130 was never added to that Ensembl version)
~~~

#I did the UCSC Browser/Table option, going for the correct hg version 'ncbi36/hg18' then 'SNPs (129)'
#Also tried doing a direct wget from the database as well

cd /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/
wget http://hgdownload.cse.ucsc.edu/goldenPath/hg18/database/snp129.txt.gz

#Creating a more generic version to be used 'regardless' of the dataset being called against the dbSNP file
cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs1.py /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs3.py
#Also creating a version, once again similar to vs2, where the order of the two input files are switched (not really sure as of 20150806 why I did not go forwad with this switch previously)
cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs3.py /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs4.py
#Maybe similar to the creation of the 'vs2' file, as I was writing up the 'vs4' one the 'vs3' files finished with around ~<4gb being used per run -- had assumed wit would go over or end up taking a larger amount of space, hence beginning to write the 'vs4' of this file. 'vs4' therefore is not finished nor tested

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs3.py --file1 /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/snp129.txt.gz --file2 /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_FastingInsulin.txt.gz | gzip > /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_FastingInsulin.wUCSCGB_snp129.vs3.txt.gz & 
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs3.py --file1 /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/snp129.txt.gz --file2 /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_FastingGlucose.txt.gz | gzip > /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_FastingGlucose.wUCSCGB_snp129.vs3.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs3.py --file1 /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/snp129.txt.gz --file2 /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_HOMA-B.txt.gz | gzip > /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_HOMA-B.wUCSCGB_snp129.vs3.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs3.py --file1 /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/snp129.txt.gz --file2 /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_HOMA-IR.txt.gz | gzip > /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_HOMA-IR.wUCSCGB_snp129.vs3.txt.gz & 


mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/MAGIC2010
mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/MAGIC2010/Vs1

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/process.MTedits.ForICBP2011.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/MAGIC2010/Vs1/process.MTedits.ForMAGIC2010.vs1.R

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_FastingInsulin.txt.gz | gzip > /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_FastingInsulin.wHapMap21.vs3.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_FastingInsulin.txt.gz | gzip > /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_FastingInsulin.wHapMap22.vs3.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_FastingInsulin.txt.gz | gzip > /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_FastingInsulin.wHapMap23.vs3.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_FastingInsulin.txt.gz | gzip > /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_FastingInsulin.wHapMap24.vs3.txt.gz &

#20150904 -- Going ahead with HapMap22 since don't really see many GWAS doing HapMap23? And nothing really obvious in manuscript/supplemental information (just statements about HapMap CEU Phase II, nothing specifically regarding release ##)
~~~
[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/MAGIC2010]$zcat /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_FastingInsulin.wHapMap21.vs3.txt.gz | awk '{ print $8 }' | grep NA | wc
26696   26696   80088
[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/MAGIC2010]$zcat /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_FastingInsulin.wHapMap22.vs3.txt.gz | awk '{ print $8 }' | grep NA | wc
430     430    1290
[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/MAGIC2010]$zcat /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_FastingInsulin.wHapMap23.vs3.txt.gz | awk '{ print $8 }' | grep NA | wc
295     295     885
[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/MAGIC2010]$zcat /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_FastingInsulin.wHapMap24.vs3.txt.gz | awk '{ print $8 }' | grep NA | wc
621     621    1863
~~~

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_FastingGlucose.txt.gz | gzip > /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_FastingGlucose.wHapMap22.vs3.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_HOMA-B.txt.gz | gzip > /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_HOMA-B.wHapMap22.vs3.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_HOMA-IR.txt.gz | gzip > /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_HOMA-IR.wHapMap22.vs3.txt.gz &

~~~
[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/MAGIC2010]$for i in `ls -lrt | awk '{ print $9 }' | grep wHapMap22.vs3.txt.gz`; do echo $i; paste -d @ <(zcat $i | awk '{ print $8 }' | grep NA | wc) <(zcat $i | awk '{ print $8 }' | grep , | wc) <(zcat $i | awk '{ print $8 }' | perl -lane 'my @vals1 = split(/,/, $F[0]); print scalar(@vals1);' | sort | uniq -c) <(zcat $i | awk '{ print $4 }' | perl -lane 'if ($F[0] !~ m/\.0?[1-9]+/) { print $F[0]; }' | sort | uniq -c) | sed 's/@/ @ /g'| column -s $'@' -t ; done
MAGIC_ln_FastingInsulin.wHapMap22.vs3.txt.gz
430     430    1290          0       0       0    2461106 1       2278 0
								3098 .004
								324 .005
								27109 .008
								14653 .009
								1 maf
MAGIC_ln_HOMA-IR.wHapMap22.vs3.txt.gz
544     544    1632          0       0       0    2458074 1       2261 0
								3087 .004
								320 .005
								26845 .008
								14548 .009
								1 maf
MAGIC_ln_HOMA-B.wHapMap22.vs3.txt.gz
544     544    1632          0       0       0    2456946 1       2239 0
								3016 .004
								316 .005
								26555 .008
								14386 .009
								1 maf
MAGIC_FastingGlucose.wHapMap22.vs3.txt.gz
678     678    2034          0       0       0    2470477 1       2425 0
								3441 .004
								365 .005
								29273 .008
								15685 .009
								1 maf
~~~

zcat /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_FastingInsulin.wHapMap22.vs3.txt.gz | perl -lane 'my @vals1 = split(/_/, $F[7]); push(@F, @vals1); print join("\t", @F);' | awk '{ print $9, "\t", $10, "\t", $1, "\t", $4, "\t", $7, "\t46186" }' | gzip > /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_FastingInsulin.wHapMap22.vs3.formatted.maxN.txt.gz
zcat /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_FastingGlucose.wHapMap22.vs3.txt.gz | perl -lane 'my @vals1 = split(/_/, $F[7]); push(@F, @vals1); print join("\t", @F);' | awk '{ print $9, "\t", $10, "\t", $1, "\t", $4, "\t", $7, "\t46186" }' | gzip > /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_FastingGlucose.wHapMap22.vs3.formatted.maxN.txt.gz
zcat /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_HOMA-B.wHapMap22.vs3.txt.gz | perl -lane 'my @vals1 = split(/_/, $F[7]); push(@F, @vals1); print join("\t", @F);' | awk '{ print $9, "\t", $10, "\t", $1, "\t", $4, "\t", $7, "\t46186" }' | gzip > /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_HOMA-B.wHapMap22.vs3.formatted.maxN.txt.gz
zcat /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_HOMA-IR.wHapMap22.vs3.txt.gz | perl -lane 'my @vals1 = split(/_/, $F[7]); push(@F, @vals1); print join("\t", @F);' | awk '{ print $9, "\t", $10, "\t", $1, "\t", $4, "\t", $7, "\t46186" }' | gzip > /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_HOMA-IR.wHapMap22.vs3.formatted.maxN.txt.gz

~~~
[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/MAGIC2010]$for i in `ls -lrt | awk '{ print $9 }' | grep wHapMap22.vs3.txt.gz`; do echo $i; zcat /mnt/gluster/data/external_public_supp/MAGIC2010/$i | perl -lane 'my $SEovrBet; if ($F[4] != 0) { $SEovrBet = $F[5] / $F[4]; } else { $SEovrBet = $F[5]; } print $F[4], "\t", $F[5], "\t", abs($SEovrBet);' | sort -k 3,3 -rg | head -n 10; done
MAGIC_ln_FastingInsulin.wHapMap22.vs3.txt.gz
-1.0e-04        4.3e-02 430
-1.0e-04        4.3e-02 430
-1.0e-04        4.2e-02 420
1.0e-04 3.9e-02 390
-1.0e-04        3.6e-02 360
-1.0e-04        3.5e-02 350
1.0e-04 3.4e-02 340
-1.0e-04        3.4e-02 340
-1.0e-04        3.4e-02 340
1.0e-04 3.3e-02 330
MAGIC_ln_HOMA-IR.wHapMap22.vs3.txt.gz
1.0e-04 4.6e-02 460
1.0e-04 4.6e-02 460
1.0e-04 4.1e-02 410
1.0e-04 4.0e-02 400
1.0e-04 3.8e-02 380
-1.0e-04        3.8e-02 380
1.0e-04 3.7e-02 370
1.0e-04 3.7e-02 370
1.0e-04 3.6e-02 360
1.0e-04 3.6e-02 360
MAGIC_ln_HOMA-B.wHapMap22.vs3.txt.gz
-1.0e-04        4.6e-02 460
1.0e-04 4.3e-02 430
1.0e-04 4.0e-02 400
-1.0e-04        3.8e-02 380
1.0e-04 3.6e-02 360
1.0e-04 3.4e-02 340
1.0e-04 3.4e-02 340
-1.0e-04        3.4e-02 340
-1.0e-04        3.3e-02 330
-1.0e-04        3.3e-02 330
MAGIC_FastingGlucose.wHapMap22.vs3.txt.gz
1.0e-04 4.8e-02 480
-1.0e-04        4.3e-02 430
-1.0e-04        3.7e-02 370
1.0e-04 3.6e-02 360
-1.0e-04        3.5e-02 350
1.0e-04 3.4e-02 340
-1.0e-04        3.4e-02 340
-1.0e-04        3.4e-02 340
1.0e-04 3.3e-02 330
1.0e-04 3.3e-02 330
~~~

zcat /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_FastingInsulin.wHapMap22.vs3.txt.gz | perl -lane 'my $scaledSE; if ($F[4] == 0) { $scaledSE = $F[5] / 430; } else { $scaledSE = ($F[5] / abs($F[4])) / 430; } push(@F, (46186 * (1 - $scaledSE))); print join("\t", @F);' | awk '{ print $9, "\t", $10, "\t", $1, "\t", $4, "\t", $7, "\t", $14 }' | gzip > /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_FastingInsulin.wHapMap22.vs3.formatted.scaledSEN.txt.gz
zcat /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_FastingGlucose.wHapMap22.vs3.txt.gz | perl -lane 'my $scaledSE; if ($F[4] == 0) { $scaledSE = $F[5] / 460; } else { $scaledSE = ($F[5] / abs($F[4])) / 460; } push(@F, (46186 * (1 - $scaledSE))); print join("\t", @F);' | awk '{ print $9, "\t", $10, "\t", $1, "\t", $4, "\t", $7, "\t", $14 }' | gzip > /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_FastingGlucose.wHapMap22.vs3.formatted.scaledSEN.txt.gz
zcat /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_HOMA-B.wHapMap22.vs3.txt.gz | perl -lane 'my $scaledSE; if ($F[4] == 0) { $scaledSE = $F[5] / 460; } else { $scaledSE = ($F[5] / abs($F[4])) / 460; } push(@F, (46186 * (1 - $scaledSE))); print join("\t", @F);' | awk '{ print $9, "\t", $10, "\t", $1, "\t", $4, "\t", $7, "\t", $14 }' | gzip > /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_HOMA-B.wHapMap22.vs3.formatted.scaledSEN.txt.gz
zcat /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_HOMA-IR.wHapMap22.vs3.txt.gz | perl -lane 'my $scaledSE; if ($F[4] == 0) { $scaledSE = $F[5] / 480; } else { $scaledSE = ($F[5] / abs($F[4])) / 480; } push(@F, (46186 * (1 - $scaledSE))); print join("\t", @F);' | awk '{ print $9, "\t", $10, "\t", $1, "\t", $4, "\t", $7, "\t", $14 }' | gzip > /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_HOMA-IR.wHapMap22.vs3.formatted.scaledSEN.txt.gz



#

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_FastingGlucose.txt.gz | gzip > /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_FastingGlucose.wHapMap22.vs3.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_HOMA-B.txt.gz | gzip > /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_HOMA-B.wHapMap22.vs3.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_HOMA-IR.txt.gz | gzip > /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_HOMA-IR.wHapMap22.vs3.txt.gz &




#20160224
#Correcting sign issue -- quick hack to get results + first full run through of MAGIC2010

#NOTE -- flipping sign on p-value to make coding downstream easier for this, ie I'm not implying something by creating negative p-values here. This is just for a quick run through.

zcat /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_FastingInsulin.wHapMap22.vs3.txt.gz | perl -lane 'print join("\t", @F[0..$#F-1]), "\t", join("\t", (split("_", $F[$#F])));' | perl -lane 'if ($F[4] < 0) { ($F[1], $F[2]) = ($F[2], $F[1]); $F[3] = 1 - $F[3]; $F[4] = $F[4] * -1; } print join("\t", @F);' | sed 's/ChrBPAFInfo/Chr BP Ref RAF Alt AAF/g' | sed 's/maf/raf/g' | grep -v -w NA | gzip > /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_FastingInsulin.wHapMap22.vs3.IncAllele.txt.gz

~~~
[  mturchin20@spudling26  ~/Lab_Stuff/StephensLab/Multivariate/MAGIC2010/Vs1]$zcat /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_FastingInsulin.wHapMap22.vs3.IncAllele.txt.gz | awk '{ print $5 }' | perl -lane 'if ($F[0] < 0) { print join("\t", @F); }'
[  mturchin20@spudling26  ~/Lab_Stuff/StephensLab/Multivariate/MAGIC2010/Vs1]$
~~~

LANG2=${LANG}
LC_ALL2=${LC_ALL}
LANG=C
LC_ALL=C
export LC_ALL LANG

join <(zcat /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_FastingInsulin.wHapMap22.vs3.IncAllele.txt.gz | awk '{ print $1, "\t", $2 }' | sort) <(zcat /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_FastingGlucose.wHapMap22.vs3.txt.gz | perl -lane 'print join("\t", @F[0..$#F-1]), "\t", join("\t", (split("_", $F[$#F])));' | perl -lane 'if ($F[4] < 0) { ($F[1], $F[2]) = ($F[2], $F[1]); $F[3] = 1 - $F[3]; $F[4] = $F[4] * -1; } print join("\t", @F);' | sort) | perl -lane 'if ($F[1] ne $F[2]) { $F[7] = $F[7] * -1; } splice(@F, 1, 1); print join("\t", @F);' | sed 's/ChrBPAFInfo/Chr BP Ref RAF Alt AAF/g' | sed 's/maf/raf/g ' | grep -v -w NA | gzip >  /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_FastingGlucose.wHapMap22.vs3.MatchedToInsln.IncAllele.txt.gz
join <(zcat /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_FastingInsulin.wHapMap22.vs3.IncAllele.txt.gz | awk '{ print $1, "\t", $2 }' | sort) <(zcat /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_HOMA-B.wHapMap22.vs3.txt.gz | perl -lane 'print join("\t", @F[0..$#F-1]), "\t", join("\t", (split("_", $F[$#F])));' | perl -lane 'if ($F[4] < 0) { ($F[1], $F[2]) = ($F[2], $F[1]); $F[3] = 1 - $F[3]; $F[4] = $F[4] * -1; } print join("\t", @F);' | sort) | perl -lane 'if ($F[1] ne $F[2]) { $F[7] = $F[7] * -1; } splice(@F, 1, 1); print join("\t", @F);' | sed 's/ChrBPAFInfo/Chr BP Ref RAF Alt AAF/g' | sed 's/maf/raf/g ' | grep -v -w NA | gzip >  /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_HOMA-B.wHapMap22.vs3.MatchedToInsln.IncAllele.txt.gz
join <(zcat /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_FastingInsulin.wHapMap22.vs3.IncAllele.txt.gz | awk '{ print $1, "\t", $2 }' | sort) <(zcat /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_HOMA-IR.wHapMap22.vs3.txt.gz | perl -lane 'print join("\t", @F[0..$#F-1]), "\t", join("\t", (split("_", $F[$#F])));' | perl -lane 'if ($F[4] < 0) { ($F[1], $F[2]) = ($F[2], $F[1]); $F[3] = 1 - $F[3]; $F[4] = $F[4] * -1; } print join("\t", @F);' | sort) | perl -lane 'if ($F[1] ne $F[2]) { $F[7] = $F[7] * -1; } splice(@F, 1, 1); print join("\t", @F);' | sed 's/ChrBPAFInfo/Chr BP Ref RAF Alt AAF/g' | sed 's/maf/raf/g ' | grep -v -w NA | gzip >  /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_HOMA-IR.wHapMap22.vs3.MatchedToInsln.IncAllele.txt.gz

LANG=${LANG2}
LC_ALL=${LC_ALL2}
export LC_ALL LANG

zcat /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_FastingInsulin.wHapMap22.vs3.IncAllele.txt.gz | awk '{ print $8, "\t", $9, "\t", $1, "\t", $4, "\t", $7, "\t46186" }' | grep -v snp | cat <(echo "Chr      BP      snp     raf     pvalue         N") - | gzip > /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_FastingInsulin.wHapMap22.vs3.IncAllele.formatted.txt.gz 

zcat /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_FastingGlucose.wHapMap22.vs3.MatchedToInsln.IncAllele.txt.gz | awk '{ print $8, "\t", $9, "\t", $1, "\t", $4, "\t", $7, "\t46186" }' | grep -v snp | cat <(echo "Chr      BP      snp     raf     pvalue         N") - |  gzip > /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_FastingGlucose.wHapMap22.vs3.MatchedToInsln.IncAllele.formatted.txt.gz
zcat /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_HOMA-B.wHapMap22.vs3.MatchedToInsln.IncAllele.txt.gz | awk '{ print $8, "\t", $9, "\t", $1, "\t", $4, "\t", $7, "\t46186" }' | grep -v snp | cat <(echo "Chr      BP      snp     raf     pvalue         N") - |  gzip > /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_HOMA-B.wHapMap22.vs3.MatchedToInsln.IncAllele.formatted.txt.gz
zcat /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_HOMA-IR.wHapMap22.vs3.MatchedToInsln.IncAllele.txt.gz | awk '{ print $8, "\t", $9, "\t", $1, "\t", $4, "\t", $7, "\t46186" }' | grep -v snp | cat <(echo "Chr      BP      snp     raf     pvalue         N") - |  gzip > /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_HOMA-IR.wHapMap22.vs3.MatchedToInsln.IncAllele.formatted.txt.gz


LANG2=${LANG}
LC_ALL2=${LC_ALL}
LANG=C
LC_ALL=C
export LC_ALL LANG

join -a 1 -a 2 -1 3 -2 3 -e NA2 -o 0 1.1 1.2 1.4 1.5 1.6 2.1 2.2 2.4 2.5 2.6 <(zcat /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_FastingInsulin.wHapMap22.vs3.IncAllele.formatted.txt.gz | sort -g -k 3,3) <(zcat /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_FastingGlucose.wHapMap22.vs3.MatchedToInsln.IncAllele.formatted.txt.gz | sort -g -k 3,3 ) | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
join -a 1 -a 2 -1 1 -2 3 -e NA2 -o 0 1.2 2.1 2.2 2.4 2.5 2.6 - <(zcat /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_HOMA-B.wHapMap22.vs3.MatchedToInsln.IncAllele.formatted.txt.gz  | sort -g -k 3,3) | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); print join("\t", @F);' | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); if ($#F == 6) { splice(@F, 1, 0, (("NA2") x 9)); } my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
join -a 1 -a 2 -1 1 -2 3 -e NA2 -o 0 1.2 2.1 2.2 2.4 2.5 2.6 - <(zcat /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_HOMA-IR.wHapMap22.vs3.MatchedToInsln.IncAllele.formatted.txt.gz | sort -g -k 3,3) | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); print join("\t", @F);' | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); if ($#F == 6) { splice(@F, 1, 0, (("NA2") x 14)); } my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
perl -lane 'print $F[0], "\t", join("\t", split(/_/, $F[1]));' | sed 's/chr//g' | gzip > /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC2010_AllPhenos.wHapMap22.vs3.MatchedToInsln.IncAllele.formatted.RAF.txt.pre1.gz

LANG=${LANG2}
LC_ALL=${LC_ALL2}
export LC_ALL LANG

~~~
[  mturchin20@spudling26  ~/Lab_Stuff/StephensLab/Multivariate/MAGIC2010/Vs1]$zcat /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC2010_AllPhenos.wHapMap22.vs3.MatchedToInsln.IncAllele.formatted.RAF.txt.pre1.gz | wc
2498320 52464720 313999785
[  mturchin20@spudling26  ~/Lab_Stuff/StephensLab/Multivariate/MAGIC2010/Vs1]$zcat /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC2010_AllPhenos.wHapMap22.vs3.MatchedToInsln.IncAllele.formatted.RAF.txt.pre1.gz | head -n 10
snp     Chr     BP      raf     pvalue  N       Chr     BP      raf     pvalue  N       Chr     BP      raf     pvalue  N       Chr     BP      raf     pvalue  N
rs1172982  1    100002699       .3      .6263   46186   1       100002699       .3      .3079   46186   1       100002699       0.7     -0.621  46186   1       100002699       .3      .3301   46186
rs1172981  1    100002785       .067    .2911   46186   1       100002785       .067    .184    46186   1       100002785       .067    .6009   46186   1       100002785       .067    .1848   46186
rs11166327 1    100003455       0.953   .2279   46186   1       100003455       0.953   .1596   46186   1       100003455       0.953   .5821   46186   1       100003455       0.953   .1334   46186
rs4908018  1    100006955       .076    .3404   46186   1       100006955       .076    .1656   46186   1       100006955       .076    .6867   46186   1       100006955       .076    .1999   46186
rs2392072  1    100007331       .341    .9666   46186   1       100007331       .341    .4242   46186   1       100007331       0.659   -0.4527 46186   1       100007331       .341    .5542   46186
rs12129588 1    100007519       0.938   .4043   46186   1       100007519       0.938   .2572   46186   1       100007519       0.938   .6031   46186   1       100007519       0.938   .3109   46186
rs10875256 1    100007856       0.942   .3275   46186   1       100007856       0.942   .1665   46186   1       100007856       0.942   .6451   46186   1       100007856       0.942   .2078   46186
rs7541580  1    100008014       .252    .7364   46186   1       100008014       0.748   -0.9864 46186   1       100008014       .252    .3875   46186   1       100008014       0.748   -0.9717 46186
rs883532   1    100008025       .143    .7463   46186   1       100008025       0.857   -0.9897 46186   1       100008025       .143    .3972   46186   1       100008025       0.857   -0.9543 46186
[  mturchin20@spudling26  ~/Lab_Stuff/StephensLab/Multivariate/MAGIC2010/Vs1]$zcat /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC2010_AllPhenos.wHapMap22.vs3.MatchedToInsln.IncAllele.formatted.RAF.txt.pre1.gz | perl -lane 'print $#F;' | sort | uniq -c
2498320 20
~~~

zcat /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC2010_AllPhenos.wHapMap22.vs3.MatchedToInsln.IncAllele.formatted.RAF.txt.pre1.gz | grep -v NA2 | gzip > /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC2010_AllPhenos.wHapMap22.vs3.MatchedToInsln.IncAllele.formatted.RAF.txt.pre2.gz

zcat /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC2010_AllPhenos.wHapMap22.vs3.MatchedToInsln.IncAllele.formatted.RAF.txt.pre2.gz | perl -lane 'splice(@F, 21, 3); splice(@F, 16, 3); splice(@F, 11, 3); splice(@F, 6, 3); print join("\t", @F);' | gzip > /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC2010_AllPhenos.wHapMap22.vs3.MatchedToInsln.IncAllele.formatted.RAF.txt.gz

zcat /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC2010_AllPhenos.wHapMap22.vs3.MatchedToInsln.IncAllele.formatted.RAF.txt.pre1.gz  | grep -w -f /mnt/lustre/home/mturchin20/Data/MAGIC2010/MAGIC2010.Table1.rsID.txt | awk '{ print $1, "\t", $2, "\t", $3 }' > /mnt/lustre/home/mturchin20/Data/MAGIC2010/MAGIC2010.Table1.MarkerChrBP.txt

~~~
[  mturchin20@spudling26  ~/Lab_Stuff/StephensLab/Multivariate/MAGIC2010/Vs1]$zcat /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC2010_AllPhenos.wHapMap22.vs3.MatchedToInsln.IncAllele.formatted.RAF.txt.pre1.gz  | grep -f /mnt/lustre/home/mturchin20/Data/MAGIC2010/MAGIC2010.Table1.rsID.txt | wc
     30     630    3995
[  mturchin20@spudling26  ~/Lab_Stuff/StephensLab/Multivariate/MAGIC2010/Vs1]$cat  /mnt/lustre/home/mturchin20/Data/MAGIC2010/MAGIC2010.Table1.rsID.txt | wc
     18      18     180
[  mturchin20@spudling26  ~/Lab_Stuff/StephensLab/Multivariate/MAGIC2010/Vs1]$zcat /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC2010_AllPhenos.wHapMap22.vs3.MatchedToInsln.IncAllele.formatted.RAF.txt.pre1.gz  | grep -w -f /mnt/lustre/home/mturchin20/Data/MAGIC2010/MAGIC2010.Table1.rsID.txt | wc
     17     357    2319
[  mturchin20@spudling26  ~/Lab_Stuff/StephensLab/Multivariate/MAGIC2010/Vs1]$zcat /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC2010_AllPhenos.wHapMap22.vs3.MatchedToInsln.IncAllele.formatted.RAF.txt.pre1.gz  | grep -w -f /mnt/lustre/home/mturchin20/Data/MAGIC2010/MAGIC2010.Table1.rsID.txt     
rs340874   1    212225879       .438    .03652  46186   1       212225879       0.562   -7.082e-08      46186   1       212225879       .438    .00003429       46186   1       212225879       .438    .326    46186
rs10885122 10   113032083       .1      .1457   46186   10      113032083       0.9     -8.414e-11      46186   10      113032083       .1      3.331e-06       46186   10      113032083       .1      .729    46186
rs4506565  10   114746031       .296    .0086   46186   10      114746031       0.704   -1.235e-08      46186   10      114746031       .296    1.391e-06       46186   10      114746031       .296    .06558  46186
rs11605924 11   45829667        0.542   .4872   46186   11      45829667        .458    -1.52e-09       46186   11      45829667        0.542   4.690e-06       46186   11      45829667        .458    -0.9304 46186
rs7944584  11   47292896        0.712   .5613   46186   11      47292896        .288    -1.466e-09      46186   11      47292896        0.712   .0000864        46186   11      47292896        0.712   .8643   46186
rs174550   11   61328054        0.633   .1464   46186   11      61328054        .367    -1.479e-08      46186   11      61328054        0.633   .00003772       46186   11      61328054        0.633   .6763   46186
rs10830963 11   92348358        .3      .3988   46186   11      92348358        0.7     -1.263e-68      46186   11      92348358        .3      8.600e-23       46186   11      92348358        0.7     -0.09074        46186
rs35767 12 101399699    0.885   9.575e-08       46186   12      101399699       0.885   .5437   46186   12      101399699       0.885   3.678e-06       46186   12      101399699       0.885   7.572e-08       46186
rs11071657 15   60221254        .408    .3601   46186   15      60221254        .408    2.827e-07       46186   15      60221254        0.592   -0.2241 46186   15      60221254        .408    .05871  46186
rs560887   2    169471394       .326    .06238  46186   2       169471394       0.674   -4.61e-75       46186   2       169471394       .326    7.674e-29       46186   2       169471394       0.674   -0.5639 46186
rs780094   2    27594741        0.606   .00009773       46186   2       27594741        0.606   2.486e-12       46186   2       27594741        0.606   .2515   46186   2       27594741        0.606   7.626e-07       46186
rs11708067 3    124548468       0.774   .5302   46186   3       124548468       .226    -8.724e-09      46186   3       124548468       0.774   .0001769        46186   3       124548468       .226    -0.8395 46186
rs11920090 3    172200215       .147    .2408   46186   3       172200215       0.853   -1.87e-06       46186   3       172200215       .147    .0001396        46186   3       172200215       .147    .9646   46186
rs2191349  7    15030834        0.533   .1875   46186   7       15030834        .467    -7.826e-17      46186   7       15030834        0.533   5.041e-11       46186   7       15030834        0.533   .6952   46186
rs4607517  7    44202193        .195    .2253   46186   7       44202193        .195    4.557e-36       46186   7       44202193        0.805   -4.836e-08      46186   7       44202193        .195    .006324 46186
rs11558471 8    118254914       0.748   .2086   46186   8       118254914       .252    -2.616e-11      46186   8       118254914       0.748   1.409e-06       46186   8       118254914       0.748   .714    46186
rs7034200  9    4279050 0.525   .0477   46186   9       4279050 .475    -0.0001204      46186   9       4279050 0.525   1.668e-06       46186   9       4279050 0.525   .06444  46186
[  mturchin20@spudling26  ~/Lab_Stuff/StephensLab/Multivariate/MAGIC2010/Vs1]$cat /mnt/lustre/home/mturchin20/Data/MAGIC2010/MAGIC2010.Table1.rsID.txt | sort | uniq -c
      1 rs10830963
      1 rs10885122
      1 rs11071657
      1 rs11558471
      1 rs11605924
      1 rs11708067
      1 rs11920090
      1 rs174550
      1 rs2191349
      1 rs340874
      1 rs35767
      1 rs4506565
      1 rs4607517
      1 rs560887
      1 rs7034200
      2 rs780094
      1 rs7944584
~~~

#Actually only 17 unique GWAS hits, not 18 -- see double count of rs780094

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/MAGIC2010/MAGIC2010.Table1.MarkerChrBP.txt --file2 /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC2010_AllPhenos.wHapMap22.vs3.MatchedToInsln.IncAllele.formatted.RAF.txt.gz | gzip > /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC2010_AllPhenos.wHapMap22.vs3.MatchedToInsln.IncAllele.formatted.RAF.wGWASannot.txt.gz

~~~
[  mturchin20@spudling26  ~/Lab_Stuff/StephensLab/Multivariate/MAGIC2010/Vs1]$zcat /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC2010_AllPhenos.wHapMap22.vs3.MatchedToInsln.IncAllele.formatted.RAF.wGWASannot.txt.gz | perl -lane 'print $F[$#F];' | sort | uniq -c
2402537 0
     17 1
  15302 2
~~~

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/MAGIC2010/Vs1/process.MTedits.ForMAGIC2010.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/MAGIC2010/Vs1/process.MTedits.ForMAGIC2010.vs1.SignCrrct.vs1.R 

LANG2=${LANG}
LC_ALL2=${LC_ALL}
LANG=C
LC_ALL=C
export LC_ALL LANG

join -a 2 -1 1 -2 1 -e NA4 -o 0 1.2 1.3 1.4 1.5 1.6 1.7 1.8 1.9 1.10 1.11 1.12 1.13 2.2 2.3 2.4 2.5 2.6 2.7 2.8 <(zcat /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC2010_AllPhenos.wHapMap22.vs3.MatchedToInsln.IncAllele.formatted.RAF.wGWASannot.txt.gz | sort -g -k 1,1 ) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/MAGIC2010/Vs1/MAGIC2010.dtlesssignif.vs1.SignCrrct.vs1.txt | perl -F, -lane 'print join("\t", @F);' | sort -g -k 1,1) | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/MAGIC2010/Vs1/MAGIC2010.dtlesssignif.vs1.SignCrrct.vs1.annot.vs1.RAF.txt.pre1.gz 

LANG=${LANG2}
LC_ALL=${LC_ALL2}
export LC_ALL LANG

zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/MAGIC2010/Vs1/MAGIC2010.dtlesssignif.vs1.SignCrrct.vs1.annot.vs1.RAF.txt.pre1.gz | grep -v NA4 | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/MAGIC2010/Vs1/MAGIC2010.dtlesssignif.vs1.SignCrrct.vs1.annot.vs1.RAF.txt.gz

#Added the following header to /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/MAGIC2010/Vs1/MAGIC2010.dtlesssignif.vs1.SignCrrct.vs1.annot.vs1.RAF.txt.gz
#snp chr pos raf p_FastIns n_FastIns p_FastGlu n_FastGlu p_HOMA_B n_HOMA_B p_HOMA_IR n_HOMA_IR annot Z.FastIns Z.FastGlu Z.HOMA_IR Z.HOMA_B mvstat mvp unip

zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/MAGIC2010/Vs1/MAGIC2010.dtlesssignif.vs1.SignCrrct.vs1.annot.vs1.RAF.txt.gz | perl -lane 'if ($F[3] > .5) { $F[3] = 1 - $F[3]; $F[13] = $F[13] * -1; $F[14] = $F[14] * -1; $F[15] = $F[15] * -1; $F[16] = $F[16] * -1; } print join("\t", @F);' | sed 's/raf/maf/g' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/MAGIC2010/Vs1/MAGIC2010.dtlesssignif.vs1.SignCrrct.vs1.annot.vs1.MAF.txt.gz

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/GLC.MTedits.ForICBP2011.vs1.SignCrrct.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/MAGIC2010/Vs1/GLC.MTedits.ForMAGIC2010.vs1.SignCrrct.vs1.R














##MAGIC2012 -- ??dbSNP (no obvious mention of specific dbSNP in main article or a few of the supplementary notes/tables...assuming same as MAGIC2010 for now?), 0 GWAS, (6 + 7 <5e-8 follow-up) N = 51750 (source: http://www.nature.com.proxy.uchicago.edu/ng/journal/v44/n6/full/ng.2274.html) 
##20150806

mkdir /mnt/gluster/data/external_public_supp/MAGIC2012
cd /mnt/gluster/data/external_public_supp/MAGIC2012

wget ftp://ftp.sanger.ac.uk/pub/magic/MAGIC_Manning*

#CHECK_0: Do FastingGlucose and FastingInsulin both w/out and w/ BMI adj? 'Interaction' file/information may not be relevant?
#CHECK_0: Apparently there is also HOMA-B and HOMA-IR but the results were not different SNP-implication wise from FastingGlucose and FastingInsulin so were not really discussed in the text (or therefore distributed either?). Data may be accessible though if contacted....if this is desired at all though?

~~~
[  mturchin20@spudling50  /mnt/gluster/data/external_public_supp/MAGIC2012]$for i in `ls -lrt | awk '{ print $9 }' | grep gz$`; do zcat $i | head -n 5; done
Snp     effect_allele   other_allele    maf     InterEffects    InterSE InterP  JMAcontBMI_MainEffect   JMAcontBMI_MainSE       JMAcontBMI_InterEffect  JMAcontBMI_InterSE      JMAcontBMI_P
rs10    a       c       0.033   6.0e-04 2.0e-03 0.7704  -3.7e-02        5.2e-02 0.001108        0.001973        0.5572
rs1000000       a       g       0.274   -1.6e-03        8.0e-04 0.0473  3.9e-02 2.0e-02 -0.001554       0.0007776       0.1257
rs10000010      t       c       0.446   -2.0e-04        7.0e-04 0.7666  1.4e-02 1.7e-02 -0.0003398      0.0006679       0.1515
rs10000012      c       g       0.146   -4.0e-04        1.0e-03 0.7043  9.0e-03 2.6e-02 -0.0002837      0.001015        0.8923
Snp     effect_allele   other_allele    maf     MainEffects     MainSE  MainP   BMIadjMainEffects       BMIadjMainSE    BMIadjMainP
rs10    a       c       0.033   1.3e-02 9.5e-03 0.1844  9.1e-03 8.2e-03 0.2655
rs1000000       a       g       0.274   4.7e-03 3.8e-03 0.2133  2.4e-03 3.2e-03 0.4582
rs10000010      t       c       0.446   8.0e-04 3.2e-03 0.8096  0.0e+00 2.7e-03 0.9901
rs10000012      c       g       0.146   6.6e-03 4.8e-03 0.169   4.0e-04 4.1e-03 0.9155
Snp     effect_allele   other_allele    maf     InterEffect     InterSE InterP  JMAcontBMI_MainEffect   JMAcontBMI_MainSE       JMAcontBMI_InterEffect  JMAcontBMI_InterSE      JMAcontBMI_P
rs10    a       c       0.033   8.0e-04 2.1e-03 0.701   -3.2e-03        5.2e-02 0.0004392       0.001975        0.5921
rs1000000       a       g       0.274   -1.0e-04        8.0e-04 0.9394  2.5e-03 2.0e-02 -4.248e-05      0.0007543       0.9126
rs10000010      t       c       0.446   4.0e-04 6.0e-04 0.5327  -9.9e-03        1.6e-02 0.0003529       0.0006084       0.8198
rs10000012      c       g       0.146   6.0e-04 1.0e-03 0.5469  -9.4e-03        2.5e-02 0.0003342       0.0009489       0.9253
Snp     effect_allele   other_allele    maf     MainEffects     MainSE  MainP   BMIadjMainEffects       BMIadjMainSE    BMIadjMainP
rs10    a       c       0.033   -3.0e-03        9.4e-03 0.7474  -3.8e-03        9.7e-03 0.6976
rs1000000       a       g       0.274   -3.0e-03        3.7e-03 0.4241  -2.7e-03        3.8e-03 0.4665
rs10000010      t       c       0.446   7.1e-03 3.1e-03 0.0218  6.9e-03 3.2e-03 0.02878
rs10000012      c       g       0.146   3.8e-03 4.6e-03 0.4133  2.0e-04 4.7e-03 0.9609
[  mturchin20@spudling50  /mnt/gluster/data/external_public_supp/MAGIC2012]$for i in `ls -lrt | awk '{ print $9 }' | grep gz$`; do zcat $i | wc | xargs echo $i; done
MAGIC_Manning_et_al_FastingGlucose_Interaction.txt.gz 2635762 31629144 230971153
MAGIC_Manning_et_al_lnFastingInsulin_MainEffect.txt.gz 2627849 26278490 176310142
MAGIC_Manning_et_al_lnFastingInsulin_Interaction.txt.gz 2634889 31618668 228394130
MAGIC_Manning_et_al_FastingGlucose_MainEffect.txt.gz 2628880 26288800 176378821
~~~

#CHECK_1: Create processed, ready files with dbSNP Chr BP snpID maf N pval

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs3.py --file1 /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/snp129.txt.gz --file2 /mnt/gluster/data/external_public_supp/MAGIC2012/MAGIC_Manning_et_al_FastingGlucose_MainEffect.txt.gz | gzip > /mnt/gluster/data/external_public_supp/MAGIC2012/MAGIC_Manning_et_al_FastingGlucose_MainEffect.wUCSCGB_snp129.vs3.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs3.py --file1 /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/snp129.txt.gz --file2 /mnt/gluster/data/external_public_supp/MAGIC2012/MAGIC_Manning_et_al_lnFastingInsulin_MainEffect.txt.gz | gzip > /mnt/gluster/data/external_public_supp/MAGIC2012/MAGIC_Manning_et_al_lnFastingInsulin_MainEffect.wUCSCGB_snp129.vs3.txt.gz &













##GEFOS2012 -- "...HapMap(37) Phase 2 release 22 using Genome Build 36..." dbSNP (source: http://www.nature.com/ng/journal/v44/n5/full/ng.2249.html), 34 <5e-8 82 <5e-6 GWAS, N=~33k for FN, ~32k for LS
##20150806

#Unsure if will be able to find male/female-specific N aside from on a 'per substudy' basis in one of the supplemental tables (Supp table 18a)? Maybe just do pooled ones?

#dbSNP may be 126?
#Source: http://hapmap.ncbi.nlm.nih.gov/
~~~
2007-03-02: HapMap Public Release #22

Genotypes and frequency data for SNPs remapped to NCBI build 36 (dbSNP b126) coordinates are now available for bulk download. These datasets will subsequently be made available in the HapMap genome browser and HapMart utility.

As detailed in the release notes, a number of SNPs were excluded from this release due to mapping inconsistencies in Build 35 relative to Build 36. Lists of excluded SNPs according to various criteria may be found here. Please refer to the README file for exclusion criteria and additional information.
~~~

cd /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/ 
wget http://hgdownload.cse.ucsc.edu/goldenPath/hg18/database/snp126.txt.gz

~~~
[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/GEFOS2012]$zcat /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/snp126.txt.gz | awk '{ print $2 }' | sort | uniq -c
993668 chr1
602274 chr10
315 chr10_random
595364 chr11
318 chr11_random
556092 chr12
426141 chr13
363 chr13_random
371075 chr14
345610 chr15
2758 chr15_random
374141 chr16
242 chr16_random
314480 chr17
7516 chr17_random
322040 chr18
246631 chr19
570 chr19_random
4991 chr1_random
932329 chr2
329300 chr20
168229 chr21
15457 chr21_random
203733 chr22
589 chr22_random
620 chr2_random
785296 chr3
2199 chr3_random
818023 chr4
3798 chr4_random
726211 chr5
416 chr5_random
787340 chr6
4106 chr6_random
635850 chr7
882 chr7_random
605604 chr8
3055 chr8_random
613690 chr9
3367 chr9_random
658 chrM
469247 chrX
3142 chrX_random
74211 chrY
~~~

mkdir /mnt/gluster/data/external_public_supp/GEFOS2012 
cd /mnt/gluster/data/external_public_supp/GEFOS2012

wget http://www.gefos.org/sites/default/files/GEFOS2_FNBMD_MEN_GC.txt.gz
wget http://www.gefos.org/sites/default/files/GEFOS2_FNBMD_WOMEN_GC.txt.gz 
wget http://www.gefos.org/sites/default/files/GEFOS2_FNBMD_POOLED_GC.txt.gz
wget http://www.gefos.org/sites/default/files/GEFOS2_LSBMD_MEN_GC.txt.gz
wget http://www.gefos.org/sites/default/files/GEFOS2_LSBMD_WOMEN_GC.txt.gz
wget http://www.gefos.org/sites/default/files/GEFOS2_LSBMD_POOLED_GC.txt.gz
wget http://www.gefos.org/sites/default/files/GEFOS2.Xchrom.zip
wget http://www.gefos.org/sites/default/files/GEFOS_COLLABORATORS.doc

~~~
[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/GEFOS2012]$for i in `ls -lrt | awk '{ print $9 }' | grep BMD`; do zcat $i | head -n 5 ; done              MarkerName      Allele1 Allele2 P-value Direction       Freq.Allele1.HapMapCEU
rs2326918       a       g       0.5354  -       0.868421
rs2439906       c       g       0.4282  +       0.0416667
rs10760160      a       c       0.6345  +       0.567797
rs977590        a       g       0.6132  -       0.958333
MarkerName      Allele1 Allele2 P-value Direction       Freq.Allele1.HapMapCEU
rs2326918       a       g       0.6202  +       0.868421
rs2439906       c       g       0.8267  -       0.0416667
rs10760160      a       c       0.2725  -       0.567797
rs977590        a       g       0.7812  +       0.958333
MarkerName      Allele1 Allele2 P-value Direction       Freq.Allele1.HapMapCEU
rs2326918       a       g       0.2035  +       0.868421
rs2439906       c       g       0.7632  -       0.0416667
rs10760160      a       c       0.04861 -       0.567797
rs977590        a       g       0.2221  +       0.958333
MarkerName      Allele1 Allele2 P-value Direction       Freq.Allele1.HapMapCEU
rs2326918       a       g       0.02072 -       0.868421
rs2439906       c       g       0.02548 +       0.0416667
rs10760160      a       c       0.3119  -       0.567797
rs977590        a       g       0.5364  +       0.958333
MarkerName      Allele1 Allele2 P-value Direction       Freq.Allele1.HapMapCEU
rs2326918       a       g       0.8546  -       0.868421
rs2439906       c       g       0.685   +       0.0416667
rs10760160      a       c       0.3734  -       0.567797
rs977590        a       g       0.9512  +       0.958333
MarkerName      Allele1 Allele2 P-value Direction       Freq.Allele1.HapMapCEU
rs2326918       a       g       0.4782  +       0.868421
rs2439906       c       g       0.8527  -       0.0416667
rs10760160      a       c       0.5397  -       0.567797
rs977590        a       g       0.3398  -       0.958333
[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/GEFOS2012]$for i in `ls -lrt | awk '{ print $9 }' | grep BMD`; do zcat $i | wc | xargs echo $i ; done     
GEFOS2_FNBMD_MEN_GC.txt.gz 2440560 14643360 76803430
GEFOS2_FNBMD_POOLED_GC.txt.gz 2478339 14870034 78011957
GEFOS2_FNBMD_WOMEN_GC.txt.gz 2506036 15036216 78876443
GEFOS2_LSBMD_MEN_GC.txt.gz 2467730 14806380 77697007
GEFOS2_LSBMD_POOLED_GC.txt.gz 2468081 14808486 77722644
GEFOS2_LSBMD_WOMEN_GC.txt.gz 2474729 14848374 77898019
~~~

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs3.py --file1 <(zcat /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/snp126.txt.gz | perl -lane 'if ($F[1] =~ m/(chr[0-9MXY]+)_.*/) { $F[1] = $1; } print join("\t", @F);') --file2 /mnt/gluster/data/external_public_supp/GEFOS2012/GEFOS2_FNBMD_MEN_GC.txt.gz | gzip > /mnt/gluster/data/external_public_supp/GEFOS2012/GEFOS2_FNBMD_MEN_GC.wUCSCGB_snp126.vs3.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs3.py --file1 <(zcat /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/snp126.txt.gz | perl -lane 'if ($F[1] =~ m/(chr[0-9MXY]+)_.*/) { $F[1] = $1; } print join("\t", @F);') --file2 /mnt/gluster/data/external_public_supp/GEFOS2012/GEFOS2_FNBMD_WOMEN_GC.txt.gz | gzip > /mnt/gluster/data/external_public_supp/GEFOS2012/GEFOS2_FNBMD_WOMEN_GC.wUCSCGB_snp126.vs3.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs3.py --file1 <(zcat /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/snp126.txt.gz | perl -lane 'if ($F[1] =~ m/(chr[0-9MXY]+)_.*/) { $F[1] = $1; } print join("\t", @F);') --file2 /mnt/gluster/data/external_public_supp/GEFOS2012/GEFOS2_FNBMD_POOLED_GC.txt.gz | gzip > /mnt/gluster/data/external_public_supp/GEFOS2012/GEFOS2_FNBMD_POOLED_GC.wUCSCGB_snp126.vs3.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs3.py --file1 <(zcat /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/snp126.txt.gz | perl -lane 'if ($F[1] =~ m/(chr[0-9MXY]+)_.*/) { $F[1] = $1; } print join("\t", @F);') --file2 /mnt/gluster/data/external_public_supp/GEFOS2012/GEFOS2_LSBMD_MEN_GC.txt.gz | gzip > /mnt/gluster/data/external_public_supp/GEFOS2012/GEFOS2_LSBMD_MEN_GC.wUCSCGB_snp126.vs3.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs3.py --file1 <(zcat /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/snp126.txt.gz | perl -lane 'if ($F[1] =~ m/(chr[0-9MXY]+)_.*/) { $F[1] = $1; } print join("\t", @F);') --file2 /mnt/gluster/data/external_public_supp/GEFOS2012/GEFOS2_LSBMD_WOMEN_GC.txt.gz | gzip > /mnt/gluster/data/external_public_supp/GEFOS2012/GEFOS2_LSBMD_WOMEN_GC.wUCSCGB_snp126.vs3.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs3.py --file1 <(zcat /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/snp126.txt.gz | perl -lane 'if ($F[1] =~ m/(chr[0-9MXY]+)_.*/) { $F[1] = $1; } print join("\t", @F);') --file2 /mnt/gluster/data/external_public_supp/GEFOS2012/GEFOS2_LSBMD_POOLED_GC.txt.gz | gzip > /mnt/gluster/data/external_public_supp/GEFOS2012/GEFOS2_LSBMD_POOLED_GC.wUCSCGB_snp126.vs3.txt.gz &








##GEFOS2015 --
##(from GEFOS 2012 "...HapMap(37) Phase 2 release 22 using Genome Build 36..." dbSNP (source: http://www.nature.com/ng/journal/v44/n5/full/ng.2249.html), 34 <5e-8 82 <5e-6 GWAS, N=~33k for FN, ~32k for LS)
##20150225

mkdir /mnt/gluster/data/external_public_supp/GEFOS2015
cd /mnt/gluster/data/external_public_supp/GEFOS2015

http://www.gefos.org/?q=content/data-release-2015

wget http://www.gefos.org/sites/default/files/README.txt
wget http://www.gefos.org/sites/default/files/GEFOS.seq_UK10K_BMD_GWAS_Terms_of_Use.pdf
wget http://www.gefos.org/sites/default/files/fa2stu.MAF0_.005.pos_.out_.gz
wget http://www.gefos.org/sites/default/files/fn2stu.MAF0_.005.pos_.out_.gz
wget http://www.gefos.org/sites/default/files/ls2stu.MAF0_.005.pos_.out_.gz

# /mnt/gluster/data/external_public_supp/GEFOS2015/fa2stu.MAF0_.005.pos_.out_.gz, /mnt/gluster/data/external_public_supp/GEFOS2015/fn2stu.MAF0_.005.pos_.out_.gz, /mnt/gluster/data/external_public_supp/GEFOS2015/ls2stu.MAF0_.005.pos_.out_.gz

mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GEFOS2015
mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1

~~~
[  mturchin20@spudling12  ~/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1]$zcat /mnt/gluster/data/external_public_supp/GEFOS2015/fa2stu.MAF0_.005.pos_.out_.gz | head -n 10
chromosome      position        rs_number       reference_allele        other_allele    eaf     beta    se      beta_95L        beta_95U        z       p-value _-log10_p-value q_statistic     q_p-value       i2      n_studies      n_samples       effects
1       12775981        chr1:12775981   G       T       0.955927        -0.062953       0.040715        -0.142754       0.016849        -1.546183       0.129581        0.887459        2.757511        0.599189        0.000000       5       -9      ---+-
1       12776140        chr1:12776140   C       T       0.997375        0.031278        0.263981        -0.486125       0.548682        0.118487        0.907511        0.042148        4.533338        0.103657        0.558824       3       -9      +--??
1       12776161        chr1:12776161   G       A       0.965792        -0.056473       0.040835        -0.136511       0.023564        -1.382954       0.175175        0.756528        2.318770        0.677354        0.000000       5       -9      ---+-
1       12776218        chr1:12776218   A       C       0.133095        0.027112        0.023202        -0.018364       0.072588        1.168497        0.251993        0.598612        4.922087        0.295386        0.187337       5       -9      --+-+
1       12776441        chr1:12776441   C       T       0.999364        0.472124        0.349409        -0.212718       1.156966        1.351206        0.185290        0.732147        0.957117        0.619676        0.000000       3       -9      -++??^
[  mturchin20@bigmem01  ~/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1]$zcat /mnt/gluster/data/external_public_supp/GEFOS2015/fa2stu.MAF0_.005.pos_.out_.gz | wc
9955366 189151954 1374529291
[  mturchin20@bigmem01  ~/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1]$zcat /mnt/gluster/data/external_public_supp/GEFOS2015/fa2stu.MAF0_.005.pos_.out_.gz | awk '{ if ($6 > .01) { print $0 } } ' | wc
8293153 157569907 1143500630
[  mturchin20@bigmem01  ~/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1]$zcat /mnt/gluster/data/external_public_supp/GEFOS2015/fa2stu.MAF0_.005.pos_.out_.gz | awk '{ if ($6 > .05) { print $0 } } ' | wc
6283522 119386918 864281277
[  mturchin20@spudling12  ~/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1]$zcat /mnt/gluster/data/external_public_supp/GEFOS2015/fa2stu.MAF0_.005.pos_.out_.gz | awk '{ if (($6 >= .05) && ($6 <= .95)) { print $0 } } ' | wc
6050596 114961324 831285723

~~~

#zcat /mnt/gluster/data/external_public_supp/GEFOS2015/fa2stu.MAF0_.005.pos_.out_.gz | perl -lane 'if ($F[6] < 0) { ($F[3], $F[4]) = ($F[4], $F[3]); $F[5] = 1 - $F[5]; $F[6] = $F[6] * -1; } print join("\t", @F);' | gzip > /mnt/gluster/data/external_public_supp/GEFOS2015/fa2stu.MAF0_.005.pos_.IncAllele.out_.gz
zcat /mnt/gluster/data/external_public_supp/GEFOS2015/fa2stu.MAF0_.005.pos_.out_.gz | awk '{ if (($6 >= .05) && ($6 <= .95)) { print $0 } } ' | perl -lane 'if ($F[6] < 0) { ($F[3], $F[4]) = ($F[4], $F[3]); $F[5] = 1 - $F[5]; $F[6] = $F[6] * -1; } print join("\t", @F);' | gzip > /mnt/gluster/data/external_public_supp/GEFOS2015/fa2stu.MAF0_.005.pos_.maf05.IncAllele.out_.gz
#From http://www.nature.com.proxy.uchicago.edu/nature/journal/v526/n7571/full/nature14878.html
#Doing >.05 maf cutoff since it's unclear whether single-snp tests or skat were using for results < .05 

LANG2=${LANG}
LC_ALL2=${LC_ALL}
LANG=C
LC_ALL=C
export LC_ALL LANG

join -1 1 -2 3 <(zcat /mnt/gluster/data/external_public_supp/GEFOS2015/fa2stu.MAF0_.005.pos_.maf05.IncAllele.out_.gz | awk '{ print $3, "\t", $4 }' | sort) <(zcat /mnt/gluster/data/external_public_supp/GEFOS2015/fn2stu.MAF0_.005.pos_.out_.gz | awk '{ if (($6 >= .05) && ($6 <= .95)) { print $0 } } ' | perl -lane 'if ($F[6] < 0) { ($F[3], $F[4]) = ($F[4], $F[3]); $F[5] = 1 - $F[5]; $F[6] = $F[6] * -1; } print join("\t", @F);' | sort -k 3,3) | perl -lane 'if ($F[1] ne $F[4]) { $F[12] = $F[12] * -1; } splice(@F, 1, 1); print join("\t", @F);' | gzip > /mnt/gluster/data/external_public_supp/GEFOS2015/fn2stu.MAF0_.005.pos_.maf05.MatchedTofa.IncAllele.out_.gz
join -1 1 -2 3 <(zcat /mnt/gluster/data/external_public_supp/GEFOS2015/fa2stu.MAF0_.005.pos_.maf05.IncAllele.out_.gz | awk '{ print $3, "\t", $4 }' | sort) <(zcat /mnt/gluster/data/external_public_supp/GEFOS2015/ls2stu.MAF0_.005.pos_.out_.gz | awk '{ if (($6 >= .05) && ($6 <= .95)) { print $0 } } ' | perl -lane 'if ($F[6] < 0) { ($F[3], $F[4]) = ($F[4], $F[3]); $F[5] = 1 - $F[5]; $F[6] = $F[6] * -1; } print join("\t", @F);' | sort -k 3,3) | perl -lane 'if ($F[1] ne $F[4]) { $F[12] = $F[12] * -1; } splice(@F, 1, 1); print join("\t", @F);' | gzip > /mnt/gluster/data/external_public_supp/GEFOS2015/ls2stu.MAF0_.005.pos_.maf05.MatchedTofa.IncAllele.out_.gz

LANG=${LANG2}
LC_ALL=${LC_ALL2}
export LC_ALL LANG

#From http://www.ncbi.nlm.nih.gov/pubmed/26367794
#53,236
#From http://www.nature.com.proxy.uchicago.edu/nature/journal/v526/n7571/full/nature14878.html
#32,965

zcat /mnt/gluster/data/external_public_supp/GEFOS2015/fa2stu.MAF0_.005.pos_.maf05.IncAllele.out_.gz | awk '{ print $1, "\t", $2, "\t", $3, "\t", $6, "\t", $12, "\t32965" }' | grep -v chromosome | cat <(echo "chromosome       position        rs_number       eaf     p-value        32965") - | gzip > /mnt/gluster/data/external_public_supp/GEFOS2015/fa2stu.MAF0_.005.pos_.maf05.IncAllele.formatted.out_.gz 

zcat /mnt/gluster/data/external_public_supp/GEFOS2015/fn2stu.MAF0_.005.pos_.maf05.MatchedTofa.IncAllele.out_.gz | awk '{ print $2, "\t", $3, "\t", $1, "\t", $6, "\t", $12, "\t32965" }' | grep -v chromosome | cat <(echo "chromosome       position        rs_number       eaf     p-value        32965") - | gzip > /mnt/gluster/data/external_public_supp/GEFOS2015/fn2stu.MAF0_.005.pos_.maf05.MatchedTofa.IncAllele.formatted.out_.gz 
zcat /mnt/gluster/data/external_public_supp/GEFOS2015/ls2stu.MAF0_.005.pos_.maf05.MatchedTofa.IncAllele.out_.gz | awk '{ print $2, "\t", $3, "\t", $1, "\t", $6, "\t", $12, "\t32965" }' | grep -v chromosome | cat <(echo "chromosome       position        rs_number       eaf     p-value        32965") - | gzip > /mnt/gluster/data/external_public_supp/GEFOS2015/ls2stu.MAF0_.005.pos_.maf05.MatchedTofa.IncAllele.formatted.out_.gz 

LANG2=${LANG}
LC_ALL2=${LC_ALL}
LANG=C
LC_ALL=C
export LC_ALL LANG

join -a 1 -a 2 -1 3 -2 3 -e NA2 -o 0 1.1 1.2 1.4 1.5 1.6 2.1 2.2 2.4 2.5 2.6 <(zcat /mnt/gluster/data/external_public_supp/GEFOS2015/fa2stu.MAF0_.005.pos_.maf05.IncAllele.formatted.out_.gz | sort -k 3,3) <(zcat /mnt/gluster/data/external_public_supp/GEFOS2015/fn2stu.MAF0_.005.pos_.maf05.MatchedTofa.IncAllele.formatted.out_.gz | sort -k 3,3 ) | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
join -a 1 -a 2 -1 1 -2 3 -e NA2 -o 0 1.2 2.1 2.2 2.4 2.5 2.6 - <(zcat /mnt/gluster/data/external_public_supp/GEFOS2015/ls2stu.MAF0_.005.pos_.maf05.MatchedTofa.IncAllele.formatted.out_.gz | sort -k 3,3) | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); print join("\t", @F);' | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); if ($#F == 6) { splice(@F, 1, 0, (("NA2") x 9)); } my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
perl -lane 'print $F[0], "\t", join("\t", split(/_/, $F[1]));' | gzip > /mnt/gluster/data/external_public_supp/GEFOS2015/AllPhenos2stu.MAF0_.005.pos_.maf05.MatchedTofa.IncAllele.formatted.RAF.out_.pre1.vs2.gz

LANG=${LANG2}
LC_ALL=${LC_ALL2}
export LC_ALL LANG

#When including sort -g vs sort (no -g); originally did -g and got odd 'NA2' pattern seen below. Reran using just 'sort -k 3,3' and not 'sort -g -k 3,3' -- maybe has something to do with there being 'chr#:#####' & 'rs#####' SNPs and not just 'rs######' SNPs?
~~~
[  mturchin20@bigmem01  ~/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1]$zcat /mnt/gluster/data/external_public_supp/GEFOS2015/AllPhenos2stu.MAF0_.005.pos_.maf05.MatchedTofa.IncAllele.formatted.RAF.out_.pre1.vs2.gz | grep -w rs1026364
rs1026364  3    113370010       0.345267        0.233148        32965   3       113370010       0.345267        0.003349        32965   3       113370010       0.345267        0.170923        32965
[  mturchin20@bigmem01  ~/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1]$zcat /mnt/gluster/data/external_public_supp/GEFOS2015/AllPhenos2stu.MAF0_.005.pos_.maf05.MatchedTofa.IncAllele.formatted.RAF.out_.pre1.gz | grep -w rs1026364
rs1026364  NA2  NA2     NA2     NA2     NA2     3       113370010       0.345267        0.003349        32965   3       113370010       0.345267        0.170923        32965
rs1026364  3    113370010       0.345267        0.233148        32965   NA2     NA2     NA2     NA2     NA2     NA2     NA2     NA2     NA2     NA2
[  mturchin20@bigmem01  ~/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1]$zcat /mnt/gluster/data/external_public_supp/GEFOS2015/AllPhenos2stu.MAF0_.005.pos_.maf05.MatchedTofa.IncAllele.formatted.RAF.out_.pre1.gz | wc
8243611 131897776 868195497
[  mturchin20@bigmem01  ~/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1]$zcat /mnt/gluster/data/external_public_supp/GEFOS2015/AllPhenos2stu.MAF0_.005.pos_.maf05.MatchedTofa.IncAllele.formatted.RAF.out_.pre1.gz | head -n 10
rs10875231 1    100000012       0.736038        0.504973        32965   1       100000012       0.736038        0.268903        32965   1       100000012       0.736038        0.439536        32965
rs6678176  1    100000827       0.671599        0.939880        32965   1       100000827       0.671599        0.393617        32965   1       100000827       0.671599        0.464566        32965
rs78286437 1    100000843       0.0640414       0.361240        32965   1       100000843       0.0640414       0.692619        32965   1       100000843       0.9359586       -0.970171       32965
rs76909621 1    100001201       0.0715195       0.643955        32965   1       100001201       0.9284805       -0.003477       32965   1       100001201       0.9284805       -0.51181        32965
rs78642210 1    100002490       0.0565632       0.387533        32965   1       100002490       0.0565632       0.723796        32965   1       100002490       0.9434368       -0.997111       32965
rs77140576 1    100002713       0.078043        0.653540        32965   1       100002713       0.921957        -0.001641       32965   1       100002713       0.921957        -0.629039       32965
rs113470118     1       100002714       0.0567224       0.394022        32965   1       100002714       0.0567224       0.679943        32965   1       100002714       0.0567224       0.993266        32965
rs7545818  1    100002882       0.65529 0.956953        32965   1       100002882       0.65529 0.404991        32965   1       100002882       0.65529 0.477079        32965
rs75635821 1    100002991       0.0770883       0.590676        32965   1       100002991       0.9229117       -0.00239        32965   1       100002991       0.9229117       -0.776492       32965
rs78948828 1    100003204       0.0550517       0.396100        32965   1       100003204       0.0550517       0.681206        32965   1       100003204       0.0550517       0.993171        32965
[  mturchin20@bigmem01  ~/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1]$zcat /mnt/gluster/data/external_public_supp/GEFOS2015/AllPhenos2stu.MAF0_.005.pos_.maf05.MatchedTofa.IncAllele.formatted.RAF.out_.pre1.gz | perl -lane 'print $#F;' | sort | uniq -c
8243611 15
[  mturchin20@bigmem01  ~/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1]$zcat /mnt/gluster/data/external_public_supp/GEFOS2015/AllPhenos2stu.MAF0_.005.pos_.maf05.MatchedTofa.IncAllele.formatted.RAF.out_.pre1.gz | grep -v NA2 | wc
3740785 59852560 443480696
[  mturchin20@spudhead  /mnt/gluster/data/external_public_supp/GEFOS2015]$join <(zcat /mnt/gluster/data/external_public_supp/GEFOS2015/AllPhenos2stu.MAF0_.005.pos_.maf05.MatchedTofa.IncAllele.formatted.RAF.out_.pre1.gz | sort) <(cat /mnt/lustre/home/mturchin20/Data/GEFOS2012/GEFOS2012.SupplTable4a4b.Marker.txt | sort) | wc
     61     976    6416
[  mturchin20@spudhead  /mnt/gluster/data/external_public_supp/GEFOS2015]$cat /mnt/lustre/home/mturchin20/Data/GEFOS2012/GEFOS2012.SupplTable4a4b.Marker.txt | wc
     61      61     616
[  mturchin20@spudling12  ~/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1]$join <(zcat /mnt/gluster/data/external_public_supp/GEFOS2015/AllPhenos2stu.MAF0_.005.pos_.maf05.MatchedTofa.IncAllele.formatted.RAF.out_.pre1.gz | sort) <(cat /mnt/lustre/home/mturchin20/Data/GEFOS2012/GEFOS2012.SupplTable4a4b.Marker.txt | sort) | awk '{ print $1, "\t", $2, "\t", $3, "\t", $7, "\t", $8, "\t", $12, "\t", $13 }' | sed 's/NA2//g' | head -n 10
rs10048146       16      86710660        16      86710660        16      86710660
rs10226308       7       37938422        7       37938422        7       37938422
rs10416218       19      33599127
rs10416218                       19      33599127        19      33599127
rs1053051        12      107367225       12      107367225       12      107367225
rs11623869       14      103883633       14      103883633       14      103883633
rs11755164       6       44639184        6       44639184        6       44639184
rs12407028       1       68647716        1       68647716        1       68647716
rs1286083        14      91442779
rs1286083                        14      91442779        14      91442779
[  mturchin20@spudling12  ~/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1]$cat /mnt/lustre/home/mturchin20/Data/GEFOS2012/GEFOS2012.SupplTable4a4b.MarkerChrBP.pre1.txt | sed 's/NA2//g' | head -n 10
rs10048146       16      86710660        16      86710660        16      86710660
rs10226308       7       37938422        7       37938422        7       37938422
rs10416218       19      33599127                                
rs10416218                       19      33599127        19      33599127
rs1053051        12      107367225       12      107367225       12      107367225
rs11623869       14      103883633       14      103883633       14      103883633
rs11755164       6       44639184        6       44639184        6       44639184
rs12407028       1       68647716        1       68647716        1       68647716
rs1286083        14      91442779                                
rs1286083                        14      91442779        14      91442779
[  mturchin20@spudling12  ~/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1]$cat /mnt/lustre/home/mturchin20/Data/GEFOS2012/GEFOS2012.SupplTable4a4b.MarkerChrBP.pre1.txt | sed 's/NA2//g' | awk '{ print $1, "\t", $2, "\t", $3 }' | head -n 10
rs10048146       16      86710660
rs10226308       7       37938422
rs10416218       19      33599127
rs10416218       19      33599127
rs1053051        12      107367225
rs11623869       14      103883633
rs11755164       6       44639184
rs12407028       1       68647716
rs1286083        14      91442779
rs1286083        14      91442779
[  mturchin20@spudling12  ~/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1]$cat /mnt/lustre/home/mturchin20/Data/GEFOS2012/GEFOS2012.SupplTable4a4b.MarkerChrBP.pre1.txt | sed 's/NA2//g' | awk '{ print $1, "\t", $2, "\t", $3 }' | perl -lane 'print $#F;' | sort | uniq -c
     61 2
[  mturchin20@bigmem01  ~/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1]$zcat /mnt/gluster/data/external_public_supp/GEFOS2015/AllPhenos2stu.MAF0_.005.pos_.maf05.MatchedTofa.IncAllele.formatted.RAF.out_.pre1.gz | wc
6050597 96809552 713179241
[  mturchin20@bigmem01  ~/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1]$zcat /mnt/gluster/data/external_public_supp/GEFOS2015/AllPhenos2stu.MAF0_.005.pos_.maf05.MatchedTofa.IncAllele.formatted.RAF.out_.pre1.gz | perl -lane 'print $#F;' | sort | uniq -c
6050597 15
[  mturchin20@bigmem01  ~/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1]$zcat /mnt/gluster/data/external_public_supp/GEFOS2015/AllPhenos2stu.MAF0_.005.pos_.maf05.MatchedTofa.IncAllele.formatted.RAF.out_.pre1.gz | grep -v NA2 | wc
5929103 94865648 702696457
~~~

zcat /mnt/gluster/data/external_public_supp/GEFOS2015/AllPhenos2stu.MAF0_.005.pos_.maf05.MatchedTofa.IncAllele.formatted.RAF.out_.pre1.gz | grep -v NA2 | gzip > /mnt/gluster/data/external_public_supp/GEFOS2015/AllPhenos2stu.MAF0_.005.pos_.maf05.MatchedTofa.IncAllele.formatted.RAF.out_.pre2.gz

zcat /mnt/gluster/data/external_public_supp/GEFOS2015/AllPhenos2stu.MAF0_.005.pos_.maf05.MatchedTofa.IncAllele.formatted.RAF.out_.pre2.gz | perl -lane 'splice(@F, 21, 3); splice(@F, 16, 3); splice(@F, 11, 3); splice(@F, 6, 3); print join("\t", @F);' | gzip > /mnt/gluster/data/external_public_supp/GEFOS2015/AllPhenos2stu.MAF0_.005.pos_.maf05.MatchedTofa.IncAllele.formatted.RAF.out_.gz

#ls -lrt | grep -E 'annot|tophit|gl' | awk '{ print $9 }' | perl -lane 'if ($F[0] =~ m/^(GEFOS2015)(\..*)/) { my $val1 = $1 . $2; my $val2 = $1 . ".GEFOS2015GWAShits" . $2; system("mv $val1 $val2"); }'
#mv GLC.MTedits.ForGEFOS2015.vs1.SignCrrct.vs1.R GLC.MTedits.ForGEFOS2015.wGEFOS2012GWAShits.vs1.SignCrrct.vs1.R
#mv GLC.MTedits.ForGEFOS2015.vs1.SignCrrct.vs1.DiscHit.vs1.R GLC.MTedits.ForGEFOS2015.wGEFOS2012GWAShitsvs1.SignCrrct.vs1.DiscHit.vs1.R

LANG2=${LANG}
LC_ALL2=${LC_ALL}
LANG=C
LC_ALL=C
export LC_ALL LANG

#join <(zcat /mnt/gluster/data/external_public_supp/GEFOS2015/AllPhenos2stu.MAF0_.005.pos_.maf05.MatchedTofa.IncAllele.formatted.RAF.out_.pre1.gz | sort -g) <(cat /mnt/lustre/home/mturchin20/Data/GEFOS2012/GEFOS2012.SupplTable4a4b.Marker.txt | sort -g) | awk '{ print $1, "\t", $2, "\t", $3, "\t", $7, "\t", $8, "\t", $12, "\t", $13 }' > /mnt/lustre/home/mturchin20/Data/GEFOS2012/GEFOS2012.SupplTable4a4b.MarkerChrBP.pre1.txt
cat /mnt/lustre/home/mturchin20/Data/GEFOS2012/GEFOS2012.SupplTable4a4b.MarkerChrBP.pre1.txt | sed 's/NA2//g' | awk '{ print $1, "\t", $2, "\t", $3 }' > /mnt/lustre/home/mturchin20/Data/GEFOS2012/GEFOS2012.SupplTable4a4b.MarkerChrBP.txt
#join <(zcat /mnt/gluster/data/external_public_supp/GEFOS2015/AllPhenos2stu.MAF0_.005.pos_.maf05.MatchedTofa.IncAllele.formatted.RAF.out_.pre1.gz | sort -g) <(cat /mnt/lustre/home/mturchin20/Data/GEFOS2012/GEFOS2012.SupplTable4a4b.Marker.txt | sort -g) | awk '{ print $1, "\t", $2, "\t", $3, "\t", $4, "\t", $5, "\t", $10, "\t", $15 }' > /mnt/lustre/home/mturchin20/Data/GEFOS2012/GEFOS2012.SupplTable4a4b.MarkerChrBPMAFpVals.txt
join <(zcat /mnt/gluster/data/external_public_supp/GEFOS2015/AllPhenos2stu.MAF0_.005.pos_.maf05.MatchedTofa.IncAllele.formatted.RAF.out_.pre1.gz | sort -g) <(cat /mnt/lustre/home/mturchin20/Data/GEFOS2015/GEFOS2015.Table1.Marker.txt | sort -g) | awk '{ print $1, "\t", $2, "\t", $3, "\t", $7, "\t", $8, "\t", $12, "\t", $13 }' > /mnt/lustre/home/mturchin20/Data/GEFOS2015/GEFOS2015.Table1.MarkerChrBP.txt

LANG=${LANG2}
LC_ALL=${LC_ALL2}
export LC_ALL LANG

#mv /mnt/gluster/data/external_public_supp/GEFOS2015/AllPhenos2stu.MAF0_.005.pos_.maf05.MatchedTofa.IncAllele.formatted.RAF.wGWASannot.out_.gz /mnt/gluster/data/external_public_supp/GEFOS2015/AllPhenos2stu.MAF0_.005.pos_.maf05.MatchedTofa.IncAllele.formatted.RAF.wGWASannot.GEFOS2012GWAShits.out_.gz
#mv /mnt/gluster/data/external_public_supp/GEFOS2015/AllPhenos2stu.MAF0_.005.pos_.maf05.MatchedTofa.IncAllele.formatted.RAF.wGWASannot.DiscHit.out_.gz /mnt/gluster/data/external_public_supp/GEFOS2015/AllPhenos2stu.MAF0_.005.pos_.maf05.MatchedTofa.IncAllele.formatted.RAF.wGWASannot.GEFOS2012GWAShits.DiscHit.out_.gz
#cat /mnt/lustre/home/mturchin20/Data/GEFOS2015/GEFOS2015.Table1.MarkerChrBP.txt /mnt/lustre/home/mturchin20/Data/GEFOS2015/GEFOS2015.SupplTable13.edited.MarkerChrBP.txt | sort | uniq > /mnt/lustre/home/mturchin20/Data/GEFOS2015/GEFOS2015.Table1SupplTable13.MarkerChrBP.txt

~~~
[  mturchin20@bigmem01  ~/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1]$cat /mnt/lustre/home/mturchin20/Data/GEFOS2015/GEFOS2015.Table1.MarkerChrBP.txt
rs11024028       11      16756873        11      16756873        11      16756873
rs6542457        2       119154872       2       119154872       2       119154872
~~~

#python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/GEFOS2012/GEFOS2012.SupplTable4a4b.MarkerChrBP.txt --file2 /mnt/gluster/data/external_public_supp/GEFOS2015/AllPhenos2stu.MAF0_.005.pos_.maf05.MatchedTofa.IncAllele.formatted.RAF.out_.gz | gzip > /mnt/gluster/data/external_public_supp/GEFOS2015/AllPhenos2stu.MAF0_.005.pos_.maf05.MatchedTofa.IncAllele.formatted.RAF.wGWASannot.out_.gz
##python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/GEFOS2012/GEFOS2012.SupplTable4a4b.MarkerChrBPMAFpVals.wGWASHitAnnot.DiscHit.MarkerChrBP.txt --file2 /mnt/gluster/data/external_public_supp/GEFOS2015/AllPhenos2stu.MAF0_.005.pos_.maf05.MatchedTofa.IncAllele.formatted.RAF.out_.gz | gzip > /mnt/gluster/data/external_public_supp/GEFOS2015/AllPhenos2stu.MAF0_.005.pos_.maf05.MatchedTofa.IncAllele.formatted.RAF.wGWASannot.DiscHit.out_.gz
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/GEFOS2015/GEFOS2015.SupplTable13.edited.MarkerChrBP.txt --file2 /mnt/gluster/data/external_public_supp/GEFOS2015/AllPhenos2stu.MAF0_.005.pos_.maf05.MatchedTofa.IncAllele.formatted.RAF.out_.gz | gzip > /mnt/gluster/data/external_public_supp/GEFOS2015/AllPhenos2stu.MAF0_.005.pos_.maf05.MatchedTofa.IncAllele.formatted.RAF.wGWASannot.out_.gz
#python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/GEFOS2015/GEFOS2015.Table1.SupplTable13.edited.MarkerChrBP.txt --file2 /mnt/gluster/data/external_public_supp/GEFOS2015/AllPhenos2stu.MAF0_.005.pos_.maf05.MatchedTofa.IncAllele.formatted.RAF.out_.gz | gzip > /mnt/gluster/data/external_public_supp/GEFOS2015/AllPhenos2stu.MAF0_.005.pos_.maf05.MatchedTofa.IncAllele.formatted.RAF.wGWASannot.DiscAndRepl.out_.gz

#A few 'hit SNPs' are not in all 3 files...hence lower 'hit SNP' count below?
~~~
#[  mturchin20@bigmem01  ~/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1]$zcat /mnt/gluster/data/external_public_supp/GEFOS2015/AllPhenos2stu.MAF0_.005.pos_.maf05.MatchedTofa.IncAllele.formatted.RAF.wGWASannot.out_.gz | perl -lane 'print $F[$#F];' | sort | uniq -c
#5823487 0
#     52 1
# 105564 2
#[  mturchin20@bigmem01  ~/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1]$zcat /mnt/gluster/data/external_public_supp/GEFOS2015/AllPhenos2stu.MAF0_.005.pos_.maf05.MatchedTofa.IncAllele.formatted.RAF.wGWASannot.DiscHit.out_.gz | perl -lane 'print $F[$#F];' | sort | uniq -c
#5883838 0
#     22 1
#  45243 2
[  mturchin20@bigmem01  ~/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1]$zcat /mnt/gluster/data/external_public_supp/GEFOS2015/AllPhenos2stu.MAF0_.005.pos_.maf05.MatchedTofa.IncAllele.formatted.RAF.wGWASannot.out_.gz | perl -lane 'print $F[$#F];' | sort | uniq -c
5879198 0
     30 1
  49875 2
~~~

#cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1/process.MTedits.ForEGG2013.Female.vs1.SignCrrct.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1/process.MTedits.ForGEFOS2015.vs1.SignCrrct.vs1.R

LANG2=${LANG}
LC_ALL2=${LC_ALL}
LANG=C
LC_ALL=C
export LC_ALL LANG

#join -a 2 -1 1 -2 1 -e NA4 -o 0 1.2 1.3 1.4 1.5 1.6 1.7 1.8 1.9 1.10 1.11 2.2 2.3 2.4 2.5 2.6 2.7 <(zcat /mnt/gluster/data/external_public_supp/GEFOS2015/AllPhenos2stu.MAF0_.005.pos_.maf05.MatchedTofa.IncAllele.formatted.RAF.wGWASannot.out_.gz | sort -g -k 1,1 ) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1/GEFOS2015.dtlesssignif.vs1.SignCrrct.vs1.txt | perl -F, -lane 'print join("\t", @F);' | sort -g -k 1,1) | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1/GEFOS2015.dtlesssignif.vs1.SignCrrct.vs1.annot.vs1.RAF.txt.pre1.gz
##join -a 2 -1 1 -2 1 -e NA4 -o 0 1.2 1.3 1.4 1.5 1.6 1.7 1.8 1.9 1.10 1.11 2.2 2.3 2.4 2.5 2.6 2.7 <(zcat /mnt/gluster/data/external_public_supp/GEFOS2015/AllPhenos2stu.MAF0_.005.pos_.maf05.MatchedTofa.IncAllele.formatted.RAF.wGWASannot.DiscHit.out_.gz | sort -g -k 1,1 ) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1/GEFOS2015.dtlesssignif.vs1.SignCrrct.vs1.txt | perl -F, -lane 'print join("\t", @F);' | sort -g -k 1,1) | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1/GEFOS2015.dtlesssignif.vs1.SignCrrct.vs1.annot.DiscHit.vs1.RAF.txt.pre1.gz
join -a 2 -1 1 -2 1 -e NA4 -o 0 1.2 1.3 1.4 1.5 1.6 1.7 1.8 1.9 1.10 1.11 2.2 2.3 2.4 2.5 2.6 2.7 <(zcat /mnt/gluster/data/external_public_supp/GEFOS2015/AllPhenos2stu.MAF0_.005.pos_.maf05.MatchedTofa.IncAllele.formatted.RAF.wGWASannot.out_.gz | sort -g -k 1,1 ) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1/GEFOS2015.dtlesssignif.vs1.SignCrrct.vs1.txt | perl -F, -lane 'print join("\t", @F);' | sort -g -k 1,1) | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1/GEFOS2015.dtlesssignif.vs1.SignCrrct.vs1.annot.vs1.RAF.txt.pre1.gz

LANG=${LANG2}
LC_ALL=${LC_ALL2}
export LC_ALL LANG

#zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1/GEFOS2015.dtlesssignif.vs1.SignCrrct.vs1.annot.vs1.RAF.txt.pre1.gz | grep -v NA4 | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1/GEFOS2015.dtlesssignif.vs1.SignCrrct.vs1.annot.vs1.RAF.txt.gz
##zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1/GEFOS2015.dtlesssignif.vs1.SignCrrct.vs1.annot.DiscHit.vs1.RAF.txt.pre1.gz | grep -v NA4 | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1/GEFOS2015.dtlesssignif.vs1.SignCrrct.vs1.annot.DiscHit.vs1.RAF.txt.gz
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1/GEFOS2015.dtlesssignif.vs1.SignCrrct.vs1.annot.vs1.RAF.txt.pre1.gz | grep -v NA4 | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1/GEFOS2015.dtlesssignif.vs1.SignCrrct.vs1.annot.vs1.RAF.txt.gz

#Added the following header to /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1/GEFOS2015.dtlesssignif.vs1.SignCrrct.vs1.annot.vs1.RAF.txt.gz
#snp chr pos raf p_fa n_fa p_fn n_fn p_ls n_ls annot Z.fa Z.fn Z.ls mvstat mvp unip
##Added the following header to /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1/GEFOS2015.dtlesssignif.vs1.SignCrrct.vs1.annot.vs1.RAF.txt.gz
##snp chr pos raf p_fa n_fa p_fn n_fn p_ls n_ls annot Z.fa Z.fn Z.ls mvstat mvp unip
##Same to /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1/GEFOS2015.dtlesssignif.vs1.SignCrrct.vs1.annot.DiscHit.vs1.RAF.txt.gz

#zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1/GEFOS2015.dtlesssignif.vs1.SignCrrct.vs1.annot.vs1.RAF.txt.gz | perl -lane 'if ($F[3] > .5) { $F[3] = 1 - $F[3]; $F[11] = $F[11] * -1; $F[12] = $F[12] * -1; $F[13] = $F[13] * -1; } print join("\t", @F);' | sed 's/raf/maf/g' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1/GEFOS2015.dtlesssignif.vs1.SignCrrct.vs1.annot.vs1.MAF.txt.gz
##zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1/GEFOS2015.dtlesssignif.vs1.SignCrrct.vs1.annot.DiscHit.vs1.RAF.txt.gz | perl -lane 'if ($F[3] > .5) { $F[3] = 1 - $F[3]; $F[11] = $F[11] * -1; $F[12] = $F[12] * -1; $F[13] = $F[13] * -1; } print join("\t", @F);' | sed 's/raf/maf/g' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1/GEFOS2015.dtlesssignif.vs1.SignCrrct.vs1.annot.DiscHit.vs1.MAF.txt.gz
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1/GEFOS2015.dtlesssignif.vs1.SignCrrct.vs1.annot.vs1.RAF.txt.gz | perl -lane 'if ($F[3] > .5) { $F[3] = 1 - $F[3]; $F[11] = $F[11] * -1; $F[12] = $F[12] * -1; $F[13] = $F[13] * -1; } print join("\t", @F);' | sed 's/raf/maf/g' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1/GEFOS2015.dtlesssignif.vs1.SignCrrct.vs1.annot.vs1.MAF.txt.gz

#cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1/GLC.MTedits.ForEGG2013.Female.vs1.SignCrrct.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1/GLC.MTedits.ForGEFOS2015.vs1.SignCrrct.vs1.R
##cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1/GLC.MTedits.ForGEFOS2015.vs1.SignCrrct.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1/GLC.MTedits.ForGEFOS2015.vs1.SignCrrct.vs1.DiscHit.vs1.R
cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1/GLC.MTedits.ForGEFOS2015.wGEFOS2012GWAShits.vs1.SignCrrct.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1/GLC.MTedits.ForGEFOS2015.vs1.SignCrrct.vs1.R

#cat /mnt/lustre/home/mturchin20/Data/GEFOS2012/GEFOS2012.SupplTable4a4b.MarkerChrBPMAFpVals.txt | sed 's/NA2/.99/g' | R -q -e "Data1 <- read.table(file('stdin'), header=F); Data1 <- cbind(Data1, apply(Data1[,5:ncol(Data1)], 1, function(x) { val1 <- max(-log10(abs(x))); val2 <- 0; if (val1 > -log10(5e-8)) { val2 <- 1; }; return(val2); })); write.table(Data1, row.names=FALSE, col.names=FALSE, quote=FALSE);" > /mnt/lustre/home/mturchin20/Data/GEFOS2012/GEFOS2012.SupplTable4a4b.MarkerChrBPMAFpVals.wGWASHitAnnot.txt 

#cat /mnt/lustre/home/mturchin20/Data/GEFOS2012/GEFOS2012.SupplTable4a4b.MarkerChrBPMAFpVals.wGWASHitAnnot.txt | perl -lane 'if ($F[$#F] == 1) { print join("\t", @F); }' > /mnt/lustre/home/mturchin20/Data/GEFOS2012/GEFOS2012.SupplTable4a4b.MarkerChrBPMAFpVals.wGWASHitAnnot.DiscHit.txt
#cat /mnt/lustre/home/mturchin20/Data/GEFOS2012/GEFOS2012.SupplTable4a4b.MarkerChrBPMAFpVals.wGWASHitAnnot.txt | perl -lane 'if ($F[$#F] != 1) { print join("\t", @F); }' | grep ^rs > /mnt/lustre/home/mturchin20/Data/GEFOS2012/GEFOS2012.SupplTable4a4b.MarkerChrBPMAFpVals.wGWASHitAnnot.ReplHit.txt
#cat /mnt/lustre/home/mturchin20/Data/GEFOS2012/GEFOS2012.SupplTable4a4b.MarkerChrBPMAFpVals.wGWASHitAnnot.DiscHit.txt | awk '{ print $1, "\t", $2, "\t", $3 }' > /mnt/lustre/home/mturchin20/Data/GEFOS2012/GEFOS2012.SupplTable4a4b.MarkerChrBPMAFpVals.wGWASHitAnnot.DiscHit.MarkerChrBP.txt
#cat /mnt/lustre/home/mturchin20/Data/GEFOS2012/GEFOS2012.SupplTable4a4b.MarkerChrBPMAFpVals.wGWASHitAnnot.ReplHit.txt | awk '{ print $1, "\t", $2, "\t", $3 }' > /mnt/lustre/home/mturchin20/Data/GEFOS2012/GEFOS2012.SupplTable4a4b.MarkerChrBPMAFpVals.wGWASHitAnnot.ReplHit.MarkerChrBP.txt

##cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1/GEFOS2015.newtophits.vs1.SignCrrct.vs1.txt | grep -v snp | awk '{ print $1, "\t", $2, "\t", $3 }' | python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/GEFOS2012/GEFOS2012.SupplTable4a4b.MarkerChrBPMAFpVals.wGWASHitAnnot.ReplHit.MarkerChrBP.txt --file2 - > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1/GEFOS2015.newtophits.vs1.SignCrrct.vs1.ReplHit.GWASannot.txt
#python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/GEFOS2012/GEFOS2012.SupplTable4a4b.MarkerChrBPMAFpVals.wGWASHitAnnot.ReplHit.MarkerChrBP.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1/GEFOS2015.newtophits.vs1.SignCrrct.vs1.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1/GEFOS2015.newtophits.vs1.SignCrrct.vs1.ReplHit.GWASannot.txt

#python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/GEFOS2012/GEFOS2012.SupplTable4a4b.MarkerChrBPMAFpVals.wGWASHitAnnot.ReplHit.MarkerChrBP.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1/GEFOS2015.DiscHit.newtophits.vs1.SignCrrct.vs1.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1/GEFOS2015.DiscHit.newtophits.vs1.SignCrrct.vs1.ReplHit.GWASannot.txt

#~~~
#[  mturchin20@spudling12  ~/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1/GEFOS2015.DiscHit.newtophits.vs1.SignCrrct.vs1.ReplHit.GWASannot.txt
#snp     chr     pos     maf     annot   Z.fa    Z.fn    Z.ls    lbfav   0
#chr10:124064829 10      124064829       0.411456        0       1.9669  4.9093  4.9436  7.4444  0
#rs1023940       6       151932778       0.465155        0       2.358   5.8998  7.1902  14.7391 2       rs7751941       6       151946658       2
#rs10946458      6       21391282        0.338425        0       -1.2118 -5.508  -4.4321 7.8613  2       rs9466056       6       21384613        2
#rs11024028      11      16756873        0.147176        0       1.6368  5.5971  5.6686  10.3758 0
#rs2291467       11      68216756        0.228162        0       -3.1525 -5.0423 -7.4457 13.7298 2       rs3736228       11      68201295        2
#rs35481967      4       88750703        0.242959        0       0.5711  4.7758  5.4164  8.7743  2       rs6532023       4       88773849        2
#~~~
















##EGG -- BW (& all studies?) 126dbSNP, BL no GWAS (7 <5e-6) -- BW no GWAS (7 <5e-8 w/ follow-up) -- Pubertal Grwoth 10 GWAS, BL N= 28459 -- BW N= 26836 -- PG= ~14k, ~11k, ~9k 
##20150807


#Seems like across all studies Hapmap Phase II release 22 was used (below source: http://hmg.oxfordjournals.org/content/24/4/1155.long#sec-9), which is db126
~~~
Genotypes within each study were obtained using high-density SNP arrays and then imputed for Ã¢ÂÂ¼2.5 M HapMap SNPs (Phase II, release 22; http://hapmap.ncbi.nlm.nih.gov/). 
~~~

#From 2013 paper maybe different HapMap? Try both?
#Also hapmap release 21 is appears like (from: http://hmg.oxfordjournals.org/content/22/13/2735.long) -- "... Imputation was performed using IMPUTE (34) or MACH (35) for roughly 2.5 million SNPs against HapMap Phase II (release 21 of 22)..."
#See below -- HapMap22 is correct based on ChrBP coordinates matching between the database and EGG files

mkdir /mnt/gluster/data/external_public_supp/EGG
cd /mnt/gluster/data/external_public_supp/EGG

#Analyze EGG-GWAS-BL.txt.gz and EGG_BW2_DISCOVERY.txt.gz together (similar sample sizes/time-points make it seem like likely to be derived from very similar datasets though 'total datasets used' is slightly diff and can also look at the papers directly to compare/confirm), as well as the various Pubertal growth phenotypes?
#Everything maybe leave alone since datasets begin seeming a bit disparate after that, e.g. varying total sample sizes therefore implying different underlying datasets used (even if maybe likely a subset of the larger sized studies? somewhat strong assumptiong to make?)

wget http://egg-consortium.org/downloads/EGG-GWAS-BL.txt.gz
wget http://egg-consortium.org/downloads/EGG_BW2_DISCOVERY.txt.gz
wget http://egg-consortium.org/downloads/EGG_Obesity_Meta_Analysis_1.txt.gz
wget http://egg-consortium.org/downloads/EGG_HC_DISCOVERY.txt.gz
wget http://egg-consortium.org/downloads/Pubertal_growth_10F.txt.gz
wget http://egg-consortium.org/downloads/Pubertal_growth_12M.txt.gz
wget http://egg-consortium.org/downloads/Pubertal_growth_10F_12M_combined.txt.gz
wget http://egg-consortium.org/downloads/Pubertal_growth_PGF.txt.gz
wget http://egg-consortium.org/downloads/Pubertal_growth_PGM.txt.gz
wget http://egg-consortium.org/downloads/Pubertal_growth_PGF_PGM_combined.txt.gz
wget http://egg-consortium.org/downloads/Pubertal_growth_PTF.txt.gz
wget http://egg-consortium.org/downloads/Pubertal_growth_PTM.txt.gz
wget http://egg-consortium.org/downloads/Pubertal_growth_PTF_PTM_combined.txt.gz
wget http://egg-consortium.org/downloads/EGG_TANNER_males.txt.gz
wget http://egg-consortium.org/downloads/EGG_TANNER_females.txt.gz
wget http://egg-consortium.org/downloads/EGG_TANNER_males_and_females_combined.txt.gz

~~~
[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/EGG]$for i in `ls -lrt | awk '{ print $9 }' | grep gz$`; do echo $i; zcat $i | head -n 5
; done
EGG_BW2_DISCOVERY.txt.gz
SNP EFFECT_ALLELE OTHER_ALLELE EFFECT SE PVAL
rs2326918 a g -2.5e-02 1.2e-02 3.6e-02
rs2439906 c g -1.7e-02 2.4e-02 4.7e-01
rs10760160 a c 1.7e-03 8.2e-03 8.4e-01
rs977590 a g -4.2e-03 1.7e-02 8.0e-01
EGG_Obesity_Meta_Analysis_1.txt.gz
SNP EFFECT_ALLELE OTHER_ALLELE EFFECT SE PVAL
rs12565286 c g -0.0673 0.1389 0.6281
rs3094315 a g -0.0042 0.0590 0.9432
rs3131968 a g 0.0049 0.0757 0.9484
rs2905035 a g 0.0138 0.0764 0.8566
EGG_HC_DISCOVERY.txt.gz
CHR     POS     SNP     EFFECT_ALLELE   NON_EFFECT_ALLELE       EFFECT_ALLELE_FREQ      BETA    SE      P_Value TotalSampleSize
1       711153  rs12565286      c       g       ,0758   -,1378  ,0574   ,01639  4376
1       713682  rs11804171      a       t       ,0757   -,1373  ,0572   ,01639  4376
1       713754  rs2977670       c       g       ,9242   ,1389   ,0568   ,01453  4376
1       742429  rs3094315       a       g       ,8241   ,0462   ,0251   ,06537  4731
EGG-GWAS-BL.txt.gz
CHR     BP      RSID    EA      NEA     BETA    SE      P       N
7       92221824        rs10    a       c       -0.0334 0.0306  0.274   18842
12      125456933       rs1000000       a       g       -0.0051 0.0115  0.6601  22142
4       21227772        rs10000010      t       c       -0.0068 0.0085  0.4238  28328
4       1347325 rs10000012      c       g       0.0369  0.0154  0.01649 21151
EGG_TANNER_males_and_females_combined.txt.gz
RSID    CHR     POS     EA      NEA     BETA    SE      P       N
rs10484380      1       740098  G       A       0.023293        0.022617        0.306296        9915
rs10484381      1       742429  G       A       -0.016051       0.029459        0.588398        9915
rs9368028       1       744055  T       G       0.013917        0.016673        0.407027        9911
rs1407496       1       758311  G       A       0.044193        0.025391        0.083836        9915
Pubertal_growth_PGM.txt.gz
RSID    CHR     POS     EA      NEA     BETA    SE      P       N
rs10    7       92221824        C       A       0.0119819003658907      0.0520387005735611      0.818264415460875       5043
rs1000000       12      125456933       G       A       -0.0206852892192123     0.0228785181735701      0.366924902456278       5040
rs10000010      4       21227772        C       T       -0.0274111402700431     0.0198324654524778      0.167840841384698       5036
rs10000012      4       1347325 C       G       0.0435422096811108      0.0299896538578145      0.147397763915379       5043
Pubertal_growth_10F_12M_combined.txt.gz
RSID    CHR     POS     EA      NEA     BETA    SE      P       N
rs10    7       92221824        C       A       0.00748506813439837     0.0337595374398832      0.828408809587153       13951
rs1000000       12      125456933       G       A       -0.00143087642675227    0.0142539714787381      0.921827254436292       13950
rs10000010      4       21227772        C       T       -0.000479587898553574   0.0121109227481301      0.969120639432438       13947
rs10000012      4       1347325 C       G       0.00435586788140434     0.0180932379728816      0.813941505431574       13960
Pubertal_growth_12M.txt.gz
RSID    CHR     POS     EA      NEA     BETA    SE      P       N
rs10    7       92221824        C       A       0.0464493452412418      0.0474312660874251      0.332245768657235       6977
rs1000000       12      125456933       G       A       -0.00726348244621842    0.0201525388678113      0.721191832336171       6983
rs10000010      4       21227772        C       T       0.00281621707900442     0.0171065505727344      0.870516021438723       6982
rs10000012      4       1347325 C       G       -0.00648376236077361    0.0254622437733485      0.800941741801866       6986
Pubertal_growth_10F.txt.gz
RSID    CHR     POS     EA      NEA     BETA    SE      P       N
rs10    7       92221824        C       A       -0.0325213185774554     0.0480613615707011      0.503971702431178       6974
rs1000000       12      125456933       G       A       0.00440824141920593     0.0201637854039662      0.82905944683406        6967
rs10000010      4       21227772        C       T       -0.00379153385231275    0.0171483884829866      0.827150985236544       6965
rs10000012      4       1347325 C       G       0.0154117993647623      0.0257150342470874      0.553927909347344       6974
Pubertal_growth_PTF_PTM_combined.txt.gz
RSID    CHR     POS     EA      NEA     BETA    SE      P       N
rs10    7       92221824        C       A       -0.119782207432755      0.0567145133956858      0.228198775910185       4282
rs1000000       12      125456933       G       A       -0.0138718556287804     0.0254230306023083      0.505976056011604       4282
rs10000010      4       21227772        C       T       -0.0277439974565636     0.0222636693000386      0.062486132297155       4276
rs10000012      4       1347325 C       G       0.0701096143634075      0.0336030210516203      0.47303059164521        4282
Pubertal_growth_PTM.txt.gz
RSID    CHR     POS     EA      NEA     BETA    SE      P       N
rs10    7       92221824        C       A       -0.119782207432755      0.0567145133956858      0.0349786977820197      4282
rs1000000       12      125456933       G       A       -0.0138718556287804     0.0254230306023083      0.585828510191988       4282
rs10000010      4       21227772        C       T       -0.0277439974565636     0.0222636693000386      0.213370488205856       4276
rs10000012      4       1347325 C       G       0.0701096143634075      0.0336030210516203      0.0372469324442491      4282
Pubertal_growth_PTF.txt.gz
RSID    CHR     POS     EA      NEA     BETA    SE      P       N
rs10    7       92221824        C       A       0.0189880419160795      0.0543612955584435      0.726970768043657       4946
rs1000000       12      125456933       G       A       0.0342814235294401      0.0239535659794243      0.152582968624631       4946
rs10000010      4       21227772        C       T       -0.0283633044970379     0.0203493020723371      0.163573522878361       4943
rs10000012      4       1347325 C       G       -0.0288718483585102     0.0308235496157442      0.349127808225971       4946
Pubertal_growth_PGF.txt.gz
RSID    CHR     POS     EA      NEA     BETA    SE      P       N
rs10    7       92221824        C       A       -0.0336646219356362     0.0485800208027845      0.49131513067265        5756
rs1000000       12      125456933       G       A       -0.00084974510854283    0.0212233200914873      0.968280946659699       5749
rs10000010      4       21227772        C       T       -0.00909563820800069    0.0180569114397476      0.616885145741547       5747
rs10000012      4       1347325 C       G       -0.00653465677303197    0.0274745422297611      0.813264536275865       5756
Pubertal_growth_PGF_PGM_combined.txt.gz
RSID    CHR     POS     EA      NEA     BETA    SE      P       N
rs10    7       92221824        C       A       -0.0124085679341998     0.0355110830689766      0.730355601933992       10799
rs1000000       12      125456933       G       A       -0.010024111124133      0.0155594364895892      0.525143090953062       10789
rs10000010      4       21227772        C       T       -0.0173969775489243     0.0133518478910542      0.198753422730644       10783
rs10000012      4       1347325 C       G       0.0163161865647863      0.0202583465890956      0.426968972292359       10799
EGG_TANNER_males.txt.gz
RSID    CHR     POS     EA      NEA     BETA    SE      P       N
rs10484380      1       740098  G       A       -0.039125       0.022617        0.322343        3769
rs10484381      1       742429  G       A       -0.001668       0.029459        0.97256 3769
rs9368028       1       744055  T       G       0.03172 0.016673        0.276221        3769
rs1407496       1       758311  G       A       0.002471        0.025391        0.955117        3769
EGG_TANNER_females.txt.gz
RSID    CHR     POS     EA      NEA     BETA    SE      P       N
rs10484380      1       740098  G       A       0.05367 0.022617        0.052603        6146
rs10484381      1       742429  G       A       -0.024457       0.029459        0.511288        6146
rs9368028       1       744055  T       G       0.005243        0.016673        0.797317        6142
rs1407496       1       758311  G       A       0.065157        0.025391        0.037084        6146
[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/EGG]$for i in `ls -lrt | awk '{ print $9 }' | grep gz$`; do zcat $i | wc | xargs echo $i; done
EGG_BW2_DISCOVERY.txt.gz 2684394 16106364 103928199
EGG_Obesity_Meta_Analysis_1.txt.gz 2442739 14656434 87229501
EGG_HC_DISCOVERY.txt.gz 2449807 24498070 142766969
EGG-GWAS-BL.txt.gz 2201972 19817612 120115669
EGG_TANNER_males_and_females_combined.txt.gz 2183229 19649061 126678946
Pubertal_growth_PGM.txt.gz 2391721 21525489 209000119
Pubertal_growth_10F_12M_combined.txt.gz 2479700 22317300 219796537
Pubertal_growth_12M.txt.gz 2500338 22503042 218776787
Pubertal_growth_10F.txt.gz 2393631 21542679 209286463
Pubertal_growth_PTF_PTM_combined.txt.gz 2401290 21611610 209716756
Pubertal_growth_PTM.txt.gz 2401290 21611610 209714794
Pubertal_growth_PTF.txt.gz 2400928 21608352 209755840
Pubertal_growth_PGF.txt.gz 2390840 21517560 208995047
Pubertal_growth_PGF_PGM_combined.txt.gz 2384832 21463488 211154357
EGG_TANNER_males.txt.gz 2183229 19649061 126682645
EGG_TANNER_females.txt.gz 2183229 19649061 126682242
~~~

#20150830 CHECK_0: EGG QC stuff

#python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs3.py --file1 <(zcat /mnt/lustre/home/mturchin20/Data/UCSC/GenomeBrowser/snp126.txt.gz | perl -lane 'if ($F[1] =~ m/(chr[0-9MXY]+)_.*/) { $F[1] = $1; } print join("\t", @F);') --file2 /mnt/gluster/data/external_public_supp/EGG/EGG_BW2_DISCOVERY.txt.gz | gzip > /mnt/gluster/data/external_public_supp/EGG/EGG_BW2_DISCOVERY.wUCSCGB_snp126.vs3.txt.gz &

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_12M.txt.gz | gzip > /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_12M.wHapMap21.txt.gz 
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_12M.txt.gz | gzip > /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_12M.wHapMap22.txt.gz 

#It is HapMap22, see the link between the reported ChrBP from HapMap files and the EGG files themselves; HapMap22 matches up and HapMap21 does not
~~~
[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/EGG]$zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_12M.wHapMap21.txt.gz | head -n 10
RSID    CHR     POS     EA      NEA     BETA    SE      P       N       ChrBPAFInfo
rs10    7       92221824        C       A       0.0464493452412418      0.0474312660874251      0.332245768657235       6977    chr7_92028539_A_0.033_C_0.967
rs1000000       12      125456933       G       A       -0.00726348244621842    0.0201525388678113      0.721191832336171       6983    chr12_125415860_C_0.627_T_0.373
rs10000010      4       21227772        C       T       0.00281621707900442     0.0171065505727344      0.870516021438723       6982    chr4_21294943_T_0.569_C_0.431
rs10000012      4       1347325 C       G       -0.00648376236077361    0.0254622437733485      0.800941741801866       6986    chr4_1347155_C_0.808_G_0.192
rs10000013      4       36901464        C       A       0.0186013590355979      0.0201550391398026      0.360833660900404       6986    chr4_37047635_C_0.167_A_0.833
rs10000017      4       84997149        C       T       0.00229841641089961     0.0205790840023647      0.911945690739412       6986    chr4_85135304_C_0.777_T_0.223
rs1000002       3       185118462       C       T       0.030436907570552       0.0170680348134758      0.0774843159455438      6986    chr3_185118470_G_0.475_A_0.525
rs10000023      4       95952929        G       T       -0.0261631374036616     0.0173535116520968      0.135526792410593       6975    chr4_96091084_G_0.408_T_0.592
rs1000003       3       99825597        G       A       0.0193292206024071      0.0229295108942842      0.40391822664983        6986    chr3_99825597_A_0.875_G_0.125
[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/EGG]$zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_12M.wHapMap22.txt.gz | head -n 10
RSID    CHR     POS     EA      NEA     BETA    SE      P       N       ChrBPAFInfo
rs10    7       92221824        C       A       0.0464493452412418      0.0474312660874251      0.332245768657235       6977    chr7_92221824_A_0.033_C_0.967
rs1000000       12      125456933       G       A       -0.00726348244621842    0.0201525388678113      0.721191832336171       6983    chr12_125456933_C_0.627_T_0.373
rs10000010      4       21227772        C       T       0.00281621707900442     0.0171065505727344      0.870516021438723       6982    chr4_21227772_T_0.575_C_0.425
rs10000012      4       1347325 C       G       -0.00648376236077361    0.0254622437733485      0.800941741801866       6986    chr4_1347325_C_0.808_G_0.192
rs10000013      4       36901464        C       A       0.0186013590355979      0.0201550391398026      0.360833660900404       6986    chr4_36901464_C_0.167_A_0.833
rs10000017      4       84997149        C       T       0.00229841641089961     0.0205790840023647      0.911945690739412       6986    chr4_84997149_C_0.777_T_0.223
rs1000002       3       185118462       C       T       0.030436907570552       0.0170680348134758      0.0774843159455438      6986    chr3_185118462_G_0.475_A_0.525
rs10000023      4       95952929        G       T       -0.0261631374036616     0.0173535116520968      0.135526792410593       6975    chr4_95952929_G_0.408_T_0.592
rs1000003       3       99825597        G       A       0.0193292206024071      0.0229295108942842      0.40391822664983        6986    chr3_99825597_A_0.875_G_0.125
~~~

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_10F.txt.gz | gzip > /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_10F.wHapMap22.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_10F_12M_combined.txt.gz | gzip > /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_10F_12M_combined.wHapMap22.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PGM.txt.gz | gzip > /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PGM.wHapMap22.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PGF.txt.gz | gzip > /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PGF.wHapMap22.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PGF_PGM_combined.txt.gz | gzip > /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PGF_PGM_combined.wHapMap22.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PTM.txt.gz | gzip > /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PTM.wHapMap22.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PTF.txt.gz | gzip > /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PTF.wHapMap22.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PTF_PTM_combined.txt.gz | gzip > /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PTF_PTM_combined.wHapMap22.txt.gz &

mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/EGG2013
mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/process.MTedits.ForICBP2011.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1/process.MTedits.ForEGG2013.Male.vs1.R
cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1/process.MTedits.ForEGG2013.Male.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1/process.MTedits.ForEGG2013.Female.vs1.R
cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1/process.MTedits.ForEGG2013.Male.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1/process.MTedits.ForEGG2013.Combined.vs1.R

#20150831 Ran script comment?

zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_12M.wHapMap22.txt.gz | perl -lane 'my @vals1 = split(/_/, $F[9]); push(@F, @vals1); print join("\t", @F);' | awk '{ print $2, "\t", $3, "\t", $1, "\t", $14, "\t", $8, "\t", $9 }' | gzip > /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_12M.wHapMap22.formatted.txt.gz & 
zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_10F.wHapMap22.txt.gz | perl -lane 'my @vals1 = split(/_/, $F[9]); push(@F, @vals1); print join("\t", @F);' | awk '{ print $2, "\t", $3, "\t", $1, "\t", $14, "\t", $8, "\t", $9 }' | gzip > /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_10F.wHapMap22.formatted.txt.gz &
zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_10F_12M_combined.wHapMap22.txt.gz | perl -lane 'my @vals1 = split(/_/, $F[9]); push(@F, @vals1); print join("\t", @F);' | awk '{ print $2, "\t", $3, "\t", $1, "\t", $14, "\t", $8, "\t", $9 }' | gzip > /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_10F_12M_combined.wHapMap22.formatted.txt.gz &
zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PGM.wHapMap22.txt.gz | perl -lane 'my @vals1 = split(/_/, $F[9]); push(@F, @vals1); print join("\t", @F);' | awk '{ print $2, "\t", $3, "\t", $1, "\t", $14, "\t", $8, "\t", $9 }' | gzip > /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PGM.wHapMap22.formatted.txt.gz &
zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PGF.wHapMap22.txt.gz | perl -lane 'my @vals1 = split(/_/, $F[9]); push(@F, @vals1); print join("\t", @F);' | awk '{ print $2, "\t", $3, "\t", $1, "\t", $14, "\t", $8, "\t", $9 }' | gzip > /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PGF.wHapMap22.formatted.txt.gz &
zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PGF_PGM_combined.wHapMap22.txt.gz | perl -lane 'my @vals1 = split(/_/, $F[9]); push(@F, @vals1); print join("\t", @F);' | awk '{ print $2, "\t", $3, "\t", $1, "\t", $14, "\t", $8, "\t", $9 }' | gzip > /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PGF_PGM_combined.wHapMap22.formatted.txt.gz &
zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PTM.wHapMap22.txt.gz | perl -lane 'my @vals1 = split(/_/, $F[9]); push(@F, @vals1); print join("\t", @F);' | awk '{ print $2, "\t", $3, "\t", $1, "\t", $14, "\t", $8, "\t", $9 }' | gzip > /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PTM.wHapMap22.formatted.txt.gz &
zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PTF.wHapMap22.txt.gz | perl -lane 'my @vals1 = split(/_/, $F[9]); push(@F, @vals1); print join("\t", @F);' | awk '{ print $2, "\t", $3, "\t", $1, "\t", $14, "\t", $8, "\t", $9 }' | gzip > /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PTF.wHapMap22.formatted.txt.gz &
zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PTF_PTM_combined.wHapMap22.txt.gz | perl -lane 'my @vals1 = split(/_/, $F[9]); push(@F, @vals1); print join("\t", @F);' | awk '{ print $2, "\t", $3, "\t", $1, "\t", $14, "\t", $8, "\t", $9 }' | gzip > /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PTF_PTM_combined.wHapMap22.formatted.txt.gz &
zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_10F.wHapMap22.txt.gz | perl -lane 'my @vals1 = split(/_/, $F[9]); push(@F, @vals1); print join("\t", @F);' | awk '{ print $2, "\t", $3, "\t", $1, "\t", $14, "\t", $8, "\t", $9 }' | gzip > /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_10F.wHapMap22.formatted.txt.gz &


##EGG2013 -- BW (& all studies?) 126dbSNP,  Pubertal Grwoth 10 GWAS, PG= ~14k, ~11k, ~9k
##20160225

#Correcting sign issue -- quick hack to get results + first full run through of TAG2010

#NOTE -- flipping sign on p-value to make coding downstream easier for this, ie I'm not implying something by creating negative p-values here. This is just for a quick run through.

#From http://egg-consortium.org/pubertal-growth.html
~~~
Analysis I:A single height measurement at age 10 in girls and 12 in boys that targets the take-off phase of the growth spurt.
Phenotype	Description	Maximum N	Downloadable file
10F	Height SDS for females at age 10.	6,974	Pubertal_growth_10F.txt.gz
12M	Height SDS for males at age 12.	6,986	Pubertal_growth_12M.txt.gz
10F + 12 M combined	Height SDS for females at age 10 and males at age 12 combined.	13,960	Pubertal_growth_10F_12M_combined.txt.gz
Analysis II:The total amount of growth across the pubertal growth period.
Phenotype	Description	Maximum N	Downloadable file
PGF	Standardized difference in height between age 8 and adult in females.	5,756	Pubertal_growth_PGF.txt.gz
PGM	Standardized difference in height between age 8 and adult in males.	5,043	Pubertal_growth_PGM.txt.gz
PGF + PGM combined	Standardized difference in height between age 8 and adult in males and females combined.	10,799	Pubertal_growth_PGF_PGM_combined.txt.gz
Analysis III:The total amount of growth in late adolescence, targeting the timing of peak height growth velocity.
Phenotype	Description	Maximum N	Downloadable file
PTF	Standardized difference in height between age 14 and adult in females.	4,946	Pubertal_growth_PTF.txt.gz
PTM	Standardized difference in height between age 14 and adult in males.	4,282	Pubertal_growth_PTM.txt.gz
PTF + PTM combined	Standardized difference in height between age 14 and adult in males and females combined.	9,228	Pubertal_growth_PTF_PTM_combined.txt.gz
~~~

#Female, Male, Combined
#Init, PubGrwth, LtAdlsc
#10/12
#PGF/M
#PTF/M

#/mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_10F.wHapMap22.txt.gz, /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PGF.wHapMap22.txt.gz, /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PTF.wHapMap22.txt.gz
#/mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_12M.wHapMap22.txt.gz, /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PGM.wHapMap22.txt.gz, /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PTM.wHapMap22.txt.gz
#/mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_10F_12M_combined.wHapMap22.txt.gz, /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PGF_PGM_combined.wHapMap22.txt.gz, /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PTF_PTM_combined.wHapMap22.txt.gz

#20160303 CHECK_0 -- Prob: Need to do a correct 'if HapMap allele matches dataset alleles' approach than what I'm doing here, in that I am only checking if reference alleles match up; if they do not then switch. Clearly need to check that dataset's ref allele matches HapMap ref allele /or/ alt allele (and if matches neither need to print warning out regarding this); also need to check 'strandedness' of dataset vs. hapmap alleles

zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_10F.wHapMap22.txt.gz | perl -lane 'print join("\t", @F[0..$#F-1]), "\t", join("\t", (split("_", $F[$#F])));' | perl -lane 'if ($F[3] ne $F[11]) { ($F[11], $F[13]) = ($F[13], $F[11]); ($F[12], $F[14]) = ($F[14], $F[12]); } print join("\t", @F);' | perl -lane 'if ($F[5] < 0) { ($F[3], $F[4]) = ($F[4], $F[3]); $F[12] = 1 - $F[12]; $F[5] = $F[5] * -1; } print join("\t", @F);' | sed 's/ChrBPAFInfo/Chr BP Ref RAF Alt AAF/g' | grep -v -w NA | gzip > /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_10F.wHapMap22.IncAllele.txt.gz 

LANG2=${LANG}
LC_ALL2=${LC_ALL}
LANG=C
LC_ALL=C
export LC_ALL LANG

join <(zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_10F.wHapMap22.IncAllele.txt.gz | awk '{ print $1, "\t", $4 }' | sort) <(zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PGF.wHapMap22.txt.gz | perl -lane 'print join("\t", @F[0..$#F-1]), "\t", join("\t", (split("_", $F[$#F])));' | perl -lane 'if ($F[3] ne $F[11]) { ($F[11], $F[13]) = ($F[13], $F[11]); ($F[12], $F[14]) = ($F[14], $F[12]); } print join("\t", @F);' | perl -lane 'if ($F[5] < 0) { ($F[3], $F[4]) = ($F[4], $F[3]); $F[12] = 1 - $F[12]; $F[5] = $F[5] * -1; } print join("\t", @F);' | sort ) | perl -lane 'if ($F[1] ne $F[4]) { $F[8] = $F[8] * -1; } splice(@F, 1, 1); print join("\t", @F);' | sed 's/ChrBPAFInfo/Chr BP Ref RAF Alt AAF/g' |  grep -v -w NA | gzip > /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PGF.wHapMap22.MatchedToInit.IncAllele.txt.gz
join <(zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_10F.wHapMap22.IncAllele.txt.gz | awk '{ print $1, "\t", $4 }' | sort) <(zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PTF.wHapMap22.txt.gz | perl -lane 'print join("\t", @F[0..$#F-1]), "\t", join("\t", (split("_", $F[$#F])));' | perl -lane 'if ($F[3] ne $F[11]) { ($F[11], $F[13]) = ($F[13], $F[11]); ($F[12], $F[14]) = ($F[14], $F[12]); } print join("\t", @F);' | perl -lane 'if ($F[5] < 0) { ($F[3], $F[4]) = ($F[4], $F[3]); $F[12] = 1 - $F[12]; $F[5] = $F[5] * -1; } print join("\t", @F);' | sort ) | perl -lane 'if ($F[1] ne $F[4]) { $F[8] = $F[8] * -1; } splice(@F, 1, 1); print join("\t", @F);' | sed 's/ChrBPAFInfo/Chr BP Ref RAF Alt AAF/g' |  grep -v -w NA | gzip > /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PTF.wHapMap22.MatchedToInit.IncAllele.txt.gz

LANG=${LANG2}
LC_ALL=${LC_ALL2}
export LC_ALL LANG

zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_10F.wHapMap22.IncAllele.txt.gz | awk '{ print $2, "\t", $3, "\t", $1, "\t", $13, "\t", $8, "\t", $9 }' | gzip > /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_10F.wHapMap22.IncAllele.formatted.txt.gz

zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PGF.wHapMap22.MatchedToInit.IncAllele.txt.gz | awk '{ print $2, "\t", $3, "\t", $1, "\t", $13, "\t", $8, "\t", $9 }' | gzip > /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PGF.wHapMap22.MatchedToInit.IncAllele.formatted.txt.gz
zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PTF.wHapMap22.MatchedToInit.IncAllele.txt.gz | awk '{ print $2, "\t", $3, "\t", $1, "\t", $13, "\t", $8, "\t", $9 }' | gzip > /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PTF.wHapMap22.MatchedToInit.IncAllele.formatted.txt.gz

LANG2=${LANG}
LC_ALL2=${LC_ALL}
LANG=C
LC_ALL=C
export LC_ALL LANG

join -a 1 -a 2 -1 3 -2 3 -e NA2 -o 0 1.1 1.2 1.4 1.5 1.6 2.1 2.2 2.4 2.5 2.6 <(zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_10F.wHapMap22.IncAllele.formatted.txt.gz | sort -g -k 3,3) <(zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PGF.wHapMap22.MatchedToInit.IncAllele.formatted.txt.gz | sort -g -k 3,3 ) | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
join -a 1 -a 2 -1 1 -2 3 -e NA2 -o 0 1.2 2.1 2.2 2.4 2.5 2.6 - <(zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PTF.wHapMap22.MatchedToInit.IncAllele.formatted.txt.gz | sort -g -k 3,3) | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); print join("\t", @F);' | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); if ($#F == 6) { splice(@F, 1, 0, (("NA2") x 9)); } my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
perl -lane 'print $F[0], "\t", join("\t", split(/_/, $F[1]));' | gzip > /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_AllPhenosF.wHapMap22.MatchedToInit.IncAllele.formatted.RAF.txt.pre1.gz

LANG=${LANG2}
LC_ALL=${LC_ALL2}
export LC_ALL LANG

~~~
[  mturchin20@bigmem01  ~/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1]$zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_AllPhenosF.wHapMap22.MatchedToInit.IncAllele.formatted.RAF.txt.pre1.gz | wc
2668730 42699680 337613537
[  mturchin20@bigmem01  ~/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1]$zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_AllPhenosF.wHapMap22.MatchedToInit.IncAllele.formatted.RAF.txt.pre1.gz | head -n 10
rs1172982  1    100002699       0.3     0.111485796472115       6974    1       100002699       0.3     0.127540707993535 5756  1       100002699 0.3   0.83225463752849        4946
rs1172981  1    100002785       0.067   0.724019715759309       6974    1       100002785       0.933   -0.655943984623986      5756    1       100002785       0.067   0.947916968503234       4946
rs11166327 1    100003455       0.953   0.77213303216568        6974    1       100003455       0.047   -0.4289219779813  5756  1       100003455 0.953 0.957367290255097       4946
rs4908018  1    100006955       0.933   0.885559813175248       6968    1       100006955       0.0669999999999999      -0.772031761489485      5750    1       100006955       0.0669999999999999      -0.951160629922146      4946
rs2392072  1    100007331       0.717   0.180631598755801       6972    1       100007331       0.717   0.0856332413322133      5756    1       100007331       0.717   0.631732162951484       4946
rs12129588 1    100007519       0.937   0.97770114960363        6974    1       100007519       0.063   -0.398867796731709      5756    1       100007519       0.063   -0.85441805810729       4946
rs10875256 1    100007856       0.942   0.825018975865846       6974    1       100007856       0.058   -0.6791783105684  5756  1       100007856 0.058 -0.954434927347006      4946
rs7541580  1    100008014       0.783   0.173661742400052       6974    1       100008014       0.783   0.0201957856190784      5756    1       100008014       0.783   0.610843129288907       4946
rs883532   1    100008025       0.857   0.172868503768936       6974    1       100008025       0.857   0.0205608337214704      5756    1       100008025       0.857   0.612633800311247       4946
rs883533   1    100008439       0.933   0.82511996178153        6974    1       100008439       0.0669999999999999      -0.679481301360411      5756    1       100008439       0.0669999999999999      -0.953690575479536      4946
[  mturchin20@bigmem01  ~/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1]$zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_AllPhenosF.wHapMap22.MatchedToInit.IncAllele.formatted.RAF.txt.pre1.gz | perl -lane 'print $#F;' | sort | uniq -c
2668730 15
~~~

zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_AllPhenosF.wHapMap22.MatchedToInit.IncAllele.formatted.RAF.txt.pre1.gz | grep -v NA2 | gzip > /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_AllPhenosF.wHapMap22.MatchedToInit.IncAllele.formatted.RAF.txt.pre2.gz

zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_AllPhenosF.wHapMap22.MatchedToInit.IncAllele.formatted.RAF.txt.pre2.gz | perl -lane 'splice(@F, 21, 3); splice(@F, 16, 3); splice(@F, 11, 3); splice(@F, 6, 3); print join("\t", @F);' | gzip > /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_AllPhenosF.wHapMap22.MatchedToInit.IncAllele.formatted.RAF.txt.gz

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/EGG2013/EGG2013.Table1.ChrBP.txt --file2 /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_AllPhenosF.wHapMap22.MatchedToInit.IncAllele.formatted.RAF.txt.gz | gzip > /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_AllPhenosF.wHapMap22.MatchedToInit.IncAllele.formatted.RAF.wGWASannot.txt.gz

~~~
[  mturchin20@bigmem01  ~/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1]$zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_AllPhenosF.wHapMap22.MatchedToInit.IncAllele.formatted.RAF.wGWASannot.txt.gz | perl -lane 'print $F[$#F];' | sort | uniq -c
[  mturchin20@bigmem01  ~/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1]$zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_AllPhenosF.wHapMap22.MatchedToInit.IncAllele.formatted.RAF.wGWASannot.txt.gz | perl -lane 'print $F[$#F];' | sort | uniq -c
2098451 0
      9 1
   6760 2
~~~

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1/process.MTedits.ForEGG2013.Female.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1/process.MTedits.ForEGG2013.Female.vs1.SignCrrct.vs1.R

LANG2=${LANG}
LC_ALL2=${LC_ALL}
LANG=C
LC_ALL=C
export LC_ALL LANG

join -a 2 -1 1 -2 1 -e NA4 -o 0 1.2 1.3 1.4 1.5 1.6 1.7 1.8 1.9 1.10 1.11 2.2 2.3 2.4 2.5 2.6 2.7 <(zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_AllPhenosF.wHapMap22.MatchedToInit.IncAllele.formatted.RAF.wGWASannot.txt.gz | sort -g -k 1,1 ) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1/EGG2013Female.dtlesslesssignif.vs1.SignCrrct.vs1.txt | perl -F, -lane 'print join("\t", @F);' | sort -g -k 1,1) | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1/EGG2013Female.dtlesslesssignif.vs1.SignCrrct.vs1.annot.vs1.RAF.txt.pre1.gz

LANG=${LANG2}
LC_ALL=${LC_ALL2}
export LC_ALL LANG

zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1/EGG2013Female.dtlesslesssignif.vs1.SignCrrct.vs1.annot.vs1.RAF.txt.pre1.gz | grep -v NA4 | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1/EGG2013Female.dtlesslesssignif.vs1.SignCrrct.vs1.annot.vs1.RAF.txt.gz

#Added the following header to /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1/EGG2013Female.dtlesslesssignif.vs1.SignCrrct.vs1.annot.vs1.RAF.txt.gz
#snp chr pos raf p_Init n_Init p_PubGrwth n_PubGrwth p_LtAdlsc n_LtAdlsc annot Z.Init Z.PubGrwth Z.LtAdlsc mvstat mvp unip

zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1/EGG2013Female.dtlesslesssignif.vs1.SignCrrct.vs1.annot.vs1.RAF.txt.gz | perl -lane 'if ($F[3] > .5) { $F[3] = 1 - $F[3]; $F[11] = $F[11] * -1; $F[12] = $F[12] * -1; $F[13] = $F[13] * -1; } print join("\t", @F);' | sed 's/raf/maf/g' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1/EGG2013Female.dtlesslesssignif.vs1.SignCrrct.vs1.annot.vs1.MAF.txt.gz

#cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/MAGIC2010/Vs1/GLC.MTedits.ForMAGIC2010.vs1.SignCrrct.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1/GLC.MTedits.ForEGG2013.Female.vs1.SignCrrct.vs1.R
p -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GLC.MTedits.ForGIANT2014_5.Orig3.vs1.SignCrrct.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1/GLC.MTedits.ForEGG2013.Female.vs1.SignCrrct.vs1.R


#Male

#/mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_10F.wHapMap22.txt.gz, /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PGF.wHapMap22.txt.gz, /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PTF.wHapMap22.txt.gz
#/mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_12M.wHapMap22.txt.gz, /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PGM.wHapMap22.txt.gz, /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PTM.wHapMap22.txt.gz
#/mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_10F_12M_combined.wHapMap22.txt.gz, /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PGF_PGM_combined.wHapMap22.txt.gz, /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PTF_PTM_combined.wHapMap22.txt.gz


#zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_12M.wHapMap22.txt.gz | perl -lane 'print join("\t", @F[0..$#F-1]), "\t", join("\t", (split("_", $F[$#F])));' | perl -lane 'if ($F[3] ne $F[11]) { ($F[11], $F[13]) = ($F[13], $F[11]); ($F[12], $F[14]) = ($F[14], $F[12]); } print join("\t", @F);' | perl -lane 'if ($F[5] < 0) { ($F[3], $F[4]) = ($F[4], $F[3]); $F[12] = 1 - $F[12]; $F[5] = $F[5] * -1; } print join("\t", @F);' | sed 's/ChrBPAFInfo/Chr BP Ref RAF Alt AAF/g' | grep -v -w NA | gzip > /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_12M.wHapMap22.IncAllele.txt.gz 
zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_12M.wHapMap22.txt.gz | perl -lane 'print join("\t", @F[0..$#F-1]), "\t", join("\t", (split("_", $F[$#F])));' | perl -lane 'if ($F[3] ne $F[11]) { ($F[11], $F[13]) = ($F[13], $F[11]); ($F[12], $F[14]) = ($F[14], $F[12]); } print join("\t", @F);' | perl -lane 'if ($F[5] < 0) { ($F[3], $F[4]) = ($F[4], $F[3]); $F[12] = 1 - $F[12]; $F[5] = $F[5] * -1; } print join("\t", @F);' | sed 's/ChrBPAFInfo/Chr BP Ref RAF Alt AAF/g' | grep -v -w NA | sort | uniq | gzip > /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_12M.wHapMap22.IncAllele.txt.gz 

#Don't know why but there are duplicates in /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_12M.wHapMap22.txt.gz and it is messing up the join functions below. Including a 'uniq' command in the usual file-prep pipeline to take care of this?
~~~
[  mturchin20@bigmem01  ~/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1]$zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_10F.wHapMap22.txt.gz |  awk '{ print $1 }' | sort | uniq -c | awk '{ if ($1 > 1) { print $0 } } ' | wc
      0       0       0
[  mturchin20@bigmem01  ~/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1]$zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_12M.wHapMap22.txt.gz |  awk '{ print $1 }' | sort | uniq -c | awk '{ if ($1 > 1) { print $0 } } ' | wc
 170191  340382 3146892
[  mturchin20@spudling12  ~/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1]$zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_12M.wHapMap22.txt.gz | sort | uniq | awk '{ print $1 }' | sort | uniq -c | awk '{ if ($1 > 1) { print $0 } } ' | wc
      0       0       0
[  mturchin20@bigmem01  ~/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1]$zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_12M.wHapMap22.txt.gz | wc
2500338 25003380 294536356
[  mturchin20@bigmem01  ~/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1]$zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PGM.wHapMap22.txt.gz |  awk '{ print $1 }' | sort | uniq -c | awk '{ if ($1 > 1) { print $0 } } ' | wc
      0       0       0
[  mturchin20@bigmem01  ~/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1]$zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PTM.wHapMap22.txt.gz |  awk '{ print $1 }' | sort | uniq -c | awk '{ if ($1 > 1) { print $0 } } ' | wc
      0       0       0
[  mturchin20@bigmem01  ~/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1]$zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_12M.wHapMap22.IncAllele.txt.gz |  awk '{ print $1 }' | sort | uniq -c | awk '{ if ($1 > 1) { print $0 } } ' | wc
      0       0       0
~~~


LANG2=${LANG}
LC_ALL2=${LC_ALL}
LANG=C
LC_ALL=C
export LC_ALL LANG

join <(zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_12M.wHapMap22.IncAllele.txt.gz | awk '{ print $1, "\t", $4 }' | sort) <(zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PGM.wHapMap22.txt.gz | perl -lane 'print join("\t", @F[0..$#F-1]), "\t", join("\t", (split("_", $F[$#F])));' | perl -lane 'if ($F[3] ne $F[11]) { ($F[11], $F[13]) = ($F[13], $F[11]); ($F[12], $F[14]) = ($F[14], $F[12]); } print join("\t", @F);' | perl -lane 'if ($F[5] < 0) { ($F[3], $F[4]) = ($F[4], $F[3]); $F[12] = 1 - $F[12]; $F[5] = $F[5] * -1; } print join("\t", @F);' | sort ) | perl -lane 'if ($F[1] ne $F[4]) { $F[8] = $F[8] * -1; } splice(@F, 1, 1); print join("\t", @F);' | sed 's/ChrBPAFInfo/Chr BP Ref RAF Alt AAF/g' |  grep -v -w NA | gzip > /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PGM.wHapMap22.MatchedToInit.IncAllele.txt.gz
join <(zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_12M.wHapMap22.IncAllele.txt.gz | awk '{ print $1, "\t", $4 }' | sort) <(zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PTM.wHapMap22.txt.gz | perl -lane 'print join("\t", @F[0..$#F-1]), "\t", join("\t", (split("_", $F[$#F])));' | perl -lane 'if ($F[3] ne $F[11]) { ($F[11], $F[13]) = ($F[13], $F[11]); ($F[12], $F[14]) = ($F[14], $F[12]); } print join("\t", @F);' | perl -lane 'if ($F[5] < 0) { ($F[3], $F[4]) = ($F[4], $F[3]); $F[12] = 1 - $F[12]; $F[5] = $F[5] * -1; } print join("\t", @F);' | sort ) | perl -lane 'if ($F[1] ne $F[4]) { $F[8] = $F[8] * -1; } splice(@F, 1, 1); print join("\t", @F);' | sed 's/ChrBPAFInfo/Chr BP Ref RAF Alt AAF/g' |  grep -v -w NA | gzip > /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PTM.wHapMap22.MatchedToInit.IncAllele.txt.gz

LANG=${LANG2}
LC_ALL=${LC_ALL2}
export LC_ALL LANG

zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_12M.wHapMap22.IncAllele.txt.gz | awk '{ print $2, "\t", $3, "\t", $1, "\t", $13, "\t", $8, "\t", $9 }' | gzip > /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_12M.wHapMap22.IncAllele.formatted.txt.gz

zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PGM.wHapMap22.MatchedToInit.IncAllele.txt.gz | awk '{ print $2, "\t", $3, "\t", $1, "\t", $13, "\t", $8, "\t", $9 }' | gzip > /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PGM.wHapMap22.MatchedToInit.IncAllele.formatted.txt.gz
zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PTM.wHapMap22.MatchedToInit.IncAllele.txt.gz | awk '{ print $2, "\t", $3, "\t", $1, "\t", $13, "\t", $8, "\t", $9 }' | gzip > /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PTM.wHapMap22.MatchedToInit.IncAllele.formatted.txt.gz

LANG2=${LANG}
LC_ALL2=${LC_ALL}
LANG=C
LC_ALL=C
export LC_ALL LANG

join -a 1 -a 2 -1 3 -2 3 -e NA2 -o 0 1.1 1.2 1.4 1.5 1.6 2.1 2.2 2.4 2.5 2.6 <(zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_12M.wHapMap22.IncAllele.formatted.txt.gz | sort -g -k 3,3) <(zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PGM.wHapMap22.MatchedToInit.IncAllele.formatted.txt.gz | sort -g -k 3,3 ) | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
join -a 1 -a 2 -1 1 -2 3 -e NA2 -o 0 1.2 2.1 2.2 2.4 2.5 2.6 - <(zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PTM.wHapMap22.MatchedToInit.IncAllele.formatted.txt.gz | sort -g -k 3,3) | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); print join("\t", @F);' | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); if ($#F == 6) { splice(@F, 1, 0, (("NA2") x 9)); } my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
perl -lane 'print $F[0], "\t", join("\t", split(/_/, $F[1]));' | gzip > /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_AllPhenosM.wHapMap22.MatchedToInit.IncAllele.formatted.RAF.txt.pre1.gz

LANG=${LANG2}
LC_ALL=${LC_ALL2}
export LC_ALL LANG

~~~
[  mturchin20@bigmem01  ~]$zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_AllPhenosM.wHapMap22.MatchedToInit.IncAllele.formatted.RAF.txt.pre1.gz | wc
2412663 38602608 315720084
[  mturchin20@bigmem01  ~]$zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_AllPhenosM.wHapMap22.MatchedToInit.IncAllele.formatted.RAF.txt.pre1.gz | head -n 10
rs1172982  1    100002699       0.7     0.748821147764486       6983    1       100002699       0.3     -0.549175066249628      5043    1       100002699       0.7     0.448938322526613       4282
rs1172981  1    100002785       0.933   0.747315220240481       6986    1       100002785       0.933   0.252418126095726 5043  1       100002785 0.933 0.598293959387293       4282
rs11166327 1    100003455       0.047   0.605916439554564       6984    1       100003455       0.047   0.435496080767843 5043  1       100003455 0.047 0.777654075332975       4282
rs4908018  1    100006955       0.0669999999999999      0.773163558756063       6984    1       100006955 0.0669999999999999    0.240631422500086 5040  1       100006955       0.0669999999999999      0.677583466060296       4282
rs2392072  1    100007331       0.283   0.686725290930294       6984    1       100007331       0.717   -0.531312742788196      5043    1       100007331       0.283   0.546887491002122       4282
rs12129588 1    100007519       0.063   0.633144017235698       6985    1       100007519       0.063   0.578388672123991 5043  1       100007519 0.063 0.756425228123899       4282
rs10875256 1    100007856       0.058   0.784311591436813       6986    1       100007856       0.058   0.243288015331042 5043  1       100007856 0.058 0.67580340322148        4282
rs7541580  1    100008014       0.217   0.663975858602998       6986    1       100008014       0.783   -0.281732781756187      5043    1       100008014       0.217   0.647752602036098       4282
rs883532   1    100008025       0.143   0.678959121488804       6986    1       100008025       0.857   -0.275650839996547      5043    1       100008025       0.143   0.656357539969864       4282
rs883533   1    100008439       0.0669999999999999      0.783296032799667       6986    1       100008439 0.0669999999999999    0.243097927421183 5043  1       100008439       0.0669999999999999      0.673601417574282       4282
[  mturchin20@bigmem01  ~]$zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_AllPhenosM.wHapMap22.MatchedToInit.IncAllele.formatted.RAF.txt.pre1.gz | perl -lane 'print $#F;' | sort | uniq -c
2412663 15
~~~

zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_AllPhenosM.wHapMap22.MatchedToInit.IncAllele.formatted.RAF.txt.pre1.gz | grep -v NA2 | gzip > /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_AllPhenosM.wHapMap22.MatchedToInit.IncAllele.formatted.RAF.txt.pre2.gz

zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_AllPhenosM.wHapMap22.MatchedToInit.IncAllele.formatted.RAF.txt.pre2.gz | perl -lane 'splice(@F, 21, 3); splice(@F, 16, 3); splice(@F, 11, 3); splice(@F, 6, 3); print join("\t", @F);' | gzip > /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_AllPhenosM.wHapMap22.MatchedToInit.IncAllele.formatted.RAF.txt.gz

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/EGG2013/EGG2013.Table1.ChrBP.txt --file2 /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_AllPhenosM.wHapMap22.MatchedToInit.IncAllele.formatted.RAF.txt.gz | gzip > /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_AllPhenosM.wHapMap22.MatchedToInit.IncAllele.formatted.RAF.wGWASannot.txt.gz

~~~
[  mturchin20@bigmem01  ~]$zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_AllPhenosM.wHapMap22.MatchedToInit.IncAllele.formatted.RAF.wGWASannot.txt.gz | perl -lane 'print $F[$#F];' | sort | uniq -c
2235481 0
      9 1
   7145 2
~~~

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1/process.MTedits.ForEGG2013.Female.vs1.SignCrrct.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1/process.MTedits.ForEGG2013.Male.vs1.SignCrrct.vs1.R

LANG2=${LANG}
LC_ALL2=${LC_ALL}
LANG=C
LC_ALL=C
export LC_ALL LANG

join -a 2 -1 1 -2 1 -e NA4 -o 0 1.2 1.3 1.4 1.5 1.6 1.7 1.8 1.9 1.10 1.11 2.2 2.3 2.4 2.5 2.6 2.7 <(zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_AllPhenosM.wHapMap22.MatchedToInit.IncAllele.formatted.RAF.wGWASannot.txt.gz | sort -g -k 1,1 ) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1/EGG2013Male.dtlesslesssignif.vs1.SignCrrct.vs1.txt | perl -F, -lane 'print join("\t", @F);' | sort -g -k 1,1) | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1/EGG2013Male.dtlesslesssignif.vs1.SignCrrct.vs1.annot.vs1.RAF.txt.pre1.gz

LANG=${LANG2}
LC_ALL=${LC_ALL2}
export LC_ALL LANG

zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1/EGG2013Male.dtlesslesssignif.vs1.SignCrrct.vs1.annot.vs1.RAF.txt.pre1.gz | grep -v NA4 | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1/EGG2013Male.dtlesslesssignif.vs1.SignCrrct.vs1.annot.vs1.RAF.txt.gz

#Added the following header to /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1/EGG2013Male.dtlesslesssignif.vs1.SignCrrct.vs1.annot.vs1.RAF.txt.gz
#snp chr pos raf p_Init n_Init p_PubGrwth n_PubGrwth p_LtAdlsc n_LtAdlsc annot Z.Init Z.PubGrwth Z.LtAdlsc mvstat mvp unip

zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1/EGG2013Male.dtlesslesssignif.vs1.SignCrrct.vs1.annot.vs1.RAF.txt.gz | perl -lane 'if ($F[3] > .5) { $F[3] = 1 - $F[3]; $F[11] = $F[11] * -1; $F[12] = $F[12] * -1; $F[13] = $F[13] * -1; } print join("\t", @F);' | sed 's/raf/maf/g' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1/EGG2013Male.dtlesslesssignif.vs1.SignCrrct.vs1.annot.vs1.MAF.txt.gz

#cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/MAGIC2010/Vs1/GLC.MTedits.ForMAGIC2010.vs1.SignCrrct.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1/GLC.MTedits.ForEGG2013.Female.vs1.SignCrrct.vs1.R
#cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GLC.MTedits.ForGIANT2014_5.Orig3.vs1.SignCrrct.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1/GLC.MTedits.ForEGG2013.Female.vs1.SignCrrct.vs1.R
cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1/GLC.MTedits.ForEGG2013.Female.vs1.SignCrrct.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1/GLC.MTedits.ForEGG2013.Male.vs1.SignCrrct.vs1.R


#Combined

#/mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_10F.wHapMap22.txt.gz, /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PGF.wHapMap22.txt.gz, /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PTF.wHapMap22.txt.gz
#/mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_12M.wHapMap22.txt.gz, /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PGM.wHapMap22.txt.gz, /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PTM.wHapMap22.txt.gz
#/mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_10F_12M_combined.wHapMap22.txt.gz, /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PGF_PGM_combined.wHapMap22.txt.gz, /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PTF_PTM_combined.wHapMap22.txt.gz


#zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_10F_12M_combined.wHapMap22.txt.gz | perl -lane 'print join("\t", @F[0..$#F-1]), "\t", join("\t", (split("_", $F[$#F])));' | perl -lane 'if ($F[3] ne $F[11]) { ($F[11], $F[13]) = ($F[13], $F[11]); ($F[12], $F[14]) = ($F[14], $F[12]); } print join("\t", @F);' | perl -lane 'if ($F[5] < 0) { ($F[3], $F[4]) = ($F[4], $F[3]); $F[12] = 1 - $F[12]; $F[5] = $F[5] * -1; } print join("\t", @F);' | sed 's/ChrBPAFInfo/Chr BP Ref RAF Alt AAF/g' | grep -v -w NA | gzip > /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_10F_12M_combined.wHapMap22.IncAllele.txt.gz 
zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_10F_12M_combined.wHapMap22.txt.gz | perl -lane 'print join("\t", @F[0..$#F-1]), "\t", join("\t", (split("_", $F[$#F])));' | perl -lane 'if ($F[3] ne $F[11]) { ($F[11], $F[13]) = ($F[13], $F[11]); ($F[12], $F[14]) = ($F[14], $F[12]); } print join("\t", @F);' | perl -lane 'if ($F[5] < 0) { ($F[3], $F[4]) = ($F[4], $F[3]); $F[12] = 1 - $F[12]; $F[5] = $F[5] * -1; } print join("\t", @F);' | sed 's/ChrBPAFInfo/Chr BP Ref RAF Alt AAF/g' | grep -v -w NA | sort | uniq | grep -v CHR | cat <(echo "RSID    CHR     POS     EA      NEA     BETA    SE      P       N       Chr BP Ref RAF Alt AAF") - | gzip > /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_10F_12M_combined.wHapMap22.IncAllele.txt.gz 



#This has the weird duplicate issue too in the 10F_12M file
~~~
[  mturchin20@bigmem01  ~/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1]$zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_10F_12M_combined.wHapMap22.txt.gz |  awk '{ print $1 }' | sort | uniq -c | awk '{ if ($1 > 1) { print $0 } } ' | wc
 168872  337744 3122335
[  mturchin20@bigmem01  ~/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1]$zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PGF_PGM_combined.wHapMap22.txt.gz |  awk '{ print $1 }' | sort | uniq -c | awk '{ if ($1 > 1) { print $0 } } ' | wc
      0       0       0
[  mturchin20@bigmem01  ~/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1]$zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PTF_PTM_combined.wHapMap22.txt.gz |  awk '{ print $1 }' | sort | uniq -c | awk '{ if ($1 > 1) { print $0 } } ' | wc
      0       0       0
[  mturchin20@bigmem01  ~/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1]$zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_10F_12M_combined.wHapMap22.txt.gz | sort | uniq | awk '{ print $1 }' | sort | uniq -c | awk '{ if ($1 > 1) { print $0 } } ' | wc
      0       0       0
[  mturchin20@bigmem01  ~/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1]$zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_10F_12M_combined.wHapMap22.IncAllele.txt.gz |  awk '{ print $1 }' | sort | uniq -c | awk '{ if ($1 > 1) { print $0 } } ' | wc
      0       0       0
~~~


LANG2=${LANG}
LC_ALL2=${LC_ALL}
LANG=C
LC_ALL=C
export LC_ALL LANG

join <(zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_10F_12M_combined.wHapMap22.IncAllele.txt.gz | awk '{ print $1, "\t", $4 }' | sort) <(zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PGF_PGM_combined.wHapMap22.txt.gz | perl -lane 'print join("\t", @F[0..$#F-1]), "\t", join("\t", (split("_", $F[$#F])));' | perl -lane 'if ($F[3] ne $F[11]) { ($F[11], $F[13]) = ($F[13], $F[11]); ($F[12], $F[14]) = ($F[14], $F[12]); } print join("\t", @F);' | perl -lane 'if ($F[5] < 0) { ($F[3], $F[4]) = ($F[4], $F[3]); $F[12] = 1 - $F[12]; $F[5] = $F[5] * -1; } print join("\t", @F);' | sort ) | perl -lane 'if ($F[1] ne $F[4]) { $F[8] = $F[8] * -1; } splice(@F, 1, 1); print join("\t", @F);' | sed 's/ChrBPAFInfo/Chr BP Ref RAF Alt AAF/g' |  grep -v -w NA | gzip > /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PGF_PGM_combined.wHapMap22.MatchedToInit.IncAllele.txt.gz
join <(zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_10F_12M_combined.wHapMap22.IncAllele.txt.gz | awk '{ print $1, "\t", $4 }' | sort) <(zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PTF_PTM_combined.wHapMap22.txt.gz | perl -lane 'print join("\t", @F[0..$#F-1]), "\t", join("\t", (split("_", $F[$#F])));' | perl -lane 'if ($F[3] ne $F[11]) { ($F[11], $F[13]) = ($F[13], $F[11]); ($F[12], $F[14]) = ($F[14], $F[12]); } print join("\t", @F);' | perl -lane 'if ($F[5] < 0) { ($F[3], $F[4]) = ($F[4], $F[3]); $F[12] = 1 - $F[12]; $F[5] = $F[5] * -1; } print join("\t", @F);' | sort ) | perl -lane 'if ($F[1] ne $F[4]) { $F[8] = $F[8] * -1; } splice(@F, 1, 1); print join("\t", @F);' | sed 's/ChrBPAFInfo/Chr BP Ref RAF Alt AAF/g' |  grep -v -w NA | gzip > /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PTF_PTM_combined.wHapMap22.MatchedToInit.IncAllele.txt.gz

LANG=${LANG2}
LC_ALL=${LC_ALL2}
export LC_ALL LANG

#Below file lses the header for some reason
zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_10F_12M_combined.wHapMap22.IncAllele.txt.gz | awk '{ print $2, "\t", $3, "\t", $1, "\t", $13, "\t", $8, "\t", $9 }' | gzip > /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_10F_12M_combined.wHapMap22.IncAllele.formatted.txt.gz
#zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_10F_12M_combined.wHapMap22.IncAllele.txt.gz | awk '{ print $2, "\t", $3, "\t", $1, "\t", $13, "\t", $8, "\t", $9 }' | grep -v CHR | cat <(echo "CHR      POS     RSID    RAF     P       N") - | gzip > /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_10F_12M_combined.wHapMap22.IncAllele.formatted.txt.gz

#~~~
#[  mturchin20@spudling12  ~/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1]$zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_10F_12M_combined.wHapMap22.IncAllele.formatted.txt.gz | head -n 10
#12       125456933       rs1000000       0.627   0.921827254436292       13950
#4        21227772        rs10000010      0.575   0.969120639432438       13947
#4        1347325         rs10000012      0.808   0.813941505431574       13960
#4        36901464        rs10000013      0.167   0.0342922524427472      13960
#4        84997149        rs10000017      0.223   0.113768777157663       13960
#3        185118462       rs1000002       0.525   0.1417885968771         13959
#4        95952929        rs10000023      0.592   0.533743038349605       13940
#4        103593179       rs10000030      0.125   0.389700571919017       13931
#4        139819348       rs10000033      0.492   0.89484102468269        13960
#3        99825597        rs1000003       0.125   0.796667663154212       13960
#[  mturchin20@spudling12  ~/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1]$zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PGF_PGM_combined.wHapMap22.MatchedToInit.IncAllele.formatted.txt.gz | head -n 10
#CHR      POS     RSID    RAF     P       N
#7        92221824        rs10    0.033   -0.730355601933992      10799
#12       125456933       rs1000000       0.627   0.525143090953062       10789
#4        21227772        rs10000010      0.575   0.198753422730644       10783
#4        1347325         rs10000012      0.808   0.426968972292359       10799
#4        36901464        rs10000013      0.833   -0.978498245816837      10799
#4        84997149        rs10000017      0.777   -0.469478265820576      10799
#3        185118462       rs1000002       0.525   0.874470525938863       10799
#4        95952929        rs10000023      0.592   0.210503763507433       10782
#3        99825597        rs1000003       0.875   -0.716320272277536      10799
#[  mturchin20@spudling12  ~/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1]$zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_10F_12M_combined.wHapMap22.IncAllele.formatted.txt.gz | grep CHR
#CHR      POS     RSID    RAF     P       N
#[  mturchin20@spudling12  ~/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1]$zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_10F_12M_combined.wHapMap22.IncAllele.formatted.txt.gz | head -n 10
#CHR      POS     RSID    RAF     P       N
#12       125456933       rs1000000       0.627   0.921827254436292       13950
#4        21227772        rs10000010      0.575   0.969120639432438       13947
#4        1347325         rs10000012      0.808   0.813941505431574       13960
#4        36901464        rs10000013      0.167   0.0342922524427472      13960
#4        84997149        rs10000017      0.223   0.113768777157663       13960
#3        185118462       rs1000002       0.525   0.1417885968771         13959
#4        95952929        rs10000023      0.592   0.533743038349605       13940
#4        103593179       rs10000030      0.125   0.389700571919017       13931
#4        139819348       rs10000033      0.492   0.89484102468269        13960
#~~~

zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PGF_PGM_combined.wHapMap22.MatchedToInit.IncAllele.txt.gz | awk '{ print $2, "\t", $3, "\t", $1, "\t", $13, "\t", $8, "\t", $9 }' | gzip > /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PGF_PGM_combined.wHapMap22.MatchedToInit.IncAllele.formatted.txt.gz
zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PTF_PTM_combined.wHapMap22.MatchedToInit.IncAllele.txt.gz | awk '{ print $2, "\t", $3, "\t", $1, "\t", $13, "\t", $8, "\t", $9 }' | gzip > /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PTF_PTM_combined.wHapMap22.MatchedToInit.IncAllele.formatted.txt.gz

LANG2=${LANG}
LC_ALL2=${LC_ALL}
LANG=C
LC_ALL=C
export LC_ALL LANG

join -a 1 -a 2 -1 3 -2 3 -e NA2 -o 0 1.1 1.2 1.4 1.5 1.6 2.1 2.2 2.4 2.5 2.6 <(zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_10F_12M_combined.wHapMap22.IncAllele.formatted.txt.gz | sort -g -k 3,3) <(zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PGF_PGM_combined.wHapMap22.MatchedToInit.IncAllele.formatted.txt.gz | sort -g -k 3,3 ) | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
join -a 1 -a 2 -1 1 -2 3 -e NA2 -o 0 1.2 2.1 2.2 2.4 2.5 2.6 - <(zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_PTF_PTM_combined.wHapMap22.MatchedToInit.IncAllele.formatted.txt.gz | sort -g -k 3,3) | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); print join("\t", @F);' | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); if ($#F == 6) { splice(@F, 1, 0, (("NA2") x 9)); } my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
perl -lane 'print $F[0], "\t", join("\t", split(/_/, $F[1]));' | gzip > /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_AllPhenosC.wHapMap22.MatchedToInit.IncAllele.formatted.RAF.txt.pre1.gz

LANG=${LANG2}
LC_ALL=${LC_ALL2}
export LC_ALL LANG

~~~
[  mturchin20@bigmem01  ~/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1]$zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_AllPhenosC.wHapMap22.MatchedToInit.IncAllele.formatted.RAF.txt.pre1.gz | wc
2379500 38072000 316832616
[  mturchin20@bigmem01  ~/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1]$zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_AllPhenosC.wHapMap22.MatchedToInit.IncAllele.formatted.RAF.txt.pre1.gz | head -n 10
rs1172982  1    100002699       0.3     0.378226807425313       13957   1       100002699       0.3     0.130556288379838       10799   1       100002699       0.7     -0.721190667447704      4282
rs1172981  1    100002785       0.067   0.98007330979874        13960   1       100002785       0.933   -0.281103207398319      10799   1       100002785       0.933   -0.764583401138596      4282
rs11166327 1    100003455       0.047   0.878883819019079       13958   1       100003455       0.047   0.271981247650781       10799   1       100003455       0.047   0.883868484876356       4282
rs4908018  1    100006955       0.0669999999999999      0.92077480613094        13952   1       100006955       0.0669999999999999      0.326654064064504       10790   1       100006955       0.0669999999999999      0.748780478605554 4282
rs2392072  1    100007331       0.717   0.518432700926805       13956   1       100007331       0.717   0.0946682726000233      10799   1       100007331       0.283   -0.953849793871725      4282
rs12129588 1    100007519       0.063   0.75473715711479        13959   1       100007519       0.063   0.321499411975321       10799   1       100007519       0.063   0.732349691261105       4282
rs10875256 1    100007856       0.058   0.9722809579372 13960   1       100007856       0.058   0.285260703596043       10799   1       100007856       0.058   0.749891691372287       4282
rs7541580  1    100008014       0.783   0.520103909703256       13960   1       100008014       0.783   0.0159185546385322      10799   1       100008014       0.217   -0.95126228341394       4282
rs883532   1    100008025       0.857   0.509478137156726       13960   1       100008025       0.857   0.0157246505697832      10799   1       100008025       0.143   -0.946258049168778      4282
rs883533   1    100008439       0.0669999999999999      0.971476593314794       13960   1       100008439       0.0669999999999999      0.285269263149643       10799   1       100008439       0.0669999999999999      0.747865216307332 4282
[  mturchin20@bigmem01  ~/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1]$zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_AllPhenosC.wHapMap22.MatchedToInit.IncAllele.formatted.RAF.txt.pre1.gz | perl -lane 'print $#F;' | sort | uniq -c
2379500 15
~~~

zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_AllPhenosC.wHapMap22.MatchedToInit.IncAllele.formatted.RAF.txt.pre1.gz | grep -v NA2 | gzip > /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_AllPhenosC.wHapMap22.MatchedToInit.IncAllele.formatted.RAF.txt.pre2.gz

zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_AllPhenosC.wHapMap22.MatchedToInit.IncAllele.formatted.RAF.txt.pre2.gz | perl -lane 'splice(@F, 21, 3); splice(@F, 16, 3); splice(@F, 11, 3); splice(@F, 6, 3); print join("\t", @F);' | gzip > /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_AllPhenosC.wHapMap22.MatchedToInit.IncAllele.formatted.RAF.txt.gz

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/EGG2013/EGG2013.Table1.ChrBP.txt --file2 /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_AllPhenosC.wHapMap22.MatchedToInit.IncAllele.formatted.RAF.txt.gz | gzip > /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_AllPhenosC.wHapMap22.MatchedToInit.IncAllele.formatted.RAF.wGWASannot.txt.gz

~~~
[  mturchin20@bigmem01  ~/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1]$zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_AllPhenosC.wHapMap22.MatchedToInit.IncAllele.formatted.RAF.wGWASannot.txt.gz | perl -lane 'print $F[$#F];' | sort | uniq -c
2231155 0
      9 1
   7067 2
~~~

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1/process.MTedits.ForEGG2013.Female.vs1.SignCrrct.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1/process.MTedits.ForEGG2013.Combined.vs1.SignCrrct.vs1.R

LANG2=${LANG}
LC_ALL2=${LC_ALL}
LANG=C
LC_ALL=C
export LC_ALL LANG

join -a 2 -1 1 -2 1 -e NA4 -o 0 1.2 1.3 1.4 1.5 1.6 1.7 1.8 1.9 1.10 1.11 2.2 2.3 2.4 2.5 2.6 2.7 <(zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_AllPhenosC.wHapMap22.MatchedToInit.IncAllele.formatted.RAF.wGWASannot.txt.gz | sort -g -k 1,1 ) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1/EGG2013Combined.dtlesslesssignif.vs1.SignCrrct.vs1.txt | perl -F, -lane 'print join("\t", @F);' | sort -g -k 1,1) | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1/EGG2013Combined.dtlesslesssignif.vs1.SignCrrct.vs1.annot.vs1.RAF.txt.pre1.gz

LANG=${LANG2}
LC_ALL=${LC_ALL2}
export LC_ALL LANG

zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1/EGG2013Combined.dtlesslesssignif.vs1.SignCrrct.vs1.annot.vs1.RAF.txt.pre1.gz | grep -v NA4 | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1/EGG2013Combined.dtlesslesssignif.vs1.SignCrrct.vs1.annot.vs1.RAF.txt.gz

#Added the following header to /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1/EGG2013Combined.dtlesslesssignif.vs1.SignCrrct.vs1.annot.vs1.RAF.txt.gz
#snp chr pos raf p_Init n_Init p_PubGrwth n_PubGrwth p_LtAdlsc n_LtAdlsc annot Z.Init Z.PubGrwth Z.LtAdlsc mvstat mvp unip

zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1/EGG2013Combined.dtlesslesssignif.vs1.SignCrrct.vs1.annot.vs1.RAF.txt.gz | perl -lane 'if ($F[3] > .5) { $F[3] = 1 - $F[3]; $F[11] = $F[11] * -1; $F[12] = $F[12] * -1; $F[13] = $F[13] * -1; } print join("\t", @F);' | sed 's/raf/maf/g' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1/EGG2013Combined.dtlesslesssignif.vs1.SignCrrct.vs1.annot.vs1.MAF.txt.gz

#cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/MAGIC2010/Vs1/GLC.MTedits.ForMAGIC2010.vs1.SignCrrct.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1/GLC.MTedits.ForEGG2013.Female.vs1.SignCrrct.vs1.R
#cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GLC.MTedits.ForGIANT2014_5.Orig3.vs1.SignCrrct.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1/GLC.MTedits.ForEGG2013.Female.vs1.SignCrrct.vs1.R
cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1/GLC.MTedits.ForEGG2013.Female.vs1.SignCrrct.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1/GLC.MTedits.ForEGG2013.Combined.vs1.SignCrrct.vs1.R



















































##GERA -- ##dbSNP, GWAS, N=?
##2015____

##BEAGESS -- ##dbSNP, GWAS, N=?
##20150818

#On mturchin20@wolfy.uchicago.edu

#cd /Users/mturchin20/clstrHme/Data/dbGaPTempDownloads/
cd /Users/mturchin20/clstrHme2/Data/dbGaPTempDownloads/

#Downloaded from Firefox browser to the above directory

#Downloading files from dbGaP is a bit of a pain, use these directions -- http://www.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=toolkit_doc&f=dbgap_use
#Need to download the project password (a '.ngc' file), upload it via vdb-config into the SRA toolkit, then use the vdb-decrypt utility /in the directory you specificy during vdb-config/ on the encrypted file of interest



##GWASofVTE -- ##dbSNP, GWAS, N=?
##20150814

#On mturchin20@wolfy.uchicago.edu

#cd /Users/mturchin20/clstrHme/Data/dbGaPTempDownloads/
cd /Users/mturchin20/clstrHme2/Data/dbGaPTempDownloads/

#Downloaded from Firefox browser to the above directory

#Downloading files from dbGaP is a bit of a pain, use these directions -- http://www.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=toolkit_doc&f=dbgap_use
#Need to download the project password (a '.ngc' file), upload it via vdb-config into the SRA toolkit, then use the vdb-decrypt utility /in the directory you specificy during vdb-config/ on the encrypted file of interest

cd /Users/mturchin20/clstrHme2/Data/dbGaPTempDownloads/45648/NHGRI/VenousThrombosis/phs000289v2/p1
#for i in `ls -lrt | awk '{ print $9 }' | grep ncbi_enc$`; do vdb-decrypt $i; done
cd /Users/mturchin20/clstrHme2/Data/dbGaPTempDownloads/dbGaP-9195/dbGaP-9195/45648/NHGRI/VenousThrombosis/phs000289v2/p1/phenotype
~~~
[  mturchin20@wolfy  ~/clstrHme2/Data/dbGaPTempDownloads/dbGaP-9195/dbGaP-9195/45648/NHGRI/VenousThrombosis/phs000289v2/p1/phenotype]$ for i in `ls -lrt | awk '{ print $9 }' | grep ncbi_enc$`; do vdb-decrypt $i; done
2015-08-14T18:02:49 vdb-decrypt.2.5.2: Decrypting file in place phs000289.v2.p1.data_dictionary.MULTI.tar.gz.ncbi_enc
2015-08-14T18:02:49 vdb-decrypt.2.5.2: copying phs000289.v2.p1.data_dictionary.MULTI.tar.gz.ncbi_enc to .phs000289.v2.p1.data_dictionary.MULTI.tar.gz.ncbi_enc.vdb-decrypt-tmp
2015-08-14T18:02:49 vdb-decrypt.2.5.2: renaming .phs000289.v2.p1.data_dictionary.MULTI.tar.gz.ncbi_enc.vdb-decrypt-tmp to phs000289.v2.p1.data_dictionary.MULTI.tar.gz
2015-08-14T18:02:49 vdb-decrypt.2.5.2: exiting: success
2015-08-14T18:02:49 vdb-decrypt.2.5.2: Decrypting file in place phs000289.v2.p1.variable_report.MULTI.tar.gz.ncbi_enc
2015-08-14T18:02:49 vdb-decrypt.2.5.2: copying phs000289.v2.p1.variable_report.MULTI.tar.gz.ncbi_enc to .phs000289.v2.p1.variable_report.MULTI.tar.gz.ncbi_enc.vdb-decrypt-tmp
2015-08-14T18:02:49 vdb-decrypt.2.5.2: renaming .phs000289.v2.p1.variable_report.MULTI.tar.gz.ncbi_enc.vdb-decrypt-tmp to phs000289.v2.p1.variable_report.MULTI.tar.gz
2015-08-14T18:02:49 vdb-decrypt.2.5.2: exiting: success
2015-08-14T18:02:49 vdb-decrypt.2.5.2: Decrypting file in place phs000289.v2.pht001883.v1.p1.Venous_Thrombosis_Subject.MULTI.txt.gz.ncbi_enc
.
.
.
~~~
cd /Users/mturchin20/clstrHme2/Data/dbGaPTempDownloads/dbGaP-9195/dbGaP-9195/45648/NHGRI/VenousThrombosis/phs000289v2/p1/genotype
#Ran code
cd /Users/mturchin20/clstrHme2/Data/dbGaPTempDownloads/dbGaP-9195/dbGaP-9195/45648/NHGRI/VenousThrombosis/phs000289v2/p1/genotype/phg000130v1
#Ran code
cd /Users/mturchin20/clstrHme2/Data/dbGaPTempDownloads/dbGaP-9195/dbGaP-9195/45648/NHGRI/VenousThrombosis/phs000289v2/p1/genotype/phg000088v1

gzip /Users/mturchin20/clstrHme2/Data/dbGaPTempDownloads/dbGaP-9195/dbGaP-9195/45648/NHGRI/VenousThrombosis/phs000289v2/p1/genotype/phg000130v1/
gzip /Users/mturchin20/clstrHme2/Data/dbGaPTempDownloads/dbGaP-9195/dbGaP-9195/45648/NHGRI/VenousThrombosis/phs000289v2/p1/genotype/phg000088v1/

mkdir /mnt/gluster/data/external_private_supp/VenousThrombosis
mv /Users/mturchin20/clstrHme2/Data/dbGaPTempDownloads/dbGaP-9195/dbGaP-9195/45648/NHGRI/VenousThrombosis/* /mnt/gluster/data/external_private_supp/VenousThrombosis/.








##Blood Traits 2012/HaemgenRBC2012 -- ChrBP provided, GWAS = 75 <10e-8 , N=provided
##20150625

#20150814 CHECK_0: Meta analysis was conducted between Europeans and South East Asians, so for MAF what do exactly?

mkdir /mnt/lustre/home/mturchin20/Data/HapMap/Releases
mkdir /mnt/lustre/home/mturchin20/Data/HapMap/Releases/Release22
mkdir /mnt/lustre/home/mturchin20/Data/HapMap/Releases/Release22/Frequencies 

cd /mnt/lustre/home/mturchin20/Data/HapMap/Releases/Release22/Frequencies

wget http://hapmap.ncbi.nlm.nih.gov/downloads/frequencies/2007-03/fwd_strand/non-redundant

mv /mnt/lustre/home/mturchin20/Data/HapMap/Releases/Release22/Frequencies/index.html /mnt/lustre/home/mturchin20/Data/HapMap/Releases/Release22/Frequencies/index.http:__hapmap.ncbi.nlm.nih.gov_downloads_frequencies_2007-03_fwd_strand_non-redundant.html

for i in `cat /mnt/lustre/home/mturchin20/Data/HapMap/Releases/Release22/Frequencies/index.http:__hapmap.ncbi.nlm.nih.gov_downloads_frequencies_2007-03_fwd_strand_non-redundant.html | perl -lane 'my $fullLine = join("", @F); if ($fullLine =~ m/.*(genotype.*.txt.gz).*/) { print $1 ; } '`; do wget http://hapmap.ncbi.nlm.nih.gov/downloads/frequencies/2007-03/fwd_strand/non-redundant/$i; done

wget http://hapmap.ncbi.nlm.nih.gov/downloads/frequencies/2007-03/rs_strand/non-redundant/

mv /mnt/lustre/home/mturchin20/Data/HapMap/Releases/Release22/Frequencies/index.html /mnt/lustre/home/mturchin20/Data/HapMap/Releases/Release22/Frequencies/index.http:__hapmap.ncbi.nlm.nih.gov_downloads_frequencies_2007-03_rs_strand_non-redundant.html

for i in `cat /mnt/lustre/home/mturchin20/Data/HapMap/Releases/Release22/Frequencies/index.http:__hapmap.ncbi.nlm.nih.gov_downloads_frequencies_2007-03_rs_strand_non-redundant.html | perl -lane 'my $fullLine = join("", @F); if ($fullLine =~ m/.*(allele.*.txt.gz).*/) { print $1 ; } '`; do wget http://hapmap.ncbi.nlm.nih.gov/downloads/frequencies/2007-03/rs_strand/non-redundant/$i; done

nbthis *

mkdir /mnt/gluster/data/external_public_supp/HaemgenRBC2012
cd /mnt/gluster/data/external_public_supp/HaemgenRBC2012

wget --user ega-box-233 --password 88fHetwh ftp://ftp.ega.ebi.ac.uk/EGAD00010000300/*

~~~
[  mturchin20@spudhead  /mnt/gluster/data/external_public_supp/HaemgenRBC2012]$zcat HaemGenRBC_RBC.txt.gz | head -n 10
SNP      Pval    Chr     Position        Effect_Allele   Non_Effect_Allele       Sample_Size     Beta    SE
rs9373124        5.11e-97     6          135464902       T       C       53337   0.0484          0.0024
rs7776054        3.51e-96     6          135460609       A       G       53403   0.0479          0.0024
rs9389269        4.54e-96     6          135468852       T       C       53402   0.0469          0.0023
rs9389268        5.98e-95     6          135461324       A       G       53403   0.0471          0.0024
rs9376090        9.04e-95     6          135452921       T       C       53537   0.0477          0.0024
rs11759553       1.84e-94     6          135463989       A       T       53398   0.0465          0.0023
rs4895441        3.67e-94     6          135468266       A       G       53384   0.0459          0.0023
rs4895440        3.75e-94     6          135468251       A       T       53400   0.0459          0.0023
rs9402686        4.36e-94     6          135469510       A       G       53400   -0.0469         0.0024
[  mturchin20@spudhead  /mnt/gluster/data/external_public_supp/HaemgenRBC2012]$zcat HaemGenRBC_PCV.txt.gz | head -n 10
SNP      Pval    Chr     Position        Effect_Allele   Non_Effect_Allele       Sample_Size     Beta    SE
rs9373124        3.869e-22    6          135464902       T       C       52764   0.1815          0.0198
rs9389269        1.966e-21    6          135468852       T       C       52829   0.1737          0.019
rs9402686        2.533e-21    6          135469510       A       G       52827   -0.1742         0.0195
rs11759553       4.364e-21    6          135463989       A       T       52825   0.1727          0.0189
rs4895441        5.04e-21     6          135468266       A       G       52811   0.1706          0.0188
rs4895440        5.451e-21    6          135468251       A       T       52827   0.1701          0.0187
rs7776054        6.178e-21    6          135460609       A       G       52830   0.1763          0.0191
rs9389268        1.32e-20     6          135461324       A       G       52830   0.1724          0.0191
rs9376092        2.143e-20    6          135468837       A       C       48671   -0.173          0.0196
[  mturchin20@spudhead  /mnt/gluster/data/external_public_supp/HaemgenRBC2012]$zcat HaemGenRBC_MCV.txt.gz | head -n 10
SNP      Pval    Chr     Position        Effect_Allele   Non_Effect_Allele       Sample_Size     Beta    SE
rs9389269        2.57e-109    6          135468852       T       C       57855   -0.6003         0.0275
rs7776054        3.13e-108    6          135460609       A       G       57856   -0.6053         0.0277
rs9373124        4.4e-108     6          135464902       T       C       57790   -0.617          0.0284
rs4895441        1.02e-107    6          135468266       A       G       57837   -0.59   0.0272
rs9376090        1.38e-107    6          135452921       T       C       57990   -0.6047         0.0279
rs9389268        2.47e-107    6          135461324       A       G       57856   -0.6004         0.0276
rs4895440        4.47e-107    6          135468251       A       T       57853   -0.5887         0.0272
rs11759553       1.32e-106    6          135463989       A       T       57851   -0.5926         0.0273
rs9402686        1.68e-106    6          135469510       A       G       57853   0.6032          0.028
[  mturchin20@spudhead  /mnt/gluster/data/external_public_supp/HaemgenRBC2012]$zcat HaemGenRBC_MCH.txt.gz | head -n 10
SNP      Pval    Chr     Position        Effect_Allele   Non_Effect_Allele       Sample_Size     Beta    SE
rs7776054        6.55e-108    6          135460609       A       G       51452   -0.2399         0.0108
rs9389269        8.02e-108    6          135468852       T       C       51452   -0.2368         0.0108
rs9389268        6.04e-107    6          135461324       A       G       51453   -0.2382         0.0108
rs9373124        1.35e-106    6          135464902       T       C       51388   -0.2436         0.0111
rs4895441        2.9e-106     6          135468266       A       G       51434   -0.2322         0.0106
rs9376090        3.04e-106    6          135452921       T       C       51588   -0.2385         0.0109
rs9402686        1.78e-105    6          135469510       A       G       51450   0.2388          0.011
rs4895440        3.28e-105    6          135468251       A       T       51450   -0.2313         0.0106
rs11759553       6.58e-105    6          135463989       A       T       51448   -0.2331         0.0107
[  mturchin20@spudhead  /mnt/gluster/data/external_public_supp/HaemgenRBC2012]$zcat HaemGenRBC_MCHC.txt.gz | head -n 10
SNP      Pval    Chr     Position        Effect_Allele   Non_Effect_Allele       Sample_Size     Beta    SE
rs10445033       1.536e-22    16         87367963        A       G       42050   -0.017          0.0038
rs837763         1.928e-22    16         87381230        T       C       37768   -0.0147         0.0037
rs475596         5.644e-21    16         87374452        C       G       42124   -0.0135         0.0038
rs198846         7.458e-21    6          26215442        A       G       56189   0.0207          0.0048
rs9932423        2.263e-19    16         87374350        A       C       41839   0.0153          0.0041
rs2608604        4.232e-19    16         87376922        A       G       37770   0.0126          0.0038
rs2932690        4.584e-18    16         87370454        A       G       42098   -0.0162         0.0041
rs855791         3.107e-17    22         35792882        A       G       41609   -0.0124         0.0037
rs2794720        3.349e-17    6          26195181        C       G       44372   0.0143          0.0037
[  mturchin20@spudhead  /mnt/gluster/data/external_public_supp/HaemgenRBC2012]$zcat HaemGenRBC_Hb.txt.gz | head -n 10
SNP      Pval    Chr     Position        Effect_Allele   Non_Effect_Allele       Sample_Size     Beta    SE
rs855791         4.645e-40    22         35792882        A       G       46184   -0.0791         0.0063
rs4820268        1.153e-31    22         35799537        A       G       43990   0.0699          0.0064
rs2413450        4.411e-31    22         35800170        T       C       49624   -0.0695         0.0064
rs198846         1.422e-30    6          26215442        A       G       60869   0.091   0.008
rs129128         8.996e-27    6          26233321        T       C       53598   -0.0925         0.0086
rs1799945        3.599e-26    6          26199158        C       G       49746   -0.0938         0.0089
rs198833         8.537e-26    6          26222487        A       G       49747   -0.0945         0.009
rs198851         6.056e-25    6          26212611        T       G       49716   0.0963          0.0093
rs130619         2.031e-24    22         35766294        T       C       55170   0.0586          0.0061
[  mturchin20@spudhead  /mnt/gluster/data/external_public_supp/HaemgenRBC2012]$ls -lrt | awk '{ print $9}'
HaemGenRBC_RBC.txt.gz
HaemGenRBC_PCV.txt.gz
HaemGenRBC_MCV.txt.gz
HaemGenRBC_MCH.txt.gz
HaemGenRBC_MCHC.txt.gz
HaemGenRBC_Hb.txt.gz
[  mturchin20@spudhead  /mnt/gluster/data/external_public_supp/HaemgenRBC2012]$for i in `ls -lrt | awk '{ print $9}'`; do zcat $i | wc; done
2589455 23305095 178560057
2591080 23319720 179098941
2591133 23320197 179175090
2586785 23281065 178736256
2588876 23299884 178629989
2593079 23337711 179093409
~~~

#20150814 20150814 CHECK_1: Need to get MAF for these datasets

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsindbSNPvcf.vs3.py /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_RBC.txt.gz | gzip > /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_RBC.wHapMap22.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_PCV.txt.gz | gzip > /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_PCV.wHapMap22.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_MCV.txt.gz | gzip > /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_MCV.wHapMap22.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_MCH.txt.gz | gzip > /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_MCH.wHapMap22.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_MCHC.txt.gz | gzip > /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_MCHC.wHapMap22.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_Hb.txt.gz | gzip > /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_Hb.wHapMap22.txt.gz &

#20150814 CHECK_0: Check that ChrBP/RefAllele/OtherAllele/etc are all the same between entries and the HapMap2 database?


mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012
mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/MAGIC2010/Vs1/process.MTedits.ForMAGIC2010.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/process.MTedits.ForHaemgenRBC2012.vs1.R

zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_RBC.wHapMap22.txt.gz | perl -lane 'my @vals1 = split(/_/, $F[9]); push(@F, @vals1); print join("\t", @F);' | awk '{ print $3, "\t", $4, "\t", $1, "\t", $14, "\t", $2, "\t", $7 }' | gzip > /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_RBC.wHapMap22.formatted.txt.gz
zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_PCV.wHapMap22.txt.gz | perl -lane 'my @vals1 = split(/_/, $F[9]); push(@F, @vals1); print join("\t", @F);' | awk '{ print $3, "\t", $4, "\t", $1, "\t", $14, "\t", $2, "\t", $7 }' | gzip > /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_PCV.wHapMap22.formatted.txt.gz
zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_MCV.wHapMap22.txt.gz | perl -lane 'my @vals1 = split(/_/, $F[9]); push(@F, @vals1); print join("\t", @F);' | awk '{ print $3, "\t", $4, "\t", $1, "\t", $14, "\t", $2, "\t", $7 }' | gzip > /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_MCV.wHapMap22.formatted.txt.gz
zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_MCH.wHapMap22.txt.gz | perl -lane 'my @vals1 = split(/_/, $F[9]); push(@F, @vals1); print join("\t", @F);' | awk '{ print $3, "\t", $4, "\t", $1, "\t", $14, "\t", $2, "\t", $7 }' | gzip > /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_MCH.wHapMap22.formatted.txt.gz
zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_MCHC.wHapMap22.txt.gz | perl -lane 'my @vals1 = split(/_/, $F[9]); push(@F, @vals1); print join("\t", @F);' | awk '{ print $3, "\t", $4, "\t", $1, "\t", $14, "\t", $2, "\t", $7 }' | gzip > /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_MCHC.wHapMap22.formatted.txt.gz
zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_Hb.wHapMap22.txt.gz | perl -lane 'my @vals1 = split(/_/, $F[9]); push(@F, @vals1); print join("\t", @F);' | awk '{ print $3, "\t", $4, "\t", $1, "\t", $14, "\t", $2, "\t", $7 }' | gzip > /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_Hb.wHapMap22.formatted.txt.gz

LANG2=${LANG}
LC_ALL2=${LC_ALL}
LANG=C
LC_ALL=C
export LC_ALL LANG

join -a 1 -a 2 -1 3 -2 3 -e NA2 -o 0 1.1 1.2 1.4 1.5 1.6 2.1 2.2 2.4 2.5 2.6 <(zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_RBC.wHapMap22.formatted.txt.gz | sort -g -k 3,3) <(zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_PCV.wHapMap22.formatted.txt.gz | sort -g -k 3,3 ) | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
join -a 1 -a 2 -1 1 -2 3 -e NA2 -o 0 1.2 2.1 2.2 2.4 2.5 2.6 - <(zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_MCV.wHapMap22.formatted.txt.gz | sort -g -k 3,3) | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); print join("\t", @F);' | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); if ($#F == 6) { splice(@F, 1, 0, (("NA2") x 9)); } my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
join -a 1 -a 2 -1 1 -2 3 -e NA2 -o 0 1.2 2.1 2.2 2.4 2.5 2.6 - <(zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_MCH.wHapMap22.formatted.txt.gz | sort -g -k 3,3) | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); print join("\t", @F);' | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); if ($#F == 6) { splice(@F, 1, 0, (("NA2") x 14)); } my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
join -a 1 -a 2 -1 1 -2 3 -e NA2 -o 0 1.2 2.1 2.2 2.4 2.5 2.6 - <(zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_MCHC.wHapMap22.formatted.txt.gz | sort -g -k 3,3) | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); print join("\t", @F);' | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); if ($#F == 6) { splice(@F, 1, 0, (("NA2") x 19)); } my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
join -a 1 -a 2 -1 1 -2 3 -e NA2 -o 0 1.2 2.1 2.2 2.4 2.5 2.6 - <(zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_Hb.wHapMap22.formatted.txt.gz | sort -g -k 3,3) | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); print join("\t", @F);' | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); if ($#F == 6) { splice(@F, 1, 0, (("NA2") x 24)); } my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
perl -lane 'print $F[0], "\t", join("\t", split(/_/, $F[1]));' | sed 's/chr//g' | gzip > /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_AllPhenos.wHapMap22.formatted.txt.pre1.gz

~~~
[  mturchin20@spudling26  ~/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1]$zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_AllPhenos.wHapMap22.formatted.txt.pre1.gz | wc
2659811 82454147 507851475
[  mturchin20@spudling26  ~/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1]$zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_AllPhenos.wHapMap22.formatted.txt.pre1.gz | perl -lane 'print $#F;' | sort | uniq -c
2659810 30
      1 36
[  mturchin20@spudling26  ~/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1]$zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_AllPhenos.wHapMap22.formatted.txt.pre1.gz | perl -lane 'if ($#F == 36) { print join("\t", @F); }'
SNP     Chr     Position        Pval    Sample  Size    NA2     Chr     Position        Pval    Sample  Size    NA2     Chr     Position        Pval    Sample  Size   NA2      Chr     Position        Pval    Sample  Size    NA2     Chr     Position        Pval    Sample  Size    NA2     Chr     Position        Pval    Sample  Size   NA2
[  mturchin20@spudling26  ~/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1]$zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_AllPhenos.wHapMap22.formatted.txt.pre1.gz | grep NA2 | wc
 262334 8132360 44885161
~~~      

zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_AllPhenos.wHapMap22.formatted.txt.pre1.gz | grep -v NA2 | gzip > /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_AllPhenos.wHapMap22.formatted.txt.pre2.gz

zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_AllPhenos.wHapMap22.formatted.txt.pre2.gz | awk '{ if (($4 != $9) && ($4 != $14) && ($4 != $19)) { print $0 } } ' | wc
zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_AllPhenos.wHapMap22.formatted.txt.pre2.gz | awk '{ if (($4 != $9) || ($4 != $14) || ($4 != $19)) { print $0 } } ' | wc
zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_AllPhenos.wHapMap22.formatted.txt.pre2.gz | awk '{ if (($3 != $8) && ($3 != $13) && ($3 != $18)) { print $0 } } ' | wc
zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_AllPhenos.wHapMap22.formatted.txt.pre2.gz | awk '{ if (($3 != $8) || ($3 != $13) || ($3 != $18)) { print $0 } } ' | wc
zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_AllPhenos.wHapMap22.formatted.txt.pre2.gz | awk '{ if (($2 != $7) && ($2 != $12) && ($2 != $17)) { print $0 } } ' | wc
zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_AllPhenos.wHapMap22.formatted.txt.pre2.gz | awk '{ if (($2 != $7) || ($2 != $12) || ($2 != $17)) { print $0 } } ' | wc

~~
[  mturchin20@spudling26  ~/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1]$zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_AllPhenos.wHapMap22.formatted.txt.pre2.gz | awk '{ if (($4 != $9) && ($4 != $14) && ($4 != $19)) { print $0 } } ' | wc
zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_AllPhenos.wHapMap22.formatted.txt.pre2.gz | awk '{ if (($2 != $7) || ($2 != $12) || ($2 != $17)) { print $0 } } ' | wc
0       0       0
[  mturchin20@spudling26  ~/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1]$zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_AllPhenos.wHapMap22.formatted.txt.pre2.gz | awk '{ if (($4 != $9) || ($4 != $14) || ($4 != $19)) { print $0 } } ' | wc
0       0       0
[  mturchin20@spudling26  ~/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1]$zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_AllPhenos.wHapMap22.formatted.txt.pre2.gz | awk '{ if (($3 != $8) && ($3 != $13) && ($3 != $18)) { print $0 } } ' | wc
0       0       0
[  mturchin20@spudling26  ~/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1]$zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_AllPhenos.wHapMap22.formatted.txt.pre2.gz | awk '{ if (($3 != $8) || ($3 != $13) || ($3 != $18)) { print $0 } } ' | wc
0       0       0
[  mturchin20@spudling26  ~/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1]$zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_AllPhenos.wHapMap22.formatted.txt.pre2.gz | awk '{ if (($2 != $7) && ($2 != $12) && ($2 != $17)) { print $0 } } ' | wc
0       0       0
[  mturchin20@spudling26  ~/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1]$zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_AllPhenos.wHapMap22.formatted.txt.pre2.gz | awk '{ if (($2 != $7) || ($2 != $12) || ($2 != $17)) { print $0 } } ' | wc
0       0       0
~~

zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_AllPhenos.wHapMap22.formatted.txt.pre2.gz | perl -lane 'splice(@F, 31, 3); splice(@F, 26, 3); splice(@F, 21, 3); splice(@F, 16, 3); splice(@F, 11, 3); splice(@F, 6, 3); print join("\t", @F);' | gzip > /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_AllPhenos.wHapMap22.formatted.txt.gz

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/HaemgenRBC2012/HaemgenRBC2012.Table1.Edited.MarkerChrBP.txt --file2 /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_AllPhenos.wHapMap22.formatted.txt.gz | gzip > /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_AllPhenos.wHapMap22.formatted.wGWASannot.txt.gz 






~~~
[  mturchin20@spudling90  ~]$python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/HaemgenRBC2012/HaemgenRBC2012.Table1.Edited.MarkerChrBP.txt --file2 /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_AllPhenos.wHapMap22.formatted.txt.gz | gzip > /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_AllPhenos.wHapMap22.formatted.wGWASannot.txt.gz
/mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_AllPhenos.wHapMap22.formatted.txt.gz
[  mturchin20@spudling90  ~]$
[  mturchin20@spudling90  ~]$zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_AllPhenos.wHapMap22.formatted.wGWASannot.txt.gz | perl -lane 'print $F[$#F];' | sort | uniq -c
2338820 0
     71 1
  58586 2
~~~

LANG2=${LANG}
LC_ALL2=${LC_ALL}
LANG=C
LC_ALL=C
export LC_ALL LANG

join -a 2 -1 1 -2 1 -e NA4 -o 0 1.2 1.3 1.4 1.5 1.6 1.7 1.8 1.9 1.10 1.11 1.12 1.13 1.14 1.15 1.16 1.17 2.2 2.3 2.4 2.5 2.6 2.7 2.8 2.9 2.10 <(zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_AllPhenos.wHapMap22.formatted.wGWASannot.txt.gz | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/HaemgenRBC2012.dtlesssignif.vs1.txt | perl -F, -lane 'print join("\t", @F);' | sort -k 1,1) | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/HaemgenRBC2012.dtlesssignif.vs1.annot.txt.pre1.gz 

~~~
[  mturchin20@spudling26  ~/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/HaemgenRBC2012.dtlesssignif.vs1.annot.txt.pre1.gz | grep NA4 | wc
    181    4344   34899
~~~    

LANG=${LANG2}
LC_ALL=${LC_ALL2}
export LC_ALL LANG

zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/HaemgenRBC2012.dtlesssignif.vs1.annot.txt.pre1.gz | grep -v NA4 | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/HaemgenRBC2012.dtlesssignif.vs1.annot.txt.gz

RBC PCV MCV MCH MCHC Hb 

#Added the following header to /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/HaemgenRBC2012.dtlesssignif.vs1.annot.txt.gz
#snp chr pos maf p_RBC n_RBC p_PCV n_PCV p_MCV n_MCV p_MCH n_MCH p_MCHC n_MCHC p_Hb n_Hb annot Z.RBC Z.PCV Z.MCV Z.MCH Z.MCHC Z.Hb mvstat mvp unip

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GLC.MTedits.ForGIANT2014_5.AllPheno.vs2.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/GLC.MTedits.ForHamegenRBC2012.AllPheno.vs1.R

#Converted /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GLC.MTedits.ForGIANT2014_5.AllPheno.vs2.R to HaemgenRBC2012 phenotypes, moving commands to RBC PCV MCV MCH MCHC and Hb instead of Height, BMI, WHRadjBMI, WHR, HIPadjBMI, HIP, WCadjBMI and WC

#Copy/pasted commands from /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/GLC.MTedits.ForHamegenRBC2012.AllPheno.vs1.R into an interactive sessions in $PWD /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/ 


#20151008
#CSHL Poster work

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/GLC.MTedits.ForHamegenRBC2012.AllPheno.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/GLC.MTedits.ForHamegenRBC2012.AllPheno.vs1.ForCSHLPoster.R

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/GLC.MTedits.ForICBP2011.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/GLC.MTedits.ForICBP2011.vs1.ForCSHLPoster.R

scp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/GLC.MTedits.ForICBP2011.vs1.ForCSHLPoster.ModelsVslbfavNewOld.vs1.jpeg mturchin20@wolfy.uchicago.edu:/Users/mturchin20/LabStuff/StephensLab/Multivariate/ICBP2011/Vs1/. 




#20160221
#Correcting sign issue -- quick hack to get results

#NOTE -- flipping sign on p-value to make coding downstream easier for this, ie I'm not implying something by creating negative p-values here. This is just for a quick run through.

#MAF ISSUE THING DO
#eg check strandedness with haemgen alleles and hapmap alleles

~~~
[  mturchin20@spudling70  ~]$zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_RBC.wHapMap22.txt.gz | perl -lane 'print join("\t", @F[0..$#F-1]), "\t", join("\t", (split("_", $F[$#F])));' | perl -lane 'if ($F[4] ne $F[11]) { ($F[11], $F[13]) = ($F[13], $F[11]); ($F[12], $F[14]) = ($F[14], $F[12]); } print join("\t", @F);' | perl -lane 'if ($F[7] < 0) { ($F[4], $F[5]) = ($F[5], $F[4]); $F[12] = 1 - $F[12]; $F[7] = $F[7] * -1; } print join("\t", @F);' | sed 's/ChrBPAFInfo/Chr BP Ref RAF Alt AAF/g' | perl -lane 'print $#F;' | sort | uniq -c
  63651 10
2462919 14
  62885 9
[  mturchin20@spudling70  ~]$zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_RBC.wHapMap22.txt.gz | perl -lane 'print join("\t", @F[0..$#F-1]), "\t", join("\t", (split("_", $F[$#F])));' | perl -lane 'if ($F[4] ne $F[11]) { ($F[11], $F[13]) = ($F[13], $F[11]); ($F[12], $F[14]) = ($F[14], $F[12]); } print join("\t", @F);' | perl -lane 'if ($F[7] < 0) { ($F[4], $F[5]) = ($F[5], $F[4]); $F[12] = 1 - $F[12]; $F[7] = $F[7] * -1; } print join("\t", @F);' | sed 's/ChrBPAFInfo/Chr BP Ref RAF Alt AAF/g' | perl -lane 'if ($#F == 10) { print join("\t", @F); } ' | head -n 10
rs351747        4.137e-09    6  109638544       G       T       11607   0.0231  0.0041  NA      1
rs9400276       8.704e-08    6  109754098       G       A       21418   0.018   0.0035  NA      1
rs642093        2.749e-07    6  32690053        G       A       30094   0.015   0.0036  NA      1
rs13011022      9.355e-07    2  60480326        G       A       21240   0.0151  0.0032  NA      1
rs198401        1.552e-06    1  11810971        G       A       21248   0.013   0.0032  NA      1
rs910050        3.793e-06    6  32423632        G       C       26754   0.0132  0.0029  NA      1
rs2414763       1.061e-05    15 60251716        G       A       13235   0.0207  0.005   NA      1
rs2816989       1.607e-05    1  198337388       C       A       13470   0.0142  0.0041  NA      1
rs1285619       2.037e-05    14 91073409        G       A       21266   0.0145  0.0039  NA      1
rs10824705      2.864e-05    10 80462534        C       A       21195   0.0134  0.0037  NA      1
[  mturchin20@spudling70  ~]$zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_RBC.wHapMap22.txt.gz | perl -lane 'print join("\t", @F[0..$#F-1]), "\t", join("\t", (split("_", $F[$#F])));' | perl -lane 'if ($F[4] ne $F[11]) { ($F[11], $F[13]) = ($F[13], $F[11]); ($F[12], $F[14]) = ($F[14], $F[12]); } print join("\t", @F);' | perl -lane 'if ($F[7] < 0) { ($F[4], $F[5]) = ($F[5], $F[4]); $F[12] = 1 - $F[12]; $F[7] = $F[7] * -1; } print join("\t", @F);' | sed 's/ChrBPAFInfo/Chr BP Ref RAF Alt AAF/g' | perl -lane 'if ($#F == 9) { print join("\t", @F); } ' | head -n 10
rs1076712       8.263e-08    6  32408130        T       C       30070   0.0142  0.0028  NA
rs830554        1.118e-07    6  109649919       A       G       21255   0.0194  0.0035  NA
rs2327586       5.586e-07    6  135536919       T       C       21259   0.0177  0.0037  NA
rs2727603       8.203e-07    5  11716510        T       C       21014   0.0234  0.0048  NA
rs351742        1.834e-06    6  109634018       A       G       21161   0.015   0.0033  NA
rs7442593       5.583e-06    4  55095835        A       G       11799   0.0429  0.0091  NA
rs6035245       6.024e-06    20 19091082        T       C       19037   0.0138  0.0036  NA
rs6554181       7.533e-06    4  55159595        A       G       11799   0.0368  0.0081  NA
rs2646814       8.303e-06    1  215590063       T       C       21194   0.0139  0.0037  NA
rs1635288       9.442e-06    17 41112752        A       G       21484   0.0161  0.0038  NA
[  mturchin20@spudling70  ~]$zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_RBC.wHapMap22.txt.gz | perl -lane 'print join("\t", @F[0..$#F-1]), "\t", join("\t", (split("_", $F[$#F])));' | perl -lane 'if ($F[4] ne $F[11]) { ($F[11], $F[13]) = ($F[13], $F[11]); ($F[12], $F[14]) = ($F[14], $F[12]); } print join("\t", @F);' | perl -lane 'if ($F[7] < 0) { ($F[4], $F[5]) = ($F[5], $F[4]); $F[12] = 1 - $F[12]; $F[7] = $F[7] * -1; } print join("\t", @F);' | sed 's/ChrBPAFInfo/Chr BP Ref RAF Alt AAF/g' | grep -v -w NA | perl -lane 'print $#F;' | sort | uniq -c
2462919 14
~~~

zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_RBC.txt.gz | perl -lane 'if ($F[7] < 0) { ($F[4], $F[5]) = ($F[5], $F[4]); 
zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_RBC.wHapMap22.txt.gz | perl -lane 'print join("\t", @F[0..$#F-1]), "\t", join("\t", (split("_", $F[$#F])));' | perl -lane 'if ($F[4] ne $F[11]) { ($F[11], $F[13]) = ($F[13], $F[11]); ($F[12], $F[14]) = ($F[14], $F[12]); } print join("\t", @F);' | perl -lane 'if ($F[7] < 0) { ($F[4], $F[5]) = ($F[5], $F[4]); $F[12] = 1 - $F[12]; $F[7] = $F[7] * -1; } print join("\t", @F);' | sed 's/ChrBPAFInfo/Chr BP Ref RAF Alt AAF/g' |  grep -v -w NA | gzip > /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_RBC.wHapMap22.IncAllele.txt.gz

LANG2=${LANG}
LC_ALL2=${LC_ALL}
LANG=C
LC_ALL=C
export LC_ALL LANG

#join <(zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_RBC.wHapMap22.IncAllele.txt.gz | awk '{ print $1, "\t", $5 }' | sort) <(zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_MCV.wHapMap22.txt.gz | perl -lane 'print join("\t", @F[0..$#F-1]), "\t", join("\t", (split("_", $F[$#F])));' | perl -lane 'if ($F[4] ne $F[11]) { ($F[11], $F[13]) = ($F[13], $F[11]); ($F[12], $F[14]) = ($F[14], $F[12]); } print join("\t", @F);' | perl -lane 'if ($F[7] < 0) { ($F[4], $F[5]) = ($F[5], $F[4]); $F[12] = 1 - $F[12]; $F[7] = $F[7] * -1; } print join("\t", @F);' | sort) | perl -lane 'if ($F[1] ne $F[5]) { $F[2] = $F[2] * -1; } splice(@F, 1, 1); print join("\t", @F);' |  sed 's/ChrBPAFInfo/Chr BP Ref RAF Alt AAF/g' |  grep -v -w NA | sort -r -k 1,1 | gzip >  /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_MCV.wHapMap22.MatchedToRBC.IncAllele.txt.gz
join <(zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_RBC.wHapMap22.IncAllele.txt.gz | awk '{ print $1, "\t", $5 }' | sort) <(zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_MCV.wHapMap22.txt.gz | perl -lane 'print join("\t", @F[0..$#F-1]), "\t", join("\t", (split("_", $F[$#F])));' | perl -lane 'if ($F[4] ne $F[11]) { ($F[11], $F[13]) = ($F[13], $F[11]); ($F[12], $F[14]) = ($F[14], $F[12]); } print join("\t", @F);' | perl -lane 'if ($F[7] < 0) { ($F[4], $F[5]) = ($F[5], $F[4]); $F[12] = 1 - $F[12]; $F[7] = $F[7] * -1; } print join("\t", @F);' | sort) | perl -lane 'if ($F[1] ne $F[5]) { $F[2] = $F[2] * -1; } splice(@F, 1, 1); print join("\t", @F);' |  sed 's/ChrBPAFInfo/Chr BP Ref RAF Alt AAF/g' |  grep -v -w NA | gzip >  /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_MCV.wHapMap22.MatchedToRBC.IncAllele.txt.gz
join <(zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_RBC.wHapMap22.IncAllele.txt.gz | awk '{ print $1, "\t", $5 }' | sort) <(zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_PCV.wHapMap22.txt.gz | perl -lane 'print join("\t", @F[0..$#F-1]), "\t", join("\t", (split("_", $F[$#F])));' | perl -lane 'if ($F[4] ne $F[11]) { ($F[11], $F[13]) = ($F[13], $F[11]); ($F[12], $F[14]) = ($F[14], $F[12]); } print join("\t", @F);' | perl -lane 'if ($F[7] < 0) { ($F[4], $F[5]) = ($F[5], $F[4]); $F[12] = 1 - $F[12]; $F[7] = $F[7] * -1; } print join("\t", @F);' | sort) | perl -lane 'if ($F[1] ne $F[5]) { $F[2] = $F[2] * -1; } splice(@F, 1, 1); print join("\t", @F);' | sed 's/ChrBPAFInfo/Chr BP Ref RAF Alt AAF/g' |  grep -v -w NA | gzip >  /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_PCV.wHapMap22.MatchedToRBC.IncAllele.txt.gz
join <(zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_RBC.wHapMap22.IncAllele.txt.gz | awk '{ print $1, "\t", $5 }' | sort) <(zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_MCH.wHapMap22.txt.gz | perl -lane 'print join("\t", @F[0..$#F-1]), "\t", join("\t", (split("_", $F[$#F])));' | perl -lane 'if ($F[4] ne $F[11]) { ($F[11], $F[13]) = ($F[13], $F[11]); ($F[12], $F[14]) = ($F[14], $F[12]); } print join("\t", @F);' | perl -lane 'if ($F[7] < 0) { ($F[4], $F[5]) = ($F[5], $F[4]); $F[12] = 1 - $F[12]; $F[7] = $F[7] * -1; } print join("\t", @F);' | sort) | perl -lane 'if ($F[1] ne $F[5]) { $F[2] = $F[2] * -1; } splice(@F, 1, 1); print join("\t", @F);' | sed 's/ChrBPAFInfo/Chr BP Ref RAF Alt AAF/g' |  grep -v -w NA | gzip >  /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_MCH.wHapMap22.MatchedToRBC.IncAllele.txt.gz
join <(zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_RBC.wHapMap22.IncAllele.txt.gz | awk '{ print $1, "\t", $5 }' | sort) <(zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_Hb.wHapMap22.txt.gz | perl -lane 'print join("\t", @F[0..$#F-1]), "\t", join("\t", (split("_", $F[$#F])));' | perl -lane 'if ($F[4] ne $F[11]) { ($F[11], $F[13]) = ($F[13], $F[11]); ($F[12], $F[14]) = ($F[14], $F[12]); } print join("\t", @F);' | perl -lane 'if ($F[7] < 0) { ($F[4], $F[5]) = ($F[5], $F[4]); $F[12] = 1 - $F[12]; $F[7] = $F[7] * -1; } print join("\t", @F);' | sort) | perl -lane 'if ($F[1] ne $F[5]) { $F[2] = $F[2] * -1; } splice(@F, 1, 1); print join("\t", @F);' | sed 's/ChrBPAFInfo/Chr BP Ref RAF Alt AAF/g' |  grep -v -w NA | gzip >  /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_Hb.wHapMap22.MatchedToRBC.IncAllele.txt.gz
join <(zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_RBC.wHapMap22.IncAllele.txt.gz | awk '{ print $1, "\t", $5 }' | sort) <(zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_MCHC.wHapMap22.txt.gz | perl -lane 'print join("\t", @F[0..$#F-1]), "\t", join("\t", (split("_", $F[$#F])));' | perl -lane 'if ($F[4] ne $F[11]) { ($F[11], $F[13]) = ($F[13], $F[11]); ($F[12], $F[14]) = ($F[14], $F[12]); } print join("\t", @F);' | perl -lane 'if ($F[7] < 0) { ($F[4], $F[5]) = ($F[5], $F[4]); $F[12] = 1 - $F[12]; $F[7] = $F[7] * -1; } print join("\t", @F);' | sort) | perl -lane 'if ($F[1] ne $F[5]) { $F[2] = $F[2] * -1; } splice(@F, 1, 1); print join("\t", @F);' | sed 's/ChrBPAFInfo/Chr BP Ref RAF Alt AAF/g' |  grep -v -w NA | gzip >  /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_MCHC.wHapMap22.MatchedToRBC.IncAllele.txt.gz

zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_RBC.wHapMap22.IncAllele.txt.gz | awk '{ print $3, "\t", $4, "\t", $1, "\t", $13, "\t", $2, "\t", $7 }' | gzip > /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_RBC.wHapMap22.IncAllele.formatted.txt.gz
zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_MCV.wHapMap22.MatchedToRBC.IncAllele.txt.gz | awk '{ print $3, "\t", $4, "\t", $1, "\t", $13, "\t", $2, "\t", $7 }' | gzip > /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_MCV.wHapMap22.MatchedToRBC.IncAllele.formatted.txt.gz
zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_PCV.wHapMap22.MatchedToRBC.IncAllele.txt.gz | awk '{ print $3, "\t", $4, "\t", $1, "\t", $13, "\t", $2, "\t", $7 }' | gzip > /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_PCV.wHapMap22.MatchedToRBC.IncAllele.formatted.txt.gz
zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_MCH.wHapMap22.MatchedToRBC.IncAllele.txt.gz | awk '{ print $3, "\t", $4, "\t", $1, "\t", $13, "\t", $2, "\t", $7 }' | gzip > /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_MCH.wHapMap22.MatchedToRBC.IncAllele.formatted.txt.gz
zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_Hb.wHapMap22.MatchedToRBC.IncAllele.txt.gz | awk '{ print $3, "\t", $4, "\t", $1, "\t", $13, "\t", $2, "\t", $7 }' | gzip > /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_Hb.wHapMap22.MatchedToRBC.IncAllele.formatted.txt.gz
zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_MCHC.wHapMap22.MatchedToRBC.IncAllele.txt.gz | awk '{ print $3, "\t", $4, "\t", $1, "\t", $13, "\t", $2, "\t", $7 }' | gzip > /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_MCHC.wHapMap22.MatchedToRBC.IncAllele.formatted.txt.gz

#LANG2=${LANG}
#LC_ALL2=${LC_ALL}
#LANG=C
#LC_ALL=C
#export LC_ALL LANG

join -a 1 -a 2 -1 3 -2 3 -e NA2 -o 0 1.1 1.2 1.4 1.5 1.6 2.1 2.2 2.4 2.5 2.6 <(zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_RBC.wHapMap22.IncAllele.formatted.txt.gz | sort -g -k 3,3) <(zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_PCV.wHapMap22.MatchedToRBC.IncAllele.formatted.txt.gz | sort -g -k 3,3 ) | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
join -a 1 -a 2 -1 1 -2 3 -e NA2 -o 0 1.2 2.1 2.2 2.4 2.5 2.6 - <(zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_MCV.wHapMap22.MatchedToRBC.IncAllele.formatted.txt.gz | sort -g -k 3,3) | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); print join("\t", @F);' | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); if ($#F == 6) { splice(@F, 1, 0, (("NA2") x 9)); } my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
join -a 1 -a 2 -1 1 -2 3 -e NA2 -o 0 1.2 2.1 2.2 2.4 2.5 2.6 - <(zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_MCH.wHapMap22.MatchedToRBC.IncAllele.formatted.txt.gz | sort -g -k 3,3) | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); print join("\t", @F);' | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); if ($#F == 6) { splice(@F, 1, 0, (("NA2") x 14)); } my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
join -a 1 -a 2 -1 1 -2 3 -e NA2 -o 0 1.2 2.1 2.2 2.4 2.5 2.6 - <(zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_MCHC.wHapMap22.MatchedToRBC.IncAllele.formatted.txt.gz | sort -g -k 3,3) | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); print join("\t", @F);' | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); if ($#F == 6) { splice(@F, 1, 0, (("NA2") x 19)); } my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
join -a 1 -a 2 -1 1 -2 3 -e NA2 -o 0 1.2 2.1 2.2 2.4 2.5 2.6 - <(zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_Hb.wHapMap22.MatchedToRBC.IncAllele.formatted.txt.gz | sort -g -k 3,3) | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); print join("\t", @F);' | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); if ($#F == 6) { splice(@F, 1, 0, (("NA2") x 24)); } my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
perl -lane 'print $F[0], "\t", join("\t", split(/_/, $F[1]));' | sed 's/chr//g' | gzip > /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.txt.pre1.gz

LANG=${LANG2}
LC_ALL=${LC_ALL2}
export LC_ALL LANG

zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.txt.pre1.gz | grep -v NA2 | gzip > /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.txt.pre2.gz

zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.txt.pre2.gz | perl -lane 'splice(@F, 31, 3); splice(@F, 26, 3); splice(@F, 21, 3); splice(@F, 16, 3); splice(@F, 11, 3); splice(@F, 6, 3); print join("\t", @F);' | gzip > /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.txt.gz

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/HaemgenRBC2012/HaemgenRBC2012.Table1.Edited.MarkerChrBP.txt --file2 /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.txt.gz | gzip > /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.wGWASannot.txt.gz

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/process.MTedits.ForHaemgenRBC2012.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/process.MTedits.ForHaemgenRBC2012.vs1.SignCrrct.vs1.R

LANG2=${LANG}
LC_ALL2=${LC_ALL}
LANG=C 
LC_ALL=C
export LC_ALL LANG

join -a 2 -1 1 -2 1 -e NA4 -o 0 1.2 1.3 1.4 1.5 1.6 1.7 1.8 1.9 1.10 1.11 1.12 1.13 1.14 1.15 1.16 1.17 2.2 2.3 2.4 2.5 2.6 2.7 2.8 2.9 2.10 <(zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.wGWASannot.txt.gz | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/HaemgenRBC2012.dtlesssignif.vs1.SignCrrct.vs1.txt | perl -F, -lane 'print join("\t", @F);' | sort -k 1,1) | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/HaemgenRBC2012.dtlesssignif.vs1.annot.vs1.SignCrrct.vs1.RAF.txt.pre1.gz

LANG=${LANG2}
LC_ALL=${LC_ALL2}
export LC_ALL LANG

zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/HaemgenRBC2012.dtlesssignif.vs1.annot.vs1.SignCrrct.vs1.RAF.txt.pre1.gz | grep -v NA4 | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/HaemgenRBC2012.dtlesssignif.vs1.annot.vs1.SignCrrct.vs1.RAF.txt.gz 

#Added the following header to /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/HaemgenRBC2012.dtlesssignif.vs1.annot.vs1.SignCrrct.vs1.RAF.txt.gz
#snp chr pos raf p_RBC n_RBC p_PCV n_PCV p_MCV n_MCV p_MCH n_MCH p_MCHC n_MCHC p_Hb n_Hb annot Z.RBC Z.PCV Z.MCV Z.MCH Z.MCHC Z.Hb mvstat mvp unip

zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/HaemgenRBC2012.dtlesssignif.vs1.annot.vs1.SignCrrct.vs1.RAF.txt.gz | perl -lane 'if ($F[3] > .5) { $F[3] = 1 - $F[3]; $F[17] = $F[17] * -1; $F[18] = $F[18] * -1; $F[19] = $F[19] * -1; $F[20] = $F[20] * -1; $F[21] = $F[21] * -1; $F[22] = $F[22] * -1; }  print join("\t", @F);' | sed 's/raf/maf/g' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/HaemgenRBC2012.dtlesssignif.vs1.annot.vs1.SignCrrct.vs1.MAF.txt.gz

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/GLC.MTedits.ForHamegenRBC2012.AllPheno.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/GLC.MTedits.ForHamegenRBC2012.AllPheno.vs1.SignCrrct.vs1.R

#20160630
cp -p HaemgenRBC2012/Vs1/GLC.MTedits.ForHamegenRBC2012.AllPheno.vs1.SignCrrct.vs1.R HaemgenRBC2012/Vs1/GLC.MTedits.ForHamegenRBC2012.AllPheno.vs1.SignCrrct.vs1.ForCSHLPoster.R



#20170123
#Vs2 -- Running things through version of bmass() code

mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs2
cd /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs2

/mnt/lustre/home/mturchin20/Data/HaemgenRBC2012/HaemgenRBC2012.Table1.Edited.MarkerChrBP.txt
/mnt/gluster/data/external_public_supp/HaemgenRBC2012/

RBC MCV PCV MCH Hb MCHC

for i in `ls -lrt /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_*.wHapMap22.txt.gz | awk '{ print $9 }'`; do echo $i; zcat $i | perl -lane 'print $#F;' | sort | uniq -c; zcat $i | grep , | wc; zcat $i | grep NA | wc; zcat $i | sed 's/_/ /g' | awk '{ print $13; }' | grep -E '^0$|^1$|^\\.$' | sort | uniq -c; done
for i in `ls -lrt /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_*.wHapMap22.txt.gz | awk '{ print $9 }'`; do echo $i; zcat $i | perl -lane 'print $#F;' | sort | uniq -c; zcat $i | sed 's/_/ /g'| grep -v NA | awk '{ if ($11 != $4) { print $0 } }' | wc; zcat $i | sed 's/_/ /g' | awk '{ print $3 "_" $4 }' | sed 's/chr//g' | sort | uniq | wc; zcat $i | sed 's/_/ /g' | awk '{ print $3 "_" $4 }' | sed 's/chr//g' | sort | uniq -d | wc; zcat $i | sed 's/_/ /g' | awk '{ print $10 "_" $11 }' | sed 's/chr//g' | sort | uniq -d | wc; zcat $i | grep , | wc; zcat $i | grep NA | wc; zcat $i | sed 's/_/ /g' | awk '{ print $13; }' | grep -E '^0$|^1$|^\\.$' | sort | uniq -c; done



need to also make sure AF is matched up with effect allele since that might not be the case -- drop those don't match for either eg ignoring strandedness issue?


zcat /data/external_public/GlobalLipid/jointGwasMc_HDL.txt.gz | sed 's/:/ /g' | sed 's/chr//g' | awk '{ print $1 "\t" $2 "\t" $12 "\t" $6 "\t" $7 "\t" $8 "\t" $11 "\t" $10 }' | grep -v hg | grep -v rs | perl -lane 'if ($F[5] > 0) { $F[5] = "+"; } elsif ($F[5] < 0) { $F[5] = "-"; } else { $F[5] = 0;} if ($F[5] eq "-") { ($F[3], $F[4]) = ($F[4], $F[3]); $F[5] = "+"; $F[2] = 1 - $F[2]; } print join("\t", @F); ' | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_HDL.IncAllele.bmass.formatted.txt.gz

zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_RBC.wHapMap22.IncAllele.txt.gz 
zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_MCV.wHapMap22.MatchedToRBC.IncAllele.txt.gz 
zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_PCV.wHapMap22.MatchedToRBC.IncAllele.txt.gz 
zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_MCH.wHapMap22.MatchedToRBC.IncAllele.txt.gz
zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_Hb.wHapMap22.MatchedToRBC.IncAllele.txt.gz
zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_MCHC.wHapMap22.MatchedToRBC.IncAllele.txt.gz

[  mturchin20@spudling89  ~]$zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_RBC.wHapMap22.txt.gz | head -n 10
SNP     Pval    Chr     Position        Effect_Allele   Non_Effect_Allele       Sample_Size     Beta    SE      ChrBPAFInfo
rs9373124       5.11e-97        6       135464902       T       C       53337   0.0484  0.0024  chr6_135464902_T_0.78_C_0.22
rs7776054       3.51e-96        6       135460609       A       G       53403   0.0479  0.0024  chr6_135460609_A_0.78_G_0.22
rs4895440       3.75e-94        6       135468251       A       T       53400   0.0459  0.0023  chr6_135468251_A_0.767_T_0.233
rs9402686       4.36e-94        6       135469510       A       G       53400   -0.0469 0.0024  chr6_135469510_G_0.783_A_0.217

#Opting to use the coordinates provided with the file and not the coordinates from HapMap22 

join <(zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_RBC.wHapMap22.txt.gz | sed 's/_/ /g' | awk '{ print $3 "_" $4 }' | sed 's/chr//g' | sort | uniq -u | grep -v , | grep -v NA | grep -v -E 'SE|ChrBP') <(zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_RBC.wHapMap22.txt.gz | sed 's/_/ /g' | grep -v , | grep -v NA | perl -lane 'my $MAF = NA2; if ($F[11] eq $F[4]) { $MAF = $F[12]; } if ($F[13] eq $F[4]) { $MAF = $F[14]; } push(@F, $MAF); print join("\t", @F);' | awk '{ print $3 "\t" $4 "\t" $16 "\t" $5 "\t $6 "\t"  





Chr     BP      MAF     A1      A2      Direction       pValue  N
10      10000134        0.449   G       A       +       0.70    253213
10      10000264        0.436   T       C       +       0.70    253213


join <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | awk '{ print $9 }' | grep -v , | grep -v NA | grep -v cox | grep -v ChrBP | sort | uniq -u) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | sed 's/_/ /g' | awk '{ print $9 "\t" $10 "\t" $4 "\t" $2 "\t" $3 "\t" $5 "\t" $7 "\t" $8 }' | perl -lane 'if ($F[5] > 0) { $F[5] = "+"; } elsif ($F[5] < 0) { $F[5] = "-"; } else { $F[5] = 0;} if ($F[5] eq "-") { ($F[3], $F[4]) = ($F[4], $F[3]); $F[5] = "+"; $F[2] = 1 - $F[2]; } print join("\t", @F); ' | grep -v Freq | awk '{ print $1 "_" $2 "\t" $0 }' | sort -k 1,1) | perl -lane 'splice(@F, 0, 1); print join("\t", @F);' | perl -lane 'if (($F[2] ne "NA") && ($F[2] > 0) && ($F[2] < 1)) { print join("\t", @F); }' | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | perl -lane 'if ($F[0] eq "X") { $F[0] = 23; } if ($F[0] eq "Y") { $F[0] = 24; } print join("\t", @F); ' | gzip > /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.noComma.noNA.noDups.noFixed.IncAllele.bmass.formatted.txt.gz  

zcat /data/external_public/GlobalLipid/jointGwasMc_HDL.txt.gz | sed 's/:/ /g' | sed 's/chr//g' | awk '{ print $1 "\t" $2 "\t" $12 "\t" $6 "\t" $7 "\t" $8 "\t" $11 "\t" $10 }' | grep -v hg | grep -v rs | perl -lane 'if ($F[5] > 0) { $F[5] = "+"; } elsif ($F[5] < 0) { $F[5] = "-"; } else { $F[5] = 0;} if ($F[5] eq "-") { ($F[3], $F[4]) = ($F[4], $F[3]); $F[5] = "+"; $F[2] = 1 - $F[2]; } print join("\t", @F); ' | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_HDL.IncAllele.bmass.formatted.txt.gz
zcat /data/external_public/GlobalLipid/jointGwasMc_LDL.txt.gz | sed 's/:/ /g' | sed 's/chr//g' | awk '{ print $1 "\t" $2 "\t" $12 "\t" $6 "\t" $7 "\t" $8 "\t" $11 "\t" $10 }' | grep -v hg | grep -v rs | perl -lane 'if ($F[5] > 0) { $F[5] = "+"; } elsif ($F[5] < 0) { $F[5] = "-"; } else { $F[5] = 0;} if ($F[5] eq "-") { ($F[3], $F[4]) = ($F[4], $F[3]); $F[5] = "+"; $F[2] = 1 - $F[2]; } print join("\t", @F); ' | cat <(echo -e "Chr\tBP\tMAF\tA1\tA2\tDirection\tpValue\tN") - | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_LDL.IncAllele.bmass.formatted.txt.gz

join <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_HDL.IncAllele.bmass.formatted.txt.gz | awk '{ print $1 "_" $2 "\t" $4 }' | grep -v NA | sort -g -k 1,1) <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_LDL.IncAllele.bmass.formatted.txt.gz | awk '{ print $1 "_" $2 "\t" $0 }' | grep -v NA | sort -g -k 1,1) | perl -lane 'if ($F[1] eq $F[6]) { ($F[5], $F[6]) = ($F[6], $F[5]); $F[4] = 1 - $F[4]; if ($F[7] eq "-") { $F[7] = "+"; } elsif ($F[7] eq "+") { $F[7] = "-"; } else { $PH = 1; } } elsif ($F[1] eq $F[5]) { $PH = 1; } else { push(@F, "NoAlleleMatch"); } print join("\t", @F[2..$#F]);' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_LDL.IncAllele.bmass.formatted.MatchedToHDL.txt.gz 
join <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_HDL.IncAllele.bmass.formatted.txt.gz | awk '{ print $1 "_" $2 "\t" $4 }' | grep -v NA | sort -g -k 1,1) <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TG.IncAllele.bmass.formatted.txt.gz | awk '{ print $1 "_" $2 "\t" $0 }' | grep -v NA | sort -g -k 1,1) | perl -lane 'if ($F[1] eq $F[6]) { ($F[5], $F[6]) = ($F[6], $F[5]); $F[4] = 1 - $F[4]; if ($F[7] eq "-") { $F[7] = "+"; } elsif ($F[7] eq "+") { $F[7] = "-"; } else { $PH = 1; } } elsif ($F[1] eq $F[5]) { $PH = 1; } else { push(@F, "NoAlleleMatch"); } print join("\t", @F[2..$#F]);' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TG.IncAllele.bmass.formatted.MatchedToHDL.txt.gz 


zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.tbl.gz | perl -lane 'if ($F[5] ne "0") { print join("\t", @F); }' | gzip > /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.No0Dir.tbl.gz
zcat /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.MatchedToHDL.tbl.gz | grep -v NoAlleleMatch | perl -lane 'if ($F[5] ne "0") { print join("\t", @F); }' | gzip > /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.MatchedToHDL.DropNoMatch.No0Dir.tbl.gz


library("devtools"); devtools::load_all("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/bmass");
HDL <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_HDL.IncAllele.bmass.formatted.txt.gz", header=T);
LDL <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_LDL.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.txt.gz", header=T);
TG <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TG.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.txt.gz", header=T);
TC <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/jointGwasMc_TC.IncAllele.bmass.formatted.MatchedToHDL.DropNoMatch.txt.gz", header=T);
GWASsnps <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.ChrBP.txt", header=F);
names(GWASsnps) <- c("Chr", "BP")
#bmassOutput1 <- bmass(c("HDL", "LDL", "TG", "TC"), GWASsnps=GWASsnps, NminThreshold = 1, bmassSeedValue=150);
#print(bmassOutput1);" > shana1.txt
DataSources = c("HDL", "LDL", "TG", "TC"); NminThreshold = 50000; bmassSeedValue=150;
ExpectedColumnNames=c("Chr", "BP", "MAF", "Direction", "pValue", "N"); GWASsnps_AnnotateWindow = 5e5; SNPMarginalUnivariateThreshold = 1e-6; SNPMarginalMultivariateThreshold = 1e-6; SigmaAlphas = c(0.005,0.0075,0.01,0.015,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.15); ProvidedPriors=NULL; UseFlatPriors=FALSE; PruneMarginalSNPs=TRUE; PruneMarginalSNPs_bpWindow=5e5;

#For all
#ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.txt

bmassOutput1 <- bmass(c("HDL", "LDL", "TG", "TC"), GWASsnps=GWASsnps, NminThreshold = 50000, PrintMergedData=FALSE, bmassSeedValue=150);
bmassOutput2 <- bmass(c("HDL", "LDL", "TG", "TC"), GWASsnps=GWASsnps, NminThreshold = 50000, PrintMergedData=FALSE, PruneMarginalSNPs=FALSE, bmassSeedValue=150);
bmassOutput3 <- bmass(c("HDL", "LDL", "TG", "TC"), GWASsnps=GWASsnps, NminThreshold = 50000, PrintMergedData=TRUE, PruneMarginalSNPs=FALSE, bmassSeedValue=150);

write.table(bmassOutput1$NewSNPs$SNPs, file="/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.txt", quote=FALSE, row.names=FALSE)
write.table(bmassOutput1$MergedDataSources, file="/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/bmass.GlobalLipids2013.Vs2.MergedDataSources.vs1.txt", quote=FALSE, row.names=FALSE)

system("gzip /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/bmass.GlobalLipids2013.Vs2.MergedDataSources.vs1.txt")



~~~
[  mturchin20@spudling89  ~]$zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_MCH.wHapMap22.txt.gz | sed 's/_/ /g' | awk '{ print $10 "_" $11 }' | sed 's/chr//g' | sort | uniq -c | awk '{ if ($1 > 1) { print $0 } }' | head -n 10
      2 15_46489749
      2 17_28342660
 126531 NA_
[  mturchin20@spudling89  ~]$zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_RBC.wHapMap22.txt.gz | sed 's/_/ /g' | grep -v NA | awk '{ if ($11 != $4) { print $0 } } ' | head -n 10
SNP     Pval    Chr     Position        Effect Allele   Non Effect Allele       Sample Size     Beta    SE      ChrBPAFInfo
rs1474827       0.002994        6       134886016       A       T       11607   -0.012  0.0045  chr6 134886011 T 1 A 0
rs2294949       0.03716 1       60087104        T       G       11607   0.0152  0.0069  chr1 60087106 A 1 C 0
rs17006223      0.08805 3       19760676        A       G       11607   0.0219  0.0116  chr3 19760677 A 0.983 G 0.017
rs4424451       0.1173  1       177620064       T       G       11607   -0.0069 0.0042  chr1 177620062 G 1 T 0
rs2905608       0.1271  5       137055936       T       C       11607   -0.0053 0.0041  chr5 137055928 T 0 C 1
rs10241692      0.1411  7       78789904        A       G       14861   -0.0106 0.0068  chr7 78789901 A 0 G 1
rs11004058      0.1788  10      55515352        A       G       11607   -0.0014 0.0063  chr10 55515351 A 0.982 G 0.018
rs13360701      0.1834  5       42038620        T       C       11607   -0.0129 0.009   chr5 42038619 T 1 C 0
rs1508337       0.1923  4       77501888        A       G       11607   -0.0052 0.0046  chr4 77501890 A 1 G 0
[  mturchin20@spudling89  ~]$for i in `ls -lrt /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_*.wHapMap22.txt.gz | awk '{ print $9 }'`; do echo $i; zcat $i | perl -lane 'print $#F;' | sort | uniq -c; zcat $i | sed 's/_/ /g'| grep -v NA | awk '{ if ($11 != $4) { print $0 } }' | wc; zcat $i | sed 's/_/ /g' | awk '{ print $3 "_" $4 }' | sed 's/chr//g' | sort | uniq | wc; zcat $i | sed 's/_/ /g' | awk '{ print $3 "_" $4 }' | sed 's/chr//g' | sort | uniq -d | wc; zcat $i | sed 's/_/ /g' | awk '{ print $10 "_" $11 }' | sed 's/chr//g' | sort | uniq -d | wc; zcat $i | grep , | wc; zcat $i | grep NA | wc; zcat $i | sed 's/_/ /g' | awk '{ print $13; }' | grep -E '^0$|^1$|^\\.$' | sort | uniq -c; done
/mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_RBC.wHapMap22.txt.gz
2589455 9
     36     539    2809
2581244 2581244 30056328
   8210    8210   94454
      3       3      28
      0       0       0
 126536 1265360 7085616
   1101 0
   4505 1
/mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_MCV.wHapMap22.txt.gz
2591133 9
    140    2099   11558
2582391 2582391 30069730
   8740    8740  100520
      3       3      28
      0       0       0
 127878 1278780 7175749
   1102 0
   4488 1
/mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_PCV.wHapMap22.txt.gz
2591080 9
    140    2099   11558
2582337 2582337 30069133
   8741    8741  100525
      3       3      28
      0       0       0
 127910 1279100 7173168
   1101 0
   4503 1
/mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_MCH.wHapMap22.txt.gz
2586785 9
     36     539    2806
2578575 2578575 30025108
   8209    8209   94452
      3       3      28
      0       0       0
 126531 1265310 7094661
   1096 0
   4453 1
/mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_Hb.wHapMap22.txt.gz
2593079 9
    140    2099   11545
2584335 2584335 30092457
   8742    8742  100537
      3       3      28
      0       0       0
 127915 1279150 7167398
   1105 0
   4543 1
/mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_MCHC.wHapMap22.txt.gz
2588876 9
    140    2099   11515
2580132 2580132 30043292
   8742    8742  100539
      3       3      28
      0       0       0
 127903 1279030 7169547
   1100 0
   4472 1




~~~









##HaemgenRBC2016
##20170123

mkdir /mnt/gluster/data/external_private_supp/HaemgenRBC2016
cd /mnt/gluster/data/external_private_supp/HaemgenRBC2016
scp -p mturchin20@wolfy.uchicago.edu:/Users/mturchin20/TempData/HaemgenRBC2016/* .
gzip *
















##GPC2012 -- ##dbSNP=provided, GWAS 2 at <5e-8, N=17375
##20150812

mkdir /mnt/gluster/data/external_public_supp/GPC2012

cd /mnt/gluster/data/external_public_supp/GPC2012

wget https://www.dropbox.com/s/df3bifmk238ks2y/GPC-1.BigFiveNEO.zip?dl=0

mv /mnt/gluster/data/external_public_supp/GPC2012/QdLIzcMSCb8NbDeKnhIY5zAj87L7urU2kJxmg9OYJhuf8H5FkA4FlyWqS3mKDHQe /mnt/gluster/data/external_public_supp/GPC2012/GPC-1.BigFiveNEO.zip

unzip /mnt/gluster/data/external_public_supp/GPC2012/GPC-1.BigFiveNEO.zip

gzip /mnt/gluster/data/external_public_supp/GPC2012/*full*

#Info below from /mnt/gluster/data/external_public_supp/GPC2012/ReadmeGPC-1.pdf
~~~
The     files   contain the     following       information     (columns):      
SNPID   CHR     BP      A1      A2      BETA    SE      PVALUE  INFO    NCOH    MAF     

SNPID   rs-number    of      the     SNP     
CHR     chromosome      number  on      which   the     SNP     is      located (build  36,     release 22)     
BP      base    pair    position        of      the     SNP     (build  36,     release 22)     
A1      effect  allele  of      the     SNP     
A2      non-ÃÂ­?<80><90>effect   allele  of      the     SNP   
BETA    pooled  effect  size    (unstandardized regression      coefficient)    
SE      standard        error   of      the     pooled  effect  size    
PVALUE  p-ÃÂ­?<80><90>value      associated      with    the     pooled  effect  size    
INFO    average info    across  cohorts (INFO,  PROPER_INFO     or      R-SQUARED)   
NCOH    number  of      cohorts for     which   SNP     association     results are     available       
MAF     minor   allele  frequency       of      the     SNP     in      the     HAPMAP  reference       set
~~~

~~~
[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/GPC2012]$for i in `ls -lrt | awk '{ print $9 }' | grep txt.gz`; do echo $i; zcat $i | head -n 5; done
GPC-1.NEO-NEUROTICISM.full.txt.gz
rs3121561       1       980243  t       c       .048    .1644   .7703   .527    8       .267
rs3813193       1       988364  c       g       .035    .1552   .8217   .784    9       .15
rs3934834       1       995669  t       c       -.017   .1541   .912    .9      9       .119
rs3766193       1       1007033 c       g       .0572   .1173   .6257   .672    9       .431
rs3766192       1       1007060 t       c       -.0399  .1126   .7231   .778    9       .424
GPC-1.NEO-EXTRAVERSION.full.txt.gz
rs3121561       1       980243  t       c       -.0071  .135    .9579   .53     8       .267
rs3813193       1       988364  c       g       .0096   .1269   .9394   .782    9       .15
rs3934834       1       995669  t       c       .0311   .1261   .8052   .901    9       .119
rs3766193       1       1007033 c       g       .0075   .0946   .9365   .672    9       .431
rs3766192       1       1007060 t       c       -.0011  .0907   .9901   .778    9       .424
GPC-1.NEO-AGREEABLENESS.full.txt.gz
rs3121561       1       980243  t       c       .0256   .1058   .8088   .532    8       .267
rs3813193       1       988364  c       g       -.0898  .1      .3692   .783    9       .15
rs3934834       1       995669  t       c       -.0868  .0993   .3825   .902    9       .119
rs3766193       1       1007033 c       g       .1615   .0765   .03479  .67     9       .431
rs3766192       1       1007060 t       c       -.1241  .0731   .08947  .777    9       .424
GPC-1.NEO-CONSCIENTIOUSNESS.full.txt.gz
rs3121561       1       980243  t       c       -.0144  .1282   .9105   .531    8       .267
rs3813193       1       988364  c       g       .0469   .1211   .6985   .782    9       .15
rs3934834       1       995669  t       c       .0759   .1203   .5282   .9      9       .119
rs3766193       1       1007033 c       g       .0419   .0918   .6485   .671    9       .431
rs3766192       1       1007060 t       c       -.0514  .0879   .5586   .777    9       .424
GPC-1.NEO-OPENNESS.full.txt.gz
rs3121561       1       980243  t       c       -.0259  .1199   .8293   .536    8       .267
rs3813193       1       988364  c       g       -.0019  .1144   .9868   .785    9       .15
rs3934834       1       995669  t       c       .0097   .1139   .9324   .902    9       .119
rs3766193       1       1007033 c       g       .0512   .087    .556    .673    9       .431
rs3766192       1       1007060 t       c       -.0503  .0831   .5451   .779    9       .424
[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/GPC2012]$for i in `ls -lrt | awk '{ print $9 }' | grep txt.gz`; do paste <(zcat $i | wc) <(zcat $i | perl -lane 'print $#F;' | sort | uniq -c) | xargs echo $i; done
GPC-1.NEO-NEUROTICISM.full.txt.gz 2305738 25363118 130839072 2305738 10
GPC-1.NEO-EXTRAVERSION.full.txt.gz 2305822 25364042 130837422 2305822 10
GPC-1.NEO-AGREEABLENESS.full.txt.gz 2305461 25360071 130825946 2305461 10
GPC-1.NEO-CONSCIENTIOUSNESS.full.txt.gz 2305682 25362502 130844185 2305682 10
GPC-1.NEO-OPENNESS.full.txt.gz 2305640 25362040 130837603 2305640 10
[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/GPC2012]$for i in `ls -lrt | awk '{ print $9 }' | grep txt.gz`; do echo $i; paste -d @ <(zcat $i | sed 's/E-//g'| perl -lane 'if ($F[0] !~ m/^rs\d+/) { print join("\t", @F); }' | awk '{ print $1 }' | sort  | uniq -c) <(zcat $i | sed 's/E-//g'| perl -lane 'if ($F[1] !~ m/(1|2|3|4|5|6|7|8|9|10|11|12|13|14|15|16|17|18|19|20|21|22|X|Y|M)/) { print join("\t", @F); }' | awk '{ print $2 }' | sort  | uniq -c) <(zcat $i | sed 's/E-//g'| perl -lane 'if ($F[2] =~ m/[^\d\.]/) { print join("\t", @F); }' | awk '{ print $3 }' | sort  | uniq -c) <(zcat $i | sed 's/E-//g'| perl -lane 'if ($F[7] =~ m/[^\d\.]/) { print join("\t", @F); }' | awk '{ print $8 }' | sort  | uniq -c) <(zcat $i | sed 's/E-//g'| perl -lane 'if ($F[10] =~ m/[^\d\.]/) { print join("\t", @F); }' | awk '{ print $11 }' | sort  | uniq -c) | sed 's/@/@ /g'| column -s $'@' -t ; done
GPC-1.NEO-NEUROTICISM.full.txt.gz
GPC-1.NEO-EXTRAVERSION.full.txt.gz
GPC-1.NEO-AGREEABLENESS.full.txt.gz
GPC-1.NEO-CONSCIENTIOUSNESS.full.txt.gz
GPC-1.NEO-OPENNESS.full.txt.gz
[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/GPC2012]$for i in `ls -lrt | awk '{ print $9 }' | grep txt.gz`; do echo $i; zcat $i | perl -lane 'print join("\t", @F);' | R -q -e "library(data.table); Data1 <- fread('file:///dev/stdin', header=FALSE); quantile(Data1\$V11); quantile(Data1\$V8); png(\""$i".QCCheck.vs1.png\", height=2000, width=2000, res=200); par(mfrow=c(2,2)); hist(Data1\$V11, breaks=100); hist(Data1\$V8, breaks=100); expP <- -log10(c(1:length(Data1\$V8["\!"is.na(Data1\$V8)])) / (length(Data1\$V8["\!"is.na(Data1\$V8)])+1)); plot(expP, -log10(sort(Data1\$V8["\!"is.na(Data1\$V8)])), xlab=\"Expected p-Values\", ylab=\"Observed p-Values\"); abline(0, 1, col=\"RED\"); dev.off();"; done
GPC-1.NEO-NEUROTICISM.full.txt.gz
> library(data.table); Data1 <- fread('file:///dev/stdin', header=FALSE); quantile(Data1$V11); quantile(Data1$V8); png("GPC-1.NEO-NEUROTICISM.full.txt.gz.QCCheck.vs1.pn
g", height=2000, width=2000, res=200); par(mfrow=c(2,2)); hist(Data1$V11, breaks=100); hist(Data1$V8, breaks=100); expP <- -log10(c(1:length(Data1$V8[!is.na(Data1$V8)])
) / (length(Data1$V8[!is.na(Data1$V8)])+1)); plot(expP, -log10(sort(Data1$V8[!is.na(Data1$V8)])), xlab="Expected p-Values", ylab="Observed p-Values"); abline(0, 1, col=
"RED"); dev.off();
Read 2305738 rows and 11 (of 11) columns from 0.122 GB file in 00:00:19
0%   25%   50%   75%  100%
0.010 0.116 0.225 0.358 0.500
0%       25%       50%       75%      100%
2.072e-06 2.446e-01 4.947e-01 7.484e-01 1.000e+00
null device
1
>
>
GPC-1.NEO-EXTRAVERSION.full.txt.gz
> library(data.table); Data1 <- fread('file:///dev/stdin', header=FALSE); quantile(Data1$V11); quantile(Data1$V8); png("GPC-1.NEO-EXTRAVERSION.full.txt.gz.QCCheck.vs1.png", height=2000, width=2000, res=200); par(mfrow=c(2,2)); hist(Data1$V11, breaks=100); hist(Data1$V8, breaks=100); expP <- -log10(c(1:length(Data1$V8[!is.na(Data1$V8)])) / (length(Data1$V8[!is.na(Data1$V8)])+1)); plot(expP, -log10(sort(Data1$V8[!is.na(Data1$V8)])), xlab="Expected p-Values", ylab="Observed p-Values"); abline(0, 1, col="RED"); dev.off();
Read 2305822 rows and 11 (of 11) columns from 0.122 GB file in 00:00:19
0%   25%   50%   75%  100%
0.010 0.116 0.225 0.358 0.500
0%       25%       50%       75%      100%
3.113e-07 2.475e-01 4.968e-01 7.486e-01 1.000e+00
null device
1
>
>
GPC-1.NEO-AGREEABLENESS.full.txt.gz
> library(data.table); Data1 <- fread('file:///dev/stdin', header=FALSE); quantile(Data1$V11); quantile(Data1$V8); png("GPC-1.NEO-AGREEABLENESS.full.txt.gz.QCCheck.vs1.png", height=2000, width=2000, res=200); par(mfrow=c(2,2)); hist(Data1$V11, breaks=100); hist(Data1$V8, breaks=100); expP <- -log10(c(1:length(Data1$V8[!is.na(Data1$V8)])) / (length(Data1$V8[!is.na(Data1$V8)])+1)); plot(expP, -log10(sort(Data1$V8[!is.na(Data1$V8)])), xlab="Expected p-Values", ylab="Observed p-Values"); abline(0, 1, col="RED"); dev.off();
Read 2305461 rows and 11 (of 11) columns from 0.122 GB file in 00:00:19
0%   25%   50%   75%  100%
0.010 0.116 0.225 0.358 0.500
0%       25%       50%       75%      100%
1.228e-06 2.486e-01 4.998e-01 7.509e-01 1.000e+00
null device
1
>
>
GPC-1.NEO-CONSCIENTIOUSNESS.full.txt.gz
> library(data.table); Data1 <- fread('file:///dev/stdin', header=FALSE); quantile(Data1$V11); quantile(Data1$V8); png("GPC-1.NEO-CONSCIENTIOUSNESS.full.txt.gz.QCCheck.vs1.png", height=2000, width=2000, res=200); par(mfrow=c(2,2)); hist(Data1$V11, breaks=100); hist(Data1$V8, breaks=100); expP <- -log10(c(1:length(Data1$V8[!is.na(Data1$V8)])) / (length(Data1$V8[!is.na(Data1$V8)])+1)); plot(expP, -log10(sort(Data1$V8[!is.na(Data1$V8)])), xlab="Expected p-Values", ylab="Observed p-Values"); abline(0, 1, col="RED"); dev.off();
Read 2305682 rows and 11 (of 11) columns from 0.122 GB file in 00:00:19
0%   25%   50%   75%  100%
0.010 0.116 0.225 0.358 0.500
0%       25%       50%       75%      100%
4.908e-08 2.445e-01 4.956e-01 7.491e-01 1.000e+00
null device
1
>
>
GPC-1.NEO-OPENNESS.full.txt.gz
> library(data.table); Data1 <- fread('file:///dev/stdin', header=FALSE); quantile(Data1$V11); quantile(Data1$V8); png("GPC-1.NEO-OPENNESS.full.txt.gz.QCCheck.vs1.png", height=2000, width=2000, res=200); par(mfrow=c(2,2)); hist(Data1$V11, breaks=100); hist(Data1$V8, breaks=100); expP <- -log10(c(1:length(Data1$V8[!is.na(Data1$V8)])) / (length(Data1$V8[!is.na(Data1$V8)])+1)); plot(expP, -log10(sort(Data1$V8[!is.na(Data1$V8)])), xlab="Expected p-Values", ylab="Observed p-Values"); abline(0, 1, col="RED"); dev.off();
Read 2305640 rows and 11 (of 11) columns from 0.122 GB file in 00:00:19
0%   25%   50%   75%  100%
0.010 0.116 0.225 0.358 0.500
0%       25%       50%       75%      100%
2.789e-08 2.414e-01 4.923e-01 7.471e-01 1.000e+00
null device
1
>
>
~~~

mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GPC2012
mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GPC2012/Vs1

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/process.MTedits.ForHaemgenRBC2012.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GPC2012/Vs1/process.MTedits.GPC2012.vs1.R

~~~
[  mturchin20@spudling26  /mnt/gluster/data/external_public_supp/GPC2012]$for i in `ls -lrt | awk '{ print $9 }' | grep full.txt.gz$`; do echo $i; zcat $i | perl -lane 'my $SEovrBet; if ($F[5] != 0) { $SEovrBet = $F[6]/ $F[5]; } else { $SEovrBet = $F[6]; } print $F[5], "\t", $F[6], "\t", abs($SEovrBet);' | sort -k 3,3 -rg | head -n 10; done
GPC-1.NEO-NEUROTICISM.full.txt.gz
-.0001  .3947   3947
-.0001  .3911   3911
-.0001  .3899   3899
-.0001  .3603   3603
.0001   .3394   3394
.0001   .3316   3316
-.0001  .3301   3301
-.0001  .3288   3288
-.0001  .299    2990
-.0001  .2979   2979
GPC-1.NEO-EXTRAVERSION.full.txt.gz
-.0001  .3082   3082
.0001   .2991   2991
-.0001  .2963   2963
-.0001  .2939   2939
-.0001  .2719   2719
-.0001  .2702   2702
-.0001  .2697   2697
-.0001  .2692   2692
-.0001  .2655   2655
.0001   .2576   2576
GPC-1.NEO-AGREEABLENESS.full.txt.gz
.0001   .2412   2412
.0001   .2397   2397
.0001   .2373   2373
.0001   .232    2320
.0001   .2315   2315
-.0001  .2283   2283
-.0001  .2269   2269
.0001   .2262   2262
.0001   .2239   2239
-.0001  .2222   2222
GPC-1.NEO-CONSCIENTIOUSNESS.full.txt.gz
-.0001  .2789   2789
.0001   .2743   2743
-.0001  .2708   2708
-.0001  .2627   2627
-.0001  .2528   2528
.0001   .2491   2491
-.0001  .2433   2433
-.0001  .2423   2423
.0001   .2416   2416
.0001   .2397   2397
GPC-1.NEO-OPENNESS.full.txt.gz
.0001   .2839   2839
-.0001  .2792   2792
-.0001  .2662   2662
.0001   .2639   2639
-.0001  .2354   2354
-.0001  .2331   2331
.0001   .2222   2222
.0001   .2205   2205
.0001   .217    2170
-.0001  .2159   2159
~~~

3947
3082
2412
2789
2839

zcat /mnt/gluster/data/external_public_supp/GPC2012/GPC-1.NEO-NEUROTICISM.full.txt.gz | awk '{ print $2, "\t", $3, "\t", $1, "\t", $11, "\t", $8, "\t17375" }' | gzip > /mnt/gluster/data/external_public_supp/GPC2012/GPC-1.NEO-NEUROTICISM.full.formatted.maxN.txt.gz 
zcat /mnt/gluster/data/external_public_supp/GPC2012/GPC-1.NEO-EXTRAVERSION.full.txt.gz | awk '{ print $2, "\t", $3, "\t", $1, "\t", $11, "\t", $8, "\t17375" }' | gzip > /mnt/gluster/data/external_public_supp/GPC2012/GPC-1.NEO-EXTRAVERSION.full.formatted.maxN.txt.gz 
zcat /mnt/gluster/data/external_public_supp/GPC2012/GPC-1.NEO-AGREEABLENESS.full.txt.gz | awk '{ print $2, "\t", $3, "\t", $1, "\t", $11, "\t", $8, "\t17375" }' | gzip > /mnt/gluster/data/external_public_supp/GPC2012/GPC-1.NEO-AGREEABLENESS.full.formatted.maxN.txt.gz 
zcat /mnt/gluster/data/external_public_supp/GPC2012/GPC-1.NEO-CONSCIENTIOUSNESS.full.txt.gz | awk '{ print $2, "\t", $3, "\t", $1, "\t", $11, "\t", $8, "\t17375" }' | gzip > /mnt/gluster/data/external_public_supp/GPC2012/GPC-1.NEO-CONSCIENTIOUSNESS.full.formatted.maxN.txt.gz 
zcat /mnt/gluster/data/external_public_supp/GPC2012/GPC-1.NEO-OPENNESS.full.txt.gz | awk '{ print $2, "\t", $3, "\t", $1, "\t", $11, "\t", $8, "\t17375" }' | gzip > /mnt/gluster/data/external_public_supp/GPC2012/GPC-1.NEO-OPENNESS.full.formatted.maxN.txt.gz 

zcat /mnt/gluster/data/external_public_supp/GPC2012/GPC-1.NEO-NEUROTICISM.full.txt.gz | perl -lane 'my $scaledSE; if ($F[5] == 0) { $scaledSE = $F[6] / 3947; } else { $scaledSE = ($F[6] / abs($F[5])) / 3947; } push(@F, (17375 * (1 - $scaledSE))); print join("\t", @F);' | awk '{ print $2, "\t", $3, "\t", $1, "\t", $11, "\t", $8, "\t", $12 }' | gzip > /mnt/gluster/data/external_public_supp/GPC2012/GPC-1.NEO-NEUROTICISM.full.formatted.scaledSEN.txt.gz
zcat /mnt/gluster/data/external_public_supp/GPC2012/GPC-1.NEO-EXTRAVERSION.full.txt.gz | perl -lane 'my $scaledSE; if ($F[5] == 0) { $scaledSE = $F[6] / 3082; } else { $scaledSE = ($F[6] / abs($F[5])) / 3082; } push(@F, (17375 * (1 - $scaledSE))); print join("\t", @F);' | awk '{ print $2, "\t", $3, "\t", $1, "\t", $11, "\t", $8, "\t", $12 }' | gzip > /mnt/gluster/data/external_public_supp/GPC2012/GPC-1.NEO-EXTRAVERSION.full.formatted.scaledSEN.txt.gz
zcat /mnt/gluster/data/external_public_supp/GPC2012/GPC-1.NEO-AGREEABLENESS.full.txt.gz | perl -lane 'my $scaledSE; if ($F[5] == 0) { $scaledSE = $F[6] / 2412; } else { $scaledSE = ($F[6] / abs($F[5])) / 2412; } push(@F, (17375 * (1 - $scaledSE))); print join("\t", @F);' | awk '{ print $2, "\t", $3, "\t", $1, "\t", $11, "\t", $8, "\t", $12 }' | gzip > /mnt/gluster/data/external_public_supp/GPC2012/GPC-1.NEO-AGREEABLENESS.full.formatted.scaledSEN.txt.gz
zcat /mnt/gluster/data/external_public_supp/GPC2012/GPC-1.NEO-CONSCIENTIOUSNESS.full.txt.gz | perl -lane 'my $scaledSE; if ($F[5] == 0) { $scaledSE = $F[6] / 2789; } else { $scaledSE = ($F[6] / abs($F[5])) / 2789; } push(@F, (17375 * (1 - $scaledSE))); print join("\t", @F);' | awk '{ print $2, "\t", $3, "\t", $1, "\t", $11, "\t", $8, "\t", $12 }' | gzip > /mnt/gluster/data/external_public_supp/GPC2012/GPC-1.NEO-CONSCIENTIOUSNESS.full.formatted.scaledSEN.txt.gz
zcat /mnt/gluster/data/external_public_supp/GPC2012/GPC-1.NEO-OPENNESS.full.txt.gz | perl -lane 'my $scaledSE; if ($F[5] == 0) { $scaledSE = $F[6] / 2839; } else { $scaledSE = ($F[6] / abs($F[5])) / 2839; } push(@F, (17375 * (1 - $scaledSE))); print join("\t", @F);' | awk '{ print $2, "\t", $3, "\t", $1, "\t", $11, "\t", $8, "\t", $12 }' | gzip > /mnt/gluster/data/external_public_supp/GPC2012/GPC-1.NEO-OPENNESS.full.formatted.scaledSEN.txt.gz






































##23andMe -- ##dbSNP, GWAS, N=?
##2015____


##WTCCC2007 -- ##dbSNP, GWAS, N=?
##20150722

#mkdir /data/internal_restricted/carbonetto_2012_turchin_2015_wtccc

mkdir /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015
mkdir /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/
mkdir /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/
mkdir /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/combined_controls
rm /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/combined_controls/snptest_BD_All.txt.gz; for i in `ls -lrt /mnt/gluster/data/external_private_supp/WTCCC/WTCCC_summary_data/7_Diseases/BD/combined_controls/snptest_BD_*.txt | awk '{ print $9 }' | perl -lane 'if ($F[0] =~ m/snptest_BD_(\d+).txt/) { print $1; }'`; do cat /mnt/gluster/data/external_private_supp/WTCCC/WTCCC_summary_data/7_Diseases/BD/combined_controls/snptest_BD_$i.txt | perl -lane 'splice (@F, 2, 0, "'$i'"); print join("\t", @F);' | gzip >> /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/combined_controls/snptest_BD_All.txt.gz; done &
rm /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/combined_controls/snptest_CD_All.txt.gz; for i in `ls -lrt /mnt/gluster/data/external_private_supp/WTCCC/WTCCC_summary_data/7_Diseases/CD/combined_controls/snptest_CD_*.txt | awk '{ print $9 }' | perl -lane 'if ($F[0] =~ m/snptest_CD_(\d+).txt/) { print $1; }'`; do cat /mnt/gluster/data/external_private_supp/WTCCC/WTCCC_summary_data/7_Diseases/CD/combined_controls/snptest_CD_$i.txt | perl -lane 'splice (@F, 2, 0, "'$i'"); print join("\t", @F);' | gzip >> /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/combined_controls/snptest_CD_All.txt.gz; done &
rm /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/combined_controls/snptest_T2D_All.txt.gz; for i in `ls -lrt /mnt/gluster/data/external_private_supp/WTCCC/WTCCC_summary_data/7_Diseases/T2D/combined_controls/snptest_T2D_*.txt | awk '{ print $9 }' | perl -lane 'if ($F[0] =~ m/snptest_T2D_(\d+).txt/) { print $1; }'`; do cat /mnt/gluster/data/external_private_supp/WTCCC/WTCCC_summary_data/7_Diseases/T2D/combined_controls/snptest_T2D_$i.txt | perl -lane 'splice (@F, 2, 0, "'$i'"); print join("\t", @F);' | gzip >> /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/combined_controls/snptest_T2D_All.txt.gz; done &
rm /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/combined_controls/snptest_RA_All.txt.gz; for i in `ls -lrt /mnt/gluster/data/external_private_supp/WTCCC/WTCCC_summary_data/7_Diseases/RA/combined_controls/snptest_RA_*.txt | awk '{ print $9 }' | perl -lane 'if ($F[0] =~ m/snptest_RA_(\d+).txt/) { print $1; }'`; do cat /mnt/gluster/data/external_private_supp/WTCCC/WTCCC_summary_data/7_Diseases/RA/combined_controls/snptest_RA_$i.txt | perl -lane 'splice (@F, 2, 0, "'$i'"); print join("\t", @F);' | gzip >> /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/combined_controls/snptest_RA_All.txt.gz; done &
rm /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/combined_controls/snptest_HT_All.txt.gz; for i in `ls -lrt /mnt/gluster/data/external_private_supp/WTCCC/WTCCC_summary_data/7_Diseases/HT/combined_controls/snptest_HT_*.txt | awk '{ print $9 }' | perl -lane 'if ($F[0] =~ m/snptest_HT_(\d+).txt/) { print $1; }'`; do cat /mnt/gluster/data/external_private_supp/WTCCC/WTCCC_summary_data/7_Diseases/HT/combined_controls/snptest_HT_$i.txt | perl -lane 'splice (@F, 2, 0, "'$i'"); print join("\t", @F);' | gzip >> /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/combined_controls/snptest_HT_All.txt.gz; done &
rm /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/combined_controls/snptest_T1D_All.txt.gz; for i in `ls -lrt /mnt/gluster/data/external_private_supp/WTCCC/WTCCC_summary_data/7_Diseases/T1D/combined_controls/snptest_T1D_*.txt | awk '{ print $9 }' | perl -lane 'if ($F[0] =~ m/snptest_T1D_(\d+).txt/) { print $1; }'`; do cat /mnt/gluster/data/external_private_supp/WTCCC/WTCCC_summary_data/7_Diseases/T1D/combined_controls/snptest_T1D_$i.txt | perl -lane 'splice (@F, 2, 0, "'$i'"); print join("\t", @F);' | gzip >> /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/combined_controls/snptest_T1D_All.txt.gz; done &
rm /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/combined_controls/snptest_CAD_All.txt.gz; for i in `ls -lrt /mnt/gluster/data/external_private_supp/WTCCC/WTCCC_summary_data/7_Diseases/CAD/combined_controls/snptest_CAD_*.txt | awk '{ print $9 }' | perl -lane 'if ($F[0] =~ m/snptest_CAD_(\d+).txt/) { print $1; }'`; do cat /mnt/gluster/data/external_private_supp/WTCCC/WTCCC_summary_data/7_Diseases/CAD/combined_controls/snptest_CAD_$i.txt | perl -lane 'splice (@F, 2, 0, "'$i'"); print join("\t", @F);' | gzip >> /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/combined_controls/snptest_CAD_All.txt.gz; done & 

~~~
[  mturchin20@spudling26  /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/combined_controls]$for i in `ls -lrt | awk '{ print $9
}' | grep snptest_`; do echo $i; zcat $i | head -n 5; done
snptest_BD_All.txt.gz
id      rsid    20      pos     allele1 allele2 average_maximum_posterior_call  controls_AA     controls_AB     controls_BB     controls_NULL   cases_AA        cases_AB
cases_BB        cases_NULL      frequentist_add frequentist_gen bayesian_add    bayesian_gen    sex_frequentist_add     sex_frequentist_gen     good_clustering
SNP_A-2163138   rs17685809      20      17408   A       G       0.999452        151     2571    11571   18      23      347     1491    7       0.334526        0.597571
-0.66192        -0.74883        0.28754 0.582069        1
SNP_A-2152189   rs7344547       20      41499   A       G       0.999627        13525   768     8       10      1780    86      1       1       0.171298        0.38123-
0.309436        -0.372097       0.128086        0.356641        1
SNP_A-2007171   rs6038013       20      56187   A       G       0.99864 13487   767     10      47      1773    84      1       10      0.116081        0.289526       -
0.191341        -0.263467       0.0651527       0.233007        1
SNP_A-2255741   rs6038037       20      57272   C       G       0.996742        1       760     13491   59      0       85      1779    4       0.152923        0.347736
-0.269832       -0.326362       0.119146        0.222081        1
snptest_CD_All.txt.gz
id      rsid    19      pos     allele1 allele2 average_maximum_posterior_call  controls_AA     controls_AB     controls_BB     controls_NULL   cases_AA        cases_AB
cases_BB        cases_NULL      frequentist_add frequentist_gen bayesian_add    bayesian_gen    sex_frequentist_add     sex_frequentist_gen     good_clustering
SNP_A-2260116   rs8105536       19      212033  A       G       0.999244        1798    5135    3664    11      287     857     602     2       0.809669        0.82941-
1.06997 -1.00963        0.483524        0.730203        1
SNP_A-2293731   rs11084928      19      228776  A       G       0.999728        1174    4769    4656    9       181     773     794     0       0.196116        0.433541
-0.70983        -0.831206       0.274287        0.627233        1
SNP_A-4277655   rs1106581       19      244934  C       T       0.99926 6588    3409    592     19      1093    543     107     5       0.971328        0.507299       -
1.016   -0.833665       0.673326        0.205112        1
SNP_A-1848123   rs10418525      19      247458  A       G       0.998399        447     3451    6660    50      72      562     1109    5       0.660506        0.905707
-0.961671       -0.674763       0.738266        0.956593        1
snptest_T2D_All.txt.gz
id      rsid    14      pos     allele1 allele2 average_maximum_posterior_call  controls_AA     controls_AB     controls_BB     controls_NULL   cases_AA        cases_AB
cases_BB        cases_NULL      frequentist_add frequentist_gen bayesian_add    bayesian_gen    sex_frequentist_add     sex_frequentist_gen     good_clustering
SNP_A-2113329   rs4562981       14      19309086        C       T       0.985262        9006    4357    287     605     1211    592     49      72      0.382706       0.315499 -0.794477       -0.885316       0.236762        0.363062        1
SNP_A-2087294   rs4080788       14      19337785        A       C       0.99091 12056   1870    0       329     1638    240     0       46      0.437927        0.437927
-0.697324       -0.760902       0.120296        0.120296        1
SNP_A-1837486   rs1780870       14      19362325        C       G       0.983984        10016   3429    149     661     1348    451     28      97      0.776168       0.239097 -0.903032       -0.808739       0.969887        0.599236        1
SNP_A-1899385   rs2318498       14      19447566        A       G       0.999566        3       448     13789   15      0       42      1879    3       0.0175519      0.0574271        0.532122        0.49189 0.0622369       0.13086 1
snptest_RA_All.txt.gz
id      rsid    21      pos     allele1 allele2 average_maximum_posterior_call  controls_AA     controls_AB     controls_BB     controls_NULL   cases_AA        cases_AB
cases_BB        cases_NULL      frequentist_add frequentist_gen bayesian_add    bayesian_gen    sex_frequentist_add     sex_frequentist_gen     good_clustering
SNP_A-2015094   rs303304        21      9941889 A       G       0.996921        9786    731     11      80      1726    117     2       15      0.365813        0.640009
-0.541533       -0.612505       0.679479        0.584454        1
SNP_A-4270130   rs456706        21      10022975        C       T       0.999336        0       794     9803    11      0       129     1728    3       0.40737 0.40737-0.569989        -0.63511        0.653915        0.653915        1
SNP_A-2007909   rs7510233       21      13291078        C       G       0.998718        10591   0       0       17      1856    0       0       4       1       1      0.0902512        0.32584 1       1       1
SNP_A-4215517   rs3875777       21      13347413        A       C       0.999367        10596   0       0       12      1859    0       0       1       1       1      0.0900234        0.330267        1       1       1
snptest_HT_All.txt.gz
id      rsid    05      pos     allele1 allele2 average_maximum_posterior_call  controls_AA     controls_AB     controls_BB     controls_NULL   cases_AA        cases_AB
cases_BB        cases_NULL      frequentist_add frequentist_gen bayesian_add    bayesian_gen    sex_frequentist_add     sex_frequentist_gen     good_clustering
SNP_A-4236757   rs7704488       05      81949   A       C       0.99813 1299    5345    5592    65      191     890     868     3       0.828643        0.215414       -1.07292 -1.04642        0.683598        0.424409        1
SNP_A-2280280   rs10065373      05      159819  C       T       0.999376        11596   683     7       15      1821    129     1       1       0.070716        0.175719
0.113993        -0.0747838      0.179098        0.413686        1
SNP_A-2025745   rs6555608       05      165712  A       G       0.993881        143     2311    9618    229     29      373     1512    38      0.353652        0.434968
-0.697044       -0.801558       0.635359        0.119975        1
SNP_A-2224179   rs10073501      05      166577  A       G       0.99976 9709    2420    164     8       1518    401     33      0       0.152076        0.28831 -0.44683
-0.611501       0.338506        0.23124 1
snptest_CAD_All.txt.gz
id      rsid    19      pos     allele1 allele2 average_maximum_posterior_call  controls_AA     controls_AB     controls_BB     controls_NULL   cases_AA        cases_AB
cases_BB        cases_NULL      frequentist_add frequentist_gen bayesian_add    bayesian_gen    sex_frequentist_add     sex_frequentist_gen     good_clustering
SNP_A-2260116   rs8105536       19      212033  A       G       0.999234        2105    5914    4268    14      329     925     670     2       0.94435 0.997147       -1.10518 -1.09272        0.996046        0.999921        1
SNP_A-2293731   rs11084928      19      228776  A       G       0.999766        1376    5456    5461    8       218     865     843     0       0.633713        0.865532
-1.03392        -1.04589        0.857118        0.971653        1
SNP_A-4277655   rs1106581       19      244934  C       T       0.999458        7688    3888    705     20      1193    627     104     2       0.859914        0.645813
-1.03044        -0.96573        0.774896        0.889331        1
SNP_A-1848123   rs10418525      19      247458  A       G       0.998498        521     3941    7786    53      89      607     1223    7       0.874153        0.694526
-1.01113        -1.00998        0.786915        0.745379        1
snptest_T1D_All.txt.gz
id      rsid    15      pos     allele1 allele2 average_maximum_posterior_call  controls_AA     controls_AB     controls_BB     controls_NULL   cases_AA        cases_AB
cases_BB        cases_NULL      frequentist_add frequentist_gen bayesian_add    bayesian_gen    sex_frequentist_add     sex_frequentist_gen     good_clustering
SNP_A-2114663   rs12916870      15      18446422        A       G       0.999376        1818    5112    3664    14      333     941     686     3       0.733095       0.936837 -1.08067        -1.08241        0.568209        0.879371        1
SNP_A-2132755   rs7179358       15      18451755        A       G       0.998239        3885    5012    1665    46      743     927     285     8       0.157878       0.340166 -0.674617       -0.8165 0.363951        0.701199        1
SNP_A-4221095   rs4114744       15      18782954        G       T       0.999331        0       0       10598   10      0       0       1962    1       1       1      0.0795725        0.886763        1       1       1
SNP_A-1847162   rs4931971       15      18846941        A       T       0.984268        106     2646    7418    438     16      509     1368    70      0.686473       0.555709 -0.883581       -0.921758       0.539368        0.670874        1
[  mturchin20@spudling26  /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/combined_controls]$for i in `ls -lrt | awk '{ print $9 }' | grep snptest_`; do paste <(zcat $i | wc) <(zcat $i | perl -lane 'print $#F;' | sort | uniq -c) | xargs echo $i; done
snptest_BD_All.txt.gz 437258 9619676 59745573 437258 21
snptest_CD_All.txt.gz 459468 10108296 62504342 459468 21
snptest_T2D_All.txt.gz 459468 10108296 62773784 459468 21
snptest_RA_All.txt.gz 459468 10108296 62496593 459468 21
snptest_HT_All.txt.gz 459468 10108296 62658163 459468 21
snptest_CAD_All.txt.gz 459468 10108296 62638982 459468 21
snptest_T1D_All.txt.gz 459468 10108296 62525644 459468 21
[  mturchin20@spudling26  /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/combined_controls]$for i in `ls -lrt | awk '{ print $9 }' | grep txt.gz`; do echo $i; paste -d @ <(zcat $i | sed 's/e-//g'| perl -lane 'if ($F[1] !~ m/^rs\d+/) { print join("\t", @F); }' | awk '{ print $2 }' | sort  | uniq -c) <(zcat $i | sed 's/e-//g'| perl -lane 'if ($F[2] !~ m/(1|2|3|4|5|6|7|8|9|10|11|12|13|14|15|16|17|18|19|20|21|22|X|Y|M)/) { print join("\t", @F); }' | awk '{ print $3 }' | sort  | uniq -c) <(zcat $i | sed 's/e-//g'| perl -lane 'if ($F[3] =~ m/[^\d\.]/) { print join("\t", @F); }' | awk '{ print $4 }' | sort  | uniq -c) <(zcat $i | sed 's/e-//g'| perl -lane 'if ($F[15] =~ m/[^\d\.]/) { print join("\t", @F); }' | awk '{ print $16 }' | sort  | uniq -c) | sed 's/@/@ /g'| column -s $'@' -t ; done
snptest_BD_All.txt.gz
5685 ---            22 pos        22 frequentist_add
22 rsid
snptest_CD_All.txt.gz
5937 ---            22 pos        22 frequentist_add
22 rsid
snptest_T2D_All.txt.gz
5937 ---            22 pos        22 frequentist_add
22 rsid
snptest_RA_All.txt.gz
5937 ---            22 pos        22 frequentist_add
22 rsid
snptest_HT_All.txt.gz
5937 ---            22 pos        22 frequentist_add
22 rsid
snptest_CAD_All.txt.gz
5937 ---            22 pos        22 frequentist_add
22 rsid
snptest_T1D_All.txt.gz
5937 ---            22 pos        22 frequentist_add
22 rsid
~~~

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/combined_controls/snptest_BD_All.txt.gz | gzip > /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/combined_controls/snptest_BD_All.wHapMap22.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/combined_controls/snptest_CD_All.txt.gz | gzip > /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/combined_controls/snptest_CD_All.wHapMap22.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/combined_controls/snptest_T2D_All.txt.gz | gzip > /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/combined_controls/snptest_T2D_All.wHapMap22.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/combined_controls/snptest_RA_All.txt.gz | gzip > /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/combined_controls/snptest_RA_All.wHapMap22.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/combined_controls/snptest_HT_All.txt.gz | gzip > /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/combined_controls/snptest_HT_All.wHapMap22.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/combined_controls/snptest_T1D_All.txt.gz | gzip > /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/combined_controls/snptest_T1D_All.wHapMap22.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/combined_controls/snptest_CAD_All.txt.gz | gzip > /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/combined_controls/snptest_CAD_All.wHapMap22.txt.gz &

#Wrong hapmap release I think -- try release 21?
~~~
[  mturchin20@spudling26  /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/combined_controls]$zcat snptest_CD_All.wHapMap22.txt.gz | perl -lane 'if (($F[1] ne "---") && ($F[$#F] ne "NA")) { my @ChrBPAFInfo1 = split(/_/, $F[$#F]); my $chr = ((split(/hr/, $ChrBPAFInfo1[0]))[1]); if (($chr ne $F[2]) && ("0$chr" ne $F[2])) { print "fail1"; } if ($ChrBPAFInfo1[1] ne $F[3]) { print "fail2"; } my ($alt1, $alt2) = ($ChrBPAFInfo1[2], $ChrBPAFInfo1[4]); $alt1 =~ tr/AGCT/TCGA/; $alt2 =~ tr/AGCT/TCGA/; if ((($ChrBPAFInfo1[2] ne $F[4]) && ($ChrBPAFInfo1[4] ne $F[4])) && (($alt1 ne $F[4]) && ($alt2 ne $F[4]))) { print "fail3"; } }' | sort | uniq -c
22 fail1
169057 fail2
22 fail3
~~~

mkdir /mnt/lustre/home/mturchin20/Data/HapMap/Releases/Release21/
mkdir /mnt/lustre/home/mturchin20/Data/HapMap/Releases/Release21/Frequencies

cd /mnt/lustre/home/mturchin20/Data/HapMap/Releases/Release21/Frequencies

wget http://hapmap.ncbi.nlm.nih.gov/downloads/frequencies/2006-07/rs_strand/non-redundant/

mv /mnt/lustre/home/mturchin20/Data/HapMap/Releases/Release21/Frequencies/index.html /mnt/lustre/home/mturchin20/Data/HapMap/Releases/Release21/Frequencies/index.http:__hapmap.ncbi.nlm.nih.gov_downloads_frequencies_2006-07_rs_strand_non-redundant.html

for i in `cat /mnt/lustre/home/mturchin20/Data/HapMap/Releases/Release21/Frequencies/index.http:__hapmap.ncbi.nlm.nih.gov_downloads_frequencies_2006-07_rs_strand_non-redundant.html | perl -lane 'my $fullLine = join("", @F); if ($fullLine =~ m/.*(allele.*.txt.gz).*/) { print $1 ; } '`; do wget http://hapmap.ncbi.nlm.nih.gov/downloads/frequencies/2006-07/rs_strand/non-redundant/$i; done

nbthis *

#Edited /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py to do HapMap21 files in /mnt/lustre/home/mturchin20/Data/HapMap/Releases/Release21/Frequencies/ instead of HapMap22 files in /mnt/lustre/home/mturchin20/Data/HapMap/Releases/Release22/Frequencies/
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/combined_controls/snptest_BD_All.txt.gz | gzip > /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/combined_controls/snptest_BD_All.wHapMap21.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/combined_controls/snptest_CD_All.txt.gz | gzip > /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/combined_controls/snptest_CD_All.wHapMap21.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/combined_controls/snptest_T2D_All.txt.gz | gzip > /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/combined_controls/snptest_T2D_All.wHapMap21.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/combined_controls/snptest_RA_All.txt.gz | gzip > /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/combined_controls/snptest_RA_All.wHapMap21.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/combined_controls/snptest_HT_All.txt.gz | gzip > /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/combined_controls/snptest_HT_All.wHapMap21.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/combined_controls/snptest_T1D_All.txt.gz | gzip > /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/combined_controls/snptest_T1D_All.wHapMap21.txt.gz &
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Python.PerlVLookup.FindrsIDsinHapMap.vs3.py --file1 placeholder.gz --file2 /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/combined_controls/snptest_CAD_All.txt.gz | gzip > /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/combined_controls/snptest_CAD_All.wHapMap21.txt.gz &

#Looks like using release 21 fixes the bp location problem from before
~~~
[  mturchin20@spudling26  /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/combined_controls]$zcat snptest_BD_All.wHapMap21.txt.gz | perl -lane 'if (($F[1] ne "---") && ($F[$#F] ne "NA")) { my @ChrBPAFInfo1 = split(/_/, $F[$#F]); my $chr = ((split(/hr/, $ChrBPAFInfo1[0]))[1]); if (($chr ne $F[2]) && ("0$chr" ne $F[2])) { print "fail1"; } if ($ChrBPAFInfo1[1] ne $F[3]) { print "fail2"; } my ($alt1, $alt2) = ($ChrBPAFInfo1[2], $ChrBPAFInfo1[4]); $alt1 =~ tr/AGCT/TCGA/; $alt2 =~ tr/AGCT/TCGA/; if ((($ChrBPAFInfo1[2] ne $F[4]) && ($ChrBPAFInfo1[4] ne $F[4])) && (($alt1 ne $F[4]) && ($alt2 ne $F[4]))) { print "fail3"; } }' | sort | uniq -c
22 fail1
40 fail2
22 fail3
[  mturchin20@spudling26  /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/combined_controls]$zcat snptest_BD_All.wHapMap21.txt.gz | perl -lane 'if (($F[1] ne "---") && ($F[$#F] ne "NA")) { my @ChrBPAFInfo1 = split(/_/, $F[$#F]); my $chr = ((split(/hr/, $ChrBPAFInfo1[0]))[1]); if (($chr ne $F[2]) && ("0$chr" ne $F[2])) { print "fail1"; } if ($ChrBPAFInfo1[1] ne $F[3]) { print "fail2\t", join("\t", @F); } my ($alt1, $alt2) = ($ChrBPAFInfo1[2], $ChrBPAFInfo1[4]); $alt1 =~ tr/AGCT/TCGA/; $alt2 =~ tr/AGCT/TCGA/; if ((($ChrBPAFInfo1[2] ne $F[4]) && ($ChrBPAFInfo1[4] ne $F[4])) && (($alt1 ne $F[4]) && ($alt2 ne $F[4]))) { print "fail3"; } }' | sort | grep fail2 | grep -v rsid | head -n 5
fail2   SNP_A-1814403   rs1939284       11      48599131        A       C       0.996962        0       5       14168   138     0       1       1861    6       0.6991 0.6991   0.153855        0.0694368       0.736127        0.736127        1       chr11_51258029_A_1_C_0
fail2   SNP_A-1870400   rs9418077       10      34076879        G       T       0.995835        11240   2718    156     197     1469    351     23      25      0.946301
0.845742        -0.894994       -0.938667       0.973762        0.985055        1       chr10_34076878_C_0.917_A_0.083
fail2   SNP_A-1893435   rs9493545       06      133530395       A       G       0.999837        14262   45      0       4       1862    5       0       1       0.732333
0.732333        -0.102934       -0.183112       0.761404        0.761404        1       chr6_133530396_T_1_C_0
fail2   SNP_A-1932857   rs2543671       02      24371923        A       G       0.999271        6       532     13750   23      0       63      1804    1       0.359722
0.507548        -0.442729       -0.499745       0.426586        0.446608        1       chr2_24371889_G_0.983_A_0.017
fail2   SNP_A-2001450   rs180945        10      115700203       C       T       0.999043        1993    6556    5727    35      246     880     739     3       0.869039
0.502869        -1.09149        -1.00948        0.981144        0.667332        1       chr10_115700192_T_0_G_0.367
[  mturchin20@spudling26  /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/combined_controls]$for i in `ls -lrt | awk '{ print $9 }' | grep snptest_ | grep wHapMap21`; do echo $i; paste -d @ <(zcat $i | grep -v rsid | perl -lane 'if (($F[1] ne "---") && ($F[$#F] ne "NA")) { my @ChrBPAFInfo1 = split(/_/, $F[$#F]); my $chr = ((split(/hr/, $ChrBPAFInfo1[0]))[1]); if (($chr ne $F[2]) && ("0$chr" ne $F[2])) { print "fail1"; } if ($ChrBPAFInfo1[1] ne $F[3]) { print "fail2"; } my ($alt1, $alt2) = ($ChrBPAFInfo1[2], $ChrBPAFInfo1[4]); $alt1 =~ tr/AGCT/TCGA/; $alt2 =~ tr/AGCT/TCGA/; if ((($ChrBPAFInfo1[2] ne $F[4]) && ($ChrBPAFInfo1[4] ne $F[4])) && (($alt1 ne $F[4]) && ($alt2 ne $F[4]))) { print "fail3"; } }' | sort | uniq -c ) <(zcat $i | grep -v rsid | perl -lane 'if (($F[1] eq "---") && ($F[$#F] ne "NA")) { print "Only_---"; } elsif (($F[1] ne "---") && ($F[$#F] eq "NA")) { print "Only_NA"; } elsif (($F[1] eq "---") && ($F[$#F] eq "NA")) { print "---_And_NA";} else { my $PH = 1; }' | sort  | uniq -c) | sed 's/@/ @ /g'| column -s $'@' -t ; done
snptest_BD_All.wHapMap21.txt.gz
18 fail2       5685 Only_---
		1718 Only_NA
snptest_T2D_All.wHapMap21.txt.gz
18 fail2       5937 Only_---
		1803 Only_NA
snptest_RA_All.wHapMap21.txt.gz
18 fail2       5937 Only_---
		1803 Only_NA
snptest_CD_All.wHapMap21.txt.gz
18 fail2       5937 Only_---
		1803 Only_NA
snptest_CAD_All.wHapMap21.txt.gz
18 fail2       5937 Only_---
		1803 Only_NA
snptest_T1D_All.wHapMap21.txt.gz
18 fail2       5937 Only_---
		1803 Only_NA
snptest_HT_All.wHapMap21.txt.gz
18 fail2       5937 Only_---
		1803 Only_NA
[  mturchin20@spudling26  /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/combined_controls]$for i in `ls -lrt | awk '{ print $9
}' | grep snptest_ | grep wHapMap21 `; do echo $i; zcat $i | grep -v rsid | perl -lane 'my @ChrBPAFInfo1 = split(/_/, $F[$#F]); my $MAF = $ChrBPAFInfo1[3]; if ($MAF >
.5) { $MAF = 1 - $MAF; } push(@F, $MAF); print join("\t", @F);' | R -q -e "library(data.table); Data1 <- fread('file:///dev/stdin', header=FALSE); quantile(Data1\$V24,
na.rm=TRUE); quantile(Data1\$V16, na.rm=TRUE); png(\""$i".QCCheck.vs1.png\", height=2000, width=2000, res=200); par(mfrow=c(2,2)); hist(Data1\$V24, breaks=100); hist(Da
ta1\$V16, breaks=100); expP <- -log10(c(1:length(Data1\$V16["\!"is.na(Data1\$V16)])) / (length(Data1\$V16["\!"is.na(Data1\$V16)])+1)); plot(expP, -log10(sort(Data1\$V16
["\!"is.na(Data1\$V16)])), xlab=\"Expected p-Values\", ylab=\"Observed p-Values\"); abline(0, 1, col=\"RED\"); dev.off();"; done
snptest_BD_All.wHapMap21.txt.gz
> library(data.table); Data1 <- fread('file:///dev/stdin', header=FALSE); quantile(Data1$V24, na.rm=TRUE); quantile(Data1$V16, na.rm=TRUE); png("snptest_BD_All.wHapMap2
1.txt.gz.QCCheck.vs1.png", height=2000, width=2000, res=200); par(mfrow=c(2,2)); hist(Data1$V24, breaks=100); hist(Data1$V16, breaks=100); expP <- -log10(c(1:length(Dat
a1$V16[!is.na(Data1$V16)])) / (length(Data1$V16[!is.na(Data1$V16)])+1)); plot(expP, -log10(sort(Data1$V16[!is.na(Data1$V16)])), xlab="Expected p-Values", ylab="Observed
p-Values"); abline(0, 1, col="RED"); dev.off();
Read 437236 rows and 24 (of 24) columns from 0.070 GB file in 00:00:10
0%   25%   50%   75%  100%
0.000 0.067 0.192 0.333 0.500
0%           25%           50%           75%          100%
5.051110e-206  2.197055e-01  4.760385e-01  7.383403e-01  1.000000e+00
null device
1
>
>
snptest_T2D_All.wHapMap21.txt.gz
> library(data.table); Data1 <- fread('file:///dev/stdin', header=FALSE); quantile(Data1$V24, na.rm=TRUE); quantile(Data1$V16, na.rm=TRUE); png("snptest_T2D_All.wHapMap
21.txt.gz.QCCheck.vs1.png", height=2000, width=2000, res=200); par(mfrow=c(2,2)); hist(Data1$V24, breaks=100); hist(Data1$V16, breaks=100); expP <- -log10(c(1:length(Da
ta1$V16[!is.na(Data1$V16)])) / (length(Data1$V16[!is.na(Data1$V16)])+1)); plot(expP, -log10(sort(Data1$V16[!is.na(Data1$V16)])), xlab="Expected p-Values", ylab="Observe
d p-Values"); abline(0, 1, col="RED"); dev.off();
Read 459446 rows and 24 (of 24) columns from 0.073 GB file in 00:00:10
0%   25%   50%   75%  100%
0.000 0.067 0.192 0.333 0.500
0%          25%          50%          75%         100%
2.73190e-177  2.34115e-01  4.88371e-01  7.45486e-01  1.00000e+00
null device
1
>
>
snptest_RA_All.wHapMap21.txt.gz
> library(data.table); Data1 <- fread('file:///dev/stdin', header=FALSE); quantile(Data1$V24, na.rm=TRUE); quantile(Data1$V16, na.rm=TRUE); png("snptest_RA_All.wHapMap21.txt.gz.QCCheck.vs1.png", height=2000, width=2000, res=200); par(mfrow=c(2,2)); hist(Data1$V24, breaks=100); hist(Data1$V16, breaks=100); expP <- -log10(c(1:length(Data1$V16[!is.na(Data1$V16)])) / (length(Data1$V16[!is.na(Data1$V16)])+1)); plot(expP, -log10(sort(Data1$V16[!is.na(Data1$V16)])), xlab="Expected p-Values", ylab="Observed p-Values"); abline(0, 1, col="RED"); dev.off();
Read 459446 rows and 24 (of 24) columns from 0.073 GB file in 00:00:10
0%   25%   50%   75%  100%
0.000 0.067 0.192 0.333 0.500
0%       25%       50%       75%      100%
0.0000000 0.2377435 0.4958145 0.7536580 1.0000000
null device
1
>
>
snptest_CD_All.wHapMap21.txt.gz
> library(data.table); Data1 <- fread('file:///dev/stdin', header=FALSE); quantile(Data1$V24, na.rm=TRUE); quantile(Data1$V16, na.rm=TRUE); png("snptest_CD_All.wHapMap21.txt.gz.QCCheck.vs1.png", height=2000, width=2000, res=200); par(mfrow=c(2,2)); hist(Data1$V24, breaks=100); hist(Data1$V16, breaks=100); expP <- -log10(c(1:length(Data1$V16[!is.na(Data1$V16)])) / (length(Data1$V16[!is.na(Data1$V16)])+1)); plot(expP, -log10(sort(Data1$V16[!is.na(Data1$V16)])), xlab="Expected p-Values", ylab="Observed p-Values"); abline(0, 1, col="RED"); dev.off();
Read 459446 rows and 24 (of 24) columns from 0.073 GB file in 00:00:10
0%   25%   50%   75%  100%
0.000 0.067 0.192 0.333 0.500
0%          25%          50%          75%         100%
3.363040e-73 2.228815e-01 4.829155e-01 7.451227e-01 1.000000e+00
null device
1
>
>
snptest_CAD_All.wHapMap21.txt.gz
> library(data.table); Data1 <- fread('file:///dev/stdin', header=FALSE); quantile(Data1$V24, na.rm=TRUE); quantile(Data1$V16, na.rm=TRUE); png("snptest_CAD_All.wHapMap21.txt.gz.QCCheck.vs1.png", height=2000, width=2000, res=200); par(mfrow=c(2,2)); hist(Data1$V24, breaks=100); hist(Data1$V16, breaks=100); expP <- -log10(c(1:length(Data1$V16[!is.na(Data1$V16)])) / (length(Data1$V16[!is.na(Data1$V16)])+1)); plot(expP, -log10(sort(Data1$V16[!is.na(Data1$V16)])), xlab="Expected p-Values", ylab="Observed p-Values"); abline(0, 1, col="RED"); dev.off();
Read 459446 rows and 24 (of 24) columns from 0.073 GB file in 00:00:10
0%   25%   50%   75%  100%
0.000 0.067 0.192 0.333 0.500
0%       25%       50%       75%      100%
0.0000000 0.2370535 0.4929550 0.7465692 1.0000000
null device
1
>
>
snptest_T1D_All.wHapMap21.txt.gz
> library(data.table); Data1 <- fread('file:///dev/stdin', header=FALSE); quantile(Data1$V24, na.rm=TRUE); quantile(Data1$V16, na.rm=TRUE); png("snptest_T1D_All.wHapMap21.txt.gz.QCCheck.vs1.png", height=2000, width=2000, res=200); par(mfrow=c(2,2)); hist(Data1$V24, breaks=100); hist(Data1$V16, breaks=100); expP <- -log10(c(1:length(Data1$V16[!is.na(Data1$V16)])) / (length(Data1$V16[!is.na(Data1$V16)])+1)); plot(expP, -log10(sort(Data1$V16[!is.na(Data1$V16)])), xlab="Expected p-Values", ylab="Observed p-Values"); abline(0, 1, col="RED"); dev.off();
Read 459446 rows and 24 (of 24) columns from 0.073 GB file in 00:00:10
0%   25%   50%   75%  100%
0.000 0.067 0.192 0.333 0.500
0%       25%       50%       75%      100%
0.0000000 0.2347780 0.4902345 0.7501537 1.0000000
null device
1
>
>
snptest_HT_All.wHapMap21.txt.gz
> library(data.table); Data1 <- fread('file:///dev/stdin', header=FALSE); quantile(Data1$V24, na.rm=TRUE); quantile(Data1$V16, na.rm=TRUE); png("snptest_HT_All.wHapMap21.txt.gz.QCCheck.vs1.png", height=2000, width=2000, res=200); par(mfrow=c(2,2)); hist(Data1$V24, breaks=100); hist(Data1$V16, breaks=100); expP <- -log10(c(1:length(Data1$V16[!is.na(Data1$V16)])) / (length(Data1$V16[!is.na(Data1$V16)])+1)); plot(expP, -log10(sort(Data1$V16[!is.na(Data1$V16)])), xlab="Expected p-Values", ylab="Observed p-Values"); abline(0, 1, col="RED"); dev.off();
Read 459446 rows and 24 (of 24) columns from 0.073 GB file in 00:00:10
0%   25%   50%   75%  100%
0.000 0.067 0.192 0.333 0.500
0%          25%          50%          75%         100%
1.004920e-86 2.358175e-01 4.901600e-01 7.474943e-01 1.000000e+00
null device
1
>
>
~~~

#Weird p-value results with -log10(pVals) going as high as 200?
#Thinking this may be due to not using any type of QC cutoff, such as MAF <.01 which is detailed in manuscript but maybe not applied to released summary data?

#Checked this, didn't seem to do much
#Also wrote a script to do a rough HWE statistic and the results don't seem to show a correlation between large p-values and large HWE values either

~~~
[  mturchin20@spudling70  /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/combined_controls]$python TestHWE.vs1.py --file1 <(zcat snptest_T1D_All.wHapMap22.txt.gz | grep -v rsid) | gzip > snptest_T1D_All.wHapMap22.wHWEStat.txt.gz
[  mturchin20@spudling70  /mnt/gluster/data/external_private_supp/WTCCC_Turchin2015/WTCCC_summary_data/7_Diseases/combined_controls]$zcat snptest_T1D_All.wHapMap22.wHWEStat.txt.gz | perl -lane 'if ($F[15] < .000000000000001) { print join("\t", @F); }' | head -n 5
SNP_A-2092234   rs11073319      15      36429803        A       T       0.998758        10598   2       0       8       1903    26      0       34      5.92701e-30    5.92701e-30      9.61996 14.89   6.36106e-29     6.36106e-29     0       chr15_36429803_A_0_T_1  0.999905660377  9.43396226415e-05       10599.0000943   2.99981132075  1.00009433962    1.33341722275   21200   0.993260756869  0.00673924313116        1904.08761016   26.8247796786   1.08761016071   1.11359084571   3858
SNP_A-2280823   rs12269329      10      16389024        A       G       0.992109        10506   1       0       101     1946    16      0       1       6.60017e-19    6.60017e-19      4.75024 8.12392 4.61846e-18     4.61846e-18     0       chr10_16389024_A_1_G_0  0.999952412677  4.75873227372e-05       10507.0000238   1.99995241268  1.00002379366    1.50008328263   21014   0.995922528033  0.00407747196738        1947.03261978   16.9347604485   1.03261977574   1.08476409261   3924
SNP_A-1915623   rs7078771       10      84679040        A       G       0.990758        0       1198    9266    144     193     8       1735    27      1.82706e-21    1.03233e-271     16.3543 176.632 2.04814e-20     1.12535e-268    0       chr10_84679040_G_0.942_A_0.058  0.057243883792  0.942756116208  35.2890863914   1130.42182722  9301.28908639    39.4628893544   20928   0.101756198347  0.898243801653  21.0459710744   354.908057851   1563.04597107   1762.93883398   3872
SNP_A-2277807   rs4959334       06      5151775 A       G       0.99938 0       0       10599   9       0       12      1948    3       7.6671e-16      7.6671e-16     3.25749  6.03659 9.46869e-15     9.46869e-15     1       chr6_5151775_T_0_C_1    0.0     1.0     1.0     1.0     10600.0 2.00009433962   21198   0.0030612244898 0.99693877551   1.01836734694   12.9632653061   1949.01836735   1.09047709531   3920
SNP_A-4220418   rs726836        06      25951905        A       G       0.999846        0       77      10526   5       0       54      1909    0       4.94622e-16    4.94622e-16      9.0171  9.12112 3.25395e-17     3.25395e-17     1       chr6_25951905_C_1_T_0   0.00363104781666        0.996368952183  1.13979534094   77.7204093181  10527.1397953    1.1465963976    21206   0.0137544574631 0.986245542537  1.3713703515    54.257259297    1910.37137035   1.37357458493   3926
~~~

#20150815 CHECK_0: As I work on this, also starting to think about how wtccc and a few other of tese datasets may really violate the notion of all phenotypes being measured in all individuals used for the study. Clearly this is not the case for the wtccc datasets as each phenotype contains a different set of cases (and say may be the case for the PGC data, for example, too). This may be a real delimiting factor for the analysis/interpretation even if 'interesting' results come out from the analysis.
#Datasets that should be fine:
#Global Lipids, GIANT (for height, BMI and WHRadjBMI ?), TAG, ICBP, MAGIC, EGG, HaemgenRBC, GPC 
#Datasets that shoudl be not fine?:
#PGC, IBD?, GEFOS?, WTCCC, 23andMe?

#PGC, IBD, TAG, BP(?dbGAP), MAGIC, GEFOS, SSGAC, EGG, GERA, BEAGESS, GWASofVTE, HaemgenRBC, GPC, 23andMe(?), WTCCC




##GIS2014
##20160330

mkdir /mnt/gluster/data/external_public_supp/GIS2014

cd /mnt/gluster/data/external_public_supp/GIS2014

#Instructions from Beben Benyamin e-mail
~~~
Dear Michael,

You can download the GWAS results here:

sftp qbisftp@sftp.qbi.uq.edu.au 
user: qbisftp 
pwd: qbisftp

and the files are in /incoming.data/Beben/

Please read the instruction in the attached pdf document.

Best wishes,
Beben
~~~

#sftp qbisftp@sftp.qbi.uq.edu.au
#cd incoming.data/Beben
#get GIS_GWAS_Summary_Benyamin2014.zip
#exit

unzip /mnt/gluster/data/external_public_supp/GIS2014/GIS_GWAS_Summary_Benyamin2014.zip

gzip /mnt/gluster/data/external_public_supp/GIS2014/GIS_GWAS_Summary_Benyamin2014/*.txt

~~~
[  mturchin20@bigmem01  /mnt/gluster/data/external_public_supp/GIS2014/GIS_GWAS_Summary_Benyamin2014]$for i in `ls -lrt | awk '{ print $9 }' | grep gz$`; do echo $i; zcat $i | wc; done
Iron_Benyamin_NatCom_2014.txt.gz
2096457 18868113 111221239
Log10Ferritin_Benyamin_NatCom_2014.txt.gz
2036124 18325116 108140823
Transferrin_Benyamin_NatCom_2014.txt.gz
2104242 18938178 111825686
Saturation_Benyamin_NatCom_2014.txt.gz
2102226 18920034 111523369
[  mturchin20@bigmem01  /mnt/gluster/data/external_public_supp/GIS2014/GIS_GWAS_Summary_Benyamin2014]$for i in `ls -lrt | awk '{ print $9 }' | grep gz$`; do echo $i; zcat $i | head -n 10; done
Iron_Benyamin_NatCom_2014.txt.gz
SNP CHR BP A1 A2 FREQ_A1 EFFECT_A1 SE P
rs1000000 12 125456933 A G 0.373 -0.0167 0.0119 0.1599
rs10000010 4 21227772 T C 0.575 -0.0094 0.0099 0.3421
rs10000012 4 1347325 C G 0.808 -0.007 0.0146 0.6309
rs10000013 4 36901464 A C 0.833 -0.0068 0.0118 0.5678
rs10000017 4 84997149 T C 0.223 0.0157 0.012 0.1881
rs1000002 3 185118462 T C 0.525 0.0171 0.01 0.08538
rs10000023 4 95952929 T G 0.592 -0.0036 0.0099 0.7207
rs1000003 3 99825597 A G 0.875 -0.0041 0.0139 0.7701
rs10000030 4 103593179 A G 0.125 -0.0224 0.0144 0.1188
Log10Ferritin_Benyamin_NatCom_2014.txt.gz
SNP CHR BP A1 A2 FREQ_A1 EFFECT_A1 SE P
rs1000000 12 125456933 A G 0.373 -0.0155 0.0112 0.1659
rs10000010 4 21227772 T C 0.575 -0.0017 0.0094 0.8587
rs10000012 4 1347325 C G 0.808 -0.0202 0.0137 0.1417
rs10000013 4 36901464 A C 0.833 -0.0029 0.0111 0.7922
rs10000017 4 84997149 T C 0.223 2e-04 0.0112 0.9867
rs1000002 3 185118462 T C 0.525 0.002 0.0093 0.8261
rs10000023 4 95952929 T G 0.592 1e-04 0.0094 0.9949
rs1000003 3 99825597 A G 0.875 -0.0168 0.0131 0.1989
rs10000030 4 103593179 A G 0.125 -0.0056 0.0135 0.6809
Transferrin_Benyamin_NatCom_2014.txt.gz
SNP CHR BP A1 A2 FREQ_A1 EFFECT_A1 SE P
rs1000000 12 125456933 A G 0.373 3e-04 0.0122 0.9807
rs10000010 4 21227772 T C 0.575 0.0104 0.0102 0.3089
rs10000012 4 1347325 C G 0.808 0.0028 0.015 0.8514
rs10000013 4 36901464 A C 0.833 -0.0155 0.0122 0.2039
rs10000017 4 84997149 T C 0.223 -0.001 0.0123 0.9329
rs1000002 3 185118462 T C 0.525 -0.0031 0.0102 0.7615
rs10000023 4 95952929 T G 0.592 -0.0165 0.0102 0.1075
rs1000003 3 99825597 A G 0.875 -0.0088 0.0143 0.5399
rs10000030 4 103593179 A G 0.125 0.0127 0.0147 0.3884
Saturation_Benyamin_NatCom_2014.txt.gz
SNP CHR BP A1 A2 FREQ_A1 EFFECT_A1 SE P
rs1000000 12 125456933 A G 0.373 -0.0112 0.0119 0.3479
rs10000010 4 21227772 T C 0.575 -0.0123 0.01 0.2183
rs10000012 4 1347325 C G 0.808 -0.0031 0.0146 0.833
rs10000013 4 36901464 A C 0.833 -0.0011 0.0119 0.9297
rs10000017 4 84997149 T C 0.223 0.0141 0.012 0.2408
rs1000002 3 185118462 T C 0.525 0.0186 0.01 0.06192
rs10000023 4 95952929 T G 0.592 0.0052 0.01 0.6043
rs1000003 3 99825597 A G 0.875 7e-04 0.014 0.9601
rs10000030 4 103593179 A G 0.125 -0.0305 0.0145 0.03482
~~~



##Shin2014
##20160420

mkdir /mnt/gluster/data/external_public_supp/Shin2014

cd /mnt/gluster/data/external_public_supp/Shin2014

#From http://metabolomics.helmholtz-muenchen.de/gwas/index.php?task=download 

wget http://metabolomics.helmholtz-muenchen.de/gwas/gwas_server/shin_et_al.associations.tar.gz
wget http://metabolomics.helmholtz-muenchen.de/gwas/gwas_server/shin_et_al.metal.out.tar.gz

mkdir /mnt/gluster/data/external_public_supp/Shin2014/shin_et_al.associations
cd /mnt/gluster/data/external_public_supp/Shin2014/shin_et_al.associations
mv /mnt/gluster/data/external_public_supp/Shin2014/shin_et_al.associations.tar.gz /mnt/gluster/data/external_public_supp/Shin2014/shin_et_al.associations/.
tar -xvzf /mnt/gluster/data/external_public_supp/Shin2014/shin_et_al.associations/shin_et_al.associations.tar.gz
gzip /mnt/gluster/data/external_public_supp/Shin2014/shin_et_al.associations/metabolites_meta/* /mnt/gluster/data/external_public_supp/Shin2014/shin_et_al.associations/xenobiotics_meta/*

mkdir /mnt/gluster/data/external_public_supp/Shin2014/shin_et_al.metal
cd /mnt/gluster/data/external_public_supp/Shin2014/shin_et_al.metal
mv /mnt/gluster/data/external_public_supp/Shin2014/shin_et_al.metal.tar.gz /mnt/gluster/data/external_public_supp/Shin2014/shin_et_al.metal/.
tar -xvzf /mnt/gluster/data/external_public_supp/Shin2014/shin_et_al.metal/shin_et_al.metal.tar.gz
#gzip /mnt/gluster/data/external_public_supp/Shin2014/shin_et_al.metal/metabolites_meta/* 

for i in `ls -lrt /mnt/gluster/data/external_public_supp/Shin2014/shin_et_al.metal/metabolites_meta/. | awk '{ print $9 }'`; do
	metabolite1=`cat /mnt/gluster/data/external_public_supp/Shin2014/shin_et_al.metal/metaboliteMap.txt | grep -f <(echo $i | perl -ane 'if ($F[0] =~ m/(.*)(\.metal.pos.txt.gz)/) { print $1; }') | awk '{ print $2}'`; 
	output1=`echo $i | perl -sane 'if ($F[0] =~ m/(.*)(\.metal.pos.txt.gz)/) { print $metabolite2, $2; } ' -- -metabolite2=$metabolite1 | sed 's/txt.gz/edited.txt.gz/g'`;
	zcat /mnt/gluster/data/external_public_supp/Shin2014/shin_et_al.metal/metabolites_meta/$i | awk '{ print $1, "\t", $2, "\t", $3, "\t", $4, "\t", $8, "\t", $9, "\t", $10, "\t", $11, "\t", $16, "\t", $17, "\t", $18 }' | gzip > /mnt/gluster/data/external_public_supp/Shin2014/shin_et_al.metal/metabolites_meta/$output1;
done

rm /mnt/gluster/data/external_public_supp/Shin2014/shin_et_al.metal/metabolites_meta/M*.metal.pos.txt.gz 














##Kettunen2016
##20160420

mkdir /mnt/gluster/data/external_public_supp/Kettunen2016

cd /mnt/gluster/data/external_public_supp/Kettunen2016

#From http://www.computationalmedicine.fi/data#NMR_GWAS

#Copy and pasted 'inspect element' -> 'console' from Chrome into /mnt/gluster/data/external_public_supp/Kettunen2016/www.computationalmedicine.fi_data.SourceCode

~~~
[  mturchin20@bigmem01  /mnt/gluster/data/external_public_supp/Kettunen2016]$cat www.computationalmedicine.fi_data.SourceCode | sed 's/<\/a>/   /g' | perl -lane 'foreach my $entry1 (@F) { print $entry1; }' | grep href | sed 's/href="//g' | perl -lane 'if ($F[0] =~ m/(.*)(">)(.*)/) { print $1; }' | head -n 10
https://computationalmedicine.blob.core.windows.net/nmrgwas/Summary_statistics_MAGNETIC_AcAce.txt.gz?se=2016-04-23&sr=b&sp=r&sig=ozMtPoHatF1LBUZBaB8cQqHagmDwWBNXIJEPPj00aw8%3D&sv=2014-02-14
https://computationalmedicine.blob.core.windows.net/nmrgwas/Summary_statistics_MAGNETIC_Ace.txt.gz?se=2016-04-23&sr=b&sp=r&sig=nAE9AZyB2LPrb56zywdUhGyq%2FInMjPBfQb02FqM%2B58E%3D&sv=2014-02-14
https://computationalmedicine.blob.core.windows.net/nmrgwas/Summary_statistics_MAGNETIC_Ala.txt.gz?se=2016-04-23&sr=b&sp=r&sig=4Jz2vVaSxxJ125ARY6rUT%2BJ%2FvuH4vBmKlTi6kX9z7cU%3D&sv=2014-02-14
https://computationalmedicine.blob.core.windows.net/nmrgwas/Summary_statistics_MAGNETIC_Alb.txt.gz?se=2016-04-23&sr=b&sp=r&sig=FKqRx4%2BIXsJ3Ov5ZzeEeTjqQI14kmkeVPh2E4c1JEo0%3D&sv=2014-02-14
https://computationalmedicine.blob.core.windows.net/nmrgwas/Summary_statistics_MAGNETIC_ApoA1.txt.gz?se=2016-04-23&sr=b&sp=r&sig=6FvlxgKSltIvy%2BILE5BqykIwpHhc41qRw7m97YsVJa0%3D&sv=2014-02-14
https://computationalmedicine.blob.core.windows.net/nmrgwas/Summary_statistics_MAGNETIC_ApoB.txt.gz?se=2016-04-23&sr=b&sp=r&sig=nPVjQizpmXe3xospWIduaJH7tExbgqgmqW0UepKZRHo%3D&sv=2014-02-14
https://computationalmedicine.blob.core.windows.net/nmrgwas/Summary_statistics_MAGNETIC_Bis.DB.ratio.txt.gz?se=2016-04-23&sr=b&sp=r&sig=pGB%2B33%2BWv9eXaKnRh%2BEsXzCn%2FMXUYxV7widxa2HLDqw%3D&sv=2014-02-14
https://computationalmedicine.blob.core.windows.net/nmrgwas/Summary_statistics_MAGNETIC_Bis.FA.ratio.txt.gz?se=2016-04-23&sr=b&sp=r&sig=Yy%2FKUV1%2FPcbjhbsjteSWQCvvCepIX0fhLx66KvWeCb4%3D&sv=2014-02-14
https://computationalmedicine.blob.core.windows.net/nmrgwas/Summary_statistics_MAGNETIC_CH2.DB.ratio.txt.gz?se=2016-04-23&sr=b&sp=r&sig=GjsfUaoli8HHkMti9yDRgBTznBXRML8Z%2BcjlJrLzp54%3D&sv=2014-02-14
https://computationalmedicine.blob.core.windows.net/nmrgwas/Summary_statistics_MAGNETIC_CH2.in.FA.txt.gz?se=2016-04-23&sr=b&sp=r&sig=V9IBG%2BihUTKjCqjax4T13j0CND%2B%2BeoPkzGhlFPhWr6k%3D&sv=2014-02-14
~~~

cat /mnt/gluster/data/external_public_supp/Kettunen2016/www.computationalmedicine.fi_data.SourceCode | sed 's/<\/a>/   /g' | perl -lane 'foreach my $entry1 (@F) { print $entry1; }' | grep href | sed 's/href="//g' | perl -lane 'if ($F[0] =~ m/(.*)(">)(.*)/) { print $1; }' > /mnt/gluster/data/external_public_supp/Kettunen2016/www.computationalmedicine.fi_data.SourceCode.JustFileURLs

for i in `cat /mnt/gluster/data/external_public_supp/Kettunen2016/www.computationalmedicine.fi_data.SourceCode.JustFileURLs`; do
	outfile1=`echo $i | perl -ane 'if ($F[0] =~ m/(.*)(Summary.*.gz)(\?.*)/) { print $2; }'`;
	wget -O $outfile1 $i; 
done

-O








##Choongwon2015
##20160211

mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2015
mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1

#Notes from e-mail chain with Choongwon regarding information
~~~
Single field trip in 2012 -- two different regions in Nepal, each district has 2 subdistricts (so 4 total)
and 6% of samples are from one district
April-June, temp differences

OxHb - Amount of oxygenated hemoglobin gram/deciliter (total hemoglobin * proportion of oxygen saturation)
LB - Live Birth
ID - Infant Death (# child die between 0-1)
JD - Juvenile Death (# child die between 5-15)
IS - Infant Survival (# of child survive up to age 1)
JS - Juvenile Survival (# of child survive 5-15)
SB - Still Birth (# of children that still birthed)
~~~

#Files downloaded from Choongwon directly

~~~
[  mturchin20@spudling26  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1]$for i in `ls -lrt | awk '{ print $9 }'`; do echo $i; zcat $i | wc; done
Tibetan_GWAS_OxHb.txt.gz
 437054 4370540 32908997
Tibetan_GWAS_LB.txt.gz
 437045 4370450 32910442
Tibetan_GWAS_ID.txt.gz
 437026 4370260 32909826
Tibetan_GWAS_JD.txt.gz
 437123 4371230 32920478
Tibetan_GWAS_IS.txt.gz
 437026 4370260 32909662
Tibetan_GWAS_JS.txt.gz
 437123 4371230 32914919
Tibetan_GWAS_SB.txt.gz
 437045 4370450 32913938
~~~

LANG2=${LANG}
LC_ALL2=${LC_ALL}
LANG=C
LC_ALL=C
export LC_ALL LANG

join -a 1 -a 2 -1 2 -2 2 -e NA2 -o 0 1.1 1.3 1.4 1.7 1.10 2.1 2.3 2.4 2.7 2.10 <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1/Tibetan_GWAS_OxHb.txt.gz | sort -g -k 2,2) <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1/Tibetan_GWAS_LB.txt.gz | sort -g -k 2,2) | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
join -a 1 -a 2 -1 1 -2 2 -e NA2 -o 0 1.2 2.1 2.3 2.4 2.7 2.10 - <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1/Tibetan_GWAS_ID.txt.gz | sort -g -k 2,2) | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); if ($#F == 6) { splice(@F, 1, 0, (("NA2") x 9)); } my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
join -a 1 -a 2 -1 1 -2 2 -e NA2 -o 0 1.2 2.1 2.3 2.4 2.7 2.10 - <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1/Tibetan_GWAS_JD.txt.gz | sort -g -k 2,2) | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); if ($#F == 6) { splice(@F, 1, 0, (("NA2") x 14)); } my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
join -a 1 -a 2 -1 1 -2 2 -e NA2 -o 0 1.2 2.1 2.3 2.4 2.7 2.10 - <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1/Tibetan_GWAS_IS.txt.gz | sort -g -k 2,2) | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); if ($#F == 6) { splice(@F, 1, 0, (("NA2") x 19)); } my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
join -a 1 -a 2 -1 1 -2 2 -e NA2 -o 0 1.2 2.1 2.3 2.4 2.7 2.10 - <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1/Tibetan_GWAS_JS.txt.gz | sort -g -k 2,2) | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); if ($#F == 6) { splice(@F, 1, 0, (("NA2") x 24)); } my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
join -a 1 -a 2 -1 1 -2 2 -e NA2 -o 0 1.2 2.1 2.3 2.4 2.7 2.10 - <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1/Tibetan_GWAS_SB.txt.gz | sort -g -k 2,2) | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); if ($#F == 6) { splice(@F, 1, 0, (("NA2") x 29)); } my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | perl -lane 'print $F[0], "\t", join("\t", split(/_/, $F[1]));' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1/Tibetan_GWAS_AllPhenos.txt.pre1.gz

zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1/Tibetan_GWAS_AllPhenos.txt.pre1.gz | grep -v NA2 | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1/Tibetan_GWAS_AllPhenos.txt.pre2.gz

zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1/Tibetan_GWAS_AllPhenos.txt.pre2.gz | awk '{ if (($5 != $10) && ($5 != $15) && ($5 != $20) && ($5 != $25) && ($5 != $30) && ($5 != $35)) { print $0 } }' | wc
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1/Tibetan_GWAS_AllPhenos.txt.pre2.gz | awk '{ if (($5 != $10) || ($5 != $15) || ($5 != $20) || ($5 != $25) || ($5 != $30) || ($5 != $35)) { print $0 } }' | wc
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1/Tibetan_GWAS_AllPhenos.txt.pre2.gz | awk '{ if (($3 != $8) && ($3 != $13) && ($3 != $18) && ($3 != $23) && ($3 != $28) && ($3 != $33)) { print $0 } }' | wc
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1/Tibetan_GWAS_AllPhenos.txt.pre2.gz | awk '{ if (($3 != $8) || ($3 != $13) || ($3 != $18) || ($3 != $23) || ($3 != $28) || ($3 != $33)) { print $0 } }' | wc
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1/Tibetan_GWAS_AllPhenos.txt.pre2.gz | awk '{ if (($2 != $7) && ($2 != $12) && ($2 != $17) && ($2 != $22) && ($2 != $27) && ($2 != $32)) { print $0 } }' | wc
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1/Tibetan_GWAS_AllPhenos.txt.pre2.gz | awk '{ if (($2 != $7) || ($2 != $12) || ($2 != $17) || ($2 != $22) || ($2 != $27) || ($2 != $32)) { print $0 } }' | wc

~~~
[  mturchin20@spudling26  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1/Tibetan_GWAS_AllPhenos.txt.pre2.gz | awk '{ if (($5 != $10) && ($5 != $15) && ($5 != $20) && ($5 != $25) && ($5 != $30) && ($5 != $35)) { print $0 } }' | wc
 312436 11247696 78964003
[  mturchin20@spudling26  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1/Tibetan_GWAS_AllPhenos.txt.pre2.gz | awk '{ if (($5 != $10) || ($5 != $15) || ($5 != $20) || ($5 != $25) || ($5 != $30) || ($5 != $35)) { print $0 } }' | wc
 433460 15604560 109556920
[  mturchin20@spudling26  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1/Tibetan_GWAS_AllPhenos.txt.pre2.gz | awk '{ if (($3 != $8) && ($3 != $13) && ($3 != $18) && ($3 != $23) && ($3 != $28) && ($3 != $33)) { print $0 } }' | wc
      0       0       0
[  mturchin20@spudling26  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1/Tibetan_GWAS_AllPhenos.txt.pre2.gz | awk '{ if (($3 != $8) || ($3 != $13) || ($3 != $18) || ($3 != $23) || ($3 != $28) || ($3 != $33)) { print $0 } }' | wc
      0       0       0
[  mturchin20@spudling26  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1/Tibetan_GWAS_AllPhenos.txt.pre2.gz | awk '{ if (($2 != $7) && ($2 != $12) && ($2 != $17) && ($2 != $22) && ($2 != $27) && ($2 != $32)) { print $0 } }' | wc
      0       0       0
[  mturchin20@spudling26  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1/Tibetan_GWAS_AllPhenos.txt.pre2.gz | awk '{ if (($2 != $7) || ($2 != $12) || ($2 != $17) || ($2 != $22) || ($2 != $27) || ($2 != $32)) { print $0 } }' | wc
      0       0       0
~~~

#Take average of AFs since they are not equal across all datasets? Which makes sense since these are direct AF estimates from the data and not from a secondary source, eg using a 1000G ancestry/population for a SNP

zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1/Tibetan_GWAS_AllPhenos.txt.pre2.gz | perl -lane '$F[4] = ($F[4] + $F[9] + $F[14] + $F[19] + $F[24] + $F[29] + $F[34]) / 7; splice(@F, 34, 1); splice(@F, 31, 2); splice(@F, 29, 1); splice(@F, 26, 2); splice(@F, 24, 1); splice(@F, 21, 2); splice(@F, 19, 1); splice(@F, 16, 2); splice(@F, 14, 1); splice(@F, 11, 2); splice(@F, 9, 1); splice(@F, 6, 2); ($F[4], $F[3]) = ($F[3], $F[4]); print join("\t", @F);' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1/Tibetan_GWAS_AllPhenos.txt.gz

zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1/Tibetan_GWAS_AllPhenos.txt.pre2.gz | perl -lane 'my $avg = ($F[4] + $F[9] + $F[14] + $F[19] + $F[24] + $F[29] + $F[34]) / 7; print join("\t", ($avg, $F[4], $F[9], $F[14], $F[19], $F[24], $F[29], $F[34]));' | head -n 120 | tail -n 10
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1/Tibetan_GWAS_AllPhenos.txt.pre2.gz | perl -lane 'my $avg = ($F[4] + $F[9] + $F[14] + $F[19] + $F[24] + $F[29] + $F[34]) / 7; my $sd = 0; my @vals1 = (4, 9, 14, 19, 24, 29, 34); foreach my $index (@vals1) { $sd += ($F[$index] - $avg) ** 2; } $sd = $sd / 7; print $avg, "\t", $sd;' | sort -k 2,2 | vi -
# | R -q -e "Data1 <- read.table(file('stdin'), header=F); quantile(Data1[,1]); quantile(Data1[,2]);"

~~~
[  mturchin20@spudling26  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1/Tibetan_GWAS_AllPhenos.txt.pre2.gz | perl -lane 'my $avg = ($F[4] + $F[9] + $F[14] + $F[19] + $F[24] + $F[29] + $F[34]) / 7; my $sd = 0;
my @vals1 = (4, 9, 14, 19, 24, 29, 34); foreach my $index (@vals1) { $sd += ($F[$index] - $avg) ** 2; } $sd = $sd / 7; print $avg, "\t", $sd;' | R -q -e "Data1 <- read.table(file('stdin'), header=F); quantile(Data1[,1]); quantile(Data1[,2]);"
> Data1 <- read.table(file('stdin'), header=F); quantile(Data1[,1]); quantile(Data1[,2]);
       0%       25%       50%       75%      100% 
0.0000000 0.1521429 0.2608571 0.3771429 0.4997143 
          0%          25%          50%          75%         100% 
0.000000e+00 3.387755e-06 9.551020e-06 2.555102e-05 5.648163e-04 
> 
> 
~~~

LANG=${LANG2}
LC_ALL=${LC_ALL2}
export LC_ALL LANG

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/process.MTedits.ForHaemgenRBC2012.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1/process.MTedits.ForChoongwon2015.vs1.R

#Converted /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/process.MTedits.ForHaemgenRBC2012.vs1.R to Choongwon2015 phenotypes, moving commands to OxHb, LB, ID, JD, IS, JS, SB from RBC, MCV, PCV, MCH, Hb, MCHC

#Copy/pasted commands from /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1/process.MTedits.ForChoongwon2015.vs1.R into an interactive sessions in $PWD /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1/

#NOTE 20160610 -- Moved main source files from /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1/ to /mnt/gluster/data/external_private_supp/Choongwon2015/ 

mkdir /mnt/gluster/data/external_private_supp/Choongwon2015

mv /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1/Tibetan_GWAS* /mnt/gluster/data/external_private_supp/Choongwon2015/.
#mv /mnt/gluster/data/external_private_supp/Choongwon2015/Tibetan_GWAS_AllPhenos.txt.* /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1/.







LANG2=${LANG}
LC_ALL2=${LC_ALL}
LANG=C
LC_ALL=C
export LC_ALL LANG




An Adaptive Association Test for Multiple Phenotypes with GWAS Summary Statistics





join -a 2 -1 1 -2 1 -e NA4 -o 0 1.2 1.3 1.4 1.5 1.6 1.7 1.8 1.9 1.10 1.11 1.12 1.13 1.14 1.15 1.16 1.17 2.2 2.3 2.4 2.5 2.6 2.7 2.8 2.9 2.10 <(zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_AllPhenos.wHapMap22.formatted.wGWASannot.txt.gz | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/HaemgenRBC2012.dtlesssignif.vs1.txt | perl -F, -lane 'print join("\t", @F);' | sort -k 1,1) | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/HaemgenRBC2012.dtlesssignif.vs1.annot.txt.pre1.gz

~
[  mturchin20@spudling26  ~/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/HaemgenRBC2012.dtlesssignif.vs1.annot.txt.pre1.gz | grep NA4 | wc
    181    4344   34899
~

LANG=${LANG2}
LC_ALL=${LC_ALL2}
export LC_ALL LANG




zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/HaemgenRBC2012.dtlesssignif.vs1.annot.txt.pre1.gz | grep -v NA4 | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/HaemgenRBC2012.dtlesssignif.vs1.annot.txt.gz

RBC PCV MCV MCH MCHC Hb 

#Added the following header to /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/HaemgenRBC2012.dtlesssignif.vs1.annot.txt.gz
#snp chr pos maf p_RBC n_RBC p_PCV n_PCV p_MCV n_MCV p_MCH n_MCH p_MCHC n_MCHC p_Hb n_Hb annot Z.RBC Z.PCV Z.MCV Z.MCH Z.MCHC Z.Hb mvstat mvp unip

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GLC.MTedits.ForGIANT2014_5.AllPheno.vs2.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/GLC.MTedits.ForHamegenRBC2012.AllPheno.vs1.R

#Converted /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GLC.MTedits.ForGIANT2014_5.AllPheno.vs2.R to HaemgenRBC2012 phenotypes, moving commands to RBC PCV MCV MCH MCHC and Hb instead of Height, BMI, WHRadjBMI, WHR, HIPadjBMI, HIP, WCadjBMI and WC

#Copy/pasted commands from /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/GLC.MTedits.ForHamegenRBC2012.AllPheno.vs1.R into an interactive sessions in $PWD /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/ 








~~~~
[  mturchin20@spudling26  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1]$join -a 1 -a 2 -1 2 -2 2 -e NA2 -o 0 1.1 1.3 1.4 1.7 1.10 2.1 2.3 2.4 2.7 2.10 <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1/Tibetan_GWAS_OxHb.txt.gz | sort -g -k 2,2) <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1/Tibetan_GWAS_LB.txt.gz | sort -g -k 2,2) | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' |  wc        
 437685  875370 34759830
[  mturchin20@spudling26  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1/Tibetan_GWAS_OxHb.txt.gz | wc
 437054 4370540 32908997
[  mturchin20@spudling26  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1/Tibetan_GWAS_LB.txt.gz | wc
 437045 4370450 32910442
[  mturchin20@spudling26  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1]$join -a 1 -a 2 -1 2 -2 2 -e NA2 -o 0 1.1 1.3 1.4 1.7 1.10 2.1 2.3 2.4 2.7 2.10 <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1/Tibetan_GWAS_OxHb.txt.gz | sort -g -k 2,2) <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1/Tibetan_GWAS_LB.txt.gz | sort -g -k 2,2) | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' |  grep NA2 | wc
   1271    2542   82514
[  mturchin20@spudling26  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1]$join -a 1 -a 2 -1 2 -2 2 -e NA2 -o 0 1.1 1.3 1.4 1.7 1.10 2.1 2.3 2.4 2.7 2.10 <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1/Tibetan_GWAS_OxHb.txt.gz | sort -g -k 2,2) <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1/Tibetan_GWAS_LB.txt.gz | sort -g -k 2,2) | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' |  grep NA2 | head -n 10
rs10158710 NA2_NA2_NA2_NA2_NA2_1_184398346_981_0.050_7.949170e-01
rs1034215  1_98284940_921_0.051_3.083157e-01_NA2_NA2_NA2_NA2_NA2
rs10443196 1_99453287_894_0.051_1.850108e-01_NA2_NA2_NA2_NA2_NA2
rs1044365  NA2_NA2_NA2_NA2_NA2_1_53361928_979_0.050_1.476610e-01
rs10489560 NA2_NA2_NA2_NA2_NA2_1_96052746_981_0.051_3.999427e-01
rs10747486 1_98253738_921_0.051_3.083157e-01_NA2_NA2_NA2_NA2_NA2
rs10747487 1_98254702_921_0.051_3.083157e-01_NA2_NA2_NA2_NA2_NA2
rs10785866 1_104887259_921_0.050_5.171624e-01_NA2_NA2_NA2_NA2_NA2
rs10874792 NA2_NA2_NA2_NA2_NA2_1_92032395_975_0.051_6.743420e-01
rs10914758 1_30101765_909_0.051_5.236349e-02_NA2_NA2_NA2_NA2_NA2
[  mturchin20@spudling26  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1/Tibetan_GWAS_AllPhenos.txt.pre1.gz | wc
 439976  879952 110905886
[  mturchin20@spudling26  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1/Tibetan_GWAS_AllPhenos.txt.pre1.gz | perl -lane 'printy $#F;' | sort | uniq -c
syntax error at -e line 1, near "printy $#F"
Execution of -e aborted due to compilation errors.
[  mturchin20@spudling26  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1/Tibetan_GWAS_AllPhenos.txt.pre1.gz | perl -lane 'print $#F;' | sort | uniq -c
 439976 1
[  mturchin20@spudling26  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2015/Vs1/Tibetan_GWAS_AllPhenos.txt.pre1.gz | grep NA2 | wc
   5827   11654 1174843
~~~








##Choongwon2016
##20160610

#Starting off with taking care of sign issue from the get go

mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016
mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1
mkdir /mnt/gluster/data/external_private_supp/Choongwon2016

#Files downloaded from Choongwon directly into /mnt/gluster/data/external_private_supp/Choongwon2016 

#Notes from e-mail chain with Choongwon regarding information
~~~
For 2016 data -- 
scp cjeong@tarbell.cri.uchicago.edu:/group/dirienzo-lab/Choongwon/Tibetan.GWAS.tar

Tibetan.dHb.assoc.txt.gz
Tibetan.Hb.assoc.txt.gz
Tibetan.OxHb.assoc.txt.gz
Tibetan.Pulse.assoc.txt.gz
Tibetan.Sat.assoc.txt.gz

3 data groups:
Hb (Hemoglobin)
Sat (Oxygen Saturation)
Pulse

OxHb (Oxygenated Hemoglobin)
dHb (Dexoygenated Hemoglobin)
Pulse

From Choongwon:
"oxygenated hemoglobin = hemoglobin * oxygen saturation, and deoxygenated hemoglobin = hemoglobin * (1 - oxygen saturation)"
~~~

gzip /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.GWAS.tar

~~~
[  mturchin20@spudhead  /mnt/gluster/data/external_private_supp/Choongwon2016]$for i in `ls -lrt | awk '{ print $9 }' `; do echo $i; zcat $i | wc; done
Tibetan.Hb.assoc.txt.gz
3507569 35075690 310256521
Tibetan.Sat.assoc.txt.gz
3507569 35075690 310251610
Tibetan.Pulse.assoc.txt.gz
3508047 35080470 310310851
Tibetan.OxHb.assoc.txt.gz
3507569 35075690 310250169
Tibetan.dHb.assoc.txt.gz
3507569 35075690 310269602
Tibetan.GWAS.tar.gz
1762621 9600204 434421760
CHR SNP POS N A1 A0 A1frq beta se p                                                                                                                                                                                            1 rs3094315:752566:G:A 752566 921 G A 0.149 7.390260e-02 9.084730e-02 4.168330e-01                                                                                                                                             1 rs3131972:752721:A:G 752721 921 A G 0.276 2.500086e-02 7.696211e-02 7.333741e-01                                                                                                                                             1 rs3131971:752894:T:C 752894 903 T C 0.149 9.224075e-02 9.128423e-02 3.131140e-01                                                                                                                                             1 rs3131967:754334:T:C 754334 897 T C 0.268 6.507564e-02 7.889252e-02 4.013396e-01                                                                                                                                             1 rs3115859:754503:G:A 754503 908 G A 0.277 1.579218e-02 7.731364e-02 8.260580e-01                                                                                                                                             1 rs3131966:754964:C:T 754964 907 C T 0.277 1.719287e-02 7.732831e-02 8.125771e-01                                                                                                                                             1 rs3115858:755890:A:T 755890 900 A T 0.139 6.934277e-02 9.393129e-02 4.649276e-01
1 rs61768170:756434:G:C 756434 889 C G 0.135 5.539038e-02 9.521388e-02 5.639026e-01
1 rs3131962:756604:A:G 756604 886 A G 0.144 4.646892e-02 9.348308e-02 6.223091e-01
[  mturchin20@bigmem02  ~]$for i in `ls -lrt /mnt/gluster/data/external_private_supp/Choongwon2016/. | awk '{ print $9 }'`; do zcat /mnt/gluster/data/external_private_supp/Choongwon2016/$i | head -n 10; done
CHR SNP POS N A1 A0 A1frq beta se p
1 rs3094315:752566:G:A 752566 921 G A 0.149 7.390260e-02 9.084730e-02 4.168330e-01
1 rs3131972:752721:A:G 752721 921 A G 0.276 2.500086e-02 7.696211e-02 7.333741e-01
1 rs3131971:752894:T:C 752894 903 T C 0.149 9.224075e-02 9.128423e-02 3.131140e-01
1 rs3131967:754334:T:C 754334 897 T C 0.268 6.507564e-02 7.889252e-02 4.013396e-01
1 rs3115859:754503:G:A 754503 908 G A 0.277 1.579218e-02 7.731364e-02 8.260580e-01
1 rs3131966:754964:C:T 754964 907 C T 0.277 1.719287e-02 7.732831e-02 8.125771e-01
1 rs3115858:755890:A:T 755890 900 A T 0.139 6.934277e-02 9.393129e-02 4.649276e-01
1 rs61768170:756434:G:C 756434 889 C G 0.135 5.539038e-02 9.521388e-02 5.639026e-01
1 rs3131962:756604:A:G 756604 886 A G 0.144 4.646892e-02 9.348308e-02 6.223091e-01
CHR SNP POS N A1 A0 A1frq beta se p
1 rs3094315:752566:G:A 752566 921 G A 0.149 3.026574e-01 2.507882e-01 2.183341e-01
1 rs3131972:752721:A:G 752721 921 A G 0.276 3.799147e-01 2.119611e-01 6.939000e-02
1 rs3131971:752894:T:C 752894 903 T C 0.149 2.800573e-01 2.516822e-01 2.548416e-01
1 rs3131967:754334:T:C 754334 897 T C 0.268 3.305025e-01 2.173547e-01 1.217874e-01
1 rs3115859:754503:G:A 754503 908 G A 0.277 3.790803e-01 2.128935e-01 7.138774e-02
1 rs3131966:754964:C:T 754964 907 C T 0.277 3.792351e-01 2.129322e-01 7.130676e-02
1 rs3115858:755890:A:T 755890 900 A T 0.139 3.556228e-01 2.589724e-01 1.619344e-01
1 rs61768170:756434:G:C 756434 889 C G 0.135 3.478426e-01 2.625003e-01 1.770711e-01
1 rs3131962:756604:A:G 756604 886 A G 0.144 3.610380e-01 2.577016e-01 1.540011e-01
CHR SNP POS N A1 A0 A1frq beta se p
1 rs3094315:752566:G:A 752566 920 G A 0.149 -8.147998e-02 7.018792e-01 8.986095e-01
1 rs3131972:752721:A:G 752721 920 A G 0.276 -1.990570e-01 5.923775e-01 7.333861e-01
1 rs3131971:752894:T:C 752894 902 T C 0.149 9.749469e-03 7.035358e-01 9.985517e-01
1 rs3131967:754334:T:C 754334 896 T C 0.268 -3.006376e-01 6.068705e-01 6.174232e-01
1 rs3115859:754503:G:A 754503 907 G A 0.277 -2.293514e-01 5.946709e-01 6.973288e-01
1 rs3131966:754964:C:T 754964 906 C T 0.276 -2.391568e-01 5.947806e-01 6.851578e-01
1 rs3115858:755890:A:T 755890 899 A T 0.140 -4.422869e-01 7.242729e-01 5.345883e-01
1 rs61768170:756434:G:C 756434 888 C G 0.135 -2.280345e-01 7.338680e-01 7.491524e-01
1 rs3131962:756604:A:G 756604 886 A G 0.144 -2.690053e-01 7.205160e-01 7.017965e-01
CHR SNP POS N A1 A0 A1frq beta se p
1 rs3094315:752566:G:A 752566 920 G A 0.149 -8.147998e-02 7.018792e-01 8.986095e-01
1 rs3131972:752721:A:G 752721 920 A G 0.276 -1.990570e-01 5.923775e-01 7.333861e-01
1 rs3131971:752894:T:C 752894 902 T C 0.149 9.749469e-03 7.035358e-01 9.985517e-01
1 rs3131967:754334:T:C 754334 896 T C 0.268 -3.006376e-01 6.068705e-01 6.174232e-01
1 rs3115859:754503:G:A 754503 907 G A 0.277 -2.293514e-01 5.946709e-01 6.973288e-01
1 rs3131966:754964:C:T 754964 906 C T 0.276 -2.391568e-01 5.947806e-01 6.851578e-01
1 rs3115858:755890:A:T 755890 899 A T 0.140 -4.422869e-01 7.242729e-01 5.345883e-01
1 rs61768170:756434:G:C 756434 888 C G 0.135 -2.280345e-01 7.338680e-01 7.491524e-01
1 rs3131962:756604:A:G 756604 886 A G 0.144 -2.690053e-01 7.205160e-01 7.017965e-01
CHR SNP POS N A1 A0 A1frq beta se p
1 rs3094315:752566:G:A 752566 921 G A 0.149 1.172253e-01 8.052830e-02 1.418002e-01
1 rs3131972:752721:A:G 752721 921 A G 0.276 7.774395e-02 6.800957e-02 2.430682e-01
1 rs3131971:752894:T:C 752894 903 T C 0.149 1.293215e-01 8.073968e-02 1.061625e-01
1 rs3131967:754334:T:C 754334 897 T C 0.268 1.016436e-01 6.962139e-02 1.375870e-01
1 rs3115859:754503:G:A 754503 908 G A 0.277 6.990500e-02 6.830048e-02 2.950650e-01
1 rs3131966:754964:C:T 754964 907 C T 0.277 7.150401e-02 6.831017e-02 2.845964e-01
1 rs3115858:755890:A:T 755890 900 A T 0.139 1.196101e-01 8.314150e-02 1.470486e-01
1 rs61768170:756434:G:C 756434 889 C G 0.135 1.098849e-01 8.427904e-02 1.883522e-01
1 rs3131962:756604:A:G 756604 886 A G 0.144 1.022105e-01 8.276942e-02 2.121807e-01
CHR SNP POS N A1 A0 A1frq beta se p
1 rs3094315:752566:G:A 752566 921 G A 0.149 -3.381448e-02 3.975178e-02 3.797220e-01
1 rs3131972:752721:A:G 752721 921 A G 0.276 -5.256199e-02 3.361495e-02 1.124069e-01
1 rs3131971:752894:T:C 752894 903 T C 0.149 -2.858749e-02 3.993963e-02 4.557214e-01
1 rs3131967:754334:T:C 754334 897 T C 0.268 -4.193312e-02 3.448905e-02 2.133919e-01
1 rs3115859:754503:G:A 754503 908 G A 0.277 -5.455136e-02 3.376451e-02 1.015270e-01
1 rs3131966:754964:C:T 754964 907 C T 0.277 -5.438106e-02 3.377120e-02 1.025686e-01
1 rs3115858:755890:A:T 755890 900 A T 0.139 -4.301473e-02 4.106936e-02 2.811686e-01
1 rs61768170:756434:G:C 756434 889 C G 0.135 -4.212731e-02 4.163961e-02 2.983531e-01
1 rs3131962:756604:A:G 756604 886 A G 0.144 -4.472352e-02 4.087081e-02 2.618215e-01
~~~

#zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Hb.assoc.txt.gz | perl -lane '$F[1] = ((split(/:/, $F[1]))[0]); if ($F[7] < 0) { ($F[4], $F[5]) = ($F[5], $F[4]); $F[6] = 1 - $F[6]; $F[7] = $F[7] * -1; } print join("\t", @F);' | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Hb.assoc.IncAllele.txt.gz
zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Hb.assoc.txt.gz | perl -lane 'my $rsID = ((split(/:/, $F[1]))[0]); my $ChrBP = $F[0] . "_" . $F[2]; splice(@F, 0, 0, ($ChrBP, $rsID)); if ($F[9] < 0) { ($F[6], $F[7]) = ($F[7], $F[6]); $F[8] = 1 - $F[8]; $F[9] = $F[9] * -1; } print join("\t", @F);' | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Hb.assoc.IncAllele.txt.gz

~~~
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Hb.assoc.IncAllele.txt.gz | head -n 10
CHR     SNP     POS     N       A1      A0      A1frq   beta    se      p
1       rs3094315       752566  921     G       A       0.149   7.390260e-02    9.084730e-02    4.168330e-01
1       rs3131972       752721  921     A       G       0.276   2.500086e-02    7.696211e-02    7.333741e-01
1       rs3131971       752894  903     T       C       0.149   9.224075e-02    9.128423e-02    3.131140e-01
1       rs3131967       754334  897     T       C       0.268   6.507564e-02    7.889252e-02    4.013396e-01
1       rs3115859       754503  908     G       A       0.277   1.579218e-02    7.731364e-02    8.260580e-01
1       rs3131966       754964  907     C       T       0.277   1.719287e-02    7.732831e-02    8.125771e-01
1       rs3115858       755890  900     A       T       0.139   6.934277e-02    9.393129e-02    4.649276e-01
1       rs61768170      756434  889     C       G       0.135   5.539038e-02    9.521388e-02    5.639026e-01
1       rs3131962       756604  886     A       G       0.144   4.646892e-02    9.348308e-02    6.223091e-01
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Hb.assoc.IncAllele.txt.gz | wc        
3507569 35075690 259172234
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Hb.assoc.IncAllele.txt.gz | awk '{ if ($8 < 0) { print $0 } } ' | wc
      0       0       0
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$join -1 1 -2 2 <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Hb.assoc.IncAllele.txt.gz | awk '{ print $2, "\t", $5 }' | sort ) <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Sat.assoc.txt.gz | perl -lane '$F[1] = ((split(/:/, $F[1]))[0]); if ($F[7] < 0) { ($F[4], $F[5]) = ($F[5], $F[4]); $F[6] = 1 - $F[6]; $F[7] = $F[7] * -1; } print join("\t", @F);' | sort -k 2,2) | head -n 10
1 A 1 100640636 919 A G 0.935 0.0793903 3.678322e-01 8.052751e-01
1 A 1 100991376 917 T C 0.111 1.564873e-01 2.829851e-01 5.725311e-01
1 A 1 101147138 910 A G 0.542 0.1638499 1.830628e-01 3.536512e-01
1 A 1 101970051 920 T C 0.251 1.534397e-01 2.085567e-01 4.620248e-01
1 A 1 102777667 885 C T 0.753 0.1817058 2.119664e-01 3.865822e-01
1 A 1 103329033 920 C G 0.828 0.4634013 2.387134e-01 5.000663e-02
1 A 1 103329034 920 T G 0.828 0.4634013 2.387134e-01 5.000663e-02
1 A 1 103329036 920 C A 0.828 0.4634013 2.387134e-01 5.000663e-02
1 A 1 103380393 897 A G 0.055 1.011388e-01 3.925120e-01 8.057365e-01
1 A 1 104395093 906 C T 0.278 1.634844e-01 2.015026e-01 4.210668e-01
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Sat.assoc.txt.gz | grep -v rs | head -n 10
CHR SNP POS N A1 A0 A1frq beta se p
1 1:1465930:C:A 1465930 912 A C 0.274 1.608274e-01 2.038613e-01 4.323969e-01
1 1:1465935:C:T 1465935 912 T C 0.274 1.608274e-01 2.038613e-01 4.323969e-01
1 1:1466230:A:T 1466230 918 T A 0.277 1.882411e-01 2.028248e-01 3.540418e-01
1 1:3253614:C:T 3253614 919 T C 0.102 8.672095e-02 2.960357e-01 7.661539e-01
1 1:4221128:A:C 4221128 880 C A 0.099 3.352135e-01 2.991641e-01 2.554131e-01
1 1:4221129:G:A 4221129 880 A G 0.099 3.352135e-01 2.991641e-01 2.554131e-01
1 1:4596009:C:T 4596009 919 T C 0.463 -2.566785e-01 1.806019e-01 1.474200e-01
1 1:4784762:C:T 4784762 906 T C 0.184 5.036829e-02 2.306971e-01 8.277443e-01
1 1:4953813:T:A 4953813 895 T A 0.085 6.793326e-01 3.266151e-01 3.669162e-02
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Hb.assoc.txt.gz | perl -lane 'my $rsID = ((split(/:/, $F[1]))[0]); my $ChrBP = $F[0] . "_" . $F[2]; splice(@F, 0, 0, ($ChrBP, $rsID)); if ($F[9] < 0) { ($F[6], $F[7]) = ($F[7], $F[6]); $F[8] = 1 - $F[8]; $F[9] = $F[9] * -1; } print join("\t", @F);' | head -n 10
CHR_POS SNP     CHR     SNP     POS     N       A1      A0      A1frq   beta    se      p
1_752566        rs3094315       1       rs3094315:752566:G:A    752566  921     G       A       0.149   7.390260e-02    9.084730e-02    4.168330e-01
1_752721        rs3131972       1       rs3131972:752721:A:G    752721  921     A       G       0.276   2.500086e-02    7.696211e-02    7.333741e-01
1_752894        rs3131971       1       rs3131971:752894:T:C    752894  903     T       C       0.149   9.224075e-02    9.128423e-02    3.131140e-01
1_754334        rs3131967       1       rs3131967:754334:T:C    754334  897     T       C       0.268   6.507564e-02    7.889252e-02    4.013396e-01
1_754503        rs3115859       1       rs3115859:754503:G:A    754503  908     G       A       0.277   1.579218e-02    7.731364e-02    8.260580e-01
1_754964        rs3131966       1       rs3131966:754964:C:T    754964  907     C       T       0.277   1.719287e-02    7.732831e-02    8.125771e-01
1_755890        rs3115858       1       rs3115858:755890:A:T    755890  900     A       T       0.139   6.934277e-02    9.393129e-02    4.649276e-01
1_756434        rs61768170      1       rs61768170:756434:G:C   756434  889     C       G       0.135   5.539038e-02    9.521388e-02    5.639026e-01
1_756604        rs3131962       1       rs3131962:756604:A:G    756604  886     A       G       0.144   4.646892e-02    9.348308e-02    6.223091e-01
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Hb.assoc.txt.gz | perl -lane 'my $rsID = ((split(/:/, $F[1]))[0]); my $ChrBP = $F[0] . "_" . $F[2]; splice(@F, 0, 0, ($ChrBP, $rsID)); if ($F[9] < 0) { ($F[6], $F[7]) = ($F[7], $F[6]); $F[8] = 1 - $F[8]; $F[9] = $F[9] * -1; } print join("\t", @F);' | perl -lane 'print $#F;' | sort | uniq -c
3507569 11
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$join <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Hb.assoc.IncAllele.txt.gz | awk '{ print $1, "\t", $7 }' | sort ) <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Sat.assoc.txt.gz | perl -lane 'my $rsID = ((split(/:/, $F[1]))[0]); my $ChrBP = $F[0] . "_" . $F[2]; splice(@F, 0, 0, ($ChrBP, $rsID)); if ($F[9] < 0) { ($F[6], $F[7]) = ($F[7], $F[6]); $F[8] = 1 - $F[8]; $F[9] = $F[9] * -1; } print join("\t", @F);' | sort) | perl -lane 'if ($F[1] ne $F[7]) { $F[12] = $F[12] * -1; } splice(@F, 1, 1); print join("\t", @F);' | head -n 10
10_100000625    rs7899632       10      rs7899632:100000625:A:G 100000625       916     G       A       0.422   2.398042e-02    1.828722e-01    -0.8755691
10_100000645    rs61875309      10      rs61875309:100000645:A:C        100000645       916     C       A       0.060   1.430280e-01    3.847636e-01    -0.7304266
10_100002399    rs8181398       10      rs8181398:100002399:A:G 100002399       918     A       G       0.055   3.641363e-01    3.919577e-01    -0.3570072
10_100003242    rs12258651      10      rs12258651:100003242:T:G        100003242       917     T       G       0.879   0.004309389     2.741558e-01    -0.999923
10_100003785    rs1359508       10      rs1359508:100003785:T:C 100003785       919     T       C       0.538   0.09766834      1.825781e-01    -0.5861611
10_100004360    rs1048754       10      rs1048754:100004360:G:A 100004360       916     A       G       0.060   1.430280e-01    3.847636e-01    -0.7304266
10_100004441    rs1048757       10      rs1048757:100004441:G:C 100004441       918     C       G       0.481   3.259236e-02    1.810009e-01    -0.8459574
10_100004906    rs3750595       10      rs3750595:100004906:C:A 100004906       919     A       C       0.421   2.382532e-02    1.826816e-01    -0.8757007
10_100004996    rs2025625       10      rs2025625:100004996:G:A 100004996       920     G       A       0.538   0.09828543      1.825797e-01    -0.5838185
10_10000514     rs6602382       10      rs6602382:10000514:T:C  10000514        912     T       C       0.532   0.1286826       1.813214e-01    -0.4672061
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Hb.assoc.txt.gz | grep rs7899632
10 rs7899632:100000625:A:G 100000625 916 G A 0.422 -1.126199e-01 6.582211e-02 8.392270e-02
zcat [  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Sat.assoc.txt.gz | grep rs7899632 
10 rs7899632:100000625:A:G 100000625 916 G A 0.422 2.398042e-02 1.828722e-01 8.755691e-01
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$join <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Hb.assoc.IncAllele.txt.gz | awk '{ print $1, "\t", $7 }' | sort ) <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Sat.assoc.txt.gz | perl -lane 'my $rsID = ((split(/:/, $F[1]))[0]); my $ChrBP = $F[0] . "_" . $F[2]; splice(@F, 0, 0, ($ChrBP, $rsID)); if ($F[9] < 0) { ($F[6], $F[7]) = ($F[7], $F[6]); $F[8] = 1 - $F[8]; $F[9] = $F[9] * -1; } print join("\t", @F);' | sort) | perl -lane 'if ($F[1] ne $F[7]) { $F[12] = $F[12] * -1; } splice(@F, 1, 1); print join("\t", @F);' | head -n 350 | tail -n 10
10_10017016     rs4617491       10      rs4617491:10017016:T:G  10017016        921     T       G       0.918   0.1044814       3.320171e-01    7.408470e-01
10_100170209    rs6584201       10      rs6584201:100170209:T:C 100170209       921     C       T       0.504   0.1240805       1.765402e-01    4.849922e-01
10_100170383    rs6584202       10      rs6584202:100170383:G:A 100170383       921     A       G       0.504   0.1240805       1.765402e-01    4.849922e-01
10_100170507    rs7896828       10      rs7896828:100170507:A:T 100170507       918     A       T       0.363   7.859986e-02    1.850275e-01    -0.6591994
10_100170762    rs3814140       10      rs3814140:100170762:C:T 100170762       921     T       C       0.504   0.1240805       1.765402e-01    4.849922e-01
10_100171028    rs3814141       10      rs3814141:100171028:G:A 100171028       895     G       A       0.601   0.03760758      1.869979e-01    -0.8336881
10_100171104    rs3814142       10      rs3814142:100171104:C:G 100171104       921     G       C       0.504   0.1240805       1.765402e-01    4.849922e-01
10_100171174    rs3829853       10      rs3829853:100171174:A:G 100171174       921     G       A       0.504   0.1240805       1.765402e-01    4.849922e-01
10_100171198    rs10748728      10      rs10748728:100171198:C:T        100171198       921     T       C       0.504   0.1240805       1.765402e-01    4.849922e-01
10_100171371    rs10748729      10      rs10748729:100171371:G:C        100171371       921     C       G       0.504   0.1240805       1.765402e-01    4.849922e-01
~~~

LANG2=${LANG}
LC_ALL2=${LC_ALL}
LANG=C
LC_ALL=C
export LC_ALL LANG

#3 data groups:
#Hb (Hemoglobin)
#Sat (Oxygen Saturation)
#Pulse
#
#OxHb (Oxygenated Hemoglobin)
#dHb (Dexoygenated Hemoglobin) 
#Pulse

#join -1 1 -2 2 <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Hb.assoc.IncAllele.txt.gz | awk '{ print $2, "\t", $5 }' | sort ) <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Sat.assoc.txt.gz | perl -lane '$F[1] = ((split(/:/, $F[1]))[0]); if ($F[7] < 0) { ($F[4], $F[5]) = ($F[5], $F[4]); $F[6] = 1 - $F[6]; $F[7] = $F[7] * -1; } print join("\t", @F);' | sort -k 2,2) | perl -lane 'if ($F[1] ne $F[5]) { $F[10] = $F[10] * -1; } splice(@F, 1, 1); print join("\t", @F);' | gzip >
#join -1 1 -2 2 <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Hb.assoc.IncAllele.txt.gz | awk '{ print $2, "\t", $8 }' | sort ) <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Sat.assoc.txt.gz | sed 's/:/ /g' | perl -lane 'if ($F[10] < 0) { ($F[7], $F[8]) = ($F[8], $F[7]); $F[9] = 1 - $F[9]; $F[10] = $F[1] * -1; } print join("\t", @F);' | sort -k 2,2) | perl -lane 'if ($F[1] ne $F[8]) { $F[13] = $F[13] * -1; } splice(@F, 1, 1); print join("\t", @F);' | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Sat.assoc.IncAllele.MatchedToHb.txt.gz
#perl -lane 'if ($F[7] < 0) { ($F[4], $F[5]) = ($F[5], $F[4]); $F[12] = 1 - $F[12]; $F[7] = $F[7] * -1; } print join("\t", @F);' | sort) | perl -lane 'if ($F[1] ne $F[5]) { $F[2] = $F[2] * -1; } splice(@F, 1, 1); print join("\t", @F);' |  sed 's/ChrBPAFInfo/Chr BP Ref RAF Alt AAF/g' |  grep -v -w NA | gzip >  /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Sat.MatchedToHb.IncAllele.txt.gz

join <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Hb.assoc.IncAllele.txt.gz | awk '{ print $1, "\t", $7 }' | sort ) <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Sat.assoc.txt.gz | perl -lane 'my $rsID = ((split(/:/, $F[1]))[0]); my $ChrBP = $F[0] . "_" . $F[2]; splice(@F, 0, 0, ($ChrBP, $rsID)); if ($F[9] < 0) { ($F[6], $F[7]) = ($F[7], $F[6]); $F[8] = 1 - $F[8]; $F[9] = $F[9] * -1; } print join("\t", @F);' | sort) | perl -lane 'if ($F[1] ne $F[7]) { $F[12] = $F[12] * -1; } splice(@F, 1, 1); print join("\t", @F);' | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Sat.assoc.IncAllele.MatchedToHb.txt.gz
join <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Hb.assoc.IncAllele.txt.gz | awk '{ print $1, "\t", $7 }' | sort ) <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Pulse.assoc.txt.gz | perl -lane 'my $rsID = ((split(/:/, $F[1]))[0]); my $ChrBP = $F[0] . "_" . $F[2]; splice(@F, 0, 0, ($ChrBP, $rsID)); if ($F[9] < 0) { ($F[6], $F[7]) = ($F[7], $F[6]); $F[8] = 1 - $F[8]; $F[9] = $F[9] * -1; } print join("\t", @F);' | sort) | perl -lane 'if ($F[1] ne $F[7]) { $F[12] = $F[12] * -1; } splice(@F, 1, 1); print join("\t", @F);' | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Pulse.assoc.IncAllele.MatchedToHb.txt.gz
join <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Hb.assoc.IncAllele.txt.gz | awk '{ print $1, "\t", $7 }' | sort ) <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.OxHb.assoc.txt.gz | perl -lane 'my $rsID = ((split(/:/, $F[1]))[0]); my $ChrBP = $F[0] . "_" . $F[2]; splice(@F, 0, 0, ($ChrBP, $rsID)); if ($F[9] < 0) { ($F[6], $F[7]) = ($F[7], $F[6]); $F[8] = 1 - $F[8]; $F[9] = $F[9] * -1; } print join("\t", @F);' | sort) | perl -lane 'if ($F[1] ne $F[7]) { $F[12] = $F[12] * -1; } splice(@F, 1, 1); print join("\t", @F);' | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.OxHb.assoc.IncAllele.MatchedToHb.txt.gz
join <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Hb.assoc.IncAllele.txt.gz | awk '{ print $1, "\t", $7 }' | sort ) <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.dHb.assoc.txt.gz | perl -lane 'my $rsID = ((split(/:/, $F[1]))[0]); my $ChrBP = $F[0] . "_" . $F[2]; splice(@F, 0, 0, ($ChrBP, $rsID)); if ($F[9] < 0) { ($F[6], $F[7]) = ($F[7], $F[6]); $F[8] = 1 - $F[8]; $F[9] = $F[9] * -1; } print join("\t", @F);' | sort) | perl -lane 'if ($F[1] ne $F[7]) { $F[12] = $F[12] * -1; } splice(@F, 1, 1); print join("\t", @F);' | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.dHb.assoc.IncAllele.MatchedToHb.txt.gz

zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Hb.assoc.IncAllele.txt.gz | awk '{ print $3, "\t", $5, "\t", $2, "\t", $9, "\t", $12, "\t", $6 }' | cat <(echo "CHR      POS     SNP     A1frq   p       N") <(cat - | grep -v A1frq) | perl -lane 'if ($F[2] !~ /\D/ || $F[2] eq "X") { $F[2] = $F[0] . "_" . $F[1]; } print join("\t", @F); ' | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Hb.assoc.IncAllele.formatted.txt.gz 
zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Sat.assoc.IncAllele.MatchedToHb.txt.gz | awk '{ print $3, "\t", $5, "\t", $2, "\t", $9, "\t", $12, "\t", $6 }' | cat <(echo "CHR      POS     SNP     A1frq   p       N") <(cat - | grep -v A1frq) | perl -lane 'if ($F[2] !~ /\D/ || $F[2] eq "X") { $F[2] = $F[0] . "_" . $F[1]; } print join("\t", @F); '  | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Sat.assoc.IncAllele.MatchedToHb.formatted.txt.gz 
zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Pulse.assoc.IncAllele.MatchedToHb.txt.gz | awk '{ print $3, "\t", $5, "\t", $2, "\t", $9, "\t", $12, "\t", $6 }' | cat <(echo "CHR      POS     SNP     A1frq   p       N") <(cat - | grep -v A1frq) | perl -lane 'if ($F[2] !~ /\D/ || $F[2] eq "X") { $F[2] = $F[0] . "_" . $F[1]; } print join("\t", @F); '  | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Pulse.assoc.IncAllele.MatchedToHb.formatted.txt.gz 
zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.OxHb.assoc.IncAllele.MatchedToHb.txt.gz | awk '{ print $3, "\t", $5, "\t", $2, "\t", $9, "\t", $12, "\t", $6 }' | cat <(echo "CHR      POS     SNP     A1frq   p       N") <(cat - | grep -v A1frq) | perl -lane 'if ($F[2] !~ /\D/ || $F[2] eq "X") { $F[2] = $F[0] . "_" . $F[1]; } print join("\t", @F); '  | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.OxHb.assoc.IncAllele.MatchedToHb.formatted.txt.gz 
zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.dHb.assoc.IncAllele.MatchedToHb.txt.gz | awk '{ print $3, "\t", $5, "\t", $2, "\t", $9, "\t", $12, "\t", $6 }' | cat <(echo "CHR      POS     SNP     A1frq   p       N") <(cat - | grep -v A1frq) | perl -lane 'if ($F[2] !~ /\D/ || $F[2] eq "X") { $F[2] = $F[0] . "_" . $F[1]; } print join("\t", @F); '  | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.dHb.assoc.IncAllele.MatchedToHb.formatted.txt.gz 



#20160626 NOTE -- Not all variants hae rsIDs, some just have a chromosome # or such in there place. This is probably messing up the join statement, so going to switch 2nd and 3rd column and make BP ChrBP 
#join on the 2nd column and make it a 'ChrBP' column of some sort

~~~
[  mturchin20@bigmem03  ~]$join -a 1 -a 2 -1 3 -2 3 -e NA2 -o 0 1.1 1.2 1.4 1.5 1.6 2.1 2.2 2.4 2.5 2.6 <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Hb.assoc.IncAllele.formatted.txt.gz | sort -g -k 3,3) <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Sat.assoc.IncAllele.MatchedToHb.formatted.txt.gz | sort -g -k 3,3 ) | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | wc
6982113 13964226 517743789
[  mturchin20@spudling28  ~]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Hb.assoc.IncAllele.formatted.txt.gz | wc
3507569 21045414 192931966
[  mturchin20@spudling28  ~]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Sat.assoc.IncAllele.MatchedToHb.formatted.txt.gz | wc
3507569 21045414 189079860
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$join -a 1 -a 2 -1 2 -2 2 -e NA2 -o 0 1.1 1.2 1.4 1.5 1.6 2.1 2.2 2.4 2.5 2.6 <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Hb.assoc.IncAllele.formatted.txt.gz | perl -lane '$F[1] = $F[0] . "_" . $F[1]; print join("\t", @F);' | sort -g -k 2,2) <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Sat.assoc.IncAllele.MatchedToHb.formatted.txt.gz | perl -lane '$F[1] = $F[0] . "_" . $F[1]; print join("\t", @F);' | sort -g -k 2,2 ) | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | wc
3507569 7015138 296719685
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$join -a 1 -a 2 -1 3 -2 3 -e NA2 -o 0 1.1 1.2 1.4 1.5 1.6 2.1 2.2 2.4 2.5 2.6 <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Hb.assoc.IncAllele.formatted.txt.gz | perl -lane '($F[2], $F[1]) = ($F[1], $F[2]); $F[2] = $F[0] . "_" . $F[2]; print join("\t", @F);' | sort -g -k 3,3) <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Sat.assoc.IncAllele.MatchedToHb.formatted.txt.gz | perl -lane '($F[2], $F[1]) = ($F[1], $F[2]); $F[2] = $F[0] . "_" . $F[2]; print join("\t", @F);' | sort -g -k 3,3 ) | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | join -a 1 -a 2 -1 1 -2 3 -e NA2 -o 0 1.2 2.1 2.2 2.4 2.5 2.6 - <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Pulse.assoc.IncAllele.MatchedToHb.formatted.txt.gz | perl -lane '($F[2], $F[1]) = ($F[1], $F[2]); $F[2] = $F[0] . "_" . $F[2]; print join("\t", @F);' | sort -g -k 3,3) | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); print join("\t", @F);' | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); if ($#F == 6) { splice(@F, 1, 0, (("NA2") x 9)); } my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | perl -lane 'print $F[0], "\t", join("\t", split(/_/, $F[1]));' | sed 's/chr//g' | grep rs | head -n 10
1_226580387     1       rs1000033       0.378   1.705824e-01    921     1       rs1000033       0.378   1.508678e-01    921     1       rs1000033       0.622   -0.1481057      920
1_162736463     1       rs1000050       0.429   7.061967e-02    920     1       rs1000050       0.571   -0.3896959      920     1       rs1000050       0.571   -0.4879369      919
1_157255396     1       rs1000073       0.126   2.015334e-01    911     1       rs1000073       0.874   -0.2014158      911     1       rs1000073       0.126   1.448819e-03    910
1_95166832      1       rs1000075       0.344   3.925941e-01    912     1       rs1000075       0.344   8.796695e-01    912     1       rs1000075       0.656   -0.6900962      911
1_8640831       1       rs1000184       0.662   5.826021e-01    913     1       rs1000184       0.662   9.268587e-04    913     1       rs1000184       0.662   4.758434e-01    912
1_209894785     1       rs1000282       0.794   8.221829e-01    921     1       rs1000282       0.794   6.688351e-01    921     1       rs1000282       0.793   6.584309e-01    920
1_209894661     1       rs1000283       0.816   7.676046e-01    911     1       rs1000283       0.816   6.995587e-01    911     1       rs1000283       0.816   3.669530e-01    910
1_15405489      1       rs1000313       0.237   6.936375e-01    898     1       rs1000313       0.237   5.742768e-01    898     1       rs1000313       0.236   5.673318e-01    897
1_15405907      1       rs1000314       0.291   8.754458e-01    903     1       rs1000314       0.291   5.786283e-01    903     1       rs1000314       0.291   4.987349e-01    902
1_15041242      1       rs1000348       0.127   6.950434e-01    913     1       rs1000348       0.127   2.753010e-01    913     1       rs1000348       0.873   -0.2354071      912
~~~

#LANG2=${LANG}
#LC_ALL2=${LC_ALL}
#LANG=C
#LC_ALL=C
#export LC_ALL LANG

join -a 1 -a 2 -1 3 -2 3 -e NA2 -o 0 1.1 1.2 1.4 1.5 1.6 2.1 2.2 2.4 2.5 2.6 <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Hb.assoc.IncAllele.formatted.txt.gz | sed 's/_/-/g' | perl -lane '($F[2], $F[1]) = ($F[1], $F[2]); $F[2] = $F[0] . "_" . $F[2]; print join("\t", @F);' | sort -g -k 3,3) <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Sat.assoc.IncAllele.MatchedToHb.formatted.txt.gz | sed 's/_/-/g' | perl -lane '($F[2], $F[1]) = ($F[1], $F[2]); $F[2] = $F[0] . "_" . $F[2]; print join("\t", @F);' | sort -g -k 3,3 ) | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
join -a 1 -a 2 -1 1 -2 3 -e NA2 -o 0 1.2 2.1 2.2 2.4 2.5 2.6 - <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Pulse.assoc.IncAllele.MatchedToHb.formatted.txt.gz | sed 's/_/-/g' | perl -lane '($F[2], $F[1]) = ($F[1], $F[2]); $F[2] = $F[0] . "_" . $F[2]; print join("\t", @F);' | sort -g -k 3,3) | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); print join("\t", @F);' | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); if ($#F == 6) { splice(@F, 1, 0, (("NA2") x 9)); } my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
perl -lane 'print $F[0], "\t", join("\t", split(/_/, $F[1]));' | sed 's/chr//g' | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.3Pheno_1.assoc.IncAllele.MatchedToHb.formatted.RAF.txt.pre1.gz

join -a 1 -a 2 -1 3 -2 3 -e NA2 -o 0 1.1 1.2 1.4 1.5 1.6 2.1 2.2 2.4 2.5 2.6 <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.OxHb.assoc.IncAllele.MatchedToHb.formatted.txt.gz | sed 's/_/-/g' | perl -lane '($F[2], $F[1]) = ($F[1], $F[2]); $F[2] = $F[0] . "_" . $F[2]; print join("\t", @F);' | sort -g -k 3,3) <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.dHb.assoc.IncAllele.MatchedToHb.formatted.txt.gz | sed 's/_/-/g' | perl -lane '($F[2], $F[1]) = ($F[1], $F[2]); $F[2] = $F[0] . "_" . $F[2]; print join("\t", @F);' | sort -g -k 3,3 ) | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
join -a 1 -a 2 -1 1 -2 3 -e NA2 -o 0 1.2 2.1 2.2 2.4 2.5 2.6 - <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Pulse.assoc.IncAllele.MatchedToHb.formatted.txt.gz | sed 's/_/-/g' | perl -lane '($F[2], $F[1]) = ($F[1], $F[2]); $F[2] = $F[0] . "_" . $F[2]; print join("\t", @F);' | sort -g -k 3,3) | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); print join("\t", @F);' | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); if ($#F == 6) { splice(@F, 1, 0, (("NA2") x 9)); } my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
perl -lane 'print $F[0], "\t", join("\t", split(/_/, $F[1]));' | sed 's/chr//g' | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.3Pheno_2.assoc.IncAllele.MatchedToHb.formatted.RAF.txt.pre1.gz

LANG=${LANG2}
LC_ALL=${LC_ALL2}
export LC_ALL LANG

~~~
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.3Pheno_1.assoc.IncAllele.MatchedToHb.formatted.RAF.txt.pre1.gz | head -n 10
CHR_POS CHR     SNP     A1frq   p       N       CHR     SNP     A1frq   p       N       CHR     SNP     A1frq   p       N
1_100640636     1       1       0.935   4.077197e-01    919     1       1       0.935   8.052751e-01    919     1       1       0.065   -0.3297514      918
1_100991376     1       1       0.889   4.654288e-02    917     1       1       0.111   -0.5725311      917     1       1       0.111   -0.9036341      916
1_101147138     1       1       0.542   9.742367e-02    910     1       1       0.542   3.536512e-01    910     1       1       0.458   -0.03433399     909
1_101970051     1       1       0.749   7.186614e-02    920     1       1       0.251   -0.4620248      920     1       1       0.251   -0.460245       919
1_102777667     1       1       0.753   2.551953e-01    885     1       1       0.753   3.865822e-01    885     1       1       0.753   7.600934e-01    884
1_103329033     1       1       0.172   4.645877e-01    920     1       1       0.828   -0.05000663     920     1       1       0.829   -0.4468977      919
1_103329034     1       1       0.172   4.645877e-01    920     1       1       0.828   -0.05000663     920     1       1       0.829   -0.4468977      919
1_103329036     1       1       0.172   4.645877e-01    920     1       1       0.828   -0.05000663     920     1       1       0.829   -0.4468977      919
1_103380393     1       1       0.945   3.685430e-01    897     1       1       0.055   -0.8057365      897     1       1       0.055   -0.5606755      896
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.3Pheno_1.assoc.IncAllele.MatchedToHb.formatted.RAF.txt.pre1.gz | perl -lane 'print $#F;' | sort | uniq -c
3510737 15
      1 30
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.3Pheno_1.assoc.IncAllele.MatchedToHb.formatted.RAF.txt.pre1.gz | wc
3510738 56171823 410228855
[  mturchin20@bigmem03  ~]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.3Pheno_2.assoc.IncAllele.MatchedToHb.formatted.RAF.txt.pre1.gz | head -n 10
CHR_POS CHR     SNP     A1frq   p       N       CHR     SNP     A1frq   p       N       CHR     SNP     A1frq   p       N
1_100640636     1       1       0.935   4.050135e-01    919     1       1       0.935   9.520115e-01    919     1       1       0.065   -0.3297514      918
1_100991376     1       1       0.889   1.069134e-01    917     1       1       0.889   3.111677e-01    917     1       1       0.111   -0.9036341      916
1_101147138     1       1       0.542   5.368052e-02    910     1       1       0.458   -0.6297548      910     1       1       0.458   -0.03433399     909
1_101970051     1       1       0.749   1.435012e-01    920     1       1       0.749   2.986489e-01    920     1       1       0.251   -0.460245       919
1_102777667     1       1       0.753   1.878538e-01    885     1       1       0.247   -0.429997       885     1       1       0.753   7.600934e-01    884
1_103329033     1       1       0.828   -0.7158805      920     1       1       0.172   3.384520e-02    920     1       1       0.829   -0.4468977      919
1_103329034     1       1       0.828   -0.7158805      920     1       1       0.172   3.384520e-02    920     1       1       0.829   -0.4468977      919
1_103329036     1       1       0.828   -0.7158805      920     1       1       0.172   3.384520e-02    920     1       1       0.829   -0.4468977      919
1_103380393     1       1       0.945   4.198021e-01    897     1       1       0.945   5.920738e-01    897     1       1       0.055   -0.5606755      896
[  mturchin20@bigmem03  ~]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.3Pheno_2.assoc.IncAllele.MatchedToHb.formatted.RAF.txt.pre1.gz | perl -lane 'print $#F;' | sort | uniq -c
3520010 15
      1 30
[  mturchin20@bigmem03  ~]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.3Pheno_2.assoc.IncAllele.MatchedToHb.formatted.RAF.txt.pre1.gz | wc
3520011 56320191 411104965
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.3Pheno_1.assoc.IncAllele.MatchedToHb.formatted.RAF.txt.pre1.gz | perl -lane 'if ($#F == 30) { print join("\t", @F); }'
2_178288061     2       P1      M       061510  2       50      M       0.499   2.595683e-01    921     2       P1      M       061510  2       50      M       0.501   -0.6665391      921     2       P1      M       0615102
        50      M       0.498   2.359597e-01    920
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Pulse.assoc.IncAllele.MatchedToHb.formatted.txt.gz | grep 178288061
2        178288061       P1_M_061510_2_50_M      0.498   2.359597e-01    920
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.3Pheno_1.assoc.IncAllele.MatchedToHb.formatted.RAF.txt.pre1.gz | perl -lane 'print $#F;' | sort | uniq -c
3510738 15
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.3Pheno_1.assoc.IncAllele.MatchedToHb.formatted.RAF.txt.pre1.gz | grep 178288061
2_178288061     2       P1-M-061510-2-50-M      0.499   2.595683e-01    921     2       P1-M-061510-2-50-M      0.501   -0.6665391      921     2       P1-M-061510-2-50-M      0.498   2.359597e-01    920
[  mturchin20@bigmem03  ~]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.3Pheno_2.assoc.IncAllele.MatchedToHb.formatted.RAF.txt.pre1.gz | perl -lane 'print $#F;' | sort | uniq -c                        3520011 15                                                                                                                                                                                                                     [  mturchin20@bigmem03  ~]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.3Pheno_2.assoc.IncAllele.MatchedToHb.formatted.RAF.txt.pre1.gz | grep 178288061                                                  2_178288061     2       P1-M-061510-2-50-M      0.499   2.006284e-01    921     2       P1-M-061510-2-50-M      0.499   5.362868e-01    921     2       P1-M-061510-2-50-M      0.498   2.359597e-01    920                   
~~~

zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.3Pheno_1.assoc.IncAllele.MatchedToHb.formatted.RAF.txt.pre1.gz | grep -v NA2 | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.3Pheno_1.assoc.IncAllele.MatchedToHb.formatted.RAF.txt.pre2.gz
zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.3Pheno_2.assoc.IncAllele.MatchedToHb.formatted.RAF.txt.pre1.gz | grep -v NA2 | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.3Pheno_2.assoc.IncAllele.MatchedToHb.formatted.RAF.txt.pre2.gz

#zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.3Pheno_1.assoc.IncAllele.MatchedToHb.formatted.RAF.txt.pre2.gz | perl -lane '($F[2], $F[0]) = ($F[0], $F[2]); $F[2] = ((split(/_/, $F[2]))[1]); print join("\t", @F);' | perl -lane 'if ($F[0] !~ /rs/) { $F[0] = $F[1] . "_" . $F[2]; } print join("\t", @F);' | perl -lane 'splice(@F, 16, 3); splice(@F, 11, 3); splice(@F, 6, 3); print join("\t", @F);' | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.3Pheno_1.assoc.IncAllele.MatchedToHb.formatted.RAF.txt.gz
#zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.3Pheno_2.assoc.IncAllele.MatchedToHb.formatted.RAF.txt.pre2.gz | perl -lane '($F[2], $F[0]) = ($F[0], $F[2]); $F[2] = ((split(/_/, $F[2]))[1]); print join("\t", @F);'  | perl -lane 'if ($F[0] !~ /rs/) { $F[0] = $F[1] . "_" . $F[2]; } print join("\t", @F);' | perl -lane 'splice(@F, 16, 3); splice(@F, 11, 3); splice(@F, 6, 3); print join("\t", @F);' | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.3Pheno_2.assoc.IncAllele.MatchedToHb.formatted.RAF.txt.gz
zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.3Pheno_1.assoc.IncAllele.MatchedToHb.formatted.RAF.txt.pre2.gz | perl -lane '($F[2], $F[0]) = ($F[0], $F[2]); $F[2] = ((split(/_/, $F[2]))[1]); print join("\t", @F);' | perl -lane 'splice(@F, 16, 3); splice(@F, 11, 3); splice(@F, 6, 3); print join("\t", @F);' | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.3Pheno_1.assoc.IncAllele.MatchedToHb.formatted.RAF.txt.gz
zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.3Pheno_2.assoc.IncAllele.MatchedToHb.formatted.RAF.txt.pre2.gz | perl -lane '($F[2], $F[0]) = ($F[0], $F[2]); $F[2] = ((split(/_/, $F[2]))[1]); print join("\t", @F);'  | perl -lane 'splice(@F, 16, 3); splice(@F, 11, 3); splice(@F, 6, 3); print join("\t", @F);' | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.3Pheno_2.assoc.IncAllele.MatchedToHb.formatted.RAF.txt.gz

#20160626 NOTE -- going to include one snp as a GWAS SNP for now? just to properly tag the region at least. May not really use the priors trained on just one SNP though -- maybe try training on like say the top 10-20 SNPs?
~~~
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.3Pheno_1.assoc.IncAllele.MatchedToHb.formatted.RAF.txt.gz | head -n 10
CHR_POS CHR     POS     A1frq   p       N       p       N       p       N
1_100640636     1       100640636       0.935   4.077197e-01    919     8.052751e-01    919     -0.3297514      918
1_100991376     1       100991376       0.889   4.654288e-02    917     -0.5725311      917     -0.9036341      916
1_101147138     1       101147138       0.542   9.742367e-02    910     3.536512e-01    910     -0.03433399     909
1_101970051     1       101970051       0.749   7.186614e-02    920     -0.4620248      920     -0.460245       919
1_102777667     1       102777667       0.753   2.551953e-01    885     3.865822e-01    885     7.600934e-01    884
1_103329033     1       103329033       0.172   4.645877e-01    920     -0.05000663     920     -0.4468977      919
1_103329034     1       103329034       0.172   4.645877e-01    920     -0.05000663     920     -0.4468977      919
1_103329036     1       103329036       0.172   4.645877e-01    920     -0.05000663     920     -0.4468977      919
1_103380393     1       103380393       0.945   3.685430e-01    897     -0.8057365      897     -0.5606755      896
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.3Pheno_1.assoc.IncAllele.MatchedToHb.formatted.RAF.txt.gz | grep rs | head -n 10
rs1000033       1       226580387       0.378   1.705824e-01    921     1.508678e-01    921     -0.1481057      920
rs1000050       1       162736463       0.429   7.061967e-02    920     -0.3896959      920     -0.4879369      919
rs1000073       1       157255396       0.126   2.015334e-01    911     -0.2014158      911     1.448819e-03    910
rs1000075       1       95166832        0.344   3.925941e-01    912     8.796695e-01    912     -0.6900962      911
rs1000184       1       8640831 0.662   5.826021e-01    913     9.268587e-04    913     4.758434e-01    912
rs1000282       1       209894785       0.794   8.221829e-01    921     6.688351e-01    921     6.584309e-01    920
rs1000283       1       209894661       0.816   7.676046e-01    911     6.995587e-01    911     3.669530e-01    910
rs1000313       1       15405489        0.237   6.936375e-01    898     5.742768e-01    898     5.673318e-01    897
rs1000314       1       15405907        0.291   8.754458e-01    903     5.786283e-01    903     4.987349e-01    902
rs1000348       1       15041242        0.127   6.950434e-01    913     2.753010e-01    913     -0.2354071      912
[  mturchin20@spudhead  ~/Software]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.*.assoc.txt.gz | awk '{ if ($10 < .00000005) { print $0 } } ' | head -n 10
2 rs374487821:46571435:G:C 46571435 881 G C 0.247 3.686509e-01 6.697345e-02 3.996904e-08
2 rs76242811:46576918:T:C 46576918 920 T C 0.251 3.742093e-01 6.567722e-02 1.343266e-08
2 rs188801636:46577251:T:C 46577251 920 T C 0.251 3.742093e-01 6.567722e-02 1.343266e-08
2 rs375554942:46579689:A:G 46579689 915 A G 0.252 3.687283e-01 6.546711e-02 1.950578e-08
2 rs189807021:46583581:G:A 46583581 912 G A 0.252 3.703709e-01 6.555872e-02 1.768995e-08
2 rs372272284:46584859:A:G 46584859 914 A G 0.248 3.860167e-01 6.603758e-02 5.713546e-09
2 rs150877473:46588019:C:G 46588019 907 C G 0.252 3.694678e-01 6.571006e-02 2.055258e-08
2 rs142826801:46588331:G:C 46588331 889 G C 0.255 3.710295e-01 6.607682e-02 2.143913e-08
[  mturchin20@spudhead  ~/Software]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.*.assoc.txt.gz | awk '{ if ($10 < .00000009) { print $0 } } ' | head -n 10
2 rs149306391:46571017:C:G 46571017 879 C G 0.247 3.608638e-01 6.692122e-02 7.388210e-08
2 rs374487821:46571435:G:C 46571435 881 G C 0.247 3.686509e-01 6.697345e-02 3.996904e-08
2 rs76242811:46576918:T:C 46576918 920 T C 0.251 3.742093e-01 6.567722e-02 1.343266e-08
2 rs188801636:46577251:T:C 46577251 920 T C 0.251 3.742093e-01 6.567722e-02 1.343266e-08
2 rs375554942:46579689:A:G 46579689 915 A G 0.252 3.687283e-01 6.546711e-02 1.950578e-08
2 rs189807021:46583581:G:A 46583581 912 G A 0.252 3.703709e-01 6.555872e-02 1.768995e-08
2 rs372272284:46584859:A:G 46584859 914 A G 0.248 3.860167e-01 6.603758e-02 5.713546e-09
2 rs150877473:46588019:C:G 46588019 907 C G 0.252 3.694678e-01 6.571006e-02 2.055258e-08
2 rs142826801:46588331:G:C 46588331 889 G C 0.255 3.710295e-01 6.607682e-02 2.143913e-08
2 rs74898705:46589032:C:T 46589032 883 C T 0.255 3.612890e-01 6.626266e-02 5.306511e-08
~~~

mkdir /mnt/lustre/home/mturchin20/Data/Choongwon2016

zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.*.assoc.txt.gz | awk '{ if ($10 < .00000005) { print $0 } } ' | perl -lane 'my $rsID = ((split(/:/, $F[1]))[0]); print $rsID, "\t", $F[0], "\t", $F[2];' > /mnt/lustre/home/mturchin20/Data/Choongwon2016/Tibetan.AllPhenos.assoc.SNPslt5eneg8.MarkerChrBP.txt 
#Choose the SNP with the most significant p-value to represent the region since the cluster were all within ~100kb of one another or so (and I use 1Mb windows to define a region for purpose of pruning/clumping anyways)
zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.*.assoc.txt.gz | awk '{ if ($10 < .00000005) { print $0 } } ' | R -q -e "Data1 <- read.table(file('stdin'), header=F); Data1[which.min(as.numeric(as.character(Data1[,10]))),];" | grep -v V | grep -v \> | perl -ane 'print join("\t", @F[1..$#F]), "\t";' | perl -lane 'print join("\t", @F);' | perl -lane 'my $rsID = ((split(/:/, $F[1]))[0]); print $rsID, "\t", $F[0], "\t", $F[2];' > /mnt/lustre/home/mturchin20/Data/Choongwon2016/Tibetan.AllPhenos.assoc.SNPslt5eneg8.MarkerChrBP.GreedyPruned.txt

~~~
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.*.assoc.txt.gz | awk '{ if ($10 < .00000005) { print $0 } } '       
2 rs374487821:46571435:G:C 46571435 881 G C 0.247 3.686509e-01 6.697345e-02 3.996904e-08
2 rs76242811:46576918:T:C 46576918 920 T C 0.251 3.742093e-01 6.567722e-02 1.343266e-08
2 rs188801636:46577251:T:C 46577251 920 T C 0.251 3.742093e-01 6.567722e-02 1.343266e-08
2 rs375554942:46579689:A:G 46579689 915 A G 0.252 3.687283e-01 6.546711e-02 1.950578e-08
2 rs189807021:46583581:G:A 46583581 912 G A 0.252 3.703709e-01 6.555872e-02 1.768995e-08
2 rs372272284:46584859:A:G 46584859 914 A G 0.248 3.860167e-01 6.603758e-02 5.713546e-09
2 rs150877473:46588019:C:G 46588019 907 C G 0.252 3.694678e-01 6.571006e-02 2.055258e-08
2 rs142826801:46588331:G:C 46588331 889 G C 0.255 3.710295e-01 6.607682e-02 2.143913e-08
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.*.assoc.txt.gz | awk '{ if ($10 < .00000005) { print $0 } } ' | R -q -e "Data1 <- read.table(file('stdin'), header=F); Data1[which.min(as.numeric(as.character(Data1[,10]))),];"
> Data1 <- read.table(file('stdin'), header=F); Data1[which.min(as.numeric(as.character(Data1[,10]))),];
  V1                       V2       V3  V4 V5 V6    V7        V8         V9
6  2 rs372272284:46584859:A:G 46584859 914  A  G 0.248 0.3860167 0.06603758
           V10
6 5.713546e-09
> 
> 
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.*.assoc.txt.gz | awk '{ if ($10 < .00000005) { print $0 } } ' | R -q -e "Data1 <- read.table(file('stdin'), header=F); Data1[which.min(as.numeric(as.character(Data1[,10]))),];" | grep -v V | grep -v \> | perl -ane 'print join("\t", @F[1..$#F]), "\t";' | perl -lane 'print join("\t", @F);'
2       rs372272284:46584859:A:G        46584859        914     A       G       0.248   0.3860167       0.06603758      5.713546e-09
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$for i in `ls -lrt /mnt/gluster/data/external_private_supp/Choongwon2016/. | grep Tibetan.*.assoc.txt.gz | awk '{ print $9 }'`; do echo $i; zcat /mnt/gluster/data/external_private_supp/Choongwon2016/$i | grep rs372272284 ; done
Tibetan.Hb.assoc.txt.gz
2 rs372272284:46584859:A:G 46584859 914 A G 0.248 3.825358e-01 7.466053e-02 2.952695e-07
Tibetan.Sat.assoc.txt.gz
2 rs372272284:46584859:A:G 46584859 914 A G 0.248 3.463506e-01 2.106146e-01 9.929316e-02
Tibetan.Pulse.assoc.txt.gz
2 rs372272284:46584859:A:G 46584859 913 A G 0.248 1.997644e-01 5.883982e-01 7.307703e-01
Tibetan.OxHb.assoc.txt.gz
2 rs372272284:46584859:A:G 46584859 914 A G 0.248 3.860167e-01 6.603758e-02 5.713546e-09
Tibetan.dHb.assoc.txt.gz
2 rs372272284:46584859:A:G 46584859 914 A G 0.248 -5.136918e-03 3.332381e-02 8.825847e-01
~~~


#python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/Choongwon2016/Tibetan.AllPhenos.assoc.SNPslt5eneg8.MarkerChrBP.txt --file2 /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.3Pheno_1.assoc.IncAllele.MatchedToHb.formatted.RAF.txt.gz | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.3Pheno_1.assoc.IncAllele.MatchedToHb.formatted.RAF.wGWASannot.txt.gz 
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/Choongwon2016/Tibetan.AllPhenos.assoc.SNPslt5eneg8.MarkerChrBP.GreedyPruned.txt --file2 /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.3Pheno_1.assoc.IncAllele.MatchedToHb.formatted.RAF.txt.gz | perl -lane '$F[10] = 0; print join("\t", @F);' | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.3Pheno_1.assoc.IncAllele.MatchedToHb.formatted.RAF.wGWASannot.txt.gz 
#python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/Choongwon2016/Tibetan.AllPhenos.assoc.SNPslt5eneg8.MarkerChrBP.txt --file2 /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.3Pheno_2.assoc.IncAllele.MatchedToHb.formatted.RAF.txt.gz | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.3Pheno_2.assoc.IncAllele.MatchedToHb.formatted.RAF.wGWASannot.txt.gz 
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/Choongwon2016/Tibetan.AllPhenos.assoc.SNPslt5eneg8.MarkerChrBP.GreedyPruned.txt --file2 /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.3Pheno_2.assoc.IncAllele.MatchedToHb.formatted.RAF.txt.gz | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.3Pheno_2.assoc.IncAllele.MatchedToHb.formatted.RAF.wGWASannot.txt.gz 

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1/process.MTedits.ForGEFOS2015.vs1.SignCrrct.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1/process.MTedits.ForChoongwon2016.3Phenos_1.vs1.SignCrrct.vs1.R 
cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1/process.MTedits.ForChoongwon2016.3Phenos_1.vs1.SignCrrct.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1/process.MTedits.ForChoongwon2016.3Phenos_2.vs1.SignCrrct.vs1.R

~~~
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.3Pheno_1.assoc.IncAllele.MatchedToHb.formatted.RAF.wGWASannot.txt.gz | wc
3503644 38547132 276906515
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.3Pheno_2.assoc.IncAllele.MatchedToHb.formatted.RAF.wGWASannot.txt.gz | wc
3494180 38443028 276380152
#[  mturchin20@bigmem03  ~]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.3Pheno_1.assoc.IncAllele.MatchedToHb.formatted.RAF.wGWASannot.txt.gz | perl -lane 'print $F[$#F];' | sort | uniq -c
#3501882 0
#      8 1
#   1754 2
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.3Pheno_1.assoc.IncAllele.MatchedToHb.formatted.RAF.wGWASannot.txt.gz | perl -lane 'print $F[10];' | sort | uniq -c
3503644 0
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.3Pheno_1.assoc.IncAllele.MatchedToHb.formatted.RAF.wGWASannot.txt.gz | perl -lane 'print $F[$#F];' | sort | uniq -c
3501909 0
      1 1
   1734 2
#[  mturchin20@bigmem03  ~]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.3Pheno_2.assoc.IncAllele.MatchedToHb.formatted.RAF.wGWASannot.txt.gz | perl -lane 'print $F[$#F];' | sort | uniq -c
#3492418 0
#      8 1
#   1754 2
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.3Pheno_2.assoc.IncAllele.MatchedToHb.formatted.RAF.wGWASannot.txt.gz | perl -lane 'print $F[10];' | sort | uniq -c
3492445 0
      1 1
   1734 2
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.3Pheno_2.assoc.IncAllele.MatchedToHb.formatted.RAF.wGWASannot.txt.gz | perl -lane 'print $F[$#F];' | sort | uniq -c
3492445 0
      1 1
   1734 2
~~~

LANG2=${LANG}
LC_ALL2=${LC_ALL}
LANG=C 
LC_ALL=C
export LC_ALL LANG

join -a 2 -1 1 -2 1 -e NA4 -o 0 1.2 1.3 1.4 1.5 1.6 1.7 1.8 1.9 1.10 1.11 2.2 2.3 2.4 2.5 2.6 2.7 <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.3Pheno_1.assoc.IncAllele.MatchedToHb.formatted.RAF.wGWASannot.txt.gz | sort -g -k 1,1 ) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1/Choongwon2016.3Phenos_1.dtlesslesssignif.vs1.SignCrrct.vs1.txt | perl -F, -lane 'print join("\t", @F);' | sort -g -k 1,1) | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1/Choongwon2016.3Phenos_1.dtlesslesssignif.vs1.SignCrrct.vs1.annot.vs1.RAF.txt.pre1.gz
join -a 2 -1 1 -2 1 -e NA4 -o 0 1.2 1.3 1.4 1.5 1.6 1.7 1.8 1.9 1.10 1.11 2.2 2.3 2.4 2.5 2.6 2.7 <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.3Pheno_2.assoc.IncAllele.MatchedToHb.formatted.RAF.wGWASannot.txt.gz | sort -g -k 1,1 ) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1/Choongwon2016.3Phenos_2.dtlesslesssignif.vs1.SignCrrct.vs1.txt | perl -F, -lane 'print join("\t", @F);' | sort -g -k 1,1) | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1/Choongwon2016.3Phenos_2.dtlesslesssignif.vs1.SignCrrct.vs1.annot.vs1.RAF.txt.pre1.gz

LANG=${LANG2}
LC_ALL=${LC_ALL2}
export LC_ALL LANG

~~~
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1/Choongwon2016.3Phenos_1.dtlesslesssignif.vs1.SignCrrct.vs1.annot.vs1.RAF.txt.pre1.gz | wc
   1491   25347  273873
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1/Choongwon2016.3Phenos_1.dtlesslesssignif.vs1.SignCrrct.vs1.txt | wc
   1491    1491  170038
[  mturchin20@bigmem03  ~]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1/Choongwon2016.3Phenos_2.dtlesslesssignif.vs1.SignCrrct.vs1.annot.vs1.RAF.txt.pre1.gz | wc
   3757   63869  685618
[  mturchin20@bigmem03  ~]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1/Choongwon2016.3Phenos_2.dtlesslesssignif.vs1.SignCrrct.vs1.txt | wc
   3757    3757  424787
~~~

zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1/Choongwon2016.3Phenos_1.dtlesslesssignif.vs1.SignCrrct.vs1.annot.vs1.RAF.txt.pre1.gz | grep -v NA4 | cat <(echo "snp chr pos raf p_Hb n_Hb p_Sat n_Sat p_Pulse n_Pulse annot Z.Hb Z.Sat Z.Pulse mvstat mvp unip") - | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1/Choongwon2016.3Phenos_1.dtlesslesssignif.vs1.SignCrrct.vs1.annot.vs1.RAF.txt.gz 
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1/Choongwon2016.3Phenos_2.dtlesslesssignif.vs1.SignCrrct.vs1.annot.vs1.RAF.txt.pre1.gz | grep -v NA4 | cat <(echo "snp chr pos raf p_OxHb n_OxHb p_dHb n_dHb p_Pulse n_Pulse annot Z.OxHb Z.dHb Z.Pulse mvstat mvp unip") - | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1/Choongwon2016.3Phenos_2.dtlesslesssignif.vs1.SignCrrct.vs1.annot.vs1.RAF.txt.gz 

zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1/Choongwon2016.3Phenos_1.dtlesslesssignif.vs1.SignCrrct.vs1.annot.vs1.RAF.txt.gz | perl -lane 'if ($F[3] > .5) { $F[3] = 1 - $F[3]; $F[11] = $F[11] * -1; $F[12] = $F[12] * -1; $F[13] = $F[13] * -1; } print join("\t", @F);' | sed 's/raf/maf/g' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1/Choongwon2016.3Phenos_1.dtlesslesssignif.vs1.SignCrrct.vs1.annot.vs1.MAF.txt.gz 
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1/Choongwon2016.3Phenos_2.dtlesslesssignif.vs1.SignCrrct.vs1.annot.vs1.RAF.txt.gz | perl -lane 'if ($F[3] > .5) { $F[3] = 1 - $F[3]; $F[11] = $F[11] * -1; $F[12] = $F[12] * -1; $F[13] = $F[13] * -1; } print join("\t", @F);' | sed 's/raf/maf/g' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1/Choongwon2016.3Phenos_2.dtlesslesssignif.vs1.SignCrrct.vs1.annot.vs1.MAF.txt.gz 

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1/GLC.MTedits.ForGEFOS2015.vs1.SignCrrct.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1/GLC.MTedits.ForChoongwon2016.3Phenos_1.vs1.SignCrrct.vs1.R 
cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1/GLC.MTedits.ForChoongwon2016.3Phenos_1.vs1.SignCrrct.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1/GLC.MTedits.ForChoongwon2016.3Phenos_2.vs1.SignCrrct.vs1.R

~~~
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$for i in `ls -lrt | tail -n 4 | awk '{ print $9 }'`; do echo $i ; cat $i | perl -lane 'print $#F;' | sort | uniq -c; done
Choongwon2016.PhenoGroup1.Pruned.logBFs.unifpriors.txt
    204 27
Choongwon2016.PhenoGroup1.Pruned.logBFs.txt
    204 27
Choongwon2016.PhenoGroup1.NotPruned.logBFs.unifpriors.txt
   1485 27
Choongwon2016.PhenoGroup1.NotPruned.logBFs.txt
   1485 27

[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$cat Choongwon2016.PhenoGroup1.Pruned.logBFs.unifpriors.txt | perl -lane 'print join("\t", @F[0..9]);' | head -n 10
rsID    0_0_0   1_0_0   2_0_0   0_1_0   1_1_0   2_1_0   0_2_0   1_2_0   2_2_0
rs10086940      0       -0.383899542533573      0       -2.24291626019867       -1.01778755158522       -2.84775689351885       0       -0.992312083284313      0
rs10105487      0       -2.12717088843192       0       0.0408995484352279      -0.254778784196617      -0.445165739566762      0       -2.66757187043917       0
rs10112319      0       -2.17412156513259       0       -2.01950204646844       -2.41091577085454       -2.63261069362767       0       -2.784853151545 0
rs10162740      0       0.0249651449716843      0       -2.13892211218601       -0.497114025372646      -2.77716922169236       0       -0.661737307419327      0
rs10184862      0       -0.464714024227872      0       -1.37787138739875       -0.00352125644187096    -2.1669362611544        0       -1.29150996927521       0
rs10211560      0       -2.1129524939705        0       -2.24032460383622       -2.54132093188855       -2.84102848175839       0       -2.71335636022791       0
rs10246790      0       -0.340819287256193      0       -2.18129228561177       -0.905204902031747      -2.79621746389889       0       -0.976050160673852      0
rs10271491      0       -2.16381280375546       0       -0.784139335885207      -1.2359497111108        -1.3500550890348        0       -2.75204782639597       0
rs10271931      0       -2.16381280375546       0       -0.784139335885207      -1.2359497111108        -1.3500550890348        0       -2.75204782639597       0
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$cat Choongwon2016.PhenoGroup1.Pruned.logBFs.txt | perl -lane 'print join("\t", @F[0..9]);' | head -n 10
rsID    0_0_0   1_0_0   2_0_0   0_1_0   1_1_0   2_1_0   0_2_0   1_2_0   2_2_0
rs10086940      0       1.71647100258399        0       -0.142545715081106      1.38361298919633        -0.145326357073324      0       1.71011845316121        0
rs10105487      0       -0.0268003433143619     0       2.14127009355279        2.14662175658493        2.25726479687876        0       0.0348586660063551      0
rs10112319      0       -0.0737510200150231     0       0.0808684986491212      -0.00951523007299609    0.0698198428178557      0       -0.0824226150994758     0
rs10162740      0       2.12533569008925        0       -0.0385515670684462     1.9042865154089 -0.0747386852468393     0       2.0406932290262 0
rs10184862      0       1.63565652088969        0       0.722499157718808       2.39787928433967        0.53549427529113        0       1.41092056717032        0
rs10211560      0       -0.0125819488529395     0       -0.13995405871866       -0.139920391107002      -0.13859794531286       0       -0.0109258237823888     0
rs10246790      0       1.75955125786137        0       -0.0809217404942063     1.4961956387498 -0.0937869274533661     0       1.72638037577167        0
rs10271491      0       -0.063442258637896      0       1.31623120923236        1.16545082967074        1.35237544741072        0       -0.0496172899504485     0
rs10271931      0       -0.063442258637896      0       1.31623120923236        1.16545082967074        1.35237544741072        0       -0.0496172899504485     0
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$cat Choongwon2016.PhenoGroup1.NotPruned.logBFs.unifpriors.txt | perl -lane 'print join("\t", @F[0..9]);' | head -n 10
rsID    0_0_0   1_0_0   2_0_0   0_1_0   1_1_0   2_1_0   0_2_0   1_2_0   2_2_0
rs10042098      0       -2.22309006215269       0       -2.21788485402604       -2.61223805064344       -2.82220896851851       0       -2.82739325094484       0
rs10069542      0       -2.22309006215269       0       -2.21788485402604       -2.61223805064344       -2.82220896851851       0       -2.82739325094484       0
rs1008264       0       -2.18143423856731       0       -2.18717779688326       -2.55482484121169       -2.79302166907504       0       -2.78731843379727       0
rs10086302      0       -2.22901323818902       0       0.403864767784846       -0.259772465567102      -0.232856380031319      0       -2.8457310572076        0
rs10086401      0       0.0861666326940878      0       -2.19714832273974       -0.524551220386128      -2.81882069257412       0       -0.563290855479418      0
rs10086462      0       -0.348281275982722      0       -2.20048372008931       -0.941009696528103      -2.81914269148059       0       -0.989396225118785      0
rs10086940      0       -0.383899542533573      0       -2.24291626019867       -1.01778755158522       -2.84775689351885       0       -0.992312083284313      0
rs10088178      0       -2.17947483812086       0       -2.22995717504615       -2.58623288182988       -2.83227866275719       0       -2.78182098120182       0
rs10089136      0       -2.24630103337055       0       -0.0348286335969847     -0.680109385580797      -0.631624245437941      0       -2.84609194420469       0
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$cat Choongwon2016.PhenoGroup1.NotPruned.logBFs.txt | perl -lane 'print join("\t", @F[0..9]);' | head -n 10
rsID    0_0_0   1_0_0   2_0_0   0_1_0   1_1_0   2_1_0   0_2_0   1_2_0   2_2_0
rs10042098      0       -0.122719517035124      0       -0.11751430890848       -0.210837509861898      -0.119778432072985      0       -0.124962714499316      0
rs10069542      0       -0.122719517035124      0       -0.11751430890848       -0.210837509861898      -0.119778432072985      0       -0.124962714499316      0
rs1008264       0       -0.0810636934497477     0       -0.0868072517656991     -0.153424300430147      -0.0905911326295124     0       -0.0848878973517462     0
rs10086302      0       -0.128642693071458      0       2.50423531290241        2.14162807521444        2.46957415641421        0       -0.143300520762077      0
rs10086401      0       2.18653717781165        0       -0.0967777776221752     1.87684932039542        -0.116390156128593      0       2.13913968096611        0
rs10086462      0       1.75208926913484        0       -0.100113174971743      1.46039084425344        -0.116712155035065      0       1.71303431132674        0
rs10086940      0       1.71647100258399        0       -0.142545715081106      1.38361298919633        -0.145326357073324      0       1.71011845316121        0
rs10088178      0       -0.0791042930033012     0       -0.129586629928589      -0.184832341048336      -0.12984812631166       0       -0.07939044475629       0
rs10089136      0       -0.145930488252985      0       2.06554191152058        1.72129115520075        2.07080629100758        0       -0.14366140775917       0



~~~

#20160727 -- Choongwon came by to scp over one other phenotype file Tibetan_livebirth_GWAS.txt.gz, and we are now considering redoing the analyses with this extra phenotype as part of each subset
#Maybe make this a 'Vs2' analysis for Choongwon2016?

~~~
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan_livebirth_GWAS.txt.gz | wc        
3507698 35076980 310268358
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan_livebirth_GWAS.txt.gz | head -n 10
CHR SNP Pos N A1 A0 A1frq beta se p
1 rs3094315:752566:G:A 752566 981 G A 0.150 -9.671835e-02 1.614483e-01 5.373019e-01
1 rs3131972:752721:A:G 752721 981 A G 0.280 -8.061955e-02 1.349635e-01 5.344548e-01
1 rs3131971:752894:T:C 752894 962 T C 0.150 -3.701661e-02 1.623434e-01 8.065017e-01
1 rs3131967:754334:T:C 754334 953 T C 0.272 -2.098065e-02 1.385687e-01 8.593784e-01
1 rs3115859:754503:G:A 754503 967 G A 0.281 -9.102407e-02 1.354852e-01 4.867832e-01
1 rs3131966:754964:C:T 754964 966 C T 0.281 -8.683694e-02 1.355087e-01 5.064970e-01
1 rs3115858:755890:A:T 755890 957 A T 0.140 4.057407e-02 1.678690e-01 8.247668e-01
1 rs61768170:756434:G:C 756434 945 C G 0.135 2.835339e-02 1.701335e-01 8.807469e-01
1 rs3131962:756604:A:G 756604 943 A G 0.145 -1.902186e-02 1.668715e-01 8.937007e-01
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan_livebirth_GWAS.txt.gz | perl -lane 'print $#F;' | sort | uniq -c
3507698 9
~~~

mv /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan_livebirth_GWAS.txt.gz /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.LvBrth.assoc.txt.gz

mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs2/

LANG2=${LANG}
LC_ALL2=${LC_ALL}
LANG=C
LC_ALL=C
export LC_ALL LANG

#2 data groups:
#Hb (Hemoglobin)
#Sat (Oxygen Saturation)
#Pulse
#
#
#OxHb (Oxygenated Hemoglobin)
#dHb (Dexoygenated Hemoglobin) 
#Pulse
#

join <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Hb.assoc.IncAllele.txt.gz | awk '{ print $1, "\t", $7 }' | sort ) <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.LvBrth.assoc.txt.gz | perl -lane 'my $rsID = ((split(/:/, $F[1]))[0]); my $ChrBP = $F[0] . "_" . $F[2]; splice(@F, 0, 0, ($ChrBP, $rsID)); if ($F[9] < 0) { ($F[6], $F[7]) = ($F[7], $F[6]); $F[8] = 1 - $F[8]; $F[9] = $F[9] * -1; } print join("\t", @F);' | sort) | perl -lane 'if ($F[1] ne $F[7]) { $F[12] = $F[12] * -1; } splice(@F, 1, 1); print join("\t", @F);' | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.LvBrth.assoc.IncAllele.MatchedToHb.txt.gz

zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.LvBrth.assoc.IncAllele.MatchedToHb.txt.gz | awk '{ print $3, "\t", $5, "\t", $2, "\t", $9, "\t", $12, "\t", $6 }' | cat <(echo "CHR      POS     SNP     A1frq   p       N") <(cat - | grep -v A1frq) | perl -lane 'if ($F[2] !~ /\D/ || $F[2] eq "X") { $F[2] = $F[0] . "_" . $F[1]; } print join("\t", @F); '  | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.LvBrth.assoc.IncAllele.MatchedToHb.formatted.txt.gz 

~~~
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.LvBrth.assoc.IncAllele.MatchedToHb.formatted.txt.gz | wc
3499632 20997792 153952720
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.LvBrth.assoc.IncAllele.MatchedToHb.formatted.txt.gz | head -n 10
CHR     POS     SNP     A1frq   p       N
10      100000625       rs7899632       0.425   -0.1777584      974
10      100000645       rs61875309      0.060   -0.1983767      975
10      100002399       rs8181398       0.947   8.854789e-01    976
10      100003242       rs12258651      0.120   4.164910e-01    977
10      100003785       rs1359508       0.539   -0.04605376     979
10      100004360       rs1048754       0.060   -0.1983767      975
10      100004441       rs1048757       0.485   -0.04806174     977
10      100004906       rs3750595       0.424   -0.17937        977
10      100004996       rs2025625       0.539   -0.04663817     980
~~~


join -a 1 -a 2 -1 3 -2 3 -e NA2 -o 0 1.1 1.2 1.4 1.5 1.6 2.1 2.2 2.4 2.5 2.6 <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Hb.assoc.IncAllele.formatted.txt.gz | sed 's/_/-/g' | perl -lane '($F[2], $F[1]) = ($F[1], $F[2]); $F[2] = $F[0] . "_" . $F[2]; print join("\t", @F);' | sort -g -k 3,3) <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Sat.assoc.IncAllele.MatchedToHb.formatted.txt.gz | sed 's/_/-/g' | perl -lane '($F[2], $F[1]) = ($F[1], $F[2]); $F[2] = $F[0] . "_" . $F[2]; print join("\t", @F);' | sort -g -k 3,3 ) | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
join -a 1 -a 2 -1 1 -2 3 -e NA2 -o 0 1.2 2.1 2.2 2.4 2.5 2.6 - <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Pulse.assoc.IncAllele.MatchedToHb.formatted.txt.gz | sed 's/_/-/g' | perl -lane '($F[2], $F[1]) = ($F[1], $F[2]); $F[2] = $F[0] . "_" . $F[2]; print join("\t", @F);' | sort -g -k 3,3) | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); print join("\t", @F);' | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); if ($#F == 6) { splice(@F, 1, 0, (("NA2") x 9)); } my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
join -a 1 -a 2 -1 1 -2 3 -e NA2 -o 0 1.2 2.1 2.2 2.4 2.5 2.6 - <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.LvBrth.assoc.IncAllele.MatchedToHb.formatted.txt.gz | sed 's/_/-/g' | perl -lane '($F[2], $F[1]) = ($F[1], $F[2]); $F[2] = $F[0] . "_" . $F[2]; print join("\t", @F);' | sort -g -k 3,3) | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); print join("\t", @F);' | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); if ($#F == 6) { splice(@F, 1, 0, (("NA2") x 9)); } my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
perl -lane 'print $F[0], "\t", join("\t", split(/_/, $F[1]));' | sed 's/chr//g' | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.4Pheno_1.assoc.IncAllele.MatchedToHb.formatted.RAF.txt.pre1.gz

join -a 1 -a 2 -1 3 -2 3 -e NA2 -o 0 1.1 1.2 1.4 1.5 1.6 2.1 2.2 2.4 2.5 2.6 <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.OxHb.assoc.IncAllele.MatchedToHb.formatted.txt.gz | sed 's/_/-/g' | perl -lane '($F[2], $F[1]) = ($F[1], $F[2]); $F[2] = $F[0] . "_" . $F[2]; print join("\t", @F);' | sort -g -k 3,3) <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.dHb.assoc.IncAllele.MatchedToHb.formatted.txt.gz | sed 's/_/-/g' | perl -lane '($F[2], $F[1]) = ($F[1], $F[2]); $F[2] = $F[0] . "_" . $F[2]; print join("\t", @F);' | sort -g -k 3,3 ) | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
join -a 1 -a 2 -1 1 -2 3 -e NA2 -o 0 1.2 2.1 2.2 2.4 2.5 2.6 - <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Pulse.assoc.IncAllele.MatchedToHb.formatted.txt.gz | sed 's/_/-/g' | perl -lane '($F[2], $F[1]) = ($F[1], $F[2]); $F[2] = $F[0] . "_" . $F[2]; print join("\t", @F);' | sort -g -k 3,3) | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); print join("\t", @F);' | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); if ($#F == 6) { splice(@F, 1, 0, (("NA2") x 9)); } my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
join -a 1 -a 2 -1 1 -2 3 -e NA2 -o 0 1.2 2.1 2.2 2.4 2.5 2.6 - <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.LvBrth.assoc.IncAllele.MatchedToHb.formatted.txt.gz | sed 's/_/-/g' | perl -lane '($F[2], $F[1]) = ($F[1], $F[2]); $F[2] = $F[0] . "_" . $F[2]; print join("\t", @F);' | sort -g -k 3,3) | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); print join("\t", @F);' | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); if ($#F == 6) { splice(@F, 1, 0, (("NA2") x 9)); } my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
perl -lane 'print $F[0], "\t", join("\t", split(/_/, $F[1]));' | sed 's/chr//g' | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.4Pheno_2.assoc.IncAllele.MatchedToHb.formatted.RAF.txt.pre1.gz

LANG=${LANG2}
LC_ALL=${LC_ALL2}
export LC_ALL LANG

zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.4Pheno_1.assoc.IncAllele.MatchedToHb.formatted.RAF.txt.pre1.gz | grep -v NA2 | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.4Pheno_1.assoc.IncAllele.MatchedToHb.formatted.RAF.txt.pre2.gz
zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.4Pheno_2.assoc.IncAllele.MatchedToHb.formatted.RAF.txt.pre1.gz | grep -v NA2 | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.4Pheno_2.assoc.IncAllele.MatchedToHb.formatted.RAF.txt.pre2.gz

#zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.4Pheno_1.assoc.IncAllele.MatchedToHb.formatted.RAF.txt.pre2.gz | perl -lane '($F[2], $F[0]) = ($F[0], $F[2]); $F[2] = ((split(/_/, $F[2]))[1]); print join("\t", @F);' | perl -lane 'if ($F[0] !~ /rs/) { $F[0] = $F[1] . "_" . $F[2]; } print join("\t", @F);' | perl -lane 'splice(@F, 16, 3); splice(@F, 11, 3); splice(@F, 6, 3); print join("\t", @F);' | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.4Pheno_1.assoc.IncAllele.MatchedToHb.formatted.RAF.txt.gz
#zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.4Pheno_2.assoc.IncAllele.MatchedToHb.formatted.RAF.txt.pre2.gz | perl -lane '($F[2], $F[0]) = ($F[0], $F[2]); $F[2] = ((split(/_/, $F[2]))[1]); print join("\t", @F);'  | perl -lane 'if ($F[0] !~ /rs/) { $F[0] = $F[1] . "_" . $F[2]; } print join("\t", @F);' | perl -lane 'splice(@F, 16, 3); splice(@F, 11, 3); splice(@F, 6, 3); print join("\t", @F);' | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.4Pheno_2.assoc.IncAllele.MatchedToHb.formatted.RAF.txt.gz
zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.4Pheno_1.assoc.IncAllele.MatchedToHb.formatted.RAF.txt.pre2.gz | perl -lane '($F[2], $F[0]) = ($F[0], $F[2]); $F[2] = ((split(/_/, $F[2]))[1]); print join("\t", @F);' | perl -lane 'splice(@F, 16, 3); splice(@F, 11, 3); splice(@F, 6, 3); print join("\t", @F);' | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.4Pheno_1.assoc.IncAllele.MatchedToHb.formatted.RAF.txt.gz
zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.4Pheno_2.assoc.IncAllele.MatchedToHb.formatted.RAF.txt.pre2.gz | perl -lane '($F[2], $F[0]) = ($F[0], $F[2]); $F[2] = ((split(/_/, $F[2]))[1]); print join("\t", @F);'  | perl -lane 'splice(@F, 16, 3); splice(@F, 11, 3); splice(@F, 6, 3); print join("\t", @F);' | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.4Pheno_2.assoc.IncAllele.MatchedToHb.formatted.RAF.txt.gz



#'vs2', eg both below, now have LvBrth
zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.*.assoc.txt.gz | awk '{ if ($10 < .00000005) { print $0 } } ' | perl -lane 'my $rsID = ((split(/:/, $F[1]))[0]); print $rsID, "\t", $F[0], "\t", $F[2];' > /mnt/lustre/home/mturchin20/Data/Choongwon2016/Tibetan.AllPhenos.assoc.SNPslt5eneg8.MarkerChrBP.vs2.txt 
#Choose the SNP with the most significant p-value to represent the region since the cluster were all within ~100kb of one another or so (and I use 1Mb windows to define a region for purpose of pruning/clumping anyways)
zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.*.assoc.txt.gz | awk '{ if ($10 < .00000005) { print $0 } } ' | R -q -e "Data1 <- read.table(file('stdin'), header=F); Data1[which.min(as.numeric(as.character(Data1[,10]))),];" | grep -v V | grep -v \> | perl -ane 'print join("\t", @F[1..$#F]), "\t";' | perl -lane 'print join("\t", @F);' | perl -lane 'my $rsID = ((split(/:/, $F[1]))[0]); print $rsID, "\t", $F[0], "\t", $F[2];' > /mnt/lustre/home/mturchin20/Data/Choongwon2016/Tibetan.AllPhenos.assoc.SNPslt5eneg8.MarkerChrBP.vs2.GreedyPruned.LvBrth.txt
cat /mnt/lustre/home/mturchin20/Data/Choongwon2016/Tibetan.AllPhenos.assoc.SNPslt5eneg8.MarkerChrBP.GreedyPruned.txt /mnt/lustre/home/mturchin20/Data/Choongwon2016/Tibetan.AllPhenos.assoc.SNPslt5eneg8.MarkerChrBP.vs2.GreedyPruned.LvBrth.txt > /mnt/lustre/home/mturchin20/Data/Choongwon2016/Tibetan.AllPhenos.assoc.SNPslt5eneg8.MarkerChrBP.vs2.GreedyPruned.OxHb_LvBrth.txt

for i in `ls -lrt /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.*.assoc.txt.gz | awk '{ print $9 }'`; do
	echo $i; zcat $i | awk '{ if ($10 < .00000005) { print $0 } } ' | perl -lane 'my $rsID = ((split(/:/, $F[1]))[0]); print $rsID, "\t", $F[0], "\t", $F[2];';
done

#20160809 NOTE -- Normally would want to do the min and then remove 1Mb window around, but since there's only one locus in each of the two files being considered, can 'cheat' and just take the min from within each file. If there were more than one significant region in any particular file, this would clearly just produce a single result and not the most significant hit for each region.
for i in `ls -lrt /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.*.assoc.txt.gz | awk '{ print $9 }'`; do
	echo $i; zcat $i | awk '{ if ($10 < .00000005) { print $0 } } ' | R -q -e "Data1 <- read.table(file('stdin'), header=F); Data1[which.min(as.numeric(as.character(Data1[,10]))),];" | grep -v V | grep -v \> | perl -ane 'print join("\t", @F[1..$#F]), "\t";' | perl -lane 'print join("\t", @F);' | perl -lane 'my $rsID = ((split(/:/, $F[1]))[0]); print $rsID, "\t", $F[0], "\t", $F[2];';
done

~~~
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$cat /mnt/lustre/home/mturchin20/Data/Choongwon2016/Tibetan.AllPhenos.assoc.SNPslt5eneg8.MarkerChrBP.txt
rs374487821     2       46571435
rs76242811      2       46576918
rs188801636     2       46577251
rs375554942     2       46579689
rs189807021     2       46583581
rs372272284     2       46584859
rs150877473     2       46588019
rs142826801     2       46588331
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.*.assoc.txt.gz | awk '{ if ($10 < .00000005) { print $0 } } ' | perl -lane 'my $rsID = ((split(/:/, $F[1]))[0]); print $rsID, "\t", $F[0], "\t", $F[2];'
rs6711319       2       179717217
rs1872203       2       179719966
rs374487821     2       46571435
rs76242811      2       46576918
rs188801636     2       46577251
rs375554942     2       46579689
rs189807021     2       46583581
rs372272284     2       46584859
rs150877473     2       46588019
rs142826801     2       46588331
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.*.assoc.txt.gz | awk '{ if ($10 < .00000005) { print $0 } } ' | perl -lane 'my $rsID = ((split(/:/, $F[1]))[0]); print $rsID, "\t", join("\t", @F);'     
rs6711319       2       rs6711319:179717217:A:G 179717217       981     G       A       0.248   -7.730317e-01   1.296529e-01    2.892655e-09
rs1872203       2       rs1872203:179719966:A:T 179719966       977     T       A       0.247   -7.684920e-01   1.299136e-01    3.820541e-09
rs374487821     2       rs374487821:46571435:G:C        46571435        881     G       C       0.247   3.686509e-01    6.697345e-02    3.996904e-08
rs76242811      2       rs76242811:46576918:T:C 46576918        920     T       C       0.251   3.742093e-01    6.567722e-02    1.343266e-08
rs188801636     2       rs188801636:46577251:T:C        46577251        920     T       C       0.251   3.742093e-01    6.567722e-02    1.343266e-08
rs375554942     2       rs375554942:46579689:A:G        46579689        915     A       G       0.252   3.687283e-01    6.546711e-02    1.950578e-08
rs189807021     2       rs189807021:46583581:G:A        46583581        912     G       A       0.252   3.703709e-01    6.555872e-02    1.768995e-08
rs372272284     2       rs372272284:46584859:A:G        46584859        914     A       G       0.248   3.860167e-01    6.603758e-02    5.713546e-09
rs150877473     2       rs150877473:46588019:C:G        46588019        907     C       G       0.252   3.694678e-01    6.571006e-02    2.055258e-08
rs142826801     2       rs142826801:46588331:G:C        46588331        889     G       C       0.255   3.710295e-01    6.607682e-02    2.143913e-08
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.*.assoc.txt.gz | awk '{ if ($10 < .00000005) { print $0 } } ' | R -q -e "Data1 <- read.table(file('stdin'), header=F); Data1[which.min(as.numeric(as.character(Data1[,10]))),];" | grep -v V | grep -v \> | perl -ane 'print join("\t", @F[1..$#F]), "\t";' | perl -lane 'print join("\t", @F);' | perl -lane 'my $rsID = ((split(/:/, $F[1]))[0]); print $rsID, "\t", $F[0], "\t", $F[2];'
rs6711319       2       179717217
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$cat /mnt/lustre/home/mturchin20/Data/Choongwon2016/Tibetan.AllPhenos.assoc.SNPslt5eneg8.MarkerChrBP.GreedyPruned.txt 
rs372272284     2       46584859
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$for i in `ls -lrt /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.*.assoc.txt.gz | awk '{ print $9 }'`; do
>         echo $i; zcat $i | awk '{ if ($10 < .00000005) { print $0 } } ' | perl -lane 'my $rsID = ((split(/:/, $F[1]))[0]); print $rsID, "\t", $F[0], "\t", $F[2];';
> done
/mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Hb.assoc.txt.gz
/mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Sat.assoc.txt.gz
/mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Pulse.assoc.txt.gz
/mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.OxHb.assoc.txt.gz
rs374487821     2       46571435
rs76242811      2       46576918
rs188801636     2       46577251
rs375554942     2       46579689
rs189807021     2       46583581
rs372272284     2       46584859
rs150877473     2       46588019
rs142826801     2       46588331
/mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.dHb.assoc.txt.gz
/mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.LvBrth.assoc.txt.gz
rs6711319       2       179717217
rs1872203       2       179719966
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$for i in `ls -lrt /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.*.assoc.txt.gz | awk '{ print $9 }'`; do
>         echo $i; zcat $i | awk '{ if ($10 < .00000005) { print $0 } } ' | R -q -e "Data1 <- read.table(file('stdin'), header=F); Data1[which.min(as.numeric(as.character(Data1[,10]))),];" | grep -v V | grep -v \> | perl -ane 'print join("\t", @F[1..$#F]), "\t";' | perl -lane 'print join("\t", @F);' | perl -lane 'my $rsID = ((split(/:/, $F[1]))[0]); print $rsID, "\t", $F[0], "\t", $F[2];';
> done
/mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Hb.assoc.txt.gz
Error in read.table(file("stdin"), header = F) :
  no lines available in input
Execution halted
/mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Sat.assoc.txt.gz
Error in read.table(file("stdin"), header = F) :
  no lines available in input
Execution halted
/mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Pulse.assoc.txt.gz
Error in read.table(file("stdin"), header = F) :
  no lines available in input
Execution halted
/mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.OxHb.assoc.txt.gz
rs372272284     2       46584859
/mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.dHb.assoc.txt.gz
Error in read.table(file("stdin"), header = F) :
  no lines available in input
Execution halted
/mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.LvBrth.assoc.txt.gz
rs6711319       2       179717217
~~~

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/Choongwon2016/Tibetan.AllPhenos.assoc.SNPslt5eneg8.MarkerChrBP.vs2.GreedyPruned.LvBrth.txt --file2 /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.4Pheno_1.assoc.IncAllele.MatchedToHb.formatted.RAF.txt.gz | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.4Pheno_1.assoc.IncAllele.MatchedToHb.formatted.RAF.wGWASannot.txt.gz 
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/Choongwon2016/Tibetan.AllPhenos.assoc.SNPslt5eneg8.MarkerChrBP.vs2.GreedyPruned.OxHb_LvBrth.txt --file2 /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.4Pheno_2.assoc.IncAllele.MatchedToHb.formatted.RAF.txt.gz | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.4Pheno_2.assoc.IncAllele.MatchedToHb.formatted.RAF.wGWASannot.txt.gz 

~~~
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs2]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.4Pheno_1.assoc.IncAllele.MatchedToHb.formatted.RAF.wGWASannot.txt.gz | perl -lane 'print $F[$#F];' | sort | uniq -c
3432579 0
      1 1
   1158 2
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs2]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.4Pheno_2.assoc.IncAllele.MatchedToHb.formatted.RAF.wGWASannot.txt.gz | perl -lane 'print $F[$#F];' | sort | uniq -c
3421434 0
      2 1
   2874 2
~~~

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1/process.MTedits.ForChoongwon2016.3Phenos_1.vs1.SignCrrct.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs2/process.MTedits.ForChoongwon2016.4Phenos_1.vs2.SignCrrct.vs1.R
cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs2/process.MTedits.ForChoongwon2016.4Phenos_1.vs2.SignCrrct.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs2/process.MTedits.ForChoongwon2016.4Phenos_2.vs2.SignCrrct.vs1.R

LANG2=${LANG}
LC_ALL2=${LC_ALL}
LANG=C 
LC_ALL=C
export LC_ALL LANG

join -a 2 -1 1 -2 1 -e NA4 -o 0 1.2 1.3 1.4 1.5 1.6 1.7 1.8 1.9 1.10 1.11 1.12 1.13 2.2 2.3 2.4 2.5 2.6 2.7 2.8 <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.4Pheno_1.assoc.IncAllele.MatchedToHb.formatted.RAF.wGWASannot.txt.gz | sort -g -k 1,1 ) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs2/Choongwon2016.4Phenos_1.dtlesslesssignif.vs2.SignCrrct.vs1.txt | perl -F, -lane 'print join("\t", @F);' | sort -g -k 1,1) | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs2/Choongwon2016.4Phenos_1.dtlesslesssignif.vs2.SignCrrct.vs1.annot.vs1.RAF.txt.pre1.gz
join -a 2 -1 1 -2 1 -e NA4 -o 0 1.2 1.3 1.4 1.5 1.6 1.7 1.8 1.9 1.10 1.11 1.12 1.13 2.2 2.3 2.4 2.5 2.6 2.7 2.8 <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.4Pheno_2.assoc.IncAllele.MatchedToHb.formatted.RAF.wGWASannot.txt.gz | sort -g -k 1,1 ) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs2/Choongwon2016.4Phenos_2.dtlesslesssignif.vs2.SignCrrct.vs1.txt | perl -F, -lane 'print join("\t", @F);' | sort -g -k 1,1) | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs2/Choongwon2016.4Phenos_2.dtlesslesssignif.vs2.SignCrrct.vs1.annot.vs1.RAF.txt.pre1.gz

LANG=${LANG2}
LC_ALL=${LC_ALL2}
export LC_ALL LANG

~~~
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs2]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs2/Choongwon2016.4Phenos_1.dtlesslesssignif.vs2.SignCrrct.vs1.txt | wc
   2248    2248  296776
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs2]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs2/Choongwon2016.4Phenos_1.dtlesslesssignif.vs2.SignCrrct.vs1.annot.vs1.RAF.txt.pre1.gz | wc
   2248   38216  436337
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs2]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs2/Choongwon2016.4Phenos_2.dtlesslesssignif.vs2.SignCrrct.vs1.txt | wc
   5967    5967  781981
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs2]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs2/Choongwon2016.4Phenos_2.dtlesslesssignif.vs2.SignCrrct.vs1.annot.vs1.RAF.txt.pre1.gz | wc
   5967  101439 1153428
~~~

zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs2/Choongwon2016.4Phenos_1.dtlesslesssignif.vs2.SignCrrct.vs1.annot.vs1.RAF.txt.pre1.gz | grep -v NA4 | cat <(echo "snp chr pos raf p_Hb n_Hb p_Sat n_Sat p_Pulse n_Pulse p_LvBrth n_LvBrth annot Z.Hb Z.Sat Z.Pulse Z.LvBrth mvstat mvp unip") - | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs2/Choongwon2016.4Phenos_1.dtlesslesssignif.vs2.SignCrrct.vs1.annot.vs1.RAF.txt.gz 
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs2/Choongwon2016.4Phenos_2.dtlesslesssignif.vs2.SignCrrct.vs1.annot.vs1.RAF.txt.pre1.gz | grep -v NA4 | cat <(echo "snp chr pos raf p_OxHb n_OxHb p_dHb n_dHb p_Pulse n_Pulse p_LvBrth n_LvBrth annot Z.OxHb Z.dHb Z.Pulse Z.LvBrth mvstat mvp unip") - | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs2/Choongwon2016.4Phenos_2.dtlesslesssignif.vs2.SignCrrct.vs1.annot.vs1.RAF.txt.gz 

zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs2/Choongwon2016.4Phenos_1.dtlesslesssignif.vs2.SignCrrct.vs1.annot.vs1.RAF.txt.gz | perl -lane 'if ($F[3] > .5) { $F[3] = 1 - $F[3]; $F[13] = $F[13] * -1; $F[14] = $F[14] * -1; $F[15] = $F[15] * -1; $F[16] = $F[16] * -1;} print join("\t", @F);' | sed 's/raf/maf/g' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs2/Choongwon2016.4Phenos_1.dtlesslesssignif.vs2.SignCrrct.vs1.annot.vs1.MAF.txt.gz 
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs2/Choongwon2016.4Phenos_2.dtlesslesssignif.vs2.SignCrrct.vs1.annot.vs1.RAF.txt.gz | perl -lane 'if ($F[3] > .5) { $F[3] = 1 - $F[3]; $F[13] = $F[13] * -1; $F[14] = $F[14] * -1; $F[15] = $F[15] * -1; $F[16] = $F[16] * -1;} print join("\t", @F);' | sed 's/raf/maf/g' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs2/Choongwon2016.4Phenos_2.dtlesslesssignif.vs2.SignCrrct.vs1.annot.vs1.MAF.txt.gz 

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1/GLC.MTedits.ForChoongwon2016.3Phenos_1.vs1.SignCrrct.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs2/GLC.MTedits.ForChoongwon2016.4Phenos_1.vs2.SignCrrct.vs1.R
cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs2/GLC.MTedits.ForChoongwon2016.4Phenos_1.vs2.SignCrrct.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs2/GLC.MTedits.ForChoongwon2016.4Phenos_2.vs2.SignCrrct.vs1.R

#20160810 NOTE -- Noticed periodically SNPs would sometimes have exact same values; unsurprisingly, these were among SNPs that were very physically close. In some example data below I show that they are distinct SNPs (eg distinct rsIDs/phys distances) among the original files but do contain the exact same summary statistics. So this was not some error on my own downstream processes that produced these exact same values

~~~
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs2]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs2/Choongwon2016.4Phenos_1.dtlesslesssignif.vs2.SignCrrct.vs1.annot.vs1.MAF.txt.gz | wc
   2201   37419  431234
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs2]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs2/Choongwon2016.4Phenos_1.dtlesslesssignif.vs2.SignCrrct.vs1.annot.vs1.MAF.txt.gz | head -n 10
snp     chr     pos     maf     p_Hb    n_Hb    p_Sat   n_Sat   p_Pulse n_Pulse p_LvBrth        n_LvBrth        annot   Z.Hb    Z.Sat   Z.Pulse mvstat  mvp     unip
rs10014388      4       83618862        0.207   4.410568e-01    914     -0.0002689743   914     1.522433e-01    913     -0.003820112    0.77040987079191        -3.64348216301859       1.43165248844303        -2.89264631971635       22.2858910534484        4.24504621482754
rs10042098      5       176143265       0.342   4.324555e-01    912     -0.995211       912     3.635102e-05    911     -0.05081795     0.78499670584331        -0.00600215744225703    -4.1295270718635        1.95301388440374        -19.7680067962334       -3.72214353050272
rs10069542      5       176143314       0.342   4.324555e-01    912     -0.995211       912     3.635102e-05    911     -0.05081795     0.78499670584331        -0.00600215744225703    -4.1295270718635        1.95301388440374        -19.7680067962334       -3.72214353050272
rs1008264       7       25618485        0.192   7.521300e-01    919     2.877942e-01    919     -8.035984e-05   918     4.052201e-01    0.315832036718987       1.06297299078072        -3.94332439207081       0.832334610286322       16.5795569593102        3.06432433893848
rs10086302      8       113467607       0.486   1.760304e-05    891     8.806490e-01    891     8.787754e-01    891     -0.6507946      4.29330818097573        0.15014654320464        -0.152521797044513      0.452658593233933       -18.7634620123193       -3.51430954360393
rs10086401      8       91572818        0.24    7.862087e-01    901     2.314939e-05    901     -0.3524451      900     -0.513254       0.271237074540256       4.2321247891004 0.929857056485969       0.653779307081217       -19.0413446840556       -3.57175231251683
rs10086462      8       91558089        0.24    8.203675e-01    891     8.043717e-05    891     -0.2268448      890     -0.5015641      0.227072329413175       3.94309381917912        1.20852652316575        0.672030782165683       -17.1071499443929       -3.17277904637203
rs10086940      8       3049799 0.473   8.326995e-01    903     -8.955387e-05   903     5.524006e-02    902     9.986558e-01    0.211240640056624       -3.91727957006403       -1.91698324716995       -0.00168470566031056    -17.7311742338376       -3.301272772209
rs10088178      8       23310904        0.271   7.863743e-01    921     -0.2185157      921     9.668874e-05    920     -0.4505058      0.271021755211515       -1.23048457425059       3.89875450623889        -0.754572048815574      16.0238992699563        2.95029293107011
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs2]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs2/Choongwon2016.4Phenos_2.dtlesslesssignif.vs2.SignCrrct.vs1.annot.vs1.MAF.txt.gz | wc
   5834   99180 1141514
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs2]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs2/Choongwon2016.4Phenos_2.dtlesslesssignif.vs2.SignCrrct.vs1.annot.vs1.MAF.txt.gz | head -n 10
snp     chr     pos     maf     p_OxHb  n_OxHb  p_dHb   n_dHb   p_Pulse n_Pulse p_LvBrth        n_LvBrth        annot   Z.OxHb  Z.dHb   Z.Pulse mvstat  mvp     unip
rs1000656       23      16922482        0.429   2.346103e-01    914     7.906188e-03    914     1.665528e-03    913     -0.0639859      1.18856649227732        2.65605004708227        3.14418027754318        -1.85227809177196       22.1914213390053        4.22538131495638
rs1000657       23      16922536        0.429   2.480660e-01    915     8.460530e-03    915     1.725427e-03    914     -0.07930957     1.15505965068038        2.63311658627783        3.13382674404997        -1.75470622910096       21.5622411286834        4.09449712917119
rs10006686      4       177993314       0.213   7.461186e-05    915     -0.6300364      915     3.517754e-01    914     3.150894e-01    3.96107893724753        -0.481675616876207      0.931151116872567       1.00460020194466        19.0833435061628        3.58043749607114
rs10007496      4       17406405        0.255   2.964302e-03    920     2.582327e-02    920     7.261546e-01    919     -0.03461353     2.97141574326057        2.22885926800862        -0.35024531684243       2.11285091752588        -25.7747815586993       -4.97342194160569
rs10009224      4       133257015       0.33    2.163151e-01    889     2.333501e-04    889     2.956022e-02    888     -0.08552347     1.23638604724268        3.67987304233749        -2.17593367941141       1.71949943742497        -26.1474866735092       -5.05145256386344
rs10011618      4       177999800       0.211   9.727020e-05    916     -0.5423901      916     3.756171e-01    915     3.832814e-01    3.89730222686746        -0.609202686261452      0.886000792786131       0.871866153728572       17.8692564985004        3.32973573273841
rs10012073      4       147015936       0.13    1.818857e-01    921     3.085042e-04    921     3.942898e-02    920     6.068551e-01    1.33497142190049        3.60805239258432        -2.05968170006247       -0.514567758504969      -24.540304204235        -4.71525544763369
rs10012313      4       17405513        0.256   3.289038e-03    921     2.670679e-02    921     7.398721e-01    920     -0.0372937      2.93934776668803        2.21577683064891        -0.332022724197703      2.08253422609799        -25.287962282729        -4.87155951922656
rs10012477      4       177994703       0.213   7.461186e-05    915     -0.6300364      915     3.517754e-01    914     3.150894e-01    3.96107893724753        -0.481675616876207      0.931151116872567       1.00460020194466        19.0833435061628        3.58043749607114
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs2]$cat Choongwon2016.PhenoGroup1.MarginalSNPsAnalyzed.txt | head -n 10
snp chr pos maf annot Z.Hb Z.Sat Z.Pulse Z.LvBrth mvstat mvp unip nmin lbfav lbfavunif
rs10014388 4 83618862 0.207 0 0.77040987079191 -3.64348216301859 1.43165248844303 -2.89264631971635 22.2858910534484 4.24504621482754 3.57028921405481 913 1.13939962839698 1.50146758288297
rs10042098 5 176143265 0.342 0 -0.78499670584331 0.00600215744225703 -4.1295270718635 1.95301388440374 19.7680067962334 3.72214353050272 4.43948339844575 911 0.227666582777915 1.56842968797506
rs10069542 5 176143314 0.342 0 -0.78499670584331 0.00600215744225703 -4.1295270718635 1.95301388440374 19.7680067962334 3.72214353050272 4.43948339844575 911 0.227666582777915 1.56842968797506
rs1008264 7 25618485 0.192 0 0.315832036718987 1.06297299078072 -3.94332439207081 0.832334610286322 16.5795569593102 3.06432433893848 4.09496093662247 918 -0.301117368734548 1.14640831074547
rs10086302 8 113467607 0.486 0 -4.29330818097573 -0.15014654320464 -0.152521797044513 0.452658593233933 18.7634620123193 3.51430954360393 4.75441232416221 891 -0.496625592434577 1.83109934634568
rs10086401 8 91572818 0.24 0 -0.271237074540256 -4.2321247891004 0.929857056485969 0.653779307081217 19.0413446840556 3.57175231251683 4.6354604484095 900 -0.375678538322612 1.61671536332671
rs10086462 8 91558089 0.24 0 -0.227072329413175 -3.94309381917912 1.20852652316575 0.672030782165683 17.1071499443929 3.17277904637203 4.09454321747614 890 -0.369326824683582 1.24004006237336
rs10086940 8 3049799 0.473 0 -0.211240640056624 3.91727957006403 -1.91698324716995 -0.00168470566031056 17.7311742338376 3.301272772209 4.04791564171524 902 -0.519817280339981 1.32138641918822
rs10088178 8 23310904 0.271 0 0.271021755211515 -1.23048457425059 3.89875450623889 -0.754572048815574 16.0238992699563 2.95029293107011 4.01462409924274 920 -0.373321917183617 1.1405764218493
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs2]$cat Choongwon2016.PhenoGroup1.MarginalSNPsAnalyzed.txt | perl -lane 'print $#F;' | sort | uniq -c
   2201 14
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs2]$cat Choongwon2016.PhenoGroup2.MarginalSNPsAnalyzed.txt | head -n 20
snp chr pos maf annot Z.OxHb Z.dHb Z.Pulse Z.LvBrth mvstat mvp unip nmin lbfav lbfavunif
rs1000656 23 16922482 0.429 0 1.18856649227732 2.65605004708227 3.14418027754318 -1.85227809177196 22.1914213390053 4.22538131495638 2.778448061776 913 0.366673969427843 1.35682664112365
rs1000657 23 16922536 0.429 0 1.15505965068038 2.63311658627783 3.13382674404997 -1.75470622910096 21.5622411286834 4.09449712917119 2.76310341027445 914 0.32144991194399 1.2866258014272
rs10006686 4 177993314 0.213 0 3.96107893724753 -0.481675616876207 0.931151116872567 1.00460020194466 19.0833435061628 3.58043749607114 4.12719213334283 914 2.52275898515011 1.44529482729611
rs10007496 4 17406405 0.255 0 -2.97141574326057 -2.22885926800862 -0.35024531684243 2.11285091752588 25.7747815586993 4.97342194160569 2.52807755296928 919 2.44440797357824 2.02580310067598
rs10009224 4 133257015 0.33 0 -1.23638604724268 -3.67987304233749 -2.17593367941141 1.71949943742497 26.1474866735092 5.05145256386344 3.63199200866595 888 0.785607368234064 2.12911145301489
rs10011618 4 177999800 0.211 0 3.89730222686746 -0.609202686261452 0.886000792786131 0.871866153728572 17.8692564985004 3.32973573273841 4.01202019115847 915 2.32124832810555 1.29175556459898
rs10012073 4 147015936 0.13 0 -1.33497142190049 -3.60805239258432 -2.05968170006247 -0.514567758504969 24.540304204235 4.71525544763369 3.51073891907188 920 0.926612882187023 1.74330605857709
rs10012313 4 17405513 0.256 0 -2.93934776668803 -2.21577683064891 -0.332022724197703 2.08253422609799 25.287962282729 4.87155951922656 2.48293110885393 920 2.38266064071958 1.95671870420089
rs10012477 4 177994703 0.213 0 3.96107893724753 -0.481675616876207 0.931151116872567 1.00460020194466 19.0833435061628 3.58043749607114 4.12719213334283 914 2.52275898515011 1.44529482729611
rs10012578 4 177994820 0.213 0 3.96107893724753 -0.481675616876207 0.931151116872567 1.00460020194466 19.0833435061628 3.58043749607114 4.12719213334283 914 2.52275898515011 1.44529482729611
rs10012654 4 177994748 0.213 0 3.96107893724753 -0.481675616876207 0.931151116872567 1.00460020194466 19.0833435061628 3.58043749607114 4.12719213334283 914 2.52275898515011 1.44529482729611
rs10015308 4 17406292 0.255 0 -2.97141574326057 -2.22885926800862 -0.35024531684243 2.11285091752588 25.7747815586993 4.97342194160569 2.52807755296928 919 2.44440797357824 2.02580310067598
rs1001602 13 61498782 0.113 0 -1.98621595499676 -3.24769820695667 -1.34449738955141 0.562340132365292 23.3752215004953 4.47203534793906 2.93426123490729 913 1.48341115830226 1.58011835562431
rs10016312 4 177996423 0.212 0 3.92592997204616 -0.455943649775475 0.930254944106193 1.08094540417086 19.0289545816412 3.56919025738 4.06350993324757 916 2.48441825968135 1.41857232232612
rs10018794 4 177996722 0.212 0 3.92592997204616 -0.455943649775475 0.930254944106193 1.08094540417086 19.0289545816412 3.56919025738 4.06350993324757 916 2.48441825968135 1.41857232232612
rs10019642 4 92645954 0.326 0 2.22672034261368 1.77720571200796 3.05812775616015 -2.15471338149348 23.2561602421026 4.44720567944971 2.65223181235875 919 1.07868655955313 1.39033389615581
rs10021126 4 150678416 0.214 0 3.12364669352189 1.82539661337389 0.700850369654038 -1.65258123357378 22.4963923300443 4.2888762336736 2.74805824446519 918 2.43537731686854 1.62152533517114
rs10022038 4 177997797 0.211 0 3.93314934681076 -0.56903144567331 0.847312968609862 0.892069092572413 18.2380217197392 3.40580115429646 4.07654816055642 916 2.40900462026373 1.35355165527685
rs10022574 4 177987600 0.211 0 3.90102137954961 -0.564680793972312 0.757321622125368 0.875691991884547 17.7931695917585 3.31405057982452 4.0186902999881 910 2.35549280925068 1.30459215654845
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs2]$for i in `ls -lrt /mnt/gluster/data/external_private_supp/Choongwon2016/. | grep Tibetan.*.assoc.txt.gz | awk '{ print $9 }'`; do echo $i; zcat /mnt/gluster/data/external_private_supp/Choongwon2016/$i | grep rs10012477; done
ibetan.Hb.assoc.txt.gz
4 rs10012477:177994703:C:T 177994703 915 T C 0.213 2.480256e-01 7.544031e-02 1.007568e-03
Tibetan.Sat.assoc.txt.gz
4 rs10012477:177994703:C:T 177994703 915 T C 0.213 3.320670e-01 2.098181e-01 1.103685e-01
Tibetan.Pulse.assoc.txt.gz
4 rs10012477:177994703:C:T 177994703 914 T C 0.212 5.510412e-01 5.907628e-01 3.517754e-01
Tibetan.OxHb.assoc.txt.gz
4 rs10012477:177994703:C:T 177994703 915 T C 0.213 2.643519e-01 6.688071e-02 7.461186e-05
Tibetan.dHb.assoc.txt.gz
4 rs10012477:177994703:C:T 177994703 915 T C 0.213 -1.555265e-02 3.320779e-02 6.300364e-01
Tibetan.LvBrth.assoc.txt.gz
4 rs10012477:177994703:C:T 177994703 975 T C 0.217 1.329911e-01 1.328198e-01 3.150894e-01
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs2]$for i in `ls -lrt /mnt/gluster/data/external_private_supp/Choongwon2016/. | grep Tibetan.*.assoc.txt.gz | awk '{ print $9 }'`; do echo $i; zcat /mnt/gluster/data/external_private_supp/Choongwon2016/$i | grep rs10012578; done
Tibetan.Hb.assoc.txt.gz
4 rs10012578:177994820:C:T 177994820 915 T C 0.213 2.480256e-01 7.544031e-02 1.007568e-03
Tibetan.Sat.assoc.txt.gz
4 rs10012578:177994820:C:T 177994820 915 T C 0.213 3.320670e-01 2.098181e-01 1.103685e-01
Tibetan.Pulse.assoc.txt.gz
4 rs10012578:177994820:C:T 177994820 914 T C 0.212 5.510412e-01 5.907628e-01 3.517754e-01
Tibetan.OxHb.assoc.txt.gz
4 rs10012578:177994820:C:T 177994820 915 T C 0.213 2.643519e-01 6.688071e-02 7.461186e-05
Tibetan.dHb.assoc.txt.gz
4 rs10012578:177994820:C:T 177994820 915 T C 0.213 -1.555265e-02 3.320779e-02 6.300364e-01
Tibetan.LvBrth.assoc.txt.gz
4 rs10012578:177994820:C:T 177994820 975 T C 0.217 1.329911e-01 1.328198e-01 3.150894e-01
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs2]$cat Choongwon2016.PhenoGroup2.MarginalSNPsAnalyzed.txt | grep rs6539167
rs6539167 12 105172975 0.375 0 -2.8118694494257 -4.42257995247372 -0.0735383849193324 1.20541482812134 44.7371564553325 8.97787450314451 5.0108651600259 907 3.92192864936103 5.74918921665757
~~~






#20160811 NOTE -- Found out that flipping Hb to all 'increasing alleles' leads to issue with 'null SNP Z-scores' biases between dHb and OxHb. As it turns out, dHb and OxHb are the components of Hb but OxHb might be a bit more weighted in the importance, so random, unassociated SNPs may still be slightly more likely to be positive, so when comparing OxHb to dHb -- oriented to Hb increasing allele -- there is an anticorrelation between OxHb and dHb that is an artifact. To deal with this, I am going to randomly flip ~half of the Hb, increasing-allele file, and then orient all files to this 'IncAllele.RanHalfFlip' version of the file. Deciding to still 'orient all positive first' to mimic the situation where effect sizes are not shared and data is given with all alleles oriented in some manner (so that sign can only be deduced by comparing files to some single standard, such as a file that is already known to be flipped to increasing allele). 
#Please see /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3/20160811.FromChoongwon.OxHb_vs_dHb.png and /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3/20160811.FromChoongwon.OxHb_vs_dHb.2.png for visuals of this phenomenon, and see below for the text from Choongwon's e-mail

~~~
Dear Michael

Hi. The issue seems to be in polarizing Z-scores based on Z-score in Hb. Here I attached two scatter plots of Z(OxHb) vs Z(dHb). One is using the original direction (based on effect size of minor alleles), and the other is flipped to make Z-scores positive in Hb. You may see the artificial negative correlation appearing after polarization.

                           Sincerely, Choongwon
~~~

#20161007 NOTE -- Going back to beginning here to investigate personally some of the correlation patterns between matching Hb to OxHb or dHb. Exploring the original 'MAF-based' orientations, the 'Increasing-Allele' based orientations, and the 'Increasing-Allele to Random Half Flipped' orientations. Mostly just going to be comparing the scatterplots of OxHb vs. dHb in all three of these scenarios to make sure I'm replicating what Choongwon originally saw, and that I am correcting it with the new 'Random Half Flipped' mechanism. Also, this new mechanism should roughly match the lack of correlation structure seen in the original 'MAF-based' orientations too.

zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Hb.assoc.txt.gz | perl -lane 'if ($F[7] < 0) { $F[9] = -1 * $F[9]; } print join("\t", @F);' | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Hb.assoc.pValDirection.txt.gz
join <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Hb.assoc.txt.gz | awk '{ print $1 "_" $3 "\t" $5 }' | sort ) <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.OxHb.assoc.txt.gz | perl -lane 'if ($F[7] < 0) { $F[9] = -1 * $F[9]; } print join("\t", @F);' | perl -lane 'my $rsID = ((split(/:/, $F[1]))[0]); my $ChrBP = $F[0] . "_" . $F[2]; splice(@F, 0, 0, ($ChrBP, $rsID)); print join("\t", @F);' | sort) | perl -lane 'if ($F[1] ne $F[7]) { $F[12] = $F[12] * -1; } splice(@F, 1, 1); print join("\t", @F);' | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.OxHb.assoc.MatchedToHb.txt.gz
join <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Hb.assoc.txt.gz | awk '{ print $1 "_" $3 "\t" $5 }' | sort ) <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.dHb.assoc.txt.gz | perl -lane 'if ($F[7] < 0) { $F[9] = -1 * $F[9]; } print join("\t", @F);' | perl -lane 'my $rsID = ((split(/:/, $F[1]))[0]); my $ChrBP = $F[0] . "_" . $F[2]; splice(@F, 0, 0, ($ChrBP, $rsID)); print join("\t", @F);' | sort) | perl -lane 'if ($F[1] ne $F[7]) { $F[12] = $F[12] * -1; } splice(@F, 1, 1); print join("\t", @F);' | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.dHb.assoc.MatchedToHb.txt.gz

join <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Hb.assoc.txt.gz | perl -lane 'print $F[0], "_", $F[2], "\t", $F[$#F];' | sort -k 1,1) <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Hb.assoc.IncAllele.txt.gz | perl -lane 'print $F[0], "\t", $F[$#F];' | sort -k 1,1) | \
join - <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Hb.assoc.IncAllele.RanHalfFlip.txt.gz | perl -lane 'print $F[0], "\t", $F[$#F];' | sort -k 1,1) | \
join - <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.OxHb.assoc.MatchedToHb.txt.gz | perl -lane 'print $F[0], "\t", $F[$#F];' | sort -k 1,1) | \
join - <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.OxHb.assoc.IncAllele.MatchedToHb.txt.gz | perl -lane 'print $F[0], "\t", $F[$#F];' | sort -k 1,1) | \
join - <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.OxHb.assoc.IncAllele.MatchedToHbRanFlip.txt.gz | perl -lane 'print $F[0], "\t", $F[$#F];' | sort -k 1,1) | \
join - <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.dHb.assoc.MatchedToHb.txt.gz | perl -lane 'print $F[0], "\t", $F[$#F];' | sort -k 1,1) | \
join - <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.dHb.assoc.IncAllele.MatchedToHb.txt.gz | perl -lane 'print $F[0], "\t", $F[$#F];' | sort -k 1,1) | \
join - <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.dHb.assoc.IncAllele.MatchedToHbRanFlip.txt.gz | perl -lane 'print $F[0], "\t", $F[$#F];' | sort -k 1,1) | grep -v p | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.TroubleShooting.Hb_OxHb_dHb.OxHbdHbAntiCorrelation.txt.gz

#zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.TroubleShooting.Hb_OxHb_dHb.OxHbdHbAntiCorrelation.txt.gz | perl -lane 'print join("\t", @F[1..$#F]);' | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.TroubleShooting.Hb_OxHb_dHb.OxHbdHbAntiCorrelation.NoChrBPColumn.txt.gz

#zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.TroubleShooting.Hb_OxHb_dHb.OxHbdHbAntiCorrelation.txt.gz | R -q -e "Data1 <- read.table(file('stdin'), header=F); Data1[,1:ncol(Data1)] <- apply(Data1[,1:ncol(Data1)], 2,  as.numeric); Data1[,1:ncol(Data1)] <- apply(Data1[,1:ncol(Data1)], 2, function(x) { val1 <- qnorm(log(abs(x)/2), lower.tail=FALSE, log.p=TRUE); if (x < 0) { val1 <- val1 * -1 }; return(val1) });
#zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.TroubleShooting.Hb_OxHb_dHb.OxHbdHbAntiCorrelation.txt.gz | perl -lane 'print join("\t", @F[1..$#F]);' | R -q -e "Data1 <- read.table(file('stdin'), header=F); Data1 <- apply(Data1, c(1,2), function(x) { val1 <- qnorm(log(abs(x)/2), lower.tail=FALSE, log.p=TRUE); if (x < 0) { val1 <- val1 * -1 }; return(val1) });
zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.TroubleShooting.Hb_OxHb_dHb.OxHbdHbAntiCorrelation.txt.gz | R -q -e "Data1 <- read.table(file('stdin'), header=F, row.name=1); Data1 <- apply(Data1, c(1,2), function(x) { val1 <- qnorm(log(abs(x)/2), lower.tail=FALSE, log.p=TRUE); if (x < 0) { val1 <- val1 * -1 }; return(val1) });
#jpeg("/mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.TroubleShooting.Hb_OxHb_dHb.OxHbdHbAntiCorrelation.Scatterplots.jpeg", height=2000, width=6000, res=300);
jpeg("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3/Tibetan.TroubleShooting.Hb_OxHb_dHb.OxHbdHbAntiCorrelation.Scatterplots.jpeg", height=2000, width=6000, res=300);
par(mfrow=c(1,3))
set.seed(10)
PlotPoints1 <- round(runif(500000, 1, nrow(Data1)))
plot(Data1[PlotPoints1,4], Data1[PlotPoints1,7], main="MAF-based", xlab="OxHb", ylab="dHb");
PlotPoints2 <- round(runif(500000, 1, nrow(Data1)))
plot(Data1[PlotPoints2,5], Data1[PlotPoints2,8], main="Increasing Allele", xlab="OxHb", ylab="dHb");
PlotPoints3 <- round(runif(500000, 1, nrow(Data1)))
plot(Data1[PlotPoints3,6], Data1[PlotPoints3,9], main="Increasing Allele Random Flipped", xlab="OxHb", ylab="dHb");
dev.off();"

#mv /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.TroubleShooting.Hb_OxHb_dHb.OxHbdHbAntiCorrelation.Scatterplots.jpeg /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3/Tibetan.TroubleShooting.Hb_OxHb_dHb.OxHbdHbAntiCorrelation.Scatterplots.jpeg 

~~~
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3]$join <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Hb.assoc.txt.gz | awk '{ print $1 "_" $3 "\t" $5 }' | sort ) <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.OxHb.assoc.txt.gz | perl -lane 'my $rsID = ((split(/:/, $F[1]))[0]); my $ChrBP = $F[0] . "_" . $F[2]; splice(@F, 0, 0, ($ChrBP, $rsID)); print join("\t", @F);' | sort) | head -n 10
10_100000625 G rs7899632 10 rs7899632:100000625:A:G 100000625 916 G A 0.422 -9.054680e-02 5.861791e-02 1.225915e-01
10_100000645 C rs61875309 10 rs61875309:100000645:A:C 100000645 916 C A 0.060 -4.178077e-02 1.236722e-01 7.340205e-01
10_100002399 A rs8181398 10 rs8181398:100002399:A:G 100002399 918 A G 0.055 -1.050833e-01 1.258591e-01 3.953881e-01
10_100003242 G rs12258651 10 rs12258651:100003242:T:G 100003242 917 G T 0.121 1.121788e-01 8.792173e-02 1.980327e-01
10_100003785 C rs1359508 10 rs1359508:100003785:T:C 100003785 919 C T 0.462 1.111696e-01 5.854501e-02 5.650671e-02
10_100004360 A rs1048754 10 rs1048754:100004360:G:A 100004360 916 A G 0.060 -4.178077e-02 1.236722e-01 7.340205e-01
10_100004441 C rs1048757 10 rs1048757:100004441:G:C 100004441 918 C G 0.481 -8.418530e-02 5.802239e-02 1.462914e-01
10_100004906 A rs3750595 10 rs3750595:100004906:C:A 100004906 919 A C 0.421 -7.352649e-02 5.857845e-02 2.093331e-01
10_100004996 A rs2025625 10 rs2025625:100004996:G:A 100004996 920 A G 0.462 1.111024e-01 5.854637e-02 5.665728e-02
10_10000514 C rs6602382 10 rs6602382:10000514:T:C 10000514 912 C T 0.468 5.076305e-03 5.830746e-02 9.520841e-01
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3]$join <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Hb.assoc.txt.gz | awk '{ print $1 "_" $3 "\t" $5 }' | sort ) <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.OxHb.assoc.txt.gz | perl -lane 'my $rsID = ((split(/:/, $F[1]))[0]); my $ChrBP = $F[0] . "_" . $F[2]; splice(@F, 0, 0, ($ChrBP, $rsID)); print join("\t", @F);' | sort) | perl -lane 'if ($F[1] eq $F[7]) { print "1"; } else { print "0"; }' | sort | uniq -c
3507376 1
[  mturchin20@bigmem03  ~]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.TroubleShooting.Hb_OxHb_dHb.OxHbdHbAntiCorrelation.txt.gz | head -n 10
10_100000625 8.392270e-02 8.392270e-02 8.392270e-02 1.225915e-01 1.225915e-01 1.225915e-01 3.937060e-01 3.937060e-01 3.937060e-01
10_100000645 6.688200e-01 6.688200e-01 -0.66882 7.340205e-01 7.340205e-01 -0.7340205 7.739307e-01 7.739307e-01 -0.7739307
10_100002399 1.520838e-01 1.520838e-01 1.520838e-01 3.953881e-01 3.953881e-01 3.953881e-01 2.350974e-01 2.350974e-01 2.350974e-01
10_100003242 2.859697e-01 2.859697e-01 2.859697e-01 1.980327e-01 1.980327e-01 1.980327e-01 8.898429e-01 -0.8898429 -0.8898429
10_100003785 2.138390e-02 2.138390e-02 2.138390e-02 5.650671e-02 5.650671e-02 5.650671e-02 2.063892e-01 2.063892e-01 2.063892e-01
10_100004360 6.688200e-01 6.688200e-01 -0.66882 7.340205e-01 7.340205e-01 -0.7340205 7.739307e-01 7.739307e-01 -0.7739307
10_100004441 1.063654e-01 1.063654e-01 -0.1063654 1.462914e-01 1.462914e-01 -0.1462914 4.393436e-01 4.393436e-01 -0.4393436
10_100004906 1.542384e-01 1.542384e-01 -0.1542384 2.093331e-01 2.093331e-01 -0.2093331 4.516756e-01 4.516756e-01 -0.4516756
10_100004996 2.131738e-02 2.131738e-02 -0.02131738 5.665728e-02 5.665728e-02 -0.05665728 2.052724e-01 2.052724e-01 -0.2052724
10_10000514 7.314249e-01 7.314249e-01 7.314249e-01 9.520841e-01 9.520841e-01 9.520841e-01 3.154640e-01 3.154640e-01 3.154640e-01
[  mturchin20@bigmem03  ~]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.TroubleShooting.Hb_OxHb_dHb.OxHbdHbAntiCorrelation.txt.gz | perl -lane 'print join("\t", @F[1..$#F]);' | head -n 100000 | R -q -e "Data1 <- read.table(file('stdin'), header=F); Data1 <- apply(Data1, c(1,2), function(x) { val1 <- qnorm(log(abs(x)/2), lower.tail=FALSE, log.p=TRUE); if (x < 0) { val1 <- val1 * -1 }; return(val1) }); head(Data1);"
> Data1 <- read.table(file('stdin'), header=F); Data1 <- apply(Data1, c(1,2), function(x) { val1 <- qnorm(log(abs(x)/2), lower.tail=FALSE, log.p=TRUE); if (x < 0) { val1 <- val1 * -1 }; return(val1) }); head(Data1);                    V1        V2         V3        V4        V5         V6        V7                                                                                                                                                   [1,] 1.7283656 1.7283656  1.7283656 1.5439869 1.5439869  1.5439869 0.8529158                                                                                                                                                   [2,] 0.4277681 0.4277681 -0.4277681 0.3397823 0.3397823 -0.3397823 0.2872372                                                                                                                                                   [3,] 1.4322098 1.4322098  1.4322098 0.8498867 0.8498867  0.8498867 1.1873302
[4,] 1.0670047 1.0670047  1.0670047 1.2871767 1.2871767  1.2871767 0.1385030
[5,] 2.3011368 2.3011368  2.3011368 1.9071072 1.9071072  1.9071072 1.2635567
[6,] 0.4277681 0.4277681 -0.4277681 0.3397823 0.3397823 -0.3397823 0.2872372
             V8         V9
[1,]  0.8529158  0.8529158
[2,]  0.2872372 -0.2872372
[3,]  1.1873302  1.1873302
[4,] -0.1385030 -0.1385030
[5,]  1.2635567  1.2635567
[6,]  0.2872372 -0.2872372
> 
> 
[  mturchin20@bigmem03  ~]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.TroubleShooting.Hb_OxHb_dHb.OxHbdHbAntiCorrelation.txt.gz | perl -lane 'print join("\t", @F[1..$#F]);' | head -n 1000000 | R -q -e "Data1 <- read.table(file('stdin'), header=F); Data1 <- apply(Data1, c(1,2), function(x) { val1 <- qnorm(log(abs(x)/2), lower.tail=FALSE, log.p=TRUE); if (x < 0) { val1 <- val1 * -1 }; return(val1) }); head(Data1);"
> Data1 <- read.table(file('stdin'), header=F); Data1 <- apply(Data1, c(1,2), function(x) { val1 <- qnorm(log(abs(x)/2), lower.tail=FALSE, log.p=TRUE); if (x < 0) { val1 <- val1 * -1 }; return(val1) }); head(Data1);
            V1        V2         V3        V4        V5         V6        V7
[1,] 1.7283656 1.7283656  1.7283656 1.5439869 1.5439869  1.5439869 0.8529158
[2,] 0.4277681 0.4277681 -0.4277681 0.3397823 0.3397823 -0.3397823 0.2872372
[3,] 1.4322098 1.4322098  1.4322098 0.8498867 0.8498867  0.8498867 1.1873302
[4,] 1.0670047 1.0670047  1.0670047 1.2871767 1.2871767  1.2871767 0.1385030
[5,] 2.3011368 2.3011368  2.3011368 1.9071072 1.9071072  1.9071072 1.2635567
[6,] 0.4277681 0.4277681 -0.4277681 0.3397823 0.3397823 -0.3397823 0.2872372
             V8         V9
[1,]  0.8529158  0.8529158
[2,]  0.2872372 -0.2872372
[3,]  1.1873302  1.1873302
[4,] -0.1385030 -0.1385030
[5,]  1.2635567  1.2635567
[6,]  0.2872372 -0.2872372
> 
> 
[  mturchin20@bigmem03  ~]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.TroubleShooting.Hb_OxHb_dHb.OxHbdHbAntiCorrelation.NoChrBPColumn.txt.gz | tail -n 10
4.014963e-01    4.014963e-01    4.014963e-01    3.142654e-01    3.142654e-01    3.142654e-01    7.215345e-01    -0.7215345      -0.7215345
9.531434e-02    9.531434e-02    -0.09531434 6.603336e-02        6.603336e-02    -0.06603336     6.927679e-01    -0.6927679      6.927679e-01
4.014963e-01    4.014963e-01    4.014963e-01    3.142654e-01    3.142654e-01    3.142654e-01    7.215345e-01    -0.7215345      -0.7215345
4.014963e-01    4.014963e-01    -0.4014963  3.142654e-01        3.142654e-01    -0.3142654      7.215345e-01    -0.7215345      7.215345e-01
1.618138e-01    1.618138e-01    -0.1618138  1.493852e-01        1.493852e-01    -0.1493852      8.000341e-01    -0.8000341      8.000341e-01
8.619916e-02    8.619916e-02    8.619916e-02    6.009375e-02    6.009375e-02    6.009375e-02    7.120184e-01    -0.7120184      -0.7120184
3.136036e-01    3.136036e-01    3.136036e-01    2.687388e-01    2.687388e-01    2.687388e-01    8.699397e-01    -0.8699397      -0.8699397
4.938913e-01    4.938913e-01    -0.4938913  4.905977e-01        4.905977e-01    -0.4905977      8.442644e-01    -0.8442644      8.442644e-01
2.633681e-01    2.633681e-01    -0.2633681  4.108792e-01        4.108792e-01    -0.4108792      6.538327e-01    6.538327e-01    -0.6538327
p       p       p       p       p       p   p   p       p
[  mturchin20@bigmem03  ~]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.TroubleShooting.Hb_OxHb_dHb.OxHbdHbAntiCorrelation.NoChrBPColumn.txt.gz | grep -v p | tail -n 10
2.633681e-01    2.633681e-01    2.633681e-01    4.108792e-01    4.108792e-01    4.108792e-01    6.538327e-01    6.538327e-01    6.538327e-01
4.014963e-01    4.014963e-01    4.014963e-01    3.142654e-01    3.142654e-01    3.142654e-01    7.215345e-01    -0.7215345      -0.7215345
9.531434e-02    9.531434e-02    -0.09531434 6.603336e-02        6.603336e-02    -0.06603336     6.927679e-01    -0.6927679      6.927679e-01
4.014963e-01    4.014963e-01    4.014963e-01    3.142654e-01    3.142654e-01    3.142654e-01    7.215345e-01    -0.7215345      -0.7215345
4.014963e-01    4.014963e-01    -0.4014963  3.142654e-01        3.142654e-01    -0.3142654      7.215345e-01    -0.7215345      7.215345e-01
1.618138e-01    1.618138e-01    -0.1618138  1.493852e-01        1.493852e-01    -0.1493852      8.000341e-01    -0.8000341      8.000341e-01
8.619916e-02    8.619916e-02    8.619916e-02    6.009375e-02    6.009375e-02    6.009375e-02    7.120184e-01    -0.7120184      -0.7120184
3.136036e-01    3.136036e-01    3.136036e-01    2.687388e-01    2.687388e-01    2.687388e-01    8.699397e-01    -0.8699397      -0.8699397
4.938913e-01    4.938913e-01    -0.4938913  4.905977e-01        4.905977e-01    -0.4905977      8.442644e-01    -0.8442644      8.442644e-01
2.633681e-01    2.633681e-01    -0.2633681  4.108792e-01        4.108792e-01    -0.4108792      6.538327e-01    6.538327e-01    -0.6538327
[  mturchin20@bigmem03  ~]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.TroubleShooting.Hb_OxHb_dHb.OxHbdHbAntiCorrelation.NoChrBPColumn.txt.gz | wc
3507311 31565799 396211829
[  mturchin20@bigmem03  ~]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.TroubleShooting.Hb_OxHb_dHb.OxHbdHbAntiCorrelation.NoChrBPColumn.txt.gz | grep -v p | wc
3507310 31565790 396211811
> Data1 <- read.table("/mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.TroubleShooting.Hb_OxHb_dHb.OxHbdHbAntiCorrelation.txt.gz", header=F, row.names=1)
> head(Data1)
                    V2        V3         V4         V5         V6          V7
10_100000625 0.0839227 0.0839227  0.0839227 0.12259150 0.12259150  0.12259150
10_100000645 0.6688200 0.6688200 -0.6688200 0.73402050 0.73402050 -0.73402050
10_100002399 0.1520838 0.1520838  0.1520838 0.39538810 0.39538810  0.39538810
10_100003242 0.2859697 0.2859697  0.2859697 0.19803270 0.19803270  0.19803270
10_100003785 0.0213839 0.0213839  0.0213839 0.05650671 0.05650671  0.05650671
10_100004360 0.6688200 0.6688200 -0.6688200 0.73402050 0.73402050 -0.73402050
                    V8         V9        V10
10_100000625 0.3937060  0.3937060  0.3937060
10_100000645 0.7739307  0.7739307 -0.7739307
10_100002399 0.2350974  0.2350974  0.2350974
10_100003242 0.8898429 -0.8898429 -0.8898429
10_100003785 0.2063892  0.2063892  0.2063892
10_100004360 0.7739307  0.7739307 -0.7739307
> Data1 <- apply(Data1, c(1,2), function(x) { val1 <- qnorm(log(abs(x)/2), lower.tail=FALSE, log.p=TRUE); if (x < 0) { val1 <- val1 * -1 }; return(val1) });
> head(Data1)
                    V2        V3         V4        V5        V6         V7
10_100000625 1.7283656 1.7283656  1.7283656 1.5439869 1.5439869  1.5439869
10_100000645 0.4277681 0.4277681 -0.4277681 0.3397823 0.3397823 -0.3397823
10_100002399 1.4322098 1.4322098  1.4322098 0.8498867 0.8498867  0.8498867
10_100003242 1.0670047 1.0670047  1.0670047 1.2871767 1.2871767  1.2871767
10_100003785 2.3011368 2.3011368  2.3011368 1.9071072 1.9071072  1.9071072
10_100004360 0.4277681 0.4277681 -0.4277681 0.3397823 0.3397823 -0.3397823
                    V8         V9        V10
10_100000625 0.8529158  0.8529158  0.8529158
10_100000645 0.2872372  0.2872372 -0.2872372
10_100002399 1.1873302  1.1873302  1.1873302
10_100003242 0.1385030 -0.1385030 -0.1385030
10_100003785 1.2635567  1.2635567  1.2635567
10_100004360 0.2872372  0.2872372 -0.2872372

~~~

#20161007 NOTE -- this is where the original 20160811 timeline continues
mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3/

~~~
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3]$perl -e 'print "nana\nnana\nnana\n";' | perl -lane 'if ($. == 1) { srand(100); } print rand();'
0.25105890266514
0.208948404851498
0.940927909958315
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Hb.assoc.txt.gz | perl -lane 'my $rsID = ((split(/:/, $F[1]))[0]); my $ChrBP = $F[0] . "_" . $F[2]; splice(@F, 0, 0, ($ChrBP, $rsID)); if ($F[9] < 0) { ($F[6], $F[7]) = ($F[7], $F[6]); $F[8] = 1 - $F[8]; $F[9] = $F[9] * -1; } print join("\t", @F);' | head -n 10
CHR_POS SNP     CHR     SNP     POS     N       A1      A0      A1frq   beta    se      p
1_752566        rs3094315       1       rs3094315:752566:G:A    752566  921     G       A       0.149   7.390260e-02    9.084730e-02    4.168330e-01
1_752721        rs3131972       1       rs3131972:752721:A:G    752721  921     A       G       0.276   2.500086e-02    7.696211e-02    7.333741e-01
1_752894        rs3131971       1       rs3131971:752894:T:C    752894  903     T       C       0.149   9.224075e-02    9.128423e-02    3.131140e-01
1_754334        rs3131967       1       rs3131967:754334:T:C    754334  897     T       C       0.268   6.507564e-02    7.889252e-02    4.013396e-01
1_754503        rs3115859       1       rs3115859:754503:G:A    754503  908     G       A       0.277   1.579218e-02    7.731364e-02    8.260580e-01
1_754964        rs3131966       1       rs3131966:754964:C:T    754964  907     C       T       0.277   1.719287e-02    7.732831e-02    8.125771e-01
1_755890        rs3115858       1       rs3115858:755890:A:T    755890  900     A       T       0.139   6.934277e-02    9.393129e-02    4.649276e-01
1_756434        rs61768170      1       rs61768170:756434:G:C   756434  889     C       G       0.135   5.539038e-02    9.521388e-02    5.639026e-01
1_756604        rs3131962       1       rs3131962:756604:A:G    756604  886     A       G       0.144   4.646892e-02    9.348308e-02    6.223091e-01
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Hb.assoc.txt.gz | perl -lane 'my $rsID = ((split(/:/, $F[1]))[0]); my $ChrBP = $F[0] . "_" . $F[2]; splice(@F, 0, 0, ($ChrBP, $rsID)); if ($F[9] < 0) { ($F[6], $F[7]) = ($F[7], $F[6]); $F[8] = 1 - $F[8]; $F[9] = $F[9] * -1; } print join("\t", @F);' | perl -lane 'if ($. == 1) { srand(100); } if (rand(1) > .5) { ($F[6], $F[7]) = ($F[7], $F[6]); $F[8] = 1 - $F[8]; $F[9] = $F[9] * -1; $F[11] = $F[11] * -1; } print join("\t", @F);' | head -n 10
CHR_POS SNP     CHR     SNP     POS     N       A1      A0      A1frq   beta    se      p
1_752566        rs3094315       1       rs3094315:752566:G:A    752566  921     G       A       0.149   7.390260e-02    9.084730e-02    4.168330e-01
1_752721        rs3131972       1       rs3131972:752721:A:G    752721  921     G       A       0.724   -0.02500086     7.696211e-02    -0.7333741
1_752894        rs3131971       1       rs3131971:752894:T:C    752894  903     T       C       0.149   9.224075e-02    9.128423e-02    3.131140e-01
1_754334        rs3131967       1       rs3131967:754334:T:C    754334  897     T       C       0.268   6.507564e-02    7.889252e-02    4.013396e-01
1_754503        rs3115859       1       rs3115859:754503:G:A    754503  908     G       A       0.277   1.579218e-02    7.731364e-02    8.260580e-01
1_754964        rs3131966       1       rs3131966:754964:C:T    754964  907     C       T       0.277   1.719287e-02    7.732831e-02    8.125771e-01
1_755890        rs3115858       1       rs3115858:755890:A:T    755890  900     T       A       0.861   -0.06934277     9.393129e-02    -0.4649276
1_756434        rs61768170      1       rs61768170:756434:G:C   756434  889     G       C       0.865   -0.05539038     9.521388e-02    -0.5639026
1_756604        rs3131962       1       rs3131962:756604:A:G    756604  886     G       A       0.856   -0.04646892     9.348308e-02    -0.6223091
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Hb.assoc.txt.gz | perl -lane 'if ($F[7] > 0) { print "1"; } elsif ($F[7] < 0 ) { print "2"; } else { print "3"; }' | sort | uniq -c
1759190 1
1748378 2
      1 3
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Hb.assoc.IncAllele.txt.gz | perl -lane 'if ($F[9] > 0) { print "1"; } elsif ($F[9] < 0 ) { print "2"; } else { print "3"; }' | sort | uniq -c
3507568 1
      1 3
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Hb.assoc.IncAllele.RanHalfFlip.txt.gz | perl -lane 'if ($F[9] > 0) { print "1"; } elsif ($F[9] < 0 ) { print "2"; } else { print "3"; }' | sort | uniq -c
1753511 1
1754057 2
      1 3
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Hb.assoc.IncAllele.RanHalfFlip.txt.gz | perl -lane 'if ($F[11] > 0) { print "1"; } elsif ($F[11] < 0 ) { print "2"; } else { print "3"; }' | sort | uniq -c
1753511 1
1754057 2
      1 3
~~~

zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Hb.assoc.txt.gz | perl -lane 'my $rsID = ((split(/:/, $F[1]))[0]); my $ChrBP = $F[0] . "_" . $F[2]; splice(@F, 0, 0, ($ChrBP, $rsID)); if ($F[9] < 0) { ($F[6], $F[7]) = ($F[7], $F[6]); $F[8] = 1 - $F[8]; $F[9] = $F[9] * -1; } print join("\t", @F);' | perl -lane 'if ($. == 1) { srand(100); } if (rand(1) > .5) { ($F[6], $F[7]) = ($F[7], $F[6]); $F[8] = 1 - $F[8]; $F[9] = $F[9] * -1; $F[11] = $F[11] * -1; } print join("\t", @F);' | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Hb.assoc.IncAllele.RanHalfFlip.txt.gz

LANG2=${LANG}
LC_ALL2=${LC_ALL}
LANG=C
LC_ALL=C
export LC_ALL LANG

#2 data groups:
#Hb (Hemoglobin)
#Sat (Oxygen Saturation)
#Pulse
#LvBrth
#
#
#OxHb (Oxygenated Hemoglobin)
#dHb (Dexoygenated Hemoglobin) 
#Pulse
#LvBrth
#


join <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Hb.assoc.IncAllele.RanHalfFlip.txt.gz | awk '{ print $1, "\t", $7 }' | sort ) <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Sat.assoc.txt.gz | perl -lane 'my $rsID = ((split(/:/, $F[1]))[0]); my $ChrBP = $F[0] . "_" . $F[2]; splice(@F, 0, 0, ($ChrBP, $rsID)); if ($F[9] < 0) { ($F[6], $F[7]) = ($F[7], $F[6]); $F[8] = 1 - $F[8]; $F[9] = $F[9] * -1; } print join("\t", @F);' | sort) | perl -lane 'if ($F[1] ne $F[7]) { $F[12] = $F[12] * -1; } splice(@F, 1, 1); print join("\t", @F);' | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Sat.assoc.IncAllele.MatchedToHbRanFlip.txt.gz
join <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Hb.assoc.IncAllele.RanHalfFlip.txt.gz | awk '{ print $1, "\t", $7 }' | sort ) <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Pulse.assoc.txt.gz | perl -lane 'my $rsID = ((split(/:/, $F[1]))[0]); my $ChrBP = $F[0] . "_" . $F[2]; splice(@F, 0, 0, ($ChrBP, $rsID)); if ($F[9] < 0) { ($F[6], $F[7]) = ($F[7], $F[6]); $F[8] = 1 - $F[8]; $F[9] = $F[9] * -1; } print join("\t", @F);' | sort) | perl -lane 'if ($F[1] ne $F[7]) { $F[12] = $F[12] * -1; } splice(@F, 1, 1); print join("\t", @F);' | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Pulse.assoc.IncAllele.MatchedToHbRanFlip.txt.gz 
join <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Hb.assoc.IncAllele.RanHalfFlip.txt.gz | awk '{ print $1, "\t", $7 }' | sort ) <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.OxHb.assoc.txt.gz | perl -lane 'my $rsID = ((split(/:/, $F[1]))[0]); my $ChrBP = $F[0] . "_" . $F[2]; splice(@F, 0, 0, ($ChrBP, $rsID)); if ($F[9] < 0) { ($F[6], $F[7]) = ($F[7], $F[6]); $F[8] = 1 - $F[8]; $F[9] = $F[9] * -1; } print join("\t", @F);' | sort) | perl -lane 'if ($F[1] ne $F[7]) { $F[12] = $F[12] * -1; } splice(@F, 1, 1); print join("\t", @F);' | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.OxHb.assoc.IncAllele.MatchedToHbRanFlip.txt.gz
join <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Hb.assoc.IncAllele.RanHalfFlip.txt.gz | awk '{ print $1, "\t", $7 }' | sort ) <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.dHb.assoc.txt.gz | perl -lane 'my $rsID = ((split(/:/, $F[1]))[0]); my $ChrBP = $F[0] . "_" . $F[2]; splice(@F, 0, 0, ($ChrBP, $rsID)); if ($F[9] < 0) { ($F[6], $F[7]) = ($F[7], $F[6]); $F[8] = 1 - $F[8]; $F[9] = $F[9] * -1; } print join("\t", @F);' | sort) | perl -lane 'if ($F[1] ne $F[7]) { $F[12] = $F[12] * -1; } splice(@F, 1, 1); print join("\t", @F);' | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.dHb.assoc.IncAllele.MatchedToHbRanFlip.txt.gz
join <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Hb.assoc.IncAllele.RanHalfFlip.txt.gz | awk '{ print $1, "\t", $7 }' | sort ) <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.LvBrth.assoc.txt.gz | perl -lane 'my $rsID = ((split(/:/, $F[1]))[0]); my $ChrBP = $F[0] . "_" . $F[2]; splice(@F, 0, 0, ($ChrBP, $rsID)); if ($F[9] < 0) { ($F[6], $F[7]) = ($F[7], $F[6]); $F[8] = 1 - $F[8]; $F[9] = $F[9] * -1; } print join("\t", @F);' | sort) | perl -lane 'if ($F[1] ne $F[7]) { $F[12] = $F[12] * -1; } splice(@F, 1, 1); print join("\t", @F);' | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.LvBrth.assoc.IncAllele.MatchedToHbRanFlip.txt.gz

zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Hb.assoc.IncAllele.RanHalfFlip.txt.gz | awk '{ print $3, "\t", $5, "\t", $2, "\t", $9, "\t", $12, "\t", $6 }' | cat <(echo "CHR      POS     SNP     A1frq   p       N") <(cat - | grep -v A1frq) | perl -lane 'if ($F[2] !~ /\D/ || $F[2] eq "X") { $F[2] = $F[0] . "_" . $F[1]; } print join("\t", @F); ' | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Hb.assoc.IncAllele.RanHalfFlip.formatted.txt.gz 
zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Sat.assoc.IncAllele.MatchedToHbRanFlip.txt.gz | awk '{ print $3, "\t", $5, "\t", $2, "\t", $9, "\t", $12, "\t", $6 }' | cat <(echo "CHR      POS     SNP     A1frq   p       N") <(cat - | grep -v A1frq) | perl -lane 'if ($F[2] !~ /\D/ || $F[2] eq "X") { $F[2] = $F[0] . "_" . $F[1]; } print join("\t", @F); '  | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Sat.assoc.IncAllele.MatchedToHbRanFlip.formatted.txt.gz 
zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Pulse.assoc.IncAllele.MatchedToHbRanFlip.txt.gz | awk '{ print $3, "\t", $5, "\t", $2, "\t", $9, "\t", $12, "\t", $6 }' | cat <(echo "CHR      POS     SNP     A1frq   p       N") <(cat - | grep -v A1frq) | perl -lane 'if ($F[2] !~ /\D/ || $F[2] eq "X") { $F[2] = $F[0] . "_" . $F[1]; } print join("\t", @F); '  | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Pulse.assoc.IncAllele.MatchedToHbRanFlip.formatted.txt.gz
zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.OxHb.assoc.IncAllele.MatchedToHbRanFlip.txt.gz | awk '{ print $3, "\t", $5, "\t", $2, "\t", $9, "\t", $12, "\t", $6 }' | cat <(echo "CHR      POS     SNP     A1frq   p       N") <(cat - | grep -v A1frq) | perl -lane 'if ($F[2] !~ /\D/ || $F[2] eq "X") { $F[2] = $F[0] . "_" . $F[1]; } print join("\t", @F); '  | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.OxHb.assoc.IncAllele.MatchedToHbRanFlip.formatted.txt.gz 
zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.dHb.assoc.IncAllele.MatchedToHbRanFlip.txt.gz | awk '{ print $3, "\t", $5, "\t", $2, "\t", $9, "\t", $12, "\t", $6 }' | cat <(echo "CHR      POS     SNP     A1frq   p       N") <(cat - | grep -v A1frq) | perl -lane 'if ($F[2] !~ /\D/ || $F[2] eq "X") { $F[2] = $F[0] . "_" . $F[1]; } print join("\t", @F); '  | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.dHb.assoc.IncAllele.MatchedToHbRanFlip.formatted.txt.gz
zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.LvBrth.assoc.IncAllele.MatchedToHbRanFlip.txt.gz | awk '{ print $3, "\t", $5, "\t", $2, "\t", $9, "\t", $12, "\t", $6 }' | cat <(echo "CHR      POS     SNP     A1frq   p       N") <(cat - | grep -v A1frq) | perl -lane 'if ($F[2] !~ /\D/ || $F[2] eq "X") { $F[2] = $F[0] . "_" . $F[1]; } print join("\t", @F); '  | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.LvBrth.assoc.IncAllele.MatchedToHbRanFlip.formatted.txt.gz

join -a 1 -a 2 -1 3 -2 3 -e NA2 -o 0 1.1 1.2 1.4 1.5 1.6 2.1 2.2 2.4 2.5 2.6 <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Hb.assoc.IncAllele.RanHalfFlip.formatted.txt.gz | sed 's/_/-/g' | perl -lane '($F[2], $F[1]) = ($F[1], $F[2]); $F[2] = $F[0] . "_" . $F[2]; print join("\t", @F);' | sort -g -k 3,3) <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Sat.assoc.IncAllele.MatchedToHbRanFlip.formatted.txt.gz | sed 's/_/-/g' | perl -lane '($F[2], $F[1]) = ($F[1], $F[2]); $F[2] = $F[0] . "_" . $F[2]; print join("\t", @F);' | sort -g -k 3,3 ) | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
join -a 1 -a 2 -1 1 -2 3 -e NA2 -o 0 1.2 2.1 2.2 2.4 2.5 2.6 - <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Pulse.assoc.IncAllele.MatchedToHbRanFlip.formatted.txt.gz | sed 's/_/-/g' | perl -lane '($F[2], $F[1]) = ($F[1], $F[2]); $F[2] = $F[0] . "_" . $F[2]; print join("\t", @F);' | sort -g -k 3,3) | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); print join("\t", @F);' | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); if ($#F == 6) { splice(@F, 1, 0, (("NA2") x 9)); } my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
join -a 1 -a 2 -1 1 -2 3 -e NA2 -o 0 1.2 2.1 2.2 2.4 2.5 2.6 - <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.LvBrth.assoc.IncAllele.MatchedToHbRanFlip.formatted.txt.gz | sed 's/_/-/g' | perl -lane '($F[2], $F[1]) = ($F[1], $F[2]); $F[2] = $F[0] . "_" . $F[2]; print join("\t", @F);' | sort -g -k 3,3) | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); print join("\t", @F);' | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); if ($#F == 6) { splice(@F, 1, 0, (("NA2") x 9)); } my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
perl -lane 'print $F[0], "\t", join("\t", split(/_/, $F[1]));' | sed 's/chr//g' | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.4Pheno_1.assoc.IncAllele.MatchedToHbRanFlip.formatted.RAF.txt.pre1.gz

join -a 1 -a 2 -1 3 -2 3 -e NA2 -o 0 1.1 1.2 1.4 1.5 1.6 2.1 2.2 2.4 2.5 2.6 <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.OxHb.assoc.IncAllele.MatchedToHbRanFlip.formatted.txt.gz | sed 's/_/-/g' | perl -lane '($F[2], $F[1]) = ($F[1], $F[2]); $F[2] = $F[0] . "_" . $F[2]; print join("\t", @F);' | sort -g -k 3,3) <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.dHb.assoc.IncAllele.MatchedToHbRanFlip.formatted.txt.gz | sed 's/_/-/g' | perl -lane '($F[2], $F[1]) = ($F[1], $F[2]); $F[2] = $F[0] . "_" . $F[2]; print join("\t", @F);' | sort -g -k 3,3 ) | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
join -a 1 -a 2 -1 1 -2 3 -e NA2 -o 0 1.2 2.1 2.2 2.4 2.5 2.6 - <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Pulse.assoc.IncAllele.MatchedToHbRanFlip.formatted.txt.gz | sed 's/_/-/g' | perl -lane '($F[2], $F[1]) = ($F[1], $F[2]); $F[2] = $F[0] . "_" . $F[2]; print join("\t", @F);' | sort -g -k 3,3) | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); print join("\t", @F);' | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); if ($#F == 6) { splice(@F, 1, 0, (("NA2") x 9)); } my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
join -a 1 -a 2 -1 1 -2 3 -e NA2 -o 0 1.2 2.1 2.2 2.4 2.5 2.6 - <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.LvBrth.assoc.IncAllele.MatchedToHbRanFlip.formatted.txt.gz | sed 's/_/-/g' | perl -lane '($F[2], $F[1]) = ($F[1], $F[2]); $F[2] = $F[0] . "_" . $F[2]; print join("\t", @F);' | sort -g -k 3,3) | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); print join("\t", @F);' | perl -lane 'splice(@F, 1, 1, split(/_/, $F[1])); if ($#F == 6) { splice(@F, 1, 0, (("NA2") x 9)); } my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | perl -lane 'my $joinFields = join("_", @F[1..$#F]); print $F[0], "\t", $joinFields;' | \
perl -lane 'print $F[0], "\t", join("\t", split(/_/, $F[1]));' | sed 's/chr//g' | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.4Pheno_2.assoc.IncAllele.MatchedToHbRanFlip.formatted.RAF.txt.pre1.gz

LANG=${LANG2}
LC_ALL=${LC_ALL2}
export LC_ALL LANG




~~~
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3]$for i in `ls -lrt /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.*.assoc.IncAllele.MatchedToHbRanFlip.txt.gz | awk '{ print $9
 }'`; do echo $i; zcat $i | wc ; zcat $i | head -n 10; done
/mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Sat.assoc.IncAllele.MatchedToHbRanFlip.txt.gz
3507569 42090828 377437884
10_100000625    rs7899632       10      rs7899632:100000625:A:G 100000625       916     G       A       0.422   2.398042e-02    1.828722e-01    -0.8755691
10_100000645    rs61875309      10      rs61875309:100000645:A:C        100000645       916     C       A       0.060   1.430280e-01    3.847636e-01    7.304266e-01
10_100002399    rs8181398       10      rs8181398:100002399:A:G 100002399       918     A       G       0.055   3.641363e-01    3.919577e-01    -0.3570072
10_100003242    rs12258651      10      rs12258651:100003242:T:G        100003242       917     T       G       0.879   0.004309389     2.741558e-01    -0.999923
10_100003785    rs1359508       10      rs1359508:100003785:T:C 100003785       919     T       C       0.538   0.09766834      1.825781e-01    -0.5861611
10_100004360    rs1048754       10      rs1048754:100004360:G:A 100004360       916     A       G       0.060   1.430280e-01    3.847636e-01    7.304266e-01
10_100004441    rs1048757       10      rs1048757:100004441:G:C 100004441       918     C       G       0.481   3.259236e-02    1.810009e-01    8.459574e-01
10_100004906    rs3750595       10      rs3750595:100004906:C:A 100004906       919     A       C       0.421   2.382532e-02    1.826816e-01    8.757007e-01
10_100004996    rs2025625       10      rs2025625:100004996:G:A 100004996       920     G       A       0.538   0.09828543      1.825797e-01    5.838185e-01
10_10000514     rs6602382       10      rs6602382:10000514:T:C  10000514        912     T       C       0.532   0.1286826       1.813214e-01    -0.4672061
/mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.dHb.assoc.IncAllele.MatchedToHbRanFlip.txt.gz
3507569 42090828 378863392
10_100000625    rs7899632       10      rs7899632:100000625:A:G 100000625       916     A       G       0.578   0.02365581      2.886626e-02    3.937060e-01
10_100000645    rs61875309      10      rs61875309:100000645:A:C        100000645       916     A       C       0.94    0.01970672      6.079258e-02    -0.7739307
10_100002399    rs8181398       10      rs8181398:100002399:A:G 100002399       918     G       A       0.945   0.07393039      6.206710e-02    2.350974e-01
10_100003242    rs12258651      10      rs12258651:100003242:T:G        100003242       917     T       G       0.879   0.005148244     4.325948e-02    -0.8898429
10_100003785    rs1359508       10      rs1359508:100003785:T:C 100003785       919     C       T       0.462   3.594422e-02    2.883051e-02    2.063892e-01
10_100004360    rs1048754       10      rs1048754:100004360:G:A 100004360       916     G       A       0.94    0.01970672      6.079258e-02    -0.7739307
10_100004441    rs1048757       10      rs1048757:100004441:G:C 100004441       918     G       C       0.519   0.02155107      2.861492e-02    -0.4393436
10_100004906    rs3750595       10      rs3750595:100004906:C:A 100004906       919     C       A       0.579   0.02070597      2.883922e-02    -0.4516756
10_100004996    rs2025625       10      rs2025625:100004996:G:A 100004996       920     A       G       0.462   3.603293e-02    2.883043e-02    -0.2052724
10_10000514     rs6602382       10      rs6602382:10000514:T:C  10000514        912     C       T       0.468   2.827712e-02    2.864803e-02    3.154640e-01
/mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.LvBrth.assoc.IncAllele.MatchedToHbRanFlip.txt.gz
3499631 41995572 376905484
10_100000625    rs7899632       10      rs7899632:100000625:A:G 100000625       974     G       A       0.425   1.583053e-01    1.167047e-01    -0.1777584
10_100000645    rs61875309      10      rs61875309:100000645:A:C        100000645       975     C       A       0.060   3.212315e-01    2.470273e-01    1.983767e-01
10_100002399    rs8181398       10      rs8181398:100002399:A:G 100002399       976     G       A       0.947   0.03973908      2.560514e-01    8.854789e-01
10_100003242    rs12258651      10      rs12258651:100003242:T:G        100003242       977     G       T       0.120   1.436057e-01    1.754573e-01    4.164910e-01
10_100003785    rs1359508       10      rs1359508:100003785:T:C 100003785       979     T       C       0.539   0.2339394       1.166442e-01    -0.04605376
10_100004360    rs1048754       10      rs1048754:100004360:G:A 100004360       975     A       G       0.060   3.212315e-01    2.470273e-01    1.983767e-01
10_100004441    rs1048757       10      rs1048757:100004441:G:C 100004441       977     C       G       0.485   2.295153e-01    1.150488e-01    4.806174e-02
10_100004906    rs3750595       10      rs3750595:100004906:C:A 100004906       977     A       C       0.424   1.575837e-01    1.165837e-01    1.793700e-01
10_100004996    rs2025625       10      rs2025625:100004996:G:A 100004996       980     G       A       0.539   0.233317        1.166472e-01    4.663817e-02
10_10000514     rs6602382       10      rs6602382:10000514:T:C  10000514        972     C       T       0.465   7.449082e-02    1.155407e-01    5.084744e-01
/mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Pulse.assoc.IncAllele.MatchedToHbRanFlip.txt.gz
3506813 42081756 376640872
10_100000625    rs7899632       10      rs7899632:100000625:A:G 100000625       915     G       A       0.422   5.309865e-01    5.109574e-01    -0.3009806
10_100000645    rs61875309      10      rs61875309:100000645:A:C        100000645       915     A       C       0.94    2.020056        1.080210e+00    -0.06235068
10_100002399    rs8181398       10      rs8181398:100002399:A:G 100002399       917     A       G       0.055   9.334268e-01    1.093651e+00    -0.3896698
10_100003242    rs12258651      10      rs12258651:100003242:T:G        100003242       916     G       T       0.121   1.245166e+00    7.648087e-01    1.011356e-01
10_100003785    rs1359508       10      rs1359508:100003785:T:C 100003785       918     T       C       0.538   0.2612783       5.101369e-01    -0.6086001
10_100004360    rs1048754       10      rs1048754:100004360:G:A 100004360       915     G       A       0.94    2.020056        1.080210e+00    -0.06235068
10_100004441    rs1048757       10      rs1048757:100004441:G:C 100004441       917     C       G       0.481   9.437970e-02    5.054949e-01    8.543304e-01
10_100004906    rs3750595       10      rs3750595:100004906:C:A 100004906       918     A       C       0.422   5.260652e-01    5.104515e-01    3.050526e-01
10_100004996    rs2025625       10      rs2025625:100004996:G:A 100004996       919     G       A       0.538   0.2627418       5.101483e-01    6.065893e-01
10_10000514     rs6602382       10      rs6602382:10000514:T:C  10000514        911     C       T       0.467   3.448256e-01    5.083672e-01    5.085889e-01
/mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.OxHb.assoc.IncAllele.MatchedToHbRanFlip.txt.gz
3507569 42090828 378399544
10_100000625    rs7899632       10      rs7899632:100000625:A:G 100000625       916     A       G       0.578   0.0905468       5.861791e-02    1.225915e-01
10_100000645    rs61875309      10      rs61875309:100000645:A:C        100000645       916     A       C       0.94    0.04178077      1.236722e-01    -0.7340205
10_100002399    rs8181398       10      rs8181398:100002399:A:G 100002399       918     G       A       0.945   0.1050833       1.258591e-01    3.953881e-01
10_100003242    rs12258651      10      rs12258651:100003242:T:G        100003242       917     G       T       0.121   1.121788e-01    8.792173e-02    1.980327e-01
10_100003785    rs1359508       10      rs1359508:100003785:T:C 100003785       919     C       T       0.462   1.111696e-01    5.854501e-02    5.650671e-02
10_100004360    rs1048754       10      rs1048754:100004360:G:A 100004360       916     G       A       0.94    0.04178077      1.236722e-01    -0.7340205
10_100004441    rs1048757       10      rs1048757:100004441:G:C 100004441       918     G       C       0.519   0.0841853       5.802239e-02    -0.1462914
10_100004906    rs3750595       10      rs3750595:100004906:C:A 100004906       919     C       A       0.579   0.07352649      5.857845e-02    -0.2093331
10_100004996    rs2025625       10      rs2025625:100004996:G:A 100004996       920     A       G       0.462   1.111024e-01    5.854637e-02    -0.05665728
10_10000514     rs6602382       10      rs6602382:10000514:T:C  10000514        912     C       T       0.468   5.076305e-03    5.830746e-02    9.520841e-01
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3]$for i in `ls -lrt /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.*.assoc.IncAllele.MatchedToHbRanFlip.txt.gz | awk '{ print $9 }'`; do echo $i; zcat $i  | perl -lane 'print $F[$#F];' | awk '{ if ($1 > 0) { print "1"; } else if ($1 < 0) { print "2"; } else { print "3"; } }' | sort | uniq -c; done
/mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Sat.assoc.IncAllele.MatchedToHbRanFlip.txt.gz
1754557 1
1753012 2
/mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.dHb.assoc.IncAllele.MatchedToHbRanFlip.txt.gz
1752325 1
1755244 2
/mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.LvBrth.assoc.IncAllele.MatchedToHbRanFlip.txt.gz
1749756 1
1749875 2
/mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.Pulse.assoc.IncAllele.MatchedToHbRanFlip.txt.gz
1754306 1
1752507 2
/mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.OxHb.assoc.IncAllele.MatchedToHbRanFlip.txt.gz
1753610 1
1753959 2
~~~

zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.4Pheno_1.assoc.IncAllele.MatchedToHbRanFlip.formatted.RAF.txt.pre1.gz | grep -v NA2 | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.4Pheno_1.assoc.IncAllele.MatchedToHbRanFlip.formatted.RAF.txt.pre2.gz
zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.4Pheno_2.assoc.IncAllele.MatchedToHbRanFlip.formatted.RAF.txt.pre1.gz | grep -v NA2 | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.4Pheno_2.assoc.IncAllele.MatchedToHbRanFlip.formatted.RAF.txt.pre2.gz

zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.4Pheno_1.assoc.IncAllele.MatchedToHbRanFlip.formatted.RAF.txt.pre2.gz | perl -lane '($F[2], $F[0]) = ($F[0], $F[2]); $F[2] = ((split(/_/, $F[2]))[1]); print join("\t", @F);' | perl -lane 'splice(@F, 16, 3); splice(@F, 11, 3); splice(@F, 6, 3); print join("\t", @F);' | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.4Pheno_1.assoc.IncAllele.MatchedToHbRanFlip.formatted.RAF.txt.gz
zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.4Pheno_2.assoc.IncAllele.MatchedToHbRanFlip.formatted.RAF.txt.pre2.gz | perl -lane '($F[2], $F[0]) = ($F[0], $F[2]); $F[2] = ((split(/_/, $F[2]))[1]); print join("\t", @F);'  | perl -lane 'splice(@F, 16, 3); splice(@F, 11, 3); splice(@F, 6, 3); print join("\t", @F);' | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.4Pheno_2.assoc.IncAllele.MatchedToHbRanFlip.formatted.RAF.txt.gz

#dfssfs

#20160830 pasued earlier, below is the last few steps, can delete this when come back just to get things up and going again, using this as mostly a bookmark
~~~
zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.4Pheno_2.assoc.IncAllele.MatchedToHbRanFlip.formatted.RAF.txt.gz | head -n 10
SNP     CHR     POS     A1frq   p       N       p       N       p       N       p       N
1-100640636     1       100640636       0.935   4.050135e-01    919     9.520115e-01    919     -0.3297514      918     -0.4731262      978
1-100991376     1       100991376       0.889   1.069134e-01    917     3.111677e-01    917     -0.9036341      916     3.085718e-01    977
1-101147138     1       101147138       0.542   5.368052e-02    910     -0.6297548      910     -0.03433399     909     5.892610e-02    970
1-101970051     1       101970051       0.749   -0.1435012      920     -0.2986489      920     4.602450e-01    919     5.713950e-01    980
1-102777667     1       102777667       0.753   -0.1878538      885     4.299970e-01    885     -0.7600934      884     -0.6617059      940
1-103329033     1       103329033       0.828   -0.7158805      920     3.384520e-02    920     -0.4468977      919     -0.03084187     980
1-103329034     1       103329034       0.828   -0.7158805      920     3.384520e-02    920     -0.4468977      919     -0.03084187     980
1-103329036     1       103329036       0.828   7.158805e-01    920     -0.0338452      920     4.468977e-01    919     3.084187e-02    980
1-103380393     1       103380393       0.945   -0.4198021      897     -0.5920738      897     5.606755e-01    896     1.330349e-01    954
~~~

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/Choongwon2016/Tibetan.AllPhenos.assoc.SNPslt5eneg8.MarkerChrBP.vs2.GreedyPruned.LvBrth.txt --file2 /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.4Pheno_1.assoc.IncAllele.MatchedToHbRanFlip.formatted.RAF.txt.gz | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.4Pheno_1.assoc.IncAllele.MatchedToHbRanFlip.formatted.RAF.wGWASannot.txt.gz
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/Choongwon2016/Tibetan.AllPhenos.assoc.SNPslt5eneg8.MarkerChrBP.vs2.GreedyPruned.OxHb_LvBrth.txt --file2 /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.4Pheno_2.assoc.IncAllele.MatchedToHbRanFlip.formatted.RAF.txt.gz | gzip > /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.4Pheno_2.assoc.IncAllele.MatchedToHbRanFlip.formatted.RAF.wGWASannot.txt.gz

~
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs2]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.4Pheno_1.assoc.IncAllele.MatchedToHbRanFlip.formatted.RAF.wGWASannot.txt.gz | perl -lane 'print $F[$#F];' | sort | uniq -c
3432579 0
      1 1
   1158 2
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs2]$zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.4Pheno_2.assoc.IncAllele.MatchedToHbRanFlip.formatted.RAF.wGWASannot.txt.gz | perl -lane 'print $F[$#F];' | sort | uniq -c
3421434 0
      2 1
   2874 2
~

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs2/process.MTedits.ForChoongwon2016.4Phenos_1.vs2.SignCrrct.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3/process.MTedits.ForChoongwon2016.4Phenos_1.vs3.SignCrrct.HbRanFlip.vs1.R
cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs2/process.MTedits.ForChoongwon2016.4Phenos_2.vs2.SignCrrct.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3/process.MTedits.ForChoongwon2016.4Phenos_2.vs3.SignCrrct.HbRanFlip.vs1.R





LANG2=${LANG}
LC_ALL2=${LC_ALL}
LANG=C
LC_ALL=C
export LC_ALL LANG

join -a 2 -1 1 -2 1 -e NA4 -o 0 1.2 1.3 1.4 1.5 1.6 1.7 1.8 1.9 1.10 1.11 1.12 1.13 2.2 2.3 2.4 2.5 2.6 2.7 2.8 <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.4Pheno_1.assoc.IncAllele.MatchedToHbRanFlip.formatted.RAF.wGWASannot.txt.gz | sort -g -k 1,1 ) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3/Choongwon2016.RanHalfFlip.4Phenos_1.dtlesslesssignif.vs3.SignCrrct.vs1.txt | perl -F, -lane 'print join("\t", @F);' | sort -g -k 1,1) | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3/Choongwon2016.RanHalfFlip.4Phenos_1.dtlesslesssignif.vs3.SignCrrct.vs1.annot.vs1.RAF.txt.pre1.gz
join -a 2 -1 1 -2 1 -e NA4 -o 0 1.2 1.3 1.4 1.5 1.6 1.7 1.8 1.9 1.10 1.11 1.12 1.13 2.2 2.3 2.4 2.5 2.6 2.7 2.8 <(zcat /mnt/gluster/data/external_private_supp/Choongwon2016/Tibetan.4Pheno_2.assoc.IncAllele.MatchedToHbRanFlip.formatted.RAF.wGWASannot.txt.gz | sort -g -k 1,1 ) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3/Choongwon2016.RanHalfFlip.4Phenos_2.dtlesslesssignif.vs3.SignCrrct.vs1.txt | perl -F, -lane 'print join("\t", @F);' | sort -g -k 1,1) | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3/Choongwon2016.RanHalfFlip.4Phenos_2.dtlesslesssignif.vs3.SignCrrct.vs1.annot.vs1.RAF.txt.pre1.gz

LANG=${LANG2}
LC_ALL=${LC_ALL2}
export LC_ALL LANG

~~~
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3/Choongwon2016.RanHalfFlip.4Phenos_1.dtlesslesssignif.vs3.SignCrrct.vs1.txt | wc
   2134    2134  282570
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3/Choongwon2016.RanHalfFlip.4Phenos_1.dtlesslesssignif.vs3.SignCrrct.vs1.annot.vs1.RAF.txt.pre1.gz | wc
   2134   42680  463435
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3/Choongwon2016.RanHalfFlip.4Phenos_2.dtlesslesssignif.vs3.SignCrrct.vs1.txt | wc
   2256    2256  299000
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3/Choongwon2016.RanHalfFlip.4Phenos_2.dtlesslesssignif.vs3.SignCrrct.vs1.annot.vs1.RAF.txt.pre1.gz | wc
   2256   45120  489615
~~~

zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3/Choongwon2016.RanHalfFlip.4Phenos_1.dtlesslesssignif.vs3.SignCrrct.vs1.annot.vs1.RAF.txt.pre1.gz | grep -v NA4 | cat <(echo "snp chr pos raf p_Hb n_Hb p_Sat n_Sat p_Pulse n_Pulse p_LvBrth n_LvBrth annot Z.Hb Z.Sat Z.Pulse Z.LvBrth mvstat mvp unip") - | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3/Choongwon2016.RanHalfFlip.4Phenos_1.dtlesslesssignif.vs3.SignCrrct.vs1.annot.vs1.RAF.txt.gz 
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3/Choongwon2016.RanHalfFlip.4Phenos_2.dtlesslesssignif.vs3.SignCrrct.vs1.annot.vs1.RAF.txt.pre1.gz | grep -v NA4 | cat <(echo "snp chr pos raf p_OxHb n_OxHb p_dHb n_dHb p_Pulse n_Pulse p_LvBrth n_LvBrth annot Z.OxHb Z.dHb Z.Pulse Z.LvBrth mvstat mvp unip") - | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3/Choongwon2016.RanHalfFlip.4Phenos_2.dtlesslesssignif.vs3.SignCrrct.vs1.annot.vs1.RAF.txt.gz 

zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3/Choongwon2016.RanHalfFlip.4Phenos_1.dtlesslesssignif.vs3.SignCrrct.vs1.annot.vs1.RAF.txt.gz | perl -lane 'if ($F[3] > .5) { $F[3] = 1 - $F[3]; $F[13] = $F[13] * -1; $F[14] = $F[14] * -1; $F[15] = $F[15] * -1; $F[16] = $F[16] * -1;} print join("\t", @F);' | sed 's/raf/maf/g' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3/Choongwon2016.RanHalfFlip.4Phenos_1.dtlesslesssignif.vs3.SignCrrct.vs1.annot.vs1.MAF.txt.gz 
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3/Choongwon2016.RanHalfFlip.4Phenos_2.dtlesslesssignif.vs3.SignCrrct.vs1.annot.vs1.RAF.txt.gz | perl -lane 'if ($F[3] > .5) { $F[3] = 1 - $F[3]; $F[13] = $F[13] * -1; $F[14] = $F[14] * -1; $F[15] = $F[15] * -1; $F[16] = $F[16] * -1;} print join("\t", @F);' | sed 's/raf/maf/g' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3/Choongwon2016.RanHalfFlip.4Phenos_2.dtlesslesssignif.vs3.SignCrrct.vs1.annot.vs1.MAF.txt.gz 

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs2/GLC.MTedits.ForChoongwon2016.4Phenos_1.vs2.SignCrrct.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3/GLC.MTedits.ForChoongwon2016.RanHalfFlip.4Phenos_1.vs3.SignCrrct.vs1.R
cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3/GLC.MTedits.ForChoongwon2016.RanHalfFlip.4Phenos_1.vs3.SignCrrct.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3/GLC.MTedits.ForChoongwon2016.RanHalfFlip.4Phenos_2.vs3.SignCrrct.vs1.R








##Stein2016
##20161106

#Files were copied over to /mnt/lustre/home/mstein3/mvbimbam_files/. by Michelle Stein on 20161105

mkdir /mnt/gluster/data/external_private_supp/Stein2016

mv /mnt/lustre/home/mstein3/mvbimbam_files/* /mnt/gluster/data/external_private_supp/Stein2016/.

##20161106 NOTE_0 -- Prob: Haven't done the below yet, need to ask Michelle to remove her original files from lustre since I don't have permissions to do so myself
#for i in `ls -lrt /mnt/gluster/data/external_private_supp/Stein2016/. | awk '{ print $9 }'`; do
#	ln -s /mnt/gluster/data/external_private_supp/Stein2016/$i /mnt/lustre/home/mstein3/mvbimbam_files/$i;
#done

mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Stein2016
mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Stein2016/Vs1

#cat /mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp2_4cyto.txt | perl -lane 'shift @F; for (my $i = 5; $i <= $#F; $i += 3) { if ($F[$i] > 0) { $F[$i] = "+"; } elsif ($F[$i] < 0) { $F[$i] = "-"; } else { $F[$i] = "NoDir"; } } print join("\t", @F);' | grep -v "NoDir" | R -q -e "Data1 <- read.table(file('stdin'), header=F); devtools::load_all(\"/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/bmass\"); for (i in seq(6, ncol(Data1), by=3)) { Data2 <- Data1[,c(i+1, i)]; colnames(Data2) <- c(\"pValue\", \"Direction\"); cbind(Data1, apply(Data2, 1, GetZScoreAndDirection)); }; print(Data1);" | head -n 10

cat /mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp2_4cyto.txt | awk '{ print $2 "\t" $3 "\t" $4 "\t" $5 "\t" $7 "\t" $8 "\t" $6 }' | perl -lane 'if ($F[4] > 0) { $F[4] = "+"; } elsif ($F[4] < 0) { $F[4] = "-"; } else { $F[4] = "NoDir"; } print join("\t", @F);' | cat <(echo -e "Chr\tBP\tMAF\tA1\tDirection\tpValue\tN") - | gzip > /mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp2_4cyto.IFN.g_lps.txt.gz
cat /mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp2_4cyto.txt | awk '{ print $2 "\t" $3 "\t" $4 "\t" $5 "\t" $10 "\t" $11 "\t" $9 }' | perl -lane 'if ($F[4] > 0) { $F[4] = "+"; } elsif ($F[4] < 0) { $F[4] = "-"; } else { $F[4] = "NoDir"; } print join("\t", @F);' | cat <(echo -e "Chr\tBP\tMAF\tA1\tDirection\tpValue\tN") - | gzip > /mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp2_4cyto.IL.12p70_lps.txt.gz
cat /mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp2_4cyto.txt | awk '{ print $2 "\t" $3 "\t" $4 "\t" $5 "\t" $13 "\t" $14 "\t" $12 }' | perl -lane 'if ($F[4] > 0) { $F[4] = "+"; } elsif ($F[4] < 0) { $F[4] = "-"; } else { $F[4] = "NoDir"; } print join("\t", @F);' | cat <(echo -e "Chr\tBP\tMAF\tA1\tDirection\tpValue\tN") - | gzip > /mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp2_4cyto.IL.23_lps.txt.gz
cat /mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp2_4cyto.txt | awk '{ print $2 "\t" $3 "\t" $4 "\t" $5 "\t" $16 "\t" $17 "\t" $15 }' | perl -lane 'if ($F[4] > 0) { $F[4] = "+"; } elsif ($F[4] < 0) { $F[4] = "-"; } else { $F[4] = "NoDir"; } print join("\t", @F);' | cat <(echo -e "Chr\tBP\tMAF\tA1\tDirection\tpValue\tN") - | gzip > /mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp2_4cyto.TNF.a_lps.txt.gz

zcat /mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp2_4cyto.IFN.g_lps.txt.gz | grep -v NoDir | gzip > /mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp2_4cyto.IFN.g_lps.No0Dir.txt.gz
zcat /mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp2_4cyto.IL.12p70_lps.txt.gz | grep -v NoDir | gzip > /mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp2_4cyto.IL.12p70_lps.No0Dir.txt.gz
zcat /mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp2_4cyto.IL.23_lps.txt.gz | grep -v NoDir | gzip > /mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp2_4cyto.IL.23_lps.No0Dir.txt.gz
zcat /mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp2_4cyto.TNF.a_lps.txt.gz | grep -v NoDir | gzip > /mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp2_4cyto.TNF.a_lps.No0Dir.txt.gz

#for i in `ls -lrt /mnt/gluster/data/external_private_supp/Stein2016/*.No0Dir.txt.gz | awk '{ print $9; }'`; do echo $i; zcat $i | awk '{ if ($6 < .00000005) { print $0 } } '; done
#for i in `ls -lrt /mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp1_7cyto.*.No0Dir.txt.gz | awk '{ print $9; }'`; do echo $i; zcat $i | awk '{ if ($6 < .00000005) { print $0 } } '; done
#for i in `ls -lrt /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp1_8cyto.*.No0Dir.txt.gz | awk '{ print $9; }'`; do echo $i; zcat $i | awk '{ if ($6 < .00000005) { print $0 } } '; done
for i in `ls -lrt /mnt/gluster/data/external_private_supp/Stein2016/*.No0Dir.txt.gz | awk '{ print $9; }'`; do echo $i; zcat $i | awk '{ if ($6 < .00000005) { print $1 "\t" $2 } } ' > $i.5e8.GWASHits.ChrBP.txt; done
for i in `ls -lrt /mnt/gluster/data/external_private_supp/Stein2016/*.No0Dir.txt.gz | awk '{ print $9; }'`; do echo $i.5e8.GWASHits.ChrBP.txt; cat $i.5e8.GWASHits.ChrBP.txt | wc; done

cat /mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp1_7cyto.*.No0Dir.txt.gz.5e8.GWASHits.ChrBP.txt > /mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp1_7cyto.AllPheno.No0Dir.txt.gz.5e8.GWASHits.ChrBP.txt
cat /mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp2_4cyto.*.No0Dir.txt.gz.5e8.GWASHits.ChrBP.txt > /mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp2_4cyto.AllPheno.No0Dir.txt.gz.5e8.GWASHits.ChrBP.txt
cat /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp1_8cyto.*.No0Dir.txt.gz.5e8.GWASHits.ChrBP.txt > /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp1_8cyto.AllPheno.No0Dir.txt.gz.5e8.GWASHits.ChrBP.txt
cat /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp2_7cyto.*.No0Dir.txt.gz.5e8.GWASHits.ChrBP.txt > /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp2_7cyto.AllPheno.No0Dir.txt.gz.5e8.GWASHits.ChrBP.txt

~~~
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate]$cat /mnt/gluster/data/external_private_supp/Stein2016/LPSCytoNames_Grp2_4cyto.txt             
"x"
"1" "IFN.g_lps"
"2" "IL.12p70_lps"
"3" "IL.23_lps"
"4" "TNF.a_lps"
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate]$cat /mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp2_4cyto.txt | head -n 10
"chr" "ps" "MAF" "ref_allele" "n" "beta" "p_val" "n" "beta" "p_val" "n" "beta" "p_val" "n" "beta" "p_val"
"1062" 1 785989 0.075 3 107 0.03018724 0.9149532 107 -0.0348633 0.8968772 107 0.15492 0.3940235 107 0.524026 0.03167614
"2548" 1 1003629 0.245 5 106 0.1911938 0.29826 106 0.3103519 0.07058205 106 0.09921486 0.4000917 106 -0.01446367 0.9281819
"3767" 1 1211292 0.168 3 107 0.03472233 0.86001 107 -0.06709015 0.721068 107 0.04177471 0.740388 107 0.05498399 0.7455785
"5323" 1 1478153 0.36 5 107 0.2579772 0.1345667 107 0.107584 0.5053216 107 0.01076524 0.9203882 107 -0.1058473 0.4586944
"5442" 1 1500941 0.36 2 107 0.2579772 0.1345667 107 0.107584 0.5053216 107 0.01076524 0.9203882 107 -0.1058473 0.4586944
"5511" 1 1510801 0.304 3 107 0.3638154 0.04009265 107 0.3378388 0.03492036 107 0.1774923 0.08384151 107 -0.05904868 0.6877749
"6774" 1 1692321 0.467 5 107 -0.2903621 0.04497574 107 -0.4338463 0.0007013064 107 -0.2900595 0.000322561 107 -0.2074867 0.07895722
"6884" 1 1721479 0.379 3 107 0.2914682 0.04644262 107 0.4142917 0.002688777 107 0.31001 0.0003383953 107 0.2261611 0.06788764
"6888" 1 1722932 0.381 3 105 0.2593161 0.07921238 105 0.3982738 0.004311464 105 0.2950783 0.0007555375 105 0.2076696 0.09586794
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate]$for i in `ls -lrt /mnt/gluster/data/external_private_supp/Stein2016/ | grep "Nov 10" | awk '{ print $9 }'`; do echo $i; zcat /mnt/gluster/data/external_private_supp/Stein2016/$i | wc; zcat /mnt/gluster/data/external_private_supp/Stein2016/$i | grep -v NoDir | wc; done
LPSGemma_Grp2_4cyto.IFN.g_lps.txt.gz
4111638 28781466 145843209
4111637 28781459 145843164
LPSGemma_Grp2_4cyto.IL.12p70_lps.txt.gz
4111638 28781466 145829039
4111637 28781459 145828994
LPSGemma_Grp2_4cyto.IL.23_lps.txt.gz
4111638 28781466 145828947
4111637 28781459 145828902
LPSGemma_Grp2_4cyto.TNF.a_lps.txt.gz
4111638 28781465 145831161
4111637 28781459 145831120
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate]$zcat /mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp2_4cyto.TNF.a_lps.txt.gz | head -n 10
Chr BP  MAF     A1      Direction       pValue  N
"ps" "MAF"      "ref_allele"    "n"     NoDir   "beta"
1  785989       0.075   3       +       0.03167614      107
1  1003629      0.245   5       -       0.9281819       106
1  1211292      0.168   3       +       0.7455785       107
1  1478153      0.36    5       -       0.4586944       107
1  1500941      0.36    2       -       0.4586944       107
1  1510801      0.304   3       -       0.6877749       107
1  1692321      0.467   5       -       0.07895722      107
1  1721479      0.379   3       +       0.06788764      107
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate]$for i in `ls -lrt /mnt/gluster/data/external_private_supp/Stein2016/*.No0Dir.txt.gz | awk '{ print $9; }'`; do echo $i; zcat $i | awk '{ if ($6 < .00000005) { print $0 } } '; done
/mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp2_4cyto.IFN.g_lps.No0Dir.txt.gz
/mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp2_4cyto.IL.12p70_lps.No0Dir.txt.gz
/mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp2_4cyto.IL.23_lps.No0Dir.txt.gz
/mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp2_4cyto.TNF.a_lps.No0Dir.txt.gz
/mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp2_7cyto.IL.10_cd.No0Dir.txt.gz
1  54381839     0.107   3       +       4.212424e-08    84
1  54382357     0.107   3       +       4.212424e-08    84
1  54390466     0.107   3       +       4.212424e-08    84
18 69130485     0.087   3       +       4.805697e-08    86
18 69131983     0.087   2       +       4.805697e-08    86
/mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp2_7cyto.IL.13_cd.No0Dir.txt.gz
/mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp2_7cyto.IL.15_cd.No0Dir.txt.gz
/mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp2_7cyto.IL.23_cd.No0Dir.txt.gz
/mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp2_7cyto.IL.4_cd.No0Dir.txt.gz
/mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp2_7cyto.IL.5_cd.No0Dir.txt.gz
/mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp2_7cyto.IL.9_cd.No0Dir.txt.gz
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate]$for i in `ls -lrt /mnt/gluster/data/external_private_supp/Stein2016/*.No0Dir.txt.gz | awk '{ print $9; }'`; do echo $i.5e8.GWASHits.ChrBP.txt; cat $i.5e8.GWASHits.ChrBP.txt | wc; done
/mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp2_4cyto.IFN.g_lps.No0Dir.txt.gz.5e8.GWASHits.ChrBP.txt
      0       0       0
/mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp2_4cyto.IL.12p70_lps.No0Dir.txt.gz.5e8.GWASHits.ChrBP.txt
      0       0       0
/mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp2_4cyto.IL.23_lps.No0Dir.txt.gz.5e8.GWASHits.ChrBP.txt
      0       0       0
/mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp2_4cyto.TNF.a_lps.No0Dir.txt.gz.5e8.GWASHits.ChrBP.txt
      0       0       0
/mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp2_7cyto.IL.10_cd.No0Dir.txt.gz.5e8.GWASHits.ChrBP.txt
      5      10      57
/mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp2_7cyto.IL.13_cd.No0Dir.txt.gz.5e8.GWASHits.ChrBP.txt
      0       0       0
/mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp2_7cyto.IL.15_cd.No0Dir.txt.gz.5e8.GWASHits.ChrBP.txt
      0       0       0
/mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp2_7cyto.IL.23_cd.No0Dir.txt.gz.5e8.GWASHits.ChrBP.txt
      0       0       0
/mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp2_7cyto.IL.4_cd.No0Dir.txt.gz.5e8.GWASHits.ChrBP.txt
      0       0       0
/mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp2_7cyto.IL.5_cd.No0Dir.txt.gz.5e8.GWASHits.ChrBP.txt
      0       0       0
/mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp2_7cyto.IL.9_cd.No0Dir.txt.gz.5e8.GWASHits.ChrBP.txt
      0       0       0
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate]$for i in `ls -lrt /mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp1_7cyto.*.No0Dir.txt.gz | awk '{ print $9; }'`; do echo $i; zcat $i | awk '{ if ($6 < .00000005) { print $0 } } '; done
/mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp1_7cyto.IL.13_cd.No0Dir.txt.gz
/mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp1_7cyto.IL.15_cd.No0Dir.txt.gz
/mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp1_7cyto.IL.22_cd.No0Dir.txt.gz
/mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp1_7cyto.IL.33_cd.No0Dir.txt.gz
/mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp1_7cyto.IL.4_cd.No0Dir.txt.gz
/mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp1_7cyto.IL.5_cd.No0Dir.txt.gz
/mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp1_7cyto.IL.9_cd.No0Dir.txt.gz
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate]$for i in `ls -lrt /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp1_8cyto.*.No0Dir.txt.gz | awk '{ print $9; }'`; do echo $i; zcat $i | awk '{ if ($6 < .00000005) { print $0 } } '; done
/mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp1_8cyto.GMCSF_cd.No0Dir.txt.gz
/mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp1_8cyto.IFN.g_cd.No0Dir.txt.gz
/mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp1_8cyto.IL.1b_cd.No0Dir.txt.gz
/mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp1_8cyto.IL.2_cd.No0Dir.txt.gz
/mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp1_8cyto.IL.6_cd.No0Dir.txt.gz
/mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp1_8cyto.MIP3A_cd.No0Dir.txt.gz
/mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp1_8cyto.TNF.a_cd.No0Dir.txt.gz
/mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp1_8cyto.TNF.b_cd.No0Dir.txt.gz
~~~

R -q -e "library("devtools"); devtools::load_all("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/bmass");
IFN.g_lps <- read.table("/mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp2_4cyto.IFN.g_lps.No0Dir.txt.gz", header=T)
IL.12p70_lps <- read.table("/mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp2_4cyto.IL.12p70_lps.No0Dir.txt.gz", header=T)
IL.23_lps <- read.table("/mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp2_4cyto.IL.23_lps.No0Dir.txt.gz", header=T)
TNF.a_lps <- read.table("/mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp2_4cyto.TNF.a_lps.No0Dir.txt.gz", header=T)
GWASsnps <- NULL
#GWASsnps <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.ChrBP.txt", header=F);
#names(GWASsnps) <- c("Chr", "BP")
#bmassOutput1 <- bmass(c("HDL", "LDL", "TG", "TC"), GWASsnps=GWASsnps, NminThreshold = 1, bmassSeedValue=150);
#print(bmassOutput1);" > shana1.txt
DataSources = c("IFN.g_lps", "IL.12p70_lps", "IL.23_lps", "TNF.a_lps"); NminThreshold = 1; bmassSeedValue=150;
ExpectedColumnNames=c("Chr", "BP", "MAF", "Direction", "pValue", "N"); GWASsnps_AnnotateWindow = 5e5; SNPMarginalUnivariateThreshold = 1e-4; SNPMarginalMultivariateThreshold = 1e-4; SigmaAlphas = c(0.005,0.0075,0.01,0.015,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.15); ProvidedPriors=NULL; UseFlatPriors=TRUE; PruneMarginalSNPs=TRUE; PruneMarginalSNPs_bpWindow=5e5;

#All steps up to Finalize

set.seed(99)
MarginalSNPs_PrunedList <- indephits(jitter(bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg, factor=.0001), bmassOutput$MarginalSNPs$SNPs$Chr, bmassOutput$MarginalSNPs$SNPs$BP, T=PruneMarginalSNPs_bpWindow)
bmassOutput$MarginalSNPs$SNPs <- bmassOutput$MarginalSNPs$SNPs[MarginalSNPs_PrunedList==1,]
bmassOutput$MarginalSNPs$logBFs <- bmassOutput$MarginalSNPs$logBFs[,MarginalSNPs_PrunedList==1]

#Posteriors <- posteriorprob(bmassOutput$MarginalSNPs$logBFs, normalize(rep(c(0,ModelPriors[-1]),length(SigmaAlphas))))
Posteriors <- posteriorprob(bmassOutput$MarginalSNPs$logBFs, bmassOutput$ModelPriors)

Posteriors.marginals <- marginal.postprobs(Posteriors, bmassOutput$Models, length(SigmaAlphas))
Posteriors.marginals.DirAssoc <- t(Posteriors.marginals[[2]])
Posteriors.marginals.DirAssoc <- data.frame(cbind(as.character(bmassOutput$MarginalSNPs$SNPs$ChrBP), Posteriors.marginals.DirAssoc))
Posteriors.marginals.DirAssoc[,2] <- as.numeric(as.character(Posteriors.marginals.DirAssoc[,2]))
Posteriors.marginals.DirAssoc[,3] <- as.numeric(as.character(Posteriors.marginals.DirAssoc[,3]))
Posteriors.marginals.DirAssoc[,4] <- as.numeric(as.character(Posteriors.marginals.DirAssoc[,4]))
Posteriors.marginals.DirAssoc[,5] <- as.numeric(as.character(Posteriors.marginals.DirAssoc[,5]))
colnames(Posteriors.marginals.DirAssoc) <- c("snp", "IFN.g_lps", "IL.12p70_lps", "IL.23_lps", "TNF.a_lps")
Posteriors.marginals.DirAssoc.Subset <- Posteriors.marginals.DirAssoc[bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg > .5,]
Posteriors.marginals.DirAssoc <- Posteriors.marginals.DirAssoc[order(bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg, decreasing=TRUE),]
Posteriors.marginals.DirAssoc.Subset <- Posteriors.marginals.DirAssoc.Subset[order(bmassOutput$MarginalSNPs$SNPs[bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg > .5,]$logBFWeightedAvg, decreasing=TRUE),]

Z <- cbind(bmassOutput$MergedDataSources$IFN.g_lps_ZScore, bmassOutput$MergedDataSources$IL.12p70_lps_ZScore, bmassOutput$MergedDataSources$IL.23_lps_ZScore, bmassOutput$MergedDataSources$TNF.a_lps_ZScore)
Z.Directions <- apply(Z, c(1,2), function(x) { y <- 0; if (x > 0) { y <- 1; } else if (x < 0) { y <- -1; } else if (x == 0) { y <- 0; } else { y <- NA; }; return(y);})
Z.Directions.marginals <- Z.Directions[as.numeric(row.names(bmassOutput$MarginalSNPs$SNPs)),][order(bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg, decreasing=TRUE),]

Posteriors.marginals.DirAssoc.wDir <- Posteriors.marginals.DirAssoc
Posteriors.marginals.DirAssoc.wDir[,2:5] <- Posteriors.marginals.DirAssoc[,2:5] * Z.Directions.marginals
Posteriors.marginals.DirAssoc.Subset.wDir <- Posteriors.marginals.DirAssoc.Subset
Posteriors.marginals.DirAssoc.Subset.wDir[,2:5] <- Posteriors.marginals.DirAssoc.Subset[,2:5] * Z.Directions.marginals[bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg > .5,][order(bmassOutput$MarginalSNPs$SNPs[bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg > .5,]$logBFWeightedAvg, decreasing=TRUE),]

library(ggplot2)
library(reshape2)

#jpeg("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Stein2016/Vs1/Stein2016.Vs1.LPSGemma_Grp2_4cyto.Marginals.HeatPlot.All.vs1.jpeg", width=6000, height=2000, res=300)
#jpeg("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Stein2016/Vs1/Stein2016.Vs1.LPSGemma_Grp2_4cyto.Marginals.HeatPlot.Subset.vs1.jpeg", width=6000, height=2000, res=300)
jpeg("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Stein2016/Vs1/Stein2016.Vs1.LPSGemma_Grp2_4cyto.Marginals.HeatPlot.Subset.wDir.vs1.jpeg", width=6000, height=2000, res=300)

#ggPlotData <- melt(Posteriors.marginals.DirAssoc[hclust(dist(Posteriors.marginals.DirAssoc[,2:5]))$order,])
#ggPlotData <- melt(Posteriors.marginals.DirAssoc.wDir[hclust(dist(Posteriors.marginals.DirAssoc.wDir[,2:5]))$order,])
#ggPlotData <- melt(Posteriors.marginals.DirAssoc.Subset[hclust(dist(Posteriors.marginals.DirAssoc.Subset[,2:5]))$order,])
ggPlotData <- melt(Posteriors.marginals.DirAssoc.Subset.wDir[hclust(dist(Posteriors.marginals.DirAssoc.Subset[,2:5]))$order,])
#ggPlotData$snp <- factor(ggPlotData$snp, levels=Posteriors.marginals.DirAssoc[,1][hclust(dist(Posteriors.marginals.DirAssoc[,2:5]))$order])
ggPlotData$snp <- factor(ggPlotData$snp, levels=Posteriors.marginals.DirAssoc.Subset[,1][hclust(dist(Posteriors.marginals.DirAssoc.Subset[,2:5]))$order])

#ggplot(ggPlotData, aes(variable, snp, value)) + geom_tile(aes(fill=value), colour="white") + scale_fill_gradient(low = "white", high="steelblue") + coord_flip() + theme(axis.text.x = element_text(angle = 90, hjust = 1))
ggplot(ggPlotData, aes(variable, snp, value)) + geom_tile(aes(fill=value), colour="white") + scale_fill_gradientn(colours=c("darkred", "white", "steelblue")) + coord_flip() + theme(axis.text.x = element_text(angle = 90, hjust = 1))

dev.off()



R output
~~~
>         bmassOutput[c("MarginalSNPs", "Models", "ModelPriors", "LogFile")] <- GetLogBFsFromData(DataSources, bmassOutput$MarginalSNPs, bmassOutput$ZScoresCorMatrix, SigmaAlphas, bmassOutput$LogFile)[c("MarginalSNPs", "Models", "ModelPriors", "LogFile")]
> length(bmassOutput$ModelPriors)                                                                                                                                                                                              [1] 81
>         bmassOutput[c("MarginalSNPs", "PreviousSNPs", "ModelPriors", "GWASlogBFMinThreshold", "LogFile")] <- DetermineAndApplyPriors(DataSources, bmassOutput$MarginalSNPs, GWASsnps, SigmaAlphas, bmassOutput$Models, bmassOutput$ModelPriors, ProvidedPriors, UseFlatPriors, bmassSeedValue, bmassOutput$LogFile)[c("MarginalSNPs", "PreviousSNPs", "ModelPriors", "GWASlogBFMinThreshold", "LogFile")]
> length(bmassOutput$ModelPriors)                                                                                                                                                                                              [1] 1134
> dim(bmassOutput$MarginalSNPs$SNPs)
[1] 3339   28
> dim(bmassOutput$MergedDataSources)
[1] 4133176      22
> Posteriors <- posteriorprob(bmassOutput$MarginalSNPs$logBFs, bmassOutput$ModelPriors)
> 
> dim(Posteriors)
[1] 1134 3339
> MarginalSNPs_PrunedList <- indephits(bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg, bmassOutput$MarginalSNPs$SNPs$Chr, bmassOutput$MarginalSNPs$SNPs$BP, T=PruneMarginalSNPs_bpWindow)
> length(MarginalSNPs_PrunedList)
[1] 3339
> length(MarginalSNPs_PrunedList[MarginalSNPs_PrunedList==1])
[1] 878
> bmassOutput$MarginalSNPs$SNPs <- bmassOutput$MarginalSNPs$SNPs[MarginalSNPs_PrunedList==1,]
> dim(bmassOutput$MarginalSNPs$SNPs)
[1] 878  28
> bmassOutput$MarginalSNPs$logBFs <- bmassOutput$MarginalSNPs$logBFs[,MarginalSNPs_PrunedList==1]
> dim(bmassOutput$MarginalSNPs$logBFs)
[1] 1134  878
> head(Posteriors.marginals.DirAssoc)
            snp IFN.g_lps IL.12p70_lps IL.23_lps TNF.a_lps
796 9_110439352 0.6455953    0.7162374 0.9539106 0.8527608
797 9_110440040 0.6455953    0.7162374 0.9539106 0.8527608
431 3_159461456 0.4590177    0.9812005 0.4314724 0.3847088
120 13_83803328 0.7664079    0.5308406 0.9483302 0.4956707
164  1_74443864 0.9352095    0.5732003 0.8019028 0.5249784
681  6_31341672 0.8386581    0.6368238 0.5243734 0.6475138
> dim(bmassOutput$MarginalSNPs$SNPs[bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg > 1,])
[1] 14 28
> dim(bmassOutput$MarginalSNPs$SNPs[bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg > .8,])
[1] 57 28
> dim(bmassOutput$MarginalSNPs$SNPs[bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg > .7,])
[1] 70 28
> dim(bmassOutput$MarginalSNPs$SNPs[bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg > .4,])
[1] 320  28
> dim(bmassOutput$MarginalSNPs$SNPs[bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg > .5,])
[1] 259  28
> dim(bmassOutput$MarginalSNPs$SNPs[bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg > .6,])
[1] 125  28
> set.seed(99)
> MarginalSNPs_PrunedList <- indephits(jitter(bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg, factor=.0001), bmassOutput$MarginalSNPs$SNPs$Chr, bmassOutput$MarginalSNPs$SNPs$BP, T=PruneMarginalSNPs_bpWindow)
> dim(bmassOutput$MarginalSNPs$SNPs[MarginalSNPs_PrunedList==1,])
[1] 322  28
> dim(Posteriors.marginals.DirAssoc[bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg > .6,])
[1] 45  5
> dim(Posteriors.marginals.DirAssoc[bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg > .5,])
[1] 75  5
> dim(Posteriors.marginals.DirAssoc[bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg > .4,])
[1] 107   5
> dim(Posteriors.marginals.DirAssoc[bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg > .3,])
[1] 163   5
.
.
.
> head(bmassOutput$MarginalSNPs$SNPs[order(bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg, decreasing=TRUE),][bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg > .5,][order(bmassOutput$MarginalSNPs$SNPs[bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg > .5,]$logBFWeightedAvg, decreasing=TRUE),grep("ZScore", colnames(bmassOutput$MarginalSNPs$SNPs))])
        IFN.g_lps_ZScore IL.12p70_lps_ZScore IL.23_lps_ZScore TNF.a_lps_ZScore
80539          0.7543900            2.508924         2.076325         4.021061
2513824       -1.4191555           -3.413513        -4.746484        -1.533014
2543392        1.6915443            4.042268         3.764491         2.218352
3746817       -2.1217845           -4.291384        -4.157887        -2.984210
891361        -0.6239513           -1.749292        -3.984162        -1.414049
289306        -1.6276081           -2.251459        -4.054636        -1.689122
> head(Z.Directions.marginals[bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg > .5,][order(bmassOutput$MarginalSNPs$SNPs[bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg > .5,]$logBFWeightedAvg, decreasing=TRUE),])
     [,1] [,2] [,3] [,4]
[1,]    1    1    1    1
[2,]   -1   -1   -1   -1
[3,]    1    1    1    1
[4,]   -1   -1   -1   -1
[5,]   -1   -1   -1   -1
[6,]   -1   -1   -1   -1
>  head(bmassOutput$MarginalSNPs$SNPs[order(bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg, decreasing=TRUE),grep("ZScore", colnames(bmassOutput$MarginalSNPs$SNPs))]) 
        IFN.g_lps_ZScore IL.12p70_lps_ZScore IL.23_lps_ZScore TNF.a_lps_ZScore
3975078        -2.914954           -3.733187       -5.1011847       -4.3081857
2447868         2.040181            5.176852        1.7913587        0.9863100
962564          3.142782            1.557326        4.3938535        0.9301112
1446412        -4.448188           -1.777171       -3.4890046       -1.7314800
3364431        -2.406673            1.500273       -0.6493787        1.8072121
3980616        -2.860371           -1.849111       -3.6156639       -4.6984744
> head(Z.Directions.marginals)
     [,1] [,2] [,3] [,4]
[1,]   -1   -1   -1   -1
[2,]    1    1    1    1
[3,]    1    1    1    1
[4,]   -1   -1   -1   -1
[5,]   -1    1   -1    1
[6,]   -1   -1   -1   -1
> bmassOutput$ZScoresCorMatrix
                    IFN.g_lps_ZScore IL.12p70_lps_ZScore IL.23_lps_ZScore
IFN.g_lps_ZScore           1.0000000           0.5298928        0.2987746
IL.12p70_lps_ZScore        0.5298928           1.0000000        0.4538220
IL.23_lps_ZScore           0.2987746           0.4538220        1.0000000
TNF.a_lps_ZScore           0.3702133           0.2987075        0.4114108
                    TNF.a_lps_ZScore
IFN.g_lps_ZScore           0.3702133
IL.12p70_lps_ZScore        0.2987075
IL.23_lps_ZScore           0.4114108
TNF.a_lps_ZScore           1.0000000
~~~

cat /mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp1_7cyto.txt | awk '{ print $2 "\t" $3 "\t" $4 "\t" $5 "\t" $7 "\t" $8 "\t" $6 }' | perl -lane 'if ($F[4] > 0) { $F[4] = "+"; } elsif ($F[4] < 0) { $F[4] = "-"; } else { $F[4] = "NoDir"; } print join("\t", @F);' | cat <(echo -e "Chr\tBP\tMAF\tA1\tDirection\tpValue\tN") - | gzip > /mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp1_7cyto.IL.13_cd.txt.gz
cat /mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp1_7cyto.txt | awk '{ print $2 "\t" $3 "\t" $4 "\t" $5 "\t" $10 "\t" $11 "\t" $9 }' | perl -lane 'if ($F[4] > 0) { $F[4] = "+"; } elsif ($F[4] < 0) { $F[4] = "-"; } else { $F[4] = "NoDir"; } print join("\t", @F);' | cat <(echo -e "Chr\tBP\tMAF\tA1\tDirection\tpValue\tN") - | gzip > /mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp1_7cyto.IL.15_cd.txt.gz
cat /mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp1_7cyto.txt | awk '{ print $2 "\t" $3 "\t" $4 "\t" $5 "\t" $13 "\t" $14 "\t" $12 }' | perl -lane 'if ($F[4] > 0) { $F[4] = "+"; } elsif ($F[4] < 0) { $F[4] = "-"; } else { $F[4] = "NoDir"; } print join("\t", @F);' | cat <(echo -e "Chr\tBP\tMAF\tA1\tDirection\tpValue\tN") - | gzip > /mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp1_7cyto.IL.22_cd.txt.gz
cat /mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp1_7cyto.txt | awk '{ print $2 "\t" $3 "\t" $4 "\t" $5 "\t" $16 "\t" $17 "\t" $15 }' | perl -lane 'if ($F[4] > 0) { $F[4] = "+"; } elsif ($F[4] < 0) { $F[4] = "-"; } else { $F[4] = "NoDir"; } print join("\t", @F);' | cat <(echo -e "Chr\tBP\tMAF\tA1\tDirection\tpValue\tN") - | gzip > /mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp1_7cyto.IL.33_cd.txt.gz
cat /mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp1_7cyto.txt | awk '{ print $2 "\t" $3 "\t" $4 "\t" $5 "\t" $19 "\t" $20 "\t" $18 }' | perl -lane 'if ($F[4] > 0) { $F[4] = "+"; } elsif ($F[4] < 0) { $F[4] = "-"; } else { $F[4] = "NoDir"; } print join("\t", @F);' | cat <(echo -e "Chr\tBP\tMAF\tA1\tDirection\tpValue\tN") - | gzip > /mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp1_7cyto.IL.4_cd.txt.gz
cat /mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp1_7cyto.txt | awk '{ print $2 "\t" $3 "\t" $4 "\t" $5 "\t" $22 "\t" $23 "\t" $21 }' | perl -lane 'if ($F[4] > 0) { $F[4] = "+"; } elsif ($F[4] < 0) { $F[4] = "-"; } else { $F[4] = "NoDir"; } print join("\t", @F);' | cat <(echo -e "Chr\tBP\tMAF\tA1\tDirection\tpValue\tN") - | gzip > /mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp1_7cyto.IL.5_cd.txt.gz
cat /mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp1_7cyto.txt | awk '{ print $2 "\t" $3 "\t" $4 "\t" $5 "\t" $25 "\t" $26 "\t" $24 }' | perl -lane 'if ($F[4] > 0) { $F[4] = "+"; } elsif ($F[4] < 0) { $F[4] = "-"; } else { $F[4] = "NoDir"; } print join("\t", @F);' | cat <(echo -e "Chr\tBP\tMAF\tA1\tDirection\tpValue\tN") - | gzip > /mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp1_7cyto.IL.9_cd.txt.gz

zcat /mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp1_7cyto.IL.13_cd.txt.gz | grep -v NoDir | gzip > /mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp1_7cyto.IL.13_cd.No0Dir.txt.gz
zcat /mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp1_7cyto.IL.15_cd.txt.gz | grep -v NoDir | gzip > /mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp1_7cyto.IL.15_cd.No0Dir.txt.gz
zcat /mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp1_7cyto.IL.22_cd.txt.gz | grep -v NoDir | gzip > /mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp1_7cyto.IL.22_cd.No0Dir.txt.gz
zcat /mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp1_7cyto.IL.33_cd.txt.gz | grep -v NoDir | gzip > /mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp1_7cyto.IL.33_cd.No0Dir.txt.gz
zcat /mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp1_7cyto.IL.4_cd.txt.gz | grep -v NoDir | gzip > /mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp1_7cyto.IL.4_cd.No0Dir.txt.gz
zcat /mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp1_7cyto.IL.5_cd.txt.gz | grep -v NoDir | gzip > /mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp1_7cyto.IL.5_cd.No0Dir.txt.gz
zcat /mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp1_7cyto.IL.9_cd.txt.gz | grep -v NoDir | gzip > /mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp1_7cyto.IL.9_cd.No0Dir.txt.gz


~~~
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate]$cat /mnt/gluster/data/external_private_supp/Stein2016/LPSCytoNames_Grp1_7cyto.txt
"x"
"1" "IL.13_lps"
"2" "IL.15_lps"
"3" "IL.22_lps"
"4" "IL.33_lps"
"5" "IL.4_lps"
"6" "IL.5_lps"
"7" "IL.9_lps"
.
.
.
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/Stein2016/Vs1]$cat /mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp1_7cyto.txt | grep 38835018
"8660543" 12 38835018 0.103 3 107 -0.05477269 0.7036978 107 -0.116289 0.3576457 107 -0.1468588 0.1730663 107 -0.01469994 0.8610959 107 0.02006067 0.8910973 107 -0.3119887 0.008564315 107 -0.02922608 0.8188888
"8660544" 12 38835018 0.057 4 105 0.09579013 0.6321584 105 0.1450891 0.396663 105 0.1244478 0.4137071 105 0.1521323 0.1867208 105 0.3513101 0.07425048 105 0.3434218 0.02119242 105 0.2985855 0.08172489

~~~


R -q -e "library("devtools"); devtools::load_all("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/bmass");
IL.13_lps <- read.table("/mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp1_7cyto.IL.13_cd.No0Dir.txt.gz", header=T)
IL.15_lps <- read.table("/mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp1_7cyto.IL.15_cd.No0Dir.txt.gz", header=T)
IL.22_lps <- read.table("/mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp1_7cyto.IL.22_cd.No0Dir.txt.gz", header=T)
IL.33_lps <- read.table("/mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp1_7cyto.IL.33_cd.No0Dir.txt.gz", header=T)
IL.4_lps <- read.table("/mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp1_7cyto.IL.4_cd.No0Dir.txt.gz", header=T)
IL.5_lps <- read.table("/mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp1_7cyto.IL.5_cd.No0Dir.txt.gz", header=T)
IL.9_lps <- read.table("/mnt/gluster/data/external_private_supp/Stein2016/LPSGemma_Grp1_7cyto.IL.9_cd.No0Dir.txt.gz", header=T)
GWASsnps <- NULL
#GWASsnps <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.ChrBP.txt", header=F);
#names(GWASsnps) <- c("Chr", "BP")
#bmassOutput1 <- bmass(c("HDL", "LDL", "TG", "TC"), GWASsnps=GWASsnps, NminThreshold = 1, bmassSeedValue=150);
#print(bmassOutput1);" > shana1.txt
DataSources = c("IL.13_lps", "IL.15_lps", "IL.22_lps", "IL.33_lps", "IL.4_lps", "IL.5_lps", "IL.9_lps"); NminThreshold = 1; bmassSeedValue=150;
ExpectedColumnNames=c("Chr", "BP", "MAF", "Direction", "pValue", "N"); GWASsnps_AnnotateWindow = 5e5; SNPMarginalUnivariateThreshold = 1e-4; SNPMarginalMultivariateThreshold = 1e-4; SigmaAlphas = c(0.005,0.0075,0.01,0.015,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.15); ProvidedPriors=NULL; UseFlatPriors=TRUE; PruneMarginalSNPs=TRUE; PruneMarginalSNPs_bpWindow=5e5;

#After merging

Z <- cbind(bmassOutput$MergedDataSources$IL.13_lps_ZScore, bmassOutput$MergedDataSources$IL.15_lps_ZScore, bmassOutput$MergedDataSources$IL.22_lps_ZScore, bmassOutput$MergedDataSources$IL.33_lps_ZScore, bmassOutput$MergedDataSources$IL.4_lps_ZScore, bmassOutput$MergedDataSources$IL.5_lps_ZScore, bmassOutput$MergedDataSources$IL.9_lps_ZScore)
dim(bmassOutput$MergedDataSources)
dim(bmassOutput$MergedDataSources[!duplicated(bmassOutput$MergedDataSources[,1]),])
bmassOutput$MergedDataSources <- bmassOutput$MergedDataSources[!duplicated(bmassOutput$MergedDataSources[,1]),]

#All steps up to Finalize

set.seed(99)
MarginalSNPs_PrunedList <- indephits(jitter(bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg, factor=.0001), bmassOutput$MarginalSNPs$SNPs$Chr, bmassOutput$MarginalSNPs$SNPs$BP, T=PruneMarginalSNPs_bpWindow)
bmassOutput$MarginalSNPs$SNPs <- bmassOutput$MarginalSNPs$SNPs[MarginalSNPs_PrunedList==1,]
bmassOutput$MarginalSNPs$logBFs <- bmassOutput$MarginalSNPs$logBFs[,MarginalSNPs_PrunedList==1]

#Posteriors <- posteriorprob(bmassOutput$MarginalSNPs$logBFs, normalize(rep(c(0,ModelPriors[-1]),length(SigmaAlphas))))
Posteriors <- posteriorprob(bmassOutput$MarginalSNPs$logBFs, bmassOutput$ModelPriors)

Posteriors.marginals <- marginal.postprobs(Posteriors, bmassOutput$Models, length(SigmaAlphas))
Posteriors.marginals.DirAssoc <- t(Posteriors.marginals[[2]])
Posteriors.marginals.DirAssoc <- data.frame(cbind(as.character(bmassOutput$MarginalSNPs$SNPs$ChrBP), Posteriors.marginals.DirAssoc))
Posteriors.marginals.DirAssoc[,2] <- as.numeric(as.character(Posteriors.marginals.DirAssoc[,2]))
Posteriors.marginals.DirAssoc[,3] <- as.numeric(as.character(Posteriors.marginals.DirAssoc[,3]))
Posteriors.marginals.DirAssoc[,4] <- as.numeric(as.character(Posteriors.marginals.DirAssoc[,4]))
Posteriors.marginals.DirAssoc[,5] <- as.numeric(as.character(Posteriors.marginals.DirAssoc[,5]))
Posteriors.marginals.DirAssoc[,6] <- as.numeric(as.character(Posteriors.marginals.DirAssoc[,6]))
Posteriors.marginals.DirAssoc[,7] <- as.numeric(as.character(Posteriors.marginals.DirAssoc[,7]))
Posteriors.marginals.DirAssoc[,8] <- as.numeric(as.character(Posteriors.marginals.DirAssoc[,8]))
colnames(Posteriors.marginals.DirAssoc) <- c("snp", "IL.13_lps", "IL.15_lps", "IL.22_lps", "IL.33_lps", "IL.4_lps", "IL.5_lps", "IL.9_lps")
Posteriors.marginals.DirAssoc.Subset <- Posteriors.marginals.DirAssoc[bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg > .4,]
Posteriors.marginals.DirAssoc <- Posteriors.marginals.DirAssoc[order(bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg, decreasing=TRUE),]
Posteriors.marginals.DirAssoc.Subset <- Posteriors.marginals.DirAssoc.Subset[order(bmassOutput$MarginalSNPs$SNPs[bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg > .4,]$logBFWeightedAvg, decreasing=TRUE),]

#Z <- cbind(bmassOutput$MergedDataSources$IL.13_lps_ZScore, bmassOutput$MergedDataSources$IL.15_lps_ZScore, bmassOutput$MergedDataSources$IL.22_lps_ZScore, bmassOutput$MergedDataSources$IL.33_lps_ZScore, bmassOutput$MergedDataSources$IL.4_lps_ZScore, bmassOutput$MergedDataSources$IL.5_lps_ZScore, bmassOutput$MergedDataSources$IL.9_lps_ZScore)
Z.Directions <- apply(Z, c(1,2), function(x) { y <- 0; if (x > 0) { y <- 1; } else if (x < 0) { y <- -1; } else if (x == 0) { y <- 0; } else { y <- NA; }; return(y);})
Z.Directions.marginals <- Z.Directions[as.numeric(row.names(bmassOutput$MarginalSNPs$SNPs)),][order(bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg, decreasing=TRUE),]

Posteriors.marginals.DirAssoc.wDir <- Posteriors.marginals.DirAssoc
Posteriors.marginals.DirAssoc.wDir[,2:8] <- Posteriors.marginals.DirAssoc[,2:8] * Z.Directions.marginals
Posteriors.marginals.DirAssoc.Subset.wDir <- Posteriors.marginals.DirAssoc.Subset
Posteriors.marginals.DirAssoc.Subset.wDir[,2:8] <- Posteriors.marginals.DirAssoc.Subset[,2:8] * Z.Directions.marginals[bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg > .4,][order(bmassOutput$MarginalSNPs$SNPs[bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg > .4,]$logBFWeightedAvg, decreasing=TRUE),]

library(ggplot2)
library(reshape2)

#jpeg("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Stein2016/Vs1/Stein2016.Vs1.LPSGemma_Grp1_7cyto.Marginals.HeatPlot.All.vs1.jpeg", width=6000, height=2000, res=300)
jpeg("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Stein2016/Vs1/Stein2016.Vs1.LPSGemma_Grp1_7cyto.Marginals.HeatPlot.Subset.vs1.jpeg", width=6000, height=2000, res=300)
#jpeg("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Stein2016/Vs1/Stein2016.Vs1.LPSGemma_Grp1_7cyto.Marginals.HeatPlot.Subset.wDir.vs1.jpeg", width=6000, height=2000, res=300)

#ggPlotData <- melt(Posteriors.marginals.DirAssoc[hclust(dist(Posteriors.marginals.DirAssoc[,2:8]))$order,])
#ggPlotData <- melt(Posteriors.marginals.DirAssoc.wDir[hclust(dist(Posteriors.marginals.DirAssoc.wDir[,2:8]))$order,])
ggPlotData <- melt(Posteriors.marginals.DirAssoc.Subset[hclust(dist(Posteriors.marginals.DirAssoc.Subset[,2:8]))$order,])
#ggPlotData <- melt(Posteriors.marginals.DirAssoc.Subset.wDir[hclust(dist(Posteriors.marginals.DirAssoc.Subset[,2:8]))$order,])
#ggPlotData$snp <- factor(ggPlotData$snp, levels=Posteriors.marginals.DirAssoc[,1][hclust(dist(Posteriors.marginals.DirAssoc[,2:8]))$order])
ggPlotData$snp <- factor(ggPlotData$snp, levels=Posteriors.marginals.DirAssoc.Subset[,1][hclust(dist(Posteriors.marginals.DirAssoc.Subset[,2:8]))$order])

ggplot(ggPlotData, aes(variable, snp, value)) + geom_tile(aes(fill=value), colour="white") + scale_fill_gradient(low = "white", high="steelblue") + coord_flip() + theme(axis.text.x = element_text(angle = 91, hjust = 1))
#ggplot(ggPlotData, aes(variable, snp, value)) + geom_tile(aes(fill=value), colour="white") + scale_fill_gradientn(colours=c("darkred", "white", "steelblue")) + coord_flip() + theme(axis.text.x = element_text(angle = 90, hjust = 1))

dev.off()


R output
~~~
> DataSources
[1] "IL.13_lps" "IL.15_lps" "IL.22_lps" "IL.33_lps" "IL.4_lps"  "IL.5_lps" 
[7] "IL.9_lps" 
> dim(bmassOutput$MergedDataSources)
[1] 4306978      33
> library("pryr")
> object_size(bmassOutput$MergedDataSources)
1.07 GB
.
.
.
> dim(IL.13_lps)
[1] 4111636       7
> length(IL.13_lps)
[1] 7
> length(IL.13_lps[,2])
[1] 4111636
> length(unique(IL.13_lps[,2]))
[1] 4065059
"IL.13_lps", "IL.15_lps", "IL.22_lps", "IL.33_lps", "IL.4_lps", "IL.5_lps", "IL.9_lps")
.
.
.
> val1 <- paste(IL.13_lps[,1], IL.13_lps[,2], sep="_")
> length(val1)
[1] 4111636
> length(unique(val1))
[1] 4110101
> for (i in round(runif(5,1,length(IL.13_lps[duplicated(val1),2])))) { print(IL.13_lps[IL.13_lps[,2]==IL.13_lps[duplicated(val1),][i,2],]);}
        Chr       BP   MAF A1 Direction    pValue   N
2996609  12 38835018 0.103  3         - 0.7036978 107
2996610  12 38835018 0.057  4         + 0.6321584 105
        Chr       BP   MAF A1 Direction    pValue   N
1415666   5 84448958 0.397  5         + 0.3122069 107
1415667   5 84448958 0.397  2         + 0.3122069 107
        Chr       BP   MAF A1 Direction    pValue   N
3309267  14 40118357 0.269  2         + 0.3767578 104
3309268  14 40118357 0.269  3         + 0.3767578 104
        Chr        BP   MAF A1 Direction    pValue   N
1439241   5 100493344 0.055  5         + 0.2971801 100
1439242   5 100493344 0.416  2         + 0.6908818 107
       Chr       BP   MAF A1 Direction    pValue   N
361176   2 29373764 0.387  2         + 0.2799848 106
361177   2 29373764 0.387  4         + 0.2799848 106
> head(bmassOutput$MarginalSNPs$SNPs[order(bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg, decreasing=TRUE),], n=5)
              ChrBP Chr       BP   MAF A1 IL.13_lps_Direction IL.13_lps_pValue
3568986  6_64412189   6 64412189 0.420  4                   +      0.038144890
991649  13_78752332  13 78752332 0.416  3                   -      0.008028971
3568978  6_64412189   6 64412189 0.420  4                   +      0.038144890
3568987  6_64412189   6 64412189 0.420  4                   +      0.038144890
991657  13_78752332  13 78752332 0.416  3                   -      0.008028971
        IL.13_lps_N IL.13_lps_ZScore IL.15_lps_Direction IL.15_lps_pValue
3568986         106         2.073294                   +       0.02950447
991649          107        -2.650849                   -       0.03426028
3568978         106         2.073294                   +       0.02950447
3568987         106         2.073294                   +       0.02950447
991657          107        -2.650849                   -       0.03426028
> length(unique(bmassOutput$MergedDataSources[,1]))
[1] 4110101
> dim(bmassOutput$MergedDataSources)
[1] 4306978      34
.
.
.
> bmassOutput$MergedDataSources[bmassOutput$MergedDataSources[,1] %in% head(unique(bmassOutput$MergedDataSources[,1])),1:5]
         ChrBP Chr        BP   MAF A1
1 10_100000625  10 100000625 0.402  2
2 10_100000645  10 100000645 0.280  2
3 10_100003785  10 100003785 0.318  5
4 10_100004360  10 100004360 0.280  4
5 10_100004441  10 100004441 0.318  3
6 10_100004906  10 100004906 0.402  3
> dim(bmassOutput$MergedDataSources[bmassOutput$MergedDataSources[,1] %in% unique(bmassOutput$MergedDataSources[,1]),])
[1] 4306978      34
> FALSE %in% c(bmassOutput$MergedDataSources[,1] %in% unique(bmassOutput$MergedDataSources[,1]))
[1] FALSE
> head(unique(bmassOutput$MergedDataSources[,1]))
[1] "10_100000625" "10_100000645" "10_100003785" "10_100004360" "10_100004441"
[6] "10_100004906"
> head(bmassOutput$MergedDataSources[,1])
[1] "10_100000625" "10_100000645" "10_100003785" "10_100004360" "10_100004441"
[6] "10_100004906"
> length(unique(bmassOutput$MergedDataSources[,1]))
[1] 4110101
> length(bmassOutput$MergedDataSources[,1])
[1] 4306978
> "nana" %in% unique(bmassOutput$MergedDataSources[,1])
[1] FALSE
> dim(bmassOutput$MergedDataSources[!duplicated(bmassOutput$MergedDataSources[,1]),])
[1] 4110101      34
> table(duplicated(bmassOutput$MergedDataSources[,1]))

  FALSE    TRUE 
4110101  196877 
.
.
.
> dim(bmassOutput$MergedDataSources)
[1] 4306978      33
> dim(bmassOutput$MergedDataSources[!duplicated(bmassOutput$MergedDataSources[,1]),])
[1] 4110101      33
> bmassOutput$MergedDataSources <- bmassOutput$MergedDataSources[!duplicated(bmassOutput$MergedDataSources[,1]),]
.
.
.
> bmassOutput$ZScoresCorMatrix
                 IL.13_lps_ZScore IL.15_lps_ZScore IL.22_lps_ZScore
IL.13_lps_ZScore       1.00000000        0.1610763        0.5510241
IL.15_lps_ZScore       0.16107632        1.0000000        0.2315951
IL.22_lps_ZScore       0.55102411        0.2315951        1.0000000
IL.33_lps_ZScore       0.08358425        0.1926497        0.3722365
IL.4_lps_ZScore        0.35730252        0.3550513        0.2187726
IL.5_lps_ZScore        0.30340116        0.2139177        0.4099843
IL.9_lps_ZScore        0.27226482        0.3704924        0.3687152
                 IL.33_lps_ZScore IL.4_lps_ZScore IL.5_lps_ZScore
IL.13_lps_ZScore       0.08358425       0.3573025       0.3034012
IL.15_lps_ZScore       0.19264970       0.3550513       0.2139177
IL.22_lps_ZScore       0.37223650       0.2187726       0.4099843
IL.33_lps_ZScore       1.00000000       0.2018183       0.3293138
IL.4_lps_ZScore        0.20181829       1.0000000       0.4059243
IL.5_lps_ZScore        0.32931375       0.4059243       1.0000000
IL.9_lps_ZScore        0.22918768       0.3697856       0.4430454
                 IL.9_lps_ZScore
IL.13_lps_ZScore       0.2722648
IL.15_lps_ZScore       0.3704924
IL.22_lps_ZScore       0.3687152
IL.33_lps_ZScore       0.2291877
IL.4_lps_ZScore        0.3697856
IL.5_lps_ZScore        0.4430454
IL.9_lps_ZScore        1.0000000
> quantile(bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg)
       0%       25%       50%       75%      100% 
0.0399808 0.1034739 0.1926883 0.3099051 1.9484681 

~~~

cat /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp1_8cyto.txt | awk '{ print $2 "\t" $3 "\t" $4 "\t" $5 "\t" $7 "\t" $8 "\t" $6 }' | perl -lane 'if ($F[4] > 0) { $F[4] = "+"; } elsif ($F[4] < 0) { $F[4] = "-"; } else { $F[4] = "NoDir"; } print join("\t", @F);' | cat <(echo -e "Chr\tBP\tMAF\tA1\tDirection\tpValue\tN") - | gzip > /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp1_8cyto.GMCSF_cd.txt.gz
cat /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp1_8cyto.txt | awk '{ print $2 "\t" $3 "\t" $4 "\t" $5 "\t" $10 "\t" $11 "\t" $9 }' | perl -lane 'if ($F[4] > 0) { $F[4] = "+"; } elsif ($F[4] < 0) { $F[4] = "-"; } else { $F[4] = "NoDir"; } print join("\t", @F);' | cat <(echo -e "Chr\tBP\tMAF\tA1\tDirection\tpValue\tN") - | gzip > /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp1_8cyto.IFN.g_cd.txt.gz
cat /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp1_8cyto.txt | awk '{ print $2 "\t" $3 "\t" $4 "\t" $5 "\t" $13 "\t" $14 "\t" $12 }' | perl -lane 'if ($F[4] > 0) { $F[4] = "+"; } elsif ($F[4] < 0) { $F[4] = "-"; } else { $F[4] = "NoDir"; } print join("\t", @F);' | cat <(echo -e "Chr\tBP\tMAF\tA1\tDirection\tpValue\tN") - | gzip > /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp1_8cyto.IL.1b_cd.txt.gz
cat /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp1_8cyto.txt | awk '{ print $2 "\t" $3 "\t" $4 "\t" $5 "\t" $16 "\t" $17 "\t" $15 }' | perl -lane 'if ($F[4] > 0) { $F[4] = "+"; } elsif ($F[4] < 0) { $F[4] = "-"; } else { $F[4] = "NoDir"; } print join("\t", @F);' | cat <(echo -e "Chr\tBP\tMAF\tA1\tDirection\tpValue\tN") - | gzip > /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp1_8cyto.IL.2_cd.txt.gz
cat /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp1_8cyto.txt | awk '{ print $2 "\t" $3 "\t" $4 "\t" $5 "\t" $19 "\t" $20 "\t" $18 }' | perl -lane 'if ($F[4] > 0) { $F[4] = "+"; } elsif ($F[4] < 0) { $F[4] = "-"; } else { $F[4] = "NoDir"; } print join("\t", @F);' | cat <(echo -e "Chr\tBP\tMAF\tA1\tDirection\tpValue\tN") - | gzip > /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp1_8cyto.IL.6_cd.txt.gz
cat /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp1_8cyto.txt | awk '{ print $2 "\t" $3 "\t" $4 "\t" $5 "\t" $22 "\t" $23 "\t" $21 }' | perl -lane 'if ($F[4] > 0) { $F[4] = "+"; } elsif ($F[4] < 0) { $F[4] = "-"; } else { $F[4] = "NoDir"; } print join("\t", @F);' | cat <(echo -e "Chr\tBP\tMAF\tA1\tDirection\tpValue\tN") - | gzip > /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp1_8cyto.MIP3A_cd.txt.gz
cat /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp1_8cyto.txt | awk '{ print $2 "\t" $3 "\t" $4 "\t" $5 "\t" $25 "\t" $26 "\t" $24 }' | perl -lane 'if ($F[4] > 0) { $F[4] = "+"; } elsif ($F[4] < 0) { $F[4] = "-"; } else { $F[4] = "NoDir"; } print join("\t", @F);' | cat <(echo -e "Chr\tBP\tMAF\tA1\tDirection\tpValue\tN") - | gzip > /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp1_8cyto.TNF.a_cd.txt.gz
cat /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp1_8cyto.txt | awk '{ print $2 "\t" $3 "\t" $4 "\t" $5 "\t" $28 "\t" $29 "\t" $27 }' | perl -lane 'if ($F[4] > 0) { $F[4] = "+"; } elsif ($F[4] < 0) { $F[4] = "-"; } else { $F[4] = "NoDir"; } print join("\t", @F);' | cat <(echo -e "Chr\tBP\tMAF\tA1\tDirection\tpValue\tN") - | gzip > /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp1_8cyto.TNF.b_cd.txt.gz

zcat /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp1_8cyto.GMCSF_cd.txt.gz | grep -v NoDir | gzip > /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp1_8cyto.GMCSF_cd.No0Dir.txt.gz
zcat /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp1_8cyto.IFN.g_cd.txt.gz | grep -v NoDir | gzip > /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp1_8cyto.IFN.g_cd.No0Dir.txt.gz
zcat /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp1_8cyto.IL.1b_cd.txt.gz | grep -v NoDir | gzip > /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp1_8cyto.IL.1b_cd.No0Dir.txt.gz
zcat /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp1_8cyto.IL.2_cd.txt.gz | grep -v NoDir | gzip > /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp1_8cyto.IL.2_cd.No0Dir.txt.gz
zcat /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp1_8cyto.IL.6_cd.txt.gz | grep -v NoDir | gzip > /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp1_8cyto.IL.6_cd.No0Dir.txt.gz
zcat /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp1_8cyto.MIP3A_cd.txt.gz | grep -v NoDir | gzip > /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp1_8cyto.MIP3A_cd.No0Dir.txt.gz
zcat /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp1_8cyto.TNF.a_cd.txt.gz | grep -v NoDir | gzip > /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp1_8cyto.TNF.a_cd.No0Dir.txt.gz
zcat /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp1_8cyto.TNF.b_cd.txt.gz | grep -v NoDir | gzip > /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp1_8cyto.TNF.b_cd.No0Dir.txt.gz

~~~
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate]$cat /mnt/gluster/data/external_private_supp/Stein2016/CDCytoNames_Grp1_8cyto.txt 
"x"
"1" "GMCSF_cd"
"2" "IFN.g_cd"
"3" "IL.1b_cd"
"4" "IL.2_cd"
"5" "IL.6_cd"
"6" "MIP3A_cd"
"7" "TNF.a_cd"
"8" "TNF.b_cd"
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate]$cat /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp1_8cyto.txt | head -n 10
"chr" "ps" "MAF" "ref_allele" "n" "beta" "p_val" "n" "beta" "p_val" "n" "beta" "p_val" "n" "beta" "p_val" "n" "beta" "p_val" "n" "beta" "p_val" "n" "beta" "p_val" "n" "beta" "p_val"
"1062" 1 785989 0.087 3 86 0.4753319 0.01061618 86 0.2197598 0.3183647 86 0.2210381 0.3255201 86 0.187842 0.3723487 86 0.2449633 0.2571314 86 0.07485174 0.678315 86 0.3543271 0.09536621 86 0.05587719 0.7710176
"2548" 1 1003629 0.276 5 85 -0.01874137 0.8859609 85 -0.02415705 0.875509 85 -0.1152311 0.4570165 85 0.05774578 0.689072 85 -0.1029904 0.4898695 85 0.00468391 0.9701497 85 -0.08071752 0.5729249 85 0.2442303 0.06176981
"3767" 1 1211292 0.186 3 86 0.06657314 0.6393984 86 0.05952447 0.7213611 86 0.09887736 0.555756 86 0.203982 0.1981636 86 -0.06036034 0.708823 86 -0.01133245 0.9331616 86 0.04546671 0.76769 86 0.1638528 0.2545144
"5323" 1 1478153 0.36 5 86 -0.06763584 0.5852594 86 0.06765619 0.625145 86 -0.003029514 0.9818628 86 0.225764 0.06889433 86 0.04086542 0.7499444 86 0.1278392 0.2288152 86 0.07449873 0.5413528 86 0.2661852 0.01698087
"5442" 1 1500941 0.36 2 86 -0.06763584 0.5852594 86 0.06765619 0.625145 86 -0.003029514 0.9818628 86 0.225764 0.06889433 86 0.04086542 0.7499444 86 0.1278392 0.2288152 86 0.07449873 0.5413528 86 0.2661852 0.01698087
"5511" 1 1510801 0.32 3 86 0.1114195 0.3469383 86 0.1488356 0.2525479 86 -0.03336247 0.8010324 86 0.3473009 0.002596369 86 -0.014909 0.9068666 86 0.09294869 0.3795356 86 0.1192796 0.3241256 86 0.2558212 0.02109778
"6774" 1 1692321 0.43 5 86 -0.1597153 0.106821 86 -0.03440904 0.7806821 86 0.06172808 0.6096533 86 -0.2517566 0.01945874 86 0.06916918 0.5521142 86 -0.06571159 0.4970195 86 -0.1194525 0.2795911 86 -0.01399911 0.8919315
"6884" 1 1721479 0.395 3 86 0.1364934 0.1578405 86 0.1333809 0.2487149 86 0.007127612 0.9492402 86 0.1584345 0.1458642 86 -0.09415163 0.3811464 86 0.05585931 0.5464744 86 0.1793559 0.07755196 86 -0.03496942 0.713697
"6888" 1 1722932 0.399 3 84 0.1242914 0.2062543 84 0.1204675 0.3034659 84 -0.01213535 0.9146448 84 0.1475215 0.1815631 84 -0.113025 0.2980433 84 0.05233822 0.576433 84 0.1682488 0.1020359 84 -0.0477868 0.6199267

~~~

R -q -e "library("devtools"); devtools::load_all("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/bmass");
GMCSF_cd <- read.table("/mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp1_8cyto.GMCSF_cd.No0Dir.txt.gz", header=T)
IFN.g_cd <- read.table("/mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp1_8cyto.IFN.g_cd.No0Dir.txt.gz", header=T)
IL.1b_cd <- read.table("/mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp1_8cyto.IL.1b_cd.No0Dir.txt.gz", header=T)
IL.2_cd <- read.table("/mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp1_8cyto.IL.2_cd.No0Dir.txt.gz", header=T)
IL.6_cd <- read.table("/mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp1_8cyto.IL.6_cd.No0Dir.txt.gz", header=T)
MIP3A_cd <- read.table("/mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp1_8cyto.MIP3A_cd.No0Dir.txt.gz", header=T)
TNF.a_cd <- read.table("/mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp1_8cyto.TNF.a_cd.No0Dir.txt.gz", header=T)
TNF.b_cd <- read.table("/mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp1_8cyto.TNF.b_cd.No0Dir.txt.gz", header=T)
GWASsnps <- NULL
#GWASsnps <- read.table("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.ChrBP.txt", header=F);
#names(GWASsnps) <- c("Chr", "BP")
#bmassOutput1 <- bmass(c("HDL", "LDL", "TG", "TC"), GWASsnps=GWASsnps, NminThreshold = 1, bmassSeedValue=150);
#print(bmassOutput1);" > shana1.txt
DataSources = c("GMCSF_cd", "IFN.g_cd", "IL.1b_cd", "IL.2_cd", "IL.6_cd", "MIP3A_cd", "TNF.a_cd", "TNF.b_cd"); NminThreshold = 1; bmassSeedValue=150;
ExpectedColumnNames=c("Chr", "BP", "MAF", "Direction", "pValue", "N"); GWASsnps_AnnotateWindow = 5e5; SNPMarginalUnivariateThreshold = 1e-4; SNPMarginalMultivariateThreshold = 1e-4; SigmaAlphas = c(0.005,0.0075,0.01,0.015,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.15); ProvidedPriors=NULL; UseFlatPriors=TRUE; PruneMarginalSNPs=TRUE; PruneMarginalSNPs_bpWindow=5e5;

#After merging

Z <- cbind(bmassOutput$MergedDataSources$GMCSF_cd_ZScore, bmassOutput$MergedDataSources$IFN.g_cd_ZScore, bmassOutput$MergedDataSources$IL.1b_cd_ZScore, bmassOutput$MergedDataSources$IL.2_cd_ZScore, bmassOutput$MergedDataSources$IL.6_cd_ZScore, bmassOutput$MergedDataSources$MIP3A_cd_ZScore, bmassOutput$MergedDataSources$TNF.a_cd_ZScore, bmassOutput$MergedDataSources$TNF.b_cd_ZScore)
dim(bmassOutput$MergedDataSources)
dim(bmassOutput$MergedDataSources[!duplicated(bmassOutput$MergedDataSources[,1]),])
bmassOutput$MergedDataSources <- bmassOutput$MergedDataSources[!duplicated(bmassOutput$MergedDataSources[,1]),]

#All steps up to Finalize

set.seed(99)
MarginalSNPs_PrunedList <- indephits(jitter(bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg, factor=.0001), bmassOutput$MarginalSNPs$SNPs$Chr, bmassOutput$MarginalSNPs$SNPs$BP, T=PruneMarginalSNPs_bpWindow)
bmassOutput$MarginalSNPs$SNPs <- bmassOutput$MarginalSNPs$SNPs[MarginalSNPs_PrunedList==1,]
bmassOutput$MarginalSNPs$logBFs <- bmassOutput$MarginalSNPs$logBFs[,MarginalSNPs_PrunedList==1]

#Posteriors <- posteriorprob(bmassOutput$MarginalSNPs$logBFs, normalize(rep(c(0,ModelPriors[-1]),length(SigmaAlphas))))
Posteriors <- posteriorprob(bmassOutput$MarginalSNPs$logBFs, bmassOutput$ModelPriors)

Posteriors.marginals <- marginal.postprobs(Posteriors, bmassOutput$Models, length(SigmaAlphas))
Posteriors.marginals.DirAssoc <- t(Posteriors.marginals[[2]])
Posteriors.marginals.DirAssoc <- data.frame(cbind(as.character(bmassOutput$MarginalSNPs$SNPs$ChrBP), Posteriors.marginals.DirAssoc))
Posteriors.marginals.DirAssoc[,2] <- as.numeric(as.character(Posteriors.marginals.DirAssoc[,2]))
Posteriors.marginals.DirAssoc[,3] <- as.numeric(as.character(Posteriors.marginals.DirAssoc[,3]))
Posteriors.marginals.DirAssoc[,4] <- as.numeric(as.character(Posteriors.marginals.DirAssoc[,4]))
Posteriors.marginals.DirAssoc[,5] <- as.numeric(as.character(Posteriors.marginals.DirAssoc[,5]))
Posteriors.marginals.DirAssoc[,6] <- as.numeric(as.character(Posteriors.marginals.DirAssoc[,6]))
Posteriors.marginals.DirAssoc[,7] <- as.numeric(as.character(Posteriors.marginals.DirAssoc[,7]))
Posteriors.marginals.DirAssoc[,8] <- as.numeric(as.character(Posteriors.marginals.DirAssoc[,8]))
Posteriors.marginals.DirAssoc[,9] <- as.numeric(as.character(Posteriors.marginals.DirAssoc[,9]))
colnames(Posteriors.marginals.DirAssoc) <- c("snp", "GMCSF_cd", "IFN.g_cd", "IL.1b_cd", "IL.2_cd", "IL.6_cd", "MIP3A_cd", "TNF.a_cd", "TNF.b_cd")
Posteriors.marginals.DirAssoc.Subset <- Posteriors.marginals.DirAssoc[bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg > .3,]
Posteriors.marginals.DirAssoc <- Posteriors.marginals.DirAssoc[order(bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg, decreasing=TRUE),]
Posteriors.marginals.DirAssoc.Subset <- Posteriors.marginals.DirAssoc.Subset[order(bmassOutput$MarginalSNPs$SNPs[bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg > .3,]$logBFWeightedAvg, decreasing=TRUE),]

#Z <- cbind(bmassOutput$MergedDataSources$GMCSF_cd_ZScore, bmassOutput$MergedDataSources$IFN.g_cd_ZScore, bmassOutput$MergedDataSources$IL.1b_cd_ZScore, bmassOutput$MergedDataSources$IL.2_cd_ZScore, bmassOutput$MergedDataSources$IL.6_cd_ZScore, bmassOutput$MergedDataSources$MIP3A_cd_ZScore, bmassOutput$MergedDataSources$TNF.a_cd_ZScore, bmassOutput$MergedDataSources$TNF.b_cd_ZScore)
Z.Directions <- apply(Z, c(1,2), function(x) { y <- 0; if (x > 0) { y <- 1; } else if (x < 0) { y <- -1; } else if (x == 0) { y <- 0; } else { y <- NA; }; return(y);})
Z.Directions.marginals <- Z.Directions[as.numeric(row.names(bmassOutput$MarginalSNPs$SNPs)),][order(bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg, decreasing=TRUE),]

Posteriors.marginals.DirAssoc.wDir <- Posteriors.marginals.DirAssoc
Posteriors.marginals.DirAssoc.wDir[,2:9] <- Posteriors.marginals.DirAssoc[,2:9] * Z.Directions.marginals
Posteriors.marginals.DirAssoc.Subset.wDir <- Posteriors.marginals.DirAssoc.Subset
Posteriors.marginals.DirAssoc.Subset.wDir[,2:9] <- Posteriors.marginals.DirAssoc.Subset[,2:9] * Z.Directions.marginals[bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg > .3,][order(bmassOutput$MarginalSNPs$SNPs[bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg > .3,]$logBFWeightedAvg, decreasing=TRUE),]

library(ggplot2)
library(reshape2)

#jpeg("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Stein2016/Vs1/Stein2016.Vs1.CDGemma_Grp1_8cyto.Marginals.HeatPlot.All.vs1.jpeg", width=6000, height=2000, res=300)
#jpeg("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Stein2016/Vs1/Stein2016.Vs1.CDGemma_Grp1_8cyto.Marginals.HeatPlot.Subset.vs1.jpeg", width=6000, height=2000, res=300)
jpeg("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Stein2016/Vs1/Stein2016.Vs1.CDGemma_Grp1_8cyto.Marginals.HeatPlot.Subset.wDir.vs1.jpeg", width=6000, height=2000, res=300)

#ggPlotData <- melt(Posteriors.marginals.DirAssoc[hclust(dist(Posteriors.marginals.DirAssoc[,2:9]))$order,])
#ggPlotData <- melt(Posteriors.marginals.DirAssoc.wDir[hclust(dist(Posteriors.marginals.DirAssoc.wDir[,2:9]))$order,])
#ggPlotData <- melt(Posteriors.marginals.DirAssoc.Subset[hclust(dist(Posteriors.marginals.DirAssoc.Subset[,2:9]))$order,])
ggPlotData <- melt(Posteriors.marginals.DirAssoc.Subset.wDir[hclust(dist(Posteriors.marginals.DirAssoc.Subset[,2:9]))$order,])
#ggPlotData$snp <- factor(ggPlotData$snp, levels=Posteriors.marginals.DirAssoc[,1][hclust(dist(Posteriors.marginals.DirAssoc[,2:9]))$order])
ggPlotData$snp <- factor(ggPlotData$snp, levels=Posteriors.marginals.DirAssoc.Subset[,1][hclust(dist(Posteriors.marginals.DirAssoc.Subset[,2:9]))$order])

#ggplot(ggPlotData, aes(variable, snp, value)) + geom_tile(aes(fill=value), colour="white") + scale_fill_gradient(low = "white", high="steelblue") + coord_flip() + theme(axis.text.x = element_text(angle = 90, hjust = 1))
ggplot(ggPlotData, aes(variable, snp, value)) + geom_tile(aes(fill=value), colour="white") + scale_fill_gradientn(colours=c("darkred", "white", "steelblue")) + coord_flip() + theme(axis.text.x = element_text(angle = 90, hjust = 1))

dev.off()

R output
~~~
> dim(bmassOutput$MergedDataSources)
[1] 4507559      37
> dim(bmassOutput$MergedDataSources[!duplicated(bmassOutput$MergedDataSources[,1]),])
[1] 4110084      37
.
.
.
> dim(bmassOutput$MarginalSNPs$SNPs[bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg > .4,])
[1] 33 44
> dim(bmassOutput$MarginalSNPs$SNPs[bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg > .5,])
[1] 19 44
> dim(bmassOutput$MarginalSNPs$SNPs[bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg > .3,])
[1] 55 44
> bmassOutput$ZScoresCorMatrix
                GMCSF_cd_ZScore IFN.g_cd_ZScore IL.1b_cd_ZScore IL.2_cd_ZScore
GMCSF_cd_ZScore       1.0000000       0.4973878       0.1525728      0.4769355
IFN.g_cd_ZScore       0.4973878       1.0000000       0.5039446      0.4070713
IL.1b_cd_ZScore       0.1525728       0.5039446       1.0000000      0.0932959
IL.2_cd_ZScore        0.4769355       0.4070713       0.0932959      1.0000000
IL.6_cd_ZScore        0.1054800       0.1945063       0.5954770      0.1631806
MIP3A_cd_ZScore       0.1883726       0.3224038       0.4036079      0.3056533
TNF.a_cd_ZScore       0.4077857       0.5412116       0.4153842      0.1585173
TNF.b_cd_ZScore       0.1777196       0.3412471       0.2898973      0.4214811
                IL.6_cd_ZScore MIP3A_cd_ZScore TNF.a_cd_ZScore TNF.b_cd_ZScore
GMCSF_cd_ZScore      0.1054800       0.1883726       0.4077857       0.1777196
IFN.g_cd_ZScore      0.1945063       0.3224038       0.5412116       0.3412471
IL.1b_cd_ZScore      0.5954770       0.4036079       0.4153842       0.2898973
IL.2_cd_ZScore       0.1631806       0.3056533       0.1585173       0.4214811
IL.6_cd_ZScore       1.0000000       0.5483300       0.2150846       0.1780315
MIP3A_cd_ZScore      0.5483300       1.0000000       0.4666557       0.4103428
TNF.a_cd_ZScore      0.2150846       0.4666557       1.0000000       0.2720482
TNF.b_cd_ZScore      0.1780315       0.4103428       0.2720482       1.0000000

~~~


cat /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp2_7cyto.txt | awk '{ print $2 "\t" $3 "\t" $4 "\t" $5 "\t" $7 "\t" $8 "\t" $6 }' | perl -lane 'if ($F[4] > 0) { $F[4] = "+"; } elsif ($F[4] < 0) { $F[4] = "-"; } else { $F[4] = "NoDir"; } print join("\t", @F);' | cat <(echo -e "Chr\tBP\tMAF\tA1\tDirection\tpValue\tN") - | gzip > /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp2_7cyto.IL.10_cd.txt.gz
cat /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp2_7cyto.txt | awk '{ print $2 "\t" $3 "\t" $4 "\t" $5 "\t" $10 "\t" $11 "\t" $9 }' | perl -lane 'if ($F[4] > 0) { $F[4] = "+"; } elsif ($F[4] < 0) { $F[4] = "-"; } else { $F[4] = "NoDir"; } print join("\t", @F);' | cat <(echo -e "Chr\tBP\tMAF\tA1\tDirection\tpValue\tN") - | gzip > /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp2_7cyto.IL.13_cd.txt.gz
cat /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp2_7cyto.txt | awk '{ print $2 "\t" $3 "\t" $4 "\t" $5 "\t" $13 "\t" $14 "\t" $12 }' | perl -lane 'if ($F[4] > 0) { $F[4] = "+"; } elsif ($F[4] < 0) { $F[4] = "-"; } else { $F[4] = "NoDir"; } print join("\t", @F);' | cat <(echo -e "Chr\tBP\tMAF\tA1\tDirection\tpValue\tN") - | gzip > /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp2_7cyto.IL.15_cd.txt.gz
cat /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp2_7cyto.txt | awk '{ print $2 "\t" $3 "\t" $4 "\t" $5 "\t" $16 "\t" $17 "\t" $15 }' | perl -lane 'if ($F[4] > 0) { $F[4] = "+"; } elsif ($F[4] < 0) { $F[4] = "-"; } else { $F[4] = "NoDir"; } print join("\t", @F);' | cat <(echo -e "Chr\tBP\tMAF\tA1\tDirection\tpValue\tN") - | gzip > /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp2_7cyto.IL.23_cd.txt.gz
cat /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp2_7cyto.txt | awk '{ print $2 "\t" $3 "\t" $4 "\t" $5 "\t" $19 "\t" $20 "\t" $18 }' | perl -lane 'if ($F[4] > 0) { $F[4] = "+"; } elsif ($F[4] < 0) { $F[4] = "-"; } else { $F[4] = "NoDir"; } print join("\t", @F);' | cat <(echo -e "Chr\tBP\tMAF\tA1\tDirection\tpValue\tN") - | gzip > /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp2_7cyto.IL.4_cd.txt.gz
cat /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp2_7cyto.txt | awk '{ print $2 "\t" $3 "\t" $4 "\t" $5 "\t" $22 "\t" $23 "\t" $21 }' | perl -lane 'if ($F[4] > 0) { $F[4] = "+"; } elsif ($F[4] < 0) { $F[4] = "-"; } else { $F[4] = "NoDir"; } print join("\t", @F);' | cat <(echo -e "Chr\tBP\tMAF\tA1\tDirection\tpValue\tN") - | gzip > /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp2_7cyto.IL.5_cd.txt.gz
cat /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp2_7cyto.txt | awk '{ print $2 "\t" $3 "\t" $4 "\t" $5 "\t" $25 "\t" $26 "\t" $24 }' | perl -lane 'if ($F[4] > 0) { $F[4] = "+"; } elsif ($F[4] < 0) { $F[4] = "-"; } else { $F[4] = "NoDir"; } print join("\t", @F);' | cat <(echo -e "Chr\tBP\tMAF\tA1\tDirection\tpValue\tN") - | gzip > /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp2_7cyto.IL.9_cd.txt.gz

zcat /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp2_7cyto.IL.10_cd.txt.gz | grep -v NoDir | gzip > /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp2_7cyto.IL.10_cd.No0Dir.txt.gz
zcat /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp2_7cyto.IL.13_cd.txt.gz | grep -v NoDir | gzip > /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp2_7cyto.IL.13_cd.No0Dir.txt.gz
zcat /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp2_7cyto.IL.15_cd.txt.gz | grep -v NoDir | gzip > /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp2_7cyto.IL.15_cd.No0Dir.txt.gz
zcat /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp2_7cyto.IL.23_cd.txt.gz | grep -v NoDir | gzip > /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp2_7cyto.IL.23_cd.No0Dir.txt.gz
zcat /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp2_7cyto.IL.4_cd.txt.gz | grep -v NoDir | gzip > /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp2_7cyto.IL.4_cd.No0Dir.txt.gz
zcat /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp2_7cyto.IL.5_cd.txt.gz | grep -v NoDir | gzip > /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp2_7cyto.IL.5_cd.No0Dir.txt.gz
zcat /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp2_7cyto.IL.9_cd.txt.gz | grep -v NoDir | gzip > /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp2_7cyto.IL.9_cd.No0Dir.txt.gz

~~~
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate]$cat /mnt/gluster/data/external_private_supp/Stein2016/CDCytoNames_Grp2_7cyto.txt | head -n 10
"x"
"1" "IL.10_cd"
"2" "IL.13_cd"
"3" "IL.15_cd"
"4" "IL.23_cd"
"5" "IL.4_cd"
"6" "IL.5_cd"
"7" "IL.9_cd"
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate]$cat /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp2_7cyto.txt | head -n 10
"chr" "ps" "MAF" "ref_allele" "n" "beta" "p_val" "n" "beta" "p_val" "n" "beta" "p_val" "n" "beta" "p_val" "n" "beta" "p_val" "n" "beta" "p_val" "n" "beta" "p_val"
"1062" 1 785989 0.087 3 86 0.04210777 0.857362 86 0.2707796 0.1450781 86 -0.08197585 0.6050011 86 -0.02302538 0.8624858 86 0.2184266 0.2367246 86 0.3500217 0.1062334 86 0.3668583 0.09947047
"2548" 1 1003629 0.276 5 85 0.3778017 0.01570569 85 0.2399258 0.05837053 85 -0.01528276 0.8886623 85 0.03871112 0.6684948 85 0.1852996 0.1343398 85 0.3566824 0.01289832 85 0.3558245 0.01642738
"3767" 1 1211292 0.186 3 86 0.08382453 0.6357063 86 0.04601755 0.7416195 86 0.03017982 0.7984058 86 0.2051074 0.04137444 86 0.1006134 0.4698077 86 0.1206348 0.4626514 86 0.3684652 0.02782147
"5323" 1 1478153 0.36 5 86 0.2872235 0.05092702 86 0.1270515 0.2463335 86 0.05717154 0.5420885 86 0.1354669 0.07697241 86 0.09230824 0.4015854 86 0.0854664 0.5208085 86 0.2033144 0.1550232
"5442" 1 1500941 0.36 2 86 0.2872235 0.05092702 86 0.1270515 0.2463335 86 0.05717154 0.5420885 86 0.1354669 0.07697241 86 0.09230824 0.4015854 86 0.0854664 0.5208085 86 0.2033144 0.1550232
"5511" 1 1510801 0.32 3 86 0.3612212 0.01231594 86 0.1590263 0.1431924 86 0.07395419 0.4262997 86 0.1345635 0.07705826 86 0.1130161 0.2983287 86 0.1437184 0.2740993 86 0.2536837 0.07353442
"6774" 1 1692321 0.43 5 86 -0.1834548 0.1608587 86 0.06161833 0.5369827 86 -0.06123552 0.4711575 86 -0.06671411 0.3417241 86 0.0470585 0.6366111 86 -0.04917271 0.6801686 86 -0.1707377 0.1749795
"6884" 1 1721479 0.395 3 86 0.03752578 0.7686483 86 -0.116953 0.2036479 86 0.06085456 0.4388749 86 0.05565388 0.4106019 86 -0.1047288 0.2441583 86 -0.06104613 0.5713475 86 0.04928359 0.6813373
"6888" 1 1722932 0.399 3 84 0.01511207 0.9066229 84 -0.1319491 0.1556098 84 0.04897511 0.5381491 84 0.06262799 0.3628721 84 -0.1222968 0.1761535 84 -0.07731144 0.4746334 84 0.01995509 0.8688484
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate]$for i in `ls -lrt /mnt/gluster/data/external_private_supp/Stein2016/ | grep CDGemma_Grp2_7cyto.IL.* | grep -v No0Dir | awk '{ print $9 }'`; do echo $i; zcat /mnt/gluster/data/external_private_supp/Stein2016/$i | wc; zcat /mnt/gluster/data/external_private_supp/Stein2016/$i | grep -v NoDir | wc; done
CDGemma_Grp2_7cyto.IL.10_cd.txt.gz
4111621 28781347 142032290
4111620 28781340 142032245
CDGemma_Grp2_7cyto.IL.13_cd.txt.gz
4111621 28781347 141950214
4111620 28781340 141950169
CDGemma_Grp2_7cyto.IL.15_cd.txt.gz
4111621 28781347 141984864
4111620 28781340 141984819
CDGemma_Grp2_7cyto.IL.23_cd.txt.gz
4111621 28781347 142003921
4111620 28781340 142003876
CDGemma_Grp2_7cyto.IL.4_cd.txt.gz
4111621 28781347 142044814
4111620 28781340 142044769
CDGemma_Grp2_7cyto.IL.5_cd.txt.gz
4111621 28781347 142053590
4111620 28781340 142053545
CDGemma_Grp2_7cyto.IL.9_cd.txt.gz
4111621 28781346 142089345
4111620 28781340 142089304
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate]$zcat /mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp2_7cyto.IL.10_cd.txt.gz | head -n 10
Chr BP  MAF     A1      Direction       pValue  N
"ps" "MAF"      "ref_allele"    "n"     NoDir   "n"     "beta"
1  785989       0.087   3       +       0.857362        86
1  1003629      0.276   5       +       0.01570569      85
1  1211292      0.186   3       +       0.6357063       86
1  1478153      0.36    5       +       0.05092702      86
1  1500941      0.36    2       +       0.05092702      86
1  1510801      0.32    3       +       0.01231594      86
1  1692321      0.43    5       -       0.1608587       86
1  1721479      0.395   3       +       0.7686483       86

~~~
"1" "IL.10_cd"  
"2" "IL.13_cd"  
"3" "IL.15_cd"  
"4" "IL.23_cd"  
"5" "IL.4_cd"   
"6" "IL.5_cd"   
"7" "IL.9_cd"

   
R -q -e "library("devtools"); devtools::load_all("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/bmass");
IL.10_cd <- read.table("/mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp2_7cyto.IL.10_cd.No0Dir.txt.gz", header=T)
IL.13_cd <- read.table("/mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp2_7cyto.IL.13_cd.No0Dir.txt.gz", header=T)
IL.15_cd <- read.table("/mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp2_7cyto.IL.15_cd.No0Dir.txt.gz", header=T)
IL.23_cd <- read.table("/mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp2_7cyto.IL.23_cd.No0Dir.txt.gz", header=T)
IL.4_cd <- read.table("/mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp2_7cyto.IL.4_cd.No0Dir.txt.gz", header=T)
IL.5_cd <- read.table("/mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp2_7cyto.IL.5_cd.No0Dir.txt.gz", header=T)
IL.9_cd <- read.table("/mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp2_7cyto.IL.9_cd.No0Dir.txt.gz", header=T)
GWASsnps <- read.table("/mnt/gluster/data/external_private_supp/Stein2016/CDGemma_Grp2_7cyto.AllPheno.No0Dir.txt.gz.5e8.GWASHits.ChrBP.txt", header=F)
names(GWASsnps) <- c("Chr", "BP")
#bmassOutput1 <- bmass(c("HDL", "LDL", "TG", "TC"), GWASsnps=GWASsnps, NminThreshold = 1, bmassSeedValue=150);
#print(bmassOutput1);" > shana1.txt
DataSources = c("IL.10_cd", "IL.13_cd", "IL.15_cd", "IL.23_cd", "IL.4_cd", "IL.5_cd", "IL.9_cd"); NminThreshold = 1; bmassSeedValue=150;
ExpectedColumnNames=c("Chr", "BP", "MAF", "Direction", "pValue", "N"); GWASsnps_AnnotateWindow = 5e5; SNPMarginalUnivariateThreshold = 1e-4; SNPMarginalMultivariateThreshold = 1e-4; SigmaAlphas = c(0.005,0.0075,0.01,0.015,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.15); ProvidedPriors=NULL; UseFlatPriors=TRUE; PruneMarginalSNPs=TRUE; PruneMarginalSNPs_bpWindow=5e5;

#After merging

Z <- cbind(bmassOutput$MergedDataSources$IL.10_cd_ZScore, bmassOutput$MergedDataSources$IL.13_cd_ZScore, bmassOutput$MergedDataSources$IL.15_cd_ZScore, bmassOutput$MergedDataSources$IL.23_cd_ZScore, bmassOutput$MergedDataSources$IL.4_cd_ZScore, bmassOutput$MergedDataSources$IL.5_cd_ZScore, bmassOutput$MergedDataSources$IL.9_cd_ZScore)
dim(bmassOutput$MergedDataSources)
dim(bmassOutput$MergedDataSources[!duplicated(bmassOutput$MergedDataSources[,1]),])
bmassOutput$MergedDataSources <- bmassOutput$MergedDataSources[!duplicated(bmassOutput$MergedDataSources[,1]),]

#All steps up to Finalize

set.seed(99)
MarginalSNPs_PrunedList <- indephits(jitter(bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg, factor=.0001), bmassOutput$MarginalSNPs$SNPs$Chr, bmassOutput$MarginalSNPs$SNPs$BP, T=PruneMarginalSNPs_bpWindow)
bmassOutput$MarginalSNPs$SNPs <- bmassOutput$MarginalSNPs$SNPs[MarginalSNPs_PrunedList==1,]
bmassOutput$MarginalSNPs$logBFs <- bmassOutput$MarginalSNPs$logBFs[,MarginalSNPs_PrunedList==1]

#Posteriors <- posteriorprob(bmassOutput$MarginalSNPs$logBFs, normalize(rep(c(0,ModelPriors[-1]),length(SigmaAlphas))))
Posteriors <- posteriorprob(bmassOutput$MarginalSNPs$logBFs, bmassOutput$ModelPriors)

Posteriors.marginals <- marginal.postprobs(Posteriors, bmassOutput$Models, length(SigmaAlphas))
Posteriors.marginals.DirAssoc <- t(Posteriors.marginals[[2]])
Posteriors.marginals.DirAssoc <- data.frame(cbind(as.character(bmassOutput$MarginalSNPs$SNPs$ChrBP), Posteriors.marginals.DirAssoc))
Posteriors.marginals.DirAssoc[,2] <- as.numeric(as.character(Posteriors.marginals.DirAssoc[,2]))
Posteriors.marginals.DirAssoc[,3] <- as.numeric(as.character(Posteriors.marginals.DirAssoc[,3]))
Posteriors.marginals.DirAssoc[,4] <- as.numeric(as.character(Posteriors.marginals.DirAssoc[,4]))
Posteriors.marginals.DirAssoc[,5] <- as.numeric(as.character(Posteriors.marginals.DirAssoc[,5]))
Posteriors.marginals.DirAssoc[,6] <- as.numeric(as.character(Posteriors.marginals.DirAssoc[,6]))
Posteriors.marginals.DirAssoc[,7] <- as.numeric(as.character(Posteriors.marginals.DirAssoc[,7]))
Posteriors.marginals.DirAssoc[,8] <- as.numeric(as.character(Posteriors.marginals.DirAssoc[,8]))
colnames(Posteriors.marginals.DirAssoc) <- c("snp", "IL.10_cd", "IL.13_cd", "IL.15_cd", "IL.23_cd", "IL.4_cd", "IL.5_cd", "IL.9_cd")
Posteriors.marginals.DirAssoc.Subset <- Posteriors.marginals.DirAssoc[bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg > .4,]
Posteriors.marginals.DirAssoc <- Posteriors.marginals.DirAssoc[order(bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg, decreasing=TRUE),]
Posteriors.marginals.DirAssoc.Subset <- Posteriors.marginals.DirAssoc.Subset[order(bmassOutput$MarginalSNPs$SNPs[bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg > .4,]$logBFWeightedAvg, decreasing=TRUE),]

#Z <- cbind(bmassOutput$MergedDataSources$IL.13_lps_ZScore, bmassOutput$MergedDataSources$IL.15_lps_ZScore, bmassOutput$MergedDataSources$IL.22_lps_ZScore, bmassOutput$MergedDataSources$IL.33_lps_ZScore, bmassOutput$MergedDataSources$IL.4_lps_ZScore, bmassOutput$MergedDataSources$IL.5_lps_ZScore, bmassOutput$MergedDataSources$IL.9_lps_ZScore)
Z.Directions <- apply(Z, c(1,2), function(x) { y <- 0; if (x > 0) { y <- 1; } else if (x < 0) { y <- -1; } else if (x == 0) { y <- 0; } else { y <- NA; }; return(y);})
Z.Directions.marginals <- Z.Directions[as.numeric(row.names(bmassOutput$MarginalSNPs$SNPs)),][order(bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg, decreasing=TRUE),]

Posteriors.marginals.DirAssoc.wDir <- Posteriors.marginals.DirAssoc
Posteriors.marginals.DirAssoc.wDir[,2:8] <- Posteriors.marginals.DirAssoc[,2:8] * Z.Directions.marginals
Posteriors.marginals.DirAssoc.Subset.wDir <- Posteriors.marginals.DirAssoc.Subset
Posteriors.marginals.DirAssoc.Subset.wDir[,2:8] <- Posteriors.marginals.DirAssoc.Subset[,2:8] * Z.Directions.marginals[bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg > .4,][order(bmassOutput$MarginalSNPs$SNPs[bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg > .4,]$logBFWeightedAvg, decreasing=TRUE),]

library(ggplot2)
library(reshape2)

#jpeg("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Stein2016/Vs1/Stein2016.Vs1.CDGemma_Grp2_7cyto.Marginals.HeatPlot.All.vs1.jpeg", width=6000, height=2000, res=300)
#jpeg("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Stein2016/Vs1/Stein2016.Vs1.CDGemma_Grp2_7cyto.Marginals.HeatPlot.Subset.vs1.jpeg", width=6000, height=2000, res=300)
jpeg("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Stein2016/Vs1/Stein2016.Vs1.CDGemma_Grp2_7cyto.Marginals.HeatPlot.Subset.wDir.vs1.jpeg", width=6000, height=2000, res=300)

#ggPlotData <- melt(Posteriors.marginals.DirAssoc[hclust(dist(Posteriors.marginals.DirAssoc[,2:8]))$order,])
#ggPlotData <- melt(Posteriors.marginals.DirAssoc.wDir[hclust(dist(Posteriors.marginals.DirAssoc.wDir[,2:8]))$order,])
#ggPlotData <- melt(Posteriors.marginals.DirAssoc.Subset[hclust(dist(Posteriors.marginals.DirAssoc.Subset[,2:8]))$order,])
ggPlotData <- melt(Posteriors.marginals.DirAssoc.Subset.wDir[hclust(dist(Posteriors.marginals.DirAssoc.Subset[,2:8]))$order,])
#ggPlotData$snp <- factor(ggPlotData$snp, levels=Posteriors.marginals.DirAssoc[,1][hclust(dist(Posteriors.marginals.DirAssoc[,2:8]))$order])
ggPlotData$snp <- factor(ggPlotData$snp, levels=Posteriors.marginals.DirAssoc.Subset[,1][hclust(dist(Posteriors.marginals.DirAssoc.Subset[,2:8]))$order])

#ggplot(ggPlotData, aes(variable, snp, value)) + geom_tile(aes(fill=value), colour="white") + scale_fill_gradient(low = "white", high="steelblue") + coord_flip() + theme(axis.text.x = element_text(angle = 90, hjust = 1))
ggplot(ggPlotData, aes(variable, snp, value)) + geom_tile(aes(fill=value), colour="white") + scale_fill_gradientn(colours=c("darkred", "white", "steelblue")) + coord_flip() + theme(axis.text.x = element_text(angle = 90, hjust = 1))

dev.off()


R output
~~~
> dim(bmassOutput$MergedDataSources)
[1] 4306961      33
> dim(bmassOutput$MergedDataSources[!duplicated(bmassOutput$MergedDataSources[,1]),])
[1] 4110084      33
.
.
.
> bmassOutput$ZScoresCorMatrix
                IL.10_cd_ZScore IL.13_cd_ZScore IL.15_cd_ZScore IL.23_cd_ZScore
IL.10_cd_ZScore      1.00000000     0.440165639     0.017431748     0.068254726
IL.13_cd_ZScore      0.44016564     1.000000000     0.001071523     0.096480984
IL.15_cd_ZScore      0.01743175     0.001071523     1.000000000     0.428313651
IL.23_cd_ZScore      0.06825473     0.096480984     0.428313651     1.000000000
IL.4_cd_ZScore       0.50768315     0.761930189    -0.081988007    -0.007962297
IL.5_cd_ZScore       0.41660541     0.665927303    -0.049924415     0.009390457
IL.9_cd_ZScore       0.41961275     0.499855689     0.093864238     0.173527288
                IL.4_cd_ZScore IL.5_cd_ZScore IL.9_cd_ZScore
IL.10_cd_ZScore    0.507683151    0.416605410     0.41961275
IL.13_cd_ZScore    0.761930189    0.665927303     0.49985569
IL.15_cd_ZScore   -0.081988007   -0.049924415     0.09386424
IL.23_cd_ZScore   -0.007962297    0.009390457     0.17352729
IL.4_cd_ZScore     1.000000000    0.737097347     0.54783958
IL.5_cd_ZScore     0.737097347    1.000000000     0.68484545
IL.9_cd_ZScore     0.547839582    0.684845453     1.00000000
.
.
.
> dim(bmassOutput$MarginalSNPs$SNPs)
[1] 4935   40
> 
> set.seed(99)
> MarginalSNPs_PrunedList <- indephits(jitter(bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg, factor=.0001), bmassOutput$MarginalSNPs$SNPs$Chr, bmassOutput$MarginalSNPs$SNPs$BP, T=PruneMarginalSNPs_bpWindow)
> bmassOutput$MarginalSNPs$SNPs <- bmassOutput$MarginalSNPs$SNPs[MarginalSNPs_PrunedList==1,]
> bmassOutput$MarginalSNPs$logBFs <- bmassOutput$MarginalSNPs$logBFs[,MarginalSNPs_PrunedList==1]
> 
> dim(bmassOutput$MarginalSNPs$SNPs)
[1] 407  40
> dim(bmassOutput$MarginalSNPs$SNPs[bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg > .5,])
[1] 39 40
> dim(bmassOutput$MarginalSNPs$SNPs[bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg > .4,])
[1] 71 40
> dim(bmassOutput$MarginalSNPs$SNPs[bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg > .3,])
[1] 128  40
> quantile(bmassOutput$MarginalSNPs$SNPs$logBFWeightedAvg)
        0%        25%        50%        75%       100% 
0.01568922 0.12746958 0.20873739 0.33201492 1.52434327 

~~~





















###########################
#
#
# Part A -- GWAS hits
#
#
###########################

##top SNP hits
#Copying/pasting/downoading/finding (however) list of top hits from each study

##GlobalLipids2010
##20160311

mkdir /mnt/lustre/home/mturchin20/Data/GlobalLipids2010

#Copy/pasted table 1 from http://www.nature.com/nature/journal/v466/n7307/fig_tab/nature09270_T1.html into /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.txt

cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.txt | grep rs[0-9] | perl -lane 'foreach my $entry1 (@F) {if ($entry1 =~ m/^rs[0-9]/) { print $entry1; }}' > /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.Marker.txt

#There were 102 GWAS hits but only 95 went on for secondary analyses, hence why title includes '95' loci but not 102
~~~
[  mturchin20@bigmem01  ~/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1/Analyses/PathwayWork]$cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.Marker.txt | wc
    102     102    1024
~~~

join <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.Marker.txt | sort) <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.tbl.gz | sort) | awk '{ print $1, "\t", $12, "\t", $13 }' > /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.MarkerChrBP.txt 

#20161007
cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.MarkerChrBP.txt | awk '{ print $2 "\t" $3 }' > /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.txt
cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.txt | grep -v NA > /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.txt
#20170112
#NA SNP rs9411489 manually replaced/given proper information from https://www.snpedia.com/index.php/Rs9411489, 9 133279427 (note `Merged into	rs635634`)
cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.MarkerChrBP.txt <(echo "rs9411489 9 133279427") | grep -v NA > /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.MarkerChrBP.noNA.NArplcd.txt
cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.txt <(echo "9 133279427") > /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.txt
#20170113
join <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.Marker.txt | sort) <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.tbl.gz | sort) | awk '{ print $1, "\t", $12, "\t", $13, "\t", $14 }' > /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.MarkerChrBPMAF.txt 
#rs9411489 has been merged into rs635634, and the CEU MAF for this SNP (from: https://www.ncbi.nlm.nih.gov/projects/SNP/snp_ref.cgi?rs=635634) is 0.18103448
cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.MarkerChrBPMAF.txt | grep -v NA | cat - <(echo "rs9411489 9 133279427 0.18103448") > /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.MarkerChrBPMAF.noNA.NArplcd.txt


##GlobalLipids2013
##20160311

#From above in file
#~~~
###NOTE_A (see below)
##Supplementary file ng.2797-S1.pdf was downloaded from http://www.nature.com/ng/journal/v45/n11/extref/ng.2797-S1.pdf and the .pdf was saved as a .txt file using Adobe Acrobat
##In excel, all tables except for supplementary tables 2 and 3 were removed manually, producing file ng.2797-S1.edited.vs2.txt
##Supplementary table 2 contains 62 novel loci and supplementary table 3 contains 157 previously identified loci
#
#cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.txt | perl -lane 'foreach my $val1 (@F) { print $val1 ; } ' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.txt
#
#cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.txt | grep ^rs > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.txt
#
#cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.txt | awk '{ print $3 }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.rsIDs.txt
#
#cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.txt | grep -v -f /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.rsIDs.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.missingSNPs.txt
#
##/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.txt was run through HapMart to get the chromosome and position locations for each rsID. 2 SNPs were missing and their information was manually included. These 2 SNPs were rs1047891 and rs9411489. Information was retrieved from UCSC Genome Browser hg18 and included in the /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.txt file
##hg18 because HapMart only had hg18 but switching to hg19 with the final annotated file
##rs1047891 chr2 211248752
##rs9411489 chr 
##NOTE -- Previously reported ABO SNP rs9411489 has merged into rs635634 (source: supplementary of http://hmg.oxfordjournals.org/content/21/6/1444.full or https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=2&cad=rja&uact=8&ved=0CCsQFjAB&url=http%3A%2F%2Fhmg.oxfordjournals.org%2Fcontent%2Fsuppl%2F2011%2F12%2F07%2Fddr581.DC1%2Fddr581supp_table1.doc&ei=HxKiU8_OIYOOyAS-tICoCg&usg=AFQjCNHO1Honaz_0R7CcVLsZan3LNJulKQ&sig2=TIdN-hlZUyRQZeMxiDGWHg&bvm=bv.69137298,d.aWw)
##rs635634 chr9 135144821
#~~~

join -1 1 -2 3 <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.txt | sort ) <(zcat /data/external_public/GlobalLipid/jointGwasMc_HDL.txt.gz | sort -k 3,3) | awk '{ print $1,"\t", $3 }' | sed 's/\:/ /g' | sed 's/chr//g' | perl -lane 'print join("\t", @F);' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.MarkerChrBP.txt

#From http://www.nature.com/ng/journal/v45/n11/full/ng.2797.html -- so expecting total of 157 loci
#~~~
#...To identify new loci and refine known loci influencing these lipids, we examined 188,577 individuals using genome-wide and custom genotyping arrays. We identify and annotate 157 loci associated with lipid levels at P < 5 ÃÂ 10Ã¢ÂÂ8, including 62 loci not previously associated with lipid levels in humans...
#~~~

#20160311 20170119 CHECK_1 -- Prob: Missing 3 SNPs/rsIDs from using any of the 3 '/data/external_public/GlobalLipid/jointGwasMc_*.txt.gz' GWAS files? Soln: At this point don't see it really as a 'problem' in-so-much that as long as the entry list of SNPs is the correct one (has proper coordinates), then will have to make do with what the 'merged' dataset looks like. As long as the full list of coordinates are there, then there won't be any regions that accidentally get called as 'new', though the average BF values may be a little 'off' since cannot analyze the full set of previous hits 
~~~
[  mturchin20@bigmem01  ~/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1/Analyses/PathwayWork]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.MarkerChrBP.txt | wc
    154     462    3351
~~~



#GIANT2010
#20170119

/mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt

#Number of SNPs expecting from each of the following publications
#Lang Allen2010 -- 180 -- "...To identify loci robustly associated with adult height, we took forward at least one SNP (Supplementary Methods) from each of the 207 loci reaching P < 5 × 10−6 into an additional 50,074 samples (stage 2) that became available after completion of our initial meta-analysis. In the joint analysis of our stage 1 and stage 2 studies, SNPs representing 180 loci reached genome-wide significance (P < 5 × 10−8; Supplementary Figs 1and 2 and Supplementary Table 1). ..."
#Speliotes2010 -- 32 -- "...We confirmed 14 known obesity susceptibility loci and identified 18 new loci associated with body mass index (P < 5 × 10−8), ..."
#Heid2010 -- 14 -- "..We identified 13 new loci in or near RSPO3, VEGFA, TBX15-WARS2, NFE2L3, GRB14, DNM3-PIGC, ITPR2-SSPN, LY86, HOXC13, ADAMTS9, ZNRF3-KREMEN1, NISCH-STAB1 and CPEB4 (P = 1.9 × 10−9 to P = 1.8 × 10−40) and the known signal at LYPLAL1..."

#In supplement of Lango Allen see reference to 'NCBI build 36.1' which is the corollary to hg18, also see reference to hg18 in a follow-up analysis in Heid2010
#Also see plenty of references to HapMap Phase 2, which is built on NCBI build 36.1 as well

#Copying/pasting a bunch from earlier in this file
```
#2010 
#Copying/pasting/downoading/finding (however) list of top hits from each study

mkdir /mnt/lustre/home/mturchin20/Data/GIANT/2010/ 

#Height -- 180 SNPs
#Copy/pasted supplementary table 1 from http://www.nature.com.proxy.uchicago.edu/nature/journal/v467/n7317/full/nature09410.html (http://www.nature.com.proxy.uchicago.edu/nature/journal/v467/n7317/extref/nature09410-s1.pdf) into /mnt/lustre/home/mturchin20/Data/GIANT/2010/LangoAllen2010.SupplTable1.txt

#BMI -- 32 SNPs 
#Copy/pasted table 1 from http://www.nature.com.proxy.uchicago.edu/ng/journal/v42/n11/full/ng.686.html into /mnt/lustre/home/mturchin20/Data/GIANT/2010/Speliotes2010.Table1.txt

#WHRadjBMI -- 14 SNPs
#Copy/pasted table 1 from http://www.nature.com.proxy.uchicago.edu/ng/journal/v42/n11/full/ng.685.html into /mnt/lustre/home/mturchin20/Data/GIANT/2010/Heid2010.Table1.txt
#Manually removed the below three lines from table1
#Further SNPs evaluated in follow up but not achieving genome-wide significance in the combined analysis
#rs2076529       6       32,471,933      BTNL2   C       0.430   2.22 Ã 10â8     0.041   34,532  0.012   0.011   92,778  3.71 Ã 10â7     0.020
#rs7081678       10      32,030,629      ZEB1    A       0.085   5.76 Ã 10â7     0.045   76,270  0.094   0.013   100,527 5.57 Ã 10â6     0.027
.
.
.
cat /mnt/lustre/home/mturchin20/Data/GIANT/2010/LangoAllen2010.SupplTable1.txt | grep rs | awk '{ print $1, "\t", $2, "\t", $3 }' > /mnt/lustre/home/mturchin20/Data/GIANT/2010/LangoAllen2010.SupplTable1.rsIDs.MarkerChrBP.txt

cat /mnt/lustre/home/mturchin20/Data/GIANT/2010/Speliotes2010.Table1.txt | grep rs | perl -ane 'print $F[0], "\t"; for (my $i = 0; $i <= $#F; $i++) { if ($F[$i] =~ m/\d,\d/) { print $F[$i-1], "\t", $F[$i], "\t"; } } print "\n";' | awk '{ print $1, "\t", $2, "\t", $3 }' | sed 's/,//g' > /mnt/lustre/home/mturchin20/Data/GIANT/2010/Speliotes2010.Table1.rsIDs.MarkerChrBP.txt

cat /mnt/lustre/home/mturchin20/Data/GIANT/2010/Heid2010.Table1.txt | grep rs | awk '{ print $1, "\t", $2, "\t", $3 }' | sed 's/,//g' > /mnt/lustre/home/mturchin20/Data/GIANT/2010/Heid2010.Table1.rsIDs.MarkerChrBP.txt

cat /mnt/lustre/home/mturchin20/Data/GIANT/2010/LangoAllen2010.SupplTable1.rsIDs.MarkerChrBP.txt /mnt/lustre/home/mturchin20/Data/GIANT/2010/Speliotes2010.Table1.rsIDs.MarkerChrBP.txt /mnt/lustre/home/mturchin20/Data/GIANT/2010/Heid2010.Table1.rsIDs.MarkerChrBP.txt > /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt

~
[  mturchin20@spudhead  ~]$cat /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt | wc
226     678    5794
[  mturchin20@spudhead  ~]$cat /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt | sort | uniq -c | wc
226     904    7602     
[  mturchin20@spudhead  ~]$cat /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt | awk '{ print $1 }' | sort | uniq -c | wc  
226     452    4074     
```

#20170120
#For whatever reason, the coordinates from the paper are 'off-by-1' compared to the coordinates in the datasets, so just going to alter the GWAS hits by 1 to account for this

cat /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt | awk '{ print $1 "\t" $2 "\t" $3-1 }' > /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.txt

~~~
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt | head -n 10
rs425277     1   2059032
rs2284746    1   17179262
rs1738475    1   23409478
rs4601530    1   24916698
rs7532866    1   26614131
rs2154319    1   41518357
rs17391694   1   78396214
rs6699417    1   88896031
rs10874746   1   93096559
rs9428104    1   118657110
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT_AllPheno_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.SignCrrct.vs1.RAF.txt.gz | grep rs4601530
rs4601530   c   t  1_24916697 NA        1       NA      NA NA   133726  NA      NA      NA      123769  NA NA  NA     77080
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT_AllPheno_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.SignCrrct.vs1.RAF.txt.gz | grep rs6699417
rs6699417   t   c  1_88896030 NA        1       NA      NA NA   133711  NA      NA      NA      123780  NA NA  NA     77126
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt | tail -n 10
rs10195252   2   165221337
rs4846567    1   217817340
rs1011731    1   170613171
rs718314     12     26344550
rs1294421    6   6688148
rs1443512    12     52628951
rs6795735    3   64680405
rs4823006    22     27781671
rs6784615    3   52481466
rs6861681    5   173295064
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT_AllPheno_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.SignCrrct.vs1.RAF.txt.gz | grep rs718314
rs7183146   g   t  15_55549035  0.4167  0       0.4167  NA NA   125943  0.4167  NA      NA      119539  0.5833 NA     NA        72854
rs718314    a   g  12_26344549  0.8167  1       0.8167  NA NA   133799  NA      NA      NA      NA  NA  NA NA  NA
rs7183143   t   c  15_59687258  0.9583  2       0.9583  NA NA   133280  0.9583  NA      NA      123117  0.9583 NA     NA        77170
rs7183141   g   a  15_86191817  0.425   0       0.425   NA NA   133788  0.425   NA      NA      123858  0.425  NA     NA        77162
rs7183145   a   g  15_67589253  0.05    2       0.05    NA NA   133770  0.05    NA      NA      123788  0.05   NA     NA        77169
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT_AllPheno_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.Annot.vs1.SignCrrct.vs1.RAF.txt.gz | grep rs4823006
rs4823006   a   g  22_27781670  0.5333  1       0.5333  NA NA   133650  0.5333  NA      NA      123766  0.5333 NA     NA        77086
~~~




#GIANT2014/5
#20170119

#Number of SNPs expecting from each of the following publications
#Wood 2014 -- 697 -- "... We identified 697 SNPs that reached genome-wide significance (P < 5 × 10−8) using an approximate conditional and joint multiple-SNP (COJO) analysis7 in GCTA8 (Online Methods), which takes linkage disequilibrium (LD) between SNPs into account (Supplementary Figs. 2 and 3, and Supplementary Table 1). The 697 SNPs clustered in 423 loci,..." 
#Locke 2015 -- 97 -- "...This analysis identifies 97 BMI-associated loci (P < 5 × 10−8), 56 of which are novel...."
#Shungin 2015 -- 49 -- "... We identify 49 loci (33 new) associated with waist-to-hip ratio adjusted for body mass index (BMI), and an additional 19 loci newly associated with related waist and hip circumference measures (P < 5 × 10−8). In total, 20 of the 49 waist-to-hip ratio adjusted for BMI loci show significant sexual dimorphism, 19 of which display a stronger effect in women...." 

#Wood 2014 has references to 1000G Phase 1 in supplement, and from 1000G websites (eg http://www.internationalgenome.org/faq/which-reference-assembly-do-you-use/) "...For the phase 1 and phase 3 analysis we mapped to GRCh37..."
 #Supplement also has this statement "To facilitate metaanalysis, each group performed genotype imputation using BIMBAM2, MACH3,4, IMPUTE5 BEAGLE6 or PLINK7 and genotypes from the Phase II CEU HapMap"
#In Locke2015 we see references to GRCh38 "...HapMap phase II CEU genotype data37 was used to compute LD and genomic coordinates were defined by genome build GRCh38....", but also "...all contributing GWAS common SNPs were imputed using the HapMap phase II CEU reference panel for European-descent studies37, and CEU+YRI+CHB+JPT HapMap release 22 for the African-American and Hispanic GWAS...."
 #Also see "...Functional variants All variants within 500 kb (HapMap release 22/1000 Genomes CEU) and in LD (r2 > 0.7) with an index SNP were annotated for functional effects based on RefSeq transcripts using Annova..."
#Shungin 2015 has direct references to NCBI Build 36 "...The GWAS scaffold in each cohort was imputed up to CEU haplotypes from HapMap resulting in ~2.5 million SNPs. Each directly typed and imputed SNP passing quality control was tested for association with each trait under an additive model in a linear regression framework (Supplementary Table 3). SNP positions are reported based on NCBI Build 36...."

#GWASsnps <- read.table("/mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt", header=F);
#/mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.w2010Annot.txt
#Should use now /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.txt

ls -lrt /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/. | awk '{ print $9 }' | grep GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP | perl -lane 'my $val1 = "/mnt/lustre/home/mturchin20/Data/GIANT/2014_5/" . $F[0]; my $val2 = "/mnt/lustre/home/mturchin20/Data/GIANT/2014_5/OLD1.WRONG1." . $F[0]; system("mv $val1 $val2");'
mv /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.AllPheno.MarkerChrBP.txt /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/OLD1.WRONG1.GIANT2014_5.AllGWASHits.AllPheno.MarkerChrBP.txt
mv /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/nature14132-s2.ST4.noCarriageReturn.WHRadjBMI.rsIDs.MarkerChrBP.txt /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/OLD1.WRONG1.nature14132-s2.ST4.noCarriageReturn.WHRadjBMI.rsIDs.MarkerChrBP.txt

cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/nature14132-s2.ST4.noCarriageReturn.txt | grep WHRadjBMI | grep -v 2nd | grep rs | awk '{ print $1, "\t", $3, "\t", $4 }' | sed 's/"//g' | sed 's/,//g' | grep -v This > /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/nature14132-s2.ST4.noCarriageReturn.WHRadjBMI.no2nds.rsIDs.MarkerChrBP.txt 
cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/ng.3097-S2.noCarriageReturn.noQuotes.noCommas.rsIDs.MarkerChrBP.txt /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/Locke2015.SupplTables1_2.wManualOCREdits.rsIDs.MarkerChrBP.txt /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/nature14132-s2.ST4.noCarriageReturn.WHRadjBMI.no2nds.rsIDs.MarkerChrBP.txt > /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.txt

#20170120
#'Off-by-1' thing occured here too, which I already knew about and was 'correcting' within R-code setup for the analysis. Just going to create a 'correct' file version here like what occured for GIANT2010

cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.txt | awk '{ print $1 "\t" $2 "\t" $3-1 }' > /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.txt

#For BMI/Locke2015, Table 2 also has 3 'previously found loci', so the total amount of '56' actually matches what the paper says. The leftover amount (97-56) were the 'previously found loci' that are mostly represented in 'extended figure 2', which I retried by using http://www.free-ocr.com/
#For WHRadjBMI/Shungin2015, just going to use 'all' results and not partitioned by ancestry since I'd have to breakdown the Locke results similarly, and their results seem more heterogenous in 'what ancestry-analysis the results came from'. Would also have to explain and justify the differing # of loci being declared as 'hits' from the paper if it differs from the paper publicized (eg what was in the abstract, intro, etc), so easier just to go with their numbers (also may be a technicality to say 'if you had only used european ancestry your true hits would have been this lesser #' and then act like we 'found new regions' when in fact we only did so if you took away a portion of their results)
#For WHRadjBMI/Shungin2015, was including 'WHRadjBMI.2nd' results as well, so total SNPs being included at the time were 68 and not the reported total of 49. Once again, including 'all SNPs the pertain to the reported total', so doing the total of 49 which also includes one 'WHRadjBMI.ALL' result, whereas there are only 48 hits for solely 'WHRadjBMI'
~~~
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/ng.3097-S2.noCarriageReturn.noQuotes.noCommas.rsIDs.MarkerChrBP.txt | wc
    697    2091   17935
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/Locke2015.SupplTables1_2.txt | head -n 75 | grep rs | awk '{ print $1 }' | sort | uniq | wc
     59      59     604
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/nature14132-s2.ST4.noCarriageReturn.txt | grep WHRadjBMI | grep rs | awk '{ print $1, "\t", $3, "\t", $4 }' | sed 's/"//g' | sed 's/,//g' | grep -v This | wc
     68     204    1760
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/nature14132-s2.ST4.noCarriageReturn.txt | grep WHRadjBMI | grep -v 2nd | grep rs | awk '{ print $1, "\t", $3, "\t", $4 }' | sed 's/"//g' | sed 's/,//g' | grep -v This | wc
     49     147    1265
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/nature14132-s2.ST4.noCarriageReturn.txt | grep WHRadjBMI | grep -v WHRadjBMI\\. | grep rs | awk '{ print $1, "\t", $3, "\t", $4 }' | sed 's/"//g' | sed 's/,//g' | grep -v This | wc
     48     144    1240
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/nature14132-s2.ST4.noCarriageReturn.WHRadjBMI.no2nds.rsIDs.MarkerChrBP.txt | wc
     49     147    1265
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/OLD1.WRONG1.GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt | wc
    862    2586   21825
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.txt | wc
    843    2529   21330
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$R -q -e "697 + 97 + 49"
> 697 + 97 + 49
[1] 843
> 
> 
~~~

#Copying/pasting sections from earlier/elsewhere in this file
```
#2014_5
#Copying/pasting/downoading/finding (however) list of top hits from each study

mkdir /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/ 

#Height -- 697 SNPs
#Downloaded Supplementary Table 1 from http://www.nature.com/ng/journal/v46/n11/full/ng.3097.html#supplementary-information and saved it as a .txt manually
scp -p mturchin20@wolfy.uchicago.edu:/Users/mturchin20/LabStuff/Data/GIANT/2014_5/ng.3097-S2.txt /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/. 
cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/ng.3097-S2.txt | sed 's/\r/\n/g' | perl -lane 'print join("\t", @F);' > /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/ng.3097-S2.noCarriageReturn.txt
cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/ng.3097-S2.noCarriageReturn.txt | sed 's/"//g' | sed 's/,//g' > /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/ng.3097-S2.noCarriageReturn.noQuotes.noCommas.txt

#BMI -- 97 SNPs
#Copy/pasted tables 1 & 2 from http://www.nature.com/nature/journal/v518/n7538/full/nature14177.html into /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/Locke2015.SupplTables1_2.txt
#As well as 'Extended Data Table 2' where I downloaded the jpg nature14177-st2.jpg and used http://www.free-ocr.com/ to extract the text from it
#Note -- for 'Extended Data Table 2' double-checked a few of the entries and corrected positions/rsIDs; feel this is still a better starting point than copying everything manually
cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/Locke2015.SupplTables1_2.txt > /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/Locke2015.SupplTables1_2.wManualOCREdits.txt

~~~
[  mturchin20@spudhead  ~/Data/GIANT/2014_5]$cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/Locke2015.SupplTables1_2.txt | grep rs | wc
     97     979    7660
[  mturchin20@spudhead  ~/Data/GIANT/2014_5]$cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/Locke2015.SupplTables1_2.txt | grep rs | awk '{ print $1 }' | sort | uniq -c | wc
     97     194    1774
~~~

#WHRadjBMI -- 49 SNPs (48 SNPs just with EUR; 1 SNP with 'ALL ancestries')
#Downlaoded Supplemetary Table 4 from http://www.nature.com/nature/journal/v518/n7538/full/nature14132.html#supplementary-information and saved it as a .txt manually
scp -p mturchin20@wolfy.uchicago.edu:/Users/mturchin20/LabStuff/Data/GIANT/2014_5/nature14132-s2.ST4.txt /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/.
cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/nature14132-s2.ST4.txt | sed 's/\r/\n/g' > /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/nature14132-s2.ST4.noCarriageReturn.txt
cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/nature14132-s2.ST4.noCarriageReturn.txt | grep WHRadjBMI | grep -vE '2nd|ALL' > /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/nature14132-s2.ST4.noCarriageReturn.WHRadjBMI.Eur.txt
cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/nature14132-s2.ST4.noCarriageReturn.txt | grep WHRadjBMI | grep -vE '2nd' > /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/nature14132-s2.ST4.noCarriageReturn.WHRadjBMI.ALL.txt
#CHECK_1: FIgure out the correct combinations for including the other waist traits -- do the '2nd' hits count if they are just conditional analyses? Solution -- just using all SNPs regardless of GWAS only, GWAS+metabochip, all vs Euro ancestries, all sex vs. male/female only, or joint/conditional. Not sure if can parse out all these factors for every trait (e.g. height results from GATC COJO directly and not sure if I can get 'non-conditional' hits so specifically like I can for WHRadjBMI or BMI)

~~~
[  mturchin20@spudhead  ~/Data/GIANT/2014_5]$cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/nature14132-s2.ST4.noCarriageReturn.WHRadjBMI.Eur.txt | grep rs | awk '{ print $1 }' | sort | uniq -c | wc
     48      96     871
~~~
.
.
.
cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/ng.3097-S2.noCarriageReturn.noQuotes.noCommas.txt | grep rs | awk '{ print $1, "\t", $2, "\t", $3 }' > /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/ng.3097-S2.noCarriageReturn.noQuotes.noCommas.rsIDs.MarkerChrBP.txt

cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/Locke2015.SupplTables1_2.wManualOCREdits.txt | grep rs | awk '{ print $1, "\t", $2 }' | sed 's/,//g' | perl -lane 'print $F[0], "\t", ((split(/:/, $F[1]))[0]), "\t", ((split(/:/, $F[1]))[1]);' > /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/Locke2015.SupplTables1_2.wManualOCREdits.rsIDs.MarkerChrBP.txt

~~~
[  mturchin20@spudhead  ~]$cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/Locke2015.SupplTables1_2.wManualOCREdits.txt| grep rs | awk '{ print $1, "\t", $2 }' | sed 's/,//g' | perl -lane 'print $F[0], "\t", ((split(/:/, $F[1]))[0]), "\t", ((split(/:/, $F[1]))[1]);' | wc                                                97     291    2130
~~~

#cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/nature14132-s2.ST4.noCarriageReturn.WHRadjBMI.Eur.txt | grep rs | awk '{ print $1, "\t", $3, "\t", $4 }' | sed 's/"//g' | sed 's/,//g' | grep -v This > /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/nature14132-s2.ST4.noCarriageReturn.WHRadjBMI.Eur.rsIDs.MarkerChrBP.txt
cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/nature14132-s2.ST4.noCarriageReturn.txt | grep WHRadjBMI | grep rs | awk '{ print $1, "\t", $3, "\t", $4 }' | sed 's/"//g' | sed 's/,//g' | grep -v This > /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/nature14132-s2.ST4.noCarriageReturn.WHRadjBMI.rsIDs.MarkerChrBP.txt
cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/nature14132-s2.ST4.noCarriageReturn.txt | grep rs | awk '{ print $1, "\t", $3, "\t", $4 }' | sed 's/"//g' | sed 's/,//g' | grep -v This > /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/nature14132-s2.ST4.noCarriageReturn.rsIDs.MarkerChrBP.txt

~~~
[  mturchin20@spudhead  ~/Data/GIANT/2014_5]$cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/nature14132-s2.ST4.noCarriageReturn.WHRadjBMI.rsIDs.MarkerChrBP.txt | wc
    68     204    1760
[  mturchin20@spudhead  ~/Data/GIANT/2014_5]$cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/nature14132-s2.ST4.noCarriageReturn.rsIDs.MarkerChrBP.txt | wc
    87     261    2245
~~~

cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/ng.3097-S2.noCarriageReturn.noQuotes.noCommas.rsIDs.MarkerChrBP.txt /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/Locke2015.SupplTables1_2.wManualOCREdits.rsIDs.MarkerChrBP.txt /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/nature14132-s2.ST4.noCarriageReturn.WHRadjBMI.rsIDs.MarkerChrBP.txt > /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt

cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/ng.3097-S2.noCarriageReturn.noQuotes.noCommas.rsIDs.MarkerChrBP.txt /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/Locke2015.SupplTables1_2.wManualOCREdits.rsIDs.MarkerChrBP.txt /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/nature14132-s2.ST4.noCarriageReturn.rsIDs.MarkerChrBP.txt > /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.AllPheno.MarkerChrBP.txt
.
.
.
##20151006 Note -- Checking to see if new hits from 2010 run are within 1Mb of the 2014_5 hits

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT2010.newtophits.vs1.txt

#python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT2010.newtophits.vs1.txt 
#cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt | perl -lane 'for (my $i = 1; $i <= 500000; $i++) { print $F[0], "\t", $F[1], "\t", $F[2] + $i; print $F[0], "\t", $F[1], "\t", $F[2] - $i; }' > /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.w1MbBorder.txt

tail -n +2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT2010.newtophits.vs1.txt | python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 - --file2 /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt > /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.w2010Annot.txt

~~~
[  mturchin20@spudling26  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1]$cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.w2010Annot.txt | awk '{ print $4 }' | sort | uniq -c
838 0
6 1
18 2
[  mturchin20@spudling26  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT2010.newtophits.vs1.txt | wc
20     160     861
[  mturchin20@spudling26  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1]$cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.w2010Annot.txt | awk '{ print $4, "\t", $5 }' | sort | uniq -c
838 0
6 1
1 2   rs10040888
1 2   rs11835818
1 2   rs12204421
2 2   rs12534698
3 2   rs1809889
1 2   rs2025151
1 2   rs2390312
2 2   rs389883
1 2   rs4735692
2 2   rs648831
1 2   rs6824258
1 2   rs7601531
1 2   rs7614120
[  mturchin20@spudling26  ~/Lab_Stuff/StephensLab/Multivariate/GIANT2010/V

~~~
```







#PGC 2013
#20150601

cd /mnt/lustre/home/mturchin20/Software/
wget ftp://ftp.foolabs.com/pub/xpdf/xpdfbin-linux-3.04.tar.gz
tar -xvzf /mnt/lustre/home/mturchin20/Software/xpdfbin-linux-3.04.tar.gz

mkdir /mnt/lustre/home/mturchin20/Data/PGC/
mkdir /mnt/lustre/home/mturchin20/Data/PGC/2013/

#ADD -- no significant loci found (that passed GWS)

#Autism -- unavailable yet (no paper?)

#Bipolar -- (2 SNPs that had primary signals and kept them during replication; more SNPs reached GWS after replication but for now don't want to include SNPs based on 'replication' results. Dropping the 2 other primary signals since we at least know they were potential FPs?)

#Copy/pasted table 2 & 3 from http://www.nature.com.proxy.uchicago.edu/ng/journal/v43/n10/fig_tab/ng.943_T3.html into /mnt/lustre/home/mturchin20/Data/PGC/2013/BP2011.Table2.txt & /mnt/lustre/home/mturchin20/Data/PGC/2013/BP2011.Table3.txt
#Manually know that rs10994397 and rs9371601 are the two SNPs that meet the paper's criteria -- "Only two of the four SNPs listed in Table 2 had replication P < 0.05; the genome-wide significant SNPs rs10994397 and rs9371601 did not have P < 0.05 (replication P = 0.12 and P = 0.10, respectively)."

cat /mnt/lustre/home/mturchin20/Data/PGC/2013/BP2011.Table2.txt | grep -E 'rs10994397|rs9371601' | awk '{ print $1, "\t", $2, "\t", $3 }' | sed 's/,//g' > /mnt/lustre/home/mturchin20/Data/PGC/2013/BP2011.Table2.GWSignificant.noCommas.MarkerChrBP.txt

#Cross-study -- 4 SNPs/loci
#Copy/pasted table 1 from http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3714010/table/T1/ (http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3714010/) into /mnt/lustre/home/mturchin20/Data/PGC/2013/Cross2014.Table1.txt

cat /mnt/lustre/home/mturchin20/Data/PGC/2013/Cross2014.Table1.txt | grep rs | awk '{ print $1, "\t", $2, "\t", $3 }' > /mnt/lustre/home/mturchin20/Data/PGC/2013/Cross2014.Table1.MarkerChrBP.txt

#MDD (CrossMDDBipolar) -- 15 SNPs
#Copy/pasted SNPs with pvals "...In the MDD-bipolar cross-disorder analysis, 15 SNPs exceeded genome-wide significance (P<5 Ã 10(-8)), and all were in a 248âkb interval of high LD on 3p21.1 (chr3:52â425â083-53â822â102, minimum P=5.9 Ã 10(-9) at rs2535629)..." in Supplemental Table 19 from http://www.nature.com.proxy.uchicago.edu/mp/journal/v18/n4/full/mp201221a.html (http://www.nature.com.proxy.uchicago.edu/mp/journal/v18/n4/suppinfo/mp201221s1.html) into /mnt/lustre/home/mturchin20/Data/PGC/2013/MDD2013.SupplTable19.txt

cat /mnt/lustre/home/mturchin20/Data/PGC/2013/MDD2013.SupplTable19.txt | awk '{ print $1, "\t", $2, "\t", $3 }' > /mnt/lustre/home/mturchin20/Data/PGC/2013/MDD2013.SupplTable19.MarkerChrBP.txt

zcat /mnt/gluster/data/external_public_supp/IBDConsortium2012/gwas_ichip_meta_release.txt.gz | grep -w -f /mnt/lustre/home/mturchin20/Data/IBDConsortium/2012/Jostins2012.SupplTable2.noCrosses.MarkerNames.txt | perl -lane 'print join("\t", @F[0..2]);' | grep -w -f /mnt/lustre/home/mturchin20/Data/IBDConsortium/2012/Jostins2012.SupplTable2.noCrosses.MarkerNames.txt > /mnt/lustre/home/mturchin20/Data/IBDConsortium/2012/Jostins2012.SupplTable2.noCrosses.MarkerNames.txt.pt1
zcat /mnt/gluster/data/external_public_supp/IBDConsortium2012/gwas_ichip_meta_release.txt.gz | grep -w -f /mnt/lustre/home/mturchin20/Data/IBDConsortium/2012/Jostins2012.SupplTable2.noCrosses.MarkerNames.txt | perl -lane 'print $F[2], "\t", $F[0], "\t", $F[1];' > /mnt/lustre/home/mturchin20/Data/IBDConsortium/2012/Jostins2012.SupplTable2.noCrosses.MarkerChrBP.txt

#I think 2 top SNPs might be missing from the maine data file? I get 161 SNPs based on the GWAS SNP column and when I try to identify the '2 SNPs missing that should be in the ICHIP column' they don't exist

~~~
[  mturchin20@spudhead  /mnt/gluster/data/external_public_supp/PGC2013]$zcat /mnt/gluster/data/external_public_supp/IBDConsortium2012/gwas_ichip_meta_release.txt.gz | grep -w -f /mnt/lustre/home/mturchin20/Data/IBDConsortium/2012/Jostins2012.SupplTable2.noCrosses.MarkerNames.txt | perl -lane 'print join("\t", @F[0..2]);' | wc
    225     675    4900
[  mturchin20@spudhead  /mnt/gluster/data/external_public_supp/PGC2013]$zcat /mnt/gluster/data/external_public_supp/IBDConsortium2012/gwas_ichip_meta_release.txt.gz | grep -w -f /mnt/lustre/home/mturchin20/Data/IBDConsortium/2012/Jostins2012.SupplTable2.noCrosses.MarkerNames.txt | perl -lane 'print join("\t", @F[0..2]);' | grep -w -f /mnt/lustre/home/mturchin20/Data/IBDConsortium/2012/Jostins2012.SupplTable2.noCrosses.MarkerNames.txt | wc
    161     483    3508
[  mturchin20@spudhead  /mnt/gluster/data/external_public_supp/PGC2013]$zcat /mnt/gluster/data/external_public_supp/IBDConsortium2012/gwas_ichip_meta_release.txt.gz | grep -w -f /mnt/lustre/home/mturchin20/Data/IBDConsortium/2012/Jostins2012.SupplTable2.noCrosses.MarkerNames.txt | grep -w -v -f <(cat /mnt/lustre/home/mturchin20/Data/IBDConsortium/2012/Jostins2012.SupplTable2.noCrosses.MarkerNames.txt.pt1 | awk '{ print $3 }') | wc
      0       0       0
[  mturchin20@spudhead  /mnt/gluster/data/external_public_supp/PGC2013]$zcat /mnt/gluster/data/external_public_supp/IBDConsortium2012/gwas_ichip_meta_release.txt.gz | grep -w -f /mnt/lustre/home/mturchin20/Data/IBDConsortium/2012/Jostins2012.SupplTable2.noCrosses.MarkerNames.txt | awk '{ print $5 }' | grep -w -v -f <(cat /mnt/lustre/home/mturchin20/Data/IBDConsortium/2012/Jostins2012.SupplTable2.noCrosses.MarkerNames.txt.pt1 | awk '{ print $3 }') | wc
    161     161    1008
[  mturchin20@spudhead  /mnt/gluster/data/external_public_supp/PGC2013]$zcat /mnt/gluster/data/external_public_supp/IBDConsortium2012/gwas_ichip_meta_release.txt.gz | grep -w -f /mnt/lustre/home/mturchin20/Data/IBDConsortium/2012/Jostins2012.SupplTable2.noCrosses.MarkerNames.txt | awk '{ print $5 }' | grep -w -v -f <(cat /mnt/lustre/home/mturchin20/Data/IBDConsortium/2012/Jostins2012.SupplTable2.noCrosses.MarkerNames.txt.pt1 | awk '{ print $3 }') | grep -w -f /mnt/lustre/home/mturchin20/Data/IBDConsortium/2012/Jostins2012.SupplTable2.noCrosses.MarkerNames.txt
[  mturchin20@spudhead  /mnt/gluster/data/external_public_supp/PGC2013]$
[  mturchin20@spudhead  /mnt/gluster/data/external_public_supp/PGC2013]$zcat /mnt/gluster/data/external_public_supp/IBDConsortium2012/gwas_ichip_meta_release.txt.gz | grep -w -f /mnt/lustre/home/mturchin20/Data/IBDConsortium/2012/Jostins2012.SupplTable2.noCrosses.MarkerNames.txt | perl -ane 'print $F[2], "\n", $F[4], "\n";' | grep -w -f /mnt/lustre/home/mturchin20/Data/IBDConsortium/2012/Jostins2012.SupplTable2.noCrosses.MarkerNames.txt | sort | uniq -c | wc
    161     322    2911
~~~

#CHECK_0: For IBD2015 'gwas hits' just explore p-value distribution of previous 163 hits in this new dataset and maybe reuse a strict p-val cutuff to see what list that garners? If same '163' are found, or more on top of them....can just use a new cutoff based on this? Maybe this has already been published on before too? Mark/those folks made it sound like this data should have been made available already which somewhat implies publications have already been out based on them?

mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/IBDConsortium2012
mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/IBDConsortium2012/Vs1

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/TAG2010/Vs1/process.MTedits.ForTAG2010.vs1.R /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/IBDConsortium2012/Vs1/process.MTedits.ForIBDConsortium2012.vs1.R




##ICBP2011 dbSNP 126, GWAS = (29 SBP/DBP w/ combined follow-up, 7 in MAP/PP w/ combined follow-up), N = provided per SNP
##201501003

mkdir /mnt/lustre/home/mturchin20/Data/ICBP2011

#Copy/pasted supplemental table 5 from http://www.nature.com/nature/journal/v478/n7367/full/nature10405.html#supplementary-information into /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.SBPDBP.SupplTable5.txt and table 1, supplemental table 2F from http://www.nature.com/ng/journal/v43/n10/full/ng.922.html into /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.PPMAP.Table1.txt & /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.PPMAP.TableSuppl2F.txt 

#Manually editing the formatting in /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.SBPDBP.SupplTable5.Edited.txt to better ease of use
cat /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.SBPDBP.SupplTable5.txt > /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.SBPDBP.SupplTable5.Edited.txt
cat /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.SBPDBP.SupplTable5.Edited.txt | awk '{ print $2 }' | grep rs > /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.SBPDBP.SupplTable5.Edited.rsIDs.txt
cat /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.SBPDBP.SupplTable5.Edited.txt | awk '{ print $2, "\t", $3, "\t", $4}' | sed 's/,//g' | tail -n +4 > /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.SBPDBP.SupplTable5.Edited.MarkerChrBP.txt

cat /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.PPMAP.Table1.txt > /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.PPMAP.Table1.Edited.txt
cat /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.PPMAP.Table1.Edited.txt | awk '{ print $1 }'  | grep rs > /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.PPMAP.Table1.Edited.rsIDs.txt
cat /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.PPMAP.Table1.Edited.txt | perl -lane 'my $ChrBP1; foreach my $i (@F) { if ($i =~ m/chr\d+:/) { $ChrBP1 = $i; } }; print $F[0], "\t", $ChrBP1;' | sed 's/chr//g' | sed 's/:/ /g' | sed 's/,//g' | grep rs > /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.PPMAP.Table1.Edited.MarkerChrBP.txt
cat /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.PPMAP.TableSuppl2F.txt > /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.PPMAP.TableSuppl2F.Edited.txt
cat /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.PPMAP.TableSuppl2F.Edited.txt | grep rs | sed 's/*//g' > /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.PPMAP.TableSuppl2F.Edited.rsIDs.noStars.txt
cat /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.PPMAP.TableSuppl2F.Edited.txt | grep -E 'rs|chr' | sed 's/:\s*/:/g' | awk '{ print $1 }' | perl -ane' if ($F[0] =~ m/rs/) { print $F[0], "\t"; } else { print $F[0], "\n"; }' | sed 's/*//g' | sed 's/chr//g' | sed 's/:/ /g' > /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.PPMAP.TableSuppl2F.Edited.MarkerChrBP.txt

cat /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.SBPDBP.SupplTable5.Edited.rsIDs.txt /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.PPMAP.Table1.Edited.rsIDs.txt /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.PPMAP.TableSuppl2F.Edited.rsIDs.noStars.txt | sort | uniq -c | sort -k 1,1 | awk '{ print $2 }' > /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.AllPheno.GWASHits.rsIDs.txt
cat /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.SBPDBP.SupplTable5.Edited.MarkerChrBP.txt /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.PPMAP.Table1.Edited.MarkerChrBP.txt /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.PPMAP.TableSuppl2F.Edited.MarkerChrBP.txt | perl -lane 'print join("\t", @F);' | sort | uniq > /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.AllPheno.GWASHits.MarkerChrBP.txt








##iGENBP2015
##20160421

mkdir /mnt/lustre/home/mturchin20/Data/iGENBP2015

#Copy/pasted supplemetnary table 4 supplementary table 6 from http://www.nature.com/ng/journal/v47/n11/full/ng.3405.html into /mnt/lustre/home/mturchin20/Data/iGENBP2015/iGENBP2015.SupplTable4.txt & /mnt/lustre/home/mturchin20/Data/iGENBP2015/iGENBP2015.SupplTable6.txt

#20160421 NOTE -- last two lines in SupplTable6 are not part of the '19' mentioned in the text, so removing them. A total of '12' were 'significant' in the combined analysis and had replication p-value <.05

~~~
[  mturchin20@bigmem01  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Data/iGENBP2015/iGENBP2015.SupplTable6.txt | sed 's/^\s*//g' | awk '{ print $1, "\t", $2, "\t", $3 }' | sed 's/,//g' | grep rs | head -n -2 | wc
     19      57     485
[  mturchin20@bigmem01  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Data/iGENBP2015/iGENBP2015.SupplTable6.txt | perl -lane 'if (($F[$#F] < .00000005) && ($F[$#F-5] < .05)) { print join("\t", @F); }' |  sed 's/^\s*//g' | awk '{ print $1, "\t", $2, "\t", $3 }' | sed 's/,//g' | grep rs | wc
     12      36     306
~~~

cat /mnt/lustre/home/mturchin20/Data/iGENBP2015/iGENBP2015.SupplTable6.txt | sed 's/^\s*//g' | awk '{ print $1, "\t", $2, "\t", $3 }' | sed 's/,//g' | grep rs | head -n -2 > /mnt/lustre/home/mturchin20/Data/iGENBP2015/iGENBP2015.SupplTable6.19Hits.MarkerChrBP.txt
cat /mnt/lustre/home/mturchin20/Data/iGENBP2015/iGENBP2015.SupplTable6.txt | perl -lane 'if (($F[$#F] < .00000005) && ($F[$#F-5] < .05)) { print join("\t", @F); }' |  sed 's/^\s*//g' | awk '{ print $1, "\t", $2, "\t", $3 }' | sed 's/,//g' | grep rs > /mnt/lustre/home/mturchin20/Data/iGENBP2015/iGENBP2015.SupplTable6.12Hits.MarkerChrBP.txt

~~~
[  mturchin20@bigmem01  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Data/iGENBP2015/iGENBP2015.SupplTable4.txt | grep No | wc
     24     897    5624
[  mturchin20@bigmem01  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Data/iGENBP2015/iGENBP2015.SupplTable4.txt | grep Yes | wc
     19     722    4515
~~~

cat /mnt/lustre/home/mturchin20/Data/iGENBP2015/iGENBP2015.SupplTable4.txt | awk '{ print $1, "\t", $2, "\t", $3 }' | sed 's/,//g' | grep rs > /mnt/lustre/home/mturchin20/Data/iGENBP2015/iGENBP2015.SupplTable4.MarkerChrBP.txt
cat /mnt/lustre/home/mturchin20/Data/iGENBP2015/iGENBP2015.SupplTable4.txt | grep No | awk '{ print $1, "\t", $2, "\t", $3 }' | sed 's/,//g' | grep rs > /mnt/lustre/home/mturchin20/Data/iGENBP2015/iGENBP2015.SupplTable4.NotNovel.MarkerChrBP.txt
cat /mnt/lustre/home/mturchin20/Data/iGENBP2015/iGENBP2015.SupplTable4.txt | grep Yes | awk '{ print $1, "\t", $2, "\t", $3 }' | sed 's/,//g' | grep rs > /mnt/lustre/home/mturchin20/Data/iGENBP2015/iGENBP2015.SupplTable4.Novel.MarkerChrBP.txt

#20160421 CHECK_0 -- Prob: Finding that a SNP in 2015 NotNovel is the same rsID from 2011 but not within 50kb of that SNP -- presumably because using a different human genome reference. May need to liftover the 2011 results?
#python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.AllPheno.GWASHits.MarkerChrBP.txt --file2 /mnt/lustre/home/mturchin20/Data/iGENBP2015/iGENBP2015.SupplTable4.NotNovel.MarkerChrBP.txt > /mnt/lustre/home/mturchin20/Data/iGENBP2015/iGENBP2015.SupplTable4.NotNovel.MarkerChrBP.w2011Hits.txt
#python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.AllPheno.GWASHits.MarkerChrBP.txt --file2 /mnt/lustre/home/mturchin20/Data/iGENBP2015/iGENBP2015.SupplTable4.Novel.MarkerChrBP.txt > /mnt/lustre/home/mturchin20/Data/iGENBP2015/iGENBP2015.SupplTable4.Novel.MarkerChrBP.w2011Hits.txt

#python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/iGENBP2015/iGENBP2015.SupplTable4.Novel.MarkerChrBP.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/newtophits.vs2.SignCrrct.vs1.txt

#Old results look to be in Hg18 and new results look to be in Hg19
~~~
[  mturchin20@bigmem01  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Data/iGENBP2015/iGENBP2015.SupplTable4.NotNovel.MarkerChrBP.txt | grep 111884608
rs3184504        12      111884608
[  mturchin20@bigmem01  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.AllPheno.GWASHits.MarkerChrBP.txt | grep rs3184504
rs3184504       12      110368991

##From UCSC GenomeBrowser
#
##Hg19
#NHGRI-EBI Catalog of Published Genome-Wide Association Studies
#
#rs3184504 at chr12:111884358-111884858
#rs3184504 at chr12:111884358-111884858
#.
#.
#.
##Hg18
#NHGRI-EBI Catalog of Published Genome-Wide Association Studies
#
#rs3184504 at chr12:110368741-110369241
#rs3184504 at chr12:110368741-110369241
#.
#.
#.

[  mturchin20@bigmem01  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.AllPheno.GWASHits.MarkerChrBP.txt | awk '{ print "chr" $2, "\t", $3, "\t", $3, "\t", $1 }' | head -n 10
chr12    113872179       113872179       rs10850411
chr10    104836168       104836168       rs11191548
chr10    104929205       104929205       rs11191593
chr11    129778440       129778440       rs11222084
chr5     32825609        32825609        rs1173756
chr5     32850785        32850785        rs1173771
chr5     157777980       157777980       rs11953630
chr10    18767965        18767965        rs12258967
chr17    44757806        44757806        rs12940887
chr2     164623454       164623454       rs13002573
~~~

#cat /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.AllPheno.GWASHits.MarkerChrBP.txt | awk '{ print "chr" $2, "\t", $3, "\t", $3, "\t", $1 }' > /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.AllPheno.GWASHits.MarkerChrBP.forUCSCGB.bed
cat /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.AllPheno.GWASHits.MarkerChrBP.txt | awk '{ print "chr"$2":"$3"-"$3 }' > /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.AllPheno.GWASHits.MarkerChrBP.forUCSCGB.liftoverFormat
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/newtophits.vs2.SignCrrct.vs1.txt | awk '{ print "chr"$2":"$3"-"$3 }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/newtophits.vs2.SignCrrct.vs1.MarkerChrBP.forUCSCGB.liftoverFormat 

#Used liftover utility in UCSC GenomeBrowser, https://genome.ucsc.edu/cgi-bin/hgLiftOver, from Hg18 to Hg19, and saved to /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.AllPheno.GWASHits.MarkerChrBP.forUCSCGB.liftoverFormat.liftoveredToHg19.output & /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/newtophits.vs2.SignCrrct.vs1.MarkerChrBP.forUCSCGB.liftoverFormat.liftoveredToHg19.output 
#49 of 49 were successful
#24 of 24 were successful

paste /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.AllPheno.GWASHits.MarkerChrBP.txt /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.AllPheno.GWASHits.MarkerChrBP.forUCSCGB.liftoverFormat /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.AllPheno.GWASHits.MarkerChrBP.forUCSCGB.liftoverFormat.liftoveredToHg19.output > /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.AllPheno.GWASHits.MarkerChrBP.forUCSCGB.liftoverFormat.liftoveredToHg19.AllFiles
paste <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/newtophits.vs2.SignCrrct.vs1.txt | perl -lane 'print join("\t", @F[0..2]);' | grep -v pos) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/newtophits.vs2.SignCrrct.vs1.MarkerChrBP.forUCSCGB.liftoverFormat | grep -v pos) /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/newtophits.vs2.SignCrrct.vs1.MarkerChrBP.forUCSCGB.liftoverFormat.liftoveredToHg19.output > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/newtophits.vs2.SignCrrct.vs1.MarkerChrBP.forUCSCGB.liftoverFormat.liftoveredToHg19.AllFiles

#Visual inspection a few quick code passes have things make sense

~~~
[  mturchin20@bigmem01  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.AllPheno.GWASHits.MarkerChrBP.forUCSCGB.liftoverFormat.liftoveredToHg19.AllFiles | sed 's/:/ /g' | sed 's/-/ /g' | sed 's/chr//g' | head -n 10
rs10850411      12      113872179       12 113872179 113872179  12 115387796 115387796
rs11191548      10      104836168       10 104836168 104836168  10 104846178 104846178
rs11191593      10      104929205       10 104929205 104929205  10 104939215 104939215
rs11222084      11      129778440       11 129778440 129778440  11 130273230 130273230
rs1173756       5       32825609        5 32825609 32825609     5 32789852 32789852
rs1173771       5       32850785        5 32850785 32850785     5 32815028 32815028
rs11953630      5       157777980       5 157777980 157777980   5 157845402 157845402
rs12258967      10      18767965        10 18767965 18767965    10 18727959 18727959
rs12940887      17      44757806        17 44757806 44757806    17 47402807 47402807
rs13002573      2       164623454       2 164623454 164623454   2 164915208 164915208
[  mturchin20@bigmem01  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.AllPheno.GWASHits.MarkerChrBP.forUCSCGB.liftoverFormat.liftoveredToHg19.AllFiles | sed 's/:/ /g' | sed 's/-/ /g' | sed 's/chr//g' | awk '{ if ($2 != $7) { print $0 } } ' | wc
      0       0       0
[  mturchin20@bigmem01  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.AllPheno.GWASHits.MarkerChrBP.forUCSCGB.liftoverFormat.liftoveredToHg19.AllFiles | sed 's/:/ /g' | sed 's/-/ /g' | sed 's/chr//g' | awk '{ if ($2 == $7) { print $0 } } ' | wc
     49     441    3178
[  mturchin20@bigmem01  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.AllPheno.GWASHits.MarkerChrBP.forUCSCGB.liftoverFormat.liftoveredToHg19.AllFiles | sed 's/:/ /g' | sed 's/-/ /g' | sed 's/chr//g' | awk '{ print $1, "\t", $7, "\t", $8 }' | head -n 10
rs10850411       12      115387796
rs11191548       10      104846178
rs11191593       10      104939215
rs11222084       11      130273230
rs1173756        5       32789852
rs1173771        5       32815028
rs11953630       5       157845402
rs12258967       10      18727959
rs12940887       17      47402807
rs13002573       2       164915208
[  mturchin20@bigmem01  ~]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/newtophits.vs2.SignCrrct.vs1.MarkerChrBP.forUCSCGB.liftoverFormat.liftoveredToHg19.AllFiles | sed 's/:/ /g' | sed 's/-/ /g' | sed 's/chr//g' | head -n 10                                                                                                                                                                                          rs10006384      4       38207753        4 38207753 38207753     4 38531358 38531358                                                                                                                                            rs10473911      5       87296415        5 87296415 87296415     5 87260659 87260659
rs10505863      12      20228698        12 20228698 20228698    12 20337431 20337431
rs10865196      2       44101281        2 44101281 44101281     2 44247777 44247777
rs11066320      12      111369135       12 111369135 111369135  12 112884752 112884752
rs11694671      2       190624076       2 190624076 190624076   2 190915831 190915831
rs13035036      2       43102039        2 43102039 43102039     2 43248535 43248535
rs13306677      16      55483696        16 55483696 55483696    16 56926195 56926195
rs17089717      13      72208616        13 72208616 72208616    13 73310615 73310615
rs17677264      8       9449460 8 9449460 9449460       8 9412050 9412050
[  mturchin20@bigmem01  ~]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/newtophits.vs2.SignCrrct.vs1.MarkerChrBP.forUCSCGB.liftoverFormat.liftoveredToHg19.AllFiles | sed 's/:/ /g' | sed 's/-/ /g' | sed 's/chr//g' | sed 's/chr//g' | awk '{ if ($2 != $7) { print $0 } } ' | wc
      0       0       0
[  mturchin20@bigmem01  ~]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/newtophits.vs2.SignCrrct.vs1.MarkerChrBP.forUCSCGB.liftoverFormat.liftoveredToHg19.AllFiles | sed 's/:/ /g' | sed 's/-/ /g' | sed 's/chr//g' | sed 's/chr//g' | awk '{ if ($2 == $7) { print $0 } } ' | wc
     24     216    1505
[  mturchin20@bigmem01  ~]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/newtophits.vs2.SignCrrct.vs1.MarkerChrBP.forUCSCGB.liftoverFormat.liftoveredToHg19.AllFiles | sed 's/:/ /g' | sed 's/-/ /g' | sed 's/chr//g' | sed 's/chr//g' | awk '{ print $1, "\t", $7, "\t", $8 }' | head -n 10
rs10006384       4       38531358
rs10473911       5       87260659
rs10505863       12      20337431
rs10865196       2       44247777
rs11066320       12      112884752
rs11694671       2       190915831
rs13035036       2       43248535
rs13306677       16      56926195
rs17089717       13      73310615
rs17677264       8       9412050
~~~

cat /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.AllPheno.GWASHits.MarkerChrBP.forUCSCGB.liftoverFormat.liftoveredToHg19.AllFiles | sed 's/:/ /g' | sed 's/-/ /g' | sed 's/chr//g' | awk '{ print $1, "\t", $7, "\t", $8 }' > /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.AllPheno.GWASHits.MarkerChrBP.Hg19.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/newtophits.vs2.SignCrrct.vs1.MarkerChrBP.forUCSCGB.liftoverFormat.liftoveredToHg19.AllFiles | sed 's/:/ /g' | sed 's/-/ /g' | sed 's/chr//g' | sed 's/chr//g' | awk '{ print $1, "\t", $7, "\t", $8 }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/newtophits.vs2.SignCrrct.vs1.MarkerChrBP.Hg19.txt

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.AllPheno.GWASHits.MarkerChrBP.Hg19.txt --file2 /mnt/lustre/home/mturchin20/Data/iGENBP2015/iGENBP2015.SupplTable4.NotNovel.MarkerChrBP.txt > /mnt/lustre/home/mturchin20/Data/iGENBP2015/iGENBP2015.SupplTable4.NotNovel.MarkerChrBP.w2011Hits.Hg19.txt
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.AllPheno.GWASHits.MarkerChrBP.Hg19.txt --file2 /mnt/lustre/home/mturchin20/Data/iGENBP2015/iGENBP2015.SupplTable4.Novel.MarkerChrBP.txt > /mnt/lustre/home/mturchin20/Data/iGENBP2015/iGENBP2015.SupplTable4.Novel.MarkerChrBP.w2011Hits.Hg19.txt

~~~
[  mturchin20@bigmem01  ~/Data/iGENBP2015]$cat iGENBP2015.SupplTable4.NotNovel.MarkerChrBP.w2011Hits.Hg19.txt
rs11105354      12      90026523        2       rs17249754      12      90060586        2
rs13149993      4       81158545        1       rs1458038       4       81164723        1
rs3184504       12      111884608       1       rs653178        12      112007756       1
rs17477177      7       106411858       1       rs17477177      7       106411858       1
rs2055450       11      100550417       2       rs633185        11      100593538       2
rs10894192      11      130266117       2       rs11222084      11      130273230       2
rs8032315       15      91418297        2       rs2521501       15      91437388        2
rs11191375      10      104464657       2       rs3824755       10      104595849       2
rs7733331       5       32828846        2       rs1173771       5       32815028        2
rs1378942       15      75077367        1       rs1378942       15      75077367        1
rs11960210      5       157817634       2       rs9313772       5       157804457       2
rs12567136      1       11883731        2       rs17367504      1       11862778        2
rs11102516      1       113161350       2       rs2932538       1       113216543       2
rs17080102      6       151004770       0
rs1887320       20      10965998        2       rs1327235       20      10969030        2
rs35444 12      115552437       2       rs10850411      12      115387796       2
rs880315        1       10796866        0
rs1347677       11      16296412        0
rs7405452       17      46674670        0
rs7629767       3       42043509        2       rs3774372       3       41877414        2
rs731141        10      95898681        2       rs9663362       10      95895177        2
rs2301597       17      43173273        0
rs2484294       10      115792062       2       rs2782980       10      115781527       2
[  mturchin20@bigmem01  ~]$cat /mnt/lustre/home/mturchin20/Data/iGENBP2015/iGENBP2015.SupplTable4.Novel.MarkerChrBP.w2011Hits.Hg19.txt
rs740406        19      2232221 0
rs12579720      12      20173764        0
rs2240736       17      59485393        0
rs1563788       6       43308363        0
rs2014912       4       86715670        0
rs2107595       7       19049388        0
rs1275988       2       26914364        0
rs10260816      7       46010100        0
rs751984        11      61278246        0
rs1558902       16      53803574        0
rs13359291      5       122476457       0
rs1344653       2       19730845        0
rs7604423       2       43382741        0
rs1474698       6       56064197        0
rs17321989      8       76869253        0
rs10931896      2       201148076       0
rs9687065       5       148391140       0
rs503341        11      63587313        0
rs490234        9       128303734       0
~~~

##Looking at 'NotNovel' SNPs that did not show up as 'hits' for the 2011 univariate results/from the papers' tables to see where they are coming from
#~~~
#rs17080102 -- 
#http://snpedia.com/index.php/Rs17080102
#PMID	[PMID 23972371OA-icon.png]
#Trait	Blood pressure
#Title	Genome-wide association analysis of blood-pressure traits in African-ancestry individuals reveals common associated genes in African and non-African populations.
#Risk Allele	C
#P-val	5E-8
#Odds Ratio	1.02 [0.53-1.51] unit decrease
#
#rs880315 -- 
#http://snpedia.com/index.php/Rs880315
#
#GWAS snp
#PMID	[PMID 21572416OA-icon.png]
#Trait	
#Title	Meta-analysis of genome-wide association studies identifies common variants associated with blood pressure variation in east Asians.
#Risk Allele	C
#P-val	3E-10
#Odds Ratio	0.5600 [0.38-0.74] mm Hg increase
#
#rs1347677 -- 
#Only found a result hitting this SNP as a bone mineral density locus? So maybe 'novel' was at-all in terms of association and not blood pressure?
#http://www.ncbi.nlm.nih.gov/pubmed/21625884
#Genetic variants in the SOX6 gene are associated with bone mineral density in both Caucasian and Chinese populations.
#RESULTS:
#We confirmed that the previously reported rs7117858-A was associated with reduced hip BMD, with combined P value of 2.45 × 10(-4). Besides this SNP, we identified another 19 SNPs within or near the SOX6 gene to be significantly associated with hip BMD after false discovery rate adjustment. The most significant SNP was rs1347677 located at the intron 3 (P = 3.15 × 10(-7)). Seven additional SNPs in high linkage disequilibrium with rs1347677 were also significantly associated with hip BMD. SNPs in SOX6 showed significant skeletal site specificity since no SNP was detected to be associated with spine BMD.
#
#Well SOX6 may have role in blood pressure so maybe they look specifically at SOX6 locus?
#http://circ.ahajournals.org/content/130/Suppl_2/A15751.abstract
#Abstract 15751: A New Role of Sox6 in Renin Regulation
#Introduction: The renin-angiotensin system (RAS) is an important component of blood pressure regulation in mammals....
#
#rs7405452 --
#can't really find anything? tags hoxb-as3 and there's maybe something with muscular development --> blood pressure? http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3750649/
#
#rs2301597 --
#can't really find anything? tags NMT1 which is associated with skeletal traits and also maybe blood pressure stuff too?
#~~~

#Concern is whether if do 'FDR' analysis some SNPs will not show up in 'new results' and these are false negatives. Probably include these above SNPs that don't overlap with ICBP2011 resutls solely

cat /mnt/lustre/home/mturchin20/Data/iGENBP2015/iGENBP2015.SupplTable4.Novel.MarkerChrBP.txt <(cat /mnt/lustre/home/mturchin20/Data/iGENBP2015/iGENBP2015.SupplTable4.NotNovel.MarkerChrBP.w2011Hits.Hg19.txt | awk '{ if ($4 == 0) { print $0 } }' | awk '{ print $1, "\t", $2, "\t", $3 }') > /mnt/lustre/home/mturchin20/Data/iGENBP2015/iGENBP2015.SupplTable4.Novel.PlusNotNovel_No2011Overlap.MarkerChrBP.txt

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/newtophits.vs2.SignCrrct.vs1.MarkerChrBP.Hg19.txt --file2 /mnt/lustre/home/mturchin20/Data/iGENBP2015/iGENBP2015.SupplTable4.Novel.MarkerChrBP.txt

cat 

#Unsure if liftovered properly to correct reference genome. See discrepancies between liftovered coordinates and what I find th main 2015 iGEN files. Original coordinates, at least here, seem to do better. But really looks like previously wanted to liftover coordinates (at least in other, previous situation).
~~~
[  mturchin20@bigmem01  ~]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/newtophits.vs2.SignCrrct.vs1.MarkerChrBP.Hg19.txt | head -n 10
rs10006384       4       38531358
rs10473911       5       87260659
rs10505863       12      20337431
rs10865196       2       44247777
rs11066320       12      112884752
rs11694671       2       190915831
rs13035036       2       43248535
rs13306677       16      56926195
rs17089717       13      73310615
rs17677264       8       9412050
[  mturchin20@bigmem01  ~]$cat /mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_SBP_summary_stats.txt.gz | grep rs10505863
[  mturchin20@bigmem01  ~]$cat /mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_SBP_summary_stats.txt.gz | grep rs13306677
[  mturchin20@bigmem01  ~]$zcat /mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_SBP_summary_stats.txt.gz | grep rs10505863
[  mturchin20@bigmem01  ~]$zcat /mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_SBP_summary_stats.txt.gz | grep rs17089717
rs17089717      13      72208616        -0.0583 71939
[  mturchin20@bigmem01  ~]$cat/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/newtophits.vs2.SignCrrc
[  mturchin20@bigmem01  ~]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/newtophits.vs2.SignCrrct.vs1.txt | grep rs17089717
rs17089717 13 72208616 0 0.02184 2.42092 4.18162 3.46505 34.15971 6.16049 4.53845 25977.9795 9.82328
[  mturchin20@bigmem01  ~]$zcat /mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_SBP_summary_stats.txt.gz | grep rs10006384
rs10006384      4       38061582        0.098   95824.4
[  mturchin20@bigmem01  ~]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/newtophits.vs2.SignCrrct.vs1.txt | grep rs10006384
rs10006384 4 38207753 0 0.46036 3.9338 4.15981 4.008 48.54274 9.13829 5.3892 26551.25684 10.64343
[  mturchin20@bigmem01  ~]$zcat /mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_SBP_summary_stats.txt.gz | grep rs10865196
rs10865196      2       44043134        0.1503  99950
[  mturchin20@bigmem01  ~]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/newtophits.vs2.SignCrrct.vs1.txt | grep rs10865196
rs10865196 2 44101281 0 0.37523 -3.28318 -3.50453 -3.44775 33.46821 6.01872 4.96282 27671.17414 8.65452
[  mturchin20@bigmem01  ~]$join <(zcat /mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_SBP_summary_stats.txt.gz | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/newtophits.vs2.SignCrrct.vs1.MarkerChrBP.Hg19.txt | sort)
rs10006384 4 38061582 0.098 95824.4 4 38531358
rs10865196 2 44043134 0.1503 99950 2 44247777
rs11066320 12 111390798 0.375 59157 12 112884752
rs11694671 2 190506815 0.0307 91123.9 2 190915831
rs13306677 16 55483696 -0.1433 82440.9 16 56926195
rs17089717 13 72208616 -0.0583 71939 13 73310615
rs2052313 7 107742212 0.2428 90597 7 107761691
rs267218 3 45608838 0.0642 90130.2 3 45633834
rs2823119 21 15469515 -0.2998 91127 21 16547644
rs2969070 7 2479071 -0.0191 99903.3 7 2319260
rs320109 11 63436621 -0.3115 92843.3 11 63680045
rs3735533 7 27212418 -0.2522 72664.8 7 27052608
rs4778850 15 78821473 0.1331 95614.6 15 81034418
rs6136464 20 18605571 -0.0415 91122.4 20 18657571
rs729820 3 196402620 0.0416 60360.8 3 194921339
rs783622 1 42139575 0.1853 99957 1 42263494
rs920644 10 62111562 -0.0289 96995.5 10 62441556
rs9288052 2 181182281 0.0461 99884 2 181591297
[  mturchin20@bigmem01  ~]$join <(zcat /mnt/gluster/data/external_public_supp/iGENBP2015/_ega-box-493_SBP_summary_stats.txt.gz | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/newtophits.vs2.SignCrrct.vs1.txt | sort) | perl -lane 'print join("\t", @F[0..9]);'
rs10006384      4       38061582        0.098   95824.4 4       38207753        0       0.46036 3.9338
rs10865196      2       44043134        0.1503  99950   2       44101281        0       0.37523 -3.28318
rs11066320      12      111390798       0.375   59157   12      111369135       0       0.43261 6.53595
rs11694671      2       190506815       0.0307  91123.9 2       190624076       0       0.24175 3.66542
rs13306677      16      55483696        -0.1433 82440.9 16      55483696        0       0.10047 -4.44518
rs17089717      13      72208616        -0.0583 71939   13      72208616        0       0.02184 2.42092
rs2052313       7       107742212       0.2428  90597   7       107548927       0       0.40212 -4.99188
rs267218        3       45608838        0.0642  90130.2 3       45608838        0       0.48792 -3.51352
rs2823119       21      15469515        -0.2998 91127   21      15469515        0       0.21574 3.26292
rs2969070       7       2479071 -0.0191 99903.3 7       2285786 0       0.35422 5.9569
rs320109        11      63436621        -0.3115 92843.3 11      63436621        0       0.43595 -2.81243
rs3735533       7       27212418        -0.2522 72664.8 7       27019133        0       0.0775  -3.15593
rs4778850       15      78821473        0.1331  95614.6 15      78821473        0       0.20508 -2.31716
rs6136464       20      18605571        -0.0415 91122.4 20      18605571        0       0.39139 2.47572
rs729820        3       196402620       0.0416  60360.8 3       196402628       0       0.08453 3.26285
rs783622        1       42139575        0.1853  99957   1       42036081        0       0.43851 3.13021
rs920644        10      62111562        -0.0289 96995.5 10      62111562        0       0.19241 -2.95709
rs9288052       2       181182281       0.0461  99884   2       181299542       0       0.34278 3.35441
~~~












###MAGIC2010 -- ChrBP not provided, GWAS = 18 w/ followup, N=not provided

mkdir /mnt/lustre/home/mturchin20/Data/MAGIC2010

#Copy/pasted table 1 from http://www.nature.com/ng/journal/v42/n2/full/ng.520.html into /mnt/lustre/home/mturchin20/Data/MAGIC2010/MAGIC2010.Table1.txt

cat /mnt/lustre/home/mturchin20/Data/MAGIC2010/MAGIC2010.Table1.txt | grep rs | awk '{ print $1 } ' > /mnt/lustre/home/mturchin20/Data/MAGIC2010/MAGIC2010.Table1.rsID.txt










##GEFOS2012
##20160225

mkdir /mnt/lustre/home/mturchin20/Data/GEFOS2012

#Copy/pasted supplementary tables 4a and 4b from http://www.nature.com/ng/journal/v44/n5/full/ng.2249.html into /mnt/lustre/home/mturchin20/Data/GEFOS2012/GEFOS2012.SupplTable4a.txt and /mnt/lustre/home/mturchin20/Data/GEFOS2012/GEFOS2012.SupplTable4b.txt

#20160305 CHECK_0 -- Prob: Check these below, not getting 56 SNPs getting more than that

cat /mnt/lustre/home/mturchin20/Data/GEFOS2012/GEFOS2012.SupplTable4a.txt | awk '{ print $1 }' | grep rs | grep -v * > /mnt/lustre/home/mturchin20/Data/GEFOS2012/GEFOS2012.SupplTable4a.Marker.noStar.txt

cat /mnt/lustre/home/mturchin20/Data/GEFOS2012/GEFOS2012.SupplTable4b.txt | awk '{ print $1 }' | grep rs | grep -v b > /mnt/lustre/home/mturchin20/Data/GEFOS2012/GEFOS2012.SupplTable4b.Marker.nob.txt

#cat /mnt/lustre/home/mturchin20/Data/GEFOS2012/GEFOS2012.SupplTable4a.Marker.noStar.txt /mnt/lustre/home/mturchin20/Data/GEFOS2012/GEFOS2012.SupplTable4b.Marker.nob.txt > /mnt/lustre/home/mturchin20/Data/GEFOS2012/GEFOS2012.SupplTable4a4b.Marker.txt

cat /mnt/lustre/home/mturchin20/Data/GEFOS2012/GEFOS2012.SupplTable4a.txt | awk '{ print $1 }' | grep rs | sed 's/\*//g' > /mnt/lustre/home/mturchin20/Data/GEFOS2012/GEFOS2012.SupplTable4a.Marker.StarRemoved.txt
cat /mnt/lustre/home/mturchin20/Data/GEFOS2012/GEFOS2012.SupplTable4b.txt | awk '{ print $1 }' | grep rs | sed 's/b//g' > /mnt/lustre/home/mturchin20/Data/GEFOS2012/GEFOS2012.SupplTable4b.Marker.bRemoved.txt

cat /mnt/lustre/home/mturchin20/Data/GEFOS2012/GEFOS2012.SupplTable4a.Marker.StarRemoved.txt /mnt/lustre/home/mturchin20/Data/GEFOS2012/GEFOS2012.SupplTable4b.Marker.bRemoved.txt > /mnt/lustre/home/mturchin20/Data/GEFOS2012/GEFOS2012.SupplTable4a4b.Marker.txt

##Checking that getting same number of total SNPS by looking across supplementary tables 5a and 5b too from http://www.nature.com/ng/journal/v44/n5/full/ng.2249.html into /mnt/lustre/home/mturchin20/Data/GEFOS2012/GEFOS2012.SupplTable5a.txt and /mnt/lustre/home/mturchin20/Data/GEFOS2012/GEFOS2012.SupplTable5b.txt
#
#cat /mnt/lustre/home/mturchin20/Data/GEFOS2012/GEFOS2012.SupplTable5a.txt | awk '{ print $1 }' | grep rs > /mnt/lustre/home/mturchin20/Data/GEFOS2012/GEFOS2012.SupplTable5a.Marker.txt
#cat /mnt/lustre/home/mturchin20/Data/GEFOS2012/GEFOS2012.SupplTable5b.txt | awk '{ print $1 }' | grep rs > /mnt/lustre/home/mturchin20/Data/GEFOS2012/GEFOS2012.SupplTable5b.Marker.txt
#
#~~~
#[  mturchin20@spudling12  ~/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1]$cat /mnt/lustre/home/mturchin20/Data/GEFOS2012/GEFOS2012.SupplTable5a.Marker.txt /mnt/lustre/home/mturchin20/Data/GEFOS2012/GEFOS2012.SupplTable5b.Marker.txt | wc
#     56      56     564
#~~~






##GEFOS2015
##20160309

#From http://www.nature.com/nature/journal/v526/n7571/full/nature14878.html
~~~
In total, we have identified multiple variants associated with BMD, including 3 genomeÃ¢ÂÂwide significant loci for forearm BMD, 14 for femoral neck and 19 for lumbar spine (Supplementary Tables 10, 13, 14, 15, and Extended Data Figs 7 and 8).
~~~

mkdir /mnt/lustre/home/mturchin20/Data/GEFOS2015

#Copy/pasted supplementary table 13 from http://www.nature.com/nature/journal/v526/n7571/full/nature14878.html#supplementary-information into /mnt/lustre/home/mturchin20/Data/GEFOS2015/GEFOS2015.SupplTable13.txt

cat /mnt/lustre/home/mturchin20/Data/GEFOS2015/GEFOS2015.SupplTable13.txt 

cat /mnt/lustre/home/mturchin20/Data/GEFOS2015/GEFOS2015.SupplTable13.txt | sed 's/^\s*//g' | grep rs | sed 's/Forearm//g' | sed 's/Femoral Neck//g' | sed 's/Lumbar Spine//g' | awk '{ print $4, "\t", $2, "\t", $3 }' | grep rs > /mnt/lustre/home/mturchin20/Data/GEFOS2015/GEFOS2015.SupplTable13.edited.MarkerChrBP.txt

#Copy/pasted table 1 from http://www.nature.com/nature/journal/v526/n7571/full/nature14878.html into /mnt/lustre/home/mturchin20/Data/GEFOS2015/GEFOS2015.Table1.txt

cat /mnt/lustre/home/mturchin20/Data/GEFOS2015/GEFOS2015.Table1.txt | perl -lane 'print $F[$#F-14];' | grep rs > /mnt/lustre/home/mturchin20/Data/GEFOS2015/GEFOS2015.Table1.Marker.txt





##EGG2013
##20160225

mkdir /mnt/lustre/home/mturchin20/Data/EGG2013

#Copy/pasted table 1 from http://hmg.oxfordjournals.org/content/22/13/2735.long into /mnt/lustre/home/mturchin20/Data/EGG2013/EGG2013.Table1.txt

cat /mnt/lustre/home/mturchin20/Data/EGG2013/EGG2013.Table1.txt | grep rs | awk '{ print $1 }' > /mnt/lustre/home/mturchin20/Data/EGG2013/EGG2013.Table1.rsID.pre1.txt

cat /mnt/lustre/home/mturchin20/Data/EGG2013/EGG2013.Table1.rsID.pre1.txt > /mnt/lustre/home/mturchin20/Data/EGG2013/EGG2013.Table1.rsID.txt

#Manually removed ?<80><83> from /mnt/lustre/home/mturchin20/Data/EGG2013/EGG2013.Table1.rsID.txt

zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_10F.txt.gz | grep -w -f /mnt/lustre/home/mturchin20/Data/EGG2013/EGG2013.Table1.rsID.txt | awk '{ print $1, "\t", $2, "\t", $3 }'  > /mnt/lustre/home/mturchin20/Data/EGG2013/EGG2013.Table1.ChrBP.txt

~~~
[  mturchin20@spudling12  ~/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1]$cat /mnt/lustre/home/mturchin20/Data/EGG2013/EGG2013.Table1.rsID.txt | wc
     10      10      99
[  mturchin20@spudling12  ~/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1]$zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_10F.txt.gz | grep -w -f /mnt/lustre/home/mturchin20/Data/EGG2013/EGG2013.Table1.rsID.txt | awk '{ print $1, "\t", $2, "\t", $3 }'
rs1172294        2       25022704
rs1346789        2       55945556
rs1365198        15      82189907
rs4788196        16      29874935
rs6507528        18      19025578
rs6764769        3       142582970
rs7628864        3       86933308
rs7759938        6       105485647
rs7846385        8       78322734
rs960273         7       2824402
[  mturchin20@spudling12  ~/Lab_Stuff/StephensLab/Multivariate/EGG2013/Vs1]$zcat /mnt/gluster/data/external_public_supp/EGG/Pubertal_growth_10F.txt.gz | grep -w -f /mnt/lustre/home/mturchin20/Data/EGG2013/EGG2013.Table1.rsID.txt | awk '{ print $1, "\t", $2, "\t", $3 }'  | wc
     10      30     253
~~~




##Blood Traits 2012/HaemgenRBC2012 -- ChrBP provided, GWAS = 75 <10e-8 , N=provided
##20151006

mkdir /mnt/lustre/home/mturchin20/Data/HaemgenRBC2012

#Copy/pasted table 1 from //www.nature.com/nature/journal/v492/n7429/full/nature11677.html into /mnt/lustre/home/mturchin20/Data/HaemgenRBC2012/HaemgenRBC2012.Table1.txt

cat /mnt/lustre/home/mturchin20/Data/HaemgenRBC2012/HaemgenRBC2012.Table1.txt | awk '{ print $2, "\t", $1, "\t", $3 }' | perl -lane 'if ($F[1] =~ m/(\d+)(p|q)\d+/) { $F[1] = $1; } print join("\t", @F);' | sed 's/,//g' | grep rs[[:digit:]]\\+ > /mnt/lustre/home/mturchin20/Data/HaemgenRBC2012/HaemgenRBC2012.Table1.Edited.MarkerChrBP.txt




##HaemgenRBC2016
##20170123








#GPC











###########################
#
#
# Part D -- Follow-up Analyses
#
#
###########################

###Part Da
###First section below are follow-up work grouped by analyses (vs. grouped by datasets as in the second-half section)


##MAF, Gene Proximity SNP matching framework/code setup
##20161226

#Given list of input SNPs, match based on MAF. Beforehand can annotate whether near gene or not, and/or nearby number of genes.
#Have 50 hash bins of MAFs, .01, .02, .03, .04, .05, .06 -- round MAFs to two significant digits? Care about being near the edges or not really? 

#Just go through list of all SNPs first and make entries in hash for each maf, then for second file (of SNPs of interest), go through X # of times where X is also an input number, and print out different version of the matched output lists X # of times. Also include an input for seed value 

file1
file2
N (number of outputs)
output1
seed

build hash of two hashes of MAFs (first prox -- 2 hashes -- then maf -- the 50 or so)

for a given entry snp build the array of +/- 5% MAF entries (taken from within prox or not) then do N random grabs and file outputs for each SNP, so not redoing these array creations N times? Eg create a line that outputs/appends to the N files for each entry SNP?

cp -p /mnt/lustre/home/mturchin20/Scripts/Python/2fileTemplate.vs1.py /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataAnalysis.MatchSNPsBy.MAF_GeneProx.vs1.py 

#Chr, BP, GWASannot, MAF, 20kb within gene or not
#Remove GWASannot == 1, but keep == 2?
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.txt.gz | perl -lane 'print $F[1], "\t", $F[2], "\t", $F[2], "\t", uc($F[4]), "\t", uc($F[5]);' | grep -v BP > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.AnnovarFormat 

zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.txt.gz | awk '{ print $1 "\t" $2 "\t" $3 }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.ChrBPChrBP.txt

#20170118
#Note -- going through and redoing a lot of the annotation-related things since I was using hg19-formatted files when GlobalLipids2010 & 2013 are in hg18 (2013 has coordinates for both and I aligned with hg18 earlier on, likely also to help with overlap analyses with 2010). So redoing a bunch of things below and renaming within code, but for a good bunch there were versions made using hg19. So leftover files may be around here and there referring to the earlier forms of things when hg19 was used below

##perl /mnt/lustre/home/mturchin20/Software/annovar/summarize_annovar.pl /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.AnnovarFormat /mnt/lustre/home/mturchin20/Software/annovar/humandb/ --ver1000g 1000g2012apr --verdbsnp 129 --veresp 6500 --buildver hg19 --remove
perl /mnt/lustre/home/mturchin20/Software/annovar/summarize_annovar.pl /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.AnnovarFormat /mnt/lustre/home/mturchin20/Software/annovar/humandb/ --ver1000g 1000g2010 --verdbsnp 129 --veresp 6500 --buildver hg18 --remove

#20170105
#File /mnt/lustre/home/mturchin20/Data/UCSC/20170105.UCSC_hg19_GeneInformation_wSymbols.NoProtAlignIDs.txt was generated by going to https://genome.ucsc.edu/cgi-bin/hgTables, selecting assembly Feb. 2009 (GRCh37/hg19), group Genes and Gene Predictions, track UCSC Genes, table knownGene, output format selected fields from primary and related tables, and selecting all of hg19.knownGene except for proteinID and alignID and kgID & geneSymbol from hg19.kgXref
#I don't see any obvious differences between the list of genes included in 20170105.UCSC_hg19_GeneInformation_wSymbols.NoProtAlignIDs.txt and those missing from UCSC_hg19_GeneInformation_wSymbols.txt, though there are clearly differences. Also have to do a bit of finageling since don't have an old, 'NoProtAlignIDs' version of the UCSC_hg19_GeneInformation_wSymbols.txt file (which is form 2013)
#Redid the above on 20170118 for the hg18 version to get 20170118.UCSC_hg18_GeneInformation_wSymbols.NoProtAlignIDs.txt -- followed exact same instructions

#20170208
#Should have either used or checked results for GeneInformation condensing against `canonical` track in UCSC, which I forgot about. May also want to consider moving to refSeq which is what annovar annotations are based on. 
#See things like https://www.biostars.org/p/93011/ and https://www.biostars.org/p/19029/, as well as http://www.psytky03.com/2015/04/difference-between-refseq-ensembl-ucsc.html & http://www.biomedcentral.com/1471-2164/16/97
#Including space below as reminder for when come back to here. May also do work related to this below, see section regarding annotating Previous, New and MergedDataSources
#Have decided to just go with refSeq as resources such as http://www.psytky03.com/2015/04/difference-between-refseq-ensembl-ucsc.html point out, refSeq is a bit more 'standardized' using more metrics vs. say UCSC knownGenes. Also going to use the refSeq data pulled from the UCSC GenomeBrowser versus the refSeq dataset pulled from Annovar. This may be better since I can more clearly account for where the refSeq information is coming from vs. not knowing was assumptions/alterations went into the Annovar version one.
#20170214 NOTE -- realized the reason why `20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.txt` appeared to have less entries than either the Annovar refSeq files or even directly downloading the refSeq hg18 via `wget -c -O 20170210DL_hg18.refGene.txt.gz http://hgdownload.soe.ucsc.edu/goldenPath/hg18/database/refGene.txt.gz` is because by linking the main refSeq table in the table browser to `hg18.kgXref` in order to get the column `geneSymbol`, many entries were being dropped out (eg if they didn't link to `hg18.kgXref` they were just not included in the final output`). However there's a default column in the original `refGene` track/table that includes `name2	Alternate name (e.g. gene_id from GTF)` which appears to often include the gene symbole name/HUGO name? Either way, it's this column that Annovar is likely using to annotate so might as well also use it to help with 1-to-1 correspondences? Also leaning towards using the copy of the hg18_refSeq file that is the default UCSC table browser `refGene` output so I have a papertrail on how I've worked with/altered the file (I don't imagine the annovar file has been preprocessed too much? but just not sure and don't think there's an obvious explanation for anything like this on the Annovar website). 

mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses
mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration

cat /mnt/gluster/home/mturchin20/Software/20130911_annovar/humandb/hg18_refGene.txt | perl -lane 'print $F[$#F-3];' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20130911_annnovar.hg18_refGene.GeneIDs.uniq.txt
cat /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/hg18_refGene.txt | perl -lane 'print $F[$#F-3];' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20170208DL_20160204Vs_annnovar.hg18_refGene.GeneIDs.uniq.txt

#Got below, eg 20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.txt, from https://genome.ucsc.edu/cgi-bin/hgTables, selecting assembly Mar. 2006 (NCBI36/hg18), group Genes and Gene Predictions, track refSeq Genes, table refGene, output format selected fields from primary and related tables, and selecting bin, name, chrom, strand, txStart, txEnd, cdsStart, cdsEnd, exonCount from hg18.refGene and kgID, geneSymbol, refseq from hg18.kgXref
#Below from MacBook Air
#scp -p /Users/mturchin20/Documents/Work/LabMisc/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.txt mturchin20@wolfy.uchicago.edu:/Users/mturchin20/LabMisc/Data/UCSC/.
#scp -p mturchin20@wolfy.uchicago.edu:/Users/mturchin20/LabMisc/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.txt /mnt/lustre/home/mturchin20/Data/UCSC/.

rm /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.Condensed.txt; for i in `cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.txt | perl -lane 'print $F[$#F-1];' | sort | sed 's/,//g' | uniq | grep -v n/a | grep -v kgXref`; do echo $i; cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.txt | perl -lane '$F[$#F-1] =~ s/,//; print join("\t", @F);' | grep [^-a-zA-Z0-9]$i[^-a-zA-Z0-9] | perl -lane 'if ($. == 1) { @refIDs = ($F[1]); $txStart = $F[4]; $txEnd = $F[5]; } else { if ($F[4] < $txStart) { $txStart = $F[4]; } if ($F[5] > $txEnd) { $txEnd = $F[5]; } push(@refIDs, $F[1]); } if (eof()) { print $F[$#F-1], "\t", $., "\t", $F[2], "\t", $F[3], "\t", $txStart, "\t", $txEnd, "\t", join(",", @refIDs); }' | sed 's/chr//g' >> /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.Condensed.txt; done 
#mv /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.Condensed.vs2.txt /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.Condensed.txt

cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.Condensed.txt | awk '{ print "chr" $3 "\t" $5 "\t" $6; }' > /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.Condensed.bed

cd /mnt/lustre/home/mturchin20/Data/UCSC
wget -c -O 20170210.wgetDirect.refSeq_hg18.refGene.txt.gz http://hgdownload.soe.ucsc.edu/goldenPath/hg18/database/refGene.txt.gz

#The below file `20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.txt.gz` was downloaded from https://genome.ucsc.edu/cgi-bin/hgTables, selecting assembly Mar. 2006 (NCBI36/hg18), group Genes and Gene Predictions, track refSeq Genes, table refGene, and output format 'all fields from selected table' 
#From MacBook Air
#scp -p /Users/mturchin20/Documents/Work/LabMisc/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.txt.gz mturchin20@wolfy.uchicago.edu:/Users/mturchin20/LabMisc/Data/UCSC/.
scp -p mturchin20@wolfy.uchicago.edu:/Users/mturchin20/LabMisc/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.txt.gz /mnt/lustre/home/mturchin20/Data/UCSC/.   

rm /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20130911_annnovar.hg18_refGene.Condensed.vs2.txt; for i in `cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20130911_annnovar.hg18_refGene.GeneIDs.uniq.txt`; do echo $i; cat /mnt/gluster/home/mturchin20/Software/20130911_annovar/humandb/hg18_refGene.txt | grep [^-a-zA-Z0-9.]$i[^-a-zA-Z0-9.] | perl -lane 'if ($. == 1) { @refIDs = ($F[1]); $txStart = $F[4]; $txEnd = $F[5]; } else { if ($F[4] < $txStart) { $txStart = $F[4]; } if ($F[5] > $txEnd) { $txEnd = $F[5]; } push(@refIDs, $F[1]); } if (eof()) { print $F[$#F-3], "\t", $., "\t", $F[2], "\t", $F[3], "\t", $txStart, "\t", $txEnd, "\t", join(",", @refIDs); }' | sed 's/chr//g' >> /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20130911_annnovar.hg18_refGene.Condensed.vs2.txt; done 
rm /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20170208DL_20160204Vs_annnovar.hg18_refGene.Condensed.vs2.txt; for i in `cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20170208DL_20160204Vs_annnovar.hg18_refGene.GeneIDs.uniq.txt`; do echo $i; cat /mnt/gluster/home/mturchin20/Software/20130911_annovar/humandb/hg18_refGene.txt | grep [^-a-zA-Z0-9.]$i[^-a-zA-Z0-9.] | perl -lane 'if ($. == 1) { @refIDs = ($F[1]); $txStart = $F[4]; $txEnd = $F[5]; } else { if ($F[4] < $txStart) { $txStart = $F[4]; } if ($F[5] > $txEnd) { $txEnd = $F[5]; } push(@refIDs, $F[1]); } if (eof()) { print $F[$#F-3], "\t", $., "\t", $F[2], "\t", $F[3], "\t", $txStart, "\t", $txEnd, "\t", join(",", @refIDs); }' | sed 's/chr//g' >> /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20170208DL_20160204Vs_annnovar.hg18_refGene.Condensed.vs2.txt; done 

rm /mnt/lustre/home/mturchin20/Data/UCSC/20170210.wgetDirect.refSeq_hg18.refGene.Condensed.vs2.txt; for i in `zcat /mnt/lustre/home/mturchin20/Data/UCSC/20170210.wgetDirect.refSeq_hg18.refGene.txt.gz | perl -lane 'print $F[$#F-3];' | sort | uniq`; do echo $i; zcat /mnt/lustre/home/mturchin20/Data/UCSC/20170210.wgetDirect.refSeq_hg18.refGene.txt.gz | grep [^-a-zA-Z0-9.]$i[^-a-zA-Z0-9.] | perl -lane 'if ($. == 1) { @refIDs = ($F[1]); $txStart = $F[4]; $txEnd = $F[5]; } else { if ($F[4] < $txStart) { $txStart = $F[4]; } if ($F[5] > $txEnd) { $txEnd = $F[5]; } push(@refIDs, $F[1]); } if (eof()) { print $F[$#F-3], "\t", $., "\t", $F[2], "\t", $F[3], "\t", $txStart, "\t", $txEnd, "\t", join(",", @refIDs); }' | sed 's/chr//g' >> /mnt/lustre/home/mturchin20/Data/UCSC/20170210.wgetDirect.refSeq_hg18.refGene.Condensed.vs2.txt; done 
rm /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.vs2.txt; for i in `zcat /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.txt.gz | perl -lane 'print $F[$#F-3];' | sort | uniq`; do echo $i; zcat /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.txt.gz | grep [^-a-zA-Z0-9.]$i[^-a-zA-Z0-9.] | perl -lane 'if ($. == 1) { @refIDs = ($F[1]); $txStart = $F[4]; $txEnd = $F[5]; } else { if ($F[4] < $txStart) { $txStart = $F[4]; } if ($F[5] > $txEnd) { $txEnd = $F[5]; } push(@refIDs, $F[1]); } if (eof()) { print $F[$#F-3], "\t", $., "\t", $F[2], "\t", $F[3], "\t", $txStart, "\t", $txEnd, "\t", join(",", @refIDs); }' | sed 's/chr//g' >> /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.vs2.txt; done 

#20170214 NOTE -- Comparisons between /mnt/lustre/home/mturchin20/Data/UCSC/20170210.wgetDirect.refSeq_hg18.refGene.Condensed.txt` & `/mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.txt` show that `/mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.txt` has a little more information than `/mnt/lustre/home/mturchin20/Data/UCSC/20170210.wgetDirect.refSeq_hg18.refGene.Condensed.txt`. Mostly the same (like 98%/99%), but probably makes more sense to use the Table Browser output as well since that has been my normal course of action thus far (eg more consistent with current practices), vs. directly downloading a file from their webservers via wget.
#20170214 NOTE -- There does seem to be more differences though between the Annovar refSeq file and refSeq files directly from UCSC. The UCSC file `/mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.txt` has more entries overall, and among overlapping entries (per-gene), has wider ranges for genes and more transcripts per-gene. There are about ~500 genes that do not show up in the UCSC files that show up in the Annovar ones. Maybe just have to deal with this for the moment? Going to be using the UCSC-based files moving forward I believe, specifically still the `/mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.txt` one.

(20170214 notes/thoughts below)
then see if bedtools intersect can get all genes within 1bp, 20kb or 1Mb of a previous GWAS snp somehow -- can also compare annovar results to 1bp results and see if getting same correspondence, should be
do bedtools stuff before annovar annotating stuff?
- so two things here, do bedtools intersect stuff, do annovar, and be sure to compare 1bp bedtools intersect stuff to annovar results
include cadd score for now and liftOver, or not? probably include...







~~~
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/gluster/home/mturchin20/Software/20130911_annovar/humandb/hg18_refGene.txt | head -n 10
1071    NM_005762       chr19   +       63747647        63753894        63747936        63753732        17      63747647,63748603,63748942,63750554,63750775,63750970,63751212,63751472,63751664,63751918,63752040,63752166,63752509,63752915,63753127,63753325,63753555,      63748276,63748716,63749075,63750690,63750892,63751085,63751359,63751587,63751771,63751955,63752089,63752419,63752829,63753039,63753214,63753463,63753894,       0       TRIM28cmpl     cmpl    0,1,0,1,2,2,0,0,1,0,1,2,0,2,0,0,0,
608     NR_033488       chr6_cox_hap1   -       3100851 3116975 3116975 3116975 20      3100851,3101400,3101544,3101726,3101943,3102151,3102625,3102917,3103483,3103980,3104441,3105481,3105706,3106929,3107261,3110849,3114845,3115177,3115975,3116513,       3101198,3101447,3101643,3101803,3102006,3102208,3102689,3103022,3103607,3104030,3104505,3105583,3105821,3107052,3107335,3110935,3114932,3115244,3116032,3116975,        0       ABHD16A unk   unk      -1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,
607     NR_033488       chr6_qbl_hap2   -       2902315 2918448 2918448 2918448 20      2902315,2902864,2903008,2903190,2903407,2903615,2904089,2904381,2904947,2905444,2905905,2906945,2907170,2908393,2908725,2912321,2916318,2916650,2917448,2917986,       2902662,2902911,2903107,2903267,2903470,2903672,2904153,2904486,2905071,2905494,2905969,2907047,2907285,2908516,2908799,2912407,2916405,2916717,2917505,2918448,        0       ABHD16A unk   unk      -1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,
869     NR_033465       chr17   -       37234859        37246049        37246049        37246049        8       37234859,37237203,37238567,37240578,37242169,37244847,37244980,37245636,        37235435,37237404,37238730,37240668,37242255,37244894,37245050,37246049,       0       NT5C3B  unk     unk     -1,-1,-1,-1,-1,-1,-1,-1,
869     NR_033464       chr17   -       37234859        37246014        37246014        37246014        9       37234859,37237203,37238567,37240578,37242169,37244847,37244980,37245636,37245856,       37235435,37237404,37238730,37240668,37242255,37244894,37245050,37245735,37246014,      0       NT5C3B  unk     unk     -1,-1,-1,-1,-1,-1,-1,-1,-1,
1603    NR_033449       chr5    +       133479248       133511819       133511819       133511819       8       133479248,133501648,133502540,133505774,133506310,133506582,133507181,133509813,        133479676,133501754,133502628,133505894,133506473,133506690,133507230,133511819,       0       TCF7    unk     unk     -1,-1,-1,-1,-1,-1,-1,-1,
902     NM_001190214    chr15   -       41612951        41669743        41614163        41663994        30      41612951,41615624,41618959,41638284,41647952,41650901,41651213,41652258,41652523,41652846,41653298,41653711,41655860,41656316,41656597,41656942,41657111,41657365,41657559,41658508,41660491,41660716,41661342,41661584,41662053,41662351,41662893,41663847,41664288,41669661, 41614966,41615744,41619073,41638422,41648043,41651066,41651330,41652366,41652638,41652974,41653522,41653853,41656043,41656438,41656729,41657025,41657221,41657441,41657646,41658610,41660613,41660878,41661444,41661739,41662139,41662442,41663089,41664022,41664382,41669743, 0       PPIP5K1
        cmpl    cmpl    1,1,1,1,0,0,0,0,2,0,1,0,0,1,1,2,0,2,2,2,0,0,0,1,2,1,0,0,-1,-1,
759     NM_000491       chr1    +       22852268        22860616        22858536        22860466        3       22852268,22858519,22859891,     22852383,22858723,22860616,     0       C1QB    cmpl    cmpl    -1,0,1,
1581    NM_003925       chr3    -       130632476       130641712       130633033       130641366       8       130632476,130634035,130634630,130635382,130635594,130637975,130639252,130641262,        130633111,130634139,130634780,130635517,130635669,130638841,130639483,130641712,       0       MBD4    cmpl    cmpl    0,1,1,1,1,2,2,0,
926     NM_003295       chr13   -       44805605        44813419        44809520        44813204        6       44805605,44810794,44811631,44812128,44812846,44813176,  44809523,44810911,44811737,44812319,44812920,44813419,0
        TPT1    cmpl    cmpl    0,0,2,0,1,0,
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/gluster/home/mturchin20/Software/20130911_annovar/humandb/hg18_refGene.txt | perl -lane 'print $#F;' | sort | uniq -c
  53123 15
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/gluster/home/mturchin20/Software/20130911_annovar/humandb/hg18_refGene.txt | perl -lane 'print $F[$#F-3];' | head -n 10
TRIM28
ABHD16A
ABHD16A
NT5C3B
NT5C3B
TCF7
PPIP5K1
C1QB
MBD4
TPT1
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/gluster/home/mturchin20/Software/20130911_annovar/humandb/hg18_refGene.txt | perl -lane 'print $F[$#F-3];' | wc
  53123   53123  370783
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/gluster/home/mturchin20/Software/20130911_annovar/humandb/hg18_refGene.txt | perl -lane 'print $F[$#F-3];' | sort | uniq | wc
  26453   26453  194253
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170118.UCSC_hg18_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.txt | head -n 10
16G2    3       +       187840889       187853491
214K23.2        13      +       31726284        31768776
44050   10      -       99364299        99383334
5-OPase 8       -       145178154       145187594
6a8b    15      -       73435185        73447994
A121/SUI1       17      +       37098652        37101424
A18hnRNP        19      +       1220266 1223603
A1BG    19      -       63549983        63556677
A1CF    10      -       52236330        52315441
A26B1   15      -       19305252        19336667
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170118.UCSC_hg18_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.txt | awk '{ print $1 }' | wc
  29348   29348  217217
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170118.UCSC_hg18_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.txt | awk '{ print $1 }' | sort | uniq | wc
  29347   29347  217205
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170118.UCSC_hg18_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.txt | awk '{ print $1 }' | sort | uniq -d | wc
      1       1      12
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170118.UCSC_hg18_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.txt | awk '{ print $1 }' | sort | uniq -d     
PEG8/IGF2AS
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20130911_annnovar.hg18_refGene.GeneIDs.uniq.txt | sort) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170118.UCSC_hg18_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.txt | awk '{ print $1 }' | sort | uniq) | wc
  17475   17475  113194
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.txt | head -n 10                 
#hg18.refGene.bin       hg18.refGene.name       hg18.refGene.chrom      hg18.refGene.strand     hg18.refGene.txStart    hg18.refGene.txEnd      hg18.refGene.cdsStart   hg18.refGene.cdsEnd     hg18.refGene.exonCount  hg18.kgXref.kgID       hg18.kgXref.geneSymbol  hg18.kgXref.refseq
0       NM_207014       chr1    -       67075869        67163158        67075923        67163102        10      uc001dcy.1,uc009wax.1,uc001dcz.1,       WDR78,  NM_207014,
0       NM_024763       chr1    -       67051159        67163158        67052400        67163102        17      uc001dcx.1,uc009waw.1,  WDR78,  NM_024763,
1       NM_001144777    chr1    +       50286272        50442044        50286423        50439437        7       n/a     n/a     n/a
1       NM_017940       chr1    -       16761508        16812687        16763024        16791103        29      uc009vos.1,uc009vot.1,uc001ayz.1,uc009wjt.1,uc001eqf.1,uc009wje.1,uc009wke.1,   NBPF1,  NM_017940,
1       NM_001127714    chr1    -       41744622        42156965        41748708        41823055        8       uc001cha.2      HIVEP3  NM_001127714
1       NM_001324208    chr1    +       50286272        50442044        50286423        50439437        7       n/a     n/a     n/a
1       NM_001042681    chr1    -       8335050 8800286 8337733 8638943 23      uc001apf.1      RERE    NM_001042681
1       NM_001080438    chr1    -       33544953        33559286        33544953        33559286        5       uc001bxd.1      A3GALT2 NM_001080438
1       NM_001042682    chr1    -       8335050 8406334 8337733 8346780 13      uc001apd.1      RERE    NM_001042682
[  mturchin20@bigmem03  /mnt/gluster/home/mturchin20/Software]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.txt | perl -lane 'print $#F;' | sort | uniq -c
  61197 11
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.txt | perl -lane 'print $F[$#F-1];' | head -n 10
hg18.kgXref.geneSymbol
WDR78,
WDR78,
n/a
NBPF1,
HIVEP3
n/a
RERE
A3GALT2
RERE
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.txt | perl -lane 'print $F[$#F-1];' | sort | wc 
  61197   61197  321339
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.txt | perl -lane 'print $F[$#F-1];' | sort | uniq | wc
  21070   21070  147554
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20130911_annnovar.hg18_refGene.GeneIDs.uniq.txt | sort) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.txt | perl -lane 'print $F[$#F-1];' | sort | uniq) | wc
  10663   10663   68994
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/gluster/home/mturchin20/Software/20130911_annovar/humandb/hg18_refGene.txt | awk '{ print $13 "\t" $2 "\t" $3 "\t" $4 "\t" $5 "\t" $6 "\t" $7 "\t" $8 "\t" $9 }' | head -n 10
TRIM28  NM_005762       chr19   +       63747647        63753894        63747936        63753732        17
ABHD16A NR_033488       chr6_cox_hap1   -       3100851 3116975 3116975 3116975 20
ABHD16A NR_033488       chr6_qbl_hap2   -       2902315 2918448 2918448 2918448 20
NT5C3B  NR_033465       chr17   -       37234859        37246049        37246049        37246049        8
NT5C3B  NR_033464       chr17   -       37234859        37246014        37246014        37246014        9
TCF7    NR_033449       chr5    +       133479248       133511819       133511819       133511819       8
PPIP5K1 NM_001190214    chr15   -       41612951        41669743        41614163        41663994        30
C1QB    NM_000491       chr1    +       22852268        22860616        22858536        22860466        3
MBD4    NM_003925       chr3    -       130632476       130641712       130633033       130641366       8
TPT1    NM_003295       chr13   -       44805605        44813419        44809520        44813204        6
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.txt | awk '{ print $11 "\t" $2 "\t" $3 "\t" $4 "\t" $5 "\t" $6 "\t" $7 "\t" $8 "\t" $9 }' | head -n 10
hg18.kgXref.geneSymbol  hg18.refGene.name       hg18.refGene.chrom      hg18.refGene.strand     hg18.refGene.txStart    hg18.refGene.txEnd      hg18.refGene.cdsStart   hg18.refGene.cdsEnd     hg18.refGene.exonCount
WDR78,  NM_207014       chr1    -       67075869        67163158        67075923        67163102        10
WDR78,  NM_024763       chr1    -       67051159        67163158        67052400        67163102        17
n/a     NM_001144777    chr1    +       50286272        50442044        50286423        50439437        7
NBPF1,  NM_017940       chr1    -       16761508        16812687        16763024        16791103        29
HIVEP3  NM_001127714    chr1    -       41744622        42156965        41748708        41823055        8
n/a     NM_001324208    chr1    +       50286272        50442044        50286423        50439437        7
RERE    NM_001042681    chr1    -       8335050 8800286 8337733 8638943 23
A3GALT2 NM_001080438    chr1    -       33544953        33559286        33544953        33559286        5
RERE    NM_001042682    chr1    -       8335050 8406334 8337733 8346780 13
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.txt | awk '{ print $11 "\t" $2 "\t" $3 "\t" $4 "\t" $5 "\t" $6 "\t" $7 "\t" $8 "\t" $9 }' | sort -k 1,1) <(cat /mnt/gluster/home/mturchin20/Software/20130911_annovar/humandb/hg18_refGene.txt | awk '{ print $13 "\t" $2 "\t" $3 "\t" $4 "\t" $5 "\t" $6 "\t" $7 "\t" $8 "\t" $9 }' | sort -k 1,1) | head -n 10
A1BG NM_130786 chr19 - 63549983 63556677 63550199 63556615 8 NM_130786 chr19 - 63549983 63556677 63550199 63556615 8
A1CF NM_138932 chr10 - 52229174 52315441 52236494 52289706 13 NM_001198818 chr10 - 52229174 52315441 52236494 52289706 14
A1CF NM_138932 chr10 - 52229174 52315441 52236494 52289706 13 NM_001198819 chr10 - 52229174 52315441 52236494 52280553 15
A1CF NM_138932 chr10 - 52229174 52315441 52236494 52289706 13 NM_001198820 chr10 - 52229174 52315441 52236494 52280553 14
A1CF NM_138932 chr10 - 52229174 52315441 52236494 52289706 13 NM_014576 chr10 - 52229174 52315441 52236494 52289706 13
A1CF NM_138932 chr10 - 52229174 52315441 52236494 52289706 13 NM_138932 chr10 - 52229174 52315441 52236494 52289706 13
A1CF NM_138932 chr10 - 52229174 52315441 52236494 52289706 13 NM_138933 chr10 - 52229174 52315441 52236494 52280553 13
A1CF NM_138933 chr10 - 52229174 52315441 52236494 52280553 13 NM_001198818 chr10 - 52229174 52315441 52236494 52289706 14
A1CF NM_138933 chr10 - 52229174 52315441 52236494 52280553 13 NM_001198819 chr10 - 52229174 52315441 52236494 52280553 15
A1CF NM_138933 chr10 - 52229174 52315441 52236494 52280553 13 NM_001198820 chr10 - 52229174 52315441 52236494 52280553 14
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.txt | awk '{ print $11 "\t" $2 "\t" $3 "\t" $4 "\t" $5 "\t" $6 "\t" $7 "\t" $8 "\t" $9 }' | sort -k 1,1) <(cat /mnt/gluster/home/mturchin20/Software/20130911_annovar/humandb/hg18_refGene.txt | awk '{ print $13 "\t" $2 "\t" $3 "\t" $4 "\t" $5 "\t" $6 "\t" $7 "\t" $8 "\t" $9 }' | sort -k 1,1) | wc
  39625  673625 4819811
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.txt | awk '{ print $11 "\t" $2 "\t" $3 "\t" $4 "\t" $5 "\t" $6 "\t" $7 "\t" $8 "\t" $9 }' | sort -k 1,1) <(cat /mnt/gluster/home/mturchin20/Software/20130911_annovar/humandb/hg18_refGene.txt | awk '{ print $13 "\t" $2 "\t" $3 "\t" $4 "\t" $5 "\t" $6 "\t" $7 "\t" $8 "\t" $9 }' | sort -k 1,1) | uniq | wc
  39625  673625 4819811
[  mturchin20@bigmem03  /mnt/gluster/home/mturchin20/Software]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.txt | perl -lane 'print $F[$#F-1];' | sort | wc     
  61197   61197  321339
[  mturchin20@bigmem03  /mnt/gluster/home/mturchin20/Software]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.txt | perl -lane 'print $F[$#F-1];' | sort | uniq | wc
  21070   21070  147554
[  mturchin20@bigmem03  /mnt/gluster/home/mturchin20/Software]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.txt | perl -lane 'print $F[$#F-1];' | sort | sed 's/,//g' | wc
  61197   61197  310556
[  mturchin20@bigmem03  /mnt/gluster/home/mturchin20/Software]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.txt | perl -lane 'print $F[$#F-1];' | sort | sed 's/,//g' | uniq | wc
  19021   19021  125540
[  mturchin20@bigmem03  /mnt/gluster/home/mturchin20/Software]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.txt | perl -lane 'print $F[$#F-1];' | sort | sed 's/,//g' | uniq -c | sort -k 1,1 | awk '{ print $1 }' | sort | uniq -c
  14664 1
      5 10
      1 11
      3 12
      1 13
      1 14
      3 15
      1 16
      2 17
      1 18
   2902 2
      2 20
      1 24
    870 3
      1 35045
    309 4
    107 5
     81 6
     26 7
     26 8
     14 9
[  mturchin20@bigmem03  /mnt/gluster/home/mturchin20/Software]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.txt | perl -lane 'print $F[$#F-1];' | sort | sed 's/,//g' | uniq -c | sort -k 1,1 | awk '{ if ($1 == 4) { print $0 } }' | head -n 10
      4 A2BP1
      4 ABI1
      4 ABLIM1
      4 ACOT7
      4 ACP5
      4 ACRV1
      4 ACVR1C
      4 ADAR
      4 ADD1
      4 ADRA1A
[  mturchin20@bigmem03  /mnt/gluster/home/mturchin20/Software]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.txt | grep ABLIM1
22 NM_002313    chr10   -  116180858    116408048       116186008       116407949       22      uc009xyi.1,uc009xyj.1,  ABLIM1, NM_002313,
22 NM_001003408 chr10   -  116180858    116434404       116186008       116434102       23      uc009xyk.1,uc009xyp.1,  ABLIM1, NM_001003408,
22 NM_001003407 chr10   -  116180858    116434404       116186008       116434102       22      uc009xyl.1,uc009xym.1,uc009xyo.1,       ABLIM1, NM_001003407,
183 NM_006720   chr10   -  116180858    116276675       116186008       116241573       17      uc009xyh.1,uc009xyg.1,uc009xyn.1,       ABLIM1, NM_006720,
[  mturchin20@bigmem03  /mnt/gluster/home/mturchin20/Software]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.txt | perl -lane 'print $F[$#F-1];' | grep \- | head -n 10
mifr-1,MMP21/22B,
mifr-1,MMP21/22B,
SNHG3-RCC1
STON1-GTF2A1L
ARPP-21,
ARPP-21,
ARPP-21
ARPP-21
NKX3-2
NKX6-1
[  mturchin20@bigmem03  /mnt/gluster/home/mturchin20/Software]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.txt | perl -lane 'print $F[$#F-1];' | grep STON1 | head -n 10
STON1,
STON1-GTF2A1L
[  mturchin20@bigmem03  /mnt/gluster/home/mturchin20/Software]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.txt | grep STON1 | head -n 10
119 NM_006873   chr2    +  48610811     48679158        48661276        48675945        4       uc002rwq.1,uc002rwo.2,uc010fbm.1,       STON1,  NM_006873,
119 NM_172311   chr2    +  48649662     48760252        48661276        48760085        11      uc002rwp.1 STON1-GTF2A1L        NM_172311
[  mturchin20@bigmem03  /mnt/gluster/home/mturchin20/Software]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.txt | grep [^-a-zA-Z0-9]STON1[^-a-zA-Z0-9] | head -n 10
119 NM_006873   chr2    +  48610811     48679158        48661276        48675945        4       uc002rwq.1,uc002rwo.2,uc010fbm.1,       STON1,  NM_006873,
[  mturchin20@bigmem03  /mnt/gluster/home/mturchin20/Software]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.txt | perl -lane 'print $F[$#F-1];' | grep \_ | sort | uniq | wc
      7       7      94
[  mturchin20@bigmem03  /mnt/gluster/home/mturchin20/Software]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.txt | perl -lane 'print $F[$#F-1];' | grep \_ | sort | uniq     
RPL13A_6_816
RPL13A_9_1211
RPL23A_20_869
RPL23A_25_1182
RPL23A_9_220
RPL31_1_108
tmp_locus_39
[  mturchin20@bigmem03  /mnt/gluster/home/mturchin20/Software]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.txt | perl -lane 'print $F[$#F-1];' | sed 's/,/ /g' | perl -lane 'print $#F;' | sort | uniq -c | head -n 10
  61190 0
      7 1
[  mturchin20@bigmem03  /mnt/gluster/home/mturchin20/Software]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.txt | perl -lane 'print $F[$#F-1];' | sed 's/,/ /g' | perl -lane 'if ($#F == 1) { print join("\t", @F); }' 
mifr-1  MMP21/22B
mifr-1  MMP21/22B
HLA-A   HLA-H
HLA-DRB1    HLA-DRB6
HLA-A   HLA-H
HLA-A   HLA-H
HLA-DRB1    HLA-DRB6
[  mturchin20@bigmem03  /mnt/gluster/home/mturchin20/Software]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.txt | perl -lane 'print $F[$#F-1];' | grep HLA-A | sort | uniq
HLA-A,
HLA-A,HLA-H,
[  mturchin20@bigmem03  /mnt/gluster/home/mturchin20/Software]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.txt | perl -lane 'print $F[$#F-1];' | grep HLA | sort | uniq
HHLA2,
HHLA3
HLA-A,
HLA-A,HLA-H,
HLA-B,
HLA-C,
HLA-DMA,
HLA-DMB,
HLA-DOA,
HLA-DOB,
HLA-DPA1,
HLA-DPB1,
HLA-DPB2,
HLA-DQA1
HLA-DQA2,
HLA-DQB1,
HLA-DRA,
HLA-DRB1
HLA-DRB1,HLA-DRB6,
HLA-DRB3
HLA-DRB5,
HLA-E,
HLA-F,
HLA-G,
[  mturchin20@bigmem03  /mnt/gluster/home/mturchin20/Software]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.txt | perl -lane 'print $F[$#F-1];' | grep mifr | sort | uniq
mifr-1,MMP21/22B,
[  mturchin20@bigmem03  /mnt/gluster/home/mturchin20/Software]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.txt | perl -lane 'print $F[$#F-1];' | grep MMP | sort | uniq
IMMP1L,
IMMP2L,
MMP1
MMP10
MMP11
MMP12
MMP13
MMP14
MMP15
MMP16
MMP17,
MMP19
MMP2
MMP2,
MMP20
MMP21
MMP23B
MMP24
MMP25,
MMP26
MMP27
MMP28
MMP28,
MMP3
MMP7,
MMP8
MMP9
mifr-1,MMP21/22B,
[  mturchin20@bigmem03  /mnt/gluster/home/mturchin20/Software]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.txt | perl -lane 'print $F[$#F-1];' | grep MMP21\/ | sort | uniq
mifr-1,MMP21/22B,
[  mturchin20@bigmem03  /mnt/gluster/home/mturchin20/Software]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.txt | perl -lane 'print $F[$#F-1];' | sort | sed 's/,//g' | uniq -c | sort -k 1,1 | awk '{ if ($1 == 24) { print $0 } }' | head -n 10
     24 MOG
[  mturchin20@bigmem03  /mnt/gluster/home/mturchin20/Software]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.txt | grep -w MOG | head -n 10
811 NM_002433   chr6    +  29732736     29748128        29732965        29747206        8       uc003nne.1,uc003qzr.1,uc003rrw.1,       MOG,    NM_002433,
811 NM_206812   chr6    +  29732736     29748128        29732965        29746941        7       uc003nnj.1,uc003qzw.1,uc003rsb.1,       MOG,    NM_206812,
811 NM_206810   chr6    +  29732736     29748128        29732965        29747206        7       uc003nnh.1,uc003qzu.1,uc003rrz.1,       MOG,    NM_206810,
811 NM_206809   chr6    +  29732736     29748128        29732965        29746941        8       uc003nnf.1,uc003nmy.1,uc003qzs.1,uc003qzl.1,uc003rrx.1,uc003rrq.1,      MOG,    NM_206809,
811 NM_001008228   chr6 +  29732736     29748128        29732965        29746941        8       uc003nni.1,uc003qzv.1,uc003rsa.1,       MOG,    NM_001008228,
811 NM_206811   chr6    +  29732736     29748128        29732965        29747206        8       uc003nng.1,uc003qzt.1,uc003rry.1,       MOG,    NM_206811,
811 NM_206814   chr6    +  29732736     29748128        29732965        29746941        7       uc003nnd.1,uc003qzq.1,uc003rrv.1,       MOG,    NM_206814,
811 NM_001008229   chr6 +  29732736     29748128        29732965        29747206        6       uc003nnk.1,uc003qzx.1,uc003qzk.1,uc003rsc.1,uc003rrp.1, MOG,    NM_001008229,
593 NM_206810   chr6_cox_hap1 + 1080065 1095458 1080294 1094528 7       uc003nnh.1,uc003qzu.1,uc003rrz.1,  MOG, NM_206810,
593 NM_001008228   chr6_cox_hap1        +       1080065 1095458 1080294 1094263 8       uc003nni.1,uc003qzv.1,uc003rsa.1,       MOG,    NM_001008228,
[  mturchin20@bigmem03  /mnt/gluster/home/mturchin20/Software]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.txt | grep -w MOG | wc        
     24     288    2865
[  mturchin20@bigmem03  /mnt/gluster/home/mturchin20/Software]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.Condensed.vs2.txt | awk '{ if ($2 >= 4) { print $0 } }' | head -n 10
A2BP1   4   16  +  6009132 7703341
ABCF1   6   6_qbl_hap2  +  1786022      30667288
ABCG1   6   21  +  42492867   42590423
ABI1 4  10  -   27075530   27190022
ABLIM1  4   10  -  116180858  116434404
ACACA   5   17  -  32516039   32841015
ACOT7   4   1   -  6246918 6376413
ACP5 4  19  -   11546474   11550823
ACRV1   4   11  -  125047438  125056003
ACVR1C  4   2   -  158091524  158193645
[  mturchin20@bigmem03  /mnt/gluster/home/mturchin20/Software]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.txt | grep ABCG1
909 NM_207627   chr21   +  42492867     42590423        42494854        42589571        16      uc002zan.1 ABCG1 NM_207627
909 NM_207629   chr21   +  42509255     42590423        42509334        42589571        15      uc002zao.1 ABCG1 NM_207629
909 NM_016818   chr21   +  42512335     42590423        42512441        42589571        15      uc002zap.1 ABCG1 NM_016818
909 NM_004915   chr21   +  42512335     42590423        42512441        42589571        15      uc002zaq.1 ABCG1 NM_004915
909 NM_207628   chr21   +  42492990     42590423        42518873        42589571        17      uc002zam.1 ABCG1 NM_207628
909 NM_207174   chr21   +  42513076     42590423        42513150        42589571        15      uc002zar.1 ABCG1 NM_207174
[  mturchin20@bigmem03  /mnt/gluster/home/mturchin20/Software]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.txt | grep ACVR1C
1791 NM_145259  chr2    -  158091524    158193645       158098675       158193402       9       uc002tzk.2 ACVR1C     NM_145259
1791 NM_001111031  chr2 -  158091524    158162318       158098675       158152098       9       uc010foe.1 ACVR1C     NM_001111031
1791 NM_001111033  chr2 -  158091524    158193645       158098675       158193402       7       uc010fof.1 ACVR1C     NM_001111033
1791 NM_001111032  chr2 -  158091524    158193645       158098675       158193402       8       uc002tzl.2 ACVR1C     NM_001111032
[  mturchin20@bigmem03  /mnt/gluster/home/mturchin20/Software]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.txt | grep -w ACVRV1
[  mturchin20@bigmem03  /mnt/gluster/home/mturchin20/Software]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.txt | grep -w ACRV1
1539 NM_020107  chr11   -  125047438    125056003       125047697       125055885       5       uc001qcn.1 ACRV1 NM_020107
1539 NM_020069  chr11   -  125047438    125056003       125047697       125055885       4       uc001qcr.1 ACRV1 NM_020069
1539 NM_001612  chr11   -  125047438    125056003       125047697       125055885       4       uc001qcs.1 ACRV1 NM_001612
1539 NM_020108  chr11   -  125047438    125056003       125047697       125055885       5       uc001qcl.1 ACRV1 NM_020108
[  mturchin20@bigmem03  /mnt/gluster/home/mturchin20/Software]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.txt | grep -w ACACA
104 NM_198839   chr17   -  32516039     32841015        32518363        32761452        60      uc002hnn.1 ACACA NM_198839
104 NM_198837   chr17   -  32516039     32730805        32518363        32730370        54      uc002hnl.1 ACACA NM_198837
104 NM_198836   chr17   -  32516039     32790172        32518363        32761452        56      uc002hnm.1 ACACA NM_198836
104 NM_198838   chr17   -  32516039     32730805        32518363        32715977        55      uc002hnk.1,uc010cuy.1,  ACACA,  NM_198838,
104 NM_198834   chr17   -  32516039     32841015        32518363        32840502        56      uc002hno.1,uc010cuz.1,  ACACA,  NM_198834,
[  mturchin20@bigmem03  /mnt/gluster/home/mturchin20/Software]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.txt | grep ACOT7   
9  NM_007274    chr1    -  6246918      6376413 6247243 6376046 9       uc001amt.1      ACOT7   NM_007274
9  NM_181866    chr1    -  6246918      6341591 6247243 6341492 9       uc001amq.1      ACOT7   NM_181866
9  NM_181865    chr1    -  6246918      6343351 6247243 6343299 9       uc001amr.1      ACOT7   NM_181865
9  NM_181864    chr1    -  6246918      6368470 6247243 6368313 9       uc001ams.1      ACOT7   NM_181864
[  mturchin20@bigmem03  /mnt/gluster/home/mturchin20/Software]$cat /mnt/gluster/home/mturchin20/Software/20130911_annovar/humandb/hg18_refGene.txt | perl -lane 'print $F[$#F-3];' | sort | grep , | wc
      0       0       0
[  mturchin20@bigmem03  /mnt/gluster/home/mturchin20/Software]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20130911_annnovar.hg18_refGene.Condensed.txt | head -n 10
A1BG 1  19  -   63549983   63556677     NM_130786
A1BG-AS1    1   19 +    63555147        63558361        NR_015380
A1CF 6  10  -   52229174   52315441     NM_001198820,NM_001198819,NM_001198818,NM_014576,NM_138933,NM_138932
A2M 1   12  -   9111570 9159825 NM_000014
A2M-AS1 1   12  +  9109039 9111918      NR_026971
A2ML1   2   12  +  8866334 8920644      NM_001282424,NM_144670
A2MP1   1   12  -  9272395 9278070      NR_040112
A3GALT2 1   1   -  33544953   33559286  NM_001080438
A4GALT  1   22  -  41418070   41446820  NM_017436
A4GNT   1   3   -  139325249  139333919 NM_016161
[  mturchin20@bigmem03  /mnt/gluster/home/mturchin20/Software]$cat /mnt/gluster/home/mturchin20/Software/20130911_annovar/humandb/hg18_refGene.txt | perl -lane 'print $F[$#F-3];' | sort | uniq -c | awk '{ print $1 }' | sort | uniq -c             
  15501 1
     73 10
     32 11
     40 12
     21 13
     10 14
     16 15
      8 16
      7 17
      8 18
      1 19
   4808 2
      4 20
      2 21
      2 22
      3 23
      2 24
      1 25
      4 27
      1 28
   2663 3
      1 30
      2 31
      1 32
      1 33
   1438 4
    776 5
    512 6
    229 7
      1 77
    158 8
    127 9
[  mturchin20@spudhead  ~]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.Condensed.txt | awk '{ print "chr" $3 "\t" $5 "\t" $6; }' | head -n 10
chr19   63549983        63556677
chr10   52229174        52315441
chr16   6009132 7703341
chr13   99980418        99984057
chr12   9111570 9160092
chr12   8866334 8920644
chr1    33544953        33559286
chr22   41418064        41446860
chr3    139325249       139333919
chr7    34724998        34840468
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20130911_annnovar.hg18_refGene.Condensed.txt | wc
  26453  185171 1446264
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.Condensed.txt | wc
  19019  133133  875931
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20130911_annnovar.hg18_refGene.Condensed.txt | head -n 10
A1BG    1       19      -       63549983        63556677        NM_130786
A1BG-AS1        1       19      +       63555147        63558361        NR_015380
A1CF    6       10      -       52229174        52315441        NM_001198820,NM_001198819,NM_001198818,NM_014576,NM_138933,NM_138932
A2M     1       12      -       9111570 9159825 NM_000014
A2M-AS1 1       12      +       9109039 9111918 NR_026971
A2ML1   2       12      +       8866334 8920644 NM_001282424,NM_144670
A2MP1   1       12      -       9272395 9278070 NR_040112
A3GALT2 1       1       -       33544953        33559286        NM_001080438
A4GALT  1       22      -       41418070        41446820        NM_017436
A4GNT   1       3       -       139325249       139333919       NM_016161
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.Condensed.txt | head -n 10
A1BG    1       19      -       63549983        63556677        NM_130786
A1CF    3       10      -       52229174        52315441        NM_138933,NM_138932,NM_014576
A2BP1   4       16      +       6009132 7703341 NM_018723,NM_145892,NM_145891,NM_145893
A2LD1   1       13      -       99980418        99984057        NM_033110
A2M     1       12      -       9111570 9160092 NM_000014
A2ML1   1       12      +       8866334 8920644 NM_144670
A3GALT2 1       1       -       33544953        33559286        NM_001080438
A4GALT  1       22      -       41418064        41446860        NM_017436
A4GNT   1       3       -       139325249       139333919       NM_016161
AAA1    1       7       -       34724998        34840468        NR_015356
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.Condensed.txt | wc
  19019  133133  875931
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20130911_annnovar.hg18_refGene.Condensed.txt | wc
  26453  185171 1446264
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20130911_annnovar.hg18_refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.Condensed.txt | sort -k 1,1) | head -n 10
A1BG 1 19 - 63549983 63556677 NM_130786 1 19 - 63549983 63556677 NM_130786
A1CF 6 10 - 52229174 52315441 NM_001198820,NM_001198819,NM_001198818,NM_014576,NM_138933,NM_138932 3 10 - 52229174 52315441 NM_138933,NM_138932,NM_014576
A2M 1 12 - 9111570 9159825 NM_000014 1 12 - 9111570 9160092 NM_000014
A2ML1 2 12 + 8866334 8920644 NM_001282424,NM_144670 1 12 + 8866334 8920644 NM_144670
A3GALT2 1 1 - 33544953 33559286 NM_001080438 1 1 - 33544953 33559286 NM_001080438
A4GALT 1 22 - 41418070 41446820 NM_017436 1 22 - 41418064 41446860 NM_017436
A4GNT 1 3 - 139325249 139333919 NM_016161 1 3 - 139325249 139333919 NM_016161
AAAS 2 12 - 51987506 52001679 NM_015665,NM_001173466 1 12 - 51987506 52001679 NM_015665
AACS 1 12 + 124115877 124193824 NM_023928 1 12 + 124115865 124193832 NM_023928
AADAC 1 3 + 153014550 153028966 NM_001086 1 3 + 153014550 153028966 NM_001086
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20130911_annnovar.hg18_refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.Condensed.txt | sort -k 1,1) | wc        
  17123  222599 1656178
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20130911_annnovar.hg18_refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.Condensed.txt | sort -k 1,1) | awk '{ if ($2 > $8) { print $0 } } ' | wc
   6518   84734  758697
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20130911_annnovar.hg18_refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.Condensed.txt | sort -k 1,1) | awk '{ if ($2 < $8) { print $0 } } ' | wc
     20     260    2616
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20130911_annnovar.hg18_refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.Condensed.txt | sort -k 1,1) | awk '{ if ($2 == $8) { print $0 } } ' | wc
  10585  137605  894865
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20130911_annnovar.hg18_refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.Condensed.txt | sort -k 1,1) | awk '{ if ($5 == $11) { print $0 } } ' | wc
  15501  201513 1489379
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20130911_annnovar.hg18_refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.Condensed.txt | sort -k 1,1) | awk '{ if ($5 < $11) { print $0 } } ' | wc
    839   10907   98550
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20130911_annnovar.hg18_refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.Condensed.txt | sort -k 1,1) | awk '{ if ($5 > $11) { print $0 } } ' | wc
    783   10179   68249
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20130911_annnovar.hg18_refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAl
ignIDs.Condensed.txt | sort -k 1,1) | awk '{ print $5 - $11 } ' | head -n 10
0
0
0
0
0
6
0
0
12
0
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20130911_annnovar.hg18_refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.Condensed.txt | sort -k 1,1) | awk '{ val1 = sqrt(($5 - $11)^2); if (val1 > 10000000) { print $0 } } ' | head -n 10
ARHGDIA 10 17 - 2385026 77422571 NM_001301240,NM_001185077,NM_001185077,NM_001185078,NM_004309,NM_001301242,NM_001301241,NM_001301241,NM_001301243,NR_125441 1 17 - 77418883 77422571 NM_004309
HLA-DQB1 4 6 - 3812911 32742444 NM_002123,NM_001243962,NM_001243962,NM_001243961 1 6 - 32735218 32742444 NM_002123
LOC283788 1 4 + 191098993 191120081 NR_027436 1 20 + 28225500 28247672 NR_003579
PRG2 4 11 - 56910835 56914706 NM_002728,NM_001302927,NM_001243245,NM_001302926 2 19 - 763487 56914706 NM_002728,NM_024888
RPL13A 3 19 + 54682622 54687376 NM_001270491,NM_012423,NR_073024 3 19 + 12919677 77826706 NR_003680,NR_003932,NM_012423
WASH2P 1 2 + 114057699 114073083 NR_024077 1 15 + 100318538 100334331 NR_003659
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20130911_annnovar.hg18_refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.Condensed.txt | sort -k 1,1) | awk '{ val1 = $5 - $11; if (val1 > 10000000) { print $0 } } '     
LOC283788 1 4 + 191098993 191120081 NR_027436 1 20 + 28225500 28247672 NR_003579
PRG2 4 11 - 56910835 56914706 NM_002728,NM_001302927,NM_001243245,NM_001302926 2 19 - 763487 56914706 NM_002728,NM_024888
RPL13A 3 19 + 54682622 54687376 NM_001270491,NM_012423,NR_073024 3 19 + 12919677 77826706 NR_003680,NR_003932,NM_012423
WASH2P 1 2 + 114057699 114073083 NR_024077 1 15 + 100318538 100334331 NR_003659
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20130911_annnovar.hg18_refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.Condensed.txt | sort -k 1,1) | awk '{ val1 = $5 - $11; if (val1 > 10000000) { print $0 } } ' | wc
      4      52     403
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20130911_annnovar.hg18_refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.Condensed.txt | sort -k 1,1) | awk '{ val1 = sqrt(($6 - $12)^2); if (val1 > 10000000) { print $0 } } '     
KIF22 4 16_random + 89350 29724207 NM_007317,NM_001256270,NM_001256270,NM_001256269 1 16_random + 89350 104032 NM_007317
LOC283788 1 4 + 191098993 191120081 NR_027436 1 20 + 28225500 28247672 NR_003579
RPL13A 3 19 + 54682622 54687376 NM_001270491,NM_012423,NR_073024 3 19 + 12919677 77826706 NR_003680,NR_003932,NM_012423
RPL31 3 2 + 100985122 101002587 NM_001098577,NM_001099693,NM_000993 4 2 + 100985122 159921666 NR_002595,NM_000993,NM_001098577,NM_001099693
WASH2P 1 2 + 114057699 114073083 NR_024077 1 15 + 100318538 100334331 NR_003659
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20130911_annnovar.hg18_refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.Condensed.txt | sort -k 1,1) | awk '{ print $5 - $11 } ' | R -q -e "Data1 <- read.table(file('stdin'), header=F); quantile(Data1[,1]); table(cut(Data1[,1], c(-1e6, -1e5, -1e4, -1e3, -1e2, 0, 1e2, 1e3, 1e4, 1e5, 1e6)));"
> Data1 <- read.table(file('stdin'), header=F); quantile(Data1[,1]); table(cut(Data1[,1], c(-1e6, -1e5, -1e4, -1e3, -1e2, 0, 1e2, 1e3, 1e4, 1e5, 1e6)));
       0%       25%       50%       75%      100%
-75033857         0         0         0 162873493

(-1e+06,-1e+05] (-1e+05,-1e+04] (-1e+04,-1e+03]   (-1e+03,-100]        (-100,0]
             32             205             237             218           15643
        (0,100]     (100,1e+03]   (1e+03,1e+04]   (1e+04,1e+05]   (1e+05,1e+06]
            476             155             134              12               2
>
>
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20130911_annnovar.hg18_refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.Condensed.txt | sort -k 1,1) | awk '{ print $5 - $11 } ' | R -q -e "Data1 <- read.table(file('stdin'), header=F); quantile(Data1[,1]); table(cut(Data1[,1], c(-1e7, -1e6, -1e5, -1e4, -1e3, -1e2, 0, 1e2, 1e3, 1e4, 1e5, 1e6, 1e7)));"
> Data1 <- read.table(file('stdin'), header=F); quantile(Data1[,1]); table(cut(Data1[,1], c(-1e7, -1e6, -1e5, -1e4, -1e3, -1e2, 0, 1e2, 1e3, 1e4, 1e5, 1e6, 1e7)));
       0%       25%       50%       75%      100%
-75033857         0         0         0 162873493

(-1e+07,-1e+06] (-1e+06,-1e+05] (-1e+05,-1e+04] (-1e+04,-1e+03]   (-1e+03,-100]
              3              32             205             237             218
       (-100,0]         (0,100]     (100,1e+03]   (1e+03,1e+04]   (1e+04,1e+05]
          15643             476             155             134              12
  (1e+05,1e+06]   (1e+06,1e+07]
              2               0
>
>
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20130911_annnovar.hg18_refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.Condensed.txt | sort -k 1,1) | awk '{ print $6 - $12 } ' | R -q -e "Data1 <- read.table(file('stdin'), header=F); quantile(Data1[,1]); table(cut(Data1[,1], c(-1e7, -1e6, -1e5, -1e4, -1e3, -1e2, 0, 1e2, 1e3, 1e4, 1e5, 1e6, 1e7)));"
> Data1 <- read.table(file('stdin'), header=F); quantile(Data1[,1]); table(cut(Data1[,1], c(-1e7, -1e6, -1e5, -1e4, -1e3, -1e2, 0, 1e2, 1e3, 1e4, 1e5, 1e6, 1e7)));
       0%       25%       50%       75%      100% 
-58919079         0         0         0 162872409 

(-1e+07,-1e+06] (-1e+06,-1e+05] (-1e+05,-1e+04] (-1e+04,-1e+03]   (-1e+03,-100] 
              1               3               9             135             153 
       (-100,0]         (0,100]     (100,1e+03]   (1e+03,1e+04]   (1e+04,1e+05] 
          16027             116             235             199             196 
  (1e+05,1e+06]   (1e+06,1e+07] 
             43               1 
> 
> 
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20130911_annnovar.hg18_refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.Condensed.txt | sort -k 1,1) | awk '{ if ($3 == $9) { print $0 } } ' | wc
  17033  221429 1639533
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20130911_annnovar.hg18_refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.Condensed.txt | sort -k 1,1) | awk '{ if ($3 != $9) { print $0 } } ' | wc
     90    1170   16645
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20130911_annnovar.hg18_refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.Condensed.txt | sort -k 1,1) | awk '{ if ($3 != $9) { print $0 } } ' | head -n 10
ABCF1 6 6_cox_hap1 + 1786022 30667288 NM_001025091,NM_001025091,NM_001025091,NM_001090,NM_001090,NM_001090 6 6_qbl_hap2 + 1786022 30667288 NM_001025091,NM_001090,NM_001025091,NM_001090,NM_001025091,NM_001090
ANAPC11 15 17 + 2409115 77451701 NM_001002248,NM_001289417,NM_001289416,NM_001289415,NM_001289414,NM_001002247,NM_001002246,NM_001002245,NM_001002244,NM_001002244,NM_016476,NM_001002249,NM_001289419,NM_001289420,NM_001289418 8 17_random + 2409115 77451701 NM_001002249,NM_016476,NM_001002248,NM_001002247,NM_001002246,NM_001002245,NM_001002244,NM_001002244
ASMT 6 X + 1674347 1721974 NM_004043,NM_001171039,NM_001171039,NM_001171038,NM_001171038,NM_004043 2 Y + 1674347 1721974 NM_004043,NM_004043
ATG4B 4 2 + 54821 242261944 NM_178326,NM_178326,NM_013325,NM_013325 4 2_random + 54821 242261944 NM_013325,NM_178326,NM_178326,NM_013325
BRD2 15 6 + 4122527 33057260 NM_001199455,NM_005104,NM_001199456,NM_001199456,NM_005104,NM_001113182,NM_005104,NM_001199455,NM_001113182,NM_001199455,NM_001113182,NM_001291986,NM_001199456,NM_001291986,NM_001291986 6 6_qbl_hap2 + 4122527 33057260 NM_005104,NM_001113182,NM_005104,NM_001113182,NM_005104,NM_001113182
C16orf54 2 16 - 41077 29664841 NM_175900,NM_175900 2 16_random - 41077 29664841 NM_175900,NM_175900
C2 18 6_cox_hap1 + 3113299 32021430 NM_000063,NM_001178063,NM_001282459,NM_001282459,NM_001282459,NM_001282458,NM_001282458,NM_001282458,NM_001282457,NM_001282457,NM_001282457,NM_001178063,NM_001145903,NM_000063,NM_000063,NM_001145903,NM_001145903,NM_001178063 3 6_qbl_hap2 + 3142993 32021430 NM_000063,NM_000063,NM_000063
C4A 6 6_cox_hap1 + 3197570 32111173 NM_001252204,NM_007293,NM_001252204,NM_001252204,NM_007293,NM_007293 3 6_qbl_hap2 + 3197570 32111173 NM_007293,NM_007293,NM_007293
C4B 8 6 + 3197570 32111174 NM_001242823,NM_001002029,NM_001242823,NM_001242823,NM_001002029,NM_001242823,NM_001002029,NM_001002029 4 6_qbl_hap2 + 3197570 32111174 NM_001002029,NM_001002029,NM_001002029,NM_001002029
C6orf25 18 6 + 2938697 31802464 NM_025260,NM_025260,NM_025260,NM_138277,NM_138277,NM_138277,NM_138275,NM_138272,NM_138273,NM_138273,NM_138273,NM_138272,NM_138272,NM_138275,NM_138274,NM_138274,NM_138275,NM_138274 18 6_qbl_hap2 + 2938697 31802464 NM_138273,NM_138274,NM_138277,NM_138272,NM_025260,NM_138275,NM_138275,NM_138274,NM_138273,NM_138272,NM_025260,NM_138277,NM_138275,NM_138274,NM_025260,NM_138272,NM_138273,NM_138277
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20130911_annnovar.hg18_refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.Condensed.txt | sort -k 1,1) | awk '{ if ($3 != $9) { print $0 } } ' | grep -v random | grep -v cox | grep -v hap | head -n 10
ASMT 6 X + 1674347 1721974 NM_004043,NM_001171039,NM_001171039,NM_001171038,NM_001171038,NM_004043 2 Y + 1674347 1721974 NM_004043,NM_004043
DHRSX 2 X - 2147552 2429015 NM_145177,NM_145177 2 Y - 2147552 2429015 NM_145177,NM_145177
IL3RA 4 X + 1415508 1461582 NM_001267713,NM_002183,NM_002183,NM_001267713 2 Y + 1415508 1461582 NM_002183,NM_002183
LOC283788 1 4 + 191098993 191120081 NR_027436 1 20 + 28225500 28247672 NR_003579
OR4F16 3 1 + 357521 180727832 NM_001005277,NM_001005277,NM_001005277 3 5 + 357521 180727832 NM_001005277,NM_001005277,NM_001005277
PPP2R3B 2 X - 214667 267690 NM_013239,NM_013239 2 Y - 214667 267690 NM_013239,NM_013239
PRG2 4 11 - 56910835 56914706 NM_002728,NM_001302927,NM_001243245,NM_001302926 2 19 - 763487 56914706 NM_002728,NM_024888
SLC25A6 2 X - 1465044 1471039 NM_001636,NM_001636 2 Y - 1465044 1471039 NM_001636,NM_001636
SPRY3 3 X + 57509790 154665313 NM_005840,NM_005840,NM_001304990 2 Y + 57509790 154665313 NM_005840,NM_005840
VAMP7 10 X + 57623336 154826627 NM_005638,NM_005638,NM_001145149,NM_001145149,NR_033715,NR_033714,NR_033714,NM_001185183,NM_001185183,NR_033715 2 Y + 57623336 154826627 NM_005638,NM_005638
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20130911_annnovar.hg18_refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170208.UCSCTableBrowser_refSeq_hg18_GeneAnnots_wSymbols.NoProtAlignIDs.Condensed.txt | sort -k 1,1) | awk '{ if ($3 != $9) { print $0 } } ' | grep -v random | grep -v cox | grep -v hap | wc        
     11     143    1239
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20130911_annnovar.hg18_refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20170208DL_20160204Vs_annnovar.hg18_refGene.Condensed.txt | sort -k 1,1) | wc
  26455  343915 2698450
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20130911_annnovar.hg18_refGene.Condensed.txt | wc
  26453  185171 1446264
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20130911_annnovar.hg18_refGene.Condensed.txt | wc
  26453  185171 1446264
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20130911_annnovar.hg18_refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20170208DL_20160204Vs_annnovar.hg18_refGene.Condensed.txt | sort -k 1,1) | awk '{ print $1 }' | sort | uniq -d
CAND1.11
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20130911_annnovar.hg18_refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20170208DL_20160204Vs_annnovar.hg18_refGene.Condensed.txt | sort -k 1,1) | grep CAND1.11
CAND1.11 1 11 + 10286435 10408796 NR_103765 1 11 + 10286435 10408796 NR_103765
CAND1.11 1 11 + 10286435 10408796 NR_103765 2 11 + 10286435 65994739 NM_018448,NR_103765
CAND1.11 2 11 + 10286435 65994739 NM_018448,NR_103765 1 11 + 10286435 10408796 NR_103765
CAND1.11 2 11 + 10286435 65994739 NM_018448,NR_103765 2 11 + 10286435 65994739 NM_018448,NR_103765
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20130911_annnovar.hg18_refGene.Condensed.txt | grep CAND1.11
CAND1.11        2       11      +       10286435        65994739        NM_018448,NR_103765
CAND1.11        1       11      +       10286435        10408796        NR_103765
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20130911_annnovar.hg18_refGene. | grep CAND1.11
20130911_annnovar.hg18_refGene.Condensed.txt     20130911_annnovar.hg18_refGene.GeneIDs.uniq.txt
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/gluster/home/mturchin20/Software/20130911_annovar/humandb/hg18_refGene.txt | grep CAND1.11
82      NR_103765       chr11   +       10286435        10408796        10408796        10408796        4       10286435,10308692,10388879,10408498,    10286796,10308832,10389099,10408796,    0       CAND1.11        unk   unk      -1,-1,-1,-1,
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/gluster/home/mturchin20/Software/20130911_annovar/humandb/hg18_refGene.txt | grep CAND1   
844     NM_033630       chr20   -       34004952        34005962        34005080        34005917        2       34004952,34005783,      34005675,34005962,      0       SCAND1  cmpl    cmpl    2,0,
1088    NM_018448       chr12   +       65949327        65994739        65949764        65992877        15      65949327,65961956,65972668,65975079,65977453,65977794,65978996,65982369,65984651,65985150,65987443,65989947,65990198,65991739,65992652,        65949832,65962100,65972823,65975203,65977710,65977900,65979142,65982662,65984793,65986644,65987539,65990117,65990363,65991847,65994739, 0       CAND1   cmpl    cmpl    0,2,2,1,2,1,2,1,0,1,1,1,0,0,0,
844     NM_016558       chr20   -       34004952        34006695        34005080        34005620        3       34004952,34005783,34006087,     34005675,34005823,34006695,     0       SCAND1  cmpl    cmpl    0,-1,-1,
82      NR_103765       chr11   +       10286435        10408796        10408796        10408796        4       10286435,10308692,10388879,10408498,    10286796,10308832,10389099,10408796,    0       CAND1.11        unk   unk      -1,-1,-1,-1,
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/gluster/home/mturchin20/Software/20130911_annovar/humandb/hg18_refGene.txt | grep [^-a-zA-Z0-9.]CAND1[^-a-zA-Z0-9\.]
1088    NM_018448       chr12   +       65949327        65994739        65949764        65992877        15      65949327,65961956,65972668,65975079,65977453,65977794,65978996,65982369,65984651,65985150,65987443,65989947,65990198,65991739,65992652,        65949832,65962100,65972823,65975203,65977710,65977900,65979142,65982662,65984793,65986644,65987539,65990117,65990363,65991847,65994739, 0       CAND1   cmpl    cmpl    0,2,2,1,2,1,2,1,0,1,1,1,0,0,0,
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/gluster/home/mturchin20/Software/20130911_annovar/humandb/hg18_refGene.txt | grep [^-a-zA-Z0-9.]CAND1[^-a-zA-Z0-9.]
1088    NM_018448       chr12   +       65949327        65994739        65949764        65992877        15      65949327,65961956,65972668,65975079,65977453,65977794,65978996,65982369,65984651,65985150,65987443,65989947,65990198,65991739,65992652,        65949832,65962100,65972823,65975203,65977710,65977900,65979142,65982662,65984793,65986644,65987539,65990117,65990363,65991847,65994739, 0       CAND1   cmpl    cmpl    0,2,2,1,2,1,2,1,0,1,1,1,0,0,0,
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/gluster/home/mturchin20/Software/20130911_annovar/humandb/hg18_refGene.txt | grep [^-a-zA-Z0-9\.]CAND1[^-a-zA-Z0-9.]
1088    NM_018448       chr12   +       65949327        65994739        65949764        65992877        15      65949327,65961956,65972668,65975079,65977453,65977794,65978996,65982369,65984651,65985150,65987443,65989947,65990198,65991739,65992652,        65949832,65962100,65972823,65975203,65977710,65977900,65979142,65982662,65984793,65986644,65987539,65990117,65990363,65991847,65994739, 0       CAND1   cmpl    cmpl    0,2,2,1,2,1,2,1,0,1,1,1,0,0,0,
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/gluster/home/mturchin20/Software/20130911_annovar/humandb/hg18_refGene.txt | grep [^-a-zA-Z0-9\.]CAND1[^-a-zA-Z0-9\.]
1088    NM_018448       chr12   +       65949327        65994739        65949764        65992877        15      65949327,65961956,65972668,65975079,65977453,65977794,65978996,65982369,65984651,65985150,65987443,65989947,65990198,65991739,65992652,        65949832,65962100,65972823,65975203,65977710,65977900,65979142,65982662,65984793,65986644,65987539,65990117,65990363,65991847,65994739, 0       CAND1   cmpl    cmpl    0,2,2,1,2,1,2,1,0,1,1,1,0,0,0,
[  mturchin20@bigmem01  ~/TempStuff]$echo -e "nana.1\nnana1" | grep a. 
nana.1
nana1
[  mturchin20@bigmem01  ~/TempStuff]$echo -e "nana.1\nnana1" | grep a\.
nana.1
nana1
[  mturchin20@bigmem01  ~/TempStuff]$echo -e "nana.1\nnana1" | grep a\\.
nana.1
[  mturchin20@bigmem01  ~/TempStuff]$echo -e "nana.1\nnana1" | grep a[^.]
nana.1
nana1
[  mturchin20@bigmem01  ~/TempStuff]$echo -e "nana.1\nnana1" | grep a[^\.]
nana.1
nana1
[  mturchin20@bigmem01  ~/TempStuff]$echo -e "nana.1\nnana1" | grep a[^\\.]
nana.1
nana1
[  mturchin20@bigmem01  ~/TempStuff]$echo -e "nana.1\nnana1" | grep ana[^\\.]
nana1
[  mturchin20@bigmem01  ~/TempStuff]$echo -e "nana.1\nnana1" | grep ana[^.]
nana1
[  mturchin20@bigmem01  ~/TempStuff]$echo -e "nana.1\nnana1" | grep ana. 
nana.1
nana1
[  mturchin20@bigmem01  ~/TempStuff]$echo -e "nana.1\nnana1" | grep ana\.
nana.1
nana1
[  mturchin20@bigmem01  ~/TempStuff]$echo -e "nana.1\nnana1" | grep ana\\.
nana.1
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20130911_annnovar.hg18_refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20170208DL_20160204Vs_annnovar.hg18_refGene.Condensed.txt | sort -k 1,1) | awk '{ if ($2 == $8) { print $0 } } ' | wc
  26453  343889 2698272
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20130911_annnovar.hg18_refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20170208DL_20160204Vs_annnovar.hg18_refGene.Condensed.txt | sort -k 1,1) | awk '{ if ($5 == $11) { print $0 } } ' | wc
  26455  343915 2698450
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20130911_annnovar.hg18_refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20170208DL_20160204Vs_annnovar.hg18_refGene.Condensed.txt | sort -k 1,1) | awk '{ if ($6 == $12) { print $0 } } ' | wc
  26453  343889 2698272
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20130911_annnovar.hg18_refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20170208DL_20160204Vs_annnovar.hg18_refGene.Condensed.txt | sort -k 1,1) | awk '{ if ($2 != $8) { print $0 } } '     
CAND1.11 1 11 + 10286435 10408796 NR_103765 2 11 + 10286435 65994739 NM_018448,NR_103765
CAND1.11 2 11 + 10286435 65994739 NM_018448,NR_103765 1 11 + 10286435 10408796 NR_103765
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20130911_annnovar.hg18_refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20170208DL_20160204Vs_annnovar.hg18_refGene.Condensed.txt | sort -k 1,1) | awk '{ if ($6 != $12) { print $0 } } ' 
CAND1.11 1 11 + 10286435 10408796 NR_103765 2 11 + 10286435 65994739 NM_018448,NR_103765
CAND1.11 2 11 + 10286435 65994739 NM_018448,NR_103765 1 11 + 10286435 10408796 NR_103765
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170210.wgetDirect.refSeq_hg18.refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.txt | sort -k 1,1) | wc
  27162  353106 2939436
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170210.wgetDirect.refSeq_hg18.refGene.Condensed.txt | wc
  27183  190281 1570715
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.txt | wc
  27293  191051 1578049
[  mturchin20@bigmem01  ~/TempStuff]$join -v 1 <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170210.wgetDirect.refSeq_hg18.refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.txt | sort -k 1,1) | wc
     21     147    1215
[  mturchin20@bigmem01  ~/TempStuff]$join -v 2 <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170210.wgetDirect.refSeq_hg18.refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.txt | sort -k 1,1) | wc
    131     917    6220
[  mturchin20@bigmem01  ~/TempStuff]$join -v 1 <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170210.wgetDirect.refSeq_hg18.refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.txt | sort -k 1,1) | head -n 10
C1QTNF9B-AS1 5 13 + 23361025 23364243 NM_001135816,NM_001014442,NM_001348113,NM_001348112,NM_001348114
CECR5 2 22 - 15998409 16026177 NM_033070,NM_017829
CRHR1-IT1 6 17 + 41053477 41079380 NR_027295,NR_026680,NR_138257,NR_138258,NR_138259,NR_138260
CRHR1-IT1-CRHR1 2 17 + 41053477 41268975 NM_001256299,NM_001303016
FAM150A 1 8 - 53609149 53640574 NM_207413
FAM150B 1 2 - 269560 278308 NM_001002919
FAM188A 2 10 - 15860173 15942538 NM_024948,NM_001318330
FAM188B 1 7 + 30777557 30898527 NM_032222
FAM63A 6 1 - 149235924 149247478 NM_001163260,NM_018379,NM_001163259,NM_001163258,NM_001319998,NM_001040217
FAM63B 2 15 + 56850682 56941390 NM_001040450,NM_001040453
[  mturchin20@bigmem01  ~/TempStuff]$join -v 2 <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170210.wgetDirect.refSeq_hg18.refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.txt | sort -k 1,1) | head -n 10
ALKAL1 1 8 - 53609149 53640574 NM_207413
ALKAL2 1 2 - 269560 278308 NM_001002919
HDHD5 2 22 - 15998409 16026177 NM_033070,NM_017829
INMT-MINDY4 1 7 + 30758275 30898527 NR_037598
LINC02210 6 17 + 41053477 41079380 NR_027295,NR_026680,NR_138257,NR_138258,NR_138259,NR_138260
LINC02210-CRHR1 2 17 + 41053477 41268975 NM_001303016,NM_001256299
LINC02211 1 5 + 25226818 25338146 NR_136209
LINC02212 1 5 - 10546638 10555840 NR_104606
LINC02213 2 5 - 10558101 10575196 NR_134289,NR_134290
LINC02214 1 5 - 116106896 116125804 NR_104678
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170210.wgetDirect.refSeq_hg18.refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.txt | sort -k 1,1) | awk '{ if ($2 == $8) { print $0 } } ' | wc
  27140  352820 2934656
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170210.wgetDirect.refSeq_hg18.refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.txt | sort -k 1,1) | awk '{ if ($2 != $8) { print $0 } } ' | wc
     22     286    4780
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170210.wgetDirect.refSeq_hg18.refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.txt | sort -k 1,1) | awk '{ if ($2 != $8) { print $0 } } ' | head -n 10
AKR7L 2 1 - 19465062 19473155 NR_040289,NR_040288 3 1 - 19465062 19473203 NM_001348421,NR_040288,NR_040289
AP3B2 3 15 - 81125087 81175714 NM_001278511,NM_004644,NM_001278512 5 15 - 81125087 81175714 NM_001278512,NM_004644,NM_001278511,NM_001348441,NM_001348440
AR 2 X + 66680598 66867186 NM_001011645,NM_000044 5 X + 66681189 66867186 NM_001348063,NM_001348061,NM_001348064,NM_000044,NM_001011645
CCDC85A 1 2 + 56264761 56466813 NM_001080433 6 2 + 56264761 56466813 NM_001080433,NM_001348512,NM_001348513,NM_001348514,NM_001348515,NM_001348516
CDH26 2 20 + 57966865 58021563 NM_177980,NM_021810 4 20 + 57966845 58022979 NM_177980,NR_145482,NM_001348204,NM_021810
CHD1L 6 1 + 145180914 145234071 NM_004284,NM_001256338,NR_046070,NM_001256337,NM_001256336,NM_024568 37 1 + 145111396 145234071 NM_001256336,NM_001256337,NM_001256338,NM_024568,NR_046070,NM_001348451,NM_001348453,NM_001348454,NM_001348455,NM_001348456,NM_001348457,NM_001348458,NM_001348459,NM_001348460,NM_001348461,NM_001348462,NM_001348463,NM_001348464,NM_001348465,NM_001348466,NM_004284,NR_145681,NR_145682,NR_145683,NR_145684,NR_145685,NR_145686,NR_145687,NR_145688,NR_145689,NR_145690,NR_145691,NR_145692,NR_145693,NR_145694,NR_145695,NM_001348452
DDX27 1 20 + 47269238 47294021 NM_017895 2 20 + 47269238 47294021 NM_017895,NM_001348187
DIMT1 1 5 - 61720107 61735485 NM_014473 3 5 - 61719536 61735544 NM_001348077,NM_014473,NM_001348076
DNAJC21 2 5 + 34965454 34994826 NM_001012339,NM_194283 3 5 + 34965454 34994826 NM_194283,NM_001012339,NM_001348420
GPR68 2 14 - 90768628 90789977 NM_001177676,NM_003485 3 14 - 90768628 90789977 NM_001177676,NM_003485,NM_001348437
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170210.wgetDirect.refSeq_hg18.refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.txt | sort -k 1,1) | awk '{ if ($2 < $8) { print $0 } } ' | wc        
     22     286    4780
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170210.wgetDirect.refSeq_hg18.refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.txt | sort -k 1,1) | awk '{ if ($5 == $11) { print $0 } } ' | wc
  27147  352911 2936858
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170210.wgetDirect.refSeq_hg18.refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.txt | sort -k 1,1) | awk '{ if ($6 == $12) { print $0 } } ' | wc
  27153  352989 2938356
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170210.wgetDirect.refSeq_hg18.refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.txt | sort -k 1,1) | awk '{ if ($5 < $11) { print $0 } } ' | wc
      2      26     811
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170210.wgetDirect.refSeq_hg18.refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.txt | sort -k 1,1) | awk '{ if ($5 > $11) { print $0 } } ' | wc
     13     169    1767
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170210.wgetDirect.refSeq_hg18.refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.txt | sort -k 1,1) | awk '{ if ($6 > $12) { print $0 } } ' | wc
      0       0       0
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170210.wgetDirect.refSeq_hg18.refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.txt | sort -k 1,1) | awk '{ if ($6 < $12) { print $0 } } ' | wc
      9     117    1080
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20170208DL_20160204Vs_annnovar.hg18_refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.txt | sort -k 1,1) | wc
  25935  337155 2739380
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20170208DL_20160204Vs_annnovar.hg18_refGene.Condensed.txt | wc
  26453  185171 1446264
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.txt | wc
  27293  191051 1578049
[  mturchin20@bigmem01  ~/TempStuff]$join -v 1 <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20170208DL_20160204Vs_annnovar.hg18_refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.txt | sort -k 1,1) | wc
    518    3626   26893
[  mturchin20@bigmem01  ~/TempStuff]$join -v 2 <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20170208DL_20160204Vs_annnovar.hg18_refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.txt | sort -k 1,1) | wc
   1359    9513   69053
[  mturchin20@bigmem01  ~/TempStuff]$join -v 1 <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20170208DL_20160204Vs_annnovar.hg18_refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.txt | sort -k 1,1) | head -n 10
ACRC 1 X + 70714598 70750158 NM_052957
ADCK3 1 1 + 225194560 225241869 NM_020247
ADCK4 2 19 - 45889273 45914630 NM_001142555,NM_024876
ADRBK1 1 11 + 66790480 66810605 NM_001619
ADRBK2 1 22 + 24290860 24455258 NM_005160
AOX2P 1 2 + 201268690 201367186 NR_001557
APITD1 2 1 + 10412745 10425459 NM_199294,NR_036462
APOA1BP 1 1 + 154828181 154830715 NM_144772
ASUN 1 12 - 26949378 26982521 NM_018164
ATHL1 1 11 + 279137 285688 NM_025092
[  mturchin20@bigmem01  ~/TempStuff]$join -v 2 <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20170208DL_20160204Vs_annnovar.hg18_refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.txt | sort -k 1,1) | head -n 10
ACBD7-DCLRE1CP1 1 10 - 15097076 15170781 NR_144471
ACOD1 1 13 + 76424624 76430777 NM_001258406
ACOXL-AS1 1 2 - 111572388 111589636 NR_122074
ADAM3B 1 16 - 48109057 48109997 NR_144553
AFDN 3 6 + 167970519 168115549 NM_001207008,NM_001040000,NM_001291964
AFDN-AS1 2 6_random - 811285 167970325 NR_027906,NR_027906
AKAIN1 2 18 - 5132909 5187751 NM_001145194,NM_001330553
ALKAL1 1 8 - 53609149 53640574 NM_207413
ALKAL2 1 2 - 269560 278308 NM_001002919
ANKRD18CP 1 9 - 98957995 99040845 NR_136286
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20170208DL_20160204Vs_annnovar.hg18_refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.txt | sort -k 1,1) | awk '{ if ($2 == $8) { print $0 } } ' | wc
  23648  307424 2405919
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20170208DL_20160204Vs_annnovar.hg18_refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.txt | sort -k 1,1) | awk '{ if ($2 < $8) { print $0 } } ' | wc
   2273   29549  330855
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20170208DL_20160204Vs_annnovar.hg18_refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.txt | sort -k 1,1) | awk '{ if ($2 > $8) { print $0 } } ' | wc
     14     182    2606
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20170208DL_20160204Vs_annnovar.hg18_refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.txt | sort -k 1,1) | awk '{ if ($5 == $11) { print $0 } } ' | wc
  24659  320567 2577833
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20170208DL_20160204Vs_annnovar.hg18_refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.txt | sort -k 1,1) | awk '{ if ($5 < $11) { print $0 } } ' | wc
     64     832    7619
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20170208DL_20160204Vs_annnovar.hg18_refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.txt | sort -k 1,1) | awk '{ if ($5 > $11) { print $0 } } ' | wc
   1212   15756  153928
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20170208DL_20160204Vs_annnovar.hg18_refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.txt | sort -k 1,1) | awk '{ if ($6 == $12) { print $0 } } ' | wc
  24694  321022 2580675
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20170208DL_20160204Vs_annnovar.hg18_refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.txt | sort -k 1,1) | awk '{ if ($6 < $12) { print $0 } } ' | wc
   1194   15522  153087
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20170208DL_20160204Vs_annnovar.hg18_refGene.Condensed.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.txt | sort -k 1,1) | awk '{ if ($6 > $12) { print $0 } } ' | wc
     47     611    5618
~~~









#rm /mnt/lustre/home/mturchin20/Data/UCSC/20170105.UCSC_hg19_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.txt; for i in `cat /mnt/lustre/home/mturchin20/Data/UCSC/20170105.UCSC_hg19_GeneInformation_wSymbols.NoProtAlignIDs.txt | perl -lane '$F[11] = join("_", @F[11..$#F]); print join("\t", @F[0..11]);' | perl -lane 'print $F[$#F];' | sort | uniq | grep -v knownGene`; do cat /mnt/lustre/home/mturchin20/Data/UCSC/20170105.UCSC_hg19_GeneInformation_wSymbols.NoProtAlignIDs.txt | grep [^-a-zA-Z0-9]$i$ | perl -lane 'if ($. == 1) { $txStart = $F[3]; $txEnd = $F[4]; } else { if ($F[3] < $txStart) { $txStart = $F[3]; } if ($F[4] > $txEnd) { $txEnd = $F[4]; } } if (eof()) { print $F[$#F], "\t", $F[1], "\t", $F[2], "\t", $txStart, "\t", $txEnd; }' | sed 's/chr//g' >> /mnt/lustre/home/mturchin20/Data/UCSC/20170105.UCSC_hg19_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.txt; done
rm /mnt/lustre/home/mturchin20/Data/UCSC/20170118.UCSC_hg18_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.txt; for i in `cat /mnt/lustre/home/mturchin20/Data/UCSC/20170118.UCSC_hg18_GeneInformation_wSymbols.NoProtAlignIDs.txt | perl -lane '$F[11] = join("_", @F[11..$#F]); print join("\t", @F[0..11]);' | perl -lane 'print $F[$#F];' | sort | uniq | grep -v knownGene`; do cat /mnt/lustre/home/mturchin20/Data/UCSC/20170118.UCSC_hg18_GeneInformation_wSymbols.NoProtAlignIDs.txt | grep [^-a-zA-Z0-9]$i$ | perl -lane 'if ($. == 1) { $txStart = $F[3]; $txEnd = $F[4]; } else { if ($F[3] < $txStart) { $txStart = $F[3]; } if ($F[4] > $txEnd) { $txEnd = $F[4]; } } if (eof()) { print $F[$#F], "\t", $F[1], "\t", $F[2], "\t", $txStart, "\t", $txEnd; }' | sed 's/chr//g' >> /mnt/lustre/home/mturchin20/Data/UCSC/20170118.UCSC_hg18_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.txt; done
#Manually removed line of 'hg18.kgXref.geneSymbol  hg18.knownGene.om       hg18.knownGene.strand   hg18.knownGene.txStart  hg18.knownGene.txEnd' from /mnt/lustre/home/mturchin20/Data/UCSC/20170118.UCSC_hg18_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.txt

R -q -e "Data1 <- read.table(\"/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.ChrBPChrBP.txt\", header=T); Data2 <- read.table(\"/mnt/lustre/home/mturchin20/Data/UCSC/20170105.UCSC_hg19_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.txt\", header=F); \
#Data1[,3] <- as.numeric(as.character(Data1[,3])); \
#Data2[,4] <- as.numeric(as.character(Data2[,4])); \
#Data2[,5] <- as.numeric(as.character(Data2[,5])); \
Data1 <- cbind(Data1, apply(Data1, 1, function(x) { NumOverlap <- nrow(Data2[x[,2] == Data2[,2] & x[,3] >= Data2[,4]+20000 & x[,3] <= Data2[,5]+20000,]); return(NumOverlap);})); \
print(head(Data1));"

wget http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/fetchChromSizes
mkdir /mnt/lustre/home/mturchin20/Software/fetchChromSizes
mv fetchChromSizes /mnt/lustre/home/mturchin20/Software/fetchChromSizes/.
bash /mnt/lustre/home/mturchin20/Software/fetchChromSizes/fetchChromSizes hg19 > /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg19.chrom.sizes
bash /mnt/lustre/home/mturchin20/Software/fetchChromSizes/fetchChromSizes hg18 > /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg18.chrom.sizes

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.ChrBPChrBP.txt | grep -v ChrBP | awk '{ print $2 "\t" $3 "\t" $3 }' | sort -k 1,1 -k 2,2 | awk '{ print "chr" $0 }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bed
cat /mnt/lustre/home/mturchin20/Data/UCSC/20170105.UCSC_hg19_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.txt | awk '{ print $2 "\t" $4 "\t" $5 }' | sort -k 1,1 -k 2,2 | awk '{ print "chr" $0 }' > /mnt/lustre/home/mturchin20/Data/UCSC/20170105.UCSC_hg19_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.bed
cat /mnt/lustre/home/mturchin20/Data/UCSC/20170105.UCSC_hg19_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.txt | awk '{ print $2 "\t" $4 "\t" $5 "\t" $1 }' | sort -k 1,1 -k 2,2 | awk '{ print "chr" $0 }' > /mnt/lustre/home/mturchin20/Data/UCSC/20170105.UCSC_hg19_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.wSymbol.bed
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/bedtools slop -i /mnt/lustre/home/mturchin20/Data/UCSC/20170105.UCSC_hg19_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.bed -g /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg19.chrom.sizes -b 20000 > /mnt/lustre/home/mturchin20/Data/UCSC/20170105.UCSC_hg19_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.20kbPadding.bed
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/bedtools slop -i /mnt/lustre/home/mturchin20/Data/UCSC/20170105.UCSC_hg19_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.wSymbol.bed -g /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg19.chrom.sizes -b 20000 > /mnt/lustre/home/mturchin20/Data/UCSC/20170105.UCSC_hg19_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.20kbPadding.wSymbol.bed
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/bedtools slop -i /mnt/lustre/home/mturchin20/Data/UCSC/20170105.UCSC_hg19_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.wSymbol.bed -g /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg19.chrom.sizes -b 250000 > /mnt/lustre/home/mturchin20/Data/UCSC/20170105.UCSC_hg19_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.250kbPadding.wSymbol.bed
##cat /mnt/lustre/home/mturchin20/Data/UCSC/20170105.UCSC_hg19_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.bed | awk '{ print $1 "\t" $2 - 20000 "\t" $3 + 20000 }' > /mnt/lustre/home/mturchin20/Data/UCSC/20170105.UCSC_hg19_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.20kbPadding.bed
cat /mnt/lustre/home/mturchin20/Data/UCSC/20170118.UCSC_hg18_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.txt | awk '{ print $2 "\t" $4 "\t" $5 }' | sort -k 1,1 -k 2,2 | awk '{ print "chr" $0 }' > /mnt/lustre/home/mturchin20/Data/UCSC/20170118.UCSC_hg18_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.bed
cat /mnt/lustre/home/mturchin20/Data/UCSC/20170118.UCSC_hg18_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.txt | awk '{ print $2 "\t" $4 "\t" $5 "\t" $1 }' | sort -k 1,1 -k 2,2 | awk '{ print "chr" $0 }' > /mnt/lustre/home/mturchin20/Data/UCSC/20170118.UCSC_hg18_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.wSymbol.bed
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/bedtools slop -i /mnt/lustre/home/mturchin20/Data/UCSC/20170118.UCSC_hg18_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.bed -g /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg18.chrom.sizes -b 20000 > /mnt/lustre/home/mturchin20/Data/UCSC/20170118.UCSC_hg18_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.20kbPadding.bed
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/bedtools slop -i /mnt/lustre/home/mturchin20/Data/UCSC/20170118.UCSC_hg18_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.wSymbol.bed -g /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg18.chrom.sizes -b 20000 > /mnt/lustre/home/mturchin20/Data/UCSC/20170118.UCSC_hg18_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.20kbPadding.wSymbol.bed
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/bedtools slop -i /mnt/lustre/home/mturchin20/Data/UCSC/20170118.UCSC_hg18_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.wSymbol.bed -g /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg18.chrom.sizes -b 250000 > /mnt/lustre/home/mturchin20/Data/UCSC/20170118.UCSC_hg18_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.250kbPadding.wSymbol.bed

/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -c -a /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170118.UCSC_hg18_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed.txt
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -c -a /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170118.UCSC_hg18_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.20kbPadding.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_20kbPadding.txt
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -wb -a /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170118.UCSC_hg18_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.wSymbol.bed | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.txt.gz
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -wb -a /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170118.UCSC_hg18_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.20kbPadding.wSymbol.bed | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.txt.gz
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -wb -a /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170118.UCSC_hg18_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.250kbPadding.wSymbol.bed | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_250kbPadding.txt.gz
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -c -a /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170118.UCSC_hg18_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.wSymbol.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Counts.txt
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -c -a /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170118.UCSC_hg18_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.20kbPadding.wSymbol.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Counts.txt
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -c -a /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170118.UCSC_hg18_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.250kbPadding.wSymbol.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_250kbPadding.Counts.txt

#20170106
#Found some duplicate entries in the original 2010 files due to use of HapMap references to get Chr & BP information per rsID# -- edited and dealt with this in the GlobalLipids2010 part of the code

#Code here -- if first line, create variables. if not first line, check if last line. If last line, check whether coords same -- if same, add last line then print, if not same, print, then print last line. If not last line, check whether coords same -- if same, add, if not same, print, then recreate variables
```
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.txt.gz | sed 's/chr//g' | sort -g -k 1,1 -k 2,2 | 
perl -lane 'if ($. == 1) { $ChrBP1 = $F[0] . "_" . $F[1]; @loci = ($F[$#F]); } 
else { $ChrBP2 = $F[0] . "_" . $F[1]; if (eof()) { if ($ChrBP1 eq $ChrBP2) { push(@loci, $F[$#F]); print $ChrBP1, "\t", scalar(@loci), "\t", join(",", @loci); } else { print $ChrBP1, "\t", scalar(@loci), "\t", join(",", @loci); print $ChrBP2, "\t1\t", $F[$#F];} } 
else { if ($ChrBP1 eq $ChrBP2) { push(@loci, $F[$#F]); } else { print $ChrBP1, "\t", scalar(@loci), "\t", join(",", @loci); $ChrBP1 = $F[0] . "_" . $F[1]; @loci = ($F[$#F]); } } }' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Condensed.txt.gz
```
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.txt.gz | sed 's/chr//g' | sort -g -k 1,1 -k 2,2 | perl -lane 'if ($. == 1) { $ChrBP1 = $F[0] . "_" . $F[1]; @loci = ($F[$#F]); } else { $ChrBP2 = $F[0] . "_" . $F[1]; if (eof()) { if ($ChrBP1 eq $ChrBP2) { push(@loci, $F[$#F]); print $ChrBP1, "\t", scalar(@loci), "\t", join(",", @loci); } else { print $ChrBP1, "\t", scalar(@loci), "\t", join(",", @loci); print $ChrBP2, "\t1\t", $F[$#F];} } else { if ($ChrBP1 eq $ChrBP2) { push(@loci, $F[$#F]); } else { print $ChrBP1, "\t", scalar(@loci), "\t", join(",", @loci); $ChrBP1 = $F[0] . "_" . $F[1]; @loci = ($F[$#F]); } } }' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Condensed.txt.gz
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.txt.gz | sed 's/chr//g' | sort -g -k 1,1 -k 2,2 | perl -lane 'if ($. == 1) { $ChrBP1 = $F[0] . "_" . $F[1]; @loci = ($F[$#F]); } else { $ChrBP2 = $F[0] . "_" . $F[1]; if (eof()) { if ($ChrBP1 eq $ChrBP2) { push(@loci, $F[$#F]); print $ChrBP1, "\t", scalar(@loci), "\t", join(",", @loci); } else { print $ChrBP1, "\t", scalar(@loci), "\t", join(",", @loci); print $ChrBP2, "\t1\t", $F[$#F];} } else { if ($ChrBP1 eq $ChrBP2) { push(@loci, $F[$#F]); } else { print $ChrBP1, "\t", scalar(@loci), "\t", join(",", @loci); $ChrBP1 = $F[0] . "_" . $F[1]; @loci = ($F[$#F]); } } }' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Condensed.txt.gz
zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_250kbPadding.txt.gz | sed 's/chr//g' | sort -g -k 1,1 -k 2,2 | perl -lane 'if ($. == 1) { $ChrBP1 = $F[0] . "_" . $F[1]; @loci = ($F[$#F]); } else { $ChrBP2 = $F[0] . "_" . $F[1]; if (eof()) { if ($ChrBP1 eq $ChrBP2) { push(@loci, $F[$#F]); print $ChrBP1, "\t", scalar(@loci), "\t", join(",", @loci); } else { print $ChrBP1, "\t", scalar(@loci), "\t", join(",", @loci); print $ChrBP2, "\t1\t", $F[$#F];} } else { if ($ChrBP1 eq $ChrBP2) { push(@loci, $F[$#F]); } else { print $ChrBP1, "\t", scalar(@loci), "\t", join(",", @loci); $ChrBP1 = $F[0] . "_" . $F[1]; @loci = ($F[$#F]); } } }' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_250kbPadding.Condensed.txt.gz

join <(cat <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Condensed.txt.gz) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | sed 's/chr//g') | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Counts.txt | sed 's/chr//g' | awk '{ print $1 "_" $2 "\t" $4 }' | sort -k 1,1) | awk '{ if ($2 == $4) { print $0 } }' | wc
join <(cat <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Condensed.txt.gz) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | sed 's/chr//g') | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Counts.txt | sed 's/chr//g' | awk '{ print $1 "_" $2 "\t" $4 }' | sort -k 1,1) | awk '{ if ($2 == $4) { print $0 } }' | wc
join <(cat <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_250kbPadding.Condensed.txt.gz) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_250kbPadding.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | sed 's/chr//g') | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_250kbPadding.Counts.txt | sed 's/chr//g' | awk '{ print $1 "_" $2 "\t" $4 }' | sort -k 1,1) | awk '{ if ($2 == $4) { print $0 } }' | wc




#join <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.txt.gz | awk '{ print $1 "\t" $2 "\t" $3 "\t" $4 }' | sort -k 1,1) <(cat <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Condensed.txt.gz) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | sed 's/chr//g') | sort -k 1,1) | join - <(cat <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Condensed.txt.gz) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | sed 's/chr//g') | sort -k 1,1) | perl -lane 'my $prox1 = 0; my $prox2 = 0; if ($F[4] > 0) { $prox1 = 1; } if ($F[6] > 0) { $prox2 = 1; } print $F[0], "\t", $F[1], "\t", $F[2], "\t", $F[3], "\t", $prox1, "\t", $F[4], "\t", $prox2, "\t", $F[6]; ' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.AnnotatedForPerms.NoPad.20kbPad.vs1.txt.gz
join <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.txt.gz | awk '{ print $1 "\t" $2 "\t" $3 "\t" $4 }' | sort -k 1,1) <(cat <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Condensed.txt.gz) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | sed 's/chr//g') | sort -k 1,1) | join - <(cat <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Condensed.txt.gz) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | sed 's/chr//g') | sort -k 1,1) | join - <(cat <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_250kbPadding.Condensed.txt.gz) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_250kbPadding.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | sed 's/chr//g') | sort -k 1,1) | perl -lane 'my $prox1 = 0; my $prox2 = 0; my $prox3 = 0; if ($F[4] > 0) { $prox1 = 1; } if ($F[6] > 0) { $prox2 = 1; } if ($F[8] > 0) { $prox3 = 1; } print $F[0], "\t", $F[1], "\t", $F[2], "\t", $F[3], "\t", $prox1, "\t", $F[4], "\t", $prox2, "\t", $F[6], "\t", $prox3, "\t", $F[8]; ' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.AnnotatedForPerms.NoPad.20kbPad.250kbPad.vs1.txt.gz

cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.txt | awk '{ print "chr"$1 "\t" $2 "\t" $2 }' > /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bed
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/bedtools slop -i /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bed -g /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg18.chrom.sizes -b 20000 > /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.20kbPadding.bed
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/bedtools slop -i /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bed -g /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg18.chrom.sizes -b 250000 > /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.250kbPadding.bed

/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -wb -a /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170118.UCSC_hg18_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.wSymbol.bed | gzip > /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.txt.gz
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -wb -a /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170118.UCSC_hg18_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.20kbPadding.wSymbol.bed | gzip > /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.txt.gz
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -wb -a /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170118.UCSC_hg18_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.250kbPadding.wSymbol.bed | gzip > /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_250kbPadding.txt.gz
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -c -a /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170118.UCSC_hg18_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.wSymbol.bed > /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Counts.txt
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -c -a /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170118.UCSC_hg18_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.20kbPadding.wSymbol.bed > /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Counts.txt
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -c -a /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170118.UCSC_hg18_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.250kbPadding.wSymbol.bed > /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_250kbPadding.Counts.txt

zcat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.txt.gz | sed 's/chr//g' | sort -g -k 1,1 -k 2,2 | perl -lane 'if ($. == 1) { $ChrBP1 = $F[0] . "_" . $F[1]; @loci = ($F[$#F]); } else { $ChrBP2 = $F[0] . "_" . $F[1]; if (eof()) { if ($ChrBP1 eq $ChrBP2) { push(@loci, $F[$#F]); print $ChrBP1, "\t", scalar(@loci), "\t", join(",", @loci); } else { print $ChrBP1, "\t", scalar(@loci), "\t", join(",", @loci); print $ChrBP2, "\t1\t", $F[$#F];} } else { if ($ChrBP1 eq $ChrBP2) { push(@loci, $F[$#F]); } else { print $ChrBP1, "\t", scalar(@loci), "\t", join(",", @loci); $ChrBP1 = $F[0] . "_" . $F[1]; @loci = ($F[$#F]); } } }' | gzip > /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Condensed.txt.gz
zcat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.txt.gz | sed 's/chr//g' | sort -g -k 1,1 -k 2,2 | perl -lane 'if ($. == 1) { $ChrBP1 = $F[0] . "_" . $F[1]; @loci = ($F[$#F]); } else { $ChrBP2 = $F[0] . "_" . $F[1]; if (eof()) { if ($ChrBP1 eq $ChrBP2) { push(@loci, $F[$#F]); print $ChrBP1, "\t", scalar(@loci), "\t", join(",", @loci); } else { print $ChrBP1, "\t", scalar(@loci), "\t", join(",", @loci); print $ChrBP2, "\t1\t", $F[$#F];} } else { if ($ChrBP1 eq $ChrBP2) { push(@loci, $F[$#F]); } else { print $ChrBP1, "\t", scalar(@loci), "\t", join(",", @loci); $ChrBP1 = $F[0] . "_" . $F[1]; @loci = ($F[$#F]); } } }' | gzip > /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Condensed.txt.gz
zcat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_250kbPadding.txt.gz | sed 's/chr//g' | sort -g -k 1,1 -k 2,2 | perl -lane 'if ($. == 1) { $ChrBP1 = $F[0] . "_" . $F[1]; @loci = ($F[$#F]); } else { $ChrBP2 = $F[0] . "_" . $F[1]; if (eof()) { if ($ChrBP1 eq $ChrBP2) { push(@loci, $F[$#F]); print $ChrBP1, "\t", scalar(@loci), "\t", join(",", @loci); } else { print $ChrBP1, "\t", scalar(@loci), "\t", join(",", @loci); print $ChrBP2, "\t1\t", $F[$#F];} } else { if ($ChrBP1 eq $ChrBP2) { push(@loci, $F[$#F]); } else { print $ChrBP1, "\t", scalar(@loci), "\t", join(",", @loci); $ChrBP1 = $F[0] . "_" . $F[1]; @loci = ($F[$#F]); } } }' | gzip > /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_250kbPadding.Condensed.txt.gz

join <(cat <(zcat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Condensed.txt.gz) <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | sed 's/chr//g') | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Counts.txt | sed 's/chr//g' | awk '{ print $1 "_" $2 "\t" $4 }' | sort -k 1,1) | awk '{ if ($2 == $4) { print $0 } }' | wc
join <(cat <(zcat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Condensed.txt.gz) <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | sed 's/chr//g') | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Counts.txt | sed 's/chr//g' | awk '{ print $1 "_" $2 "\t" $4 }' | sort -k 1,1) | awk '{ if ($2 == $4) { print $0 } }' | wc
join <(cat <(zcat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_250kbPadding.Condensed.txt.gz) <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_250kbPadding.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | sed 's/chr//g') | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_250kbPadding.Counts.txt | sed 's/chr//g' | awk '{ print $1 "_" $2 "\t" $4 }' | sort -k 1,1) | awk '{ if ($2 == $4) { print $0 } }' | wc

join <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.MarkerChrBPMAF.noNA.NArplcd.txt | awk '{ print $2 "_" $3 "\t" $2 "\t" $3 "\t" $4 }' | sort -k 1,1) <(cat <(zcat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Condensed.txt.gz) <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | sed 's/chr//g') | sort -k 1,1) | join - <(cat <(zcat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Condensed.txt.gz) <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | sed 's/chr//g') | sort -k 1,1) | join - <(cat <(zcat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_250kbPadding.Condensed.txt.gz) <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_250kbPadding.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | sed 's/chr//g') | sort -k 1,1) | perl -lane 'my $prox1 = 0; my $prox2 = 0; my $prox3 = 0; if ($F[4] > 0) { $prox1 = 1; } if ($F[6] > 0) { $prox2 = 1; } if ($F[8] > 0) { $prox3 = 1; } print $F[0], "\t", $F[1], "\t", $F[2], "\t", $F[3], "\t", $prox1, "\t", $F[4], "\t", $prox2, "\t", $F[6], "\t", $prox3, "\t", $F[8]; ' | gzip > /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.AnnotatedForPerms.NoPad.20kbPad.250kbPad.vs1.txt.gz

zcat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.AnnotatedForPerms.NoPad.20kbPad.250kbPad.vs1.txt.gz | head -n 10 > /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.AnnotatedForPerms.NoPad.20kbPad.250kbPad.vs1.txt.top10

mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses
mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/PermutationAnalyses 
mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/PermutationAnalyses/Permutations 

#python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataAnalysis.MatchSNPsBy.MAF_GeneProx.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.AnnotatedForPerms.NoPad.20kbPad.250kbPad.vs1.txt.top10 --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.AnnotatedForPerms.NoPad.20kbPad.250kbPad.vs1.txt.gz --N 10 --outfile1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/PermutationAnalyses/Permutations/TEST.20170113.GlobalLipids2010.Table1.bmass_GlobalLipids2010_Vs2_MergedDataSources_vs1_AnnotatedForPerms --seed 49734597345
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataAnalysis.MatchSNPsBy.MAF_GeneProx.vs1.py --file1 /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.AnnotatedForPerms.NoPad.20kbPad.250kbPad.vs1.txt.gz --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.AnnotatedForPerms.NoPad.20kbPad.250kbPad.vs1.txt.gz --N 10000 --outfile1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/PermutationAnalyses/Permutations/GlobalLipids2010.Table1.bmass_GlobalLipids2010_Vs2_MergedDataSources_vs1_AnnotatedForPerms.vs1 --seed 49734597345

zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/PermutationAnalyses/Permutations/GlobalLipids2010.Table1.bmass_GlobalLipids2010_Vs2_MergedDataSources_vs1_AnnotatedForPerms.vs1Permutation1.txt.gz | awk '{ print $11 }' | R -q -e "Data1 <- read.table(file('stdin'), header=F); quantile(Data1[,1]);"

#20170117 CHECK_0 -- Prob: Randomly check a few entries from permutations to see if chosen matched SNPs are as expected compared to original seed SNP?



cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.MarkerChrBP.noNA.NArplcd.txt | awk '{ print "chr" $2 "\t" $3 "\t" $3 }' > /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.MarkerChrBP.noNA.NArplcd.bed

mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/PermutationAnalyses/DataFileAnnotations 

#Two main files being used here
/mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.MarkerChrBP.noNA.NArplcd.bed
/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bed

do intersect, and then when joining together make into binary within the <(... | sort) of overall join functions

#locus defs
#/mnt/lustre/home/mturchin20/Data/UCSC/20170105.UCSC_hg19_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.wSymbol.bed
#/mnt/lustre/home/mturchin20/Data/UCSC/20170105.UCSC_hg19_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.20kbPadding.wSymbol.bed
/mnt/lustre/home/mturchin20/Data/UCSC/20170118.UCSC_hg18_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.wSymbol.bed
/mnt/lustre/home/mturchin20/Data/UCSC/20170118.UCSC_hg18_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.20kbPadding.wSymbol.bed

/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -c -a /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170118.UCSC_hg18_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.wSymbol.bed | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/PermutationAnalyses/DataFileAnnotations/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Counts.txt.gz
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -c -a /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170118.UCSC_hg18_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.20kbPadding.wSymbol.bed | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/PermutationAnalyses/DataFileAnnotations/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_20kbPadding_wSymbol.Counts.txt.gz
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -c -a /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.MarkerChrBP.noNA.NArplcd.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170118.UCSC_hg18_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.wSymbol.bed | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/PermutationAnalyses/DataFileAnnotations/GlobalLipids2010.Table1.MarkerChrBP.noNA.NArplcd.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Counts.txt.gz
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -c -a /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.MarkerChrBP.noNA.NArplcd.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170118.UCSC_hg18_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.20kbPadding.wSymbol.bed | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/PermutationAnalyses/DataFileAnnotations/GlobalLipids2010.Table1.MarkerChrBP.noNA.NArplcd.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_20kbPadding_wSymbol.Counts.txt.gz

<(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/PermutationAnalyses/DataFileAnnotations/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Counts.txt.gz | sed 's/chr//g' | perl -lane 'my $val1 = NA; if ($F[3] > 0) { $val1 = 1; } if ($F[3] == 0) { $val1 = 0; } print $F[0], "_", $F[1], "\t", $val1;' | sort -k 1,1) 


#nonsyn in merged (
join <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.Marker.txt | sort) <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.tbl.gz | sort) | awk '{ print $12, "\t", $13, "\t", $13, "\t", $2, "\t", $3 }' | sed 's/a/A/g' | sed 's/g/G/' | sed 's/c/C/g' | sed 's/t/T/g' | grep -v NA | cat - <(echo "9 133279427 133279427 T C") > /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.Marker.noNA.NArplcd.AnnovarFormat
##perl /mnt/lustre/home/mturchin20/Software/annovar/summarize_annovar.pl /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.Marker.noNA.NArplcd.AnnovarFormat /mnt/lustre/home/mturchin20/Software/annovar/humandb/ --ver1000g 1000g2012apr --verdbsnp 129 --veresp 6500 --buildver hg19 --remove
mv /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.Marker.noNA.NArplcd.AnnovarFormat.genome_summary.csv /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/OLD1.WRONG1.hg19.GlobalLipids2010.Table1.Marker.noNA.NArplcd.AnnovarFormat.genome_summary.csv
perl /mnt/lustre/home/mturchin20/Software/annovar/summarize_annovar.pl /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.Marker.noNA.NArplcd.AnnovarFormat /mnt/lustre/home/mturchin20/Software/annovar/humandb/ --ver1000g 1000g2010 --verdbsnp 129 --veresp 6500 --buildver hg18 --remove

/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.AnnovarFormat.genome_summary.csv
/mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.Marker.noNA.NArplcd.AnnovarFormat.genome_summary.csv

cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.Marker.noNA.NArplcd.AnnovarFormat.genome_summary.csv | grep -E 'nonsynonymous|splicing|stopgain|stoploss' | grep -v ncRNA | perl -F, -lane 'print "chr", $F[$#F-4], "\t", $F[$#F-3], "\t", $F[$#F-3];' > /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.Marker.noNA.NArplcd.AnnovarFormat.genome_summary.NonsynSpliceStopgainStoploss.no_ncRNA.bed
#Then use concatenated /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.Marker.noNA.NArplcd.AnnovarFormat.genome_summary.NonsynSpliceStopgainStoploss.no_ncRNA.bed & the below 'MergedDataSources' files to get the 'padding' files? The point would be to include any nonsyn SNPs that don't make it into the mergeddatasources file but are known gwas hit SNPs so they don't get lost when we expand. Eg
#/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -c -a file1 -b <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.Marker.noNA.NArplcd.AnnovarFormat.genome_summary.NonsynSpliceStopgainStoploss.no_ncRNA.bed /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.AnnovarFormat.genome_summary.NonsynSpliceStopgainStoploss.no_ncRNA.500bpPadding.bed) > ... 

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.AnnovarFormat.genome_summary.csv | grep -E 'nonsynonymous|splicing|stopgain|stoploss' | grep -v ncRNA | perl -F, -lane 'print "chr", $F[$#F-4], "\t", $F[$#F-3], "\t", $F[$#F-3];' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.AnnovarFormat.genome_summary.NonsynSpliceStopgainStoploss.no_ncRNA.bed
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/bedtools slop -i /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.AnnovarFormat.genome_summary.NonsynSpliceStopgainStoploss.no_ncRNA.bed -g /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg18.chrom.sizes -b 500 > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.AnnovarFormat.genome_summary.NonsynSpliceStopgainStoploss.no_ncRNA.500bpPadding.bed 
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/bedtools slop -i /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.AnnovarFormat.genome_summary.NonsynSpliceStopgainStoploss.no_ncRNA.bed -g /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg18.chrom.sizes -b 1000 > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.AnnovarFormat.genome_summary.NonsynSpliceStopgainStoploss.no_ncRNA.1kbPadding.bed 
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/bedtools slop -i /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.AnnovarFormat.genome_summary.NonsynSpliceStopgainStoploss.no_ncRNA.bed -g /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg18.chrom.sizes -b 10000 > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.AnnovarFormat.genome_summary.NonsynSpliceStopgainStoploss.no_ncRNA.10kbPadding.bed 
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/bedtools slop -i /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.AnnovarFormat.genome_summary.NonsynSpliceStopgainStoploss.no_ncRNA.bed -g /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg18.chrom.sizes -b 20000 > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.AnnovarFormat.genome_summary.NonsynSpliceStopgainStoploss.no_ncRNA.20kbPadding.bed 

/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -c -a /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bed -b /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.AnnovarFormat.genome_summary.NonsynSpliceStopgainStoploss.no_ncRNA.bed | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/PermutationAnalyses/DataFileAnnotations/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.AnnovarFormat_genome_summary_NonsynSpliceStopgainStoploss_no_ncRNA.Counts.txt.gz
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -c -a /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bed -b /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.AnnovarFormat.genome_summary.NonsynSpliceStopgainStoploss.no_ncRNA.500bpPadding.bed | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/PermutationAnalyses/DataFileAnnotations/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.AnnovarFormat_genome_summary_NonsynSpliceStopgainStoploss_no_ncRNA_500bpPadding.Counts.txt.gz
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -c -a /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bed -b /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.AnnovarFormat.genome_summary.NonsynSpliceStopgainStoploss.no_ncRNA.1kbPadding.bed | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/PermutationAnalyses/DataFileAnnotations/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.AnnovarFormat_genome_summary_NonsynSpliceStopgainStoploss_no_ncRNA_1kbPadding.Counts.txt.gz
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -c -a /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bed -b /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.AnnovarFormat.genome_summary.NonsynSpliceStopgainStoploss.no_ncRNA.10kbPadding.bed | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/PermutationAnalyses/DataFileAnnotations/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.AnnovarFormat_genome_summary_NonsynSpliceStopgainStoploss_no_ncRNA_10kbPadding.Counts.txt.gz
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -c -a /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bed -b /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.AnnovarFormat.genome_summary.NonsynSpliceStopgainStoploss.no_ncRNA.200kbPadding.bed | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/PermutationAnalyses/DataFileAnnotations/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.AnnovarFormat_genome_summary_NonsynSpliceStopgainStoploss_no_ncRNA_200kbPadding.Counts.txt.gz


#SNP/Marker all these qualities (yay/nay, so binary) -- a bunch of bedtools intersect files then just one major join file for the major 'annotation file' ? do counts then edit as yay/nay for the final, total join? 
#nhgri

mkdir /mnt/lustre/home/mturchin20/Data/NHGRI
cd /mnt/lustre/home/mturchin20/Data/NHGRI
wget https://www.ebi.ac.uk/gwas/api/search/downloads/alternative
wget https://www.ebi.ac.uk/gwas/api/search/downloads/studies_alternative
wget https://www.ebi.ac.uk/gwas/api/search/downloads/ancestry

mv /mnt/lustre/home/mturchin20/Data/NHGRI/alternative /mnt/lustre/home/mturchin20/Data/NHGRI/20170118Accessed_gwas_catalog_v1.0.1-associations_e87_r2017-01-09.tsv
mv /mnt/lustre/home/mturchin20/Data/NHGRI/studies_alternative /mnt/lustre/home/mturchin20/Data/NHGRI/20170118Accessed_gwas_catalog_v1.0.1-studies_r2017-01-09.tsv
mv /mnt/lustre/home/mturchin20/Data/NHGRI/ancestry /mnt/lustre/home/mturchin20/Data/NHGRI/20170118Accessed_gwas_catalog-ancestry_r2017-01-09.tsv

#From https://www.ebi.ac.uk/gwas/docs/file-downloads
```
.
.
.
File name conventions:
v1.0 indicates the columns originally proposed by the NCBI
v1.0.1 indicates the addition of ontology annotations and other new columns
eXX indicates the Ensembl release version that the data is mapped to
rYYYY-MM-DD indicates the date on which the GWAS Catalog was released
Older versions of the GWAS Catalog, with or without ontology annotations, can be found on our FTP server.
We are working to provide richer, fully customise-able downloads of GWAS data and these will be available soon.
.
.
.
```

cat /mnt/lustre/home/mturchin20/Data/NHGRI/20170118Accessed_gwas_catalog_v1.0.1-associations_e87_r2017-01-09.tsv | perl -F\\t -lane 'if (!$F[11]) { @altvals = split(/\:/, $F[21]); $F[11] = $altvals[0]; $F[12] = $altvals[1]; } my @traits = split(/\s+/, $F[7]); print join("_", @traits), "\t", $F[11], "\t", $F[12], "\t", $F[27];' | perl -lane 'if ($#F == 3) { print join("\t", @F); }' | R -q -e "Data1 <- read.table(file('stdin'), header=T); Data1 <- Data1[Data1[,4] < 5e-08,]; write.table(Data1, row.name=FALSE, quote=FALSE);" > /mnt/lustre/home/mturchin20/Data/NHGRI/20170118Accessed_gwas_catalog_v1.0.1-associations_e87_r2017-01-09.Edited.pVal_lt_5e-8.tsv
cat /mnt/lustre/home/mturchin20/Data/NHGRI/20170118Accessed_gwas_catalog_v1.0.1-associations_e87_r2017-01-09.Edited.pVal_lt_5e-8.tsv | grep -v \> | grep -v CHR_ID | awk '{ print $2 "\t" $3 "\t" $3 }' | sed 's/chr//g' > /mnt/lustre/home/mturchin20/Data/NHGRI/20170118Accessed_gwas_catalog_v1.0.1-associations_e87_r2017-01-09.Edited.pVal_lt_5e-8.bed

#From https://www.ebi.ac.uk/gwas/docs/file-downloads
```
The full NHGRI GWAS catalogue (http://www.genome.gov/26525384; accessed on 13th
January 2013) was downloaded and all SNPs associated with diseases and traits other than
anthropometric (height, BMI, Waist-Hip-Ratio and obesity) at genome-wide significance level
(P<5x10-8) were extracted. We next identified all SNPs in strong LD (r2 ≥ 0.8) with the 697
height-associated SNPs. Thirty-six of the 697 height SNPs tagged 45 NHGRI GWAS
catalogue associations, of which 9 were the height SNP itself (Supplementary Table 10).
Two height SNPs rs7980687 and rs8756 (two proxies at r2>0.8) were associated with
childhood head circumference and adult brain structure, where the height increasing allele
was also the trait increasing allele. Next we applied the same enrichment assessment
framework (as described in section 2.1.1 above) to estimate if the height SNPs tend to show
increased overlap with trait associated loci. Among the 10,000 random sets, we observed an
average of 13 (range 0 to 26) trait associations from the NHGRI GWAS catalogue tagged (r2≥ 0.8) by the matched SNPs compared to 33 with the pruned set of 628 height SNPs. The
observed distribution resulted in an empirical enrichment P-value <0.0001.
```

#Also see http://nar.oxfordjournals.org/content/45/D1/D896

#ldscore

mkdir /mnt/lustre/home/mturchin20/Data/PriceLab
mkdir /mnt/lustre/home/mturchin20/Data/PriceLab/LDScore
cd /mnt/lustre/home/mturchin20/Data/PriceLab/LDScore
wget https://data.broadinstitute.org/alkesgroup/LDSCORE/baseline_bedfiles.tgz
wget https://data.broadinstitute.org/alkesgroup/LDSCORE/ct_and_ctg_bedfiles.tgz
tar -xvzf /mnt/lustre/home/mturchin20/Data/PriceLab/LDScore/baseline_bedfiles.tgz
tar -xvzf /mnt/lustre/home/mturchin20/Data/PriceLab/LDScore/ct_and_ctg_bedfiles.tgz

rm -f /mnt/lustre/home/mturchin20/Data/PriceLab/LDScore/baseline/20170118.LDScore.RegulatoryRegions.Collapsed.bed
for i in `ls -lrt /mnt/lustre/home/mturchin20/Data/PriceLab/LDScore/baseline/. | grep -v 500 | grep -v UCSC | awk '{ print $9 }'`; do
	echo $i
	cat /mnt/lustre/home/mturchin20/Data/PriceLab/LDScore/baseline/$i >> /mnt/lustre/home/mturchin20/Data/PriceLab/LDScore/baseline/20170118.LDScore.RegulatoryRegions.Collapsed.bed
done

#From http://www.nature.com/ng/journal/v47/n11/full/ng.3404.html
```
.
.
.
To apply stratified LD score regression (or REML), we must first specify which categories are included in our model. We created a 'full baseline model' from 24 publicly available main annotations that are not specific to any cell type (Supplementary Table 1; see URLs and the Online Methods). Below, we show that including many categories in our model leads to more accurate estimates of enrichment. The 24 main annotations include coding, UTR, promoter and intronic regions14, 17; the histone marks monomethylation (H3K4me1) and trimethylation (H3K4me3) of histone H3 at lysine 4 (refs. 3,4,5), acetylation of histone H3 at lysine 9 (H3K9ac)3, 4, 5 and two versions of acetylation of histone H3 at lysine 27 (H3K27ac)18, 19; open chromatin, as reflected by DNase I hypersensitivity sites (DHSs)5, 14; combined chromHMM and Segway predictions20, which make use of many Encyclopedia of DNA Elements (ENCODE) annotations to produce a single partition of the genome into seven underlying chromatin states; regions that are conserved in mammals21, 22; super-enhancers, which are large clusters of highly active enhancers19; and enhancers with balanced bidirectional capped transcripts identified using cap analysis of gene expression in the FANTOM5 panel of samples, which we call FANTOM5 enhancers23. For histone marks and other annotations that differ among cell types, we combined the data from the different cell types into a single annotation for the full baseline model by taking a union (except for the repressed category, for which we took an intersection). To prevent our estimates from being biased upward by enrichment in nearby regions14, we also included 500-bp windows around each of the 24 main annotations in the full baseline model, as well as 100-bp windows around chromatin immunoprecipitation and sequencing (ChIP-seq) peaks when appropriate (Online Methods). This yielded a total of 53 (overlapping) functional categories in the full baseline model, including a category containing all SNPs.

In addition to the analyses using the full baseline model, we performed analyses using cell type–specific annotations for the four histone marks H3K4me1, H3K4me3, H3K9ac and H3K27ac. Each cell type–specific annotation corresponds to a histone mark in a single cell type—for example, H3K27ac in liver cells—and there were 220 such annotations in total (Online Methods and Supplementary Table 2). When generating these 220 cell type–specific annotations, we wanted to control for overlap with the functional categories in the full baseline model but not for overlap with the 219 other cell type–specific annotations. Thus, we added these annotations individually to the baseline model, creating 220 separate models, each with 54 annotations. Then, for a given phenotype, we ran LD score regression once each on the 220 models and ranked the cell type–specific annotations by the P value of the coefficient τC of the annotation in the corresponding analysis. This P value tests whether the annotation contributes significantly to SNP heritability after controlling for the effects of the annotations in the full baseline model.

We also divided the 220 cell type–specific annotations into 10 groups: adrenal and pancreas, CNS, cardiovascular, connective and bone, gastrointestinal, immune and hematopoietic, kidney, liver, skeletal muscle and other. We took a union of the cell type–specific annotations within each group, resulting in ten new cell type group annotations (for example, SNPs with any of the four histone modifications in any CNS cell type). We then repeated the cell type–specific analysis described above with these ten cell type groups instead of the 220 cell type–specific annotations.
.
.
.
Full baseline model
The 53 functional categories, derived from 24 main annotations, were obtained as follows:

Coding, 3′ UTR, 5′ UTR, promoter and intron annotations from RefSeq gene models were obtained from the UCSC Genome Browser17 and post-processed by Gusev et al.14.
Digital genomic footprint and transcription factor binding site annotations were obtained from ENCODE3 and post-processed by Gusev et al.14.
The combined chromHMM and Segway annotations for six cell lines were obtained from Hoffman et al.20. The CTCF, promoter-flanking, transcribed, transcription start site (TSS), strong enhancer and weak enhancer categories are each a union over the six cell lines; the repressed category is an intersection over the six cell lines.
DHSs are a combination of ENCODE and Roadmap Epigenomics data, post-processed by Trynka et al.5. We combined the cell type–specific annotations into two annotations for inclusion in the full baseline model: a union of all cell types and a union of only fetal cell types.
Cell type–specific H3K4me1, H3K4me3 and H3K9ac data were all obtained from Roadmap Epigenomics and post-processed by Trynka et al.5. For each mark, we took a union over cell types for the full baseline model and used the individual cell types for our cell type–specific analyses.
Cell type–specific H3K27ac data were obtained from Roadmap Epigenomics and post-processed18. A second version of H3K27ac data was obtained from Hnisz et al.19. For each mark, we took a union over cell types for the full baseline model. We also used the individual cell types of the Roadmap Epigenomics H3K27ac data for our cell type–specific analyses.
Super-enhancers were also obtained from Hnisz et al.19 and comprise a subset of the H3K27ac annotation from that paper. We took a union over cell types for the full baseline model.
Regions conserved in mammals were obtained from Lindblad-Toh et al.21 and post-processed by Ward and Kellis22.
FANTOM5 enhancers were obtained from Andersson et al.23.
For each of these 24 categories, we added a 500-bp window around the category as an additional category to keep our heritability estimates from being inflated by heritability in flanking regions14.
For each DHS, H3K4me1, H3K4me3 and H3K9ac site, we added a 100-bp window around the ChIP-seq peak as an additional category.
We added an additional category containing all SNPs.
When we report results in Supplementary Tables 5–7, we do not report results from the category containing all SNPs, as it has 100% of the heritability with zero standard error. (It might have a coefficient τc that is non-trivial, but in these tables we report proportions of heritability.)

According to our simulations (Fig. 2), including these 53 categories in our baseline model allows us to obtain unbiased or nearly unbiased estimates of enrichment for a wide range of potential new categories. To estimate the enrichment of a new annotation, we perform analyses using a model with these 53 annotations plus the new annotation. For example, for the cell type–specific analysis, we add each cell type–specific annotation to the baseline model, one at a time, and asses enrichment using the z score of the cell type–specific annotation.
.
.
.
```

#eqtl thing

mkdir /mnt/lustre/home/mturchin20/Data/GTEX/Vs6p
cd /mnt/lustre/home/mturchin20/Data/GTEX/Vs6p
wget http://www.gtexportal.org/static/datasets/gtex_analysis_v6p/single_tissue_eqtl_data/README_eQTL_v6p.txt
wget http://www.gtexportal.org/static/datasets/gtex_analysis_v6p/single_tissue_eqtl_data/GTEx_Analysis_v6p_eQTL.tar
tar -xvf /mnt/lustre/home/mturchin20/Data/GTEX/Vs6p/GTEx_Analysis_v6p_eQTL.tar
nbthis /mnt/lustre/home/mturchin20/Data/GTEX/Vs6p/GTEx_Analysis_v6p_eQTL.tar

#20170118 NOTE -- gtex shows SNP mapping as being based on build 'b37' which corresponds to hg19 (with some minor differences that shouldn't really affect things here)

zcat /mnt/lustre/home/mturchin20/Data/GTEX/Vs6p/GTEx_Analysis_v6p_eQTL/*signif*gz | grep -v variant | awk '{ print $1 }' | sed 's/_/ /g' | awk '{ print "chr" $1 "\t" $2 "\t" $2 }' > /mnt/lustre/home/mturchin20/Data/GTEX/Vs6p/GTEx_Analysis_v6p_eQTL/AllTraits_Analysis.v6p.signif_snpgene_pairs.bed
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/bedtools slop -i /mnt/lustre/home/mturchin20/Data/GTEX/Vs6p/GTEx_Analysis_v6p_eQTL/AllTraits_Analysis.v6p.signif_snpgene_pairs.bed -g /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg19.chrom.sizes -b 1000 > /mnt/lustre/home/mturchin20/Data/GTEX/Vs6p/GTEx_Analysis_v6p_eQTL/AllTraits_Analysis.v6p.signif_snpgene_pairs.1kbPadding.bed
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/bedtools slop -i /mnt/lustre/home/mturchin20/Data/GTEX/Vs6p/GTEx_Analysis_v6p_eQTL/AllTraits_Analysis.v6p.signif_snpgene_pairs.bed -g /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg19.chrom.sizes -b 5000 > /mnt/lustre/home/mturchin20/Data/GTEX/Vs6p/GTEx_Analysis_v6p_eQTL/AllTraits_Analysis.v6p.signif_snpgene_pairs.5kbPadding.bed
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/bedtools slop -i /mnt/lustre/home/mturchin20/Data/GTEX/Vs6p/GTEx_Analysis_v6p_eQTL/AllTraits_Analysis.v6p.signif_snpgene_pairs.bed -g /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg19.chrom.sizes -b 10000 > /mnt/lustre/home/mturchin20/Data/GTEX/Vs6p/GTEx_Analysis_v6p_eQTL/AllTraits_Analysis.v6p.signif_snpgene_pairs.10kbPadding.bed
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/bedtools slop -i /mnt/lustre/home/mturchin20/Data/GTEX/Vs6p/GTEx_Analysis_v6p_eQTL/AllTraits_Analysis.v6p.signif_snpgene_pairs.bed -g /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg19.chrom.sizes -b 20000 > /mnt/lustre/home/mturchin20/Data/GTEX/Vs6p/GTEx_Analysis_v6p_eQTL/AllTraits_Analysis.v6p.signif_snpgene_pairs.20kbPadding.bed

#mkdir /mnt/gluster/home/mturchin20/Data/GTEX
#mkdir /mnt/gluster/home/mturchin20/Data/GTEX/Vs6p
#mkdir /mnt/gluster/home/mturchin20/Data/GTEX/Vs6p/GTEx_Analysis_v6p_eQTL
#/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/bedtools slop -i /mnt/lustre/home/mturchin20/Data/GTEX/Vs6p/GTEx_Analysis_v6p_eQTL/AllTraits_Analysis.v6p.signif_snpgene_pairs.bed -g /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg19.chrom.sizes -b 5000 > /mnt/gluster/home/mturchin20/Data/GTEX/Vs6p/GTEx_Analysis_v6p_eQTL/AllTraits_Analysis.v6p.signif_snpgene_pairs.5kbPadding.bed
#mv /mnt/gluster/home/mturchin20/Data/GTEX/Vs6p/GTEx_Analysis_v6p_eQTL/AllTraits_Analysis.v6p.signif_snpgene_pairs.5kbPadding.bed /mnt/lustre/home/mturchin20/Data/GTEX/Vs6p/GTEx_Analysis_v6p_eQTL/AllTraits_Analysis.v6p.signif_snpgene_pairs.5kbPadding.bed
#/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/bedtools slop -i /mnt/lustre/home/mturchin20/Data/GTEX/Vs6p/GTEx_Analysis_v6p_eQTL/AllTraits_Analysis.v6p.signif_snpgene_pairs.bed -g /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg19.chrom.sizes -b 10000 > /mnt/gluster/home/mturchin20/Data/GTEX/Vs6p/GTEx_Analysis_v6p_eQTL/AllTraits_Analysis.v6p.signif_snpgene_pairs.10kbPadding.bed
#mv /mnt/gluster/home/mturchin20/Data/GTEX/Vs6p/GTEx_Analysis_v6p_eQTL/AllTraits_Analysis.v6p.signif_snpgene_pairs.10kbPadding.bed /mnt/lustre/home/mturchin20/Data/GTEX/Vs6p/GTEx_Analysis_v6p_eQTL/AllTraits_Analysis.v6p.signif_snpgene_pairs.10kbPadding.bed


#From http://www.gtexportal.org/home/datasets
```
.
.
.
Single Tissue cis-eQTL Data
Description
Name
Size
README file describing the contents of the eQTL files.	README_eQTL_v6p.txt	4.4K
eGene and significant snp-gene associations based on permutations.	GTEx_Analysis_v6p_eQTL.tar	782M
Every SNP-gene association test (including non-significant test). NOTE: This is a very large file and downloading will take hours. If you only need the significant associations, please download the file GTEx_Analysis_v6p_eQTL.tar instead. If you need every SNP gene association, but only for a particular tissue, then download the appropriate file in the Tissue-Specific All SNP Gene Associations fileset below.	GTEx_Analysis_v6p_all-associations.tar	108G
Fully processed, filtered and normalized gene expression matrices (in BED format) for each tissue, which were used as input into FastQTL for eQTL discovery.	GTEx_Analysis_v6p_eQTL_expression_matrices.tar	481M
Covariates used in eQTL analysis. Includes genotyping principal components and PEER factors.	GTEx_Analysis_v6p_eQTL_covariates.tar.gz	1.7M
Tissue-Specific All SNP Gene Associations
.
.
.
```

#omim genes (spec to giant2010/2014_5)

#From 


```
Simulations:		loci defs
-20kb locus		nonsyn in merged dataset
-locus		
-non-synonymous		nhgri gwas hits
-500bp non-syn		ldscore func reg categories
-1kb non-syn		eqtls based on pval thresh? Or just top hit in region?	
-10kb non-syn			
-20kn non-syn			
(-Wood et al. talks about being near non-synonymous but pretty sure not going to do this)			
			
-nhgri GWAS hit			
-nearby ldscore functional reg categories		https://data.broadinstitute.org/alkesgroup/LDSCORE/	https://data.broadinstitute.org/alkesgroup/LDSCORE/baseline_bedfiles.tgz
-eQTL for any tissue?			https://data.broadinstitute.org/alkesgroup/LDSCORE/ct_and_ctg_bedfiles.tgz

-eQTL specific tissues			
-ldscore cell-type specific categories			
-for height specifically OMIM list from wood 2014			

From wood 2014:	http://www.nature.com/ng/journal/v46/n11/full/ng.3097.html	http://www.nature.com/ng/journal/v46/n11/extref/ng.3097-S1.pdf	
Similar to the 180 variants identified in our previous analysis, the 697 variants were non-randomly distributed with respect to functional and putatively functional regions of the genome (Online Methods). We observed that the height-associated variants were enriched for nonsynonymous SNPs (empirical enrichment of 1.2-fold; P = 0.02), cis-regulatory effects in blood (empirical enrichment of 1.5-fold; P = 0.03), overlap with a curated list of genes that underlie monogenic syndromes of abnormal skeletal growth12 (empirical enrichment of 1.4-fold; P = 0.013), associations with apparently unrelated complex traits in the National Human Genome Research Institute (NHGRI) GWAS catalog (empirical enrichment of 2.6-fold; P < 1 × 10−4) and functional chromatin annotations in multiple tissues and cell types (empirical enrichment of 1.8-fold; P < 1 × 10−3) (Supplementary Tables 7,8,9,10,11 and Supplementary Note).			
```







~~~
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170105.UCSC_hg19_GeneInformation_wSymbols.NoProtAlignIDs.txt | perl -lane 'print $#F;' | sort | uniq -c
  82930 11
     11 12
      9 13
      6 14
      4 16
      1 17
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170105.UCSC_hg19_GeneInformation_wSymbols.NoProtAlignIDs.txt | perl -lane 'if ($#F == 12) { print join("\t", @F);}' | head -n 10
uc002stj.1      chr2    -       89156708        89165653        89156708        89156708        3       89156708,89161397,89165130,     89157196,89161595,89165653,     uc002stj.1      Ig      kappa
uc021zil.1      chr6    +       168265223       168265426       168265223       168265223       1       168265223,      168265426,      uc021zil.1      MLL/AF6 fusion
uc003tfw.3      chr7    -       38288826        38295990        38288826        38288826        2       38288826,38295937,      38289173,38295990,      uc003tfw.3      TCR     V3-JP2-Cgamma2
uc022bqu.2      chrM    +       10059   10404   10059   10059   1       10059,  10404,  uc022bqu.2      AD      1
uc022bqw.1      chrM    +       14856   15888   14856   14856   1       14856,  15888,  uc022bqw.1      cytochrome      b
uc022bqx.1      chrM    -       15959   16024   15959   15959   1       15959,  16024,  uc022bqx.1      tRNA    Pro
uc001rmp.1      chr12   +       41289751        41289889        41289751        41289751        1       41289751,       41289889,       uc001rmp.1      contactin       1
uc009zme.2      chr12   +       52473479        52502034        52473479        52473479        3       52473479,52477677,52501162,     52473965,52477752,52502034,     uc009zme.2      bpl     41-16
uc010axp.1      chr14   -       106064027       106064317       106064027       106064027       2       106064027,106064195,    106064111,106064317,    uc010axp.1      epsilon IgE
uc001ysd.1      chr14   -       106185758       106186048       106185758       106185758       2       106185758,106185926,    106185842,106186048,    uc001ysd.1      epsilon IgE
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170105.UCSC_hg19_GeneInformation_wSymbols.NoProtAlignIDs.txt | perl -lane '$F[11] = join("_", @F[11..$#F]); if ($#F > 11) { print join("\t", @F[0..11]); }' | wc
     31     372    3812
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170105.UCSC_hg19_GeneInformation_wSymbols.NoProtAlignIDs.txt | perl -lane '$F[11] = join("_", @F[11..$#F]); if ($#F > 11) { print join("\t", @F[0..11]); }' | head -n 10
uc002stj.1      chr2    -       89156708        89165653        89156708        89156708        3       89156708,89161397,89165130,     89157196,89161595,89165653,     uc002stj.1      Ig_kappa
uc021zil.1      chr6    +       168265223       168265426       168265223       168265223       1       168265223,      168265426,      uc021zil.1      MLL/AF6_fusion
uc003tfw.3      chr7    -       38288826        38295990        38288826        38288826        2       38288826,38295937,      38289173,38295990,      uc003tfw.3      TCR_V3-JP2-Cgamma2
uc003vxl.2      chr7    +       142028196       142028611       142028196       142028196       2       142028196,142028315,    142028226,142028611,    uc003vxl.2      T-cell_receptor_IGR_b_14_Vbeta_13
uc022ani.1      chr7    -       142161932       142162280       142161932       142161932       1       142161932,      142162280,      uc022ani.1      T-cell_receptor_IGR_b_16_Vbeta
uc022bqu.2      chrM    +       10059   10404   10059   10059   1       10059,  10404,  uc022bqu.2      AD_1
uc022bqw.1      chrM    +       14856   15888   14856   14856   1       14856,  15888,  uc022bqw.1      cytochrome_b
uc022bqx.1      chrM    -       15959   16024   15959   15959   1       15959,  16024,  uc022bqx.1      tRNA_Pro
uc011nad.1      chrX    -       155255322       155257848       155255322       155255322       3       155255322,155257000,155257494,  155256502,155257109,155257848,  uc011nad.1      DDX11L_family_member
uc011ncc.1      chrY    -       59358328        59360854        59358328        59358328        3       59358328,59360006,59360500,     59359508,59360115,59360854,     uc011ncc.1      DDX11L_family_member
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170105.UCSC_hg19_GeneInformation_wSymbols.NoProtAlignIDs.txt | perl -lane '$F[11] = join("_", @F[11..$#F]); print join("\t", @F[0..11]);' | perl -lane 'print $F[$#F];' | sort | uniq | grep uc
RNaseP_nuc
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170105.UCSC_hg19_GeneInformation_wSymbols.NoProtAlignIDs.txt | perl -lane '$F[11] = join("_", @F[11..$#F]); print join("\t", @F[0..11]);' | perl -lane 'print $F[$#F];' | sort | uniq | wc     
  28518   28518  210135
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join -v 1 <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170105.UCSC_hg19_GeneInformation_wSymbols.NoProtAlignIDs.txt | perl -lane '$F[11] = join("_", @F[11..$#F]); print join("\t", @F[0..11]);' | perl -lane 'print $F[$#F];' | sort | uniq) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/UCSC_hg19_GeneInformation_wSymbols.txt | perl -lane 'print $F[$#F];' | sort | uniq) | head -n 10
A1BG
A1BG-AS1
A2M
A2M-AS1
A2ML1
A2MP1
A4GALT
AA06
AAAS
AACS
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/UCSC/UCSC_hg19_GeneInformation_wSymbols.txt | grep AA06
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/UCSC/UCSC_hg19_GeneInformation_wSymbols.txt | grep AAAS
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join -v 1 <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170105.UCSC_hg19_GeneInformation_wSymbols.NoProtAlignIDs.txt | perl -lane '$F[11] = join("_", @F[11..$#F]); print join("\t", @F[0..11]);' | perl -lane 'print $F[$#F];' | sort | uniq) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/UCSC_hg19_GeneInformation_wSymbols.txt | perl -lane 'print $F[$#F];' | sort | uniq) | wc        
  12655   12655   94260
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$for i in `cat /mnt/lustre/home/mturchin20/Data/UCSC/20170105.UCSC_hg19_GeneInformation_wSymbols.NoProtAlignIDs.txt | perl -lane '$F[11] = join("_", @F[11..$#F]); print join("\t", @F[0..11]);' | perl -lane 'print $F[$#F];' | sort | uniq | head -n 10`; do cat /mnt/lustre/home/mturchin20/Data/UCSC/20170105.UCSC_hg19_GeneInformation_wSymbols.NoProtAlignIDs.txt | grep $i | perl -lane 'if ($. == 1) { $txStart = $F[3]; $txEnd = $F[4]; } else { if ($F[3] < $txStart) { $txStart = $F[3]; } if ($F[4] > $txEnd) { $txEnd = $F[4]; } } if (eof()) { print $F[$#F], "\t", $F[1], "\t", $F[2], "\t", $txStart, "\t", $txEnd; }'; done
5_8S_rRNA       chr9    -       111754688       111754849
5S_rRNA chr22   +       327954  242524490
6M1-18  chr6_ssto_hap7  -       698397  29424848
7M1-2   chr7    +       143656156       143657063
7SK chr22       -       1323794 237284409
A1BG-AS1        chr19   +       58858171        58874214
A1BG-AS1        chr19   +       58859116        58866549
A1CF chr10      -       52559168        52645435
A2MP1   chr12   -       8975149 9386803
A2M-AS1 chr12   +       9217772 9220651
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$for i in `cat /mnt/lustre/home/mturchin20/Data/UCSC/20170105.UCSC_hg19_GeneInformation_wSymbols.NoProtAlignIDs.txt | perl -lane '$F[11] = join("_", @F[11..$#F]); print join("\t", @F[0..11]);' | perl -lane 'print $F[$#F];' | sort | uniq | head -n 10`; do cat /mnt/lustre/home/mturchin20/Data/UCSC/20170105.UCSC_hg19_GeneInformation_wSymbols.NoProtAlignIDs.txt | grep -w $i | perl -lane 'if ($. == 1) { $txStart = $F[3]; $txEnd = $F[4]; } else { if ($F[3] < $txStart) { $txStart = $F[3]; } if ($F[4] > $txEnd) { $txEnd = $F[4]; } } if (eof()) { print $F[$#F], "\t", $F[1], "\t", $F[2], "\t", $txStart, "\t", $txEnd; }'; done
5_8S_rRNA       chr9    -       111754688       111754849
5S_rRNA chr22   +       327954  242524490
6M1-18  chr6_ssto_hap7  -       698397  29424848
7M1-2   chr7    +       143656156       143657063
7SK chr22       -       1323794 237284409
A1BG-AS1        chr19   +       58858171        58874214
A1BG-AS1        chr19   +       58859116        58866549
A1CF chr10      -       52559168        52645435
A2M chr12       -       9217772 9268558
A2M-AS1 chr12   +       9217772 9220651
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$for i in `cat /mnt/lustre/home/mturchin20/Data/UCSC/20170105.UCSC_hg19_GeneInformation_wSymbols.NoProtAlignIDs.txt | perl -lane '$F[11] = join("_", @F[11..$#F]); print join("\t", @F[0..11]);' | perl -lane 'print $F[$#F];' | sort | uniq | head -n 10`; do cat /mnt/lustre/home/mturchin20/Data/UCSC/20170105.UCSC_hg19_GeneInformation_wSymbols.NoProtAlignIDs.txt | grep [^-a-zA-Z0-9]$i[^-a-zA-Z0-9] | perl -lane 'if ($. == 1) { $txStart = $F[3]; $txEnd = $F[4]; } else { if ($F[3] < $txStart) { $txStart = $F[3]; } if ($F[4] > $txEnd) { $txEnd = $F[4]; } } if (eof()) { print $F[$#F], "\t", $F[1], "\t", $F[2], "\t", $txStart, "\t", $txEnd; }'; done
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170105.UCSC_hg19_GeneInformation_wSymbols.NoProtAlignIDs.txt | grep [^-_a-zA-Z0-9]A1BG$
uc002qsd.4      chr19   -       58858171        58864865        58858387        58864803        8       58858171,58858718,58861735,58862756,58863648,58864293,58864657,58864769,        58858395,58859006,58862017,58863053,58863921,58864563,58864693,58864865,        uc002qsd.4      A1BG
uc002qsf.2      chr19   -       58859831        58874214        58859831        58859831        11      58859831,58860933,58862756,58863648,58864293,58864657,58868950,58869317,58869854,58870562,58874042,     58860494,58862017,58863053,58863921,58864563,58864693,58869015,58869652,58869951,58870689,58874214,     uc002qsf.2      A1BG
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$for i in `cat /mnt/lustre/home/mturchin20/Data/UCSC/20170105.UCSC_hg19_GeneInformation_wSymbols.NoProtAlignIDs.txt | perl -lane '$F[11] = join("_", @F[11..$#F]); print join("\t", @F[0..11]);' | perl -lane 'print $F[$#F];' | sort | uniq | head -n 10`; do cat /mnt/lustre/home/mturchin20/Data/UCSC/20170105.UCSC_hg19_GeneInformation_wSymbols.NoProtAlignIDs.txt | grep [^-a-zA-Z0-9]$i$ | perl -lane 'if ($. == 1) { $txStart = $F[3]; $txEnd = $F[4]; } else { if ($F[3] < $txStart) { $txStart = $F[3]; } if ($F[4] > $txEnd) { $txEnd = $F[4]; } } if (eof()) { print $F[$#F], "\t", $F[1], "\t", $F[2], "\t", $txStart, "\t", $txEnd; }'; done
5_8S_rRNA       chr9    -       111754688       111754849
5S_rRNA chr22   +       327954  242524490
6M1-18  chr6_ssto_hap7  -       698397  29424848
7M1-2   chr7    +       143656156       143657063
7SK chr22       -       1323794 237284409
A1BG chr19      -       58858171        58874214
A1BG-AS1        chr19   +       58859116        58866549
A1CF chr10      -       52559168        52645435
A2M chr12       -       9220303 9268558
A2M-AS1 chr12   +       9217772 9220651
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170105.UCSC_hg19_GeneInformation_wSymbols.NoProtAlignIDs.txt | grep A1BG$
uc002qsd.4      chr19   -       58858171        58864865        58858387        58864803        8       58858171,58858718,58861735,58862756,58863648,58864293,58864657,58864769,        58858395,58859006,58862017,58863053,58863921,58864563,58864693,58864865,        uc002qsd.4      A1BG
uc002qsf.2      chr19   -       58859831        58874214        58859831        58859831        11      58859831,58860933,58862756,58863648,58864293,58864657,58868950,58869317,58869854,58870562,58874042,     58860494,58862017,58863053,58863921,58864563,58864693,58869015,58869652,58869951,58870689,58874214,     uc002qsf.2 A1BG
[  mturchin20@spudhead  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs1]$cat /mnt/lustre/home/mturchin20/Data/UCSC/UCSC_hg19_GeneInformation_wSymbols.txt | grep -v Known | perl -lane 'for (my $i = $#F-1; $i > 0; $i--) { if ($F[$i] =~ m/uc/) { $F[$i+1] = join("_", @F[$i+1..$#F]); print join("\t", @F[0..$i+1]); last;} }' | perl -lane 'print $#F;' | sort | uniq -c
  11409 12
  33585 13
[  mturchin20@spudhead  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs1]$cat /mnt/lustre/home/mturchin20/Data/UCSC/UCSC_hg19_GeneInformation_wSymbols.txt | perl -lane 'print $#F;' | sort | uniq -c
  11397 12
  33592 13
      3 14
      2 17
      1 18
      1 8
[  mturchin20@spudhead  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs1]$cat /mnt/lustre/home/mturchin20/Data/UCSC/UCSC_hg19_GeneInformation_wSymbols.txt | perl -lane 'if ($#F == 18) { print join("\t", @F); }'
uc003vxl.2      chr7    +       142028196       142028611       142028196       142028196       2       142028196,142028315,    142028226,142028611,    uc003vxl.2      uc003vxl.2      T-cell  receptor        IGR     b       14      Vbeta   13
[  mturchin20@spudhead  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs1]$cat /mnt/lustre/home/mturchin20/Data/UCSC/UCSC_hg19_GeneInformation_wSymbols.txt | grep -v Known | perl -lane 'for (my $i = $#F-1; $i > 0; $i--) { if ($F[$i] =~ m/uc/) { $F[$i+1] = join("_", @F[$i+1..$#F]); print join("\t", @F[0..$i+1]); last;} }' | grep Vbeta
uc003vxl.2      chr7    +       142028196       142028611       142028196       142028196       2       142028196,142028315,    142028226,142028611,    uc003vxl.2      uc003vxl.2      T-cell_receptor_IGR_b_14_Vbeta_13
uc022ani.1      chr7    -       142161932       142162280       142161932       142161932       1       142161932,      142162280,      uc022ani.1      uc022ani.1      T-cell_receptor_IGR_b_16_Vbeta
[  mturchin20@spudhead  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs1]$join -v 1 <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170105.UCSC_hg19_GeneInformation_wSymbols.NoProtAlignIDs.txt | perl -lane '$F[11] = join("_", @F[11..$#F]); print join("\t", @F[0..11]);' | perl -lane 'print $F[$#F];' | sort | uniq) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/UCSC_hg19_GeneInformation_wSymbols.txt | grep -v Known | perl -lane 'for (my $i = $#F-1; $i > 0; $i--) { if ($F[$i] =~ m/uc/) { $F[$i+1] = join("_", @F[$i+1..$#F]); print join("\t", @F[0..$i+1]); last;} }' | perl -lane 'print $F[$#F];' | sort | uniq ) | wc
  12645   12645   94065
[  mturchin20@spudhead  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs1]$join -v 1 <(cat /mnt/lustre/home/mturchin20/Data/UCSC/20170105.UCSC_hg19_GeneInformation_wSymbols.NoProtAlignIDs.txt | perl -lane '$F[11] = join("_", @F[11..$#F]); print join("\t", @F[0..11]);' | perl -lane 'print $F[$#F];' | sort | uniq) <(cat /mnt/lustre/home/mturchin20/Data/UCSC/UCSC_hg19_GeneInformation_wSymbols.txt | grep -v Known | perl -lane 'for (my $i = $#F-1; $i > 0; $i--) { if ($F[$i] =~ m/uc/) { $F[$i+1] = join("_", @F[$i+1..$#F]); print join("\t", @F[0..$i+1]); last;} }' | perl -lane 'print $F[$#F];' | sort | uniq ) | head -n 10
A1BG
A1BG-AS1
A2M
A2M-AS1
A2ML1
A2MP1
A4GALT
AA06
AAAS
AACS
[  mturchin20@spudhead  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs1]$cat /mnt/lustre/home/mturchin20/Data/UCSC/UCSC_hg19_GeneInformation_wSymbols.txt | grep AAAS
[  mturchin20@spudhead  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs1]$cat /mnt/lustre/home/mturchin20/Data/UCSC/UCSC_hg19_GeneInformation_wSymbols.txt | grep AACS
uc011dgk.2      chr5    -       178191863       178203277       178191863       178191863       7       178191863,178194130,178195470,178199429,178201498,178202267,178203142,  178193176,178194336,178195665,178199632,178201549,178202337,178203277,          uc011dgk.2      uc011dgk.2      AACSP1
uc011dgl.2      chr5    -       178191863       178245436       178191863       178191863       11      178191863,178194130,178195470,178199429,178201498,178202267,178203142,178224531,178238174,178241898,178245094,  178193176,178194336,178195668,178199632,178201549,178202337,178203277,178224612,178238287,178242009,178245436,             uc011dgl.2      uc011dgl.2      AACSP1
uc003mjk.3      chr5    -       178191863       178245436       178191863       178191863       11      178191863,178194130,178195470,178199429,178201498,178202267,178203142,178224531,178238174,178241898,178245094,  178193176,178194336,178195665,178199632,178201549,178202337,178203277,178224612,178238287,178242009,178245436,             uc003mjk.3      uc003mjk.3      AACSP1
[  mturchin20@spudhead  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs1]$cat /mnt/lustre/home/mturchin20/Data/UCSC/UCSC_hg19_GeneInformation_wSymbols.txt | grep -w AACS
[  mturchin20@spudhead  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs1]$
[  mturchin20@spudhead  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs1]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.AnnovarFormat.genome_summary.csv | perl -F, -lane 'my $flag1 = 0; if ($F[0] !~ m/(intergenic|ncRNA)/) { $flag1 = 1; } elsif ($F[0] =~ m/intergenic/) { if ($F[2] =~ m/dist=(\d+)/) { if ($1 <= 20000) { $flag1 = 1; } } if ($F[3] =~ m/dist=(\d+)/) { if ($1 <= 20000) { $flag1 = 1; } } } else { $PH = 1; } print $F[$#F-4], "\t", $F[$#F-3], "\t", $F[$#F-1], "\t", $F[$#F], "\t", $F[0], "\t", $F[1], "\t", $F[2], "\t", $flag1;' | perl -lane 'print $#F;' | sort | uniq -c
1117066 6
1456967 7
  26715 8
[  mturchin20@spudhead  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs1]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.AnnovarFormat.genome_summary.csv | perl -F, -lane 'my $flag1 = 0; if ($F[0] !~ m/(intergenic|ncRNA)/) { $flag1 = 1; } elsif ($F[0] =~ m/intergenic/) { if ($F[2] =~ m/dist=(\d+)/) { if ($1 <= 20000) { $flag1 = 1; } } if ($F[3] =~ m/dist=(\d+)/) { if ($1 <= 20000) { $flag1 = 1; } } } else { $PH = 1; } print $F[$#F-4], "\t", $F[$#F-3], "\t", $F[$#F-1], "\t", $F[$#F], "\t", $F[0], "\t", $F[1], "\t", $F[2], "\t", $flag1;' | perl -lane 'if ($#F == 8) { print join("\t", @F); }' | head -n 10
10      100020758       C       T       "exonic"        "LOXL4" "nonsynonymous  SNV"    1
10      100146980       A       T       "exonic"        "PYROXD2"       "nonsynonymous  SNV"    1
10      100150389       C       T       "exonic"        "PYROXD2"       "nonsynonymous  SNV"    1
10      100152753       A       G       "exonic"        "PYROXD2"       "synonymous     SNV"    1
10      100154979       T       C       "exonic"        "PYROXD2"       "synonymous     SNV"    1
10      100177443       G       A       "exonic"        "HPS1"  "nonsynonymous  SNV"    1
10      100183516       A       C       "exonic"        "HPS1"  "nonsynonymous  SNV"    1
10      100189612       C       T       "exonic"        "HPS1"  "nonsynonymous  SNV"    1
10      100219496       G       C       "exonic"        "HPSE2" "synonymous     SNV"    1
10      100374669       T       C       "exonic"        "HPSE2" "nonsynonymous  SNV"    1
[  mturchin20@spudhead  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs1]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.AnnovarFormat.genome_summary.csv | perl -F, -lane 'my $flag1 = 0; if ($F[0] !~ m/(intergenic|ncRNA)/) { $flag1 = 1; } elsif ($F[0] =~ m/intergenic/) { if ($F[2] =~ m/dist=(\d+)/) { if ($1 <= 20000) { $flag1 = 1; } } if ($F[3] =~ m/dist=(\d+)/) { if ($1 <= 20000) { $flag1 = 1; } } } else { $PH = 1; } print $F[$#F-4], "\t", $F[$#F-3], "\t", $F[$#F-1], "\t", $F[$#F], "\t", $F[0], "\t", $F[1], "\t", $F[2], "\t", $flag1;' | perl -lane 'if ($#F == 7) { print join("\t", @F); }' | head -n 10
Start   End     Obs     Otherinfo       Func    Gene    ExonicFunc      1
10      10000135        G       A       "intergenic"    "LINC00709(dist=671116) SFTA1P(dist=826267)"    0
10      10000265        T       C       "intergenic"    "LINC00709(dist=671246) SFTA1P(dist=826137)"    0
10      10000459        T       G       "intergenic"    "LINC00709(dist=671440) SFTA1P(dist=825943)"    0
10      100006186       C       G       "intergenic"    "R3HCC1L(dist=1532)     LOXL4(dist=1257)"       1
10      100006303       T       A       "intergenic"    "R3HCC1L(dist=1649)     LOXL4(dist=1140)"       1
10      100006329       T       C       "intergenic"    "R3HCC1L(dist=1675)     LOXL4(dist=1114)"       1
10      10000897        T       C       "intergenic"    "LINC00709(dist=671878) SFTA1P(dist=825505)"    0
10      10001617        A       G       "intergenic"    "LINC00709(dist=672598) SFTA1P(dist=824785)"    0
10      10001673        G       T       "intergenic"    "LINC00709(dist=672654) SFTA1P(dist=824729)"    0
[  mturchin20@spudhead  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs1]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.AnnovarFormat.genome_summary.csv | perl -F, -lane 'my $flag1 = 0; if ($F[0] !~ m/(intergenic|ncRNA)/) { $flag1 = 1; } elsif ($F[0] =~ m/intergenic/) { if ($F[2] =~ m/dist=(\d+)/) { if ($1 <= 20000) { $flag1 = 1; } } if ($F[3] =~ m/dist=(\d+)/) { if ($1 <= 20000) { $flag1 = 1; } } } else { $PH = 1; } print $F[$#F-4], "\t", $F[$#F-3], "\t", $F[$#F-1], "\t", $F[$#F], "\t", $F[0], "\t", $F[1], "\t", $F[2], "\t", $flag1;' | perl -lane 'if ($#F == 6) { print join("\t", @F); }' | head -n 10
10      100002729       G       A       "intronic"      "R3HCC1L"       1
10      100002880       G       A       "intronic"      "R3HCC1L"       1
10      100003553       T       C       "intronic"      "R3HCC1L"       1
10      100003805       T       C       "intronic"      "R3HCC1L"       1
10      100003967       A       T       "UTR3"  "R3HCC1L"       1
10      100005553       C       G       "downstream"    "R3HCC1L"       1
10      100005593       G       T       "downstream"    "R3HCC1L"       1
10      100007443       G       T       "UTR3"  "LOXL4" 1
10      100008579       G       T       "UTR3"  "LOXL4" 1
10      100009029       T       G       "intronic"      "LOXL4" 1
[  mturchin20@spudhead  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs1]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.AnnovarFormat.genome_summary.csv | perl -F, -lane 'my $flag1 = 0; if ($F[0] !~ m/(intergenic|ncRNA)/) { $flag1 = 1; } elsif ($F[0] =~ m/intergenic/) { if ($F[2] =~ m/dist=(\d+)/) { if ($1 <= 20000) { $flag1 = 1; } } if ($F[3] =~ m/dist=(\d+)/) { if ($1 <= 20000) { $flag1 = 1; } } } else { $PH = 1; } print $F[$#F-4], "\t", $F[$#F-3], "\t", $F[$#F-1], "\t", $F[$#F], "\t", $F[0], "\t", $F[1], "\t", $F[2], "\t", $flag1;' | grep dist | perl -lane 'my $val1 = join("_", @F); if ($val1 =~ m/.*dist=.*dist=(\d+).*/) { if (($1 < 20000) && ($1 > 15000)) { print join("\t", @F); } }' | head -n 10
10      100123436       G       A       "intergenic"    "LOXL4(dist=95429)      PYROXD2(dist=19886)"    1
10      100123921       C       A       "intergenic"    "LOXL4(dist=95914)      PYROXD2(dist=19401)"    1
10      100124026       C       T       "intergenic"    "LOXL4(dist=96019)      PYROXD2(dist=19296)"    1
10      100125512       T       C       "intergenic"    "LOXL4(dist=97505)      PYROXD2(dist=17810)"    1
10      100127818       T       C       "intergenic"    "LOXL4(dist=99811)      PYROXD2(dist=15504)"    1
10      101069132       G       A       "intergenic"    "HPSE2(dist=73500)      CNNM1(dist=19724)"      1
10      101069378       A       G       "intergenic"    "HPSE2(dist=73746)      CNNM1(dist=19478)"      1
10      101069484       T       A       "intergenic"    "HPSE2(dist=73852)      CNNM1(dist=19372)"      1
10      101069486       A       G       "intergenic"    "HPSE2(dist=73854)      CNNM1(dist=19370)"      1
10      101069780       T       C       "intergenic"    "HPSE2(dist=74148)      CNNM1(dist=19076)"      1
[  mturchin20@spudhead  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs1]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.AnnovarFormat.genome_summary.csv | perl -F, -lane 'my $flag1 = 0; if ($F[0] !~ m/(intergenic|ncRNA)/) { $flag1 = 1; } elsif ($F[0] =~ m/intergenic/) { if ($F[2] =~ m/dist=(\d+)/) { if ($1 <= 20000) { $flag1 = 1; } } if ($F[3] =~ m/dist=(\d+)/) { if ($1 <= 20000) { $flag1 = 1; } } } else { $PH = 1; } print $F[$#F-4], "\t", $F[$#F-3], "\t", $F[$#F-1], "\t", $F[$#F], "\t", $F[0], "\t", $F[1], "\t", $F[2], "\t", $flag1;' | grep dist | perl -lane 'my $val1 = join("_", @F); if ($val1 =~ m/.*dist=(\d+).*dist=.*/) { if (($1 < 20000) && ($1 > 15000)) { print join("\t", @F); } }' | head -n 10
10      100045634       C       T       "intergenic"    "LOXL4(dist=17627)      PYROXD2(dist=97688)"    0
10      100046035       T       C       "intergenic"    "LOXL4(dist=18028)      PYROXD2(dist=97287)"    0
10      100047451       G       A       "intergenic"    "LOXL4(dist=19444)      PYROXD2(dist=95871)"    0
10      100047697       T       C       "intergenic"    "LOXL4(dist=19690)      PYROXD2(dist=95625)"    0
10      100047807       C       T       "intergenic"    "LOXL4(dist=19800)      PYROXD2(dist=95515)"    0
10      101205892       A       T       "intergenic"    "GOT1(dist=15362)       NKX2-3(dist=86798)"     0
10      101208688       C       T       "intergenic"    "GOT1(dist=18158)       NKX2-3(dist=84002)"     0
10      101208889       C       T       "intergenic"    "GOT1(dist=18359)       NKX2-3(dist=83801)"     0
10      101209202       C       T       "intergenic"    "GOT1(dist=18672)       NKX2-3(dist=83488)"     0
10      101210050       C       T       "intergenic"    "GOT1(dist=19520)       NKX2-3(dist=82640)"     0
.
.
.
> system.time(Data1b <- cbind(head(Data1,n=1000), apply(head(Data1,n=1000), 1, function(x) { NumOverlap <- nrow(Data2[x[2] == Data2[,2] & x[3] >= Data2[,4]-20000 & x[3] <= Data2[,5]+20000,]); return(NumOverlap);})))
   user  system elapsed
100.034   0.000 100.041
> system.time(Data1b <- cbind(head(Data1,n=10000), apply(head(Data1,n=10000), 1, function(x) { NumOverlap <- nrow(Data2[x[2] == Data2[,2] & x[3] >= Data2[,4]-20000 & x[3] <= Data2[,5]+20000,]); return(NumOverlap);})))


Timing stopped at: 585.509 0 585.551
> Data1b <- cbind(head(Data1,n=10000), apply(head(Data1,n=10000), 1, function(x) { NumOverlap <- nrow(Data2[x[2] == Data2[,2] & x[3] >= Data2[,4]-20000 & x[3] <= Data2[,5]+20000,]); return(NumOverlap);}))
> head(Data1b)
         ChrBP Chr        BP apply(head(Data1, n = 10000), 1, function(x) {
1  10_10000135  10  10000135                                              0
2  10_10000265  10  10000265                                              0
3 10_100002729  10 100002729                                              0
4 10_100002880  10 100002880                                              0
5 10_100003553  10 100003553                                              0
6 10_100003805  10 100003805                                              0
.
.
.
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170105.UCSC_hg19_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.bed | awk '{ print $1 }' | sort | uniq -c
   2644 1
   1136 10
   1777 11
   1347 12
    540 13
    998 14
   1407 15
   1121 16
   1603 17
     26 17_ctg5_hap1
      1 17_gl000204_random
      1 17_gl000205_random
    441 18
   1853 19
     10 19_gl000209_random
      3 1_gl000191_random
      7 1_gl000192_random
   1703 2
    747 20
    352 21
    726 22
   1434 3
   1021 4
      5 4_ctg9_hap1
      1 4_gl000193_random
      3 4_gl000194_random
   1192 5
   1304 6
      1 6_apd_hap1
      7 6_cox_hap2
      2 6_dbb_hap3
     25 6_mann_hap4
      4 6_mcf_hap5
     20 6_qbl_hap6
    182 6_ssto_hap7
   1421 7
      3 7_gl000195_random
    970 8
   1185 9
      1 9_gl000201_random
     14 M
      2 Un_gl000211
      2 Un_gl000212
      3 Un_gl000213
      1 Un_gl000214
      2 Un_gl000218
      3 Un_gl000219
      6 Un_gl000220
      2 Un_gl000221
      4 Un_gl000222
      7 Un_gl000223
      1 Un_gl000227
      7 Un_gl000228
      1 Un_gl000229
      1 Un_gl000237
      1 Un_gl000241
      1 Un_gl000243
   1090 X
    117 Y
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170105.UCSC_hg19_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.20kbPadding.bed | awk '{ print $1 }' | sort | uniq -c
   2644 chr1
   1136 chr10
   1777 chr11
   1347 chr12
    540 chr13
    998 chr14
   1407 chr15
   1121 chr16
   1603 chr17
     26 chr17_ctg5_hap1
      1 chr17_gl000204_random
      1 chr17_gl000205_random
    441 chr18
   1853 chr19
     10 chr19_gl000209_random
      3 chr1_gl000191_random
      7 chr1_gl000192_random
   1703 chr2
    747 chr20
    352 chr21
    726 chr22
   1434 chr3
   1021 chr4
      5 chr4_ctg9_hap1
      1 chr4_gl000193_random
      3 chr4_gl000194_random
   1192 chr5
   1304 chr6
      1 chr6_apd_hap1
      7 chr6_cox_hap2
      2 chr6_dbb_hap3
     25 chr6_mann_hap4
      4 chr6_mcf_hap5
     20 chr6_qbl_hap6
    182 chr6_ssto_hap7
   1421 chr7
      3 chr7_gl000195_random
    970 chr8
   1185 chr9
      1 chr9_gl000201_random
     14 chrM
      2 chrUn_gl000211
      2 chrUn_gl000212
      3 chrUn_gl000213
      1 chrUn_gl000214
      2 chrUn_gl000218
      3 chrUn_gl000219
      6 chrUn_gl000220
      2 chrUn_gl000221
      4 chrUn_gl000222
      7 chrUn_gl000223
      1 chrUn_gl000227
      7 chrUn_gl000228
      1 chrUn_gl000229
      1 chrUn_gl000237
      1 chrUn_gl000241
      1 chrUn_gl000243
   1090 chrX
    117 chrY
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170105.UCSC_hg19_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.bed | head -n 10                         
chr1 10003485   10045556
chr1 100111430  100160097
chr1 100174258  100231349
chr1 100178485  100178513
chr1 100178485  100178513
chr1 100315639  100389579
chr1 100434000  100435404
chr1 100435344  100492534
chr1 100503788  100548929
chr1 100549101  100598511
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170105.UCSC_hg19_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.20kbPadding.bed | head -n 10
chr1 9983485    10065556
chr1 100091430  100180097
chr1 100154258  100251349
chr1 100158485  100198513
chr1 100158485  100198513
chr1 100295639  100409579
chr1 100414000  100455404
chr1 100415344  100512534
chr1 100483788  100568929
chr1 100529101  100618511
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170105.UCSC_hg19_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.bed | head -n 20000 | tail -n 10
chr4 118004709  118006736
chr4 118954772  119179789
chr4 1189570    1202750
chr4 119199916  119200978
chr4 119200344  128433548
chr4 119201192  119273922
chr4 1193485    1195204
chr4 119412048  243352032
chr4 119434236  119435886
chr4 119437494  119475359
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170105.UCSC_hg19_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.20kbPadding.bed | head -n 20000 | tail -n 10
chr4 117984709  118026736
chr4 118934772  119199789
chr4 1169570    1222750
chr4 119179916  119220978
chr4 119180344  128453548
chr4 119181192  119293922
chr4 1173485    1215204
chr4 119392048  191154276
chr4 119414236  119455886
chr4 119417494  119495359
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170105.UCSC_hg19_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.bed | tail -n 10
chrY 9167488    9172441
chrY 9175072    9347389
chrY 9175072    9368285
chrY 9175118    9218486
chrY 9187188    9362885
chrY 9195451    9198202
chrY 9207465    9322271
chrY 9225730    9233425
chrY 9448329    9452762
chrY 9638761    9650854
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170105.UCSC_hg19_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.20kbPadding.bed | tail -n 10
chrY 9147488    9192441
chrY 9155072    9367389
chrY 9155072    9388285
chrY 9155118    9238486
chrY 9167188    9382885
chrY 9175451    9218202
chrY 9187465    9342271
chrY 9205730    9253425
chrY 9428329    9472762
chrY 9618761    9670854
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_20kbPadding.txt | awk '{ print $4 }' | sort | uniq -c | head -n 50
 210611 0
 421840 1
  69700 10
      5 101
      8 102
      6 103
      5 104
      3 105
      4 106
      3 107
      4 108
      5 109
  61554 11
      9 110
      1 111
      3 112
      1 113
      1 114
      1 118
      1 119
  41382 12
      1 121
      2 123
      4 124
      2 125
      2 126
      3 128
      1 129
  41742 13
      2 130
      2 131
      1 132
      1 134
      2 135
      1 137
  41219 14
      1 143
  41052 15
  31319 16
  19402 17
  10988 18
   4412 19
 335202 2
   2311 20
   2034 21
   6355 22
   6261 23
   3620 24
   4869 25
  10179 26
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.txt.gz | awk '{ print $1 "\t" $2 "\t" $3 "\t" $4 }' | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_20kbPadding.txt | sed 's/chr//g' | awk '{ print $1 "_" $2 "\t" $1 "\t" $2 "\t" $4 }' | sort -k 1,1) | wc
3096191 21673337 135247332
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_20kbPadding.txt | sed 's/chr//g' | awk '{ print $1 "_" $2 "\t" $1 "\t" $2 "\t" $4 }' | awk '{ print $1 }' | wc
2600747 2600747 30283888
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_20kbPadding.txt | sed 's/chr//g' | awk '{ print $1 "_" $2 "\t" $1 "\t" $2 "\t" $4 }' | awk '{ print $1 }' | sort | uniq | wc
2596774 2596774 30237669
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_20kbPadding.txt | sed 's/chr//g' | awk '{ print $1 "_" $2 "\t" $1 "\t" $2 "\t" $4 }' | awk '{ print $1 }' | sort | uniq -d | head -n 10
10_107974310
10_44190021
1_151269919
1_181792961
1_181801167
1_181822914
1_243218867
1_244679662
13_104850499
13_27851104
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.txt.gz | grep 10_107974310 | wc
    128    2944   21440
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.No0Dir.tbl.gz | awk '{ print $1 "_" $2 }' | sort | uniq | wc
2627727 2627727 30597828
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.No0Dir.tbl.gz | awk '{ print $1 "_" $2 }' | sort | wc       
2692298 2692298 30985457
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.No0Dir.tbl.gz | awk '{ print $1 "_" $2 }' | sort | uniq -d | head -n 10
10_107974310
10_44190021
1_151269919
1_181792961
1_181801167
1_181822914
1_243218867
1_244679662
13_104850499
13_27851104
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.tbl.gz | head -n 10
MarkerName      Allele1 Allele2 Weight  GC.Zscore       GC.Pvalue       Overall Direction       Effect  StdErr  ChrBPAFInfo
rs10 a  c       85853   -0.096  0.9232  -       ---+-+++-+?+--??--++??+++       -0.0019 0.0147  chr7_92221824_A_0.033_C_0.967
rs1000000       a       g       99900   -0.937  0.349   -       ---+-+---++-+---+++-++++-       -0.0056 0.0058  chr12_125456933_G_0.627_A_0.373
rs10000010      t       c       99900   0.185   0.8532  +       -+++-+-+--+++-++++--+----       0.0013  0.0049  chr4_21227772_T_0.575_C_0.425
rs10000012      c       g       99843   -0.752  0.4523  -       +-+-+-+--+----+--+++++-++       -0.0065 0.0071  chr4_1347325_C_0.808_G_0.192
rs10000013      a       c       99900   0.642   0.5209  +       +++-++--+--++--+---+-++--       3e-04   0.0058  chr4_36901464_C_0.167_A_0.833
rs10000017      t       c       99900   2.043   0.04106 +       +-++++++---++++---++--+++       0.0105  0.0062  chr4_84997149_C_0.777_T_0.223
rs1000002       t       c       99797   2.249   0.02453 +       -+-+---+-+-+++--++-+-+-++       0.0101  0.0049  chr3_185118462_C_0.475_T_0.525
rs10000023      t       g       99900   -0.008  0.9935  -       -+-+--+--+++++--++------+       4e-04   0.005   chr4_95952929_G_0.408_T_0.592
rs10000027      c       g       7807    -0.686  0.4925  -       ?????????????-???????????       -0.0141 0.0205  chr4_156533822_T_0_G_0.267
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.tbl.gz | grep 107974310
rs35532567      t       c       7807    -0.735  0.4622  -       ?????????????-???????????       -0.0141 0.0191  chr10_107974310_C_0.308_T_0.692
rs3861988       t       c       92001   0.118   0.9062  +       +++-++--+-+-++++++-+-+-++       -0.0022 0.0052  chr10_107974310_C_0.695_T_0.305
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.No0Dir.tbl.gz | awk '{ print $1 "_" $2 }' | sort | uniq -u | wc
2627690 2627690 30597403
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.No0Dir.tbl.gz | awk '{ print $1 "_" $2 "\t" $0 }' | sort -k 1,1) <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.No0Dir.tbl.gz | awk '{ print $1 "_" $2 }' | sort | uniq -u) | wc
2627690 23649210 127286752
#20170106
#This is where I began to realize the duplication problem in the original GlobalLipids2010 files, and then dealt with it. Certain filenames remained the same afterwards, hence why different wc's for files with the same names below here.
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.txt.gz | awk '{ print $1 "\t" $2 "\t" $3 "\t" $4 }' | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_20kbPadding.txt | sed 's/chr//g' | awk '{ print $1 "_" $2 "\t" $1 "\t" $2 "\t" $4 }' | sort -k 1,1) | wc
2593966 18157762 113672104
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.txt.gz | wc
2593967 59661241 438572406
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.txt.gz | head -n 10
ChrBP Chr BP MAF A1 HDL_A2 HDL_Direction HDL_pValue HDL_N HDL_ZScore LDL_Direction LDL_pValue LDL_N LDL_ZScore TG_Direction TG_pValue TG_N TG_ZScore TC_Direction TC_pValue TC_N TC_ZScore GWASannot
10_10000135 10 10000135 0.449 g a + 0.411 99150 -0.822135118817414 - 0.9712 94704 0.0361032887447841 - 0.4344 95848 0.781684522266763 - 0.8399 99434 0.202021392729858 0
10_10000265 10 10000265 0.436 t c + 0.3933 99150 0.853648097868868 + 0.9683 94704 0.039740516098603 - 0.4232 95848 -0.800881933637729 - 0.8803 99434 -0.15058892288133 0
10_100002729 10 100002729 0.367 g a + 0.03387 99900 -2.12161597232425 - 0.03202 95454 2.14416085995566 - 0.04005 96598 2.05323284873217 - 0.1812 100184 1.33706935239798 0
10_100002880 10 100002880 0.358 g a + 0.03461 99900 -2.11289214901574 - 0.06781 95454 1.82626735507002 - 0.05027 96598 1.95765933421239 - 0.2776 100184 1.08572652902347 0
10_100003553 10 100003553 0.12 t c + 0.8488 99900 0.190649762933283 - 0.2675 95454 -1.1088381719739 - 0.1312 96598 -1.50938666045622 - 0.2265 100184 -1.20942398128564 0
10_100003805 10 100003805 0.3 t c + 0.1194 99900 1.55729694674125 - 0.1734 95454 -1.3613598428411 - 0.4822 96598 -0.702768549495165 - 0.6764 100184 -0.417380658208539 0
10_100003967 10 100003967 0.356 a t + 0.03739 99900 2.0814799219582 - 0.05586 95454 -1.91212627071685 - 0.05034 96598 -1.95706352623053 - 0.2426 100184 -1.16851274834855 0
10_10000459 10 10000459 0.427 t g + 0.4015 99150 0.838945310119727 - 0.9956 94704 -0.00551461015479343 - 0.4362 95848 -0.778626099569515 - 0.8693 99434 -0.164547699082322 0
10_100005553 10 100005553 0.144 c g + 0.1038 99900 1.62670390028542 + 0.4128 95454 0.818976186983752 - 0.549 96598 -0.599259276137027 + 0.3909 100184 0.857986351425547 0
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_20kbPadding.txt | wc
2593966 10375864 67617105
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.txt.gz | awk '{ print $1 "\t" $2 "\t" $3 "\t" $4 }' | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_20kbPadding.txt | sed 's/chr//g' | awk '{ print $1 "_" $2 "\t" $1 "\t" $2 "\t" $4 }' | sort -k 1,1) | head -n 10
10_10000135 10 10000135 0.449 10 10000135 0
10_10000265 10 10000265 0.436 10 10000265 0
10_100002729 10 100002729 0.367 10 100002729 6
10_100002880 10 100002880 0.358 10 100002880 6
10_100003553 10 100003553 0.12 10 100003553 6
10_100003805 10 100003805 0.3 10 100003805 6
10_100003967 10 100003967 0.356 10 100003967 6
10_10000459 10 10000459 0.427 10 10000459 0
10_100005553 10 100005553 0.144 10 100005553 6
10_100005593 10 100005593 0.217 10 100005593 6
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.txt.gz | awk '{ print $1 "\t" $2 "\t" $3 "\t" $4 }' | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed.txt | sed 's/chr//g' | awk '{ print $1 "_" $2 "\t" $1 "\t" $2 "\t" $4 }' | sort -k 1,1) | join - <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_20kbPadding.txt | sed 's/chr//g' | awk '{ print $1 "_" $2 "\t" $4 }' | sort -k 1,1) | head -n 10
10_10000135 10 10000135 0.449 10 10000135 0 0
10_10000265 10 10000265 0.436 10 10000265 0 0
10_100002729 10 100002729 0.367 10 100002729 5 6
10_100002880 10 100002880 0.358 10 100002880 5 6
10_100003553 10 100003553 0.12 10 100003553 5 6
10_100003805 10 100003805 0.3 10 100003805 5 6
10_100003967 10 100003967 0.356 10 100003967 5 6
10_10000459 10 10000459 0.427 10 10000459 0 0
10_100005553 10 100005553 0.144 10 100005553 4 6
10_100005593 10 100005593 0.217 10 100005593 4 6
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.txt.gz | awk '{ print $1 "\t" $2 "\t" $3 "\t" $4 }' | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed.txt | sed 's/chr//g' | awk '{ print $1 "_" $2 "\t" $1 "\t" $2 "\t" $4 }' | sort -k 1,1) | join - <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_20kbPadding.txt | sed 's/chr//g' | awk '{ print $1 "_" $2 "\t" $4 }' | sort -k 1,1) | sort -g -k 1,1 | head -n 10
1_100002699 1 100002699 0.3 1 100002699 0 0
1_100002785 1 100002785 0.0669999999999999 1 100002785 0 0
1_100003455 1 100003455 0.047 1 100003455 0 0
1_100006955 1 100006955 0.0669999999999999 1 100006955 0 0
1_100007331 1 100007331 0.283 1 100007331 0 0
1_100007519 1 100007519 0.0620000000000001 1 100007519 0 0
1_100007856 1 100007856 0.0580000000000001 1 100007856 0 0
1_100008014 1 100008014 0.217 1 100008014 0 0
1_100008025 1 100008025 0.143 1 100008025 0 0
1_100008439 1 100008439 0.0669999999999999 1 100008439 0 0
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.txt.gz | awk '{ print $1 "\t" $2 "\t" $3 "\t" $4 }' | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed.txt | sed 's/chr//g' | awk '{ print $1 "_" $2 "\t" $1 "\t" $2 "\t" $4 }' | sort -k 1,1) | join - <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_20kbPadding.txt | sed 's/chr//g' | awk '{ print $1 "_" $2 "\t" $4 }' | sort -k 1,1) | sort -g -k 1,1 | wc
Done
2593966 20751728 119301659
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170105.UCSC_hg19_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.wSymbol.bed | head -n 10
chr1 10003485   10045556        NMNAT1
chr1 100111430  100160097       PALMD
chr1 100174258  100231349       FRRS1
chr1 100178485  100178513       MIR548AA1
chr1 100178485  100178513       MIR548D1
chr1 100315639  100389579       AGL
chr1 100434000  100435404       BC112312
chr1 100435344  100492534       SLC35A3
chr1 100503788  100548929       HIAT1
chr1 100549101  100598511       SASS6
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/bedtools slop -i /mnt/lustre/home/mturchin20/Data/UCSC/20170105.UCSC_hg19_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.wSymbol.bed -g /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg19.chrom.sizes -b 20000 > /mnt/lustre/home/mturchin20/Data/UCSC/20170105.UCSC_hg19_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.20kbPadding.wSymbol.bed
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170105.UCSC_hg19_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.20kbPadding.wSymbol.bed | head -n 10
chr1 9983485    10065556        NMNAT1
chr1 100091430  100180097       PALMD
chr1 100154258  100251349       FRRS1
chr1 100158485  100198513       MIR548AA1
chr1 100158485  100198513       MIR548D1
chr1 100295639  100409579       AGL
chr1 100414000  100455404       BC112312
chr1 100415344  100512534       SLC35A3
chr1 100483788  100568929       HIAT1
chr1 100529101  100618511       SASS6
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -wb -a /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170105.UCSC_hg19_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.wSymbol.bed | head -n 10
chr1 10005008   10005008        chr1    10003485        10045556        NMNAT1
chr1 10010658   10010658        chr1    10003485        10045556        NMNAT1
chr1 100112098  100112098       chr1    100111430       100160097       PALMD
chr1 100112230  100112230       chr1    100111430       100160097       PALMD
chr1 100112247  100112247       chr1    100111430       100160097       PALMD
chr1 100112297  100112297       chr1    100111430       100160097       PALMD
chr1 100112394  100112394       chr1    100111430       100160097       PALMD
chr1 100112813  100112813       chr1    100111430       100160097       PALMD
chr1 100113375  100113375       chr1    100111430       100160097       PALMD
chr1 100113415  100113415       chr1    100111430       100160097       PALMD
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.txt.gz | awk '{ print $1 "_" $2 }' | sed 's/chr//g' | sort | uniq | wc
2321276 2321276 27168768
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join -v 1 <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bed | awk '{ print $1 "_" $2 }' | sed 's/chr//g' | sort) <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.txt.gz | awk '{ print $1 "_" $2 }' | sed 's/chr//g' | sort | uniq) | wc
 272690  272690 3040332
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bed | wc
2593966 7781898 61963243
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$R -q -e "2321276 + 272690"
> 2321276 + 272690
[1] 2593966
>
>
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed.txt | awk '{ if ($4 == 0) { print $0 } } ' | wc
 272690 1090760 6878968
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed.txt | awk '{ if ($4 > 0) { print $0 } } ' | wc
2321276 9285104 60713830
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.txt.gz | sed 's/chr//g' | sort -g -k 1,1 -k 2,2 | gzip > pinsdfinsdinsdf.gz
#Code here -- if first line, create variables. if not first line, check if last line. If last line, check whether coords same -- if same, add last line then print, if not same, print, then print last line. If not last line, check whether coords same -- if same, add, if not same, print, then recreate variables
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat pinsdfinsdinsdf.gz | perl -lane 'if ($. == 1) { $ChrBP1 = $F[0] . "_" . $F[1]; @loci = ($F[$#F]); } else { $ChrBP2 = $F[0] . "_" . $F[1]; if (eof()) { if ($ChrBP1 eq $ChrBP2) { push(@loci, $F[$#F]); print $ChrBP1, "\t", scalar(@loci), "\t", join(",", @loci); } else { print $ChrBP1, "\t", scalar(@loci), "\t", join(",", @loci); print $ChrBP2, "\t1\t", $F[$#F];} } else { if ($ChrBP1 eq $ChrBP2) { push(@loci, $F[$#F]); } else { print $ChrBP1, "\t", scalar(@loci), "\t", join(",", @loci); $ChrBP1 = $F[0] . "_" . $F[1]; @loci = ($F[$#F]); } } }' | wc
2321276 6963828 153802591
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat pinsdfinsdinsdf.gz | perl -lane 'if ($. == 1) { $ChrBP1 = $F[0] . "_" . $F[1]; @loci = ($F[$#F]); } else { $ChrBP2 = $F[0] . "_" . $F[1]; if (eof()) { if ($ChrBP1 eq $ChrBP2) { push(@loci, $F[$#F]); print $ChrBP1, "\t", scalar(@loci), "\t", join(",", @loci); } else { print $ChrBP1, "\t", scalar(@loci), "\t", join(",", @loci); print $ChrBP2, "\t1\t", $F[$#F];} } else { if ($ChrBP1 eq $ChrBP2) { push(@loci, $F[$#F]); } else { print $ChrBP1, "\t", scalar(@loci), "\t", join(",", @loci); $ChrBP1 = $F[0] . "_" . $F[1]; @loci = ($F[$#F]); } } }' | wc
2321276 6963828 153802591
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.txt.gz | head -n 10
chr1 10005008   10005008        chr1    10003485        10045556        NMNAT1
chr1 10010658   10010658        chr1    10003485        10045556        NMNAT1
chr1 100112098  100112098       chr1    100111430       100160097       PALMD
chr1 100112230  100112230       chr1    100111430       100160097       PALMD
chr1 100112247  100112247       chr1    100111430       100160097       PALMD
chr1 100112297  100112297       chr1    100111430       100160097       PALMD
chr1 100112394  100112394       chr1    100111430       100160097       PALMD
chr1 100112813  100112813       chr1    100111430       100160097       PALMD
chr1 100113375  100113375       chr1    100111430       100160097       PALMD
chr1 100113415  100113415       chr1    100111430       100160097       PALMD
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.txt.gz | wc
14266648 99866536 805941198
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat pinsdfinsdinsdf.gz | wc
14266648 99866536 720341310
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.txt.gz | sed 's/chr//g' | sort -g -k 1,1 -k 2,2 | perl -lane 'if ($. == 1) { $ChrBP1 = $F[0] . "_" . $F[1]; @loci = ($F[$#F]); } else { $ChrBP2 = $F[0] . "_" . $F[1]; if (eof()) { if ($ChrBP1 eq $ChrBP2) { push(@loci, $F[$#F]); print $ChrBP1, "\t", scalar(@loci), "\t", join(",", @loci); } else { print $ChrBP1, "\t", scalar(@loci), "\t", join(",", @loci); print $ChrBP2, "\t1\t", $F[$#F];} } else { if ($ChrBP1 eq $ChrBP2) { push(@loci, $F[$#F]); } else { print $ChrBP1, "\t", scalar(@loci), "\t", join(",", @loci); $ChrBP1 = $F[0] . "_" . $F[1]; @loci = ($F[$#F]); } } }' | wc
2321276 6963828 153802591
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.txt.gz | sed 's/chr//g' | sort -g -k 1,1 -k 2,2 | perl -lane 'if ($. == 1) { $ChrBP1 = $F[0] . "_" . $F[1]; @loci = ($F[$#F]); }
> else { $ChrBP2 = $F[0] . "_" . $F[1]; if (eof()) { if ($ChrBP1 eq $ChrBP2) { push(@loci, $F[$#F]); print $ChrBP1, "\t", scalar(@loci), "\t", join(",", @loci); } else { print $ChrBP1, "\t", scalar(@loci), "\t", join(",", @loci); print $ChrBP2, "\t1\t", $F[$#F];} }
> else { if ($ChrBP1 eq $ChrBP2) { push(@loci, $F[$#F]); } else { print $ChrBP1, "\t", scalar(@loci), "\t", join(",", @loci); $ChrBP1 = $F[0] . "_" . $F[1]; @loci = ($F[$#F]); } } }' | wc
2321276 6963828 153802591
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.txt.gz | sed 's/chr//g' | sort -g -k 1,1 -k 2,2 |
> perl -lane 'if ($. == 1) { $ChrBP1 = $F[0] . "_" . $F[1]; @loci = ($F[$#F]); }
> else { $ChrBP2 = $F[0] . "_" . $F[1]; if (eof()) { if ($ChrBP1 eq $ChrBP2) { push(@loci, $F[$#F]); print $ChrBP1, "\t", scalar(@loci), "\t", join(",", @loci); } else { print $ChrBP1, "\t", scalar(@loci), "\t", join(",", @loci); print $ChrBP2, "\t1\t", $F[$#F];} }
> else { if ($ChrBP1 eq $ChrBP2) { push(@loci, $F[$#F]); } else { print $ChrBP1, "\t", scalar(@loci), "\t", join(",", @loci); $ChrBP1 = $F[0] . "_" . $F[1]; @loci = ($F[$#F]); } } }' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Condensed.txt.gz
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Condensed.txt.gz | wc
2321276 6963828 153802591
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.txt.gz | sed 's/chr//g' | sort -g -k 1,1 -k 2,2 |
> perl -lane 'if ($. == 1) { $ChrBP1 = $F[0] . "_" . $F[1]; @loci = ($F[$#F]); }
> else { $ChrBP2 = $F[0] . "_" . $F[1]; if (eof()) { if ($ChrBP1 eq $ChrBP2) { push(@loci, $F[$#F]); print $ChrBP1, "\t", scalar(@loci), "\t", join(",", @loci); } else { print $ChrBP1, "\t", scalar(@loci), "\t", join(",", @loci); print $ChrBP2, "\t1\t", $F[$#F];} }
> else { if ($ChrBP1 eq $ChrBP2) { push(@loci, $F[$#F]); } else { print $ChrBP1, "\t", scalar(@loci), "\t", join(",", @loci); $ChrBP1 = $F[0] . "_" . $F[1]; @loci = ($F[$#F]); } } }' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Condensed.txt.gz
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(cat <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Condensed.txt.gz) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | sed 's/chr//g') | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Counts.txt | sed 's/chr//g' | awk '{ print $1 "_" $2 "\t" $4 }' | sort -k 1,1) | wc        
2593966 10375864 163563238
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Counts.txt | wc
2593966 10375864 67592798
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Condensed.txt.gz | wc
2321276 6963828 153802591
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | sed 's/chr//g' | wc
 272690  818070 4131092
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$R -q -e "2321276 + 272690"
> 2321276 + 272690
[1] 2593966
> 
> 
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(cat <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Condensed.txt.gz) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | sed 's/chr//g') | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Counts.txt | sed 's/chr//g' | awk '{ print $1 "_" $2 "\t" $4 }' | sort -k 1,1) | awk '{ if ($2 == $4) { print $0 } }' | wc
2593966 10375864 163563238
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(cat <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Condensed.txt.gz) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | sed 's/chr//g') | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Counts.txt | sed 's/chr//g' | awk '{ print $1 "_" $2 "\t" $4 }' | sort -k 1,1) | awk '{ if ($2 == $4) { print $0 } }' | wc
2593966 10375864 163563238
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(cat <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Condensed.txt.gz) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | sed 's/chr//g') | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Counts.txt | sed 's/chr//g' | awk '{ print $1 "_" $2 "\t" $4 }' | sort -k 1,1) | wc
2593966 10375864 169782577
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Condensed.txt.gz | wc
2384461 7153383 160956733
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | wc                                209505  628515 3800497
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Counts.txt | wc
2593966 10375864 67617105
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$R -q -e "2384461 + 209505"
> 2384461 + 209505
[1] 2593966
>
>
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(cat <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Condensed.txt.gz) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | sed 's/chr//g') | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Counts.txt | sed 's/chr//g' | awk '{ print $1 "_" $2 "\t" $4 }' | sort -k 1,1) | awk '{ if ($2 == $4) { print $0 } }' | wc
2593966 10375864 169782577
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.txt.gz | awk '{ print $1 "\t" $2 "\t" $3 "\t" $4 }' | sort -k 1,1) <(cat <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Condensed.txt.gz) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | sed 's/chr//g') | sort -k 1,1) | join - <(cat <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Condensed.txt.gz) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | sed 's/chr//g') | sort -k 1,1) | head -n 10
10_10000135 10 10000135 0.449 0 0 0 0
10_10000265 10 10000265 0.436 0 0 0 0
10_100002729 10 100002729 0.367 5 DQ596633,DQ577099,MIR1256,DQ582762,R3HCC1L 6 DQ596633,DQ577099,MIR1256,DQ582762,R3HCC1L,LOXL4
10_100002880 10 100002880 0.358 5 DQ596633,DQ577099,MIR1256,DQ582762,R3HCC1L 6 DQ596633,DQ577099,MIR1256,DQ582762,R3HCC1L,LOXL4
10_100003553 10 100003553 0.12 5 DQ596633,DQ577099,MIR1256,DQ582762,R3HCC1L 6 DQ596633,DQ577099,MIR1256,DQ582762,R3HCC1L,LOXL4
10_100003805 10 100003805 0.3 5 DQ596633,DQ577099,MIR1256,DQ582762,R3HCC1L 6 DQ596633,DQ577099,MIR1256,DQ582762,R3HCC1L,LOXL4
10_100003967 10 100003967 0.356 5 DQ596633,DQ577099,MIR1256,DQ582762,R3HCC1L 6 DQ596633,DQ577099,MIR1256,DQ582762,R3HCC1L,LOXL4
10_10000459 10 10000459 0.427 0 0 0 0
10_100005553 10 100005553 0.144 4 DQ596633,DQ577099,MIR1256,DQ582762 6 DQ596633,DQ577099,MIR1256,DQ582762,R3HCC1L,LOXL4
10_100005593 10 100005593 0.217 4 DQ596633,DQ577099,MIR1256,DQ582762 6 DQ596633,DQ577099,MIR1256,DQ582762,R3HCC1L,LOXL4
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.txt.gz | awk '{ print $1 "\t" $2 "\t" $3 "\t" $4 }' | sort -k 1,1) <(cat <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Condensed.txt.gz) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | sed 's/chr//g') | sort -k 1,1) | join - <(cat <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Condensed.txt.gz) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | sed 's/chr//g') | sort -k 1,1) | perl -lane 'my $prox1 = 0; my $prox2 = 0; if ($F[4] > 0) { $prox1 = 1; } if ($F[6] > 0) { $prox2 = 1; } print $F[0], "\t", $F[1], "\t", $F[2], "\t", $F[3], "\t", $prox1, "\t", $F[4], "\t", $prox2, "\t", $F[6]; ' | head -n 10 
10_10000135     10      10000135        0.449   0       0       0       0
10_10000265     10      10000265        0.436   0       0       0       0
10_100002729    10      100002729       0.367   1       5       1       6
10_100002880    10      100002880       0.358   1       5       1       6
10_100003553    10      100003553       0.12    1       5       1       6
10_100003805    10      100003805       0.3     1       5       1       6
10_100003967    10      100003967       0.356   1       5       1       6
10_10000459     10      10000459        0.427   0       0       0       0
10_100005553    10      100005553       0.144   1       4       1       6
10_100005593    10      100005593       0.217   1       4       1       6
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.txt.gz | awk '{ print $1 "\t" $2 "\t" $3 "\t" $4 }' | sort -k 1,1) <(cat <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Condensed.txt.gz) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | sed 's/chr//g') | sort -k 1,1) | join - <(cat <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Condensed.txt.gz) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | sed 's/chr//g') | sort -k 1,1) | perl -lane 'my $prox1 = 0; my $prox2 = 0; if ($F[4] > 0) { $prox1 = 1; } if ($F[6] > 0) { $prox2 = 1; } print $F[0], "\t", $F[1], "\t", $F[2], "\t", $F[3], "\t", $prox1, "\t", $F[4], "\t", $prox2, "\t", $F[6]; ' | wc
2593966 20751728 99468423
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.txt.gz | awk '{ print $1 "\t" $2 "\t" $3 "\t" $4 }' | sort -k 1,1) <(cat <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Condensed.txt.gz) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | sed 's/chr//g') | sort -k 1,1) | join - <(cat <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Condensed.txt.gz) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | sed 's/chr//g') | sort -k 1,1) | perl -lane 'my $prox1 = 0; my $prox2 = 0; if ($F[4] > 0) { $prox1 = 1; } if ($F[6] > 0) { $prox2 = 1; } print $F[0], "\t", $F[1], "\t", $F[2], "\t", $F[3], "\t", $prox1, "\t", $F[4], "\t", $prox2, "\t", $F[6]; ' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.AnnotatedForPerms.NoPad.20kbPad.vs1.txt.gz
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.AnnotatedForPerms.NoPad.20kbPad.vs1.txt.gz | head -n 10
10_10000135     10      10000135        0.449   0       0       0       0
10_10000265     10      10000265        0.436   0       0       0       0
10_100002729    10      100002729       0.367   1       5       1       6
10_100002880    10      100002880       0.358   1       5       1       6
10_100003553    10      100003553       0.12    1       5       1       6
10_100003805    10      100003805       0.3     1       5       1       6
10_100003967    10      100003967       0.356   1       5       1       6
10_10000459     10      10000459        0.427   0       0       0       0
10_100005553    10      100005553       0.144   1       4       1       6
10_100005593    10      100005593       0.217   1       4       1       6
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.AnnotatedForPerms.NoPad.20kbPad.vs1.txt.gz | wc        
2593966 20751728 99468423
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.txt | awk '{ print $1 "_" $2 }' | sort) <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.AnnotatedForPerms.NoPad.20kbPad.vs1.txt.gz | sort -k 1,1) | wc
     99     792    3774
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.txt | wc
    101     202    1190
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.txt | head -n 10
11 18589560
2  165221337
19 19268718
2  21078786
10 64697616
6  161009807
12 110556807
8  145115531
11 125749162
12 56078847
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join -v 1 <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.txt | awk '{ print $1 "_" $2 }' | sort) <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.AnnotatedForPerms.NoPad.20kbPad.vs1.txt.gz | sort -k 1,1)     
11_10345358
15_42033223
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.No0Dir.tbl.gz | awk '{ print $1 "_" $2 "\t" $3 }' | head -n 10
Chr_BP  MAF
10_10000135     0.551
10_10000265     0.436
10_100002729    0.633
10_100002880    0.642
10_100003553    0.12
10_100003805    0.3
10_100003967    0.356
10_10000459     0.427
10_100005553    0.144
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.txt | awk '{ print $1 "_" $2 }' | sort) <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.No0Dir.tbl.gz | awk '{ print $1 "_" $2 "\t" $3 }' | sort -k 1,1) | wc
    101     202    1769
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.txt | wc
    101     202    1190
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.txt | awk '{ print $1 "_" $2 }' | sort) <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.No0Dir.tbl.gz | awk '{ print $1 "_" $2 "\t" $3 }' | sort -k 1,1) | head -n 10
10_113923876 0.3
10_64697616 0.536
10_94829632 0.433
1_109619829 0.702
11_10345358 0.15
11_116154127 0.108
11_122027585 0.617
11_125749162 0.842
11_18589560 0.683
11_46699823 0.964
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.txt | awk '{ print $1 "_" $2 }' | sort) <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.IncAllele.bmass.formatted.NoDups.No0Dir.tbl.gz | awk '{ print $1 "_" $2 "\t" $3 }' | perl -lane 'if ($F[1] > .5) { $F[1] = 1 - $F[5]; } print join("\t", @F);' | sort -k 1,1) | head -n 10
10_113923876 0.3
10_64697616 1
10_94829632 0.433
1_109619829 1
11_10345358 0.15
11_116154127 0.108
11_122027585 1
11_125749162 1
11_18589560 1
11_46699823 1
[  mturchin20@spudhead  ~/Lab_Stuff/StephensLab/Multivariate]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_250kbPadding.txt.gz | wc        
25464348 178250436 1417704006
[  mturchin20@spudhead  ~/Lab_Stuff/StephensLab/Multivariate]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.txt.gz | wc
15128292 105898044 850069663
[  mturchin20@spudhead  ~/Lab_Stuff/StephensLab/Multivariate]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_250kbPadding.Condensed.txt.gz | wc
2528918 7586754 239521166
[  mturchin20@spudhead  ~/Lab_Stuff/StephensLab/Multivariate]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Condensed.txt.gz | wc
2384461 7153383 160956733
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(cat <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_250kbPadding.Condensed.txt.gz) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_250kbPadding.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | sed 's/chr//g') | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_250kbPadding.Counts.txt | sed 's/chr//g' | awk '{ print $1 "_" $2 "\t" $4 }' | sort -k 1,1) | awk '{ if ($2 == $4) { print $0 } }' | wc
2593966 10375864 246652707
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_250kbPadding.Condensed.txt.gz | wc
2528918 7586754 239521166
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_250kbPadding.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | wc
  65048  195144 1184375
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$R -q -e "2528918 + 65048"
> 2528918 + 65048
[1] 2593966
> 
> 
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.txt.gz | awk '{ print $1 "\t" $2 "\t" $3 "\t" $4 }' | sort -k 1,1) <(cat <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Condensed.txt.gz) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | sed 's/chr//g') | sort -k 1,1) | join - <(cat <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Condensed.txt.gz) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | sed 's/chr//g') | sort -k 1,1) | head -n 10
10_10000135 10 10000135 0.449 0 0 0 0
10_10000265 10 10000265 0.436 0 0 0 0
10_100002729 10 100002729 0.367 5 DQ596633,DQ577099,MIR1256,DQ582762,R3HCC1L 6 DQ596633,DQ577099,MIR1256,DQ582762,R3HCC1L,LOXL4
10_100002880 10 100002880 0.358 5 DQ596633,DQ577099,MIR1256,DQ582762,R3HCC1L 6 DQ596633,DQ577099,MIR1256,DQ582762,R3HCC1L,LOXL4
10_100003553 10 100003553 0.12 5 DQ596633,DQ577099,MIR1256,DQ582762,R3HCC1L 6 DQ596633,DQ577099,MIR1256,DQ582762,R3HCC1L,LOXL4
10_100003805 10 100003805 0.3 5 DQ596633,DQ577099,MIR1256,DQ582762,R3HCC1L 6 DQ596633,DQ577099,MIR1256,DQ582762,R3HCC1L,LOXL4
10_100003967 10 100003967 0.356 5 DQ596633,DQ577099,MIR1256,DQ582762,R3HCC1L 6 DQ596633,DQ577099,MIR1256,DQ582762,R3HCC1L,LOXL4
10_10000459 10 10000459 0.427 0 0 0 0
10_100005553 10 100005553 0.144 4 DQ596633,DQ577099,MIR1256,DQ582762 6 DQ596633,DQ577099,MIR1256,DQ582762,R3HCC1L,LOXL4
10_100005593 10 100005593 0.217 4 DQ596633,DQ577099,MIR1256,DQ582762 6 DQ596633,DQ577099,MIR1256,DQ582762,R3HCC1L,LOXL4
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.txt.gz | awk '{ print $1 "\t" $2 "\t" $3 "\t" $4 }' | sort -k 1,1) <(cat <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Condensed.txt.gz) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | sed 's/chr//g') | sort -k 1,1) | join - <(cat <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Condensed.txt.gz) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | sed 's/chr//g') | sort -k 1,1) | join - <(cat <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_250kbPadding.Condensed.txt.gz) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_250kbPadding.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | sed 's/chr//g') | sort -k 1,1) | head -n 10
10_10000135 10 10000135 0.449 0 0 0 0 1 BC032914
10_10000265 10 10000265 0.436 0 0 0 0 1 BC032914
10_100002729 10 100002729 0.367 5 DQ596633,DQ577099,MIR1256,DQ582762,R3HCC1L 6 DQ596633,DQ577099,MIR1256,DQ582762,R3HCC1L,LOXL4 12 DQ596633,DQ577099,MIR1256,DQ582762,CRTAC1,R3HCC1L,LOXL4,PYROXD2,MIR1287,HPS1,MIR4685,HPSE2
10_100002880 10 100002880 0.358 5 DQ596633,DQ577099,MIR1256,DQ582762,R3HCC1L 6 DQ596633,DQ577099,MIR1256,DQ582762,R3HCC1L,LOXL4 12 DQ596633,DQ577099,MIR1256,DQ582762,CRTAC1,R3HCC1L,LOXL4,PYROXD2,MIR1287,HPS1,MIR4685,HPSE2
10_100003553 10 100003553 0.12 5 DQ596633,DQ577099,MIR1256,DQ582762,R3HCC1L 6 DQ596633,DQ577099,MIR1256,DQ582762,R3HCC1L,LOXL4 12 DQ596633,DQ577099,MIR1256,DQ582762,CRTAC1,R3HCC1L,LOXL4,PYROXD2,MIR1287,HPS1,MIR4685,HPSE2
10_100003805 10 100003805 0.3 5 DQ596633,DQ577099,MIR1256,DQ582762,R3HCC1L 6 DQ596633,DQ577099,MIR1256,DQ582762,R3HCC1L,LOXL4 12 DQ596633,DQ577099,MIR1256,DQ582762,CRTAC1,R3HCC1L,LOXL4,PYROXD2,MIR1287,HPS1,MIR4685,HPSE2
10_100003967 10 100003967 0.356 5 DQ596633,DQ577099,MIR1256,DQ582762,R3HCC1L 6 DQ596633,DQ577099,MIR1256,DQ582762,R3HCC1L,LOXL4 12 DQ596633,DQ577099,MIR1256,DQ582762,CRTAC1,R3HCC1L,LOXL4,PYROXD2,MIR1287,HPS1,MIR4685,HPSE2
10_10000459 10 10000459 0.427 0 0 0 0 1 BC032914
10_100005553 10 100005553 0.144 4 DQ596633,DQ577099,MIR1256,DQ582762 6 DQ596633,DQ577099,MIR1256,DQ582762,R3HCC1L,LOXL4 12 DQ596633,DQ577099,MIR1256,DQ582762,CRTAC1,R3HCC1L,LOXL4,PYROXD2,MIR1287,HPS1,MIR4685,HPSE2
10_100005593 10 100005593 0.217 4 DQ596633,DQ577099,MIR1256,DQ582762 6 DQ596633,DQ577099,MIR1256,DQ582762,R3HCC1L,LOXL4 12 DQ596633,DQ577099,MIR1256,DQ582762,CRTAC1,R3HCC1L,LOXL4,PYROXD2,MIR1287,HPS1,MIR4685,HPSE2
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.AnnotatedForPerms.NoPad.20kbPad.250kbPad.vs1.txt.gz | awk '{ print $8 }' | sort | uniq -c | sort -g -k 2,2 | head -n 10
 209505 0
 420100 1
 334282 2
 185642 3
 197034 4
 238554 5
 227076 6
 161800 7
  88154 8
  65974 9
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.AnnotatedForPerms.NoPad.20kbPad.250kbPad.vs1.txt.gz | awk '{ print $10 }' | sort | uniq -c | sort -g -k 2,2 | head -n 10
  65048 0
 139030 1
 182023 2
 174516 3
 187751 4
 196234 5
 210428 6
 191264 7
 157318 8
 138202 9
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(cat <(zcat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Condensed.txt.gz) <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | sed 's/chr//g') | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Counts.txt | sed 's/chr//g' | awk '{ print $1 "_" $2 "\t" $4 }' | sort -k 1,1) | awk '{ if ($2 == $4) { print $0 } }' | wc
    101     404    7762
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Condensed.txt.gz | wc
     91     273    7384
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | wc
     10      30     181
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(cat <(zcat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Condensed.txt.gz) <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | sed 's/chr//g') | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Counts.txt | sed 's/chr//g' | awk '{ print $1 "_" $2 "\t" $4 }' | sort -k 1,1) | awk '{ if ($2 == $4) { print $0 } }' | wc
    101     404    8204
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Condensed.txt.gz | wc
     94     282    7868
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | wc 
      7      21     127
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(cat <(zcat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_250kbPadding.Condensed.txt.gz) <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_250kbPadding.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | sed 's/chr//g') | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_250kbPadding.Counts.txt | sed 's/chr//g' | awk '{ print $1 "_" $2 "\t" $4 }' | sort -k 1,1) | awk '{ if ($2 == $4) { print $0 } }' | wc
    101     404   13826
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_250kbPadding.Condensed.txt.gz | wc
    100     300   13550
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_250kbPadding.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | wc
      1       3      19
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_250kbPadding.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }'     
chr19_59484573  0       0
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(cat <(zcat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Condensed.txt.gz) <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | sed 's/chr//g') | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Counts.txt | sed 's/chr//g' | awk '{ print $1 "_" $2 "\t" $4 }' | sort -k 1,1) | awk '{ if ($2 == $4) { print $0 } }' | wc
    102     408    7792
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Condensed.txt.gz | wc
     92     276    7412
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | wc
     10      30     181
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(cat <(zcat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Condensed.txt.gz) <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | sed 's/chr//g') | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Counts.txt | sed 's/chr//g' | awk '{ print $1 "_" $2 "\t" $4 }' | sort -k 1,1) | awk '{ if ($2 == $4) { print $0 } }' | wc
    102     408    8234
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Condensed.txt.gz | wc
     95     285    7896
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat  /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | wc
      7      21     127
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(cat <(zcat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_250kbPadding.Condensed.txt.gz) <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_250kbPadding.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | sed 's/chr//g') | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_250kbPadding.Counts.txt | sed 's/chr//g' | awk '{ print $1 "_" $2 "\t" $4 }' | sort -k 1,1) | awk '{ if ($2 == $4) { print $0 } }' | wc
    102     408   13889
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_250kbPadding.Condensed.txt.gz | wc
    101     303   13611
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_250kbPadding.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | wc
      1       3      19
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.AnnotatedForPerms.NoPad.20kbPad.250kbPad.vs1.txt.gz | wc
    102    1020    4356
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.AnnotatedForPerms.NoPad.20kbPad.250kbPad.vs1.txt.gz | head -n 10
10_113923876    10      113923876       0.3     1       4  1    4       1       7
10_64697616 10  64697616        0.464   1       8       1  8    1       12
10_94829632 10  94829632        0.433   1       4       1  7    1       8
11_10345358 11  10345358        0.15    1       6       1  8    1       15
11_116154127    11      116154127       0.108   1       4  1    4       1       4
11_122027585    11      122027585       0.383   1       5  1    7       1       13
11_125749162    11      125749162       0.158   1       4  1    6       1       15
11_18589560 11  18589560        0.317   1       7       1  8    1       19
11_46699823 11  46699823        0.036   1       7       1  8    1       19
11_61326406 11  61326406        0.367   1       6       1  6    1       26
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.AnnotatedForPerms.NoPad.20kbPad.250kbPad.vs1.txt.gz | grep NA | wc
      0       0       0
[  mturchin20@spudhead  ~/Software]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/PermutationAnalyses/Permutations/GlobalLipids2010.Table1.bmass_GlobalLipids2010_Vs2_MergedDataSources_vs1_AnnotatedForPerms.vs1Permutation1.txt.gz | awk '{ print $11 }' | R -q -e "Data1 <- read.table(file('stdin'), header=F); quantile(Data1[,1]);"
> Data1 <- read.table(file('stdin'), header=F); quantile(Data1[,1]);
     0%     25%     50%     75%    100% 
 1047.0 12522.5 28262.0 35388.5 73416.0 
> 
> 
[  mturchin20@spudhead  ~/Software]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/PermutationAnalyses/Permutations/GlobalLipids2010.Table1.bmass_GlobalLipids2010_Vs2_MergedDataSources_vs1_AnnotatedForPerms.vs1Permutation10000.txt.gz | awk '{ print $11 }' | R -q -e "Data1 <- read.table(file('stdin'), header=F); quantile(Data1[,1]);"
> Data1 <- read.table(file('stdin'), header=F); quantile(Data1[,1]);
     0%     25%     50%     75%    100% 
 1047.0 12522.5 28262.0 35388.5 73416.0 
> 
> 



[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.MarkerChrBPMAF.noNA.NArplcd.txt | wc
    102     408    3449
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.Marker.AnnovarFormat.genome_summary.csv | wc
    102     103    8732
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.Marker.AnnovarFormat.genome_summary.csv | head -n 5
Func,Gene,ExonicFunc,AAChange,Conserved,SegDup,ESP6500_ALL,1000g2012apr_ALL,dbSNP129,AVSIFT,LJB_PhyloP,LJB_PhyloP_Pred,LJB_SIFT,LJB_SIFT_Pred,LJB_PolyPhen2,LJB_PolyPhen2_Pred,LJB_LRT,LJB_LRT_Pred,LJB_MutationTaster,LJB_MutationTaster_Pred,LJB_GERP++,Chr,Start,End,Ref,Obs,Otherinfo
"intronic","UEVLD",,,,,,,,,,,,,,,,,,,,11,18589560,18589560,T,C,
"intergenic","FIGN(dist=628824),GRB14(dist=127986)",,,,,,,,,,,,,,,,,,,,2,165221337,165221337,T,C,
"intronic","MEF2B,MEF2BNB-MEF2B",,,,,,,,,,,,,,,,,,,,19,19268718,19268718,T,C,
"intergenic","C2orf43(dist=55896),APOB(dist=145515)",,,,,,,,,,,,,,,,,,,,2,21078786,21078786,T,C,
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.txt | grep rs9411489
ABO 9   rs9411489       LDL     TC      C/T/0.20        +2.24   6â????â??10â??13             Y       ????
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.Marker.txt | sort) <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.tbl.gz | sort) | awk '{ print $12, "\t", $13, "\t", $13, "\t", $2, "\t", $3 }' | sed 's/a/A/g' | sed 's/g/G/' | sed 's/c/C/g' | sed 's/t/T/g' | grep NA
NA   NA      NA          T       C
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.Marker.txt | sort) <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.tbl.gz | sort) | awk '{ print $12, "\t", $13, "\t", $13, "\t", $2, "\t", $3 }' | sed 's/a/A/g' | sed 's/g/G/' | sed 's/c/C/g' | sed 's/t/T/g' | head -n 10
11   18589560    18589560        T       C
2   165221337    165221337       T       C
19   19268718    19268718        T       C
2   21078786     21078786        T       C
10   64697616    64697616        A       T
6   161009807    161009807       A       G
12   110556807   110556807       A       G
8   145115531    145115531       A       G
11   125749162   125749162       A       G
12   56078847    56078847        T       C
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.tbl.gz | grep rs9411489
rs9411489   t   c       38680   7.204   5.844e-13       +  ?++?+++++-??++??++++??++?    0.0728  0.01    NA NA  NA     NA
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /data/external_public/GlobalLipids2010/LDL_with_Effect.wHapMap23.expanded.tbl.gz | head -n 5     
MarkerName  Allele1     Allele2 Weight  GC.Zscore       GC.Pvalue       Overall Direction       Effect  StdErr ChrBPAFInfo      Chr     BP      MAF
rs10 a  c   81680       2.051   0.04027 +       ++-+++++-+?++-??-++-??++-       0.0294  0.0152  chr7_92221824_A_0.033_C_0.967   7       92221824        0.033
rs1000000   a   g       95454   0.662   0.5077  +       -----++-+--+-+++-+++-+-++       0.0044  0.0063  chr12_125456933_G_0.627_A_0.373 12      125456933       0.373
rs10000010  t   c       95454   1.583   0.1135  +       ++--++---+-+-++--+++---++       0.0073  0.0052  chr4_21227772_T_0.575_C_0.425   4       21227772        0.425
rs10000012  c   g       95397   0.155   0.877   +       +-+-----++-++---+++---+-+       -2e-04  0.0076  chr4_1347325_C_0.808_G_0.192    4       1347325 0.192
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.Marker.noNA.NArplcd.AnnovarFormat | wc
    102     510    3367
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.Marker.noNA.NArplcd.AnnovarFormat | grep NA | wc
      0       0       0
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$perl /mnt/lustre/home/mturchin20/Software/annovar/summarize_annovar.pl /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.Marker.noNA.NArplcd.AnnovarFormat /mnt/lustre/home/mturchin20/Software/annovar/humandb/ --ver1000g 1000g2012apr --verdbsnp 129 --veresp 6500 --buildver hg19 --remove

NOTICE: Running step 1 with system command <annotate_variation.pl -geneanno -buildver hg19 -dbtype refgene -outfile /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.Marker.noNA.NArplcd.AnnovarFormat -exonsort /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.Marker.noNA.NArplcd.AnnovarFormat /mnt/lustre/home/mturchin20/Software/annovar/humandb/>
NOTICE: Reading gene annotation from /mnt/lustre/home/mturchin20/Software/annovar/humandb/hg19_refGene.txt ...
.
.
.
---------------------------------------------------------------------------------------
WARNING: 1 exonic SNPs have WRONG reference alleles specified in your input file!
WARNING: An example input line is <12    56078847        56078847        T       C>
WARNING: ANNOVAR can still annotate exonic_variant_function for the mutation correctly!
WARNING: you may have used wrong -buildver, or specified incorrect reference allele, or used outdated mRNA FASTA file!
---------------------------------------------------------------------------------------
.
.
.
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.AnnovarFormat.genome_summary.csv | awk -F, '{ print $1 }' | sort | uniq -c
  21836 "UTR3"
   3874 "UTR5"
     10 "UTR5;UTR3"
  15110 "downstream"
  26990 "exonic"
      7 "exonic;splicing"
1438718 "intergenic"
 963454 "intronic"
    528 "ncRNA_UTR3"
     85 "ncRNA_UTR5"
      1 "ncRNA_UTR5;ncRNA_UTR3"
   5995 "ncRNA_exonic"
 102023 "ncRNA_intronic"
     38 "ncRNA_splicing"
    656 "splicing"
  14225 "upstream"
    416 "upstream;downstream"
      1 Func
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.AnnovarFormat.genome_summary.csv | grep -E 'exonic|splicing' | grep -v ncRNA | perl -F, -lane 'print $F[$#F-23];' | sort | uniq -c
    656 
  14707 "nonsynonymous SNV"
    806 "stopgain SNV"
     41 "stoploss SNV"
  11203 "synonymous SNV"
    240 "unknown"
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.AnnovarFormat.genome_summary.csv | grep -E 'exonic|splicing' | grep -v ncRNA | awk -F, '{ print $1 }' | sort | uniq -c
9_133279427,9,133279427,0.18103448,1,1,1,1,1,5  0.13103448,0.23103448,1,4.5,5.5
  26990 "exonic"
      7 "exonic;splicing"
    656 "splicing"
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.AnnovarFormat.genome_summary.csv | grep -E 'nonsynonymous|splicing|stopgain|stoploss' | grep -v ncRNA | awk -F, '{ print $1 }' | sort | uniq -c
  15550 "exonic"
      7 "exonic;splicing"
    656 "splicing"
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.AnnovarFormat.genome_summary.csv | grep -E 'nonsynonymous|splicing|stopgain|stoploss' | grep -v ncRNA | perl -F, -lane 'print $F[$#F-23];' | sort | uniq -c
    656 
  14707 "nonsynonymous SNV"
    806 "stopgain SNV"
     41 "stoploss SNV"
      3 "synonymous SNV"
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.AnnovarFormat.genome_summary.csv | grep -E 'nonsynonymous|splicing|stopgain|stoploss' | grep -v ncRNA | perl -F, -lane 'print $F[$#F-4], "_", $F[$#F-3], "\t", $F[$#F-3], "\t", $F[$#F-3];' | head -n 10
10_100020758    100020758  100020758
10_100146980    100146980  100146980
10_100150389    100150389  100150389
10_100177443    100177443  100177443
10_100183516    100183516  100183516
10_100185573    100185573  100185573
10_100189612    100189612  100189612
10_100374669    100374669  100374669
10_100503774    100503774  100503774
10_100995527    100995527  100995527
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.AnnovarFormat.genome_summary.csv | grep -E 'nonsynonymous|splicing|stopgain|stoploss' | grep -v ncRNA | perl -F, -lane 'print "chr", $F[$#F-4], "\t", $F[$#F-3], "\t", $F[$#F-3];' | head -n 10
chr10   100020758  100020758
chr10   100146980  100146980
chr10   100150389  100150389
chr10   100177443  100177443
chr10   100183516  100183516
chr10   100185573  100185573
chr10   100189612  100189612
chr10   100374669  100374669
chr10   100503774  100503774
chr10   100995527  100995527
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.AnnovarFormat.genome_summary.csv | grep -E 'nonsynonymous|splicing|stopgain|stoploss' | grep -v ncRNA | perl -F, -lane 'print "chr", $F[$#F-4], "\t", $F[$#F-3], "\t", $F[$#F-3];' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.AnnovarFormat.genome_summary.NonsynSpliceStopgainStoploss.no_ncRNA.bed
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.AnnovarFormat.genome_summary.csv | wc
2593967 2620724 223286308
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.AnnovarFormat.genome_summary.NonsynSpliceStopgainStoploss.no_ncRNA.bed | wc
  16213   48639  387479
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -c -a <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.AnnovarFormat.genome_summary.NonsynSpliceStopgainStoploss.no_ncRNA.bed) -b /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.AnnovarFormat.genome_summary.NonsynSpliceStopgainStoploss.no_ncRNA.20kbPadding.bed | head -n 10
chr10   100020758  100020758  1
chr10   100146980  100146980  2
chr10   100150389  100150389  2
chr10   100177443  100177443  4
chr10   100183516  100183516  4
chr10   100185573  100185573  4
chr10   100189612  100189612  4
chr10   100374669  100374669  1
chr10   100503774  100503774  1
chr10   100995527  100995527  1
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/PermutationAnalyses/DataFileAnnotations/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_20kbPadding_wSymbol.Counts.txt.gz | awk '{ print $4 }' | sort | uniq -c | sort -r -k 1,1 | head -n 20
 420100 1
 334282 2
 238554 5
 227076 6
 209505 0
 197034 4
 185642 3
 161800 7
  88154 8
  69564 10
  65974 9
  61552 11
  41613 13
  41381 12
  41027 14
  40870 15
  31191 16
  19373 17
  12393 31
  10988 18
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/PermutationAnalyses/DataFileAnnotations/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Counts.txt.gz | sed 's/chr//g' | perl -lane 'my $val1 = NA; if ($F[3] > 0) { $val1 = 1; } if ($F[3] == 0) { $val1 = 0; } print $F[0], "_", $F[1], "\t", $val1;' | sort -k 1,1 | head -n 10
10_10000135 0
10_10000265 0
10_100002729    1
10_100002880    1
10_100003553    1
10_100003805    1
10_100003967    1
10_10000459 0
10_100005553    1
10_100005593    1
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/PermutationAnalyses/DataFileAnnotations/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Counts.txt.gz | sed 's/chr//g' | perl -lane 'my $val1 = NA; if ($F[3] > 0) { $val1 = 1; } if ($F[3] == 0) { $val1 = 0; } print $F[0], "_", $F[1], "\t", $val1;' | sort -k 1,1 | wc        
2593966 5187932 35397032
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/PermutationAnalyses/DataFileAnnotations/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Counts.txt.gz | wc
2593966 10375864 67592798
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/PermutationAnalyses/DataFileAnnotations/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170105_UCSC_hg19_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Counts.txt.gz | sed 's/chr//g' | perl -lane 'my $val1 = NA; if ($F[3] > 0) { $val1 = 1; } if ($F[3] == 0) { $val1 = 0; } print $F[0], "_", $F[1], "\t", $val1;' | sort -k 1,1 | grep NA | wc
      0       0       0
.
.
.
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/NHGRI/20170118Accessed_gwas_catalog_v1.0.1-associations_e87_r2017-01-09.tsv | perl -F\\t -lane 'print $#F;' | sort | uniq -c
  34398 36
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/NHGRI/20170118Accessed_gwas_catalog_v1.0.1-associations_e87_r2017-01-09.tsv | perl -F\\t -lane 'my @traits = split(/\s+/, $F[7]); print join("_", @traits), "\t", $F[11], "\t", $F[12];' | head -n 10
DISEASE/TRAIT   CHR_ID  CHR_POS
YKL-40_levels   1  203186754
Psoriasis   13  39776775
Lung_cancer 15  78513681
Lung_cancer 1   159711078
Lung_cancer 3   190632672
Lung_cancer 15  78513681
Nicotine_dependence 15  78601997
Colorectal_cancer  8    127412547
Colorectal_cancer  11   111300984
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/NHGRI/20170118Accessed_gwas_catalog_v1.0.1-associations_e87_r2017-01-09.tsv | perl -F\\t -lane 'my @traits = split(/\s+/, $F[7]); print join("_", @traits), "\t", $F[11], "\t", $F[12];' | wc        
  34398  102146 1382826
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/NHGRI/20170118Accessed_gwas_catalog_v1.0.1-associations_e87_r2017-01-09.tsv | perl -F\\t -lane 'my @traits = split(/\s+/, $F[7]); print join("_", @traits), "\t", $F[11], "\t", $F[12];' | awk '{ print $1 }' | sort | uniq -c | sort -r -k 1,1 | head -n 20
   1631 Post_bronchodilator_FEV1/FVC_ratio
   1182 Post_bronchodilator_FEV1
    957 Obesity-related_traits
    899 Schizophrenia
    841 Body_mass_index
    823 Height
    699 IgG_glycosylation
    473 Post_bronchodilator_FEV1/FVC_ratio_in_COPD
    396 Type_2_diabetes
    390 Crohn's_disease
    319 Inflammatory_bowel_disease
    309 Rheumatoid_arthritis
    296 Waist-to-hip_ratio_adjusted_for_body_mass_index
    283 Systemic_lupus_erythematosus
    278 HDL_cholesterol
    273 Prostate_cancer
    249 Blood_metabolite_levels
    246 3-hydroxypropylmercapturic_acid_levels_in_smokers
    240 3-hydroxy-1-methylpropylmercapturic_acid_levels_in_smokers
    238 LDL_cholesterol
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/NHGRI/20170118Accessed_gwas_catalog_v1.0.1-associations_e87_r2017-01-09.tsv | perl -F\\t -lane 'my @traits = split(/\s+/, $F[7]); print join("_", @traits), "\t", $F[11], "\t", $F[12];' | awk '{ print $1 }' | sort | uniq -c | wc                         
   1684    3368   71873
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/NHGRI/20170118Accessed_gwas_catalog_v1.0.1-associations_e87_r2017-01-09.tsv | perl -F\\t -lane 'my @traits = split(/\s+/, $F[7]); print join("_", @traits), "\t", $F[11], "\t", $F[12], "\t", $F[27];' | wc                      
  34398  136533 1564180
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/NHGRI/20170118Accessed_gwas_catalog_v1.0.1-associations_e87_r2017-01-09.tsv | perl -F\\t -lane 'my @traits = split(/\s+/, $F[7]); print join("_", @traits), "\t", $F[11], "\t", $F[12], "\t", $F[27];' | perl -lane 'if ($#F == 3) { print join("\t", @F); }' | wc
  33429  133716 1524496
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/NHGRI/20170118Accessed_gwas_catalog_v1.0.1-associations_e87_r2017-01-09.tsv | wc
  34398 2524701 22151760
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/NHGRI/20170118Accessed_gwas_catalog_v1.0.1-associations_e87_r2017-01-09.tsv | perl -F\\t -lane 'my @traits = split(/\s+/, $F[7]); print join("_", @traits), "\t", $F[11], "\t", $F[12], "\t", $F[27];' | perl -lane 'if ($#F == 3) { print join("\t", @F); }' | wc
  33429  133716 1524496
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/NHGRI/20170118Accessed_gwas_catalog_v1.0.1-associations_e87_r2017-01-09.tsv | perl -F\\t -lane 'if (!$F[11]) { @altvals = split(/\:/, $F[21]); $F[11] = $altvals[0]; $F[12] = $altvals[1]; } my @traits = split(/\s+/, $F[7]); print join("_", @traits), "\t", $F[11], "\t", $F[12], "\t", $F[27];' | perl -lane 'if ($#F == 3) { print join("\t", @F); }' | wc
  33977  135908 1548233
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/NHGRI/20170118Accessed_gwas_catalog_v1.0.1-associations_e87_r2017-01-09.tsv | perl -F\\t -lane 'if (!$F[11]) { print $F[21]; }' | wc        
    814     814   10701
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/NHGRI/20170118Accessed_gwas_catalog_v1.0.1-associations_e87_r2017-01-09.tsv | perl -F\\t -lane 'if (!$F[11]) { print $F[21]; }' | grep chr | wc
    476     476    6987
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/NHGRI/20170118Accessed_gwas_catalog_v1.0.1-associations_e87_r2017-01-09.tsv | perl -F\\t -lane 'if (!$F[11]) { print $F[21]; }' | grep rs | wc
     81      83     813
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/NHGRI/20170118Accessed_gwas_catalog_v1.0.1-associations_e87_r2017-01-09.tsv | perl -F\\t -lane 'if (!$F[11]) { print $F[21]; }' | grep -v -E 'chr|rs' | head -n 10
Chr11:5232745

NR
NR
6:32588205
i4000416

NR
NR
H1H2
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/NHGRI/20170118Accessed_gwas_catalog_v1.0.1-associations_e87_r2017-01-09.tsv | perl -F\\t -lane 'if (!$F[11]) { print $F[21]; }' | grep -v -E 'chr|rs' | wc        
    257     255    2901
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/NHGRI/20170118Accessed_gwas_catalog_v1.0.1-associations_e87_r2017-01-09.tsv | perl -F\\t -lane 'if (!$F[11]) { print $F[21]; }' | grep \: | grep -v chr | wc
     74      76     877
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/NHGRI/20170118Accessed_gwas_catalog_v1.0.1-associations_e87_r2017-01-09.tsv | perl -F\\t -lane 'if (!$F[11]) { print $F[21]; }' | grep chr | grep \: | wc
    475     475    6972
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/NHGRI/20170118Accessed_gwas_catalog_v1.0.1-associations_e87_r2017-01-09.tsv | perl -F\\t -lane 'if (!$F[11]) { print $F[21]; }' | grep chr | grep \_ | wc
      2       2      35
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/NHGRI/20170118Accessed_gwas_catalog_v1.0.1-associations_e87_r2017-01-09.tsv | perl -F\\t -lane 'if (!$F[11]) { @altvals = split(/\:/, $F[21]); $F[11] = $altvals[0]; $F[12] = $altvals[1]; } my @traits = split(/\s+/, $F[7]); print join("_", @traits), "\t", $F[11], "\t", $F[12], "\t", $F[27];' | perl -lane 'if ($#F == 3) { print join("\t", @F); }' | R -q -e "Data1 <- read.table(file('stdin'), header=T); Data2 <- Data1[Data1[,4] <= 5e-08,]; print(head(Data1)); print(c(nrow(Data1), nrow(Data2)));"
> Data1 <- read.table(file('stdin'), header=T); Data2 <- Data1[Data1[,4] <= 5e-08,]; print(head(Data1)); print(c(nrow(Data1), nrow(Data2)));
  DISEASE.TRAIT CHR_ID   CHR_POS P.VALUE
1 YKL-40_levels      1 203186754   1e-13
2     Psoriasis     13  39776775   2e-06
3   Lung_cancer     15  78513681   3e-18
4   Lung_cancer      1 159711078   7e-06
5   Lung_cancer      3 190632672   8e-06
6   Lung_cancer     15  78513681   5e-20
[1] 33976 13483
> 
> 
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/NHGRI/20170118Accessed_gwas_catalog_v1.0.1-associations_e87_r2017-01-09.tsv | perl -F\\t -lane 'if (!$F[11]) { @altvals = split(/\:/, $F[21]); $F[11] = $altvals[0]; $F[12] = $altvals[1]; } my @traits = split(/\s+/, $F[7]); print join("_", @traits), "\t", $F[11], "\t", $F[12], "\t", $F[27];' | perl -lane 'if ($#F == 3) { print join("\t", @F); }' | R -q -e "Data1 <- read.table(file('stdin'), header=T); Data2 <- Data1[Data1[,4] < 5e-08,]; Data3 <- Data1[Data1[,4] <= 5e-08,]; print(head(Data1)); print(c(nrow(Data1), nrow(Data2), nrow(Data3)));"
> Data1 <- read.table(file('stdin'), header=T); Data2 <- Data1[Data1[,4] < 5e-08,]; Data3 <- Data1[Data1[,4] <= 5e-08,]; print(head(Data1)); print(c(nrow(Data1), nrow(Data2), nrow(Data3)));
  DISEASE.TRAIT CHR_ID   CHR_POS P.VALUE
1 YKL-40_levels      1 203186754   1e-13
2     Psoriasis     13  39776775   2e-06
3   Lung_cancer     15  78513681   3e-18
4   Lung_cancer      1 159711078   7e-06
5   Lung_cancer      3 190632672   8e-06
6   Lung_cancer     15  78513681   5e-20
[1] 33976 13198 13483
> 
> 
.
.
.
[  mturchin20@spudhead  ~/Data/PriceLab/LDScore/baseline]$ls -lrt /mnt/lustre/home/mturchin20/Data/PriceLab/LDScore/baseline/. | grep UCSC
-rw-rw-r-- 1 mturchin20 mturchin20  4680349 Apr 12  2016 Coding_UCSC.bed
-rw-rw-r-- 1 mturchin20 mturchin20  2989152 Apr 12  2016 Coding_UCSC.extend.500.bed
-rw-rw-r-- 1 mturchin20 mturchin20   427789 Apr 12  2016 Intron_UCSC.extend.500.bed
-rw-rw-r-- 1 mturchin20 mturchin20  4115451 Apr 12  2016 Intron_UCSC.bed
-rw-rw-r-- 1 mturchin20 mturchin20   524360 Apr 12  2016 Promoter_UCSC.extend.500.bed
-rw-rw-r-- 1 mturchin20 mturchin20   535603 Apr 12  2016 Promoter_UCSC.bed
-rw-rw-r-- 1 mturchin20 mturchin20  1512478 Apr 12  2016 UTR_5_UCSC.extend.500.bed
-rw-rw-r-- 1 mturchin20 mturchin20  2241626 Apr 12  2016 UTR_5_UCSC.bed
-rw-rw-r-- 1 mturchin20 mturchin20  1120869 Apr 12  2016 UTR_3_UCSC.extend.500.bed
-rw-rw-r-- 1 mturchin20 mturchin20  1471595 Apr 12  2016 UTR_3_UCSC.bed
[  mturchin20@spudhead  ~/Data/PriceLab/LDScore/baseline]$ls -lrt /mnt/lustre/home/mturchin20/Data/PriceLab/LDScore/baseline/. | grep -v 500 | grep -v UCSC
total 650716
-rw-rw-r-- 1 mturchin20 mturchin20 84072024 Apr 12  2016 Conserved_LindbladToh.bed
-rw-rw-r-- 1 mturchin20 mturchin20  3262823 Apr 12  2016 CTCF_Hoffman.bed
-rw-rw-r-- 1 mturchin20 mturchin20 45317629 Apr 12  2016 DGF_ENCODE.bed
-rw-rw-r-- 1 mturchin20 mturchin20 42809087 Apr 12  2016 DHS_peaks_Trynka.bed
-rw-rw-r-- 1 mturchin20 mturchin20 32929947 Apr 12  2016 DHS_Trynka.bed
-rw-rw-r-- 1 mturchin20 mturchin20 17875890 Apr 12  2016 FetalDHS_Trynka.bed
-rw-rw-r-- 1 mturchin20 mturchin20  4445956 Apr 12  2016 Enhancer_Hoffman.bed
-rw-rw-r-- 1 mturchin20 mturchin20  1028556 Apr 12  2016 Enhancer_Andersson.bed
-rw-rw-r-- 1 mturchin20 mturchin20 17573287 Apr 12  2016 H3K27ac_PGC2.bed
-rw-rw-r-- 1 mturchin20 mturchin20  2143370 Apr 12  2016 H3K27ac_Hnisz.bed
-rw-rw-r-- 1 mturchin20 mturchin20 65983033 Apr 12  2016 H3K4me1_peaks_Trynka.bed
-rw-rw-r-- 1 mturchin20 mturchin20 19606778 Apr 12  2016 H3K4me3_peaks_Trynka.bed
-rw-rw-r-- 1 mturchin20 mturchin20 10077188 Apr 12  2016 H3K4me3_Trynka.bed
-rw-rw-r-- 1 mturchin20 mturchin20 17726338 Apr 12  2016 H3K4me1_Trynka.bed
-rw-rw-r-- 1 mturchin20 mturchin20 19651510 Apr 12  2016 H3K9ac_peaks_Trynka.bed
-rw-rw-r-- 1 mturchin20 mturchin20  9336764 Apr 12  2016 H3K9ac_Trynka.bed
-rw-rw-r-- 1 mturchin20 mturchin20  2438493 Apr 12  2016 PromoterFlanking_Hoffman.bed
-rw-rw-r-- 1 mturchin20 mturchin20 17860191 Apr 12  2016 TFBS_ENCODE.bed
-rw-rw-r-- 1 mturchin20 mturchin20   206622 Apr 12  2016 SuperEnhancer_Hnisz.bed
-rw-rw-r-- 1 mturchin20 mturchin20 53217162 Apr 12  2016 Repressed_Hoffman.bed
-rw-rw-r-- 1 mturchin20 mturchin20 50555834 Apr 12  2016 Transcribed_Hoffman.bed
-rw-rw-r-- 1 mturchin20 mturchin20  1628292 Apr 12  2016 TSS_Hoffman.bed
-rw-rw-r-- 1 mturchin20 mturchin20  5758664 Apr 12  2016 WeakEnhancer_Hoffman.bed
[  mturchin20@spudhead  ~/Data/PriceLab/LDScore/baseline]$ls -lrt /mnt/lustre/home/mturchin20/Data/PriceLab/LDScore/baseline/. | grep -v 500 | grep -v UCSC | wc
     24     209    1803
[  mturchin20@spudhead  ~/Data/PriceLab/LDScore/baseline]$for i in `ls -lrt /mnt/lustre/home/mturchin20/Data/PriceLab/LDScore/baseline/. | grep -v 500 | grep -v UCSC | awk '{ print $9 }'`; do
>         echo $i
>         cat /mnt/lustre/home/mturchin20/Data/PriceLab/LDScore/baseline/$i >> /mnt/lustre/home/mturchin20/Data/PriceLab/LDScore/baseline/20170118.LDScore.RegulatoryRegions.Collapsed.bed
> done
Conserved_LindbladToh.bed
CTCF_Hoffman.bed
DGF_ENCODE.bed
DHS_peaks_Trynka.bed
DHS_Trynka.bed
FetalDHS_Trynka.bed
Enhancer_Hoffman.bed
Enhancer_Andersson.bed
H3K27ac_PGC2.bed
H3K27ac_Hnisz.bed
H3K4me1_peaks_Trynka.bed
H3K4me3_peaks_Trynka.bed
H3K4me3_Trynka.bed
H3K4me1_Trynka.bed
H3K9ac_peaks_Trynka.bed
H3K9ac_Trynka.bed
PromoterFlanking_Hoffman.bed
TFBS_ENCODE.bed
SuperEnhancer_Hnisz.bed
Repressed_Hoffman.bed
Transcribed_Hoffman.bed
TSS_Hoffman.bed
WeakEnhancer_Hoffman.bed
[  mturchin20@spudhead  ~/Data/PriceLab/LDScore/baseline]$cat /mnt/lustre/home/mturchin20/Data/PriceLab/LDScore/baseline/20170118.LDScore.RegulatoryRegions.Collapsed.bed | wc
21967944 65903832 525505438
[  mturchin20@spudhead  ~/Data/PriceLab/LDScore/baseline]$ls -lrth /mnt/lustre/home/mturchin20/Data/PriceLab/LDScore/baseline/20170118.LDScore.RegulatoryRegions.Collapsed.bed     
-rw-rw-r-- 1 mturchin20 mturchin20 502M Jan 18 15:42 /mnt/lustre/home/mturchin20/Data/PriceLab/LDScore/baseline/20170118.LDScore.RegulatoryRegions.Collapsed.bed
[  mturchin20@spudhead  ~/Data/GTEX/Vs6p/GTEx_Analysis_v6p_eQTL]$ls -lrt *signif* | wc
     44     396    5121
[  mturchin20@spudhead  ~/Data/GTEX/Vs6p/GTEx_Analysis_v6p_eQTL]$ls -lrt . | wc       
     89     794    9639
[  mturchin20@spudhead  ~/Data/GTEX/Vs6p/GTEx_Analysis_v6p_eQTL]$zcat *signif* | head -n 10
variant_id      gene_id tss_distance    pval_nominal    slope   slope_se        slope_fpkm      slope_fpkm_se   pval_nominal_threshold  min_pval_nominal        pval_beta
1_98929_A_G_b37 ENSG00000237094.7       -218791 6.08809e-06     0.713632        0.15447 0.195573        0.0445252       0.000161646     2.14627e-12     1.38918e-08
1_662622_G_A_b37        ENSG00000237094.7       344902  1.07414e-05     0.516278        0.114963        0.169576        0.0326117       0.000161646     2.14627e-12     1.38918e-08
1_693731_A_G_b37        ENSG00000237094.7       376011  8.16602e-06     0.426614        0.0936897       0.130506        0.0267584       0.000161646     2.14627e-12     1.38918e-08
1_727841_G_A_b37        ENSG00000237094.7       410121  5.75046e-05     0.653994        0.159848        0.240116        0.0449424       0.000161646     2.14627e-12     1.38918e-08
1_729679_C_G_b37        ENSG00000237094.7       411959  7.59141e-07     -0.401875       0.0792457       -0.135562       0.0223193       0.000161646     2.14627e-12     1.38918e-08
1_730087_T_C_b37        ENSG00000237094.7       412367  5.75712e-05     0.653365        0.159706        0.239827        0.0449037       0.000161646     2.14627e-12     1.38918e-08
1_731718_T_C_b37        ENSG00000237094.7       413998  7.93067e-08     0.475891        0.0860707       0.147388        0.0244757       0.000161646     2.14627e-12     1.38918e-08
1_734349_T_C_b37        ENSG00000237094.7       416629  4.26455e-08     0.490069        0.0867381       0.152384        0.0246401       0.000161646     2.14627e-12     1.38918e-08
1_736289_T_A_b37        ENSG00000237094.7       418569  9.15939e-09     0.495733        0.0834283       0.157718        0.0235801       0.000161646     2.14627e-12     1.38918e-08
[  mturchin20@spudhead  ~/Data/GTEX/Vs6p/GTEx_Analysis_v6p_eQTL]$zcat *signif* | wc        
26393329 290326619 3431417429
[  mturchin20@spudhead  ~/Data/GTEX/Vs6p/GTEx_Analysis_v6p_eQTL]$zcat *signif* | awk '{ print $4 }' | grep -v pval | R -q -e "Data1 <- read.table(file('stdin'), header=F); quantile(Data1[,1]);"
> Data1 <- read.table(file('stdin'), header=F); quantile(Data1[,1]);
          0%          25%          50%          75%         100% 
4.53555e-151  9.43964e-11  3.10780e-07  9.36221e-06  4.06250e-02 
> 
> 
[  mturchin20@spudhead  ~/Data/GTEX/Vs6p/GTEx_Analysis_v6p_eQTL]$cat /mnt/lustre/home/mturchin20/Data/GTEX/Vs6p/GTEx_Analysis_v6p_eQTL/AllTraits_Analysis.v6p.signif_snpgene_pairs.bed | wc
26393285 79179855 626117578
[  mturchin20@spudhead  ~/Data/GTEX/Vs6p/GTEx_Analysis_v6p_eQTL]$cat /mnt/lustre/home/mturchin20/Data/GTEX/Vs6p/GTEx_Analysis_v6p_eQTL/AllTraits_Analysis.v6p.signif_snpgene_pairs.bed | head -n 10
chr1    98929   98929
chr1    662622  662622
chr1    693731  693731
chr1    727841  727841
chr1    729679  729679
chr1    730087  730087
chr1    731718  731718
chr1    734349  734349
chr1    736289  736289
chr1    748878  748878
[  mturchin20@spudling91  ~]$cat /mnt/lustre/home/mturchin20/Data/GTEX/Vs6p/GTEx_Analysis_v6p_eQTL/AllTraits_Analysis.v6p.signif_snpgene_pairs.bed | wc
26393285 79179855 626117578
[  mturchin20@spudling91  ~]$cat /mnt/lustre/home/mturchin20/Data/GTEX/Vs6p/GTEx_Analysis_v6p_eQTL/AllTraits_Analysis.v6p.signif_snpgene_pairs.1kbPadding.bed | wc
26393285 79179855 626116895
[  mturchin20@spudling91  ~]$cat /mnt/lustre/home/mturchin20/Data/GTEX/Vs6p/GTEx_Analysis_v6p_eQTL/AllTraits_Analysis.v6p.signif_snpgene_pairs.5kbPadding.bed | wc
26393285 79179855 626117111
[  mturchin20@spudling91  ~]$cat /mnt/lustre/home/mturchin20/Data/GTEX/Vs6p/GTEx_Analysis_v6p_eQTL/AllTraits_Analysis.v6p.signif_snpgene_pairs.10kbPadding.bed | wc
26393285 79179855 626116990
[  mturchin20@spudling91  ~]$cat /mnt/lustre/home/mturchin20/Data/GTEX/Vs6p/GTEx_Analysis_v6p_eQTL/AllTraits_Analysis.v6p.signif_snpgene_pairs.20kbPadding.bed | wc
26393285 79179855 626118729
.
.
.
#20170118 -- in relation to redoing annotations via annovar using hg18 instead of hg19 for the approriate datasets
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.Marker.noNA.NArplcd.AnnovarFormat.genome_summary.csv | grep APOB
"exonic","APOB","synonymous SNV","NM_000384:c.G13013G:p.S4338S",,,,,,,,,,,,,,,,,,,,2,21078786,21078786,T,C,
"exonic","APOB","synonymous SNV","NM_000384:c.C293C:p.T98T","450;Name=lod=89",,,,,,,,,,,,,,,,,,,2,21117405,21117405,A,G,
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/OLD1.WRONG1.hg19.GlobalLipids2010.Table1.Marker.noNA.NArplcd.AnnovarFormat.genome_summary.csv | grep APOB
"intergenic","C2orf43(dist=55896),APOB(dist=145515)",,,,,,,,,,,,,,,,,,,,2,21078786,21078786,T,C,
"intergenic","C2orf43(dist=94515),APOB(dist=106896)",,,"507;Name=lod=153",,,,,,,,,,,,,,,,,2,21117405,21117405,A,G,
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.Marker.noNA.NArplcd.AnnovarFormat.genome_summary.csv | grep FRK 
"intronic","FRK",,,,,,0.31,0.32,0.16,rs9488822,,,,,,,,,,,,,6,116419586,116419586,A,T,
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/OLD1.WRONG1.hg19.GlobalLipids2010.Table1.Marker.noNA.NArplcd.AnnovarFormat.genome_summary.csv | grep FRK 
"intergenic","FRK(dist=37665),NT5DC1(dist=2413)",,,,,,,,,,,,,,,,,,,,6,116419586,116419586,A,T,
#20170118 -- going through everything again now and using/replacing/setting up files with hg18 standard instead
[  mturchin20@spudhead  ~/Data/GTEX/Vs6p]$join <(cat /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg18.chrom.sizes | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg19.chrom.sizes | sort -k 1,1) | wc
     25      75     614
[  mturchin20@spudhead  ~/Data/GTEX/Vs6p]$join <(cat /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg18.chrom.sizes | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg19.chrom.sizes | sort -k 1,1) | head -n 10
chr1 247249719 249250621
chr10 135374737 135534747
chr11 134452384 135006516
chr12 132349534 133851895
chr13 114142980 115169878
chr14 106368585 107349540
chr15 100338915 102531392
chr16 88827254 90354753
chr17 78774742 81195210
chr18 76117153 78077248
[  mturchin20@spudhead  ~/Data/GTEX/Vs6p]$join -v2 <(cat /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg18.chrom.sizes | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg19.chrom.sizes | sort -k 1,1) | wc        
     68     136    1595
[  mturchin20@spudhead  ~/Data/GTEX/Vs6p]$join -v2 <(cat /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg18.chrom.sizes | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg19.chrom.sizes | sort -k 1,1) | head -n 10
chr11_gl000202_random 40103
chr17_ctg5_hap1 1680828
chr17_gl000203_random 37498
chr17_gl000204_random 81310
chr17_gl000205_random 174588
chr17_gl000206_random 41001
chr18_gl000207_random 4262
chr19_gl000208_random 92689
chr19_gl000209_random 159169
chr1_gl000191_random 106433
[  mturchin20@spudhead  ~/Data/GTEX/Vs6p]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170105.UCSC_hg19_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.txt | wc
  28489  142445  863923
[  mturchin20@spudhead  ~/Data/GTEX/Vs6p]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170118.UCSC_hg18_GeneInformation_wSymbols.NoProtAlignIDs.Condensed.txt | wc
  29348  146740  891200
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(cat <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Condensed.txt.gz) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | sed 's/chr//g') | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Counts.txt | sed 's/chr//g' | awk '{ print $1 "_" $2 "\t" $4 }' | sort -k 1,1) | awk '{ if ($2 == $4) { print $0 } }' | wc
2593966 10375864 116306714
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Condensed.txt.gz | wc
2384946 7154838 107662272
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | wc
 209020  627060 3907296
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$R -q -e "2384946 + 209020"
> 2384946 + 209020
[1] 2593966
>
>
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(cat <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Condensed.txt.gz) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | sed 's/chr//g') | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Counts.txt | sed 's/chr//g' | awk '{ print $1 "_" $2 "\t" $4 }' | sort -k 1,1) | awk '{ if ($2 == $4) { print $0 } }' | wc
2593966 10375864 122736499
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Condensed.txt.gz | wc
2427319 7281957 114726421
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | wc
 166647  499941 3119359
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$R -q -e "2427319 + 166647"
> 2427319 + 166647
[1] 2593966
> 
> 
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(cat <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_250kbPadding.Condensed.txt.gz) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_250kbPadding.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | sed 's/chr//g') | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_250kbPadding.Counts.txt | sed 's/chr//g' | awk '{ print $1 "_" $2 "\t" $4 }' | sort -k 1,1) | awk '{ if ($2 == $4) { print $0 } }' | wc
2593966 10375864 201330748
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_250kbPadding.Condensed.txt.gz | wc
2541422 7624266 194588080
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_250kbPadding.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | wc
  52544  157632  985278
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$R -q -e "2541422 + 52544"
> 2541422 + 52544
[1] 2593966
> 
> 
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.AnnotatedForPerms.NoPad.20kbPad.250kbPad.vs1.txt.gz | wc
2593966 25939660 110042826
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.AnnotatedForPerms.NoPad.20kbPad.250kbPad.vs1.txt.gz | perl -lane 'print $#F;' | sort | uniq -c
2593966 9
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(cat <(zcat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Condensed.txt.gz) <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | sed 's/chr//g') | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Counts.txt | sed 's/chr//g' | awk '{ print $1 "_" $2 "\t" $4 }' | sort -k 1,1) | awk '{ if ($2 == $4) { print $0 } }' | wc
    102     408    5660
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Condensed.txt.gz | wc
     99     297    5393
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | wc
      3       9      57
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(cat <(zcat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Condensed.txt.gz) <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | sed 's/chr//g') | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Counts.txt | sed 's/chr//g' | awk '{ print $1 "_" $2 "\t" $4 }' | sort -k 1,1) | awk '{ if ($2 == $4) { print $0 } }' | wc
    102     408    6356
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Condensed.txt.gz | wc
    100     300    6101
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_20kbPadding.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | wc
      2       6      38
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(cat <(zcat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_250kbPadding.Condensed.txt.gz) <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_250kbPadding.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | sed 's/chr//g') | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_250kbPadding.Counts.txt | sed 's/chr//g' | awk '{ print $1 "_" $2 "\t" $4 }' | sort -k 1,1) | awk '{ if ($2 == $4) { print $0 } }' | wc
    102     408   12796
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_250kbPadding.Condensed.txt.gz | wc
    102     306   12536
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.20170118_UCSC_hg18_GeneInformation_wSymbols_NoProtAlignIDs_Condensed_wSymbol_250kbPadding.Counts.txt | awk '{ if ($4 == 0) { print $1 "_" $2 "\t0\t0" } }' | wc
      0       0       0






~~~











##Checking new 'old' bmass hits replicate in new 'orig'(/'prev') hits
##20161228

#GlobalLipids2010
#20161228

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.txt | awk '{ print $1 "\t" $2 "\t" $3 }' | grep -v SNP > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.MarkerChrBP.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.MarkerChrBP.txt | awk '{ print $2 "_" $3 "\t" $2 "\t" $3 }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.ChrBPChrBP.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.MarkerChrBP.txt | awk '{ print $2 "_" $3 "\t" $2 "\t" $3 }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.ChrBPChrBP.txt

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.MarkerChrBP.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.MarkerChrBP.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.OverlapWith.ng_2797-S1_edited_vs2_carriageRemoved_rsIDs_HapMart.Marker.vs1.txt
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.ChrBPChrBP.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.ChrBPChrBP.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.OverlapWith.ng_2797-S1_edited_vs2_carriageRemoved_rsIDs_HapMart.ChrBP.vs1.txt

[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.OverlapWith.ng_2797-S1_edited_vs2_carriageRemoved_rsIDs_HapMart.Marker.vs1.txt | awk '{ if ($4 == 1) { print $0 } } '
rs10904908      10      17300296        1       rs10904908      10      17300296        1
rs11246602      11      51368666        1       rs11246602      11      51368666        1
rs13326165      3       52507158        1       rs13326165      3       52507158        1
rs267733        1       149225460       1       rs267733        1       149225460       1
rs4722551       7       25958351        1       rs4722551       7       25958351        1
rs499974        11      75132669        1       rs499974        11      75132669        1
rs970548        10      45333283        1       rs970548        10      45333283        1
rs998584        6       43865874        1       rs998584        6       43865874        1
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.OverlapWith.ng_2797-S1_edited_vs2_carriageRemoved_rsIDs_HapMart.Marker.vs1.txt | awk '{ if ($4 == 2) { print $0 } } '
rs10490626      2       118552311       2       rs10490632      2       118295555       2
rs12748152      1       27010980        2       rs12739698      1       27102620        2
rs17695224      19      57016028        2       rs10422101      19      57011927        2
rs1832007       10      5244847 2       rs17134533      10      5237098 2
rs1997243       7       1050303 2       rs6951245       7       1024719 2
rs4942486       13      31851388        2       rs4942505       13      31861707        2
rs6831256       4       3442937 2       rs762861        4       3411809 2

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.OverlapWith.ng_2797-S1_edited_vs2_carriageRemoved_rsIDs_HapMart.Marker.vs1.txt | awk '{ print $5 "\t" $6 "\t" $7 "\t" $4 }' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.OverlapWith.ng_2797-S1_edited_vs2_carriageRemoved_rsIDs_HapMart.Marker.vs1.Hits.Uniq.txt

cat <(join -v 1 <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.MarkerChrBP.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.OverlapWith.ng_2797-S1_edited_vs2_carriageRemoved_rsIDs_HapMart.Marker.vs1.Hits.Uniq.txt | sort -k 1,1) | awk '{ print $0 "\t0" }') /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.OverlapWith.ng_2797-S1_edited_vs2_carriageRemoved_rsIDs_HapMart.Marker.vs1.Hits.Uniq.txt | grep rs > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.OverlapWith.ng_2797-S1_edited_vs2_carriageRemoved_rsIDs_HapMart.Marker.vs1.Hits.Uniq.wNoHits.txt 

[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.OverlapWith.ng_2797-S1_edited_vs2_carriageRemoved_rsIDs_HapMart.Marker.vs1.txt | awk '{ if ($4 == 2) { print $0 } } ' | perl -lane 'my $diff = $F[6] - $F[2]; print join("\t", @F), "\t", $diff;' | sort -g -k 9,9
rs10490626      2       118552311       2       rs10490632      2       118295555       2       -256756
rs6831256       4       3442937 2       rs762861        4       3411809 2       -31128
rs1997243       7       1050303 2       rs6951245       7       1024719 2       -25584
rs1832007       10      5244847 2       rs17134533      10      5237098 2       -7749
rs17695224      19      57016028        2       rs10422101      19      57011927        2       -4101
rs4942486       13      31851388        2       rs4942505       13      31861707        2       10319
rs12748152      1       27010980        2       rs12739698      1       27102620        2       91640

[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.OverlapWith.ng_2797-S1_edited_vs2_carriageRemoved_rsIDs_HapMart.Marker.vs1.Hits.Uniq.wNoHits.txt | sort -k 4,4 -gk 2,2
rs2746150 6 29550680    0
rs1408579 10 101902184  0
rs11227638 11 55776161  0
rs11229252 11 54886216  0
rs267733        1       149225460       1
rs13326165      3       52507158        1
rs998584        6       43865874        1
rs4722551       7       25958351        1
rs10904908      10      17300296        1
rs970548        10      45333283        1
rs11246602      11      51368666        1
rs499974        11      75132669        1
rs12739698      1       27102620        2
rs10490632      2       118295555       2
rs762861        4       3411809 2
rs6951245       7       1024719 2
rs17134533      10      5237098 2
rs4942505       13      31861707        2
rs10422101      19      57011927        2

join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.MarkerChrBP.txt | sort -k 1,1) <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.tbl.gz | awk '{ print $1 "\t" $2 "\t" $3 }' | sort -k 1,1) | perl -lane 'print $F[1], "\t", $F[2], "\t", $F[2], "\t", uc($F[3]), "\t", uc($F[4]);' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.AnnovarFormat

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.txt | awk '{ print $1 "\t" $2 "\t" $2 "\t" $4 "\t" $6 }' | sed 's/chr//g' | sed 's/\// /g' | perl -lane 'if ($F[$#F-1] eq $F[$#F]) { ($F[4], $F[3]) = ($F[3], $F[4]); } print join("\t", @F[0..$#F-1]);' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.AnnovarFormat
#20170216 NOTE -- The last two entries of this file are lacking the alleles (originally lacking in the HapMart file), so manually putting them in -- rs1047891 is C/A (https://www.snpedia.com/index.php/Rs1047891) and https://www.snpedia.com/index.php/Rs635634 is G/A (https://www.snpedia.com/index.php/Rs635634)


##perl /mnt/lustre/home/mturchin20/Software/annovar/summarize_annovar.pl /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.AnnovarFormat /mnt/lustre/home/mturchin20/Software/annovar/humandb/ --ver1000g 1000g2012apr --verdbsnp 129 --veresp 6500 --buildver hg19 --remove
mv /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.AnnovarFormat.genome_summary.csv /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/OLD1.WRONG1.hg19.GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.AnnovarFormat.genome_summary.csv
perl /mnt/lustre/home/mturchin20/Software/annovar/summarize_annovar.pl /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.AnnovarFormat /mnt/lustre/home/mturchin20/Software/annovar/humandb/ --ver1000g 1000g2010 --verdbsnp 129 --veresp 6500 --buildver hg18 --remove

##perl /mnt/lustre/home/mturchin20/Software/annovar/summarize_annovar.pl /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.AnnovarFormat /mnt/lustre/home/mturchin20/Software/annovar/humandb/ --ver1000g 1000g2012apr --verdbsnp 129 --veresp 6500 --buildver hg19 --remove
mv /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.AnnovarFormat.genome_summary.csv /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/OLD1.WRONG1.hg19.ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.AnnovarFormat.genome_summary.csv
perl /mnt/lustre/home/mturchin20/Software/annovar/summarize_annovar.pl /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.AnnovarFormat /mnt/lustre/home/mturchin20/Software/annovar/humandb/ --ver1000g 1000g2010 --verdbsnp 129 --veresp 6500 --buildver hg18 --remove

#From Stephens 2013 PLoS ONE http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0065245
```
The associations reported in Table 2 were identified by first ranking SNPs by their Bayes Factor for association (), and then removing multiple associations that were likely due to LD by eliminating any SNP that was within 0.5Mb of a higher-ranked SNP. We also removed a SNP (rs2746150) on chromosome 6, which we judged likely a secondary association due to LD with previously-reported associations in the MHC region.
```
#Additionally Table 2 only shows 18 SNPs, not 19. So in terms of reported results, rs2746150 was not even included. So 15/18 results.

#re: rs1408579 (note -- I bring up rs2862954 on my own earlier in the my work here but can't discern why I thought it mattered/why I did so before. Can probably just ignore for now)
#here is what Matthew says in Stephens 2013 main text re: ERLIN1
```
ERLIN1 codes for the protein Erlin-1, which is a member of the prohibitin family of proteins that define lipid-raft-like domains of the endoplasmic reticulum [47]. SNPs near ERLIN1 have previously been associated with plasma levels of alanine-aminotransferase [48], an important liver enzyme.

47-Browman DT, Resek ME, Zajchowski LD, Robbins SM (2006) Erlin-1 and erlin-2 are novel members of the prohibitin family of proteins that define lipid-raft-like domains of the er. J Cell Sci 119: 3149–60.
48-Yuan X, Waterworth D, Perry JRB, Lim N, Song K, et al. (2008) Population-based genome-wide association studies reveal six loci inuencing plasma levels of liver enzymes. Am J Hum Genet 83: 520–8.
```

#There are two other, more recent sources connecting ERLIN1 to NAFLD
Feitosa et al. 2013 Atherosclerosis The ERLIN1-CHUK-CWF19L1 gene cluster influences liver fat deposition and hepatic inflammation in the NHLBI Family Heart Study. https://www.ncbi.nlm.nih.gov/pubmed/23477746
Edelman et al. 2015 Molecular Genetics & Genomic Medicine Genetic analysis of nonalcoholic fatty liver disease within a Caribbean–Hispanic population http://onlinelibrary.wiley.com/doi/10.1002/mgg3.168/full

Feitosa has both rs1408579 & rs2862954, whereas Edelman only seems to refer to rs2862954 and it's unclear whether that SNP specifically is significant in their study? But they do indicate ERLIN1, as part of a complex, is related to NAFLD. The former study is more explicit in linking all these things together.

#notes from earlier work on this
```
#rs2862954 is not within .5Mb of a hit, but it is in ERLIN1 which is upstream of BLOC1S2 and CHUK, both of which come up as 'enriched genes' in the pathway analyses. BF is on the lower end (4.8) in the Matthew paper table 2, though other loci with lower BFs did get picked up, e.g. VEGFA with 4.5 and STAB1 with 4.7

#Both rs11229252 and rs11227638 are in the OR4* (olfactory receptor) region of chromosome 11 -- they are also not showing up via those rsID #s in the new dataset, but presumably they are being considered 'tagged' by being nearby any type of OR4*, which has already shown up in the 2013 paper in the form of OR4C46
```

#re: rs11227638 & rs11229252
#See OR4* references above
#rs11227638 11 55776161  0
#rs11229252 11 54886216  0

#Looks like neither SNP reaches GW significance in the univariate results from 2013 data (see below)...so just not significant. rs11246602 is significant though and it is 42kb nearby OR4A5...? Don't see OR4C46 anywhere as I suggest above in some earlier work I did on this? Checked the 2013 GlobalLipids manuscript for this as well.
#Didn't see OR4C46 because of the hg19/hg18 annotation messup before. See it now (see below, the SNP is within ~4kb of the locus, and the annotation now picks up that the SNP is rs11246602 too). So while the other 2 SNPs don't reach genome-wide significance anyways, encouraging that they tag olfactory loci that presumably are also far away enough that it isn't solely secondary signals produced from LD? Unless olfactory regions produce long-range LD blocks

#From Table 1 of http://www.nature.com/ng/journal/v45/n11/full/ng.2797.html 
```
.
.
.
OR4C46	rs11246602	11	51.51	HDL	0.15	C/T	0.034	176	2 × 10−10
.
.
.
```

#20170118 -- GlobalLipids2010 was in hg18 and for the analyses done so far, used hg18 for GlobalLipids2013 as well (see notes from earlier in logfile, printed below). Used HapMart to get info from GlobalLipids2013 results and got output in hg18 format.
```
.
.
.
#/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.txt was run through HapMart to get the chromosome and position locations for each rsID. 2 SNPs were missing and their information was manually included. These 2 SNPs were rs1047891 and rs9411489. Information was retrieved from UCSC Genome Browser hg18 and included in the /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.txt file
#hg18 because HapMart only had hg18 but switching to hg19 with the final annotated file
#rs1047891 chr2 211248752
#rs9411489 chr
#NOTE -- Previously reported ABO SNP rs9411489 has merged into rs635634 (source: supplementary of http://hmg.oxfordjournals.org/content/21/6/1444.full or https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=2&cad=rja&uact=8&ved=0CCsQFjAB&url=http%3A%2F%2Fhmg.oxfordjournals.org%2Fcontent%2Fsuppl%2F2011%2F12%2F07%2Fddr581.DC1%2Fddr581supp_table1.doc&ei=HxKiU8_OIYOOyAS-tICoCg&usg=AFQjCNHO1Honaz_0R7CcVLsZan3LNJulKQ&sig2=TIdN-hlZUyRQZeMxiDGWHg&bvm=bv.69137298,d.aWw)
#rs635634 chr9 135144821
.
.
.
```

perl /mnt/lustre/home/mturchin20/Software/annovar/annotate_variation.pl --downdb -webfrom annovar refGene /mnt/lustre/home/mturchin20/Software/annovar/humandb/ -build hg18
perl /mnt/lustre/home/mturchin20/Software/annovar/annotate_variation.pl --downdb genomicSuperDups /mnt/lustre/home/mturchin20/Software/annovar/humandb/ -build hg18
perl /mnt/lustre/home/mturchin20/Software/annovar/annotate_variation.pl --downdb -webfrom annovar snp129 /mnt/lustre/home/mturchin20/Software/annovar/humandb/ -build hg18
perl /mnt/lustre/home/mturchin20/Software/annovar/annotate_variation.pl --downdb -webfrom annovar avsift /mnt/lustre/home/mturchin20/Software/annovar/humandb/ -build hg18
perl /mnt/lustre/home/mturchin20/Software/annovar/annotate_variation.pl --downdb -webfrom annovar ljb_all /mnt/lustre/home/mturchin20/Software/annovar/humandb/ -build hg18
perl /mnt/lustre/home/mturchin20/Software/annovar/annotate_variation.pl --downdb -webfrom annovar ljb23_all /mnt/lustre/home/mturchin20/Software/annovar/humandb/ -build hg18
perl /mnt/lustre/home/mturchin20/Software/annovar/annotate_variation.pl --downdb -webfrom annovar esp6500_all /mnt/lustre/home/mturchin20/Software/annovar/humandb/ -build hg18
perl /mnt/lustre/home/mturchin20/Software/annovar/annotate_variation.pl --downdb -webfrom annovar 1000g2012apr /mnt/lustre/home/mturchin20/Software/annovar/humandb/ -build hg18
perl /mnt/lustre/home/mturchin20/Software/annovar/annotate_variation.pl --downdb -webfrom annovar 1000g2010 /mnt/lustre/home/mturchin20/Software/annovar/humandb/ -build hg18
perl /mnt/lustre/home/mturchin20/Software/annovar/annotate_variation.pl --downdb phastConsElements44way /mnt/lustre/home/mturchin20/Software/annovar/humandb/ -build hg18
#20170123
perl /mnt/lustre/home/mturchin20/Software/annovar/annotate_variation.pl --downdb -webfrom annovar ljb26_all /mnt/lustre/home/mturchin20/Software/annovar/humandb/ -build hg18
perl /mnt/lustre/home/mturchin20/Software/annovar/annotate_variation.pl --downdb -webfrom annovar dbnsfp33a /mnt/lustre/home/mturchin20/Software/annovar/humandb/ -build hg18

~~~
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.AnnovarFormat.genome_summary.csv | perl -F, -lane 'if ($F[$#F-4] == 11) { print join(",", @F); }'
"intronic","UEVLD",,,,,,,,,,,,,,,,,,,,11,18589560,18589560,T,C
"intergenic","PATE4(dist=39195),HYLS1(dist=4347)",,,,,,,,,,,,,,,,,,,,11,125749162,125749162,G,A
"intergenic","LOC646813(dist=988864),OR4A5(dist=42712)",,,,,,,,,,,,,,,,,,,,11,51368666,51368666,T,C
"downstream","TMPRSS4",,,,,,,,,,,,,,,,,,,,11,117991277,117991277,T,C
"intronic","SLC25A45",,,,,,,,,,,,,,,,,,,,11,65147893,65147893,G,A
"intronic","SYT7",,,,,,,,,,,,,,,,,,,,11,61326406,61326406,C,T
"ncRNA_intronic","CAND1.11",,,,,,,,,,,,,,,,,,,,11,10345358,10345358,A,G
"UTR3","ARHGAP1",,,,,,,,,,,,,,,,,,,,11,46699823,46699823,T,C
"intronic","RPS3",,,,,,,,,,,,,,,,,,,,11,75132669,75132669,C,A
"ncRNA_intronic","MIR100HG",,,,,,,,,,,,,,,,,,,,11,122027585,122027585,T,C
"intergenic","LINC00900(dist=523209),BUD13(dist=464759)",,,,,,,,,,,,,,,,,,,,11,116154127,116154127,G,C
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /data/external_public/GlobalLipid/jointGwasMc_*.txt.gz | grep rs11246602
chr11:51368666  chr11:51512090  rs11246602      c       t       0.034   0.0052  176028.03       1.681e-10       0.1332
chr11:51368666  chr11:51512090  rs11246602      c       t       0.0019  0.0056  162022.00       0.5259  0.1332
chr11:51368666  chr11:51512090  rs11246602      c       t       0.0112  0.0054  176253.43       0.01944 0.1332
chr11:51368666  chr11:51512090  rs11246602      t       c       0.0091  0.0052  166782.00       0.1924  0.8668
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /data/external_public/GlobalLipid/jointGwasMc_*.txt.gz | grep rs11227638
chr11:55776161  chr11:56019585  rs11227638      t       a       0.0369  0.0074  88834.00        1.785e-06       1
chr11:55776161  chr11:56019585  rs11227638      a       t       0.0029  0.0081  84452.00        0.754   NA
chr11:55776161  chr11:56019585  rs11227638      t       a       0.0135  0.0079  89074.00        0.03422 1
chr11:55776161  chr11:56019585  rs11227638      t       a       0.0007  0.0078  85492.00        0.8241  1
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /data/external_public/GlobalLipid/jointGwasMc_*.txt.gz | grep rs11229252
chr11:54886216  chr11:55129640  rs11229252      t       c       0.0363  0.0082  91826.00        4.602e-06       0.1201
chr11:54886216  chr11:55129640  rs11229252      t       c       0.0095  0.0090  87403.00        0.1372  0.1201
chr11:54886216  chr11:55129640  rs11229252      t       c       0.0253  0.0087  92110.00        0.001403        0.1201
chr11:54886216  chr11:55129640  rs11229252      t       c       0.0051  0.0086  88528.00        0.5878  0.1201
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.txt | awk '{ if ($2 == 11) { print $0 } }'
rs11227638 11 55776161 0.118 94423 0.03701 0.007065 90018 0.001169 0.007677 94663 0.01605 0.007535 91077 0.004328 0.007456 0 n.a. rs11227638 -0.637 -2.285 -0.445 -5.026 32.5420089289469 5.82908985750609 6.30032190381785 11_55776161 90018 4.79187824293809
rs11229252 11 54886216 0.091 97415 0.03709 0.007882 92969 0.01194 0.008648 97699 0.02784 0.00846 94113 0.006356 0.008351 0 n.a. rs11229252 0.538 3.373 1.54 4.966 33.862372338347 6.09951170574259 6.16527447867204 11_54886216 92969 4.9459696690849
rs11246602 11 51368666 0.126 99900 -0.03649 0.00687 95454 -0.003593 0.00745 100184 -0.01722 0.007312 96598 0.001288 0.006708 0 n.a. rs11246602 -0.168 -2.536 -0.708 -5.395 35.0428664214368 6.34178716857721 7.16416012975714 11_51368666 95454 5.3070092019464
rs499974 11 75132669 0.175 99852 0.02471 0.006034 95409 -0.002567 0.006509 100136 0.01451 0.006405 96550 0.01389 0.005872 0 n.a. rs499974 -2.098 -2.383 0.266 -4.11 34.1074338093301 6.14976851951062 4.40267882216328 11_75132669 95409 4.45521629192124
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.txt | grep rs11246602
chr11   51368666        rs11246602      C/T     2       T
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.AnnovarFormat.genome_summary.csv | perl -F, -lane 'if ($F[$#F-4] == 11) { print join(",", @F); }'
"intergenic","OR5F1(dist=14060),OR5AS1(dist=21734)",,,,,,,,,,,,,,,,,,,,11,55776161,55776161,A,T
"intergenic","NONE(dist=NONE),TRIM48(dist=143442)",,,,,,,,,,,,,,,,,,,,11,54886216,54886216,T,C
"intergenic","LOC646813(dist=988864),OR4A5(dist=42712)",,,,,,,,,,,,,,,,,,,,11,51368666,51368666,T,C
"intronic","RPS3",,,,,,,,,,,,,,,,,,,,11,75132669,75132669,A,C
.
.
.
#20170118
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/OLD1.WRONG1.hg19.ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.AnnovarFormat.genome_summary.csv | grep OR4
"intergenic","LOC646813(dist=988864),OR4A5(dist=42712)",,,,,,,,,,,,,,,,,,,,11,51368666,51368666,T,C,
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.AnnovarFormat.genome_summary.csv | grep OR4
"intergenic","OR4A5(dist=99642),OR4C46(dist=3192)",,,,"0.96",,,,,rs11246602,,,,,,,,,,,,,11,51368666,51368666,T,C,

~~~

#Edit /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py 
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.MarkerChrBP.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.MarkerChrBP.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.OverlapWith.ng_2797-S1_edited_vs2_carriageRemoved_rsIDs_HapMart.Marker.2MbWindow.vs1.txt
#Edit /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py 
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.MarkerChrBP.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.MarkerChrBP.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.OverlapWith.ng_2797-S1_edited_vs2_carriageRemoved_rsIDs_HapMart.Marker.4MbWindow.vs1.txt

~~~
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.MarkerChrBP.txt | head -n 10
rs10019888       4       25672088
rs10102164       8       55584167
rs10128711       11      18589560
rs10401969       19      19268718
rs10490626       2       118552311
rs10761731       10      64697616
rs1077514        1       23638820
rs10904908       10      17300296
rs11065987       12      110556807
rs11136341       8       145115531
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.OverlapWith.ng_2797-S1_edited_vs2_carriageRemoved_rsIDs_HapMart.Marker.vs1.txt | awk '{ if ($4 > 0) { print $0 } } ' | wc
     15     120     706
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.OverlapWith.ng_2797-S1_edited_vs2_carriageRemoved_rsIDs_HapMart.ChrBP.vs1.txt | awk '{ if ($4 > 0) { print $0 } } ' | wc
     15     120     744
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.MarkerChrBP.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.OverlapWith.ng_2797-S1_edited_vs2_carriageRemoved_rsIDs_HapMart.Marker.Hits.Uniq.vs1.txt | sort -k 1,1) | wc
     15      90     524
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join -v 1 <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.MarkerChrBP.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.OverlapWith.ng_2797-S1_edited_vs2_carriageRemoved_rsIDs_HapMart.Marker.Hits.Uniq.vs1.txt | sort -k 1,1) | wc
      4      12      90
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.OverlapWith.ng_2797-S1_edited_vs2_carriageRemoved_rsIDs_HapMart.Marker.4MbWindow.vs1.txt | awk '{ if ($4 > 0) { print $0 } } ' | wc
     16     128     754
[  mturchin20@spudhead  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs1]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.MarkerChrBP.txt | sort -k 1,1) <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.tbl.gz | awk '{ print $1 "\t" $2 "\t" $3 }' | sort -k 1,1) | head -n 10
rs10422101 19 57011927 c g
rs10490632 2 118295555 a g
rs10904908 10 17300296 a g
rs11227638 11 55776161 a t
rs11229252 11 54886216 t c
rs11246602 11 51368666 t c
rs12739698 1 27102620 a g
rs13326165 3 52507158 a g
rs1408579 10 101902184 t c
rs17134533 10 5237098 a g
[  mturchin20@spudhead  ~/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs1]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.MarkerChrBP.txt | sort -k 1,1) <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.tbl.gz | awk '{ print $1 "\t" $2 "\t" $3 }' | sort -k 1,1) | wc        
     19      95     489

~~~





#GIANT2010
#20161229

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT2010.newtophits.vs1.SignCrrct.vs1.txt | awk '{ print $1 "\t" $2 "\t" $3 }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT2010.newtophits.vs1.SignCrrct.vs1.MarkerChrBP.txt

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT2010.newtophits.vs1.SignCrrct.vs1.MarkerChrBP.txt | grep -v pos | python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 - --file2 /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT2010.newtophits.vs1.SignCrrct.vs1.OverlapWith.GIANT2014_5_AllGWASHits_HeightBMIWHRadjBMI_MarkerChrBP.vs1.txt 

~~~
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT2010.newtophits.vs1.SignCrrct.vs1.OverlapWith.GIANT2014_5_AllGWASHits_HeightBMIWHRadjBMI_MarkerChrBP.vs1.txt | awk '{ if ($4 > 0) { print $0 } } ' | awk '{ print $5 }' | sort | uniq | wc
     19      19     192
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT2010.newtophits.vs1.SignCrrct.vs1.OverlapWith.GIANT2014_5_AllGWASHits_HeightBMIWHRadjBMI_MarkerChrBP.vs1.txt | awk '{ if ($4 == 1) { print $0 } } ' | awk '{ print $5 }' | sort | uniq | wc
      7       7      70
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT2010.newtophits.vs1.SignCrrct.vs1.OverlapWith.GIANT2014_5_AllGWASHits_HeightBMIWHRadjBMI_MarkerChrBP.vs1.txt | awk '{ if ($4 == 2) { print $0 } } ' | awk '{ print $5 }' | sort | uniq | wc
     16      16     163
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT2010.newtophits.vs1.SignCrrct.vs1.OverlapWith.GIANT2014_5_AllGWASHits_HeightBMIWHRadjBMI_MarkerChrBP.vs1.txt | awk '{ if ($4 == 2) { print $0 } } ' | awk '{ print $5 }' | sort | uniq | grep -v -f <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT2010.newtophits.vs1.SignCrrct.vs1.OverlapWith.GIANT2014_5_AllGWASHits_HeightBMIWHRadjBMI_MarkerChrBP.vs1.txt | awk '{ if ($4 == 1) { print $0 } } ' | awk '{ print $5 }' | sort | uniq) | wc
     12      12     122
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT2010.newtophits.vs1.SignCrrct.vs1.OverlapWith.GIANT2014_5_AllGWASHits_HeightBMIWHRadjBMI_MarkerChrBP.vs1.txt | awk '{ if ($4 == 1) { print $0 } } ' | awk '{ print $5 }' | sort | uniq     
rs11835818
rs12204421
rs17783015
rs1809889
rs34651
rs6441170
rs648831
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT2010.newtophits.vs1.SignCrrct.vs1.OverlapWith.GIANT2014_5_AllGWASHits_HeightBMIWHRadjBMI_MarkerChrBP.vs1.txt | awk '{ if ($4 == 2) { print $0 } } ' | awk '{ print $5 }' | sort | uniq | grep -v -f <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT2010.newtophits.vs1.SignCrrct.vs1.OverlapWith.GIANT2014_5_AllGWASHits_HeightBMIWHRadjBMI_MarkerChrBP.vs1.txt | awk '{ if ($4 == 1) { print $0 } } ' | awk '{ print $5 }' | sort | uniq)     
rs10040888
rs12534698
rs12538581
rs1260326
rs2025151
rs3747579
rs389883
rs4735692
rs6824258
rs7601531
rs8016570
rs8071778
~~~

#20170123 -- coming back into this now
/mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.txt










#ICBP2011??
#20161229??


#HaemgenRBC2010
#20161229










##Checking whether expanding out window around new hits leads to overlap with previous hits (to just check/take a look at proximity of new hits to old hits even if originally determined to be outside 1Mb window)
##20170123

#GlobalLipids2010
#20170123

mv /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs2.py /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/20170113_NEVERUSED_DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs2.py

#NOTE -- /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs2.py was copied and intended to become something (possibly as an idea), but was never followed up on (or had yet to be at least). So for the time being moving the file out away from the file hierarchy and creating a new vs3 for what is needed/we're interested in doing below. For the idea originally proposed for `vs2`, if we come back to it we'll rename a script `vs4` to keep in line chronologically (at least that's what's being proposed in my head as I type this right now).

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs3.py

cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.txt | awk '{ print $1 "_" $2 "\t" $1 "\t" $2 }' > /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.ChrBPChrBP.txt

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.NewSNPs.SNPs.vs1.txt | tail -n +2 | awk '{ print $1 "\t" $2 "\t" $3 }' | python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs3.py --file1 - --file2 /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.ChrBPChrBP.txt --window 500000 > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.OverlapWith.bmass_GlobalLipids2010_Vs2_NewSNPs_SNPs_vs1_txt.Window500000.vs1.txt

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.NewSNPs.SNPs.vs1.txt | tail -n +2 | awk '{ print "chr" $2 "\t" $3 "\t" $3 }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.NewSNPs.SNPs.vs1.bed

/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/bedtools slop -i /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.NewSNPs.SNPs.vs1.bed -g /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg18.chrom.sizes -b 500000 > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bed
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/bedtools slop -i /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.NewSNPs.SNPs.vs1.bed -g /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg18.chrom.sizes -b 1000000 > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.NewSNPs.SNPs.vs1.1MbPadding.bed
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/bedtools slop -i /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.NewSNPs.SNPs.vs1.bed -g /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg18.chrom.sizes -b 2000000 > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bed

#20170207 NOTE -- added in `-f 1` to options for all below commands
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -f 1 -wb -a /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bed -b /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.bmass_GlobalLipids2010_Vs2_NewSNPs_SNPs_vs1_500kbPadding_bed.vs1.txt 
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -f 1 -wb -a /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bed -b /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.NewSNPs.SNPs.vs1.1MbPadding.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.bmass_GlobalLipids2010_Vs2_NewSNPs_SNPs_vs1_1MbPadding_bed.vs1.txt 
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -f 1 -wb -a /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bed -b /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.bmass_GlobalLipids2010_Vs2_NewSNPs_SNPs_vs1_2MbPadding_bed.vs1.txt 
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -f 1 -c -a /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bed -b /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.bmass_GlobalLipids2010_Vs2_NewSNPs_SNPs_vs1_500kbPadding_bed.vs1.Counts.txt 
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -f 1 -c -a /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bed -b /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.NewSNPs.SNPs.vs1.1MbPadding.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.bmass_GlobalLipids2010_Vs2_NewSNPs_SNPs_vs1_1MbPadding_bed.vs1.Counts.txt 
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -f 1 -c -a /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bed -b /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.bmass_GlobalLipids2010_Vs2_NewSNPs_SNPs_vs1_2MbPadding_bed.vs1.Counts.txt 

~~~
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.NewSNPs.SNPs.vs1.txt | tail -n +2 | awk '{ print "chr" $2 "\t" $3 "\t" $3 }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.NewSNPs.SNPs.vs1.bed
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.NewSNPs.SNPs.vs1.bed | head -n 10
chr10   101902184  101902184
chr10   17300296   17300296
chr10   45333283   45333283
chr10   5236085 5236085
chr11   47215429   47215429
chr1 149225460  149225460
chr11   51368666   51368666
chr11   54886216   54886216
chr11   55776161   55776161
chr11   75132669   75132669
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.OverlapWith.bmass_GlobalLipids2010_Vs2_NewSNPs_SNPs_vs1_txt.Window500000.vs1.txt | awk '{ print $4 }' | sort | uniq -c
    102 0
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.bmass_GlobalLipids2010_Vs2_NewSNPs_SNPs_vs1_1MbPadding_bed.vs1.txt
chr2 21078786   21078786   chr2 19234648        21234648
chr12   110556807  110556807  chr12     110076069       112076069
chr8 10721339   10721339   chr8 8831779 10831779
chr2 21117405   21117405   chr2 19234648        21234648
chr11   46699823   46699823   chr11     46215429        48215429
chr12   122362191  122362191  chr12     120747808       122747808
chr1 92782026   92782026   chr1 92472800        94472800
chr8 9220768    9220768 chr8  8831779   10831779
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.bmass_GlobalLipids2010_Vs2_NewSNPs_SNPs_vs1_2MbPadding_bed.vs1.txt
chr2 21078786   21078786   chr2 18234648        22234648
chr12   110556807  110556807  chr12     109076069       113076069
chr12   119901033  119901033  chr12     119747808       123747808
chr8 10721339   10721339   chr8 7831779 11831779
chr1 25648320   25648320   chr1 25102620        29102620
chr2 21117405   21117405   chr2 18234648        22234648
chr6 31373469   31373469   chr6 27550680        31550680
chr11   46699823   46699823   chr11     45215429        49215429
chr12   122362191  122362191  chr12     119747808       123747808
chr12   123026120  123026120  chr12     119747808       123747808
chr1 92782026   92782026   chr1 91472800        95472800
chr8 9220768    9220768 chr8  7831779   11831779
chr9 133279427  133279427  chr9 133143989       137143989
#20170125 -- forceMAF update
#[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.bmass_GlobalLipids2010_Vs2_NewSNPs_SNPs_vs1_500kbPadding_bed.vs1.Counts.txt | awk '{ print $4 }' | sort | uniq -c
#    102 0
#[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.bmass_GlobalLipids2010_Vs2_NewSNPs_SNPs_vs1_1MbPadding_bed.vs1.Counts.txt | awk '{ print $4 }' | sort | uniq -c
#     94 0
#      8 1
#[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.bmass_GlobalLipids2010_Vs2_NewSNPs_SNPs_vs1_2MbPadding_bed.vs1.Counts.txt | awk '{ print $4 }' | sort | uniq -c
#     89 0
#     13 1
#20170207 -- MinorAllele update
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.bmass_GlobalLipids2010_Vs2_NewSNPs_SNPs_vs1_500kbPadding_bed.vs1.Counts.txt | awk '{ print $4 }' | sort | uniq -c
    102 0
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.bmass_GlobalLipids2010_Vs2_NewSNPs_SNPs_vs1_1MbPadding_bed.vs1.Counts.txt | awk '{ print $4 }' | sort | uniq -c
     94 0
      8 1
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Table1.ChrBP.noNA.NArplcd.bedIntersect.bmass_GlobalLipids2010_Vs2_NewSNPs_SNPs_vs1_2MbPadding_bed.vs1.Counts.txt | awk '{ print $4 }' | sort | uniq -c
     89 0
     13 1
~~~


#GlobalLipids2013
#20170123

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.ChrBP.txt | awk '{ print "chr" $1 "\t" $2 "\t" $2 }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.bed
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.txt | tail -n +2 | awk '{ print "chr" $2 "\t" $3 "\t" $3 }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.bed

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.txt | tail -n +2 | awk '{ print $1 "\t" $2 "\t" $3 }' | python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs3.py --file1 - --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.ChrBPChrBP.txt --window 500000 > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.OverlapWith.bmass_GlobalLipids2013_Vs2_NewSNPs_SNPs_vs1_txt.Window500000.vs1.txt
 
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/bedtools slop -i /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.bed -g /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg18.chrom.sizes -b 500000 > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bed
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/bedtools slop -i /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.bed -g /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg18.chrom.sizes -b 1000000 > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.1MbPadding.bed
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/bedtools slop -i /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.bed -g /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg18.chrom.sizes -b 2000000 > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bed

/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -f 1 -wb -a /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.bed -b /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.bedIntersect.bmass_GlobalLipids2013_Vs2_NewSNPs_SNPs_vs1_500kbPadding_bed.vs1.txt 
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -f 1 -wb -a /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.bed -b /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.1MbPadding.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.bedIntersect.bmass_GlobalLipids2013_Vs2_NewSNPs_SNPs_vs1_1MbPadding_bed.vs1.txt 
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -f 1 -wb -a /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.bed -b /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.bedIntersect.bmass_GlobalLipids2013_Vs2_NewSNPs_SNPs_vs1_2MbPadding_bed.vs1.txt 
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -f 1 -c -a /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.bed -b /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.bedIntersect.bmass_GlobalLipids2013_Vs2_NewSNPs_SNPs_vs1_500kbPadding_bed.vs1.Counts.txt 
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -f 1 -c -a /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.bed -b /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.1MbPadding.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.bedIntersect.bmass_GlobalLipids2013_Vs2_NewSNPs_SNPs_vs1_1MbPadding_bed.vs1.Counts.txt 
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -f 1 -c -a /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.bed -b /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.bedIntersect.bmass_GlobalLipids2013_Vs2_NewSNPs_SNPs_vs1_2MbPadding_bed.vs1.Counts.txt 

~~~
#20170125 -- forceMAF update
#[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.OverlapWith.bmass_GlobalLipids2013_Vs2_NewSNPs_SNPs_vs1_txt.Window500000.vs1.txt | awk '{ print $4 }' | sort | uniq -c
#    157 0
#[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.bedIntersect.bmass_GlobalLipids2013_Vs2_NewSNPs_SNPs_vs1_500kbPadding_bed.vs1.Counts.txt | awk '{ print $4 }' | sort | uniq -c
#    157 0
#[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.bedIntersect.bmass_GlobalLipids2013_Vs2_NewSNPs_SNPs_vs1_1MbPadding_bed.vs1.Counts.txt | awk '{ print $4 }' | sort | uniq -c
#    136 0
#     19 1
#      2 2
#[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.bedIntersect.bmass_GlobalLipids2013_Vs2_NewSNPs_SNPs_vs1_2MbPadding_bed.vs1.Counts.txt | awk '{ print $4 }' | sort | uniq -c
#    125 0
#     24 1
#      7 2
#      1 3
#20170207 -- MinorAllele update
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.bedIntersect.bmass_GlobalLipids2013_Vs2_NewSNPs_SNPs_vs1_500kbPadding_bed.vs1.Counts.txt | awk '{ print $4 }' | sort | uniq -c
    157 0
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.bedIntersect.bmass_GlobalLipids2013_Vs2_NewSNPs_SNPs_vs1_1MbPadding_bed.vs1.Counts.txt | awk '{ print $4 }' | sort | uniq -c
    136 0
     19 1
      2 2
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.bedIntersect.bmass_GlobalLipids2013_Vs2_NewSNPs_SNPs_vs1_2MbPadding_bed.vs1.Counts.txt | awk '{ print $4 }' | sort | uniq -c
    125 0
     24 1
      7 2
      1 3
~~~



#GIANT2010
#20170123

/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.txt
/mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.txt

cat /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.txt | awk '{ print "chr" $2 "\t" $3 "\t" $3 }' > /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.BPMinus1.bed
cat /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.txt | awk '{ print $2 "_" $3 "\t" $2 "\t" $3 }' > /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.ChrBPChrBP.BPMinus1.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.txt | tail -n +2 | awk '{ print "chr" $2 "\t" $3 "\t" $3 }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.bed

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.txt | tail -n +2 | awk '{ print $1 "\t" $2 "\t" $3 }' | python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs3.py --file1 - --file2 /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.ChrBPChrBP.BPMinus1.txt --window 500000 > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.BPMinus1.OverlapWith.bmass_GIANT2010_Vs2_NewSNPs_SNPs_vs1_txt.Window500000.vs1.txt 

/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/bedtools slop -i /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.bed -g /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg18.chrom.sizes -b 500000 > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bed
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/bedtools slop -i /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.bed -g /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg18.chrom.sizes -b 1000000 > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.1MbPadding.bed
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/bedtools slop -i /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.bed -g /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg18.chrom.sizes -b 2000000 > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bed

/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -f 1 -wb -a /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.BPMinus1.bed -b /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.BPMinus1.bedIntersect.bmass_GIANT2010_Vs2_NewSNPs_SNPs_vs1_500kbPadding_bed.vs1.txt
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -f 1 -wb -a /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.BPMinus1.bed -b /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.1MbPadding.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.BPMinus1.bedIntersect.bmass_GIANT2010_Vs2_NewSNPs_SNPs_vs1_1MbPadding_bed.vs1.txt
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -f 1 -wb -a /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.BPMinus1.bed -b /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.BPMinus1.bedIntersect.bmass_GIANT2010_Vs2_NewSNPs_SNPs_vs1_2MbPadding_bed.vs1.txt
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -f 1 -c -a /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.BPMinus1.bed -b /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.BPMinus1.bedIntersect.bmass_GIANT2010_Vs2_NewSNPs_SNPs_vs1_500kbPadding_bed.vs1.Counts.txt
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -f 1 -c -a /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.BPMinus1.bed -b /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.1MbPadding.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.BPMinus1.bedIntersect.bmass_GIANT2010_Vs2_NewSNPs_SNPs_vs1_1MbPadding_bed.vs1.Counts.txt
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -c -a /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.BPMinus1.bed -b /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.BPMinus1.bedIntersect.bmass_GIANT2010_Vs2_NewSNPs_SNPs_vs1_2MbPadding_bed.vs1.Counts.txt

~~~
#20170125 -- forceMAF update
#[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.BPMinus1.OverlapWith.bmass_GIANT2010_Vs2_NewSNPs_SNPs_vs1_txt.Window500000.vs1.txt | awk '{ print $4 }' | sort | uniq -c
#    226 0
#[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.BPMinus1.bedIntersect.bmass_GIANT2010_Vs2_NewSNPs_SNPs_vs1_500kbPadding_bed.vs1.Counts.txt | awk '{ print $4 }' | sort | uniq -c
#    226 0
#[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.BPMinus1.bedIntersect.bmass_GIANT2010_Vs2_NewSNPs_SNPs_vs1_1MbPadding_bed.vs1.Counts.txt | awk '{ print $4 }' | sort | uniq -c
#    218 0
#      8 1
#[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.BPMinus1.bedIntersect.bmass_GIANT2010_Vs2_NewSNPs_SNPs_vs1_2MbPadding_bed.vs1.Counts.txt | awk '{ print $4 }' | sort | uniq -c
#    215 0
#     10 1
#      1 2
#20170208 -- MinorAllele update
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.BPMinus1.bedIntersect.bmass_GIANT2010_Vs2_NewSNPs_SNPs_vs1_500kbPadding_bed.vs1.Counts.txt | awk '{ print $4 }' | sort | uniq -c
    226 0
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.BPMinus1.bedIntersect.bmass_GIANT2010_Vs2_NewSNPs_SNPs_vs1_1MbPadding_bed.vs1.Counts.txt | awk '{ print $4 }' | sort | uniq -c
    218 0
      8 1
[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.BPMinus1.bedIntersect.bmass_GIANT2010_Vs2_NewSNPs_SNPs_vs1_2MbPadding_bed.vs1.Counts.txt | awk '{ print $4 }' | sort | uniq -c
    215 0
     10 1
      1 2
~~~



#GIANT2014_5
#20170123

/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.txt
/mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.txt

cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.txt | awk '{ print "chr" $2 "\t" $3 "\t" $3 }' > /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.bed
cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.txt | awk '{ print $2 "_" $3 "\t" $2 "\t" $3 }' > /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.ChrBPChrBP.BPMinus1.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.txt | tail -n +2 | awk '{ print "chr" $2 "\t" $3 "\t" $3 }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.bed

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.txt | tail -n +2 | awk '{ print $1 "\t" $2 "\t" $3 }' | python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs3.py --file1 - --file2 /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.ChrBPChrBP.BPMinus1.txt --window 500000 > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.BPMinus1.OverlapWith.bmass_GIANT2014_5_Vs2_NewSNPs_SNPs_vs1_txt.Window500000.vs1.txt 

/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/bedtools slop -i /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.bed -g /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg18.chrom.sizes -b 500000 > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bed
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/bedtools slop -i /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.bed -g /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg18.chrom.sizes -b 1000000 > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.1MbPadding.bed
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/bedtools slop -i /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.bed -g /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg18.chrom.sizes -b 2000000 > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bed

/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -f 1 -wb -a /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.bed -b /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.bedIntersect.bmass_GIANT2014_5_Vs2_NewSNPs_SNPs_vs1_500kbPadding_bed.vs1.txt
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -f 1 -wb -a /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.bed -b /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.1MbPadding.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.bedIntersect.bmass_GIANT2014_5_Vs2_NewSNPs_SNPs_vs1_1MbPadding_bed.vs1.txt
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -f 1 -wb -a /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.bed -b /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.bedIntersect.bmass_GIANT2014_5_Vs2_NewSNPs_SNPs_vs1_2MbPadding_bed.vs1.txt
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -f 1 -c -a /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.bed -b /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.bedIntersect.bmass_GIANT2014_5_Vs2_NewSNPs_SNPs_vs1_500kbPadding_bed.vs1.Counts.txt
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -f 1 -c -a /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.bed -b /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.1MbPadding.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.bedIntersect.bmass_GIANT2014_5_Vs2_NewSNPs_SNPs_vs1_1MbPadding_bed.vs1.Counts.txt
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -c -a /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.bed -b /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.bedIntersect.bmass_GIANT2014_5_Vs2_NewSNPs_SNPs_vs1_2MbPadding_bed.vs1.Counts.txt

~~~
#20170208 -- MinorAllele update
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.bedIntersect.bmass_GIANT2014_5_Vs2_NewSNPs_SNPs_vs1_500kbPadding_bed.vs1.Counts.txt | awk '{ print $4 }' | sort | uniq -c
    843 0
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.bedIntersect.bmass_GIANT2014_5_Vs2_NewSNPs_SNPs_vs1_1MbPadding_bed.vs1.Counts.txt | awk '{ print $4 }' | sort | uniq -c
    759 0
     78 1
      6 2
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.bedIntersect.bmass_GIANT2014_5_Vs2_NewSNPs_SNPs_vs1_2MbPadding_bed.vs1.Counts.txt | awk '{ print $4 }' | sort | uniq -c
    652 0
    165 1
     24 2
      2 3
~~~













##20170214
##Annotations, first set, using refSeq files as base + bedtools intersect setup; annotating previous & new snps atm; also including a bit of Annovar annotation information too
##Also checking whether 1st set of 'new loci' completely accounted for in 2nd set 'previous + new loci', eg does first dataset contain any hits that are completely different from the 2nd dataset set

#GlobalLipids2010
#20170214

mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap

cat /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.txt | grep -v txStart | awk '{ print "chr" $3 "\t" $5 "\t" $6 "\t" $1 ";" $2 ";" $4 ";" $7 }' > /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.bed 

cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.AnnovarFormat | awk '{ print "chr" $1 "\t" $2 "\t" $3 }' > /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.bed

/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/bedtools slop -i /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.bed -g /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg18.chrom.sizes -b 10000 > /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.10kbPadding.bed
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/bedtools slop -i /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.bed -g /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg18.chrom.sizes -b 500000 > /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.500kbPadding.bed
#2Mb just as a positive control, not going to be normally interested in this range of output
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/bedtools slop -i /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.bed -g /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg18.chrom.sizes -b 2000000 > /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.2MbPadding.bed

/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -f 1 -wb -a /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -f 1 -c -a /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.Counts.vs1.txt
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -wa -wb -a /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.10kbPadding.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -wa -wb -a /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.500kbPadding.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -wa -wb -a /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.2MbPadding.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt 
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | grep -v MIR | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt 
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | grep -v MIR | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | grep -v MIR | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | grep -v MIR | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.AnnovarFormat | awk '{ print "chr" $1 "\t" $2 "\t" $3 }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.bed

/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/bedtools slop -i /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.bed -g /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg18.chrom.sizes -b 10000 > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.10kbPadding.bed
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/bedtools slop -i /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.bed -g /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg18.chrom.sizes -b 500000 > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.500kbPadding.bed
#2Mb just as a positive control, not going to be normally interested in this range of output
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/bedtools slop -i /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.bed -g /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg18.chrom.sizes -b 2000000 > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.2MbPadding.bed

/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -f 1 -wb -a /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -f 1 -c -a /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.Counts.vs1.txt
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -wa -wb -a /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.10kbPadding.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -wa -wb -a /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.500kbPadding.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -wa -wb -a /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.2MbPadding.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt 
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | grep -v MIR | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt 
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | grep -v MIR | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | grep -v MIR | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | grep -v MIR | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt

join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) | wc
join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) | wc
join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) | wc
join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) | wc
join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) | wc
join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) | wc
join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) | wc
join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) | wc

perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/table_annovar.pl /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.AnnovarFormat /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -buildver hg18 -out /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.AnnovarFormat.20170208DL_20160204Vs_annovar.TableAnnovar.refGeneOnly -remove -protocol refGene -operation g -nastring NA 
perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/table_annovar.pl /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.AnnovarFormat /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -buildver hg18 -out /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.AnnovarFormat.TableAnnovar.refGeneOnly -remove -protocol refGene -operation g -nastring NA










#perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/table_annovar.pl /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.AnnovarFormat /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -buildver hg18 -out /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.AnnovarFormat.20170208DL_20160204Vs_annovar.TableAnnovar -remove -protocol refGene,genomicSuperDups,phastConsElements44way,cytoBand,tfbsConsSites,wgRna,targetScanS,gwasCatalog,snp129,snp138,1000g2014oct_all,esp6500siv2_all,exac03,dbnsfp33a -operation g,r,r,r,r,r,r,r,f,f,f,f,f,f -nastring NA
#
##/mnt/lustre/home/mturchin20/Software/liftOver/hg18ToHg19.over.chain.gz
##
##usage:
##   liftOver oldFile map.chain newFile unMapped
#
#
#
#
#
#
#
#cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.AnnovarFormat | awk '{ print "chr"$1"\t"$2"\t"$3+1"\t"$1"_"$2}' > /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.liftoverBed
#
#/mnt/lustre/home/mturchin20/Software/liftOver/liftOver /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.liftoverBed /mnt/lustre/home/mturchin20/Software/liftOver/hg18ToHg19.over.chain.gz /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.liftoverBed.hg19.bed /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.liftoverBed.hg19.bed.dropped
#
#join <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.liftoverBed.hg19.bed | sed 's/chr//g' | awk '{ print $4 "\t" $1 "\t" $2 "\t" $2 }' | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.AnnovarFormat | awk '{ print $1 "_" $2 "\t" $4 "\t" $5 }' | sort -k 1,1) | perl -lane 'push(@F, $F[0]); splice (@F,0,1); print join("\t", @F);' > /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.liftoverBed.hg19.AnnovarFormat
#
#perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/table_annovar.pl /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.liftoverBed.hg19.AnnovarFormat /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -buildver hg19 -out /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.hg19.AnnovarFormat.20170208DL_20160204Vs_annovar.TableAnnovar -remove -protocol cadd -operation f -nastring NA -otherinfo
#
#
#want to know if expand new SNPs 1Mb, 2Mb out, what additional loci do they overlap -- do loci that are now showing up already exist among the previous snps? 
#
#















join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt | sed 's/chr//g' | awk '{ print $1 "_" $2 "\t" $4 }' | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.AnnovarFormat.20170208DL_20160204Vs_annovar.TableAnnovar.refGeneOnly.hg18_multianno.txt | awk '{ print $1 "_" $2 "\t" $7 "\t" $8}' | sort -k 1,1)

#20170215 NOTE -- Find that overlap between bedtools intersect and Annovar is fairly good if not almost complete. Use of 'upstream/downstream' by annovar (I thinkkkk meaning <1kb) makes a few things appear not overlapping at first, but it mostly due to discrepancies in how the two approaches choose to output things.

~~~
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.vs2.txt | head -n 10
A1BG    1       19      -       63549983        63556677        NM_130786
A1BG-AS1        1       19      +       63555147        63558361        NR_015380
A1CF    6       10      -       52229174        52315441        NM_138933,NM_001198818,NM_138932,NM_001198820,NM_001198819,NM_014576
A2M     4       12      -       9111570 9160092 NM_000014,NM_001347423,NM_001347424,NM_001347425
A2M-AS1 3       12      +       9109039 9111918 NR_026971,NR_137424,NR_137425
A2ML1   2       12      +       8866334 8920644 NM_001282424,NM_144670
A2MP1   1       12      -       9272395 9278070 NR_040112
A3GALT2 1       1       -       33544953        33559286        NM_001080438
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.bed | head -n 10
chr19   63549983        63556677        A1BG;1;-;NM_130786
chr19   63555147        63558361        A1BG-AS1;1;+;NR_015380
chr10   52229174        52315441        A1CF;6;-;NM_138933,NM_001198818,NM_138932,NM_001198820,NM_001198819,NM_014576
chr12   9111570 9160092 A2M;4;-;NM_000014,NM_001347423,NM_001347424,NM_001347425
chr12   9109039 9111918 A2M-AS1;3;+;NR_026971,NR_137424,NR_137425
chr12   8866334 8920644 A2ML1;2;+;NM_001282424,NM_144670
chr12   9272395 9278070 A2MP1;1;-;NR_040112
chr1    33544953        33559286        A3GALT2;1;-;NM_001080438
chr22   41418064        41447251        A4GALT;2;-;NM_017436,NM_001318038
chr3    139325249       139333919       A4GNT;1;-;NM_016161
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.AnnovarFormat | head -n 10
11      18589560        18589560        T       C
2       165221337       165221337       T       C
19      19268718        19268718        T       C
2       21078786        21078786        T       C
10      64697616        64697616        A       T
6       161009807       161009807       A       G
12      110556807       110556807       A       G
8       145115531       145115531       A       G
11      125749162       125749162       A       G
12      56078847        56078847        T       C
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.bed | head -n 10
chr11   18589560        18589560
chr2    165221337       165221337
chr19   19268718        19268718
chr2    21078786        21078786
chr10   64697616        64697616
chr6    161009807       161009807
chr12   110556807       110556807
chr8    145115531       145115531
chr11   125749162       125749162
chr12   56078847        56078847
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.bed | head -n 10
chr11   18589560        18589560
chr2    165221337       165221337
chr19   19268718        19268718
chr2    21078786        21078786
chr10   64697616        64697616
chr6    161009807       161009807
chr12   110556807       110556807
chr8    145115531       145115531
chr11   125749162       125749162
chr12   56078847        56078847
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.10kbPadding.bed | head -n 10
chr11   18579560        18599560
chr2    165211337       165231337
chr19   19258718        19278718
chr2    21068786        21088786
chr10   64687616        64707616
chr6    160999807       161019807
chr12   110546807       110566807
chr8    145105531       145125531
chr11   125739162       125759162
chr12   56068847        56088847
[  mturchin20@bigmem01  ~/TempStuff]$/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -f 1 -wb -a /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt
***** ERROR: illegal character 't' found in integer conversion of string "txStart". Exiting...
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.bed | head -n 10
chr11   18589560        18589560
chr2    165221337       165221337
chr19   19268718        19268718
chr2    21078786        21078786
chr10   64697616        64697616
chr6    161009807       161009807
chr12   110556807       110556807
chr8    145115531       145115531
chr11   125749162       125749162
chr12   56078847        56078847
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.bed | grep txStart
chrom   txStart txEnd   name2;1;strand;name
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | head -n 10
chr11   18589560        18589560        chr11   18584523        18612596        SPTY2D1;1;-;NM_194285
chr19   19268718        19268718        chr19   19248321        19292321        SUGP1;1;-;NM_172231
chr19   19268718        19268718        chr19   24395   114052897       FAM138D;4;-;NR_026823,NR_026823,NR_026823,NR_026823
chr19   19268718        19268718        chr19   3693815 130464236       MIR1268A;3;+;NR_031672,NR_031672,NR_031672
chr19   19268718        19268718        chr19   20143   100318322       MIR1302-10;4;+;NR_036267,NR_036267,NR_036267,NR_036267
chr19   19268718        19268718        chr19   20143   100318322       MIR1302-11;4;+;NR_036268,NR_036268,NR_036268,NR_036268
chr19   19268718        19268718        chr19   20143   100318322       MIR1302-2;4;+;NR_036051,NR_036051,NR_036051,NR_036051
chr19   19268718        19268718        chr19   20143   100318322       MIR1302-9;4;+;NR_036266,NR_036266,NR_036266,NR_036266
chr19   19268718        19268718        chr19   844483  182432325       RNU6-2;11;+;NR_125730,NR_125730,NR_125730,NR_125730,NR_125730,NR_125730,NR_125730,NR_125730,NR_125730,NR_125730,NR_125730
chr2    21078786        21078786        chr2    21077805        21120450        APOB;1;-;NM_000384
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | awk '{ print $1 "_" $2 }' | sort | uniq -c | awk '{ print $1 }' | sort | uniq -c
     39 1
      1 19
     18 2
      2 3
      1 4
      3 5
      5 6
      7 7
      7 8
      4 9
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | awk '{ print $1 "_" $2 }' | sort | uniq -c | awk '{ if ($1 == 19) { print $0 } } '
     19 chr19_53898229
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | grep 53898229 | head -n 10                                                        
chr19   53898229        53898229        chr19   53891039        53901003        FUT2;2;+;NM_001097638,NM_000511
chr19   53898229        53898229        chr19   53897518        53898917        LOC105447645;1;-;NR_131188
chr19   53898229        53898229        chr19   53118847        55323591        SNAR-A10;10;-;NR_024229,NR_024229,NR_024229,NR_024229,NR_024229,NR_024229,NR_024229,NR_024229,NR_024229,NR_024229
chr19   53898229        53898229        chr19   53118847        55323591        SNAR-A11;10;-;NR_024225,NR_024225,NR_024225,NR_024225,NR_024225,NR_024225,NR_024225,NR_024225,NR_024225,NR_024225
chr19   53898229        53898229        chr19   53118847        55323591        SNAR-A14;10;-;NR_024242,NR_024242,NR_024242,NR_024242,NR_024242,NR_024242,NR_024242,NR_024242,NR_024242,NR_024242
chr19   53898229        53898229        chr19   53118847        55323591        SNAR-A3;10;-;NR_024214,NR_024214,NR_024214,NR_024214,NR_024214,NR_024214,NR_024214,NR_024214,NR_024214,NR_024214
chr19   53898229        53898229        chr19   53118847        55323591        SNAR-A4;10;-;NR_024215,NR_024215,NR_024215,NR_024215,NR_024215,NR_024215,NR_024215,NR_024215,NR_024215,NR_024215
chr19   53898229        53898229        chr19   53118847        55323591        SNAR-A5;10;-;NR_024223,NR_024223,NR_024223,NR_024223,NR_024223,NR_024223,NR_024223,NR_024223,NR_024223,NR_024223
chr19   53898229        53898229        chr19   53118847        55323591        SNAR-A6;10;-;NR_024227,NR_024227,NR_024227,NR_024227,NR_024227,NR_024227,NR_024227,NR_024227,NR_024227,NR_024227
chr19   53898229        53898229        chr19   53118847        55323591        SNAR-A7;10;-;NR_024224,NR_024224,NR_024224,NR_024224,NR_024224,NR_024224,NR_024224,NR_024224,NR_024224,NR_024224
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | wc
    290    2030   29340
[  mturchin20@bigmem01  ~/TempStuff]$/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -wb -a /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.bed | wc
    290    2030   29340
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.Counts.vs1.txt | wc
    102     408    2662
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.Counts.vs1.txt | head -n 10
chr11   18589560        18589560        1
chr2    165221337       165221337       0
chr19   19268718        19268718        8
chr2    21078786        21078786        1
chr10   64697616        64697616        2
chr6    161009807       161009807       0
chr12   110556807       110556807       5
chr8    145115531       145115531       1
chr11   125749162       125749162       2
chr12   56078847        56078847        6
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.Counts.vs1.txt | awk '{ print $4 }' | sort | uniq -c
     15 0
     39 1
      1 19
     18 2
      2 3
      1 4
      3 5
      5 6
      7 7
      7 8
      4 9
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt | head -n 10
chr11   18589560        SPTY2D1
chr19   19268718        SUGP1;FAM138D;MIR1268A;MIR1302-10;MIR1302-11;MIR1302-2;MIR1302-9;RNU6-2
chr2    21078786        APOB
chr10   64697616        JMJD1C;MIR4454
chr12   110556807       MIR1244-1;MIR1244-2;MIR1244-3;MIR1244-4;MIR548T
chr8    145115531       PLEC
chr11   125749162       ST3GAL4;MIR3654
chr12   56078847        R3HDM2;CAND1;MIR1244-1;MIR1244-2;MIR1244-3;MIR1244-4
chr16   30825988        MIR1972-1;MIR1972-2;MIR6859-1;MIR6859-2;MIR6859-3;MIR6859-4
chr12   119901033       HNF1A;MIR1244-1;MIR1244-2;MIR1244-3;MIR1244-4;MIR548T
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt | wc        
     87     261    3838
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt | awk '{ print $3 }' | sort | uniq -c
     39 1
      1 19
     18 2
      2 3
      1 4
      3 5
      5 6
      7 7
      7 8
      4 9
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.Counts.vs1.txt | awk '{ print $1 "_" $2 "\t" $0 }' | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt | awk '{ print $1 "_" $2 "\t" $0 }' | sort -k 1,1) | wc
     87     783    7573
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.Counts.vs1.txt | awk '{ print $1 "_" $2 "\t" $0 }' | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt | awk '{ print $1 "_" $2 "\t" $0 }' | sort -k 1,1) | head -n 10
chr10_113923876 chr10 113923876 113923876 2 chr10 113923876 2 GPAM;MIR4454
chr10_64697616 chr10 64697616 64697616 2 chr10 64697616 2 JMJD1C;MIR4454
chr10_94829632 chr10 94829632 94829632 1 chr10 94829632 1 MIR4454
chr1_109619829 chr1 109619829 109619829 1 chr1 109619829 1 CELSR2
chr11_10345358 chr11 10345358 10345358 1 chr11 10345358 1 CAND1.11
chr11_116154127 chr11 116154127 116154127 2 chr11 116154127 2 ZPR1;MIR3654
chr11_122027585 chr11 122027585 122027585 1 chr11 122027585 1 MIR3654
chr11_125749162 chr11 125749162 125749162 2 chr11 125749162 2 ST3GAL4;MIR3654
chr11_18589560 chr11 18589560 18589560 1 chr11 18589560 1 SPTY2D1
chr11_46699823 chr11 46699823 46699823 1 chr11 46699823 1 F2
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.Counts.vs1.txt | awk '{ print $1 "_" $2 "\t" $0 }' | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt | awk '{ print $1 "_" $2 "\t" $0 }' | sort -k 1,1) | awk '{ if ($8 == $5) { print $0 } } ' | wc
     87     783    7573
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt | wc  
     75     300    2428
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | wc
    339    2373   33291
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | grep -v MIR | wc
    193    1351   18368
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | wc
   2067   14469  179371
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | grep -v MIR | wc
   1795   12565  155323
[  mturchin20@bigmem01  ~/TempStuff]$/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -wa -wb -a /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.10kbPadding.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.bed | head -n 20
chr11   18579560        18599560        chr11   18577926        18588378        SPTY2D1-AS1;1;+;NR_038360
chr11   18579560        18599560        chr11   18584523        18612596        SPTY2D1;1;-;NM_194285
chr19   19258718        19278718        chr19   19248321        19292321        SUGP1;1;-;NM_172231
chr19   19258718        19278718        chr19   24395   114052897       FAM138D;4;-;NR_026823,NR_026823,NR_026823,NR_026823
chr19   19258718        19278718        chr19   3693815 130464236       MIR1268A;3;+;NR_031672,NR_031672,NR_031672
chr19   19258718        19278718        chr19   20143   100318322       MIR1302-10;4;+;NR_036267,NR_036267,NR_036267,NR_036267
chr19   19258718        19278718        chr19   20143   100318322       MIR1302-11;4;+;NR_036268,NR_036268,NR_036268,NR_036268
chr19   19258718        19278718        chr19   20143   100318322       MIR1302-2;4;+;NR_036051,NR_036051,NR_036051,NR_036051
chr19   19258718        19278718        chr19   20143   100318322       MIR1302-9;4;+;NR_036266,NR_036266,NR_036266,NR_036266
chr19   19258718        19278718        chr19   844483  182432325       RNU6-2;11;+;NR_125730,NR_125730,NR_125730,NR_125730,NR_125730,NR_125730,NR_125730,NR_125730,NR_125730,NR_125730,NR_125730
chr2    21068786        21088786        chr2    21077805        21120450        APOB;1;-;NM_000384
chr10   64687616        64707616        chr10   64596986        64951841        JMJD1C;8;-;NM_001322258,NM_001282948,NM_032776,NM_001318154,NR_134512,NM_001322254,NM_001322252,NM_001318153
chr10   64687616        64707616        chr10   6717037 227377171       MIR4454;8;+;NR_039659,NR_039659,NR_039659,NR_039659,NR_039659,NR_039659,NR_039659,NR_039659
chr6    160999807       161019807       chr6    160872504       161007397       LPA;1;-;NM_005577
chr12   110546807       110566807       chr12   110564332       110608183       BRAP;1;-;NM_006768
chr12   110546807       110566807       chr12   9283332 232286349       MIR1244-1;4;+;NR_036052,NR_036052,NR_036052,NR_036052
chr12   110546807       110566807       chr12   9283332 232286349       MIR1244-2;4;+;NR_036262,NR_036262,NR_036262,NR_036262
chr12   110546807       110566807       chr12   9283332 232286349       MIR1244-3;4;+;NR_036263,NR_036263,NR_036263,NR_036263
chr12   110546807       110566807       chr12   9283330 232286349       MIR1244-4;4;+;NR_128710,NR_128710,NR_128710,NR_128710
chr12   110546807       110566807       chr12   83063265        147674860       MIR548T;2;-;NR_036093,NR_036093
[  mturchin20@bigmem01  ~/TempStuff]$/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -wb -a /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.10kbPadding.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.bed | head -n 20
chr11   18579560        18588378        chr11   18577926        18588378        SPTY2D1-AS1;1;+;NR_038360
chr11   18584523        18599560        chr11   18584523        18612596        SPTY2D1;1;-;NM_194285
chr19   19258718        19278718        chr19   19248321        19292321        SUGP1;1;-;NM_172231
chr19   19258718        19278718        chr19   24395   114052897       FAM138D;4;-;NR_026823,NR_026823,NR_026823,NR_026823
chr19   19258718        19278718        chr19   3693815 130464236       MIR1268A;3;+;NR_031672,NR_031672,NR_031672
chr19   19258718        19278718        chr19   20143   100318322       MIR1302-10;4;+;NR_036267,NR_036267,NR_036267,NR_036267
chr19   19258718        19278718        chr19   20143   100318322       MIR1302-11;4;+;NR_036268,NR_036268,NR_036268,NR_036268
chr19   19258718        19278718        chr19   20143   100318322       MIR1302-2;4;+;NR_036051,NR_036051,NR_036051,NR_036051
chr19   19258718        19278718        chr19   20143   100318322       MIR1302-9;4;+;NR_036266,NR_036266,NR_036266,NR_036266
chr19   19258718        19278718        chr19   844483  182432325       RNU6-2;11;+;NR_125730,NR_125730,NR_125730,NR_125730,NR_125730,NR_125730,NR_125730,NR_125730,NR_125730,NR_125730,NR_125730
chr2    21077805        21088786        chr2    21077805        21120450        APOB;1;-;NM_000384
chr10   64687616        64707616        chr10   64596986        64951841        JMJD1C;8;-;NM_001322258,NM_001282948,NM_032776,NM_001318154,NR_134512,NM_001322254,NM_001322252,NM_001318153
chr10   64687616        64707616        chr10   6717037 227377171       MIR4454;8;+;NR_039659,NR_039659,NR_039659,NR_039659,NR_039659,NR_039659,NR_039659,NR_039659
chr6    160999807       161007397       chr6    160872504       161007397       LPA;1;-;NM_005577
chr12   110564332       110566807       chr12   110564332       110608183       BRAP;1;-;NM_006768
chr12   110546807       110566807       chr12   9283332 232286349       MIR1244-1;4;+;NR_036052,NR_036052,NR_036052,NR_036052
chr12   110546807       110566807       chr12   9283332 232286349       MIR1244-2;4;+;NR_036262,NR_036262,NR_036262,NR_036262
chr12   110546807       110566807       chr12   9283332 232286349       MIR1244-3;4;+;NR_036263,NR_036263,NR_036263,NR_036263
chr12   110546807       110566807       chr12   9283330 232286349       MIR1244-4;4;+;NR_128710,NR_128710,NR_128710,NR_128710
chr12   110546807       110566807       chr12   83063265        147674860       MIR548T;2;-;NR_036093,NR_036093
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt | wc
     93     372    4472
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt | wc
     88     352    2974
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt | wc
    101     404   17015
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt | wc
    101     404   14590
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | wc
    134     134    1052
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | wc
     99      99     719
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | wc
    181     181    1394
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | wc
    142     142    1030
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | wc
   1780    1780   12790
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | wc
   1626    1626   11507
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.Counts.vs1.txt | awk '{ print $4 }' | sort | uniq -c
      8 0
      5 1
      5 2
      1 8
[  mturchin20@bigmem01  ~/TempStuff]$for i in `ls -lrt /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1*bedIntersect.2017021* | awk '{ print $9 }'`; do echo $i; cat $i | wc; done
/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt
     23     161    2352
/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.Counts.vs1.txt
     19      76     485
/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt
     35     245    3461
/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt
    377    2639   31176
/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt
     11      44     364
/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt
     10      40     256
/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt
     18      72     555
/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt
     16      64     416
/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt
     18      72    3053
/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt
     18      72    2823
/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt
     21      21     164
/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt
     13      13      89
/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt
    369     369    2685
/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt
    344     344    2472
/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt
     33      33     241
/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt
     23      23     151
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt) | wc
     11      11      98
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt)
FAM138D
MIR1268A
MIR1302-10
MIR1302-11
MIR1302-2
MIR1302-9
MIR3654
MIR4454
MIR6730
RNU6-2
SRGAP2D
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt) | wc
      3       3      23
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt)     
FAM138D
RNU6-2
SRGAP2D
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt | grep FAM138D
chr19   19268718        3       SUGP1;FAM138D;RNU6-2
chr19   59484573        2       FAM138D;RNU6-2
chr19   50106291        2       FAM138D;RNU6-2
chr19   50114786        2       FAM138D;RNU6-2
chr19   53898229        14      FUT2;LOC105447645;SNAR-A10;SNAR-A11;SNAR-A14;SNAR-A3;SNAR-A4;SNAR-A5;SNAR-A6;SNAR-A7;SNAR-A8;SNAR-A9;FAM138D;RNU6-2
chr19   11063306        3       LDLR;FAM138D;RNU6-2
chr19   8339196 3       ANGPTL4;FAM138D;RNU6-2
chr19   11208493        3       DOCK6;FAM138D;RNU6-2
chr19   57011927        3       FPR3;FAM138D;RNU6-2
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt) | wc
     11      11      98
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt) | wc
      3       3      23
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt) | wc
     11      11      98
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt) | wc
      3       3      23
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | wc
   1626    1626   11507
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | wc
    344     344    2472
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.bed | grep -E 'FAM138D|RNU6-2|SRGAP2D'
chr19   24395   114052897       FAM138D;4;-;NR_026823,NR_026823,NR_026823,NR_026823
chr19   844483  182432325       RNU6-2;11;+;NR_125730,NR_125730,NR_125730,NR_125730,NR_125730,NR_125730,NR_125730,NR_125730,NR_125730,NR_125730,NR_125730
chr1    120808674       204647924       SRGAP2D;2;+;NR_120535,NR_120535
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) | wc
    241     241    1857
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Vs2.w_20161109_edits_dtlesssignif_file.NewSNPs.SNPs.vs1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) | wc
    175     175    1306
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt | sed 's/chr//g' | awk '{ print $1 "_" $2 "\t" $4 }' | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.AnnovarFormat.20170208DL_20160204Vs_annovar.TableAnnovar.refGeneOnly.hg18_multianno.txt | awk '{ print $1 "_" $2 "\t" $7 "\t" $8}' | sort -k 1,1) | wc
     87     348    5091
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt | wc
     87     348    4013
[  mturchin20@bigmem01  ~/TempStuff]$join -v 2 <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt | sed 's/chr//g' | awk '{ print $1 "_" $2 "\t" $4 }' | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.AnnovarFormat.20170208DL_20160204Vs_annovar.TableAnnovar.refGeneOnly.hg18_multianno.txt | awk '{ print $1 "_" $2 "\t" $7 "\t" $8}' | sort -k 1,1) | head -n 10
1_232925220 LOC101927787,LINC01132 dist=61927;dist=1192
1_55277238 PCSK9 NA
18_45414951 LIPG,ACAA2 dist=41675;dist=148921
18_56000003 PMAIP1,MC4R dist=277485;dist=189541
2_165221337 GRB14,COBLL1 dist=34731;dist=23604
2_226808942 LOC646736,MIR5702 dist=55920;dist=422728
6_139871359 LINC01625,LOC100132735 dist=33933;dist=262544
6_161009807 LPA,PLG dist=2410;dist=33408
6_16235386 DTNBP1,MYLIP dist=464118;dist=1910
6_31373469 HLA-C,HLA-B dist=25577;dist=56159
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.Counts.vs1.txt | grep 55277238
chr1    55277238        55277238        0
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.AnnovarFormat.20170208DL_20160204Vs_annovar.TableAnnovar.refGeneOnly.hg18_multianno.txt | grep 55277238
1       55277238        55277238        A       G       upstream        PCSK9   NA      NA      NA
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt | sed 's/chr//g' | awk '{ print $1 "_" $2 "\t" $4 }' | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.AnnovarFormat.20170208DL_20160204Vs_annovar.TableAnnovar.refGeneOnly.hg18_multianno.txt | awk '{ print $1 "_" $2 "\t" $7 "\t" $8}' | sort -k 1,1) | head -n 10
10_113923876 GPAM;MIR4454 GPAM NA
10_64697616 JMJD1C;MIR4454 JMJD1C NA
10_94829632 MIR4454 CYP26A1,MYOF dist=2001;dist=226544
1_109619829 CELSR2 CELSR2 NM_001408:c.*1635T>G
11_10345358 CAND1.11 CAND1.11 NA
11_116154127 ZPR1;MIR3654 ZPR1 NM_001317086:c.*724G>C,NM_003904:c.*724G>C
11_122027585 MIR3654 MIR100HG,UBASH3B dist=448605;dist=4023
11_125749162 ST3GAL4;MIR3654 ST3GAL4 NA
11_18589560 SPTY2D1 SPTY2D1 NA
11_46699823 F2 F2 NA
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt | sed 's/chr//g' | awk '{ print $1 "_" $2 "\t" $4 }' | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.AnnovarFormat.20170208DL_20160204Vs_annovar.TableAnnovar.refGeneOnly.hg18_multianno.txt | awk '{ print $1 "_" $2 "\t" $7 "\t" $8}' | sort -k 1,1) | grep -v dist | head -n 10
10_113923876 GPAM;MIR4454 GPAM NA
10_64697616 JMJD1C;MIR4454 JMJD1C NA
1_109619829 CELSR2 CELSR2 NM_001408:c.*1635T>G
11_10345358 CAND1.11 CAND1.11 NA
11_116154127 ZPR1;MIR3654 ZPR1 NM_001317086:c.*724G>C,NM_003904:c.*724G>C
11_125749162 ST3GAL4;MIR3654 ST3GAL4 NA
11_18589560 SPTY2D1 SPTY2D1 NA
11_46699823 F2 F2 NA
11_61326406 FADS1 FADS1 NM_013402:c.*53A>G
12_108484576 MMAB;MIR1244-1;MIR1244-2;MIR1244-3;MIR1244-4;MIR548T MMAB NA
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt | sed 's/chr//g' | awk '{ print $1 "_" $2 "\t" $4 }' | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.AnnovarFormat.20170208DL_20160204Vs_annovar.TableAnnovar.refGeneOnly.hg18_multianno.txt | awk '{ print $1 "_" $2 "\t" $7 "\t" $8}' | sort -k 1,1) | grep -v dist | perl -lane '$F[1] = ((split(/;/, $F[1]))[0]); print join("\t", @F);' | awk '{ if ($2 == $3) { print $0 } } ' | wc
     63     252    1917
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt | sed 's/chr//g' | awk '{ print $1 "_" $2 "\t" $4 }' | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.AnnovarFormat.20170208DL_20160204Vs_annovar.TableAnnovar.refGeneOnly.hg18_multianno.txt | awk '{ print $1 "_" $2 "\t" $7 "\t" $8}' | sort -k 1,1) | grep -v dist | perl -lane '$F[1] = ((split(/;/, $F[1]))[0]); print join("\t", @F);' | awk '{ if ($2 != $3) { print $0 } } ' | wc
      5      20     168
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt | sed 's/chr//g' | awk '{ print $1 "_" $2 "\t" $4 }' | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.AnnovarFormat.20170208DL_20160204Vs_annovar.TableAnnovar.refGeneOnly.hg18_multianno.txt | awk '{ print $1 "_" $2 "\t" $7 "\t" $8}' | sort -k 1,1) | grep -v dist | perl -lane '$F[1] = ((split(/;/, $F[1]))[0]); print join("\t", @F);' | awk '{ if ($2 != $3) { print $0 } } '     
12_123026120    ZNF664  ZNF664,ZNF664-FAM101A   NA
12_123827546    MIR1244-1       SCARB1  NA
19_50114786     FAM138D APOC1   NA
5_156322875     LOC100132062    TIMD4   NA
7_72620810      GTF2IP1 TBL2    NA
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/Analyses/GeneAnnotationOverlap/GlobalLipids2010.Table1.noNA.NArplcd.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt | grep -E '123026120|123827546|50114786|156322875|72620810'
chr7    72620810        8       GTF2IP1;GTF2IP4;LOC541473;PMS2P5;PMS2P7;SPDYE8P;STAG3L1;TRIM73
chr19   50114786        7       FAM138D;MIR1268A;MIR1302-10;MIR1302-11;MIR1302-2;MIR1302-9;RNU6-2
chr12   123026120       8       ZNF664;ZNF664-RFLNA;MIR1244-1;MIR1244-2;MIR1244-3;MIR1244-4;MIR548AQ;MIR548T
chr5    156322875       9       LOC100132062;LOC100132287;LOC100133331;MIR548F3;OR4F16;OR4F29;OR4F3;SNORA105A;SNORA105B
chr12   123827546       6       MIR1244-1;MIR1244-2;MIR1244-3;MIR1244-4;MIR548AQ;MIR548T
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.bed | grep APOC1
chr19   50109343        50114446        APOC1;3;+;NM_001645,NM_001321066,NM_001321065
chr19   50121899        50126483        APOC1P1;3;+;NR_028413,NR_028412,NR_028414
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.bed | grep TIMD4
chr5    156278870       156322844       TIMD4;2;-;NM_138379,NM_001146726
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.bed | grep TBL2
chr5    56811599        56814393        ACTBL2;1;-;NM_001017992
chr7    72621209        72630949        TBL2;1;-;NM_012453
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Analyses/Annovar_refSeq_UCSCs_Exploration/20130911_annnovar.hg18_refGene.Condensed.vs2.txt | grep TBL2
ACTBL2  1       5       -       56811599        56814393        NM_001017992
TBL2    1       7       -       72621209        72630949        NM_012453
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.AnnovarFormat.20170208DL_20160204Vs_annovar.TableAnnovar.refGeneOnly.hg18_multianno.txt | grep TBL2
7       72620810        72620810        T       C       downstream      TBL2    NA      NA      NA
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.AnnovarFormat.20170208DL_20160204Vs_annovar.TableAnnovar.refGeneOnly.hg18_multianno.txt | grep TIMD4
5       156322875       156322875       T       C       upstream        TIMD4   NA      NA      NA
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.AnnovarFormat.20170208DL_20160204Vs_annovar.TableAnnovar.refGeneOnly.hg18_multianno.txt | grep APOC1
19      50106291        50106291        T       C       intergenic      APOE,APOC1      dist=1799;dist=3470     NA      NA
19      50114786        50114786        A       G       downstream      APOC1   NA      NA      NA
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.AnnovarFormat.20170208DL_20160204Vs_annovar.TableAnnovar.refGeneOnly.hg18_multianno.txt | grep SCARB1
12      123827546       123827546       T       C       downstream      SCARB1  NA      NA      NA
~~~


#GlobalLipids2013
#20170215

mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses
mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap

#/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.bed
#/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.bed

/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/bedtools slop -i /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.bed -g /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg18.chrom.sizes -b 10000 > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.10kbPadding.bed
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/bedtools slop -i /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.bed -g /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg18.chrom.sizes -b 500000 > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.500kbPadding.bed
#2Mb just as a positive control, not going to be normally interested in this range of output
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/bedtools slop -i /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.bed -g /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg18.chrom.sizes -b 2000000 > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.2MbPadding.bed

/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -f 1 -wb -a /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -f 1 -c -a /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.Counts.vs1.txt
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -wa -wb -a /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.10kbPadding.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -wa -wb -a /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.500kbPadding.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -wa -wb -a /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.2MbPadding.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt 
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | grep -v MIR | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt 
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | grep -v MIR | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | grep -v MIR | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | grep -v MIR | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt

/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/bedtools slop -i /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.bed -g /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg18.chrom.sizes -b 10000 > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.10kbPadding.bed
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/bedtools slop -i /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.bed -g /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg18.chrom.sizes -b 500000 > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bed
#2Mb just as a positive control, not going to be normally interested in this range of output
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/bedtools slop -i /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.bed -g /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg18.chrom.sizes -b 2000000 > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bed

/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -f 1 -wb -a /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -f 1 -c -a /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.Counts.vs1.txt
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -wa -wb -a /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.10kbPadding.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -wa -wb -a /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -wa -wb -a /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt 
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | grep -v MIR | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt 
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | grep -v MIR | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | grep -v MIR | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | grep -v MIR | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt

join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) | wc
join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) | wc
join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) | wc
join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) | wc
join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) | wc
join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) | wc
join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) | wc
join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) | wc

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.txt | awk '{ print $2 "\t" $3 "\t" $3 "\t" $5 "\t" $6 }' | grep -v Chr > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.AnnovarFormat

perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/table_annovar.pl /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.AnnovarFormat /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -buildver hg18 -out /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.AnnovarFormat.20170208DL_20160204Vs_annovar.TableAnnovar.refGeneOnly -remove -protocol refGene -operation g -nastring NA 
perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/table_annovar.pl /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.AnnovarFormat /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -buildver hg18 -out /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.AnnovarFormat.TableAnnovar.refGeneOnly -remove -protocol refGene -operation g -nastring NA










#cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.AnnovarFormat.20170208DL_20160204Vs_annovar.TableAnnovar.refGeneOnly.hg18_multianno.txt | perl -lane 'if ($F[5] eq "intergenic") { my @genes1 = split(/,/, $F[6]); my @dists1 = split(/;/, $F[7]); if (((split(/=/, $dists1[0]))[1]) < 10000) { print $F[5], "\t", $genes1[0], "\t", $dists1[0]; } if (((split(/=/, $dists1[1]))[1]) < 10000) { print $F[5], "\t", $genes1[1], "\t", $dists1[1]; } } else { print $F[5], "\t", $F[6], "\t", $F[7]; } ' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.AnnovarFormat.20170208DL_20160204Vs_annovar.TableAnnovar.refGeneOnly.hg18_multianno.RegGeneDist_lt10kb.txt 
#cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.AnnovarFormat.20170208DL_20160204Vs_annovar.TableAnnovar.refGeneOnly.hg18_multianno.txt | perl -lane 'if ($F[5] eq "intergenic") { my @genes1 = split(/,/, $F[6]); my @dists1 = (((split(/=/, ((split(/;/, $F[7]))[0])))[1]), ((split(/=/, ((split(/;/, $F[7]))[1])))[1])); print join("\t", @dists1); }'
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.AnnovarFormat.20170208DL_20160204Vs_annovar.TableAnnovar.refGeneOnly.hg18_multianno.txt | grep -v intergenic | awk '{ print $6 "\t" $7 "\t" $8 }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.AnnovarFormat.20170208DL_20160204Vs_annovar.TableAnnovar.refGeneOnly.hg18_multianno.RegGeneDist_lt10kb.txt  
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.AnnovarFormat.20170208DL_20160204Vs_annovar.TableAnnovar.refGeneOnly.hg18_multianno.RegGeneDist_lt10kb.txt | awk '{ print $2 }' | perl -lane 'if ($F[0] =~ m/,/) { print $F[0]; print join("\n", split(/,/, $F[0])); } else { print join("\t", @F); }' | sort > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.AnnovarFormat.20170208DL_20160204Vs_annovar.TableAnnovar.refGeneOnly.hg18_multianno.RegGeneDist_lt10kb.GeneOnly.txt

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.AnnovarFormat.TableAnnovar.refGeneOnly.hg18_multianno.txt | grep -v intergenic | awk '{ print $6 "\t" $7 "\t" $8 }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.AnnovarFormat.TableAnnovar.refGeneOnly.hg18_multianno.RegGeneDist_lt10kb.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.AnnovarFormat.TableAnnovar.refGeneOnly.hg18_multianno.RegGeneDist_lt10kb.txt | awk '{ print $2 }' | perl -lane 'if ($F[0] =~ m/,/) { print $F[0]; print join("\n", split(/,/, $F[0])); } else { print join("\t", @F); }' | sort > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.AnnovarFormat.TableAnnovar.refGeneOnly.hg18_multianno.RegGeneDist_lt10kb.GeneOnly.txt

join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.AnnovarFormat.20170208DL_20160204Vs_annovar.TableAnnovar.refGeneOnly.hg18_multianno.RegGeneDist_lt10kb.GeneOnly.txt) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.AnnovarFormat.TableAnnovar.refGeneOnly.hg18_multianno.RegGeneDist_lt10kb.GeneOnly.txt) | wc

~~~
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) | wc
     48      48     445
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) | wc   
     25      25     222
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) | wc
     48      48     445
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) | wc
     25      25     222
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) | wc
    183     183    1387
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) | wc
    153     153    1108
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) | wc
   1963    1963   14260
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) | wc
   1816    1816   13026
[  mturchin20@bigmem03  /mnt/gluster/home/mturchin20/Software]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs
1.Condensed.noMIR.uniq.GeneIDs.txt | wc
    152     152    1070
[  mturchin20@bigmem03  /mnt/gluster/home/mturchin20/Software]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed
_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.u
niq.GeneIDs.txt | sort)
CAND1
FAM138D
LOC100132062
LOC100132287
LOC100133331
OR4F16
OR4F29
OR4F3
PGM5P3-AS1
RNU6-2
SNAR-A10
SNAR-A11
SNAR-A14
SNAR-A3
SNAR-A4
SNAR-A5
SNAR-A6
SNAR-A7
SNAR-A8
SNAR-A9
SNORA105A
SNORA105B
SNORD141A
SNORD141B
SRGAP2D
[  mturchin20@bigmem03  /mnt/gluster/home/mturchin20/Software]$for i in `join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort)`; do echo $i; cat /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.bed | grep $i; done
CAND1
chr12   10286435   65994739   CAND1;5;+;NR_103765,NM_018448,NM_001329676,NM_001329675,NM_001329674
chr11   10286435   10408796   CAND1.11;1;+;NR_103765
chr20   34004952   34006695   SCAND1;2;-;NM_016558,NM_033630
FAM138D
chr19   24395   114052897  FAM138D;4;-;NR_026823,NR_026823,NR_026823,NR_026823
LOC100132062
chr5 313754 180687802   LOC100132062;2;+;NR_028325,NR_028325
LOC100132287
chr5 313754 180687802   LOC100132287;2;+;NR_028322,NR_028322
LOC100133331
chr5 313754 180687802   LOC100133331;3;+;NR_028327,NR_028327,NR_028327
OR4F16
chr5 357521 180727832   OR4F16;3;+;NM_001005277,NM_001005277,NM_001005277
OR4F29
chr5 357521 180727832   OR4F29;3;+;NM_001005221,NM_001005221,NM_001005221
OR4F3
chr5 357521 180727832   OR4F3;3;+;NM_001005224,NM_001005224,NM_001005224
PGM5P3-AS1
chr9 62689  114016720   PGM5P3-AS1;4;+;NR_121190,NR_121190,NR_121189,NR_121188
RNU6-2
chr19   844483  182432325  RNU6-2;11;+;NR_125730,NR_125730,NR_125730,NR_125730,NR_125730,NR_125730,NR_125730,NR_125730,NR_125730,NR_125730,NR_125730
SNAR-A10
chr19   53118847   55323591   SNAR-A10;10;-;NR_024229,NR_024229,NR_024229,NR_024229,NR_024229,NR_024229,NR_024229,NR_024229,NR_024229,NR_024229
SNAR-A11
chr19   53118847   55323591   SNAR-A11;10;-;NR_024225,NR_024225,NR_024225,NR_024225,NR_024225,NR_024225,NR_024225,NR_024225,NR_024225,NR_024225
SNAR-A14
chr19   53118847   55323591   SNAR-A14;10;-;NR_024242,NR_024242,NR_024242,NR_024242,NR_024242,NR_024242,NR_024242,NR_024242,NR_024242,NR_024242
SNAR-A3
chr19   53118847   55323591   SNAR-A3;10;-;NR_024214,NR_024214,NR_024214,NR_024214,NR_024214,NR_024214,NR_024214,NR_024214,NR_024214,NR_024214
SNAR-A4
chr19   53118847   55323591   SNAR-A4;10;-;NR_024215,NR_024215,NR_024215,NR_024215,NR_024215,NR_024215,NR_024215,NR_024215,NR_024215,NR_024215
SNAR-A5
chr19   53118847   55323591   SNAR-A5;10;-;NR_024223,NR_024223,NR_024223,NR_024223,NR_024223,NR_024223,NR_024223,NR_024223,NR_024223,NR_024223
SNAR-A6
chr19   53118847   55323591   SNAR-A6;10;-;NR_024227,NR_024227,NR_024227,NR_024227,NR_024227,NR_024227,NR_024227,NR_024227,NR_024227,NR_024227
SNAR-A7
chr19   53118847   55323591   SNAR-A7;10;-;NR_024224,NR_024224,NR_024224,NR_024224,NR_024224,NR_024224,NR_024224,NR_024224,NR_024224,NR_024224
SNAR-A8
chr19   53118847   55323591   SNAR-A8;10;-;NR_024228,NR_024228,NR_024228,NR_024228,NR_024228,NR_024228,NR_024228,NR_024228,NR_024228,NR_024228
SNAR-A9
chr19   53118847   55323591   SNAR-A9;10;-;NR_024226,NR_024226,NR_024226,NR_024226,NR_024226,NR_024226,NR_024226,NR_024226,NR_024226,NR_024226
SNORA105A
chr5 21920319   198059872  SNORA105A;2;+;NR_132787,NR_132787
SNORA105B
chr5 21920319   198059872  SNORA105B;2;+;NR_132788,NR_132788
SNORD141A
chr9 14705386   134885742  SNORD141A;3;+;NR_132980,NR_132980,NR_132980
SNORD141B
chr9 14705386   134885742  SNORD141B;3;+;NR_132981,NR_132981,NR_132981
SRGAP2D
chr1 120808674  204647924  SRGAP2D;2;+;NR_120535,NR_120535
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/Analyses/GeneAnnotationOverlap/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) | head -n 10
ABCB9
ABCC10
ACACB
ACAD10
ACOT11
ADAM1A
ALDH2
ALKBH2
ANAPC5
ARFGAP2
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.txt | awk '{ print $2 "\t" $3 "\t" $3 "\t" $5 "\t" $6 }' | grep -v Chr | head -n 10
10      101902054       101902054       C       T
10      5839619 5839619 G       A
11      109521729       109521729       T       G
11      13313759        13313759        G       A
11      45696596        45696596        T       C
11      47251202        47251202        C       T
11      47857845        47857845        T       C
11      49295647        49295647        C       T
11      50052671        50052671        T       G
1       152511766       152511766       G       T
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.AnnovarFormat.20170208DL_20160204Vs_annovar.TableAnnovar.refGeneOnly.hg18_multianno.txt | perl -lane 'if ($F[5] eq "intergenic") { my @genes1 = split(/,/, $F[6]); my @dists1 = split(/;/, $F[7]); if (((split(/=/, $dists1[0]))[1]) < 10000) { print $F[5], "\t", $genes1[0], "\t", $dists1[0]; } if (((split(/=/, $dists1[1]))[1]) < 10000) { print $F[5], "\t", $genes1[1], "\t", $dists1[1]; } } else { print $F[5], "\t", $F[6], "\t", $F[7]; } ' | head -n 10 
Func.refGene    Gene.refGene    GeneDetail.refGene
intronic        SPTY2D1 NA
intronic        SUGP1   NA
intronic        JMJD1C  NA
intronic        ASAP3   NA
ncRNA_intronic  VIM-AS1 NA
intergenic      BRAP    dist=7526
intronic        PLEC    NA
intronic        FTO     NA
intronic        ST3GAL4 NA
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.AnnovarFormat.20170208DL_20160204Vs_annovar.TableAnnovar.refGeneOnly.hg18_multianno.RegGeneDist_lt10kb.txt | awk '{ print $2 }' | sed 's/,/\n/g' | head -n 20
Gene.refGene
SPTY2D1
SUGP1
JMJD1C
ASAP3
VIM-AS1
BRAP
PLEC
FTO
ST3GAL4
OR4C46
UGT1A1
UGT1A10
UGT1A3
UGT1A4
UGT1A5
UGT1A6
UGT1A7
UGT1A8
UGT1A9
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.AnnovarFormat.20170208DL_20160204Vs_annovar.TableAnnovar.refGeneOnly.hg18_multianno.RegGeneDist_lt10kb.GeneOnly.txt) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.AnnovarFormat.TableAnnovar.refGeneOnly.hg18_multianno.RegGeneDist_lt10kb.GeneOnly.txt) | wc
      1       1      13
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.AnnovarFormat.20170208DL_20160204Vs_annovar.TableAnnovar.refGeneOnly.hg18_multianno.RegGeneDist_lt10kb.GeneOnly.txt) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.AnnovarFormat.TableAnnovar.refGeneOnly.hg18_multianno.RegGeneDist_lt10kb.GeneOnly.txt) | head -n 10
Gene.refGene
~~~




#GIANT2010

mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses
mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap

cat /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.AnnovarFormat | awk '{ print $1 "\t" $2-1 "\t" $3-1 "\t" $4 "\t" $5 }' > /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.AnnovarFormat
cat /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.AnnovarFormat | awk '{ print "chr" $1 "\t" $2 "\t" $3 }' > /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.bed

/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/bedtools slop -i /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.bed -g /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg18.chrom.sizes -b 10000 > /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.10kbPadding.bed
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/bedtools slop -i /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.bed -g /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg18.chrom.sizes -b 500000 > /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.500kbPadding.bed
#2Mb just as a positive control, not going to be normally interested in this range of output
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/bedtools slop -i /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.bed -g /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg18.chrom.sizes -b 2000000 > /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.2MbPadding.bed

/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -f 1 -wb -a /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -f 1 -c -a /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.Counts.vs1.txt
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -wa -wb -a /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.10kbPadding.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -wa -wb -a /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.500kbPadding.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -wa -wb -a /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.2MbPadding.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt 
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | grep -v MIR | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt 
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | grep -v MIR | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | grep -v MIR | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | grep -v MIR | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt

#cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.AnnovarFormat | awk '{ print "chr" $1 "\t" $2 "\t" $3 }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.bed

/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/bedtools slop -i /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.bed -g /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg18.chrom.sizes -b 10000 > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.10kbPadding.bed
#/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/bedtools slop -i /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.bed -g /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg18.chrom.sizes -b 500000 > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bed
#2Mb just as a positive control, not going to be normally interested in this range of output
#/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/bedtools slop -i /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.bed -g /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg18.chrom.sizes -b 2000000 > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bed

/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -f 1 -wb -a /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -f 1 -c -a /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.Counts.vs1.txt
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -wa -wb -a /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.10kbPadding.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -wa -wb -a /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -wa -wb -a /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt 
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | grep -v MIR | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt 
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | grep -v MIR | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | grep -v MIR | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | grep -v MIR | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt

join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) | wc
join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) | wc
join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) | wc
join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) | wc
join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) | wc
join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) | wc
join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) | wc
join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) | wc

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.txt | awk '{ print $2 "\t" $3 "\t" $3 "\t" $5 "\t" $6 }' | grep -v Chr > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.AnnovarFormat

perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/table_annovar.pl /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.AnnovarFormat /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -buildver hg18 -out /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.AnnovarFormat.20170208DL_20160204Vs_annovar.TableAnnovar.refGeneOnly -remove -protocol refGene -operation g -nastring NA 
perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/table_annovar.pl /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.AnnovarFormat /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -buildver hg18 -out /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.AnnovarFormat.TableAnnovar.refGeneOnly -remove -protocol refGene -operation g -nastring NA









cat /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.AnnovarFormat.20170208DL_20160204Vs_annovar.TableAnnovar.refGeneOnly.hg18_multianno.txt | grep -v intergenic | awk '{ print $6 "\t" $7 "\t" $8 }' > /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.AnnovarFormat.20170208DL_20160204Vs_annovar.TableAnnovar.refGeneOnly.hg18_multianno.RegGeneDist_lt10kb.txt  
cat /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.AnnovarFormat.20170208DL_20160204Vs_annovar.TableAnnovar.refGeneOnly.hg18_multianno.RegGeneDist_lt10kb.txt | awk '{ print $2 }' | perl -lane 'if ($F[0] =~ m/,/) { print $F[0]; print join("\n", split(/,/, $F[0])); } else { print join("\t", @F); }' | sort > /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.AnnovarFormat.20170208DL_20160204Vs_annovar.TableAnnovar.refGeneOnly.hg18_multianno.RegGeneDist_lt10kb.GeneOnly.txt

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.AnnovarFormat.TableAnnovar.refGeneOnly.hg18_multianno.txt | grep -v intergenic | awk '{ print $6 "\t" $7 "\t" $8 }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.AnnovarFormat.TableAnnovar.refGeneOnly.hg18_multianno.RegGeneDist_lt10kb.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.AnnovarFormat.TableAnnovar.refGeneOnly.hg18_multianno.RegGeneDist_lt10kb.txt | awk '{ print $2 }' | perl -lane 'if ($F[0] =~ m/,/) { print $F[0]; print join("\n", split(/,/, $F[0])); } else { print join("\t", @F); }' | sort > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.AnnovarFormat.TableAnnovar.refGeneOnly.hg18_multianno.RegGeneDist_lt10kb.GeneOnly.txt

join <(cat /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.AnnovarFormat.20170208DL_20160204Vs_annovar.TableAnnovar.refGeneOnly.hg18_multianno.RegGeneDist_lt10kb.GeneOnly.txt) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.AnnovarFormat.TableAnnovar.refGeneOnly.hg18_multianno.RegGeneDist_lt10kb.GeneOnly.txt) | wc

~~~
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) | wc
     29      29     284
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) | wc
     11      11     110
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) | wc
     29      29     284
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) | wc
     11      11     110
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) | wc
     40      40     369
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) | wc
     20      20     179
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) | wc
    382     382    2913
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) | wc
    331     331    2475
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort)
LOC100132062
LOC100132287
LOC100133331
OR4F16
OR4F29
OR4F3
PGM5P3-AS1
SNORA105A
SNORA105B
SNORD141A
SNORD141B
[  mturchin20@bigmem01  ~/TempStuff]$for i in `join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort)`; do echo $i; cat /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.bed | grep $i; done
LOC100132062
chr5    313754  180687802       LOC100132062;2;+;NR_028325,NR_028325
LOC100132287
chr5    313754  180687802       LOC100132287;2;+;NR_028322,NR_028322
LOC100133331
chr5    313754  180687802       LOC100133331;3;+;NR_028327,NR_028327,NR_028327
OR4F16
chr5    357521  180727832       OR4F16;3;+;NM_001005277,NM_001005277,NM_001005277
OR4F29
chr5    357521  180727832       OR4F29;3;+;NM_001005221,NM_001005221,NM_001005221
OR4F3
chr5    357521  180727832       OR4F3;3;+;NM_001005224,NM_001005224,NM_001005224
PGM5P3-AS1
chr9    62689   114016720       PGM5P3-AS1;4;+;NR_121190,NR_121190,NR_121189,NR_121188
SNORA105A
chr5    21920319        198059872       SNORA105A;2;+;NR_132787,NR_132787
SNORA105B
chr5    21920319        198059872       SNORA105B;2;+;NR_132788,NR_132788
SNORD141A
chr9    14705386        134885742       SNORD141A;3;+;NR_132980,NR_132980,NR_132980
SNORD141B
chr9    14705386        134885742       SNORD141B;3;+;NR_132981,NR_132981,NR_132981
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort)
DNAH10
ERCC6L2
GRM4
IP6K3
LEMD2
LINC01016
LINC02120
LOC100132062
LOC100132287
LOC100133331
LOC101927668
MLN
OR4F16
OR4F29
OR4F3
PGM5P3-AS1
SNORA105A
SNORA105B
SNORD141A
SNORD141B
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.AnnovarFormat.20170208DL_20160204Vs_annovar.TableAnnovar.refGeneOnly.hg18_multianno.RegGeneDist_lt10kb.GeneOnly.txt) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.AnnovarFormat.TableAnnovar.refGeneOnly.hg18_multianno.RegGeneDist_lt10kb.GeneOnly.txt) | wc
      1       1      13
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.AnnovarFormat.20170208DL_20160204Vs_annovar.TableAnnovar.refGeneOnly.hg18_multianno.RegGeneDist_lt10kb.GeneOnly.txt) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs2/bmass.GIANT2010.Vs2.NewSNPs.SNPs.vs1.AnnovarFormat.TableAnnovar.refGeneOnly.hg18_multianno.RegGeneDist_lt10kb.GeneOnly.txt)     
Gene.refGene
~~~




#GIANT2014/5

mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses
mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap

join <(cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.txt | sort -k 1,1) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.IncAllele.txt.gz | sort -k 1,1) | awk '{ print $2, "\t", $3, "\t", $3, "\t", $4, "\t", $5 }' > /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.AnnovarFormat

/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/bedtools slop -i /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.bed -g /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg18.chrom.sizes -b 10000 > /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.10kbPadding.bed
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/bedtools slop -i /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.bed -g /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg18.chrom.sizes -b 500000 > /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.500kbPadding.bed
#2Mb just as a positive control, not going to be normally interested in this range of output
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/bedtools slop -i /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.bed -g /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg18.chrom.sizes -b 2000000 > /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.2MbPadding.bed

/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -f 1 -wb -a /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -f 1 -c -a /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.Counts.vs1.txt
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -wa -wb -a /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.10kbPadding.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -wa -wb -a /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.500kbPadding.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -wa -wb -a /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.2MbPadding.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt 
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | grep -v MIR | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt 
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | grep -v MIR | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | grep -v MIR | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | grep -v MIR | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt

#cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.AnnovarFormat | awk '{ print "chr" $1 "\t" $2 "\t" $3 }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.bed

/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/bedtools slop -i /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.bed -g /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg18.chrom.sizes -b 10000 > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.10kbPadding.bed
#/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/bedtools slop -i /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.bed -g /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg18.chrom.sizes -b 500000 > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bed
#2Mb just as a positive control, not going to be normally interested in this range of output
#/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/bedtools slop -i /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.bed -g /mnt/lustre/home/mturchin20/Software/fetchChromSizes/hg18.chrom.sizes -b 2000000 > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bed

/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -f 1 -wb -a /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -f 1 -c -a /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.Counts.vs1.txt
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -wa -wb -a /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.10kbPadding.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -wa -wb -a /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt
/mnt/lustre/home/mturchin20/Software/bedtools2-2.25.0/bin/intersectBed -wa -wb -a /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bed -b /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.bed > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt 
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | grep -v MIR | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt 
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | grep -v MIR | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | grep -v MIR | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.txt | grep -v MIR | perl -lane 'if ($. == 1) { $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } elsif (eof()) { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); print $F[0], "\t", $F[1], "\t", scalar(@genes), "\t", join(";", @genes); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); print $F[0], "\t", $F[1], "\t1\t", join("", ((split(/;/, $F[6]))[0])); } } else { if ($chrBP eq $F[0] . "_" . $F[1]) { push(@genes, ((split(/;/, $F[6]))[0])); } else { print join("", ((split(/_/, $chrBP))[0])), "\t", join("", ((split(/_/, $chrBP))[1])), "\t", scalar(@genes), "\t", join(";", @genes); $chrBP = $F[0] . "_" . $F[1]; @genes = ((split(/;/, $F[6]))[0]); } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.txt | perl -lane 'print join("\n", split(/;/, $F[$#F]));' | sort | uniq > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt

join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) | wc
join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) | wc
join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) | wc
join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) | wc
join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) | wc
join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) | wc
join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) | wc
join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) | wc

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.txt | awk '{ print $2 "\t" $3 "\t" $3 "\t" $5 "\t" $6 }' | grep -v Chr > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.AnnovarFormat

perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/table_annovar.pl /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.AnnovarFormat /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -buildver hg18 -out /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.AnnovarFormat.20170208DL_20160204Vs_annovar.TableAnnovar.refGeneOnly -remove -protocol refGene -operation g -nastring NA 
perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/table_annovar.pl /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.AnnovarFormat /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -buildver hg18 -out /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.AnnovarFormat.TableAnnovar.refGeneOnly -remove -protocol refGene -operation g -nastring NA









cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.AnnovarFormat.20170208DL_20160204Vs_annovar.TableAnnovar.refGeneOnly.hg18_multianno.txt | grep -v intergenic | awk '{ print $6 "\t" $7 "\t" $8 }' > /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.AnnovarFormat.20170208DL_20160204Vs_annovar.TableAnnovar.refGeneOnly.hg18_multianno.RegGeneDist_lt10kb.txt  
cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.AnnovarFormat.20170208DL_20160204Vs_annovar.TableAnnovar.refGeneOnly.hg18_multianno.RegGeneDist_lt10kb.txt | awk '{ print $2 }' | perl -lane 'if ($F[0] =~ m/,/) { print $F[0]; print join("\n", split(/,/, $F[0])); } else { print join("\t", @F); }' | sort > /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.AnnovarFormat.20170208DL_20160204Vs_annovar.TableAnnovar.refGeneOnly.hg18_multianno.RegGeneDist_lt10kb.GeneOnly.txt

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.AnnovarFormat.TableAnnovar.refGeneOnly.hg18_multianno.txt | grep -v intergenic | awk '{ print $6 "\t" $7 "\t" $8 }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.AnnovarFormat.TableAnnovar.refGeneOnly.hg18_multianno.RegGeneDist_lt10kb.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.AnnovarFormat.TableAnnovar.refGeneOnly.hg18_multianno.RegGeneDist_lt10kb.txt | awk '{ print $2 }' | perl -lane 'if ($F[0] =~ m/,/) { print $F[0]; print join("\n", split(/,/, $F[0])); } else { print join("\t", @F); }' | sort > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.AnnovarFormat.TableAnnovar.refGeneOnly.hg18_multianno.RegGeneDist_lt10kb.GeneOnly.txt

join <(cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.AnnovarFormat.20170208DL_20160204Vs_annovar.TableAnnovar.refGeneOnly.hg18_multianno.RegGeneDist_lt10kb.GeneOnly.txt) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.AnnovarFormat.TableAnnovar.refGeneOnly.hg18_multianno.RegGeneDist_lt10kb.GeneOnly.txt) | wc

~~~
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.txt | wc
    843    2529   18346
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.AnnovarFormat | wc
    843    4215   27710
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.AnnovarFormat | awk '{ print $4 }' | sort | uniq -c
    189 A
    223 C
    223 G
    208 T
[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.AnnovarFormat | awk '{ print $5 }' | sort | uniq -c
    223 A
    206 C
    227 G
    187 T
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) | wc
     54      54     519
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) | wc
     16      16     152
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) | wc
     54      54     519
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) | wc
     16      16     152
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) | wc
    343     343    2521
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) | wc
    292     292    2048
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.uniq.GeneIDs.txt | sort) | wc
   5247    5247   38266
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.2MbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) | wc
   4858    4858   35025
[  mturchin20@bigmem01  ~/TempStuff]$for i in `join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/Analyses/GeneAnnotationOverlap/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.10kbPadding.bedIntersect.20170214.UCSCTableBrowser_refSeq_hg18_DefaultOutput_Condensed_bed.vs1.Condensed.noMIR.uniq.GeneIDs.txt | sort)`; do echo $i; cat /mnt/lustre/home/mturchin20/Data/UCSC/20170214.UCSCTableBrowser_refSeq_hg18.DefaultOutput.Condensed.bed | grep $i; done
CAND1
chr12   10286435        65994739        CAND1;5;+;NR_103765,NM_018448,NM_001329676,NM_001329675,NM_001329674
chr11   10286435        10408796        CAND1.11;1;+;NR_103765
chr20   34004952        34006695        SCAND1;2;-;NM_016558,NM_033630
FAM138D
chr19   24395   114052897       FAM138D;4;-;NR_026823,NR_026823,NR_026823,NR_026823
LOC100132062
chr5    313754  180687802       LOC100132062;2;+;NR_028325,NR_028325
LOC100132287
chr5    313754  180687802       LOC100132287;2;+;NR_028322,NR_028322
LOC100133331
chr5    313754  180687802       LOC100133331;3;+;NR_028327,NR_028327,NR_028327
LOC100288778
chr15   4224    100334331       LOC100288778;2;+;NR_130745,NR_130745
OR4F16
chr5    357521  180727832       OR4F16;3;+;NM_001005277,NM_001005277,NM_001005277
OR4F29
chr5    357521  180727832       OR4F29;3;+;NM_001005221,NM_001005221,NM_001005221
OR4F3
chr5    357521  180727832       OR4F3;3;+;NM_001005224,NM_001005224,NM_001005224
PGM5P3-AS1
chr9    62689   114016720       PGM5P3-AS1;4;+;NR_121190,NR_121190,NR_121189,NR_121188
RNU6-2
chr19   844483  182432325       RNU6-2;11;+;NR_125730,NR_125730,NR_125730,NR_125730,NR_125730,NR_125730,NR_125730,NR_125730,NR_125730,NR_125730,NR_125730
SNORA105A
chr5    21920319        198059872       SNORA105A;2;+;NR_132787,NR_132787
SNORA105B
chr5    21920319        198059872       SNORA105B;2;+;NR_132788,NR_132788
SNORD141A
chr9    14705386        134885742       SNORD141A;3;+;NR_132980,NR_132980,NR_132980
SNORD141B
chr9    14705386        134885742       SNORD141B;3;+;NR_132981,NR_132981,NR_132981
SRGAP2D
chr1    120808674       204647924       SRGAP2D;2;+;NR_120535,NR_120535
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.AnnovarFormat.20170208DL_20160204Vs_annovar.TableAnnovar.refGeneOnly.hg18_multianno.RegGeneDist_lt10kb.GeneOnly.txt) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.AnnovarFormat.TableAnnovar.refGeneOnly.hg18_multianno.RegGeneDist_lt10kb.GeneOnly.txt) | wc
      1       1      13
[  mturchin20@bigmem01  ~/TempStuff]$join <(cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.AnnovarFormat.20170208DL_20160204Vs_annovar.TableAnnovar.refGeneOnly.hg18_multianno.RegGeneDist_lt10kb.GeneOnly.txt) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/bmass.GIANT2014_5.Vs2.NewSNPs.SNPs.vs1.AnnovarFormat.TableAnnovar.refGeneOnly.hg18_multianno.RegGeneDist_lt10kb.GeneOnly.txt)     
Gene.refGene

~~~









#HaemgenRBC2012



#HaemgenRBC2016









##20170218
##little bit of multivariate modeling setup work

#ModelPriorMatrix <- cbind(Models,ModelPriors)[order(ModelPriors, decreasing=TRUE),]
#ModelPriorMatrix <- data.frame(cbind(ModelPriorMatrix, cumsum(ModelPriorMatrix[,5])))
#colnames(ModelPriorMatrix) <- c(DataSources, "pValue", "Cumm_pValue")


GetModelPriorMatrix(bmassOutput1$Models, bmassOutput1$ModelPriors, bmassOutput1$LogFile)$ModelPriorMatrix)

GetTopModelsPerSNPViaPosteriors(DataSources, bmassOutput1$PreviousSNPs, GetModelPriorMatrix(bmassOutput1$Models, bmassOutput1$ModelPriors, bmassOutput1$LogFile)$ModelPriorMatrix, bmassOutput1$LogFile)
GetTopModelsPerSNPViaPosteriors(DataSources, bmassOutput1$NewSNPs, GetModelPriorMatrix(bmassOutput1$Models, bmassOutput1$ModelPriors, bmassOutput1$LogFile)$ModelPriorMatrix, bmassOutput1$LogFile)

#marginal.postprobs(bmassOutput1$NewSNPs$Posteriors[,5:ncol(bmassOutput1$NewSNPs$Posteriors)], bmassOutput1$Models, 1)

nana2 <- GetMarginalPosteriors(DataSources, bmassOutput1$NewSNPs, bmassOutput1$Models, SigmaAlphas = c(0.005,0.0075,0.01,0.015,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.15), bmassOutput1$LogFile)
nana2$ListSNPs$Marginals$pU[,1:10]

#Make 'Analysis folder for marginal stuff?'
jpeg("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.NewSNPs.MarginalHeatplots.DirectAssoc.vs1.jpg", width=4000, height=2000, res=300)
#	NewSNPs.Marginals.DirectAssoc.Clustered <- PlotMarginalPosteriors(DataSources, nana2$ListSNPs, "D", Main="BlahBlah")
	NewSNPs.Marginals.DirectAssoc.Clustered <- PlotMarginalPosteriors(DataSources, bmassOutput1$NewSNPs, "D", Main="BlahBlah")
	#PlotMarginalPosteriors <- function (DataSources, ListSNPs, Marginal) {
dev.off()

jpeg("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.NewSNPs.MarginalHeatplots.DirectAssoc.wDirection.vs1.jpg", width=4000, height=2000, res=300)
	NewSNPs.Marginals.DirectAssoc.Clustered <- PlotMarginalPosteriors.withDirection(DataSources, bmassOutput1$NewSNPs, "D", Main="BlahBlah")
dev.off()



#ModelPriorMatrix.Paste <- apply(ModelPriorMatrix[,1:4], 1, function(x) paste(x, collapse="_")) 
#PerSNPTopModels <- cbind(Models.Paste[apply(ListSNPs$Posteriors, 2, which.max)], apply(ListSNPs$Posteriors, 2, max))
#SummaryOfTopModels <- c()
#
#for (i in unique(PerSNPTopModels[,1])) { 
#	PerSNPTopModels.Subset <- matrix(PerSNPTopModels[PerSNPTopModels[,1]==i,], nrow=length(which(PerSNPTopModels[,1]==i)))
#	SummaryOfTopModels <- rbind(SummaryOfTopModels, cbind(i, nrow(PerSNPTopModels.Subset), round(mean(as.numeric(PerSNPTopModels.Subset[,2])), digits=3), ModelPriorMatrix[ModelPriorMatrix.Paste == i, 5]))
#}
#
#SummaryOfTopModels <- SummaryOfTopModels[order(as.numeric(SummaryOfTopModels[,2]), decreasing=TRUE),]
#}  


#From `/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GLC.MTedits.For2013.vs3.SignCrrct.vs1.ForCSHLPoster.R` 
~~~
library(ggplot2)
library(reshape2)
set.seed(100)
.
.
.
sub.pp <- posteriorprob(lbf.bigmat[,l==1],ebprior.glhits)
sub.pp.marginals <- marginal.postprobs(sub.pp, lbf$gamma, length(sigmaa))
#sub.pp.marginals.DirAssoc <- sub.pp.marginals[[2]]
sub.pp.marginals.DirAssoc <- t(sub.pp.marginals[[2]])
#colnames(sub.pp.marginals.DirAssoc) <- sub$snp
#sub.pp.marginals.DirAssoc <- rbind(as.character(sub$snp), sub.pp.marginals.DirAssoc)
sub.pp.marginals.DirAssoc <- data.frame(cbind(as.character(sub$snp), sub.pp.marginals.DirAssoc))
sub.pp.marginals.DirAssoc[,2] <- as.numeric(as.character(sub.pp.marginals.DirAssoc[,2]))
sub.pp.marginals.DirAssoc[,3] <- as.numeric(as.character(sub.pp.marginals.DirAssoc[,3]))
sub.pp.marginals.DirAssoc[,4] <- as.numeric(as.character(sub.pp.marginals.DirAssoc[,4]))
sub.pp.marginals.DirAssoc[,5] <- as.numeric(as.character(sub.pp.marginals.DirAssoc[,5]))
#rownames(sub.pp.marginals.DirAssoc) <- c("TC", "TG", "HDL", "LDL")
#rownames(sub.pp.marginals.DirAssoc) <- c("LDL", "HDL", "TC", "TG")
#rownames(sub.pp.marginals.DirAssoc) <- c("snp", "LDL", "HDL", "TC", "TG")
colnames(sub.pp.marginals.DirAssoc) <- c("snp", "LDL", "HDL", "TC", "TG")
sub.pp.marginals.DirAssoc <- sub.pp.marginals.DirAssoc[order(sub$lbfav, decreasing=TRUE),]
sub.pp.marginals.DirAssoc$snp <- factor(sub.pp.marginals.DirAssoc$snp, levels=sub[order(sub$lbfav),][,4])

#newhits$snp <- factor(newhits$snp, levels=newhits[order(newhits$lbfav),][,1])

sub.pp.marginals.DirAssoc.newhits <- sub.pp.marginals.DirAssoc[sub[order(sub$lbfav, decreasing=TRUE),]$annot==0 & sub[order(sub$lbfav, decreasing=TRUE),]$lbfav>3.989499 & sub[order(sub$lbfav, decreasing=TRUE),]$nmin>50000,]
sub.pp.marginals.DirAssoc.newhits$snp <- factor(sub.pp.marginals.DirAssoc.newhits$snp, levels=sub[sub$annot==0 & sub$lbfav>3.989499 & sub$nmin>50000,][order(sub[sub$annot==0 & sub$lbfav>3.989499 & sub$nmin>50000,]$lbfav),][,4])

.
.
.

#melt(t(sub.pp.marginals.DirAssoc[1:4,1:5]))
#ggplot(t(sub.pp.marginals.DirAssoc[1:4,1:5]))
#ggplot(melt(prevhits2[,c(1,5:8)]), aes(variable, snp, value)) + geom_tile(aes(fill=value), colour="white") + scale_fill_gradient(low = "white", high="steelblue")


##jpeg("GlobalLipids2013.vs3.ForCSHLPoster.20160830PreProbGenom.vs1.jpg", width=2000, height=4000, res=300)
#jpeg("GlobalLipids2013.vs3.ForCSHLPoster.20160830PreProbGenom.MarginalHeatplots.NewHits.vs1.jpg", width=2000, height=4000, res=300)
jpeg("GlobalLipids2013.vs3.ForCSHLPoster.20160830PreProbGenom.MarginalHeatplots.NewHits.vs1.jpg", width=4000, height=2000, res=300)

##ggplot(melt(sub.pp.marginals.DirAssoc), aes(variable, snp, value)) + geom_tile(aes(fill=value), colour="white") + scale_fill_gradient(low = "white", high="steelblue")
##ggplot(melt(newhits[,c(1,5:8)]), aes(variable, snp, value)) + geom_tile(aes(fill=value), colour="white") + scale_fill_gradient(low = "white", high="steelblue")
#ggplot(melt(sub.pp.marginals.DirAssoc.newhits), aes(variable, snp, value)) + geom_tile(aes(fill=value), colour="white") + scale_fill_gradient(low = "white", high="steelblue")
#ggPlotData_newhits <- melt(sub.pp.marginals.DirAssoc.newhits)
#ggPlotData_newhits$snp <- factor(ggPlotData_newhits$snp, levels=sub.pp.marginals.DirAssoc.newhits[,1][hclust(dist(sub.pp.marginals.DirAssoc.newhits[,2:5]))$order])
ggPlotData_newhits <- melt(sub.pp.marginals.DirAssoc.newhits[hclust(dist(sub.pp.marginals.DirAssoc.newhits[,2:5]))$order,][c(1:20,60:92),])
ggPlotData_newhits$snp <- factor(ggPlotData_newhits$snp, levels=sub.pp.marginals.DirAssoc.newhits[,1][hclust(dist(sub.pp.marginals.DirAssoc.newhits[,2:5]))$order][c(1:20,60:92)])

#ggplot(melt(sub.pp.marginals.DirAssoc.newhits), aes(variable, snp, value)) + geom_tile(aes(fill=value), colour="white") + scale_fill_gradient(low = "white", high="steelblue") + coord_flip()
ggplot(ggPlotData_newhits, aes(variable, snp, value)) + geom_tile(aes(fill=value), colour="white") + scale_fill_gradient(low = "white", high="steelblue") + coord_flip() + theme(axis.text.x = element_text(angle = 90, hjust = 1))


dev.off()

jpeg("GlobalLipids2013.vs3.ForCSHLPoster.20160830PreProbGenom.MarginalHeatplots.PrevHits.vs1.jpg", width=2000, height=4000, res=300)

ggplot(melt(prevhits.pp.marginals.DirAssoc), aes(variable, snp, value)) + geom_tile(aes(fill=value), colour="white") + scale_fill_gradient(low = "white", high="steelblue")

dev.off()

jpeg("GlobalLipids2013.vs3.ForCSHLPoster.20160830PreProbGenom.MarginalHeatplots.MargSNPs.vs1.jpg", width=2000, height=4000, res=300)

ggplot(melt(sub.pp.marginals.DirAssoc), aes(variable, snp, value)) + geom_tile(aes(fill=value), colour="white") + scale_fill_gradient(low = "white", high="steelblue")

dev.off()
.
.
.
~~~



#From `/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GLC.MTedits.For2013.vs3.ForCSHLPoster.R`
~~~
.
.
.
lbf=compute.allBFs.fromZscores(Z,VYY,gl$nmin,gl$maf,sigmaa)
lbf.bigmat=do.call(rbind,lbf$lbf) #lbf$lbf is a list of matrices, with one element (a matrix) for each sigma; this stacks these matrices together into a big matrix with nsnp columns, and nsigma*nmodels rows

#collapse takes a vector that is nsigmmaa stacked m-vectors, and adds them together to produce a single m vector (averages over values of sigmaa)
collapse = function(x,nsigmaa){return(apply(matrix(x,ncol=nsigmaa),1,sum))}

#compute empirical bayes priors from global lipids hit SNPs
gl.glhits = gl[gl$annot==1,] # subset of SNPs reported in Teslovich 2010 paper
lbf.glhits = lbf.bigmat[,gl$annot==1]
ebprior.glhits = em.priorprobs(lbf.glhits,lbf$prior,100)
ebprior.glhits2 = em.priorprobs(lbf.glhits,lbf$prior*runif(length(lbf$prior)),100)
ebprior.glhits3 = em.priorprobs(lbf.glhits,lbf$prior*runif(length(lbf$prior)),100)
ebprior.glhits4 = em.priorprobs(lbf.glhits,lbf$prior*runif(length(lbf$prior)),100)

ebprior.glhits.collapse =collapse(ebprior.glhits,length(sigmaa))
ebprior.glhits.collapse2 =collapse(ebprior.glhits2,length(sigmaa))
ebprior.glhits.collapse3 =collapse(ebprior.glhits3,length(sigmaa))
ebprior.glhits.collapse4 =collapse(ebprior.glhits4,length(sigmaa))

pp.glhits = posteriorprob(lbf.glhits,ebprior.glhits) #posterior prob on models for gl hits
pp.glhits.collapse =  apply(pp.glhits,2,collapse, nsigmaa=length(sigmaa))

# this returns a list with elements  pU pD and pI
marginal.glhits = marginal.postprobs(pp.glhits, lbf$gamma,length(sigmaa))

#looking at which models are favored by the prior
cumsum(sort(ebprior.glhits.collapse,decreasing=TRUE))
lbf$gamma[order(ebprior.glhits.collapse,decreasing=TRUE),]
modelmatrix = cbind(lbf$gamma,ebprior.glhits.collapse)[order(ebprior.glhits.collapse,decreasing=TRUE),]
modelmatrix = data.frame(cbind(modelmatrix,cumsum(modelmatrix[,5])))
colnames(modelmatrix)= c("TC","TG","HDL","LDL","p","cump")
.
.
.
#compute for each SNP which class it is most likely assigned to
#(all assoc, or notLDL, or notHDL or notTG)
ppmatrix = cbind(pp.glhits.collapse)[order(ebprior.glhits.collapse,decreasing=TRUE),]
pp.classmatrix = rbind(colSums(ppmatrix[allassoc,]),colSums(ppmatrix[allbut1assoc & modelmatrix[,4]==0,]),colSums(ppmatrix[allbut1assoc & modelmatrix[,3]==0,]),colSums(ppmatrix[allbut1assoc & modelmatrix[,2]==0,]),colSums(ppmatrix[allbut1assoc & modelmatrix[,1]==0,]))
bestclass= apply(pp.classmatrix, 2,which.max)
.
.
.
#compute for each SNP which class it is most likely assigned to
#(all assoc, or notLDL, or notHDL or notTG)
ppmatrix.newhits = cbind(pp.newhits.collapse)[order(ebprior.glhits.collapse,decreasing=TRUE),]
pp.newhits.classmatrix = rbind(colSums(ppmatrix.newhits[allassoc,]),colSums(ppmatrix.newhits[allbut1assoc & modelmatrix[,4]==0,]),colSums(ppmatrix.newhits[allbut1assoc & modelmatrix[,3]==0,]),colSums(ppmatrix.newhits[allbut1assoc & modelmatrix[,2]==0,]),colSums(ppmatrix.newhits[allbut1assoc & modelmatrix[,1]==0,]))
bestclass= apply(pp.newhits.classmatrix, 2,which.max)
cbind(as.character(newhits$snp),bestclass,apply(pp.newhits.classmatrix,2,max))
.
.
.
head(modelmatrix)
modelmatrix.paste <- apply(modelmatrix[,1:4], 1, function(x) paste(x, collapse="_"))
table(modelmatrix.paste[apply(ppmatrix.newhits, 2, which.max)])
modelmatrix[apply(as.matrix(names(table(modelmatrix.paste[apply(ppmatrix.newhits, 2, which.max)]))), 1, function(x) which(modelmatrix.paste == x)),5]

snpmodels <- cbind(modelmatrix.paste[apply(ppmatrix.newhits, 2, which.max)], apply(ppmatrix.newhits, 2, max))
topmodels <- c()

for (i in unique(snpmodels[,1])) {
        snpmodels.subset <- matrix(snpmodels[snpmodels[,1]==i,], nrow=length(which(snpmodels[,1]==i)))
        topmodels <- rbind(topmodels, cbind(i, nrow(snpmodels.subset), mean(as.numeric(snpmodels.subset[,2])), sd(as.numeric(snpmodels.subset[,2])), modelmatrix[modelmatrix.paste == i,5]))
}

topmodels <- topmodels[order(as.numeric(topmodels[,2]), decreasing=TRUE),]
 
#topmodels <- cbind(names(table(modelmatrix.paste[apply(ppmatrix.newhits, 2, which.max)])), table(modelmatrix.paste[apply(ppmatrix.newhits, 2, which.max)]), modelmatrix[apply(as.matrix(names(table(modelmatrix.paste[apply(ppmatrix.newhits, 2, which.max)]))), 1, function(x) which(modelmatrix.paste == x)),5])
#topmodels <- topmodels[order(as.numeric(topmodels[,2]), decreasing=FALSE),]


oldsnpmodels <- cbind(modelmatrix.paste[apply(ppmatrix, 2, which.max)], apply(ppmatrix, 2, max))
oldtopmodels <- c() 

for (i in unique(oldsnpmodels[,1])) {
        oldsnpmodels.subset <- matrix(oldsnpmodels[oldsnpmodels[,1]==i,], nrow=length(which(oldsnpmodels[,1]==i)))
        oldtopmodels <- rbind(oldtopmodels, cbind(i, nrow(oldsnpmodels.subset), mean(as.numeric(oldsnpmodels.subset[,2])), sd(as.numeric(oldsnpmodels.subset[,2])), modelmatrix[modelmatrix.paste == i,5]))
}  

oldtopmodels <- oldtopmodels[order(as.numeric(oldtopmodels[,2]), decreasing=TRUE),]
.
.
.
#~~~
#> head(modelmatrix)
#  TC TG HDL LDL         p      cump
#1  1  1   1   1 0.2384938 0.2384938
#2  1  2   1   1 0.1538137 0.3923075
#3  2  1   1   2 0.1359187 0.5282262
#4  1  1   2   1 0.1169160 0.6451422
#5  2  1   1   1 0.1061912 0.7513333
#6  1  2   1   2 0.0662903 0.8176236
#> modelmatrix.paste <- apply(modelmatrix[,1:4], 1, function(x) paste(x, collapse="_"))
#> table(modelmatrix.paste[apply(ppmatrix.newhits, 2, which.max)])
#
# 1_1_1_1 1_1_1_2 1_1_2_1 1_2_1_0 1_2_1_1 1_2_1_2 2_1_1_1 2_1_1_2
#       6       4      11       2       8      10      10      23
#>
#> oldtopmodels
#i
#[1,] "2_1_1_2" "27" "0.649805935158347" "0.126514408595193"
#[2,] "1_1_2_1" "25" "0.560267755605778" "0.144186211707476"
#[3,] "1_1_1_1" "23" "0.695426592023571" "0.245764342956809"
#[4,] "1_2_1_1" "18" "0.542869375102103" "0.170171215709513"
#[5,] "2_1_1_1" "16" "0.538959150076247" "0.114011285745139"
#[6,] "1_2_1_2" "14" "0.426076489932265" "0.13374214470218"
#[7,] "1_2_2_1" "10" "0.455935405804636" "0.142378648270453"
#[8,] "1_2_1_0" "9"  "0.546339652410049" "0.180107361222572"
#[9,] "1_1_1_2" "2"  "0.420209033811409" "0.0364324249667825"
#[10,] "0_1_1_0" "1"  "0.976129541178598" NA
#[11,] "1_1_2_2" "1"  "0.551973978945456" NA
#[12,] "1_0_1_2" "1"  "0.867490268910215" NA
#[13,] "1_1_1_0" "1"  "0.540316620623034" NA
#
#[1,] "0.135918699667997"
#[2,] "0.116915972449258"
#[3,] "0.238493786373925"
#[4,] "0.153813713870225"
#[5,] "0.106191169098223"
#[6,] "0.0662903020181461"
#[7,] "0.054043193014096"
#[8,] "0.04360787170365"
#[9,] "0.0526273773446415"
.
.
.
~~~



#From /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3/GLC.MTedits.ForChoongwon2016.RanHalfFlip.4Phenos_1.vs3.SignCrrct.vs1.R
~~~
.
.
.
Z = cbind(gl$Z.Sat,gl$Z.Hb,gl$Z.Pulse,gl$Z.LvBrth)
.
.
.
Z.Directions <- apply(Z, c(1,2), function(x) { y <- 0; if (x > 0) { y <- 1; } else if (x < 0) { y <- -1; } else if (x == 0) { y <- 0; } else { y <- NA; }; return(y);})

#Z.Directions.sub.lbfavunif <- Z.Directions[l.lbfavunif==1,]
Z.Directions.sub.lbfavunif <- Z.Directions[l.lbfavunif==1,][order(sub.lbfavunif$lbfavunif, decreasing=TRUE),]
.
.
.
posterior.sub.NoSum.marginals.DirAssoc.wDir <- posterior.sub.NoSum.marginals.DirAssoc
posterior.sub.NoSum.marginals.DirAssoc.wDir[,2:5] <- posterior.sub.NoSum.marginals.DirAssoc[,2:5] * Z.Directions.sub.lbfavunif

jpeg("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3/Choongwon2016.RanHalfFlip.PhenoGroup1.DirAssocMarginalPosteriors.HeatPlot.wDir.hclust.vs1.jpeg", width=6000, height=2000, res=300)
#jpeg("/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs3/Choongwon2016.RanHalfFlip.PhenoGroup1.DirAssocMarginalPosteriors.HeatPlot.wDir.hclust_onDirVals.vs1.jpeg", width=6000, height=2000, res=300)

ggPlotData <- melt(posterior.sub.NoSum.marginals.DirAssoc.wDir[hclust(dist(posterior.sub.NoSum.marginals.DirAssoc.wDir[,2:5]))$order,])
ggPlotData$snp <- factor(ggPlotData$snp, levels=posterior.sub.NoSum.marginals.DirAssoc[,1][hclust(dist(posterior.sub.NoSum.marginals.DirAssoc[,2:5]))$order])
#ggPlotData$snp <- factor(ggPlotData$snp, levels=posterior.sub.NoSum.marginals.DirAssoc.wDir[,1][hclust(dist(posterior.sub.NoSum.marginals.DirAssoc.wDir[,2:5]))$order])

#ggplot(ggPlotData, aes(variable, snp, value)) + geom_tile(aes(fill=value), colour="white") + scale_fill_gradient(low = "darkred", high="steelblue") + coord_flip() + theme(axis.text.x = element_text(angle = 90, hjust = 1))
ggplot(ggPlotData, aes(variable, snp, value)) + geom_tile(aes(fill=value), colour="white") + scale_fill_gradientn(colours=c("darkred", "white", "steelblue")) + coord_flip() + theme(axis.text.x = element_text(angle = 90, hjust = 1))

#scale_fill_gradientn(colours=c("red","white","green"),

dev.off()
.
.
.
~~~












##20170218
##2nd level/stage annotation work -- first looking at expanded annovar results, then omim, then DisGeNET

mkdir /mnt/gluster/home/mturchin20/Data/OMIM
mkdir /mnt/gluster/home/mturchin20/Data/DisGeNET
mkdir OMIM list from GIANT?
mkdir GIANT 2017 dataset paper?

cd /mnt/gluster/home/mturchin20/Data/OMIM

#20170218 NOTE -- the below URLs were retreived by registering with OMIM and receiving an email with the links. Access via these links should be available until Feb 2018 apparently. Also including a small bit from the email.
~~~
.
.
.
You can view a list of all the data you have access with the following URL:


https://omim.org/downloads/qaHhLEotTnSo0n7tMSTP6Q
.
.
.
~~~

wget https://omim.org/static/omim/data/mim2gene.txt  
wget --no-check-certificate https://data.omim.org/downloads/qaHhLEotTnSo0n7tMSTP6Q/mimTitles.txt
wget --no-check-certificate https://data.omim.org/downloads/qaHhLEotTnSo0n7tMSTP6Q/genemap.txt
wget --no-check-certificate https://data.omim.org/downloads/qaHhLEotTnSo0n7tMSTP6Q/morbidmap.txt
wget --no-check-certificate https://data.omim.org/downloads/qaHhLEotTnSo0n7tMSTP6Q/genemap2.txt


cd /mnt/gluster/home/mturchin20/Data/DisGeNET

#20170218 NOTE -- below URLs were retreived from http://www.disgenet.org/web/DisGeNET/menu/downloads. Additionally, the below information is from the webpage
~~~
.
.
.
Important notice
The DisGeNET database is made available under the Open Database License. For more information, see the Legal Notices page.

DATA in tab separated files
1. Curated gene-disease associations
The file contains gene-disease associations from UNIPROT, CTD (human subset), ClinVar, Orphanet, and the GWAS Catalog.

2. BeFree gene-disease associations
The file contains gene-disease associations obtained by text mining MEDLINE abstracts using the BeFree system.

3. ALL gene-disease associations
The file contains all gene-disease associations in DisGeNET.

4. BeFree SNP-gene-disease associations
The file contains SNP-gene-disease associations obtained by text mining MEDLINE abstracts using the BeFree system.

5. All SNP-gene-disease associations
The file contains All SNP-gene-disease associations in DisGeNET.

6.Publications with gene-disease associations from BeFree System
The file contains the publications supporting the gene-disease associations obtained by text mining MEDLINE abstracts (2015-2016) using the BeFree system.

7. README file
.
.
.
~~~

wget http://www.disgenet.org/ds/DisGeNET/results/curated_gene_disease_associations.tsv.gz  
wget http://www.disgenet.org/ds/DisGeNET/results/befree_gene_disease_associations.tsv.gz
wget http://www.disgenet.org/ds/DisGeNET/results/all_gene_disease_associations.tsv.gz
wget http://www.disgenet.org/ds/DisGeNET/results/befree_snps_sentences_pubmeds.tsv.gz
wget http://www.disgenet.org/ds/DisGeNET/results/all_snps_sentences_pubmeds.tsv.gz
wget http://www.disgenet.org/ds/DisGeNET/results/befree_results_only_version_4.0.tar.gz
wget http://www.disgenet.org/ds/DisGeNET/results/readme.txt




#GlobalLipids2010







#GlobalLipids2013


#GIANT2010


#GIANT2014_%


#HaemgenRBC2012


#HaemgenRBC2016










##2017
##Specific GIANT annotation stuff? eg OMIM list from Wood et al. 2014 and that other paper?

#Marouli et al. 2017 "Rare and low-frequency coding variants alter human adult height" http://www.nature.com/nature/journal/v542/n7640/full/nature21039.html





##2017
##PathwayWork









#####Looking up OMIM and individual loci results
#####20170125
#
#
##20170216 NOTE -- thinking the below section is somewhat getting divided as being part of the 'annovar loci prev/new comparisons part' and then also later simulations/enrichment part...nore sure yet, but kind of freezing this section right here
##20170218 NOTE -- Furthermore, thinking I'll have 3 kind of 'annotation' sections -- the first with mostly just the geneIDs for the 'checking prev/new loci lists against one another for overlap' thing, another set for a deeper dive into the gene list/variant qualities in order to start investigating possible biological stories via OMIM, DisGeNET, etc, then last after this section with a full-on annovar setup in expectation for the simulation/enrichment things?
#####Annotating prev and new hits to find loci, dbl checking loci in new hits don't overlap with prev hit loci (and/or also checking how close/far away they are/may be)
#####(also going to check out varying expanding windows to see if/what get other loci too doing this)
#####(also checking if any loci that are not picked up from earlier dataset still are not picked up by 'prev' of new data)
#####20170123
#
#
##GlobalLipids2010
#
#Previous, New
#
##Getting NA SNP rs9411489 info from https://www.ncbi.nlm.nih.gov/projects/SNP/snp_ref.cgi?rs=635634 as before
#join <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.Marker.txt | sort) <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.tbl.gz | sort) | perl -lane 'print $F[11], "\t", $F[12], "\t", $F[12], "\t", uc($F[1]), "\t", uc($F[2]);' > /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.AnnovarFormat
#cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.AnnovarFormat | grep -v NA | cat - <(echo "9	133279427	133279427	G	A") > /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.AnnovarFormat
#ln -s /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.AnnovarFormat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.AnnovarFormat.20170123_ljbedits
#
#cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.NewSNPs.SNPs.vs1.txt | awk '{ print $2 "\t" $3 "\t" $3 "\t" $5 "\t" $6 }' | grep -v Chr > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.NewSNPs.SNPs.vs1.AnnovarFormat
#
#mv /mnt/gluster/home/mturchin20/Lab_Stuff /mnt/gluster/home/mturchin20/LabMisc
#mkdir /mnt/gluster/home/mturchin20/Software
#mv /mnt/lustre/home/mturchin20/Software/annovar /mnt/gluster/home/mturchin20/Software/20130911_annovar
##Downloaded 'latest' annovar via a link provided in an e-mail (request page -- http://www.openbioinformatics.org/annovar/annovar_download_form.php)
#scp -p mturchin20@wolfy.uchicago.edu:/Users/mturchin20/LabMisc/Software/20170208DL_annovar.latest.tar.gz /mnt/gluster/home/mturchin20/Software
#tar -xvzf /mnt/gluster/home/mturchin20/Software/20170208DL_annovar.latest.tar.gz
#mv /mnt/gluster/home/mturchin20/Software/annovar /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar
#
##Did the below with instructions from http://stackoverflow.com/questions/2920301/clear-a-file-without-changing-its-timestamp
#cd /mnt/lustre/home/mturchin20/Software/annovar/humandb/
#for i in `ls -lrt | grep -v ^d | awk '{ print $9 }'`; do TMPFILE=`mktemp`; touch -r /mnt/lustre/home/mturchin20/Software/annovar/humandb/$i $TMPFILE; cat /dev/null > /mnt/lustre/home/mturchin20/Software/annovar/humandb/$i; touch -r $TMPFILE /mnt/lustre/home/mturchin20/Software/annovar/humandb/$i; rm $TMPFILE; done
##for i in `ls -lrt /mnt/lustre/home/mturchin20/Software/annovar/humandb/. | grep -v ^d | awk '{ print $9 }'`; do TMPFILE=`mktemp`; touch -r $i $TMPFILE; cat /dev/null > $i; touch -r $TMPFILE $i; rm $TMPFILE; done
#
##~perl /mnt/lustre/home/mturchin20/Software/annovar/20170123_ljbedits_summarize_annovar.pl /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.AnnovarFormat /mnt/lustre/home/mturchin20/Software/annovar/humandb/ --ver1000g 1000g2010 --verdbsnp 129 --veresp 6500 --buildver hg18 --outfile /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.AnnovarFormat.20170123_ljbedits --remove 
##~rm /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.AnnovarFormat.20170123_ljbedits*dropped /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.AnnovarFormat.20170123_ljbedits*hg18_phastConsElements44way /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.AnnovarFormat.20170123_ljbedits*genomicSuperDups
##~perl /mnt/lustre/home/mturchin20/Software/annovar/20170123_ljbedits_summarize_annovar.pl /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.AnnovarFormat.20170123_ljbedits /mnt/lustre/home/mturchin20/Software/annovar/humandb/ --ver1000g 1000g2010 --verdbsnp 129 --veresp 6500 --buildver hg18 --remove
##perl /mnt/lustre/home/mturchin20/Software/annovar/20170123_ljbedits_summarize_annovar.pl /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.AnnovarFormat /mnt/lustre/home/mturchin20/Software/annovar/humandb/ --ver1000g 1000g2010 --verdbsnp 129 --veresp 6500 --buildver hg18 --outfile /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.AnnovarFormat.20170123_ljbedits --remove
##~perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/annotate_variation.pl -filter -dbtype dbnsfp33a -buildver hg18 -out /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.AnnovarFormat.20170208DL_20160204Vs_annovar.dbsfp33a /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.AnnovarFormat /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb -otherinfo
#
##20170209
#perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/annotate_variation.pl --downdb -webfrom annovar refGene /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -build hg18
#perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/annotate_variation.pl --downdb genomicSuperDups /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -build hg18
#perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/annotate_variation.pl --downdb phastConsElements44way /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -build hg18
#perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/annotate_variation.pl --downdb tfbsConsSites /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -build hg18  
#perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/annotate_variation.pl --downdb cytoBand /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -build hg18
#perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/annotate_variation.pl --downdb wgRna /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -build hg18
#perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/annotate_variation.pl --downdb targetScanS /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -build hg18
#perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/annotate_variation.pl --downdb gwasCatalog /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -build hg18
#perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/annotate_variation.pl --downdb -webfrom annovar snp129 /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -build hg18
#perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/annotate_variation.pl --downdb -webfrom annovar snp138 /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -build hg18
#perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/annotate_variation.pl --downdb -webfrom annovar avsift /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -build hg18
#perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/annotate_variation.pl --downdb -webfrom annovar ljb_all /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -build hg18
#perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/annotate_variation.pl --downdb -webfrom annovar ljb23_all /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -build hg18
#perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/annotate_variation.pl --downdb -webfrom annovar ljb26_all /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -build hg18
#perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/annotate_variation.pl --downdb -webfrom annovar dbnsfp33a /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -build hg18
##perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/annotate_variation.pl --downdb -webfrom annovar esp6500_all /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -build hg18
#perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/annotate_variation.pl --downdb -webfrom annovar esp6500siv2_all /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -build hg18
#perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/annotate_variation.pl --downdb -webfrom annovar exac03 /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -build hg18
##perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/annotate_variation.pl --downdb -webfrom annovar 1000g2012apr /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -build hg18
##perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/annotate_variation.pl --downdb -webfrom annovar 1000g2010 /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -build hg18
#perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/annotate_variation.pl --downdb -webfrom annovar 1000g2014oct /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -build hg18
#
##hg19 only
##clinvar_20161128
#
#perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/annotate_variation.pl --downdb -webfrom annovar refGene /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -build hg19
#perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/annotate_variation.pl --downdb genomicSuperDups /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -build hg19
#perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/annotate_variation.pl --downdb phastConsElements46way /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -build hg19
#perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/annotate_variation.pl --downdb tfbsConsSites /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -build hg19  
#perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/annotate_variation.pl --downdb cytoBand /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -build hg19
#perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/annotate_variation.pl --downdb wgRna /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -build hg19
#perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/annotate_variation.pl --downdb targetScanS /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -build hg19
#perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/annotate_variation.pl --downdb gwasCatalog /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -build hg19
#perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/annotate_variation.pl --downdb -webfrom annovar snp129 /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -build hg19
#perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/annotate_variation.pl --downdb -webfrom annovar snp138 /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -build hg19
#perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/annotate_variation.pl --downdb -webfrom annovar avsift /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -build hg19
##~perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/annotate_variation.pl --downdb -webfrom annovar ljb_all /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -build hg19
##~perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/annotate_variation.pl --downdb -webfrom annovar ljb23_all /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -build hg19
#perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/annotate_variation.pl --downdb -webfrom annovar ljb26_all /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -build hg19
#perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/annotate_variation.pl --downdb -webfrom annovar dbnsfp33a /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -build hg19
##perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/annotate_variation.pl --downdb -webfrom annovar esp6500_all /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -build hg19
#perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/annotate_variation.pl --downdb -webfrom annovar esp6500siv2_all /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -build hg19
#perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/annotate_variation.pl --downdb -webfrom annovar exac03 /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -build hg19
##perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/annotate_variation.pl --downdb -webfrom annovar 1000g2012apr /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -build hg19
##perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/annotate_variation.pl --downdb -webfrom annovar 1000g2010 /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -build hg19
#perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/annotate_variation.pl --downdb -webfrom annovar 1000g2014oct /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -build hg19
#perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/annotate_variation.pl --downdb -webfrom annovar clinvar_20161128 /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -build hg19
#perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/annotate_variation.pl --downdb -webfrom annovar cadd /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -build hg19
#cadd
#
#
#
#
#
#
##annotate_variation.pl -filter -dbtype 1000g2012apr_eur -buildver hg19 -out ex1 example/ex1.avinput humandb/
##table_annovar.pl example/ex1.avinput humandb/ -buildver hg19 -out myanno -remove -protocol refGene,cytoBand,genomicSuperDups,esp6500si_all,1000g2012apr_all,snp138,ljb23_all -operation g,r,r,f,f,f,f -nastring 
##-csvout
#
#perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/table_annovar.pl /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.AnnovarFormat /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -buildver hg18 -out /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.AnnovarFormat.20170208DL_20160204Vs_annovar.TableAnnovar -remove -protocol refGene,genomicSuperDups,phastConsElements44way,cytoBand,tfbsConsSites,wgRna,targetScanS,gwasCatalog,snp129,snp138,1000g2014oct_all,esp6500siv2_all,exac03,dbnsfp33a -operation g,r,r,r,r,r,r,r,f,f,f,f,f,f -nastring NA
#
##/mnt/lustre/home/mturchin20/Software/liftOver/hg18ToHg19.over.chain.gz
##
##usage:
##   liftOver oldFile map.chain newFile unMapped
#
#
#
#
#
#
#
#cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.AnnovarFormat | awk '{ print "chr"$1"\t"$2"\t"$3+1"\t"$1"_"$2}' > /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.liftoverBed
#
#/mnt/lustre/home/mturchin20/Software/liftOver/liftOver /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.liftoverBed /mnt/lustre/home/mturchin20/Software/liftOver/hg18ToHg19.over.chain.gz /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.liftoverBed.hg19.bed /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.liftoverBed.hg19.bed.dropped
#
#join <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.liftoverBed.hg19.bed | sed 's/chr//g' | awk '{ print $4 "\t" $1 "\t" $2 "\t" $2 }' | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.AnnovarFormat | awk '{ print $1 "_" $2 "\t" $4 "\t" $5 }' | sort -k 1,1) | perl -lane 'push(@F, $F[0]); splice (@F,0,1); print join("\t", @F);' > /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.liftoverBed.hg19.AnnovarFormat
#
#perl /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/table_annovar.pl /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.liftoverBed.hg19.AnnovarFormat /mnt/gluster/home/mturchin20/Software/20170208DL_20160204Vs_annovar/humandb/ -buildver hg19 -out /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.hg19.AnnovarFormat.20170208DL_20160204Vs_annovar.TableAnnovar -remove -protocol cadd -operation f -nastring NA -otherinfo
#
#
#want to know if expand new SNPs 1Mb, 2Mb out, what additional loci do they overlap -- do loci that are now showing up already exist among the previous snps? 
#
#
#/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.NewSNPs.SNPs.vs1.500kbPadding.bed
#
#
#
##refGene
##
##Conserved genomic elements annotation
##Transcription factor binding site
##Identify cytogenetic band
##disrupting microRNAs and snoRNAs
##disrupting predicted microRNA binding sites
##Identify variants located in segmental duplications
##Identify variants reported in previously published GWAS
##
##esp
##1000g
##snp138/9/whatev
##ljbsomething
#
#
#
#
#
#/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/.
#bmass.GlobalLipids2010.Vs2.NewSNPs.SNPs.vs1.txt
#bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.txt.gz
#
#
#look at 5kb, 10kb, 20kb border windows for calling something related to a locus? or just be conservative?...probably see how different the results are
#bedtools intersect thing....slop too....
#take list from prev and doublecheck it's different from new results, and not even (hopefully?) nearby new results or vice-versa
#
#~~~
#[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$join <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.Marker.txt | sort) <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.tbl.gz | sort) | perl -lane 'print $F[11], "\t", $F[12], "\t", $F[12], "\t", uc($F[1]), "\t", uc($F[2]);' | wc
#    102     510    2546
#[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$join <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.Marker.txt | sort) <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.tbl.gz | sort) | perl -lane 'print $F[11], "\t", $F[12], "\t", $F[12], "\t", uc($F[1]), "\t", uc($F[2]);' | grep -v NA | wc
#    101     505    2533
#[  mturchin20@bigmem03  /mnt/gluster/data/external_private_supp/HaemgenRBC2016]$cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.Marker.txt | wc
#    102     102    1024
#[  mturchin20@bigmem01  ~/TempStuff]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.NewSNPs.SNPs.vs1.txt | awk '{ print $2 "\t" $3 "\t" $3 "\t" $5 "\t" $6 }' | grep -v Chr | head -n 10
#10      101902184       101902184       C       T
#10      17300296        17300296        G       A
#10      45333283        45333283        C       A
#10      5236085 5236085 A       G
#11      47215429        47215429        T       C
#1       149225460       149225460       G       A
#11      51368666        51368666        C       T
#11      54886216        54886216        T       C
#11      55776161        55776161        T       A
#11      75132669        75132669        A       C
#[  mturchin20@spudhead  ~/Software/annovar/humandb]$ls -lrt
#total 64954160
#-rw-rw-r-- 1 mturchin20 mturchin20    10626173 Aug  3  2006 hg18_genomicSuperDups.txt
#-rw-rw-r-- 1 mturchin20 mturchin20   193606817 Feb  2  2009 hg18_phastConsElements44way.txt
#-rw-rw-r-- 1 mturchin20 mturchin20   205139082 Dec  6  2009 hg19_phastConsElements46way.txt
#-rw-rw-r-- 1 mturchin20 mturchin20  2609192331 Sep 12  2011 hg19_avsift.txt
#-rw-rw-r-- 1 mturchin20 mturchin20    10570294 Oct 25  2011 hg19_genomicSuperDups.txt
#-rw-rw-r-- 1 mturchin20 mturchin20     6317127 Feb 22  2012 hg19_avsift.txt.idx
#-rw-rw-r-- 1 mturchin20 mturchin20     5233500 Feb 22  2012 hg19_ljb_all.txt.idx
#-rw-r--r-- 1 mturchin20 mturchin20    73382469 Feb 23  2012 hg18_CEU.sites.2010_03.txt.idx
#-rw-r--r-- 1 mturchin20 mturchin20   196121521 Feb 23  2012 hg18_CEU.sites.2010_03.txt
#-rw-r--r-- 1 mturchin20 mturchin20    68307610 Feb 23  2012 hg18_JPTCHB.sites.2010_03.txt.idx
#-rw-r--r-- 1 mturchin20 mturchin20   158204020 Feb 23  2012 hg18_JPTCHB.sites.2010_03.txt
#-rw-r--r-- 1 mturchin20 mturchin20   263387600 Feb 23  2012 hg18_YRI.sites.2010_03.txt
#-rw-r--r-- 1 mturchin20 mturchin20    77895206 Feb 23  2012 hg18_YRI.sites.2010_03.txt.idx
#-rw-rw-r-- 1 mturchin20 mturchin20  5795736715 Feb 24  2012 hg19_ljb_all.txt
#-rw-r--r-- 1 mturchin20 mturchin20    89456008 May 25  2012 hg19_ALL.sites.2012_04.txt.idx
#-rw-r--r-- 1 mturchin20 mturchin20    87006131 May 25  2012 hg19_AMR.sites.2012_04.txt.idx
#-rw-r--r-- 1 mturchin20 mturchin20    85553308 May 25  2012 hg19_ASN.sites.2012_04.txt.idx
#-rw-r--r-- 1 mturchin20 mturchin20    86463107 May 25  2012 hg19_EUR.sites.2012_04.txt.idx
#-rw-r--r-- 1 mturchin20 mturchin20  1372992404 May 25  2012 hg19_ALL.sites.2012_04.txt
#-rw-r--r-- 1 mturchin20 mturchin20   691158224 May 25  2012 hg19_AMR.sites.2012_04.txt
#-rw-r--r-- 1 mturchin20 mturchin20   502418754 May 25  2012 hg19_ASN.sites.2012_04.txt
#-rw-r--r-- 1 mturchin20 mturchin20   598097961 May 25  2012 hg19_EUR.sites.2012_04.txt
#-rw-r--r-- 1 mturchin20 mturchin20   894204297 May 25  2012 hg19_AFR.sites.2012_04.txt
#-rw-r--r-- 1 mturchin20 mturchin20    87520670 May 25  2012 hg19_AFR.sites.2012_04.txt.idx
#-rw-rw-r-- 1 mturchin20 mturchin20    63392958 Jun 26  2012 hg19_esp6500_all.txt
#-rw-rw-r-- 1 mturchin20 mturchin20     5558185 Jun 26  2012 hg19_esp6500_all.txt.idx
#-rw-rw-r-- 1 mturchin20 mturchin20  1513192372 Aug  5  2012 hg19_snp129.txt
#-rw-rw-r-- 1 mturchin20 mturchin20    94950668 Aug  5  2012 hg19_snp129.txt.idx
#-rw-r--r-- 1 mturchin20 mturchin20    89345455 Aug 21  2012 hg18_ALL.sites.2012_04.txt.idx
#-rw-r--r-- 1 mturchin20 mturchin20  1370830422 Aug 21  2012 hg18_ALL.sites.2012_04.txt
#-rw-r--r-- 1 mturchin20 mturchin20        5965 Sep 11  2013 hg18_example_db_generic.txt
#-rw-r--r-- 1 mturchin20 mturchin20       30748 Sep 11  2013 hg18_cytoBand.txt
#-rw-r--r-- 1 mturchin20 mturchin20     2076545 Sep 11  2013 hg18_example_db_gff3.txt
#-rw-r--r-- 1 mturchin20 mturchin20       17094 Sep 11  2013 GRCh37_MT_ensGeneMrna.fa
#-rw-r--r-- 1 mturchin20 mturchin20        3131 Sep 11  2013 GRCh37_MT_ensGene.txt
#drwxr-xr-x 2 mturchin20 mturchin20        4096 Sep 11  2013 genometrax-sample-files-gff/
#-rw-rw-r-- 1 mturchin20 mturchin20     5924981 Jan  7  2014 hg18_avsift.txt.idx
#-rw-rw-r-- 1 mturchin20 mturchin20  2286099588 Jan  7  2014 hg18_avsift.txt
#-rw-rw-r-- 1 mturchin20 mturchin20     6139175 Jan  7  2014 hg18_esp6500_all.txt.idx
#-rw-rw-r-- 1 mturchin20 mturchin20    68983971 Jan  7  2014 hg18_esp6500_all.txt
#-rw-rw-r-- 1 mturchin20 mturchin20     5243750 Jan  7  2014 hg18_ljb_all.txt.idx
#-rw-rw-r-- 1 mturchin20 mturchin20  5795099570 Jan  7  2014 hg18_ljb_all.txt
#-rw-rw-r-- 1 mturchin20 mturchin20  1685177611 Jan  7  2014 hg18_snp129.txt
#-rw-rw-r-- 1 mturchin20 mturchin20    96928355 Jan  7  2014 hg18_snp129.txt.idx
#-rw-rw-r-- 1 mturchin20 mturchin20     6611074 Feb 23  2014 hg19_ljb23_all.txt.idx
#-rw-rw-r-- 1 mturchin20 mturchin20 12073254842 Feb 23  2014 hg19_ljb23_all.txt
#-rw-rw-r-- 1 mturchin20 mturchin20   151652203 Feb 24  2014 hg19_refGeneMrna.fa
#-rw-rw-r-- 1 mturchin20 mturchin20    13817961 Feb 24  2014 hg19_refGene.txt
#-rw-rw-r-- 1 mturchin20 mturchin20    31869378 Feb 24  2014 hg19_refLink.txt
#-rw-rw-r-- 1 mturchin20 mturchin20 12066978717 May 27  2014 hg18_ljb23_all.txt
#-rw-rw-r-- 1 mturchin20 mturchin20     6615864 May 27  2014 hg18_ljb23_all.txt.idx
#-rw-rw-r-- 1 mturchin20 mturchin20 11045700932 Sep 25  2014 hg18_ljb26_all.txt
#-rw-rw-r-- 1 mturchin20 mturchin20    18738649 Sep 25  2014 hg18_ljb26_all.txt.idx
#-rw-rw-r-- 1 mturchin20 mturchin20    32674836 Mar 22  2015 hg18_refLink.txt
#-rw-rw-r-- 1 mturchin20 mturchin20    14642615 Dec 11  2015 hg18_refGene.txt
#-rw-rw-r-- 1 mturchin20 mturchin20   165910269 Dec 11  2015 hg18_refGeneMrna.fa
#-rw-rw-r-- 1 mturchin20 mturchin20    18955411 Jan 23 23:57 hg18_dbnsfp33a.txt.idx
#-rw-rw-r-- 1 mturchin20 mturchin20  3607350277 Jan 24 00:05 hg18_dbnsfp33a.txt.gz
#-rw-rw-r-- 1 mturchin20 mturchin20        5601 Feb  8 16:53 annovar_downdb.log
#[  mturchin20@spudhead  ~/Software/annovar/humandb]$for i in `ls -lrt | grep -v ^d | awk '{ print $9 }'`; do TMPFILE=`mktemp`; touch -r /mnt/lustre/home/mturchin20/Software/annovar/humandb/$i $TMPFILE; cat /dev/null > /mnt/lustre/home/mturchin20/Software/annovar/humandb/$i; touch -r $TMPFILE /mnt/lustre/home/mturchin20/Software/annovar/humandb/$i; rm $TMPFILE; done
#[  mturchin20@spudhead  ~/Software/annovar/humandb]$ls -lrt                                                                                                                                                                                                                                                                total 4
#-rw-rw-r-- 1 mturchin20 mturchin20    0 Feb  2  2009 hg18_phastConsElements44way.txt
#-rw-rw-r-- 1 mturchin20 mturchin20    0 Dec  6  2009 hg19_phastConsElements46way.txt
#-rw-rw-r-- 1 mturchin20 mturchin20    0 Sep 12  2011 hg19_avsift.txt
#-rw-rw-r-- 1 mturchin20 mturchin20    0 Oct 25  2011 hg19_genomicSuperDups.txt
#-rw-rw-r-- 1 mturchin20 mturchin20    0 Feb 22  2012 hg19_avsift.txt.idx
#-rw-rw-r-- 1 mturchin20 mturchin20    0 Feb 22  2012 hg19_ljb_all.txt.idx
#-rw-r--r-- 1 mturchin20 mturchin20    0 Feb 23  2012 hg18_CEU.sites.2010_03.txt.idx
#-rw-r--r-- 1 mturchin20 mturchin20    0 Feb 23  2012 hg18_CEU.sites.2010_03.txt
#-rw-r--r-- 1 mturchin20 mturchin20    0 Feb 23  2012 hg18_JPTCHB.sites.2010_03.txt.idx
#-rw-r--r-- 1 mturchin20 mturchin20    0 Feb 23  2012 hg18_JPTCHB.sites.2010_03.txt
#-rw-r--r-- 1 mturchin20 mturchin20    0 Feb 23  2012 hg18_YRI.sites.2010_03.txt
#-rw-r--r-- 1 mturchin20 mturchin20    0 Feb 23  2012 hg18_YRI.sites.2010_03.txt.idx
#-rw-rw-r-- 1 mturchin20 mturchin20    0 Feb 24  2012 hg19_ljb_all.txt
#-rw-r--r-- 1 mturchin20 mturchin20    0 May 25  2012 hg19_ALL.sites.2012_04.txt.idx
#-rw-r--r-- 1 mturchin20 mturchin20    0 May 25  2012 hg19_AMR.sites.2012_04.txt.idx
#-rw-r--r-- 1 mturchin20 mturchin20    0 May 25  2012 hg19_ASN.sites.2012_04.txt.idx
#-rw-r--r-- 1 mturchin20 mturchin20    0 May 25  2012 hg19_EUR.sites.2012_04.txt.idx
#-rw-r--r-- 1 mturchin20 mturchin20    0 May 25  2012 hg19_ALL.sites.2012_04.txt
#-rw-r--r-- 1 mturchin20 mturchin20    0 May 25  2012 hg19_AMR.sites.2012_04.txt
#-rw-r--r-- 1 mturchin20 mturchin20    0 May 25  2012 hg19_ASN.sites.2012_04.txt
#-rw-r--r-- 1 mturchin20 mturchin20    0 May 25  2012 hg19_EUR.sites.2012_04.txt
#-rw-r--r-- 1 mturchin20 mturchin20    0 May 25  2012 hg19_AFR.sites.2012_04.txt
#-rw-r--r-- 1 mturchin20 mturchin20    0 May 25  2012 hg19_AFR.sites.2012_04.txt.idx
#-rw-rw-r-- 1 mturchin20 mturchin20    0 Jun 26  2012 hg19_esp6500_all.txt
#-rw-rw-r-- 1 mturchin20 mturchin20    0 Jun 26  2012 hg19_esp6500_all.txt.idx
#-rw-rw-r-- 1 mturchin20 mturchin20    0 Aug  5  2012 hg19_snp129.txt
#-rw-rw-r-- 1 mturchin20 mturchin20    0 Aug  5  2012 hg19_snp129.txt.idx
#-rw-r--r-- 1 mturchin20 mturchin20    0 Aug 21  2012 hg18_ALL.sites.2012_04.txt.idx
#-rw-r--r-- 1 mturchin20 mturchin20    0 Aug 21  2012 hg18_ALL.sites.2012_04.txt
#-rw-r--r-- 1 mturchin20 mturchin20    0 Sep 11  2013 hg18_example_db_generic.txt
#-rw-r--r-- 1 mturchin20 mturchin20    0 Sep 11  2013 hg18_cytoBand.txt
#-rw-r--r-- 1 mturchin20 mturchin20    0 Sep 11  2013 hg18_example_db_gff3.txt
#-rw-r--r-- 1 mturchin20 mturchin20    0 Sep 11  2013 GRCh37_MT_ensGeneMrna.fa
#-rw-r--r-- 1 mturchin20 mturchin20    0 Sep 11  2013 GRCh37_MT_ensGene.txt
#drwxr-xr-x 2 mturchin20 mturchin20 4096 Sep 11  2013 genometrax-sample-files-gff/
#-rw-rw-r-- 1 mturchin20 mturchin20    0 Jan  7  2014 hg18_avsift.txt.idx
#-rw-rw-r-- 1 mturchin20 mturchin20    0 Jan  7  2014 hg18_avsift.txt
#-rw-rw-r-- 1 mturchin20 mturchin20    0 Jan  7  2014 hg18_esp6500_all.txt.idx
#-rw-rw-r-- 1 mturchin20 mturchin20    0 Jan  7  2014 hg18_esp6500_all.txt
#-rw-rw-r-- 1 mturchin20 mturchin20    0 Jan  7  2014 hg18_ljb_all.txt.idx
#-rw-rw-r-- 1 mturchin20 mturchin20    0 Jan  7  2014 hg18_ljb_all.txt
#-rw-rw-r-- 1 mturchin20 mturchin20    0 Jan  7  2014 hg18_snp129.txt
#-rw-rw-r-- 1 mturchin20 mturchin20    0 Jan  7  2014 hg18_snp129.txt.idx
#-rw-rw-r-- 1 mturchin20 mturchin20    0 Feb 23  2014 hg19_ljb23_all.txt.idx
#-rw-rw-r-- 1 mturchin20 mturchin20    0 Feb 23  2014 hg19_ljb23_all.txt
#-rw-rw-r-- 1 mturchin20 mturchin20    0 Feb 24  2014 hg19_refGeneMrna.fa
#-rw-rw-r-- 1 mturchin20 mturchin20    0 Feb 24  2014 hg19_refGene.txt
#-rw-rw-r-- 1 mturchin20 mturchin20    0 Feb 24  2014 hg19_refLink.txt
#-rw-rw-r-- 1 mturchin20 mturchin20    0 May 27  2014 hg18_ljb23_all.txt
#-rw-rw-r-- 1 mturchin20 mturchin20    0 May 27  2014 hg18_ljb23_all.txt.idx
#-rw-rw-r-- 1 mturchin20 mturchin20    0 Sep 25  2014 hg18_ljb26_all.txt
#-rw-rw-r-- 1 mturchin20 mturchin20    0 Sep 25  2014 hg18_ljb26_all.txt.idx
#-rw-rw-r-- 1 mturchin20 mturchin20    0 Mar 22  2015 hg18_refLink.txt
#-rw-rw-r-- 1 mturchin20 mturchin20    0 Dec 11  2015 hg18_refGene.txt
#-rw-rw-r-- 1 mturchin20 mturchin20    0 Dec 11  2015 hg18_refGeneMrna.fa
#-rw-rw-r-- 1 mturchin20 mturchin20    0 Jan 23 23:57 hg18_dbnsfp33a.txt.idx
#-rw-rw-r-- 1 mturchin20 mturchin20    0 Jan 24 00:05 hg18_dbnsfp33a.txt.gz
#-rw-rw-r-- 1 mturchin20 mturchin20    0 Feb  8 16:53 annovar_downdb.log
#-rw-rw-r-- 1 mturchin20 mturchin20    0 Feb  9 16:10 hg18_genomicSuperDups.txt
#[  mturchin20@bigmem03  /mnt/gluster/home/mturchin20/Software]$cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.liftoverBed.hg19.bed | head -n 10
#chr11   18632984   18632985   11_18589560
#chr2 165513091  165513092  2_165221337
#chr19   19407718   19407719   19_19268718
#chr2 21225281   21225282   2_21078786
#chr10   65027610   65027611   10_64697616
#chr6 161089817  161089818  6_161009807
#chr12   112072424  112072425  12_110556807
#chr8 145043543  145043544  8_145115531
#chr11   126243952  126243953  11_125749162
#chr12   57792580   57792581   12_56078847
#[  mturchin20@bigmem03  /mnt/gluster/home/mturchin20/Software]$join <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.liftoverBed.hg19.bed | sed 's/chr//g' | awk '{ print $4 "\t" $1 "\t" $2 "\t" $2 }' | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.AnnovarFormat | awk '{ print $1 "_" $2 "\t" $4 "\t" $5 }' | sort -k 1,1) | head -n 10
#10_113923876 10 113933886 113933886 A G
#10_64697616 10 65027610 65027610 A T
#10_94829632 10 94839642 94839642 A G
#11_10345358 11 10388782 10388782 A G
#11_116154127 11 116648917 116648917 C G
#11_122027585 11 122522375 122522375 T C
#11_125749162 11 126243952 126243952 A G
#11_18589560 11 18632984 18632984 T C
#11_46699823 11 46743247 46743247 T C
#11_61326406 11 61569830 61569830 T C
#[  mturchin20@bigmem03  /mnt/gluster/home/mturchin20/Software]$join <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.liftoverBed.hg19.bed | sed 's/chr//g' | awk '{ print $4 "\t" $1 "\t" $2 "\t" $2 }' | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.AnnovarFormat | awk '{ print $1 "_" $2 "\t" $4 "\t" $5 }' | sort -k 1,1) | perl -lane 'push(@F, $F[0]); splice (@F,0,1); print join("\t", @F);' > /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.liftoverBed.hg19.AnnovarFormat 
#[  mturchin20@bigmem03  /mnt/gluster/home/mturchin20/Software]$cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.noNA.NArplcd.liftoverBed.hg19.AnnovarFormat | head -n 10
#10 113933886    113933886  A  G 10_113923876
#10 65027610 65027610    A  T  10_64697616
#10 94839642 94839642    A  G  10_94829632
#11 10388782 10388782    A  G  11_10345358
#11 116648917    116648917  C  G 11_116154127
#11 122522375    122522375  T  C 11_122027585
#11 126243952    126243952  A  G 11_125749162
#11 18632984 18632984    T  C  11_18589560
#11 46743247 46743247    T  C  11_46699823
#11 61569830 61569830    T  C  11_61326406
#
#~~~
#
#
##GlobalLipids2013
#
#/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.ChrBP.txt
#/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs2/bmass.GlobalLipids2013.Vs2.NewSNPs.SNPs.vs1.txt
#
#cp -p /mnt/lustre/home/mturchin20/Software/annovar/summarize_annovar.pl /mnt/lustre/home/mturchin20/Software/annovar/20170123_ljbedits_summarize_annovar.pl
#
#
# --outfile
#
#perl /mnt/lustre/home/mturchin20/Software/annovar/summarize_annovar.pl /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs2/bmass.GlobalLipids2010.Vs2.MergedDataSources.vs1.AnnovarFormat /mnt/lustre/home/mturchin20/Software/annovar/humandb/ --ver1000g 1000g2010 --verdbsnp 129 --veresp 6500 --buildver hg18 --remove
#--outfile
#
#cat 
#
#
#
##GIANT2010
#
#/mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.BPMinus1.txt
#
#
##GIANT2014_5
#
#/mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.WHRadjBMI_no2nds.MarkerChrBP.BPMinus1.txt
#/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/.
#
#
##HaemgenRBC2012
#
#
##HaemgenRBC2016
#
#
#~~~
#
#~~~



##
##





























###
###
###Part Db
###Below are analyses that are best grouped by dataset, versus above where analyses are best grouped by that analysis (eg if an analysis is particularly short or straight forward, it may be better to just clump all the runs of that analysis together in one spot than spread across disparate, study-specific sections -- somewhat subjective here).
###
###

##GlobalLipids2010
##20160311



#Pathway work
#20160311
mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs1/Analyses
mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs1/Analyses/PathwayWork/

join <(cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.Marker.txt | sort) <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.tbl.gz | sort) | awk '{ print $12, "\t", $13, "\t", $13, "\t", $2, "\t", $3 }' | sed 's/a/A/g' | sed 's/g/G/' | sed 's/c/C/g' | sed 's/t/T/g' > /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.Marker.AnnovarFormat 

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/multivariate/globallipids/newtophits.txt | awk '{ print $1, "\t", $3, "\t", $5 }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/multivariate/globallipids/newtophits.MarkerChrBP.txt

join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/multivariate/globallipids/newtophits.MarkerChrBP.txt | sort) <(zcat /data/external_public/GlobalLipids2010/HDL_with_Effect.wHapMap23.expanded.tbl.gz | sort) | awk '{ print $2, "\t", $3, "\t", $3, "\t", $4, "\t", $5 }' | sed 's/a/A/g' | sed 's/g/G/' | sed 's/c/C/g' | sed 's/t/T/g' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs1/Analyses/PathwayWork/newtophits.AnnovarFormat

perl /mnt/lustre/home/mturchin20/Software/annovar/summarize_annovar.pl /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.Marker.AnnovarFormat /mnt/lustre/home/mturchin20/Software/annovar/humandb/ --ver1000g 1000g2012apr --verdbsnp 129 --veresp 6500 --buildver hg19 --remove

perl /mnt/lustre/home/mturchin20/Software/annovar/summarize_annovar.pl /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs1/Analyses/PathwayWork/newtophits.AnnovarFormat /mnt/lustre/home/mturchin20/Software/annovar/humandb/ --ver1000g 1000g2012apr --verdbsnp 129 --veresp 6500 --buildver hg19 --remove

cat /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.Marker.AnnovarFormat.genome_summary.csv | grep -v intergenic | grep -v ncRNA | awk -F, '{ print $2 }' | sed 's/"//g' | grep -v Gene > /mnt/lustre/home/mturchin20/Data/GlobalLipids2010/GlobalLipids2010.Table1.Marker.AnnovarFormat.genome_summary.GeneIDs.no_Intergenic_ncRNA

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs1/Analyses/PathwayWork/newtophits.AnnovarFormat.genome_summary.csv | grep -v intergenic | grep -v ncRNA | awk -F, '{ print $2 }' | sed 's/"//g' | grep -v Gene > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2010/Vs1/Analyses/PathwayWork/newtophits.AnnovarFormat.genome_summary.GeneIDs.no_Intergenic_ncRNA



cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/PathwayWork/HaemgenRBC2012.Table1.Edited.MarkerChrBP.AnnovarFormat.genome_summary.csv | grep -v intergenic | grep -v ncRNA | awk -F, '{ print $2 }' | sed 's/"//g' | grep -v Gene > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/PathwayWork/HaemgenRBC2012.Table1.Edited.MarkerChrBP.AnnovarFormat.genome_summary.GeneIDs.no_Intergenic_ncRNA






##GlobalLipids2013
##20160311

mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses
mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/PathwayWork/

join -1 1 -2 3 <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.txt | sort ) <(zcat /data/external_public/GlobalLipid/jointGwasMc_HDL.txt.gz | sort -k 3,3) | awk '{ print $3, "\t", $4, "\t", $5 }' | sed 's/\:/ /g' | sed 's/chr//g' | sed 's/a/A/g' | sed 's/g/G/' | sed 's/c/C/g' | sed 's/t/T/g' | perl -lane 'splice(@F, 1, 0, $F[1]); print join("\t", @F);' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/PathwayWork/ng.2797-S1.edited.vs2.carriageRemoved.AnnovarFormat 

join -1 1 -2 3 <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.txt | awk '{ print $1, "\t", $2, "\t", $3 }' | sort ) <(zcat /data/external_public/GlobalLipid/jointGwasMc_HDL.txt.gz | sort -k 3,3) | awk '{ print $2, "\t", $3, "\t", $3, "\t", $6, "\t", $7 }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/PathwayWork/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.AnnovarFormat 

perl /mnt/lustre/home/mturchin20/Software/annovar/summarize_annovar.pl /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/PathwayWork/ng.2797-S1.edited.vs2.carriageRemoved.AnnovarFormat /mnt/lustre/home/mturchin20/Software/annovar/humandb/ --ver1000g 1000g2012apr --verdbsnp 129 --veresp 6500 --buildver hg19 --remove

perl /mnt/lustre/home/mturchin20/Software/annovar/summarize_annovar.pl /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/PathwayWork/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.AnnovarFormat /mnt/lustre/home/mturchin20/Software/annovar/humandb/ --ver1000g 1000g2012apr --verdbsnp 129 --veresp 6500 --buildver hg19 --remove

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/PathwayWork/ng.2797-S1.edited.vs2.carriageRemoved.AnnovarFormat.genome_summary.csv | grep -v intergenic | grep -v ncRNA | awk -F, '{ print $2 }' | sed 's/"//g' | grep -v Gene > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/PathwayWork/ng.2797-S1.edited.vs2.carriageRemoved.AnnovarFormat.genome_summary.GeneIDs.no_Intergenic_ncRNA

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/PathwayWork/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.AnnovarFormat.genome_summary.csv | grep -v intergenic | grep -v ncRNA | awk -F, '{ print $2 }' | sed 's/"//g' | grep -v Gene > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/PathwayWork/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.AnnovarFormat.genome_summary.GeneIDs.no_Intergenic_ncRNA


#20160616
#Followup analyses based on comments from NHS presentation -- looking further down into univariate results (how much less signif are the new hits from the original data), uniBF vs other BFs, heatmap of orig, new, miss Zscore hits
#Also in prep for committee meeting
#Continuing to use the 'ForCSHL' versions of scripts to keep track of code and trying things out

~~~
[  mturchin20@spudling07  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.txt | awk '{ print $1 }' | head -n 10
snp
rs895953
rs4808802
rs12525163
rs9646133
rs3741782
rs10733608
rs10054063
rs12588415
rs6968554
[  mturchin20@spudling07  ~/Data/iGENBP2015]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.txt | awk '{ print $1 }' | sort) <(zcat /data/external_public/GlobalLipid/jointGwasMc_LDL.txt.gz | awk '{ print $3, "\t", $9 }' | sort) | head -n 10
rs10054063 0.6767
rs1035744 0.1581
rs10408163 0.2384
rs10460587 0.2448
rs1062219 0.0762
rs10733608 0.0221
rs10757056 2.139e-06
rs10832027 0.04054
rs10861661 0.9286
rs10947207 2.448e-12
[  mturchin20@spudling07  ~/Data/iGENBP2015]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.txt | awk '{ print $1 }' | sort) <(zcat /data/external_public/GlobalLipid/jointGwasMc_LDL.txt.gz | awk '{ print $3, "\t", $9 }' | sort) | cat <(echo "rsID LDL_pVal") - | head -n 10
rsID LDL_pVal
rs10054063 0.6767
rs1035744 0.1581
rs10408163 0.2384
rs10460587 0.2448
rs1062219 0.0762
rs10733608 0.0221
rs10757056 2.139e-06
rs10832027 0.04054
rs10861661 0.9286
~~~

/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals

join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.txt | awk '{ print $1 }' | sort) <(zcat /data/external_public/GlobalLipid/jointGwasMc_LDL.txt.gz | awk '{ print $3, "\t", $9 }' | sort) | cat <(echo "rsID LDL_pVal") - | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_LDL.rsIDpVal.txt.gz
join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.txt | awk '{ print $1 }' | sort) <(zcat /data/external_public/GlobalLipid/jointGwasMc_HDL.txt.gz | awk '{ print $3, "\t", $9 }' | sort) | cat <(echo "rsID HDL_pVal") - | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_HDL.rsIDpVal.txt.gz
join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.txt | awk '{ print $1 }' | sort) <(zcat /data/external_public/GlobalLipid/jointGwasMc_TG.txt.gz | awk '{ print $3, "\t", $9 }' | sort) | cat <(echo "rsID TG_pVal") - | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_TG.rsIDpVal.txt.gz
join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.txt | awk '{ print $1 }' | sort) <(zcat /data/external_public/GlobalLipid/jointGwasMc_TC.txt.gz | awk '{ print $3, "\t", $9 }' | sort) | cat <(echo "rsID TC_pVal") - | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_TC.rsIDpVal.txt.gz

join -a 1 -a 2 -1 1 -2 1 -e NA -o 0 1.2 2.2 <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_LDL.rsIDpVal.txt.gz | sort) <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_HDL.rsIDpVal.txt.gz | sort) | perl -lane 'print $F[0], "\t", join("_", @F[1..$#F]);' | \
join -a 1 -a 2 -1 1 -2 1 -e NA -o 0 1.2 2.2 - <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_TG.rsIDpVal.txt.gz | sort) | perl -lane 'print $F[0], "\t", join("_", @F[1..$#F]);' | \
join -a 1 -a 2 -1 1 -2 1 -e NA -o 0 1.2 2.2 - <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_TC.rsIDpVal.txt.gz | sort) | perl -lane 'print $F[0], "\t", join("\t", split("_", $F[1])), "\t", $F[2];' | gzip > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.rsIDpVal.txt.gz

#20160618 CHECK_0 -- Prob: Check that all SNPs are actually being captured across the first four 'join' statements, eg across all four phenos. Think maybe one SNP is being lost somehow? count is 83 in the final file being used below and not 84 as expected?

~~~
[  mturchin20@spudling07  ~/Data/iGENBP2015]$join -a 1 -a 2 -1 1 -2 1 -e NA -o 0 1.2 2.2 <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_LDL.rsIDpVal.txt.gz | sort) <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_HDL.rsIDpVal.txt.gz | sort) | perl -lane 'print $F[0], "\t", join("_", @F[1..$#F]);' | join -a 1 -a 2 -1 1 -2 1 -e NA -o 0 1.2 2.2 - <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_TG.rsIDpVal.txt.gz | sort) | perl -lane 'print $F[0], "\t", join("_", @F[1..$#F]);' | join -a 1 -a 2 -1 1 -2 1 -e NA -o 0 1.2 2.2 - <(zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_TC.rsIDpVal.txt.gz | sort) | perl -lane 'print $F[0], "\t", join("\t", split("_", $F[1])), "\t", $F[2];' | head -n 10
rs10054063      0.6767  0.00181 5.273e-07       0.3111
rs1035744       0.1581  0.1548  1.448e-07       0.02108
rs10408163      0.2384  5.106e-07       6.759e-07       0.3324
rs10460587      0.2448  4.638e-07       0.06647 0.3685
rs1062219       0.0762  0.1831  1.686e-09       0.6772
rs10733608      0.0221  7.396e-06       0.6587  3.75e-05
rs10757056      2.139e-06       0.00556 0.8861  5.361e-08
rs10832027      0.04054 7.015e-07       1.256e-06       0.05323
rs10861661      0.9286  5.046e-07       2.598e-07       0.6563
rs10947207      2.448e-12       1.892e-05       0.9551  3.322e-15
[  mturchin20@spudling07  ~/Data/iGENBP2015]$zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.rsIDpVal.txt.gz | grep -v HDL | perl -lane 'print join("\t", @F[0..3]);' | R -q -e "Data1 <- read.table(file('stdin'), header=F); print(matrix(apply(Data1, 1, min, na.rm=TRUE), ncol=1));" | awk '{ print $2; }' | sed 's/"//g' | grep e- | head -n 10
1.810e-03
1.448e-07
2.384e-01
2.448e-01
1.686e-09
2.210e-02
2.139e-06
1.256e-06
2.598e-07
1.892e-05
~~~

zcat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.rsIDpVal.txt.gz | grep -v HDL | perl -lane 'print join("\t", @F[0..3]);' | R -q -e "Data1 <- read.table(file('stdin'), header=F); print(matrix(apply(Data1, 1, min, na.rm=TRUE), ncol=1));" | awk '{ print $2; }' | sed 's/"//g' | grep e- > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.rsIDpVal.MinpValPerSNP.txt

~~~
[  mturchin20@spudling07  ~/Data/iGENBP2015]$zcat /data/external_public/GlobalLipid/jointGwasMc_*.txt.gz | awk '{ if ($9 <= .00000005) { print $0 } } ' | awk '{ print $3, "\t", $2 }' | sed 's/:/ /g' | sed 's/chr//g' | sort | uniq | awk '{ print $0 } ' | head -n 10
.        6 30996132
.        6 30997334
.        6 31085770
.        6 31086048
.        6 31087354
.        6 31089631
.        6 31091070
.        6 31091862
.        6 31100869
.        6 31353689
~~~

zcat /data/external_public/GlobalLipid/jointGwasMc_*.txt.gz | awk '{ if ($9 <= .00000005) { print $0 } } ' | awk '{ print $3, "\t", $2 }' | sed 's/:/ /g' | sed 's/chr//g' | sort | uniq | awk '{ print $0 } ' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBP.txt
zcat /data/external_public/GlobalLipid/jointGwasMc_*.txt.gz | awk '{ if ($9 <= .0000005) { print $0 } } ' | awk '{ print $3, "\t", $2 }' | sed 's/:/ /g' | sed 's/chr//g' | sort | uniq | awk '{ print $0 } ' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e7Cutoff.SNPChrBP.txt
zcat /data/external_public/GlobalLipid/jointGwasMc_*.txt.gz | awk '{ if ($9 <= .000005) { print $0 } } ' | awk '{ print $3, "\t", $2 }' | sed 's/:/ /g' | sed 's/chr//g' | sort | uniq | awk '{ print $0 } ' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e6Cutoff.SNPChrBP.txt
zcat /data/external_public/GlobalLipid/jointGwasMc_*.txt.gz | awk '{ if ($9 <= .00005) { print $0 } } ' | awk '{ print $3, "\t", $2 }' | sed 's/:/ /g' | sed 's/chr//g' | sort | uniq | awk '{ print $0 } ' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e5Cutoff.SNPChrBP.txt
zcat /data/external_public/GlobalLipid/jointGwasMc_*.txt.gz | awk '{ if ($9 <= .0005) { print $0 } } ' | awk '{ print $3, "\t", $2 }' | sed 's/:/ /g' | sed 's/chr//g' | sort | uniq | awk '{ print $0 } ' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e4Cutoff.SNPChrBP.txt
zcat /data/external_public/GlobalLipid/jointGwasMc_*.txt.gz | awk '{ if ($9 <= .00000005) { print $0 } } ' | awk '{ print $3, "\t", $2, "\t", $9 }' | sed 's/:/ /g' | sed 's/chr//g' | sort | uniq | awk '{ print $0 } ' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBPpVal.txt
zcat /data/external_public/GlobalLipid/jointGwasMc_*.txt.gz | awk '{ if ($9 <= .0000005) { print $0 } } ' | awk '{ print $3, "\t", $2, "\t", $9 }' | sed 's/:/ /g' | sed 's/chr//g' | sort | uniq | awk '{ print $0 } ' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e7Cutoff.SNPChrBPpVal.txt
zcat /data/external_public/GlobalLipid/jointGwasMc_*.txt.gz | awk '{ if ($9 <= .000005) { print $0 } } ' | awk '{ print $3, "\t", $2, "\t", $9 }' | sed 's/:/ /g' | sed 's/chr//g' | sort | uniq | awk '{ print $0 } ' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e6Cutoff.SNPChrBPpVal.txt
zcat /data/external_public/GlobalLipid/jointGwasMc_*.txt.gz | awk '{ if ($9 <= .00005) { print $0 } } ' | awk '{ print $3, "\t", $2, "\t", $9 }' | sed 's/:/ /g' | sed 's/chr//g' | sort | uniq | awk '{ print $0 } ' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e5Cutoff.SNPChrBPpVal.txt
zcat /data/external_public/GlobalLipid/jointGwasMc_*.txt.gz | awk '{ if ($9 <= .0005) { print $0 } } ' | awk '{ print $3, "\t", $2, "\t", $9 }' | sed 's/:/ /g' | sed 's/chr//g' | sort | uniq | awk '{ print $0 } ' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e4Cutoff.SNPChrBPpVal.txt

~~~
[  mturchin20@spudling07  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBP.txt | wc
   8607   25821  204802
[  mturchin20@spudling07  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e7Cutoff.SNPChrBP.txt | wc
  10471   31413  249193
[  mturchin20@spudling07  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e6Cutoff.SNPChrBP.txt | wc
  13155   39465  313088
[  mturchin20@spudling07  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e5Cutoff.SNPChrBP.txt | wc
  18009   54027  428813
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e4Cutoff.SNPChrBP.txt | wc
  29101   87303  693407
~~~

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.txt | awk '{ print $1, "\t", $2, "\t" , $3 }' | grep -v snp > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.SNPChrBP.txt

#python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBP.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.SNPChrBP.txt 
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.SNPChrBP.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBP.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBP.wbmassHits1Mb.txt
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.SNPChrBP.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e7Cutoff.SNPChrBP.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e7Cutoff.SNPChrBP.wbmassHits1Mb.txt
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.SNPChrBP.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e6Cutoff.SNPChrBP.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e6Cutoff.SNPChrBP.wbmassHits1Mb.txt
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.SNPChrBP.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e5Cutoff.SNPChrBP.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e5Cutoff.SNPChrBP.wbmassHits1Mb.txt
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.SNPChrBP.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e4Cutoff.SNPChrBP.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e4Cutoff.SNPChrBP.wbmassHits1Mb.txt

~~~
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBP.wbmassHits1Mb.txt | awk '{ print $5 }' | sort | uniq | wc
     22      21     213
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e7Cutoff.SNPChrBP.wbmassHits1Mb.txt | awk '{ print $5 }' | sort | uniq | wc
     54      53     537
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBP.wbmassHits1Mb.txt | awk '{ print $5 }' | sort | uniq | wc
     22      21     213
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e6Cutoff.SNPChrBP.wbmassHits1Mb.txt | awk '{ print $5 }' | sort | uniq | wc
     76      75     763
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e5Cutoff.SNPChrBP.wbmassHits1Mb.txt | awk '{ print $5 }' | sort | uniq | wc
     82      81     825
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e4Cutoff.SNPChrBP.wbmassHits1Mb.txt | awk '{ print $5 }' | sort | uniq | wc                                                                                                                                     84      83     846                     
~~~

#20160618 CHECK_0 -- Prob: Make version of Proxy script that allows multiple SNPs to exist/represent a region? Call it /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs2.py
#20160618 CHECK_0 -- Prob: Why do some 'new hits' have p-values <= 5e-8 from the univariate GWAS? B/c within region with a previous hit? Too close? 

cp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs2.py 

#Replaced 5e8 to 5e7, 5e6, 5e5 and 5e4
for i in {1..22}; do
#for i in {X..X}; do
	cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e4Cutoff.SNPChrBPpVal.txt | awk '{ if ($2 == '$i') { print $0 } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e4Cutoff.SNPChrBPpVal.Chr${i}.txt
	cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e4Cutoff.SNPChrBPpVal.Chr${i}.txt | \
	R -q -e "Data1 <- read.table(file('stdin'), header=F); Data1 <- Data1[order(Data1[,4], decreasing=FALSE),]; Data2 <- c(); for (i in 1:nrow(Data1)) { Data2 <- rbind(Data2, Data1[1,]); Data1 <- Data1[Data1[,3] < Data1[1,3] - 500000 | Data1[,3] > Data1[1,3] + 500000,]; }; Data2;" | grep -v NA | grep -v V | grep -v \> | perl -lane 'print join("\t", @F[1..$#F]);' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e4Cutoff.SNPChrBPpVal.Chr${i}.1MbWindowPruned.txt
done


rm /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e6Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt
rm /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e6Cutoff.SNPChrBPpVal.ChrAll.txt
for i in {1..22}; do
#for i in {X..X}; do
	cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e4Cutoff.SNPChrBPpVal.Chr${i}.txt >> /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.txt
	cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e4Cutoff.SNPChrBPpVal.Chr${i}.1MbWindowPruned.txt	>> /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt
done

#20160618 CHECK_0 -- Prob: Go back and check/include X in the 5e7s and later....was not present in 5e8 for sure

#cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBPpVal.Chr*.1MbWindowPruned.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt
#cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBPpVal.Chr*.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.txt

#R -q -e "Data1 <- read.table(file('stdin'), header=F); Data1 <- Data1[order(Data1[,4], decreasing=FALSE),]; Data2 <- c(); for (i in 1:nrow(Data1)) { Data2 <- rbind(Data2, Data1[i,]); Data1 <- Data1[Data1[,3] < Data1[i,3] - 500000 | Data1[,3] > Data1[i,3] + 500000,]; }; Data2;" | grep -v NA | grep -v V | grep -v \> | perl -lane 'print join("\t", @F[1..$#F]);' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBPpVal.Chr${i}.1MbWindowPruned.txt
#R -q -e "Data1 <- read.table(file('stdin'), header=F); Data1 <- Data1[order(Data1[,4], decreasing=FALSE),]; for (i in 1:50) { Data1 <- rbind(Data1[1:i,], Data1[Data1[i+1:nrow(Data1),3] < Data1[i,3] - 500000,]); }; Data1;"
#R -q -e "Data1 <- read.table(file('stdin'), header=F); Data1 <- Data1[order(Data1[,4], decreasing=FALSE),]; Data2 <- c(); for (i in 1:50) { Data2 <- rbind(Data2, Data1[i,]); Data1 <- Data1[Data1[i+1:nrow(Data1),3] < Data1[i,3] - 500000 | Data1[i+1:nrow(Data1),3] > Data1[i,3] + 500000,]; }; Data2;"
#R -q -e "Data1 <- read.table(file('stdin'), header=F); Data1 <- Data1[order(Data1[,4], decreasing=FALSE),]; for (i in 1:50) { Data1 <- rbind(Data1[1:i,], Data1[Data1[1:i | i+1:nrow(Data1),3] < Data1[i,3] - 500000 | Data1[i+1:nrow(Data1),3] > Data1[i,3] + 500000,]); }; Data1;"
#R -q -e "Data1 <- read.table(file('stdin'), header=F); Data1 <- Data1[order(Data1[,4], decreasing=FALSE),]; for (i in 1:nrow(Data1)) { Data1 <- rbind(Data1[i,], Data1[Data1[i+1:nrow(Data1),3] < Data1[i,3] - 500000 | Data1[i+1:nrow(Data1),3] > Data1[i,3] + 500000,]); }; Data1;";
#R -q -e "Data1 <- read.table(file('stdin'), header=F); Data1 <- Data1[order(Data1[,4], decreasing=FALSE),]; Data2 <- c(); BPs <- Data1[,3]; for (i in 1:nrow(Data1)) { Data2 <- rbind(Data2, Data1[i,]); Data1 <- Data1[Data1[,3] < BPs[i] - 500000 | Data1[,3] > BPs[i] + 500000,]; print(BPs[i]);}; Data2; " | grep -v NA | grep -v V | grep -v \> | perl -lane 'print join("\t", @F[1..$#F]);' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBPpVal.Chr${i}.1MbWindowPruned.txt

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBPpVal.Chr1.1MbWindowPruned.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBPpVal.Chr1.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBPpVal.Chr1.MatchedTo1MbWindowPruned.txt 
#python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.MatchedTo1MbWindowPruned.txt

rm /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.MatchedTo1MbWindowPruned.NoMatch.txt
for i in {1..22}; do
	echo $i >> /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.MatchedTo1MbWindowPruned.NoMatch.txt;
	python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e4Cutoff.SNPChrBPpVal.Chr${i}.1MbWindowPruned.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e4Cutoff.SNPChrBPpVal.Chr${i}.txt | perl -lane 'if (!$F[5]) { print join("\t", @F); }' | wc >> /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.MatchedTo1MbWindowPruned.NoMatch.txt
done

~~~
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBPpVal.Chr1.1MbWindowPruned.txt
rs646776        1       109818530       1.630e-272
rs11591147      1       55505647        8.580e-143
rs4587594       1       63133930        3.503e-82
rs4846914       1       230295691       3.513e-41
rs558971        1       234853406       7.025e-28
rs1689797       1       182150978       2.852e-21
rs2642438       1       220970028       1.283e-18
rs4660293       1       40028180        2.863e-18
rs10903129      1       25768937        3.030e-17
rs12748152      1       27138393        9.737e-16
rs6603981       1       92993807        7.846e-15
rs12133576      1       93816400        6.150e-11
rs333947        1       110470764       3.166e-09
rs267733        1       150958836       5.285e-09
rs1077514       1       23766233        6.400e-09
rs4650994       1       178515312       6.696e-09
rs12066643      1       56240278        1.063e-08
rs12145743      1       156700651       1.803e-08
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBPpVal.Chr1.txt | wc                                                                                                                                                                           1725    6900   61186
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBPpVal.Chr1.1MbWindowPruned.txt | wc
     18      72     571
[  mturchin20@spudling96  ~/Data/iGENBP2015]$python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBPpVal.Chr1.1MbWindowPruned.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBPpVal.Chr1.txt | perl -lane 'if ($F[5]) { print join("\t", @F); }' | wc
   1725   17250  115580
[  mturchin20@spudling96  ~/Data/iGENBP2015]$python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBPpVal.Chr1.1MbWindowPruned.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBPpVal.Chr1.txt | perl -lane 'if (!$F[5]) { print join("\t", @F); }' | wc 
      0       0       0
[  mturchin20@spudhead  ~/Software]$for i in {1..22}; do cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBPpVal.Chr${i}.txt | wc; done
   1725    6900   61186
   2134    8536   75022
    134     536    4714
     93     372    3242
    978    3912   34462
   1253    5012   44004
    259    1036    9092
   1661    6644   58647
    314    1256   10956
    143     572    5222
   1809    7236   66154
    380    1520   14072
     10      40     359
      6      24     217
    376    1504   13559
    739    2956   26727
    263    1052    9486
    228     912    8227
   1135    4540   40905
    341    1364   12255
      0       0       0
     32     128    1151
[  mturchin20@spudhead  ~/Software]$R -q -e "1725 + 2134 + 134 + 93 + 978 + 1253 + 259 + 1661 + 314 + 143 + 1809 + 380 + 10 + 6 + 376 + 739 + 263 + 228 + 1135 + 341 + 32"
> 1725 + 2134 + 134 + 93 + 978 + 1253 + 259 + 1661 + 314 + 143 + 1809 + 380 + 10 + 6 + 376 + 739 + 263 + 228 + 1135 + 341 + 32
[1] 14013
> 
> 
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt | wc
    194     776    6184
#[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.txt | wc
#  15932   74078  621423
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBPpVal.txt | wc
  14013   56052  499659
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.txt | wc
  14013   56052  499659
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt | wc
    194     776    6184
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGw
asMc_AllPhenos.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.MatchedTo1MbWindowPruned.NoMatch.txt
1
      0       0       0
2
      0       0       0
3
      0       0       0
4
      0       0       0
5
      0       0       0
6
      0       0       0
7
      0       0       0
8
      0       0       0
9
      0       0       0
10
      0       0       0
11
      0       0       0
12
      0       0       0
13
      0       0       0
14
      0       0       0
15
      0       0       0
16
      0       0       0
17
      0       0       0
18
      0       0       0
19
      0       0       0
20
      0       0       0
21
      0       0       0
22
      0       0       0
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e7Cutoff.SNPChrBPpVal.ChrAll.txt | wc
  17087   68348  609472
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e7Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt | wc
    241     964    7687
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e7Cutoff.SNPChrBPpVal.txt | wc
  17087   68348  609472
~~~

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.SNPChrBP.txt /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.MarkerChrBP.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.SNPChrBP.AndOrigUniHits.txt

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.SNPChrBP.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassHits1Mb.txt
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.SNPChrBP.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e7Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e7Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassHits1Mb.txt
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.SNPChrBP.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e6Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e6Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassHits1Mb.txt
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.SNPChrBP.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e5Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e5Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassHits1Mb.txt
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.SNPChrBP.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassHits1Mb.txt

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.MarkerChrBP.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHits1Mb.txt 
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.MarkerChrBP.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e7Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e7Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHits1Mb.txt 
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.MarkerChrBP.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e6Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e6Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHits1Mb.txt 
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.MarkerChrBP.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e5Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e5Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHits1Mb.txt 
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.MarkerChrBP.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHits1Mb.txt 

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.SNPChrBP.AndOrigUniHits.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHits1Mb.txt 
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.SNPChrBP.AndOrigUniHits.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e7Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e7Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHits1Mb.txt 
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.SNPChrBP.AndOrigUniHits.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e6Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e6Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHits1Mb.txt 
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.SNPChrBP.AndOrigUniHits.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e5Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e5Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHits1Mb.txt 
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.SNPChrBP.AndOrigUniHits.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHits1Mb.txt 

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e7Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e6Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e5Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e7Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e6Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e5Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e7Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e6Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e5Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc

#20160618 20160619 CHECK_1 -- Prob: Check basepair positions of original hits from file /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.MarkerChrBP.txt since getting error that a SNP has >1Mb difference in BP between that file and the multivariate results file -- may be due to using hg18 vs hg19 mapping coordinates that the original source files (but maybe not what the NG article from which I got the SNPs/coordinates from) has? example SNP -- rs4846914, see below Soln: yes makes a difference, went ahead and switched to the hg19 coords from the original source pheno files (though 2 SNPs are missing from those files too now??) 
#20160618 CHECK_0 -- Prob: Check that one ...llPhenos.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassHits1Mb.txt file issue with the low hit count, much lower than expected, the one prefaced by the ???
#20160619 CHECK_0 -- Prob: 2 orig hit SNPs missing from original source files? Figure out where they are going or coming from and get their hg19 coords too??
~~~
[  mturchin20@spudhead  ~/Software]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt | head -n 5
rs646776        1       109818530       1.630e-272
rs11591147      1       55505647        8.580e-143
rs4587594       1       63133930        3.503e-82
rs4846914       1       230295691       3.513e-41
rs558971        1       234853406       7.025e-28
[  mturchin20@spudhead  ~/Software]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.MarkerChrBP.txt | grep rs4846914
rs4846914        1       228362314
[  mturchin20@spudhead  ~/Software]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt | wc
    194     776    6184
[  mturchin20@spudhead  ~/Software]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e7Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt | wc
    241     964    7687
[  mturchin20@spudhead  ~/Software]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e6Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt | wc
    318    1272   10150
[  mturchin20@spudhead  ~/Software]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e5Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt | wc
    502    2008   15984
[  mturchin20@spudhead  ~/Software]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt | wc
   1190    4760   38071
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
     19      18     185
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e7Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
     53      52     526
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e6Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
     76      75     763
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e5Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
     82      81     825
???[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
     36      35     350
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
     40      39     386
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e7Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
     46      45     447
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e6Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
     48      47     466
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e5Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
     55      54     539
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
     71      70     695
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
     48      47     469
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e7Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
     63      62     619
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e6Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
     73      72     722
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e5Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
     82      81     815
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
    101     100    1002
~~~

#20160618 20160619 CHECK_1 -- Prob: Include univariate gwas hits in the list of markers to use in the 'compare to pruned p-val thresh' data? Soln: Did it -- still need to redo list of 'Orig hit' variants though to use the hg19 coords and not the ones from the paper? See the other CHECK_# referencing this above

~~~
[  mturchin20@spudling96  ~/Data/iGENBP2015]$join -a 1 -1 1 -2 3 -e NA -o 0 1.1 1.2 1.3 2.1 2.2 2.3 <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.MarkerChrBP.txt | sort) <(zcat /data/external_public/GlobalLipid/jointGwasMc_HDL.txt.gz | sort -k 3,3) | head -n 10
rs10019888 rs10019888 4 25672088 chr4:25672088 chr4:26062990 rs10019888
rs10102164 rs10102164 8 55584167 chr8:55584167 chr8:55421614 rs10102164
rs10128711 rs10128711 11 18589560 chr11:18589560 chr11:18632984 rs10128711
rs10401969 rs10401969 19 19268718 chr19:19268718 chr19:19407718 rs10401969
rs1047891 rs1047891 2 211248752 chr2:211248752 chr2:211540507 rs1047891
rs10490626 rs10490626 2 118552311 chr2:118552311 chr2:118835841 rs10490626
rs10761731 rs10761731 10 64697616 chr10:64697616 chr10:65027610 rs10761731
rs1077514 rs1077514 1 23638820 chr1:23638820 chr1:23766233 rs1077514
rs10904908 rs10904908 10 17300296 chr10:17300296 chr10:17260290 rs10904908
rs11065987 rs11065987 12 110556807 chr12:110556807 chr12:112072424 rs11065987
[  mturchin20@spudling96  ~/Data/iGENBP2015]$join -a 1 -1 1 -2 3 -e NA -o 0 1.1 1.2 1.3 2.1 2.2 2.3 <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.MarkerChrBP.txt | sort) <(zcat /data/external_public/GlobalLipid/jointGwasMc_HDL.txt.gz | sort -k 3,3) | wc   
    157    1099   11127
[  mturchin20@spudling96  ~/Data/iGENBP2015]$join -a 1 -1 1 -2 3 -e NA -o 0 1.1 1.2 1.3 2.1 2.2 2.3 <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.MarkerChrBP.txt | sort) <(zcat /data/external_public/GlobalLipid/jointGwasMc_HDL.txt.gz | sort -k 3,3) | grep NA | wc
      2      14      81
[  mturchin20@spudling96  ~/Data/iGENBP2015]$join -a 1 -1 1 -2 3 -e NA -o 0 1.1 1.2 1.3 2.1 2.2 2.3 <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.MarkerChrBP.txt | sort) <(zcat /data/external_public/GlobalLipid/jointGwasMc_LDL.txt.gz | sort -k 3,3) | grep NA | wc
      2      14      81
[  mturchin20@spudling96  ~/Data/iGENBP2015]$join -a 1 -1 1 -2 3 -e NA -o 0 1.1 1.2 1.3 2.1 2.2 2.3 <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.MarkerChrBP.txt | sort) <(zcat /data/external_public/GlobalLipid/jointGwasMc_TG.txt.gz | sort -k 3,3) | grep NA | wc
      2      14      81
[  mturchin20@spudling96  ~/Data/iGENBP2015]$join -a 1 -1 1 -2 3 -e NA -o 0 1.1 1.2 1.3 2.1 2.2 2.3 <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.MarkerChrBP.txt | sort) <(zcat /data/external_public/GlobalLipid/jointGwasMc_TC.txt.gz | sort -k 3,3) | grep NA | wc
      2      14      81
[  mturchin20@spudling96  ~/Data/iGENBP2015]$join -a 1 -1 1 -2 3 -e NA -o 0 1.1 1.2 1.3 2.1 2.2 2.3 <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.MarkerChrBP.txt | sort) <(zcat /data/external_public/GlobalLipid/jointGwasMc_TC.txt.gz | sort -k 3,3) | grep NA     
rs181362 rs181362 22 20262068 NA NA NA
rs7941030 rs7941030 11 122027585 NA NA NA
[  mturchin20@spudling96  ~/Data/iGENBP2015]$join -a 1 -1 1 -2 3 -e NA -o 0 2.2 <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.MarkerChrBP.txt | sort) <(zcat /data/external_public/GlobalLipid/jointGwasMc_HDL.txt.gz | sort -k 3,3) | grep -v NA | sed 's/:/ /g' | sed 's/chr//g' | awk '{ print $0 }' | head -n 10
rs10019888 4 26062990
rs10102164 8 55421614
rs10128711 11 18632984
rs10401969 19 19407718
rs1047891 2 211540507
rs10490626 2 118835841
rs10761731 10 65027610
rs1077514 1 23766233
rs10904908 10 17260290
rs11065987 12 112072424
~~~

#20160619 NOTE -- So yes, the coords from the paper directly are in hg18 -- need to use the hg19 coords for this analysis so making a new file that has those coords instead of the original hg18 ones

join -a 1 -1 1 -2 3 -e NA -o 0 2.2 <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.MarkerChrBP.txt | sort) <(zcat /data/external_public/GlobalLipid/jointGwasMc_HDL.txt.gz | sort -k 3,3) | grep -v NA | sed 's/:/ /g' | sed 's/chr//g' | awk '{ print $0 }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.MatchedTohg19Coords.MarkerChrBP.txt

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.SNPChrBP.txt /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.MatchedTohg19Coords.MarkerChrBP.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.SNPChrBP.AndOrigUniHitshg19.txt 

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.MatchedTohg19Coords.MarkerChrBP.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHitshg191Mb.txt 
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.MatchedTohg19Coords.MarkerChrBP.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e7Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e7Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHitshg191Mb.txt 
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.MatchedTohg19Coords.MarkerChrBP.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e6Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e6Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHitshg191Mb.txt 
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.MatchedTohg19Coords.MarkerChrBP.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e5Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e5Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHitshg191Mb.txt 
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/ng.2797-S1.edited.vs2.carriageRemoved.rsIDs.HapMart.MatchedTohg19Coords.MarkerChrBP.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHitshg191Mb.txt 

python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.SNPChrBP.AndOrigUniHitshg19.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHitshg191Mb.txt 
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.SNPChrBP.AndOrigUniHitshg19.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e7Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e7Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHitshg191Mb.txt 
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.SNPChrBP.AndOrigUniHitshg19.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e6Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e6Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHitshg191Mb.txt 
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.SNPChrBP.AndOrigUniHitshg19.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e5Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e5Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHitshg191Mb.txt 
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.SNPChrBP.AndOrigUniHitshg19.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHitshg191Mb.txt 

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHitshg191Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e7Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHitshg191Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e6Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHitshg191Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e5Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHitshg191Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHitshg191Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHitshg191Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e7Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHitshg191Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e6Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHitshg191Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e5Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHitshg191Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHitshg191Mb.txt | awk '{ print $6 }' | sort | uniq | wc

~~~
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHitshg191Mb.txt | awk '{ print $6 }' | sort | uniq | wc
    152     151    1519
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e7Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHitshg191Mb.txt | awk '{ print $6 }' | sort | uniq | wc
    154     153    1540
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e6Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHitshg191Mb.txt | awk '{ print $6 }' | sort | uniq | wc
    154     153    1540
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e5Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHitshg191Mb.txt | awk '{ print $6 }' | sort | uniq | wc
    154     153    1540
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHitshg191Mb.txt | awk '{ print $6 }' | sort | uniq | wc
    154     153    1540
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHitshg191Mb.txt | awk '{ print $6 }' | sort | uniq | wc
    170     169    1703
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e7Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHitshg191Mb.txt | awk '{ print $6 }' | sort | uniq | wc
    206     205    2065
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e6Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHitshg191Mb.txt | awk '{ print $6 }' | sort | uniq | wc
    229     228    2302
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e5Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHitshg191Mb.txt | awk '{ print $6 }' | sort | uniq | wc
    235     234    2364
~~~



#20160830
#Work prior to ProbGenom but likely applicable to 































##GIANT2010
##20160313

mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/Analyses
mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/Analyses/PathwayWork

join <(cat /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt | sort ) <(zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wHapMap22.vs3.txt.gz | sort) | awk '{ print $2, "\t", $3, "\t", $3, "\t", $4, "\t", $5 }' | sed 's/a/A/g' | sed 's/g/G/' | sed 's/c/C/g' | sed 's/t/T/g' > /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.AnnovarFormat

join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/GIANT2010.newtophits.vs1.SignCrrct.vs1.txt | awk '{ print $1, "\t", $2, "\t", $3 }' | sort) <(zcat /mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wHapMap22.vs3.txt.gz | sort) | awk '{ print $2, "\t", $3, "\t", $3, "\t", $4, "\t", $5 }' | sed 's/a/A/g' | sed 's/g/G/' | sed 's/c/C/g' | sed 's/t/T/g' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/Analyses/PathwayWork/GIANT2010.newtophits.vs1.SignCrrct.vs1.AnnovarFormat

perl /mnt/lustre/home/mturchin20/Software/annovar/summarize_annovar.pl /mnt/lustre/home/mturchin20/Data/GIANT/2010/GIANT2010.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.AnnovarFormat /mnt/lustre/home/mturchin20/Software/annovar/humandb/ --ver1000g 1000g2012apr --verdbsnp 129 --veresp 6500 --buildver hg19 --remove

perl /mnt/lustre/home/mturchin20/Software/annovar/summarize_annovar.pl /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2010/Vs1/Analyses/PathwayWork/GIANT2010.newtophits.vs1.SignCrrct.vs1.AnnovarFormat /mnt/lustre/home/mturchin20/Software/annovar/humandb/ --ver1000g 1000g2012apr --verdbsnp 129 --veresp 6500 --buildver hg19 --remove

/mnt/gluster/data/external_public_supp/GIANT2010/GIANT_HEIGHT_LangoAllen2010_publicrelease_HapMapCeuFreq.wHapMap22.vs3.txt.gz




##GIANT2014_5
##20160313

mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses
mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/PathwayWork

join <(cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt | sort ) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.IncAllele.txt.gz | sort) | awk '{ print $2, "\t", $3, "\t", $3, "\t", $4, "\t", $5 }' > /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.AnnovarFormat

join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5.Orig3.newtophits.vs1.SignCrrct.vs1.txt | awk '{ print $1, "\t", $2, "\t", $3 }' | sort) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.IncAllele.txt.gz | sort) | awk '{ print $2, "\t", $3, "\t", $3, "\t", $4, "\t", $5 }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/PathwayWork/GIANT2014_5.Orig3.newtophits.vs1.SignCrrct.vs1.AnnovarFormat

perl /mnt/lustre/home/mturchin20/Software/annovar/summarize_annovar.pl /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.AnnovarFormat /mnt/lustre/home/mturchin20/Software/annovar/humandb/ --ver1000g 1000g2012apr --verdbsnp 129 --veresp 6500 --buildver hg19 --remove

perl /mnt/lustre/home/mturchin20/Software/annovar/summarize_annovar.pl /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/PathwayWork/GIANT2014_5.Orig3.newtophits.vs1.SignCrrct.vs1.AnnovarFormat /mnt/lustre/home/mturchin20/Software/annovar/humandb/ --ver1000g 1000g2012apr --verdbsnp 129 --veresp 6500 --buildver hg19 --remove

cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.AnnovarFormat.genome_summary.csv | grep -v intergenic | grep -v ncRNA | awk -F, '{ print $2 }' | sed 's/"//g' | grep -v Gene > /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.AnnovarFormat.genome_summary.GeneIDs.no_Intergenic_ncRNA

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/PathwayWork/GIANT2014_5.Orig3.newtophits.vs1.SignCrrct.vs1.AnnovarFormat.genome_summary.csv | grep -v intergenic | grep -v ncRNA | awk -F, '{ print $2 }' | sed 's/"//g' | grep -v Gene > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/PathwayWork/GIANT2014_5.Orig3.newtophits.vs1.SignCrrct.vs1.AnnovarFormat.genome_summary.GeneIDs.no_Intergenic_ncRNA



cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/PathwayWork/HaemgenRBC2012.Table1.Edited.MarkerChrBP.AnnovarFormat.genome_summary.csv | grep -v intergenic | grep -v ncRNA | awk -F, '{ print $2 }' | sed 's/"//g' | grep -v Gene > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/PathwayWork/HaemgenRBC2012.Table1.Edited.MarkerChrBP.AnnovarFormat.genome_summary.GeneIDs.no_Intergenic_ncRNA







#20160619
#Followup analyses based on comments from NHS presentation -- looking further down into univariate results (how much less signif are the new hits from the original data), uniBF vs other BFs, heatmap of orig, new, miss Zscore hits
#Also in prep for committee meeting
#Continuing to use the 'ForCSHL' versions of scripts to keep track of code and trying things out

mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/

#zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.IncAllele.txt.gz /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.txt.gz /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.MatchedToHeight.IncAllele.gz | awk '{ if ($7 <= .00000005) { print $0 } } ' | awk '{ print $1, "\t", $9, "\t", $7 }' | sed 's/_/ /g' | sort | uniq | awk '{ print $0 } ' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e8Cutoff.SNPChrBPpVal.txt 
zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.IncAllele.txt.gz /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.txt.gz /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.MatchedToHeight.IncAllele.gz | perl -lane ' if (abs($F[6]) <= .00000005) { print join("\t", @F); } ' | awk '{ print $1, "\t", $9, "\t", $7 }' | sed 's/_/ /g' | sort | uniq | awk '{ print $0 } ' | grep -v ChrBP | grep -v , > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e8Cutoff.SNPChrBPpVal.txt 
zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.IncAllele.txt.gz /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.txt.gz /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.MatchedToHeight.IncAllele.gz | perl -lane ' if (abs($F[6]) <= .0000005) { print join("\t", @F); } ' | awk '{ print $1, "\t", $9, "\t", $7 }' | sed 's/_/ /g' | sort | uniq | awk '{ print $0 } ' | grep -v ChrBP | grep -v , > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e7Cutoff.SNPChrBPpVal.txt 
zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.IncAllele.txt.gz /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.txt.gz /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.MatchedToHeight.IncAllele.gz | perl -lane ' if (abs($F[6]) <= .000005) { print join("\t", @F); } ' | awk '{ print $1, "\t", $9, "\t", $7 }' | sed 's/_/ /g' | sort | uniq | awk '{ print $0 } ' | grep -v ChrBP | grep -v , > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e6Cutoff.SNPChrBPpVal.txt 
zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.IncAllele.txt.gz /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.txt.gz /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.MatchedToHeight.IncAllele.gz | perl -lane ' if (abs($F[6]) <= .00005) { print join("\t", @F); } ' | awk '{ print $1, "\t", $9, "\t", $7 }' | sed 's/_/ /g' | sort | uniq | awk '{ print $0 } ' | grep -v ChrBP | grep -v , > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e5Cutoff.SNPChrBPpVal.txt 
zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.IncAllele.txt.gz /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.txt.gz /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.MatchedToHeight.IncAllele.gz | perl -lane ' if (abs($F[6]) <= .0005) { print join("\t", @F); } ' | awk '{ print $1, "\t", $9, "\t", $7 }' | sed 's/_/ /g' | sort | uniq | awk '{ print $0 } ' | grep -v ChrBP | grep -v , > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.txt 

#20160619 NOTE -- dropping markers that have multiple entries for ChrBP from dbSNP130, hence the grep -v , call; this might be a bit anticonservative but prefering it atm to 'arbitrarily' choosing say the first entry of any/all possible entries for coordinates. The first set of wordcounts below are the results when not dropping multiple entries/commas from the files

~~~
#[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e8Cutoff.SNPChrBPpVal.txt | wc
#  29176  126282 1041630
#[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e7Cutoff.SNPChrBPpVal.txt | wc
#  38064  164580 1358723
#[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e6Cutoff.SNPChrBPpVal.txt | wc
#  51990  223189 1848049
#[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e5Cutoff.SNPChrBPpVal.txt | wc
#  75803  321900 2677454
#[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.txt | wc
# 125202  524314 4396088
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e8Cutoff.SNPChrBPpVal.txt | wc
  27487  109921  926119
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e7Cutoff.SNPChrBPpVal.txt | wc
  35897  143556 1210192
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e6Cutoff.SNPChrBPpVal.txt | wc
  49314  197214 1664378
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e5Cutoff.SNPChrBPpVal.txt | wc
  72524  290028 2452006
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.txt | wc
 121071  484179 4111714
~~~

#Replaced 5e8 to 5e7, 5e6, 5e5 and 5e4
#for i in {1..22}; do
#for i in {X..X}; do
for i in `echo {1..22} "X"`; do
	cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.txt | awk '{ if ($2 == '$i') { print $0 } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.Chr${i}.txt
	cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.Chr${i}.txt | \
	R -q -e "Data1 <- read.table(file('stdin'), header=F); Data1 <- Data1[order(Data1[,4], decreasing=FALSE),]; Data2 <- c(); for (i in 1:nrow(Data1)) { Data2 <- rbind(Data2, Data1[1,]); Data1 <- Data1[Data1[,3] < Data1[1,3] - 500000 | Data1[,3] > Data1[1,3] + 500000,]; }; Data2;" | grep -v NA | grep -v V | grep -v \> | perl -lane 'print join("\t", @F[1..$#F]);' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.Chr${i}.1MbWindowPruned.txt
done

rm /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e6Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt
rm /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e6Cutoff.SNPChrBPpVal.ChrAll.txt
#for i in {1..22}; do
#for i in {X..X}; do
for i in `echo {1..22} "X"`; do
	cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.Chr${i}.txt >> /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.txt
	cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.Chr${i}.1MbWindowPruned.txt	>> /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt
done

#20160619 NOTE -- the below comment is from when pasting over GlobalLipids2013 code. During the GIANT2014_5 run I realized I coudl so the echo {1..2} "X" thing to get both 1..2 and X, so will be incorporating this in future efforts of similar problem. As of this comment still need to go back and do GlobalLipids2013 X stuff too, so the CHECK_0 is not complete #20160618 CHECK_0 -- Prob: Go back and check/include X in the 5e7s and later....was not present in 5e8 for sure

rm /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.MatchedTo1MbWindowPruned.NoMatch.txt
for i in `echo {1..22} "X"`; do
	echo $i >> /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.MatchedTo1MbWindowPruned.NoMatch.txt;
	python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.Chr${i}.1MbWindowPruned.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.Chr${i}.txt | perl -lane 'if (!$F[5]) { print join("\t", @F); }' | wc >> /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.MatchedTo1MbWindowPruned.NoMatch.txt
done
	
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.Chr1.1MbWindowPruned.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.Chr1.txt | perl -lane 'if ($F[5]) { print join("\t", @F); }' | wc 
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.Chr1.1MbWindowPruned.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.Chr1.txt | perl -lane 'if (!$F[5]) { print join("\t", @F); }' | wc 

~~~
[  mturchin20@spudling07  ~]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e8Cutoff.SNPChrBPpVal.Chr1.1MbWindowPruned.txt | wc
     63     252    1997
[  mturchin20@spudling07  ~]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e8Cutoff.SNPChrBPpVal.Chr1.txt | wc
   2835   11340   95856
[  mturchin20@spudling07  ~]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.txt | wc
  27460  109840  925432
[  mturchin20@spudling07  ~]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt | wc
    659    2636   20925
[  mturchin20@spudling07  ~]$for i in `echo {1..22} "X"`; do cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e8Cutoff.SNPChrBPpVal.Chr${i}.1MbWindowPruned.txt | wc; done
     63     252    1997
     55     220    1746
     42     168    1333
     31     124     980
     40     160    1252
     54     216    1706
     29     116     903
     26     104     815
     34     136    1080
     23      92     755
     37     148    1197
     32     128    1039
     16      64     519
     22      88     710
     30     120     966
     25     100     797
     37     148    1182
     10      40     318
     23      92     731
     20      80     599
      5      20     151
      5      20     149
      0       0       0
[  mturchin20@spudling07  ~]$R -q -e "63 + 55 + 42 + 31 + 40 + 54 + 29 + 26 + 34 + 23 + 37 + 32 + 16 + 22 + 30 + 25 + 37 + 10 + 23 + 20 + 5 + 5"
> 63 + 55 + 42 + 31 + 40 + 54 + 29 + 26 + 34 + 23 + 37 + 32 + 16 + 22 + 30 + 25 + 37 + 10 + 23 + 20 + 5 + 5
[1] 659
> 
> 
[  mturchin20@spudling07  ~]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.Chr1.txt | wc
  10840   43360  367751
[  mturchin20@spudling07  ~]$python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.Chr1.1MbWindowPruned.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.Chr1.txt | perl -lane 'if ($F[5]) { print join("\t", @F); }' | wc
  10840  108400  714567
[  mturchin20@spudling07  ~]$python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.Chr1.1MbWindowPruned.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.Chr1.txt | perl -lane 'if (!$F[5]) { print join("\t", @F); }' | wc
      0       0       0
~~~

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5.Orig3.newtophits.vs1.SignCrrct.vs1.txt | awk '{ print $1, "\t", $2, "\t", $3 }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5.Orig3.newtophits.vs1.SignCrrct.vs1.MarkerChrBP.txt

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5.Orig3.newtophits.vs1.SignCrrct.vs1.MarkerChrBP.txt /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5.Orig3.newtophits.vs1.SignCrrct.vs1.MarkerChrBP.AndOrigUniHits.txt

~~~
[  mturchin20@spudhead  ~/Software]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5.Orig3.newtophits.vs1.SignCrrct.vs1.txt | awk '{ print $1, "\t", $2, "\t", $3 }' | wc
    218     654    5624
[  mturchin20@spudhead  ~/Software]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5.Orig3.newtophits.vs1.SignCrrct.vs1.txt | awk '{ print $1, "\t", $2, "\t", $3 }' | head -n 10snp      chr     pos
rs6719451        2       238001540
rs7651617        3       169321372
rs2741856        17      39182364
rs12899850       15      63838352
rs615632         8       9833730
rs2347709        7       134054280
rs10737541       1       166480721
rs9424148        10      12186863
rs11135932       8       26182377
~~~

for i in `echo {1..22} "X"`; do
	cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5.Orig3.newtophits.vs1.SignCrrct.vs1.MarkerChrBP.txt | awk '{ if ($2 == '$i') { print $0 } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.Orig3.newtophits.vs1.SignCrrct.vs1.Chr${i}.MarkerChrBP.txt
	cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt | awk '{ if ($2 == '$i') { print $0 } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.Chr${i}.MarkerChrBP.txt
	cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.Orig3.newtophits.vs1.SignCrrct.vs1.Chr${i}.MarkerChrBP.txt /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.Chr${i}.MarkerChrBP.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.Orig3.newtophits.vs1.SignCrrct.vs1.Chr${i}.MarkerChrBP.AndOrigUniHits.txt	
done

#Replaced 5e8 to 5e7, 5e6, 5e5 and 5e4
for i in `echo {1..22} "X"`; do
	if [ $i == "1" ] ; then
		rm -f /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassHits1Mb.txt
		rm -f /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHits1Mb.txt
		rm -f /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHits1Mb.txt
	fi
	python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.Orig3.newtophits.vs1.SignCrrct.vs1.Chr${i}.MarkerChrBP.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt >> /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassHits1Mb.txt 
	python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.Chr${i}.MarkerChrBP.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt >> /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHits1Mb.txt
	python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.Orig3.newtophits.vs1.SignCrrct.vs1.Chr${i}.MarkerChrBP.AndOrigUniHits.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt >> /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHits1Mb.txt 
done

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e7Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e6Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e5Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e7Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e6Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e5Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e7Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e6Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e5Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc

#20160620 CHECK_0 -- Prob: Think through this problem of doing the 1Mb window pruning leading to an upper-bound of regions available that is below the total number of univariate GWAS hits, eg getting like ~610 regions pruned at 5e8 threshold but there are ~820 GWAS hits, so clearly not going to get all of them represented at that threshold/area. And we never seem to reach ~820 even after raising threshold -- likely due to multiple hits being within 1Mb of one another and LD not really matching up with the simplifying 1Mb physical distance assumption
~~~
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
     45      44     456
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e7Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
    138     137    1403
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e6Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
    203     202    2060
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e5Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
    216     215    2187
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
    216     215    2187
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5.Orig3.newtophits.vs1.SignCrrct.vs1.txt | wc
    218    1744   12211
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
    575     574    5817
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e7Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
    610     609    6172
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e6Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
    624     623    6315
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e5Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
    626     625    6336
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
    637     636    6448
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.Chr*.MarkerChrBP.txt | wc
    862    2586   21825
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
    618     617    6263
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e7Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
    746     745    7565
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e6Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
    823     822    8346
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e5Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
    838     837    8494
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
    848     847    8594
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt | wc
    659    2636   20925
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e7Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt | wc
    809    3236   25785
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e6Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt | wc
   1066    4264   34064
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e5Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt | wc
   1510    6040   48370
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt | wc
   2356    9424   75681
~~~









zcat /data/external_public/GlobalLipid/jointGwasMc_*.txt.gz | awk '{ if ($9 <= .00000005) { print $0 } } ' | awk '{ print $3, "\t", $2, "\t", $9 }' | sed 's/:/ /g' | sed 's/chr//g' | sort | uniq | awk '{ print $0 } ' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBPpVal.txt
zcat /data/external_public/GlobalLipid/jointGwasMc_*.txt.gz | awk '{ if ($9 <= .0000005) { print $0 } } ' | awk '{ print $3, "\t", $2, "\t", $9 }' | sed 's/:/ /g' | sed 's/chr//g' | sort | uniq | awk '{ print $0 } ' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e7Cutoff.SNPChrBPpVal.txt
zcat /data/external_public/GlobalLipid/jointGwasMc_*.txt.gz | awk '{ if ($9 <= .000005) { print $0 } } ' | awk '{ print $3, "\t", $2, "\t", $9 }' | sed 's/:/ /g' | sed 's/chr//g' | sort | uniq | awk '{ print $0 } ' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e6Cutoff.SNPChrBPpVal.txt
zcat /data/external_public/GlobalLipid/jointGwasMc_*.txt.gz | awk '{ if ($9 <= .00005) { print $0 } } ' | awk '{ print $3, "\t", $2, "\t", $9 }' | sed 's/:/ /g' | sed 's/chr//g' | sort | uniq | awk '{ print $0 } ' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e5Cutoff.SNPChrBPpVal.txt
zcat /data/external_public/GlobalLipid/jointGwasMc_*.txt.gz | awk '{ if ($9 <= .0005) { print $0 } } ' | awk '{ print $3, "\t", $2, "\t", $9 }' | sed 's/:/ /g' | sed 's/chr//g' | sort | uniq | awk '{ print $0 } ' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e4Cutoff.SNPChrBPpVal.txt




















##ICBP2011
##20160313

mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/Analyses
mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/Analyses/PathwayWork

join <(cat /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.AllPheno.GWASHits.MarkerChrBP.txt | sort ) <(zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_DBP_results_v2.wHapMap21.vs3.MatchedToSBP.IncAllele.txt.gz | sort) | awk '{ print $2, "\t", $3, "\t", $3, "\t", $11, "\t", $12 }' > /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.AllPheno.GWASHits.MarkerChrBP.AnnovarFormat

join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/newtophits.vs2.SignCrrct.vs1.txt | awk '{ print $1, "\t", $2, "\t", $3 }' | sort) <(zcat /mnt/gluster/data/external_private_supp/ICBP2011/PhenoGenotypeFiles/RootStudyConsentSet_phs000585.ICBP_GWAS.v1.p1.c1.DS-CVD-IRB/AnalysisFiles/release_3_2013/data/ICBP2011_DBP_results_v2.wHapMap21.vs3.MatchedToSBP.IncAllele.txt.gz | sort) | awk '{ print $2, "\t", $3, "\t", $3, "\t", $11, "\t", $12 }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/Analyses/PathwayWork/newtophits.vs2.SignCrrct.vs1.AnnovarFormat

perl /mnt/lustre/home/mturchin20/Software/annovar/summarize_annovar.pl /mnt/lustre/home/mturchin20/Data/ICBP2011/ICBP2011.AllPheno.GWASHits.MarkerChrBP.AnnovarFormat /mnt/lustre/home/mturchin20/Software/annovar/humandb/ --ver1000g 1000g2012apr --verdbsnp 129 --veresp 6500 --buildver hg19 --remove

perl /mnt/lustre/home/mturchin20/Software/annovar/summarize_annovar.pl /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ICBP2011/Vs1/Analyses/PathwayWork/newtophits.vs2.SignCrrct.vs1.AnnovarFormat /mnt/lustre/home/mturchin20/Software/annovar/humandb/ --ver1000g 1000g2012apr --verdbsnp 129 --veresp 6500 --buildver hg19 --remove










##HaemgenRBC2012
##20160309

mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses
mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/PathwayWork

join <(cat /mnt/lustre/home/mturchin20/Data/HaemgenRBC2012/HaemgenRBC2012.Table1.Edited.MarkerChrBP.txt | sort) <(zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_RBC.wHapMap22.IncAllele.txt.gz | sort) | awk '{ print $2, "\t", $3, "\t", $3, "\t", $7, "\t", $8 }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/PathwayWork/HaemgenRBC2012.Table1.Edited.MarkerChrBP.AnnovarFormat

join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/GLC.MTedits.ForHaemgenRBC2012.AllPheno.vs1.newtophits.vs1.SignCrrct.vs1.txt | awk '{ print $1, "\t", $2, "\t", $3 }' | sort) <(zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_RBC.wHapMap22.IncAllele.txt.gz | sort) | awk '{ print $2, "\t", $3, "\t", $3, "\t", $7, "\t", $8 }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/PathwayWork/GLC.MTedits.ForHaemgenRBC2012.AllPheno.vs1.newtophits.vs1.SignCrrct.vs1.AnnovarFormat

perl /mnt/lustre/home/mturchin20/Software/annovar/summarize_annovar.pl /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/PathwayWork/HaemgenRBC2012.Table1.Edited.MarkerChrBP.AnnovarFormat /mnt/lustre/home/mturchin20/Software/annovar/humandb/ --ver1000g 1000g2012apr --verdbsnp 129 --veresp 6500 --buildver hg19 --remove

perl /mnt/lustre/home/mturchin20/Software/annovar/summarize_annovar.pl /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/PathwayWork/GLC.MTedits.ForHaemgenRBC2012.AllPheno.vs1.newtophits.vs1.SignCrrct.vs1.AnnovarFormat /mnt/lustre/home/mturchin20/Software/annovar/humandb/ --ver1000g 1000g2012apr --verdbsnp 129 --veresp 6500 --buildver hg19 --remove

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/PathwayWork/HaemgenRBC2012.Table1.Edited.MarkerChrBP.AnnovarFormat.genome_summary.csv | grep -v intergenic | grep -v ncRNA | awk -F, '{ print $2 }' | sed 's/"//g' | grep -v Gene > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/PathwayWork/HaemgenRBC2012.Table1.Edited.MarkerChrBP.AnnovarFormat.genome_summary.GeneIDs.no_Intergenic_ncRNA

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/PathwayWork/HaemgenRBC2012.Table1.Edited.MarkerChrBP.AnnovarFormat.genome_summary.csv | grep intergenic | sed 's/),/);/g' | perl -F, -lane 'print $F[1];' | perl -lane 'my @genes1 = split(/;/, $F[0]); foreach my $gene (@genes1) { print $gene; }' | sed 's/"//g' | sed 's/(/ /g' | awk '{ print $1 }' | grep -v Gene | grep -v NONE > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/PathwayWork/HaemgenRBC2012.Table1.Edited.MarkerChrBP.AnnovarFormat.genome_summary.GeneIDs.Intergenic

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/PathwayWork/GLC.MTedits.ForHaemgenRBC2012.AllPheno.vs1.newtophits.vs1.SignCrrct.vs1.AnnovarFormat.genome_summary.csv | grep -v intergenic | grep -v ncRNA | awk -F, '{ print $2 }' | sed 's/"//g' | grep -v Gene > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/PathwayWork/GLC.MTedits.ForHaemgenRBC2012.AllPheno.vs1.newtophits.vs1.SignCrrct.vs1.AnnovarFormat.genome_summary.GeneIDs.no_Intergenic_ncRNA

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/PathwayWork/GLC.MTedits.ForHaemgenRBC2012.AllPheno.vs1.newtophits.vs1.SignCrrct.vs1.AnnovarFormat.genome_summary.csv | grep intergenic | sed 's/),/);/g' | perl -F, -lane 'print $F[1];' | perl -lane 'my @genes1 = split(/;/, $F[0]); foreach my $gene (@genes1) { print $gene; }' | sed 's/"//g' | sed 's/(/ /g' | awk '{ print $1 }' | grep -v Gene | grep -v NONE > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/PathwayWork/GLC.MTedits.ForHaemgenRBC2012.AllPheno.vs1.newtophits.vs1.SignCrrct.vs1.AnnovarFormat.genome_summary.GeneIDs.Intergenic

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/PathwayWork/HaemgenRBC2012.Table1.Edited.MarkerChrBP.AnnovarFormat.genome_summary.GeneIDs.no_Intergenic_ncRNA /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/PathwayWork/GLC.MTedits.ForHaemgenRBC2012.AllPheno.vs1.newtophits.vs1.SignCrrct.vs1.AnnovarFormat.genome_summary.GeneIDs.no_Intergenic_ncRNA > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/PathwayWork/HaemgenRBC2012.UniHits.MultiHits.GeneIDs.no_Intergenic_ncRNA 

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/PathwayWork/HaemgenRBC2012.Table1.Edited.MarkerChrBP.AnnovarFormat.genome_summary.GeneIDs.Intergenic /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/PathwayWork/GLC.MTedits.ForHaemgenRBC2012.AllPheno.vs1.newtophits.vs1.SignCrrct.vs1.AnnovarFormat.genome_summary.GeneIDs.Intergenic > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/PathwayWork/HaemgenRBC2012.UniHits.MultiHits.GeneIDs.Intergenic

join <(zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.txt.gz | awk '{ print $1, "\t", $2, "\t", $3 }' | sort) <(zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_RBC.wHapMap22.IncAllele.txt.gz | sort) | awk '{ print $2, "\t", $3, "\t", $3, "\t", $7, "\t", $8 }' > /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.AnnovarFormat

~~~
[  mturchin20@bigmem01  ~/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1/Analyses/PathwayWork]$join <(zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.txt.gz | awk '{ print $1, "\t", $2, "\t", $3 }' | sort) <(zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_RBC.wHapMap22.IncAllele.txt.gz | sort) | awk '{ print $2, "\t", $3, "\t", $3, "\t", $7, "\t", $8 }' > /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.AnnovarFormat
[  mturchin20@bigmem01  ~/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1/Analyses/PathwayWork]$cat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.AnnovarFormat | wc
2425038 12125190 79750377
[  mturchin20@bigmem01  ~/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1/Analyses/PathwayWork]$zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.txt.gz | wc
2425038 38800611 262808152
~~~

perl /mnt/lustre/home/mturchin20/Software/annovar/summarize_annovar.pl /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.AnnovarFormat /mnt/lustre/home/mturchin20/Software/annovar/humandb/ --ver1000g 1000g2012apr --verdbsnp 129 --veresp 6500 --buildver hg19 --remove

cat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.AnnovarFormat.genome_summary.csv | grep -v intergenic | grep -v ncRNA | awk -F, '{ print $2 }' | sed 's/"//g' | grep -v Gene > /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.AnnovarFormat.genome_summary.GeneIDs.no_Intergenic_ncRNA

mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/PathwayWork/Permutations/

for i in {1..10}; do
	cat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.AnnovarFormat.genome_summary.GeneIDs.no_Intergenic_ncRNA | sort | uniq | R -q -e "Data1 <- read.table(file('stdin'), stringsAsFactors=FALSE, header=F); write.table(matrix(sample(Data1[,1], size=300), ncol=1), quote=FALSE, row.names=FALSE, col.names=FALSE);" | grep -v ^\> | perl -lane 'print +(split(/\(/, $F[0]))[0];' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/PathwayWork/Permutations/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.AnnovarFormat.genome_summary.GeneIDs.no_Intergenic_ncRNA.ran300.${i}
done

#scp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/PathwayWork/Permutations/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.AnnovarFormat.genome_summary.GeneIDs.no_Intergenic_ncRNA.ran300.* /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/PathwayWork/Permutations/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.AnnovarFormat.genome_summary.GeneIDs.no_Intergenic_ncRNA.ran100.* mturchin20@wolfy.uchicago.edu:/Users/mturchin20/LabMisc/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/PathwayWork/Permutations/.

shuf 



#20160619
#Followup analyses based on comments from NHS presentation -- looking further down into univariate results (how much less signif are the new hits from the original data), uniBF vs other BFs, heatmap of orig, new, miss Zscore hits
#Also in prep for committee meeting
#Continuing to use the 'ForCSHL' versions of scripts to keep track of code and trying things out

mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/

for i in `ls -lrt /mnt/gluster/data/external_public_supp/HaemgenRBC2012/. | grep IncAllele.formatted.txt.gz$ | grep -v AllPhenos | awk '{ print $9 }'`; do zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/$i; done | perl -lane ' if (abs($F[4]) <= .00000005) { print join("\t", @F); } ' | awk '{ print $3, "\t", $1, "\t", $2, "\t", $5 }' | grep -v Chr > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e8Cutoff.SNPChrBPpVal.txt 
for i in `ls -lrt /mnt/gluster/data/external_public_supp/HaemgenRBC2012/. | grep IncAllele.formatted.txt.gz$ | grep -v AllPhenos | awk '{ print $9 }'`; do zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/$i; done | perl -lane ' if (abs($F[4]) <= .0000005) { print join("\t", @F); } ' | awk '{ print $3, "\t", $1, "\t", $2, "\t", $5 }' | grep -v Chr > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e7Cutoff.SNPChrBPpVal.txt 
for i in `ls -lrt /mnt/gluster/data/external_public_supp/HaemgenRBC2012/. | grep IncAllele.formatted.txt.gz$ | grep -v AllPhenos | awk '{ print $9 }'`; do zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/$i; done | perl -lane ' if (abs($F[4]) <= .000005) { print join("\t", @F); } ' | awk '{ print $3, "\t", $1, "\t", $2, "\t", $5 }' | grep -v Chr > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e6Cutoff.SNPChrBPpVal.txt 
for i in `ls -lrt /mnt/gluster/data/external_public_supp/HaemgenRBC2012/. | grep IncAllele.formatted.txt.gz$ | grep -v AllPhenos | awk '{ print $9 }'`; do zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/$i; done | perl -lane ' if (abs($F[4]) <= .00005) { print join("\t", @F); } ' | awk '{ print $3, "\t", $1, "\t", $2, "\t", $5 }' | grep -v Chr > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e5Cutoff.SNPChrBPpVal.txt 
for i in `ls -lrt /mnt/gluster/data/external_public_supp/HaemgenRBC2012/. | grep IncAllele.formatted.txt.gz$ | grep -v AllPhenos | awk '{ print $9 }'`; do zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/$i; done | perl -lane ' if (abs($F[4]) <= .0005) { print join("\t", @F); } ' | awk '{ print $3, "\t", $1, "\t", $2, "\t", $5 }' | grep -v Chr > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e4Cutoff.SNPChrBPpVal.txt 

~~~
[  mturchin20@spudling96  ~/Data/iGENBP2015]$zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_RBC.wHapMap22.IncAllele.formatted.txt.gz /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_*.wHapMap22.MatchedToRBC.IncAllele.formatted.txt.gz | head -n 10
7        92221824        rs10    0.967   0.1006          13846
12       125456933       rs1000000       0.627   0.03907         53336
4        21227772        rs10000010      0.575   0.3052          45011
4        1347325         rs10000012      0.808   0.06836         53594
4        36901464        rs10000013      0.167   0.6573          53364
4        84997149        rs10000017      0.223   0.1169          45343
3        185118462       rs1000002       0.475   0.2828          53542
4        95952929        rs10000023      0.408   0.4534          49284
4        138905074       rs10000029      0.025   0.04834         12853
3        99825597        rs1000003       0.125   0.4191          53439
[  mturchin20@spudling96  ~/Data/iGENBP2015]$zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_RBC.wHapMap22.IncAllele.formatted.txt.gz /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_*.wHapMap22.MatchedToRBC.IncAllele.formatted.txt.gz | grep Chr  
Chr      Position        SNP     RAF     Pval    Sample_Size
SNP     Chr     Position        RAF     Pval    Sample  RAF     Pval    Position        RAF     Chr     Position        Size    Chr     Sample  Size    Pval    Sample  Size
Chr      Position        SNP     RAF     Pval    Sample_Size
Chr      Position        SNP     RAF     Pval    Sample_Size
Chr      Position        SNP     RAF     Pval    Sample_Size
Chr      Position        SNP     RAF     Pval    Sample_Size
Chr      Position        SNP     RAF     Pval    Sample_Size
[  mturchin20@spudling96  ~/Data/iGENBP2015]$ls -lrt /mnt/gluster/data/external_public_supp/HaemgenRBC2012/. | grep IncAllele.formatted.txt.gz$
-rw-rw-r-- 1 mturchin20 mturchin20  39187955 Feb 23 02:06 HaemGenRBC_RBC.wHapMap22.IncAllele.formatted.txt.gz
-rw-rw-r-- 1 mturchin20 mturchin20  40099342 Feb 23 10:19 HaemGenRBC_MCV.wHapMap22.MatchedToRBC.IncAllele.formatted.txt.gz
-rw-rw-r-- 1 mturchin20 mturchin20  39614307 Feb 23 10:24 HaemGenRBC_PCV.wHapMap22.MatchedToRBC.IncAllele.formatted.txt.gz
-rw-rw-r-- 1 mturchin20 mturchin20  39880429 Feb 23 10:24 HaemGenRBC_MCH.wHapMap22.MatchedToRBC.IncAllele.formatted.txt.gz
-rw-rw-r-- 1 mturchin20 mturchin20  39884607 Feb 23 10:25 HaemGenRBC_Hb.wHapMap22.MatchedToRBC.IncAllele.formatted.txt.gz
-rw-rw-r-- 1 mturchin20 mturchin20  40023913 Feb 23 10:25 HaemGenRBC_MCHC.wHapMap22.MatchedToRBC.IncAllele.formatted.txt.gz
-rw-rw-r-- 1 mturchin20 mturchin20 102988762 Feb 23 10:32 HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.txt.gz
[  mturchin20@spudling96  ~/Data/iGENBP2015]$ls -lrt /mnt/gluster/data/external_public_supp/HaemgenRBC2012/. | grep IncAllele.formatted.txt.gz$ | grep -v AllPhenos
-rw-rw-r-- 1 mturchin20 mturchin20  39187955 Feb 23 02:06 HaemGenRBC_RBC.wHapMap22.IncAllele.formatted.txt.gz
-rw-rw-r-- 1 mturchin20 mturchin20  40099342 Feb 23 10:19 HaemGenRBC_MCV.wHapMap22.MatchedToRBC.IncAllele.formatted.txt.gz
-rw-rw-r-- 1 mturchin20 mturchin20  39614307 Feb 23 10:24 HaemGenRBC_PCV.wHapMap22.MatchedToRBC.IncAllele.formatted.txt.gz
-rw-rw-r-- 1 mturchin20 mturchin20  39880429 Feb 23 10:24 HaemGenRBC_MCH.wHapMap22.MatchedToRBC.IncAllele.formatted.txt.gz
-rw-rw-r-- 1 mturchin20 mturchin20  39884607 Feb 23 10:25 HaemGenRBC_Hb.wHapMap22.MatchedToRBC.IncAllele.formatted.txt.gz
-rw-rw-r-- 1 mturchin20 mturchin20  40023913 Feb 23 10:25 HaemGenRBC_MCHC.wHapMap22.MatchedToRBC.IncAllele.formatted.txt.gz
[  mturchin20@spudling96  ~/Data/iGENBP2015]$for i in `ls -lrt /mnt/gluster/data/external_public_supp/HaemgenRBC2012/. | grep IncAllele.formatted.txt.gz$ | grep -v AllPhenos | awk '{ print $9 }'`; do zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/$i; done | wc        
14769120 88614720 757114750
[  mturchin20@spudling96  ~/Data/iGENBP2015]$for i in `ls -lrt /mnt/gluster/data/external_public_supp/HaemgenRBC2012/. | grep IncAllele.formatted.txt.gz$ | grep -v AllPhenos | awk '{ print $9 }'`; do zcat /mnt/gluster/data/external_public_supp/HaemgenRBC2012/$i | wc; done     
2462919 14777514 125316393
2461975 14771850 126850797
2461822 14770932 125866809
2459829 14758974 126724531
2462518 14775108 125979767
2460057 14760342 126376453
[  mturchin20@spudling96  ~/Data/iGENBP2015]$R -q -e "2461975 * 6"
> 2461975 * 6
[1] 14771850
> 
> 
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e8Cutoff.SNPChrBPpVal.txt | wc
   5458   21832  208599
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e7Cutoff.SNPChrBPpVal.txt | wc
   7639   30556  291707
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e6Cutoff.SNPChrBPpVal.txt | wc
  10904   43616  416160
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e5Cutoff.SNPChrBPpVal.txt | wc
  17566   70264  670202
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e4Cutoff.SNPChrBPpVal.txt | wc
  38951  155804 1484748
~~~

#Replaced 5e8 to 5e7, 5e6, 5e5 and 5e4
#for i in {1..22}; do
#for i in {X..X}; do
for i in `echo {1..22} "X"`; do
	cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e4Cutoff.SNPChrBPpVal.txt | awk '{ if ($2 == '$i') { print $0 } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e4Cutoff.SNPChrBPpVal.Chr${i}.txt
	cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e4Cutoff.SNPChrBPpVal.Chr${i}.txt | \
	R -q -e "Data1 <- read.table(file('stdin'), header=F); Data1 <- Data1[order(Data1[,4], decreasing=FALSE),]; Data2 <- c(); for (i in 1:nrow(Data1)) { Data2 <- rbind(Data2, Data1[1,]); Data1 <- Data1[Data1[,3] < Data1[1,3] - 500000 | Data1[,3] > Data1[1,3] + 500000,]; }; Data2;" | grep -v NA | grep -v V | grep -v \> | perl -lane 'print join("\t", @F[1..$#F]);' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e4Cutoff.SNPChrBPpVal.Chr${i}.1MbWindowPruned.txt
done

#rm /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e6Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt
#rm /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e6Cutoff.SNPChrBPpVal.ChrAll.txt
#for i in {1..22}; do
#for i in {X..X}; do
for i in `echo {1..22} "X"`; do
	if [ $i == "1" ] ; then
		rm -f /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt
		rm -f /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.txt
	fi	
	cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e4Cutoff.SNPChrBPpVal.Chr${i}.txt >> /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.txt
	cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e4Cutoff.SNPChrBPpVal.Chr${i}.1MbWindowPruned.txt >> /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt
done

#20160619 NOTE -- the below comment is from when pasting over GlobalLipids2013 code. During the GIANT2014_5 run I realized I coudl so the echo {1..2} "X" thing to get both 1..2 and X, so will be incorporating this in future efforts of similar problem. As of this comment still need to go back and do GlobalLipids2013 X stuff too, so the CHECK_0 is not complete #20160618 CHECK_0 -- Prob: Go back and check/include X in the 5e7s and later....was not present in 5e8 for sure

#rm /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.MatchedTo1MbWindowPruned.NoMatch.txt
for i in `echo {1..22} "X"`; do
	if [ $i == "1" ] ; then
		rm -f /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.MatchedTo1MbWindowPruned.NoMatch.txt
	fi
	echo $i >> /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.MatchedTo1MbWindowPruned.NoMatch.txt;
	python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e4Cutoff.SNPChrBPpVal.Chr${i}.1MbWindowPruned.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e4Cutoff.SNPChrBPpVal.Chr${i}.txt | perl -lane 'if (!$F[5]) { print join("\t", @F); }' | wc >> /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.MatchedTo1MbWindowPruned.NoMatch.txt
done
	
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e8Cutoff.SNPChrBPpVal.Chr1.1MbWindowPruned.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e8Cutoff.SNPChrBPpVal.Chr1.txt | perl -lane 'if ($F[5]) { print join("\t", @F); }' | wc 
python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e8Cutoff.SNPChrBPpVal.Chr1.1MbWindowPruned.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e8Cutoff.SNPChrBPpVal.Chr1.txt | perl -lane 'if (!$F[5]) { print join("\t", @F); }' | wc 


~~~
[  mturchin20@bigmem03  ~]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e8Cutoff.SNPChrBPpVal.Chr1.txt | wc
    140     560    5349
[  mturchin20@bigmem03  ~]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e8Cutoff.SNPChrBPpVal.Chr1.1MbWindowPruned.txt | wc
      8      32     260
[  mturchin20@bigmem03  ~]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.txt | wc
   5458   21832  208599
[  mturchin20@bigmem03  ~]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt | wc
     82     328    2652
[  mturchin20@bigmem03  ~]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e4Cutoff.SNPChrBPpVal.Chr1.txt | wc
   2665   10660  101284
[  mturchin20@bigmem03  ~]$for i in `echo {1..22} "X"`; do cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e8Cutoff.SNPChrBPpVal.Chr${i}.1MbWindowPruned.txt | wc; done 
      8      32     260
      3      12      95
      6      24     192
      3      12      96
      0       0       0
     16      64     516
      3      12      98
      2       8      63
      2       8      62
      3      12     100
      4      16     128
      6      24     196
      0       0       0
      2       8      67
      3      12      99
      4      16     127
      8      32     260
      1       4      33
      3      12      96
      1       4      33
      0       0       0
      4      16     131
      0       0       0
[  mturchin20@bigmem03  ~]$R -q -e "8 + 3 + 6 + 3 + 0 + 16 + 3 + 2 + 2 + 3 + 4 + 6 + 0 + 2 + 3 +4 + 8 + 1 + 3 + 1 + 0 + 4 + 0"
> 8 + 3 + 6 + 3 + 0 + 16 + 3 + 2 + 2 + 3 + 4 + 6 + 0 + 2 + 3 +4 + 8 + 1 + 3 + 1 + 0 + 4 + 0
[1] 82
> 
> 
[  mturchin20@bigmem03  ~]$python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e8Cutoff.SNPChrBPpVal.Chr1.1MbWindowPruned.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e8Cutoff.SNPChrBPpVal.Chr1.txt | perl -lane 'if ($F[5]) { print join("\t", @F); }' | wc
    140    1400    9673
[  mturchin20@bigmem03  ~]$python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e8Cutoff.SNPChrBPpVal.Chr1.1MbWindowPruned.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e8Cutoff.SNPChrBPpVal.Chr1.txt | perl -lane 'if (!$F[5]) { print join("\t", @F); }' | wc
      0       0       0
~~~

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/GLC.MTedits.ForHaemgenRBC2012.AllPheno.vs1.newtophits.vs1.SignCrrct.vs1.txt | awk '{ print $1, "\t", $2, "\t", $3 }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/GLC.MTedits.ForHaemgenRBC2012.AllPheno.vs1.newtophits.vs1.SignCrrct.vs1.MarkerChrBP.txt

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/GLC.MTedits.ForHaemgenRBC2012.AllPheno.vs1.newtophits.vs1.SignCrrct.vs1.MarkerChrBP.txt /mnt/lustre/home/mturchin20/Data/HaemgenRBC2012/HaemgenRBC2012.Table1.Edited.MarkerChrBP.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/GLC.MTedits.ForHaemgenRBC2012.AllPheno.vs1.newtophits.vs1.SignCrrct.vs1.MarkerChrBP.AndOrigUniHits.txt 

~~~
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/GLC.MTedits.ForHaemgenRBC2012.AllPheno.vs1.newtophits.vs1.SignCrrct.vs1.txt | wc
     71    1207    8381
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/GLC.MTedits.ForHaemgenRBC2012.AllPheno.vs1.newtophits.vs1.SignCrrct.vs1.txt | awk '{ print $1, "\t", $2, "\t", $3 }' | head -n 10
snp      chr     pos
rs10073340       5       1374873
rs10086987       8       48208696
rs10184575       2       53090876
rs10444523       12      94656231
rs10503716       8       21922608
rs11119328       1       207951941
rs11210927       1       44039007
rs11672923       19      50493862
rs11757235       6       28597258
~~~

for i in `echo {1..22} "X"`; do
	cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/GLC.MTedits.ForHaemgenRBC2012.AllPheno.vs1.newtophits.vs1.SignCrrct.vs1.MarkerChrBP.txt | awk '{ if ($2 == '$i') { print $0 } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/GLC.MTedits.ForHaemgenRBC2012.AllPheno.vs1.newtophits.vs1.SignCrrct.vs1.Chr${i}.MarkerChrBP.txt
	cat /mnt/lustre/home/mturchin20/Data/HaemgenRBC2012/HaemgenRBC2012.Table1.Edited.MarkerChrBP.txt | awk '{ if ($2 == '$i') { print $0 } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemgenRBC2012.Table1.Edited.Chr${i}.MarkerChrBP.txt
	cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/GLC.MTedits.ForHaemgenRBC2012.AllPheno.vs1.newtophits.vs1.SignCrrct.vs1.Chr${i}.MarkerChrBP.txt /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemgenRBC2012.Table1.Edited.Chr${i}.MarkerChrBP.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/GLC.MTedits.ForHaemgenRBC2012.AllPheno.vs1.newtophits.vs1.SignCrrct.vs1.Chr${i}.MarkerChrBP.AndOrigUniHits.txt	
done

#Replaced 5e8 to 5e7, 5e6, 5e5 and 5e4
for i in `echo {1..22} "X"`; do
	if [ $i == "1" ] ; then
		rm -f /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e5Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassHits1Mb.txt
		rm -f /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e5Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHits1Mb.txt
		rm -f /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e5Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHits1Mb.txt
	fi
	python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/GLC.MTedits.ForHaemgenRBC2012.AllPheno.vs1.newtophits.vs1.SignCrrct.vs1.Chr${i}.MarkerChrBP.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e5Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt >> /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e5Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassHits1Mb.txt 
	python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemgenRBC2012.Table1.Edited.Chr${i}.MarkerChrBP.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e5Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt >> /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e5Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHits1Mb.txt
	python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/GLC.MTedits.ForHaemgenRBC2012.AllPheno.vs1.newtophits.vs1.SignCrrct.vs1.Chr${i}.MarkerChrBP.AndOrigUniHits.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e5Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt >> /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e5Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHits1Mb.txt 
done

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e7Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e6Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e5Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e7Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e6Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e5Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e7Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e6Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e5Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt | wc



[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt | wc

~~~
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
     13      12     123
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e7Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
     28      27     279
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e6Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
     50      49     508
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e5Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
     67      66     680
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
     70      69     711
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
     68      67     672
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e7Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
     76      75     755
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e6Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
     76      75     755
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e5Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
     76      75     755
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
     76      75     755
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
     80      79     794
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e7Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
    103     102    1033
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e6Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
    125     124    1262
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e5Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc 
    142     141    1434
[  mturchin20@spudling96  ~/Data/iGENBP2015]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
    144     143    1455
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt | wc
     82     328    2652
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e7Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt | wc
    114     456    3701
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e6Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt | wc
    204     816    6620
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e5Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt | wc
    620    2480   20058
[  mturchin20@spudling96  ~/Lab_Stuff/StephensLab/Multivariate/Choongwon2016/Vs1]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/ComparingNewBFSToOrigPVals/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt | wc
   1961    7844   63733
~~~













cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/HaemgenRBC2012.Orig3.newtophits.vs1.SignCrrct.vs1.txt | awk '{ print $1, "\t", $2, "\t", $3 }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/HaemgenRBC2012.Orig3.newtophits.vs1.SignCrrct.vs1.MarkerChrBP.txt

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5.Orig3.newtophits.vs1.SignCrrct.vs1.MarkerChrBP.txt /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5.Orig3.newtophits.vs1.SignCrrct.vs1.MarkerChrBP.AndOrigUniHits.txt

~~~
[  mturchin20@spudhead  ~/Software]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5.Orig3.newtophits.vs1.SignCrrct.vs1.txt | awk '{ print $1, "\t", $2, "\t", $3 }' | wc
    218     654    5624
[  mturchin20@spudhead  ~/Software]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5.Orig3.newtophits.vs1.SignCrrct.vs1.txt | awk '{ print $1, "\t", $2, "\t", $3 }' | head -n 10snp      chr     pos
rs6719451        2       238001540
rs7651617        3       169321372
rs2741856        17      39182364
rs12899850       15      63838352
rs615632         8       9833730
rs2347709        7       134054280
rs10737541       1       166480721
rs9424148        10      12186863
rs11135932       8       26182377
~~~

for i in `echo {1..22} "X"`; do
	cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5.Orig3.newtophits.vs1.SignCrrct.vs1.MarkerChrBP.txt | awk '{ if ($2 == '$i') { print $0 } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.Orig3.newtophits.vs1.SignCrrct.vs1.Chr${i}.MarkerChrBP.txt
	cat /mnt/lustre/home/mturchin20/Data/GIANT/2014_5/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.MarkerChrBP.txt | awk '{ if ($2 == '$i') { print $0 } }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.Chr${i}.MarkerChrBP.txt
	cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.Orig3.newtophits.vs1.SignCrrct.vs1.Chr${i}.MarkerChrBP.txt /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.Chr${i}.MarkerChrBP.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.Orig3.newtophits.vs1.SignCrrct.vs1.Chr${i}.MarkerChrBP.AndOrigUniHits.txt	
done

#Replaced 5e8 to 5e7, 5e6, 5e5 and 5e4
for i in `echo {1..22} "X"`; do
	if [ $i == "1" ] ; then
		rm -f /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassHits1Mb.txt
		rm -f /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHits1Mb.txt
		rm -f /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHits1Mb.txt
	fi
	python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.Orig3.newtophits.vs1.SignCrrct.vs1.Chr${i}.MarkerChrBP.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt >> /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassHits1Mb.txt 
	python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllGWASHits.HeightBMIWHRadjBMI.Chr${i}.MarkerChrBP.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt >> /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHits1Mb.txt
	python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.Orig3.newtophits.vs1.SignCrrct.vs1.Chr${i}.MarkerChrBP.AndOrigUniHits.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.txt >> /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHits1Mb.txt 
done

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e7Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e6Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e5Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e7Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e6Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e5Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e8Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e7Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e6Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e5Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.ChrAll.1MbWindowPruned.wbmassAndOrigHits1Mb.txt | awk '{ print $6 }' | sort | uniq | wc

#20160620 CHECK_0 -- Prob: Think through this problem of doing the 1Mb window pruning leading to an upper-bound of regions available that is below the total number of univariate GWAS hits, eg getting like ~610 regions pruned at 5e8 threshold but there are ~820 GWAS hits, so clearly not going to get all of them represented at that threshold/area. And we never seem to reach ~820 even after raising threshold -- likely due to multiple hits being within 1Mb of one another and LD not really matching up with the simplifying 1Mb physical distance assumption
~~~

[  mturchin20@spudling07  ~]$for i in `echo {1..22} "X"`; do cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e8Cutoff.SNPChrBPpVal.Chr${i}.1MbWindowPruned.txt | wc; done
     63     252    1997
     55     220    1746
     42     168    1333
     31     124     980
     40     160    1252
     54     216    1706
     29     116     903
     26     104     815
     34     136    1080
     23      92     755
     37     148    1197
     32     128    1039
     16      64     519
     22      88     710
     30     120     966
     25     100     797
     37     148    1182
     10      40     318
     23      92     731
     20      80     599
      5      20     151
      5      20     149
      0       0       0
[  mturchin20@spudling07  ~]$R -q -e "63 + 55 + 42 + 31 + 40 + 54 + 29 + 26 + 34 + 23 + 37 + 32 + 16 + 22 + 30 + 25 + 37 + 10 + 23 + 20 + 5 + 5"
> 63 + 55 + 42 + 31 + 40 + 54 + 29 + 26 + 34 + 23 + 37 + 32 + 16 + 22 + 30 + 25 + 37 + 10 + 23 + 20 + 5 + 5
[1] 659
> 
> 
[  mturchin20@spudling07  ~]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.Chr1.txt | wc
  10840   43360  367751
[  mturchin20@spudling07  ~]$python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.Chr1.1MbWindowPruned.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.Chr1.txt | perl -lane 'if ($F[5]) { print join("\t", @F); }' | wc
  10840  108400  714567
[  mturchin20@spudling07  ~]$python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.Chr1.1MbWindowPruned.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e4Cutoff.SNPChrBPpVal.Chr1.txt | perl -lane 'if (!$F[5]) { print join("\t", @F); }' | wc
      0       0       0

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5.Orig3.newtophits.vs1.SignCrrct.vs1.txt | awk '{ print $1, "\t", $2, "\t", $3 }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/GIANT2014_5.Orig3.newtophits.vs1.SignCrrct.vs1.MarkerChrBP.txt


zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.IncAllele.txt.gz /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.txt.gz /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.MatchedToHeight.IncAllele.gz | perl -lane ' if (abs($F[6]) <= .00000005) { print join("\t", @F); } ' | awk '{ print $1, "\t", $9, "\t", $7 }' | sed 's/_/ /g' | sort | uniq | awk '{ print $0 } ' | grep -v ChrBP | grep -v , > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e8Cutoff.SNPChrBPpVal.txt 
zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.IncAllele.txt.gz /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.txt.gz /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.MatchedToHeight.IncAllele.gz | perl -lane ' if (abs($F[6]) <= .0000005) { print join("\t", @F); } ' | awk '{ print $1, "\t", $9, "\t", $7 }' | sed 's/_/ /g' | sort | uniq | awk '{ print $0 } ' | grep -v ChrBP | grep -v , > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e7Cutoff.SNPChrBPpVal.txt 
zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.IncAllele.txt.gz /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_2015_WHRadjBMI_COMBINED_EUR.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.txt.gz /mnt/gluster/data/external_public_supp/GIANT2014_5/SNP_gwas_mc_merge_nogc.tbl.uniq.wUCSCGenomeBrowser_dbSNP130.vs1.MatchedToHeight.IncAllele.gz | perl -lane ' if (abs($F[6]) <= .000005) { print join("\t", @F); } ' | awk '{ print $1, "\t", $9, "\t", $7 }' | sed 's/_/ /g' | sort | uniq | awk '{ print $0 } ' | grep -v ChrBP | grep -v , > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs1/Analyses/ComparingNewBFSToOrigPVals/GIANT2014_5.AllPhenos.wUCSCGenomeBrowser_dbSNP130.vs1.HeaderFix.MatchedToHeight.IncAllele.pVal5e6Cutoff.SNPChrBPpVal.txt 
zcat /data/external_public/GlobalLipid/jointGwasMc_*.txt.gz | awk '{ if ($9 <= .00000005) { print $0 } } ' | awk '{ print $3, "\t", $2, "\t", $9 }' | sed 's/:/ /g' | sed 's/chr//g' | sort | uniq | awk '{ print $0 } ' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e8Cutoff.SNPChrBPpVal.txt
zcat /data/external_public/GlobalLipid/jointGwasMc_*.txt.gz | awk '{ if ($9 <= .0000005) { print $0 } } ' | awk '{ print $3, "\t", $2, "\t", $9 }' | sed 's/:/ /g' | sed 's/chr//g' | sort | uniq | awk '{ print $0 } ' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e7Cutoff.SNPChrBPpVal.txt
zcat /data/external_public/GlobalLipid/jointGwasMc_*.txt.gz | awk '{ if ($9 <= .000005) { print $0 } } ' | awk '{ print $3, "\t", $2, "\t", $9 }' | sed 's/:/ /g' | sed 's/chr//g' | sort | uniq | awk '{ print $0 } ' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e6Cutoff.SNPChrBPpVal.txt
zcat /data/external_public/GlobalLipid/jointGwasMc_*.txt.gz | awk '{ if ($9 <= .00005) { print $0 } } ' | awk '{ print $3, "\t", $2, "\t", $9 }' | sed 's/:/ /g' | sed 's/chr//g' | sort | uniq | awk '{ print $0 } ' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e5Cutoff.SNPChrBPpVal.txt
zcat /data/external_public/GlobalLipid/jointGwasMc_*.txt.gz | awk '{ if ($9 <= .0005) { print $0 } } ' | awk '{ print $3, "\t", $2, "\t", $9 }' | sed 's/:/ /g' | sed 's/chr//g' | sort | uniq | awk '{ print $0 } ' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GlobalLipids2013/Vs1/Analyses/ComparingNewBFSToOrigPVals/GlobalLipids2013.newtophits.vs3.SignCrrct.vs1.wjointGwasMc_AllPhenos.pVal5e4Cutoff.SNPChrBPpVal.txt






















##MAGIC2010
##20160313

mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/MAGIC2010/Vs1/Analyses
mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/MAGIC2010/Vs1/Analyses/PathwayWork


join <(cat /mnt/lustre/home/mturchin20/Data/MAGIC2010/MAGIC2010.Table1.MarkerChrBP.txt | sort) <(zcat /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_FastingInsulin.wHapMap22.vs3.IncAllele.txt.gz | sort) | awk '{ print $2, "\t", $3, "\t", $3, "\t", $4, "\t", $5 }' | sed 's/a/A/g' | sed 's/g/G/' | sed 's/c/C/g' | sed 's/t/T/g' > /mnt/lustre/home/mturchin20/Data/MAGIC2010/MAGIC2010.Table1.MarkerChrBP.AnnovarFormat 

join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/MAGIC2010/Vs1/newtophits.vs2.SignCrrct.vs1.txt | awk '{ print $1, "\t", $2, "\t", $3 }' | sort) <(zcat /mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_FastingInsulin.wHapMap22.vs3.IncAllele.txt.gz | sort) | awk '{ print $2, "\t", $3, "\t", $3, "\t", $4, "\t", $5 }' | sed 's/a/A/g' | sed 's/g/G/' | sed 's/c/C/g' | sed 's/t/T/g' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/MAGIC2010/Vs1/Analyses/PathwayWork/newtophits.vs2.SignCrrct.vs1.AnnovarFormat

perl /mnt/lustre/home/mturchin20/Software/annovar/summarize_annovar.pl /mnt/lustre/home/mturchin20/Data/MAGIC2010/MAGIC2010.Table1.MarkerChrBP.AnnovarFormat /mnt/lustre/home/mturchin20/Software/annovar/humandb/ --ver1000g 1000g2012apr --verdbsnp 129 --veresp 6500 --buildver hg19 --remove

perl /mnt/lustre/home/mturchin20/Software/annovar/summarize_annovar.pl /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/MAGIC2010/Vs1/Analyses/PathwayWork/newtophits.vs2.SignCrrct.vs1.AnnovarFormat /mnt/lustre/home/mturchin20/Software/annovar/humandb/ --ver1000g 1000g2012apr --verdbsnp 129 --veresp 6500 --buildver hg19 --remove




perl /mnt/lustre/home/mturchin20/Software/annovar/summarize_annovar.pl /mnt/gluster/data/external_public_supp/HaemgenRBC2012/HaemGenRBC_AllPhenos.wHapMap22.MatchedToRBC.IncAllele.formatted.AnnovarFormat /mnt/lustre/home/mturchin20/Software/annovar/humandb/ --ver1000g 1000g2012apr --verdbsnp 129 --veresp 6500 --buildver hg19 --remove


/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/MAGIC2010/Vs1/newtophits.vs2.SignCrrct.vs1.txt
/mnt/gluster/data/external_public_supp/MAGIC2010/MAGIC_ln_FastingInsulin.wHapMap22.vs3.IncAllele.txt.gz







##GEFOS2015
##20160309

mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1/Analyses
mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1/Analyses/PathwayWork

join -1 1 -2 3 <(cat /mnt/lustre/home/mturchin20/Data/GEFOS2015/GEFOS2015.SupplTable13.edited.MarkerChrBP.txt | sort) <(zcat /mnt/gluster/data/external_public_supp/GEFOS2015/fa2stu.MAF0_.005.pos_.maf05.IncAllele.out_.gz | sort -k 3,3) | awk '{ print $2, "\t", $3, "\t", $3, "\t", $6, "\t", $7 }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1/Analyses/PathwayWork/GEFOS2015.SupplTable13.edited.MarkerChrBP.AnnovarFormat

join -1 1 -2 3 <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1/GEFOS2015.newtophits.vs1.SignCrrct.vs1.txt | awk '{ print $1, "\t", $2, "\t", $3 }' | sort) <(zcat /mnt/gluster/data/external_public_supp/GEFOS2015/fa2stu.MAF0_.005.pos_.maf05.IncAllele.out_.gz | sort -k 3,3) | awk '{ print $2, "\t", $3, "\t", $3, "\t", $6, "\t", $7 }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GEFOS2015/Vs1/Analyses/PathwayWork/GEFOS2015.newtophits.vs1.SignCrrct.vs1.AnnovarFormat













##Added here 20160726 -- but done sometime in mid/late March?
#Able to grab this line (the last one of a group of lines) from some holdover histories in a few older, opened screens
#mysql -h genome-mysql.cse.ucsc.edu -u genome -D hg19 -N -A -e 'select chrom,strand, txStart,cdsStart,geneSymbol from knownGene INNER JOIN kgXref ON knownGene.name = kgXref.kgID where chrom = "chr1" AND cdsstart < '$val1'' > utrs.txt
#cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/PathwayWork/utrs.txt | head -n 20
#[  mturchin20@spudhead  ~/Lab_Stuff/StephensLab/bmass]$find $HOME/ -name "utrs.txt"
#/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/HaemgenRBC2012/Vs1/Analyses/PathwayWork/utrs.txt

#[  mturchin20@spudhead  ~/Lab_Stuff/StephensLab/Multivariate]$grep --include=\*.{R,sh,py} -rnw '/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/' -e "mysql"







###########################
#
#
# Part E -- bmass Code writeup
#
#
###########################

cd /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/bmass
R -q -e "library(\"devtools\"); devtools::create(\"/mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/bmass\");" 
mkdir R
mkdir data

#Log file suggestion -- http://stackoverflow.com/questions/22843775/how-to-create-periodically-send-text-to-a-log-file-while-printing-normal-outpu
#Post about cat vs print -- http://stackoverflow.com/questions/31843662/what-is-the-difference-between-cat-and-print



###########################
#
#
# Part F -- Per-dataset analyses for specific people/projects/reasons
#
#
###########################






##GIANT 2014/5
##20161222


#Sarah U MASH result comparisons
#20161222

mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople
mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU
mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons
mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant
mkdir /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec

cd /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant

wget https://github.com/surbut/mashgwas/tree/master/Data/significant

#for i in `cat significant | grep txt | sed 's/=/ /g' | awk '{ print $6 }' | sed 's/\"//g'`; do wget --no-check-certificate https://github.com$i; done
for i in `cat significant | grep txt | sed 's/=/ /g' | awk '{ print $6 }' | sed 's/\"//g' | perl -F/\ -lane 'print $F[$#F];'`; do wget --no-check-certificate https://raw.githubusercontent.com/surbut/mashgwas/master/Data/significant/$i; done

cd /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec

wget https://github.com/surbut/mashgwas/tree/master/Data/traitspec

for i in `cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspec | grep txt | sed 's/=/ /g' | awk '{ print $6 }' | sed 's/\"//g' | perl -F/\ -lane 'print $F[$#F];'`; do wget --no-check-certificate https://raw.githubusercontent.com/surbut/mashgwas/master/Data/traitspec/$i; done

join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecheight.txt | awk '{ print $2 }' | sed 's/"//g' | sort) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | sort -k 1,1) | perl -lane 'print $F[0], "\t", $F[$#F];' | sed 's/_/ /g' | perl -F, -lane 'print $F[0];' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecheight.MarkerChrBP.txt
join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecbmi.txt | awk '{ print $2 }' | sed 's/"//g' | sort) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | sort -k 1,1) | perl -lane 'print $F[0], "\t", $F[$#F];' | sed 's/_/ /g' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecbmi.MarkerChrBP.txt

join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightsignificantin.txt | awk '{ print $2 }' | sed 's/"//g' | sort) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | sort -k 1,1) | perl -lane 'print $F[0], "\t", $F[$#F];' | sed 's/_/ /g' | perl -F, -lane 'print $F[0];' | grep -v NA > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightsignificantin.MarkerChrBP.txt
join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/bmisignificantin.txt | awk '{ print $2 }' | sed 's/"//g' | sort) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | sort -k 1,1) | perl -lane 'print $F[0], "\t", $F[$#F];' | sed 's/_/ /g' | perl -F, -lane 'print $F[0];' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/bmisignificantin.MarkerChrBP.txt

~~~
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec]$cat traitspecheight.txt | wc
  36288   72575  720066
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec]$join <(cat traitspecheight.txt | awk '{ print $2 }' | sed 's/"//g' | sort) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | sort -k 1,1) | wc
  36287  326583 2167198
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec]$join <(cat traitspecheight.txt | awk '{ print $2 }' | sed 's/"//g' | sort) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | sort -k 1,1) | perl -lane 'print $F[0], "\t", $F[$#F];' | sed 's/_/ /g' | head -n 10
rs10000502      4 141084206
rs10000610      4 159252624
rs1000071       11 4557877
rs10000747      4 83037841
rs10000864      4 187936416
rs10000940      4 44689543
rs10000960      4 129302182
rs10001054      4 48103577
rs10001241      4 132410977
rs1000176       20 21181552
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec]$join <(cat traitspecheight.txt | awk '{ print $2 }' | sed 's/"//g' | sort) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | sort -k 1,1) | perl -lane 'print $F[0], "\t", $F[$#F];' | sed 's/_/ /g' | perl -lane 'print $#F;' | sort | uniq -c
  36276 2
      1 3
      2 4
      1 5
      1 6
      6 8
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec]$join <(cat traitspecheight.txt | awk '{ print $2 }' | sed 's/"//g' | sort) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | sort -k 1,1) | perl -lane 'print $F[0], "\t", $F[$#F];' | sed 's/_/ /g' | perl -lane 'if ($#F == 8) { print join("\t", @F);}'
rs1044043       6       32901958,6      cox     hap1    4174836,6       qbl     hap2    3980033
rs11955148      5       127585712,5     h2      hap1    798644,5        h2      hap1    1302625
rs1826833       11      50176398,6      cox     hap1    221488,6        qbl     hap2    19336
rs2235498       6       33238407,6      cox     hap1    4510593,6       qbl     hap2    4316599
rs2272874       6       29804223,6      cox     hap1    1148935,6       qbl     hap2    951052
rs7941169       11      50174233,6      cox     hap1    219189,6        qbl     hap2    17020
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec]$join <(cat traitspecheight.txt | awk '{ print $2 }' | sed 's/"//g' | sort) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | sort -k 1,1) | perl -lane 'print $F[0], "\t", $F[$#F];' | sed 's/_/ /g' | grep , | head -n 10              
rs1044043       6 32901958,6 cox hap1 4174836,6 qbl hap2 3980033
rs10992512      9 94692771,9 98950389
rs11955148      5 127585712,5 h2 hap1 798644,5 h2 hap1 1302625
rs12992721      2 148840166,Y 57656780,6 qbl hap2 3442364
rs16900380      5 71978016,6 random 565629
rs1826833       11 50176398,6 cox hap1 221488,6 qbl hap2 19336
rs2235498       6 33238407,6 cox hap1 4510593,6 qbl hap2 4316599
rs2272874       6 29804223,6 cox hap1 1148935,6 qbl hap2 951052
rs5996135       22 40971401,22 h2 hap1 46549
rs7941169       11 50174233,6 cox hap1 219189,6 qbl hap2 17020
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec]$join <(cat traitspecheight.txt | awk '{ print $2 }' | sed 's/"//g' | sort) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | sort -k 1,1) | perl -lane 'print $F[0], "\t", $F[$#F];' | sed 's/_/ /g' | perl -F, -lane 'print $F[0];' | perl -lane 'print $#F;' | sort | uniq -c
  36287 2
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec]$cat traitspecbmi.txt | wc
    500     999    8957
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec]$join <(cat traitspecbmi.txt | awk '{ print $2 }' | sed 's/"//g' | sort) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | sort -k 1,1) | perl -lane 'print $F[0], "\t", $F[$#F];' | sed 's/_/ /g' | wc
    499    1497   10830
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec]$join <(cat traitspecbmi.txt | awk '{ print $2 }' | sed 's/"//g' | sort) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | sort -k 1,1) | perl -lane 'print $F[0], "\t", $F[$#F];' | sed 's/_/ /g' | head -n 10
rs1005552       17 32023710
rs10057590      5 124363145
rs10057967      5 75033511
rs10092489      8 77299142
rs1009468       9 15815229
rs10161145      12 87869167
rs10166736      2 227608662
rs1016756       2 58530852
rs10170606      2 565563
rs10204803      2 58758358
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec]$join <(cat traitspecbmi.txt | awk '{ print $2 }' | sed 's/"//g' | sort) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | sort -k 1,1) | perl -lane 'print $F[0], "\t", $F[$#F];' | sed 's/_/ /g' | perl -lane 'print $#F;' | sort | uniq -c 
    499 2
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec]$join <(cat traitspecbmi.txt | awk '{ print $2 }' | sed 's/"//g' | sort) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | sort -k 1,1) | perl -lane 'print $F[0], "\t", $F[$#F];' | sed 's/_/ /g' | grep ,                              
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec]$
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(cat heightsignificantin.txt | awk '{ print $2 }' | sed 's/"//g' | sort) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | sort -k 1,1) | perl -lane 'print $F[0], "\t", $F[$#F];' | sed 's/_/ /g' | grep , | head -n 10
rs10419174      19 42297180,6 cox hap1 110605
rs1042147       6 31191134,6 cox hap1 2534366,6 qbl hap2 2332920
rs1044043       6 32901958,6 cox hap1 4174836,6 qbl hap2 3980033
rs10992512      9 94692771,9 98950389
rs1150753       6 32167844,6 cox hap1 3467036
rs1150754       6 32158735,6 cox hap1 3457930,6 qbl hap2 3265727
rs1150755       6 32146527,6 cox hap1 3445724,6 qbl hap2 3253532
rs11955148      5 127585712,5 h2 hap1 798644,5 h2 hap1 1302625
rs12198173      6 32134785,6 cox hap1 3433965,6 qbl hap2 3241808
rs12360875      11 50190697,6 cox hap1 230198,6 cox hap1 244639,6 qbl hap2 28167,6 qbl hap2 42601
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(cat heightsignificantin.txt | awk '{ print $2 }' | sed 's/"//g' | sort) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | sort -k 1,1) | perl -lane 'print $F[0], "\t", $F[$#F];' | sed 's/_/ /g' | perl -lane 'if ($#F == 8) { print join("\t", @F);}' | head -n 10
rs1042147       6       31191134,6      cox     hap1    2534366,6       qbl     hap2    2332920
rs1044043       6       32901958,6      cox     hap1    4174836,6       qbl     hap2    3980033
rs1150754       6       32158735,6      cox     hap1    3457930,6       qbl     hap2    3265727
rs1150755       6       32146527,6      cox     hap1    3445724,6       qbl     hap2    3253532
rs11955148      5       127585712,5     h2      hap1    798644,5        h2      hap1    1302625
rs12198173      6       32134785,6      cox     hap1    3433965,6       qbl     hap2    3241808
rs1265048       6       31189387,6      cox     hap1    2532619,6       qbl     hap2    2331173
rs1265061       6       31186235,6      cox     hap1    2529467,6       qbl     hap2    2328021
rs1265063       6       31185916,6      cox     hap1    2529148,6       qbl     hap2    2327702
rs1265087       6       31217788,6      cox     hap1    2561030,6       qbl     hap2    2359590
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(cat heightsignificantin.txt | awk '{ print $2 }' | sed 's/"//g' | sort) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | sort -k 1,1) | perl -lane 'print $F[0], "\t", $F[$#F];' | sed 's/_/ /g' |
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat heightsignificantin.txt | wc
  75699  151397 1512936
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(cat heightsignificantin.txt | awk '{ print $2 }' | sed 's/"//g' | sort) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | sort -k 1,1) | perl -lane 'print $F[0], "\t", $F[$#F];' | sed 's/_/ /g' | wc
  75698  227917 1654161
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(cat heightsignificantin.txt | awk '{ print $2 }' | sed 's/"//g' | sort) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | sort -k 1,1) | perl -lane 'print $F[0], "\t", $F[$#F];' | sed 's/_/ /g' | head -n 10
rs10000502      4 141084206
rs10000531      4 10903454
rs1000055       18 44835310
rs10000610      4 159252624
rs1000071       11 4557877
rs10000747      4 83037841
rs10000864      4 187936416
rs10000940      4 44689543
rs10000960      4 129302182
rs10001054      4 48103577
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(cat heightsignificantin.txt | awk '{ print $2 }' | sed 's/"//g' | sort) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | sort -k 1,1) | perl -lane 'print $F[0], "\t", $F[$#F];' | sed 's/_/ /g' | perl -lane 'print $#F;' | sort | uniq -c
      1 1
      1 11
      1 14
  75544 2
      2 3
      4 4
     25 5
      1 6
    119 8
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(cat heightsignificantin.txt | awk '{ print $2 }' | sed 's/"//g' | sort) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | sort -k 1,1) | perl -lane 'print $F[0], "\t", $F[$#F];' | sed 's/_/ /g' | perl -F, -lane 'print $F[0];' | perl -lane 'print $#F;' | sort | uniq -c 
      1 1
  75697 2
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(cat heightsignificantin.txt | awk '{ print $2 }' | sed 's/"//g' | sort) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | sort -k 1,1) | perl -lane 'print $F[0], "\t", $F[$#F];' | sed 's/_/ /g' | perl -F, -lane 'print $F[0];' | perl -lane 'if ($#F == 1) { print join("\t", @F); }'
rs138544        NA
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(cat heightsignificantin.txt | awk '{ print $2 }' | sed 's/"//g' | sort) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | sort -k 1,1) | perl -lane 'print $F[0], "\t", $F[$#F];' | sed 's/_/ /g' | perl -F, -lane 'print $F[0];' | grep -v NA | perl -lane 'print $#F;' | sort | uniq -c
  75697 2
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat bmisignificantin.txt | wc
   4884    9767   92305
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(cat bmisignificantin.txt | awk '{ print $2 }' | sed 's/"//g' | sort) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | sort -k 1,1) | perl -lane 'print $F[0], "\t", $F[$#F];' | sed 's/_/ /g' | wc
   4883   14727  106511
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(cat bmisignificantin.txt | awk '{ print $2 }' | sed 's/"//g' | sort) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | sort -k 1,1) | perl -lane 'print $F[0], "\t", $F[$#F];' | sed 's/_/ /g' | head -n 10
rs10009336      4 44175539
rs10010305      4 11656391
rs10013571      4 28821637
rs1001870       15 65938501
rs10020270      4 16069344
rs1002922       5 40422311
rs10029408      4 60257431
rs10029732      4 48193435
rs10033946      4 20661466
rs10040401      5 91512801
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(cat bmisignificantin.txt | awk '{ print $2 }' | sed 's/"//g' | sort) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | sort -k 1,1) | perl -lane 'print $F[0], "\t", $F[$#F];' | sed 's/_/ /g' | perl -lane 'print $#F;' | sort | uniq -c
   4869 2
      2 5
     12 8
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(cat bmisignificantin.txt | awk '{ print $2 }' | sed 's/"//g' | sort) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | sort -k 1,1) | perl -lane 'print $F[0], "\t", $F[$#F];' | sed 's/_/ /g' | grep , | head -n 10           
rs17421624      6 32174154,6 cox hap1 3473345
rs2071293       6 32170664,6 cox hap1 3469855
rs2071295       6 32146677,6 cox hap1 3445874,6 qbl hap2 3253682
rs2233956       6 31189183,6 cox hap1 2532415,6 qbl hap2 2330969
rs2299851       6 31826580,6 cox hap1 3164698,6 qbl hap2 2966190
rs2516049       6 32678377,6 cox hap1 3957888,6 qbl hap2 3761787
rs2734335       6 32001922,6 cox hap1 3340206,6 qbl hap2 3141683
rs3117574       6 31833208,6 cox hap1 3171322,6 qbl hap2 2972817
rs3130484       6 31823860,6 cox hap1 3161979,6 qbl hap2 2963471
rs577272        6 31945941,6 cox hap1 3284223,6 qbl hap2 3085698
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(cat bmisignificantin.txt | awk '{ print $2 }' | sed 's/"//g' | sort) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | sort -k 1,1) | perl -lane 'print $F[0], "\t", $F[$#F];' | sed 's/_/ /g' | perl -lane 'if ($#F == 8) { print join("\t", @F); }' | head -n 10
rs2071295       6       32146677,6      cox     hap1    3445874,6       qbl     hap2    3253682
rs2233956       6       31189183,6      cox     hap1    2532415,6       qbl     hap2    2330969
rs2299851       6       31826580,6      cox     hap1    3164698,6       qbl     hap2    2966190
rs2516049       6       32678377,6      cox     hap1    3957888,6       qbl     hap2    3761787
rs2734335       6       32001922,6      cox     hap1    3340206,6       qbl     hap2    3141683
rs3117574       6       31833208,6      cox     hap1    3171322,6       qbl     hap2    2972817
rs3130484       6       31823860,6      cox     hap1    3161979,6       qbl     hap2    2963471
rs577272        6       31945941,6      cox     hap1    3284223,6       qbl     hap2    3085698
rs589428        6       31956198,6      cox     hap1    3294480,6       qbl     hap2    3095958
rs605203        6       31954990,6      cox     hap1    3293272,6       qbl     hap2    3094750
[  mturchin20@spudling07  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(cat bmisignificantin.txt | awk '{ print $2 }' | sed 's/"//g' | sort) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | sort -k 1,1) | perl -lane 'print $F[0], "\t", $F[$#F];' | sed 's/_/ /g' | perl -F, -lane 'print $F[0];' | perl -lane 'print $#F;' | sort | uniq -c
   4883 2
~~~

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightsignificantin.MarkerChrBP.txt /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/bmisignificantin.MarkerChrBP.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightbmisignificantin.MarkerChrBP.txt

join -v 1 <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightsignificantin.MarkerChrBP.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/bmisignificantin.MarkerChrBP.txt | sort -k 1,1) > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightsignificantin.MarkerChrBP.NoOverlapBMI.txt
join -v 2 <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightsignificantin.MarkerChrBP.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/bmisignificantin.MarkerChrBP.txt | sort -k 1,1) > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/bmisignificantin.MarkerChrBP.NoOverlapHeight.txt
join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightsignificantin.MarkerChrBP.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/bmisignificantin.MarkerChrBP.txt | sort -k 1,1) > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightbmisignificantin.MarkerChrBP.OverlapHeightBMI.txt

join -v 1 <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecheight.MarkerChrBP.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecbmi.MarkerChrBP.txt | sort -k 1,1) > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecheight.MarkerChrBP.NoOverlapBMI.txt
join -v 2 <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecheight.MarkerChrBP.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecbmi.MarkerChrBP.txt | sort -k 1,1) > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecbmi.MarkerChrBP.NoOverlapHeight.txt
join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecheight.MarkerChrBP.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecbmi.MarkerChrBP.txt | sort -k 1,1) > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecheightbmi.MarkerChrBP.OverlapHeightBMI.txt

#python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightsignificantin.MarkerChrBP.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/GIANT2014_5.Vs2.HeightBMI.NewSNPs.SNPs.vs1.ChrBPChrBP.txt 
##python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/GIANT2014_5.Vs2.HeightBMI.NewSNPs.SNPs.vs1.ChrBPChrBP.txt --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightsignificantin.MarkerChrBP.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/GIANT2014_5.Vs2.HeightBMI.NewSNPs.SNPs.vs1.ChrBPChrBP.txt | grep -v BP | python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 - --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightsignificantin.MarkerChrBP.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightsignificantin.MarkerChrBP.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/GIANT2014_5.Vs2.HeightBMI.NewSNPs.SNPs.vs1.ChrBPChrBP.txt | grep -v BP | python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 - --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/bmisignificantin.MarkerChrBP.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/bmisignificantin.MarkerChrBP.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/GIANT2014_5.Vs2.HeightBMI.NewSNPs.SNPs.vs1.ChrBPChrBP.txt | grep -v BP | python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 - --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightsignificantin.MarkerChrBP.NoOverlapBMI.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightsignificantin.MarkerChrBP.NoOverlapBMI.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/GIANT2014_5.Vs2.HeightBMI.NewSNPs.SNPs.vs1.ChrBPChrBP.txt | grep -v BP | python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 - --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/bmisignificantin.MarkerChrBP.NoOverlapHeight.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/bmisignificantin.MarkerChrBP.NoOverlapHeight.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/GIANT2014_5.Vs2.HeightBMI.NewSNPs.SNPs.vs1.ChrBPChrBP.txt | grep -v BP | python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 - --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightbmisignificantin.MarkerChrBP.OverlapHeightBMI.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightbmisignificantin.MarkerChrBP.OverlapHeightBMI.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/GIANT2014_5.Vs2.HeightBMI.NewSNPs.SNPs.vs1.ChrBPChrBP.txt | grep -v BP | python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 - --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecheight.MarkerChrBP.NoOverlapBMI.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecheight.MarkerChrBP.NoOverlapBMI.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/GIANT2014_5.Vs2.HeightBMI.NewSNPs.SNPs.vs1.ChrBPChrBP.txt | grep -v BP | python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 - --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecbmi.MarkerChrBP.NoOverlapHeight.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecbmi.MarkerChrBP.NoOverlapHeight.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt


cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightsignificantin.MarkerChrBP.NoOverlapBMI.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | awk '{ print $5 }' | grep _ | sort | uniq -c | wc
join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/GIANT2014_5.Vs2.HeightBMI.NewSNPs.SNPs.vs1.ChrBPChrBP.txt | sort -k 1,1) <(cat <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightsignificantin.MarkerChrBP.NoOverlapBMI.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | awk '{ print $5 }' | grep _ | sort | uniq -c) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/bmisignificantin.MarkerChrBP.NoOverlapHeight.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | awk '{ print $5 }' | grep _ | sort | uniq -c) | awk '{ print $2 }' | sort | uniq) | wc

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecheight.MarkerChrBP.NoOverlapBMI.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | awk '{ print $5 }' | grep _ | sort | uniq -c | wc
join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/GIANT2014_5.Vs2.HeightBMI.NewSNPs.SNPs.vs1.ChrBPChrBP.txt | sort -k 1,1) <(cat <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecheight.MarkerChrBP.NoOverlapBMI.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | awk '{ print $5 }' | grep _ | sort | uniq -c) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecbmi.MarkerChrBP.NoOverlapHeight.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | awk '{ print $5 }' | grep _ | sort | uniq -c) | awk '{ print $2 }' | sort | uniq) | wc


###Edit /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py to do 2Mb window
##cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/GIANT2014_5.Vs2.HeightBMI.NewSNPs.SNPs.vs1.ChrBPChrBP.txt | grep -v BP | python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 - --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightsignificantin.MarkerChrBP.NoOverlapBMI.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightsignificantin.MarkerChrBP.NoOverlapBMI.OverlapWith.GIANT2014_5.HeightBMI.2MbWindow.vs1.txt
##cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/GIANT2014_5.Vs2.HeightBMI.NewSNPs.SNPs.vs1.ChrBPChrBP.txt | grep -v BP | python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 - --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/bmisignificantin.MarkerChrBP.NoOverlapHeight.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/bmisignificantin.MarkerChrBP.NoOverlapHeight.OverlapWith.GIANT2014_5.HeightBMI.2MbWindow.vs1.txt
##cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/GIANT2014_5.Vs2.HeightBMI.NewSNPs.SNPs.vs1.ChrBPChrBP.txt | grep -v BP | python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 - --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightbmisignificantin.MarkerChrBP.OverlapHeightBMI.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightbmisignificantin.MarkerChrBP.OverlapHeightBMI.OverlapWith.GIANT2014_5.HeightBMI.2MbWindow.vs1.txt
##
##cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/GIANT2014_5.Vs2.HeightBMI.NewSNPs.SNPs.vs1.ChrBPChrBP.txt | grep -v BP | python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 - --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecheight.MarkerChrBP.NoOverlapBMI.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecheight.MarkerChrBP.NoOverlapBMI.OverlapWith.GIANT2014_5.HeightBMI.2MbWindow.vs1.txt
##cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/GIANT2014_5.Vs2.HeightBMI.NewSNPs.SNPs.vs1.ChrBPChrBP.txt | grep -v BP | python /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/DataOrganization.AllStudies.PythonVLookUp.GWASHitProximityAnnot.vs1.py --file1 - --file2 /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecbmi.MarkerChrBP.NoOverlapHeight.txt > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecbmi.MarkerChrBP.NoOverlapHeight.OverlapWith.GIANT2014_5.HeightBMI.2MbWindow.vs1.txt 


cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightsignificantin.MarkerChrBP.NoOverlapBMI.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | awk '{ if ($4 == 0) { print $0 } }' | awk '{ print $1 "\t" $2 "\t" $3 }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightsignificantin.MarkerChrBP.NoOverlapBMI.OverlapWith.GIANT2014_5.HeightBMI.NoHits.vs1.txt
ln -s /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightsignificantin.MarkerChrBP.NoOverlapBMI.OverlapWith.GIANT2014_5.HeightBMI.NoHits.vs1.txt /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightsignificantin.NoOverlapWith.bmass.GIANT2014_5.HeightBMI.vs1.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/bmisignificantin.MarkerChrBP.NoOverlapHeight.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | awk '{ if ($4 == 0) { print $0 } }' | awk '{ print $1 "\t" $2 "\t" $3 }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/bmisignificantin.MarkerChrBP.NoOverlapHeight.OverlapWith.GIANT2014_5.HeightBMI.NoHits.vs1.txt
ln -s /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/bmisignificantin.MarkerChrBP.NoOverlapHeight.OverlapWith.GIANT2014_5.HeightBMI.NoHits.vs1.txt /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/bmisignificantin.NoOverlapWith.bmass.GIANT2014_5.HeightBMI.vs1.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightbmisignificantin.MarkerChrBP.OverlapHeightBMI.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | awk '{ if ($6 == 0) { print $0 } }' | awk '{ print $1 "\t" $2 "\t" $3 }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightbmisignificantin.MarkerChrBP.OverlapHeightBMI.OverlapWith.GIANT2014_5.HeightBMI.NoHits.vs1.txt
ln -s /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightbmisignificantin.MarkerChrBP.OverlapHeightBMI.OverlapWith.GIANT2014_5.HeightBMI.NoHits.vs1.txt /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightbmisignificantin.NoOverlapWith.bmass.GIANT2014_5.HeightBMI.vs1.txt

cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecheight.MarkerChrBP.NoOverlapBMI.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | awk '{ if ($4 == 0) { print $0 } }' | awk '{ print $1 "\t" $2 "\t" $3 }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecheight.MarkerChrBP.NoOverlapBMI.OverlapWith.GIANT2014_5.HeightBMI.NoHits.vs1.txt
ln -s /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecheight.MarkerChrBP.NoOverlapBMI.OverlapWith.GIANT2014_5.HeightBMI.NoHits.vs1.txt /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecheight.NoOverlapWith.bmass.GIANT2014_5.HeightBMI.vs1.txt
cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecbmi.MarkerChrBP.NoOverlapHeight.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | awk '{ if ($4 == 0) { print $0 } }' | awk '{ print $1 "\t" $2 "\t" $3 }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecbmi.MarkerChrBP.NoOverlapHeight.OverlapWith.GIANT2014_5.HeightBMI.NoHits.vs1.txt
ln -s /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecbmi.MarkerChrBP.NoOverlapHeight.OverlapWith.GIANT2014_5.HeightBMI.NoHits.vs1.txt /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecbmi.NoOverlapWith.bmass.GIANT2014_5.HeightBMI.vs1.txt

join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/GIANT2014_5.Vs2.HeightBMI.NewSNPs.SNPs.vs1.ChrBPChrBP.txt | sort -k 1,1) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | perl -lane 'print $F[$#F], "\t", $F[0];' | sort -k 1,1) | awk '{ print $4 "\t" $2 "\t" $3 }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/GIANT2014_5.Vs2.HeightBMI.NewSNPs.SNPs.vs1.MarkerChrBP.txt
ln -s /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/GIANT2014_5.Vs2.HeightBMI.NewSNPs.SNPs.vs1.MarkerChrBP.txt /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/bmass.GIANT2014_5.Vs2.HeightBMI.NewSNPs.vs1.txt

#On wolfy.uchicago.edu
#mkdir /Users/mturchin20/LabMisc/StephensLab/Multivariate/ForPeople
#mkdir /Users/mturchin20/LabMisc/StephensLab/Multivariate/ForPeople/SarahU
#mkdir /Users/mturchin20/LabMisc/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons

scp -p /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightsignificantin.NoOverlapWith.bmass.GIANT2014_5.HeightBMI.vs1.txt /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/bmisignificantin.NoOverlapWith.bmass.GIANT2014_5.HeightBMI.vs1.txt /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightbmisignificantin.NoOverlapWith.bmass.GIANT2014_5.HeightBMI.vs1.txt /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecheight.NoOverlapWith.bmass.GIANT2014_5.HeightBMI.vs1.txt /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecbmi.NoOverlapWith.bmass.GIANT2014_5.HeightBMI.vs1.txt /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/bmass.GIANT2014_5.Vs2.HeightBMI.NewSNPs.vs1.txt mturchin20@wolfy.uchicago.edu:/Users/mturchin20/LabMisc/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/.

#From MacBook Air
#mkdir /Users/mturchin20/Documents/Work/LabMisc/StephensLab/Multivariate/ForPeople
#mkdir /Users/mturchin20/Documents/Work/LabMisc/StephensLab/Multivariate/ForPeople/SarahU
#mkdir /Users/mturchin20/Documents/Work/LabMisc/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons
#scp -p mturchin20wolfy.uchicago.edu:/Users/mturchin20/LabMisc/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/* /Users/mturchin20/Documents/Work/LabMisc/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/.


~~~
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join -v 1 <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightsignificantin.MarkerChrBP.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/bmisignificantin.MarkerChrBP.txt | sort -k 1,1) | wc
  72592  217776 1581554
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join -v 2 <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightsignificantin.MarkerChrBP.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/bmisignificantin.MarkerChrBP.txt | sort -k 1,1) | wc 
   1778    5334   38634
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightsignificantin.MarkerChrBP.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/bmisignificantin.MarkerChrBP.txt | sort -k 1,1) | wc      3105   15525  103375
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightsignificantin.MarkerChrBP.txt /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/bmisignificantin.MarkerChrBP.txt | wc
  80580  241740 1754954
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$R -q -e "72592 + 1778 + 3105 + 3105" 
> 72592 + 1778 + 3105 + 3105
[1] 80580
> 
> 
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecheight.MarkerChrBP.txt /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecbmi.MarkerChrBP.txt | wc
  36786  110358  801516
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join -v 1 <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecheight.MarkerChrBP.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecbmi.MarkerChrBP.txt | sort -k 1,1) | wc
  36287  108861  790686
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join -v 2 <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecheight.MarkerChrBP.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecbmi.MarkerChrBP.txt | sort -k 1,1) | wc
    499    1497   10830
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecheight.MarkerChrBP.txt | sort -k 1,1) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecbmi.MarkerChrBP.txt | sort -k 1,1) | wc
      0       0       0
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightsignificantin.MarkerChrBP.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | awk '{ print $5 }' | sort | uniq -c | wc  
    199     397    3903
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightsignificantin.MarkerChrBP.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | awk '{ print $5 }' | sort | uniq -c | awk '{ print $1 }' | R -q -e "Data1 <- read.table(file('stdin'), header=F); table(cut(Data1[,1], c(0,10,20,30,40,50,60,70,80,100))); quantile(Data1[,1]);"
> Data1 <- read.table(file('stdin'), header=F); table(cut(Data1[,1], c(0,10,20,30,40,50,60,70,80,100))); quantile(Data1[,1]);

  (0,10]  (10,20]  (20,30]  (30,40]  (40,50]  (50,60]  (60,70]  (70,80] 
      10       19       39       31       27       31       18       14 
(80,100] 
       3 
     0%     25%     50%     75%    100% 
    3.0    26.5    41.0    57.5 67100.0 
> 
> 
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightsignificantin.MarkerChrBP.NoOverlapBMI.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | awk '{ print $5 }' | sort | uniq -c | wc
    199     397    3903
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightsignificantin.MarkerChrBP.NoOverlapBMI.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | awk '{ print $5 }' | sort | uniq -c | head -n 10
  64284 
     44 10_12186863
     20 10_125050937
     69 10_22911128
     38 10_30507810
     76 10_31157363
     36 10_63250872
     27 10_64988771
     44 10_79281772
     34 10_81696303
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightsignificantin.MarkerChrBP.NoOverlapBMI.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | awk '{ print $5 }' | grep _ | sort | uniq -c | wc
    198     396    3894
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightsignificantin.MarkerChrBP.NoOverlapBMI.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | awk '{ print $5 }' | grep _ | sort | uniq -c | head -n 10
     44 10_12186863
     20 10_125050937
     69 10_22911128
     38 10_30507810
     76 10_31157363
     36 10_63250872
     27 10_64988771
     44 10_79281772
     34 10_81696303
     67 10_89379557
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/bmisignificantin.MarkerChrBP.NoOverlapHeight.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | awk '{ print $5 }' | grep _ | sort | uniq -c | wc
     25      50     493
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightsignificantin.MarkerChrBP.NoOverlapBMI.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | awk '{ print $5 }' | grep _ | sort | uniq -c) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/bmisignificantin.MarkerChrBP.NoOverlapHeight.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | awk '{ print $5 }' | grep _ | sort | uniq -c) | wc                                      
    223     446    4387
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightsignificantin.MarkerChrBP.NoOverlapBMI.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | awk '{ print $5 }' | grep _ | sort | uniq -c) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/bmisignificantin.MarkerChrBP.NoOverlapHeight.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | awk '{ print $5 }' | grep _ | sort | uniq -c) | awk '{ print $2 }' | sort | uniq -c | wc
    198     396    3894
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/GIANT2014_5.Vs2.HeightBMI.NewSNPs.SNPs.vs1.ChrBPChrBP.txt | sort -k 1,1) <(cat <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightsignificantin.MarkerChrBP.NoOverlapBMI.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | awk '{ print $5 }' | grep _ | sort | uniq -c) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/bmisignificantin.MarkerChrBP.NoOverlapHeight.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | awk '{ print $5 }' | grep _ | sort | uniq -c) | awk '{ print $2 }' | sort | uniq) | wc
    198     594    4620
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join -a 1 <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/GIANT2014_5.Vs2.HeightBMI.NewSNPs.SNPs.vs1.ChrBPChrBP.txt | sort -k 1,1) <(cat <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightsignificantin.MarkerChrBP.NoOverlapBMI.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | awk '{ print $5 }' | grep _ | sort | uniq -c) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/bmisignificantin.MarkerChrBP.NoOverlapHeight.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | awk '{ print $5 }' | grep _ | sort | uniq -c) | awk '{ print $2 }' | sort | uniq) | wc
    201     603    4679
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/GIANT2014_5.Vs2.HeightBMI.NewSNPs.SNPs.vs1.ChrBPChrBP.txt | wc
    201     603    4679
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join -v 1 <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/GIANT2014_5.Vs2.HeightBMI.NewSNPs.SNPs.vs1.ChrBPChrBP.txt | sort -k 1,1) <(cat <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightsignificantin.MarkerChrBP.NoOverlapBMI.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | awk '{ print $5 }' | grep _ | sort | uniq -c) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/bmisignificantin.MarkerChrBP.NoOverlapHeight.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | awk '{ print $5 }' | grep _ | sort | uniq -c) | awk '{ print $2 }' | sort | uniq) | wc
      3       9      59
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join -v 1 <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/GIANT2014_5.Vs2.HeightBMI.NewSNPs.SNPs.vs1.ChrBPChrBP.txt | sort -k 1,1) <(cat <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightsignificantin.MarkerChrBP.NoOverlapBMI.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | awk '{ print $5 }' | grep _ | sort | uniq -c) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/bmisignificantin.MarkerChrBP.NoOverlapHeight.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | awk '{ print $5 }' | grep _ | sort | uniq -c) | awk '{ print $2 }' | sort | uniq) 
11_51270793 11 51270793
2_95249470 2 95249470
ChrBP Chr BP
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep 51270793
rs5017948       A       T       0.212   0.026   0.0040  1.5e-10 238963  11_51270793
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep 2_95249470
rs13007001      A       G       0.958   0.046   0.0090  2.3e-07 229695  2_95249470
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecheight.MarkerChrBP.NoOverlapBMI.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | awk '{ print $5 }' | grep _ | sort | uniq -c | wc
    194     388    3813
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecbmi.MarkerChrBP.NoOverlapHeight.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | awk '{ print $5 }' | grep _ | sort | uniq -c | wc
     11      22     218
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecheight.MarkerChrBP.NoOverlapBMI.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | awk '{ print $5 }' | grep _ | sort | uniq -c) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecbmi.MarkerChrBP.NoOverlapHeight.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | awk '{ print $5 }' | grep _ | sort | uniq -c) | awk '{ print $2 }' | sort | uniq | wc
    196     196    2285
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecheight.MarkerChrBP.NoOverlapBMI.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | awk '{ print $5 }' | grep _ | sort | uniq -c) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecbmi.MarkerChrBP.NoOverlapHeight.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | awk '{ print $5 }' | grep _ | sort | uniq -c) | awk '{ print $2 }' | sort | uniq | head -n 10
10_12186863
10_125050937
10_22911128
10_30507810
10_31157363
10_63250872
10_64988771
10_79281772
10_81696303
10_89379557
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/GIANT2014_5.Vs2.HeightBMI.NewSNPs.SNPs.vs1.ChrBPChrBP.txt | sort -k 1,1) <(cat <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecheight.MarkerChrBP.NoOverlapBMI.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | awk '{ print $5 }' | grep _ | sort | uniq -c) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecbmi.MarkerChrBP.NoOverlapHeight.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | awk '{ print $5 }' | grep _ | sort | uniq -c) | awk '{ print $2 }' | sort | uniq) | wc
    196     588    4570
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join -v 1 <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/GIANT2014_5.Vs2.HeightBMI.NewSNPs.SNPs.vs1.ChrBPChrBP.txt | sort -k 1,1) <(cat <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecheight.MarkerChrBP.NoOverlapBMI.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | awk '{ print $5 }' | grep _ | sort | uniq -c) <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecbmi.MarkerChrBP.NoOverlapHeight.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | awk '{ print $5 }' | grep _ | sort | uniq -c) | awk '{ print $2 }' | sort | uniq)     
11_51270793 11 51270793
14_104348253 14 104348253
2_95249470 2 95249470
4_131036509 4 131036509
ChrBP Chr BP
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep 104348253
rs4983559       A       G       0.633   -0.015  0.0032  2.6e-06 233438  14_104348253
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | grep 131036509
rs2391518       T       C       0.3     0.0075  0.0032  0.017   252193  4_131036509
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecheight.MarkerChrBP.NoOverlapBMI.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | awk '{ if ($2 == 11) { if (($3 >= 49270793) && ($3 <= 53270793)) { print $0 } } }' | sort -g -k 3,3
rs17556442      11      49282000        2       11_49264086     11      49264086        2
rs3960835       11      49283109        2       11_49264086     11      49264086        2
rs10501325      11      49347194        2       11_49264086     11      49264086        2
rs10839287      11      49455498        2       11_49264086     11      49264086        2
rs7117247       11      49503350        2       11_49264086     11      49264086        2
rs11040421      11      49527162        2       11_49264086     11      49264086        2
rs7925397       11      49572091        2       11_49264086     11      49264086        2
rs17473628      11      49576942        2       11_49264086     11      49264086        2
rs4466833       11      49590125        2       11_49264086     11      49264086        2
rs12798376      11      49632663        2       11_49264086     11      49264086        2
rs10839357      11      49726568        2       11_49264086     11      49264086        2
rs4881649       11      49802135        0
rs7120100       11      49805081        0
rs10839398      11      49816619        0
rs10769621      11      49817038        0
rs11040588      11      49817337        0
rs7928714       11      49844197        0
rs10769630      11      49857387        0
rs1983448       11      49890431        0
rs4628673       11      49894154        0
rs10839448      11      49895505        0
rs1851869       11      49917127        2       11_50402841     11      50402841        2
rs10769643      11      49938205        2       11_50402841     11      50402841        2
rs4316513       11      49941290        2       11_50402841     11      50402841        2
rs1390677       11      50026907        2       11_50402841     11      50402841        2
rs1845861       11      50077715        2       11_50402841     11      50402841        2
rs7928188       11      50080636        2       11_50402841     11      50402841        2
rs1995042       11      50089949        2       11_50402841     11      50402841        2
rs10794212      11      50128452        2       11_50402841     11      50402841        2
rs7941169       11      50174233        2       11_50402841     11      50402841        2
rs1826833       11      50176398        2       11_50402841     11      50402841        2
rs7120107       11      50204907        2       11_50402841     11      50402841        2
rs7101625       11      50205109        2       11_50402841     11      50402841        2
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecheight.MarkerChrBP.NoOverlapBMI.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | awk '{ if ($2 == 14) { if (($3 >= 103348253) && ($3 <= 105348253)) { print $0 } } }' | sort -g -k 3,3
rs2497295       14      103721149       0
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecheight.MarkerChrBP.NoOverlapBMI.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | awk '{ if ($2 == 14) { if (($3 >= 102348253) && ($3 <= 106348253)) { print $0 } } }' | sort -g -k 3,3
rs1190202       14      102764264       0
rs959083        14      102892439       0
rs10144051      14      102955683       0
rs10148970      14      102961752       0
rs6575988       14      102970579       0
rs2497295       14      103721149       0
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecheight.MarkerChrBP.NoOverlapBMI.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | awk '{ if ($2 == 2) { if (($3 >= 94249470) && ($3 <= 96249470)) { print $0 } } }' | sort -g -k 3,3
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecheight.MarkerChrBP.NoOverlapBMI.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | awk '{ if ($2 == 2) { if (($3 >= 93249470) && ($3 <= 97249470)) { print $0 } } }' | sort -g -k 3,3
rs12477963      2       96625319        2       2_96697218      2       96697218        2
rs17636359      2       97024345        2       2_96697218      2       96697218        2
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecheight.MarkerChrBP.NoOverlapBMI.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | awk '{ if ($2 == 4) { if (($3 >= 130036509) && ($3 <= 132036509)) { print $0 } } }' | sort -g -k 3,3
rs477926        4       130352821       0
rs1709425       4       130358389       0
rs1875799       4       130361130       0
rs306373        4       131900553       0
rs3113500       4       131914880       0
rs306377        4       131919663       0
rs11727622      4       131943240       0
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightsignificantin.MarkerChrBP.NoOverlapBMI.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | wc        
  72592  323600 1937286
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightsignificantin.MarkerChrBP.NoOverlapBMI.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | awk '{ if ($4 == 0) { print $0 } }' | wc
  64284  257136 1529045
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightsignificantin.MarkerChrBP.NoOverlapBMI.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | grep -v _ | wc
  64284  257136 1529045
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightsignificantin.MarkerChrBP.NoOverlapBMI.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | awk '{ if ($4 == 0) { print $0 } }' | awk '{ print $1 "\t" $2 "\t" $3 }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightsignificantin.MarkerChrBP.NoOverlapBMI.OverlapWith.GIANT2014_5.HeightBMI.NoHits.vs1.txt
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$ln -s /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightsignificantin.MarkerChrBP.NoOverlapBMI.OverlapWith.GIANT2014_5.HeightBMI.NoHits.vs1.txt /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightsignificantin.NoOverlapWith.bmass.GIANT2014_5.HeightBMI.vs1.txt
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/bmisignificantin.MarkerChrBP.NoOverlapHeight.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | awk '{ if ($4 == 0) { print $0 } }' | awk '{ print $1 "\t" $2 "\t" $3 }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/bmisignificantin.MarkerChrBP.NoOverlapHeight.OverlapWith.GIANT2014_5.HeightBMI.NoHits.vs1.txt
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$ln -s /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/bmisignificantin.MarkerChrBP.NoOverlapHeight.OverlapWith.GIANT2014_5.HeightBMI.NoHits.vs1.txt /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/bmisignificantin.NoOverlapWith.bmass.GIANT2014_5.HeightBMI.vs1.txt
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightbmisignificantin.MarkerChrBP.OverlapHeightBMI.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | awk '{ if ($6 == 0) { print $0 } }' | awk '{ print $1 "\t" $2 "\t" $3 }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightbmisignificantin.MarkerChrBP.OverlapHeightBMI.OverlapWith.GIANT2014_5.HeightBMI.NoHits.vs1.txt
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$ln -s /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightbmisignificantin.MarkerChrBP.OverlapHeightBMI.OverlapWith.GIANT2014_5.HeightBMI.NoHits.vs1.txt /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant/heightbmisignificantin.NoOverlapWith.bmass.GIANT2014_5.HeightBMI.vs1.txt
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat heightsignificantin.MarkerChrBP.NoOverlapBMI.OverlapWith.GIANT2014_5.HeightBMI.NoHits.vs1.txt | wc
  64284  257136 1529045
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat heightbmisignificantin.MarkerChrBP.OverlapHeightBMI.OverlapWith.GIANT2014_5.HeightBMI.NoHits.vs1.txt | wc
   2816   16896   99354
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat bmisignificantin.MarkerChrBP.NoOverlapHeight.OverlapWith.GIANT2014_5.HeightBMI.NoHits.vs1.txt | wc
   1639    6556   38842
  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecheight.MarkerChrBP.NoOverlapBMI.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | awk '{ if ($4 == 0) { print $0 } }' | awk '{ print $1 "\t" $2 "\t" $3 }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecheight.MarkerChrBP.NoOverlapBMI.OverlapWith.GIANT2014_5.HeightBMI.NoHits.vs1.txt
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$ln -s /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecheight.MarkerChrBP.NoOverlapBMI.OverlapWith.GIANT2014_5.HeightBMI.NoHits.vs1.txt /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecheight.NoOverlapWith.bmass.GIANT2014_5.HeightBMI.vs1.txt
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecbmi.MarkerChrBP.NoOverlapHeight.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | awk '{ if ($4 == 0) { print $0 } }' | awk '{ print $1 "\t" $2 "\t" $3 }' > /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecbmi.MarkerChrBP.NoOverlapHeight.OverlapWith.GIANT2014_5.HeightBMI.NoHits.vs1.txt
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$ln -s /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecbmi.MarkerChrBP.NoOverlapHeight.OverlapWith.GIANT2014_5.HeightBMI.NoHits.vs1.txt /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/traitspec/traitspecbmi.NoOverlapWith.bmass.GIANT2014_5.HeightBMI.vs1.txt
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat ../traitspec/traitspecheight.MarkerChrBP.NoOverlapBMI.OverlapWith.GIANT2014_5.HeightBMI.NoHits.vs1.txt | wc
  32232   96696  702344
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat ../traitspec/traitspecbmi.MarkerChrBP.NoOverlapHeight.OverlapWith.GIANT2014_5.HeightBMI.NoHits.vs1.txt | wc
    461    1383    9988
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat ../traitspec/traitspecheightbmi.MarkerChrBP.OverlapHeightBMI.txt | wc
      0       0       0
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat ../traitspec/traitspecheight.MarkerChrBP.NoOverlapBMI.txt | wc
  36287  108861  790686
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat ../traitspec/traitspecbmi.MarkerChrBP.NoOverlapHeight.txt | wc
    499    1497   10830
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat ../traitspec/traitspecheight.MarkerChrBP.NoOverlapBMI.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | wc
  36287  161368  965834
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat ../traitspec/traitspecbmi.MarkerChrBP.NoOverlapHeight.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | wc
    499    2148   12818
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat ../traitspec/traitspecheight.MarkerChrBP.NoOverlapBMI.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | grep _ | wc
   4055   32440  199026
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat ../traitspec/traitspecbmi.MarkerChrBP.NoOverlapHeight.OverlapWith.GIANT2014_5.HeightBMI.vs1.txt | grep _ | wc
     38     304    1908
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | perl -lane 'print $F[$#F], "\t", $F[0];' | head -n 10
ChrBP   MarkerName
10_10000134     rs4747841
10_10000264     rs4749917
10_100002728    rs737656
10_100002879    rs737657
10_100003552    rs7086391
10_100003804    rs878177
10_100003966    rs878178
10_10000458     rs12219605
10_100005552    rs3763688
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/GIANT2014_5.Vs2.HeightBMI.NewSNPs.SNPs.vs1.ChrBPChrBP.txt | head -n 10
ChrBP   Chr     BP
10_12186863     10      12186863
10_125050937    10      125050937
10_22911128     10      22911128
10_30507810     10      30507810
10_31157363     10      31157363
10_63250872     10      63250872
10_64988771     10      64988771
10_79281772     10      79281772
10_81696303     10      81696303
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/GIANT2014_5.Vs2.HeightBMI.NewSNPs.SNPs.vs1.ChrBPChrBP.txt | sort -k 1,1) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | perl -lane 'print $F[$#F], "\t", $F[0];' | sort -k 1,1) | head -n 10
10_12186863 10 12186863 rs9424148
10_125050937 10 125050937 rs10510128
10_22911128 10 22911128 rs12770214
10_30507810 10 30507810 rs2488003
10_31157363 10 31157363 rs11008282
10_63250872 10 63250872 rs1470374
10_64988771 10 64988771 rs10761785
10_79281772 10 79281772 rs1248690
10_81696303 10 81696303 rs721917
10_89379557 10 89379557 rs4934353
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$join <(cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/GIANT2014_5.Vs2.HeightBMI.NewSNPs.SNPs.vs1.ChrBPChrBP.txt | sort -k 1,1) <(zcat /mnt/gluster/data/external_public_supp/GIANT2014_5/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.wUCSCGenomeBrowser_dbSNP130.vs1.txt.gz | perl -lane 'print $F[$#F], "\t", $F[0];' | sort -k 1,1) | wc
    202     808    6757
[  mturchin20@bigmem03  ~/Lab_Stuff/StephensLab/Multivariate/ForPeople/SarahU/MASHComparisons/significant]$cat /mnt/lustre/home/mturchin20/Lab_Stuff/StephensLab/Multivariate/GIANT2014_5/Vs2/GIANT2014_5.Vs2.HeightBMI.NewSNPs.SNPs.vs1.ChrBPChrBP.txt | wc
    201     603    4679


~~~


















